diff -ur gnotime-2.2.0.orig/configure gnotime-2.2.0.new/configure
--- gnotime-2.2.0.orig/configure	Sun Apr 25 13:51:27 2004
+++ gnotime-2.2.0.new/configure	Fri Apr 30 16:40:16 2004
@@ -12519,23 +12519,23 @@
   else
      PKG_CONFIG_MIN_VERSION=0.9.0
      if $PKG_CONFIG --atleast-pkgconfig-version $PKG_CONFIG_MIN_VERSION; then
-        echo "$as_me:$LINENO: checking for libgtkhtml-3.1 >= $LIBGTKHTML_REQUIRED" >&5
-echo $ECHO_N "checking for libgtkhtml-3.1 >= $LIBGTKHTML_REQUIRED... $ECHO_C" >&6
+        echo "$as_me:$LINENO: checking for libgtkhtml-3.0 >= $LIBGTKHTML_REQUIRED" >&5
+echo $ECHO_N "checking for libgtkhtml-3.0 >= $LIBGTKHTML_REQUIRED... $ECHO_C" >&6
 
-        if $PKG_CONFIG --exists "libgtkhtml-3.1 >= $LIBGTKHTML_REQUIRED" ; then
+        if $PKG_CONFIG --exists "libgtkhtml-3.0 >= $LIBGTKHTML_REQUIRED" ; then
             echo "$as_me:$LINENO: result: yes" >&5
 echo "${ECHO_T}yes" >&6
             succeeded=yes
 
             echo "$as_me:$LINENO: checking LIBGTKHTML_CFLAGS" >&5
 echo $ECHO_N "checking LIBGTKHTML_CFLAGS... $ECHO_C" >&6
-            LIBGTKHTML_CFLAGS=`$PKG_CONFIG --cflags "libgtkhtml-3.1 >= $LIBGTKHTML_REQUIRED"`
+            LIBGTKHTML_CFLAGS=`$PKG_CONFIG --cflags "libgtkhtml-3.0 >= $LIBGTKHTML_REQUIRED"`
             echo "$as_me:$LINENO: result: $LIBGTKHTML_CFLAGS" >&5
 echo "${ECHO_T}$LIBGTKHTML_CFLAGS" >&6
 
             echo "$as_me:$LINENO: checking LIBGTKHTML_LIBS" >&5
 echo $ECHO_N "checking LIBGTKHTML_LIBS... $ECHO_C" >&6
-            LIBGTKHTML_LIBS=`$PKG_CONFIG --libs "libgtkhtml-3.1 >= $LIBGTKHTML_REQUIRED"`
+            LIBGTKHTML_LIBS=`$PKG_CONFIG --libs "libgtkhtml-3.0 >= $LIBGTKHTML_REQUIRED"`
             echo "$as_me:$LINENO: result: $LIBGTKHTML_LIBS" >&5
 echo "${ECHO_T}$LIBGTKHTML_LIBS" >&6
         else
@@ -12543,7 +12543,7 @@
             LIBGTKHTML_LIBS=""
             ## If we have a custom action on failure, don't print errors, but
             ## do set a variable so people can do so.
-            LIBGTKHTML_PKG_ERRORS=`$PKG_CONFIG --errors-to-stdout --print-errors "libgtkhtml-3.1 >= $LIBGTKHTML_REQUIRED"`
+            LIBGTKHTML_PKG_ERRORS=`$PKG_CONFIG --errors-to-stdout --print-errors "libgtkhtml-3.0 >= $LIBGTKHTML_REQUIRED"`
             echo $LIBGTKHTML_PKG_ERRORS
         fi
 
@@ -12558,8 +12558,8 @@
   if test $succeeded = yes; then
      :
   else
-     { { echo "$as_me:$LINENO: error: Library requirements (libgtkhtml-3.1 >= $LIBGTKHTML_REQUIRED) not met; consider adjusting the PKG_CONFIG_PATH environment variable if your libraries are in a nonstandard prefix so pkg-config can find them." >&5
-echo "$as_me: error: Library requirements (libgtkhtml-3.1 >= $LIBGTKHTML_REQUIRED) not met; consider adjusting the PKG_CONFIG_PATH environment variable if your libraries are in a nonstandard prefix so pkg-config can find them." >&2;}
+     { { echo "$as_me:$LINENO: error: Library requirements (libgtkhtml-3.0 >= $LIBGTKHTML_REQUIRED) not met; consider adjusting the PKG_CONFIG_PATH environment variable if your libraries are in a nonstandard prefix so pkg-config can find them." >&5
+echo "$as_me: error: Library requirements (libgtkhtml-3.0 >= $LIBGTKHTML_REQUIRED) not met; consider adjusting the PKG_CONFIG_PATH environment variable if your libraries are in a nonstandard prefix so pkg-config can find them." >&2;}
    { (exit 1); exit 1; }; }
   fi
 
Only in gnotime-2.2.0.new: configure~
diff -ur gnotime-2.2.0.orig/src/main.c gnotime-2.2.0.new/src/main.c
--- gnotime-2.2.0.orig/src/main.c	Mon Apr 26 07:46:20 2004
+++ gnotime-2.2.0.new/src/main.c	Fri Apr 30 18:01:53 2004
@@ -33,7 +33,7 @@
 #include <unistd.h>
 #include <utime.h>
 #if defined(HAVE_DECL_WNOHANG) && defined(HAVE_WAITPID)
-#include <wait.h>
+#include <sys/wait.h>
 #endif
 
 #include <qof.h>
Only in gnotime-2.2.0.new/src: main.c~
