diff -Naur gphoto2-2.1.0.orig/camlibs/sierra/nikon-desc.c gphoto2-2.1.0/camlibs/sierra/nikon-desc.c
--- gphoto2-2.1.0.orig/camlibs/sierra/nikon-desc.c	Tue Jun 25 21:52:12 2002
+++ gphoto2-2.1.0/camlibs/sierra/nikon-desc.c	Fri Nov 22 23:43:31 2002
@@ -19,7 +19,7 @@
  */
 #include <stdio.h>
 #include <sys/types.h>
-#include <malloc.h>
+//#include <malloc.h>
 #include <string.h>
 #include <gphoto2-library.h>
 #include "sierra.h"
diff -Naur gphoto2-2.1.0.orig/configure gphoto2-2.1.0/configure
--- gphoto2-2.1.0.orig/configure	Tue Jun 25 21:59:38 2002
+++ gphoto2-2.1.0/configure	Fri Nov 22 23:37:42 2002
@@ -9155,7 +9155,7 @@
     #        cross-compilation, but unfortunately the echo tests do not
     #        yet detect zsh echo's removal of \ escapes.  Also zsh mangles
     #	     `"' quotes if we put them in here... so don't!
-    archive_cmds='$nonopt $(test .$module = .yes && echo -bundle || echo -dynamiclib) $allow_undefined_flag -o $lib $libobjs $deplibs$linker_flags -install_name $rpath/$soname $verstring'
+    archive_cmds='$nonopt $(test .$module = .yes && echo -bundle || echo -dynamiclib) $allow_undefined_flag -o $lib $libobjs $deplibs$linkopts $(test .$module != .yes && echo -install_name $rpath/$soname $tmp_verstring)'
     # We need to add '_' to the symbols in $export_symbols first
     #archive_expsym_cmds="$archive_cmds"' && strip -s $export_symbols'
     hardcode_direct=yes
diff -Naur gphoto2-2.1.0.orig/frontends/command-line/actions.c gphoto2-2.1.0/frontends/command-line/actions.c
--- gphoto2-2.1.0.orig/frontends/command-line/actions.c	Tue Jun 25 21:52:15 2002
+++ gphoto2-2.1.0/frontends/command-line/actions.c	Tue Nov 26 10:10:54 2002
@@ -337,16 +337,16 @@
         printf ("%-59.59s", _("Value"));
         printf ("\n");
         print_hline ();
-        if (ed->ifd0)
-                show_ifd (ed->ifd0);
-        if (ed->ifd1)
-                show_ifd (ed->ifd1);
-        if (ed->ifd_exif)
-                show_ifd (ed->ifd_exif);
-        if (ed->ifd_gps)
-                show_ifd (ed->ifd_gps);
-        if (ed->ifd_interoperability)
-                show_ifd (ed->ifd_interoperability);
+        if (ed->ifd[EXIF_IFD_0])
+                show_ifd (ed->ifd[EXIF_IFD_0]);
+        if (ed->ifd[EXIF_IFD_1])
+                show_ifd (ed->ifd[EXIF_IFD_1]);
+        if (ed->ifd[EXIF_IFD_EXIF])
+                show_ifd (ed->ifd[EXIF_IFD_EXIF]);
+        if (ed->ifd[EXIF_IFD_GPS])
+                show_ifd (ed->ifd[EXIF_IFD_GPS]);
+        if (ed->ifd[EXIF_IFD_INTEROPERABILITY])
+                show_ifd (ed->ifd[EXIF_IFD_INTEROPERABILITY]);
         print_hline ();
         if (ed->size) {
                 printf (_("EXIF data contains a thumbnail (%i bytes)."),
diff -Naur gphoto2-2.1.0.orig/libgphoto2/gphoto2-filesys.c gphoto2-2.1.0/libgphoto2/gphoto2-filesys.c
--- gphoto2-2.1.0.orig/libgphoto2/gphoto2-filesys.c	Tue Jun 25 21:52:15 2002
+++ gphoto2-2.1.0/libgphoto2/gphoto2-filesys.c	Tue Nov 26 10:07:17 2002
@@ -98,12 +98,12 @@
         /*
          * HP PhotoSmart C30 has the date and time in ifd_exif.
          */
-	e = exif_content_get_entry (ed->ifd0, EXIF_TAG_DATE_TIME);
+	e = exif_content_get_entry (ed->ifd[EXIF_IFD_0], EXIF_TAG_DATE_TIME);
 	if (!e)
-	        e = exif_content_get_entry (ed->ifd_exif,
+	        e = exif_content_get_entry (ed->ifd[EXIF_IFD_EXIF],
 					    EXIF_TAG_DATE_TIME_ORIGINAL);
 	if (!e)
-		e = exif_content_get_entry (ed->ifd_exif, 
+		e = exif_content_get_entry (ed->ifd[EXIF_IFD_EXIF], 
 					    EXIF_TAG_DATE_TIME_DIGITIZED);
         if (!e) {
                 GP_DEBUG ("EXIF data has not date/time tag.");
diff -Naur gphoto2-2.1.0.orig/libgphoto2_port/configure gphoto2-2.1.0/libgphoto2_port/configure
--- gphoto2-2.1.0.orig/libgphoto2_port/configure	Tue Jun 25 21:57:33 2002
+++ gphoto2-2.1.0/libgphoto2_port/configure	Fri Nov 22 23:37:53 2002
@@ -7994,7 +7994,7 @@
     #        cross-compilation, but unfortunately the echo tests do not
     #        yet detect zsh echo's removal of \ escapes.  Also zsh mangles
     #	     `"' quotes if we put them in here... so don't!
-    archive_cmds='$nonopt $(test .$module = .yes && echo -bundle || echo -dynamiclib) $allow_undefined_flag -o $lib $libobjs $deplibs$linker_flags -install_name $rpath/$soname $verstring'
+    archive_cmds='$nonopt $(test .$module = .yes && echo -bundle || echo -dynamiclib) $allow_undefined_flag -o $lib $libobjs $deplibs$linkopts $(test .$module != .yes && echo -install_name $rpath/$soname $tmp_verstring)'
     # We need to add '_' to the symbols in $export_symbols first
     #archive_expsym_cmds="$archive_cmds"' && strip -s $export_symbols'
     hardcode_direct=yes
diff -Naur gphoto2-2.1.0.orig/libgphoto2_port/ltmain.sh gphoto2-2.1.0/libgphoto2_port/ltmain.sh
--- gphoto2-2.1.0.orig/libgphoto2_port/ltmain.sh	Tue Jun 25 21:56:37 2002
+++ gphoto2-2.1.0/libgphoto2_port/ltmain.sh	Tue Nov 26 11:18:00 2002
@@ -1758,7 +1758,7 @@
 
 	  if test "$installed" = no; then
 	    notinst_deplibs="$notinst_deplibs $lib"
-	    need_relink=yes
+	    need_relink=no
 	  fi
 
 	  if test -n "$old_archive_from_expsyms_cmds"; then
@@ -2940,6 +2940,11 @@
 	if test -n "$export_symbols" && test -n "$archive_expsym_cmds"; then
 	  eval cmds=\"$archive_expsym_cmds\"
 	else
+     if test "x$verstring" = "x0.0"; then
+      tmp_verstring=
+     else
+      tmp_verstring="$verstring"
+     fi
 	  save_deplibs="$deplibs"
 	  for conv in $convenience; do
 	    tmp_deplibs=
diff -Naur gphoto2-2.1.0.orig/ltmain.sh gphoto2-2.1.0/ltmain.sh
--- gphoto2-2.1.0.orig/ltmain.sh	Tue Jun 25 21:57:40 2002
+++ gphoto2-2.1.0/ltmain.sh	Tue Nov 26 11:16:50 2002
@@ -1758,7 +1758,7 @@
 
 	  if test "$installed" = no; then
 	    notinst_deplibs="$notinst_deplibs $lib"
-	    need_relink=yes
+	    need_relink=no
 	  fi
 
 	  if test -n "$old_archive_from_expsyms_cmds"; then
@@ -2940,6 +2940,11 @@
 	if test -n "$export_symbols" && test -n "$archive_expsym_cmds"; then
 	  eval cmds=\"$archive_expsym_cmds\"
 	else
+	 if test "x$verstring" = "x0.0"; then
+	   tmp_verstring=
+	 else
+	   tmp_verstring="$verstring"
+	 fi	
 	  save_deplibs="$deplibs"
 	  for conv in $convenience; do
 	    tmp_deplibs=
