diff -Naur gphoto2-2.1.0.orig/frontends/command-line/actions.c gphoto2-2.1.0/frontends/command-line/actions.c
--- gphoto2-2.1.0.orig/frontends/command-line/actions.c	Tue Jun 25 21:52:15 2002
+++ gphoto2-2.1.0/frontends/command-line/actions.c	Tue Nov 26 10:10:54 2002
@@ -337,16 +337,16 @@
         printf ("%-59.59s", _("Value"));
         printf ("\n");
         print_hline ();
-        if (ed->ifd0)
-                show_ifd (ed->ifd0);
-        if (ed->ifd1)
-                show_ifd (ed->ifd1);
-        if (ed->ifd_exif)
-                show_ifd (ed->ifd_exif);
-        if (ed->ifd_gps)
-                show_ifd (ed->ifd_gps);
-        if (ed->ifd_interoperability)
-                show_ifd (ed->ifd_interoperability);
+        if (ed->ifd[EXIF_IFD_0])
+                show_ifd (ed->ifd[EXIF_IFD_0]);
+        if (ed->ifd[EXIF_IFD_1])
+                show_ifd (ed->ifd[EXIF_IFD_1]);
+        if (ed->ifd[EXIF_IFD_EXIF])
+                show_ifd (ed->ifd[EXIF_IFD_EXIF]);
+        if (ed->ifd[EXIF_IFD_GPS])
+                show_ifd (ed->ifd[EXIF_IFD_GPS]);
+        if (ed->ifd[EXIF_IFD_INTEROPERABILITY])
+                show_ifd (ed->ifd[EXIF_IFD_INTEROPERABILITY]);
         print_hline ();
         if (ed->size) {
                 printf (_("EXIF data contains a thumbnail (%i bytes)."),
