diff -Naur gphoto2-2.1.0.orig/libgphoto2_port/ltmain.sh gphoto2-2.1.0/libgphoto2_port/ltmain.sh
--- gphoto2-2.1.0.orig/libgphoto2_port/ltmain.sh	Tue Jun 25 21:56:37 2002
+++ gphoto2-2.1.0/libgphoto2_port/ltmain.sh	Tue Nov 26 11:18:00 2002
@@ -1758,7 +1758,7 @@
 
 	  if test "$installed" = no; then
 	    notinst_deplibs="$notinst_deplibs $lib"
-	    need_relink=yes
+	    need_relink=no
 	  fi
 
 	  if test -n "$old_archive_from_expsyms_cmds"; then
@@ -2940,6 +2940,11 @@
 	if test -n "$export_symbols" && test -n "$archive_expsym_cmds"; then
 	  eval cmds=\"$archive_expsym_cmds\"
 	else
+     if test "x$verstring" = "x0.0"; then
+      tmp_verstring=
+     else
+      tmp_verstring="$verstring"
+     fi
 	  save_deplibs="$deplibs"
 	  for conv in $convenience; do
 	    tmp_deplibs=
diff -Naur gphoto2-2.1.0.orig/ltmain.sh gphoto2-2.1.0/ltmain.sh
--- gphoto2-2.1.0.orig/ltmain.sh	Tue Jun 25 21:57:40 2002
+++ gphoto2-2.1.0/ltmain.sh	Tue Nov 26 11:16:50 2002
@@ -1758,7 +1758,7 @@
 
 	  if test "$installed" = no; then
 	    notinst_deplibs="$notinst_deplibs $lib"
-	    need_relink=yes
+	    need_relink=no
 	  fi
 
 	  if test -n "$old_archive_from_expsyms_cmds"; then
@@ -2940,6 +2940,11 @@
 	if test -n "$export_symbols" && test -n "$archive_expsym_cmds"; then
 	  eval cmds=\"$archive_expsym_cmds\"
 	else
+	 if test "x$verstring" = "x0.0"; then
+	   tmp_verstring=
+	 else
+	   tmp_verstring="$verstring"
+	 fi	
 	  save_deplibs="$deplibs"
 	  for conv in $convenience; do
 	    tmp_deplibs=
--- libgphoto2-2.1.1/configure.orig	2002-12-06 22:37:49.000000000 -0800
+++ libgphoto2-2.1.1/configure	2002-12-06 22:37:52.000000000 -0800
@@ -10091,7 +10091,7 @@
     #        cross-compilation, but unfortunately the echo tests do not
     #        yet detect zsh echo's removal of \ escapes.  Also zsh mangles
     #	     `"' quotes if we put them in here... so don't!
-    archive_cmds='$CC -r -keep_private_externs -nostdlib -o ${lib}-master.o $libobjs && $CC $(test .$module = .yes && echo -bundle || echo -dynamiclib) $allow_undefined_flag -o $lib ${lib}-master.o $deplibs$linker_flags $(test .$module != .yes && echo -install_name $rpath/$soname $verstring)'
+    archive_cmds='$CC -r -keep_private_externs -nostdlib -o ${lib}-master.o $libobjs && $CC $(test .$module = .yes && echo -bundle || echo -dynamiclib) $allow_undefined_flag -o $lib ${lib}-master.o $deplibs$linkopts $(test .$module != .yes && echo -install_name $rpath/$soname $verstring)'
     # We need to add '_' to the symbols in $export_symbols first
     #archive_expsym_cmds="$archive_cmds"' && strip -s $export_symbols'
     hardcode_direct=yes
--- libgphoto2-2.1.1/libgphoto2_port/configure.orig	2002-12-06 23:06:06.000000000 -0800
+++ libgphoto2-2.1.1/libgphoto2_port/configure	2002-12-06 23:06:29.000000000 -0800
@@ -9204,7 +9204,7 @@
     #        cross-compilation, but unfortunately the echo tests do not
     #        yet detect zsh echo's removal of \ escapes.  Also zsh mangles
     #	     `"' quotes if we put them in here... so don't!
-    archive_cmds='$CC -r -keep_private_externs -nostdlib -o ${lib}-master.o $libobjs && $CC $(test .$module = .yes && echo -bundle || echo -dynamiclib) $allow_undefined_flag -o $lib ${lib}-master.o $deplibs$linker_flags $(test .$module != .yes && echo -install_name $rpath/$soname $verstring)'
+    archive_cmds='$CC -r -keep_private_externs -nostdlib -o ${lib}-master.o $libobjs && $CC $(test .$module = .yes && echo -bundle || echo -dynamiclib) $allow_undefined_flag -o $lib ${lib}-master.o $deplibs$linkopts $(test .$module != .yes && echo -install_name $rpath/$soname $verstring)'
     # We need to add '_' to the symbols in $export_symbols first
     #archive_expsym_cmds="$archive_cmds"' && strip -s $export_symbols'
     hardcode_direct=yes
