--- libusb-0.1.7/darwin.c.orig	Wed Nov 20 23:29:49 2002
+++ libusb-0.1.7/darwin.c	Wed Nov 20 23:31:03 2002
@@ -393,8 +393,12 @@
   if (!device->interface)
     USB_ERROR_STR(-EACCES, "interface used without being claimed");
 
+/*
   if ((pipeRef = ep_to_pipeRef(device, ep)) == -1)
     USB_ERROR(-EINVAL);
+  Don't ask me why, but this always fails for me.... but the following works
+*/
+  pipeRef = ep;
 
   if (usb_debug > 3)
     fprintf(stderr, "usb_bulk_write: endpoint=0x%02x size=%i TO=%i\n", ep, size, timeout);
@@ -437,10 +441,12 @@
   /* interface is not open */
   if (!device->interface)
     USB_ERROR_STR(-EACCES, "interface used without being claimed");
-
+/*
   if ((pipeRef = ep_to_pipeRef(device, ep)) == -1)
     USB_ERROR(-EINVAL);
-
+  Don't ask me why, but this always fails for me.... but the following works
+*/
+  pipeRef = ep;
   if (usb_debug > 3)
     fprintf(stderr, "usb_bulk_read: endpoint=0x%02x size=%i TO=%i\n", ep, size, timeout);
 
