diff -uNr gst-plugins-0.8.8/ext/cdparanoia/gstcdparanoia.c gst-plugins-0.8.8-new/ext/cdparanoia/gstcdparanoia.c
--- gst-plugins-0.8.8/ext/cdparanoia/gstcdparanoia.c	Sat Mar  5 11:33:30 2005
+++ gst-plugins-0.8.8-new/ext/cdparanoia/gstcdparanoia.c	Wed Mar 16 10:23:50 2005
@@ -43,8 +43,9 @@
 #define CD_SECS              60 /* seconds per minute */
 #define CD_FRAMES            75 /* frames per second */
 
-#include "gstcdparanoia.h"
+#undef VERSION
 
+#include "gstcdparanoia.h"
 
 static GstElementDetails cdparanoia_details = {
   "CD Audio (cdda) Source, Paranoia IV",
@@ -1227,4 +1228,4 @@
     GST_VERSION_MINOR,
     "cdparanoia",
     "Read audio from CD in paranoid mode",
-    plugin_init, VERSION, "GPL", GST_PACKAGE, GST_ORIGIN)
+    plugin_init, GST_PLUGINS_VERSION, "GPL", GST_PACKAGE, GST_ORIGIN)
diff -uNr gst-plugins-0.8.8/gst/audioconvert/Makefile.in gst-plugins-0.8.8-new/gst/audioconvert/Makefile.in
--- gst-plugins-0.8.8/gst/audioconvert/Makefile.in	Mon Mar  7 11:43:21 2005
+++ gst-plugins-0.8.8-new/gst/audioconvert/Makefile.in	Tue Mar 15 22:44:49 2005
@@ -41,7 +41,7 @@
 build_triplet = @build@
 host_triplet = @host@
 target_triplet = @target@
-noinst_PROGRAMS = channelmixtest$(EXEEXT)
+noinst_PROGRAMS = 
 subdir = gst/audioconvert
 DIST_COMMON = $(noinst_HEADERS) $(srcdir)/Makefile.am \
 	$(srcdir)/Makefile.in
