diff -uNbr PerlQt-3/smoke/qt/qtguess.pl.in PerlQt-3-new/smoke/qt/qtguess.pl.in
--- PerlQt-3/smoke/qt/qtguess.pl.in	Thu Dec 12 11:25:16 2002
+++ PerlQt-3-new/smoke/qt/qtguess.pl.in	Fri Dec 13 19:19:05 2002
@@ -51,7 +51,7 @@
 
  -e "$qtinc/qglobal.h" or die "Invalid Qt directory.\n";
 
-my $ccmd = "$cc $ccflags $allinc $alllib -o $tmp $tmp.cpp $qtflags 2>/dev/null";
+my $ccmd = "$cc $ccflags $allinc $alllib -o $tmp $tmp.cpp $qtflags >/dev/null 2>\&1";
 
 my $treshold = defined($opt_t)?$opt_t : $default_treshold;
 $treshold >= 0 or die "invalid testing treshold: $treshold\n";
