diff -uNbr XML-RSS-1.00/lib/RSS.pm XML-RSS-1.00-new/lib/RSS.pm
--- XML-RSS-1.00/lib/RSS.pm	Fri Jan 31 12:19:22 2003
+++ XML-RSS-1.00-new/lib/RSS.pm	Sun Feb  2 22:38:17 2003
@@ -565,8 +565,8 @@
     $output .= qq|<?xml version="1.0"$encoding?>\n\n|;
 
     # RDF root element
-    $output .= '<rdf:RDF'."\n".'xmlns:rdf="http://www.w3.org/1999/02/22-rdf-syntax-ns#"'."\n";
-    $output .= 'xmlns="http://my.netscape.com/rdf/simple/0.9/">'."\n\n";
+    $output .= '<rdf:RDF'."\n".'xmlns:rdf="http://www.w3.org/1999/02/22-rdf-syntax-ns#"';
+    $output .= 'xmlns="http://my.netscape.com/rdf/simple/0.9/">'."\n";
 
     ###################
     # Channel Element #
@@ -797,8 +797,8 @@
 
     # RDF namespaces declaration
     $output .="<rdf:RDF"."\n";
-    $output .=' xmlns:rdf="http://www.w3.org/1999/02/22-rdf-syntax-ns#"'."\n";
-    $output .=' xmlns="http://purl.org/rss/1.0/"'."\n";
+    $output .=' xmlns:rdf="http://www.w3.org/1999/02/22-rdf-syntax-ns#"';
+    $output .=' xmlns="http://purl.org/rss/1.0/"';
 
     # print all imported namespaces
     while (my($k, $v) = each %{$self->{modules}}) {
