diff -uNr openssl-0.9.7g/Makefile openssl-0.9.7g-new/Makefile
--- openssl-0.9.7g/Makefile	2005-04-11 11:17:45.000000000 -0400
+++ openssl-0.9.7g-new/Makefile	2005-04-26 09:55:26.000000000 -0400
@@ -325,7 +325,7 @@
 	fi; \
 	( set -x; ${CC} --verbose -dynamiclib -o lib$$i${SHLIB_EXT} \
 		lib$$i.a $$libs -all_load -current_version ${SHLIB_MAJOR}.${SHLIB_MINOR} \
-		-compatibility_version ${SHLIB_MAJOR}.`echo ${SHLIB_MINOR} | cut -d. -f1` \
+		-compatibility_version ${SHLIB_MAJOR}.${SHLIB_MINOR}` \
 		-install_name ${INSTALLTOP}/lib/lib$$i${SHLIB_EXT} ) || exit 1; \
 	libs="-l`basename $$i${SHLIB_EXT} .dylib` $$libs"; \
 	echo "" ; \
diff -uNr openssl-0.9.7g/Makefile.bak openssl-0.9.7g-new/Makefile.bak
--- openssl-0.9.7g/Makefile.bak	2005-04-11 11:17:39.000000000 -0400
+++ openssl-0.9.7g-new/Makefile.bak	2005-04-26 09:56:59.000000000 -0400
@@ -325,7 +325,7 @@
 	fi; \
 	( set -x; ${CC} --verbose -dynamiclib -o lib$$i${SHLIB_EXT} \
 		lib$$i.a $$libs -all_load -current_version ${SHLIB_MAJOR}.${SHLIB_MINOR} \
-		-compatibility_version ${SHLIB_MAJOR}.`echo ${SHLIB_MINOR} | cut -d. -f1` \
+		-compatibility_version ${SHLIB_MAJOR}.${SHLIB_MINOR} \
 		-install_name ${INSTALLTOP}/lib/lib$$i${SHLIB_EXT} ) || exit 1; \
 	libs="-l`basename $$i${SHLIB_EXT} .dylib` $$libs"; \
 	echo "" ; \
@@ -888,10 +888,7 @@
 		$(INSTALL_PREFIX)$(MANDIR)/man7
 	@pod2man="`cd util; ./pod2mantest $(PERL)`"; \
 	here="`pwd`"; \
-	filecase=; \
-	if [ "$(PLATFORM)" = "DJGPP" -o "$(PLATFORM)" = "Cygwin" -o "$(PLATFORM)" = "mingw" ]; then \
-		filecase=-i; \
-	fi; \
+	filecase=-i; \
 	for i in doc/apps/*.pod; do \
 		fn=`basename $$i .pod`; \
 		if [ "$$fn" = "config" ]; then sec=5; else sec=1; fi; \
diff -uNr openssl-0.9.7g/Makefile.org openssl-0.9.7g-new/Makefile.org
--- openssl-0.9.7g/Makefile.org	2005-03-15 04:46:13.000000000 -0500
+++ openssl-0.9.7g-new/Makefile.org	2005-04-26 09:56:14.000000000 -0400
@@ -323,7 +323,7 @@
 	fi; \
 	( set -x; ${CC} --verbose -dynamiclib -o lib$$i${SHLIB_EXT} \
 		lib$$i.a $$libs -all_load -current_version ${SHLIB_MAJOR}.${SHLIB_MINOR} \
-		-compatibility_version ${SHLIB_MAJOR}.`echo ${SHLIB_MINOR} | cut -d. -f1` \
+		-compatibility_version ${SHLIB_MAJOR}.${SHLIB_MINOR} \
 		-install_name ${INSTALLTOP}/lib/lib$$i${SHLIB_EXT} ) || exit 1; \
 	libs="-l`basename $$i${SHLIB_EXT} .dylib` $$libs"; \
 	echo "" ; \
@@ -886,10 +886,7 @@
 		$(INSTALL_PREFIX)$(MANDIR)/man7
 	@pod2man="`cd util; ./pod2mantest $(PERL)`"; \
 	here="`pwd`"; \
-	filecase=; \
-	if [ "$(PLATFORM)" = "DJGPP" -o "$(PLATFORM)" = "Cygwin" -o "$(PLATFORM)" = "mingw" ]; then \
-		filecase=-i; \
-	fi; \
+	filecase=-i; \
 	for i in doc/apps/*.pod; do \
 		fn=`basename $$i .pod`; \
 		if [ "$$fn" = "config" ]; then sec=5; else sec=1; fi; \
diff -uNr openssl-0.9.7g/apps/Makefile openssl-0.9.7g-new/apps/Makefile
--- openssl-0.9.7g/apps/Makefile	2005-03-12 07:15:19.000000000 -0500
+++ openssl-0.9.7g-new/apps/Makefile	2005-04-26 09:57:30.000000000 -0400
@@ -31,8 +31,8 @@
 
 DLIBCRYPTO=../libcrypto.a
 DLIBSSL=../libssl.a
-LIBCRYPTO=-L.. -lcrypto
-LIBSSL=-L.. -lssl
+LIBCRYPTO=$(DLIBCRYPTO)
+LIBSSL=$(DLIBSSL)
 
 PROGRAM= openssl
 
