diff -uNr postgresql-8.0.1/contrib/Makefile postgresql-8.0.1-new/contrib/Makefile
--- postgresql-8.0.1/contrib/Makefile	Tue Dec 14 17:50:23 2004
+++ postgresql-8.0.1-new/contrib/Makefile	Sat Apr  9 17:35:12 2005
@@ -37,7 +37,6 @@
 		string		\
 		tablefunc	\
 		tips		\
-		tsearch		\
 		tsearch2	\
 		userlock	\
 		vacuumlo
diff -uNr postgresql-8.0.1/pgsql.sh postgresql-8.0.1-new/pgsql.sh
--- postgresql-8.0.1/pgsql.sh	Wed Dec 31 19:00:00 1969
+++ postgresql-8.0.1-new/pgsql.sh	Sat Apr  9 17:35:12 2005
@@ -0,0 +1,21 @@
+#!/bin/sh
+
+for flag in "-n" "-d" "-s" "-u" "-v"; do
+	ulimit $flag 65536 >/dev/null 2>&1 || :
+done
+
+PREFIX="@INSTPREFIX@"
+DATADIR="${PREFIX}/var/postgresql-8.0/data"
+LOGFILE="${PREFIX}/var/postgresql-8.0/pgsql.log"
+
+case "$1" in
+	start)
+		unset LC_ALL; LANG=C sudo -u postgres "${PREFIX}/bin/pg_ctl-8.0" start -D "$DATADIR" -l "$LOGFILE"
+		;;
+	restart)
+		unset LC_ALL; LANG=C sudo -u postgres "${PREFIX}/bin/pg_ctl-8.0" restart -D "$DATADIR" -m fast
+		;;
+	stop)
+		unset LC_ALL; LANG=C sudo -u postgres "${PREFIX}/bin/pg_ctl-8.0" stop -D "$DATADIR" -m fast
+		;;
+esac
diff -uNr postgresql-8.0.1/src/Makefile.global.in postgresql-8.0.1-new/src/Makefile.global.in
--- postgresql-8.0.1/src/Makefile.global.in	Wed Jan 26 14:24:15 2005
+++ postgresql-8.0.1-new/src/Makefile.global.in	Sat Apr  9 17:35:12 2005
@@ -65,21 +65,21 @@
 libexecdir := @libexecdir@
 ifeq "$(findstring pgsql, $(libexecdir))" ""
 ifeq "$(findstring postgres, $(libexecdir))" ""
-override libexecdir := $(libexecdir)/postgresql
+override libexecdir := $(libexecdir)/postgresql-8.0
 endif
 endif
 
 datadir := @datadir@
 ifeq "$(findstring pgsql, $(datadir))" ""
 ifeq "$(findstring postgres, $(datadir))" ""
-override datadir := $(datadir)/postgresql
+override datadir := $(datadir)/postgresql-8.0
 endif
 endif
 
 sysconfdir := @sysconfdir@
 ifeq "$(findstring pgsql, $(sysconfdir))" ""
 ifeq "$(findstring postgres, $(sysconfdir))" ""
-override sysconfdir := $(sysconfdir)/postgresql
+override sysconfdir := $(sysconfdir)/postgresql-8.0
 endif
 endif
 
@@ -87,7 +87,7 @@
 pkglibdir = $(libdir)
 ifeq "$(findstring pgsql, $(pkglibdir))" ""
 ifeq "$(findstring postgres, $(pkglibdir))" ""
-override pkglibdir := $(pkglibdir)/postgresql
+override pkglibdir := $(pkglibdir)/postgresql-8.0
 endif
 endif
 
@@ -95,7 +95,7 @@
 pkgincludedir = $(includedir)
 ifeq "$(findstring pgsql, $(pkgincludedir))" ""
 ifeq "$(findstring postgres, $(pkgincludedir))" ""
-override pkgincludedir := $(pkgincludedir)/postgresql
+override pkgincludedir := $(pkgincludedir)/postgresql-8.0
 endif
 endif
 includedir_server = $(pkgincludedir)/server
@@ -109,7 +109,7 @@
 ifneq (,$(docdir))
 ifeq "$(findstring pgsql, $(docdir))" ""
 ifeq "$(findstring postgres, $(docdir))" ""
-override docdir := $(docdir)/postgresql
+override docdir := $(docdir)/postgresql-8.0
 endif
 endif
 endif
diff -uNr postgresql-8.0.1/src/Makefile.shlib postgresql-8.0.1-new/src/Makefile.shlib
--- postgresql-8.0.1/src/Makefile.shlib	Sat Nov 20 16:13:04 2004
+++ postgresql-8.0.1-new/src/Makefile.shlib	Sat Apr  9 17:35:12 2005
@@ -101,11 +101,11 @@
   ifeq ($(DLTYPE), library)
     # linkable library
     DLSUFFIX		:= .dylib
-    LINK.shared		= $(COMPILER) -dynamiclib -install_name $(libdir)/lib$(NAME).$(SO_MAJOR_VERSION)$(DLSUFFIX) $(version_link) -multiply_defined suppress
+    LINK.shared		= $(CXX) $(CFLAGS) -dynamiclib -install_name $(libdir)/lib$(NAME).$(SO_MAJOR_VERSION)$(DLSUFFIX) $(version_link) -multiply_defined suppress
   else
     # loadable module (default case)
     DLSUFFIX		:= .so
-    LINK.shared		= $(COMPILER) -bundle
+    LINK.shared		= $(CXX) $(CFLAGS) -bundle
   endif
   shlib			= lib$(NAME).$(SO_MAJOR_VERSION).$(SO_MINOR_VERSION)$(DLSUFFIX)
   shlib_major		= lib$(NAME).$(SO_MAJOR_VERSION)$(DLSUFFIX)
diff -uNr postgresql-8.0.1/src/backend/main/main.c postgresql-8.0.1-new/src/backend/main/main.c
--- postgresql-8.0.1/src/backend/main/main.c	Fri Dec 31 16:59:53 2004
+++ postgresql-8.0.1-new/src/backend/main/main.c	Sat Apr  9 17:35:12 2005
@@ -56,6 +56,7 @@
 	struct passwd *pw;
 #endif
 	char	   *pw_name_persist;
+	char	   *progname;
 
 	/*
 	 * Place platform-specific startup hacks here.	This is the right
@@ -258,7 +259,26 @@
 	 */
 	len = strlen(argv[0]);
 
-	if ((len >= 10 && strcmp(argv[0] + len - 10, "postmaster") == 0)
+	/*
+	 * Find the program name, which is after the final slash in the
+	 * program argument.
+	 */
+	progname = strrchr(argv[0], '/');
+	if (progname != NULL)
+	{
+		progname = progname + 1;
+	  }
+	else
+	{
+		progname = argv[0];
+	}
+
+	/*
+	 * Now check if the program name starts with the
+	 * word "postmaster", in which case PostmasterMain is called.
+	 */
+	len = strlen(progname);
+	if ((len >= 10 && strncmp(progname, "postmaster", 10) == 0)
 #ifdef WIN32
 	  || (len >= 14 && strcmp(argv[0] + len - 14, "postmaster.exe") == 0)
 #endif
diff -uNr postgresql-8.0.1/src/backend/utils/fmgr/dfmgr.c postgresql-8.0.1-new/src/backend/utils/fmgr/dfmgr.c
--- postgresql-8.0.1/src/backend/utils/fmgr/dfmgr.c	Fri Dec 31 17:01:31 2004
+++ postgresql-8.0.1-new/src/backend/utils/fmgr/dfmgr.c	Sat Apr  9 17:35:12 2005
@@ -305,6 +305,26 @@
 		pfree(full);
 	}
 
+	new = palloc(strlen(name) + strlen(".so") + 1);
+	strcpy(new, name);
+	strcat(new, ".so");
+
+	if (!have_slash)
+	{
+		full = find_in_dynamic_libpath(new);
+		pfree(new);
+		if (full)
+			return full;
+	}
+	else
+	{
+		full = substitute_libpath_macro(new);
+		pfree(new);
+		if (file_exists(full))
+			return full;
+		pfree(full);
+	}
+
 	new = palloc(strlen(name) + strlen(DLSUFFIX) + 1);
 	strcpy(new, name);
 	strcat(new, DLSUFFIX);
diff -uNr postgresql-8.0.1/src/backend/utils/misc/postgresql.conf.sample postgresql-8.0.1-new/src/backend/utils/misc/postgresql.conf.sample
--- postgresql-8.0.1/src/backend/utils/misc/postgresql.conf.sample	Fri Nov  5 14:16:16 2004
+++ postgresql-8.0.1-new/src/backend/utils/misc/postgresql.conf.sample	Sat Apr  9 17:35:12 2005
@@ -61,7 +61,7 @@
 # - Security & Authentication -
 
 #authentication_timeout = 60	# 1-600, in seconds
-#ssl = false
+ssl = false
 #password_encryption = true
 #krb_server_keyfile = ''
 #db_user_namespace = false
diff -uNr postgresql-8.0.1/src/interfaces/ecpg/compatlib/Makefile postgresql-8.0.1-new/src/interfaces/ecpg/compatlib/Makefile
--- postgresql-8.0.1/src/interfaces/ecpg/compatlib/Makefile	Mon Jul 12 20:06:39 2004
+++ postgresql-8.0.1-new/src/interfaces/ecpg/compatlib/Makefile	Sun Apr 10 00:20:15 2005
@@ -20,6 +20,7 @@
 override CPPFLAGS := -I$(top_srcdir)/src/interfaces/ecpg/include -I$(libpq_srcdir) \
 	-I$(top_srcdir)/src/include/utils $(CPPFLAGS)
 override CFLAGS += $(PTHREAD_CFLAGS)
+override LDFLAGS := -L../ecpglib -L../pgtypeslib -L../../libpq $(LDFLAGS)
 SHLIB_LINK = -L../ecpglib -lecpg -L../pgtypeslib -lpgtypes $(libpq) \
 	$(filter -lintl -lssl -lcrypto -lkrb5 -lcrypt -lm, $(LIBS)) $(PTHREAD_LIBS)
 
diff -uNr postgresql-8.0.1/src/interfaces/ecpg/ecpglib/Makefile postgresql-8.0.1-new/src/interfaces/ecpg/ecpglib/Makefile
--- postgresql-8.0.1/src/interfaces/ecpg/ecpglib/Makefile	Wed Jan 26 14:24:17 2005
+++ postgresql-8.0.1-new/src/interfaces/ecpg/ecpglib/Makefile	Sun Apr 10 00:19:39 2005
@@ -19,6 +19,7 @@
 
 override CPPFLAGS := -DFRONTEND -I$(top_srcdir)/src/interfaces/ecpg/include \
 	-I$(libpq_srcdir) -I$(top_builddir)/src/port $(CPPFLAGS)
+override LDFLAGS := -L../ecpglib -L../pgtypeslib -L../../libpq $(LDFLAGS)
 override CFLAGS += $(PTHREAD_CFLAGS)
 
 # Need to recomple any libpgport object files
diff -uNr postgresql-8.0.1/src/makefiles/Makefile.darwin postgresql-8.0.1-new/src/makefiles/Makefile.darwin
--- postgresql-8.0.1/src/makefiles/Makefile.darwin	Thu Dec 16 22:52:48 2004
+++ postgresql-8.0.1-new/src/makefiles/Makefile.darwin	Sat Apr  9 17:35:12 2005
@@ -13,6 +13,6 @@
 # Rule for building shared libs (currently used only for regression test
 # shlib ... should go away, since this is not really enough knowledge)
 %.so: %.o
-	$(CC) -bundle -o $@ $< $(BE_DLLLIBS)
+	$(CXX) -bundle -undefined dynamic_lookup -o $@ $< $(BE_DLLLIBS)
 
 sqlmansect = 7
