diff -ru Guppi-0.40.3/configure Guppi-0.40.3-patched/configure
--- Guppi-0.40.3/configure	2002-01-21 23:15:34.000000000 -0500
+++ Guppi-0.40.3-patched/configure	2002-05-15 18:12:26.000000000 -0400
@@ -4784,7 +4784,7 @@
     ;;
 
   darwin* | rhapsody*)
-    allow_undefined_flag='-undefined suppress'
+    allow_undefined_flag='-flat_namespace -undefined suppress'
     # FIXME: Relying on posixy $() will cause problems for
     #        cross-compilation, but unfortunately the echo tests do not
     #        yet detect zsh echo's removal of \ escapes.
