diff -Naur libxpg4.old/Makefile libxpg4.new/Makefile
--- libxpg4.old/Makefile	Tue Jun  5 18:33:44 2001
+++ libxpg4.new/Makefile	Mon May 13 23:47:11 2002
@@ -21,7 +21,7 @@
 	ranlib /usr/local/lib/${TARGET}.a
 
 ${TARGET}.dylib: ${OBJECTS}
-	${CC} -dynamiclib -o $@ ${OBJECTS}
+	${CC} -dynamiclib -flat_namespace -o $@ ${OBJECTS}
 
 ${TARGET}.a: ${OBJECTS}
 	ar cru $@ ${OBJECTS}
diff -Naur libxpg4.old/fink/Makefile libxpg4.new/fink/Makefile
--- libxpg4.old/fink/Makefile	Thu Jan  1 09:00:00 1970
+++ libxpg4.new/fink/Makefile	Fri Jun 28 23:42:52 2002
@@ -0,0 +1,40 @@
+WRAPPER	= bash cvs cvs-proxy galeon gnome-panel gnuplotx hugs mplayer octave \
+	rbash scigraphica xchat
+PROFILE	= libxpg4.sh libxpg4.csh
+CLEAN	= $(WRAPPER) stamp-build
+PREFIX	= @PREFIX@
+DESTDIR	=
+
+CFLAGS	= -O3
+
+all: stamp-build
+
+stamp-build: wrapper.c
+	for i in $(WRAPPER); do \
+	  if [ $$i = bash ]; then \
+	    cc -o $$i -DPROGNAME=\"$(PREFIX)/bin/$$i.real\" -DSHNAME=\"$(PREFIX)/bin/libxpg4/sh\" wrapper.c; \
+	  elif [ $$i = cvs ]; then \
+	    cc -o $$i -DPROGNAME=\"$(PREFIX)/bin/$$i.real\" -DCVSNAME=\"/usr/bin/cvs\" wrapper.c; \
+	  else \
+	    cc -o $$i -DPROGNAME=\"$(PREFIX)/bin/$$i.real\" wrapper.c; \
+	  fi; \
+	done
+	touch $@
+
+install: all install-wrapper install-profile
+
+install-wrapper: stamp-build
+	install -d -m 755 $(DESTDIR)$(PREFIX)/bin
+	for i in $(WRAPPER); do \
+	  install -c -m 755 $$i $(DESTDIR)$(PREFIX)/bin/; \
+	done
+
+install-profile:
+	install -d -m 755 $(DESTDIR)$(PREFIX)/etc/profile.d
+	for i in $(PROFILE); do \
+	  install -c -m 755 $$i $(DESTDIR)$(PREFIX)/etc/profile.d/; \
+	done
+
+clean:
+	rm -rf $(CLEAN)
+
diff -Naur libxpg4.old/fink/libxpg4.csh libxpg4.new/fink/libxpg4.csh
--- libxpg4.old/fink/libxpg4.csh	Thu Jan  1 09:00:00 1970
+++ libxpg4.new/fink/libxpg4.csh	Mon May 13 23:47:11 2002
@@ -0,0 +1,20 @@
+# libxpg4.csh
+
+# This instructs dyld to load the libxpg4 library before loading
+# any other libraries. This way the functions in the libxpg4 library
+# replace the functions in libSystem.
+
+set libxpg4="@PREFIX@/lib/xpg4/libxpg4.dylib"
+
+if ( -e "$libxpg4" ) then
+	set libpath="$libxpg4"
+	if ( $?DYLD_INSERT_LIBRARIES ) then
+		if ( "$DYLD_INSERT_LIBRARIES" != "") then
+			set libpath=`echo "${libxpg4}:${DYLD_INSERT_LIBRARIES}" | tr ':' '\n' | sort | uniq | paste -s -d ':' -`
+		endif
+	endif
+
+	setenv DYLD_INSERT_LIBRARIES "$libpath"
+	setenv DYLD_FORCE_FLAT_NAMESPACE 1
+endif
+
diff -Naur libxpg4.old/fink/libxpg4.sh libxpg4.new/fink/libxpg4.sh
--- libxpg4.old/fink/libxpg4.sh	Thu Jan  1 09:00:00 1970
+++ libxpg4.new/fink/libxpg4.sh	Mon May 13 23:47:11 2002
@@ -0,0 +1,23 @@
+# libxpg4.sh
+
+# This instructs dyld to load the libxpg4 library before loading
+# any other libraries. This way the functions in the libxpg4 library
+# replace the functions in libSystem.
+
+libxpg4="@PREFIX@/lib/xpg4/libxpg4.dylib"
+
+if [ -f "$libxpg4" ]; then
+	if [ -z "$DYLD_INSERT_LIBRARIES" ]; then
+		DYLD_INSERT_LIBRARIES="$libxpg4"
+	else
+		DYLD_INSERT_LIBRARIES=`				\
+			echo "$libxpg4:$DYLD_INSERT_LIBRARIES"	\
+			| tr ':' '\n' | sort | uniq		\
+			| paste -s -d ':' -`
+	fi
+	DYLD_FORCE_FLAT_NAMESPACE=1
+
+	export DYLD_INSERT_LIBRARIES
+	export DYLD_FORCE_FLAT_NAMESPACE
+fi
+
diff -Naur libxpg4.old/fink/wrapper.c libxpg4.new/fink/wrapper.c
--- libxpg4.old/fink/wrapper.c	Thu Jan  1 09:00:00 1970
+++ libxpg4.new/fink/wrapper.c	Mon May 13 23:51:08 2002
@@ -0,0 +1,58 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#ifdef SHNAME
+char *
+get_progname (const char *name)
+{
+	int len;
+
+	len = strlen (name);
+	if (len < 2)
+		return PROGNAME;
+	if (name[len - 2] != 's' || name[len - 1] != 'h')
+		return PROGNAME;
+	if (len > 2 && (name [len - 3] != '/'))
+		return PROGNAME;
+
+	return SHNAME;
+}
+#elif defined(CVSNAME)
+#include <sys/types.h>
+#include <sys/stat.h>
+char *
+get_progname (const char *name)
+{
+	struct stat sb;
+	if (stat(PROGNAME, &sb) == -1)
+		return CVSNAME;
+	return PROGNAME;
+}
+#else
+char *
+get_progname (const char *name)
+{
+	return PROGNAME;
+}
+#endif
+
+int
+main(int argc, char *argv[])
+{
+	char **new_argv;
+
+	new_argv = calloc (argc + 1, sizeof (char*));
+	if (new_argv == NULL)
+		return -1;
+
+	memcpy (new_argv, argv, sizeof (char*) * argc);
+	new_argv[0] = get_progname (argv[0]);
+
+	unsetenv ("DYLD_FORCE_FLAT_NAMESPACE");
+	execv (new_argv[0], new_argv);
+
+	fprintf (stderr, "%s is not installed.\n", argv[0]);
+	exit (1);
+}
+
diff -Naur libxpg4.old/setlocale.c libxpg4.new/setlocale.c
--- libxpg4.old/setlocale.c	Tue Jun  5 16:23:41 2001
+++ libxpg4.new/setlocale.c	Mon May 13 23:47:11 2002
@@ -51,7 +51,7 @@
 #include <locale.h>
 #include <rune.h>
 #undef _PATH_LOCALE
-#define _PATH_LOCALE "/usr/local/share/locale"
+#define _PATH_LOCALE "@PREFIX@/share/locale"
 #include <stdlib.h>
 #include <string.h>
 #include <unistd.h>
diff -Naur libxpg4.old/setrunelocale.c libxpg4.new/setrunelocale.c
--- libxpg4.old/setrunelocale.c	Tue Jun  5 18:30:19 2001
+++ libxpg4.new/setrunelocale.c	Mon May 13 23:47:11 2002
@@ -36,7 +36,7 @@
 
 #include <rune.h>
 #undef _PATH_LOCALE
-#define _PATH_LOCALE "/usr/local/share/locale"
+#define _PATH_LOCALE "@PREFIX@/share/locale"
 #include <errno.h>
 #include <limits.h>
 #include <string.h>
