diff -ru sawfish-1.0.1/configure sawfish-1.0.1-patched/configure
--- sawfish-1.0.1/configure	Mon Oct 15 06:55:25 2001
+++ sawfish-1.0.1-patched/configure	Thu Dec 27 15:28:50 2001
@@ -3286,7 +3286,7 @@
 
 if test "$doug_lea_malloc" = "no"; then
   case ${host} in
-    *-dec-osf*|*-sparc-sun*)
+    *-dec-osf*|*-sparc-sun*|*-darwin*)
                   cat >> confdefs.h <<\EOF
 #define LIBC_MALLOC 1
 EOF
