diff -ru xchat-1.8.7/configure xchat-1.8.7-patched/configure
--- xchat-1.8.7/configure	Mon Jan  7 11:16:55 2002
+++ xchat-1.8.7-patched/configure	Tue Jan 29 19:13:29 2002
@@ -4357,8 +4357,8 @@
 		fi
 		PERL_LDFLAGS=`echo $PERL_LDFLAGS |$sedpath 's/-lc //'`
 		echo "$ac_t""ok" 1>&6
-		CFLAGS="$CFLAGS $PERL_CFLAGS"
-		LIBS="$LIBS $PERL_LDFLAGS"
+		CFLAGS="$PERL_CFLAGS $CFLAGS"
+		LIBS="$PERL_LDFLAGS $LIBS"
 		cat >> confdefs.h <<\EOF
 #define USE_PERL 1
 EOF
