
--- fung-calc-1.3.2b/Makefile.am	Sun Jul 13 15:04:45 2003
+++ fung-calc-1.3.2b-new/Makefile.am	Fri May  5 17:28:17 2006
@@ -3,27 +3,27 @@
 EXTRA_DIST = admin AUTHORS COPYING ChangeLog INSTALL README TODO samplegraphs.fgc
 
 install-data-local:
-	$(mkinstalldirs) $(prefix)/share/fung-calc/
-	$(INSTALL_DATA) $(srcdir)/AUTHORS $(prefix)/share/fung-calc/AUTHORS
-	$(mkinstalldirs) $(prefix)/share/fung-calc/
-	$(INSTALL_DATA) $(srcdir)/COPYING $(prefix)/share/fung-calc/COPYING
-	$(mkinstalldirs) $(prefix)/share/fung-calc/
-	$(INSTALL_DATA) $(srcdir)/ChangeLog $(prefix)/share/fung-calc/ChangeLog
-	$(mkinstalldirs) $(prefix)/share/fung-calc/
-	$(INSTALL_DATA) $(srcdir)/README $(prefix)/share/fung-calc/README
-	$(mkinstalldirs) $(prefix)/share/fung-calc/
-	$(INSTALL_DATA) $(srcdir)/TODO $(prefix)/share/fung-calc/TODO
-	$(mkinstalldirs) $(prefix)/share/fung-calc/
-	$(INSTALL_DATA) $(srcdir)/samplegraphs.fgc $(prefix)/share/fung-calc/samplegraphs.fgc
+	$(mkinstalldirs) $(DESTDIR)$(prefix)/share/fung-calc/
+	$(INSTALL_DATA) $(srcdir)/AUTHORS $(DESTDIR)$(prefix)/share/fung-calc/AUTHORS
+	$(mkinstalldirs) $(DESTDIR)$(prefix)/share/fung-calc/
+	$(INSTALL_DATA) $(srcdir)/COPYING $(DESTDIR)$(prefix)/share/fung-calc/COPYING
+	$(mkinstalldirs) $(DESTDIR)$(prefix)/share/fung-calc/
+	$(INSTALL_DATA) $(srcdir)/ChangeLog $(DESTDIR)$(prefix)/share/fung-calc/ChangeLog
+	$(mkinstalldirs) $(DESTDIR)$(prefix)/share/fung-calc/
+	$(INSTALL_DATA) $(srcdir)/README $(DESTDIR)$(prefix)/share/fung-calc/README
+	$(mkinstalldirs) $(DESTDIR)$(prefix)/share/fung-calc/
+	$(INSTALL_DATA) $(srcdir)/TODO $(DESTDIR)$(prefix)/share/fung-calc/TODO
+	$(mkinstalldirs) $(DESTDIR)$(prefix)/share/fung-calc/
+	$(INSTALL_DATA) $(srcdir)/samplegraphs.fgc $(DESTDIR)$(prefix)/share/fung-calc/samplegraphs.fgc
 
 uninstall-local:
-	-rm -f ${prefix)/share/fung-calc
-	-rm -f $(prefix)/share/fung-calc/AUTHORS
-	-rm -f $(prefix)/share/fung-calc/COPYING
-	-rm -f $(prefix)/share/fung-calc/ChangeLog
-	-rm -f $(prefix)/share/fung-calc/README
-	-rm -f $(prefix)/share/fung-calc/TODO
-	-rm -f $(prefix)/share/fung-calc/samplegraphs.fgc
+	-rm -f $(DESTDIR)$(prefix)/share/fung-calc
+	-rm -f $(DESTDIR)$(prefix)/share/fung-calc/AUTHORS
+	-rm -f $(DESTDIR)$(prefix)/share/fung-calc/COPYING
+	-rm -f $(DESTDIR)$(prefix)/share/fung-calc/ChangeLog
+	-rm -f $(DESTDIR)$(prefix)/share/fung-calc/README
+	-rm -f $(DESTDIR)$(prefix)/share/fung-calc/TODO
+	-rm -f $(DESTDIR)$(prefix)/share/fung-calc/samplegraphs.fgc
 
 $(top_srcdir)/configure.in: configure.in.in $(top_srcdir)/subdirs
 	cd $(top_srcdir) && $(MAKE) -f admin/Makefile.common configure.in ;
--- fung-calc-1.3.2b/src/libfungcalc/3D/Makefile.am	Thu Jul  3 19:22:01 2003
+++ fung-calc-1.3.2b-new/src/libfungcalc/3D/Makefile.am	Fri May  5 17:29:44 2006
@@ -17,38 +17,3 @@
 	glgraphevent.h	\
 	glparametricgraph.h	\
 	glpolargraph.h
-	
-
-#>+ 2
-docs-am:
-
-#>+ 6
-force-reedit:
-	
-
-	cd $(top_srcdir) && perl admin/am_edit fung_calc/libfungcalc/graphs/3D/Makefile.am
-
-
-#>+ 2
-final:
-	$(MAKE) all-am
-#>+ 2
-final-install:
-	$(MAKE) install-am
-#>+ 2
-no-final:
-	$(MAKE) all-am
-#>+ 2
-no-final-install:
-	$(MAKE) install-am
-#>+ 3
-cvs-clean:
-	$(MAKE) -f $(top_srcdir)/admin/Makefile.common cvs-clean
-
-#>+ 3
-kde-rpo-clean:
-	-rm -f *.rpo
-
-#>+ 3
-clean: kde-rpo-clean 
-all: docs-am 
--- fung-calc-1.3.2b/src/translations/Makefile.am	Thu Jul  3 19:22:01 2003
+++ fung-calc-1.3.2b-new/src/translations/Makefile.am	Fri May  5 17:28:43 2006
@@ -1,10 +1,10 @@
 EXTRA_DIST = fung-calc.es.qm 
 
 install-data-local:
-	$(mkinstalldirs) $(prefix)/share/fung-calc/translations/
-	$(INSTALL_DATA) $(srcdir)/fung-calc.es.qm $(prefix)/share/fung-calc/translations/fung-calc.es.qm
+	$(mkinstalldirs) $(DESTDIR)$(prefix)/share/fung-calc/translations/
+	$(INSTALL_DATA) $(srcdir)/fung-calc.es.qm $(DESTDIR)$(prefix)/share/fung-calc/translations/fung-calc.es.qm
 
 uninstall-local:
-	-rm -f $(prefix)/share/fung-calc/translations/fung-calc.es.qm
-	-rm -f $(prefix)/share/fung-calc/translations
+	-rm -f $(DESTDIR)$(prefix)/share/fung-calc/translations/fung-calc.es.qm
+	-rm -f $(DESTDIR)$(prefix)/share/fung-calc/translations
 
