diff -uNbr bash_completion/bash-completion.sh bash_completion-new/bash-completion.sh
--- bash_completion/bash-completion.sh	Wed Dec 31 19:00:00 1969
+++ bash_completion-new/bash-completion.sh	Sun Jul 28 14:30:17 2002
@@ -0,0 +1,11 @@
+#!/bin/sh
+
+if test ${SHELL#*bin/} = "bash"; then
+	bash=${BASH_VERSION%.*}; bmajor=${bash%.*}; bminor=${bash#*.}
+	if [ "$PS1" ] && [ $bmajor -eq 2 ] && [ $bminor '>' 04 ] \
+		&& [ -f @PREFIX@/etc/bash_completion ]; then # interactive shell
+		# Source completion code
+        	. @PREFIX@/etc/bash_completion
+	fi
+	unset bash bmajor bminor
+fi
diff -uNbr bash_completion/bash_completion bash_completion-new/bash_completion
--- bash_completion/bash_completion	Sat Jul 27 05:12:57 2002
+++ bash_completion-new/bash_completion	Sun Jul 28 14:14:59 2002
@@ -24,8 +24,8 @@
 
 # Alter the following to reflect the location of this file
 #
-declare -r BASH_COMPLETION=${BASH_COMPLETION:-/etc/bash_completion} 2>/dev/null
-declare -r BASH_COMPLETION_DIR=${BASH_COMPLETION_DIR:=/etc/bash_completion.d} 2>/dev/null
+declare -r BASH_COMPLETION=${BASH_COMPLETION:-@PREFIX@/etc/bash_completion} 2>/dev/null
+declare -r BASH_COMPLETION_DIR=${BASH_COMPLETION_DIR:=@PREFIX@/etc/bash_completion.d} 2>/dev/null
 
 # Set a couple of useful vars
 #
