diff -ruN ghc-6.4-orig/ghc/driver/mangler/ghc-asm.lprl ghc-6.4/ghc/driver/mangler/ghc-asm.lprl
--- ghc-6.4-orig/ghc/driver/mangler/ghc-asm.lprl	2005-06-20 01:09:59.000000000 -0700
+++ ghc-6.4/ghc/driver/mangler/ghc-asm.lprl	2005-06-20 01:15:06.000000000 -0700
@@ -550,6 +550,24 @@
             # code.
             # The .no_dead_strip directives are actually put there by
             # the gcc3 "used" attribute on entry points.
+	} elsif ( $TargetPlatform =~ /^powerpc-apple-.*/ && ( 
+		   /^\.picsymbol_stub/
+		|| /^\.section __TEXT,__picsymbol_stub1,.*/
+		|| /^\.section __TEXT,__picsymbolstub1,.*/
+		|| /^\.symbol_stub/
+		|| /^\.section __TEXT,__symbol_stub1,.*/
+		|| /^\.section __TEXT,__symbolstub1,.*/
+		|| /^\.lazy_symbol_pointer/
+		|| /^\.non_lazy_symbol_pointer/ ))
+	{
+	    $chk[++$i]   = $_;
+	    $chkcat[$i]  = 'dyld';
+	    $chksymb[$i] = '';
+	} elsif ( $TargetPlatform =~ /^powerpc-apple-.*/ && /^\.data/ && $chkcat[$i] eq 'dyld')
+	{	# non_lazy_symbol_ptrs that point to local symbols
+	    $chk[++$i]   = $_;
+	    $chkcat[$i]  = 'dyld';
+	    $chksymb[$i] = '';
 	} elsif ( /^\s+/ ) { # most common case first -- a simple line!
 	    # duplicated from the bottom
 
@@ -696,24 +714,6 @@
 	    	$chkcat[$i]  = 'unknown';
 	    }
 
-	} elsif ( $TargetPlatform =~ /^powerpc-apple-.*/ && ( 
-		   /^\.picsymbol_stub/
-		|| /^\.section __TEXT,__picsymbol_stub1,.*/
-		|| /^\.section __TEXT,__picsymbolstub1,.*/
-		|| /^\.symbol_stub/
-		|| /^\.section __TEXT,__symbol_stub1,.*/
-		|| /^\.section __TEXT,__symbolstub1,.*/
-		|| /^\.lazy_symbol_pointer/
-		|| /^\.non_lazy_symbol_pointer/ ))
-	{
-	    $chk[++$i]   = $_;
-	    $chkcat[$i]  = 'dyld';
-	    $chksymb[$i] = '';
-	} elsif ( $TargetPlatform =~ /^powerpc-apple-.*/ && /^\.data/ && $chkcat[$i] eq 'dyld')
-	{	# non_lazy_symbol_ptrs that point to local symbols
-	    $chk[++$i]   = $_;
-	    $chkcat[$i]  = 'dyld';
-	    $chksymb[$i] = '';
 	} elsif ( $TargetPlatform =~ /^powerpc-.*-linux/ && /^\.LCTOC1 = /o ) {
 		# PowerPC Linux's large-model PIC (-fPIC) generates a gobal offset
 		# table "by hand". Be sure to copy it over.
diff -ruN ghc-6.4-orig/ghc/rts/GC.c ghc-6.4/ghc/rts/GC.c
--- ghc-6.4-orig/ghc/rts/GC.c	2005-06-20 01:09:58.000000000 -0700
+++ ghc-6.4/ghc/rts/GC.c	2005-06-20 01:12:22.000000000 -0700
@@ -80,7 +80,7 @@
  * We build up a static object list while collecting generations 0..N,
  * which is then appended to the static object list of generation N+1.
  */
-static StgClosure* static_objects;      // live static objects
+StgClosure* static_objects;      // live static objects
 StgClosure* scavenged_static_objects;   // static objects scavenged so far
 
 /* N is the oldest generation being collected, where the generations
diff -ruN ghc-6.4-orig/mk/build.mk ghc-6.4/mk/build.mk
--- ghc-6.4-orig/mk/build.mk	1969-12-31 16:00:00.000000000 -0800
+++ ghc-6.4/mk/build.mk	2005-06-20 01:11:46.000000000 -0700
@@ -0,0 +1,9 @@
+#
+# Local configuration overrides for DarwinPorts
+#
+ReadlineIncludePath=${realprefix}/include
+SRC_CC_OPTS += -I${realprefix}/include
+SRC_HC_OPTS += -I${realprefix}/include -L${realprefix}/lib
+EXTRA_HSC2HS_OPTS += -I${realprefix}/include
+EXTRA_LD_OPTS += -L${realprefix}/lib
+EXTRA_HC_OPTS += -H128M
