diff -uNbr qsa-x11-free-20030513/examples/scriptbutton/scriptbutton.pro qsa-x11-free-20030513-new/examples/scriptbutton/scriptbutton.pro
--- qsa-x11-free-20030513/examples/scriptbutton/scriptbutton.pro	Tue May 13 08:35:47 2003
+++ qsa-x11-free-20030513-new/examples/scriptbutton/scriptbutton.pro	Sat May 17 02:39:25 2003
@@ -3,3 +3,5 @@
 
 # Leave away .. in your own code. See comment in ../qsa.prf.
 load( ../qsa )
+
+LIBS += -leditor
diff -uNbr qsa-x11-free-20030513/src/qsa/qsa.prf qsa-x11-free-20030513-new/src/qsa/qsa.prf
--- qsa-x11-free-20030513/src/qsa/qsa.prf	Tue May 13 08:35:47 2003
+++ qsa-x11-free-20030513-new/src/qsa/qsa.prf	Sat May 17 11:30:55 2003
@@ -1,3 +1,3 @@
 CONFIG		+= qt warn_on
-unix:LIBS	+= -lqsa
+unix:LIBS	+= -L$${QSA_BUILD_PREFIX}/lib -lqsa
 win32:LIBS	+= qsa100.lib
diff -uNbr qsa-x11-free-20030513/src/qsa/qsa.pro qsa-x11-free-20030513-new/src/qsa/qsa.pro
--- qsa-x11-free-20030513/src/qsa/qsa.pro	Tue May 13 08:35:47 2003
+++ qsa-x11-free-20030513-new/src/qsa/qsa.pro	Sat May 17 11:31:05 2003
@@ -55,6 +55,7 @@
 win32-msvc.net:DEFINES+=Q_TEMPLATE_EXTERN=extern
 
 LIBS += -L$${QT_INSTALL_PREFIX}/lib
+
 unix {
 	!hpux-acc:LIBS += -leditor -lqui
 	hpux-acc:LIBS += -Wl,-a,default -leditor
