diff -uNbr qt-x11-free-3.1.0-b2/include/qglobal.h qt-x11-free-3.1.0-b2-new/include/qglobal.h
--- qt-x11-free-3.1.0-b2/include/qglobal.h	Mon Sep 30 10:21:34 2002
+++ qt-x11-free-3.1.0-b2-new/include/qglobal.h	Wed Oct 23 22:26:36 2002
@@ -78,9 +78,9 @@
      UNIX	- Any UNIX BSD/SYSV system
 */
 
-#if defined(__APPLE__) && defined(__GNUC__)
+#if defined(__APPLE__) && defined(__GNUC__) && !defined(Q_OS_DARWIN)
 #  define Q_OS_MACX
-#elif defined(__MACOSX__)
+#elif defined(__MACOSX__) && !defined(Q_OS_DARWIN)
 #  define Q_OS_MACX
 #elif defined(macintosh)
 #  define Q_OS_MAC9
@@ -150,10 +150,12 @@
 #  define Q_OS_UNIXWARE
 #  define Q_OS_UNIXWARE7
 #else
+#  if !defined(Q_OS_DARWIN)
 #  error "Qt has not been ported to this OS - talk to qt-bugs@trolltech.com"
+#  endif
 #endif
 
-#if defined(Q_OS_MAC9) || defined(Q_OS_MACX)
+#if defined(Q_OS_MAC9) || defined(Q_OS_MACX) || defined(Q_OS_DARWIN)
 #  define Q_OS_MAC
 #endif
 
@@ -649,7 +651,7 @@
 #define Q_UINT64		Q_ULONG
 #endif
 
-#if defined(Q_OS_MACX) && !defined(QT_LARGEFILE_SUPPORT)
+#if (defined(Q_OS_MACX) || defined(Q_OS_DARWIN)) && !defined(QT_LARGEFILE_SUPPORT)
 #  define QT_LARGEFILE_SUPPORT 64
 #endif
 #if defined(QT_LARGEFILE_SUPPORT)
@@ -852,7 +854,7 @@
 #  define QT_DEBUG				// display debug messages
 #  if !defined(QT_NO_COMPAT)			// compatibility with Qt 2
 #    if !defined(NO_DEBUG) && !defined(DEBUG)
-#      if !defined(Q_OS_MACX)			// clash with MacOS X headers
+#      if !defined(Q_OS_MACX) && !defined(Q_OS_DARWIN)	// clash with MacOS X headers
 #        define DEBUG
 #      endif
 #    endif
diff -uNbr qt-x11-free-3.1.0-b2/include/qt.h qt-x11-free-3.1.0-b2-new/include/qt.h
--- qt-x11-free-3.1.0-b2/include/qt.h	Mon Sep  9 06:00:28 2002
+++ qt-x11-free-3.1.0-b2-new/include/qt.h	Wed Oct 23 21:37:18 2002
@@ -323,7 +323,7 @@
 #include <qfontfactoryttf_qws.h>
 #include <qgfx_qws.h>
 #include <qgfxvnc_qws.h>
-#if !defined( QT_NO_SOUND ) && !defined( Q_OS_MACX )
+#if !defined( QT_NO_SOUND ) && !defined( Q_OS_MACX ) && !defined( Q_OS_DARWIN )
 # include <qsoundqss_qws.h>
 #endif
 #include "qwsdisplay_qws.h"
diff -uNbr qt-x11-free-3.1.0-b2/mkspecs/darwin-g++/qmake.conf qt-x11-free-3.1.0-b2-new/mkspecs/darwin-g++/qmake.conf
--- qt-x11-free-3.1.0-b2/mkspecs/darwin-g++/qmake.conf	Mon Sep 30 10:22:13 2002
+++ qt-x11-free-3.1.0-b2-new/mkspecs/darwin-g++/qmake.conf	Thu Oct 24 08:19:28 2002
@@ -6,18 +6,18 @@
 
 MAKEFILE_GENERATOR	= UNIX
 TEMPLATE		= app
-CONFIG			+= qt warn_on release resource_fork
+CONFIG			+= qt warn_on release lib_version_first
 
 QMAKE_CC		= cc
 QMAKE_LEX		= flex
 QMAKE_LEXFLAGS		= 
 QMAKE_YACC		= yacc
 QMAKE_YACCFLAGS		= -d
-QMAKE_CFLAGS		= -pipe
+QMAKE_CFLAGS		= -fPIC -fno-common -pipe -Ddlsym=dlsym_auto_underscore
 QMAKE_CFLAGS_DEPS	= -M
 QMAKE_CFLAGS_WARN_ON	= -Wall -W
 QMAKE_CFLAGS_WARN_OFF	= -w
-QMAKE_CFLAGS_RELEASE	= -O3
+QMAKE_CFLAGS_RELEASE	= -Os -frename-registers -mmultiple -mcpu=750 -mtune=750
 QMAKE_CFLAGS_DEBUG	= -g
 QMAKE_CFLAGS_SHLIB	= -fPIC
 QMAKE_EXTENSION_SHLIB	= dylib
@@ -35,8 +35,8 @@
 QMAKE_CXXFLAGS_YACC	= $$QMAKE_CFLAGS_YACC
 QMAKE_CXXFLAGS_THREAD	=
 
-QMAKE_INCDIR		= /usr/local/include
-QMAKE_LIBDIR		=
+QMAKE_INCDIR		= @PREFIX@/include
+QMAKE_LIBDIR		= @PREFIX@/lib
 QMAKE_INCDIR_X11	= /usr/X11R6/include
 QMAKE_LIBDIR_X11	= /usr/X11R6/lib
 QMAKE_INCDIR_QT		= $(QTDIR)/include
@@ -46,22 +46,22 @@
 
 QMAKE_LINK		= c++
 QMAKE_LINK_SHLIB	= c++
-QMAKE_LFLAGS		=
+QMAKE_LFLAGS		= -framework CoreServices
 QMAKE_LFLAGS_RELEASE	=
 QMAKE_LFLAGS_DEBUG	=
-QMAKE_LFLAGS_SHLIB	= -dynamiclib
+QMAKE_LFLAGS_SHLIB	= -dynamiclib -install_name $$QMAKE_LIBDIR/$(TARGET) -bind_at_load
 QMAKE_LFLAGS_PLUGIN	= -bundle
 QMAKE_LFLAGS_THREAD	= 
 QMAKE_RPATH		= 
 
-QMAKE_LIBS_DYNLOAD	=
+QMAKE_LIBS_DYNLOAD	= -ldl
 QMAKE_LIBS_X11		= -lXext -lX11 -lm
 QMAKE_LIBS_X11SM	= -lSM -lICE
 QMAKE_LIBS_QT		= -lqt
 QMAKE_LIBS_QT_THREAD	= -lqt-mt
 QMAKE_LIBS_OPENGL	= -lGLU -lGL -lXmu
 QMAKE_LIBS_OPENGL_QT	= -lGL -lXmu
-QMAKE_LIBS_THREAD	=
+QMAKE_LIBS_THREAD	= -lXt
 
 QMAKE_MOC		= $(QTDIR)/bin/moc
 QMAKE_UIC		= $(QTDIR)/bin/uic
diff -uNbr qt-x11-free-3.1.0-b2/qmake/generators/mac/metrowerks_xml.cpp qt-x11-free-3.1.0-b2-new/qmake/generators/mac/metrowerks_xml.cpp
--- qt-x11-free-3.1.0-b2/qmake/generators/mac/metrowerks_xml.cpp	Mon Sep 30 10:21:58 2002
+++ qt-x11-free-3.1.0-b2-new/qmake/generators/mac/metrowerks_xml.cpp	Wed Oct 23 22:35:51 2002
@@ -42,7 +42,7 @@
 #include <qregexp.h>
 #include <stdlib.h>
 #include <time.h>
-#ifdef Q_OS_MAC
+#ifdef Q_OS_MACX
 #include <Carbon/Carbon.h>
 #include <sys/types.h>
 #include <sys/stat.h>
@@ -664,7 +664,7 @@
     static QString st_volume;
     if(st_volume.isEmpty()) {
 	st_volume = var("QMAKE_VOLUMENAME");
-#ifdef Q_OS_MAC
+#ifdef Q_OS_MACX
 	if(st_volume.isEmpty()) {
 	    uchar foo[512];
 	    HVolumeParam pb;
diff -uNbr qt-x11-free-3.1.0-b2/qmake/generators/unix/unixmake.cpp qt-x11-free-3.1.0-b2-new/qmake/generators/unix/unixmake.cpp
--- qt-x11-free-3.1.0-b2/qmake/generators/unix/unixmake.cpp	Mon Sep 30 10:21:58 2002
+++ qt-x11-free-3.1.0-b2-new/qmake/generators/unix/unixmake.cpp	Wed Oct 23 22:52:39 2002
@@ -134,14 +134,6 @@
 	    !project->variables()["QMAKE_LIB_FLAG"].isEmpty() &&
 	    project->isActiveConfig("dll"))
 	project->variables()["QMAKE_LFLAGS"] += project->variables()["QMAKE_LFLAGS_PREBIND"];
-    if(!project->isEmpty("QMAKE_INCDIR"))
-	project->variables()["INCLUDEPATH"] += project->variables()["QMAKE_INCDIR"];
-    if(!project->isEmpty("QMAKE_LIBDIR")) {
-	if ( !project->isEmpty("QMAKE_RPATH") )
-	    project->variables()["QMAKE_LIBDIR_FLAGS"] += varGlue("QMAKE_LIBDIR", " " + var("QMAKE_RPATH"), 
-								  " " + var("QMAKE_RPATH"), "");
-	project->variables()["QMAKE_LIBDIR_FLAGS"] += varGlue( "QMAKE_LIBDIR", "-L", " -L", "" );
-    }
     if ( extern_libs && (project->isActiveConfig("qt") || project->isActiveConfig("opengl")) ) {
 	if(configs.findIndex("x11lib") == -1)
 	    configs.append("x11lib");
@@ -175,6 +167,14 @@
 	    else
 		project->variables()["QMAKE_LIBS"] += project->variables()["QMAKE_LIBS_QT"];
 	}
+    }
+    if(!project->isEmpty("QMAKE_INCDIR"))
+	project->variables()["INCLUDEPATH"] += project->variables()["QMAKE_INCDIR"];
+    if(!project->isEmpty("QMAKE_LIBDIR")) {
+	if ( !project->isEmpty("QMAKE_RPATH") )
+	    project->variables()["QMAKE_LIBDIR_FLAGS"] += varGlue("QMAKE_LIBDIR", " " + var("QMAKE_RPATH"), 
+								  " " + var("QMAKE_RPATH"), "");
+	project->variables()["QMAKE_LIBDIR_FLAGS"] += varGlue( "QMAKE_LIBDIR", "-L", " -L", "" );
     }
     if ( project->isActiveConfig("thread") ) {
 	if(project->isActiveConfig("qt"))
diff -uNbr qt-x11-free-3.1.0-b2/src/kernel/qprocess_unix.cpp qt-x11-free-3.1.0-b2-new/src/kernel/qprocess_unix.cpp
--- qt-x11-free-3.1.0-b2/src/kernel/qprocess_unix.cpp	Mon Sep 30 10:21:42 2002
+++ qt-x11-free-3.1.0-b2-new/src/kernel/qprocess_unix.cpp	Wed Oct 23 21:42:08 2002
@@ -746,7 +746,7 @@
 
 	if ( env == 0 ) { // inherit environment and start process
 	    QString command = _arguments[0];
-#if defined(Q_OS_MACX) //look in a bundle
+#if defined(Q_OS_MACX) || defined(Q_OS_DARWIN) //look in a bundle
 	    const QString mac_bundle_suffix = ".app/Contents/MacOS/";
 	    if(!QFile::exists(command) && QFile::exists(command + mac_bundle_suffix)) {
 		QString exec = command;
@@ -766,7 +766,7 @@
 	} else { // start process with environment settins as specified in env
 	    // construct the environment for exec
 	    int numEntries = env->count();
-#if defined(Q_OS_MACX)
+#if defined(Q_OS_MACX) || defined(Q_OS_DARWIN)
 	    QString ld_library_path("DYLD_LIBRARY_PATH");
 #else
 	    QString ld_library_path("LD_LIBRARY_PATH");
@@ -798,7 +798,7 @@
 		    QStringList pathList = QStringList::split( ':', getenv( "PATH" ) );
 		    for (QStringList::Iterator it = pathList.begin(); it != pathList.end(); ++it ) {
 			QString dir = *it;
-#if defined(Q_OS_MACX) //look in a bundle
+#if defined(Q_OS_MACX) || defined(Q_OS_DARWIN) //look in a bundle
 			if(!QFile::exists(dir + "/" + command) && QFile::exists(dir + "/" + command + ".app")) 
 			    dir += "/" + command + ".app/Contents/MacOS";
 #endif
@@ -808,7 +808,7 @@
 			QFileInfo fileInfo( dir + "/" + command );
 #endif
 			if ( fileInfo.isExecutable() ) {
-#if defined(Q_OS_MACX)
+#if defined(Q_OS_MACX) || defined(Q_OS_DARWIN)
 			    arglistQ[0] = fileInfo.absFilePath().local8Bit();
 #else
 			    arglistQ[0] = fileInfo.filePath().local8Bit();
@@ -819,7 +819,7 @@
 		    }
 		}
 	    }
-#if defined(Q_OS_MACX)
+#if defined(Q_OS_MACX) || defined(Q_OS_DARWIN)
 	    if(!QFile::exists(arglist[0])) {
 		QString command = arglist[0];
 		const QString mac_bundle_suffix = ".app/Contents/MacOS/";
diff -uNbr qt-x11-free-3.1.0-b2/src/kernel/qt.h qt-x11-free-3.1.0-b2-new/src/kernel/qt.h
--- qt-x11-free-3.1.0-b2/src/kernel/qt.h	Mon Sep  9 06:00:28 2002
+++ qt-x11-free-3.1.0-b2-new/src/kernel/qt.h	Wed Oct 23 21:37:18 2002
@@ -323,7 +323,7 @@
 #include <qfontfactoryttf_qws.h>
 #include <qgfx_qws.h>
 #include <qgfxvnc_qws.h>
-#if !defined( QT_NO_SOUND ) && !defined( Q_OS_MACX )
+#if !defined( QT_NO_SOUND ) && !defined( Q_OS_MACX ) && !defined( Q_OS_DARWIN )
 # include <qsoundqss_qws.h>
 #endif
 #include "qwsdisplay_qws.h"
diff -uNbr qt-x11-free-3.1.0-b2/src/tools/qdir_unix.cpp qt-x11-free-3.1.0-b2-new/src/tools/qdir_unix.cpp
--- qt-x11-free-3.1.0-b2/src/tools/qdir_unix.cpp	Mon Sep 30 10:21:35 2002
+++ qt-x11-free-3.1.0-b2-new/src/tools/qdir_unix.cpp	Wed Oct 23 21:43:39 2002
@@ -86,7 +86,7 @@
 
 bool QDir::mkdir( const QString &dirName, bool acceptAbsPath ) const
 {
-#if defined(Q_OS_MACX)  // Mac X doesn't support trailing /'s
+#if defined(Q_OS_MACX) || defined(Q_OS_DARWIN) // Mac X doesn't support trailing /'s
     QString name = dirName;
     if (dirName[dirName.length() - 1] == "/")
 	name = dirName.left( dirName.length() - 1 );
diff -uNbr qt-x11-free-3.1.0-b2/src/tools/qglobal.h qt-x11-free-3.1.0-b2-new/src/tools/qglobal.h
--- qt-x11-free-3.1.0-b2/src/tools/qglobal.h	Mon Sep 30 10:21:34 2002
+++ qt-x11-free-3.1.0-b2-new/src/tools/qglobal.h	Wed Oct 23 22:26:36 2002
@@ -78,9 +78,9 @@
      UNIX	- Any UNIX BSD/SYSV system
 */
 
-#if defined(__APPLE__) && defined(__GNUC__)
+#if defined(__APPLE__) && defined(__GNUC__) && !defined(Q_OS_DARWIN)
 #  define Q_OS_MACX
-#elif defined(__MACOSX__)
+#elif defined(__MACOSX__) && !defined(Q_OS_DARWIN)
 #  define Q_OS_MACX
 #elif defined(macintosh)
 #  define Q_OS_MAC9
@@ -150,10 +150,12 @@
 #  define Q_OS_UNIXWARE
 #  define Q_OS_UNIXWARE7
 #else
+#  if !defined(Q_OS_DARWIN)
 #  error "Qt has not been ported to this OS - talk to qt-bugs@trolltech.com"
+#  endif
 #endif
 
-#if defined(Q_OS_MAC9) || defined(Q_OS_MACX)
+#if defined(Q_OS_MAC9) || defined(Q_OS_MACX) || defined(Q_OS_DARWIN)
 #  define Q_OS_MAC
 #endif
 
@@ -649,7 +651,7 @@
 #define Q_UINT64		Q_ULONG
 #endif
 
-#if defined(Q_OS_MACX) && !defined(QT_LARGEFILE_SUPPORT)
+#if (defined(Q_OS_MACX) || defined(Q_OS_DARWIN)) && !defined(QT_LARGEFILE_SUPPORT)
 #  define QT_LARGEFILE_SUPPORT 64
 #endif
 #if defined(QT_LARGEFILE_SUPPORT)
@@ -852,7 +854,7 @@
 #  define QT_DEBUG				// display debug messages
 #  if !defined(QT_NO_COMPAT)			// compatibility with Qt 2
 #    if !defined(NO_DEBUG) && !defined(DEBUG)
-#      if !defined(Q_OS_MACX)			// clash with MacOS X headers
+#      if !defined(Q_OS_MACX) && !defined(Q_OS_DARWIN)	// clash with MacOS X headers
 #        define DEBUG
 #      endif
 #    endif
diff -uNbr qt-x11-free-3.1.0-b2/src/tools/qgpluginmanager.cpp qt-x11-free-3.1.0-b2-new/src/tools/qgpluginmanager.cpp
--- qt-x11-free-3.1.0-b2/src/tools/qgpluginmanager.cpp	Mon Sep 30 10:21:34 2002
+++ qt-x11-free-3.1.0-b2-new/src/tools/qgpluginmanager.cpp	Wed Oct 23 21:48:05 2002
@@ -316,7 +316,7 @@
 
 #if defined(Q_OS_WIN32)
     QString filter = "dll";
-#elif defined(Q_OS_MACX)
+#elif defined(Q_OS_MACX) || defined(Q_OS_DARWIN)
     QString filter = "dylib";
 #elif defined(Q_OS_UNIX)
     QString filter = "so";
@@ -329,6 +329,17 @@
 	    continue;
 	libList.append( lib );
     }
+
+#if defined(Q_OS_DARWIN)
+    /* do it again on darwin, since .dylib and .so both exist :P */
+    plugins = QDir(path).entryList( "*.so" );
+    for ( QStringList::Iterator p = plugins.begin(); p != plugins.end(); ++p ) {
+	QString lib = QDir::cleanDirPath( path + "/" + *p );
+	if ( libList.contains( lib ) )
+	    continue;
+	libList.append( lib );
+    }
+#endif
 }
 
 const QLibrary* QGPluginManager::library( const QString& feature ) const
diff -uNbr qt-x11-free-3.1.0-b2/src/tools/qlibrary.cpp qt-x11-free-3.1.0-b2-new/src/tools/qlibrary.cpp
--- qt-x11-free-3.1.0-b2/src/tools/qlibrary.cpp	Mon Sep 30 10:21:34 2002
+++ qt-x11-free-3.1.0-b2-new/src/tools/qlibrary.cpp	Wed Oct 23 21:53:31 2002
@@ -319,6 +319,7 @@
 
     QString filename = libfile;
 
+/* FIXME - do we have to search out both .dylib and .so in darwin? */
 #if defined(Q_WS_WIN)
     if ( filename.findRev( '.' ) <= filename.findRev( '/' ) )
 	filename += ".dll";
diff -uNbr qt-x11-free-3.1.0-b2/src/tools/qsettings.cpp qt-x11-free-3.1.0-b2-new/src/tools/qsettings.cpp
--- qt-x11-free-3.1.0-b2/src/tools/qsettings.cpp	Mon Sep 30 10:21:33 2002
+++ qt-x11-free-3.1.0-b2-new/src/tools/qsettings.cpp	Thu Oct 24 09:31:04 2002
@@ -455,7 +455,7 @@
 QSettingsPrivate::QSettingsPrivate( QSettings::Format format )
     : groupDirty( TRUE ), modified(FALSE), globalScope(TRUE)
 {
-#if defined(Q_WS_WIN) || defined(Q_OS_MAC)
+#if defined(Q_WS_WIN) || defined(Q_OS_MACX)
     if ( format != QSettings::Ini )
 	return;
 #endif
@@ -757,7 +757,7 @@
 */
 void QSettings::insertSearchPath( System s, const QString &path)
 {
-#if defined(Q_WS_WIN) || defined(Q_OS_MAC)
+#if defined(Q_WS_WIN) || defined(Q_OS_MACX)
     if ( d->sysd ) {
 	d->sysInsertSearchPath( s, path );
 	return;
@@ -771,12 +771,12 @@
 	return;
     }
 
-#if defined(Q_WS_WIN) || defined(Q_OS_MAC)
+#if defined(Q_WS_WIN) || defined(Q_OS_MACX)
     if ( d->sysd && s != Unix ) {
 #else
     if ( s != Unix ) {
 #endif
-#ifdef Q_OS_MAC
+#ifdef Q_OS_MACX
 	if(s != Mac) //mac is respected on the mac as well
 #endif
 	    return;
@@ -811,12 +811,12 @@
 	return;
     }
 #endif
-#if defined(Q_WS_WIN) || defined(Q_OS_MAC)
+#if defined(Q_WS_WIN) || defined(Q_OS_MACX)
     if ( d->sysd && s != Unix ) {
 #else
     if ( s != Unix ) {
 #endif
-#ifdef Q_OS_MAC
+#ifdef Q_OS_MACX
 	if(s != Mac) //mac is respected on the mac as well
 #endif
 	    return;
@@ -837,7 +837,7 @@
     d = new QSettingsPrivate( Native );
     Q_CHECK_PTR(d);
 
-#if defined(Q_WS_WIN) || defined(Q_OS_MAC)
+#if defined(Q_WS_WIN) || defined(Q_OS_MACX)
     d->sysd = 0;
     d->sysInit();
 #endif
@@ -854,7 +854,7 @@
     d = new QSettingsPrivate( format );
     Q_CHECK_PTR(d);
 
-#if defined(Q_WS_WIN) || defined(Q_OS_MAC)
+#if defined(Q_WS_WIN) || defined(Q_OS_MACX)
     d->sysd = 0;
     if ( format == Native )
 	d->sysInit();
@@ -872,7 +872,7 @@
 {
     sync();
 
-#if defined(Q_WS_WIN) || defined(Q_OS_MAC)
+#if defined(Q_WS_WIN) || defined(Q_OS_MACX)
     if ( d->sysd )
 	d->sysClear();
 #endif
@@ -887,7 +887,7 @@
 */
 bool QSettings::sync()
 {
-#if defined(Q_WS_WIN) || defined(Q_OS_MAC)
+#if defined(Q_WS_WIN) || defined(Q_OS_MACX)
     if ( d->sysd )
 	return d->sysSync();
 #endif
@@ -1019,7 +1019,7 @@
     }
 
     QString theKey = groupKey( group(), key );
-#if defined(Q_WS_WIN) || defined(Q_OS_MAC)
+#if defined(Q_WS_WIN) || defined(Q_OS_MACX)
     if ( d->sysd )
 	return d->sysReadBoolEntry( theKey, def, ok );
 #endif
@@ -1071,7 +1071,7 @@
     }
 
     QString theKey = groupKey( group(), key );
-#if defined(Q_WS_WIN) || defined(Q_OS_MAC)
+#if defined(Q_WS_WIN) || defined(Q_OS_MACX)
     if ( d->sysd )
 	return d->sysReadDoubleEntry( theKey, def, ok );
 #endif
@@ -1117,7 +1117,7 @@
 
     QString theKey = groupKey( group(), key );
 
-#if defined(Q_WS_WIN) || defined(Q_OS_MAC)
+#if defined(Q_WS_WIN) || defined(Q_OS_MACX)
     if ( d->sysd )
 	return d->sysReadNumEntry( theKey, def, ok );
 #endif
@@ -1164,7 +1164,7 @@
 
     QString theKey = groupKey( group(), key );
 
-#if defined(Q_WS_WIN) || defined(Q_OS_MAC)
+#if defined(Q_WS_WIN) || defined(Q_OS_MACX)
     if ( d->sysd )
 	return d->sysReadEntry( theKey, def, ok );
 #endif
@@ -1237,7 +1237,7 @@
 
     QString theKey = groupKey( group(), key );
 
-#if defined(Q_WS_WIN) || defined(Q_OS_MAC)
+#if defined(Q_WS_WIN) || defined(Q_OS_MACX)
     if ( d->sysd )
 	return d->sysWriteEntry( theKey, value );
 #endif
@@ -1269,7 +1269,7 @@
 
     QString theKey = groupKey( group(), key );
 
-#if defined(Q_WS_WIN) || defined(Q_OS_MAC)
+#if defined(Q_WS_WIN) || defined(Q_OS_MACX)
     if ( d->sysd )
 	return d->sysWriteEntry( theKey, value );
 #endif
@@ -1300,7 +1300,7 @@
 
     QString theKey = groupKey( group(), key );
 
-#if defined(Q_WS_WIN) || defined(Q_OS_MAC)
+#if defined(Q_WS_WIN) || defined(Q_OS_MACX)
     if ( d->sysd )
 	return d->sysWriteEntry( theKey, value );
 #endif
@@ -1363,7 +1363,7 @@
 
     QString theKey = groupKey( group(), key );
 
-#if defined(Q_WS_WIN) || defined(Q_OS_MAC)
+#if defined(Q_WS_WIN) || defined(Q_OS_MACX)
     if ( d->sysd )
 	return d->sysWriteEntry( theKey, value );
 #endif
@@ -1424,7 +1424,7 @@
 
     QString theKey = groupKey( group(), key );
 
-#if defined(Q_WS_WIN) || defined(Q_OS_MAC)
+#if defined(Q_WS_WIN) || defined(Q_OS_MACX)
     if ( d->sysd )
 	return d->sysRemoveEntry( theKey );
 #endif
@@ -1503,7 +1503,7 @@
 
     QString theKey = groupKey( group(), key );
 
-#if defined(Q_WS_WIN) || defined(Q_OS_MAC)
+#if defined(Q_WS_WIN) || defined(Q_OS_MACX)
     if ( d->sysd )
 	return d->sysEntryList( theKey );
 #endif
@@ -1600,7 +1600,7 @@
 
     QString theKey = groupKey( group(), key );
 
-#if defined(Q_WS_WIN) || defined(Q_OS_MAC)
+#if defined(Q_WS_WIN) || defined(Q_OS_MACX)
     if ( d->sysd )
 	return d->sysSubkeyList( theKey );
 #endif
@@ -1679,7 +1679,7 @@
 
     QString theKey = groupKey( group(), key );
 
-#if defined(Q_WS_WIN) || defined(Q_OS_MAC)
+#if defined(Q_WS_WIN) || defined(Q_OS_MACX)
     if ( d->sysd )
 	return QDateTime();
 #endif
diff -uNbr qt-x11-free-3.1.0-b2/tools/designer/app/main.cpp qt-x11-free-3.1.0-b2-new/tools/designer/app/main.cpp
--- qt-x11-free-3.1.0-b2/tools/designer/app/main.cpp	Thu Sep 19 08:18:07 2002
+++ qt-x11-free-3.1.0-b2-new/tools/designer/app/main.cpp	Thu Oct 24 10:56:27 2002
@@ -174,7 +174,7 @@
 #ifdef Q_WS_WIN
     extern void qInitImages_designerlib();
     qInitImages_designerlib();
-#else
+#elif !defined(Q_OS_DARWIN)
     extern void qInitImages_designer();
     qInitImages_designer();
 #endif
diff -uNbr qt-x11-free-3.1.0-b2/tools/designer/designer/designer.pro qt-x11-free-3.1.0-b2-new/tools/designer/designer/designer.pro
--- qt-x11-free-3.1.0-b2/tools/designer/designer/designer.pro	Thu Sep 26 11:50:24 2002
+++ qt-x11-free-3.1.0-b2-new/tools/designer/designer/designer.pro	Thu Oct 24 13:49:05 2002
@@ -1,5 +1,5 @@
 TEMPLATE	= lib
-CONFIG		+= qt warn_on staticlib
+CONFIG		+= qt warn_on
 CONFIG 		-= dll
 !force_static:!win32:contains(QT_PRODUCT,qt-internal) {
    CONFIG          -= staticlib
diff -uNbr qt-x11-free-3.1.0-b2/tools/designer/designer/mainwindow.cpp qt-x11-free-3.1.0-b2-new/tools/designer/designer/mainwindow.cpp
--- qt-x11-free-3.1.0-b2/tools/designer/designer/mainwindow.cpp	Thu Sep 26 07:50:02 2002
+++ qt-x11-free-3.1.0-b2-new/tools/designer/designer/mainwindow.cpp	Thu Oct 24 10:57:07 2002
@@ -121,7 +121,7 @@
 #ifdef Q_WS_WIN
     extern void qInitImages_designerlib();
     qInitImages_designerlib();
-#else
+#elif !defined(Q_OS_DARWIN)
     extern void qInitImages_designer();
     qInitImages_designer();
 #endif
diff -uNbr qt-x11-free-3.1.0-b2/tools/designer/editor/editor.pro qt-x11-free-3.1.0-b2-new/tools/designer/editor/editor.pro
--- qt-x11-free-3.1.0-b2/tools/designer/editor/editor.pro	Wed Sep 25 04:07:06 2002
+++ qt-x11-free-3.1.0-b2-new/tools/designer/editor/editor.pro	Thu Oct 24 13:48:28 2002
@@ -1,5 +1,5 @@
 TEMPLATE	= lib
-CONFIG		+= qt warn_on staticlib
+CONFIG		+= qt warn_on
 CONFIG		-= dll
 HEADERS		= editor.h  \
 		  parenmatcher.h  \
