--- ExtUtils-F77-1.14/F77.pm	2001-06-15 22:04:17.000000000 +0200
+++ ExtUtils-F77-1.14/F77.pm	2003-07-02 17:09:24.000000000 +0200
@@ -158,7 +158,7 @@
 };
 $F77config{Generic}{G77}{Trail_} = 1;
 $F77config{Generic}{G77}{Compiler} = find_in_path('g77','f77','fort77');
-$F77config{Generic}{G77}{Cflags} = '-O';
+$F77config{Generic}{G77}{Cflags} = '-O2';
 $F77config{Generic}{DEFAULT} = 'G77';
 $F77config{Generic}{F2c}     = $F77config{Generic}{G77};
 
@@ -507,6 +507,7 @@
     my $ret;
     open(OUT,">$file.f");
     print OUT "      print *, 'Hello World'\n";
+    print OUT "      STOP 'test'\n";
     print OUT "      end\n";
     close(OUT);
     print "Compiling the test Fortran program...\n";
