diff -Naur GConf-2.5.90-old/doc/gconf/Makefile.in GConf-2.5.90/doc/gconf/Makefile.in
--- GConf-2.5.90-old/doc/gconf/Makefile.in	Tue Feb 24 11:01:11 2004
+++ GConf-2.5.90/doc/gconf/Makefile.in	Thu Mar 18 11:14:50 2004
@@ -177,7 +177,7 @@
 
 # CFLAGS and LDFLAGS for compiling scan program. Only needed
 # if $(DOC_MODULE).types is non-empty.
-GTKDOC_CFLAGS = -I$(top_srcdir) $(CFLAGS) $(DEPENDENT_CFLAGS)
+GTKDOC_CFLAGS = -I$(top_srcdir) $(CFLAGS) $(CPPFLAGS) $(DEPENDENT_CFLAGS)
 
 GTKDOC_LIBS = $(LDFLAGS) $(DEPENDENT_LIBS) $(top_builddir)/gconf/libgconf-$(MAJOR_VERSION).la
 
diff -Naur GConf-2.5.90-old/gconf/Makefile.in GConf-2.5.90/gconf/Makefile.in
--- GConf-2.5.90-old/gconf/Makefile.in	Tue Feb 24 11:00:06 2004
+++ GConf-2.5.90/gconf/Makefile.in	Thu Mar 18 11:14:50 2004
@@ -210,24 +210,24 @@
 	gconfd.c
 
 
-gconfd_2_LDADD = $(EFENCE) $(INTLLIBS) $(DEPENDENT_LIBS) libgconf-$(MAJOR_VERSION).la
+gconfd_2_LDADD = libgconf-$(MAJOR_VERSION).la $(EFENCE) $(INTLLIBS) $(DEPENDENT_LIBS)
 
 # gconf_testclient_SOURCES = \
 # 	testclient.c
 
-# gconf_testclient_LDADD = $(GLIB_LIBS) $(OAF_LIBS) libgconf-client.la
+# gconf_testclient_LDADD = libgconf-client.la $(GLIB_LIBS) $(OAF_LIBS)
 
 gconftool_2_SOURCES = \
 	gconftool.c
 
 
-gconftool_2_LDADD = $(EFENCE) $(INTLLIBS) $(DEPENDENT_WITH_XML_LIBS) $(POPT_LIBS) libgconf-$(MAJOR_VERSION).la
+gconftool_2_LDADD = libgconf-$(MAJOR_VERSION).la $(EFENCE) $(INTLLIBS) $(DEPENDENT_WITH_XML_LIBS) $(POPT_LIBS)
 
 gconf_sanity_check_2_SOURCES = \
 	gconf-sanity-check.c
 
 
-gconf_sanity_check_2_LDADD = $(EFENCE) $(INTLLIBS) $(DEPENDENT_WITH_XML_AND_GTK_LIBS) $(POPT_LIBS) libgconf-$(MAJOR_VERSION).la
+gconf_sanity_check_2_LDADD = libgconf-$(MAJOR_VERSION).la $(EFENCE) $(INTLLIBS) $(DEPENDENT_WITH_XML_AND_GTK_LIBS) $(POPT_LIBS)
 
 libgconf_2_la_SOURCES = \
 	gconf-internals.c	\
diff -Naur GConf-2.5.90-old/gconf/gconf-internals.c GConf-2.5.90/gconf/gconf-internals.c
--- GConf-2.5.90-old/gconf/gconf-internals.c	Thu Oct 23 10:58:37 2003
+++ GConf-2.5.90/gconf/gconf-internals.c	Thu Mar 18 11:14:50 2004
@@ -2581,6 +2581,9 @@
         }
     }
 
+  /* needed on Darwin */
+  fsync (lock->lock_fd);
+
   return lock;
 }
 
diff -Naur GConf-2.5.90-old/ltmain.sh GConf-2.5.90/ltmain.sh
--- GConf-2.5.90-old/ltmain.sh	Tue Feb 24 10:50:18 2004
+++ GConf-2.5.90/ltmain.sh	Fri Mar 19 17:51:55 2004
@@ -3745,6 +3745,11 @@
 	  eval test_cmds=\"$archive_expsym_cmds\"
 	  cmds=$archive_expsym_cmds
 	else
+	  if test "x$verstring" = "x0.0"; then
+            tmp_verstring=
+          else
+            tmp_verstring="$verstring"
+          fi
 	  eval test_cmds=\"$archive_cmds\"
 	  cmds=$archive_cmds
 	  fi
