diff -Naur gnome-applets-2.6.0-old/Makefile.in gnome-applets-2.6.0/Makefile.in
--- gnome-applets-2.6.0-old/Makefile.in	Sun Mar 21 14:04:29 2004
+++ gnome-applets-2.6.0/Makefile.in	Sat Apr 10 10:38:45 2004
@@ -194,7 +194,6 @@
 
 @APPLET_ACCESSX_TRUE@accessx_status_SUBDIR = accessx-status
 
-@APPLET_GSWITCHIT_TRUE@gswitchit_SUBDIRS = libgswitchit gswitchit
 
 always_built_SUBDIRS = \
 	drivemount	\
@@ -202,7 +201,6 @@
 	geyes		\
 	mailcheck	\
 	mini-commander	\
-	gkb-new		\
 	wireless	\
 	stickynotes
 
@@ -217,8 +215,7 @@
 	$(gtop_SUBDIRS)		\
 	$(vfs_SUBDIRS)		\
 	$(battstat_SUBDIR)	\
-	$(accessx_status_SUBDIR)\
-	$(gswitchit_SUBDIRS)
+	$(accessx_status_SUBDIR)
 
 
 DIST_SUBDIRS = \
@@ -232,7 +229,6 @@
 	charpick	\
 	geyes		\
 	mini-commander	\
-	gkb-new		\
 	battstat	\
 	multiload	\
 	gtik		\
@@ -241,9 +237,7 @@
 	man		\
 	wireless	\
 	accessx-status	\
-	stickynotes	\
-	libgswitchit	\
-	gswitchit
+	stickynotes
 
 
 EXTRA_DIST = \
diff -Naur gnome-applets-2.6.0-old/configure gnome-applets-2.6.0/configure
--- gnome-applets-2.6.0-old/configure	Sun Mar 21 14:08:37 2004
+++ gnome-applets-2.6.0/configure	Sat Apr 10 10:37:55 2004
@@ -24230,6 +24230,9 @@
 	echo "warning: ${host} is not supported by battstat_applet, not building" >&2
 	build_battstat_applet=no
 	;;
+  *darwin*)
+	build_battstat_applet=no
+	;;
 esac
 
 
diff -Naur gnome-applets-2.6.0-old/gkb-new/preset.c gnome-applets-2.6.0/gkb-new/preset.c
--- gnome-applets-2.6.0-old/gkb-new/preset.c	Thu Jun 12 17:38:04 2003
+++ gnome-applets-2.6.0/gkb-new/preset.c	Sat Apr 10 10:37:56 2004
@@ -29,6 +29,7 @@
  * USA
  */
 
+#include <sys/types.h>
 #include <dirent.h>
 #include "gkb.h"
 
diff -Naur gnome-applets-2.6.0-old/multiload/Makefile.in gnome-applets-2.6.0/multiload/Makefile.in
--- gnome-applets-2.6.0-old/multiload/Makefile.in	Sun Mar 21 14:07:41 2004
+++ gnome-applets-2.6.0/multiload/Makefile.in	Sat Apr 10 10:37:56 2004
@@ -196,7 +196,8 @@
 multiload_applet_2_LDADD = \
 	$(top_builddir)/screen-exec/libscreen-exec.la	\
 				$(GNOME_APPLETS_LIBS) 	\
-				$(GTOP_APPLETS_LIBS)
+				$(GTOP_APPLETS_LIBS)	\
+				$(X_LIBS)
 
 
 pixmaps = gnome-cpu.png
