diff -uNr gst-plugins-0.8.3/ext/cdparanoia/gstcdparanoia.c gst-plugins-0.8.3-new/ext/cdparanoia/gstcdparanoia.c
--- gst-plugins-0.8.3/ext/cdparanoia/gstcdparanoia.c	Tue Jul 27 12:13:51 2004
+++ gst-plugins-0.8.3-new/ext/cdparanoia/gstcdparanoia.c	Sun Jan 23 09:20:07 2005
@@ -20,9 +20,6 @@
 
 /* #define GST_DEBUG_ENABLED */
 
-#ifdef HAVE_CONFIG_H
-#include "config.h"
-#endif
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
@@ -35,7 +32,6 @@
 #include <sys/types.h>
 #include <dirent.h>
 #include <errno.h>
-#include "gst/gst-i18n-plugin.h"
 
 /* taken from linux/cdrom.h */
 #define CD_MSF_OFFSET       150 /* MSF numbering offset of first frame */
@@ -44,6 +40,11 @@
 
 #include "gstcdparanoia.h"
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
+#include "gst/gst-i18n-plugin.h"
 
 static GstElementDetails cdparanoia_details = {
   "CD Audio (cdda) Source, Paranoia IV",
diff -uNr gst-plugins-0.8.3/ext/cdparanoia/gstcdparanoia.h gst-plugins-0.8.3-new/ext/cdparanoia/gstcdparanoia.h
--- gst-plugins-0.8.3/ext/cdparanoia/gstcdparanoia.h	Tue Jun 15 05:11:57 2004
+++ gst-plugins-0.8.3-new/ext/cdparanoia/gstcdparanoia.h	Sun Jan 23 09:20:07 2005
@@ -21,7 +21,6 @@
 #ifndef __CDPARANOIA_H__
 #define __CDPARANOIA_H__
 
-
 #include <glib.h>
 #include <gst/gst.h>
 
