diff -Naur gucharmap-1.3.2-old/ltmain.sh gucharmap-1.3.2/ltmain.sh
--- gucharmap-1.3.2-old/ltmain.sh	Sun Jan 25 09:59:27 2004
+++ gucharmap-1.3.2/ltmain.sh	Sun Apr  4 21:20:12 2004
@@ -3742,6 +3742,11 @@
 	  eval test_cmds=\"$archive_expsym_cmds\"
 	  cmds=$archive_expsym_cmds
 	else
+	  if test "x$verstring" = "x0.0"; then
+             tmp_verstring=
+           else
+             tmp_verstring="$verstring"
+           fi
 	  eval test_cmds=\"$archive_cmds\"
 	  cmds=$archive_cmds
 	  fi
@@ -5244,10 +5249,10 @@
 
 # Directory that this library needs to be installed in:
 libdir='$install_libdir'"
-	  if test "$installed" = no && test "$need_relink" = yes; then
-	    $echo >> $output "\
-relink_command=\"$relink_command\""
-	  fi
+#	  if test "$installed" = no && test "$need_relink" = yes; then
+#	    $echo >> $output "\
+#relink_command=\"$relink_command\""
+#	  fi
 	done
       fi
 
