diff -Naur librsvg-2.6.3-old/configure librsvg-2.6.3/configure
--- librsvg-2.6.3-old/configure	Thu Mar 18 20:18:09 2004
+++ librsvg-2.6.3/configure	Sun Apr  4 16:23:45 2004
@@ -8282,11 +8282,11 @@
             echo "$as_me:$LINENO: checking LIBRSVG_CFLAGS" >&5
 echo $ECHO_N "checking LIBRSVG_CFLAGS... $ECHO_C" >&6
             LIBRSVG_CFLAGS=`$PKG_CONFIG --cflags "\
+	pangoft2 >= $PANGOFT2_REQUIRED \
 	gdk-pixbuf-2.0 >= $GDK_PIXBUF_REQUIRED \
 	glib-2.0 >= $GLIB_REQUIRED \
 	libart-2.0 >= $LIBART_REQUIRED \
-	libxml-2.0 >= $LIBXML_REQUIRED \
-	pangoft2 >= $PANGOFT2_REQUIRED"`
+	libxml-2.0 >= $LIBXML_REQUIRED"`
             echo "$as_me:$LINENO: result: $LIBRSVG_CFLAGS" >&5
 echo "${ECHO_T}$LIBRSVG_CFLAGS" >&6
 
diff -Naur librsvg-2.6.3-old/ltmain.sh librsvg-2.6.3/ltmain.sh
--- librsvg-2.6.3-old/ltmain.sh	Mon Aug 19 21:13:37 2002
+++ librsvg-2.6.3/ltmain.sh	Sun Apr  4 19:37:59 2004
@@ -4006,10 +4006,10 @@
 
 # Directory that this library needs to be installed in:
 libdir='$install_libdir'"
-	  if test "$installed" = no && test "$need_relink" = yes; then
-	    $echo >> $output "\
-relink_command=\"$relink_command\""
-	  fi
+#	  if test "$installed" = no && test "$need_relink" = yes; then
+#	    $echo >> $output "\
+#relink_command=\"$relink_command\""
+#	  fi
 	done
       fi
 
