diff -uNr autoconf-2.13/Makefile.in autoconf-2.13-new/Makefile.in
--- autoconf-2.13/Makefile.in	Tue Jan  5 08:27:16 1999
+++ autoconf-2.13-new/Makefile.in	Sat Sep 20 13:55:33 2003
@@ -49,7 +49,7 @@
 
 # Directory in which to install library files.
 datadir = @datadir@
-acdatadir = $(datadir)/autoconf
+acdatadir = $(datadir)/autoconf-2.13
 
 # Directory in which to install documentation info files.
 infodir = @infodir@
