diff -uNbr distcc-0.12/distcc-helper distcc-0.12-new/distcc-helper
--- distcc-0.12/distcc-helper	Wed Dec 31 19:00:00 1969
+++ distcc-0.12-new/distcc-helper	Sun Oct 13 12:13:22 2002
@@ -0,0 +1,13 @@
+#!/bin/sh
+
+if test -f @PREFIX@/etc/distcc-hosts; then
+	export DISTCC_HOSTS=`cat @PREFIX@/etc/distcc-hosts`
+	export DISTCC_LOG=@PREFIX@/var/log/distcc.log
+	export DISTCC_NUMHOSTS=`echo $DISTCC_HOSTS | wc -w`
+	let DISTCC_NUMCPUS="($DISTCC_NUMHOSTS * 2)"
+	export DISTCC_NUMCPUS
+	export DISTCC_MAKEFLAGS="-j${DISTCC_NUMCPUS} CC=\"distcc cc\" CXX=\"distcc c++\" CCLD=\"cc\" CXXLD=\"c++\""
+
+	echo "- using distcc with $DISTCC_NUMCPUS hosts"
+	echo "  you can now run make as 'make \$DISTCC_MAKEFLAGS'"
+fi
diff -uNbr distcc-0.12/src/trace.h distcc-0.12-new/src/trace.h
--- distcc-0.12/src/trace.h	Sun Oct  6 22:51:21 2002
+++ distcc-0.12-new/src/trace.h	Sun Oct 13 11:09:36 2002
@@ -93,7 +93,7 @@
 /** Default trace callback that writes to stderr.  Implements
  * ::rs_trace_fn_t, and may be passed to rs_trace_to(). */
 void rs_trace_to_file(int level, const char *fn, char const *fmt, va_list va);
-int rs_trace_fd;
+extern int rs_trace_fd;
 
 
 void
