diff -uNbr mono-0.17/config.h.in mono-0.17-patched/config.h.in
--- mono-0.17/config.h.in	Fri Dec  6 12:32:17 2002
+++ mono-0.17-patched/config.h.in	Tue Dec 10 14:28:39 2002
@@ -114,3 +114,6 @@
 
 /* Version number of package */
 #undef VERSION
+
+/* Define the real type of socklen_t */
+#undef socklen_t
diff -uNbr mono-0.17/configure mono-0.17-patched/configure
--- mono-0.17/configure	Mon Dec  9 13:02:55 2002
+++ mono-0.17-patched/configure	Tue Dec 10 14:28:39 2002
@@ -1779,6 +1779,14 @@
 		CPPFLAGS="$CPPFLAGS -DGC_WIN32_THREADS"
 		libdl=
 		;;
+	*-*-*darwin*)
+		platform_win32=no
+		CPPFLAGS="$CPPFLAGS -no-cpp-precomp"
+		cat >>confdefs.h <<\_ACEOF
+#define NEED_LINK_UNLINK 1
+_ACEOF
+		libdl="-ldl"
+		;;
 	*-*-*bsd*)
 		platform_win32=no
 		CPPFLAGS="$CPPFLAGS -D_THREAD_SAFE"
@@ -8773,6 +8781,118 @@
 echo "$as_me: WARNING: No 64 bit file size support available" >&2;}
 	fi
 
+                                echo "$as_me:$LINENO: checking for socklen_t" >&5
+echo $ECHO_N "checking for socklen_t... $ECHO_C" >&6
+        if test "${ac_cv_socklen_t+set}" = set; then
+  echo $ECHO_N "(cached) $ECHO_C" >&6
+else
+
+                cat >conftest.$ac_ext <<_ACEOF
+#line $LINENO "configure"
+#include "confdefs.h"
+
+#include <sys/types.h>
+#include <sys/socket.h>
+
+#ifdef F77_DUMMY_MAIN
+#  ifdef __cplusplus
+     extern "C"
+#  endif
+   int F77_DUMMY_MAIN() { return 1; }
+#endif
+int
+main ()
+{
+
+socklen_t a=0;
+getsockname(0,(struct sockaddr*)0, &a);
+
+  ;
+  return 0;
+}
+_ACEOF
+rm -f conftest.$ac_objext
+if { (eval echo "$as_me:$LINENO: \"$ac_compile\"") >&5
+  (eval $ac_compile) 2>&5
+  ac_status=$?
+  echo "$as_me:$LINENO: \$? = $ac_status" >&5
+  (exit $ac_status); } &&
+         { ac_try='test -s conftest.$ac_objext'
+  { (eval echo "$as_me:$LINENO: \"$ac_try\"") >&5
+  (eval $ac_try) 2>&5
+  ac_status=$?
+  echo "$as_me:$LINENO: \$? = $ac_status" >&5
+  (exit $ac_status); }; }; then
+  ac_cv_socklen_t=socklen_t
+else
+  echo "$as_me: failed program was:" >&5
+cat conftest.$ac_ext >&5
+cat >conftest.$ac_ext <<_ACEOF
+#line $LINENO "configure"
+#include "confdefs.h"
+
+#include <sys/types.h>
+#include <sys/socket.h>
+
+#ifdef F77_DUMMY_MAIN
+#  ifdef __cplusplus
+     extern "C"
+#  endif
+   int F77_DUMMY_MAIN() { return 1; }
+#endif
+int
+main ()
+{
+
+int a=0;
+getsockname(0,(struct sockaddr*)0, &a);
+
+  ;
+  return 0;
+}
+_ACEOF
+rm -f conftest.$ac_objext
+if { (eval echo "$as_me:$LINENO: \"$ac_compile\"") >&5
+  (eval $ac_compile) 2>&5
+  ac_status=$?
+  echo "$as_me:$LINENO: \$? = $ac_status" >&5
+  (exit $ac_status); } &&
+         { ac_try='test -s conftest.$ac_objext'
+  { (eval echo "$as_me:$LINENO: \"$ac_try\"") >&5
+  (eval $ac_try) 2>&5
+  ac_status=$?
+  echo "$as_me:$LINENO: \$? = $ac_status" >&5
+  (exit $ac_status); }; }; then
+  ac_cv_socklen_t=int
+else
+  echo "$as_me: failed program was:" >&5
+cat conftest.$ac_ext >&5
+ac_cv_socklen_t=size_t
+
+fi
+rm -f conftest.$ac_objext conftest.$ac_ext
+
+fi
+rm -f conftest.$ac_objext conftest.$ac_ext
+
+fi
+
+
+        echo "$as_me:$LINENO: result: $ac_cv_socklen_t" >&5
+echo "${ECHO_T}$ac_cv_socklen_t" >&6
+        if test "$ac_cv_socklen_t" != "socklen_t"; then
+
+cat >>confdefs.h <<_ACEOF
+#define socklen_t $ac_cv_socklen_t
+_ACEOF
+
+        fi
+
+cat >>confdefs.h <<_ACEOF
+#define ksize_t socklen_t
+_ACEOF
+
+
 				echo "$as_me:$LINENO: checking for socket in -lsocket" >&5
 echo $ECHO_N "checking for socket in -lsocket... $ECHO_C" >&6
 if test "${ac_cv_lib_socket_socket+set}" = set; then
@@ -9918,7 +10038,7 @@
 sparc*-*-*) TARGET=SPARC; arch_target=sparc; ACCESS_UNALIGNED="no";;
 #alpha*-*-linux* | alpha*-*-osf*) TARGET=ALPHA; ACCESS_UNALIGNED="no";;
 #m68k-*-linux*) TARGET=M68K;;
-macppc-*-openbsd* | powerpc-*-linux* | powerpc-*-openbsd* | powerpc-*-sysv*) TARGET=POWERPC; arch_target=ppc;;
+powerpc-*-darwin* | macppc-*-openbsd* | powerpc-*-linux* | powerpc-*-openbsd* | powerpc-*-sysv*) TARGET=POWERPC; arch_target=ppc;;
 arm-*-linux-* | armv4l-*-linux-*) TARGET=ARM; arch_target=arm; ACCESS_UNALIGNED="no";;
 s390-*-linux*) TARGET=S390; arch_target=s390; ACCESS_UNALIGNED="no";;
 esac
diff -uNbr mono-0.17/configure.in mono-0.17-patched/configure.in
--- mono-0.17/configure.in	Mon Dec  9 13:02:42 2002
+++ mono-0.17-patched/configure.in	Tue Dec 10 14:28:39 2002
@@ -15,6 +15,12 @@
 		CPPFLAGS="$CPPFLAGS -DGC_WIN32_THREADS"
 		libdl=
 		;;
+	*-*-*darwin*)
+		platform_win32=no
+		CPPFLAGS="$CPPFLAGS -no-cpp-precomp"
+		AC_DEFINE(NEED_LINK_UNLINK)
+		libdl="-ldl"
+		;;
 	*-*-*bsd*)
 		platform_win32=no
 		CPPFLAGS="$CPPFLAGS -D_THREAD_SAFE"
@@ -212,6 +218,40 @@
 	fi
 	
 	dnl *****************************
+	dnl *** Checks for socklen_t  ***
+	dnl *****************************
+	AC_MSG_CHECKING(for socklen_t)
+	AC_CACHE_VAL(ac_cv_socklen_t, [
+		AC_TRY_COMPILE([
+#include <sys/types.h>
+#include <sys/socket.h>
+			],[
+socklen_t a=0;
+getsockname(0,(struct sockaddr*)0, &a);
+			],
+			ac_cv_socklen_t=socklen_t,
+			AC_TRY_COMPILE([
+#include <sys/types.h>
+#include <sys/socket.h>
+				],[
+int a=0;
+getsockname(0,(struct sockaddr*)0, &a);
+				],
+				ac_cv_socklen_t=int,
+				ac_cv_socklen_t=size_t
+			)
+		)
+		AC_LANG_RESTORE
+	])
+
+	AC_MSG_RESULT($ac_cv_socklen_t)
+	if test "$ac_cv_socklen_t" != "socklen_t"; then
+		AC_DEFINE_UNQUOTED(socklen_t, $ac_cv_socklen_t,
+			[Define the real type of socklen_t])
+	fi
+	AC_DEFINE_UNQUOTED(ksize_t, socklen_t, [Compatibility define])
+
+	dnl *****************************
 	dnl *** Checks for libsocket  ***
 	dnl *****************************
 	AC_CHECK_LIB(socket, socket, LIBS="$LIBS -lsocket")
@@ -478,7 +518,7 @@
 sparc*-*-*) TARGET=SPARC; arch_target=sparc; ACCESS_UNALIGNED="no";;
 #alpha*-*-linux* | alpha*-*-osf*) TARGET=ALPHA; ACCESS_UNALIGNED="no";;
 #m68k-*-linux*) TARGET=M68K;;
-macppc-*-openbsd* | powerpc-*-linux* | powerpc-*-openbsd* | powerpc-*-sysv*) TARGET=POWERPC; arch_target=ppc;;
+powerpc-*-darwin* | macppc-*-openbsd* | powerpc-*-linux* | powerpc-*-openbsd* | powerpc-*-sysv*) TARGET=POWERPC; arch_target=ppc;;
 arm-*-linux-* | armv4l-*-linux-*) TARGET=ARM; arch_target=arm; ACCESS_UNALIGNED="no";;
 s390-*-linux*) TARGET=S390; arch_target=s390; ACCESS_UNALIGNED="no";;
 esac
diff -uNbr mono-0.17/mono/interpreter/interp.h mono-0.17-patched/mono/interpreter/interp.h
--- mono-0.17/mono/interpreter/interp.h	Fri Aug  2 16:41:50 2002
+++ mono-0.17-patched/mono/interpreter/interp.h	Tue Dec 10 14:28:39 2002
@@ -1,4 +1,4 @@
-
+#include <config.h>
 #include <glib.h>
 #include <mono/metadata/loader.h>
 #include <mono/metadata/object.h>
diff -uNbr mono-0.17/mono/io-layer/Makefile.am mono-0.17-patched/mono/io-layer/Makefile.am
--- mono-0.17/mono/io-layer/Makefile.am	Thu Aug  1 11:31:07 2002
+++ mono-0.17-patched/mono/io-layer/Makefile.am	Tue Dec 10 14:28:39 2002
@@ -5,7 +5,8 @@
 	$(GMODULE_CFLAGS)	\
 	-D_WAPI_BUILDING=1	\
 	-DMONO_BINDIR=\""$(bindir)"\"	\
-	-I$(top_srcdir)	
+	-I$(top_srcdir)		\
+	-I$(prefix)/include
 
 libwapiincludedir = $(includedir)/mono/io-layer
 
diff -uNbr mono-0.17/mono/io-layer/Makefile.in mono-0.17-patched/mono/io-layer/Makefile.in
--- mono-0.17/mono/io-layer/Makefile.in	Mon Dec  9 16:50:31 2002
+++ mono-0.17-patched/mono/io-layer/Makefile.in	Tue Dec 10 14:28:39 2002
@@ -103,7 +103,8 @@
 	$(GMODULE_CFLAGS)	\
 	-D_WAPI_BUILDING=1	\
 	-DMONO_BINDIR=\""$(bindir)"\"	\
-	-I$(top_srcdir)	
+	-I$(top_srcdir)	\
+	-I$(prefix)/include
 
 
 libwapiincludedir = $(includedir)/mono/io-layer
diff -uNbr mono-0.17/mono/metadata/decimal.c mono-0.17-patched/mono/metadata/decimal.c
--- mono-0.17/mono/metadata/decimal.c	Thu Nov 28 12:11:24 2002
+++ mono-0.17-patched/mono/metadata/decimal.c	Tue Dec 10 14:28:39 2002
@@ -13,6 +13,7 @@
  * CSharp value type System.Decimal
  */
 
+#include <config.h>
 #include <mono/metadata/exception.h>
 #include <stdio.h>
 #include <memory.h>
diff -uNbr mono-0.17/mono/metadata/exception.c mono-0.17-patched/mono/metadata/exception.c
--- mono-0.17/mono/metadata/exception.c	Wed Aug 14 12:17:24 2002
+++ mono-0.17-patched/mono/metadata/exception.c	Tue Dec 10 14:28:39 2002
@@ -10,6 +10,7 @@
  * (C) 2001, 2002 Ximian, Inc.
  */
 
+#include <config.h>
 #include <mono/metadata/exception.h>
 #include <mono/metadata/class.h>
 #include <mono/metadata/appdomain.h>
diff -uNbr mono-0.17/mono/metadata/monosn.c mono-0.17-patched/mono/metadata/monosn.c
--- mono-0.17/mono/metadata/monosn.c	Thu Dec  5 11:08:19 2002
+++ mono-0.17-patched/mono/metadata/monosn.c	Tue Dec 10 14:28:39 2002
@@ -7,6 +7,7 @@
  * (C) 2002 Ximian, Inc.
  *
  */
+#include <config.h>
 #include <mono/metadata/class.h>
 #include <mono/metadata/debug-helpers.h>
 #include <mono/metadata/tokentype.h>
diff -uNbr mono-0.17/mono/metadata/profiler.c mono-0.17-patched/mono/metadata/profiler.c
--- mono-0.17/mono/metadata/profiler.c	Sat Aug 24 19:04:39 2002
+++ mono-0.17-patched/mono/metadata/profiler.c	Tue Dec 10 14:28:39 2002
@@ -1,4 +1,5 @@
 
+#include <config.h>
 #include "mono/metadata/profiler-private.h"
 #include "mono/metadata/debug-helpers.h"
 #include <string.h>
diff -uNbr mono-0.17/mono/metadata/threads.c mono-0.17-patched/mono/metadata/threads.c
--- mono-0.17/mono/metadata/threads.c	Fri Dec  6 00:31:42 2002
+++ mono-0.17-patched/mono/metadata/threads.c	Tue Dec 10 14:49:39 2002
@@ -1097,6 +1097,10 @@
 
 
 	/* fixme: store the state somewhere */
+#ifndef SIGRTMIN
+#define SIGRTMIN SIGUSR1
+#endif
+
 #ifndef __MINGW32__
 #ifdef PTHREAD_POINTER_ID
 	pthread_kill (GUINT_TO_POINTER(thread->tid), SIGRTMIN);
diff -uNbr mono-0.17/mono/metadata/verify.c mono-0.17-patched/mono/metadata/verify.c
--- mono-0.17/mono/metadata/verify.c	Mon Sep 16 14:51:17 2002
+++ mono-0.17-patched/mono/metadata/verify.c	Tue Dec 10 14:28:39 2002
@@ -1,4 +1,4 @@
-
+#include <config.h>
 #include <mono/metadata/object.h>
 #include <mono/metadata/verify.h>
 #include <mono/metadata/opcodes.h>
diff -uNbr mono-0.17/mono/monograph/monograph.c mono-0.17-patched/mono/monograph/monograph.c
--- mono-0.17/mono/monograph/monograph.c	Tue Oct  8 01:52:55 2002
+++ mono-0.17-patched/mono/monograph/monograph.c	Tue Dec 10 14:28:39 2002
@@ -1,3 +1,4 @@
+#include <config.h>
 #include <glib.h>
 #include <string.h>
 #include "mono/metadata/class.h"
diff -uNbr mono-0.17/scripts/Makefile.am mono-0.17-patched/scripts/Makefile.am
--- mono-0.17/scripts/Makefile.am	Sat Feb 16 03:36:15 2002
+++ mono-0.17-patched/scripts/Makefile.am	Tue Dec 10 15:26:00 2002
@@ -4,6 +4,12 @@
 
 CLEANFILES = mcs
 
+if X86
+MONO_INTERP = mono
+else
+MONO_INTERP = mint
+endif
+
 mcs: mcs.in
-	sed -e 's^\@bindir\@^$(bindir)^g' < $(srcdir)/mcs.in > mcs.tmp \
+	sed -e 's^\@bindir\@^$(bindir)^g' -e 's^\@mono_interp\@^$(MONO_INTERP)^g' < $(srcdir)/mcs.in > mcs.tmp \
 	&& mv mcs.tmp mcs 
diff -uNbr mono-0.17/scripts/Makefile.in mono-0.17-patched/scripts/Makefile.in
--- mono-0.17/scripts/Makefile.in	Mon Dec  9 16:50:37 2002
+++ mono-0.17-patched/scripts/Makefile.in	Tue Dec 10 15:26:00 2002
@@ -224,9 +224,11 @@
 mostlyclean-generic distclean-generic clean-generic \
 maintainer-clean-generic clean mostlyclean distclean maintainer-clean
 
+@X86_TRUE@MONO_INTERP = mono
+@X86_FALSE@MONO_INTERP = mint
 
 mcs: mcs.in
-	sed -e 's^\@bindir\@^$(bindir)^g' < $(srcdir)/mcs.in > mcs.tmp \
+	sed -e 's^\@bindir\@^$(bindir)^g' -e 's^\@mono_interp\@^$(MONO_INTERP)^g' < $(srcdir)/mcs.in > mcs.tmp \
 	&& mv mcs.tmp mcs 
 
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
diff -uNbr mono-0.17/scripts/mcs.in mono-0.17-patched/scripts/mcs.in
--- mono-0.17/scripts/mcs.in	Tue May  7 17:25:30 2002
+++ mono-0.17-patched/scripts/mcs.in	Tue Dec 10 15:26:00 2002
@@ -1,2 +1,2 @@
 #!/bin/sh
-@bindir@/mono @bindir@/mcs.exe "$@"
+@bindir@/@mono_interp@ @bindir@/mcs.exe "$@"
