diff -uNbr mono-0.24/configure.in mono-0.24-patched/configure.in
--- mono-0.24/configure.in	Tue May  6 17:04:31 2003
+++ mono-0.24-patched/configure.in	Wed May  7 21:49:38 2003
@@ -51,11 +51,9 @@
 		platform_win32=no
 		CPPFLAGS="$CPPFLAGS -no-cpp-precomp -D_THREAD_SAFE"
 		libmono_cflags="-D_THREAD_SAFE"
-		LDFLAGS="$LDFLAGS -pthread"
-		libmono_ldflags="-pthread"
 		AC_DEFINE(NEED_LINK_UNLINK)
 		AC_DEFINE(PTHREAD_POINTER_ID)
-		libdl=
+		libdl="-ldl"
 		;;
 	*)
 		AC_MSG_WARN([*** Please add $host to configure.in checks!])
