diff -Naur gconf-editor-0.1.old/src/gconf-util.c gconf-editor-0.1.new/src/gconf-util.c
--- gconf-editor-0.1.old/src/gconf-util.c	Sat Feb  9 20:11:55 2002
+++ gconf-editor-0.1.new/src/gconf-util.c	Wed May 22 23:11:46 2002
@@ -49,6 +49,7 @@
 	return g_strdup (ptr);
 }
 
+#if !defined(__APPLE__)
 gchar *
 gconf_value_type_to_string (GConfValueType value_type)
 {
@@ -69,3 +70,4 @@
 		return g_strdup_printf ("UNKNOWN, %d", value_type);
 	}
 }
+#endif
diff -Naur gconf-editor-0.1.old/src/gconf-util.h gconf-editor-0.1.new/src/gconf-util.h
--- gconf-editor-0.1.old/src/gconf-util.h	Sat Feb  9 20:11:52 2002
+++ gconf-editor-0.1.new/src/gconf-util.h	Wed May 22 23:11:46 2002
@@ -29,6 +29,8 @@
 
 GType gconf_value_get_type (void);
 gchar *gconf_get_key_name_from_path (const gchar *path);
+#if !defined(__APPLE__)
 gchar *gconf_value_type_to_string (GConfValueType value_type);
+#endif
 
 #endif /* __GCONF_UTIL_H__ */
