diff -Naur gedit-1.117.0.old/configure gedit-1.117.0.new/configure
--- gedit-1.117.0.old/configure	Tue Apr 16 12:44:40 2002
+++ gedit-1.117.0.new/configure	Thu May 23 00:46:11 2002
@@ -4564,7 +4564,7 @@
     # FIXME: Relying on posixy $() will cause problems for
     #        cross-compilation, but unfortunately the echo tests do not
     #        yet detect zsh echo's removal of \ escapes.
-    archive_cmds='$nonopt $(test "x$module" = xyes && echo -bundle || echo -dynamiclib) $allow_undefined_flag -o $lib $libobjs $deplibs$linker_flags -install_name $rpath/$soname $verstring'
+    archive_cmds='$nonopt $(test x$module = xyes && echo -bundle || echo -dynamiclib) $allow_undefined_flag -o $lib $libobjs $deplibs$linker_flags -install_name $rpath/$soname $verstring'
     # We need to add '_' to the symbols in $export_symbols first
     #archive_expsym_cmds="$archive_cmds"' && strip -s $export_symbols'
     hardcode_direct=yes
diff -Naur gedit-1.117.0.old/plugins/ASCII/Makefile.in gedit-1.117.0.new/plugins/ASCII/Makefile.in
--- gedit-1.117.0.old/plugins/ASCII/Makefile.in	Tue Apr 16 12:45:16 2002
+++ gedit-1.117.0.new/plugins/ASCII/Makefile.in	Wed May 22 23:08:06 2002
@@ -143,7 +143,7 @@
 CPPFLAGS = @CPPFLAGS@
 LDFLAGS = @LDFLAGS@
 LIBS = @LIBS@
-libasciitable_la_LDFLAGS = 
+libasciitable_la_LDFLAGS = -module -avoid-version
 libasciitable_la_LIBADD = 
 libasciitable_la_OBJECTS =  ascii.lo
 CFLAGS = @CFLAGS@
diff -Naur gedit-1.117.0.old/plugins/cvschangelog/Makefile.in gedit-1.117.0.new/plugins/cvschangelog/Makefile.in
--- gedit-1.117.0.old/plugins/cvschangelog/Makefile.in	Tue Apr 16 12:45:17 2002
+++ gedit-1.117.0.new/plugins/cvschangelog/Makefile.in	Wed May 22 23:08:06 2002
@@ -138,7 +138,7 @@
 CPPFLAGS = @CPPFLAGS@
 LDFLAGS = @LDFLAGS@
 LIBS = @LIBS@
-libcvschangelog_la_LDFLAGS = 
+libcvschangelog_la_LDFLAGS = -module -avoid-version
 libcvschangelog_la_LIBADD = 
 libcvschangelog_la_OBJECTS =  cvschangelog.lo
 CFLAGS = @CFLAGS@
diff -Naur gedit-1.117.0.old/plugins/diff/Makefile.in gedit-1.117.0.new/plugins/diff/Makefile.in
--- gedit-1.117.0.old/plugins/diff/Makefile.in	Tue Apr 16 12:45:16 2002
+++ gedit-1.117.0.new/plugins/diff/Makefile.in	Wed May 22 23:08:06 2002
@@ -143,7 +143,7 @@
 CPPFLAGS = @CPPFLAGS@
 LDFLAGS = @LDFLAGS@
 LIBS = @LIBS@
-libdiff_la_LDFLAGS = 
+libdiff_la_LDFLAGS = -module -avoid-version
 libdiff_la_LIBADD = 
 libdiff_la_OBJECTS =  diff.lo
 CFLAGS = @CFLAGS@
diff -Naur gedit-1.117.0.old/plugins/docinfo/Makefile.in gedit-1.117.0.new/plugins/docinfo/Makefile.in
--- gedit-1.117.0.old/plugins/docinfo/Makefile.in	Tue Apr 16 12:45:16 2002
+++ gedit-1.117.0.new/plugins/docinfo/Makefile.in	Wed May 22 23:08:06 2002
@@ -143,7 +143,7 @@
 CPPFLAGS = @CPPFLAGS@
 LDFLAGS = @LDFLAGS@
 LIBS = @LIBS@
-libdocinfo_la_LDFLAGS = 
+libdocinfo_la_LDFLAGS = -module -avoid-version
 libdocinfo_la_LIBADD = 
 libdocinfo_la_OBJECTS =  docinfo.lo
 CFLAGS = @CFLAGS@
diff -Naur gedit-1.117.0.old/plugins/sample/Makefile.in gedit-1.117.0.new/plugins/sample/Makefile.in
--- gedit-1.117.0.old/plugins/sample/Makefile.in	Tue Apr 16 12:45:17 2002
+++ gedit-1.117.0.new/plugins/sample/Makefile.in	Wed May 22 23:08:06 2002
@@ -138,7 +138,7 @@
 CPPFLAGS = @CPPFLAGS@
 LDFLAGS = @LDFLAGS@
 LIBS = @LIBS@
-libsample_la_LDFLAGS = 
+libsample_la_LDFLAGS = -module -avoid-version
 libsample_la_LIBADD = 
 libsample_la_OBJECTS =  sample.lo
 CFLAGS = @CFLAGS@
diff -Naur gedit-1.117.0.old/plugins/shell_output/Makefile.in gedit-1.117.0.new/plugins/shell_output/Makefile.in
--- gedit-1.117.0.old/plugins/shell_output/Makefile.in	Tue Apr 16 12:45:18 2002
+++ gedit-1.117.0.new/plugins/shell_output/Makefile.in	Wed May 22 23:08:06 2002
@@ -143,7 +143,7 @@
 CPPFLAGS = @CPPFLAGS@
 LDFLAGS = @LDFLAGS@
 LIBS = @LIBS@
-libshell_output_la_LDFLAGS = 
+libshell_output_la_LDFLAGS = -module -avoid-version
 libshell_output_la_LIBADD = 
 libshell_output_la_OBJECTS =  shell_output.lo
 CFLAGS = @CFLAGS@
diff -Naur gedit-1.117.0.old/plugins/taglist/Makefile.in gedit-1.117.0.new/plugins/taglist/Makefile.in
--- gedit-1.117.0.old/plugins/taglist/Makefile.in	Tue Apr 16 12:45:17 2002
+++ gedit-1.117.0.new/plugins/taglist/Makefile.in	Wed May 22 23:08:06 2002
@@ -145,7 +145,7 @@
 CPPFLAGS = @CPPFLAGS@
 LDFLAGS = @LDFLAGS@
 LIBS = @LIBS@
-libtaglist_la_LDFLAGS = 
+libtaglist_la_LDFLAGS = -module -avoid-version
 libtaglist_la_LIBADD = 
 libtaglist_la_OBJECTS =  gedit-taglist-plugin-parser.lo \
 gedit-taglist-plugin-window.lo gedit-taglist-plugin.lo
diff -Naur gedit-1.117.0.old/plugins/time/Makefile.in gedit-1.117.0.new/plugins/time/Makefile.in
--- gedit-1.117.0.old/plugins/time/Makefile.in	Tue Apr 16 12:45:17 2002
+++ gedit-1.117.0.new/plugins/time/Makefile.in	Wed May 22 23:08:06 2002
@@ -143,7 +143,7 @@
 CPPFLAGS = @CPPFLAGS@
 LDFLAGS = @LDFLAGS@
 LIBS = @LIBS@
-libtime_la_LDFLAGS = 
+libtime_la_LDFLAGS = -module -avoid-version
 libtime_la_LIBADD = 
 libtime_la_OBJECTS =  time.lo
 CFLAGS = @CFLAGS@
diff -Naur gedit-1.117.0.old/src/gedit-plugins-engine.c gedit-1.117.0.new/src/gedit-plugins-engine.c
--- gedit-1.117.0.old/src/gedit-plugins-engine.c	Tue Mar 19 00:21:49 2002
+++ gedit-1.117.0.new/src/gedit-plugins-engine.c	Wed May 22 22:56:05 2002
@@ -31,6 +31,7 @@
 #include <config.h>
 #endif
 
+#include <sys/types.h> 
 #include <dirent.h> 
 #include <string.h>
 
