diff -Naur gnome-desktop-2.0.2.old/libgnome-desktop/gnome-theme-parser.c gnome-desktop-2.0.2.new/libgnome-desktop/gnome-theme-parser.c
--- gnome-desktop-2.0.2.old/libgnome-desktop/gnome-theme-parser.c	Fri May 17 03:12:46 2002
+++ gnome-desktop-2.0.2.new/libgnome-desktop/gnome-theme-parser.c	Fri Jun 28 19:39:44 2002
@@ -763,7 +763,11 @@
 {
   char *p, *lang;
 
+#ifdef LC_MESSAGES
   lang = g_strdup (setlocale (LC_MESSAGES, NULL));
+#else
+  lang = g_strdup (setlocale (LC_ALL, NULL));
+#endif
   
   if (lang)
     {
