diff -Naur libgtkhtml-1.99.5.old/ltmain.sh libgtkhtml-1.99.5.new/ltmain.sh
--- libgtkhtml-1.99.5.old/ltmain.sh	Tue Mar 19 05:01:27 2002
+++ libgtkhtml-1.99.5.new/ltmain.sh	Wed May 22 02:23:52 2002
@@ -2894,7 +2894,18 @@
 	if test -n "$export_symbols" && test -n "$archive_expsym_cmds"; then
 	  eval cmds=\"$archive_expsym_cmds\"
 	else
+	  save_deplibs="$deplibs"
+	  for conv in $convenience; do
+	    tmp_deplibs=
+	    for test_deplib in $deplibs; do
+	      if test "$test_deplib" != "$conv"; then
+	        tmp_deplibs="$tmp_deplibs $test_deplib"
+	      fi
+	    done
+	    deplibs="$tmp_deplibs"
+	  done
 	  eval cmds=\"$archive_cmds\"
+	  deplibs="$save_deplibs"
 	fi
 	save_ifs="$IFS"; IFS='~'
 	for cmd in $cmds; do
