diff -uNbr opennms/README.Darwin opennms-new/README.Darwin
--- opennms/README.Darwin	Wed Dec 31 19:00:00 1969
+++ opennms-new/README.Darwin	Sun Feb  2 13:01:39 2003
@@ -0,0 +1,37 @@
+Post-Installation Tasks
+------------------------------------------------------------------------------
+On MacOSX, to be able to give PostgreSQL enough buffers and such to run
+properly, you will need to change the shared memory buffers in the startup
+item:
+
+  /System/Library/StartupItems/SystemTuning/SystemTuning
+
+On our test system we multiplied the supplied values by 4 and all seems to
+run OK.  Just look for the sysctl lines near the bottom and set them to:
+
+  sysctl -w kern.sysv.shmmax=16777216
+  sysctl -w kern.sysv.shmmin=1
+  sysctl -w kern.sysv.shmmni=128
+  sysctl -w kern.sysv.shmseg=32
+  sysctl -w kern.sysv.shmall=4096
+
+Once you do that, you will need to edit
+@PREFIX@/var/postgresql/data/postgresql.conf (as root, or pgsql), and
+make sure that the following values are set:
+
+  tcpip_socket    = true
+  max_connections = 256
+  shared_buffers  = 512
+
+Then, you can run @PREFIX@/var/opennms/bin/install.pl to configure the
+database, like so:
+
+  @PREFIX@/var/opennms/bin/install.pl -q -U pgsql @PREFIX@/var/opennms/etc/create.sql
+
+Now you're ready to start everything up!  Check the configuration files
+in @PREFIX@/var/opennms/etc and set them up for your network (as per the
+quick start guide at http://www.opennms.org/users/docs/) and run:
+
+  sudo -u tomcat /sw/bin/tomcat start
+  sudo @PREFIX@/var/opennms/bin/opennms.sh start
+
diff -uNbr opennms/build.properties opennms-new/build.properties
--- opennms/build.properties	Fri Nov 29 10:47:10 2002
+++ opennms-new/build.properties	Sun Feb  2 12:43:26 2003
@@ -5,25 +5,25 @@
 # passed to ant when the build is run.
 
 # root.install: the top directory to install OpenNMS to
-root.install		= ${root.source}/dist
+root.install		= @PREFIX@/var/opennms
 # root.install.initdir: the location of the init.d directory
-root.install.initdir	= ${root.source}/dist/bin
+root.install.initdir	= @PREFIX@/bin
 # install.servlets: the location of the servlet root
-root.install.servlets	= ${root.source}/dist/webapps/opennms
+root.install.servlets	= @PREFIX@/var/tomcat4/webapps/opennms
 # root.install.share: the location of shared data
-root.install.share	= ${root.source}/dist/share
+root.install.share	= @PREFIX@/var/opennms/data
 # root.install.rrdtool: the location of rrdtool libraries
-root.install.rrdtool	= ${root.source}/dist/lib
+root.install.rrdtool	= @PREFIX@/var/opennms/lib
 # root.install.postgresql: the location of postgresql libraries
-root.install.postgresql	= ${root.source}/dist/lib
+root.install.postgresql	= @PREFIX@/var/opennms/lib
 # root.install.contrib:	the location to put the contributed source
-root.install.contrib	= ${root.source}/dist/contrib
+root.install.contrib	= @PREFIX@/var/opennms/contrib
 # root.install.logs: the directory to log to
-root.install.logs	= ${root.source}/dist/logs
+root.install.logs	= @PREFIX@/var/log/opennms
 # root.install.pid: the name of the pid file to use
-root.install.pid	= ${root.source}/dist/logs/opennms.pid
+root.install.pid	= @PREFIX@/var/log/opennms/opennms.pid
 # root.installprefix: a prefix used for installation in or chroot
-root.installprefix	= 
+root.installprefix	= @DESTDIR@
 # root.build: the top working directory for compiling/etc.
 root.build		= ${root.source}/work
 # root.build.classes:	where OpenNMS classes go when building
@@ -79,8 +79,8 @@
 rpm.buildtests		= false
 
 install.database.driver   = org.postgresql.Driver
-install.database.url      = jdbc:postgresql://localhost:5432/opennms?compatible=7.1
+install.database.url      = jdbc:postgresql://localhost:5432/opennms
 install.database.name     = opennms
 install.database.user     = opennms
 install.database.password = opennms
-install.rrdtool.bin       = /usr/bin/rrdtool
+install.rrdtool.bin       = @PREFIX@/bin/rrdtool
diff -uNbr opennms/tools/infrastructure/compiler_setup.sh opennms-new/tools/infrastructure/compiler_setup.sh
--- opennms/tools/infrastructure/compiler_setup.sh	Thu Jan  2 14:03:22 2003
+++ opennms-new/tools/infrastructure/compiler_setup.sh	Sun Feb  2 12:43:54 2003
@@ -26,14 +26,14 @@
 	LD="/usr/bin/cc"
 fi
 
-for dir in /usr/local/pgsql/lib /usr/local/lib /usr/lib /sw/lib; do
+for dir in /usr/local/pgsql/lib /usr/local/lib /usr/lib @PREFIX@/lib; do
 	if [ -f "$dir/libpq.so" ] || [ -f "$dir/libpq.dylib" ]; then
 		PG_LIBDIR="$dir"
 		break
 	fi
 done
 
-for dir in /usr/local/pgsql/include /usr/local/include /usr/include /sw/include; do
+for dir in /usr/local/pgsql/include /usr/local/include /usr/include @PREFIX@/include; do
 	if [ -f "$dir/postgres.h" ] || [ -f "$dir/server/postgres.h" ]; then
 		PG_INCDIR="$dir"
 		break
@@ -46,7 +46,7 @@
 	fi
 done
 
-for dir in /usr/local/lib /usr/local/rrdtool-*/lib /usr/lib /sw/lib; do
+for dir in /usr/local/lib /usr/local/rrdtool-*/lib /usr/lib @PREFIX@/lib; do
 	if [ -f "$dir/librrd.so" ]; then
 		RRD_LIBDIR="$dir"
 		break
@@ -57,7 +57,7 @@
 	fi
 done
 
-for dir in /usr/local/include /usr/local/rrdtool-*/include /usr/include /sw/include; do
+for dir in /usr/local/include /usr/local/rrdtool-*/include /usr/include @PREFIX@/include; do
 	if [ -f "$dir/rrd.h" ]; then
 		RRD_INCDIR="$dir"
 		break
diff -uNbr opennms/tools/infrastructure/platform_darwin.sh opennms-new/tools/infrastructure/platform_darwin.sh
--- opennms/tools/infrastructure/platform_darwin.sh	Thu Jan  2 14:03:24 2003
+++ opennms-new/tools/infrastructure/platform_darwin.sh	Sun Feb  2 12:44:11 2003
@@ -4,8 +4,8 @@
 PACKAGES="$PACKAGES PLATFORM_DARWIN"
 
 if [ `uname` = 'Darwin' ]; then
-	PATH="/usr/local/bin:/sw/bin:$PATH" export PATH
-	DYLD_LIBRARY_PATH="/sw/lib:/usr/local/rrdtool-1.0.33/lib:$DYLD_LIBRARY_PATH" export DYLD_LIBRARY_PATH
+	PATH="/usr/local/bin:@PREFIX@/bin:$PATH" export PATH
+	DYLD_LIBRARY_PATH="@PREFIX@/lib:/usr/local/rrdtool-1.0.33/lib:$DYLD_LIBRARY_PATH" export DYLD_LIBRARY_PATH
 	PLATFORM=darwin export PLATFORM
 	# compile.javainc = the arch-specific path for headers in the JDK
 	DEFINES="$DEFINES -Dcompile.otherlibs=/usr/lib/libc.dylib -Dcompile.platform=darwin -Dcompile.soext=so"
Binary files opennms/tools/run/.install.pl.swp and opennms-new/tools/run/.install.pl.swp differ
diff -uNbr opennms/tools/run/CVS/Entries opennms-new/tools/run/CVS/Entries
--- opennms/tools/run/CVS/Entries	Sun Feb  2 14:04:31 2003
+++ opennms-new/tools/run/CVS/Entries	Sun Feb  2 12:01:27 2003
@@ -7,5 +7,5 @@
 /parse-status.pl/1.1.1.1/Mon May  6 15:12:28 2002//
 /report.sh/1.1.1.1/Mon May  6 15:12:28 2002//
 /webreport.pl/1.2/Thu Oct 24 16:03:10 2002//
-/install.pl/1.9/Sun Feb  2 19:04:19 2003//
+/install.pl/1.7/Sun Feb  2 16:29:07 2003//
 D
diff -uNbr opennms/tools/run/install.pl opennms-new/tools/run/install.pl
--- opennms/tools/run/install.pl	Sun Feb  2 14:04:19 2003
+++ opennms-new/tools/run/install.pl	Sun Feb  2 14:04:42 2003
@@ -630,7 +630,7 @@
 
 if ($TOMCAT) {
 
-	for my $dir ('/var/tomcat4', '/sw/var/tomcat4', '/usr/local/tomcat4') {
+	for my $dir ('/var/tomcat4', '@PREFIX@/var/tomcat4', '/usr/local/tomcat4') {
 
 		my $serverxml = $dir . '/server.xml';
 	
@@ -826,7 +826,7 @@
 	my $plpgsql_failed = 0;
 	my $plpgsql_sofile;
 	print "- adding PL/pgSQL call handler... ";
-	for my $dir ('/usr/lib', '/usr/local/lib', '/sw/lib/postgresql') {
+	for my $dir ('/usr/lib', '/usr/local/lib', '@PREFIX@/lib/postgresql') {
 		for ($SOEXT, 'so') {
 			if (-d "$dir") {
 				if (-f "$dir/pgsql/plpgsql.$_") {
