Index: CHANGELOG
===================================================================
RCS file: /cvsroot/opennms/opennms/CHANGELOG,v
retrieving revision 1.45
retrieving revision 1.46
diff -u -r1.45 -r1.46
--- CHANGELOG	7 Feb 2003 04:25:09 -0000	1.45
+++ CHANGELOG	14 Feb 2003 13:30:30 -0000	1.46
@@ -1,3 +1,7 @@
+1.1.1 to 1.1.2
+----------------------------------------------------------------------
+- Fixed a small bug with RRAs. Bug 707. Thx Volker
+
 1.1.0 to 1.1.1
 ----------------------------------------------------------------------
 - Slightly improved the output of mib2opennms. Thx John and Rick
Index: build.properties
===================================================================
RCS file: /cvsroot/opennms/opennms/build.properties,v
retrieving revision 1.7
retrieving revision 1.8
diff -u -r1.7 -r1.8
--- build.properties	6 Feb 2003 01:55:49 -0000	1.7
+++ build.properties	14 Feb 2003 13:30:30 -0000	1.8
@@ -35,7 +35,7 @@
 root.release		= ${root.source}/work/releases
 
 # opennms.version:	the version number of this install
-opennms.version		= 1.1.1
+opennms.version		= 1.1.2
 # opennms.snapshot:	whether or not this should be a "snapshot" release
 # please do not check this in as false!  set false in
 # ~/.opennms-global.properties or on the command-line, as in,
Index: etc/log4j.properties
===================================================================
RCS file: /cvsroot/opennms/opennms/etc/log4j.properties,v
retrieving revision 1.1.1.1
diff -u -r1.1.1.1 log4j.properties
--- etc/log4j.properties	6 May 2002 15:10:45 -0000	1.1.1.1
+++ etc/log4j.properties	15 Feb 2003 16:25:35 -0000
@@ -33,7 +33,7 @@
 log4j.appender.UNCATEGORIZED.MaxBackupIndex=4
 log4j.appender.UNCATEGORIZED.File=@root.install.logs@/uncategorized.log
 log4j.appender.UNCATEGORIZED.layout=org.apache.log4j.PatternLayout
-log4j.appender.UNCATEGORIZED.layout.ConversionPattern=%d %-5p [%t] %c{1}: %m%n
+log4j.appender.UNCATEGORIZED.layout.ConversionPattern=%d %-5p [%t] %c: %m%n
 
 # Eventd
 log4j.category.OpenNMS.Eventd=DEBUG, EVENTD
@@ -235,3 +235,15 @@
 log4j.appender.JMX.File=@root.install.logs@/jmx.log
 log4j.appender.JMX.layout=org.apache.log4j.PatternLayout
 log4j.appender.JMX.layout.ConversionPattern=%d %-5p [%t] %c{1}: %m%n
+
+# Tomcat internal logs (xml readers, etc)
+log4j.category.org.apache.commons=WARN, TOMCAT-INTERNAL 
+log4j.additivity.org.apache.commons=false
+log4j.category.org.apache.coyote=WARN, TOMCAT-INTERNAL
+log4j.additivity.org.apache.coyote=false
+log4j.appender.TOMCAT-INTERNAL=org.apache.log4j.RollingFileAppender
+log4j.appender.TOMCAT-INTERNAL.MaxFileSize=100MB
+log4j.appender.TOMCAT-INTERNAL.MaxBackupIndex=2
+log4j.appender.TOMCAT-INTERNAL.File=@root.install.logs@/tomcat-internal.log
+log4j.appender.TOMCAT-INTERNAL.layout=org.apache.log4j.PatternLayout
+log4j.appender.TOMCAT-INTERNAL.layout.ConversionPattern=%d %-5p [%t] %c: %m%n
Index: src/services/org/opennms/netmgt/rrd/rrd_jinterface.c
===================================================================
RCS file: /cvsroot/opennms/opennms/src/services/org/opennms/netmgt/rrd/rrd_jinterface.c,v
retrieving revision 1.3
retrieving revision 1.4
diff -u -r1.3 -r1.4
--- src/services/org/opennms/netmgt/rrd/rrd_jinterface.c	6 Feb 2003 03:40:59 -0000	1.3
+++ src/services/org/opennms/netmgt/rrd/rrd_jinterface.c	14 Feb 2003 13:30:31 -0000	1.4
@@ -192,6 +192,8 @@
 #ifdef DEBUG
 			printf("hit null, creating token.\n");
 #endif
+			// terminate buffer
+			buffer[x] = '\0';
 			
 			/// allocate storage for new token string
 			token = (char *)malloc(strlen(buffer)+1);
Index: src/services/org/opennms/protocols/icmp/IcmpSocket.c
===================================================================
RCS file: /cvsroot/opennms/opennms/src/services/org/opennms/protocols/icmp/IcmpSocket.c,v
retrieving revision 1.6
diff -u -r1.6 IcmpSocket.c
--- src/services/org/opennms/protocols/icmp/IcmpSocket.c	9 Feb 2003 19:46:00 -0000	1.6
+++ src/services/org/opennms/protocols/icmp/IcmpSocket.c	15 Feb 2003 16:25:36 -0000
@@ -162,22 +162,22 @@
  * Macros for doing byte swapping
  */
 
-#if defined(__LITTLE_ENDIAN) || defined(_LITTLE_ENDIAN) || defined(LITTLE_ENDIAN)
+#if defined(__LITTLE_ENDIAN) || defined(_LITTLE_ENDIAN) || defined(__LITTLE_ENDIAN__)
 # ifndef ntohll
 #  if defined(__DARWIN__)
-#   define ntohll(_x_) NXSwapLongLong(_x_)
+#   define ntohll(_x_) NXSwapBigLongLongToHost(_x_)
 #  else
 #   define ntohll(_x_) __bswap_64(_x_)
 #  endif
 # endif
 # ifndef htonll
 #  if defined(__DARWIN__)
-#   define htonll(_x_) NXSwapLongLong(_x_)
+#   define htonll(_x_) NXSwapHostLongLongToBig(_x_)
 #  else
 #   define htonll(_x_) __bswap_64(_x_)
 #  endif
 # endif
-#elif  defined(__BIG_ENDIAN) || defined(_BIG_ENDIAN) || defined(BIG_ENDIAN)
+#elif  defined(__BIG_ENDIAN) || defined(_BIG_ENDIAN) || defined(__BIG_ENDIAN__)
 # ifndef ntohll
 #  define ntohll(_x_) _x_
 # endif
Index: tools/packages/fink/README.Darwin
===================================================================
RCS file: /cvsroot/opennms/opennms/tools/packages/fink/README.Darwin,v
retrieving revision 1.1
retrieving revision 1.2
diff -u -r1.1 -r1.2
--- tools/packages/fink/README.Darwin	7 Feb 2003 16:57:02 -0000	1.1
+++ tools/packages/fink/README.Darwin	11 Feb 2003 00:12:43 -0000	1.2
@@ -9,6 +9,7 @@
 On our test system we multiplied the supplied values by 4 and all seems to
 run OK.  Just look for the sysctl lines near the bottom and set them to:
 
+  sysctl -w kern.maxfiles=65536
   sysctl -w kern.sysv.shmmax=16777216
   sysctl -w kern.sysv.shmmin=1
   sysctl -w kern.sysv.shmmni=128
@@ -20,8 +21,8 @@
 make sure that the following values are set:
 
   tcpip_socket    = true
-  max_connections = 256
-  shared_buffers  = 512
+  max_connections = 192
+  shared_buffers  = 384
 
 Then, you can run @PREFIX@/var/opennms/bin/install.pl to configure the
 database, like so:
Index: tools/run/install.pl
===================================================================
RCS file: /cvsroot/opennms/opennms/tools/run/install.pl,v
retrieving revision 1.11
diff -u -r1.11 install.pl
--- tools/run/install.pl	7 Feb 2003 04:08:23 -0000	1.11
+++ tools/run/install.pl	15 Feb 2003 16:25:36 -0000
@@ -18,10 +18,12 @@
 	$PG_USER
 	$PG_PASS
 	$PG_LIBDIR
+	$PG_VERSION
 
 	$USER
 	$PASS
 
+	$CASCADE
 	$DATABASE
 	$SQL_FILE
 
@@ -55,6 +57,7 @@
 
 $OPENNMS_HOME	= '@root.install@';
 $SOEXT		= '@compile.soext@';
+$TOMCAT		= 1;
 $VERBOSE	= 1;
 $LOG		= 0;
 $LOG_FILE	= '/tmp/opennms-install.log';
@@ -115,7 +118,7 @@
 			$NOINSERT = 1 if ($arg eq "i");
 			$RPM			= 1 if ($arg eq "r");
 			$NOSO		 = 1 if ($arg eq "s");
-			$TOMCAT	 = 1 if ($arg eq "t");
+			$TOMCAT	 = 0 if ($arg eq "t");
 			$DEBUG		= 1 if ($arg eq "x");
 			$DODROPS	= 1 if ($arg eq "z");
 
@@ -255,6 +258,17 @@
 
 END
 
+	my $verquery = $template->prepare('select version()') or die "couldn't prepare version from the database: " . $template->errstr;
+	$verquery->execute() or die "couldn't execute version from database: " . $verquery->errstr;
+	if ($verquery->rows) {
+		my ($version) = $verquery->fetchrow_array();
+		($PG_VERSION) = $version =~ /PostgreSQL (\d+\.\d+)/;
+	}
+
+	if ($PG_VERSION >= 7.3) {
+		$CASCADE = " $CASCADE";
+	}
+
 	# dbi will return an OK even if the user exists, so we
 	# always do a create, and let it silently fail if they do
 
@@ -412,7 +426,7 @@
 
 		eval {
 			stderr_off();
-			$database->do("DROP TABLE $table") and print "DROPPED ";
+			$database->do("DROP TABLE $table $CASCADE") and print "DROPPED ";
 			stderr_on();
 		};
 		undef $@;
@@ -1611,7 +1625,7 @@
 	my $revert_table = sub {
 		my $errormessage = shift;
 		stderr_on();
-		$database->do("DROP TABLE $table");
+		$database->do("DROP TABLE $table $CASCADE");
 		$database->do("CREATE TABLE $table AS SELECT " . join(', ', @db_names) . " FROM temp");
 		die "FAILED: $errormessage";
 	};
@@ -1643,14 +1657,14 @@
 	}
 
 	# This line throws a harmless error if the table doesn't exist (normally it shouldn't between upgrades)
-	$database->do('DROP TABLE temp');
+	$database->do("DROP TABLE temp $CASCADE");
 
 	print "  - creating temporary table... ";
 	$text = "CREATE TABLE temp AS SELECT " . join(', ', @db_names) . " FROM $table";
 	$database->do($text) or die "FAILED: unable to create temporary table $table (as 'temp'): " . $database->errstr;
 	print "done\n";
 
-	$database->do("DROP TABLE $table");
+	$database->do("DROP TABLE $table $CASCADE");
 
 	print "  - creating new '$table' table... ";
 	$text = "CREATE TABLE $table (" . get_table_from_sql($table, 1) . ")";
@@ -1741,7 +1755,7 @@
 	print "  - transforming data into the new table... done     \n";
 
 	print "  - dropping temporary table... ";
-	$text = "DROP TABLE temp";
+	$text = "DROP TABLE temp $CASCADE";
 	$database->do($text) or die "FAILED: " . $database->errstr;
 	print "done\n";
 
@@ -1801,7 +1815,7 @@
 	}
 
 	$database->do("CREATE TABLE temp AS SELECT $columns FROM $table") or die "unable to create temporary table: " . $database->errstr;
-	$database->do("DROP TABLE $table") or die "unable to drop old table: " . $database->errstr;
+	$database->do("DROP TABLE $table $CASCADE") or die "unable to drop old table: " . $database->errstr;
 	my $new     = get_table_from_sql($table);
 	my $newtext;
 	for my $col (@{$tabledata}) {
@@ -1812,7 +1826,7 @@
 	$database->do("CREATE TABLE $table ($newtext)") or die "unable to create new table: " . $database->errstr;
 	$database->do("GRANT ALL ON $table TO $USER") or die(scalar $database->errstr);
 	$database->do("INSERT INTO $table ($columns) SELECT $columns FROM temp") or die "unable to fill new table: " . $database->errstr;
-	$database->do("DROP TABLE temp") or die "unable to drop temporary table: " . $database->errstr;
+	$database->do("DROP TABLE temp $CASCADE") or die "unable to drop temporary table: " . $database->errstr;
 
 	return 1;
 }
