Index: src/services/org/opennms/protocols/icmp/IcmpSocket.c
===================================================================
RCS file: /cvsroot/opennms/opennms/src/services/org/opennms/protocols/icmp/IcmpSocket.c,v
retrieving revision 1.5
diff -u -r1.5 IcmpSocket.c
--- old/src/services/org/opennms/protocols/icmp/IcmpSocket.c	2 Feb 2003 17:07:06 -0000	1.5
+++ new/src/services/org/opennms/protocols/icmp/IcmpSocket.c	8 Feb 2003 04:10:37 -0000
@@ -32,6 +32,7 @@
 #include <sys/socket.h>
 #if defined(__DARWIN__) || defined(__SOLARIS__)
 #include <netinet/in_systm.h>
+#include <architecture/byte_order.h>
 #endif
 #include <netinet/in.h>
 #include <netinet/ip.h>
@@ -163,14 +164,14 @@
 #if defined(__LITTLE_ENDIAN) || defined(_LITTLE_ENDIAN) || defined(LITTLE_ENDIAN)
 # ifndef ntohll
 #  if defined(__DARWIN__)
-#   define ntohll(_x_) ntohl(_x_)
+#   define ntohll(_x_) NXSwapBigLongLongToHost(_x_)
 #  else
 #   define ntohll(_x_) __bswap_64(_x_)
 #  endif
 # endif
 # ifndef htonll
 #  if defined(__DARWIN__)
-#   define htonll(_x_) htonl(_x_)
+#   define htonll(_x_) NXSwapHostLongLongToBig(_x_)
 #  else
 #   define htonll(_x_) __bswap_64(_x_)
 #  endif
