diff -Naur desktop-file-utils-0.8.old/src/menu-method.c desktop-file-utils-0.8.new/src/menu-method.c
--- desktop-file-utils-0.8.old/src/menu-method.c	Sun Aug 29 23:52:37 2004
+++ desktop-file-utils-0.8.new/src/menu-method.c	Sun Sep 19 23:23:21 2004
@@ -1688,7 +1688,7 @@
 		    GnomeVFSOpenMode   mode,
 		    FileHandle       **handle,
 		    gboolean           exclusive,
-		    mode_t             perms)
+		    unsigned int       perms)
 {
 	int unix_flags;
 
