--- flac-1.0.2/ltmain.sh	Mon Dec  3 23:10:14 2001
+++ flac-1.0.2-patched/ltmain.sh	Tue May 21 05:44:53 2002
@@ -1884,7 +1884,11 @@
 	      add="-l$name"
 	    else
 	      # We cannot seem to hardcode it, guess we'll fake it.
-	      add_dir="-L$libdir"
+ 	      if test "X$installed" = Xyes; then
+ 	        add_dir="-L$libdir"
+ 	      else
+ 	        add_dir="-L$DESTDIR$libdir"
+ 	      fi
 	      add="-l$name"
 	    fi
 
@@ -4141,12 +4145,21 @@
 	esac
 
 	# Add the libdir to current_libdirs if it is the destination.
+	DESTDIR=
 	if test "X$destdir" = "X$libdir"; then
 	  case "$current_libdirs " in
 	  *" $libdir "*) ;;
 	  *) current_libdirs="$current_libdirs $libdir" ;;
 	  esac
 	else
+ 	  case "$destdir" in
+ 	    *"$libdir")
+ 	      DESTDIR=`$echo "$destdir" | sed -e 's!'"$libdir"'$!!'`
+ 	      if test "X$destdir" != "X$DESTDIR$libdir"; then
+ 		DESTDIR=
+ 	      fi
+ 	      ;;
+ 	  esac
 	  # Note the libdir as a future libdir.
 	  case "$future_libdirs " in
 	  *" $libdir "*) ;;
@@ -4160,6 +4173,7 @@
 
 	if test -n "$relink_command"; then
 	  $echo "$modename: warning: relinking \`$file'" 1>&2
+ 	  export DESTDIR
 	  $show "$relink_command"
 	  if $run eval "$relink_command"; then :
 	  else
@@ -4167,6 +4181,7 @@
 	    continue
 	  fi
 	fi
+	unset DESTDIR
 
 	# See the names of the shared library.
 	set dummy $library_names
--- flac-1.0.2/configure	Mon Dec  3 23:10:10 2001
+++ flac-1.0.2-patched/configure	Tue May 21 05:54:08 2002
@@ -4555,7 +4555,7 @@
     # FIXME: Relying on posixy $() will cause problems for
     #        cross-compilation, but unfortunately the echo tests do not
     #        yet detect zsh echo's removal of \ escapes.
-    archive_cmds='$nonopt $(test "x$module" = xyes && echo -bundle || echo -dynamiclib) $allow_undefined_flag -o $lib $libobjs $deplibs$linker_flags -install_name $rpath/$soname $verstring'
+    archive_cmds='$nonopt $(test x$module = xyes && echo -bundle || echo -dynamiclib) $allow_undefined_flag -o $lib $libobjs $deplibs$linker_flags $(test .$module != .yes && echo -install_name $rpath/$soname $tmp_verstring)'
     # We need to add '_' to the symbols in $export_symbols first
     #archive_expsym_cmds="$archive_cmds"' && strip -s $export_symbols'
     hardcode_direct=yes

