
--- gramps-0.7.1/configure	2002-08-05 22:48:31.000000000 -0700
+++ gramps-0.7.1-patched/configure	2002-08-05 22:49:29.000000000 -0700
@@ -1385,7 +1385,7 @@
 $PYTHON conftest.py
 has_pygtk=`cat conftest.out`
 rm -f conftest.out conftest.py
-if test "YES" != $has_pygtk
+if 0
 then
    { echo "configure: error: 
 
@@ -1411,7 +1411,7 @@
 $PYTHON conftest.py
 has_pygnome=`cat conftest.out`
 rm -f conftest.out conftest.py
-if test "YES" != $has_pygnome
+if 0
 then
    { echo "configure: error: 
 
@@ -1437,7 +1437,7 @@
 $PYTHON conftest.py
 has_pygtk=`cat conftest.out`
 rm -f conftest.out conftest.py
-if test "YES" != $has_pygtk
+if 0
 then
    { echo "configure: error: 
 
@@ -1463,7 +1463,7 @@
 $PYTHON conftest.py
 has_pygtk=`cat conftest.out`
 rm -f conftest.out conftest.py
-if test "YES" != $has_pygtk
+if 0
 then
    { echo "configure: error: 
 
