--- popularity-contest-1.3/debian/cron.weekly	2002-02-21 06:10:58.000000000 -0800
+++ popularity-contest-1.3-patched/debian/cron.weekly	2002-11-02 15:06:33.000000000 -0800
@@ -1,7 +1,7 @@
 #!/bin/sh
 
 # don't run if this package is removed but not purged
-if [ ! -f /usr/sbin/popularity-contest ]; then
+if [ ! -f @PREFIX@/sbin/popularity-contest ]; then
 	exit 0
 fi
 
@@ -11,7 +11,7 @@
 unset PARTICIPATE
 
 # get configuration information
-. /etc/popularity-contest.conf
+. @PREFIX@/etc/popularity-contest.conf
 
 # don't run if MAILTO address is blank!
 if [ -z "$MAILTO" ]; then exit 0; fi
@@ -20,7 +20,7 @@
 [ "$PARTICIPATE" = "no" -o -z "$PARTICIPATE" ] && exit 0
 
 # keep old logs
-cd /var/log
+cd @PREFIX@/var/log
 umask 022
 savelog -c 7 popularity-contest >/dev/null
 
@@ -33,7 +33,7 @@
 	echo "To: $MAILTO"
 	echo "Subject: popularity-contest submission"
 	echo
-	su nobody -pc "sh -c /usr/sbin/popularity-contest"
+	sh -c "@PREFIX@/sbin/popularity-contest"
 }
 
 do_sendmail()
@@ -48,5 +48,5 @@
 # generate and transmit the popularity contest data
 
 run_popcon \
-	| tee /var/log/popularity-contest \
+	| tee @PREFIX@/var/log/popularity-contest \
 	| do_sendmail
--- popularity-contest-1.3-patched/popularity-contest	2002-11-02 02:11:39.000000000 -0800
+++ popularity-contest-1.3/popularity-contest	2002-11-02 15:12:38.000000000 -0800
@@ -6,17 +6,17 @@
 #
 
 # the set of installed packages is the set of all .list files in
-# /var/lib/dpkg/info, with pathnames and .list suffix removed.
+# @PREFIX@/var/lib/dpkg/info, with pathnames and .list suffix removed.
 #
 # Store that list in $PACKAGES.
 #
-. /etc/popularity-contest.conf
+. @PREFIX@/etc/popularity-contest.conf
 if [ -z "$MY_HOSTID" ]; then
-	echo "You must set MY_HOSTID in /etc/popularity-contest.conf!" >&2
+	echo "You must set MY_HOSTID in @PREFIX@/etc/popularity-contest.conf!" >&2
 	exit 1
 fi
 
-DI=/var/lib/dpkg/info
+DI=@PREFIX@/var/lib/dpkg/info
 PACKAGES=$(dpkg-awk "Status: .* installed" -- Package \
 		| grep '^Package:' | sed 's/^Package: //')
 
