--- ToutDoux-1.2.7/configure.orig	2002-10-13 01:34:36.000000000 -0700
+++ ToutDoux-1.2.7/configure	2002-10-13 01:36:16.000000000 -0700
@@ -4606,12 +4606,12 @@
 	PGSQL_INCDIR=/usr/include/pgsql
 	PGSQL_LIBDIR=/usr/lib
 else
-	if test -f /usr/include/postgresql/libpq-fe.h; then
+	if test -f @PREFIX@/include/postgresql/libpq-fe.h; then
 		echo "$ac_t""yes" 1>&6
 		pgsql_msg=yes
 		have_pgsql=true
-		PGSQL_INCDIR=/usr/include/postgresql
-		PGSQL_LIBDIR=/usr/lib
+		PGSQL_INCDIR=@PREFIX@/include/postgresql
+		PGSQL_LIBDIR=@PREFIX@/lib
 	else
 		if test -f /usr/include/libpq-fe.h; then
 			echo "$ac_t""yes" 1>&6
--- ToutDoux-1.2.7/core-plugins/dialog_messages/Makefile.in.orig	2002-10-13 01:47:30.000000000 -0700
+++ ToutDoux-1.2.7/core-plugins/dialog_messages/Makefile.in	2002-10-13 01:49:42.000000000 -0700
@@ -188,13 +188,13 @@
 # Compilation
 # *****************************************************************************
 
-CFLAGS = -DHAVE_CONFIG_H -I$(top_srcdir) -I/. -I$(top_srcdir)/intl -I$(top_srcdir)/lib $(XML_CFLAGS) $(GNOME_CFLAGS)
+CFLAGS = -DHAVE_CONFIG_H -I$(top_srcdir) -I/. -I$(top_srcdir)/intl -I$(top_srcdir)/lib $(XML_CFLAGS) $(GNOME_CFLAGS) -I@PREFIX@/include
 
-LDFLAGS = $(INTLLIBS) $(GNOME_LIBDIR) $(GNOME_LIBS) $(top_srcdir)/lib/.libs/libtoutdoux.so
+LDFLAGS = $(INTLLIBS) $(GNOME_LIBDIR) $(GNOME_LIBS) $(top_srcdir)/lib/.libs/libtoutdoux.so -L@PREFIX@/lib
 
 COMPILE = $(CC) -g -O2 -Wall -Wno-unused -c -fPIC -DPIC -DHAVE_CONFIG_H -I$(top_srcdir) -I/.
 
-LINK = $(CC) -g -O2 -Wall -shared
+LINK = $(CC) -g -O2 -Wall -bundle -bundle_loader ../../app/toutdoux
 mkinstalldirs = $(SHELL) $(top_srcdir)/mkinstalldirs
 CONFIG_HEADER = ../../config.h
 CONFIG_CLEAN_FILES = 
