diff -Naur docbook-dtd-catalog.old/build_catalog docbook-dtd-catalog.new/build_catalog
--- docbook-dtd-catalog.old/build_catalog	Thu Jan  1 09:00:00 1970
+++ docbook-dtd-catalog.new/build_catalog	Fri Sep 27 01:15:11 2002
@@ -0,0 +1,99 @@
+#!/bin/sh
+
+catalog=docbook.cat
+xmlcatalog --noout --create $catalog
+
+for version in 4.0 4.1 4.1.1 4.1.2; do
+	dir=@PREFIX@/share/xml/dtd/docbookx/$version
+
+	xmlcatalog --noout --add "public" \
+		"-//OASIS//ELEMENTS DocBook XML Information Pool V$version//EN" \
+		"file://$dir/dbpoolx.mod" $catalog
+	xmlcatalog --noout --add "public" \
+		"-//OASIS//DTD DocBook XML V$version//EN" \
+		"file://$dir/docbookx.dtd" $catalog
+	xmlcatalog --noout --add "public" \
+		"-//OASIS//ENTITIES DocBook XML Character Entities V$version//EN" \
+		"file://$dir/dbcentx.mod" $catalog
+	xmlcatalog --noout --add "public" \
+		"-//OASIS//ENTITIES DocBook XML Notations V$version//EN" \
+		"file://$dir/dbnotnx.mod" $catalog
+	xmlcatalog --noout --add "public" \
+		"-//OASIS//ENTITIES DocBook XML Additional General Entities V$version//EN" \
+		"file://$dir/dbgenent.mod" $catalog
+	xmlcatalog --noout --add "public" \
+		"-//OASIS//ELEMENTS DocBook XML Document Hierarchy V$version//EN" \
+		"file://$dir/dbhierx.mod" $catalog
+	xmlcatalog --noout --add "public" \
+		"-//OASIS//DTD XML Exchange Table Model 19990315//EN" \
+		"file://$dir/soextblx.dtd" $catalog
+	xmlcatalog --noout --add "public" \
+		"-//OASIS//DTD DocBook XML CALS Table Model V$version//EN" \
+		"file://$dir/calstblx.dtd" $catalog
+	xmlcatalog --noout --add "rewriteSystem" \
+		"http://www.oasis-open.org/docbook/xml/$version" \
+		"file://$dir" $catalog
+	xmlcatalog --noout --add "rewriteURI" \
+		"http://www.oasis-open.org/docbook/xml/$version" \
+		"file://$dir" $catalog
+
+	xmlcatalog --noout --add "public" \
+		"ISO 8879:1986//ENTITIES Publishing//EN" \
+		"file://$isodir/iso-pub.ent" $catalog
+	xmlcatalog --noout --add "public" \
+		"ISO 8879:1986//ENTITIES Greek Letters//EN" \
+		"file://$isodir/iso-grk1.ent" $catalog
+	xmlcatalog --noout --add "public" \
+		"ISO 8879:1986//ENTITIES Box and Line Drawing//EN" \
+		"file://$isodir/iso-box.ent" $catalog
+	xmlcatalog --noout --add "public" \
+		"ISO 8879:1986//ENTITIES Greek Symbols//EN" \
+		"file://$isodir/iso-grk3.ent" $catalog
+	xmlcatalog --noout --add "public" \
+		"ISO 8879:1986//ENTITIES Added Math Symbols: Negated Relations//EN" \
+		"file://$isodir/iso-amsn.ent" $catalog
+	xmlcatalog --noout --add "public" \
+		"ISO 8879:1986//ENTITIES Numeric and Special Graphic//EN" \
+		"file://$isodir/iso-num.ent" $catalog
+	xmlcatalog --noout --add "public" \
+		"ISO 8879:1986//ENTITIES Alternative Greek Symbols//EN" \
+		"file://$isodir/iso-grk4.ent" $catalog
+	xmlcatalog --noout --add "public" \
+		"ISO 8879:1986//ENTITIES Diacritical Marks//EN" \
+		"file://$isodir/iso-dia.ent" $catalog
+	xmlcatalog --noout --add "public" \
+		"ISO 8879:1986//ENTITIES Monotoniko Greek//EN" \
+		"file://$isodir/iso-grk2.ent" $catalog
+	xmlcatalog --noout --add "public" \
+		"ISO 8879:1986//ENTITIES Added Math Symbols: Arrow Relations//EN" \
+		"file://$isodir/iso-amsa.ent" $catalog
+	xmlcatalog --noout --add "public" \
+		"ISO 8879:1986//ENTITIES Added Math Symbols: Ordinary//EN" \
+		"file://$isodir/iso-amso.ent" $catalog
+	xmlcatalog --noout --add "public" \
+		"ISO 8879:1986//ENTITIES Russian Cyrillic//EN" \
+		"file://$isodir/iso-cyr1.ent" $catalog
+	xmlcatalog --noout --add "public" \
+		"ISO 8879:1986//ENTITIES General Technical//EN" \
+		"file://$isodir/iso-tech.ent" $catalog
+	xmlcatalog --noout --add "public" \
+		"ISO 8879:1986//ENTITIES Added Math Symbols: Delimiters//EN" \
+		"file://$isodir/iso-amsc.ent" $catalog
+	xmlcatalog --noout --add "public" \
+		"ISO 8879:1986//ENTITIES Added Latin 1//EN" \
+		"file://$isodir/iso-lat1.ent" $catalog
+	xmlcatalog --noout --add "public" \
+		"ISO 8879:1986//ENTITIES Added Math Symbols: Binary Operators//EN" \
+		"file://$isodir/iso-amsb.ent" $catalog
+	xmlcatalog --noout --add "public" \
+		"ISO 8879:1986//ENTITIES Added Latin 2//EN" \
+		"file://$isodir/iso-lat2.ent" $catalog
+	xmlcatalog --noout --add "public" \
+		"ISO 8879:1986//ENTITIES Added Math Symbols: Relations//EN" \
+		"file://$isodir/iso-amsr.ent" $catalog
+	xmlcatalog --noout --add "public" \
+		"ISO 8879:1986//ENTITIES Non-Russian Cyrillic//EN" \
+		"file://$isodir/iso-cyr2.ent" $catalog
+done
+
+exit 0
