diff -Naur gnome-system-monitor-2.0.2.old/help/C/Makefile.in gnome-system-monitor-2.0.2.new/help/C/Makefile.in
--- gnome-system-monitor-2.0.2.old/help/C/Makefile.in	Tue Aug 27 08:55:54 2002
+++ gnome-system-monitor-2.0.2.new/help/C/Makefile.in	Fri Sep 27 04:23:28 2002
@@ -197,7 +197,7 @@
 GZIP_ENV = --best
 all: all-redirect
 .SUFFIXES:
-$(srcdir)/Makefile.in: Makefile.am $(top_srcdir)/configure.in $(ACLOCAL_M4) $(top_srcdir)/xmldocs.make $(top_srcdir)/omf.make
+$(srcdir)/Makefile.in: Makefile.am $(top_srcdir)/configure.in $(ACLOCAL_M4) $(top_srcdir)/xmldocs.make
 	cd $(top_srcdir) && $(AUTOMAKE) --gnu --include-deps help/C/Makefile
 
 Makefile: $(srcdir)/Makefile.in  $(top_builddir)/config.status
diff -Naur gnome-system-monitor-2.0.2.old/omf.make gnome-system-monitor-2.0.2.new/omf.make
--- gnome-system-monitor-2.0.2.old/omf.make	Tue Aug 27 08:55:17 2002
+++ gnome-system-monitor-2.0.2.new/omf.make	Fri Sep 27 04:07:13 2002
@@ -42,7 +42,7 @@
 	for file in $(omffile); do \
 		$(INSTALL_DATA) $$file.out $(DESTDIR)$(omf_dest_dir)/$$file; \
 	done
-	-scrollkeeper-update -p $(scrollkeeper_localstate_dir) -o $(DESTDIR)$(omf_dest_dir)
+#	-scrollkeeper-update -p $(scrollkeeper_localstate_dir) -o $(DESTDIR)$(omf_dest_dir)
 
 uninstall-local-omf:
 	-for file in $(srcdir)/*.omf; do \
diff -Naur gnome-system-monitor-2.0.2.old/src/Makefile.in gnome-system-monitor-2.0.2.new/src/Makefile.in
--- gnome-system-monitor-2.0.2.old/src/Makefile.in	Tue Aug 27 08:55:49 2002
+++ gnome-system-monitor-2.0.2.new/src/Makefile.in	Sat Sep 21 01:09:20 2002
@@ -108,7 +108,7 @@
 gnome_system_monitor_SOURCES =  	procman.c  procman.h 	interface.c interface.h 	callbacks.c callbacks.h 	load-graph.c load-graph.h 	proctable.c proctable.h 	prettytable.c prettytable.h 	util.c util.h 	infoview.c infoview.h 	procactions.c procactions.h 	procdialogs.c procdialogs.h 	memmaps.c memmaps.h 	favorites.c favorites.h 	defaulttable.h
 
 
-gnome_system_monitor_LDADD = @PROCMAN_LIBS@
+gnome_system_monitor_LDADD = @PROCMAN_LIBS@ -lXau -lX11
 
 specdir = $(datadir)/procman
 
diff -Naur gnome-system-monitor-2.0.2.old/src/util.c gnome-system-monitor-2.0.2.new/src/util.c
--- gnome-system-monitor-2.0.2.old/src/util.c	Tue Aug 27 08:55:12 2002
+++ gnome-system-monitor-2.0.2.new/src/util.c	Sat Sep 21 01:09:20 2002
@@ -40,7 +40,7 @@
  */
 #define OPEN_TTY() getpt()
 
-#ifdef __FreeBSD__
+#if defined(__FreeBSD__) || defined(__APPLE__)
 /* FreeBSD doesn't have getpt(). This function emulates it's behaviour. */
 int getpt (void);
 
@@ -144,8 +144,15 @@
 		}
 
 		if ((env = getenv ("XAUTHORITY")) == NULL) {
+#ifdef HAVE_ASPRINTF
 			if (asprintf (&env, "XAUTHORITY=%s/.Xauthority", home) > 0)
 				putenv (env);
+#else
+			if ((env = malloc (1024))) {
+				snprintf (env, 1024, "XAUTHORITY=%s/.Xauthority", home);
+				env[1023] = '\0';
+			}
+#endif
 			else {
 				perror ("Unable to allocate memory chunk");
 				_exit (-1);
@@ -153,8 +160,15 @@
 		}
 
 		if ((env = getenv ("ICEAUTHORITY")) == NULL) {
+#ifdef HAVE_ASPRINTF
 			if (asprintf (&env, "ICEAUTHORITY=%s/.ICEauthority", home) > 0)
 				putenv (env);
+#else
+			if ((env = malloc (1024))) {
+				snprintf (env, 1024, "ICEAUTHORITY=%s/.ICEauthority", home);
+				env[1023] = '\0';
+			}
+#endif
 			else {
 				perror ("Unable to allocate memory chunk");
 				_exit (-1);
