diff -Naur gnome-terminal-2.0.1.old/help/C/Makefile.in gnome-terminal-2.0.1.new/help/C/Makefile.in
--- gnome-terminal-2.0.1.old/help/C/Makefile.in	Mon Aug  5 02:05:29 2002
+++ gnome-terminal-2.0.1.new/help/C/Makefile.in	Fri Sep 27 04:10:29 2002
@@ -312,7 +312,7 @@
 	for file in $(omffile); do \
 		$(INSTALL_DATA) $$file.out $(DESTDIR)$(omf_dest_dir)/$$file; \
 	done
-	-scrollkeeper-update -p $(scrollkeeper_localstate_dir) -o $(DESTDIR)$(omf_dest_dir)
+#	-scrollkeeper-update -p $(scrollkeeper_localstate_dir) -o $(DESTDIR)$(omf_dest_dir)
 
 uninstall-local-omf:
 	-for file in $(srcdir)/*.omf; do \
diff -Naur gnome-terminal-2.0.1.old/help/omf.make gnome-terminal-2.0.1.new/help/omf.make
--- gnome-terminal-2.0.1.old/help/omf.make	Tue May 28 07:08:24 2002
+++ gnome-terminal-2.0.1.new/help/omf.make	Fri Sep 27 04:10:07 2002
@@ -42,7 +42,7 @@
 	for file in $(omffile); do \
 		$(INSTALL_DATA) $$file.out $(DESTDIR)$(omf_dest_dir)/$$file; \
 	done
-	-scrollkeeper-update -p $(scrollkeeper_localstate_dir) -o $(DESTDIR)$(omf_dest_dir)
+#	-scrollkeeper-update -p $(scrollkeeper_localstate_dir) -o $(DESTDIR)$(omf_dest_dir)
 
 uninstall-local-omf:
 	-for file in $(srcdir)/*.omf; do \
