diff -Naur linc-0.5.3.old/Makefile.in linc-0.5.3.new/Makefile.in
--- linc-0.5.3.old/Makefile.in	Sat Aug 24 01:49:23 2002
+++ linc-0.5.3.new/Makefile.in	Mon Sep 16 15:29:23 2002
@@ -90,7 +90,7 @@
 VERSION = @VERSION@
 WARN_CFLAGS = @WARN_CFLAGS@
 
-SUBDIRS = include src test docs
+SUBDIRS = include src docs
 
 m4datadir = $(datadir)/aclocal
 m4data_DATA = linc.m4
diff -Naur linc-0.5.3.old/config.h.in linc-0.5.3.new/config.h.in
--- linc-0.5.3.old/config.h.in	Wed Jul 31 19:26:22 2002
+++ linc-0.5.3.new/config.h.in	Mon Sep 16 15:29:23 2002
@@ -10,7 +10,9 @@
  * Copyright 2001, Red Hat, Inc., Ximian, Inc.,
  *                 Sun Microsystems, Inc.
  */
+#if !defined (__APPLE__)
 #undef socklen_t
+#endif
 #undef HAVE_SOCKADDR_SA_LEN
 #undef HAVE_INET_PTON
 
diff -Naur linc-0.5.3.old/include/linc/linc-protocol.h linc-0.5.3.new/include/linc/linc-protocol.h
--- linc-0.5.3.old/include/linc/linc-protocol.h	Thu Jun 13 01:46:39 2002
+++ linc-0.5.3.new/include/linc/linc-protocol.h	Mon Sep 16 15:29:23 2002
@@ -14,6 +14,10 @@
 
 #include <glib/gmacros.h>
 
+#if defined(__APPLE__) && !defined(socklen_t)
+#  define socklen_t int
+#endif
+
 G_BEGIN_DECLS
 
 #include <linc/linc-types.h>
diff -Naur linc-0.5.3.old/src/linc-protocols.c linc-0.5.3.new/src/linc-protocols.c
--- linc-0.5.3.old/src/linc-protocols.c	Wed Jul 31 19:45:22 2002
+++ linc-0.5.3.new/src/linc-protocols.c	Mon Sep 16 15:29:23 2002
@@ -11,12 +11,17 @@
  */
 #include <config.h>
 #include "linc-compat.h"
+#include <sys/types.h>
 #include <dirent.h>
 #include <linc/linc-protocol.h>
 #include <linc/linc-connection.h>
 
 #include "linc-private.h"
 #include "linc-debug.h"
+
+#if defined(__APPLE__) && !defined(RES_USE_INET6)
+#  undef AF_INET6
+#endif
 
 #undef LOCAL_DEBUG
 
