diff -Naur mozilla-1.2.1.old/.mozconfig mozilla-1.2.1.new/.mozconfig
--- mozilla-1.2.1.old/.mozconfig	Thu Jan  1 09:00:00 1970
+++ mozilla-1.2.1.new/.mozconfig	Sun Mar  2 13:55:16 2003
@@ -0,0 +1,17 @@
+ac_add_options --enable-macos-target=10.2
+ac_add_options --enable-default-toolkit=gtk2
+ac_add_options --with-pthreads
+ac_add_options --with-system-jpeg=@PREFIX@
+ac_add_options --with-system-zlib=@PREFIX@
+ac_add_options --with-system-png=@PREFIX@
+ac_add_options --disable-tests
+ac_add_options --disable-debug
+ac_add_options --disable-short-wchar
+ac_add_options --enable-optimize
+ac_add_options --enable-strip
+ac_add_options --enable-xft
+ac_add_options --enable-crypto
+ac_add_options --enable-mathml
+ac_add_options --enable-extensions=all
+ac_add_options --enable-xinerama
+ac_add_options --with-default-mozilla-five-home=@PREFIX@/lib/mozilla
diff -Naur mozilla-1.2.1.old/build/unix/mozilla-config.in mozilla-1.2.1.new/build/unix/mozilla-config.in
--- mozilla-1.2.1.old/build/unix/mozilla-config.in	Thu Apr 25 11:51:15 2002
+++ mozilla-1.2.1.new/build/unix/mozilla-config.in	Fri Feb 28 21:17:48 2003
@@ -55,7 +55,7 @@
       echo_exec_prefix=yes
       ;;
     --version)
-      echo -1
+      echo "%MOZILLA_VERSION%"
       exit 0
       ;;
     --cflags)
diff -Naur mozilla-1.2.1.old/config/autoconf.mk.in mozilla-1.2.1.new/config/autoconf.mk.in
--- mozilla-1.2.1.old/config/autoconf.mk.in	Wed Oct 16 08:34:42 2002
+++ mozilla-1.2.1.new/config/autoconf.mk.in	Sat Mar  1 03:45:23 2003
@@ -30,14 +30,14 @@
 prefix		= @prefix@
 exec_prefix	= @exec_prefix@
 bindir		= @bindir@
-includedir	= @includedir@/mozilla-$(MOZILLA_VERSION)
+includedir	= @includedir@/mozilla
 libdir		= @libdir@
 datadir		= @datadir@
 mandir		= @mandir@
-idldir		= @datadir@/idl/mozilla-$(MOZILLA_VERSION)
+idldir		= @datadir@/idl/mozilla
 
-mozappdir	= $(libdir)/mozilla-$(MOZILLA_VERSION)
-mredir		= $(libdir)/mre/mre-$(MOZILLA_VERSION)
+mozappdir	= $(libdir)/mozilla
+mredir		= $(libdir)/mre
 mrelibdir	= $(mredir)/lib
 
 DIST		= $(DEPTH)/dist
diff -Naur mozilla-1.2.1.old/config/rules.mk mozilla-1.2.1.new/config/rules.mk
--- mozilla-1.2.1.old/config/rules.mk	Fri Oct 25 14:59:03 2002
+++ mozilla-1.2.1.new/config/rules.mk	Fri Feb 28 21:17:48 2003
@@ -447,7 +447,7 @@
 ifdef IS_COMPONENT
 EXTRA_DSO_LDOPTS	+= -bundle
 else
-EXTRA_DSO_LDOPTS	+= -dynamiclib -install_name @executable_path/\$@ -compatibility_version 1 -current_version 1
+EXTRA_DSO_LDOPTS	+= -dynamiclib -install_name @PREFIX@/lib/mozilla/\$@ -compatibility_version 1 -current_version 1
 endif
 endif
 
diff -Naur mozilla-1.2.1.old/configure mozilla-1.2.1.new/configure
--- mozilla-1.2.1.old/configure	Sun Dec  1 04:59:50 2002
+++ mozilla-1.2.1.new/configure	Sat Mar  1 17:50:54 2003
@@ -4663,34 +4663,34 @@
 
 *-darwin*) 
     _PEDANTIC=
-    CFLAGS="$CFLAGS -fpascal-strings -traditional-cpp -fno-common"
-    CXXFLAGS="$CXXFLAGS -fpascal-strings -traditional-cpp -fno-common"
+    CFLAGS="$CFLAGS -no-cpp-precomp -fno-common"
+    CXXFLAGS="$CXXFLAGS -no-cpp-precomp -fno-common"
     DLL_SUFFIX=".dylib"
     DSO_LDOPTS=''
     STRIP="$STRIP -x -S"
     case "${target_cpu}" in
     powerpc*)
-        cat >> confdefs.h <<\EOF
-#define XP_MACOSX 1
-EOF
-
-        cat >> confdefs.h <<\EOF
-#define TARGET_CARBON 1
-EOF
-
-        cat >> confdefs.h <<\EOF
-#define TARGET_API_MAC_CARBON 1
-EOF
-
-        no_x=yes
+#        cat >> confdefs.h <<\EOF
+##define XP_MACOSX 1
+#EOF
+#
+#        cat >> confdefs.h <<\EOF
+##define TARGET_CARBON 1
+#EOF
+#
+#        cat >> confdefs.h <<\EOF
+##define TARGET_API_MAC_CARBON 1
+#EOF
+#
+#        no_x=yes
         MACOSX=1
-        _PLATFORM_DEFAULT_TOOLKIT='mac'
+#        _PLATFORM_DEFAULT_TOOLKIT='mac'
         DSO_LDOPTS="$DSO_LDOPTS -arch ppc"
-        TK_LIBS='-framework Carbon /System/Library/Frameworks/Carbon.framework/Carbon'
-        TK_CFLAGS="-I/Developer/Headers/FlatCarbon -F/System/Library/Frameworks"
-        CFLAGS="$CFLAGS $TK_CFLAGS"
-        CXXFLAGS="$CXXFLAGS $TK_CFLAGS"
-        MOZ_ENABLE_POSTSCRIPT=
+#        TK_LIBS='-framework Carbon /System/Library/Frameworks/Carbon.framework/Carbon'
+#        TK_CFLAGS="-I/Developer/Headers/FlatCarbon -F/System/Library/Frameworks"
+#        CFLAGS="$CFLAGS $TK_CFLAGS"
+#        CXXFLAGS="$CXXFLAGS $TK_CFLAGS"
+#        MOZ_ENABLE_POSTSCRIPT=
         ;;
     esac
     ;;
@@ -7542,7 +7542,7 @@
 fi
 if eval "test \"`echo '$ac_cv_lib_'$ac_lib_var`\" = yes"; then
   echo "$ac_t""yes" 1>&6
-  MOZ_XINERAMA_LIBS="-lXinerama"
+  MOZ_XINERAMA_LIBS="-lXinerama $XLIBS $XEXT_LIBS"
 else
   echo "$ac_t""no" 1>&6
 fi
@@ -14425,9 +14425,9 @@
 #define XP_UNIX 1
 EOF
 
-  cat >> confdefs.h <<\EOF
-#define UNIX_EMBED 1
-EOF
+#  cat >> confdefs.h <<\EOF
+##define UNIX_EMBED 1
+#EOF
 
   cat >> confdefs.h <<\EOF
 #define UNIX_LDAP 1
diff -Naur mozilla-1.2.1.old/directory/c-sdk/config/config.mk mozilla-1.2.1.new/directory/c-sdk/config/config.mk
--- mozilla-1.2.1.old/directory/c-sdk/config/config.mk	Wed Mar 27 06:51:57 2002
+++ mozilla-1.2.1.new/directory/c-sdk/config/config.mk	Fri Feb 28 21:17:48 2003
@@ -60,10 +60,10 @@
 
 NFSPWD		= $(MOD_DEPTH)/config/nfspwd
 
-CFLAGS		= $(CC_ONLY_FLAGS) $(OPTIMIZER) $(OS_CFLAGS)\
-		  $(XP_DEFINE) $(DEFINES) $(INCLUDES) $(XCFLAGS)
-CCCFLAGS	= $(CCC_ONLY_FLAGS) $(OPTIMIZER) $(OS_CFLAGS)\
-		  $(XP_DEFINE) $(DEFINES) $(INCLUDES) $(XCFLAGS)
+CFLAGS		= $(CC_ONLY_FLAGS) $(OPTIMIZER) $(XP_DEFINE)\
+		  $(DEFINES) $(INCLUDES) $(XCFLAGS) $(OS_CFLAGS)
+CCCFLAGS	= $(CCC_ONLY_FLAGS) $(OPTIMIZER) $(XP_DEFINE)\
+		  $(DEFINES) $(INCLUDES) $(XCFLAGS) $(OS_CFLAGS)
 # For purify
 NOMD_CFLAGS	= $(CC_ONLY_FLAGS) $(OPTIMIZER) $(NOMD_OS_CFLAGS)\
 		  $(XP_DEFINE) $(DEFINES) $(INCLUDES) $(XCFLAGS)
diff -Naur mozilla-1.2.1.old/directory/c-sdk/configure mozilla-1.2.1.new/directory/c-sdk/configure
--- mozilla-1.2.1.old/directory/c-sdk/configure	Fri Aug 23 07:44:21 2002
+++ mozilla-1.2.1.new/directory/c-sdk/configure	Fri Feb 28 21:17:48 2003
@@ -3483,7 +3483,7 @@
 
         CPU_ARCH=ppc
     fi
-    DSO_LDOPTS='-dynamiclib -compatibility_version 1 -current_version 1 -all_load -install_name @executable_path/$@'
+    DSO_LDOPTS='-dynamiclib -compatibility_version 1 -current_version 1 -all_load -install_name @PREFIX@/lib/mozilla/$@'
     # Use the standard preprocessor (cpp)
     CFLAGS="$CFLAGS -no-cpp-precomp"
     MKSHLIB='$(CC) -arch $(CPU_ARCH) $(DSO_LDOPTS) -o $@'
diff -Naur mozilla-1.2.1.old/fink/Applications/mozilla-editor.desktop mozilla-1.2.1.new/fink/Applications/mozilla-editor.desktop
--- mozilla-1.2.1.old/fink/Applications/mozilla-editor.desktop	Thu Jan  1 09:00:00 1970
+++ mozilla-1.2.1.new/fink/Applications/mozilla-editor.desktop	Fri Feb 28 21:17:48 2003
@@ -0,0 +1,8 @@
+[Desktop Entry]
+Name=Mozilla Editor
+Comment=HTML Composer
+Exec=mozilla -edit
+Icon=gnome-word.png
+Terminal=false
+MultipleArgs=false
+Type=Application
diff -Naur mozilla-1.2.1.old/fink/Internet/mozilla-address.desktop mozilla-1.2.1.new/fink/Internet/mozilla-address.desktop
--- mozilla-1.2.1.old/fink/Internet/mozilla-address.desktop	Thu Jan  1 09:00:00 1970
+++ mozilla-1.2.1.new/fink/Internet/mozilla-address.desktop	Fri Feb 28 21:17:48 2003
@@ -0,0 +1,8 @@
+[Desktop Entry]
+Name=Mozilla Addressbook
+Comment=Addressbook from Mozilla Mailnews
+Exec=mozilla -addressbook
+Icon=gnome-calendar.png
+Terminal=false
+MultipleArgs=false
+Type=Application
diff -Naur mozilla-1.2.1.old/fink/Internet/mozilla-chatzilla.desktop mozilla-1.2.1.new/fink/Internet/mozilla-chatzilla.desktop
--- mozilla-1.2.1.old/fink/Internet/mozilla-chatzilla.desktop	Thu Jan  1 09:00:00 1970
+++ mozilla-1.2.1.new/fink/Internet/mozilla-chatzilla.desktop	Fri Feb 28 21:17:48 2003
@@ -0,0 +1,8 @@
+[Desktop Entry]
+Name=Chatzilla
+Comment=IRC client
+Exec=mozilla -chat
+Icon=gnome-irc.png
+Terminal=false
+MultipleArgs=false
+Type=Application
diff -Naur mozilla-1.2.1.old/fink/Internet/mozilla-mail-composer.desktop mozilla-1.2.1.new/fink/Internet/mozilla-mail-composer.desktop
--- mozilla-1.2.1.old/fink/Internet/mozilla-mail-composer.desktop	Thu Jan  1 09:00:00 1970
+++ mozilla-1.2.1.new/fink/Internet/mozilla-mail-composer.desktop	Fri Feb 28 21:17:48 2003
@@ -0,0 +1,8 @@
+[Desktop Entry]
+Name=Mozilla Mail Composer
+Comment=Mozilla Mail Composer
+Exec=mozilla -compose
+Icon=mozilla-mail-composer.gif
+Terminal=false
+MultipleArgs=false
+Type=Application
diff -Naur mozilla-1.2.1.old/fink/Internet/mozilla-mail.desktop mozilla-1.2.1.new/fink/Internet/mozilla-mail.desktop
--- mozilla-1.2.1.old/fink/Internet/mozilla-mail.desktop	Thu Jan  1 09:00:00 1970
+++ mozilla-1.2.1.new/fink/Internet/mozilla-mail.desktop	Fri Feb 28 21:17:48 2003
@@ -0,0 +1,8 @@
+[Desktop Entry]
+Name=Mozilla Mail
+Comment=Mozilla Mail
+Exec=mozilla -mail
+Icon=mozilla-mail.gif
+Terminal=false
+MultipleArgs=false
+Type=Application
diff -Naur mozilla-1.2.1.old/fink/Internet/mozilla-news.desktop mozilla-1.2.1.new/fink/Internet/mozilla-news.desktop
--- mozilla-1.2.1.old/fink/Internet/mozilla-news.desktop	Thu Jan  1 09:00:00 1970
+++ mozilla-1.2.1.new/fink/Internet/mozilla-news.desktop	Fri Feb 28 21:17:48 2003
@@ -0,0 +1,8 @@
+[Desktop Entry]
+Name=Mozilla News
+Comment=Mozilla Mail
+Exec=mozilla -news
+Icon=gnome-news.png
+Terminal=false
+MultipleArgs=false
+Type=Application
diff -Naur mozilla-1.2.1.old/fink/Internet/mozilla.desktop mozilla-1.2.1.new/fink/Internet/mozilla.desktop
--- mozilla-1.2.1.old/fink/Internet/mozilla.desktop	Thu Jan  1 09:00:00 1970
+++ mozilla-1.2.1.new/fink/Internet/mozilla.desktop	Fri Feb 28 21:17:48 2003
@@ -0,0 +1,8 @@
+[Desktop Entry]
+Name=Mozilla
+Comment=Mozilla Web Browser
+Exec=mozilla 
+Icon=mozilla.png
+Terminal=false
+MultipleArgs=false
+Type=Application
diff -Naur mozilla-1.2.1.old/fink/chrome.d/default mozilla-1.2.1.new/fink/chrome.d/default
--- mozilla-1.2.1.old/fink/chrome.d/default	Thu Jan  1 09:00:00 1970
+++ mozilla-1.2.1.new/fink/chrome.d/default	Fri Feb 28 21:17:48 2003
@@ -0,0 +1,2 @@
+skin,install,select,modern/1.0
+locale,install,select,en-US
diff -Naur mozilla-1.2.1.old/fink/mozilla mozilla-1.2.1.new/fink/mozilla
--- mozilla-1.2.1.old/fink/mozilla	Thu Jan  1 09:00:00 1970
+++ mozilla-1.2.1.new/fink/mozilla	Fri Feb 28 21:17:48 2003
@@ -0,0 +1,99 @@
+#!/bin/sh
+#
+# The contents of this file are subject to the Netscape Public
+# License Version 1.1 (the "License"); you may not use this file
+# except in compliance with the License. You may obtain a copy of
+# the License at http://www.mozilla.org/NPL/
+#
+# Software distributed under the License is distributed on an "AS
+# IS" basis, WITHOUT WARRANTY OF ANY KIND, either express or
+# implied. See the License for the specific language governing
+# rights and limitations under the License.
+#
+# The Original Code is mozilla.org code.
+#
+# The Initial Developer of the Original Code is Netscape
+# Communications Corporation.  Portions created by Netscape are
+# Copyright (C) 1998 Netscape Communications Corporation. All
+# Rights Reserved.
+#
+# Contributor(s): 
+#
+
+## 
+## Usage:
+##
+## $ mozilla
+##
+## This script is meant to run a mozilla program from the mozilla
+## rpm installation.
+##
+## The script will setup all the environment voodoo needed to make
+## mozilla work.
+
+##
+## Variables
+##
+MOZ_PROGRAM="@PREFIX@/lib/mozilla/mozilla-bin"
+
+##
+## Set MOZILLA_FIVE_HOME
+##
+MOZILLA_FIVE_HOME="@PREFIX@/lib/mozilla"
+export MOZILLA_FIVE_HOME
+
+##
+## Set DYLD_LIBRARY_PATH
+##
+DYLD_LIBRARY_PATH=@PREFIX@/lib/mozilla:$DYLD_LIBRARY_PATH
+export DYLD_LIBRARY_PATH
+
+# If there is no command line argument at all then try to open a new
+# window in an already running instance.
+
+if [ -z "$1" ]; then
+  $MOZ_PROGRAM -remote "openurl(about:blank,new-window)" 2>/dev/null 1>/dev/null
+  RETURN_VAL=$?
+  if [ "$RETURN_VAL" -eq "2" ]; then
+    $MOZ_PROGRAM ${1+"$@"}
+    RETURN_VAL=$?
+  fi
+  exit $RETURN_VAL
+fi
+
+# If there's a command line argument but it doesn't begin with a -
+# it's probably a url.  Try to send it to a running instance.
+
+USE_EXIST=0
+opt="$1"
+case "$opt" in
+  -*) ;;
+  *) USE_EXIST=1 ;;
+esac
+
+if [ "${USE_EXIST}" -eq "1" ]; then
+  # check to make sure that the command contains at least a :/ in it.
+  echo $opt | grep -e ':/' 2>/dev/null > /dev/null
+  RETURN_VAL=$?
+  if [ "$RETURN_VAL" -eq "1" ]; then
+    # does it begin with a / ?
+    echo $opt | grep -e '^/' 2>/dev/null > /dev/null
+    RETURN_VAL=$?
+    if [ "$RETURN_VAL" -eq "0" ]; then
+      opt="file:$opt"
+    elif [ -e `pwd`/$opt ]; then
+      opt="`pwd`/$opt"
+    else
+      opt="http://$opt"
+    fi
+  fi
+  $MOZ_PROGRAM -remote "openurl($opt,new-window)" 2>/dev/null >/dev/null
+  RETURN_VAL=$?
+  if [ "$RETURN_VAL" -eq "2" ]; then
+    $MOZ_PROGRAM ${1+"$@"}
+    RETURN_VAL=$?
+  fi
+  exit $RETURN_VAL
+fi
+
+$MOZ_PROGRAM ${1+"$@"}
diff -Naur mozilla-1.2.1.old/fink/prefs.js mozilla-1.2.1.new/fink/prefs.js
--- mozilla-1.2.1.old/fink/prefs.js	Thu Jan  1 09:00:00 1970
+++ mozilla-1.2.1.new/fink/prefs.js	Fri Feb 28 21:17:48 2003
@@ -0,0 +1,17 @@
+// XIM style
+pref("xim.input_style", "over-the-spot");
+
+// TryeType
+pref("font.FreeType2.enable", true);
+pref("font.freetype2.shared-library", "@PREFIX@/lib/mozilla/libfreetype.6.so");
+pref("font.FreeType2.autohinted", false);
+pref("font.FreeType2.unhinted", false);
+pref("font.antialias.min",        10);
+pref("font.directory.truetype.1", "/usr/X11R6/lib/X11/fonts/TTF");
+pref("font.directory.truetype.2", "@PREFIX@/share/fonts/truetype");
+
+// AA with Bitmap scaling.
+pref("font.scale.aa_bitmap.enable", true);
+pref("font.scale.aa_bitmap.always", false);
+pref("font.scale.aa_bitmap.min", 6);
+
diff -Naur mozilla-1.2.1.old/fink/update-mozilla-chrome mozilla-1.2.1.new/fink/update-mozilla-chrome
--- mozilla-1.2.1.old/fink/update-mozilla-chrome	Thu Jan  1 09:00:00 1970
+++ mozilla-1.2.1.new/fink/update-mozilla-chrome	Fri Feb 28 21:17:48 2003
@@ -0,0 +1,22 @@
+#!/bin/sh -e
+
+ulimit -s unlimited || true
+
+MOZILLA_FIVE_HOME=@PREFIX@/lib/mozilla
+CHROME_DIR=@PREFIX@/var/lib/mozilla/chrome.d
+
+TMPFILE=`tempfile`
+cat ${CHROME_DIR}/* >>${TMPFILE}
+mv ${TMPFILE} ${MOZILLA_FIVE_HOME}/chrome/installed-chrome.txt
+chmod 644 ${MOZILLA_FIVE_HOME}/chrome/installed-chrome.txt
+
+rm -rf ${MOZILLA_FIVE_HOME}/chrome/overlayinfo
+rm -f  ${MOZILLA_FIVE_HOME}/chrome/*.rdf
+rm -f  ${MOZILLA_FIVE_HOME}/component.reg
+rm -f  ${MOZILLA_FIVE_HOME}/components/*.dat
+
+sudo -H ${MOZILLA_FIVE_HOME}/regxpcom || true
+sudo -H ${MOZILLA_FIVE_HOME}/regchrome || true
+
+exit 0
+
diff -Naur mozilla-1.2.1.old/gfx/src/gtk/Makefile.in mozilla-1.2.1.new/gfx/src/gtk/Makefile.in
--- mozilla-1.2.1.old/gfx/src/gtk/Makefile.in	Tue Oct 15 12:58:05 2002
+++ mozilla-1.2.1.new/gfx/src/gtk/Makefile.in	Sat Mar  1 06:05:57 2003
@@ -136,7 +136,6 @@
 		$(LIBS_DIR) \
 		-lgkgfx \
 		-lgfxshared_s \
-		$(GFX_XINERAMA_LIBS) \
 		-lgtksuperwin \
 		$(MOZ_COMPONENT_LIBS) \
 		$(MOZ_JS_LIBS) \
@@ -147,7 +146,6 @@
 EXTRA_DSO_LDOPTS += $(MOZ_COMPONENT_LIBS) \
 		-lgkgfx \
 		-lgfxshared_s \
-		$(GFX_XINERAMA_LIBS) \
 		$(MOZ_JS_LIBS) \
 		$(NULL)
 endif
@@ -157,7 +155,7 @@
 CFLAGS		+= $(MOZ_XFT_CFLAGS) $(MOZ_GTK_CFLAGS) $(MOZ_GTK2_CFLAGS)
 
 ifdef MOZ_ENABLE_XINERAMA
-GFX_XINERAMA_LIBS += $(MOZ_XINERAMA_LIBS)
+EXTRA_DSO_LDOPTS += $(MOZ_XINERAMA_LIBS)
 endif
 
 ifeq ($(OS_ARCH), Linux)
@@ -193,6 +191,8 @@
 ifdef MOZ_ENABLE_FREETYPE2
 DEFINES         += -DMOZ_ENABLE_FREETYPE2
 INCLUDES        += $(FT2_CFLAGS)
+CC		+= $(FT2_CFLAGS)
+CXX		+= $(FT2_CFLAGS)
 # due to runtime linking with PR_LoadLibrary
 # EXTRA_DSO_LDOPTS does not need $(FT2_LIBS)
 endif
diff -Naur mozilla-1.2.1.old/modules/libpref/src/nsPrefService.cpp mozilla-1.2.1.new/modules/libpref/src/nsPrefService.cpp
--- mozilla-1.2.1.old/modules/libpref/src/nsPrefService.cpp	Sun Sep  8 02:11:33 2002
+++ mozilla-1.2.1.new/modules/libpref/src/nsPrefService.cpp	Fri Feb 28 21:17:48 2003
@@ -530,6 +530,7 @@
 #elif defined(XP_BEOS)
       "beos.js"
 #endif
+      , "fink.js"
   };
 
   rv = NS_GetSpecialDirectory(NS_APP_PREF_DEFAULTS_50_DIR, getter_AddRefs(defaultPrefDir));
diff -Naur mozilla-1.2.1.old/modules/plugin/base/src/Makefile.in mozilla-1.2.1.new/modules/plugin/base/src/Makefile.in
--- mozilla-1.2.1.old/modules/plugin/base/src/Makefile.in	Thu Oct 17 15:46:16 2002
+++ mozilla-1.2.1.new/modules/plugin/base/src/Makefile.in	Sat Mar  1 18:26:14 2003
@@ -104,6 +104,10 @@
 		$(MOZ_UNICHARUTIL_LIBS) \
 		$(NULL)
 
+ifdef MOZ_ENABLE_XINERAMA
+EXTRA_DSO_LDOPTS += $(MOZ_XINERAMA_LIBS)  
+endif
+
 ifneq (,$(filter mac cocoa,$(MOZ_WIDGET_TOOLKIT)))
 EXTRA_DSO_LDOPTS += $(TK_LIBS)
 endif
@@ -118,13 +122,13 @@
 ifdef MOZ_ENABLE_GTK
 CXXFLAGS        += $(MOZ_GTK_CFLAGS)
 CFLAGS          += $(MOZ_GTK_CFLAGS)
-EXTRA_DSO_LDOPTS += -lgtkxtbin -lgtksuperwin $(XLDFLAGS) $(XT_LIBS) $(MOZ_GTK_LDFLAGS)
+EXTRA_DSO_LDOPTS += -lgtkxtbin -lgtksuperwin $(XLDFLAGS) $(XT_LIBS) $(XLIBS) $(MOZ_GTK_LDFLAGS)
 endif		#MOZ_ENABLE_GTK
 
 ifdef MOZ_ENABLE_GTK2
 CXXFLAGS        += $(MOZ_GTK2_CFLAGS)
 CFLAGS          += $(MOZ_GTK2_CFLAGS)
-EXTRA_DSO_LDOPTS += -lgtkxtbin $(XLDFLAGS) $(XT_LIBS) $(MOZ_GTK2_LIBS)
+EXTRA_DSO_LDOPTS += -lgtkxtbin $(XLDFLAGS) $(XT_LIBS) $(XLIBS) $(MOZ_GTK2_LIBS)
 endif           #MOZ_ENABLE_GTK2
 
 ifdef MOZ_ENABLE_XLIB
diff -Naur mozilla-1.2.1.old/nsprpub/configure mozilla-1.2.1.new/nsprpub/configure
--- mozilla-1.2.1.old/nsprpub/configure	Thu Sep 26 21:46:21 2002
+++ mozilla-1.2.1.new/nsprpub/configure	Fri Feb 28 21:17:49 2003
@@ -3278,7 +3278,7 @@
 
         CPU_ARCH=ppc
     fi
-    DSO_LDOPTS='-dynamiclib -compatibility_version 1 -current_version 1 -all_load -install_name @executable_path/$@'
+    DSO_LDOPTS='-dynamiclib -compatibility_version 1 -current_version 1 -all_load -install_name @PREFIX@/lib/mozilla/$@'
     # Use the standard preprocessor (cpp)
     CFLAGS="$CFLAGS -no-cpp-precomp"
     MKSHLIB='$(CC) -arch $(CPU_ARCH) $(DSO_LDOPTS) -o $@'
diff -Naur mozilla-1.2.1.old/other-licenses/Xft/fontconfig/src/Makefile.in mozilla-1.2.1.new/other-licenses/Xft/fontconfig/src/Makefile.in
--- mozilla-1.2.1.old/other-licenses/Xft/fontconfig/src/Makefile.in	Fri Apr 26 04:03:35 2002
+++ mozilla-1.2.1.new/other-licenses/Xft/fontconfig/src/Makefile.in	Fri Feb 28 21:17:49 2003
@@ -46,3 +46,5 @@
 ifdef MOZ_XFT_SYSTEM_FREETYPE2
 CFLAGS		+= $(FT2_CFLAGS)
 endif
+
+CFLAGS := -I$(MOZ_BUILD_ROOT)/dist/include/expat-1.95 $(CFLAGS)
diff -Naur mozilla-1.2.1.old/other-licenses/expat/xmlwf/Makefile.in mozilla-1.2.1.new/other-licenses/expat/xmlwf/Makefile.in
--- mozilla-1.2.1.old/other-licenses/expat/xmlwf/Makefile.in	Mon Feb 25 15:53:11 2002
+++ mozilla-1.2.1.new/other-licenses/expat/xmlwf/Makefile.in	Fri Feb 28 21:17:49 2003
@@ -23,7 +23,7 @@
 LIBDIR= ../lib/.libs
 INCDIR= ../lib
 
-LDFLAGS= @LDFLAGS@ -static
+LDFLAGS= @LDFLAGS@
 CFLAGS= @CFLAGS@ -I$(INCDIR)
 CC = @CC@
 
diff -Naur mozilla-1.2.1.old/other-licenses/freetype/builds/unix/configure mozilla-1.2.1.new/other-licenses/freetype/builds/unix/configure
--- mozilla-1.2.1.old/other-licenses/freetype/builds/unix/configure	Fri Apr 26 04:00:54 2002
+++ mozilla-1.2.1.new/other-licenses/freetype/builds/unix/configure	Fri Feb 28 21:17:49 2003
@@ -5715,7 +5715,7 @@
     # FIXME: Relying on posixy $() will cause problems for
     #        cross-compilation, but unfortunately the echo tests do not
     #        yet detect zsh echo's removal of \ escapes.
-    archive_cmds='$nonopt $(test "x$module" = xyes && echo -bundle || echo -dynamiclib) $allow_undefined_flag -o $lib $libobjs $deplibs$linker_flags -install_name $rpath/$soname $verstring'
+    archive_cmds='$nonopt $(test .$module = .yes && echo -bundle || echo -dynamiclib) $allow_undefined_flag -o $lib $libobjs $deplibs$linker_flags $(test .$module != .yes && echo -install_name $rpath/$soname $verstring)'
     # We need to add '_' to the symbols in $export_symbols first
     #archive_expsym_cmds="$archive_cmds"' && strip -s $export_symbols'
     hardcode_direct=yes
diff -Naur mozilla-1.2.1.old/other-licenses/freetype/builds/unix/unix-cc.in mozilla-1.2.1.new/other-licenses/freetype/builds/unix/unix-cc.in
--- mozilla-1.2.1.old/other-licenses/freetype/builds/unix/unix-cc.in	Mon Feb 25 15:57:25 2002
+++ mozilla-1.2.1.new/other-licenses/freetype/builds/unix/unix-cc.in	Fri Feb 28 21:17:49 2003
@@ -76,6 +76,6 @@
 # Library linking
 #
 LINK_LIBRARY = $(LIBTOOL) --mode=link $(CCraw) -o $@ $(OBJECTS_LIST) \
-                          -rpath $(libdir) -version-info $(version_info)
+                          -rpath $(libdir) -version-info $(version_info) -module
 
 # EOF
diff -Naur mozilla-1.2.1.old/other-licenses/freetype/include/freetype/config/ftoption.h mozilla-1.2.1.new/other-licenses/freetype/include/freetype/config/ftoption.h
--- mozilla-1.2.1.old/other-licenses/freetype/include/freetype/config/ftoption.h	Fri Apr 26 03:57:53 2002
+++ mozilla-1.2.1.new/other-licenses/freetype/include/freetype/config/ftoption.h	Fri Feb 28 21:17:49 2003
@@ -311,7 +311,7 @@
   /* embedded bitmaps in all formats using the SFNT module (namely         */
   /* TrueType & OpenType).                                                 */
   /*                                                                       */
-#define TT_CONFIG_OPTION_EMBEDDED_BITMAPS
+#undef TT_CONFIG_OPTION_EMBEDDED_BITMAPS
 
 
   /*************************************************************************/
@@ -375,7 +375,7 @@
   /* By undefining this, you will only compile the code necessary to load  */
   /* TrueType glyphs without hinting.                                      */
   /*                                                                       */
-#undef TT_CONFIG_OPTION_BYTECODE_INTERPRETER
+#define TT_CONFIG_OPTION_BYTECODE_INTERPRETER
 
 
   /*************************************************************************/
diff -Naur mozilla-1.2.1.old/security/coreconf/Darwin.mk mozilla-1.2.1.new/security/coreconf/Darwin.mk
--- mozilla-1.2.1.old/security/coreconf/Darwin.mk	Wed Jun 12 09:21:19 2002
+++ mozilla-1.2.1.new/security/coreconf/Darwin.mk	Fri Feb 28 21:17:49 2003
@@ -57,7 +57,7 @@
 # definitions so that the linker can catch multiply-defined symbols.
 # Also, common symbols are not allowed with Darwin dynamic libraries.
 
-OS_CFLAGS	= $(DSO_CFLAGS) $(OS_REL_CFLAGS) -Wmost -fpascal-strings -traditional-cpp -fno-common -pipe -DDARWIN -DHAVE_STRERROR -DHAVE_BSD_FLOCK
+OS_CFLAGS	= $(DSO_CFLAGS) $(OS_REL_CFLAGS) -Wmost -no-cpp-precomp -fno-common -pipe -DDARWIN -DHAVE_STRERROR -DHAVE_BSD_FLOCK
 
 ifdef BUILD_OPT
 OPTIMIZER	= -O2
@@ -66,7 +66,7 @@
 ARCH		= darwin
 
 # May override this with -bundle to create a loadable module.
-DSO_LDOPTS	= -dynamiclib -compatibility_version 1 -current_version 1 -install_name @executable_path/$(notdir $@)
+DSO_LDOPTS	= -dynamiclib -compatibility_version 1 -current_version 1 -install_name @PREFIX@/lib/mozilla/$(notdir $@)
 
 MKSHLIB		= $(CC) -arch $(CPU_ARCH) $(DSO_LDOPTS)
 DLL_SUFFIX	= dylib
diff -Naur mozilla-1.2.1.old/security/nss/cmd/platlibs.mk mozilla-1.2.1.new/security/nss/cmd/platlibs.mk
--- mozilla-1.2.1.old/security/nss/cmd/platlibs.mk	Thu Sep 19 08:25:27 2002
+++ mozilla-1.2.1.new/security/nss/cmd/platlibs.mk	Fri Feb 28 21:17:49 2003
@@ -177,7 +177,7 @@
 endif
 
 ifeq ($(OS_ARCH), Darwin)
-EXTRA_SHARED_LIBS += -dylib_file @executable_path/libsoftokn3.dylib:$(DIST)/lib/libsoftokn3.dylib
+EXTRA_SHARED_LIBS += -dylib_file @PREFIX@/lib/mozilla/libsoftokn3.dylib:$(DIST)/lib/libsoftokn3.dylib
 endif
 
 # Use the Solaris ld, which knows where to find libsoftokn3.so.
diff -Naur mozilla-1.2.1.old/security/nss/lib/smime/config.mk mozilla-1.2.1.new/security/nss/lib/smime/config.mk
--- mozilla-1.2.1.old/security/nss/lib/smime/config.mk	Sat Feb 16 08:38:43 2002
+++ mozilla-1.2.1.new/security/nss/lib/smime/config.mk	Fri Feb 28 21:17:49 2003
@@ -60,7 +60,7 @@
 	$(NULL)
 
 ifeq ($(OS_ARCH), Darwin)
-EXTRA_SHARED_LIBS += -dylib_file @executable_path/libsoftokn3.dylib:$(DIST)/lib/libsoftokn3.dylib
+EXTRA_SHARED_LIBS += -dylib_file @PREFIX@/lib/mozilla/libsoftokn3.dylib:$(DIST)/lib/libsoftokn3.dylib
 endif
 
 endif
diff -Naur mozilla-1.2.1.old/security/nss/lib/ssl/config.mk mozilla-1.2.1.new/security/nss/lib/ssl/config.mk
--- mozilla-1.2.1.old/security/nss/lib/ssl/config.mk	Fri Feb 22 13:23:22 2002
+++ mozilla-1.2.1.new/security/nss/lib/ssl/config.mk	Fri Feb 28 21:17:49 2003
@@ -64,7 +64,7 @@
 endif
 
 ifeq ($(OS_ARCH), Darwin)
-EXTRA_SHARED_LIBS += -dylib_file @executable_path/libsoftokn3.dylib:$(DIST)/lib/libsoftokn3.dylib
+EXTRA_SHARED_LIBS += -dylib_file @PREFIX@/lib/mozilla/libsoftokn3.dylib:$(DIST)/lib/libsoftokn3.dylib
 endif
 
 endif
diff -Naur mozilla-1.2.1.old/widget/src/gtk2/Makefile.in mozilla-1.2.1.new/widget/src/gtk2/Makefile.in
--- mozilla-1.2.1.old/widget/src/gtk2/Makefile.in	Wed Oct 30 10:04:59 2002
+++ mozilla-1.2.1.new/widget/src/gtk2/Makefile.in	Sat Mar  1 06:15:49 2003
@@ -77,6 +77,10 @@
 		$(XLIBS) \
 		$(MOZ_GTK2_LIBS)
 
+ifdef MOZ_ENABLE_XINERAMA
+EXTRA_DSO_LDOPTS += $(MOZ_XINERAMA_LIBS)   
+endif
+
 EXPORTS		= \
 		mozdrawingarea.h \
 		mozcontainer.h \
diff -Naur mozilla-1.2.1.old/widget/src/gtk2/mai/Makefile.in mozilla-1.2.1.new/widget/src/gtk2/mai/Makefile.in
--- mozilla-1.2.1.old/widget/src/gtk2/mai/Makefile.in	Sun Sep 29 13:51:20 2002
+++ mozilla-1.2.1.new/widget/src/gtk2/mai/Makefile.in	Sat Mar  1 06:27:32 2003
@@ -78,7 +78,7 @@
                 nsMaiHyperlink.cpp \
 		$(NULL)
 
-EXTRA_DSO_LDOPTS += $(MOZ_GTK2_LIBS)
+EXTRA_DSO_LDOPTS += $(NSPR_LIBS) $(XPCOM_LIBS) $(MOZ_GTK2_LIBS)
 
 include $(topsrcdir)/config/rules.mk
 
diff -Naur mozilla-1.2.1.old/widget/src/gtkxtbin/Makefile.in mozilla-1.2.1.new/widget/src/gtkxtbin/Makefile.in
--- mozilla-1.2.1.old/widget/src/gtkxtbin/Makefile.in	Tue Oct  1 07:18:27 2002
+++ mozilla-1.2.1.new/widget/src/gtkxtbin/Makefile.in	Sat Mar  1 05:51:46 2003
@@ -33,16 +33,19 @@
 CSRCS		= \
 		gtk2xtbin.c \
 		$(NULL)
-EXTRA_DSO_LDOPTS += $(MOZ_GTK2_LIBS) $(XLDFLAGS) $(XT_LIBS)
+EXTRA_DSO_LDOPTS += $(MOZ_GTK2_LIBS) $(XLDFLAGS) $(XT_LIBS) $(XLIBS)
 endif
 
 ifdef MOZ_ENABLE_GTK
 CSRCS           = \
                 gtkxtbin.c \
 		$(NULL)
-EXTRA_DSO_LDOPTS += $(MOZ_GTK_LDFLAGS) $(XT_LIBS)
+EXTRA_DSO_LDOPTS += $(MOZ_GTK_LDFLAGS) $(XT_LIBS) $(XLIBS)
 endif
 
+ifdef MOZ_ENABLE_XINERAMA
+EXTRA_DSO_LDOPTS += $(MOZ_XINERAMA_LIBS)  
+endif
 
 EXPORTS         = gtkxtbin.h gtk2xtbin.h 
 
diff -Naur mozilla-1.2.1.old/xpfe/bootstrap/Makefile.in mozilla-1.2.1.new/xpfe/bootstrap/Makefile.in
--- mozilla-1.2.1.old/xpfe/bootstrap/Makefile.in	Sat Nov  9 00:33:48 2002
+++ mozilla-1.2.1.new/xpfe/bootstrap/Makefile.in	Fri Feb 28 21:17:49 2003
@@ -250,7 +250,7 @@
 endif
 
 ifeq ($(OS_ARCH),Darwin)
-EXTRA_DSO_LDOPTS := $(subst -dynamiclib -install_name @executable_path/\$@ -compatibility_version 1 -current_version 1,-execute,$(EXTRA_DSO_LDOPTS))
+EXTRA_DSO_LDOPTS := $(subst -dynamiclib -install_name @PREFIX@/lib/mozilla/\$@ -compatibility_version 1 -current_version 1,-execute,$(EXTRA_DSO_LDOPTS))
 
 ifdef BUILD_STATIC_LIBS
 LDFLAGS	+= -framework QuickTime $(TK_LIBS)
diff -Naur mozilla-1.2.1.old/xpfe/browser/resources/content/unix/platformNavigationBindings.xul mozilla-1.2.1.new/xpfe/browser/resources/content/unix/platformNavigationBindings.xul
--- mozilla-1.2.1.old/xpfe/browser/resources/content/unix/platformNavigationBindings.xul	Mon Sep  2 12:35:28 2002
+++ mozilla-1.2.1.new/xpfe/browser/resources/content/unix/platformNavigationBindings.xul	Mon Mar  3 03:38:31 2003
@@ -12,8 +12,12 @@
 
     <!-- Some people apparently use this combination too on Unix...
          we live and learn.  -->
+<!-- FIXME: I don't know why it does not work.
     <key key="&goBackCmd.commandKey;" command="Browser:Back" modifiers="accel"/>
     <key key="&goForwardCmd.commandKey;" command="Browser:Forward" modifiers="accel"/>
+-->
+    <key key="[" command="Browser:Back" modifiers="accel"/>
+    <key key="]" command="Browser:Forward" modifiers="accel"/>
 
     <key id="goHome" keycode="VK_HOME" command="Browser:Home" modifiers="alt"/>
     <key id="key_fullScreen" keycode="VK_F11" command="View:FullScreen"/>
diff -Naur mozilla-1.2.1.old/xpfe/communicator/resources/content/unix/platformCommunicatorOverlay.xul mozilla-1.2.1.new/xpfe/communicator/resources/content/unix/platformCommunicatorOverlay.xul
--- mozilla-1.2.1.old/xpfe/communicator/resources/content/unix/platformCommunicatorOverlay.xul	Mon Sep  2 12:35:39 2002
+++ mozilla-1.2.1.new/xpfe/communicator/resources/content/unix/platformCommunicatorOverlay.xul	Mon Mar  3 03:38:49 2003
@@ -14,15 +14,22 @@
      xmlns="http://www.mozilla.org/keymaster/gatekeeper/there.is.only.xul">
 
     <!-- close -->
+<!-- FIXME: I don't know why it does not work.
     <menuitem id="menu_close" label="&closeCmd.label;" key="key_close" accesskey="&closeCmd.accesskey;" command="cmd_close"/>
+-->
+    <menuitem id="menu_close" label="&closeCmd.label;" key="key_close" accesskey="c" command="cmd_close"/>
     <key id="key_close"  key="&closeCmd.key;" command="cmd_close" modifiers="accel"/>
     <key id="key_closeWindow"  key="&closeCmd.key;" command="cmd_closeWindow" modifiers="accel,shift"/>
 
     <!-- quit -->
     <menupopup id="menu_FilePopup">
       <menuseparator id="menu_FileQuitSeparator"/>
+<!-- FIXME: I don't know why it does not work.
       <menuitem label="&quitApplicationCmd.label;" id="menu_FileQuitItem"
                 key="key_quit" accesskey="&quitApplicationCmd.accesskey;" command="cmd_quit"/>
+-->
+      <menuitem label="&quitApplicationCmd.label;" id="menu_FileQuitItem"
+                key="key_quit" accesskey="q" command="cmd_quit"/>
     </menupopup>
     <key id="key_quit"  key="&quitApplicationCmd.key;" command="cmd_quit" modifiers="accel"/>
     <command id="cmd_quit" oncommand="goQuitApplication()"/>
