diff -Naur GConf-2.2.0.old/doc/gconf/Makefile.in GConf-2.2.0.new/doc/gconf/Makefile.in
--- GConf-2.2.0.old/doc/gconf/Makefile.in	Tue Jan 21 12:30:30 2003
+++ GConf-2.2.0.new/doc/gconf/Makefile.in	Sun Mar  9 09:03:03 2003
@@ -172,7 +172,7 @@
 
 # CFLAGS and LDFLAGS for compiling scan program. Only needed
 # if $(DOC_MODULE).types is non-empty.
-GTKDOC_CFLAGS = -I$(top_srcdir) $(CFLAGS) $(DEPENDENT_CFLAGS)
+GTKDOC_CFLAGS = -I$(top_srcdir) $(CFLAGS) $(CPPFLAGS) $(DEPENDENT_CFLAGS)
 
 GTKDOC_LIBS = $(LDFLAGS) $(DEPENDENT_LIBS) $(top_builddir)/gconf/libgconf-$(MAJOR_VERSION).la
 
diff -Naur GConf-2.2.0.old/gconf/gconf-internals.c GConf-2.2.0.new/gconf/gconf-internals.c
--- GConf-2.2.0.old/gconf/gconf-internals.c	Sun Nov  3 00:36:10 2002
+++ GConf-2.2.0.new/gconf/gconf-internals.c	Sun Mar  9 09:03:04 2003
@@ -2573,6 +2573,9 @@
         }
     }
 
+  /* needed on Darwin */
+  fsync (lock->lock_fd);
+
   return lock;
 }
 
