diff -Naur gnome-themes-2.2.old/po/Makefile.in.in gnome-themes-2.2.new/po/Makefile.in.in
--- gnome-themes-2.2.old/po/Makefile.in.in	Tue Feb  4 17:08:32 2003
+++ gnome-themes-2.2.new/po/Makefile.in.in	Mon Mar 10 23:23:22 2003
@@ -28,7 +28,7 @@
 exec_prefix = @exec_prefix@
 datadir = @datadir@
 libdir = @libdir@
-localedir = $(libdir)/locale
+localedir = $(datadir)/locale
 gnulocaledir = $(datadir)/locale
 gettextsrcdir = $(datadir)/glib-2.0/gettext/po
 subdir = po
