diff -Naur gnome-icon-theme-1.0.0.old/po/Makefile.in.in gnome-icon-theme-1.0.0.new/po/Makefile.in.in
--- gnome-icon-theme-1.0.0.old/po/Makefile.in.in	Tue Jan  7 07:25:15 2003
+++ gnome-icon-theme-1.0.0.new/po/Makefile.in.in	Mon Mar 10 20:59:02 2003
@@ -28,7 +28,7 @@
 exec_prefix = @exec_prefix@
 datadir = @datadir@
 libdir = @libdir@
-localedir = $(libdir)/locale
+localedir = $(datadir)/locale
 gnulocaledir = $(datadir)/locale
 gettextsrcdir = $(datadir)/glib-2.0/gettext/po
 subdir = po
