diff -Naur gnome-applets-2.2.0.old/configure gnome-applets-2.2.0.new/configure
--- gnome-applets-2.2.0.old/configure	Mon Jan 27 02:40:12 2003
+++ gnome-applets-2.2.0.new/configure	Tue Mar 11 18:31:54 2003
@@ -12467,6 +12467,9 @@
   *solaris*)
 	build_battstat_applet=no
 	;;
+  *darwin*)
+	build_battstat_applet=no
+	;;
 esac
 
 
diff -Naur gnome-applets-2.2.0.old/gkb-new/preset.c gnome-applets-2.2.0.new/gkb-new/preset.c
--- gnome-applets-2.2.0.old/gkb-new/preset.c	Mon Jan 27 02:39:38 2003
+++ gnome-applets-2.2.0.new/gkb-new/preset.c	Tue Mar 11 18:31:55 2003
@@ -29,6 +29,7 @@
  * USA
  */
 
+#include <sys/types.h>
 #include <dirent.h>
 #include "gkb.h"
 
diff -Naur gnome-applets-2.2.0.old/multiload/Makefile.in gnome-applets-2.2.0.new/multiload/Makefile.in
--- gnome-applets-2.2.0.old/multiload/Makefile.in	Mon Jan 27 02:44:03 2003
+++ gnome-applets-2.2.0.new/multiload/Makefile.in	Tue Mar 11 18:50:05 2003
@@ -173,7 +173,8 @@
 multiload_applet_2_LDADD = \
 	$(top_builddir)/screen-exec/libscreen-exec.la	\
 				$(GNOME_APPLETS_LIBS) 	\
-				$(GTOP_APPLETS_LIBS)
+				$(GTOP_APPLETS_LIBS)	\
+				$(X_LIBS)
 
 
 pixmaps = gnome-cpu.png
diff -Naur gnome-applets-2.2.0.old/omf-install/Makefile.in gnome-applets-2.2.0.new/omf-install/Makefile.in
--- gnome-applets-2.2.0.old/omf-install/Makefile.in	Mon Jan 27 02:44:52 2003
+++ gnome-applets-2.2.0.new/omf-install/Makefile.in	Tue Mar 11 18:31:55 2003
@@ -265,7 +265,7 @@
 	-for file in $(srcdir)/*.omf; do \
 		$(INSTALL_DATA) $(srcdir)/$$file $(DESTDIR)$(omf_dest_dir); \
 	done
-	-scrollkeeper-update -p $(scrollkeeper_localstate_dir) 
+#	-scrollkeeper-update -p $(scrollkeeper_localstate_dir) 
 
 uninstall-local:
 	-for file in $(srcdir)/*.omf; do \
