diff -Naur libwnck-2.2.0.old/libwnck/Makefile.in libwnck-2.2.0.new/libwnck/Makefile.in
--- libwnck-2.2.0.old/libwnck/Makefile.in	Tue Jan 21 12:11:07 2003
+++ libwnck-2.2.0.new/libwnck/Makefile.in	Mon Mar 10 13:14:46 2003
@@ -140,9 +140,9 @@
 libwnck_1_la_LIBADD =  	@LIBWNCK_LIBS@
 
 
-test_wnck_LDADD = @LIBWNCK_LIBS@ libwnck-1.la
-test_tasklist_LDADD = @LIBWNCK_LIBS@ libwnck-1.la
-test_pager_LDADD = @LIBWNCK_LIBS@ libwnck-1.la
+test_wnck_LDADD = libwnck-1.la @LIBWNCK_LIBS@
+test_tasklist_LDADD = libwnck-1.la @LIBWNCK_LIBS@
+test_pager_LDADD = libwnck-1.la @LIBWNCK_LIBS@
 
 IMAGES = default_icon.png stock_delete.png stock_maximize.png stock_minimize.png
 VARIABLES = default_icon_data $(srcdir)/default_icon.png 			stock_delete_data $(srcdir)/stock_delete.png 			stock_maximize_data $(srcdir)/stock_maximize.png 		stock_minimize_data $(srcdir)/stock_minimize.png
diff -Naur libwnck-2.2.0.old/ltmain.sh libwnck-2.2.0.new/ltmain.sh
--- libwnck-2.2.0.old/ltmain.sh	Tue Jan 21 12:10:02 2003
+++ libwnck-2.2.0.new/ltmain.sh	Mon Mar 10 13:13:51 2003
@@ -4047,10 +4047,10 @@
 
 # Directory that this library needs to be installed in:
 libdir='$install_libdir'"
-	  if test "$installed" = no && test $need_relink = yes; then
-	    $echo >> $output "\
-relink_command=\"$relink_command\""
-	  fi
+#	  if test "$installed" = no && test $need_relink = yes; then
+#	    $echo >> $output "\
+#relink_command=\"$relink_command\""
+#	  fi
 	done
       fi
 
