diff -Naur pango-1.2.1.old/Makefile.in pango-1.2.1.new/Makefile.in
--- pango-1.2.1.old/Makefile.in	Thu Jan 30 02:59:20 2003
+++ pango-1.2.1.new/Makefile.in	Wed Mar 12 23:08:46 2003
@@ -118,7 +118,7 @@
 XFT_LIBS = @XFT_LIBS@
 ms_librarian = @ms_librarian@
 
-SUBDIRS = pango modules examples docs tools tests
+SUBDIRS = pango modules docs tools
 
 EXTRA_DIST = \
 	pango.spec.in	\
diff -Naur pango-1.2.1.old/configure pango-1.2.1.new/configure
--- pango-1.2.1.old/configure	Thu Jan 30 02:53:01 2003
+++ pango-1.2.1.new/configure	Wed Mar 12 23:08:46 2003
@@ -9122,7 +9122,7 @@
 
   if $have_freetype ; then
     pango_save_cppflags="$CPPFLAGS"
-    CPPFLAGS="$CPPFLAGS $FREETYPE_CFLAGS"
+    CPPFLAGS="$FREETYPE_CFLAGS $CPPFLAGS"
 
     echo "$as_me:$LINENO: checking For sufficiently new FreeType (at least 2.0.1)" >&5
 echo $ECHO_N "checking For sufficiently new FreeType (at least 2.0.1)... $ECHO_C" >&6
@@ -9177,7 +9177,7 @@
 
 ## see which freetype we have, 2.0.3 vs. 2.0.4
 pango_save_cppflags="$CPPFLAGS"
-CPPFLAGS="$CPPFLAGS $FREETYPE_CFLAGS"
+CPPFLAGS="$FREETYPE_CFLAGS $CPPFLAGS"
 if test "${ac_cv_header_freetype_internal_tterrors_h+set}" = set; then
   echo "$as_me:$LINENO: checking for freetype/internal/tterrors.h" >&5
 echo $ECHO_N "checking for freetype/internal/tterrors.h... $ECHO_C" >&6
diff -Naur pango-1.2.1.old/ltmain.sh pango-1.2.1.new/ltmain.sh
--- pango-1.2.1.old/ltmain.sh	Sat Nov  9 05:40:11 2002
+++ pango-1.2.1.new/ltmain.sh	Wed Mar 12 23:08:46 2003
@@ -4047,10 +4047,10 @@
 
 # Directory that this library needs to be installed in:
 libdir='$install_libdir'"
-	  if test "$installed" = no && test $need_relink = yes; then
-	    $echo >> $output "\
-relink_command=\"$relink_command\""
-	  fi
+#	  if test "$installed" = no && test $need_relink = yes; then
+#	    $echo >> $output "\
+#relink_command=\"$relink_command\""
+#	  fi
 	done
       fi
 
diff -Naur pango-1.2.1.old/modules/arabic/Makefile.in pango-1.2.1.new/modules/arabic/Makefile.in
--- pango-1.2.1.old/modules/arabic/Makefile.in	Thu Jan 30 02:59:24 2003
+++ pango-1.2.1.new/modules/arabic/Makefile.in	Thu Mar 13 00:10:03 2003
@@ -120,8 +120,8 @@
 
 pangolibs = $(top_builddir)/pango/libpango-$(PANGO_API_VERSION).la $(FRIBIDI_LIBS) $(GLIB_LIBS)
 pangoxlibs = $(top_builddir)/pango/libpangox-$(PANGO_API_VERSION).la $(X_LIBS) $(pangolibs)
-pangoxftlibs = $(top_builddir)/pango/libpangoxft-$(PANGO_API_VERSION).la $(XFT_LIBS) $(pangolibs)
-pangoft2libs = $(top_builddir)/pango/libpangoft2-$(PANGO_API_VERSION).la $(FREETYPE_LIBS) $(pangolibs)
+pangoxftlibs = $(XFTLIBS) $(top_builddir)/pango/libpangoxft-$(PANGO_API_VERSION).la $(pangolibs)
+pangoft2libs = $(FREETYPE_LIBS) $(top_builddir)/pango/libpangoft2-$(PANGO_API_VERSION).la $(pangolibs)
 @HAVE_XFT_TRUE@@INCLUDE_ARABIC_XFT_TRUE@XFT_MODULES = 
 @HAVE_XFT_TRUE@@INCLUDE_ARABIC_XFT_FALSE@XFT_MODULES = @HAVE_XFT_TRUE@@INCLUDE_ARABIC_XFT_FALSE@pango-arabic-xft.la
 @HAVE_XFT_FALSE@XFT_MODULES = 
diff -Naur pango-1.2.1.old/modules/basic/Makefile.in pango-1.2.1.new/modules/basic/Makefile.in
--- pango-1.2.1.old/modules/basic/Makefile.in	Thu Jan 30 02:59:25 2003
+++ pango-1.2.1.new/modules/basic/Makefile.in	Thu Mar 13 00:13:03 2003
@@ -122,8 +122,8 @@
 
 pangolibs = $(top_builddir)/pango/libpango-$(PANGO_API_VERSION).la $(FRIBIDI_LIBS) $(GLIB_LIBS)
 pangoxlibs = $(top_builddir)/pango/libpangox-$(PANGO_API_VERSION).la $(X_LIBS) $(pangolibs)
-pangoxftlibs = $(top_builddir)/pango/libpangoxft-$(PANGO_API_VERSION).la $(XFT_LIBS) $(pangolibs)
-pangoft2libs = $(top_builddir)/pango/libpangoft2-$(PANGO_API_VERSION).la $(FREETYPE_LIBS) $(pangolibs)
+pangoxftlibs = $(XFT_LIBS) $(top_builddir)/pango/libpangoxft-$(PANGO_API_VERSION).la $(pangolibs)
+pangoft2libs = $(FREETYPE_LIBS) $(top_builddir)/pango/libpangoft2-$(PANGO_API_VERSION).la $(pangolibs)
 pangowin32libs = $(top_builddir)/pango/libpangowin32-$(PANGO_API_VERSION).la $(pangolibs)
 @HAVE_XFT_TRUE@@INCLUDE_BASIC_XFT_TRUE@XFT_INCLUDED = @HAVE_XFT_TRUE@@INCLUDE_BASIC_XFT_TRUE@libpango-basic-xft.la
 @HAVE_XFT_TRUE@@INCLUDE_BASIC_XFT_FALSE@XFT_INCLUDED = 
diff -Naur pango-1.2.1.old/modules/hangul/Makefile.in pango-1.2.1.new/modules/hangul/Makefile.in
--- pango-1.2.1.old/modules/hangul/Makefile.in	Thu Jan 30 02:59:26 2003
+++ pango-1.2.1.new/modules/hangul/Makefile.in	Thu Mar 13 00:13:15 2003
@@ -120,8 +120,8 @@
 
 pangolibs = $(top_builddir)/pango/libpango-$(PANGO_API_VERSION).la $(FRIBIDI_LIBS) $(GLIB_LIBS)
 pangoxlibs = $(top_builddir)/pango/libpangox-$(PANGO_API_VERSION).la $(X_LIBS) $(pangolibs)
-pangoxftlibs = $(top_builddir)/pango/libpangoxft-$(PANGO_API_VERSION).la $(XFT_LIBS) $(pangolibs)
-pangoft2libs = $(top_builddir)/pango/libpangoft2-$(PANGO_API_VERSION).la $(FREETYPE_LIBS) $(pangolibs)
+pangoxftlibs = $(XFT_LIBS) $(top_builddir)/pango/libpangoxft-$(PANGO_API_VERSION).la $(pangolibs)
+pangoft2libs = $(FREETYPE_LIBS) $(top_builddir)/pango/libpangoft2-$(PANGO_API_VERSION).la $(pangolibs)
 
 x_sources = hangul-x.c hangul-defs.h
 xft_sources = hangul-xft.c hangul-defs.h
diff -Naur pango-1.2.1.old/modules/hebrew/Makefile.in pango-1.2.1.new/modules/hebrew/Makefile.in
--- pango-1.2.1.old/modules/hebrew/Makefile.in	Thu Jan 30 02:59:29 2003
+++ pango-1.2.1.new/modules/hebrew/Makefile.in	Thu Mar 13 00:43:55 2003
@@ -125,8 +125,8 @@
 
 pangolibs = $(top_builddir)/pango/libpango-$(PANGO_API_VERSION).la $(FRIBIDI_LIBS) $(GLIB_LIBS)
 pangoxlibs = $(top_builddir)/pango/libpangox-$(PANGO_API_VERSION).la $(X_LIBS) $(pangolibs)
-pangoxftlibs = $(top_builddir)/pango/libpangoxft-$(PANGO_API_VERSION).la $(XFT_LIBS) $(pangolibs)
-pangoft2libs = $(top_builddir)/pango/libpangoft2-$(PANGO_API_VERSION).la $(FREETYPE_LIBS) $(pangolibs)
+pangoxftlibs = $(XFT_LIBS) $(top_builddir)/pango/libpangoxft-$(PANGO_API_VERSION).la $(pangolibs)
+pangoft2libs = $(FREETYPE_LIBS) $(top_builddir)/pango/libpangoft2-$(PANGO_API_VERSION).la $(pangolibs)
 @HAVE_XFT_TRUE@@INCLUDE_BASIC_XFT_TRUE@XFT_INCLUDED = @HAVE_XFT_TRUE@@INCLUDE_BASIC_XFT_TRUE@libpango-hebrew-xft.la
 @HAVE_XFT_TRUE@@INCLUDE_BASIC_XFT_FALSE@XFT_INCLUDED = 
 @HAVE_XFT_FALSE@XFT_INCLUDED = 
@@ -136,7 +136,7 @@
 @HAVE_XFT_TRUE@@INCLUDE_BASIC_XFT_TRUE@XFT_PREFIX = @HAVE_XFT_TRUE@@INCLUDE_BASIC_XFT_TRUE@-DXFT_MODULE_PREFIX
 @HAVE_XFT_TRUE@@INCLUDE_BASIC_XFT_FALSE@XFT_PREFIX = 
 @HAVE_XFT_FALSE@XFT_PREFIX = 
-@HAVE_XFT_TRUE@@INCLUDE_BASIC_XFT_FALSE@hebrew_xft_libadd = @HAVE_XFT_TRUE@@INCLUDE_BASIC_XFT_FALSE@$(pangoxlibs) $(pangoxftlibs)
+@HAVE_XFT_TRUE@@INCLUDE_BASIC_XFT_FALSE@hebrew_xft_libadd = @HAVE_XFT_TRUE@@INCLUDE_BASIC_XFT_FALSE@$(pangoxftlibs) $(pangoxlibs)
 @HAVE_X_TRUE@@INCLUDE_BASIC_X_TRUE@X_INCLUDED = @HAVE_X_TRUE@@INCLUDE_BASIC_X_TRUE@libpango-hebrew-x.la
 @HAVE_X_TRUE@@INCLUDE_BASIC_X_FALSE@X_INCLUDED = 
 @HAVE_X_FALSE@X_INCLUDED = 
diff -Naur pango-1.2.1.old/modules/indic/Makefile.in pango-1.2.1.new/modules/indic/Makefile.in
--- pango-1.2.1.old/modules/indic/Makefile.in	Thu Jan 30 02:59:26 2003
+++ pango-1.2.1.new/modules/indic/Makefile.in	Thu Mar 13 00:13:32 2003
@@ -136,8 +136,8 @@
 
 pangolibs = $(top_builddir)/pango/libpango-$(PANGO_API_VERSION).la $(FRIBIDI_LIBS) $(GLIB_LIBS)
 pangoxlibs = $(top_builddir)/pango/libpangox-$(PANGO_API_VERSION).la $(X_LIBS) $(pangolibs)
-pangoxftlibs = $(top_builddir)/pango/libpangoxft-$(PANGO_API_VERSION).la $(XFT_LIBS) $(pangolibs)
-pangoft2libs = $(top_builddir)/pango/libpangoft2-$(PANGO_API_VERSION).la $(FREETYPE_LIBS) $(pangolibs)
+pangoxftlibs = $(XFT_LIBS) $(top_builddir)/pango/libpangoxft-$(PANGO_API_VERSION).la $(pangolibs)
+pangoft2libs = $(FREETYPE_LIBS) $(top_builddir)/pango/libpangoft2-$(PANGO_API_VERSION).la $(pangolibs)
 @HAVE_XFT_TRUE@@INCLUDE_INDIC_XFT_TRUE@indic_xft_inst = 
 @HAVE_XFT_TRUE@@INCLUDE_INDIC_XFT_FALSE@indic_xft_inst = @HAVE_XFT_TRUE@@INCLUDE_INDIC_XFT_FALSE@pango-indic-xft.la
 @HAVE_XFT_FALSE@indic_xft_inst = 
diff -Naur pango-1.2.1.old/modules/tamil/Makefile.in pango-1.2.1.new/modules/tamil/Makefile.in
--- pango-1.2.1.old/modules/tamil/Makefile.in	Thu Jan 30 02:59:27 2003
+++ pango-1.2.1.new/modules/tamil/Makefile.in	Thu Mar 13 00:13:41 2003
@@ -120,7 +120,7 @@
 
 pangolibs = $(top_builddir)/pango/libpango-$(PANGO_API_VERSION).la $(FRIBIDI_LIBS) $(GLIB_LIBS)
 pangoxlibs = $(top_builddir)/pango/libpangox-$(PANGO_API_VERSION).la $(X_LIBS) $(pangolibs)
-pangoft2libs = $(top_builddir)/pango/libpangoft2-$(PANGO_API_VERSION).la $(FREETYPE_LIBS) $(pangolibs)
+pangoft2libs = $(FREETYPE_LIBS) $(top_builddir)/pango/libpangoft2-$(PANGO_API_VERSION).la $(pangolibs)
 
 x_sources = \
 	tamil-x.c \
diff -Naur pango-1.2.1.old/modules/thai/Makefile.in pango-1.2.1.new/modules/thai/Makefile.in
--- pango-1.2.1.old/modules/thai/Makefile.in	Thu Jan 30 02:59:28 2003
+++ pango-1.2.1.new/modules/thai/Makefile.in	Thu Mar 13 00:13:49 2003
@@ -124,8 +124,8 @@
 
 pangolibs = $(top_builddir)/pango/libpango-$(PANGO_API_VERSION).la $(FRIBIDI_LIBS) $(GLIB_LIBS)
 pangoxlibs = $(top_builddir)/pango/libpangox-$(PANGO_API_VERSION).la $(X_LIBS) $(pangolibs)
-pangoxftlibs = $(top_builddir)/pango/libpangoxft-$(PANGO_API_VERSION).la $(XFT_LIBS) $(pangolibs)
-pangoft2libs = $(top_builddir)/pango/libpangoft2-$(PANGO_API_VERSION).la $(FREETYPE_LIBS) $(pangolibs)
+pangoxftlibs = $(XFT_LIBS) $(top_builddir)/pango/libpangoxft-$(PANGO_API_VERSION).la $(pangolibs)
+pangoft2libs = $(FREETYPE_LIBS) $(top_builddir)/pango/libpangoft2-$(PANGO_API_VERSION).la $(pangolibs)
 @HAVE_XFT_TRUE@@INCLUDE_THAI_XFT_TRUE@XFT_MODULES = 
 @HAVE_XFT_TRUE@@INCLUDE_THAI_XFT_FALSE@XFT_MODULES = @HAVE_XFT_TRUE@@INCLUDE_THAI_XFT_FALSE@pango-thai-xft.la
 @HAVE_XFT_FALSE@XFT_MODULES = 
diff -Naur pango-1.2.1.old/pango/Makefile.in pango-1.2.1.new/pango/Makefile.in
--- pango-1.2.1.old/pango/Makefile.in	Thu Jan 30 02:59:21 2003
+++ pango-1.2.1.new/pango/Makefile.in	Thu Mar 13 01:04:20 2003
@@ -138,10 +138,10 @@
 	-DLIBDIR=\"$(libdir)\"				\
 	-DMODULE_VERSION=\"$(PANGO_MODULE_VERSION)\"	\
 	-DG_DISABLE_DEPRECATED				\
-	$(GLIB_CFLAGS)					\
 	$(XFT_CFLAGS)					\
 	$(FONTCONFIG_CFLAGS)				\
 	$(FREETYPE_CFLAGS)				\
+	$(GLIB_CFLAGS)					\
 	$(X_CFLAGS)					\
 	-I$(top_srcdir)
 
@@ -309,10 +309,10 @@
 pango_querymodules_SOURCES = \
 	querymodules.c
 
-@HAVE_XFT_TRUE@pango_querymodules_LDADD = @HAVE_XFT_TRUE@$(FT2_BUILD_LIBS) $(X_BUILD_LIBS) libpangoxft-$(PANGO_API_VERSION).la $(GLIB_LIBS) $(FRIBIDI_LIBS) $(XFT_LIBS) $(FONTCONFIG_LIBS)
-@HAVE_XFT_FALSE@pango_querymodules_LDADD = @HAVE_XFT_FALSE@$(FT2_BUILD_LIBS) $(X_BUILD_LIBS) $(WIN32_BUILD_LIBS) $(GLIB_LIBS) $(FRIBIDI_LIBS) $(X_LIBS) $(FONTCONFIG_LIBS) $(X_EXTRA_LIBS)
-@HAVE_XFT_TRUE@pango_querymodules_DEPENDENCIES = @HAVE_XFT_TRUE@$(FT2_BUILD_LIBS) $(X_BUILD_LIBS) libpangoxft-$(PANGO_API_VERSION).la libpango-$(PANGO_API_VERSION).la
-@HAVE_XFT_FALSE@pango_querymodules_DEPENDENCIES = @HAVE_XFT_FALSE@$(FT2_BUILD_LIBS) $(X_BUILD_LIBS) $(WIN32_BUILD_LIBS) libpango-$(PANGO_API_VERSION).la
+@HAVE_XFT_TRUE@pango_querymodules_LDADD = $(X_BUILD_LIBS) libpangoxft-$(PANGO_API_VERSION).la $(GLIB_LIBS) $(FRIBIDI_LIBS)
+@HAVE_XFT_FALSE@pango_querymodules_LDADD = $(X_BUILD_LIBS) $(WIN32_BUILD_LIBS) $(GLIB_LIBS) $(FRIBIDI_LIBS)
+@HAVE_XFT_TRUE@pango_querymodules_DEPENDENCIES = $(X_BUILD_LIBS) libpangoxft-$(PANGO_API_VERSION).la libpango-$(PANGO_API_VERSION).la
+@HAVE_XFT_FALSE@pango_querymodules_DEPENDENCIES = $(X_BUILD_LIBS) $(WIN32_BUILD_LIBS) libpango-$(PANGO_API_VERSION).la
 
 EXTRA_DIST = \
 	$(pango_built_headers)			\
diff -Naur pango-1.2.1.old/update-pango-modules pango-1.2.1.new/update-pango-modules
--- pango-1.2.1.old/update-pango-modules	Thu Jan  1 09:00:00 1970
+++ pango-1.2.1.new/update-pango-modules	Wed Mar 12 23:08:46 2003
@@ -0,0 +1,20 @@
+#!/bin/sh -e
+#
+# update-pango-modules
+#
+#   Package which install pango module(s) should call update-pango-modules in
+#   it's PostInstScript/PostRmScript to update pango.module file.
+#
+
+CONFDIR=@PREFIX@/etc/pango
+CONFFILE=$CONFDIR/pango.modules
+
+mkdir -p $CONFDIR
+
+TMPFILE=`tempfile`
+pango-querymodules >$TMPFILE
+
+mv -f $TMPFILE $CONFFILE
+chmod 0644 $CONFFILE
+
+exit 0
