diff -Naur texlive-20080816-source.orig/texk/web2c/doc/Makefile.in texlive-20080816-source/texk/web2c/doc/Makefile.in
--- texlive-20080816-source.orig/texk/web2c/doc/Makefile.in	2008-04-30 07:40:29.000000000 +0900
+++ texlive-20080816-source/texk/web2c/doc/Makefile.in	2009-05-04 00:26:14.000000000 +0900
@@ -54,7 +54,7 @@
 #
 	$(POST_INSTALL)
 	if $(SHELL) -c 'install-info --version' >/dev/null 2>&1; then \
-	  install-info --info-dir=$(infodir) $(infodir)/web2c.info; \
+	  install-info --info-dir=$(infodir) $(infodir)/web2c.info || true; \
 	else true; fi
 
 check: all
diff -udNr texlive-20080816-source.ep-tl/texk/web2c/ptexdir/ptex.mk texlive-20080816-source/texk/web2c/ptexdir/ptex.mk
--- texlive-20080816-source.ep-tl/texk/web2c/ptexdir/ptex.mk	2009-02-23 06:52:53.618335326 +0900
+++ texlive-20080816-source/texk/web2c/ptexdir/ptex.mk	2009-02-23 06:53:07.534346816 +0900
@@ -6,7 +6,7 @@
 Makefile: $(srcdir)/ptexdir/ptex.mk
 
 # We build ptex.
-ptex = @JPTEX@ ptex ptftopl ppltotf pdvitype
+ptex = @JPTEX@ ptex ptftopl ppltotf pdvitype jbibtex
 
 # Extract ptex version
 ptexdir/ptex.version: ptexdir/ptex-base.ch
@@ -23,8 +23,13 @@
 # Making ptex.
 ptex: $(ptex_o) $(plib) $(ptexenc)
 	$(kpathsea_link) $(ptex_o) $(plib) $(ptexenc) $(socketlibs) $(LOADLIBES)
+jbibtex: jbibtex.o  $(plib) $(ptexenc)
+	$(kpathsea_link) jbibtex.o $(plib) $(ptexenc) $(socketlibs) $(LOADLIBES)
 
 # C file dependencies
+jbibtex.c: jbibtex.p web2c/cvtbib.sed 
+	$(web2c) jbibtex
+	cat $(srcdir)/ptexdir/jbibextra.c >> jbibtex.c 
 $(ptex_c) ptexcoerce.h ptexd.h: ptex.p $(web2c_texmf)
 	$(web2c) ptex
 ptexextra.c: ptexdir/ptexextra.h lib/texmfmp.c
@@ -36,6 +41,8 @@
 # Tangling
 ptex.p ptex.pool: tangle ptex.web ptex.ch
 	$(TANGLE) ptex.web ptex.ch
+jbibtex.p jbibtex.pool: tangle jbibtex.web
+	$(TANGLE) jbibtex.web
 
 # Generation of the web and ch file.
 #   Sources for ptex.web:
@@ -45,7 +52,11 @@
 ptex_ch_srcs = ptex.web \
   $(srcdir)/ptexdir/ptex-base.ch \
   $(srcdir)/ptexdir/ptex-include.ch
+jbibtex_srcs = $(srcdir)/ptexdir/jbibtex.ch $(srcdir)/bibtex.web
+
 #   Rules:
+jbibtex.web: tie $(srcdir)/ptexdir/jbibtex.ch $(srcdir)/bibtex.web
+	$(TIE) -m jbibtex.web $(srcdir)/bibtex.web $(srcdir)/ptexdir/jbibtex.ch
 ptex.web: tie ptexdir/ptex.mk $(ptex_web_srcs)
 	$(TIE) -m ptex.web $(ptex_web_srcs)
 ptex.ch: $(ptex_ch_srcs)
diff -udNr texlive-20080816-source.ep-tl/texk/web2c/web2c/convert texlive-20080816-source/texk/web2c/web2c/convert
--- texlive-20080816-source.ep-tl/texk/web2c/web2c/convert	2009-02-23 06:52:53.630335525 +0900
+++ texlive-20080816-source/texk/web2c/web2c/convert	2009-02-23 06:53:07.534346816 +0900
@@ -65,6 +65,11 @@
   bibtex)
     midcmd="| sed -f $srcdir/web2c/cvtbib.sed"
     ;;
+  jbibtex)
+    more_defines="$srcdir/ptexdir/jbibtex.defines"
+    hfile=$srcdir/ptexdir/jbibextra.h
+    midcmd="| sed -f $srcdir/web2c/cvtbib.sed"
+    ;;
 
   mf) # MF and MP are almost the same.
     more_defines="$srcdir/web2c/texmf.defines $srcdir/web2c/mfmp.defines"
@@ -162,6 +167,10 @@
     sed -e 's/(buftype)//g' -e 's/(pdstype)//g' <bibtex.h >xbibtex.h
     mv xbibtex.h bibtex.h
     ;;
+  jbibtex)
+    sed -e 's/(buftype)//g' -e 's/(pdstype)//g' <jbibtex.h >xjbibtex.h
+    mv xjbibtex.h jbibtex.h
+    ;;
   tex|eomega|etex|luatex|pdftex|pdfetex|omega|aleph|mf|mp|xetex|ptex)
     sleep 2 # so timestamps are definitely later, to avoid make weirdness
     cat ${basefile}coerce.h $srcdir/web2c/coerce.h >x${basefile}coerce.h
diff -Naur texlive-20080816-source.orig/texk/tetex/fmtutil.cnf texlive-20080816-source/texk/tetex/fmtutil.cnf
--- texlive-20080816-source.orig/bin.duplex	2009-05-11 00:38:42.000000000 +0900
+++ texlive-20080816-source/bin.duplex	2009-05-11 00:36:56.000000000 +0900
@@ -0,0 +1,43 @@
+ aleph 
+ bibtex
+ ctangle
+ ctie
+ cweave
+ dmp
+ dvicopy
+ dvitomp
+ dvitype
+ gftodvi
+ gftopk
+ gftype
+ makempx
+ mf
+ mft
+ mkocp
+ mkofm
+ mpost
+ mpto
+ newer
+ odvicopy
+ odvitype
+ omega
+ omfonts
+ otangle
+ otp2ocp
+ outocp
+ patgen
+ pdftex
+ pdftosrc
+ pktogf
+ pktype
+ pltotf
+ pooltype
+ synctex
+ tangle
+ tex
+ tftopl
+ tie
+ ttf2afm
+ vftovp
+ vptovf
+ weave
