diff -uNr kdelibs-3.3.91/admin/ltmain.sh kdelibs-3.3.91-new/admin/ltmain.sh
--- kdelibs-3.3.91/admin/ltmain.sh	2005-01-03 09:53:16.000000000 -0500
+++ kdelibs-3.3.91-new/admin/ltmain.sh	2005-01-31 19:22:52.000000000 -0500
@@ -4453,6 +4453,7 @@
 	# We should set the shlibpath_var
 	rpath=
 	for dir in $temp_rpath; do
+     if test "$dir" != "@FINKPREFIX@/lib"; then
 	  case $dir in
 	  [\\/]* | [A-Za-z]:[\\/]*)
 	    # Absolute path.
@@ -4463,6 +4464,7 @@
 	    rpath="$rpath\$thisdir/$dir:"
 	    ;;
 	  esac
+     fi
 	done
 	temp_rpath="$rpath"
       fi
diff -uNr kdelibs-3.3.91/configure kdelibs-3.3.91-new/configure
--- kdelibs-3.3.91/configure	2005-01-07 07:49:25.000000000 -0500
+++ kdelibs-3.3.91-new/configure	2005-01-31 19:37:41.000000000 -0500
@@ -8528,6 +8528,19 @@
     lt_cv_sys_max_cmd_len=8192;
     ;;
 
+  netbsd* | freebsd* | openbsd* | darwin* )
+    # This has been around since 386BSD, at least.  Likely further.
+    if test -x /sbin/sysctl; then
+      lt_cv_sys_max_cmd_len=`/sbin/sysctl -n kern.argmax`
+    elif test -x /usr/sbin/sysctl; then
+      lt_cv_sys_max_cmd_len=`/usr/sbin/sysctl -n kern.argmax`
+    else
+      lt_cv_sys_max_cmd_len=65536 # usable default for *BSD
+    fi
+    # And add a safety zone
+    lt_cv_sys_max_cmd_len=`expr $lt_cv_sys_max_cmd_len \/ 4`
+    ;;
+
  *)
     # If test is not a shell built-in, we'll probably end up computing a
     # maximum length that is only half of the actual maximum length, but
@@ -10171,10 +10184,21 @@
       archive_cmds_need_lc=no
       case "$host_os" in
       rhapsody* | darwin1.[012])
-	allow_undefined_flag='-undefined suppress'
+	allow_undefined_flag='-Wl,-undefined -Wl,suppress'
 	;;
       *) # Darwin 1.3 on
-	test -z ${LD_TWOLEVEL_NAMESPACE} && allow_undefined_flag='-flat_namespace -undefined suppress'
+	if test -z ${MACOSX_DEPLOYMENT_TARGET} ; then
+	  allow_undefined_flag='-Wl,-flat_namespace -Wl,-undefined -Wl,suppress'
+	else
+	  case ${MACOSX_DEPLOYMENT_TARGET} in
+	    10.[012])
+	      allow_undefined_flag='-Wl,-flat_namespace -Wl,-undefined -Wl,suppress'
+	      ;;
+	    10.*)
+	      allow_undefined_flag='-Wl,-undefined -Wl,dynamic_lookup'
+	      ;;
+	  esac
+	fi
 	;;
       esac
     	lt_int_apple_cc_single_mod=no
@@ -12024,7 +12048,18 @@
       allow_undefined_flag='-undefined suppress'
       ;;
     *) # Darwin 1.3 on
-      test -z ${LD_TWOLEVEL_NAMESPACE} && allow_undefined_flag='-flat_namespace -undefined suppress'
+	if test -z ${MACOSX_DEPLOYMENT_TARGET} ; then
+	  allow_undefined_flag='-Wl,-flat_namespace -Wl,-undefined -Wl,suppress'
+	else
+	  case ${MACOSX_DEPLOYMENT_TARGET} in
+	    10.[012])
+	      allow_undefined_flag='-Wl,-flat_namespace -Wl,-undefined -Wl,suppress'
+	      ;;
+	    10.*)
+	      allow_undefined_flag='-Wl,-undefined -Wl,dynamic_lookup'
+	      ;;
+	  esac
+	fi
       ;;
     esac
     output_verbose_link_cmd='echo'
@@ -13075,7 +13110,18 @@
       allow_undefined_flag_CXX='-undefined suppress'
       ;;
     *) # Darwin 1.3 on
-      test -z ${LD_TWOLEVEL_NAMESPACE} && allow_undefined_flag_CXX='-flat_namespace -undefined suppress'
+	if test -z ${MACOSX_DEPLOYMENT_TARGET} ; then
+	  allow_undefined_flag_CXX='-Wl,-flat_namespace -Wl,-undefined -Wl,suppress'
+	else
+	  case ${MACOSX_DEPLOYMENT_TARGET} in
+	    10.[012])
+	      allow_undefined_flag_CXX='-Wl,-flat_namespace -Wl,-undefined -Wl,suppress'
+	      ;;
+	    10.*)
+	      allow_undefined_flag_CXX='-Wl,-undefined -Wl,dynamic_lookup'
+	      ;;
+	  esac
+	fi
       ;;
     esac
     lt_int_apple_cc_single_mod=no
@@ -16898,7 +16944,18 @@
 	allow_undefined_flag_F77='-undefined suppress'
 	;;
       *) # Darwin 1.3 on
-	test -z ${LD_TWOLEVEL_NAMESPACE} && allow_undefined_flag_F77='-flat_namespace -undefined suppress'
+	if test -z ${MACOSX_DEPLOYMENT_TARGET} ; then
+	  allow_undefined_flag_F77='-Wl,-flat_namespace -Wl,-undefined -Wl,suppress'
+	else
+	  case ${MACOSX_DEPLOYMENT_TARGET} in
+	    10.[012])
+	      allow_undefined_flag_F77='-Wl,-flat_namespace -Wl,-undefined -Wl,suppress'
+	      ;;
+	    10.*)
+	      allow_undefined_flag_F77='-Wl,-undefined -Wl,dynamic_lookup'
+	      ;;
+	  esac
+	fi
 	;;
       esac
     	lt_int_apple_cc_single_mod=no
@@ -19180,7 +19237,18 @@
 	allow_undefined_flag_GCJ='-undefined suppress'
 	;;
       *) # Darwin 1.3 on
-	test -z ${LD_TWOLEVEL_NAMESPACE} && allow_undefined_flag_GCJ='-flat_namespace -undefined suppress'
+	if test -z ${MACOSX_DEPLOYMENT_TARGET} ; then
+	  allow_undefined_flag_GCJ='-Wl,-flat_namespace -Wl,-undefined -Wl,suppress'
+	else
+	  case ${MACOSX_DEPLOYMENT_TARGET} in
+	    10.[012])
+	      allow_undefined_flag_GCJ='-Wl,-flat_namespace -Wl,-undefined -Wl,suppress'
+	      ;;
+	    10.*)
+	      allow_undefined_flag_GCJ='-Wl,-undefined -Wl,dynamic_lookup'
+	      ;;
+	  esac
+	fi
 	;;
       esac
     	lt_int_apple_cc_single_mod=no
@@ -23040,6 +23108,9 @@
 	darwin*)
 	  libltdl_cv_sys_dlopen_deplibs=yes
 	  ;;
+	darwin*)
+	  libltdl_cv_sys_dlopen_deplibs=yes
+	  ;;
 	linux*)
 	  libltdl_cv_sys_dlopen_deplibs=yes
 	  ;;
@@ -43996,8 +44067,8 @@
 kde_save_libs="$LIBS"
 LIBS="$all_libraries $USER_LDFLAGS $kde_cups_libs"
 CFLAGS="$CFLAGS $all_includes $USER_INCLUDES"
-echo "$as_me:$LINENO: checking CUPS version >= 1.0120" >&5
-echo $ECHO_N "checking CUPS version >= 1.0120... $ECHO_C" >&6
+echo "$as_me:$LINENO: checking CUPS version >= 1.0121" >&5
+echo $ECHO_N "checking CUPS version >= 1.0121... $ECHO_C" >&6
 kde_save_link=$ac_link
 ac_link='${CC} $CFLAGS -o conftest conftest.c $LIBS 1>&5'
 cat > conftest.c <<EOF
@@ -44007,7 +44078,7 @@
 {
   if (CUPS_VERSION <
 EOF
-echo "1.0120" >> conftest.c
+echo "1.0121" >> conftest.c
 cat >> conftest.c <<EOF
 )
     printf("no\n");
@@ -46595,7 +46666,7 @@
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
   ac_check_lib_save_LIBS=$LIBS
-LIBS="-lqt-mt  $LIBS"
+LIBS="-lqt-mt -lXft $LIBS"
 cat >conftest.$ac_ext <<_ACEOF
 /* confdefs.h.  */
 _ACEOF
@@ -48588,7 +48659,7 @@
 no) echo "$as_me:$LINENO: result: no" >&5
 echo "${ECHO_T}no" >&6 ;;
 framework)
-  GSSAPI_LIBS="-Xlinker -framework -Xlinker GSSAPI"
+  GSSAPI_LIBS="-Xlinker -framework -Xlinker Kerberos"
 
 cat >>confdefs.h <<_ACEOF
 #define HAVE_LIBGSSAPI 1
diff -uNr kdelibs-3.3.91/kdecore/Makefile.in kdelibs-3.3.91-new/kdecore/Makefile.in
--- kdelibs-3.3.91/kdecore/Makefile.in	2005-01-07 07:48:59.000000000 -0500
+++ kdelibs-3.3.91-new/kdecore/Makefile.in	2005-01-31 19:22:52.000000000 -0500
@@ -653,7 +653,7 @@
 xdg_menudir = @xdg_menudir@
 INCLUDES = -I$(top_srcdir)/kdefx -I$(top_srcdir)/kdecore/network $(all_includes) $(LIBART_CFLAGS)
 @include_SVGICONS_TRUE@SVGICONS = svgicons
-@include_SVGICONS_TRUE@SVGICON_LIB = svgicons/libkdesvgicons.la
+@include_SVGICONS_TRUE@SVGICON_LIB = svgicons/libkdesvgicons.la $(LIBZ)
 SUBDIRS = malloc network $(SVGICONS) . kconfig_compiler tests
 AM_LDFLAGS = $(LDFLAGS_AS_NEEDED)
 lib_LTLIBRARIES = libkdefakes.la libkdecore.la
diff -uNr kdelibs-3.3.91/kdecore/kglobalsettings.cpp kdelibs-3.3.91-new/kdecore/kglobalsettings.cpp
--- kdelibs-3.3.91/kdecore/kglobalsettings.cpp	2004-10-26 13:30:49.000000000 -0400
+++ kdelibs-3.3.91-new/kdecore/kglobalsettings.cpp	2005-01-31 19:22:52.000000000 -0500
@@ -460,7 +460,7 @@
       s_desktopPath->append('/');
 
     // Trash Path - TODO remove in KDE4 (kio_trash can't use it for interoperability reasons)
-    *s_trashPath = *s_desktopPath + i18n("Trash") + "/";
+    *s_trashPath = *s_desktopPath + "." + i18n("Trash") + "/";
     *s_trashPath = g.readPathEntry( "Trash" , *s_trashPath);
     *s_trashPath = QDir::cleanDirPath( *s_trashPath );
     if ( !s_trashPath->endsWith("/") )
diff -uNr kdelibs-3.3.91/kdecore/ksocks.cpp kdelibs-3.3.91-new/kdecore/ksocks.cpp
--- kdelibs-3.3.91/kdecore/ksocks.cpp	2004-05-11 04:31:44.000000000 -0400
+++ kdelibs-3.3.91-new/kdecore/ksocks.cpp	2005-01-31 19:22:52.000000000 -0500
@@ -475,7 +475,7 @@
                                                    ksocklen_t addrlen) {
    if (_useSocks && F_connect)
       return (*F_connect)(sockfd, serv_addr, addrlen);
-   else return ::connect(sockfd, (sockaddr*) serv_addr, (socklen_t)addrlen);
+   else return ::connect(sockfd, (sockaddr*) serv_addr, (ksocklen_t)addrlen);
 }
 
 
@@ -498,7 +498,7 @@
    if (_useSocks && F_recvfrom) {
       return (*F_recvfrom)(s, buf, len, flags, from, fromlen);
    } else {
-      socklen_t casted_len = (socklen_t) *fromlen;
+      kde_socklen_t casted_len = (kde_socklen_t) *fromlen;
       int rc = ::recvfrom(s, (char*) buf, len, flags, from, &casted_len);
       *fromlen = casted_len;
       return rc;
@@ -510,7 +510,7 @@
                              const sockaddr *to, ksocklen_t tolen) {
    if (_useSocks && F_sendto)
       return (*F_sendto)(s, msg, len, flags, to, tolen);
-   else return ::sendto(s, (char*) msg, len, flags, to, (socklen_t)tolen);
+   else return ::sendto(s, (char*) msg, len, flags, to, (ksocklen_t)tolen);
 }
 
 
@@ -532,7 +532,7 @@
    if (_useSocks && F_getsockname) {
       return (*F_getsockname)(s, name, namelen);
    } else {
-     socklen_t casted_len = *namelen;
+     kde_socklen_t casted_len = *namelen;
      int rc = ::getsockname(s, name, &casted_len);
      *namelen = casted_len;
      return rc;
@@ -544,7 +544,7 @@
    if (_useSocks && F_getpeername) {
       return (*F_getpeername)(s, name, namelen);
    } else {
-      socklen_t casted_len = *namelen;
+      kde_socklen_t casted_len = *namelen;
       int rc = ::getpeername(s, name, &casted_len);
       *namelen = casted_len;
       return rc;
@@ -556,7 +556,7 @@
    if (_useSocks && F_accept) {
      return (*F_accept)(s, addr, addrlen);
    } else {
-      socklen_t casted_len = *addrlen;
+      kde_socklen_t casted_len = *addrlen;
       int rc = ::accept(s, addr, &casted_len);
       *addrlen = casted_len;
       return rc;
@@ -582,13 +582,13 @@
 int KSocks::bind (int sockfd, const sockaddr *my_addr, ksocklen_t addrlen) {
    if (_useSocks && F_bind)
       return (*F_bind)(sockfd, my_addr, addrlen);
-   else return ::bind(sockfd, my_addr, (socklen_t)addrlen);
+   else return ::bind(sockfd, my_addr, (ksocklen_t)addrlen);
 }
 
 int KSocks::bind (int sockfd, sockaddr *my_addr, ksocklen_t addrlen) {
    if (_useSocks && F_bind)
       return (*F_bind)(sockfd, my_addr, addrlen);
-   else return ::bind(sockfd, my_addr, (socklen_t)addrlen);
+   else return ::bind(sockfd, my_addr, (ksocklen_t)addrlen);
 }
 
 
diff -uNr kdelibs-3.3.91/kdecore/kstandarddirs.cpp kdelibs-3.3.91-new/kdecore/kstandarddirs.cpp
--- kdelibs-3.3.91/kdecore/kstandarddirs.cpp	2005-01-03 09:53:18.000000000 -0500
+++ kdelibs-3.3.91-new/kdecore/kstandarddirs.cpp	2005-01-31 19:22:52.000000000 -0500
@@ -1020,7 +1020,7 @@
     if (!strcmp(type, "data"))
 	return "share/apps/";
     if (!strcmp(type, "html"))
-	return "share/doc/HTML/";
+	return "share/doc/kde/";
     if (!strcmp(type, "icon"))
 	return "share/icons/";
     if (!strcmp(type, "config"))
diff -uNr kdelibs-3.3.91/kdecore/network/kresolvermanager.cpp kdelibs-3.3.91-new/kdecore/network/kresolvermanager.cpp
--- kdelibs-3.3.91/kdecore/network/kresolvermanager.cpp	2004-12-04 13:07:46.000000000 -0500
+++ kdelibs-3.3.91-new/kdecore/network/kresolvermanager.cpp	2005-01-31 19:22:52.000000000 -0500
@@ -31,6 +31,7 @@
 
 #ifdef HAVE_RES_INIT
 # include <sys/stat.h>
+# include <nameser.h>
 # include <resolv.h>
 # include <time.h>
 #endif
diff -uNr kdelibs-3.3.91/kdecore/network/ksocketdevice.cpp kdelibs-3.3.91-new/kdecore/network/ksocketdevice.cpp
--- kdelibs-3.3.91/kdecore/network/ksocketdevice.cpp	2004-12-04 13:07:46.000000000 -0500
+++ kdelibs-3.3.91-new/kdecore/network/ksocketdevice.cpp	2005-01-31 19:22:52.000000000 -0500
@@ -311,7 +311,7 @@
     }
 
   struct sockaddr sa;
-  socklen_t len = sizeof(sa);
+  kde_socklen_t len = sizeof(sa);
   int newfd = kde_accept(m_sockfd, &sa, &len);
   if (newfd == -1)
     {
@@ -385,7 +385,7 @@
 
 static int do_read_common(int sockfd, char *data, Q_ULONG maxlen, KSocketAddress* from, ssize_t &retval, bool peek = false)
 {
-  socklen_t len;
+  kde_socklen_t len;
   if (from)
     {
       from->setLength(len = 128); // arbitrary length
@@ -523,7 +523,7 @@
   if (m_sockfd == -1)
     return KSocketAddress();	// not open, empty value
 
-  socklen_t len;
+  kde_socklen_t len;
   KSocketAddress localAddress;
   localAddress.setLength(len = 32);	// arbitrary value
   if (kde_getsockname(m_sockfd, localAddress.address(), &len) == -1)
@@ -552,7 +552,7 @@
   if (m_sockfd == -1)
     return KSocketAddress();	// not open, empty value
 
-  socklen_t len;
+  kde_socklen_t len;
   KSocketAddress peerAddress;
   peerAddress.setLength(len = 32);	// arbitrary value
   if (kde_getpeername(m_sockfd, peerAddress.address(), &len) == -1)
diff -uNr kdelibs-3.3.91/kdecore/network/syssocket.h kdelibs-3.3.91-new/kdecore/network/syssocket.h
--- kdelibs-3.3.91/kdecore/network/syssocket.h	2004-05-25 22:34:35.000000000 -0400
+++ kdelibs-3.3.91-new/kdecore/network/syssocket.h	2005-01-31 19:22:52.000000000 -0500
@@ -53,7 +53,7 @@
   }
 
   // bind
-  inline int kde_bind(int fd, const struct sockaddr* sa, socklen_t len)
+  inline int kde_bind(int fd, const struct sockaddr* sa, kde_socklen_t len)
   {
     return ::bind(fd, sa, len);
   }
@@ -65,25 +65,25 @@
   }
 
   // connect
-  inline int kde_connect(int fd, const struct sockaddr* sa, socklen_t len)
+  inline int kde_connect(int fd, const struct sockaddr* sa, kde_socklen_t len)
   {
     return ::connect(fd, sa, len);
   }
 
   // accept
-  inline int kde_accept(int fd, struct sockaddr* sa, socklen_t* len)
+  inline int kde_accept(int fd, struct sockaddr* sa, kde_socklen_t* len)
   {
     return ::accept(fd, sa, len);
   }
 
   // getpeername
-  inline int kde_getpeername(int fd, struct sockaddr* sa, socklen_t* len)
+  inline int kde_getpeername(int fd, struct sockaddr* sa, kde_socklen_t* len)
   {
     return ::getpeername(fd, sa, len);
   }
 
   // getsockname
-  inline int kde_getsockname(int fd, struct sockaddr* sa, socklen_t* len)
+  inline int kde_getsockname(int fd, struct sockaddr* sa, kde_socklen_t* len)
   {
     return ::getsockname(fd, sa, len);
   }
diff -uNr kdelibs-3.3.91/kdesu/su.cpp kdelibs-3.3.91-new/kdesu/su.cpp
--- kdelibs-3.3.91/kdesu/su.cpp	2004-04-13 09:33:21.000000000 -0400
+++ kdelibs-3.3.91-new/kdesu/su.cpp	2005-01-31 19:22:52.000000000 -0500
@@ -74,13 +74,14 @@
 	setTerminal(true);
 
     QCStringList args;
+    args += "-c";
+    args += "staff";
     if ((m_Scheduler != SchedNormal) || (m_Priority > 50))
 	args += "root";
     else
 	args += m_User;
     args += "-c";
     args += QCString(__KDE_BINDIR) + "/kdesu_stub";
-    args += "-";
 
     QCString command = __PATH_SU;
     if (::access(__PATH_SU, X_OK) != 0)
diff -uNr kdelibs-3.3.91/kdeui/Makefile.in kdelibs-3.3.91-new/kdeui/Makefile.in
--- kdelibs-3.3.91/kdeui/Makefile.in	2005-01-07 07:49:03.000000000 -0500
+++ kdelibs-3.3.91-new/kdeui/Makefile.in	2005-01-31 19:22:52.000000000 -0500
@@ -614,11 +614,11 @@
 lib_LTLIBRARIES = libkdeui.la libkspell.la
 #>- libkdeui_la_LDFLAGS = $(KDE_MT_LDFLAGS) -no-undefined -version-info 6:0:2
 #>+ 1
-libkdeui_la_LDFLAGS = $(KDE_MT_LDFLAGS) -no-undefined $(KDE_NO_UNDEFINED) -version-info 6:0:2
+libkdeui_la_LDFLAGS = $(KDE_MT_LDFLAGS) $(all_libraries) -no-undefined $(KDE_NO_UNDEFINED) -version-info 6:0:2
 libkdeui_la_LIBADD = ../kdecore/libkdecore.la
 #>- libkspell_la_LDFLAGS = $(KDE_MT_LDFLAGS) -version-info 6:0:2 -no-undefined
 #>+ 1
-libkspell_la_LDFLAGS = $(KDE_MT_LDFLAGS) -version-info 6:0:2 -no-undefined $(KDE_NO_UNDEFINED)
+libkspell_la_LDFLAGS = $(KDE_MT_LDFLAGS) $(all_libraries) -version-info 6:0:2 -no-undefined $(KDE_NO_UNDEFINED)
 libkspell_la_LIBADD = libkdeui.la
 libkspell_la_SOURCES = dummy.cpp
 picsdir = $(kde_datadir)/kdeui/pics
diff -uNr kdelibs-3.3.91/kimgio/Makefile.in kdelibs-3.3.91-new/kimgio/Makefile.in
--- kdelibs-3.3.91/kimgio/Makefile.in	2005-01-07 07:49:06.000000000 -0500
+++ kdelibs-3.3.91-new/kimgio/Makefile.in	2005-01-31 19:23:08.000000000 -0500
@@ -581,7 +581,7 @@
              kimg_pcx.la kimg_tga.la kimg_rgb.la kimg_xcf.la kimg_dds.la $(KIMGIO_EXR_MODULES) \
 			 kimg_psd.la
 
-KIMGIO_PLUGIN = -avoid-version -export-symbols-regex 'kimgio_.*_(read|write)'
+KIMGIO_PLUGIN = -avoid-version
 kimg_tiff_la_SOURCES = tiffr.cpp
 #>- kimg_tiff_la_LDFLAGS = -module $(KIMGIO_PLUGIN) -no-undefined $(all_libraries)
 #>+ 1
diff -uNr kdelibs-3.3.91/kinit/Makefile.in kdelibs-3.3.91-new/kinit/Makefile.in
--- kdelibs-3.3.91/kinit/Makefile.in	2005-01-07 07:49:06.000000000 -0500
+++ kdelibs-3.3.91-new/kinit/Makefile.in	2005-01-31 19:22:52.000000000 -0500
@@ -596,7 +596,7 @@
 xdg_appsdir = @xdg_appsdir@
 xdg_directorydir = @xdg_directorydir@
 xdg_menudir = @xdg_menudir@
-INCLUDES = -I$(srcdir)/../libltdl/ $(all_includes) $(KDEINIT_XFT_INCLUDES)
+INCLUDES = -I$(srcdir)/../libltdl/ $(KDEINIT_XFT_INCLUDES) $(all_includes)
 SUBDIRS = . tests
 #>- lib_LTLIBRARIES = 
 #>+ 1
diff -uNr kdelibs-3.3.91/kinit/setproctitle.cpp kdelibs-3.3.91-new/kinit/setproctitle.cpp
--- kdelibs-3.3.91/kinit/setproctitle.cpp	2004-11-12 19:01:44.000000000 -0500
+++ kdelibs-3.3.91-new/kinit/setproctitle.cpp	2005-01-31 19:57:27.000000000 -0500
@@ -76,7 +76,9 @@
 # endif
 # if SPT_TYPE == SPT_PSSTRINGS
 #  include <machine/vmparam.h>
-#  include <sys/exec.h>
+#  ifdef HAVE_SYS_EXEC_H
+#    include <sys/exec.h>
+#  endif
 #  ifndef PS_STRINGS	/* hmmmm....  apparently not available after all */
 #   undef SPT_TYPE
 #   define SPT_TYPE	SPT_REUSEARGV
diff -uNr kdelibs-3.3.91/kio/kio/job.cpp kdelibs-3.3.91-new/kio/kio/job.cpp
--- kdelibs-3.3.91/kio/kio/job.cpp	2005-01-03 09:53:25.000000000 -0500
+++ kdelibs-3.3.91-new/kio/kio/job.cpp	2005-01-31 19:22:52.000000000 -0500
@@ -3260,10 +3260,10 @@
                            f.close();
                            KSimpleConfig config( path );
                            config.setDesktopGroup();
-                           config.writePathEntry( QString::fromLatin1("URL"), (*it).uSource.url() );
-                           KURL urlName = (*it).uSource;
-                           urlName.setPass( "" );
-                           config.writeEntry( QString::fromLatin1("Name"), urlName.url() );
+                           KURL url = (*it).uSource;
+                           url.setPass( "" );
+                           config.writePathEntry( QString::fromLatin1("URL"), url.url() );
+                           config.writeEntry( QString::fromLatin1("Name"), url.url() );
                            config.writeEntry( QString::fromLatin1("Type"), QString::fromLatin1("Link") );
                            QString protocol = (*it).uSource.protocol();
                            if ( protocol == QString::fromLatin1("ftp") )
diff -uNr kdelibs-3.3.91/kio/kssl/kopenssl.cc kdelibs-3.3.91-new/kio/kssl/kopenssl.cc
--- kdelibs-3.3.91/kio/kssl/kopenssl.cc	2004-05-11 04:31:45.000000000 -0400
+++ kdelibs-3.3.91-new/kio/kssl/kopenssl.cc	2005-01-31 19:22:52.000000000 -0500
@@ -322,6 +322,7 @@
              #elif defined(_AIX)
              << "libcrypto.a(libcrypto.so.0)"
 	     #elif defined(__APPLE__)
+	     << "libcrypto.0.9.7.dylib"
 	     << "libcrypto.dylib"
 	     << "libcrypto.0.9.dylib"
 	     #else
diff -uNr kdelibs-3.3.91/kio/misc/kpac/Makefile.in kdelibs-3.3.91-new/kio/misc/kpac/Makefile.in
--- kdelibs-3.3.91/kio/misc/kpac/Makefile.in	2005-01-07 07:49:09.000000000 -0500
+++ kdelibs-3.3.91-new/kio/misc/kpac/Makefile.in	2005-01-31 19:22:52.000000000 -0500
@@ -533,7 +533,7 @@
                              downloader.cpp discovery.cpp proxyscout_skel.cpp
 
 kded_proxyscout_la_LDFLAGS = $(all_libraries) -module -avoid-version
-kded_proxyscout_la_LIBADD = $(LIB_KIO) $(top_builddir)/kjs/libkjs.la
+kded_proxyscout_la_LIBADD = $(LIB_KIO) $(top_builddir)/kjs/libkjs.la $(LIBRESOLV)
 kpac_dhcp_helper_SOURCES = kpac_dhcp_helper.c
 kpac_dhcp_helper_LDADD = $(LIBSOCKET)
 noinst_HEADERS = proxyscout.h script.h downloader.h discovery.h
