diff -ruN Fink/Config.pm Fink.new/Config.pm
--- Fink/Config.pm	Sun Sep  8 20:11:38 2002
+++ Fink.new/Config.pm	Thu Nov  7 21:38:04 2002
@@ -36,12 +36,12 @@
   @ISA         = qw(Exporter Fink::Base);
   @EXPORT      = qw();
   @EXPORT_OK   = qw($config $basepath $libpath $debarch $darwin_version $macosx_version $distribution
-                    &get_option &set_options &verbosity_level);
+                    &get_option &set_options &verbosity_level $buildpath);
   %EXPORT_TAGS = ( );   # eg: TAG => [ qw!name1 name2! ],
 }
 our @EXPORT_OK;
 
-our ($config, $basepath, $libpath, $debarch, $darwin_version, $macosx_version, $distribution);
+our ($config, $basepath, $libpath, $debarch, $darwin_version, $macosx_version, $distribution, $buildpath);
 $debarch = "darwin-powerpc";
 
 my %globals = ();
@@ -87,6 +87,12 @@
   $basepath = $self->param("Basepath");
   die "Basepath not set in config file \"".$self->{_path}."\"!\n"
     unless (defined $basepath and $basepath);
+
+  #hmm...a hidden option atm ;-)
+  $buildpath = $config->param_default("Buildpath", "$basepath/src");
+print $buildpath;
+
+
   $libpath = "$basepath/lib/fink";
   $distribution = $self->param("Distribution");
 
diff -ruN Fink/Depends.pm Fink.new/Depends.pm
--- Fink/Depends.pm	Wed Dec 31 19:00:00 1969
+++ Fink.new/Depends.pm	Thu Nov  7 21:38:04 2002
@@ -0,0 +1,135 @@
+#
+# Fink::Depends class
+#
+# Fink - a package manager that downloads source and installs it
+# Copyright (c) 2001 Christoph Pfisterer
+# Copyright (c) 2001-2002 The Fink Package Manager Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
+#
+
+$|++;
+
+package Fink::Depends;
+use Fink::Base;
+use Fink::Config qw($basepath);
+
+use strict;
+use warnings;
+
+BEGIN {
+  use Exporter ();
+  our ($VERSION, @ISA, @EXPORT, @EXPORT_OK, %EXPORT_TAGS);
+  $VERSION = 1.00;
+  @ISA         = qw(Exporter Fink::Base);
+  @EXPORT      = qw();
+  @EXPORT_OK   = qw(&depends_version &run_dependscheck);
+  %EXPORT_TAGS = ( );
+}
+our @EXPORT_OK;
+our %PACKAGES;
+
+# this is the one and only version number
+our $depends_version = "0.1.0.cvs";
+
+END { }       # module clean-up code here (global destructor)
+
+### return depends version
+
+sub depends_version {
+  return $depends_version;
+}
+
+sub run_dependscheck {
+  my $self = shift;
+  my $pkgname = shift;
+  my (@depends, $depend);
+
+  unless (defined $pkgname) {
+    print "NOP\n";
+    return;
+  } 
+
+  print "Scanning ${pkgname} for a list of lib depends\n";
+  @depends = &check_pkg($pkgname);
+
+  foreach $depend (@depends) {
+    chomp($depend);
+    $PACKAGES{$depend} = 1;  
+  }
+
+  if (keys %PACKAGES) {    
+    print "\nDepends: ", join(', ', sort keys %PACKAGES), "\n\n";
+  } else {
+    print "\nHas no lib depends!\n\n";
+  }
+
+  return;
+}
+
+sub check_pkg {
+  my $pkgname = shift;
+  my (@depends, $depend, @files, $file, @matches, $match);
+
+  # get files to test
+  open(DPKG, "dpkg --listfiles $pkgname 2>/dev/null |") or die "can't run dpkg: $!\n";
+  # iterate through the output of dpkg, looking for bins and libs
+  while (<DPKG>) {
+    chomp();
+    next if (-d "$_");			# Drop  directories
+    if ("$_" =~ /\.dylib/) {		# Check for (.dylib) libs
+      push(@files, $_);
+    } elsif ("$_" =~ /\.a/) {		# Check for (.a) libs
+      push(@files, $_);
+    } elsif ("$_" =~ /\.so/) {		# Check for (.so) libs
+      push(@files, $_);
+    } elsif ("$_" =~ /\/bin\//) {	# Check for anything in %p/bin
+      push(@files, $_);
+    } elsif ("$_" =~ /\/sbin\//) {	# Check for anything in %p/sbin
+      push(@files, $_);
+    } else {				# Drop everything else
+      next;
+    }
+  }
+  close (DPKG);
+
+  # get a list of linked files to the pkg files
+  foreach $file (@files) {
+    chomp($file);
+    open(OTOOL, "otool -L $file 2>/dev/null |") or die "can't run otool: $!\n";
+      # need to drop all links to system libs and the first two lines
+      while (<OTOOL>) {
+        chomp();
+        next if ("$_" =~ /\:/);		# Nuke first line or errors
+        next if ("$_" =~ /\/usr\/lib\/libSystem/);	# Nuke system links
+        $_ =~ s/\ \(.*$//;		# Nuke the end
+        push(@matches, $_);
+      }
+    close (OTOOL);
+   }
+
+   # get list of depend pkgs
+   foreach $match (@matches) {
+    if (length($match) > 1) {
+      $depend = `dpkg --search $match`;
+      $depend =~ s/\:.*$//;		# strip out everything after the colon
+      chomp($depend);
+      next if ($depend eq $pkgname);	# Can't dpend on it's self
+      push(@depends, $depend);
+    }
+  }
+ 
+  return @depends;
+}
diff -ruN Fink/Engine.pm Fink.new/Engine.pm
--- Fink/Engine.pm	Tue Oct 15 18:34:13 2002
+++ Fink.new/Engine.pm	Thu Nov  7 21:38:04 2002
@@ -28,7 +28,7 @@
                       &file_MD5_checksum);
 use Fink::Package;
 use Fink::PkgVersion;
-use Fink::Config qw($config $basepath);
+use Fink::Config qw($config $basepath $distribution);
 use File::Find;
 
 use strict;
@@ -453,7 +453,7 @@
 
   chdir "$basepath/fink";
   foreach $tree (@treelist) {
-    $treedir = "dists/$tree/binary-darwin-powerpc";
+    $treedir = "$distribution/$tree/binary-darwin-powerpc";
     if ($tree =~ /^([^\/]+)\/(.+)$/) {
       $archive = $1;
       $component = $2;
@@ -672,7 +672,7 @@
   }
   
   # Now search through all .deb files in /sw/fink/dists/
-  find (\&kill_obsolete_debs, "$basepath/fink/dists");
+  find (\&kill_obsolete_debs, "$basepath/fink/$distribution");
   
   # Remove broken symlinks in /sw/fink/debs (i.e. those that pointed to 
   # the .deb files we deleted above).
diff -ruN Fink/FinkVersion.pm Fink.new/FinkVersion.pm
--- Fink/FinkVersion.pm	Thu Nov  7 19:23:41 2002
+++ Fink.new/FinkVersion.pm	Wed Dec 31 19:00:00 1969
@@ -1,86 +0,0 @@
-#
-# Fink::FinkVersion package
-#
-# Fink - a package manager that downloads source and installs it
-# Copyright (c) 2001 Christoph Pfisterer
-# Copyright (c) 2001-2002 The Fink Package Manager Team
-#
-# This program is free software; you can redistribute it and/or
-# modify it under the terms of the GNU General Public License
-# as published by the Free Software Foundation; either version 2
-# of the License, or (at your option) any later version.
-#
-# This program is distributed in the hope that it will be useful,
-# but WITHOUT ANY WARRANTY; without even the implied warranty of
-# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-# GNU General Public License for more details.
-#
-# You should have received a copy of the GNU General Public License
-# along with this program; if not, write to the Free Software
-# Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
-#
-
-package Fink::FinkVersion;
-
-use strict;
-use warnings;
-
-BEGIN {
-  use Exporter ();
-  our ($VERSION, @ISA, @EXPORT, @EXPORT_OK, %EXPORT_TAGS);
-  $VERSION = 1.00;
-  @ISA         = qw(Exporter);
-  @EXPORT      = qw();
-  @EXPORT_OK   = qw(&fink_version &distribution_version &pkginfo_version);
-  %EXPORT_TAGS = ( );   # eg: TAG => [ qw!name1 name2! ],
-}
-our @EXPORT_OK;
-
-
-# this is the one and only version number
-our $fink_version = "0.11.0";
-
-
-END { }       # module clean-up code here (global destructor)
-
-
-### return package manager version
-
-sub fink_version {
-  return $fink_version;
-}
-
-### retreive distribution version
-
-sub distribution_version {
-  my $dv;
-  if (-f "/sw/fink/VERSION") {
-    chomp($dv = `cat /sw/fink/VERSION`);
-  } elsif (-f "/sw/etc/fink-release") {
-    chomp($dv = `cat /sw/etc/fink-release`);
-  } else {
-    $dv = "unknown";
-  }
-  return $dv;
-}
-
-### retreive version stamp of package info
-
-sub pkginfo_version {
-  my $pv = "0";
-  my ($fn, $v);
-  foreach $fn (glob("/sw/fink/stamp-*")) {
-    if ($fn =~ /\/stamp-cvs/) {
-      return "cvs";
-    } elsif ($fn =~ /\/stamp-rel-(.+)$/) {
-      $v = $1;
-      $pv = $v if ($v gt $pv);
-    }
-  }
-  $pv = &distribution_version() if $pv eq "0";
-  return $pv;
-}
-
-
-### EOF
-1;
diff -ruN Fink/NetAccess.pm Fink.new/NetAccess.pm
--- Fink/NetAccess.pm	Sat Sep 28 15:53:38 2002
+++ Fink.new/NetAccess.pm	Thu Nov  7 21:38:04 2002
@@ -52,11 +52,11 @@
 
 sub fetch_url {
   my $url = shift;
-  my $destdir = shift || "$basepath/src";
+  my $downloaddir = shift || "$basepath/src";
   my ($file, $cmd);
 
   $file = &filename($url);
-  return &fetch_url_to_file($url, $file, 0, 0, 0, $destdir);
+  return &fetch_url_to_file($url, $file, 0, 0, 0, $downloaddir);
 }
 
 ### download a file to the designated directory and save it under the
@@ -69,18 +69,18 @@
   my $custom_mirror = shift || 0;
   my $tries = shift || 0;
   my $cont = shift || 0;  
-  my $destdir = shift || "$basepath/src";
+  my $downloaddir = shift || "$basepath/src";
   my ($http_proxy, $ftp_proxy);
   my ($url, $cmd, $cont_cmd, $result);
 
   # create destination directory if necessary
-  if (not -d $destdir) {
-    &execute("mkdir -p $destdir");
-    if (not -d $destdir) {
-      die "Download directory \"$destdir\" can not be created!\n";
+  if (not -d $downloaddir) {
+    &execute("mkdir -p $downloaddir");
+    if (not -d $downloaddir) {
+      die "Download directory \"$downloaddir\" can not be created!\n";
     }
   }
-  chdir $destdir;
+  chdir $downloaddir;
 
   # determine download command
   $cmd = &download_cmd($origurl, $file);
diff -ruN Fink/Package.pm Fink.new/Package.pm
--- Fink/Package.pm	Thu Sep 19 16:03:02 2002
+++ Fink.new/Package.pm	Thu Nov  7 21:38:04 2002
@@ -24,7 +24,7 @@
 use Fink::Base;
 use Fink::Services qw(&read_properties &latest_version &version_cmp
                       &print_breaking &execute);
-use Fink::Config qw($config $basepath);
+use Fink::Config qw($config $basepath $distribution);
 use Fink::PkgVersion;
 use File::Find;
 
@@ -342,7 +342,7 @@
 	  or ((stat("$basepath/etc/fink.conf"))[9] > $db_mtime)) {
 	  $db_outdated = 1;
 	} else {
-	  find (\&process_find, "$basepath/fink/dists");
+	  find (\&process_find, "$basepath/fink/$distribution");
 	}
       }
       
@@ -409,7 +409,7 @@
   
   # read data from descriptions
   foreach $tree ($config->get_treelist()) {
-    $dir = "$basepath/fink/dists/$tree/finkinfo";
+    $dir = "$basepath/fink/$distribution/$tree/finkinfo";
     Fink::Package->scan($dir);
   }
   
diff -ruN Fink/PkgVersion.pm Fink.new/PkgVersion.pm
--- Fink/PkgVersion.pm	Sun Oct 13 19:37:23 2002
+++ Fink.new/PkgVersion.pm	Thu Nov  7 21:38:04 2002
@@ -28,7 +28,7 @@
 		      &prompt_boolean &prompt_selection
 		      &collapse_space &read_properties_var
 		      &file_MD5_checksum);
-use Fink::Config qw($config $basepath $libpath $debarch);
+use Fink::Config qw($config $basepath $libpath $debarch $buildpath  $distribution);
 use Fink::NetAccess qw(&fetch_url_to_file);
 use Fink::Mirror;
 use Fink::Package;
@@ -101,7 +101,7 @@
     }
     
     # determine the package tree ("stable", "unstable", etc.)
-	@parts = split(/\//, substr($filename,length("$basepath/fink/dists/")));
+	@parts = split(/\//, substr($filename,length("$basepath/fink/$distribution/")));
     $self->{_tree}  = $parts[0];
   } else {
     # for dummy descriptions generated from dpkg status data alone
@@ -122,11 +122,11 @@
   # percent-expansions
   $configure_params = "--prefix=\%p ".
     $self->param_default("ConfigureParams", "");
-  $destdir = "$basepath/src/root-".$self->{_fullname};
+  $destdir = "$buildpath/root-".$self->{_fullname};
   if ($self->{_type} eq "splitoff") {
     my $parent = $self->{parent};
     $parentpkgname = $parent->{_name};
-    $parentdestdir = "$basepath/src/root-".$parent->{_fullname};
+    $parentdestdir = "$buildpath/root-".$parent->{_fullname};
   } else {
     $parentpkgname = $pkgname;
     $parentdestdir = $destdir;
@@ -306,13 +306,13 @@
   my ($destdir);
   my $splitoff;
 
-  $destdir = "$basepath/src/root-".$self->{_fullname};
+  $destdir = "$buildpath/root-".$self->{_fullname};
   $self->{_expand}->{p} = $basepath;
   $self->{_expand}->{d} = $destdir;
   $self->{_expand}->{i} = $destdir.$basepath;
   if ($self->{_type} eq "splitoff") {
     my $parent = $self->{parent};
-    my $parentdestdir = "$basepath/src/root-".$parent->{_fullname};
+    my $parentdestdir = "$buildpath/root-".$parent->{_fullname};
     $self->{_expand}->{D} = $parentdestdir;
     $self->{_expand}->{I} = $parentdestdir.$basepath;
   } else {
@@ -474,7 +474,7 @@
     $self->{_builddir} = $self->get_fullname()."/".$dir;
   }
 
-  $self->{_expand}->{b} = "$basepath/src/".$self->{_builddir};
+  $self->{_expand}->{b} = "$buildpath/".$self->{_builddir};
   return $self->{_builddir};
 }
 
@@ -934,7 +934,7 @@
   }
 
   # remove dir if it exists
-  chdir "$basepath/src";
+  chdir "$buildpath";
   if (-e $bdir) {
     if (&execute("rm -rf $bdir")) {
       die "can't remove existing directory $bdir\n";
@@ -942,7 +942,7 @@
   }
 
   if ($self->{_type} eq "nosource") {
-    $destdir = "$basepath/src/$bdir";
+    $destdir = "$buildpath/$bdir";
     if (&execute("mkdir -p $destdir")) {
       die "can't create directory $destdir\n";
     }
@@ -1045,7 +1045,7 @@
     }
 
     # calculate destination directory
-    $destdir = "$basepath/src/$bdir";
+    $destdir = "$buildpath/$bdir";
     if ($i > 1) {
       if ($self->has_param("Source".$i."ExtractDir")) {
 	$destdir .= "/".&expand_percent($self->param("Source".$i."ExtractDir"), $self->{_expand});
@@ -1103,10 +1103,10 @@
   }
 
   $dir = $self->get_build_directory();
-  if (not -d "$basepath/src/$dir") {
-    die "directory $basepath/src/$dir doesn't exist, check the package description\n";
+  if (not -d "$buildpath/$dir") {
+    die "directory $buildpath/$dir doesn't exist, check the package description\n";
   }
-  chdir "$basepath/src/$dir";
+  chdir "$buildpath/$dir";
 
   $patch_script = "";
 
@@ -1189,10 +1189,10 @@
   }
 
   $dir = $self->get_build_directory();
-  if (not -d "$basepath/src/$dir") {
-    die "directory $basepath/src/$dir doesn't exist, check the package description\n";
+  if (not -d "$buildpath/$dir") {
+    die "directory $buildpath/$dir doesn't exist, check the package description\n";
   }
-  chdir "$basepath/src/$dir";
+  chdir "$buildpath/$dir";
 
   # generate compilation script
   if ($self->has_param("CompileScript")) {
@@ -1234,10 +1234,10 @@
     } else {
       $dir = $self->get_build_directory();
     }
-    if (not -d "$basepath/src/$dir") {
-      die "directory $basepath/src/$dir doesn't exist, check the package description\n";
+    if (not -d "$buildpath/$dir") {
+      die "directory $buildpath/$dir doesn't exist, check the package description\n";
     }
-    chdir "$basepath/src/$dir";
+    chdir "$buildpath/$dir";
   }
 
   # generate installation script
@@ -1394,7 +1394,7 @@
 
   if (not $do_splitoff) {
     $bdir = $self->get_fullname();
-    chdir "$basepath/src";
+    chdir "$buildpath";
     if (not $config->param_boolean("KeepBuildDir") and -e $bdir) {
       if (&execute("rm -rf $bdir")) {
 	&print_breaking("WARNING: Can't remove build directory $bdir. ".
@@ -1427,9 +1427,9 @@
     return;
   }
 
-  chdir "$basepath/src";
+  chdir "$buildpath";
   $ddir = "root-".$self->get_fullname();
-  $destdir = "$basepath/src/$ddir";
+  $destdir = "$buildpath/$ddir";
 
   if (not -d "$destdir/DEBIAN") {
     if (&execute("mkdir -p $destdir/DEBIAN")) {
diff -ruN Fink/SelfUpdate.pm Fink.new/SelfUpdate.pm
--- Fink/SelfUpdate.pm	Sun Sep  8 20:11:38 2002
+++ Fink.new/SelfUpdate.pm	Thu Nov  7 21:38:04 2002
@@ -356,7 +356,7 @@
 
 sub do_tarball {
   my $newversion = shift;
-  my ($destdir, $dir);
+  my ($downloaddir, $dir);
   my ($pkgtarball, $url, $verbosity, $unpack_cmd);
 
   print "\n";
@@ -366,8 +366,8 @@
 		  "using commands like 'fink update-all'.");
   print "\n";
 
-  $destdir = "$basepath/src";
-  chdir $destdir or die "Can't cd to $destdir: $!\n";
+  $downloaddir = "$basepath/src";
+  chdir $downloaddir or die "Can't cd to $downloaddir: $!\n";
 
   # go ahead and upgrade
   # first, download the packages tarball
@@ -376,7 +376,7 @@
   $url = "mirror:sourceforge:fink/$pkgtarball";
 
   if (not -f $pkgtarball) {
-    if (&fetch_url($url, $destdir)) {
+    if (&fetch_url($url, $downloaddir)) {
       die "Downloading the update tarball '$pkgtarball' from the URL '$url' failed.\n";
     }
   }
@@ -402,7 +402,7 @@
   if (&execute("./inject.pl $basepath -quiet")) {
     die "injecting the new package definitions from $pkgtarball failed\n";
   }
-  chdir $destdir or die "Can't cd to $destdir: $!\n";
+  chdir $downloaddir or die "Can't cd to $downloaddir: $!\n";
   if (-e $dir) {
     &execute("rm -rf $dir");
   }
diff -ruN Fink/Validation.pm Fink.new/Validation.pm
--- Fink/Validation.pm	Sun Oct 13 21:36:43 2002
+++ Fink.new/Validation.pm	Thu Nov  7 21:38:04 2002
@@ -23,7 +23,7 @@
 package Fink::Validation;
 
 use Fink::Services qw(&read_properties &expand_percent);
-use Fink::Config qw($config $basepath);
+use Fink::Config qw($config $basepath $buildpath);
 
 use strict;
 use warnings;
@@ -172,7 +172,7 @@
   my ($properties, @parts);
   my ($pkgname, $pkgversion, $pkgrevision, $pkgfullname, $pkgdestdir, $pkgpatchpath);
   my ($field, $value);
-  my ($basepath, $expand);
+  my ($basepath, $expand, $buildpath);
   my $looks_good = 1;
   my $error_found = 0;
 
@@ -185,12 +185,13 @@
   
   # determine the base path
   $basepath = $config->param("basepath");
+  $buildpath = $config->param("buildpath");
 
   $pkgname = $properties->{package};
   $pkgversion = $properties->{version};
   $pkgrevision = $properties->{revision};
   $pkgfullname = "$pkgname-$pkgversion-$pkgrevision";
-  $pkgdestdir = "$basepath/src/root-".$pkgfullname;
+  $pkgdestdir = "$buildpath/root-".$pkgfullname;
   
   @parts = split(/\//, $filename);
   $filename = pop @parts;   # remove filename
