diff -ruN Fink.orig/PkgVersion.pm Fink/PkgVersion.pm
--- Fink.orig/PkgVersion.pm     Sun Oct 13 19:37:23 2002
+++ Fink/PkgVersion.pm  Fri Nov  8 00:08:51 2002
@@ -1735,6 +1736,7 @@
   foreach $varname ("CC", "CFLAGS",
                    "CPP", "CPPFLAGS",
                    "CXX", "CXXFLAGS",
+                   "CXXCPP", "CXXCPPFLAGS",
                    "LD", "LDFLAGS", "LIBS",
                    "MAKE", "MFLAGS") {
     if ($self->has_param("Set$varname")) {
