diff -ruN Fink.orig/Configure.pm Fink/Configure.pm
--- Fink.orig/Configure.pm	Mon Aug  5 18:08:16 2002
+++ Fink/Configure.pm	Fri Nov  8 00:46:31 2002
@@ -62,6 +62,14 @@
 	    $config->param_default("FetchAltDir", ""));
   if ($otherdir) {
     $config->set_param("FetchAltDir", $otherdir);
+  }
+
+  $builddir =
+    &prompt("In what directory should Fink use to build packages? \(Most ".
+	    "people should just leave this the default.\)",
+	    $config->param_default("Buildpath", ""));
+  if ($builddir) {
+    $config->set_param("Buildpath", $builddir);
   }
 
   $verbose = $config->param_default("Verbose", 3);
