diff -Naur gnome-games-2.8.0-old/libgames-support/games-network.h gnome-games-2.8.0/libgames-support/games-network.h
--- gnome-games-2.8.0-old/libgames-support/games-network.h	Tue Jul 20 07:35:02 2004
+++ gnome-games-2.8.0/libgames-support/games-network.h	Sun Oct 10 09:44:52 2004
@@ -3,8 +3,8 @@
 
 extern char *game_server;
 extern guint whose_turn;
-const char *player_name;
-const char *opponent_name;
+extern const char *player_name;
+extern const char *opponent_name;
 extern guint game_in_progress;
 
 typedef struct {
