diff -Naur librsvg-2.8.1-old/Makefile.in librsvg-2.8.1/Makefile.in
--- librsvg-2.8.1-old/Makefile.in	Sun Sep 12 15:16:35 2004
+++ librsvg-2.8.1/Makefile.in	Wed Oct  6 16:46:25 2004
@@ -350,6 +350,7 @@
 	librsvg-features.h
 
 INCLUDES = \
+	-I/usr/X11R6/include/freetype2  \
 	$(LIBGNOME_VFS_CFLAGS)		\
 	$(LIBGSF_CFLAGS)		\
 	$(LIBCROCO_CFLAGS)		\
diff -Naur librsvg-2.8.1-old/configure librsvg-2.8.1/configure
--- librsvg-2.8.1-old/configure	Sun Sep 12 15:16:43 2004
+++ librsvg-2.8.1/configure	Wed Oct  6 16:30:27 2004
@@ -20871,11 +20871,11 @@
             echo "$as_me:$LINENO: checking LIBRSVG_CFLAGS" >&5
 echo $ECHO_N "checking LIBRSVG_CFLAGS... $ECHO_C" >&6
             LIBRSVG_CFLAGS=`$PKG_CONFIG --cflags "\
+	pangoft2 >= $PANGOFT2_REQUIRED \
 	gdk-pixbuf-2.0 >= $GDK_PIXBUF_REQUIRED \
 	glib-2.0 >= $GLIB_REQUIRED \
 	libart-2.0 >= $LIBART_REQUIRED \
-	libxml-2.0 >= $LIBXML_REQUIRED \
-	pangoft2 >= $PANGOFT2_REQUIRED"`
+	libxml-2.0 >= $LIBXML_REQUIRED"`
             echo "$as_me:$LINENO: result: $LIBRSVG_CFLAGS" >&5
 echo "${ECHO_T}$LIBRSVG_CFLAGS" >&6
 
diff -Naur librsvg-2.8.1-old/ltmain.sh librsvg-2.8.1/ltmain.sh
--- librsvg-2.8.1-old/ltmain.sh	Tue May  4 23:35:36 2004
+++ librsvg-2.8.1/ltmain.sh	Wed Oct  6 16:30:27 2004
@@ -5263,10 +5263,10 @@
 
 # Directory that this library needs to be installed in:
 libdir='$install_libdir'"
-	  if test "$installed" = no && test "$need_relink" = yes; then
-	    $echo >> $output "\
-relink_command=\"$relink_command\""
-	  fi
+#	  if test "$installed" = no && test "$need_relink" = yes; then
+#	    $echo >> $output "\
+#relink_command=\"$relink_command\""
+#	  fi
 	done
       fi
 
