diff -Naur ccl.orig/level-1/version.lisp ccl/level-1/version.lisp
--- ccl.orig/level-1/version.lisp	2004-05-11 15:22:54.000000000 -0700
+++ ccl/level-1/version.lisp	2004-10-03 22:03:54.000000000 -0700
@@ -17,7 +17,7 @@
 (defparameter *openmcl-major-version* 0)
 (defparameter *openmcl-minor-version* 14)
 (defparameter *openmcl-revision* 2)
-(defparameter *openmcl-suffix* "p1")
+(defparameter *openmcl-suffix* "p1+@DATE@")
 
 (defparameter *openmcl-version* (format nil "(Beta: ~~A) ~d.~d~@[.~d~]~@[-~a~]"
 					*openmcl-major-version*
diff -Naur ccl.orig/tools/asdf.lisp ccl/tools/asdf.lisp
--- ccl.orig/tools/asdf.lisp	2004-02-07 21:08:11.000000000 -0800
+++ ccl/tools/asdf.lisp	2004-10-01 21:58:07.000000000 -0700
@@ -1101,4 +1101,20 @@
   
   (pushnew 'module-provide-asdf sb-ext:*module-provider-functions*))
 
+#+openmcl
+(progn
+  (defun module-provide-asdf (name)
+    (handler-bind ((style-warning #'muffle-warning))
+      (let* ((*verbose-out* (make-broadcast-stream))
+	     (system (asdf:find-system name nil)))
+	(when system
+	  (asdf:operate 'asdf:load-op name)
+	  t))))
+
+  (pushnew
+   #P"@PREFIX@/lib/common-lisp/openmcl/systems/"
+   *central-registry*)
+  
+  (pushnew 'module-provide-asdf ccl:*module-provider-functions*))
+
 (provide 'asdf)
