diff -x CVS -x .cvsignore -x '#*' -x '*~' -Naur ccl.orig/level-1/version.lisp ccl/level-1/version.lisp
--- ccl.orig/level-1/version.lisp	2004-10-09 22:20:26.000000000 -0700
+++ ccl/level-1/version.lisp	2004-10-09 23:35:07.000000000 -0700
@@ -19,7 +19,7 @@
 (defparameter *openmcl-revision* 2)
 (defparameter *openmcl-suffix* "p1")
 
-(defparameter *openmcl-version* (format nil "(Beta: ~~A) ~d.~d~@[.~d~]~@[-~a~]"
+(defparameter *openmcl-version* (format nil "(Beta: ~~A) ~d.~d~@[.~d~]~@[-~a~] CVS ccl-0.14 @DATE@"
 					*openmcl-major-version*
 					*openmcl-minor-version*
 					(unless (zerop *openmcl-revision*)
diff -x CVS -x .cvsignore -x '#*' -x '*~' -Naur ccl.orig/tools/asdf.lisp ccl/tools/asdf.lisp
--- ccl.orig/tools/asdf.lisp	2004-10-09 22:20:29.000000000 -0700
+++ ccl/tools/asdf.lisp	2004-10-09 23:36:32.000000000 -0700
@@ -1101,4 +1101,16 @@
   
   (pushnew 'module-provide-asdf sb-ext:*module-provider-functions*))
 
+#+openmcl
+(progn
+  (defun module-provide-asdf (name)
+    (handler-bind ((style-warning #'muffle-warning))
+      (let* ((*verbose-out* (make-broadcast-stream))
+	     (system (asdf:find-system name nil)))
+	(when system
+	  (asdf:operate 'asdf:load-op name)
+	  t))))
+
+  (pushnew 'module-provide-asdf ccl:*module-provider-functions*))
+
 (provide 'asdf)
