diff -Naur sbcl.orig/contrib/asdf/asdf.lisp sbcl/contrib/asdf/asdf.lisp
--- sbcl.orig/contrib/asdf/asdf.lisp	2004-07-19 14:29:10.000000000 -0700
+++ sbcl/contrib/asdf/asdf.lisp	2004-09-30 18:26:15.000000000 -0700
@@ -1099,4 +1099,6 @@
   
   (pushnew 'module-provide-asdf sb-ext:*module-provider-functions*))
 
+  (pushnew #P"@PREFIX@/lib/common-lisp/sbcl/systems/" *central-registry*) 
+
 (provide 'asdf)
diff -Naur sbcl.orig/install.sh sbcl/install.sh
--- sbcl.orig/install.sh	2004-08-18 05:58:41.000000000 -0700
+++ sbcl/install.sh	2004-09-30 14:47:47.000000000 -0700
@@ -82,18 +82,7 @@
 # info
 for info in doc/manual/*.info
 do
-  cp $info $BUILD_ROOT$INFO_DIR/ \
-      && echo -n " info $BUILD_ROOT$INFO_DIR/`basename $info`" \
-      && ( install-info $BUILD_ROOT$INFO_DIR/`basename $info` > /dev/null 2>&1 \
-           || echo -n " (could not add to system catalog)" ) \
-      && echo
-done
-
-# pdf
-for pdf in doc/manual/*.pdf
-do
-  cp $pdf $BUILD_ROOT$DOC_DIR/ \
-      && echo " pdf $BUILD_ROOT$DOC_DIR/`basename $pdf`"
+  cp $info $BUILD_ROOT$INFO_DIR/ 
 done
 
 # html
