diff -uNr ../xc/config/cf/darwin.cf ./config/cf/darwin.cf
--- ../xc/config/cf/darwin.cf	Fri Nov 21 00:55:04 2003
+++ ./config/cf/darwin.cf	Sun Nov 23 08:17:03 2003
@@ -163,8 +163,8 @@
 #endif
 
 /* we have a gcc compatible compiler, but its not called gcc */
-#define CcCmd                   /usr/bin/cc
-#define CplusplusCmd            /usr/bin/c++
+#define CcCmd                   gcc-3.3
+#define CplusplusCmd            g++-3.3
 
 /* default flags to pass to cc */
 #ifndef DefaultCCOptions
diff -uNr ../xc/programs/Xserver/Imakefile ./programs/Xserver/Imakefile
--- ../xc/programs/Xserver/Imakefile	Fri Nov 21 00:58:14 2003
+++ ./programs/Xserver/Imakefile	Sun Nov 23 08:16:18 2003
@@ -1233,8 +1233,7 @@
 /*
  * IOKit X server
  */
-SetUIDServerTarget(XDarwin,$(DARWINDIRS),$(DARWINOBJS), \
-	$(DARWINLIBS) $(IOKITLIB) $(DARWINEXTLIBS),$(DARWINSYSLIBS))
+SetUIDServerTarget(XDarwin,$(DARWINDIRS),$(DARWINOBJS),$(DARWINLIBS) $(IOKITLIB) $(DARWINEXTLIBS),$(DARWINSYSLIBS))
 
 #if DarwinQuartzSupport
 
@@ -1267,9 +1266,7 @@
 /*
  * Quartz X server (installed in its application bundle)
  */
-SetUIDServerTarget(XDarwinApp,$(DARWINDIRS),$(QUARTZOBJS), \
-	$(DARWINLIBS) $(QUARTZLIB) $(QUARTZEXTLIBS), \
-	$(DARWINSYSLIBS) $(QUARTZSYSLIBS) -u _miDCInitialize)
+SetUIDServerTarget(XDarwinApp,$(DARWINDIRS),$(QUARTZOBJS),$(DARWINLIBS) $(QUARTZLIB) $(QUARTZEXTLIBS),$(DARWINSYSLIBS) $(QUARTZSYSLIBS) -u _miDCInitialize)
 
 install::
 	$(MKDIRHIER) $(DESTDIR)$(XDARWINAPPDIR)/MacOS
@@ -1286,41 +1283,30 @@
 XPRDIRS = $(QUARTZDIR)/xpr $(ROOTLESSDIR) $(ROOTLESSDIR)/safeAlpha
 XPRLIBS = $(QUARTZDIR)/xpr/LibraryTargetName(xpr) $(ROOTLESSLIB)
 
-BundleProgramTarget(xpr,XDarwinApp,$(XPRDIRS),$(QUARTZDIR)/xpr/xprScreen.o, \
-	$(XPRLIBS),-framework ApplicationServices $(XPLUGINLIB), \
-	$(XDARWINAPPDIR)/Resources)
+BundleProgramTarget(xpr,XDarwinApp,$(XPRDIRS),$(QUARTZDIR)/xpr/xprScreen.o,$(XPRLIBS),-framework ApplicationServices $(XPLUGINLIB),$(XDARWINAPPDIR)/Resources)
 #endif
 
 CRDIRS = $(QUARTZDIR)/cr $(ROOTLESSDIR) $(ROOTLESSDIR)/safeAlpha
 CRLIBS = $(QUARTZDIR)/cr/LibraryTargetName(cr) $(ROOTLESSLIB)
 
-BundleProgramTarget(cr,XDarwinApp,$(CRDIRS),$(QUARTZDIR)/cr/crScreen.o, \
-	$(CRLIBS),-framework ApplicationServices -framework Cocoa \
-	-framework Carbon -ObjC, $(XDARWINAPPDIR)/Resources)
+BundleProgramTarget(cr,XDarwinApp,$(CRDIRS),$(QUARTZDIR)/cr/crScreen.o,$(CRLIBS),-framework ApplicationServices -framework Cocoa -framework Carbon -ObjC,$(XDARWINAPPDIR)/Resources)
 
 FSDIRS = $(QUARTZDIR)/fullscreen $(SHADOWDIR)
 FSLIBS = $(QUARTZDIR)/fullscreen/LibraryTargetName(fullscreen) $(SHADOW)
 
-BundleProgramTarget(fullscreen,XDarwinApp,$(FSDIRS), \
-	$(QUARTZDIR)/fullscreen/fullscreen.o,$(FSLIBS), \
-	-framework ApplicationServices,$(XDARWINAPPDIR)/Resources)
+BundleProgramTarget(fullscreen,XDarwinApp,$(FSDIRS),$(QUARTZDIR)/fullscreen/fullscreen.o,$(FSLIBS),-framework ApplicationServices,$(XDARWINAPPDIR)/Resources)
 
 /*
  * GLX bundles for Quartz
  * (installed in their own bundles inside XDarwin's)
  */
-BundleProgramTarget(glxAGL,XDarwinApp,GL,GL/glx/glxext.o,GL/glx/libglx.a \
-	GL/apple/libAGLcore.a,-framework AGL -framework OpenGL, \
-	$(XDARWINAPPDIR)/Resources)
+BundleProgramTarget(glxAGL,XDarwinApp,GL,GL/glx/glxext.o,GL/glx/libglx.a GL/apple/libAGLcore.a,-framework AGL -framework OpenGL,$(XDARWINAPPDIR)/Resources)
 
 #if BuildAppleDRI
-BundleProgramTarget(glxCGL,XDarwinApp,GL,GL/glx/glxext.o,GL/glx/libglx.a \
-	GL/apple/libCGLcore.a,-framework OpenGL $(XPLUGINLIB), \
-	$(XDARWINAPPDIR)/Resources)
+BundleProgramTarget(glxCGL,XDarwinApp,GL,GL/glx/glxext.o,GL/glx/libglx.a GL/apple/libCGLcore.a,-framework OpenGL $(XPLUGINLIB),$(XDARWINAPPDIR)/Resources)
 #endif
 
-BundleProgramTarget(glxMesa,XDarwinApp,GL,GL/glx/glxext.o,GL/glx/libglx.a \
-	GL/mesa/GLcore/libGLcore.a,NullParameter,$(XDARWINAPPDIR)/Resources)
+BundleProgramTarget(glxMesa,XDarwinApp,GL,GL/glx/glxext.o,GL/glx/libglx.a GL/mesa/GLcore/libGLcore.a,NullParameter,$(XDARWINAPPDIR)/Resources)
 
 #else /* !DarwinQuartzSupport */
 
