--- config/cf/darwin.cf	Fri Aug 13 17:18:27 2004
+++ config/cf/darwin.cf	Fri Aug 13 17:20:38 2004
@@ -159,7 +159,9 @@
  * The default cpp-3.3 that ships with Panther inserts spurious #pragmas,
  * so we use the 3.1-based version.
  */
-#if OSMajorVersion >= 7
+#if OSMajorVersion >= 8
+# define CppCmd                 /usr/bin/cpp-3.5
+#elif OSMajorVersion >= 7
 # define CppCmd                 /usr/bin/cpp3
 # define StandardCppOptions     -traditional -D__GNUC__
 #else
--- extras/freetype2/include/freetype/freetype.h.orig	Thu Feb 10 17:19:38 2005
+++ extras/freetype2/include/freetype/freetype.h	Thu Feb 10 17:19:51 2005
@@ -15,14 +15,7 @@
 /*                                                                         */
 /***************************************************************************/
 
-
-#ifndef FT_FREETYPE_H
-#error "`ft2build.h' hasn't been included yet!"
-#error "Please always use macros to include FreeType header files."
-#error "Example:"
-#error "  #include <ft2build.h>"
-#error "  #include FT_FREETYPE_H"
-#endif
+#include <ft2build.h>
 
 
 #ifndef __FREETYPE_H__
