diff -ru2N --exclude=.DS_Store apt-spy-3.0/Makefile apt-spy-3.0-patched/Makefile
--- apt-spy-3.0/Makefile	Tue May 20 13:07:55 2003
+++ apt-spy-3.0-patched/Makefile	Sun Jun 22 18:11:08 2003
@@ -1,5 +1,5 @@
 CC = cc
 CFILES = benchmark.c  file.c  protocol.c  main.c  parse.c  update.c
-CFLAGS = -lcurl -g -Wall -pedantic-errors
+CFLAGS = -lcurl -g -O3 -Wall -I/sw/include -L/sw/lib
 TARGET = apt-spy
 
diff -ru2N --exclude=.DS_Store apt-spy-3.0/README apt-spy-3.0-patched/README
--- apt-spy-3.0/README	Tue May 20 13:07:55 2003
+++ apt-spy-3.0-patched/README	Sun Jun 22 18:11:08 2003
@@ -3,14 +3,14 @@
 re-write of the original.
 
-You need to create a directory /usr/share/apt-spy
+You need to create a directory @PREFIX@/share/apt-spy
 
 Then run: 
-./apt-spy update
+sudo apt-spy update
 
 Then: 
-cp apt-spy.conf /etc
+skip = cp apt-spy.conf @PREFIX@/etc
 
 Then:
-./apt-spy -a <area> -d <distribution>
+sudo apt-spy -a <area> -d <distribution>
 
 valid values for <area> are: 
@@ -27,6 +27,6 @@
 woody, sarge or sid.
 
-The program will write a new /etc/apt/sources.list
-Your old /etc/apt/sources.list will be saved to /etc/apt/sources.list.bak
+The program will write a new @PREFIX@/etc/apt/sources.list
+Your old @PREFIX@/etc/apt/sources.list will be saved to @PREFIX@/etc/apt/sources.list.bak
 
 
diff -ru2N --exclude=.DS_Store apt-spy-3.0/apt-spy.8 apt-spy-3.0-patched/apt-spy.8
--- apt-spy-3.0/apt-spy.8	Tue May 20 13:07:55 2003
+++ apt-spy-3.0-patched/apt-spy.8	Sun Jun 22 18:11:08 2003
@@ -17,5 +17,5 @@
 .\" for manpage-specific macros, see man(7)
 .SH NAME
-apt-spy \- program to generate a /etc/apt/sources.list file
+apt-spy \- program to generate a @PREFIX@/etc/apt/sources.list file
 .SH SYNOPSIS
 .BR apt-spy " -d " \fIdist\fP 
@@ -45,5 +45,5 @@
 \fBapt-spy\fP is a program that tests a series of Debian archive mirrors for
 bandwith.
-It writes an \fI/etc/apt/sources.list\fP with the fastest server.
+It writes an \fI@PREFIX@/etc/apt/sources.list\fP with the fastest server.
 .PP
 You can select which of the many Debian sites to benchmark by geographical
@@ -53,6 +53,5 @@
 .TP
 .BI -d " dist" 
-Debian distribution to use. Sensible values would be `stable', `testing'
-or `unstable'. 
+Fink distribution to use. Sensible values would be `stable' or `unstable'.
 This value is directly used in the generated sources.list
 .TP
@@ -65,5 +64,5 @@
 .TP
 .BI -c " config"
-configuration file to use instead of \fI/etc/apt-spy.conf.
+configuration file to use instead of \fI@PREFIX@/lib/fink/mirror/_keys.
 .TP
 .BI -e " number"
@@ -85,5 +84,5 @@
 .BI -m " mirror-list"
 where to load the mirrors list from or where to save it to.
-The default location is \fB/usr/share/apt-spy/README.mirrors.txt\fP.
+The default location is \fB@PREFIX@/lib/fink/mirror/sourceforge\fP.
 Using this option allows you to specify a different location. 
 When updating, this defines where the updated mirrors list will be saved.
@@ -91,5 +90,5 @@
 .BI -o " output-file"
 location for the new \fBsources.list\fP.
-By default the new file is created as \fB/etc/apt/sources.list\fP.
+By default the new file is created as \fB@PREFIX@/etc/apt/sources.list\fP.
 .TP
 .BI -p " proxy"
@@ -120,5 +119,5 @@
 .BI update
 connect to \fBhttp.us.debian.org\fP and download the mirror list to
-\fB/usr/share/apt-spy/README.mirrors.txt\fP.
+\fB@PREFIX@/lib/fink/mirror/sourceforge\fP.
 .TP
 .BI -v
diff -ru2N --exclude=.DS_Store apt-spy-3.0/apt-spy.conf.5 apt-spy-3.0-patched/apt-spy.conf.5
--- apt-spy-3.0/apt-spy.conf.5	Tue May 20 13:12:42 2003
+++ apt-spy-3.0-patched/apt-spy.conf.5	Sun Jun 22 18:11:08 2003
@@ -38,5 +38,5 @@
 This is the same as the ISO3166 code.
 You can find the country code for a specific country either by looking at
-\fB/usr/share/apt-spy/README.mirrors.txt,\fP or by looking at the commented
+\fB@PREFIX@/lib/fink/mirror/sourceforge,\fP or by looking at the commented
 examples in the default \fBapt-spy.conf\fP.
 .PP
diff -ru2N --exclude=.DS_Store apt-spy-3.0/file.c apt-spy-3.0-patched/file.c
--- apt-spy-3.0/file.c	Tue May 20 13:07:55 2003
+++ apt-spy-3.0-patched/file.c	Sun Jun 22 18:59:38 2003
@@ -101,5 +101,4 @@
 	/* We now read the line into the newly created buffer */
 	fgets(temp, buffsize + 1, infile_p);
-
 	return temp;
 }
diff -ru2N --exclude=.DS_Store apt-spy-3.0/main.c apt-spy-3.0-patched/main.c
--- apt-spy-3.0/main.c	Tue May 20 13:07:55 2003
+++ apt-spy-3.0-patched/main.c	Sun Jun 22 18:14:50 2003
@@ -20,9 +20,9 @@
 
 char d_area[] = "All";
-char d_out[] = "/etc/apt/sources.list";		/* Default outfile */
-char d_config[] = "/etc/apt-spy.conf";		/* Default configuration file */
-char d_mirror[] = "/usr/share/apt-spy/README.mirrors.txt";	/* Default mirror list */
-char d_file[] = "ls-lR";			/* Default file to grab when benchmarking */
-char d_update_url[] = "http://http.us.debian.org/debian/README.mirrors.txt";
+char d_out[] = "@PREFIX@/etc/apt/sources.list";		/* Default outfile */
+char d_config[] = "@PREFIX@/lib/fink/mirror/_keys";		/* Default configuration file */
+char d_mirror[] = "@PREFIX@/lib/fink/mirror/sourceforge";	/* Default mirror list */
+char d_file[] = "Packages.gz";			/* Default file to grab when benchmarking */
+char d_update_url[] = "http://cvs.sourceforge.net/cgi-bin/viewcvs.cgi/fink/mirrors/mirror/sourceforge?rev=HEAD";
 
 int main(int argc, char *argv[])
@@ -101,4 +101,7 @@
 	argv += optind;
 
+	/* Fink currently only provides debs in stable. */
+	if (distrib == NULL)
+		distrib = "stable";
 	/* We require an area and distribution argument if we are not updating */
 	if ((argc == 0) && (distrib == NULL))
diff -ru2N --exclude=.DS_Store apt-spy-3.0/parse.c apt-spy-3.0-patched/parse.c
--- apt-spy-3.0/parse.c	Tue May 20 13:07:55 2003
+++ apt-spy-3.0-patched/parse.c	Sun Jun 22 19:35:54 2003
@@ -23,6 +23,6 @@
 	while((line = next_entry(config_p)) != NULL) {
 		/* test for area string */
+		fprintf(stderr, "while((line = next_entry %s\n", line);
 		if ((tmp = strstr(line, area)) != NULL) {
-		
 			if ((strchr(tmp,':')) == NULL) { 	/* And for trailing colon.. */
 				free(line);
@@ -55,22 +55,36 @@
 		/* Skip blank lines */
 		country_code = line;
+fprintf(stderr, "skip blank lines %s\n", line);
 
 		/* We check for either a non-space or a '\n'. Has the useful side-effect of */
 		/* fast-forwarding country_code past any preceding whitespace.*/
-		while ((*country_code != '\n') && (isspace(*country_code) != 0))
+while ((*country_code != '\n') && (isspace(*country_code) != 0)) {
+fprintf(stderr, "fast-forward %s\n", line);
 			++country_code;
-		
+}
+fprintf(stderr, "fast-forward country %s\n", country_code);
+fprintf(stderr, "fast-forward line %s\n", line);
 		/* If country_code points to a '\n', there were no other characters. It was a blank line. */
 		/* If it points to a '#', there is a comment. We skip it too. */
 		if ((*country_code == '\n') || (*country_code == '#'))
 			continue;
+fprintf(stderr, "skip \\n # country %s\n", country_code);
+fprintf(stderr, "line %s\n", line);
 
-		if ((strchr(line, ':')) != NULL)
+		if ((strchr(line, ':')) == NULL)
 			return 0;			/* End of list. Return. */
+fprintf(stderr, "end of list.\n");
 
 		/* We do a little fiddling to get the country code down to 2 letters and a space */
+//*(country_code + 0) = *(country_code + 4);
+//*(country_code + 1) = *(country_code + 5);
+fprintf(stderr, "c= %s.\n", country_code);
+/*
 		*(country_code + 2) = ' ';
 		*(country_code + 3) = '\0';
-
+*/
+*(country_code + 6) = '\0';
+fprintf(stderr, "c truncated= '%s'.\n", country_code);
+ 
 		/* Sigh. The country code is in "country_code". We now parse the mirrors file for this */
 		/* and set the file position so that the next read will return the first */
@@ -111,12 +125,16 @@
 
 	/* Read until we find the country code */
-	while ((line = next_entry(mirror_list)) != NULL)
+	while ((line = next_entry(mirror_list)) != NULL) {
+		fprintf(stderr, "find_country line=%s", line);
 		if (strstr(line, country_code) != NULL)
 			break;
-
+	}
+	fprintf(stderr, "find_country, return= line=%s", line);
 	if (line == NULL)
 		return 1;
+	fprintf(stderr, "find_country2, return= line=%s", line);
 	
 	next_entry(mirror_list);		/* Skip a line */
+	fprintf(stderr, "find_country3, return= line=%s", line);
 	
 	if (ferror(mirror_list)) {
@@ -124,4 +142,5 @@
 		return 1;
 	}
+	fprintf(stderr, "find_country4, return= line=%s", line);
 	return 0;		/* We're positioned nicely for the next read */
 }
