diff -rU2 Extractor/ExtractorCommon/SensorsFile.cpp Extractor-patched/ExtractorCommon/SensorsFile.cpp
--- Extractor/ExtractorCommon/SensorsFile.cpp	Thu Oct 10 18:59:45 2002
+++ Extractor-patched/ExtractorCommon/SensorsFile.cpp	Tue Nov 12 01:38:11 2002
@@ -345,7 +346,11 @@
 		return -1;
 #ifndef _WIN32
-        if (lockf (fd, F_LOCK, 0) != 0)
+#    ifndef __APPLE__
+            if (lockf (fd, F_LOCK, 0) != 0)
+#    else
+            if (flock (fd, LOCK_EX) != 0)
+#    endif
 #else
-	if (_locking (fd, _LK_LOCK, 1) != 0)
+            if (_locking (fd, _LK_LOCK, 1) != 0)
 #endif
 		return -1;
@@ -358,5 +363,9 @@
         {
 #ifndef _WIN32
+#    ifndef __APPLE__
           lockf (fd, F_ULOCK, 0);
+#    else
+          flock (fd, LOCK_UN);
+#    endif
 #else
 	  _locking (fd, _LK_UNLCK, 1);
@@ -377,5 +386,10 @@
 		ftruncate (fd, nFilePos);
         lseek (fd, 0, SEEK_SET);
+#    ifndef __APPLE__
         lockf (fd, F_ULOCK, 0);
+#    else
+        flock (fd, LOCK_UN);
+#    endif
+        
 #else
         nFilePos = _tell (fd);
diff -rU2 Extractor/Makefile Extractor-patched/Makefile
--- Extractor/Makefile	Fri Oct  4 20:40:19 2002
+++ Extractor-patched/Makefile	Tue Nov 12 01:04:05 2002
@@ -2,6 +2,6 @@
 # All Rights Reserved
 
-EXTRACTORDIR=/usr/local/extractor
-BINDIR=/usr/local/bin
+EXTRACTORDIR=@PREFIX@/share/extractor
+BINDIR=@PREFIX@/bin
 PLATFORM=$(shell uname)
 
diff -rU2 Extractor/UnixExtractor/Makefile Extractor-patched/UnixExtractor/Makefile
--- Extractor/UnixExtractor/Makefile	Thu Oct 10 18:58:41 2002
+++ Extractor-patched/UnixExtractor/Makefile	Tue Nov 12 01:43:35 2002
@@ -36,5 +36,5 @@
 
 SOParser.o: ../ExtractorCommon/SOParser.cpp
-	$(CXX) -c -o SOParser.o ../ExtractorCommon/SOParser.cpp
+	$(CXX) -c $(IOPT) -o SOParser.o ../ExtractorCommon/SOParser.cpp
 
 ProfilesList.o: ProfilesList.cpp
diff -rU2 Extractor/UnixExtractor/bsd.mk Extractor-patched/UnixExtractor/bsd.mk
--- Extractor/UnixExtractor/bsd.mk	Fri Sep  6 16:05:13 2002
+++ Extractor-patched/UnixExtractor/bsd.mk	Tue Nov 12 01:56:33 2002
@@ -2,9 +2,10 @@
 # All Rights Reserved
 
-CC = gcc -Wall -O2
-CXX = g++ -Wall -O2
+CC = gcc -Wall -O2 -flat_namespace
+CXX = g++ -Wall -O2 -flat_namespace
 IOPT = -I/usr/local/include
 LOPT = -L/usr/local/lib
-LDSHARED = ld -shared
+#LDSHARED = ld -shared
+LDSHARED = ld -dynamic -flat_namespace -undefined suppress
 LIBDL =
 NEEDSSL = 0
diff -rU2 Extractor/UnixExtractor/platform.mk Extractor-patched/UnixExtractor/platform.mk
--- Extractor/UnixExtractor/platform.mk	Fri Sep  6 16:05:13 2002
+++ Extractor-patched/UnixExtractor/platform.mk	Tue Nov 12 01:03:45 2002
@@ -14,4 +14,7 @@
 include ../UnixExtractor/bsd.mk
 endif
+ifeq ($(PLATFORM),Darwin)
+include ../UnixExtractor/bsd.mk
+endif
 ifeq ($(PLATFORM),OpenBSD)
 include ../UnixExtractor/bsd.mk
