--- ./ltconfig	Thu Dec 12 01:51:15 2002
+++ ./ltconfig-prebind	Tue Feb 11 16:48:37 2003
@@ -1382,5 +1382,5 @@
       ;;
     esac
-    archive_cmds='$nonopt $(test .$module = .yes && echo -bundle || echo -dynamiclib) $allow_undefined_flag -o $lib $libobjs $deplibs $linkopts $(test .$module != .yes && echo -install_name $rpath/$soname $verstring)'
+    archive_cmds='$nonopt $(test .$module = .yes && echo -bundle || echo -dynamiclib) -o $lib $libobjs $deplibs $linkopts $(test .$module != .yes && echo -install_name $rpath/$soname $verstring) -prebind -seg1addr 0x00200000'
     # We need to add '_' to the symbols in $export_symbols first
     #archive_expsym_cmds="$archive_cmds"' && strip -s $export_symbols $lib'
