diff -ru1N localepurge-0.0.48/debian/README.debian localepurge-0.0.48-patched/debian/README.debian
--- localepurge-0.0.48/debian/README.debian	Thu May 16 18:14:48 2002
+++ localepurge-0.0.48-patched/debian/README.debian	Wed Jun 11 16:16:01 2003
@@ -21,4 +21,4 @@
 installation run.  It uses the dpkg "Post-Invoke" function to execute
-the script "/usr/sbin/localepurge", which automagically deletes the
-directory of every locale *not* defined in "/etc/locale.nopurge".  You
+the script "@PREFIX@/sbin/localepurge", which automagically deletes the
+directory of every locale *not* defined in "@PREFIX@/etc/locale.nopurge".  You
 have to put in the locale directory names you want to keep from removal
@@ -27,3 +27,3 @@
 system's locale directories won't be touched at all.  Any other
-directory (or file) in /usr/share/locale not containing a subdirectory
+directory (or file) in @PREFIX@/share/locale not containing a subdirectory
 named LC_MESSAGES will be discreetly ignored.
@@ -43,3 +43,3 @@
  All locale directories containing a subdirectory named LC_MESSAGES
- which are *not* listed in "/etc/locale.nopurge" will be irreversibly 
+ which are *not* listed in "@PREFIX@/etc/locale.nopurge" will be irreversibly 
  deleted.  The only way to reinstall any lost locales then is a complete 
@@ -50,3 +50,3 @@
 
- See "/usr/share/doc/localepurge/reinstall_debs.sh" for a ready made recipe.
+ See "@PREFIX@/share/doc/localepurge/reinstall_debs.sh" for a ready made recipe.
 
diff -ru1N localepurge-0.0.48/debian/localepurge.8 localepurge-0.0.48-patched/debian/localepurge.8
--- localepurge-0.0.48/debian/localepurge.8	Sun Jul 28 17:47:32 2002
+++ localepurge-0.0.48-patched/debian/localepurge.8	Wed Jun 11 16:16:01 2003
@@ -18,3 +18,3 @@
 in the 
-.BR /etc/locale.nopurge
+.BR @PREFIX@/etc/locale.nopurge
 configuration file.  Unless 
@@ -25,3 +25,3 @@
 Any other directory (or file) in 
-.BR /usr/share/locale
+.BR @PREFIX@/share/locale
 not containing a subdirectory named
@@ -33,3 +33,3 @@
 addition to locale files.  All actions are specified in the
-.BR /etc/locale.nopurge 
+.BR @PREFIX@/etc/locale.nopurge 
 configuration file. Since 
@@ -82,3 +82,3 @@
 which are either commented out or not even listed at all in
-.BR /etc/locale.nopurge 
+.BR @PREFIX@/etc/locale.nopurge 
 will be irreversibly deleted.  The only way to
@@ -86,3 +86,3 @@
 Debian packages containing them. See 
-.BR /usr/share/doc/localepurge/reinstall_debs.sh
+.BR @PREFIX@/share/doc/localepurge/reinstall_debs.sh
 for a recipe.
@@ -103,3 +103,3 @@
 .SH FILES
-.BR /etc/locale.nopurge
+.BR @PREFIX@/etc/locale.nopurge
 .SH SEE ALSO
@@ -109,5 +109,5 @@
 .br
-.BR /usr/share/doc/localepurge/README.debian
+.BR @PREFIX@/share/doc/localepurge/README.debian
 .br
-.BR /usr/share/doc/localepurge/reinstall_debs.sh
+.BR @PREFIX@/share/doc/localepurge/reinstall_debs.sh
 .SH AUTHOR
diff -ru1N localepurge-0.0.48/debian/localepurge.config localepurge-0.0.48-patched/debian/localepurge.config
--- localepurge-0.0.48/debian/localepurge.config	Sun Jan 26 10:44:53 2003
+++ localepurge-0.0.48-patched/debian/localepurge.config	Wed Jun 11 16:16:01 2003
@@ -7,7 +7,7 @@
 
-if [ ! -f /var/cache/localepurge/defaultlist ]; then
-    if [ ! -d /var/cache/localepurge ]; then
-	    mkdir -m 755 /var/cache/localepurge
+if [ ! -f @PREFIX@/var/cache/localepurge/defaultlist ]; then
+    if [ ! -d @PREFIX@/var/cache/localepurge ]; then
+	    mkdir -m 755 @PREFIX@/var/cache/localepurge
     fi
-    cat << ENDOFLIST > /var/cache/localepurge/defaultlist
+    cat << ENDOFLIST > @PREFIX@/var/cache/localepurge/defaultlist
 af
@@ -281,3 +281,5 @@
 
-. /usr/share/debconf/confmodule
+if [ -f @PREFIX@/share/debconf/confmodule ]; then
+. @PREFIX@/share/debconf/confmodule
+fi
 
@@ -286,3 +288,3 @@
 TEMPFILE=`tempfile`.$$
-LOCALELIST=/var/cache/localepurge/localelist
+LOCALELIST=@PREFIX@/var/cache/localepurge/localelist
 
@@ -291,5 +293,5 @@
 
-if [ -d /usr/share/locale ]; then
- for locale in `cd /usr/share/locale; ls .` 
-  do if [ -d /usr/share/locale/$locale/LC_MESSAGES ]; then 
+if [ -d @PREFIX@/share/locale ]; then
+ for locale in `cd @PREFIX@/share/locale; ls .` 
+  do if [ -d @PREFIX@/share/locale/$locale/LC_MESSAGES ]; then 
     echo $locale; fi; done >> "$TEMPFILE"
@@ -297,4 +299,4 @@
 
-if [ -d /usr/share/i18n/locales ]; then
- ls -1 /usr/share/i18n/locales/??_* | cut -d '/' -f 6 >> "$TEMPFILE"
+if [ -d @PREFIX@/share/i18n/locales ]; then
+ ls -1 @PREFIX@/share/i18n/locales/??_* | cut -d '/' -f 6 >> "$TEMPFILE"
 fi
@@ -306,4 +308,4 @@
 
-if [ -f /etc/locale.gen ]; then
-    grep ^[a-z] /etc/locale.gen | cut -d" " -f1 > "$LOCALEGEN"
+if [ -f @PREFIX@/etc/locale.gen ]; then
+    grep ^[a-z] @PREFIX@/etc/locale.gen | cut -d" " -f1 > "$LOCALEGEN"
 fi
@@ -312,4 +314,4 @@
 
-if [ -f /etc/locale.nopurge ]; then
-    grep ^[a-z] /etc/locale.nopurge | cut -d" " -f1 >> "$LOCALEGEN"
+if [ -f @PREFIX@/etc/locale.nopurge ]; then
+    grep ^[a-z] @PREFIX@/etc/locale.nopurge | cut -d" " -f1 >> "$LOCALEGEN"
 fi
@@ -327,4 +329,4 @@
 
-if [ -f /var/cache/localepurge/defaultlist ]; then
-    cat /var/cache/localepurge/defaultlist >> "$TEMPFILE"
+if [ -f @PREFIX@/etc/locale.nopurge/var/cache/localepurge/defaultlist ]; then
+    cat @PREFIX@/etc/locale.nopurge/var/cache/localepurge/defaultlist >> "$TEMPFILE"
 fi
@@ -353,4 +355,4 @@
 if [ ! -f "$LOCALELIST" ]; then
-    if [ ! -d /var/cache/localepurge ]; then
-	    mkdir -m 755 /var/cache/localepurge
+    if [ ! -d @PREFIX@/etc/locale.nopurge/var/cache/localepurge ]; then
+	    mkdir -m 755 @PREFIX@/etc/locale.nopurge/var/cache/localepurge
     fi
diff -ru1N localepurge-0.0.48/debian/localepurge.templates localepurge-0.0.48-patched/debian/localepurge.templates
--- localepurge-0.0.48/debian/localepurge.templates	Sun Apr 14 10:22:01 2002
+++ localepurge-0.0.48-patched/debian/localepurge.templates	Wed Jun 11 16:16:01 2003
@@ -10,3 +10,3 @@
  So please select both for best support of your national 
- language settings.  The entries from /etc/locale.gen will be 
+ language settings.  The entries from @PREFIX@/etc/locale.gen will be 
  preselected if no prior configuration has been successfully completed.
@@ -38,3 +38,3 @@
  it with the command "dpkg-reconfigure localepurge". The configured entries 
- from /etc/locale.gen of the locales package will then be automagically 
+ from @PREFIX@/etc/locale.gen of the locales package will then be automagically 
  preselected.
diff -ru1N localepurge-0.0.48/debian/postinst localepurge-0.0.48-patched/debian/postinst
--- localepurge-0.0.48/debian/postinst	Thu May 16 18:16:05 2002
+++ localepurge-0.0.48-patched/debian/postinst	Wed Jun 11 16:16:01 2003
@@ -7,3 +7,3 @@
 
-CACHEDIR=/var/cache/localepurge
+CACHEDIR=@PREFIX@/var/cache/localepurge
 DEFAULTLIST=$CACHEDIR/defaultlist
@@ -21,5 +21,7 @@
  
-CONFIGFILE=/etc/locale.nopurge
+CONFIGFILE=@PREFIX@/etc/locale.nopurge
 
-. /usr/share/debconf/confmodule
+if [ -f @PREFIX@/share/debconf/confmodule ]; then
+. @PREFIX@/share/debconf/confmodule
+fi
 
@@ -114,3 +116,3 @@
 # present on the system and this will result in the definite loss 
-# of all locale data in "/usr/share/locale/". You should know what
+# of all locale data in "@PREFIX@/share/locale/". You should know what
 # you are doing or reconfigure "localepurge" as described above.
diff -ru1N localepurge-0.0.48/debian/postrm localepurge-0.0.48-patched/debian/postrm
--- localepurge-0.0.48/debian/postrm	Sun Apr 14 10:32:02 2002
+++ localepurge-0.0.48-patched/debian/postrm	Wed Jun 11 16:16:01 2003
@@ -4,7 +4,7 @@
 
-if [ "$1" = purge -a -e /usr/share/debconf/confmodule ]; then
-    . /usr/share/debconf/confmodule
+if [ "$1" = purge -a -e @PREFIX@/share/debconf/confmodule ]; then
+    . @PREFIX@/share/debconf/confmodule
     db_purge
-    rm -rf /var/cache/localepurge
-    rm -f /etc/locale.nopurge
+    rm -rf @PREFIX@/var/cache/localepurge
+    rm -f @PREFIX@/etc/locale.nopurge
 
diff -ru1N localepurge-0.0.48/debian/preinst localepurge-0.0.48-patched/debian/preinst
--- localepurge-0.0.48/debian/preinst	Sun Aug  4 19:42:24 2002
+++ localepurge-0.0.48-patched/debian/preinst	Wed Jun 11 16:16:01 2003
@@ -4,12 +4,12 @@
 
-if [ -f /etc/locale.purge ]
+if [ -f @PREFIX@/etc/locale.purge ]
   then 
-    /bin/mv /etc/locale.purge /etc/locale.purge-obsolete
-    /usr/bin/mail -n -s "Upgrade of localepurge" root@localhost << EOF
-    The configuration file /etc/locale.purge has become obsolete and
+    mv @PREFIX@/etc/locale.purge @PREFIX@/etc/locale.purge-obsolete
+    echo '
+    The configuration file @PREFIX@/etc/locale.purge has become obsolete and
     can be safely removed from your system.  As reminder, it has been
-    renamed to /etc/locale.purge-obsolete.
+    renamed to @PREFIX@/etc/locale.purge-obsolete.
 
     Note please, that the removal mechanism of localepurge has been
-    inverted.  It now keeps locales as listed in "/etc/locale.nopurge"
+    inverted.  It now keeps locales as listed in "@PREFIX@/etc/locale.nopurge"
     and deletes the files contained in all else locale directories having 
@@ -18,3 +18,3 @@
                                 Yours truly, Root
-EOF
+'
 fi
diff -ru1N localepurge-0.0.48/etc/apt/apt.conf.d/99-localepurge localepurge-0.0.48-patched/etc/apt/apt.conf.d/99-localepurge
--- localepurge-0.0.48/etc/apt/apt.conf.d/99-localepurge	Wed Jun 26 06:50:21 2002
+++ localepurge-0.0.48-patched/etc/apt/apt.conf.d/99-localepurge	Wed Jun 11 16:16:01 2003
@@ -4,3 +4,3 @@
 {
-Post-Invoke {"if [ -x /usr/sbin/localepurge ] && [ `ps aux | grep "$PPID" | grep -c remove` != 1 ]; then /usr/sbin/localepurge; else exit 0; fi";};
+Post-Invoke {"if [ -x @PREFIX@/sbin/localepurge ] && [ `ps aux | grep "$PPID" | grep -c remove` != 1 ]; then @PREFIX@/sbin/localepurge; else exit 0; fi";};
 };
diff -ru1N localepurge-0.0.48/usr/sbin/localepurge localepurge-0.0.48-patched/usr/sbin/localepurge
--- localepurge-0.0.48/usr/sbin/localepurge	Thu Aug  8 05:16:08 2002
+++ localepurge-0.0.48-patched/usr/sbin/localepurge	Wed Jun 11 16:16:01 2003
@@ -3,3 +3,3 @@
 # Deleting all locale files on system
-# *not* listed in /etc/locale.nopurge
+# *not* listed in @PREFIX@/etc/locale.nopurge
 
@@ -12,3 +12,3 @@
 
-if [ "`grep -x ^SHOWFREEDSPACE /etc/locale.nopurge`" ]; then
+if [ "`grep -x ^SHOWFREEDSPACE @PREFIX@/etc/locale.nopurge`" ]; then
     SHOWFREEDSPACE=enabled
@@ -16,3 +16,3 @@
 
-if [ "`grep -x ^VERBOSE /etc/locale.nopurge`" ] \
+if [ "`grep -x ^VERBOSE @PREFIX@/etc/locale.nopurge`" ] \
     || [ "$1" = "-verbose" ] || [ "$1" = "-v" ] \
@@ -22,4 +22,4 @@
 
-LOCALEDIR=/usr/share/locale
-MANPAGEDIR=/usr/share/man
+LOCALEDIR=@PREFIX@/share/locale
+MANPAGEDIR=@PREFIX@/share/man
 
@@ -27,8 +27,8 @@
 
-if [ ! -f /etc/locale.nopurge ]
+if [ ! -f @PREFIX@/etc/locale.nopurge ]
     then
-        echo " No /etc/locale.nopurge file present, exiting ..."
+        echo " No @PREFIX@/etc/locale.nopurge file present, exiting ..."
         exit 0
     else
-        if [ "`grep -x ^NEEDSCONFIGFIRST /etc/locale.nopurge`" ]
+        if [ "`grep -x ^NEEDSCONFIGFIRST @PREFIX@/etc/locale.nopurge`" ]
             then
@@ -49,3 +49,3 @@
 
-LOCALELIST=/var/cache/localepurge/localelist
+LOCALELIST=@PREFIX@/var/cache/localepurge/localelist
 NEWLOCALELIST="$LOCALELIST"-new
@@ -56,5 +56,5 @@
 
-for NEWLOCALE in $(cd /usr/share/locale; ls .)
+for NEWLOCALE in $(cd @PREFIX@/share/locale; ls .)
  do 
- if [ -d /usr/share/locale/$NEWLOCALE/LC_MESSAGES ]; then
+ if [ -d @PREFIX@/share/locale/$NEWLOCALE/LC_MESSAGES ]; then
      if [ ! "$(grep -cx $NEWLOCALE $LOCALELIST)" = "1" ]; then
@@ -96,3 +96,3 @@
          do
-          if [ ! "`grep -x ^$LOCALE /etc/locale.nopurge`" ] && \
+          if [ ! "`grep -x ^$LOCALE @PREFIX@/etc/locale.nopurge`" ] && \
              [ "`grep -x ^$LOCALE $LOCALELIST`" ]; then
@@ -106,3 +106,3 @@
          do
-          if [ ! "`grep -x ^$LOCALE /etc/locale.nopurge`" ] && \
+          if [ ! "`grep -x ^$LOCALE @PREFIX@/etc/locale.nopurge`" ] && \
              [ "`grep -x ^$LOCALE $LOCALELIST`" ]; then
@@ -126,3 +126,3 @@
 
-if [ -d $MANPAGEDIR ] && [ "`grep -x ^MANDELETE /etc/locale.nopurge`" ]; then
+if [ -d $MANPAGEDIR ] && [ "`grep -x ^MANDELETE @PREFIX@/etc/locale.nopurge`" ]; then
 
@@ -136,3 +136,3 @@
          do
-          if [ ! "`grep -x ^$LOCALE /etc/locale.nopurge`" ] && \
+          if [ ! "`grep -x ^$LOCALE @PREFIX@/etc/locale.nopurge`" ] && \
              [ "`grep -x ^$LOCALE $LOCALELIST`" ]; then
@@ -146,3 +146,3 @@
          do
-          if [ ! "`grep -x ^$LOCALE /etc/locale.nopurge`" ] && \
+          if [ ! "`grep -x ^$LOCALE @PREFIX@/etc/locale.nopurge`" ] && \
              [ "`grep -x ^$LOCALE $LOCALELIST`" ]; then
