diff -ru2N ./Makefile ./Makefile-patched
--- ./Makefile	Wed Feb 22 22:53:10 1995
+++ ./Makefile-patched	Sun May  4 04:50:51 2003
@@ -18,5 +18,5 @@
 #
 # Optimization/debug flags as needed
-CDEBUGFLAGS=	-O
+CDEBUGFLAGS=	-O3 -prebind
 # 
 # If you are using a system where malloc(0) or realloc(ptr, 0) returns
@@ -24,5 +24,5 @@
 # -DMALLOC_0_RETURNS_NULL.  If you don't know whether or not to define
 # it, run "make malloc-test".
-MALLOC_DEFINE=	-DMALLOC_0_RETURNS_NULL
+#MALLOC_DEFINE=	-DMALLOC_0_RETURNS_NULL
 #
 # If you are compiling on a system that has an st_blocks field in
@@ -35,6 +35,6 @@
 # Change these as needed.
 CC= 		cc
-DEPEND=		/usr/bin/X11/makedepend
-COMPILE_ET= 	com_err/compile_et
+DEPEND=		/usr/X11R6/bin/makedepend
+COMPILE_ET= 	/System/Library/Frameworks/Kerberos.framework/Versions/A/Support/compile_et
 LINT= 		lint
 # Command used to create the symlink from purge to expunge
@@ -56,9 +56,9 @@
 # This is the installation directory for binaries. The executables
 # are installed here.
-BINDIR= 	/usr/bin
+BINDIR= 	@PREFIX@/bin
 #
 # This is the installation directory for manual pages. The manual pages are
 # installed here.
-MANDIR=		/usr/man
+MANDIR=		@PREFIX@/share/man
 #
 # This is the directory for any temporary file.
@@ -92,6 +92,6 @@
 # library is stored.
 # 
-ETINCS=		-Icom_err
-ETLDFLAGS=	-Lcom_err
+ETINCS=		-I/usr/include
+ETLDFLAGS=	-L/usr/lib/libcom_err.dylib
 
 ###########################################################
diff -ru2N ./util.c ./util.c-patched
--- ./util.c	Wed Feb 22 22:42:40 1995
+++ ./util.c-patched	Sun May  4 04:20:08 2003
@@ -15,5 +15,5 @@
 #include <sys/param.h>
 #include <sys/types.h>
-#if defined(SYSV) && !(defined(hpux) || defined(sun) || defined(linux)) /* SYSV doesn't define uid_t */
+#if defined(SYSV) && !(defined(hpux) || defined(sun) || defined(linux)||defined(__APPLE__)) /* SYSV doesn't define uid_t */
 typedef unsigned short uid_t;
 #endif
