diff -ru2N ./usermin-1.020/config-macos ./usermin-1.020-patched/config-macos
--- ./usermin-1.020/config-macos	Sun May 11 19:49:14 2003
+++ ./usermin-1.020-patched/config-macos	Thu May 22 16:32:16 2003
@@ -1,4 +1,4 @@
 find_pid_command=ps auxwwww | grep NAME | grep -v grep | awk '{ print $2 }'
-path=/bin:/usr/bin:/sbin:/usr/sbin:/usr/local/bin:/usr/libexec
+path=@PREFIX@/bin:@PREFIX@/sbin:/bin:/usr/bin:/sbin:/usr/sbin:/usr/local/bin:/usr/libexec
 ld_env=LD_LIBRARY_PATH
 passwd_file=nidump passwd . |
diff -ru2N ./usermin-1.020/setup.sh ./usermin-1.020-patched/setup.sh
--- ./usermin-1.020/setup.sh	Sun May 11 19:49:13 2003
+++ ./usermin-1.020-patched/setup.sh	Thu May 22 16:30:42 2003
@@ -55,10 +55,10 @@
 echo "you can just accept the defaults."
 echo ""
-printf "Config file directory [/etc/usermin]: "
+printf "Config file directory [@PREFIX@/etc/usermin]: "
 if [ "$config_dir" = "" ]; then
 	read config_dir
 fi
 if [ "$config_dir" = "" ]; then
-	config_dir=/etc/usermin
+	config_dir=@PREFIX@/etc/usermin
 fi
 abspath=`echo $config_dir | grep "^/"`
@@ -148,5 +148,5 @@
 	# Config directory exists .. make sure it is not in use
 	ls $config_dir | grep -v rpmsave >/dev/null 2>&1
-	if [ "$?" = "0" -a "$config_dir" != "/etc/usermin" ]; then
+	if [ "$?" = "0" -a "$config_dir" != "@PREFIX@/etc/usermin" ]; then
 		echo "ERROR: Config directory $config_dir is not empty"
 		echo ""
@@ -155,10 +155,10 @@
 
 	# Ask for log directory
-	printf "Log file directory [/var/usermin]: "
+	printf "Log file directory [@PREFIX@/var/usermin]: "
 	if [ "$var_dir" = "" ]; then
 		read var_dir
 	fi
 	if [ "$var_dir" = "" ]; then
-		var_dir=/var/usermin
+		var_dir=@PREFIX@/var/usermin
 	fi
 	abspath=`echo $var_dir | grep "^/"`
@@ -530,5 +530,5 @@
 	chown -R root:bin "$wadir"
 	chmod -R og-w "$wadir"
-	if [ $var_dir != "/var" ]; then
+	if [ $var_dir != "@PREFIX@/var" ]; then
 		chown -R root:bin $var_dir
 		chmod -R og-w $var_dir
