--- ./usermin-0.970/config-macos	Mon Jul 15 11:40:13 2002
+++ ./usermin-0.970-patched/config-macos	Sun Sep  1 02:18:06 2002
@@ -1,4 +1,4 @@
 find_pid_command=ps auxwwww | grep NAME | grep -v grep | awk '{ print $2 }'
-path=/bin:/usr/bin:/sbin:/usr/sbin:/usr/local/bin:/usr/libexec
+path=@PREFIX@/bin:@PREFIX@/sbin:/bin:/usr/bin:/sbin:/usr/sbin:/usr/local/bin:/usr/libexec
 ld_env=LD_LIBRARY_PATH
 passwd_file=nidump passwd . |
--- ./usermin-0.970/setup.sh	Mon Jul 15 11:40:13 2002
+++ ./usermin-0.970-patched/setup.sh	Mon Sep  2 19:31:24 2002
@@ -47,10 +47,10 @@
 echo "you can just accept the defaults."
 echo ""
-printf "Config file directory [/etc/usermin]: "
+printf "Config file directory [@PREFIX@/etc/usermin]: "
 if [ "$config_dir" = "" ]; then
 	read config_dir
 fi
 if [ "$config_dir" = "" ]; then
-	config_dir=/etc/usermin
+	config_dir=@PREFIX@/etc/usermin
 fi
 abspath=`echo $config_dir | grep "^/"`
@@ -135,5 +135,5 @@
 	# Config directory exists .. make sure it is not in use
 	ls $config_dir | grep -v rpmsave >/dev/null 2>&1
-	if [ "$?" = "0" -a "$config_dir" != "/etc/usermin" ]; then
+	if [ "$?" = "0" -a "$config_dir" != "@PREFIX@/etc/usermin" ]; then
 		echo "ERROR: Config directory $config_dir is not empty"
 		echo ""
@@ -142,10 +142,10 @@
 
 	# Ask for log directory
-	printf "Log file directory [/var/usermin]: "
+	printf "Log file directory [@PREFIX@/var/usermin]: "
 	if [ "$var_dir" = "" ]; then
 		read var_dir
 	fi
 	if [ "$var_dir" = "" ]; then
-		var_dir=/var/usermin
+		var_dir=@PREFIX@/var/usermin
 	fi
 	abspath=`echo $var_dir | grep "^/"`
