diff -ru2N ./common/HttpClient/Makefile ./patched/common/HttpClient/Makefile
--- ./common/HttpClient/Makefile	Fri Jan  3 18:21:58 2003
+++ ./patched/common/HttpClient/Makefile	Thu Jan 30 09:18:26 2003
@@ -7,5 +7,5 @@
 DEFS= -D_REENTRANT
 AR= ar cq
-RANLIB= ranlib
+RANLIB= ranlib -s
 LIBNAME= libhttp.a
 
@@ -15,5 +15,5 @@
 
 DEFINES= $(INCLUDES) $(DEFS)
-CFLAGS= -g $(DEFINES)
+CFLAGS= -O3 $(DEFINES)
 
 SRCS = HttpClient.cpp HttpClientResponse.cpp HttpSession.cpp ../Base64.cpp
diff -ru2N ./common/Makefile ./patched/common/Makefile
--- ./common/Makefile	Fri Jan  3 18:21:58 2003
+++ ./patched/common/Makefile	Thu Jan 30 09:18:34 2003
@@ -7,5 +7,5 @@
 DEFS= -D_REENTRANT
 AR= ar cq
-RANLIB= ranlib
+RANLIB= ranlib -s
 LIBNAME= libcommon.a
 
@@ -15,5 +15,5 @@
 
 DEFINES= $(INCLUDES) $(DEFS)
-CFLAGS= -g $(DEFINES)
+CFLAGS= -O3 $(DEFINES)
 
 SRCS = Base64.cpp LogFile.cpp vsscanf.cpp YahooStuff.cpp YahooSessionCache.cpp SmtpListener.cpp
diff -ru2N ./common/lib/mimepp/Makefile ./patched/common/lib/mimepp/Makefile
--- ./common/lib/mimepp/Makefile	Fri Jan  3 18:21:55 2003
+++ ./patched/common/lib/mimepp/Makefile	Thu Jan 30 09:19:35 2003
@@ -23,15 +23,15 @@
 CXX = c++
 AR = ar
-RANLIB = ranlib
+RANLIB = ranlib -s
 
 ###### Compiler/linker flags
 
-DBG_CXX_FLAGS = -DDW_DEBUG_VERSION -g -D_REENTRANT -fpic -fexceptions -I./src
+DBG_CXX_FLAGS = -DDW_DEBUG_VERSION -g -D_REENTRANT -fPIC -fexceptions -I./src
 DBG_LINK_FLAGS = 
 
-DEV_CXX_FLAGS = -DDW_DEVELOPMENT_VERSION -g -D_REENTRANT -fpic -fexceptions -I./src
+DEV_CXX_FLAGS = -DDW_DEVELOPMENT_VERSION -g -D_REENTRANT -fPIC -fexceptions -I./src
 DEV_LINK_FLAGS = 
 
-REL_CXX_FLAGS = -DDW_PRODUCTION_VERSION -D_REENTRANT -fpic -fexceptions -I./src
+REL_CXX_FLAGS = -DDW_PRODUCTION_VERSION -O3 -D_REENTRANT -fPIC -fexceptions -I./src
 REL_LINK_FLAGS = 
 
diff -ru2N ./common/lib/re_lib/Makefile ./patched/common/lib/re_lib/Makefile
--- ./common/lib/re_lib/Makefile	Fri Jan  3 18:21:58 2003
+++ ./patched/common/lib/re_lib/Makefile	Thu Jan 30 09:20:49 2003
@@ -25,5 +25,5 @@
 
 #CFLAGS= -O $(ENV) $(TEST) $(PROF)
-CFLAGS= -g $(ENV) $(TEST) $(PROF) -I../../
+CFLAGS= -O3 $(ENV) $(TEST) $(PROF) -I../../
 LDFLAGS=
 
@@ -92,5 +92,5 @@
 lib:	regexp.o regsub.o regerror.o relib_enhance.o 
 	ar cq libre_lib.a regexp.o regsub.o regerror.o relib_enhance.o ../../vsscanf.o
-	ranlib libre_lib.a
+	ranlib -s libre_lib.a
 
 depend:
diff -ru2N ./unix/Makefile ./patched/unix/Makefile
--- ./unix/Makefile	Fri Jan  3 18:22:56 2003
+++ ./patched/unix/Makefile	Thu Jan 30 09:16:07 2003
@@ -7,6 +7,6 @@
 DEFS= -D_REENTRANT 
 PROGNAME= yahoopops
-INCLUDES=  -I. -I../common -I../common/lib/re_lib -I/home/users/o/os/osdndevel/ypmac/openssl/include -I../common/lib/mimepp/src -I../common/lib/mimepp/examples/email
-LIBS= ../common/lib/mimepp/rel/libmimepp_core.a ../common/lib/mimepp/rel/libmimepp_util.a ../common/lib/mimepp/rel/libmimepp_net.a ../common/libcommon.a ../common/HttpClient/libhttp.a ../common/lib/re_lib/libre_lib.a -L/home/users/o/os/osdndevel/ypmac/openssl/lib -lcrypto -L/home/users/o/os/osdndevel/ypmac/openssl/lib -lssl -lpthread -lstdc++
+INCLUDES=  -I. -I../common -I../common/lib/re_lib -I@PREFIX@/include -I../common/lib/mimepp/src -I../common/lib/mimepp/examples/email
+LIBS= ../common/lib/mimepp/rel/libmimepp_core.a ../common/lib/mimepp/rel/libmimepp_util.a ../common/lib/mimepp/rel/libmimepp_net.a ../common/libcommon.a ../common/HttpClient/libhttp.a ../common/lib/re_lib/libre_lib.a -lcrypto -L@PREFIX@/lib -lssl -lpthread -lstdc++
 
 
@@ -14,5 +14,5 @@
 
 DEFINES= $(INCLUDES) $(DEFS)
-CFLAGS= -g $(DEFINES)
+CFLAGS= -O3 $(DEFINES)
 
 SRCS = ConfigAdvancedPage.cpp ConfigEmailPage.cpp ConfigEmailPreferencesPage.cpp ConfigLogPage.cpp ConfigMiscPage.cpp ConfigNetworkPage.cpp ConfigProxyPage.cpp ConfigSecurityPage.cpp CustomPropertyPage.cpp DataHolder.cpp DisclaimerDlg.cpp Email.cpp HyperLink.cpp mainfrm.cpp OptionsTreeDlg.cpp ProgressWnd.cpp ServiceClient.cpp stdafx.cpp trayicon.cpp WebBrowser.cpp YAHOOPOPS.cpp
