diff -uNbr libcapsinetwork-0.2.0/Makefile.am libcapsinetwork-0.2.0-new/Makefile.am
--- libcapsinetwork-0.2.0/Makefile.am	Thu Oct 17 18:57:50 2002
+++ libcapsinetwork-0.2.0-new/Makefile.am	Sun Oct 27 20:52:17 2002
@@ -2,7 +2,7 @@
 
 lib_LTLIBRARIES = libcapsinetwork.la
 
-libcapsinetwork_la_SOURCES = listener.cc listenport.cc socket.cc
+libcapsinetwork_la_SOURCES = listener.cc listenport.cc socket.cc server.cc
 libcapsinetwork_la_LDFLAGS = -no-undefined -version-info 0:1:0
 
 libcapsinetworkinc_HEADERS = listener.h listenport.h socket.h
diff -uNbr libcapsinetwork-0.2.0/Makefile.in libcapsinetwork-0.2.0-new/Makefile.in
--- libcapsinetwork-0.2.0/Makefile.in	Tue Oct 22 16:27:24 2002
+++ libcapsinetwork-0.2.0-new/Makefile.in	Sun Oct 27 20:56:44 2002
@@ -89,7 +89,7 @@
 
 lib_LTLIBRARIES = libcapsinetwork.la
 
-libcapsinetwork_la_SOURCES = listener.cc listenport.cc socket.cc
+libcapsinetwork_la_SOURCES = listener.cc listenport.cc socket.cc server.cc
 libcapsinetwork_la_LDFLAGS = -no-undefined -version-info 0:1:0
 
 libcapsinetworkinc_HEADERS = listener.h listenport.h socket.h
@@ -108,7 +108,7 @@
 LTLIBRARIES = $(lib_LTLIBRARIES)
 
 libcapsinetwork_la_LIBADD =
-am_libcapsinetwork_la_OBJECTS = listener.lo listenport.lo socket.lo
+am_libcapsinetwork_la_OBJECTS = listener.lo listenport.lo socket.lo server.lo
 libcapsinetwork_la_OBJECTS = $(am_libcapsinetwork_la_OBJECTS)
 noinst_PROGRAMS = exampleserver$(EXEEXT)
 PROGRAMS = $(noinst_PROGRAMS)
diff -uNbr libcapsinetwork-0.2.0/configure libcapsinetwork-0.2.0-new/configure
--- libcapsinetwork-0.2.0/configure	Tue Oct 22 16:27:26 2002
+++ libcapsinetwork-0.2.0-new/configure	Sun Oct 27 20:57:10 2002
@@ -5559,7 +5559,7 @@
     # FIXME: Relying on posixy $() will cause problems for
     #        cross-compilation, but unfortunately the echo tests do not
     #        yet detect zsh echo's removal of \ escapes.
-    archive_cmds='$CC $(test .$module = .yes && echo -bundle || echo -dynamiclib) $allow_undefined_flag -o $lib $libobjs $deplibs$linkopts -install_name $rpath/$soname $(test -n "$verstring" -a x$verstring != x0.0 && echo $verstring)'
+    archive_cmds='g++ $(test .$module = .yes && echo -bundle || echo -dynamiclib) $allow_undefined_flag -o $lib $libobjs $deplibs$linkopts -install_name $rpath/$soname $(test -n "$verstring" -a x$verstring != x0.0 && echo $verstring)'
     # We need to add '_' to the symbols in $export_symbols first
     #archive_expsym_cmds="$archive_cmds"' && strip -s $export_symbols'
     hardcode_direct=yes
diff -uNbr libcapsinetwork-0.2.0/listener.cc libcapsinetwork-0.2.0-new/listener.cc
--- libcapsinetwork-0.2.0/listener.cc	Tue Oct 22 16:23:26 2002
+++ libcapsinetwork-0.2.0-new/listener.cc	Sun Oct 27 20:14:30 2002
@@ -14,17 +14,18 @@
 // the Free Software Foundation, Inc., 59 Temple Place - Suite 330,
 // Boston, MA 02111-1307, USA.
 
-//#include <netinet/in.h>
+#include <sys/types.h>
+#include <netinet/in.h>
 #include <arpa/inet.h>
 #include <netdb.h>
 #include <sys/socket.h>
-#include <sys/types.h>
 #include <unistd.h>
 
 #include "listener.h"
 #include "listenport.h"
 #include "socket.h"
 
+#define socklen_t int
 #define	MAXLINE	1024
 
 extern int errno;
diff -uNbr libcapsinetwork-0.2.0/listenport.cc libcapsinetwork-0.2.0-new/listenport.cc
--- libcapsinetwork-0.2.0/listenport.cc	Thu Oct 17 22:56:36 2002
+++ libcapsinetwork-0.2.0-new/listenport.cc	Sun Oct 27 20:15:07 2002
@@ -18,9 +18,9 @@
 #include <arpa/inet.h>
 #include <fcntl.h>
 #include <netdb.h>
-//#include <netinet/in.h>
-#include <sys/socket.h>
 #include <sys/types.h>
+#include <netinet/in.h>
+#include <sys/socket.h>
 #include <unistd.h>
 
 #include "listenport.h"
