diff -uNbr libmath++-0.0.3/examples/Makefile.in libmath++-0.0.3-new/examples/Makefile.in
--- libmath++-0.0.3/examples/Makefile.in	2002-07-01 22:46:04.000000000 -0500
+++ libmath++-0.0.3-new/examples/Makefile.in	2002-07-30 09:24:25.000000000 -0500
@@ -92,7 +92,7 @@
 am__quote = @am__quote@
 install_sh = @install_sh@
 
-CXXFLAGS = -pedantic -ansi -Wall -Wno-long-long
+CXXFLAGS = -pedantic -ansi -Wall -Wno-long-long -I$(top_srcdir)/math++
 
 LIBS = ../math++/libmath++.la
 
diff -uNbr libmath++-0.0.3/math++/Makefile.in libmath++-0.0.3-new/math++/Makefile.in
--- libmath++-0.0.3/math++/Makefile.in	2002-07-01 22:46:05.000000000 -0500
+++ libmath++-0.0.3-new/math++/Makefile.in	2002-07-30 09:24:25.000000000 -0500
@@ -92,7 +92,7 @@
 am__quote = @am__quote@
 install_sh = @install_sh@
 
-CXXFLAGS = -pedantic -ansi -Wall -Wno-long-long
+CXXFLAGS = -pedantic -ansi -Wall -Wno-long-long -I$(top_srcdir)/math++
 
 lib_LTLIBRARIES = libmath++.la
 
