diff -uNbr libxslt-1.0.29/breakpoint/Makefile.in libxslt-1.0.29-new/breakpoint/Makefile.in
--- libxslt-1.0.29/breakpoint/Makefile.in	Tue Apr  1 08:40:05 2003
+++ libxslt-1.0.29-new/breakpoint/Makefile.in	Mon May 26 03:41:24 2003
@@ -129,7 +129,7 @@
 	deprecated.c
 
 
-libxsltbreakpoint_la_LIBADD = 
+libxsltbreakpoint_la_LIBADD = ../libxslt/libxslt.la
 libxsltbreakpoint_la_LDFLAGS = -version-info @LIBXSLTBREAK_VERSION_INFO@
 mkinstalldirs = $(SHELL) $(top_srcdir)/mkinstalldirs
 CONFIG_HEADER = ../config.h
diff -uNbr libxslt-1.0.29/libexslt/Makefile.in libxslt-1.0.29-new/libexslt/Makefile.in
--- libxslt-1.0.29/libexslt/Makefile.in	Tue Apr  1 08:39:25 2003
+++ libxslt-1.0.29-new/libexslt/Makefile.in	Mon May 26 03:42:55 2003
@@ -153,7 +153,7 @@
 # The following DOES NOT WORK reliably. Sorry no prelinking to uninstalled
 # yet libraries.
 # libexslt_la_LIBADD = $(top_builddir)/libxslt/libxslt.la $(EXTRA_LIBS)
-libexslt_la_LIBADD = $(INSTALLED_XSLT_LIB) $(EXTRA_LIBS)
+libexslt_la_LIBADD = $(INSTALLED_XSLT_LIB) $(EXTRA_LIBS) ../libxslt/libxslt.la
 libexslt_la_LDFLAGS = -version-info @LIBEXSLT_VERSION_INFO@
 
 man_MANS = libexslt.4
diff -uNbr libxslt-1.0.29/python/Makefile.in libxslt-1.0.29-new/python/Makefile.in
--- libxslt-1.0.29/python/Makefile.in	Tue Apr  1 08:39:28 2003
+++ libxslt-1.0.29-new/python/Makefile.in	Mon May 26 03:45:25 2003
@@ -150,7 +150,7 @@
 
 
 libxsltmod_la_LDFLAGS = -module -avoid-version -L$(top_builddir)/libxslt/.libs \
-                        -L$(top_builddir)/libexslt/.libs
+                        -L$(top_builddir)/libexslt/.libs -Wl,-undefined -Wl,suppress -Wl,-flat_namespace
 
 
 @WITH_PYTHON_TRUE@mylibs = @WITH_PYTHON_TRUE@\
