diff -uNr arts-1.3.92/admin/ltmain.sh arts-1.3.92-new/admin/ltmain.sh
--- arts-1.3.92/admin/ltmain.sh	Mon Jan  3 09:24:20 2005
+++ arts-1.3.92-new/admin/ltmain.sh	Sat Feb  5 15:34:40 2005
@@ -4453,6 +4453,7 @@
 	# We should set the shlibpath_var
 	rpath=
 	for dir in $temp_rpath; do
+     if test "$dir" != "@FINKPREFIX@/lib"; then
 	  case $dir in
 	  [\\/]* | [A-Za-z]:[\\/]*)
 	    # Absolute path.
@@ -4463,6 +4464,7 @@
 	    rpath="$rpath\$thisdir/$dir:"
 	    ;;
 	  esac
+     fi
 	done
 	temp_rpath="$rpath"
       fi
diff -uNr arts-1.3.92/configure arts-1.3.92-new/configure
--- arts-1.3.92/configure	Fri Feb  4 05:44:16 2005
+++ arts-1.3.92-new/configure	Sat Feb  5 15:34:41 2005
@@ -22718,6 +22718,9 @@
 	darwin*)
 	  libltdl_cv_sys_dlopen_deplibs=yes
 	  ;;
+	darwin*)
+	  libltdl_cv_sys_dlopen_deplibs=yes
+	  ;;
 	linux*)
 	  libltdl_cv_sys_dlopen_deplibs=yes
 	  ;;
diff -uNr arts-1.3.92/examples/Makefile.in arts-1.3.92-new/examples/Makefile.in
--- arts-1.3.92/examples/Makefile.in	Fri Feb  4 05:44:13 2005
+++ arts-1.3.92-new/examples/Makefile.in	Sat Feb  5 15:34:41 2005
@@ -555,7 +555,7 @@
 testasubsys_SOURCES = testasubsys.cc
 testasubsys_LDADD = $(FLOWLIBS)
 testthreads_SOURCES = testthreads.cc
-testthreads_LDFLAGS = $(LDFLAGS)
+testthreads_LDFLAGS = $(all_libraries)
 testthreads_LDADD = $(top_builddir)/mcop_mt/libmcop_mt.la
 testnothreads_SOURCES = testthreads.cc
 testaggregation_SOURCES = aggregation.cc testaggregation.cc
@@ -575,7 +575,7 @@
 artsccat_SOURCES = artsccat.c
 artscrec_LDADD = $(top_builddir)/artsc/libartsc.la
 artscrec_SOURCES = artscrec.c
-artscmt_LDFLAGS = $(LDFLAGS) $(USE_THREADS)
+artscmt_LDFLAGS = $(all_libraries) $(USE_THREADS)
 artscmt_LDADD = $(top_builddir)/artsc/libartsc.la -lm $(LIBPTHREAD) $(LIBPOSIX4)
 artscmt_SOURCES = artscmt.c
 playtofile_SOURCES = playtofile.cc playtofile_impl.cc playtofile_main.cc
diff -uNr arts-1.3.92/mcop/tcpserver.cc arts-1.3.92-new/mcop/tcpserver.cc
--- arts-1.3.92/mcop/tcpserver.cc	Fri Nov 29 11:51:13 2002
+++ arts-1.3.92-new/mcop/tcpserver.cc	Sat Feb  5 15:34:41 2005
@@ -129,7 +129,7 @@
 		return false;
     }
 
-    socklen_t sz = sizeof(struct sockaddr_in);
+    kde_socklen_t sz = sizeof(struct sockaddr_in);
 	int r = getsockname (theSocket,(struct sockaddr *)&socket_addr, &sz);
 	if(r == 0) {
 		thePort = ntohs(socket_addr.sin_port);
@@ -170,7 +170,7 @@
 	{
 		int clientfd;
 		struct sockaddr_in incoming;
-		socklen_t size_in = sizeof(struct sockaddr_in);
+		kde_socklen_t size_in = sizeof(struct sockaddr_in);
 
 		clientfd = accept(theSocket, (struct sockaddr*) &incoming, &size_in );
 		if(clientfd > 0)
diff -uNr arts-1.3.92/mcop/unixserver.cc arts-1.3.92-new/mcop/unixserver.cc
--- arts-1.3.92/mcop/unixserver.cc	Fri Mar  8 15:30:20 2002
+++ arts-1.3.92-new/mcop/unixserver.cc	Sat Feb  5 15:34:41 2005
@@ -130,7 +130,7 @@
 	{
 		int clientfd;
 		struct sockaddr_un incoming;
-		socklen_t size_in = sizeof(struct sockaddr_un);
+		kde_socklen_t size_in = sizeof(struct sockaddr_un);
 
 		clientfd = accept(theSocket, (struct sockaddr*) &incoming, &size_in );
 		if(clientfd > 0)
diff -uNr arts-1.3.92/x11/Makefile.in arts-1.3.92-new/x11/Makefile.in
--- arts-1.3.92/x11/Makefile.in	Fri Feb  4 05:44:13 2005
+++ arts-1.3.92-new/x11/Makefile.in	Sat Feb  5 15:34:41 2005
@@ -371,7 +371,7 @@
 @include_x11_TRUE@lib_LTLIBRARIES = libx11globalcomm.la
 @include_x11_TRUE@libx11globalcomm_la_SOURCES = x11globalcomm.cc x11globalcomm_impl.cc
 @include_x11_TRUE@libx11globalcomm_la_LIBADD = $(top_builddir)/mcop/libmcop.la $(LIB_X11)
-@include_x11_TRUE@libx11globalcomm_la_LDFLAGS = -no-undefined -module -version-info 1:0 $(X_LDFLAGS) $(all_libraries)
+@include_x11_TRUE@libx11globalcomm_la_LDFLAGS = -no-undefined -module -version-info 1:0 $(all_libraries) $(X_LDFLAGS)
 @include_x11_TRUE@DISTCLEANFILES = x11globalcomm.cc x11globalcomm.h \
 @include_x11_TRUE@                 x11globalcomm.mcoptype x11globalcomm.mcopclass
 
