diff -uNr kdeedu-3.3.91/admin/ltmain.sh kdeedu-3.3.91-new/admin/ltmain.sh
--- kdeedu-3.3.91/admin/ltmain.sh	Mon Jan  3 09:52:29 2005
+++ kdeedu-3.3.91-new/admin/ltmain.sh	Tue Jan 18 16:26:48 2005
@@ -4453,6 +4453,7 @@
 	# We should set the shlibpath_var
 	rpath=
 	for dir in $temp_rpath; do
+     if test "$dir" != "@FINKPREFIX@/lib"; then
 	  case $dir in
 	  [\\/]* | [A-Za-z]:[\\/]*)
 	    # Absolute path.
@@ -4463,6 +4464,7 @@
 	    rpath="$rpath\$thisdir/$dir:"
 	    ;;
 	  esac
+     fi
 	done
 	temp_rpath="$rpath"
       fi
diff -uNr kdeedu-3.3.91/configure kdeedu-3.3.91-new/configure
--- kdeedu-3.3.91/configure	Fri Jan  7 06:40:40 2005
+++ kdeedu-3.3.91-new/configure	Tue Jan 18 16:26:49 2005
@@ -8232,6 +8232,19 @@
     lt_cv_sys_max_cmd_len=8192;
     ;;
 
+  netbsd* | freebsd* | openbsd* | darwin* )
+    # This has been around since 386BSD, at least.  Likely further.
+    if test -x /sbin/sysctl; then
+      lt_cv_sys_max_cmd_len=`/sbin/sysctl -n kern.argmax`
+    elif test -x /usr/sbin/sysctl; then
+      lt_cv_sys_max_cmd_len=`/usr/sbin/sysctl -n kern.argmax`
+    else
+      lt_cv_sys_max_cmd_len=65536 # usable default for *BSD
+    fi
+    # And add a safety zone
+    lt_cv_sys_max_cmd_len=`expr $lt_cv_sys_max_cmd_len \/ 4`
+    ;;
+
  *)
     # If test is not a shell built-in, we'll probably end up computing a
     # maximum length that is only half of the actual maximum length, but
@@ -9875,10 +9888,21 @@
       archive_cmds_need_lc=no
       case "$host_os" in
       rhapsody* | darwin1.[012])
-	allow_undefined_flag='-undefined suppress'
+	allow_undefined_flag='-Wl,-undefined -Wl,suppress'
 	;;
       *) # Darwin 1.3 on
-	test -z ${LD_TWOLEVEL_NAMESPACE} && allow_undefined_flag='-flat_namespace -undefined suppress'
+	if test -z ${MACOSX_DEPLOYMENT_TARGET} ; then
+	  allow_undefined_flag='-Wl,-flat_namespace -Wl,-undefined -Wl,suppress'
+	else
+	  case ${MACOSX_DEPLOYMENT_TARGET} in
+	    10.[012])
+	      allow_undefined_flag='-Wl,-flat_namespace -Wl,-undefined -Wl,suppress'
+	      ;;
+	    10.*)
+	      allow_undefined_flag='-Wl,-undefined -Wl,dynamic_lookup'
+	      ;;
+	  esac
+	fi
 	;;
       esac
     	lt_int_apple_cc_single_mod=no
@@ -11728,7 +11752,18 @@
       allow_undefined_flag='-undefined suppress'
       ;;
     *) # Darwin 1.3 on
-      test -z ${LD_TWOLEVEL_NAMESPACE} && allow_undefined_flag='-flat_namespace -undefined suppress'
+	if test -z ${MACOSX_DEPLOYMENT_TARGET} ; then
+	  allow_undefined_flag='-Wl,-flat_namespace -Wl,-undefined -Wl,suppress'
+	else
+	  case ${MACOSX_DEPLOYMENT_TARGET} in
+	    10.[012])
+	      allow_undefined_flag='-Wl,-flat_namespace -Wl,-undefined -Wl,suppress'
+	      ;;
+	    10.*)
+	      allow_undefined_flag='-Wl,-undefined -Wl,dynamic_lookup'
+	      ;;
+	  esac
+	fi
       ;;
     esac
     output_verbose_link_cmd='echo'
@@ -12779,7 +12814,18 @@
       allow_undefined_flag_CXX='-undefined suppress'
       ;;
     *) # Darwin 1.3 on
-      test -z ${LD_TWOLEVEL_NAMESPACE} && allow_undefined_flag_CXX='-flat_namespace -undefined suppress'
+	if test -z ${MACOSX_DEPLOYMENT_TARGET} ; then
+	  allow_undefined_flag_CXX='-Wl,-flat_namespace -Wl,-undefined -Wl,suppress'
+	else
+	  case ${MACOSX_DEPLOYMENT_TARGET} in
+	    10.[012])
+	      allow_undefined_flag_CXX='-Wl,-flat_namespace -Wl,-undefined -Wl,suppress'
+	      ;;
+	    10.*)
+	      allow_undefined_flag_CXX='-Wl,-undefined -Wl,dynamic_lookup'
+	      ;;
+	  esac
+	fi
       ;;
     esac
     lt_int_apple_cc_single_mod=no
@@ -16602,7 +16648,18 @@
 	allow_undefined_flag_F77='-undefined suppress'
 	;;
       *) # Darwin 1.3 on
-	test -z ${LD_TWOLEVEL_NAMESPACE} && allow_undefined_flag_F77='-flat_namespace -undefined suppress'
+	if test -z ${MACOSX_DEPLOYMENT_TARGET} ; then
+	  allow_undefined_flag_F77='-Wl,-flat_namespace -Wl,-undefined -Wl,suppress'
+	else
+	  case ${MACOSX_DEPLOYMENT_TARGET} in
+	    10.[012])
+	      allow_undefined_flag_F77='-Wl,-flat_namespace -Wl,-undefined -Wl,suppress'
+	      ;;
+	    10.*)
+	      allow_undefined_flag_F77='-Wl,-undefined -Wl,dynamic_lookup'
+	      ;;
+	  esac
+	fi
 	;;
       esac
     	lt_int_apple_cc_single_mod=no
@@ -18884,7 +18941,18 @@
 	allow_undefined_flag_GCJ='-undefined suppress'
 	;;
       *) # Darwin 1.3 on
-	test -z ${LD_TWOLEVEL_NAMESPACE} && allow_undefined_flag_GCJ='-flat_namespace -undefined suppress'
+	if test -z ${MACOSX_DEPLOYMENT_TARGET} ; then
+	  allow_undefined_flag_GCJ='-Wl,-flat_namespace -Wl,-undefined -Wl,suppress'
+	else
+	  case ${MACOSX_DEPLOYMENT_TARGET} in
+	    10.[012])
+	      allow_undefined_flag_GCJ='-Wl,-flat_namespace -Wl,-undefined -Wl,suppress'
+	      ;;
+	    10.*)
+	      allow_undefined_flag_GCJ='-Wl,-undefined -Wl,dynamic_lookup'
+	      ;;
+	  esac
+	fi
 	;;
       esac
     	lt_int_apple_cc_single_mod=no
diff -uNr kdeedu-3.3.91/kstars/kstars/Makefile.in kdeedu-3.3.91-new/kstars/kstars/Makefile.in
--- kdeedu-3.3.91/kstars/kstars/Makefile.in	Fri Jan  7 06:40:36 2005
+++ kdeedu-3.3.91-new/kstars/kstars/Makefile.in	Tue Jan 18 16:26:49 2005
@@ -132,8 +132,7 @@
 	fitshistogram.$(OBJEXT) opssolarsystem.$(OBJEXT) \
 	streamwg.$(OBJEXT) indifitsconf.$(OBJEXT) \
 	imagesequence.$(OBJEXT) kstarsinterface_skel.$(OBJEXT) simclockinterface_skel.$(OBJEXT) addcatdialogui.$(OBJEXT) addlinkdialogui.$(OBJEXT) detaildialogui.$(OBJEXT) devmanager.$(OBJEXT) focusdialogdlg.$(OBJEXT) indiconf.$(OBJEXT) indihostconf.$(OBJEXT) opsadvancedui.$(OBJEXT) opscatalogui.$(OBJEXT) opscolorsui.$(OBJEXT) opsguides.$(OBJEXT) opssolarsystemui.$(OBJEXT) telescopewizard.$(OBJEXT) fovdialogui.$(OBJEXT) newfovui.$(OBJEXT) kswizardui.$(OBJEXT) contrastbrightnessgui.$(OBJEXT) imagereductionui.$(OBJEXT) histdialog.$(OBJEXT) statform.$(OBJEXT) fitsheaderdialog.$(OBJEXT) streamformui.$(OBJEXT) imgsequencedlgui.$(OBJEXT) Options.$(OBJEXT)
-@KDE_USE_FINAL_FALSE@kstars_OBJECTS = $(kstars_nofinal_OBJECTS)
-@KDE_USE_FINAL_TRUE@kstars_OBJECTS = $(kstars_final_OBJECTS)
+kstars_OBJECTS = $(kstars_nofinal_OBJECTS)
 kstars_DEPENDENCIES = indi/liblilxml.a indi/libindicom.a \
 	tools/libkstarstools.a \
 	$(top_builddir)/libkdeedu/kdeeduplot/libkdeeduplot.la \
@@ -254,8 +253,6 @@
 KDE_RPATH = @KDE_RPATH@
 KDE_USE_CLOSURE_FALSE = @KDE_USE_CLOSURE_FALSE@
 KDE_USE_CLOSURE_TRUE = @KDE_USE_CLOSURE_TRUE@
-KDE_USE_FINAL_FALSE = @KDE_USE_FINAL_FALSE@
-KDE_USE_FINAL_TRUE = @KDE_USE_FINAL_TRUE@
 KDE_USE_NMCHECK_FALSE = @KDE_USE_NMCHECK_FALSE@
 KDE_USE_NMCHECK_TRUE = @KDE_USE_NMCHECK_TRUE@
 KDE_XSL_STYLESHEET = @KDE_XSL_STYLESHEET@
@@ -1594,11 +1591,11 @@
 
 #>+ 3
 final:
-	$(MAKE) kstars_OBJECTS="$(kstars_final_OBJECTS)" all-am
+	$(MAKE) kstars_OBJECTS="$(kstars_nofinal_OBJECTS)" all-am
 
 #>+ 3
 final-install:
-	$(MAKE) kstars_OBJECTS="$(kstars_final_OBJECTS)" install-am
+	$(MAKE) kstars_OBJECTS="$(kstars_nofinal_OBJECTS)" install-am
 
 #>+ 3
 no-final:
diff -uNr kdeedu-3.3.91/kstars/kstars/indi/fli_ccd.c kdeedu-3.3.91-new/kstars/kstars/indi/fli_ccd.c
--- kdeedu-3.3.91/kstars/kstars/indi/fli_ccd.c	Mon Jan  3 09:52:58 2005
+++ kdeedu-3.3.91-new/kstars/kstars/indi/fli_ccd.c	Tue Jan 18 16:26:49 2005
@@ -19,6 +19,7 @@
 
 #endif
 
+#include <config.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
@@ -1696,7 +1697,7 @@
  
 	/* get a private connection to new client */
 	cli_len = sizeof(cli_socket);
-	s = accept (lsocket, (struct sockaddr *)&cli_socket, (socklen_t *)&cli_len);
+	s = accept (lsocket, (struct sockaddr *)&cli_socket, (kde_socklen_t *)&cli_len);
 	if(s < 0) {
 	    fprintf (stderr, "%s: accept: %s", me, strerror(errno));
 	    exit (1);
