diff -uNr kdemultimedia-3.3.91/kioslave/audiocd/audiocd.cpp kdemultimedia-3.3.91-new/kioslave/audiocd/audiocd.cpp
--- kdemultimedia-3.3.91/kioslave/audiocd/audiocd.cpp	Mon Jan  3 09:53:36 2005
+++ kdemultimedia-3.3.91-new/kioslave/audiocd/audiocd.cpp	Tue Jan 25 09:32:20 2005
@@ -24,6 +24,8 @@
 
 #include <config.h>
 
+#undef VERSION
+
 extern "C"
 {
 #include <cdda_interface.h>
diff -uNr kdemultimedia-3.3.91/noatun/library/noatunarts/Makefile.in kdemultimedia-3.3.91-new/noatun/library/noatunarts/Makefile.in
--- kdemultimedia-3.3.91/noatun/library/noatunarts/Makefile.in	Fri Jan  7 06:15:42 2005
+++ kdemultimedia-3.3.91-new/noatun/library/noatunarts/Makefile.in	Mon Jan 24 20:09:32 2005
@@ -515,7 +515,7 @@
 
 #>- libnoatunarts_la_LDFLAGS = $(all_libraries) -module -avoid-version -no-undefined
 #>+ 1
-libnoatunarts_la_LDFLAGS = $(all_libraries) -module -avoid-version -no-undefined $(KDE_NO_UNDEFINED)
+libnoatunarts_la_LDFLAGS = $(all_libraries) -no-undefined $(KDE_NO_UNDEFINED)
 libnoatunarts_la_LIBADD = -lkmedia2_idl -lsoundserver_idl -lartsflow
 #>- libnoatunarts_la_METASOURCES = AUTO
 mcoptypedir = $(libdir)/mcop
diff -uNr kdemultimedia-3.3.91/noatun/modules/winskin/vis/Makefile.in kdemultimedia-3.3.91-new/noatun/modules/winskin/vis/Makefile.in
--- kdemultimedia-3.3.91/noatun/modules/winskin/vis/Makefile.in	Fri Jan  7 06:15:43 2005
+++ kdemultimedia-3.3.91-new/noatun/modules/winskin/vis/Makefile.in	Mon Jan 24 22:55:26 2005
@@ -507,7 +507,7 @@
 #>- libwinskinvis_la_LDFLAGS = $(all_libraries) -module -avoid-version \
 #>- 				-no-undefined
 #>+ 2
-libwinskinvis_la_LDFLAGS = $(all_libraries) -module -avoid-version \
+libwinskinvis_la_LDFLAGS = $(all_libraries) \
 				-no-undefined $(KDE_NO_UNDEFINED)
 
 libwinskinvis_la_LIBADD = -lkmedia2_idl -lsoundserver_idl -lartsflow
diff -uNr kdemultimedia-3.3.91/oggvorbis_artsplugin/oggPlayObject_impl.h kdemultimedia-3.3.91-new/oggvorbis_artsplugin/oggPlayObject_impl.h
--- kdemultimedia-3.3.91/oggvorbis_artsplugin/oggPlayObject_impl.h	Sat Oct 26 21:10:28 2002
+++ kdemultimedia-3.3.91-new/oggvorbis_artsplugin/oggPlayObject_impl.h	Mon Jan 24 18:16:23 2005
@@ -1,7 +1,7 @@
 #ifndef OGGPLAYER_IMPL_H
 #define OGGPLAYER_IMPL_H "$Id$"
 
-#if defined(__GNU_LIBRARY__) && !defined(_SEM_SEMUN_UNDEFINED)
+#if (defined(__GNU_LIBRARY__) && !defined(_SEM_SEMUN_UNDEFINED)) || defined(__APPLE__)
 /* union semun is defined by including <sys/sem.h> */
 #else
 /* according to X/OPEN we have to define it ourselves */
