diff -Naur deborphan-old/configure deborphan-new/configure
--- deborphan-old/configure	2003-02-24 10:44:14.000000000 -0500
+++ deborphan-new/configure	2003-07-24 21:34:34.000000000 -0400
@@ -902,8 +902,8 @@
 
 
 
-cat >> confdefs.h <<\EOF
-#define STATUS_FILE "/var/lib/dpkg/status"
+cat >> confdefs.h <<EOF
+#define STATUS_FILE "${prefix}/var/lib/dpkg/status"
 EOF
 
 cat >> confdefs.h <<\EOF
@@ -4395,8 +4395,8 @@
 
 echo $ac_n "checking debfoster's keepers file""... $ac_c" 1>&6
 echo "configure:4398: checking debfoster's keepers file" >&5
-if [ -r /var/state/debfoster/keepers ]; then
-  dffile="/var/state/debfoster/keepers"
+if [ -r ${prefix}/var/lib/debfoster/keepers ]; then
+  dffile="${prefix}/var/lib/debfoster/keepers"
   echo "$ac_t""$dffile" 1>&6
 else 
   if [ -r /etc/apt/keepers ]; then
diff -Naur deborphan-old/dpkg-rebuild deborphan-new/dpkg-rebuild
--- deborphan-old/dpkg-rebuild	1969-12-31 19:00:00.000000000 -0500
+++ deborphan-new/dpkg-rebuild	2003-07-24 22:41:32.000000000 -0400
@@ -0,0 +1,93 @@
+#!/usr/bin/perl
+
+# Rebuild the Debian '/var/lib/dpkg/status' file from information in
+# '/var/lib/dpkg/available' and '/var/lib/dpkg/info/*.list'.  This is
+# useful if your 'status' file got corrupted if the system crashed during
+# package maintenance, for example.
+#
+# Copyright 2002 by Patrick Reynolds <reynolds@cs.duke.edu>
+# Distributed under the terms of the GNU General Public License (GPL).
+#
+# Usage:
+#   dpkg-rebuild
+# It takes no arguments and generates output in /tmp/status.
+# Move /tmp/status to /var/lib/dpkg if it looks acceptable.
+#
+# Limitations:
+#   1) Packages that are no longer available will not show up in the
+#   rebuilt 'status' file.  This means installed-but-obsolete packages
+#   can't be managed after a rebuild.
+#
+#   2) The 'Conffiles:' keys in the 'status' file are not rebuilt.
+#   Configuration files may not be completely removed when you purge
+#   packages, and package upgrades may clobber existing configuration
+#   files without asking.
+#
+#   3) The 'Essential:' keys in the 'status' file now appear after, not
+#   before, the 'Status:' keys.  I believe this is harmless.
+#
+#   4) Packages in the 'deinstall' state will appear to be in the 'purge'
+#   state.  Their configuration files will remain, but dpkg won't know
+#   about them.
+#
+#   5) Packages in transitional or error states will be misreported.
+
+($prefix = `which fink`) =~ s,/bin/fink\s$,,;
+$available = "$prefix/var/lib/dpkg/available";
+$status = "/tmp/new-status";
+$info_dir = "$prefix/var/lib/dpkg/info";
+
+foreach (<$info_dir/*.list>) {
+	s#.*/([^/]+)\.list$#$1#;
+	$installed{$_} = 1;
+}
+
+$state = 0;   # 0=between, 1=copying-installed, 2=copying-not-installed
+open(AVAILABLE, "<$available") || die "no $available";
+open(STATUS, ">$status") || die "no $status";
+while (<AVAILABLE>) {
+	chomp;
+	if ($state == 0) {
+		if (/^Package: (\S+)$/) {
+			print STATUS "$_\n";
+			if ($installed{$1}) {
+				$state = 1;
+				print STATUS "Status: install ok installed\n";
+				delete $installed{$1};
+			}
+			else {
+				$state = 2;
+				print STATUS "Status: purge ok not-installed\n";
+			}
+		}
+		else {
+			die "Expected 'Package:' at $.";
+		}
+	}
+	elsif ($state == 1) {
+		if ($_ eq "") {
+			print STATUS "\n";
+			$state = 0;
+		}
+		elsif (!/^Architecture: / && !/^Filename: / && !/^Size: / && !/^MD5sum: /) {
+			print STATUS "$_\n";
+		}
+	}
+	elsif ($state == 2) {
+		if ($_ eq "") {
+			print STATUS "\n";
+			$state = 0;
+		}
+		elsif (/^Priority: / || /^Section: /) {
+			print STATUS "$_\n";
+		}
+	}
+	else {
+		die "Invalid state $state";
+	}
+}
+
+printf "Installed packages not found in $available:\n";
+foreach (sort keys %installed) {
+	print "  $_\n";
+}
diff -Naur deborphan-old/include/deborphan.h deborphan-new/include/deborphan.h
--- deborphan-old/include/deborphan.h	2003-02-24 11:06:04.000000000 -0500
+++ deborphan-new/include/deborphan.h	2003-07-24 21:33:09.000000000 -0400
@@ -29,9 +29,6 @@
 
 #ifndef _POSIX_SOURCE
 #  include <stdio.h>
-#  ifndef off_t
-     typedef int off_t;
-#  endif
 int ftruncate(int, off_t);
 extern int fileno __P ((FILE *__stream));
 #endif
diff -Naur deborphan-old/src/Makefile.in deborphan-new/src/Makefile.in
--- deborphan-old/src/Makefile.in	2003-02-24 10:44:48.000000000 -0500
+++ deborphan-new/src/Makefile.in	2003-07-24 21:33:09.000000000 -0400
@@ -91,7 +91,7 @@
 deborphan_SOURCES = deborphan.c exit.c libdeps.c pkginfo.c xalloc.c string.c keep.c file.c set.c
 
 KEEPNAME = keep
-STATEDIR = /var/lib/deborphan
+STATEDIR = ${prefix}/var/lib/deborphan
 KEEPFILE = $(STATEDIR)/$(KEEPNAME)
 
 localedir = $(datadir)/locale
@@ -302,8 +302,8 @@
 
 
 install-data-hook:
-	test -e $(STATEDIR) || mkdir $(STATEDIR)
-	test -e $(KEEPFILE) || $(INSTALL_PROGRAM) ../keep $(KEEPFILE)
+	mkdir -p $(DESTDIR)$(STATEDIR)
+	$(INSTALL_PROGRAM) ../keep $(DESTDIR)$(KEEPFILE)
 
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
diff -Naur deborphan-old/src/pkginfo.c deborphan-new/src/pkginfo.c
--- deborphan-old/src/pkginfo.c	2003-02-24 11:25:07.000000000 -0500
+++ deborphan-new/src/pkginfo.c	2003-07-24 21:33:09.000000000 -0400
@@ -9,6 +9,7 @@
 
 #include <stdlib.h>
 #include <string.h>
+#include <sys/types.h>
 #include <regex.h>
 
 #ifdef HAVE_CONFIG_H
diff -Naur deborphan-old/util/Makefile.in deborphan-new/util/Makefile.in
--- deborphan-old/util/Makefile.in	2003-02-24 10:45:13.000000000 -0500
+++ deborphan-new/util/Makefile.in	2003-07-24 21:33:09.000000000 -0400
@@ -213,7 +213,7 @@
 
 
 install-exec-hook:
-	-ln -s $(sbindir)/orphaner $(sbindir)/editkeep
+	-ln -s $(sbindir)/orphaner $(DESTDIR)$(sbindir)/editkeep
 
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
diff -Naur deborphan-old/util/orphaner deborphan-new/util/orphaner
--- deborphan-old/util/orphaner	2003-02-24 10:36:43.000000000 -0500
+++ deborphan-new/util/orphaner	2003-07-24 21:32:53.000000000 -0400
@@ -25,8 +25,8 @@
 set -e
 
 OPTIONS=$@
-VALIDOPTIONS='^-([aDHns]|-guess-(.+)|-nice-mode|-all-packages|-priority(.+)|p(.+)|-show-section|-force-hold)[[:space:]]$'
-VALIDKEEPOPTIONS='^-([aDHns]|-guess-(.+)|-nice-mode|-all-packages|-priority(.+)|p(.+)|-show-section|-force-hold)[[:space:]]$'
+VALIDOPTIONS='^-([aDHns]|-guess-(.+)|-find-(.+)|-nice-mode|-all-packages|-priority(.+)|p(.+)|-show-section|-force-hold)[[:space:]]$'
+VALIDKEEPOPTIONS='^-([aDHns]|-guess-(.+)|-find-(.+)|-nice-mode|-all-packages|-priority(.+)|p(.+)|-show-section|-force-hold)[[:space:]]$'
 
 if [ -x /usr/bin/dialog ]; then
   DIALOG=/usr/bin/dialog
