diff -ruN mozilla-1.9.2-orig/.mozconfig mozilla-1.9.2/.mozconfig
--- mozilla-1.9.2-orig/.mozconfig	1969-12-31 19:00:00.000000000 -0500
+++ mozilla-1.9.2/.mozconfig	2009-04-27 16:56:30.000000000 -0400
@@ -0,0 +1,56 @@
+# sh
+# Build configuration script
+#
+# See https://developer.mozilla.org/en/Configuring_Build_Options for build options.
+#
+
+. $topsrcdir/browser/config/mozconfig
+
+# Options for client.mk.
+# environment variables to set
+
+mk_add_options MOZ_OBJDIR=@TOPSRCDIR@/obj-@FINK_ARCH_TARGET@-apple-darwin`uname -r`
+mk_add_options AUTOCONF=autoconf
+CC="gcc-4.0"
+CXX="g++-4.0"
+ac_add_options --target=@FINK_ARCH_TARGET@-apple-darwin`uname -r`
+ac_add_options --with-macos-sdk=/Developer/SDKs/MacOSX10.4u.sdk
+HOST_CC="gcc-4.0"
+HOST_CXX="g++-4.0"
+RANLIB=/usr/bin/ranlib
+AR=/usr/bin/ar
+AS=$CC
+LD=/usr/bin/ld
+#STRIP="strip -x -S"
+CROSS_COMPILE=1
+#PKG_CONFIG_PATH="@FINKPREFIX@/lib/pango-ft219/lib/pkgconfig:@FINKPREFIX@/lib/fontconfig2/lib/pkgconfig:@FINKPREFIX@/lib/freetype219/lib/pkgconfig:$PKG_CONFIG_PATH"	
+
+# Options for 'configure' (same as command-line options).
+ac_add_options --prefix=@FINKPREFIX@
+ac_add_options --enable-official-branding
+ac_add_options --enable-default-toolkit=cairo-gtk2
+ac_add_options --x-libraries=/usr/X11R6/lib
+ac_add_options --x-includes=/usr/X11R6/include
+#ac_add_options --enable-xft
+#ac_add_options --disable-xft
+#ac_add_options --enable-freetype2
+#ac_add_options --disable-freetype2
+#ac_add_options --enable-xinerama
+#ac_add_options --with-pthreads
+ac_add_options --disable-gnomevfs
+ac_add_options --disable-gnomeui
+ac_add_options --enable-dbus
+ac_add_options --enable-extensions=default,-gnomevfs
+
+ac_add_options --disable-dependency-tracking
+ac_add_options --disable-crashreporter
+ac_add_options --disable-necko-wifi
+
+ac_add_options --disable-debug
+ac_add_options --disable-tests
+ac_add_options --enable-strip
+ac_add_options --enable-macos-target=10.4
+#ac_add_options --with-default-mozilla-five-home=@FINKPREFIX@/lib/firefox3
+ac_add_options --with-system-jpeg=@FINKPREFIX@
+ac_add_options --disable-system-cairo
+ac_add_options --disable-system-sqlite
diff -ruN mozilla-1.9.2-orig/browser/installer/package-manifest.in mozilla-1.9.2/browser/installer/package-manifest.in
--- mozilla-1.9.2-orig/browser/installer/package-manifest.in
+++ mozilla-1.9.2/browser/installer/package-manifest.in
@@ -77,7 +77,8 @@
 @BINPATH@/plugins/MRJPlugin.plugin/
 #endif
 #elifdef XP_UNIX
-@BINPATH@/plugins/libnullplugin.so
+@BINPATH@/plugins/libnullplugin.dylib
+@BINPATH@/plugins/libunixprintplugin.dylib
 #elifdef XP_WIN32
 #ifndef WINCE
 @BINPATH@/plugins/npnul32.dll
@@ -189,7 +189,7 @@
 @BINPATH@/components/layout_xul.xpt
 #ifdef XP_UNIX
 #ifndef XP_MACOSX
-@BINPATH@/components/libimgicon.so
+@BINPATH@/components/libimgicon.dylib
 #endif
 #endif
 @BINPATH@/components/locale.xpt
diff -ruN mozilla-1.9.2-orig/browser/locales/generic/profile/bookmarks.html.in mozilla-1.9.2/browser/locales/generic/profile/bookmarks.html.in
--- mozilla-1.9.2-orig/browser/locales/generic/profile/bookmarks.html.in	2009-04-23 21:46:37.000000000 -0400
+++ mozilla-1.9.2/browser/locales/generic/profile/bookmarks.html.in	2009-04-27 16:56:30.000000000 -0400
@@ -15,6 +15,7 @@
     <DL><p>
         <DT><A HREF="http://www.mozilla.com/@AB_CD@/firefox/central/" ID="rdf:#$GvPhC3">@getting_started@</A>
         <DT><A HREF="http://fxfeeds.mozilla.com/@AB_CD@/firefox/livebookmarks/" FEEDURL="http://fxfeeds.mozilla.com/@AB_CD@/firefox/headlines.xml" ID="rdf:#$HvPhC3">@latest_headlines@</A>
+        <DT><A HREF="http://www.finkproject.org/" ICON_URI="http://www.finkproject.org/favicon.ico" ICON="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAA+0lEQVQ4jWNgGDLg/6sTqv9OVe75uzv43J+TpUv/MzAwEqfxfEfU3y3O///u9P//d5v7/7/HCir/vz8n//9ir9+fffF+eDX/O5yxCqLR4///28u0GRgYGP6uN/sPN/zNBWlMG9+e4Pt7tq0bbutO//9/t3v9///qFQ9e2/7uCZ3/b3fwvX9HMg//PZpz7e9O//9/9ye8+Lc34ubf7V7/f1+e4Irdj3dWVf//f5UNzn91VeLvzoD/f0+U1OK1EadLtrr8/3+uoYcszf+PZHX/OxB/jizNDAwMDP92BX0hW/P/o1k7ydbMwMDA8P98D/ZQJkrz+6sJFNk+oAAAPwyICxl3w8YAAAAASUVORK5CYII=">Fink - Home</A>
     </DL><p>
     <HR>
     <DT><H3 ID="rdf:#$ZvPhC3">@firefox_heading@</H3>
diff -ruN mozilla-1.9.2-orig/config/autoconf.mk.in mozilla-1.9.2/config/autoconf.mk.in
--- mozilla-1.9.2-orig/config/autoconf.mk.in	2009-04-23 21:46:37.000000000 -0400
+++ mozilla-1.9.2/config/autoconf.mk.in	2009-04-27 16:56:30.000000000 -0400
@@ -58,16 +58,17 @@
 prefix		= @prefix@
 exec_prefix	= @exec_prefix@
 bindir		= @bindir@
-includedir	= @includedir@/$(MOZ_APP_NAME)-$(MOZ_APP_VERSION)
+includedir	= @includedir@/$(MOZ_APP_NAME)3
 libdir		= @libdir@
 datadir		= @datadir@
 mandir		= @mandir@
-idldir		= $(datadir)/idl/$(MOZ_APP_NAME)-$(MOZ_APP_VERSION)
+idldir		= $(datadir)/idl/$(MOZ_APP_NAME)3
 
-installdir	= $(libdir)/$(MOZ_APP_NAME)-$(MOZ_APP_VERSION)
-sdkdir		= $(libdir)/$(MOZ_APP_NAME)-devel-$(MOZ_APP_VERSION)
+installdir	= $(libdir)/$(MOZ_APP_NAME)3
+sdkdir		= $(libdir)/$(MOZ_APP_NAME)3-devel
 
 DIST		= $(DEPTH)/dist
+DISTLIB		= $(DIST)/lib
 LIBXUL_SDK      = @LIBXUL_SDK@
 
 MOZ_FS_LAYOUT = @MOZ_FS_LAYOUT@
@@ -195,8 +196,16 @@
 MOZ_FIX_LINK_PATHS=@MOZ_FIX_LINK_PATHS@
 
 XPCOM_FROZEN_LDOPTS=@XPCOM_FROZEN_LDOPTS@
+XPCOM_DYLIB_FILES = \
+	$(NSS_DYLIB_FILES) \
+	$(SQLITE_DYLIB_FILES) \
+	$(call EXPAND_DYLIB_NAME,xul) \
+	$(call EXPAND_DYLIB_NAME,mozjs)
+XPCOM_FROZEN_LDOPTS += $(XPCOM_DYLIB_FILES)
 XPCOM_LIBS=@XPCOM_LIBS@
+XPCOM_LIBS += $(XPCOM_DYLIB_FILES)
 LIBXUL_LIBS=@LIBXUL_LIBS@
+LIBXUL_LIBS += $(XPCOM_DYLIB_FILES)
 MOZ_TIMELINE=@MOZ_TIMELINE@
 
 ENABLE_STRIP	= @ENABLE_STRIP@
@@ -464,14 +473,28 @@
 MOZ_NATIVE_SQLITE = @MOZ_NATIVE_SQLITE@
 SQLITE_CFLAGS     = @SQLITE_CFLAGS@
 SQLITE_LIBS       = @SQLITE_LIBS@
+SQLITE_DYLIB_FILES = \
+	$(call EXPAND_DYLIB_NAME,sqlite3)
+# SQLITE_LIBS += $(SQLITE_DYLIB_FILES)
 
 NSPR_CONFIG	= @NSPR_CONFIG@
 NSPR_CFLAGS	= @NSPR_CFLAGS@
 NSPR_LIBS	= @NSPR_LIBS@
+NSPR_DYLIB_FILES = \
+	$(call EXPAND_DYLIB_NAME,plds4) \
+	$(call EXPAND_DYLIB_NAME,plc4) \
+	$(call EXPAND_DYLIB_NAME,nspr4)
+NSPR_LIBS += $(NSPR_DYLIB_FILES)
 
 NSS_CONFIG	= @NSS_CONFIG@
 NSS_CFLAGS	= @NSS_CFLAGS@
 NSS_LIBS	= @NSS_LIBS@
+NSS_DYLIB_FILES = \
+	$(call EXPAND_DYLIB_NAME,smime3) \
+	$(call EXPAND_DYLIB_NAME,ssl3) \
+	$(call EXPAND_DYLIB_NAME,nss3) \
+	$(call EXPAND_DYLIB_NAME,nssutil3)
+NSS_LIBS += $(NSS_DYLIB_FILES)
 NSS_DEP_LIBS	= @NSS_DEP_LIBS@
 NSS_DISABLE_DBM = @NSS_DISABLE_DBM@
 
@@ -521,7 +544,10 @@
 MOZ_ENABLE_XREMOTE	= @MOZ_ENABLE_XREMOTE@
 
 MOZ_GTK2_CFLAGS		= @MOZ_GTK2_CFLAGS@
-MOZ_GTK2_LIBS		= @MOZ_GTK2_LIBS@
+MOZ_GTK2_LIBS_CAIRO		= @MOZ_GTK2_LIBS@
+# remove -lcairo to avoid symbol conflict with mozcairo
+MOZ_GTK2_LIBS_NOCAIRO		= $(filter-out -lcairo,$(MOZ_GTK2_LIBS_CAIRO))
+MOZ_GTK2_LIBS		= $(MOZ_GTK2_LIBS_NOCAIRO)
 
 MOZ_QT_CFLAGS		= @MOZ_QT_CFLAGS@
 MOZ_QT_LIBS		= @MOZ_QT_LIBS@
diff -ruN mozilla-1.9.2-orig/configure mozilla-1.9.2/configure
--- mozilla-1.9.2-orig/configure	2009-04-23 21:51:07.000000000 -0400
+++ mozilla-1.9.2/configure	2009-04-27 16:56:30.000000000 -0400
@@ -5956,8 +5956,8 @@
 if test "$GNU_CC"; then
     # FIXME: Let us build with strict aliasing. bug 414641.
     CFLAGS="$CFLAGS -fno-strict-aliasing"
-    MKSHLIB='$(CXX) $(CXXFLAGS) $(DSO_PIC_CFLAGS) $(DSO_LDOPTS) -Wl,-h,$@ -o $@'
-    MKCSHLIB='$(CC) $(CFLAGS) $(DSO_PIC_CFLAGS) $(DSO_LDOPTS) -Wl,-h,$@ -o $@'
+    MKSHLIB='$(CXX) $(DSO_PIC_CFLAGS) $(DSO_LDOPTS) -Wl,-h,$@ -o $@'
+    MKCSHLIB='$(CC) $(DSO_PIC_CFLAGS) $(DSO_LDOPTS) -Wl,-h,$@ -o $@'
     DSO_LDOPTS='-shared'
     if test "$GCC_USE_GNU_LD"; then
         # Don't allow undefined symbols in libraries
@@ -6381,7 +6381,7 @@
     ;;
 
 *-darwin*)
-    HOST_CFLAGS="$HOST_CFLAGS -DXP_UNIX -DXP_MACOSX -DNO_X11"
+    HOST_CFLAGS="$HOST_CFLAGS -DXP_UNIX"
     HOST_NSPR_MDCPUCFG='\"md/_darwin.cfg\"'
     HOST_OPTIMIZE_FLAGS="${HOST_OPTIMIZE_FLAGS=-O3}"
     MOZ_FIX_LINK_PATHS='-Wl,-executable_path,$(LIBXUL_DIST)/bin'
@@ -6676,14 +6676,14 @@
     ;;
 
 *-darwin*) 
-    MKSHLIB='$(CXX) $(CXXFLAGS) $(DSO_PIC_CFLAGS) $(DSO_LDOPTS) -o $@'
-    MKCSHLIB='$(CC) $(CFLAGS) $(DSO_PIC_CFLAGS) $(DSO_LDOPTS) -o $@'
+    MKSHLIB='$(CXX) $(DSO_PIC_CFLAGS) $(DSO_LDOPTS) -o $@'
+    MKCSHLIB='$(CC) $(DSO_PIC_CFLAGS) $(DSO_LDOPTS) -o $@'
     MOZ_OPTIMIZE_FLAGS="-O3"
     _PEDANTIC=
-    CFLAGS="$CFLAGS -fpascal-strings -fno-common"
-    CXXFLAGS="$CXXFLAGS -fpascal-strings -fno-common"
+    CFLAGS="$CFLAGS -fno-common"
+    CXXFLAGS="$CXXFLAGS -fno-common"
     DLL_SUFFIX=".dylib"
-    DSO_LDOPTS=''
+    DSO_LDOPTS='-framework CoreFoundation'
     STRIP="$STRIP -x -S"
     _PLATFORM_DEFAULT_TOOLKIT='cairo-cocoa'
     MOZ_ENABLE_POSTSCRIPT=
diff -ruN mozilla-1.9.2-orig/browser/installer/Makefile.in mozilla-1.9.2/browser/installer/Makefile.in
--- mozilla-1.9.2-orig/browser/installer/Makefile.in	2009-11-10 12:53:40.000000000 -0800
+++ mozilla-1.9.2/browser/installer/Makefile.in	2009-11-10 13:01:38.000000000 -0800
@@ -120,7 +120,7 @@ endif
 
 include $(topsrcdir)/toolkit/mozapps/installer/packager.mk
 
-ifeq (Darwin, $(OS_ARCH))
+ifeq (bundle, $(MOZ_FS_LAYOUT))
 DEFINES += \
   -DBINPATH=$(_BINPATH) \
   -DAPPNAME=$(_APPNAME) \

diff -ruN mozilla-1.9.2-orig/fink/applications/firefox.desktop mozilla-1.9.2/fink/applications/firefox.desktop
--- mozilla-1.9.2-orig/fink/applications/firefox.desktop	1969-12-31 19:00:00.000000000 -0500
+++ mozilla-1.9.2/fink/applications/firefox.desktop	2009-04-27 16:56:30.000000000 -0400
@@ -0,0 +1,10 @@
+[Desktop Entry]
+Encoding=UTF-8
+Name=Firefox Web Browser
+Comment=Browse the World Wide Web
+Exec=firefox
+Icon=firefox.png
+Terminal=false
+Type=Application
+Categories=Application;Network
+StartupNotify=true
diff -ruN mozilla-1.9.2-orig/fink/fink.js mozilla-1.9.2/fink/fink.js
--- mozilla-1.9.2-orig/fink/fink.js	1969-12-31 19:00:00.000000000 -0500
+++ mozilla-1.9.2/fink/fink.js	2009-04-27 16:56:30.000000000 -0400
@@ -0,0 +1 @@
+pref("general.useragent.extra.vendorComment", "Fink Community Edition");
diff -ruN mozilla-1.9.2-orig/netwerk/wifi/src/nsWifiMonitor.h mozilla-1.9.2/netwerk/wifi/src/nsWifiMonitor.h
--- mozilla-1.9.2-orig/netwerk/wifi/src/nsWifiMonitor.h	2009-10-16 11:14:14.000000000 -0400
+++ mozilla-1.9.2/netwerk/wifi/src/nsWifiMonitor.h	2009-10-28 09:30:59.000000000 -0400
@@ -88,7 +88,7 @@
 
   nsresult DoScan();
 
-#if defined(XP_MACOSX)
+#if defined(XP_UNIX)
   nsresult DoScanWithCoreWLAN();
   nsresult DoScanOld();
 #endif
diff -ruN mozilla-1.9.2-orig/xpcom/base/nsStackWalk.cpp mozilla-1.9.2/xpcom/base/nsStackWalk.cpp
--- mozilla-1.9.2-orig/xpcom/base/nsStackWalk.cpp	2009-04-23 21:47:11.000000000 -0400
+++ mozilla-1.9.2/xpcom/base/nsStackWalk.cpp	2009-04-27 16:56:31.000000000 -0400
@@ -1132,7 +1132,7 @@
 #define __USE_GNU
 #endif
 
-#if defined(HAVE_LIBDL) || defined(XP_MACOSX)
+#if defined(HAVE_LIBDL) || defined(XP_UNIX)
 #include <dlfcn.h>
 #endif
 
diff -ruN mozilla-1.9.2-orig/config/config.mk mozilla-1.9.2/config/config.mk
--- mozilla-1.9.2-orig/config/config.mk	2010-03-12 13:08:56.000000000 -0800
+++ mozilla-1.9.2/config/config.mk	2010-03-12 13:09:41.000000000 -0800
@@ -590,8 +590,9 @@
 OS_COMPILE_CMMFLAGS += -fobjc-exceptions
 endif
 
-COMPILE_CFLAGS	= $(VISIBILITY_FLAGS) $(DEFINES) $(INCLUDES) $(XCFLAGS) $(PROFILER_CFLAGS) $(DSO_CFLAGS) $(DSO_PIC_CFLAGS) $(CFLAGS) $(RTL_FLAGS) $(OS_COMPILE_CFLAGS)
-COMPILE_CXXFLAGS = $(VISIBILITY_FLAGS) $(DEFINES) $(INCLUDES) $(XCFLAGS) $(PROFILER_CFLAGS) $(DSO_CFLAGS) $(DSO_PIC_CFLAGS)  $(CXXFLAGS) $(RTL_FLAGS) $(OS_COMPILE_CXXFLAGS)
+# patch: reordered XCFLAGS later to find alternate fontconfig2 headers first
+COMPILE_CFLAGS	= $(VISIBILITY_FLAGS) $(DEFINES) $(INCLUDES) $(PROFILER_CFLAGS) $(DSO_CFLAGS) $(DSO_PIC_CFLAGS) $(CFLAGS) $(RTL_FLAGS) $(XCFLAGS) $(OS_COMPILE_CFLAGS)
+COMPILE_CXXFLAGS = $(VISIBILITY_FLAGS) $(DEFINES) $(INCLUDES) $(PROFILER_CFLAGS) $(DSO_CFLAGS) $(DSO_PIC_CFLAGS)  $(CXXFLAGS) $(RTL_FLAGS) $(XCFLAGS) $(OS_COMPILE_CXXFLAGS)
 COMPILE_CMFLAGS = $(OS_COMPILE_CMFLAGS)
 COMPILE_CMMFLAGS = $(OS_COMPILE_CMMFLAGS)
 
diff -ruN mozilla-1.9.2-orig/config/rules.mk mozilla-1.9.2/config/rules.mk
--- mozilla-1.9.2-orig/config/rules.mk.orig	2010-03-22 12:47:26.000000000 -0700
+++ mozilla-1.9.2/config/rules.mk	2010-03-22 12:47:35.000000000 -0700
@@ -121,6 +121,11 @@
 EXPAND_MOZLIBNAME = $(addprefix -l,$(1))
 endif
 
+# patch for darwin8:
+# EXPAND_DYLIB_NAME - $(call EXPAND_DYLIB_NAME,foo)
+# expands to -dylib_file $(installdir)/libfoo.dylib:$(DISTLIB)/libfoo.dylib
+EXPAND_DYLIB_NAME = -dylib_file $(installdir)/$(DLL_PREFIX)$(1)$(DLL_SUFFIX):$(DISTLIB)/$(DLL_PREFIX)$(1)$(DLL_SUFFIX)
+
 ifdef EXTRA_DSO_LIBS
 EXTRA_DSO_LIBS	:= $(call EXPAND_MOZLIBNAME,$(EXTRA_DSO_LIBS))
 endif
@@ -990,7 +995,7 @@
 endif	# MSVC with manifest tool
 else
 ifeq ($(CPP_PROG_LINK),1)
-	$(CCC) -o $@ $(CXXFLAGS) $(WRAP_MALLOC_CFLAGS) $(PROGOBJS) $(RESFILE) $(WIN32_EXE_LDFLAGS) $(SOLARIS_JEMALLOC_LDFLAGS) $(LDFLAGS) $(LIBS_DIR) $(LIBS) $(OS_LIBS) $(EXTRA_LIBS) $(BIN_FLAGS) $(WRAP_MALLOC_LIB) $(PROFILER_LIBS) $(EXE_DEF_FILE)
+	$(CCC) -o $@ $(WRAP_MALLOC_CFLAGS) $(PROGOBJS) $(RESFILE) $(WIN32_EXE_LDFLAGS) $(SOLARIS_JEMALLOC_LDFLAGS) $(LDFLAGS) $(LIBS_DIR) $(LIBS) $(OS_LIBS) $(EXTRA_LIBS) $(BIN_FLAGS) $(WRAP_MALLOC_LIB) $(PROFILER_LIBS) $(EXE_DEF_FILE)
 else # ! CPP_PROG_LINK
 	$(CC) -o $@ $(CFLAGS) $(PROGOBJS) $(RESFILE) $(WIN32_EXE_LDFLAGS) $(SOLARIS_JEMALLOC_LDFLAGS) $(LDFLAGS) $(LIBS_DIR) $(LIBS) $(OS_LIBS) $(EXTRA_LIBS) $(BIN_FLAGS) $(EXE_DEF_FILE)
 endif # CPP_PROG_LINK
diff -ruN mozilla-1.9.2-orig/toolkit/library/Makefile.in mozilla-1.9.2/toolkit/library/Makefile.in
--- mozilla-1.9.2-orig/toolkit/library/Makefile.in	2010-03-30 09:58:07.000000000 -0700
+++ mozilla-1.9.2/toolkit/library/Makefile.in	2010-04-07 00:51:47.000000000 -0700
@@ -181,7 +181,9 @@
 export:: $(RDF_UTIL_SRC_CPPSRCS) $(INTL_UNICHARUTIL_UTIL_CPPSRCS)
 	$(INSTALL) $^ .
 
+# patch: because -lcairo (shlib) was filtered out of link command
+EXTRA_DSO_LDOPTS += -Wl,-undefined,dynamic_lookup
 EXTRA_DSO_LDOPTS += $(LIBS_DIR) $(EXTRA_DSO_LIBS)
 
 ifdef MOZ_ENABLE_LIBXUL
 include $(srcdir)/libxul-rules.mk
diff -ruN mozilla-1.9.2-orig/xpcom/stub/Makefile.in mozilla-1.9.2/xpcom/stub/Makefile.in
--- mozilla-1.9.2-orig/xpcom/stub/Makefile.in	2010-04-07 01:48:44.000000000 -0700
+++ mozilla-1.9.2/xpcom/stub/Makefile.in	2010-04-07 02:11:28.000000000 -0700
@@ -80,6 +80,7 @@
 	$(LIB_PREFIX)nspr4$(DLL_SUFFIX) \
 	$(LIB_PREFIX)plc4$(DLL_SUFFIX) \
 	$(LIB_PREFIX)plds4$(DLL_SUFFIX) \
+	$(NSPR_DYLIB_FILES) \
 	$(NULL)
 
 ifdef MOZ_ENABLE_LIBXUL
@@ -119,7 +120,10 @@
 
 endif
 
+# patch: because -lcairo (shlib) was filtered out of link command
+EXTRA_DSO_LDOPTS += -Wl,-undefined,dynamic_lookup
 EXTRA_DSO_LDOPTS += $(EXTRA_DSO_LIBS) $(NSPR_LIBS)
+EXTRA_DSO_LDOPTS += $(XPCOM_DYLIB_FILES)
 
 include $(topsrcdir)/config/rules.mk
 
diff -ruN mozilla-1.9.2-orig/js/src/xpconnect/shell/Makefile.in mozilla-1.9.2/js/src/xpconnect/shell/Makefile.in
--- mozilla-1.9.2-orig/js/src/xpconnect/shell/Makefile.in	2010-04-08 19:02:44.000000000 -0700
+++ mozilla-1.9.2/js/src/xpconnect/shell/Makefile.in	2010-04-08 19:08:17.000000000 -0700
@@ -78,6 +78,8 @@
 include $(topsrcdir)/config/rules.mk
 
 DEFINES		+= -DJS_THREADSAFE
+# patch: because -lcairo (shlib) was filtered out of link command
+LDFLAGS += -Wl,-undefined,dynamic_lookup
 
 ifdef MOZ_SHARK
 DEFINES += -DMOZ_SHARK
diff -ruN mozilla-1.9.2-orig/browser/components/build/Makefile.in mozilla-1.9.2/browser/components/build/Makefile.in
--- mozilla-1.9.2-orig/browser/components/build/Makefile.in	2010-04-08 19:26:50.000000000 -0700
+++ mozilla-1.9.2/browser/components/build/Makefile.in	2010-04-08 19:28:31.000000000 -0700
@@ -68,6 +68,8 @@
 SHARED_LIBRARY_LIBS += ../shell/src/$(LIB_PREFIX)shellservice_s.$(LIB_SUFFIX)
 endif
 
+# patch: because -lcairo (shlib) was filtered out of link command
+EXTRA_DSO_LDOPTS += -Wl,-undefined,dynamic_lookup
 EXTRA_DSO_LDOPTS += $(call EXPAND_LIBNAME_PATH,unicharutil_external_s,$(LIBXUL_DIST)/lib)
 
 # migration requires mozreg, which doesn't build on WINCE; only include
diff -ruN mozilla-1.9.2-orig/browser/app/Makefile.in mozilla-1.9.2/browser/app/Makefile.in
--- mozilla-1.9.2-orig/browser/app/Makefile.in	2010-04-08 20:08:13.000000000 -0700
+++ mozilla-1.9.2/browser/app/Makefile.in	2010-04-08 20:14:06.000000000 -0700
@@ -363,6 +363,9 @@
 endif
 endif
 
+# patch: because -lcairo (shlib) was filtered out of link command
+LDFLAGS += -Wl,-undefined,dynamic_lookup
+
 libs::
 ifeq ($(OS_ARCH),WINNT)
 	$(PERL) -pe 's/(?<!\r)\n/\r\n/g;' < $(topsrcdir)/LICENSE > $(DIST)/bin/LICENSE
