--- gmp-5.0.0/mpn/x86_64/darwin.m4	2010-01-14 20:12:26.000000000 -0500
+++ gmp-5.0.0-patched/mpn/x86_64/darwin.m4	2010-01-14 20:15:09.000000000 -0500
@@ -22,4 +22,8 @@
 dnl
 dnl  Simply override the definition in x86_64-defs.m4.
 
+define(`LEA',`
+       lea     $1(%rip), $2
+')
+
 define(`CALL',`call	GSYM_PREFIX`'$1')
--- gmp-5.0.0/mpn/x86/invert_limb.asm	Tue Jan 12 23:50:39 2010 +0100
+++ gmp-5.0.0-patched/mpn/x86/invert_limb.asm	Fri Jan 15 00:27:41 2010 +0100
@@ -33,6 +33,13 @@
 
 defframe(PARAM_DIVISOR,4)
 
+ASM_START()
+
+C Make approx_tab global to work around Apple relocation bug.
+ifdef(`DARWIN',`
+	define(`approx_tab', MPN(invert_limb_tab))
+	GLOBL	approx_tab')
+
 	TEXT
 	ALIGN(16)
 PROLOGUE(mpn_invert_limb)
@@ -95,9 +102,7 @@
 
 EPILOGUE()
 
-	RODATA
-	ALIGN(2)
-approx_tab:
+DEF_OBJECT(approx_tab,2)
 	.value	0x7fe1,0x7fa1,0x7f61,0x7f22,0x7ee3,0x7ea4,0x7e65,0x7e27
 	.value	0x7de9,0x7dab,0x7d6d,0x7d30,0x7cf3,0x7cb6,0x7c79,0x7c3d
 	.value	0x7c00,0x7bc4,0x7b89,0x7b4d,0x7b12,0x7ad7,0x7a9c,0x7a61
@@ -162,3 +167,4 @@
 	.value	0x4179,0x4168,0x4157,0x4147,0x4136,0x4125,0x4115,0x4104
 	.value	0x40f4,0x40e3,0x40d3,0x40c2,0x40b2,0x40a2,0x4091,0x4081
 	.value	0x4071,0x4061,0x4050,0x4040,0x4030,0x4020,0x4010,0x4000
+END_OBJECT(approx_tab)
