diff -uNbr dlcompat-20020929/Makefile dlcompat-20020929-new/Makefile
--- dlcompat-20020929/Makefile	Tue Aug 27 09:06:52 2002
+++ dlcompat-20020929-new/Makefile	Sat Nov  9 00:47:29 2002
@@ -63,7 +63,7 @@
 	ln -sf libdl.1.dylib $@
 
 libdl.1.dylib: $(OBJS)
-	$(CC) -dynamiclib -o $@ $(OBJS) -install_name $(prefix)/lib/$@ -compatibility_version 1 -current_version 1
+	gcc-prebind $(CC) -dynamiclib -o $@ $(OBJS) -install_name $(prefix)/lib/$@ -compatibility_version 1 -current_version 1
 
 clean:
 	rm -f $(OBJS) libdl.* *~ *.o
diff -uNbr dlcompat-20020929/fink.mk dlcompat-20020929-new/fink.mk
--- dlcompat-20020929/fink.mk	Sat Sep 21 11:20:00 2002
+++ dlcompat-20020929-new/fink.mk	Sat Nov  9 00:45:03 2002
@@ -65,7 +65,7 @@
 	ln -sf libdl.0.dylib $@
 
 libdl.0.dylib: $(OBJS)
-	$(CC) -dynamiclib -o $@ $(OBJS) -install_name $(prefix)/lib/$@
+	gcc-prebind $(CC) -dynamiclib -o $@ $(OBJS) -install_name $(prefix)/lib/$@
 
 clean:
 	rm -f $(OBJS) libdl.* *~ *.o
