diff -uNbr kmplayer/kmplayer/Makefile.am kmplayer-new/kmplayer/Makefile.am
--- kmplayer/kmplayer/Makefile.am	Fri Feb 28 09:57:17 2003
+++ kmplayer-new/kmplayer/Makefile.am	Sat Mar 22 17:06:07 2003
@@ -2,27 +2,24 @@
 
 SUBDIRS=
 
-lib_LTLIBRARIES=kdeinit_kmplayer.la
-
+lib_LTLIBRARIES=libkparts_kmplayer.la
 kde_module_LTLIBRARIES = kparts_kmplayer.la
 
-kparts_kmplayer_la_SOURCES= kmplayer_part.cpp kmplayerview.cpp configdialog.ui kmplayerconfig.cpp
-
-kparts_kmplayer_la_LDFLAGS= $(all_libraries) -module $(KDE_PLUGIN) -lartskde -lm
-kparts_kmplayer_la_LIBADD= $(LIB_KFILE) $(LIB_KPARTS) $(LIB_KIO) -lkmediaplayer
-kparts_kmplayer_la_METASOURCES= kmplayer_part.moc kmplayerview.moc configdialog.moc kmplayerconfig.moc
-
-kdeinit_kmplayer_la_SOURCES= main.cpp kmplayerapp.cpp kmplayerdoc.cpp
-
-kdeinit_kmplayer_la_LIBADD= kparts_kmplayer.la
-kdeinit_kmplayer_la_LDFLAGS= $(all_libraries) $(KDE_RPATH) -module -avoid-version
-kdeinit_kmplayer_la_METASOURCES= kmplayer.moc kmplayerdoc.moc
+libkparts_kmplayer_la_SOURCES = kmplayer_part.cpp kmplayerview.cpp configdialog.ui kmplayerconfig.cpp
+libkparts_kmplayer_la_LDFLAGS = $(all_libraries) $(KDE_RPATH)
+libkparts_kmplayer_la_LIBADD  = $(LIB_KFILE) $(LIB_KPARTS) $(LIB_KIO) -lkmediaplayer -lartskde -lm
+libkparts_kmplayer_la_METASOURCES= kmplayer_part.moc kmplayerview.moc configdialog.moc kmplayerconfig.moc
+
+kparts_kmplayer_la_SOURCES = dummy.cpp
+kparts_kmplayer_la_LDFLAGS = $(all_libraries) $(KDE_PLUGIN) -module -avoid-version
+kparts_kmplayer_la_LIBADD  = libkparts_kmplayer.la
 
 bin_PROGRAMS= kmplayer
 
-kmplayer_LDADD= kdeinit_kmplayer.la
-kmplayer_LDFLAGS= $(all_libraries) $(KDE_RPATH)
-kmplayer_SOURCES= dummy.cpp
+kmplayer_SOURCES = main.cpp kmplayerapp.cpp kmplayerdoc.cpp
+kmplayer_LDADD   = libkparts_kmplayer.la
+kmplayer_LDFLAGS = $(all_libraries) $(KDE_RPATH) -avoid-version
+kmplayer_METASOURCES= kmplayer.moc kmplayerdoc.moc
 
 KDE_ICON = AUTO
 CLEANFILES = dummy.cpp
