Index: kopete/protocols/icq_new/Makefile.am
===================================================================
RCS file: /home/kde/kdenonbeta/kopete/protocols/icq_new/Makefile.am,v
retrieving revision 1.15
diff -u -u -b -r1.15 Makefile.am
--- kopete/protocols/icq_new/Makefile.am	11 Mar 2003 19:29:26 -0000	1.15
+++ kopete/protocols/icq_new/Makefile.am	23 Mar 2003 22:51:51 -0000
@@ -7,16 +7,24 @@
 	$(all_includes)
 
 kde_module_LTLIBRARIES = kopete_icq.la
+lib_LTLIBRARIES = libkopete_icq_shared.la
 
-kopete_icq_la_SOURCES = icqprotocol.cpp icqpreferences.cpp \
+libkopete_icq_shared_la_SOURCES = icqprotocol.cpp icqpreferences.cpp \
     icqaddcontactpage.cpp icqcontact.cpp  \
     icqfiletransfer.cpp icqchangestatus.cpp icquserinfo.cpp \
 	icquserreason.cpp enable.cpp client.cpp
-kopete_icq_la_LDFLAGS = -no-undefined -module $(KDE_PLUGIN) $(all_libraries)
-kopete_icq_la_LIBADD  = ./ui/libkopeteicqui.la \
+libkopete_icq_shared_la_LDFLAGS = -no-undefined $(KDE_RPATH) $(all_libraries)
+libkopete_icq_shared_la_LIBADD  = ./ui/libkopeteicqui.la \
 		       ./libicq/libicq.la \
 		       	$(top_builddir)/kopete/libkopete/libkopete.la \
 			$(LIB_KIO)
+
+kopete_icq_la_SOURCES = dummy.cpp
+kopete_icq_la_LDFLAGS = -no-undefined -module $(KDE_PLUGIN) $(all_libraries)
+kopete_icq_la_LIBADD  = libkopete_icq_shared.la
+
+dummy.cpp:
+	echo > dummy.cpp
 
 service_DATA = icq.desktop
 servicedir = $(kde_servicesdir)
Index: kopete/protocols/icq_new/libicq/Makefile.am
===================================================================
RCS file: /home/kde/kdenonbeta/kopete/protocols/icq_new/libicq/Makefile.am,v
retrieving revision 1.7
diff -u -u -b -r1.7 Makefile.am
--- kopete/protocols/icq_new/libicq/Makefile.am	27 Jan 2003 20:35:15 -0000	1.7
+++ kopete/protocols/icq_new/libicq/Makefile.am	23 Mar 2003 22:51:51 -0000
@@ -1,5 +1,7 @@
 noinst_LTLIBRARIES = libicq.la
 
+KDE_OPTIONS = nofinal
+
 libicq_la_METASOURCES = AUTO
 libicq_la_SOURCES = proxy.cpp icqssl.cpp direct.cpp \
 genrtf.cc clearhtml.cc country.cpp rtf.cc xml.cpp \
Index: kopete/protocols/icq_new/libicq/message.cpp
===================================================================
RCS file: /home/kde/kdenonbeta/kopete/protocols/icq_new/libicq/message.cpp,v
retrieving revision 1.14
diff -u -u -b -r1.14 message.cpp
--- kopete/protocols/icq_new/libicq/message.cpp	27 Feb 2003 21:02:56 -0000	1.14
+++ kopete/protocols/icq_new/libicq/message.cpp	23 Mar 2003 22:51:51 -0000
@@ -879,7 +879,7 @@
             }
             if (f->files.size() > 1){
                 char b[32];
-                snprintf(b, sizeof(b), "%u Files", f->files.size());
+                snprintf(b, sizeof(b), "%u Files", (unsigned int)f->files.size());
                 f->Name = b;
             }else{
                 f->Name = f->shortName();
Index: kopete/protocols/irc/libkirc/kirc.cpp
===================================================================
RCS file: /home/kde/kdenonbeta/kopete/protocols/irc/libkirc/kirc.cpp,v
retrieving revision 1.53
diff -u -u -b -r1.53 kirc.cpp
--- kopete/protocols/irc/libkirc/kirc.cpp	11 Mar 2003 21:33:03 -0000	1.53
+++ kopete/protocols/irc/libkirc/kirc.cpp	23 Mar 2003 22:51:51 -0000
@@ -15,6 +15,7 @@
     *************************************************************************
 */
 
+#include <config.h>
 #include "kirc.h"
 
 #include <kdebug.h>
Index: kopete/protocols/msn/Makefile.am
===================================================================
RCS file: /home/kde/kdenonbeta/kopete/protocols/msn/Makefile.am,v
retrieving revision 1.41
diff -u -u -b -r1.41 Makefile.am
--- kopete/protocols/msn/Makefile.am	11 Mar 2003 19:29:27 -0000	1.41
+++ kopete/protocols/msn/Makefile.am	23 Mar 2003 22:51:51 -0000
@@ -6,14 +6,22 @@
 	$(all_includes)
 
 kde_module_LTLIBRARIES = kopete_msn.la
+lib_LTLIBRARIES = libkopete_msn_shared.la
 
-kopete_msn_la_SOURCES = msnprotocol.cpp msnaccount.cpp \
+libkopete_msn_shared_la_SOURCES = msnprotocol.cpp msnaccount.cpp \
 	msnpreferences.cpp msnaddcontactpage.cpp \
 	msncontact.cpp msnsocket.cpp msnauthsocket.cpp msndispatchsocket.cpp msnmessagemanager.cpp\
 	msndebugrawcmddlg.cpp msnnotifysocket.cpp msnswitchboardsocket.cpp msnfiletransfersocket.cpp
-kopete_msn_la_LDFLAGS = -no-undefined -module $(KDE_PLUGIN) $(all_libraries)
-kopete_msn_la_LIBADD  = ./ui/libkopetemsnui.la \
+libkopete_msn_shared_la_LDFLAGS = -no-undefined $(KDE_RPATH) $(all_libraries)
+libkopete_msn_shared_la_LIBADD  = ./ui/libkopetemsnui.la \
 	../../libkopete/libkopete.la  $(LIB_KIO)
+
+kopete_msn_la_SOURCES = dummy.cpp
+kopete_msn_la_LDFLAGS = -no-undefined -module $(KDE_PLUGIN) $(all_libraries)
+kopete_msn_la_LIBADD  = libkopete_msn_shared.la
+
+dummy.cpp:
+	echo > dummy.cpp
 
 service_DATA = msn.desktop
 servicedir = $(kde_servicesdir)
Index: kopete/protocols/oscar/Makefile.am
===================================================================
RCS file: /home/kde/kdenonbeta/kopete/protocols/oscar/Makefile.am,v
retrieving revision 1.12
diff -u -u -b -r1.12 Makefile.am
--- kopete/protocols/oscar/Makefile.am	14 Mar 2003 22:04:29 -0000	1.12
+++ kopete/protocols/oscar/Makefile.am	23 Mar 2003 22:51:52 -0000
@@ -7,6 +7,8 @@
     -I$(srcdir)/oscarsocket \
     $(all_includes)
 
+KDE_OPTIONS = nofinal
+
 kde_module_LTLIBRARIES = kopete_oscar.la
 
 #kopete_aim_la_SOURCES = aimprotocol.cpp \
Index: kopete/protocols/oscar/oscarcontact.cpp
===================================================================
RCS file: /home/kde/kdenonbeta/kopete/protocols/oscar/oscarcontact.cpp,v
retrieving revision 1.69
diff -u -u -b -r1.69 oscarcontact.cpp
--- kopete/protocols/oscar/oscarcontact.cpp	23 Mar 2003 19:36:01 -0000	1.69
+++ kopete/protocols/oscar/oscarcontact.cpp	23 Mar 2003 22:51:52 -0000
@@ -15,6 +15,7 @@
   *************************************************************************
   */
 
+#include <sys/types.h>
 #include "oscarcontact.h"
 
 #include <time.h>
Index: kopete/plugins/importer/Makefile.am
===================================================================
RCS file: /home/kde/kdenonbeta/kopete/plugins/importer/Makefile.am,v
retrieving revision 1.8
diff -u -u -b -r1.8 Makefile.am
--- kopete/plugins/importer/Makefile.am	11 Mar 2003 19:29:26 -0000	1.8
+++ kopete/plugins/importer/Makefile.am	24 Mar 2003 00:26:53 -0000
@@ -5,7 +5,7 @@
 kopete_importer_la_SOURCES     = importerplugin.cpp importer.cpp imp.cpp importeruibase.ui
 kopete_importer_la_METASOURCES = AUTO
 kopete_importer_la_LDFLAGS     = -module
-kopete_importer_la_LIBADD      = ../../libkopete/libkopete.la ../../protocols/msn/kopete_msn.la ../../protocols/icq_new/kopete_icq.la -lkio
+kopete_importer_la_LIBADD      = ../../libkopete/libkopete.la ../../protocols/msn/libkopete_msn_shared.la ../../protocols/icq_new/libkopete_icq_shared.la -lkio
 
 service_DATA = importer.desktop
 servicedir = $(kde_servicesdir)
