Index: admin/Makefile.common
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/admin/Makefile.common,v
retrieving revision 1.1.1.2
retrieving revision 1.3
diff -u -b -u -r1.1.1.2 -r1.3
--- admin/Makefile.common	25 Jun 2002 03:35:55 -0000	1.1.1.2
+++ admin/Makefile.common	25 Jun 2002 03:36:46 -0000	1.3
@@ -1,24 +1,12 @@
 
 SHELL=/bin/sh
 
+ACLOCAL=aclocal-1.5
+AUTOMAKE=automake-1.5
+AUTOCONF=autoconf
+
 cvs:
-	if test -x /usr/bin/autoconf2.50 ; then \
-	   AUTOCONF="/usr/bin/autoconf2.50" ;\
-	else \
-	   AUTOCONF="autoconf" ;\
-	fi; \
-	AUTOCONF_VERSION=`$$AUTOCONF --version` ;\
-	if test "x$$AUTOCONF_VERSION" = "xAutoconf version 2.13"; then \
-	   echo "*** YOU'RE USING AUTOCONF 2.13. We suggest updating." ;\
-	   echo "*** autoconf 2.5x works best for now, we will drop" ;\
-	   echo "*** support for autoconf 2.13 soon."; \
-	   test ! -f admin/libtool.m4.in || mv admin/libtool.m4.in admin/new-libtool.m4.in ;\
-	   test ! -f admin/ltmain.sh || mv admin/ltmain.sh admin/new-ltmain.sh ;\
-	   for file in libtool.m4.in ltcf-c.sh ltcf-cxx.sh ltcf-gcj.sh ltconfig ltmain.sh; do \
-		rm -f admin/$$file ;\
-		ln -s old-$$file admin/$$file ;\
-	   done ;\
-	fi 
+	@chmod a+x admin/*
 	@if grep '$$(top_srcdir)/acinclude.m4:' Makefile.am >/dev/null; then \
 	  echo "*** Creating acinclude.m4" ;\
 	  rm -f acinclude.m4 configure.files ;\
@@ -34,7 +22,7 @@
 	  $(MAKE) -f Makefile.am top_srcdir=. configure.in ;\
 	fi
 	@echo "*** Creating aclocal.m4"
-	@aclocal
+	@$(ACLOCAL)
 	@echo "*** Creating configure"
 	@autoconf
 	@if test -r configure.in.in ; then \
@@ -52,7 +40,7 @@
 		autoheader ;\
 	fi
 	@echo "*** Creating Makefile templates"
-	@automake
+	@$(AUTOMAKE)
 	@echo "*** Postprocessing Makefile templates"
 	@perl admin/am_edit
 	@if egrep "^cvs-local:" Makefile.am >/dev/null; then \
@@ -71,9 +59,9 @@
 	@if test -r configure.in.in; then \
 	  $(MAKE) -f Makefile.am top_srcdir=. subdirs configure.in; \
 	fi ;\
-	aclocal; \
+	$(ACLOCAL); \
 	autoheader; \
-	automake --foreign --include-deps; \
+	$(AUTOMAKE) --foreign --include-deps; \
 	perl admin/am_edit; \
 	autoconf; \
 	if test -r configure.in.in ; then \
@@ -281,7 +269,9 @@
 	done ;\
 	rm -f _SUBDIRS ;\
 	for i in $$dirs; do \
+	if [ $$i != "kdm" ]; then \
 	echo $$i >> ./_SUBDIRS; \
+        fi \
 	done
 	@if test -r subdirs && diff subdirs _SUBDIRS > /dev/null; then \
 	  rm -f _SUBDIRS; \
Index: admin/acinclude.m4.in
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/admin/acinclude.m4.in,v
retrieving revision 1.1.1.1
retrieving revision 1.2
diff -u -b -u -r1.1.1.1 -r1.2
--- admin/acinclude.m4.in	25 Jun 2002 03:34:27 -0000	1.1.1.1
+++ admin/acinclude.m4.in	25 Jun 2002 03:35:36 -0000	1.2
@@ -693,6 +693,19 @@
    AC_CHECK_FUNC(shmat, ,
      AC_CHECK_LIB(ipc, shmat, X_EXTRA_LIBS="$X_EXTRA_LIBS -lipc"))
    
+   # darwin needs this to initialize the environment
+   AC_CHECK_HEADERS([crt_externs.h sys/types.h])
+   AC_CHECK_FUNC(_NSGetEnviron, [AC_DEFINE(HAVE_NSGETENVIRON, 1, [Define if your system needs _NSGetEnviron to set up the environment])])
+
+   # darwin requires a poll.h emulation library
+   AC_CHECK_LIB(poll, poll, LIB_POLL="-lpoll")
+
+   # CoreAudio framework
+   AC_CHECK_HEADER(CoreAudio/CoreAudio.h, [
+     AC_DEFINE(HAVE_COREAUDIO, 1, [Define if you have the CoreAudio API])
+     FRAMEWORK_COREAUDIO="-framework CoreAudio"
+   ])
+
    # Solaris 2.6 and others need -lresolv for res_init
    AC_CHECK_FUNCS(res_init, , [
      kde_libs_safe="$LIBS"
@@ -713,6 +726,8 @@
 
    LIBSOCKET="$X_EXTRA_LIBS"
    AC_SUBST(LIBSOCKET)
+   AC_SUBST(FRAMEWORK_COREAUDIO)
+   AC_SUBST(LIB_POLL)
    AC_SUBST(LIBRESOLV)
    AC_SUBST(X_EXTRA_LIBS)
    AC_CHECK_LIB(ucb, killpg, [LIBUCB="-lucb"]) dnl for Solaris2.4
@@ -1826,6 +1841,32 @@
 if test $kde_qtver = 3; then
    LIB_KDECORE='-lkdecore'
    AC_SUBST(LIB_KDECORE)
+   LIB_KDEFX='-lkdefx'
+   AC_SUBST(LIB_KDEFX)
+   LIB_ARTSKDE='-lartskde'
+   AC_SUBST(LIB_ARTSKDE)
+   LIB_KABC='-lkabc'
+   AC_SUBST(LIB_KABC)
+   LIB_KDESASL='-lkdesasl'
+   AC_SUBST(LIB_KDESASL)
+   LIB_KDESU='-lkdesu'
+   AC_SUBST(LIB_KDESU)
+   LIB_KJAVA='-lkjava'
+   AC_SUBST(LIB_KJAVA)
+   LIB_KJS='-lkjs'
+   AC_SUBST(LIB_KJS)
+   LIB_KMID='-lkmid'
+   AC_SUBST(LIB_KMID)
+   LIB_KSCRIPT='-lkscript'
+   AC_SUBST(LIB_KSCRIPT)
+   LIB_KTEXTEDITOR='-lktexteditor'
+   AC_SUBST(LIB_KTEXTEDITOR)
+   LIB_SHELLSCRIPT='-lshellscript'
+   AC_SUBST(LIB_SHELLSCRIPT)
+   LIB_VCARD='-lvcard'
+   AC_SUBST(LIB_VCARD)
+   LIB_DCOP='-lDCOP'
+   AC_SUBST(LIB_DCOP)
    LIB_KDEUI='-lkdeui'
    AC_SUBST(LIB_KDEUI)
    LIB_KIO='-lkio'
@@ -1842,6 +1883,8 @@
    AC_SUBST(LIB_KPARTS)
    LIB_KDEPRINT='-lkdeprint'
    AC_SUBST(LIB_KDEPRINT)
+   LIB_KDED='@PREFIX@/lib/kded.la'
+   AC_SUBST(LIB_KDED)
 # these are for backward compatibility
    LIB_KSYCOCA='-lkio'
    AC_SUBST(LIB_KSYCOCA)
@@ -4158,7 +4201,7 @@
 AC_DEFUN(KDE_CREATE_SUBDIRSLIST,
 [
 
-DO_NOT_COMPILE="$DO_NOT_COMPILE CVS debian bsd-port admin"
+DO_NOT_COMPILE="$DO_NOT_COMPILE CVS debian bsd-port admin kdm ksysguard kscd"
 
 if test ! -s $srcdir/subdirs; then
   dnl Note: Makefile.common creates subdirs, so this is just a fallback
@@ -4530,9 +4573,12 @@
            [initgroups may exist but not its prototype (e.g. AIX<4.3.3:8)])
 ])
 
-
 AC_DEFUN(KDE_CHECK_JAVA_DIR,
 [
+
+# needed by the MacOS X JVM in certain circumstances
+AC_CHECK_HEADERS([TargetConditionals.h])
+
 AC_MSG_CHECKING([for Java directory])
 
 AC_ARG_WITH(java,
@@ -4541,6 +4587,7 @@
 ], ac_java_dir=""
 )
 
+
 dnl at this point ac_java_dir is either a dir, 'no' to disable, or '' to say look in $PATH
 if test "x$ac_java_dir" = "xno"; then
    kde_cv_java_bindir=no
@@ -4572,10 +4619,20 @@
 dnl At this point kde_cv_java_bindir and kde_cv_java_includedir are either set or "no"
 if test "x$kde_cv_java_bindir" != "xno"; then
 
+  case $host_os in
+    darwin*)
+      dnl Look for libjvm.dylib
+      kde_java_libjvmdir=`find /System/Library/Frameworks/JavaVM.framework/ -name libjvm.dylib | sed 's,libjvm.dylib,,'|head -n 1`
+      kd_java_libhpidir=''
+      dnl Darwin doesn't have libhpi
+      ;;
+    *)
   dnl Look for libjvm.so
   kde_java_libjvmdir=`find $kde_cv_java_bindir/.. -name libjvm.so | sed 's,libjvm.so,,'|head -n 1`
   dnl Look for libhpi.so and avoid green threads
   kde_java_libhpidir=`find $kde_cv_java_bindir/.. -name libhpi.so | grep -v green | sed 's,libhpi.so,,'`
+      ;;
+  esac
 
   dnl Now check everything's fine under there
 
@@ -4591,12 +4648,26 @@
   if test ! -r "$kde_cv_java_includedir/jni.h"; then
     AC_MSG_ERROR([jni.h not found under $kde_cv_java_includedir. Use --with-java or --without-java.])
   fi
+  case $host_os in
+    darwin*)
+      if test ! -r "$kde_java_libjvmdir/libjvm.dylib"; then
+        AC_MSG_ERROR([libjvm.dylib not found under $kde_java_libjvmdir. Use --without-java.])
+      fi
+      dnl Darwin doesn't have libhpi, but it's not fatal
+      LIB_JVM="-framework JavaVM"
+      LIB_HPI=""
+      ;;
+    *)
   if test ! -r "$kde_java_libjvmdir/libjvm.so"; then
     AC_MSG_ERROR([libjvm.so not found under $kde_java_libjvmdir. Use --without-java.])
   fi
+      LIB_JVM="-L$kde_java_libjvmdir -ljvm"
   if test ! -r "$kde_java_libhpidir/libhpi.so"; then
     AC_MSG_ERROR([libhpi.so not found under $kde_java_libhpidir. Use --without-java.])
   fi
+      LIB_HPI="-L$kde_java_libhpidir -lhpi"
+      ;;
+  esac
 
   jni_includes="-I$kde_cv_java_includedir"
   dnl Strange thing, jni.h requires jni_md.h which is under genunix here..
@@ -4643,7 +4714,7 @@
   JAR=$kde_cv_java_bindir/jar
   AC_SUBST(JAR)
   AC_SUBST(jni_includes)
-  JVMLIBS="-L$kde_java_libjvmdir -ljvm -L$kde_java_libhpidir -lhpi"
+  JVMLIBS="$LIB_JVM $LIB_HPI"
   AC_SUBST(JVMLIBS)
 fi
 ])
Index: admin/libtool.m4.in
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/admin/libtool.m4.in,v
retrieving revision 1.1.1.1
retrieving revision 1.2
diff -u -b -u -r1.1.1.1 -r1.2
--- admin/libtool.m4.in	25 Jun 2002 03:34:27 -0000	1.1.1.1
+++ admin/libtool.m4.in	25 Jun 2002 03:35:36 -0000	1.2
@@ -593,6 +593,12 @@
     lt_cv_sys_max_cmd_len=-1;
     ;;
 
+  darwin*)
+    # this should be enough... the length test freaks out some versions
+    # of OSX's `expr`
+    lt_cv_sys_max_cmd_len=16384;
+    ;;
+
   *)
     # If test is not a shell built-in, we'll probably end up computing a
     # maximum length that is only half of the actual maximum length, but
@@ -1876,16 +1882,7 @@
   ;;
 
 darwin* | rhapsody*)
-  lt_cv_deplibs_check_method='file_magic Mach-O dynamically linked shared library'
-  lt_cv_file_magic_cmd='/usr/bin/file -L'
-  case "$host_os" in
-  rhapsody* | darwin1.[[012]])
-    lt_cv_file_magic_test_file=`/System/Library/Frameworks/System.framework/System`
-    ;;
-  *) # Darwin 1.3 on
-    lt_cv_file_magic_test_file='/usr/lib/libSystem.dylib'
-    ;;
-  esac
+  lt_cv_deplibs_check_method='pass_all'
   ;;
 
 freebsd*)
@@ -2535,6 +2532,11 @@
 	;;
     esac
     ;;
+  darwin*)
+    shared_flag='-dynamiclib'
+    _LT_AC_TAGVAR(archive_cmds, $1)='$CC -r -keep_private_externs -nostdlib -o ${lib}-master.o $libobjs && $CC -dynamiclib -install_name $rpath/$soname $predep_objects ${lib}-master.o $deplibs $postdep_objects $compiler_flags -o $lib'
+    output_verbose_link_cmd='$CC -dynamiclib $CFLAGS -v conftest.$objext 2>&1 | egrep "\-L"'
+    ;;
   dgux*)
     case $cc_basename in
       ec++)
@@ -4725,7 +4727,9 @@
 	_LT_AC_TAGVAR(allow_undefined_flag, $1)='-undefined suppress'
 	;;
       *) # Darwin 1.3 on
-	_LT_AC_TAGVAR(allow_undefined_flag, $1)='-flat_namespace -undefined suppress'
+	#_LT_AC_TAGVAR(allow_undefined_flag, $1)='-flat_namespace -undefined suppress'
+        # testing with twolevel namespaces
+	_LT_AC_TAGVAR(allow_undefined_flag, $1)=unsupported
 	;;
       esac
 
@@ -4733,7 +4737,7 @@
       #        cross-compilation, but unfortunately the echo tests do not
       #        yet detect zsh echo's removal of \ escapes.  Also zsh mangles
       #	       `"' quotes if we put them in here... so don't!
-      _LT_AC_TAGVAR(archive_cmds, $1)='$CC $(test .$module = .yes && echo -bundle || echo -dynamiclib) $allow_undefined_flag -o $lib $libobjs $deplibs$linker_flags -install_name $rpath/$soname $verstring'
+      _LT_AC_TAGVAR(archive_cmds, $1)='$CC -r -keep_private_externs -nostdlib -o ${lib}-master.o $libobjs && $CC $(test .$module = .yes && echo -bundle || echo -dynamiclib) -install_name $rpath/$soname $allow_undefined_flag -o $lib ${lib}-master.o $deplibs $linker_flags $verstring'
       # We need to add '_' to the symbols in $export_symbols first
       #_LT_AC_TAGVAR(archive_expsym_cmds, $1)="$_LT_AC_TAGVAR(archive_cmds, $1)"' && strip -s $export_symbols'
       _LT_AC_TAGVAR(hardcode_direct, $1)=yes
Index: admin/ltmain.sh
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/admin/ltmain.sh,v
retrieving revision 1.1.1.1
retrieving revision 1.2
diff -u -b -u -r1.1.1.1 -r1.2
--- admin/ltmain.sh	25 Jun 2002 03:34:27 -0000	1.1.1.1
+++ admin/ltmain.sh	25 Jun 2002 03:35:36 -0000	1.2
@@ -3324,7 +3324,18 @@
 	if test -n "$export_symbols" && test -n "$archive_expsym_cmds"; then
 	  eval cmds=\"$archive_expsym_cmds\"
 	else
+ 	  save_deplibs="$deplibs"
+ 	  for conv in $convenience; do
+            tmp_deplibs=
+            for test_deplib in $deplibs; do
+              if test "$test_deplib" != "$conv"; then
+                tmp_deplibs="$tmp_deplibs $test_deplib"
+              fi
+            done
+            deplibs="$tmp_deplibs"
+ 	  done
 	  eval cmds=\"$archive_cmds\"
+ 	  deplibs="$save_deplibs"
 	fi
 
 	if len=`expr "X$cmds" : ".*"` &&
@@ -3635,6 +3646,14 @@
 	compile_deplibs=`$echo "X $compile_deplibs" | $Xsed -e 's/ -lc / -framework System /'`
 	finalize_deplibs=`$echo "X $finalize_deplibs" | $Xsed -e 's/ -lc / -framework System /'`
 	;;
+      esac
+
+      case $host in
+      *darwin*)
+        # Don't allow lazy linking, it breaks C++ global constructors
+        compile_command="$compile_command ${wl}-bind_at_load"
+        finalize_command="$finalize_command ${wl}-bind_at_load"
+        ;;
       esac
 
       compile_command="$compile_command $compile_deplibs"
Index: admin/missing
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/admin/missing,v
retrieving revision 1.1.1.1
retrieving revision 1.2
diff -u -b -u -r1.1.1.1 -r1.2
--- admin/missing	25 Jun 2002 03:34:27 -0000	1.1.1.1
+++ admin/missing	25 Jun 2002 03:35:36 -0000	1.2
@@ -87,7 +87,7 @@
     exit 1
     ;;
 
-  aclocal)
+  aclocal*)
     echo 1>&2 "\
 WARNING: \`$1' is missing on your system.  You should only need it if
          you modified \`acinclude.m4' or \`${configure_ac}'.  You might want
@@ -124,7 +124,7 @@
     touch $touch_files
     ;;
 
-  automake)
+  automake*)
     echo 1>&2 "\
 WARNING: \`$1' is missing on your system.  You should only need it if
          you modified \`Makefile.am', \`acinclude.m4' or \`${configure_ac}'.
Index: ark/Makefile.am
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/kdeutils/ark/Makefile.am,v
retrieving revision 1.1.1.2
retrieving revision 1.3
diff -u -b -u -r1.1.1.2 -r1.3
--- ark/Makefile.am	25 Jun 2002 03:26:08 -0000	1.1.1.2
+++ ark/Makefile.am	25 Jun 2002 03:26:40 -0000	1.3
@@ -20,7 +20,7 @@
 libark_la_LDFLAGS = $(all_libraries) -version-info 1:0:0 -module $(LIB_KIO)
 
 # KParts
-libark_la_LIBADD   = $(LIB_KFILE) -lkparts
+libark_la_LIBADD   = $(LIB_KFILE) -lkparts $(LIB_KDEUI) $(LIB_QT) $(LIB_DCOP) $(LIB_KDECORE)
 
 # Which sources should be compiled for ark.
 ark_SOURCES    = dummy.cpp
Index: debian/kregexpeditor.install
===================================================================
RCS file: debian/kregexpeditor.install
diff -N debian/kregexpeditor.install
--- /dev/null	1 Jan 1970 00:00:00 -0000
+++ debian/kregexpeditor.install	25 Jun 2002 03:24:54 -0000	1.1.1.1
@@ -0,0 +1,5 @@
+debian/tmp/usr/lib/kde3/libkregexpeditorgui.la
+debian/tmp/usr/lib/kde3/libkregexpeditorgui.so
+debian/tmp/usr/share/apps/kregexpeditor
+debian/tmp/usr/share/doc/kde/HTML/en/KRegExpEditor
+debian/tmp/usr/share/services/kregexpeditorgui.desktop
Index: kcalc/Makefile.am
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/kdeutils/kcalc/Makefile.am,v
retrieving revision 1.1.1.1
retrieving revision 1.2
diff -u -b -u -r1.1.1.1 -r1.2
--- kcalc/Makefile.am	25 Jun 2002 03:24:54 -0000	1.1.1.1
+++ kcalc/Makefile.am	25 Jun 2002 03:25:51 -0000	1.2
@@ -11,7 +11,7 @@
 
 INCLUDES= $(all_includes)
 kcalc_la_LDFLAGS = $(all_libraries) -module -avoid-version
-kcalc_la_LIBADD = $(LIB_KDEUI) -lm
+kcalc_la_LIBADD = $(LIB_KDEUI) -lm $(LIB_KDECORE) $(LIB_QT)
 
 kcalc_SOURCES = dummy.cpp
 kcalc_LDADD = kcalc.la
Index: kedit/Makefile.am
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/kdeutils/kedit/Makefile.am,v
retrieving revision 1.1.1.1
retrieving revision 1.2
diff -u -b -u -r1.1.1.1 -r1.2
--- kedit/Makefile.am	25 Jun 2002 03:24:54 -0000	1.1.1.1
+++ kedit/Makefile.am	25 Jun 2002 03:25:51 -0000	1.2
@@ -12,7 +12,7 @@
 
 kedit_la_SOURCES = kedit.cpp ktextfiledlg.cpp\
 		mail.cpp  optiondialog.cpp 
-kedit_la_LIBADD = -lkdeprint -lkspell $(LIB_KFILE) $(LIB_KIO)
+kedit_la_LIBADD = -lkdeprint -lkspell $(LIB_KFILE) $(LIB_KIO) $(LIB_QT) $(LIB_KDECORE) $(LIB_KDEUI)
 kedit_la_LDFLAGS = $(all_libraries) -module -avoid-version
 
 kedit_SOURCES = dummy.cpp
Index: khexedit/chartabledialog.cc
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/kdeutils/khexedit/chartabledialog.cc,v
retrieving revision 1.1.1.1
retrieving revision 1.2
diff -u -b -u -r1.1.1.1 -r1.2
--- khexedit/chartabledialog.cc	25 Jun 2002 03:24:54 -0000	1.1.1.1
+++ khexedit/chartabledialog.cc	25 Jun 2002 03:25:51 -0000	1.2
@@ -19,6 +19,7 @@
  */
 
 #include <limits.h>
+#define MAXINT 2147483647
 
 #include <qlabel.h>
 #include <qheader.h>
Index: khexedit/dialog.cc
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/kdeutils/khexedit/dialog.cc,v
retrieving revision 1.1.1.1
retrieving revision 1.2
diff -u -b -u -r1.1.1.1 -r1.2
--- khexedit/dialog.cc	25 Jun 2002 03:24:54 -0000	1.1.1.1
+++ khexedit/dialog.cc	25 Jun 2002 03:25:51 -0000	1.2
@@ -18,9 +18,11 @@
  *
  */
 
+#include <limits.h>
+#define MAXINT 2147483647
+
 #include <ctype.h>
 #include <iostream>
-#include <limits.h>
 
 #include <qbuttongroup.h> 
 #include <qfileinfo.h> 
Index: khexedit/exportdialog.cc
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/kdeutils/khexedit/exportdialog.cc,v
retrieving revision 1.1.1.1
retrieving revision 1.2
diff -u -b -u -r1.1.1.1 -r1.2
--- khexedit/exportdialog.cc	25 Jun 2002 03:24:54 -0000	1.1.1.1
+++ khexedit/exportdialog.cc	25 Jun 2002 03:25:51 -0000	1.2
@@ -19,6 +19,8 @@
  */
 
 #include <limits.h>
+#define MAXINT 2147483647
+
 #include <qbuttongroup.h>
 #include <qlineedit.h> 
 
Index: khexedit/optiondialog.cc
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/kdeutils/khexedit/optiondialog.cc,v
retrieving revision 1.1.1.1
retrieving revision 1.2
diff -u -b -u -r1.1.1.1 -r1.2
--- khexedit/optiondialog.cc	25 Jun 2002 03:24:54 -0000	1.1.1.1
+++ khexedit/optiondialog.cc	25 Jun 2002 03:25:51 -0000	1.2
@@ -19,6 +19,7 @@
  */
 
 #include <limits.h>
+#define MAXINT 2147483647
 
 #include <qvbox.h>
 #include <qvbuttongroup.h>
Index: khexedit/printerdialog.cc
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/kdeutils/khexedit/printerdialog.cc,v
retrieving revision 1.1.1.2
retrieving revision 1.3
diff -u -b -u -r1.1.1.2 -r1.3
--- khexedit/printerdialog.cc	25 Jun 2002 03:26:12 -0000	1.1.1.2
+++ khexedit/printerdialog.cc	25 Jun 2002 03:26:41 -0000	1.3
@@ -19,6 +19,7 @@
  */
 
 #include <limits.h>
+#define MAXINT 2147483647
 
 #include <qbuttongroup.h>
 #include <qfileinfo.h> 
Index: klaptopdaemon/Makefile.am
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/kdeutils/klaptopdaemon/Makefile.am,v
retrieving revision 1.1.1.1
retrieving revision 1.2
diff -u -b -u -r1.1.1.1 -r1.2
--- klaptopdaemon/Makefile.am	25 Jun 2002 03:24:54 -0000	1.1.1.1
+++ klaptopdaemon/Makefile.am	25 Jun 2002 03:25:51 -0000	1.2
@@ -5,7 +5,7 @@
 
 klaptopdaemon_la_SOURCES = laptop_daemon.cpp kpcmcia.cpp kpcmciainfo.cpp daemondock.cpp notify.cpp portable.cpp
 klaptopdaemon_la_LDFLAGS = $(all_libraries) -module -avoid-version
-klaptopdaemon_la_LIBADD   = $(LIB_KDEUI)
+klaptopdaemon_la_LIBADD   = $(LIB_KDEUI) $(LIB_QT) $(LIB_DCOP) $(LIB_KDECORE)
 
 klaptopdaemon_SOURCES = dummy.cpp
 klaptopdaemon_LDFLAGS = $(all_libraries)
Index: klprfax/Makefile.am
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/kdeutils/klprfax/Makefile.am,v
retrieving revision 1.1.1.1
retrieving revision 1.2
diff -u -b -u -r1.1.1.1 -r1.2
--- klprfax/Makefile.am	25 Jun 2002 03:24:55 -0000	1.1.1.1
+++ klprfax/Makefile.am	25 Jun 2002 03:25:51 -0000	1.2
@@ -1,6 +1,6 @@
 ####### kdevelop will overwrite this part!!! (begin)##########
 
-SUBDIRS = klprfax efax 
+SUBDIRS = klprfax
 
 EXTRA_DIST = klprfax.kdevprj AUTHORS COPYING ChangeLog INSTALL README TODO klprfax.lsm 
 
