diff -uNr glitz-0.5.3/configure glitz-0.5.3-patched/configure
--- glitz-0.5.3/configure	2006-02-15 22:20:58.000000000 -0500
+++ glitz-0.5.3-patched/configure	2006-03-16 11:16:53.000000000 -0500
@@ -20218,7 +20218,7 @@
 
 
 
-GLITZ_LIB='-L$(top_builddir)/src -lglitz'
+GLITZ_LIB='$(top_builddir)/src/libglitz.la'
 GLITZ_INC='-I$(top_builddir)/src -I$(top_srcdir)/src'
 
 
diff -uNr glitz-0.5.3/src/agl/glitz_agl_format.c glitz-0.5.3-patched/src/agl/glitz_agl_format.c
--- glitz-0.5.3/src/agl/glitz_agl_format.c	2006-02-10 04:10:35.000000000 -0500
+++ glitz-0.5.3-patched/src/agl/glitz_agl_format.c	2006-03-16 11:16:09.000000000 -0500
@@ -231,7 +231,7 @@
 
     format.types	= GLITZ_DRAWABLE_TYPE_WINDOW_MASK;
     format.d.id		= 0;
-    format.color.fourcc = GLITZ_FOURCC_RGB;
+    format.d.color.fourcc = GLITZ_FOURCC_RGB;
 
     n_attribs_list = sizeof (_attribs_list) / sizeof (GLint *);
 
