diff -Naur anjuta-1.2.2/manuals/C/anjuta-advanced-tutorial/Makefile.in anjuta-1.2.2.fink/manuals/C/anjuta-advanced-tutorial/Makefile.in
--- anjuta-1.2.2/manuals/C/anjuta-advanced-tutorial/Makefile.in	Wed Apr 14 14:07:39 2004
+++ anjuta-1.2.2.fink/manuals/C/anjuta-advanced-tutorial/Makefile.in	Wed Oct 27 05:49:55 2004
@@ -373,7 +373,7 @@
 	for file in $(omffile); do \
 		$(INSTALL_DATA) $$file.out $(DESTDIR)$(omf_dest_dir)/$$file; \
 	done
-	-scrollkeeper-update -p $(scrollkeeper_localstate_dir) -o $(DESTDIR)$(omf_dest_dir)
+	#-scrollkeeper-update -p $(scrollkeeper_localstate_dir) -o $(DESTDIR)$(omf_dest_dir)
 
 uninstall-local-omf:
 	-for file in $(srcdir)/*.omf; do \
@@ -381,7 +381,7 @@
 		rm -f $(omf_dest_dir)/$$basefile; \
 	done
 	-rmdir $(omf_dest_dir)
-	-scrollkeeper-update -p $(scrollkeeper_localstate_dir)
+	#-scrollkeeper-update -p $(scrollkeeper_localstate_dir)
 
 clean-local-omf:
 	-for file in $(omffile); do \
diff -Naur anjuta-1.2.2/manuals/C/anjuta-faqs/Makefile.in anjuta-1.2.2.fink/manuals/C/anjuta-faqs/Makefile.in
--- anjuta-1.2.2/manuals/C/anjuta-faqs/Makefile.in	Wed Apr 14 14:07:40 2004
+++ anjuta-1.2.2.fink/manuals/C/anjuta-faqs/Makefile.in	Wed Oct 27 05:50:09 2004
@@ -373,7 +373,7 @@
 	for file in $(omffile); do \
 		$(INSTALL_DATA) $$file.out $(DESTDIR)$(omf_dest_dir)/$$file; \
 	done
-	-scrollkeeper-update -p $(scrollkeeper_localstate_dir) -o $(DESTDIR)$(omf_dest_dir)
+	#-scrollkeeper-update -p $(scrollkeeper_localstate_dir) -o $(DESTDIR)$(omf_dest_dir)
 
 uninstall-local-omf:
 	-for file in $(srcdir)/*.omf; do \
@@ -381,7 +381,7 @@
 		rm -f $(omf_dest_dir)/$$basefile; \
 	done
 	-rmdir $(omf_dest_dir)
-	-scrollkeeper-update -p $(scrollkeeper_localstate_dir)
+	#-scrollkeeper-update -p $(scrollkeeper_localstate_dir)
 
 clean-local-omf:
 	-for file in $(omffile); do \
diff -Naur anjuta-1.2.2/manuals/C/anjuta-manual/Makefile.in anjuta-1.2.2.fink/manuals/C/anjuta-manual/Makefile.in
--- anjuta-1.2.2/manuals/C/anjuta-manual/Makefile.in	Wed Apr 14 14:07:41 2004
+++ anjuta-1.2.2.fink/manuals/C/anjuta-manual/Makefile.in	Wed Oct 27 05:50:18 2004
@@ -380,7 +380,7 @@
 	for file in $(omffile); do \
 		$(INSTALL_DATA) $$file.out $(DESTDIR)$(omf_dest_dir)/$$file; \
 	done
-	-scrollkeeper-update -p $(scrollkeeper_localstate_dir) -o $(DESTDIR)$(omf_dest_dir)
+	#-scrollkeeper-update -p $(scrollkeeper_localstate_dir) -o $(DESTDIR)$(omf_dest_dir)
 
 uninstall-local-omf:
 	-for file in $(srcdir)/*.omf; do \
@@ -388,7 +388,7 @@
 		rm -f $(omf_dest_dir)/$$basefile; \
 	done
 	-rmdir $(omf_dest_dir)
-	-scrollkeeper-update -p $(scrollkeeper_localstate_dir)
+	#-scrollkeeper-update -p $(scrollkeeper_localstate_dir)
 
 clean-local-omf:
 	-for file in $(omffile); do \
diff -Naur anjuta-1.2.2/manuals/C/anjuta-tutorial/Makefile.in anjuta-1.2.2.fink/manuals/C/anjuta-tutorial/Makefile.in
--- anjuta-1.2.2/manuals/C/anjuta-tutorial/Makefile.in	Wed Apr 14 14:07:39 2004
+++ anjuta-1.2.2.fink/manuals/C/anjuta-tutorial/Makefile.in	Wed Oct 27 05:50:25 2004
@@ -373,7 +373,7 @@
 	for file in $(omffile); do \
 		$(INSTALL_DATA) $$file.out $(DESTDIR)$(omf_dest_dir)/$$file; \
 	done
-	-scrollkeeper-update -p $(scrollkeeper_localstate_dir) -o $(DESTDIR)$(omf_dest_dir)
+	#-scrollkeeper-update -p $(scrollkeeper_localstate_dir) -o $(DESTDIR)$(omf_dest_dir)
 
 uninstall-local-omf:
 	-for file in $(srcdir)/*.omf; do \
@@ -381,7 +381,7 @@
 		rm -f $(omf_dest_dir)/$$basefile; \
 	done
 	-rmdir $(omf_dest_dir)
-	-scrollkeeper-update -p $(scrollkeeper_localstate_dir)
+	#-scrollkeeper-update -p $(scrollkeeper_localstate_dir)
 
 clean-local-omf:
 	-for file in $(omffile); do \
diff -Naur anjuta-1.2.2/manuals/de/anjuta-tutorial/Makefile.in anjuta-1.2.2.fink/manuals/de/anjuta-tutorial/Makefile.in
--- anjuta-1.2.2/manuals/de/anjuta-tutorial/Makefile.in	Wed Apr 14 14:07:42 2004
+++ anjuta-1.2.2.fink/manuals/de/anjuta-tutorial/Makefile.in	Wed Oct 27 05:50:32 2004
@@ -373,7 +373,7 @@
 	for file in $(omffile); do \
 		$(INSTALL_DATA) $$file.out $(DESTDIR)$(omf_dest_dir)/$$file; \
 	done
-	-scrollkeeper-update -p $(scrollkeeper_localstate_dir) -o $(DESTDIR)$(omf_dest_dir)
+	#-scrollkeeper-update -p $(scrollkeeper_localstate_dir) -o $(DESTDIR)$(omf_dest_dir)
 
 uninstall-local-omf:
 	-for file in $(srcdir)/*.omf; do \
@@ -381,7 +381,7 @@
 		rm -f $(omf_dest_dir)/$$basefile; \
 	done
 	-rmdir $(omf_dest_dir)
-	-scrollkeeper-update -p $(scrollkeeper_localstate_dir)
+	#-scrollkeeper-update -p $(scrollkeeper_localstate_dir)
 
 clean-local-omf:
 	-for file in $(omffile); do \
diff -Naur anjuta-1.2.2/manuals/ja/anjuta-faqs/Makefile.in anjuta-1.2.2.fink/manuals/ja/anjuta-faqs/Makefile.in
--- anjuta-1.2.2/manuals/ja/anjuta-faqs/Makefile.in	Wed Apr 14 14:07:44 2004
+++ anjuta-1.2.2.fink/manuals/ja/anjuta-faqs/Makefile.in	Wed Oct 27 05:50:38 2004
@@ -373,7 +373,7 @@
 	for file in $(omffile); do \
 		$(INSTALL_DATA) $$file.out $(DESTDIR)$(omf_dest_dir)/$$file; \
 	done
-	-scrollkeeper-update -p $(scrollkeeper_localstate_dir) -o $(DESTDIR)$(omf_dest_dir)
+	#-scrollkeeper-update -p $(scrollkeeper_localstate_dir) -o $(DESTDIR)$(omf_dest_dir)
 
 uninstall-local-omf:
 	-for file in $(srcdir)/*.omf; do \
@@ -381,7 +381,7 @@
 		rm -f $(omf_dest_dir)/$$basefile; \
 	done
 	-rmdir $(omf_dest_dir)
-	-scrollkeeper-update -p $(scrollkeeper_localstate_dir)
+	#-scrollkeeper-update -p $(scrollkeeper_localstate_dir)
 
 clean-local-omf:
 	-for file in $(omffile); do \
diff -Naur anjuta-1.2.2/manuals/ja/anjuta-manual/Makefile.in anjuta-1.2.2.fink/manuals/ja/anjuta-manual/Makefile.in
--- anjuta-1.2.2/manuals/ja/anjuta-manual/Makefile.in	Wed Apr 14 14:07:43 2004
+++ anjuta-1.2.2.fink/manuals/ja/anjuta-manual/Makefile.in	Wed Oct 27 05:50:44 2004
@@ -380,7 +380,7 @@
 	for file in $(omffile); do \
 		$(INSTALL_DATA) $$file.out $(DESTDIR)$(omf_dest_dir)/$$file; \
 	done
-	-scrollkeeper-update -p $(scrollkeeper_localstate_dir) -o $(DESTDIR)$(omf_dest_dir)
+	#-scrollkeeper-update -p $(scrollkeeper_localstate_dir) -o $(DESTDIR)$(omf_dest_dir)
 
 uninstall-local-omf:
 	-for file in $(srcdir)/*.omf; do \
@@ -388,7 +388,7 @@
 		rm -f $(omf_dest_dir)/$$basefile; \
 	done
 	-rmdir $(omf_dest_dir)
-	-scrollkeeper-update -p $(scrollkeeper_localstate_dir)
+	#-scrollkeeper-update -p $(scrollkeeper_localstate_dir)
 
 clean-local-omf:
 	-for file in $(omffile); do \
diff -Naur anjuta-1.2.2/manuals/omf.make anjuta-1.2.2.fink/manuals/omf.make
--- anjuta-1.2.2/manuals/omf.make	Thu Nov  6 21:12:04 2003
+++ anjuta-1.2.2.fink/manuals/omf.make	Wed Oct 27 05:50:51 2004
@@ -45,7 +45,7 @@
 	for file in $(omffile); do \
 		$(INSTALL_DATA) $$file.out $(DESTDIR)$(omf_dest_dir)/$$file; \
 	done
-	-scrollkeeper-update -p $(scrollkeeper_localstate_dir) -o $(DESTDIR)$(omf_dest_dir)
+	#-scrollkeeper-update -p $(scrollkeeper_localstate_dir) -o $(DESTDIR)$(omf_dest_dir)
 
 uninstall-local-omf:
 	-for file in $(srcdir)/*.omf; do \
@@ -53,7 +53,7 @@
 		rm -f $(omf_dest_dir)/$$basefile; \
 	done
 	-rmdir $(omf_dest_dir)
-	-scrollkeeper-update -p $(scrollkeeper_localstate_dir)
+	#-scrollkeeper-update -p $(scrollkeeper_localstate_dir)
 
 clean-local-omf:
 	-for file in $(omffile); do \
diff -Naur anjuta-1.2.2/src/Makefile.in anjuta-1.2.2.fink/src/Makefile.in
--- anjuta-1.2.2/src/Makefile.in	Wed Apr 14 14:07:37 2004
+++ anjuta-1.2.2.fink/src/Makefile.in	Wed Oct 27 05:08:29 2004
@@ -467,7 +467,7 @@
 
 
 anjuta_LDADD = \
-	$(ANJUTA_LIBS) -lutil\
+	$(ANJUTA_LIBS) -lSystem\
 	@LIBICONV@\
 	$(PCRE_LIBS)\
 	$(top_builddir)/scintilla/libscintilla.a\
diff -Naur anjuta-1.2.2/src/launcher.c anjuta-1.2.2.fink/src/launcher.c
--- anjuta-1.2.2/src/launcher.c	Thu Apr  8 13:15:31 2004
+++ anjuta-1.2.2.fink/src/launcher.c	Wed Oct 27 05:08:06 2004
@@ -23,10 +23,10 @@
 #include <unistd.h>
 #include <fcntl.h>
 #include <signal.h>
-#include <pty.h>
 #include <assert.h>
 #include <gnome.h>
 #include <termios.h>
+#include <vte/pty.h>
 
 #include <string.h>
 
@@ -36,6 +36,14 @@
 #include "utilities.h"
 #include "anjuta-marshalers.h"
 #include "anjuta.h"
+
+#ifndef __MAX_BAUD
+#  define __MAX_BAUD B115200
+#endif
+
+#ifndef O_SYNC
+#  define O_SYNC O_FSYNC
+#endif
 
 #define FILE_BUFFER_SIZE 1024
 /*
