diff -Naru mysql-5.0.67/mysys/default.c mysql-5.0.67.fink/mysys/default.c
--- mysql-5.0.67/mysys/default.c	2008-11-29 04:54:49.000000000 +0900
+++ mysql-5.0.67.fink/mysys/default.c	2008-11-29 05:15:35.000000000 +0900
@@ -1077,7 +1077,6 @@
 
 #else
 
-  errors += add_directory(alloc, "/etc/", dirs);
 
 #if defined(__EMX__) || defined(OS2)
   if ((env= getenv("ETC")))
diff -Naru mysql-5.0.67/scripts/mysql_upgrade_shell.sh mysql-5.0.67.fink/scripts/mysql_upgrade_shell.sh
--- mysql-5.0.67/scripts/mysql_upgrade_shell.sh	2008-11-29 04:54:53.000000000 +0900
+++ mysql-5.0.67.fink/scripts/mysql_upgrade_shell.sh	2008-11-29 14:50:37.000000000 +0900
@@ -150,18 +150,7 @@
 
 if test -z "$DATADIR"
 then
-  # Try where the binary installs put it
-  if test -d $MY_BASEDIR_VERSION/data/mysql
-  then
-    DATADIR=$MY_BASEDIR_VERSION/data
-  # Next try where the source installs put it
-  elif test -d $MY_BASEDIR_VERSION/var/mysql
-  then
-    DATADIR=$MY_BASEDIR_VERSION/var
-  # Or just give up and use our compiled-in default
-  else
     DATADIR=@localstatedir@
-  fi
 fi
 
 if test ! -x "$bindir/mysqlcheck"
diff -Naru mysql-5.0.67/scripts/mysqld_safe.sh mysql-5.0.67.fink/scripts/mysqld_safe.sh
--- mysql-5.0.67/scripts/mysqld_safe.sh	2008-11-29 04:54:53.000000000 +0900
+++ mysql-5.0.67.fink/scripts/mysqld_safe.sh	2008-11-29 05:11:31.000000000 +0900
@@ -133,22 +133,7 @@
 # Second, try to find the data directory
 #
 
-# Try where the binary installs put it
-if test -d $MY_BASEDIR_VERSION/data/mysql
-then
-  DATADIR=$MY_BASEDIR_VERSION/data
-  if test -z "$defaults" -a -r "$DATADIR/my.cnf"
-  then
-    defaults="--defaults-extra-file=$DATADIR/my.cnf"
-  fi
-# Next try where the source installs put it
-elif test -d $MY_BASEDIR_VERSION/var/mysql
-then
-  DATADIR=$MY_BASEDIR_VERSION/var
-# Or just give up and use our compiled-in default
-else
   DATADIR=@localstatedir@
-fi
 
 if test -z "$MYSQL_HOME"
 then 
