### scp2 patches to use correct system libxml
Index: scp2/source/base/file_base.scp
===================================================================
RCS file: /cvs/installation/scp2/source/base/file_base.scp,v
retrieving revision 1.3
diff -u -r1.3 file_base.scp
--- scp2/source/base/file_base.scp	30 Jun 2005 16:38:33 -0000	1.3
+++ scp2/source/base/file_base.scp	13 Aug 2005 14:40:02 -0000
@@ -75,12 +75,16 @@
 
 #endif
 
+#ifdef SOLAR_JAVA
+
 File gid_File_Help_Sdatabase_Zip
     Dir = gid_Dir_Help_Isolanguage;
     ARCHIVE_TXT_FILE_BODY_PATCH;
     EXTRA_ALL_LANG(sdatabase);
 End
 
+#endif
+
 File gid_File_Registry_Spool_Oo_Setup_Base_Xcu
     TXT_FILE_BODY;
     Styles = (PACKED);
Index: scp2/source/ure/makefile.mk
===================================================================
RCS file: /cvs/installation/scp2/source/ure/makefile.mk,v
retrieving revision 1.3
diff -u -r1.3 makefile.mk
--- scp2/source/ure/makefile.mk	18 Jul 2005 08:45:25 -0000	1.3
+++ scp2/source/ure/makefile.mk	13 Aug 2005 14:40:02 -0000
@@ -2,9 +2,9 @@
 #
 #   $RCSfile$
 #
-#   $Revision$
+#   $Revision$
 #
-#   last change: $Author$ $Date$
+#   last change: $Author$ $Date$
 #
 #   The Contents of this file are made available subject to the terms of
 #   either of the following licenses
@@ -72,4 +72,8 @@
 SCPDEFS+=-DUSE_SYSTEM_STL
 .ENDIF
 
+.IF "$(SYSTEM_LIBXML)" == "YES"
+SCPDEFS+=-DSYSTEM_LIBXML
+.ENDIF
+
 .INCLUDE: target.mk
Index: scp2/source/ure/ure.scp
===================================================================
RCS file: /cvs/installation/scp2/source/ure/ure.scp,v
retrieving revision 1.5
diff -u -r1.5 ure.scp
--- scp2/source/ure/ure.scp	20 Jul 2005 12:31:24 -0000	1.5
+++ scp2/source/ure/ure.scp	13 Aug 2005 14:40:02 -0000
@@ -2,9 +2,9 @@
  *
  *  $RCSfile$
  *
- *  $Revision$
+ *  $Revision$
  *
- *  last change: $Author$ $Date$
+ *  last change: $Author$ $Date$
  *
  *  The Contents of this file are made available subject to the terms of
  *  either of the following licenses
@@ -439,6 +439,7 @@
     RegistryID = gid_Starregistry_Services;
 End
 
+
 #if defined SOLAR_JAVA
 File gid_File_Dl_Javaloader
     TXT_FILE_BODY;
@@ -749,9 +750,12 @@
 
 // Private Miscellaneous Files:
 
+#if defined SOLAR_JAVA
 File gid_File_Misc_JavavendorsXml
     TXT_FILE_BODY;
     Dir = gid_Dir_Misc;
     Name = "javavendors.xml";
     Styles = (PACKED);
 End
+#endif
+


### ucb patches to use system neon installed in non-standard directory; IssueZilla #53335
Index: ucb/source/ucp/webdav/LinkSequence.cxx
===================================================================
RCS file: /cvs/ucb/ucb/source/ucp/webdav/LinkSequence.cxx,v
retrieving revision 1.5
diff -u -r1.5 LinkSequence.cxx
--- ucb/source/ucp/webdav/LinkSequence.cxx	27 Jan 2005 12:12:09 -0000	1.5
+++ ucb/source/ucp/webdav/LinkSequence.cxx	14 Aug 2005 02:51:49 -0000
@@ -62,8 +62,12 @@
 #include <string.h>
 
 #ifndef NE_XML_H
+#ifdef SYSTEM_NEON
+#include <ne_xml.h>
+#else
 #include <neon/ne_xml.h>
 #endif
+#endif
 
 #ifndef _LINKSEQUENCE_HXX_
 #include "LinkSequence.hxx"
Index: ucb/source/ucp/webdav/LockEntrySequence.cxx
===================================================================
RCS file: /cvs/ucb/ucb/source/ucp/webdav/LockEntrySequence.cxx,v
retrieving revision 1.5
diff -u -r1.5 LockEntrySequence.cxx
--- ucb/source/ucp/webdav/LockEntrySequence.cxx	27 Jan 2005 12:12:40 -0000	1.5
+++ ucb/source/ucp/webdav/LockEntrySequence.cxx	14 Aug 2005 02:51:49 -0000
@@ -62,8 +62,12 @@
 #include <string.h>
 
 #ifndef NE_XML_H
+#ifdef SYSTEM_NEON
+#include <ne_xml.h>
+#else
 #include <neon/ne_xml.h>
 #endif
+#endif
 
 #ifndef _LOCKENTRYSEQUENCE_HXX_
 #include "LockEntrySequence.hxx"
Index: ucb/source/ucp/webdav/LockSequence.cxx
===================================================================
RCS file: /cvs/ucb/ucb/source/ucp/webdav/LockSequence.cxx,v
retrieving revision 1.6
diff -u -r1.6 LockSequence.cxx
--- ucb/source/ucp/webdav/LockSequence.cxx	27 Jan 2005 12:13:09 -0000	1.6
+++ ucb/source/ucp/webdav/LockSequence.cxx	14 Aug 2005 02:51:49 -0000
@@ -62,8 +62,12 @@
 #include <string.h>
 
 #ifndef NE_XML_H
+#ifdef SYSTEM_NEON
+#include <ne_xml.h>
+#else
 #include <neon/ne_xml.h>
 #endif
+#endif
 
 #ifndef _LOCKSEQUENCE_HXX_
 #include "LockSequence.hxx"
Index: ucb/source/ucp/webdav/NeonSession.cxx
===================================================================
RCS file: /cvs/ucb/ucb/source/ucp/webdav/NeonSession.cxx,v
retrieving revision 1.34
diff -u -r1.34 NeonSession.cxx
--- ucb/source/ucp/webdav/NeonSession.cxx	27 Jan 2005 12:13:44 -0000	1.34
+++ ucb/source/ucp/webdav/NeonSession.cxx	14 Aug 2005 02:51:50 -0000
@@ -62,17 +62,33 @@
 #include <string.h>
 
 #ifndef NE_SOCKET_H
+#ifdef SYSTEM_NEON
+#include <ne_socket.h>
+#else
 #include <neon/ne_socket.h>
 #endif
+#endif
 #ifndef NE_AUTH_H
+#ifdef SYSTEM_NEON
+#include <ne_auth.h>
+#else
 #include <neon/ne_auth.h>
 #endif
+#endif
 #ifndef NE_REDIRECT_H
+#ifdef SYSTEM_NEON
+#include <ne_redirect.h>
+#else
 #include <neon/ne_redirect.h>
 #endif
+#endif
 #ifndef NE_LOCKS_H
+#ifdef SYSTEM_NEON
+#include <ne_locks.h>
+#else
 #include <neon/ne_locks.h>
 #endif
+#endif
 
 #ifndef _RTL_USTRBUF_HXX_
 #include <rtl/ustrbuf.hxx>
Index: ucb/source/ucp/webdav/NeonTypes.hxx
===================================================================
RCS file: /cvs/ucb/ucb/source/ucp/webdav/NeonTypes.hxx,v
retrieving revision 1.9
diff -u -r1.9 NeonTypes.hxx
--- ucb/source/ucp/webdav/NeonTypes.hxx	27 Jan 2005 12:13:58 -0000	1.9
+++ ucb/source/ucp/webdav/NeonTypes.hxx	14 Aug 2005 02:51:50 -0000
@@ -63,17 +63,33 @@
 #define _NEONTYPES_HXX_
 
 #ifndef NE_SESSION_H
+#ifdef SYSTEM_NEON
+#include <ne_session.h>
+#else
 #include <neon/ne_session.h> // for ne_session
 #endif
+#endif
 #ifndef NE_UTILS_H
+#ifdef SYSTEM_NEON
+#include <ne_utils.h>
+#else
 #include <neon/ne_utils.h> // for ne_status
 #endif
+#endif
 #ifndef NE_BASIC_H
+#ifdef SYSTEM_NEON
+#include <ne_basic.h>
+#else
 #include <neon/ne_basic.h> // for ne_server_capabilities
 #endif
+#endif
 #ifndef NE_PROPS_H
+#ifdef SYSTEM_NEON
+#include <ne_props.h>
+#else
 #include <neon/ne_props.h> // for ne_propname, ne_prop_result_set
 #endif
+#endif
 
 typedef ne_session                  HttpSession;
 typedef ne_status                   HttpStatus;
Index: ucb/source/ucp/webdav/NeonUri.hxx
===================================================================
RCS file: /cvs/ucb/ucb/source/ucp/webdav/NeonUri.hxx,v
retrieving revision 1.8
diff -u -r1.8 NeonUri.hxx
--- ucb/source/ucp/webdav/NeonUri.hxx	21 Aug 2002 07:34:53 -0000	1.8
+++ ucb/source/ucp/webdav/NeonUri.hxx	14 Aug 2005 02:51:50 -0000
@@ -62,8 +62,12 @@
 #define _NEONURI_HXX_
 
 #ifndef NE_URI_H
+#ifdef SYSTEM_NEON
+#include <ne_uri.h>
+#else
 #include <neon/ne_uri.h>
 #endif
+#endif
 
 #ifndef _RTL_USTRING_HXX_
 #include <rtl/ustring.hxx>
Index: ucb/source/ucp/webdav/UCBDeadPropertyValue.cxx
===================================================================
RCS file: /cvs/ucb/ucb/source/ucp/webdav/UCBDeadPropertyValue.cxx,v
retrieving revision 1.6
diff -u -r1.6 UCBDeadPropertyValue.cxx
--- ucb/source/ucp/webdav/UCBDeadPropertyValue.cxx	27 Jan 2005 12:14:13 -0000	1.6
+++ ucb/source/ucp/webdav/UCBDeadPropertyValue.cxx	14 Aug 2005 02:51:50 -0000
@@ -62,8 +62,12 @@
 #include <string.h>
 
 #ifndef NE_XML_H
+#ifdef SYSTEM_NEON
+#include <ne_xml.h>
+#else
 #include <neon/ne_xml.h>
 #endif
+#endif
 
 #ifndef _OSL_DIAGNOSE_H_
 #include <osl/diagnose.h>
Index: ucb/source/ucp/webdav/makefile.mk
===================================================================
RCS file: /cvs/ucb/ucb/source/ucp/webdav/makefile.mk,v
retrieving revision 1.23
diff -u -r1.23 makefile.mk
--- ucb/source/ucp/webdav/makefile.mk	27 Jan 2005 12:14:44 -0000	1.23
+++ ucb/source/ucp/webdav/makefile.mk	14 Aug 2005 02:51:50 -0000
@@ -85,6 +85,7 @@
 .INCLUDE: sv.mk
 
 .IF "$(SYSTEM_NEON)" == "YES"
+CFLAGS+= -DSYSTEM_NEON
 CFLAGS+= $(NEON_CFLAGS)
 .ELSE
 NEONINCDIR=external$/neon


### patch to use system jpeg
Index: solenv/inc/libs.mk
===================================================================
RCS file: /cvs/tools/solenv/inc/libs.mk,v
retrieving revision 1.92
diff -u -u -r1.92 libs.mk
--- solenv/inc/libs.mk	5 Aug 2005 15:04:49 -0000	1.92
+++ solenv/inc/libs.mk	15 Aug 2005 06:21:39 -0000
@@ -208,6 +208,8 @@
 #i34482# Blackdown jdk is in the libsearch patch and has a libjpeg
 .IF "$(OS)" == "FREEBSD"
 JPEG3RDLIB=/usr/local/lib/libjpeg.so
+.ELIF "$(OS)" == "MACOSX"
+JPEG3RDLIB=-ljpeg
 .ELSE
 JPEG3RDLIB=/usr/lib/libjpeg.so
 .ENDIF


# patch to use system python and system portaudio; IssueZilla #53280 & #53271
Index: config_office/configure.in
===================================================================
RCS file: /cvs/tools/config_office/configure.in,v
retrieving revision 1.129
diff -u -r1.129 configure.in
--- config_office/configure.in	5 Aug 2005 15:05:56 -0000	1.129
+++ config_office/configure.in	15 Aug 2005 08:14:48 -0000
@@ -2436,7 +2436,8 @@
    python_include=`$PYTHON -c "import distutils.sysconfig; print distutils.sysconfig.get_config_var('INCLUDEPY');"`
    python_version=`$PYTHON -c "import distutils.sysconfig; print distutils.sysconfig.get_config_var('VERSION');"`
    PYTHON_CFLAGS="-I$python_include"
-   PYTHON_LIBS="-lpython$python_version"
+   PYTHON_LIBS=`$PYTHON -c "import distutils.sysconfig; print distutils.sysconfig.get_config_var('LIBPL');"`
+   PYTHON_LIBS="-L$PYTHON_LIBS -lpython$python_version"
    
    dnl check if the headers really work:
    save_CPPFLAGS="$CPPFLAGS"
@@ -3047,8 +3048,13 @@
 	   AC_MSG_RESULT(external)
 	   AC_CHECK_HEADER(portaudio.h, [],
 		[AC_MSG_ERROR(portaudio header not found. install portaudio)], [])
-	   AC_CHECK_LIB(portaudio, Pa_Initialize, [],
-		[AC_MSG_ERROR(portaudio library not found or functional)], [])
+	   if test "$_os" = "Darwin"; then
+		AC_CHECK_LIB(portaudio, Pa_Initialize, [],
+		    [AC_MSG_ERROR(portaudio library not found or functional)], [-framework CoreAudio -framework AudioToolBox])
+	   else
+		AC_CHECK_LIB(portaudio, Pa_Initialize, [],
+		    [AC_MSG_ERROR(portaudio library not found or functional)], [])
+	   fi
 	   AC_MSG_CHECKING(portaudio version)
 	   AC_LANG(C)
 	   AC_TRY_LINK([#include <portaudio.h>], [


### patch to remove unwanted symlinks of python; IssueZilla #46314
Index: shellscripts_pyuno.txt
===================================================================
RCS file: /cvs/installation/instsetoo_native/inc_openoffice/unix/shellscripts_pyuno.txt,v
retrieving revision 1.5
diff -c -r1.5 shellscripts_pyuno.txt
*** instsetoo_native/inc_openoffice/unix/shellscripts_pyuno.txt	11 Feb 2005 16:51:58 -0000	1.5
--- shellscripts_pyuno.txt	17 Jul 2005 11:37:03 -0000
***************
*** 59,63 ****
--- 59,74 ----
    ;;
  esac
  
+ if [ -h $$PRODUCTINSTALLLOCATION/PRODUCTDIRECTORYNAME/program/python-core ]
+ then
+   rm -f $$PRODUCTINSTALLLOCATION/PRODUCTDIRECTORYNAME/program/python-core >/dev/null 2>&1
+ fi
+ 
+ if [ -h $$PRODUCTINSTALLLOCATION/PRODUCTDIRECTORYNAME/program/python ]
+ then
+   rm -f $$PRODUCTINSTALLLOCATION/PRODUCTDIRECTORYNAME/program/python >/dev/null 2>&1
+ fi
+ 
+ 
  exit 0
  END


### patches for enabling the use of a scanner on Mac OSX; IssueZilla #48285
--- extensions/source/scanner/sane.cxx	2003-04-17 17:14:24.000000000 +0200
+++ sane_new.cxx	2005-07-15 17:36:55.000000000 +0200
@@ -213,12 +213,22 @@
 
 void Sane::Init()
 {
+#ifdef MACOSX
+	pSaneLib = dlopen( "libsane.dylib", RTLD_LAZY );
+        if( ! pSaneLib )
+                pSaneLib = dlopen( "libsane.dylib.1", RTLD_LAZY );
+	// try reasonable places that might not be in the library search path
+	if( ! pSaneLib )
+		pSaneLib = dlopen( "/usr/local/lib/libsane.dylib", RTLD_LAZY );
+#else
 	pSaneLib = dlopen( "libsane.so", RTLD_LAZY );
-    if( ! pSaneLib )
-        pSaneLib = dlopen( "libsane.so.1", RTLD_LAZY );
+       if( ! pSaneLib )
+               pSaneLib = dlopen( "libsane.so.1", RTLD_LAZY );
 	// try reasonable places that might not be in the library search path
 	if( ! pSaneLib )
 		pSaneLib = dlopen( "/usr/local/lib/libsane.so", RTLD_LAZY );
+#endif
+
 	if( pSaneLib )
 	{
 		bSaneSymbolLoadFailed = FALSE;
@@ -268,7 +278,11 @@
 	}
 #if (OSL_DEBUG_LEVEL > 1) || defined DBG_UTIL
 	else
+#ifdef MACOSX
+		fprintf( stderr, "libsane.dylib could not be opened: %s\n",
+#else
 		fprintf( stderr, "libsane.so could not be opened: %s\n",
+#endif
 				 dlerror() );
 #endif
 }


### patches to fix incorrect UI language detection on Mac OS X; IssueZilla #51246
--- shell/source/backends/localebe/localebackend.cxx	Fri Oct 22 18:16:56 2004
+++ localebackend.cxx.new	Fri Jun 24 07:53:16 2005
@@ -76,9 +76,9 @@
 
 #include <stdio.h>
 
-#ifdef UNX
-#include <rtl/ustrbuf.hxx>
+#if defined(LINUX) || defined(SOLARIS) || defined(IRIX) || defined(NETBSD) || defined(FREEBSD)
 
+#include <rtl/ustrbuf.hxx>
 #include <locale.h>
 #include <string.h>
 
@@ -126,8 +126,105 @@
     return aLocaleBuffer.makeStringAndClear();
 }
 
-#endif // UNX
+#elif defined(MACOSX) 
 
+#include <rtl/ustrbuf.hxx>
+#include <locale.h>
+#include <string.h>
+
+#include <premac.h>
+#include <CoreServices/CoreServices.h>
+#include <CoreFoundation/CoreFoundation.h>
+#include <postmac.h>
+
+namespace /* private */
+{
+
+	void OUStringBufferAppendCFString(rtl::OUStringBuffer& buffer, const CFStringRef s)
+	{
+		CFIndex lstr = CFStringGetLength(s);	
+		for (CFIndex i = 0; i < lstr; i++)
+			buffer.append(CFStringGetCharacterAtIndex(s, i));
+	}
+
+	template <typename T>
+	class CFGuard
+	{
+	public:
+		explicit CFGuard(T* pT) : pT_(pT) {}
+		~CFGuard() { if (pT_) CFRelease(*pT_); }
+	private:
+		T* pT_;
+	};
+
+	typedef CFGuard<CFArrayRef> CFArrayGuard;
+	typedef CFGuard<CFStringRef> CFStringGuard;
+	typedef CFGuard<CFTypeRef> CFTypeRefGuard;
+
+	/* For more information on the Apple locale concept please refer to 
+	http://developer.apple.com/documentation/CoreFoundation/Conceptual/CFLocales/Articles/CFLocaleConcepts.html
+	According to this documentation a locale identifier has the format: language[_country][_variant]*
+	e.g. es_ES_PREEURO -> spain prior Euro support
+	Note: The calling code should be able to handle locales with only language information e.g. 'en' for certain
+	UI languages just the language code will be returned.
+	*/
+
+	CFStringRef ImplGetAppPreference(const char* pref)
+	{
+		CFStringRef csPref = CFStringCreateWithCString(NULL, pref, kCFStringEncodingASCII);
+		CFStringGuard csRefGuard(&csPref);
+		
+		CFTypeRef ref = CFPreferencesCopyAppValue(csPref, kCFPreferencesCurrentApplication);
+		CFTypeRefGuard refGuard(&ref);
+	
+		if (ref == NULL)
+			return NULL;
+	
+		CFStringRef sref = (CFGetTypeID(ref) == CFArrayGetTypeID()) ? (CFStringRef)CFArrayGetValueAtIndex((CFArrayRef)ref, 0) : (CFStringRef)ref;
+		
+		CFRetain(sref); // caller is responsible for releasing reference
+		return sref;
+	}
+
+	rtl::OUString ImplGetLocale(const char* pref)
+	{
+		CFStringRef sref = ImplGetAppPreference(pref);
+		CFStringGuard srefGuard(&sref);
+		
+		rtl::OUStringBuffer aLocaleBuffer;
+		aLocaleBuffer.appendAscii("en-US"); // initialize with fallback value
+		
+		if (sref != NULL)
+		{
+			// split the string into substrings; the first two (if there are two) substrings 
+			// are language and country
+			CFArrayRef subs = CFStringCreateArrayBySeparatingStrings(NULL, sref, CFSTR("_"));
+			CFArrayGuard subsGuard(&subs);
+				
+			if (subs != NULL)
+			{
+				aLocaleBuffer.setLength(0); // clear buffer which still contains fallback value
+				
+				CFStringRef lang = (CFStringRef)CFArrayGetValueAtIndex(subs, 0);
+				OUStringBufferAppendCFString(aLocaleBuffer, lang);
+				
+				// country also available? Assumption: if the array contains more than one
+				// value the second value is always the country!
+				if (CFArrayGetCount(subs) > 1) 
+				{	
+					aLocaleBuffer.appendAscii("-");
+					CFStringRef country = (CFStringRef)CFArrayGetValueAtIndex(subs, 1);
+					OUStringBufferAppendCFString(aLocaleBuffer, country);
+				}					
+			}
+		}
+		return aLocaleBuffer.makeStringAndClear();
+	}
+
+} // namespace /* private */
+
+#endif
+
 // -------------------------------------------------------------------------------
 
 #ifdef WNT
@@ -187,8 +284,10 @@
 
 rtl::OUString LocaleBackend::getLocale(void) 
 {
-#ifdef UNX
+#if defined(LINUX) || defined(SOLARIS) || defined(IRIX) || defined(NETBSD) || defined(FREEBSD)
     return ImplGetLocale(LC_CTYPE);
+#elif defined (MACOSX)
+	return ImplGetLocale("AppleLocale");
 #elif defined WNT
     return ImplGetLocale( GetUserDefaultLCID() );
 #endif
@@ -198,8 +297,10 @@
 
 rtl::OUString LocaleBackend::getUILocale(void) 
 {
-#ifdef UNX
+#if defined(LINUX) || defined(SOLARIS) || defined(IRIX) || defined(NETBSD) || defined(FREEBSD)
     return ImplGetLocale(LC_MESSAGES);
+#elif defined(MACOSX)
+	return ImplGetLocale("AppleLanguages");
 #elif defined WNT
     return ImplGetLocale( MAKELCID(GetUserDefaultUILanguage(), SORT_DEFAULT) );
 #endif
--- shell/source/backends/localebe/makefile.mk	Mon Sep 20 18:28:36 2004
+++ makefile.mk.new	Fri Jun 24 07:46:44 2005
@@ -92,10 +92,19 @@
 SHL1DEF=$(MISC)$/$(SHL1TARGET).def
 
 SHL1IMPLIB=i$(SHL1TARGET)
+
+.IF "$(OS)"=="MACOSX"
+SHL1STDLIBS= \
+        $(CPPUHELPERLIB) \
+        $(CPPULIB) \
+        $(SALLIB) \
+		-framework CoreServices	
+.ELSE
 SHL1STDLIBS=    \
         $(CPPUHELPERLIB) \
         $(CPPULIB) \
         $(SALLIB)
+.ENDIF
 
 SHL1VERSIONMAP=exports.map
 SHL1DEF=$(MISC)$/$(SHL1TARGET).def


### fix bad name for libnpsoplugin on Mac OSX; IssueZilla #51851
diff -ur extensions/source/nsplugin/source/so_env.cxx extensions_new/source/nsplugin/source/so_env.cxx
--- extensions/source/nsplugin/source/so_env.cxx	2005-07-15 13:44:22.000000000 +0200
+++ extensions_new/source/nsplugin/source/so_env.cxx	2005-07-15 13:39:13.000000000 +0200
@@ -88,6 +88,7 @@
 #include <errno.h>
 #include "so_env.hxx"
 #include "ns_debug.hxx"
+#include <sal/config.h>
 
 #define PLUGIN_NAME         "OpenOffice.org"
 
@@ -159,9 +160,9 @@
     // Filename of lnk file, eg. "soffice"
     char lnkFileName[NPP_PATH_MAX] = {0};
     char* pTempZero = NULL;
-    sprintf(lnkFileName, "%s/.mozilla/plugins/%s", getenv("HOME"), "libnpsoplugin.so");
+    sprintf(lnkFileName, "%s/.mozilla/plugins/libnpsoplugin%s", getenv("HOME"), SAL_DLLEXTENSION);
     if ((0 > readlink(lnkFileName, realFileName, NPP_PATH_MAX)) ||
-       (NULL == (pTempZero = strstr(realFileName, "/program/libnpsoplugin.so"))))
+       (NULL == (pTempZero = strstr(realFileName, "/program/libnpsoplugin" SAL_DLLEXTENSION))))
     {
         *realFileName = 0;
         return -1;
diff -ur svx/source/dialog/optinet2.cxx svx_new/source/dialog/optinet2.cxx
--- svx/source/dialog/optinet2.cxx	2005-07-15 13:48:58.000000000 +0200
+++ svx_new/source/dialog/optinet2.cxx	2005-07-15 09:22:24.000000000 +0200
@@ -271,6 +271,8 @@
 #define RET_ALL				((short)200)
 #define CFG_READONLY_DEFAULT    FALSE
 
+#include <sal/config.h>
+
 SV_IMPL_PTRARR( SfxFilterPtrArr, SfxFilterPtr )
 
 const char* SEARCHENGINE_INI	= "srcheng.ini";
@@ -1757,7 +1759,7 @@
     char lnkReferFilePath[NPP_PATH_MAX] = {0};
     char* pHome = getpwuid(getuid())->pw_dir;
 	strcat(lnkFilePath, pHome);
-	strcat(lnkFilePath, "/.mozilla/plugins/libnpsoplugin.so");
+	strcat(lnkFilePath, "/.mozilla/plugins/libnpsoplugin" SAL_DLLEXTENSION);
 
 	struct stat sBuf;
 	if (0 > lstat(lnkFilePath, &sBuf))
@@ -1775,7 +1777,7 @@
     ::rtl::OString tempString;
     tempString = getDllURL();
     strncpy(realFilePath, tempString.getStr(), NPP_PATH_MAX);
-	strcat(realFilePath, "/libnpsoplugin.so");
+	strcat(realFilePath, "/libnpsoplugin" SAL_DLLEXTENSION);
 
     if (0 != strcmp(lnkReferFilePath, realFilePath))
         return false;
@@ -1803,7 +1805,7 @@
     char lnkFilePath[NPP_PATH_MAX] = {0};
     char* pHome = getpwuid(getuid())->pw_dir;
 	strcat(lnkFilePath, pHome);
-    strcat(lnkFilePath, "/.mozilla/plugins/libnpsoplugin.so");
+    strcat(lnkFilePath, "/.mozilla/plugins/libnpsoplugin" SAL_DLLEXTENSION);
     remove(lnkFilePath);
 
     // create the dirs if necessary
@@ -1822,7 +1824,7 @@
     ::rtl::OString tempString;
     tempString = getDllURL();
     strncpy(realFilePath, tempString.getStr(), NPP_PATH_MAX);
-	strcat(realFilePath, "/libnpsoplugin.so");
+	strcat(realFilePath, "/libnpsoplugin" SAL_DLLEXTENSION);
 
     // create the link
     if (0 != symlink(realFilePath, lnkFilePath))
@@ -1848,7 +1850,7 @@
     char lnkFilePath[NPP_PATH_MAX] = {0};
     char* pHome = getpwuid(getuid())->pw_dir;
 	strcat(lnkFilePath, pHome);
-	strcat(lnkFilePath, "/.mozilla/plugins/libnpsoplugin.so");
+	strcat(lnkFilePath, "/.mozilla/plugins/libnpsoplugin" SAL_DLLEXTENSION);
 
 	if(0 > remove(lnkFilePath))
 	    return false;


### patch to enable gtkplugin on Mac OS X; IssueZilla #52381
--- scp2/source/ooo/file_library_ooo.scp	2005-07-14 12:37:01.000000000 +0200
+++ file_library_ooo_new.scp	2005-07-20 17:27:27.000000000 +0200
@@ -666,7 +666,6 @@
 #ifdef UNX
 #ifdef ENABLE_GTK
 #ifdef GTK_TWO_FOUR
-#ifndef MACOSX
 File gid_File_Lib_Fps_Gnome
     TXT_FILE_BODY;
     Styles = (PACKED,PATCH);
@@ -678,7 +677,6 @@
 #endif
 #endif
 #endif
-#endif
 
 SPECIAL_UNO_COMPONENT_LIB_FILE_PATCH( gid_File_Lib_Fps_Office, fps_office.uno )
 SPECIAL_UNO_COMPONENT_LIB_FILE( gid_File_Lib_Fpicker, fpicker.uno )
@@ -1408,7 +1406,6 @@
 
 #ifdef UNX
 #ifdef ENABLE_GTK
-#ifndef MACOSX
 File gid_File_Lib_Vclplug_Gtk
     Name = LIBNAME(vclplug_gtk);
     TXT_FILE_BODY;
@@ -1416,7 +1413,6 @@
     Dir = gid_Dir_Program;
 End
 #endif
-#endif
 #ifdef ENABLE_KDE
 #ifndef MACOSX
 File gid_File_Lib_Vclplug_Kde


### fix for the break in package with gcc-3.3 on Mac OSX ZipPackageFolder.hxx; IssueZilla #51940
diff -ur package/inc/ZipPackageFolder.hxx package_new/inc/ZipPackageFolder.hxx
--- package/inc/ZipPackageFolder.hxx	2005-07-12 14:28:10.000000000 +0200
+++ package_new/inc/ZipPackageFolder.hxx	2005-07-14 17:30:52.000000000 +0200
@@ -121,7 +121,7 @@
 	}
 
 	void setPackageFormat_Impl( sal_Bool bPackageFormat ) { m_bPackageFormat = bPackageFormat; }
-	void setRemoveOnInsertMode_Impl( sal_Bool bRemove ) { ZipPackageEntry::mbAllowRemoveOnInsert = bRemove; }
+	void setRemoveOnInsertMode_Impl( sal_Bool bRemove ) { this->mbAllowRemoveOnInsert = bRemove; }
 
 	// Recursive functions
 	void  saveContents(rtl::OUString &rPath, std::vector < com::sun::star::uno::Sequence < com::sun::star::beans::PropertyValue > > &rManList, ZipOutputStream & rZipOut, com::sun::star::uno::Sequence < sal_Int8 > &rEncryptionKey, rtlRandomPool & rRandomPool)
diff -ur package/source/zippackage/ZipPackageFolder.cxx package_new/source/zippackage/ZipPackageFolder.cxx
--- package/source/zippackage/ZipPackageFolder.cxx	2005-07-12 14:31:33.000000000 +0200
+++ package_new/source/zippackage/ZipPackageFolder.cxx	2005-07-14 17:31:14.000000000 +0200
@@ -128,7 +128,7 @@
 {
 	OSL_ENSURE( m_xFactory.is(), "No factory is provided to the package folder!" );
 
-	ZipPackageEntry::mbAllowRemoveOnInsert = bAllowRemoveOnInsert;
+	this->mbAllowRemoveOnInsert = bAllowRemoveOnInsert;
 
 	SetFolder ( sal_True );
 	aEntry.nVersion		= -1;
diff -ur package/source/zippackage/ZipPackageStream.cxx package_new/source/zippackage/ZipPackageStream.cxx
--- package/source/zippackage/ZipPackageStream.cxx	2005-07-12 14:31:48.000000000 +0200
+++ package_new/source/zippackage/ZipPackageStream.cxx	2005-07-14 17:32:10.000000000 +0200
@@ -137,7 +137,7 @@
 { 
 	OSL_ENSURE( m_xFactory.is(), "No factory is provided to ZipPackageStream!\n" );
 
-	ZipPackageEntry::mbAllowRemoveOnInsert = bAllowRemoveOnInsert;
+	this->mbAllowRemoveOnInsert = bAllowRemoveOnInsert;
 
 	SetFolder ( sal_False );
 	aEntry.nVersion		= -1;


### patches to detect locale correctly; IsuueZilla #25416
# http://www.openoffice.org/nonav/issues/showattachment.cgi/27507/sal_systools_macxp_extras_x11osx_osxlocale.c.patch
# http://www.openoffice.org/nonav/issues/showattachment.cgi/27508/sal_osl_unx_file.cxx.patch
# build failed in udkapi
