### A patch to utilize Osaka font; IssueZilla #54535
--- officecfg/registry/data/org/openoffice/VCL.xcu	2005-09-08 23:46:30.000000000 +0900
+++ officecfg/registry/data/org/openoffice/VCL.xcu.fink	2005-10-17 03:08:38.000000000 +0900
@@ -434,31 +434,31 @@
     </node>
     <node oor:name="ja" oor:op="replace">
       <prop oor:name="CJK_TEXT" oor:type="xs:string" oor:op="replace">
-        <value>HG 明朝L Sun;HG MinchoL Sun;HG PゴシックB Sun;HG-PGothicB Sun;HG PゴシックB;HG-PGothicB;HG Mincho Light J;MS P 明朝;TLP明朝;LX明朝;IPA P明朝;東風明朝;Kochi Mincho;さざなみ明朝;Mincho;Serif</value>
+        <value>HG 明朝L Sun;HG MinchoL Sun;HG PゴシックB Sun;HG-PGothicB Sun;HG PゴシックB;HG-PGothicB;HG Mincho Light J;MS P 明朝;TLP明朝;LX明朝;IPA P明朝;Osaka;東風明朝;Kochi Mincho;さざなみ明朝;Mincho;Serif</value>
       </prop>
        <prop oor:name="CJK_HEADING" oor:type="xs:string" oor:op="replace">
-        <value>HG 明朝L Sun;HG 明朝L;HG Mincho Light J;MS P ゴシック;TLPゴシック;LXゴシック;IPA Pゴシック;東風ゴシック;さざなみゴシック;Gothic;MS 明朝;HG Mincho J;HG Mincho L;HG Mincho;Mincho;MS P 明朝;HG Mincho Light J;MS ゴシック;HG Gothic J;HG Gothic B;HG Gothic;Gothic;MS P ゴシック;Andale Sans UI</value>
+        <value>HG 明朝L Sun;HG 明朝L;HG Mincho Light J;MS P ゴシック;TLPゴシック;LXゴシック;IPA Pゴシック;Osaka;東風ゴシック;さざなみゴシック;Gothic;MS 明朝;HG Mincho J;HG Mincho L;HG Mincho;Mincho;MS P 明朝;HG Mincho Light J;MS ゴシック;HG Gothic J;HG Gothic B;HG Gothic;Gothic;MS P ゴシック;Andale Sans UI</value>
       </prop>
       <prop oor:name="CJK_PRESENTATION" oor:type="xs:string" oor:op="replace">
-        <value>HG PゴシックB Sun;HG-PGothicB Sun;HG PゴシックB;HG-PGothicB;HG-GothicB;HG Mincho Light J;MS P ゴシック;TLPゴシック;LXゴシック;IPA Pゴシック;東風ゴシック;さざなみゴシック;MS ゴシック;MS P ゴシック;HG Gothic;HG Gothic B;Gothic;Andale Sans UI</value>
+        <value>HG PゴシックB Sun;HG-PGothicB Sun;HG PゴシックB;HG-PGothicB;HG-GothicB;HG Mincho Light J;MS P ゴシック;TLPゴシック;LXゴシック;IPA Pゴシック;Osaka;東風ゴシック;さざなみゴシック;MS ゴシック;MS P ゴシック;HG Gothic;HG Gothic B;Gothic;Andale Sans UI</value>
       </prop>
       <prop oor:name="LATIN_PRESENTATION" oor:type="xs:string" oor:op="replace">
         <value>Andale;Arial</value>
       </prop>
       <prop oor:name="CJK_SPREADSHEET" oor:type="xs:string" oor:op="replace">
-        <value>HG PゴシックB Sun;HG-PGothicB Sun;HG PゴシックB;HG-PGothicB;MS P ゴシック;TLPゴシック;LXゴシック;IPA Pゴシック;東風ゴシック;さざなみゴシック;Andale Sans UI;Kochi Gothic;HG Gothic J;HG Gothic B;HG Gothic;Gothic</value>
+        <value>HG PゴシックB Sun;HG-PGothicB Sun;HG PゴシックB;HG-PGothicB;MS P ゴシック;TLPゴシック;LXゴシック;IPA Pゴシック;Osaka;東風ゴシック;さざなみゴシック;Andale Sans UI;Kochi Gothic;HG Gothic J;HG Gothic B;HG Gothic;Gothic</value>
       </prop>
       <prop oor:name="LATIN_SPREADSHEET" oor:type="xs:string" oor:op="replace">
-        <value>HG PゴシックB Sun;HG-PGothicB Sun;HG PゴシックB;HG-PGothicB;Andale Sans UI;MS P ゴシック;TLPゴシック;LXゴシック;IPA Pゴシック;東風ゴシック;MS Gothic;HG Gothic J;HG Gothic B;HG Gothic;Kochi Gothic;MS PGothic;Gothic</value>
+        <value>HG PゴシックB Sun;HG-PGothicB Sun;HG PゴシックB;HG-PGothicB;Andale Sans UI;MS P ゴシック;TLPゴシック;LXゴシック;IPA Pゴシック;Osaka;東風ゴシック;MS Gothic;HG Gothic J;HG Gothic B;HG Gothic;Kochi Gothic;MS PGothic;Gothic</value>
       </prop>
       <prop oor:name="UI_FIXED" oor:type="xs:string" oor:op="replace">
-        <value>HG ゴシックB Sun;HG-GothicB Sun;HG Mincho Light J;MS P ゴシック;TLPゴシック;LXゴシック;IPA Pゴシック;東風ゴシック;さざなみゴシック;Kochi Gothic;Gothic</value>
+        <value>HG ゴシックB Sun;HG-GothicB Sun;HG Mincho Light J;MS P ゴシック;TLPゴシック;LXゴシック;IPA Pゴシック;Osaka;東風ゴシック;さざなみゴシック;Kochi Gothic;Gothic</value>
        </prop>
        <prop oor:name="FIXED" oor:type="xs:string" oor:op="replace">
-        <value>HG ゴシックB Sun;HG-GothicB Sun;HG ゴシックB;HG Mincho Light J;MS P ゴシック;TLPゴシック;LXゴシック;IPA Pゴシック;東風ゴシック;さざなみゴシック;Kochi Gothic;Gothic</value>
+        <value>HG ゴシックB Sun;HG-GothicB Sun;HG ゴシックB;HG Mincho Light J;MS P ゴシック;TLPゴシック;LXゴシック;IPA Pゴシック;Osaka;東風ゴシック;さざなみゴシック;Kochi Gothic;Gothic</value>
        </prop>
       <prop oor:name="UI_SANS" oor:type="xs:string" oor:op="replace">
-        <value>HG PゴシックB Sun;HG-PGothicB Sun;HG PゴシックB;HG-PGothicB;HG-GothicB;Andale Sans UI;HG Mincho Light J;標準;TLPゴシック;LXゴシック;IPA Pゴシック;東風ゴシック;さざなみゴシック;Kochi Gothic;Gothic;Gnu-Unifont</value>
+        <value>HG PゴシックB Sun;HG-PGothicB Sun;HG PゴシックB;HG-PGothicB;HG-GothicB;Andale Sans UI;HG Mincho Light J;標準;TLPゴシック;LXゴシック;IPA Pゴシック;Osaka;東風ゴシック;さざなみゴシック;Kochi Gothic;Gothic;Gnu-Unifont</value>
       </prop>
       <prop oor:name="LATIN_FIXED" oor:type="xs:string" oor:op="replace">
         <value>hgmincholightj;cumberland;couriernew;courier;lucidasanstypewriter;lucidatypewriter;monaco;monospaced</value>
# Script to create language packs
--- /dev/null	2006-01-22 00:03:20.000000000 +0900
+++ languagepack-splitoff	2006-01-22 00:12:29.000000000 +0900
@@ -0,0 +1,37 @@
+#!/bin/bash
+
+set -e
+
+lang1=$1
+lang2=$2
+fink_p=$3
+fink_d=$4
+fink_i=$5
+# Setting up environment
+. ./MacosxEnv.Set.sh
+
+# Variables needed to execute make_installer.pl
+export OUT="../$INPATH"
+export LOCAL_OUT="$OUT"
+export LOCAL_COMMON_OUT="$OUT"
+solarlibdir=$SOLARVER/$UPD/$INPATH/lib
+export PYTHONPATH=$SRC_ROOT/instsetoo_native/$INPATH/bin:$solarlibdir:$solarlibdir/python:$solarlibdir/python/lib-dynload
+build=`sed -n 's/^BUILD=\(.*\)$/\1/p' $SOLARENVINC/minor.mk`
+
+set -v
+
+# Install language pack
+pushd instsetoo_native/util
+/usr/bin/perl -w $SOLARENV/bin/make_installer.pl \
+  -f openoffice.lst -l $lang2 -p OpenOffice \
+  -packagelist ../inc_openoffice/unix/packagelist.txt \
+  -buildid $build -languagepack \
+  -destdir $fink_d \
+  -simple $fink_p/lib/openoffice.org
+popd
+
+# Remove the list files
+/bin/rm -v $fink_d/gid_Module*
+
+/usr/bin/install -m 755 -d $fink_i/share/doc/openoffice.org-$lang1
+/bin/ln -sf $fink_p/lib/openoffice.org/licenses/LICENSE_$lang2 $fink_i/share/doc/openoffice.org-$lang1/LICENSE
# No-op if --with-system-python
--- python/makefile.mk	2005-12-29 01:57:47.000000000 +0900
+++ python/makefile.mk.fink	2006-01-15 20:59:52.000000000 +0900
@@ -105,7 +105,7 @@
 .ENDIF			# "$(SYSBASE)"!=""
 
 CONFIGURE_ACTION=./configure --prefix=$(MYCWD)/python-inst --enable-shared CFLAGS="$(python_CFLAGS)" LDFLAGS="$(python_LDFLAGS)"
-.IF "$(OS)$(CPU)" == "SOLARISI"
+.IF "$(OS)$(CPU)" == "SOLARISI" || "$(OS)$(BUILD_OS_MAJOR)$(BUILD_OS_MINOR)" == "MACOSX103"
 CONFIGURE_ACTION += --disable-ipv6
 .ENDIF
 .IF "$(COMNAME)"=="sunpro5"
# 10.4 (Xcode 2.x) and 3.3; IssueZilla #49044
--- stlport/STLport-4.5-macxp-panther.patch	2005-10-20 15:24:35.000000000 +0200
+++ stlport/STLport-4.5-macxp-panther.patch.fink	2005-11-29 22:05:56.000000000 +0100
@@ -1,3 +1,17 @@
+--- misc/STLport-4.5/stlport/stdarg.h	2001-01-27 02:39:59.000000000 +0100
++++ misc/build/STLport-4.5/stlport/stdarg.h	2005-05-01 19:37:22.000000000 +0200
+@@ -20,7 +20,11 @@
+ #  define _STLP_DONT_POP_0x261
+ # endif
+ 
++#if defined(__GNUC__) && defined(__APPLE__) && (BUILD_OS_MAJOR >= 10) && (BUILD_OS_MINOR >= 4) && (__GNUC__ == 3)
++#  include _STLP_NATIVE_HEADER(../stdarg.h)
++#else
+ # include _STLP_NATIVE_C_HEADER(stdarg.h)
++#endif
+ 
+ # if (_STLP_OUTERMOST_HEADER_ID == 0x261)
+ #  if ! defined (_STLP_DONT_POP_0x261)
 --- misc/STLport-4.5/stlport/stl/c_locale.h	2001-05-31 05:24:41.000000000 +0200
 +++ misc/build/STLport-4.5/stlport/stl/c_locale.h	2005-05-01 23:19:20.000000000 +0200
 @@ -290,16 +290,28 @@
# IssueZilla #53414, #60580
--- config_office/configure.in.orig	2006-01-10 23:33:04.000000000 +0900
+++ config_office/configure.in	2006-01-18 23:38:56.000000000 +0900
@@ -229,6 +229,16 @@
 AC_ARG_WITH(system-python,
 [  --with-system-python    Use python already on system
 ],,)
+AC_ARG_WITH(python-libs,
+[[  --with-python-libs[=auto|LIBS|none]
+                          Specify linker options to link the python library. [default=auto]
+                          This option does nothing without --with-system-python.
+]],[
+case $withval in
+   yes) with_python_libs=auto;;
+   no|"") with_python_libs=none;;
+esac
+],with_python_libs=auto)
 #AC_ARG_WITH(system-icu,
 #[  --with-system-icu       Use icu already on system
 #],,)
@@ -2098,7 +2108,7 @@
    AC_PROG_CPP
 fi
 dnl Check whether there's a C++ pre-processor.
-if test "$_os" = "Linux" -o "$_os" = "FreeBSD" -o "$_os" = "NetBSD" ; then
+if test "$_os" = "Linux" -o "$_os" = "FreeBSD" -o "$_os" = "NetBSD" -o "$_os" = "Darwin"; then
    AC_PROG_CXXCPP
 fi
 
@@ -2424,6 +2434,9 @@
 dnl ===================================================================
 dnl Check for system zlib
 dnl ===================================================================
+if test "$_os" = "Darwin" -o "$_os" = "Linux" && test "$with_system_zlib" != "no"; then
+   with_system_zlib=yes
+fi
 AC_MSG_CHECKING([which zlib to use])
 if test -n "$with_system_zlib" -o -n "$with_system_libs" && \
 	test "$with_system_zlib" != "no"; then
@@ -2536,11 +2549,11 @@
 dnl ===================================================================
 dnl Check for system python
 dnl ===================================================================
-if test "$_os" = "Darwin" && test -z "$with_system_python"; then
+if test "$_os" = "Darwin" && test "$with_system_python" != "no"; then
    with_system_python=yes
 fi
 AC_MSG_CHECKING([which python to use])
-if test -n "$with_system_python"; then
+if test -n "$with_system_python" && test "$with_system_python" != "no"; then
    SYSTEM_PYTHON=YES
    AC_MSG_RESULT([external])
    AM_PATH_PYTHON([2.2])
@@ -2548,7 +2561,16 @@
    python_include=`$PYTHON -c "import distutils.sysconfig; print distutils.sysconfig.get_config_var('INCLUDEPY');"`
    python_version=`$PYTHON -c "import distutils.sysconfig; print distutils.sysconfig.get_config_var('VERSION');"`
    PYTHON_CFLAGS="-I$python_include"
-   PYTHON_LIBS="-lpython$python_version"
+   case $with_python_libs in
+      auto)
+      case $_os:$python_include in
+         Darwin:/System/*) PYTHON_LIBS="-framework Python" ;;
+         *) PYTHON_LIBS="-lpython$python_version" ;;
+      esac
+      ;;
+      none) PYTHON_LIBS= ;;
+      *) PYTHON_LIBS=$with_python_libs ;;
+   esac
    
    dnl check if the headers really work:
    save_CPPFLAGS="$CPPFLAGS"
@@ -2706,10 +2728,13 @@
 if test -n "$with_system_boost" && test "$with_system_boost" != "no"; then
    AC_MSG_RESULT([external])
    SYSTEM_BOOST=YES
+   AC_LANG_SAVE
+   AC_LANG_CPLUSPLUS
    AC_CHECK_HEADER(boost/shared_ptr.hpp, [],
        [AC_MSG_ERROR(boost/shared_ptr.hpp not found. install boost)], [])
    AC_CHECK_HEADER(boost/spirit/core.hpp, [],
        [AC_MSG_ERROR(boost/spirit/core.hpp not found. install boost)], [])
+   AC_LANG_RESTORE
 else
    AC_MSG_RESULT([internal])
    BUILD_TYPE="$BUILD_TYPE BOOST"
# Workaround gcc 4.0.0's bug (2.0 <= Xcode < 2.2)
--- rsc/source/parser/rsclex.cxx.orig	2005-09-08 22:47:43.000000000 +0900
+++ rsc/source/parser/rsclex.cxx	2006-01-16 04:09:20.000000000 +0900
@@ -86,7 +86,7 @@
 RscFileInst*	pFI;
 RscTypCont* 	pTC;
 RscExpression * pExp;
-struct {
+struct RscKeyVal { // workaround gcc 4.0.0's bug (gcc bugzilla #15049)
 	int 	nKeyWord;
 	YYSTYPE aYYSType;
 } aKeyVal[ 1 ];
--- framework/source/loadenv/loadenv.cxx.orig	2005-11-11 21:06:12.000000000 +0900
+++ framework/source/loadenv/loadenv.cxx	2006-01-17 22:17:03.000000000 +0900
@@ -617,7 +617,13 @@
             // <- BREAK SAFE ------------------------------
             xAsyncLoader->cancel();
             // <- RESTART SAFE ----------------------------
+#if defined(__GNUC__) && (__GNUC__ == 4) && (__GNUC_MINOR__ == 0) && (__GNUC_PATCHLEVEL__ < 1)
+            try {
+                aReadLock.lock();
+            } catch (...) {} // avoid an internal compiler error
+#else
             aReadLock.lock();
+#endif
             /* Attention:
                 After returning from any cancel/dispose call, neither the frame nor weself
                 may be called back. Because only we can cancel this job, we already know
--- framework/source/services/autorecovery.cxx.orig	2005-12-22 01:08:39.000000000 +0900
+++ framework/source/services/autorecovery.cxx	2006-01-20 04:54:11.000000000 +0900
@@ -825,7 +825,13 @@
         // <- SAFE ------------------------------
         aReadLock.unlock();
         xListener->statusChanged(aEvent);
+#if defined(__GNUC__) && (__GNUC__ == 4) && (__GNUC_MINOR__ == 0) && (__GNUC_PATCHLEVEL__ < 1)
+        try {
+            aReadLock.lock();
+        } catch (...) {} // avoid an internal compiler error
+#else
         aReadLock.lock();
+#endif
         // SAFE -> ------------------------------
     }
 
--- binfilter/bf_sw/source/core/txtnode/makefile.mk	9 Sep 2005 02:37:27 -0000	1.4
+++ binfilter/bf_sw/source/core/txtnode/makefile.mk	18 Jan 2006 17:53:49 -0000
@@ -63,10 +63,6 @@
 LIBFLAGS=/NOI /NOE /PAGE:512
 .ENDIF
 
-.IF "$(COMID)" == "gcc3"
-ENVCFLAGS=-fno-access-control
-.ENDIF
-
 # --- Files --------------------------------------------------------
 
 CXXFILES = \
--- binfilter/bf_sw/source/core/txtnode/sw_thints.cxx	9 Sep 2005 02:41:10 -0000	1.7
+++ binfilter/bf_sw/source/core/txtnode/sw_thints.cxx	18 Jan 2006 17:53:50 -0000
@@ -1853,11 +1853,7 @@
 /*N*/ 			( SFX_ITEM_SET == rNode.GetpSwAttrSet()->GetItemState( nWhich,
 /*N*/ 				FALSE, &pParaItem ) ) && ( pParaItem == &pHint->GetAttr() );
 /*N*/ 		BOOL bReplace = !( SETATTR_DONTREPLACE & nMode );
-#if defined(_MSC_VER) && (_MSC_VER >= 1310 )
-			::binfilter::SwpHtStart_SAR *pTmpHints = 0;
-#else
-/*N*/ 		SwpHtStart_SAR *pTmpHints = 0;
-#endif
+/*N*/ 		::binfilter::SwpHtStart_SAR *pTmpHints = 0;
 /*N*/
 /*N*/ 		USHORT i;
 /*N*/ 		// Wir wollen zwar von nHtStart bis nMaxEnd, muessen aber ggf.
@@ -2034,11 +2030,7 @@
 /*N*/ 									// Beim Einfuegen spaltet es sich selbst.
 /*?*/ 										if( pHistory ) pHistory->Add( pOther );
 /*?*/ 										if( !pTmpHints )
-#if defined(_MSC_VER) && (_MSC_VER >= 1310 )
-												pTmpHints = new ::binfilter::SwpHtStart_SAR();
-#else
-/*?*/ 											pTmpHints = new SwpHtStart_SAR();
-#endif
+/*?*/ 											pTmpHints = new ::binfilter::SwpHtStart_SAR();
 /*?*/ 										pTmpHints->C40_INSERT( SwTxtAttr, pOther,
 /*?*/ 											pTmpHints->Count() );
 /*?*/ 										Cut( i );
@@ -2087,11 +2079,7 @@
 /*?*/ 									// Beim Einfuegen spaltet es sich selbst.
 /*?*/ 										if( pHistory ) pHistory->Add( pOther );
 /*?*/ 										if( !pTmpHints )
-#if defined(_MSC_VER) && (_MSC_VER >= 1310 )
-												pTmpHints = new ::binfilter::SwpHtStart_SAR();
-#else
-/*?*/ 											pTmpHints = new SwpHtStart_SAR();
-#endif
+/*?*/ 											pTmpHints = new ::binfilter::SwpHtStart_SAR();
 /*?*/ 										pTmpHints->C40_INSERT( SwTxtAttr, pOther,
 /*?*/ 											pTmpHints->Count() );
 /*?*/ 										Cut( i );
--- binfilter/bf_xmloff/source/text/xmloff_XMLIndexBibliographySourceContext.cxx.orig	2005-09-09 19:21:35.000000000 +0900
+++ binfilter/bf_xmloff/source/text/xmloff_XMLIndexBibliographySourceContext.cxx	2006-01-19 02:07:39.000000000 +0900
@@ -144,12 +144,22 @@
 	if ( ( XML_NAMESPACE_TEXT == nPrefix ) &&
 		 ( IsXMLToken( rLocalName, XML_BIBLIOGRAPHY_ENTRY_TEMPLATE ) ) )
 	{
+#if defined(__GNUC__) && (__GNUC__ == 4) && (__GNUC_MINOR__ == 0) && (__GNUC_PATCHLEVEL__ < 1)
+                static const SvXMLEnumMapEntry *p = aLevelNameBibliographyMap;
+		return new XMLIndexTemplateContext(GetImport(), rIndexPropertySet, 
+										   nPrefix, rLocalName,
+										   p,
+										   XML_BIBLIOGRAPHY_TYPE,
+										   aLevelStylePropNameBibliographyMap,
+										   aAllowedTokenTypesBibliography);
+#else
 		return new XMLIndexTemplateContext(GetImport(), rIndexPropertySet, 
 										   nPrefix, rLocalName,
 										   aLevelNameBibliographyMap,
 										   XML_BIBLIOGRAPHY_TYPE,
 										   aLevelStylePropNameBibliographyMap,
 										   aAllowedTokenTypesBibliography);
+#endif
 	}
 	else 
 	{
--- sw/source/core/txtnode/makefile.mk	9 Sep 2005 05:12:57 -0000	1.7
+++ sw/source/core/txtnode/makefile.mk	19 Jan 2006 05:42:26 -0000
@@ -48,10 +48,6 @@
 CDEFS+=-Dmydebug
 .ENDIF
 
-.IF "$(COMID)" == "gcc3"
-ENVCFLAGS=-fno-access-control
-.ENDIF
-
 # --- Files --------------------------------------------------------
 
 SLOFILES =  \
# list of help images is misgenerated; IssueZilla #60490
--- helpcontent2/helpers/create_ilst.pl	17 Jan 2006 12:23:19 -0000	1.4
+++ helpcontent2/helpers/create_ilst.pl	21 Jan 2006 13:26:15 -0000
@@ -45,8 +45,7 @@
 ($params =~ /-dir/) ? ($startdir = $params) =~ (s/.*-dir=([^\|]*).*$/$1/gs) : (($startdir = `pwd`) =~ s/\n//gs);
 ($params =~ /-pre/) ? ($pre = $params) =~ (s/.*-pre=([^\|]*).*$/$1/gs) : ($pre = "helpimg");
 
-my $startdir_regexp = $startdir;
-$startdir_regexp =~ s/\\/\\\\/g;
+my $startdir_regexp = quotemeta($startdir);
 
 
 if ( -d $startdir ) {
# MacOS X <= 10.3 wants -bundle; IssueZilla #61137
--- pyuno/source/module/makefile.mk	16 Jan 2006 15:54:58 -0000	1.12
+++ pyuno/source/module/makefile.mk	24 Jan 2006 11:03:08 -0000
@@ -129,7 +129,7 @@
 .ELIF "$(OS)" == "NETBSD"
 	@+ echo $(LINK) $(LINKFLAGSSHLCUI) -o $@ $(SLO)$/pyuno_dlopenwrapper.o > $(MISC)$/$(@:b).cmd
 .ELIF "$(OS)" == "MACOSX"
-	@+ echo $(CC) -dynamiclib -ldl -o $@ $(SLO)$/pyuno_dlopenwrapper.o > $(MISC)$/$(@:b).cmd
+	@+ echo $(CC) -bundle -ldl -o $@ $(SLO)$/pyuno_dlopenwrapper.o > $(MISC)$/$(@:b).cmd
 .ELSE
 	@+ echo $(LINK) $(LINKFLAGSSHLCUI) -o $@ $(SLO)$/pyuno_dlopenwrapper.o > $(MISC)$/$(@:b).cmd
 .ENDIF
