--- PerlMagick/Makefile.PL.orig	Fri Feb  4 11:27:26 2005
+++ PerlMagick/Makefile.PL	Fri Feb  4 11:50:51 2005
@@ -44,7 +44,7 @@
    'DEFINE'	=> ' -D_FILE_OFFSET_BITS=64 -DHAVE_CONFIG_H',     # e.g., '-DHAVE_SOMETHING' 
 
    # Header search specfication and preprocessor flags
-   'INC'	=> '-I../ -I.. -I/usr/include/freetype2 -I/usr/X11R6/include -I/usr/X11R6/include/X11 -I/usr/include/libxml2',
+   'INC'	=> '-I../ -I.. -I/usr/X11R6/include/freetype2 -I/usr/X11R6/include -I/usr/X11R6/include/X11 -I@PREFIX@/include/libxml2 -I@PREFIX@/include',
 
    # C pre-processor flags (e.g. -I & -D options)
    # 'CPPFLAGS'    => "$Config{'cppflags'} -I/usr/include/freetype2 -I/usr/X11R6/include -I/usr/X11R6/include/X11 -I/usr/include/libxml2",
@@ -53,16 +53,16 @@
    'CCFLAGS'     => "$Config{'ccflags'} -O3 -g -Wall -pthread",
 
    # Linker flags for building an executable
-   'LDFLAGS'    =>  "-L/usr/local/lib $Config{'ldflags'}",
+   'LDFLAGS'    =>  "$Config{'ldflags'}",
 
    # Linker flags for building a dynamically loadable module
-   'LDDLFLAGS'  => "-L/usr/local/lib $Config{'lddlflags'}",
+   'LDDLFLAGS'  => "$Config{'lddlflags'}",
 
    # Install PerlMagick binary into ImageMagick bin directory
-   'INSTALLBIN'	=> '/usr/local/bin',
+   'INSTALLBIN'	=> '@PREFIX@/bin',
 
    # Library specification
-   'LIBS'	=> ['-L/usr/local/lib -lMagick -L/usr/X11R6/lib -lfreetype -lz -L/usr/lib -llcms -ltiff -lfreetype -ljpeg -lpng -ldpstk -ldps -lXext -lXt -lSM -lICE -lX11 -lbz2 -lxml2 -lz -lpthread -lm -lpthread'],
+   'LIBS'	=> ['-L@PREFIX@/lib -lMagick -L/usr/X11R6/lib -lfreetype -lz -L/usr/lib -llcms -ltiff -lfreetype -ljpeg -lpng -ldpstk -ldps -lXext -lXt -lSM -lICE -lX11 -lbz2 -lxml2 -lz -lpthread -lm -lpthread'],
 
    # Perl binary name (if a Perl binary is built)
    'MAP_TARGET'	=> 'PerlMagick',
@@ -96,7 +96,7 @@
      # tries to insert our library path first. Also, use the same compiler used
      # to build perlmain.c to link so that a C++ compiler may be used if
      # necessary.
-     $inherited =~ s:MAP_LINKCMD\s.*\s*\$\(CC\):MAP_LINKCMD   = \$(PERLMAINCC) -L/usr/local/lib: ;
+     $inherited =~ s:MAP_LINKCMD\s.*\s*\$\(CC\):MAP_LINKCMD   = \$(PERLMAINCC):;
      $inherited;
  }
 
