--- configure.orig	2006-12-30 08:56:26.000000000 +0900
+++ configure	2007-05-11 17:10:49.000000000 +0900
@@ -19836,7 +19836,10 @@
 # Test for the Windows version first because if if QTDIR is mounted via
 # smbfs to an actual Windows installation, the /include/Qt is found
 # case-insensitively
-if test -f $QTDIR/include/Qt/qglobal.h -a -f $QTDIR/src/corelib/global/qglobal.h; then
+if true; then
+  QTINC=$QTDIR/include
+  QTVERTEST=$QTDIR/include/Qt
+elif test -f $QTDIR/include/Qt/qglobal.h -a -f $QTDIR/src/corelib/global/qglobal.h; then
   # Windows: $QTDIR/include/Qt/qglobal.h includes $QTDIR/src/corelib/global/qglobal.h
   QTINC=$QTDIR/include
   QTVERTEST=$QTDIR/src/corelib/global/
@@ -19866,7 +19869,7 @@
     QT_MAJOR="${QT_VER:0:1}"
     QT_MINOR="${QT_VER:1:1}"
     case "${host}" in
-    *-darwin*)
+    *-disabled*)
       QT4_3SUPPORTINC=$QTDIR/lib/Qt3Support.framework/Headers
       QT4_COREINC=$QTDIR/lib/QtCore.framework/Headers
       QT4_GUIINC=$QTDIR/lib/QtGui.framework/Headers
@@ -20200,9 +20203,9 @@
     ;;
   *-darwin*)
     # determin static or dynamic -- prefer dynamic
-    QT_IS_DYNAMIC=`ls $QTDIR/lib/libqt*.dylib $QTDIR/lib/QtCore.framework/QtCore 2> /dev/null`
+    QT_IS_DYNAMIC=`ls $QTDIR/lib/libQt*.dylib 2> /dev/null`
     if test "x$QT_IS_DYNAMIC" = x;  then
-      QT_IS_STATIC=`ls $QTDIR/lib/libqt*.a 2> /dev/null`
+      QT_IS_STATIC=`ls $QTDIR/lib/libQt*.a 2> /dev/null`
       if test "x$QT_IS_STATIC" = x; then
         QT_IS_STATIC="no"
         { { echo "$as_me:$LINENO: error: *** Couldn't find any Qt libraries in $QTDIR/lib" >&5
@@ -20232,13 +20235,9 @@
       QT_LIB="-lqte-mt"
       QT_IS_MT="yes"
       QT_IS_EMBEDDED="yes"
-    elif test "x`ls $QTDIR/lib/QtCore.framework/QtCore 2> /dev/null`" != x ; then
-      QT_LIB="-Xlinker -F$QTDIR/lib -framework Qt3Support -framework QtCore -framework QtGui -framework QtNetwork -framework QtXml -framework QtSvg"
-      if test $QT_MINOR -lt "2"; then
-        QT_LIB="$QT_LIB -L$QTDIR/lib -lQtDesigner -lQtTest"
-      else
-        QT_LIB="$QT_LIB -framework QtDesigner -framework QtTest"
-      fi
+    elif test "x`ls $QTDIR/lib/libQtCore.*.dylib 2> /dev/null`" != x ; then
+      QT_LIB="-L$QTDIR/lib -lQt3Support -lQtCore -lQtGui -lQtNetwork -lQtXml -lQtSvg"
+      QT_LIB="$QT_LIB -lQtDesigner -lQtTest"
       QT_CXXFLAGS="-DQT3_SUPPORT -I$QT4_DEFAULTINC -I$QT4_3SUPPORTINC -I$QT4_COREINC -I$QT4_GUIINC -I$QT4_NETWORKINC -I$QT4_OPENGLINC -I$QT4_SQLINC -I$QT4_SVGINC -I$QT4_XMLINC -I$QT4_DESIGNERINC -I$QT4_TESTINC -I$QTDIR/include"
       QT_IS_MT="yes"
     fi
