diff -ruN giFT/FastTrack/Makefile giFT.new/FastTrack/Makefile
--- giFT/FastTrack/Makefile	2003-05-20 19:51:09.000000000 -0400
+++ giFT.new/FastTrack/Makefile	2003-05-20 20:51:48.000000000 -0400
@@ -30,7 +30,7 @@
 
 mac: $(FST_OBJS) $(CRYPT_OBJS)
 	@echo
-	$(CC) -dynamiclib -flat_namespace -undefined suppress -o FastTrack.so  $(CFLAGS) $(LDFLAGS) $(DEBUGFLAGS) $(FST_OBJS) $(CRYPT_OBJS)
+	$(CC) -dynamiclib -flat_namespace -undefined suppress -o FastTrack.dylib  $(CFLAGS) $(LDFLAGS) $(DEBUGFLAGS) $(FST_OBJS) $(CRYPT_OBJS)
 	@echo
 	@echo "*************************************************************"
 	@echo Your giFT settings path is $(GIFT_INSTALL_DIR) and the plugin will be installed into $(FST_INSTALL_DIR).
diff -ruN giFT/autogen.sh giFT.new/autogen.sh
--- giFT/autogen.sh	2002-10-29 19:36:21.000000000 -0500
+++ giFT.new/autogen.sh	2003-05-20 19:54:45.000000000 -0400
@@ -9,7 +9,7 @@
 : ${AUTOHEADER=autoheader}
 : ${AUTOMAKE=automake}
 : ${ACLOCAL=aclocal}
-: ${LIBTOOLIZE=libtoolize}
+: ${LIBTOOLIZE=glibtoolize}
 
 srcdir=`dirname $0`
 test -z "$srcdir" && srcdir=.
@@ -57,6 +57,6 @@
 ) || exit 1
 
 #conf_flags="--enable-maintainer-mode --enable-compile-warnings"
-echo Running $srcdir/configure $conf_flags "$@" ...
-$srcdir/configure --cache-file=config.cache $conf_flags "$@" && \
-  echo "Now type \`make' to compile giFT."
+#echo Running $srcdir/configure $conf_flags "$@" ...
+#$srcdir/configure --cache-file=config.cache $conf_flags "$@" && \
+#  echo "Now type \`make' to compile giFT."
diff -ruN giFT/src/Makefile.am giFT.new/src/Makefile.am
--- giFT/src/Makefile.am	2003-05-03 13:03:59.000000000 -0400
+++ giFT.new/src/Makefile.am	2003-05-20 19:54:45.000000000 -0400
@@ -60,6 +60,7 @@
 
 giFT_LDFLAGS = \
 	$(GIFT_LDFLAGS) \
+	-L$(top_builddir)/lib/.libs \
 	$(IMAGEMAGICK_LDFLAGS) \
 	-export-dynamic
 
