diff -ruN giFT/Makefile.am giFT.new/Makefile.am
--- giFT/Makefile.am	2003-05-20 19:51:25.000000000 -0400
+++ giFT.new/Makefile.am	2003-05-20 19:55:08.000000000 -0400
@@ -1,4 +1,4 @@
-SUBDIRS = lib plugin OpenFT Gnutella src data
+SUBDIRS = lib plugin OpenFT Gnutella src data FastTrack
 
 man_MANS = giFT.1 gift.conf.5 OpenFT.conf.5
 
diff -ruN giFT/autogen.sh giFT.new/autogen.sh
--- giFT/autogen.sh	2002-10-29 19:36:21.000000000 -0500
+++ giFT.new/autogen.sh	2003-05-20 19:54:45.000000000 -0400
@@ -9,7 +9,7 @@
 : ${AUTOHEADER=autoheader}
 : ${AUTOMAKE=automake}
 : ${ACLOCAL=aclocal}
-: ${LIBTOOLIZE=libtoolize}
+: ${LIBTOOLIZE=glibtoolize}
 
 srcdir=`dirname $0`
 test -z "$srcdir" && srcdir=.
@@ -57,6 +57,6 @@
 ) || exit 1
 
 #conf_flags="--enable-maintainer-mode --enable-compile-warnings"
-echo Running $srcdir/configure $conf_flags "$@" ...
-$srcdir/configure --cache-file=config.cache $conf_flags "$@" && \
-  echo "Now type \`make' to compile giFT."
+#echo Running $srcdir/configure $conf_flags "$@" ...
+#$srcdir/configure --cache-file=config.cache $conf_flags "$@" && \
+#  echo "Now type \`make' to compile giFT."
diff -ruN giFT/src/Makefile.am giFT.new/src/Makefile.am
--- giFT/src/Makefile.am	2003-05-03 13:03:59.000000000 -0400
+++ giFT.new/src/Makefile.am	2003-05-20 19:54:45.000000000 -0400
@@ -60,6 +60,7 @@
 
 giFT_LDFLAGS = \
 	$(GIFT_LDFLAGS) \
+	-L$(top_builddir)/lib/.libs \
 	$(IMAGEMAGICK_LDFLAGS) \
 	-export-dynamic
 
