diff -ruN hammerhead-2.1.1/configure hammerhead-2.1.1.new/configure
--- hammerhead-2.1.1/configure	Wed Jul 18 08:56:58 2001
+++ hammerhead-2.1.1.new/configure	Thu Oct  3 12:42:39 2002
@@ -716,6 +716,18 @@
 	TARGET_OS=FreeBSD
 	
 	;;
+*darwin*)
+	cat >> confdefs.h <<\EOF
+#define BSD 1
+EOF
+
+	cat >> confdefs.h <<\EOF
+#define USE_BPF 1
+EOF
+
+	TARGET_OS=FreeBSD
+	
+	;;
 *linux*) 
 	cat >> confdefs.h <<\EOF
 #define LINUX 1
