diff -ruN kdegift-0.7/Makefile.am kdegift-0.7.new/Makefile.am
--- kdegift-0.7/Makefile.am	2003-04-14 16:38:41.000000000 -0400
+++ kdegift-0.7.new/Makefile.am	2003-06-14 15:08:33.000000000 -0400
@@ -27,3 +27,5 @@
 dist-hook:
 	cd $(top_distdir) && perl admin/am_edit -padmin
 	cd $(top_distdir) && $(MAKE) -f admin/Makefile.common subdirs
+
+.PHONY: dist-hook package-messages install
diff -ruN kdegift-0.7/kdegift/Makefile.am kdegift-0.7.new/kdegift/Makefile.am
--- kdegift-0.7/kdegift/Makefile.am	2003-04-14 16:39:21.000000000 -0400
+++ kdegift-0.7.new/kdegift/Makefile.am	2003-06-14 14:53:30.000000000 -0400
@@ -7,7 +7,7 @@
 install-data-local:
 	$(mkinstalldirs) $(kde_appsdir)/Applications/
 	$(INSTALL_DATA) $(srcdir)/KdeGiFT.desktop $(kde_appsdir)/Applications/KdeGiFT.desktop
-	$(mkinstalldirs) $(kde_icondir)/hicolor/32x32/apps/
+	$(mkinstalldirs) $(kde_icondir)/hicolor/16x16/apps/
 	$(INSTALL_DATA) $(srcdir)/kdegift16x16.png $(kde_icondir)/hicolor/16x16/apps/kdegift.png
 	$(mkinstalldirs) $(kde_icondir)/hicolor/32x32/apps/
 	$(INSTALL_DATA) $(srcdir)/kdegift32x32.png $(kde_icondir)/hicolor/32x32/apps/kdegift.png
