diff -ruN kdegift-0.5/kdegift/Makefile.am kdegift-0.5.new/kdegift/Makefile.am
--- kdegift-0.5/kdegift/Makefile.am	2003-03-30 01:26:49.000000000 -0500
+++ kdegift-0.5.new/kdegift/Makefile.am	2003-04-04 22:41:32.000000000 -0500
@@ -8,9 +8,9 @@
 install-data-local:
 	$(mkinstalldirs) $(kde_appsdir)/Applications/
 	$(INSTALL_DATA) $(srcdir)/kdegift.desktop $(kde_appsdir)/Applications/kdegift.desktop
-	$(mkinstalldirs) $(kde_icondir)/hicolor/32x32/apps/
+	$(mkinstalldirs) $(kde_icondir)/hicolor/16x16/apps/
 	$(INSTALL_DATA) $(srcdir)/kdegift16x16.png $(kde_icondir)/hicolor/16x16/apps/kdegift.png
-	$(mkinstalldirs) $(kde_icondir)/locolor/32x32/apps/
+	$(mkinstalldirs) $(kde_icondir)/hicolor/32x32/apps/
 	$(INSTALL_DATA) $(srcdir)/kdegift32x32.png $(kde_icondir)/hicolor/32x32/apps/kdegift.png
 
 uninstall-local:
diff -ruN kdegift-0.5/kdegift/Makefile.in kdegift-0.5.new/kdegift/Makefile.in
--- kdegift-0.5/kdegift/Makefile.in	2003-04-01 13:15:28.000000000 -0500
+++ kdegift-0.5.new/kdegift/Makefile.in	2003-04-04 22:42:01.000000000 -0500
@@ -620,13 +620,13 @@
 	$(INSTALL_DATA) $(srcdir)/kdegift.desktop $(DESTDIR)$(kde_appsdir)/Applications/kdegift.desktop
 #>- 	$(mkinstalldirs) $(kde_icondir)/hicolor/32x32/apps/
 #>+ 1
-	$(mkinstalldirs) $(DESTDIR)$(kde_icondir)/hicolor/32x32/apps/
+	$(mkinstalldirs) $(DESTDIR)$(kde_icondir)/hicolor/16x16/apps/
 #>- 	$(INSTALL_DATA) $(srcdir)/kdegift16x16.png $(kde_icondir)/hicolor/16x16/apps/kdegift.png
 #>+ 1
 	$(INSTALL_DATA) $(srcdir)/kdegift16x16.png $(DESTDIR)$(kde_icondir)/hicolor/16x16/apps/kdegift.png
 #>- 	$(mkinstalldirs) $(kde_icondir)/locolor/32x32/apps/
 #>+ 1
-	$(mkinstalldirs) $(DESTDIR)$(kde_icondir)/locolor/32x32/apps/
+	$(mkinstalldirs) $(DESTDIR)$(kde_icondir)/hicolor/32x32/apps/
 #>- 	$(INSTALL_DATA) $(srcdir)/kdegift32x32.png $(kde_icondir)/hicolor/32x32/apps/kdegift.png
 #>+ 1
 	$(INSTALL_DATA) $(srcdir)/kdegift32x32.png $(DESTDIR)$(kde_icondir)/hicolor/32x32/apps/kdegift.png
