diff -ruN kift-0.8.1/kift/kiftdownload.cpp kift-0.8.1.new/kift/kiftdownload.cpp
--- kift-0.8.1/kift/kiftdownload.cpp	Sun Mar 24 09:10:19 2002
+++ kift-0.8.1.new/kift/kiftdownload.cpp	Sun Aug 11 13:02:50 2002
@@ -41,7 +41,7 @@
 #include "clistviewitem.h"
 #include "kiftsearch.h"
 
-
+using namespace std
 
 Kiftdownload::Kiftdownload()
 {
@@ -501,4 +501,4 @@
   cerr<<query<<endl;
   search->startSearch(query,id);
   cerr<<"Not fully implemented now (findmoresources)!"<<endl;
-}
\ No newline at end of file
+}
