diff -ruN kift-0.8.1/kift/kiftdownload.cpp kift-0.8.1.new/kift/kiftdownload.cpp
--- kift-0.8.1/kift/kiftdownload.cpp	Sun Mar 24 09:10:19 2002
+++ kift-0.8.1.new/kift/kiftdownload.cpp	Sun Aug 11 13:02:50 2002
@@ -41,7 +41,7 @@
 #include "clistviewitem.h"
 #include "kiftsearch.h"
 
-
+using namespace std;
 
 Kiftdownload::Kiftdownload()
 {
@@ -501,4 +501,4 @@
   cerr<<query<<endl;
   search->startSearch(query,id);
   cerr<<"Not fully implemented now (findmoresources)!"<<endl;
-}
\ No newline at end of file
+}
--- kift-0.8.1/kift/kifftid3editor.h	2002-03-12 18:23:34.000000000 -0800
+++ kift-0.8.1-patched/kift/kifftid3editor.h	2002-08-11 18:23:07.000000000 -0700
@@ -36,6 +36,8 @@
 
 #include "id3propsui.h"
 
+using namespace std;
+
 /**
   *@author kift1
   */
@@ -67,7 +69,7 @@
     void slotUpButton();
     void slotHomeButton();
     void slotSaveButton();
-    void slotProgramOutput(KProcess *proc, char *stdout, int len);
+    void slotProgramOutput(KProcess *proc, char *so, int len);
     void slotProcessExited(KProcess *);
     void slotGotoURL();
     void slotGotoID3HPURL();
--- kift-0.8.1/kift/kifftid3editor.cpp	2002-04-05 03:46:10.000000000 -0800
+++ kift-0.8.1-patched/kift/kifftid3editor.cpp	2002-08-11 18:24:23.000000000 -0700
@@ -31,7 +31,7 @@
 #include <kstdaction.h>
 #include <kstdaccel.h>
 #include <kaction.h>
-#include <kapp.h>
+#include <kapplication.h>
 #include <krun.h>
 
 #include <qvaluelist.h>
@@ -212,11 +212,11 @@
 }
 
 
-void kifftID3Editor::slotProgramOutput(KProcess *proc, char *stdout, int len)
+void kifftID3Editor::slotProgramOutput(KProcess *proc, char *so, int len)
 {
-  cerr<<stdout<<endl;
-  QString output = stdout;
-  output = output.left( len );
+  cerr<<so<<endl;
+  QString output = so;
+  so = output.left( len );
   id3toolOut=id3toolOut+output;
 }
 
--- kift-0.8.1/kift/founditem.h	2002-03-11 05:37:03.000000000 -0800
+++ kift-0.8.1-patched/kift/founditem.h	2002-08-11 18:28:56.000000000 -0700
@@ -23,6 +23,8 @@
 #include <kurl.h>
 #include <baseitemclass.h>
 
+using namespace std;
+
 /**
   *@author Kift01
   */
--- kift-0.8.1/kift/downloadmanager.h	2002-03-12 03:45:17.000000000 -0800
+++ kift-0.8.1-patched/kift/downloadmanager.h	2002-08-11 18:32:58.000000000 -0700
@@ -18,6 +18,8 @@
 #ifndef DOWNLOADMANAGER_H
 #define DOWNLOADMANAGER_H
 
+using namespace std;
+
 #include <iostream>
 
 #include <qlistview.h>
--- kift-0.8.1/kift/baseitemclass.h	2002-03-11 05:45:00.000000000 -0800
+++ kift-0.8.1-patched/kift/baseitemclass.h	2002-08-11 18:37:27.000000000 -0700
@@ -18,6 +18,8 @@
 #ifndef BASEITEMCLASS_H
 #define BASEITEMCLASS_H
 
+using namespace std;
+
 #include <qobject.h>
 #include <qsocket.h>
 #include <qvaluelist.h>
--- kift-0.8.1/kift/clistviewitem.h	2002-03-11 06:11:44.000000000 -0800
+++ kift-0.8.1-patched/kift/clistviewitem.h	2002-08-11 18:38:04.000000000 -0700
@@ -18,6 +18,8 @@
 #ifndef CLISTVIEWITEM_H
 #define CLISTVIEWITEM_H
 
+using namespace std;
+
 #include <qlistview.h>
 #include <qstring.h>
 
--- kift-0.8.1/kift/kift.h	2002-03-12 18:30:51.000000000 -0800
+++ kift-0.8.1-patched/kift/kift.h	2002-08-11 18:45:27.000000000 -0700
@@ -22,6 +22,8 @@
 #include <config.h>
 #endif
 
+using namespace std;
+
 #include <qwidget.h>
 #include <qmultilineedit.h>
 #include <qlabel.h>
@@ -39,7 +41,7 @@
 #include <kio/job.h>
 #include <kicondialog.h>
 
-#include <kapp.h>
+#include <kapplication.h>
 #include "kifftmainForm.h"
 #include "founditem.h"
 #include "kiftsearch.h"
