diff -uNr arts-1.3.91/admin/ltmain.sh arts-1.3.91-new/admin/ltmain.sh
--- arts-1.3.91/admin/ltmain.sh	Mon Jan  3 09:24:20 2005
+++ arts-1.3.91-new/admin/ltmain.sh	Fri Jan 14 16:00:51 2005
@@ -4453,6 +4453,7 @@
 	# We should set the shlibpath_var
 	rpath=
 	for dir in $temp_rpath; do
+     if test "$dir" != "@FINKPREFIX@/lib"; then
 	  case $dir in
 	  [\\/]* | [A-Za-z]:[\\/]*)
 	    # Absolute path.
@@ -4463,6 +4464,7 @@
 	    rpath="$rpath\$thisdir/$dir:"
 	    ;;
 	  esac
+     fi
 	done
 	temp_rpath="$rpath"
       fi
diff -uNr arts-1.3.91/configure arts-1.3.91-new/configure
--- arts-1.3.91/configure	Tue Jan  4 03:50:55 2005
+++ arts-1.3.91-new/configure	Fri Jan 14 16:00:51 2005
@@ -9793,10 +9793,21 @@
       archive_cmds_need_lc=no
       case "$host_os" in
       rhapsody* | darwin1.[012])
-	allow_undefined_flag='-undefined suppress'
+	allow_undefined_flag='-Wl,-undefined -Wl,suppress'
 	;;
       *) # Darwin 1.3 on
-	test -z ${LD_TWOLEVEL_NAMESPACE} && allow_undefined_flag='-flat_namespace -undefined suppress'
+	if test -z ${MACOSX_DEPLOYMENT_TARGET} ; then
+	  allow_undefined_flag='-Wl,-flat_namespace -Wl,-undefined -Wl,suppress'
+	else
+	  case ${MACOSX_DEPLOYMENT_TARGET} in
+	    10.[012])
+	      allow_undefined_flag='-Wl,-flat_namespace -Wl,-undefined -Wl,suppress'
+	      ;;
+	    10.*)
+	      allow_undefined_flag='-Wl,-undefined -Wl,dynamic_lookup'
+	      ;;
+	  esac
+	fi
 	;;
       esac
     	lt_int_apple_cc_single_mod=no
@@ -11643,10 +11654,21 @@
     archive_cmds_need_lc=no
     case "$host_os" in
     rhapsody* | darwin1.[012])
-      allow_undefined_flag='-undefined suppress'
+      allow_undefined_flag='-Wl,-undefined -Wl,suppress'
       ;;
     *) # Darwin 1.3 on
-      test -z ${LD_TWOLEVEL_NAMESPACE} && allow_undefined_flag='-flat_namespace -undefined suppress'
+	if test -z ${MACOSX_DEPLOYMENT_TARGET} ; then
+	  allow_undefined_flag='-Wl,-flat_namespace -Wl,-undefined -Wl,suppress'
+	else
+	  case ${MACOSX_DEPLOYMENT_TARGET} in
+	    10.[012])
+	      allow_undefined_flag='-Wl,-flat_namespace -Wl,-undefined -Wl,suppress'
+	      ;;
+	    10.*)
+	      allow_undefined_flag='-Wl,-undefined -Wl,dynamic_lookup'
+	      ;;
+	  esac
+	fi
       ;;
     esac
     output_verbose_link_cmd='echo'
@@ -12694,10 +12716,21 @@
     archive_cmds_need_lc_CXX=no
     case "$host_os" in
     rhapsody* | darwin1.[012])
-      allow_undefined_flag_CXX='-undefined suppress'
+      allow_undefined_flag_CXX='-Wl,-undefined -Wl,suppress'
       ;;
     *) # Darwin 1.3 on
-      test -z ${LD_TWOLEVEL_NAMESPACE} && allow_undefined_flag_CXX='-flat_namespace -undefined suppress'
+	if test -z ${MACOSX_DEPLOYMENT_TARGET} ; then
+	  allow_undefined_flag='-Wl,-flat_namespace -Wl,-undefined -Wl,suppress'
+	else
+	  case ${MACOSX_DEPLOYMENT_TARGET} in
+	    10.[012])
+	      allow_undefined_flag='-Wl,-flat_namespace -Wl,-undefined -Wl,suppress'
+	      ;;
+	    10.*)
+	      allow_undefined_flag='-Wl,-undefined -Wl,dynamic_lookup'
+	      ;;
+	  esac
+	fi
       ;;
     esac
     lt_int_apple_cc_single_mod=no
@@ -16517,10 +16550,21 @@
       archive_cmds_need_lc_F77=no
       case "$host_os" in
       rhapsody* | darwin1.[012])
-	allow_undefined_flag_F77='-undefined suppress'
+	allow_undefined_flag_F77='-Wl,-undefined -Wl,suppress'
 	;;
       *) # Darwin 1.3 on
-	test -z ${LD_TWOLEVEL_NAMESPACE} && allow_undefined_flag_F77='-flat_namespace -undefined suppress'
+	if test -z ${MACOSX_DEPLOYMENT_TARGET} ; then
+	  allow_undefined_flag='-Wl,-flat_namespace -Wl,-undefined -Wl,suppress'
+	else
+	  case ${MACOSX_DEPLOYMENT_TARGET} in
+	    10.[012])
+	      allow_undefined_flag='-Wl,-flat_namespace -Wl,-undefined -Wl,suppress'
+	      ;;
+	    10.*)
+	      allow_undefined_flag='-Wl,-undefined -Wl,dynamic_lookup'
+	      ;;
+	  esac
+	fi
 	;;
       esac
     	lt_int_apple_cc_single_mod=no
@@ -18799,10 +18843,21 @@
       archive_cmds_need_lc_GCJ=no
       case "$host_os" in
       rhapsody* | darwin1.[012])
-	allow_undefined_flag_GCJ='-undefined suppress'
+	allow_undefined_flag_GCJ='-Wl,-undefined -Wl,suppress'
 	;;
       *) # Darwin 1.3 on
-	test -z ${LD_TWOLEVEL_NAMESPACE} && allow_undefined_flag_GCJ='-flat_namespace -undefined suppress'
+	if test -z ${MACOSX_DEPLOYMENT_TARGET} ; then
+	  allow_undefined_flag='-Wl,-flat_namespace -Wl,-undefined -Wl,suppress'
+	else
+	  case ${MACOSX_DEPLOYMENT_TARGET} in
+	    10.[012])
+	      allow_undefined_flag='-Wl,-flat_namespace -Wl,-undefined -Wl,suppress'
+	      ;;
+	    10.*)
+	      allow_undefined_flag='-Wl,-undefined -Wl,dynamic_lookup'
+	      ;;
+	  esac
+	fi
 	;;
       esac
     	lt_int_apple_cc_single_mod=no
@@ -22657,6 +22712,9 @@
 	libltdl_cv_sys_dlopen_deplibs=unknown
 	case "$host_os" in
 	cygwin*)
+	  libltdl_cv_sys_dlopen_deplibs=yes
+	  ;;
+	darwin*)
 	  libltdl_cv_sys_dlopen_deplibs=yes
 	  ;;
 	darwin*)
diff -uNr arts-1.3.91/examples/Makefile.in arts-1.3.91-new/examples/Makefile.in
--- arts-1.3.91/examples/Makefile.in	Tue Jan  4 03:50:50 2005
+++ arts-1.3.91-new/examples/Makefile.in	Fri Jan 14 16:00:51 2005
@@ -555,7 +555,7 @@
 testasubsys_SOURCES = testasubsys.cc
 testasubsys_LDADD = $(FLOWLIBS)
 testthreads_SOURCES = testthreads.cc
-testthreads_LDFLAGS = $(LDFLAGS)
+testthreads_LDFLAGS = $(all_libraries)
 testthreads_LDADD = $(top_builddir)/mcop_mt/libmcop_mt.la
 testnothreads_SOURCES = testthreads.cc
 testaggregation_SOURCES = aggregation.cc testaggregation.cc
@@ -575,7 +575,7 @@
 artsccat_SOURCES = artsccat.c
 artscrec_LDADD = $(top_builddir)/artsc/libartsc.la
 artscrec_SOURCES = artscrec.c
-artscmt_LDFLAGS = $(LDFLAGS) $(USE_THREADS)
+artscmt_LDFLAGS = $(all_libraries) $(USE_THREADS)
 artscmt_LDADD = $(top_builddir)/artsc/libartsc.la -lm $(LIBPTHREAD) $(LIBPOSIX4)
 artscmt_SOURCES = artscmt.c
 playtofile_SOURCES = playtofile.cc playtofile_impl.cc playtofile_main.cc
diff -uNr arts-1.3.91/mcop/tcpserver.cc arts-1.3.91-new/mcop/tcpserver.cc
--- arts-1.3.91/mcop/tcpserver.cc	Fri Nov 29 11:51:13 2002
+++ arts-1.3.91-new/mcop/tcpserver.cc	Fri Jan 14 16:01:16 2005
@@ -129,7 +129,7 @@
 		return false;
     }
 
-    socklen_t sz = sizeof(struct sockaddr_in);
+    kde_socklen_t sz = sizeof(struct sockaddr_in);
 	int r = getsockname (theSocket,(struct sockaddr *)&socket_addr, &sz);
 	if(r == 0) {
 		thePort = ntohs(socket_addr.sin_port);
@@ -170,7 +170,7 @@
 	{
 		int clientfd;
 		struct sockaddr_in incoming;
-		socklen_t size_in = sizeof(struct sockaddr_in);
+		kde_socklen_t size_in = sizeof(struct sockaddr_in);
 
 		clientfd = accept(theSocket, (struct sockaddr*) &incoming, &size_in );
 		if(clientfd > 0)
diff -uNr arts-1.3.91/mcop/unixserver.cc arts-1.3.91-new/mcop/unixserver.cc
--- arts-1.3.91/mcop/unixserver.cc	Fri Mar  8 15:30:20 2002
+++ arts-1.3.91-new/mcop/unixserver.cc	Fri Jan 14 16:01:16 2005
@@ -130,7 +130,7 @@
 	{
 		int clientfd;
 		struct sockaddr_un incoming;
-		socklen_t size_in = sizeof(struct sockaddr_un);
+		kde_socklen_t size_in = sizeof(struct sockaddr_un);
 
 		clientfd = accept(theSocket, (struct sockaddr*) &incoming, &size_in );
 		if(clientfd > 0)
diff -uNr arts-1.3.91/x11/Makefile.in arts-1.3.91-new/x11/Makefile.in
--- arts-1.3.91/x11/Makefile.in	Tue Jan  4 03:50:50 2005
+++ arts-1.3.91-new/x11/Makefile.in	Fri Jan 14 16:00:51 2005
@@ -371,7 +371,7 @@
 @include_x11_TRUE@lib_LTLIBRARIES = libx11globalcomm.la
 @include_x11_TRUE@libx11globalcomm_la_SOURCES = x11globalcomm.cc x11globalcomm_impl.cc
 @include_x11_TRUE@libx11globalcomm_la_LIBADD = $(top_builddir)/mcop/libmcop.la $(LIB_X11)
-@include_x11_TRUE@libx11globalcomm_la_LDFLAGS = -no-undefined -module -version-info 1:0 $(X_LDFLAGS) $(all_libraries)
+@include_x11_TRUE@libx11globalcomm_la_LDFLAGS = -no-undefined -module -version-info 1:0 $(all_libraries) $(X_LDFLAGS)
 @include_x11_TRUE@DISTCLEANFILES = x11globalcomm.cc x11globalcomm.h \
 @include_x11_TRUE@                 x11globalcomm.mcoptype x11globalcomm.mcopclass
 
