diff -uNr kdeartwork-3.2.92/kscreensaver/kdesavers/Makefile.am kdeartwork-3.2.92-new/kscreensaver/kdesavers/Makefile.am
--- kdeartwork-3.2.92/kscreensaver/kdesavers/Makefile.am	2004-06-22 13:13:59.000000000 -0400
+++ kdeartwork-3.2.92-new/kscreensaver/kdesavers/Makefile.am	2004-07-26 13:25:57.000000000 -0400
@@ -22,7 +22,7 @@
 LIBART_KSAVERS_LDADD = $(AM_LDADD) $(LIBART_LIBS)
 
 if COMPILE_GL_KSAVERS
-GL_KSAVERS=kfountain.kss kwave.kss kgravity.kss kflux.kss keuphoria.kss ksolarwinds.kss krotation.kss kpendulum.kss kfiresaver.kss
+GL_KSAVERS=kfountain.kss kwave.kss kgravity.kss kflux.kss keuphoria.kss ksolarwinds.kss kpendulum.kss kfiresaver.kss
 GL_DESKTOP_FILES=KGravity.desktop KWave.desktop KFountain.desktop KFlux.desktop KEuphoria.desktop KSolarWinds.desktop KRotation.desktop KPendulum.desktop KFiresaver.desktop
 endif
 GL_KSAVERS_LDADD = $(AM_LDADD) $(GLLIB)
@@ -62,8 +62,8 @@
 keuphoria_kss_LDADD = $(GL_KSAVERS_LDADD) -lm
 ksolarwinds_kss_SOURCES = SolarWinds.cpp
 ksolarwinds_kss_LDADD = $(GL_KSAVERS_LDADD) -lm
-krotation_kss_SOURCES = rotation.cpp rotationcfg.ui sspreviewarea.cpp rkodesolver.cpp vec3.cpp
-krotation_kss_LDADD = $(GL_KSAVERS_LDADD) -lm
+#krotation_kss_SOURCES = rotation.cpp rotationcfg.ui sspreviewarea.cpp rkodesolver.cpp vec3.cpp
+#krotation_kss_LDADD = $(GL_KSAVERS_LDADD) -lm
 kpendulum_kss_SOURCES = pendulum.cpp pendulumcfg.ui sspreviewarea.cpp rkodesolver.cpp
 kpendulum_kss_LDADD = $(GL_KSAVERS_LDADD) -lm
 kfiresaver_kss_SOURCES = firesaversetup.ui firesaverparticle.cpp firesaverwriter.cpp firesaver.cpp
