diff -uNr kdebase-3.2.92/kioslave/info/kde-info2html.conf kdebase-3.2.92-new/kioslave/info/kde-info2html.conf
--- kdebase-3.2.92/kioslave/info/kde-info2html.conf	2004-05-22 16:26:52.000000000 -0400
+++ kdebase-3.2.92-new/kioslave/info/kde-info2html.conf	2004-07-19 09:31:18.000000000 -0400
@@ -25,6 +25,7 @@
 
 #-- location of info files.
 our @INFODIR = (
+	    "@PREFIX@/share/info",
 	    "/usr/share/info",
 	    "/usr/info",
 	    "/usr/lib/info",
diff -uNr kdebase-3.2.92/kioslave/man/kio_man.cpp kdebase-3.2.92-new/kioslave/man/kio_man.cpp
--- kdebase-3.2.92/kioslave/man/kio_man.cpp	2004-06-22 13:32:52.000000000 -0400
+++ kdebase-3.2.92-new/kioslave/man/kio_man.cpp	2004-07-19 09:31:18.000000000 -0400
@@ -50,7 +50,7 @@
 
 MANProtocol *MANProtocol::_self = 0;
 
-#define SGML2ROFF_DIRS "/usr/lib/sgml"
+#define SGML2ROFF_DIRS "@PREFIX@/share/sgml"
 
 /*
  * Drop trailing ".section[.gz]" from name
diff -uNr kdebase-3.2.92/libkonq/Makefile.am kdebase-3.2.92-new/libkonq/Makefile.am
--- kdebase-3.2.92/libkonq/Makefile.am	2004-05-22 16:27:06.000000000 -0400
+++ kdebase-3.2.92-new/libkonq/Makefile.am	2004-07-19 09:31:49.000000000 -0400
@@ -21,7 +21,7 @@
 
 lib_LTLIBRARIES = libkonq.la
 libkonq_la_LDFLAGS = $(all_libraries) -version-info 6:0:2 -no-undefined
-libkonq_la_LIBADD = $(LIB_KPARTS)
+libkonq_la_LIBADD = $(LIB_KPARTS) $(LIBZ)
 
 libkonq_la_SOURCES = konq_popupmenu.cc knewmenu.cc \
    konq_xmlguiclient.cc\
diff -uNr kdebase-3.2.92/startkde kdebase-3.2.92-new/startkde
--- kdebase-3.2.92/startkde	2004-07-17 13:51:54.000000000 -0400
+++ kdebase-3.2.92-new/startkde	2004-07-19 09:31:49.000000000 -0400
@@ -26,6 +26,8 @@
   esac
 fi
 
+source "@PREFIX@/bin/init.sh"
+
 # Boot sequence:
 #
 # kdeinit is used to fork off processes which improves memory usage
@@ -86,6 +88,11 @@
 # Where X is the original font database that was set up before this script
 # runs.
 
+/usr/X11R6/bin/xmodmap -e "keycode 74 = Right"
+/usr/X11R6/bin/xmodmap -e "keycode 78 = Left"
+/usr/X11R6/bin/xmodmap -e "keycode 80 = Down"
+/usr/X11R6/bin/xmodmap -e "keycode 85 = Up"
+
 usr_odir=$HOME/.fonts/kde-override
 usr_fdir=$HOME/.fonts
 
@@ -190,7 +197,7 @@
     done
     dcopserver_shutdown
     # shutdown will also make kwin quit, give it time to do so
-    sleep 1
+    sleep 5
 fi
 
 # the splashscreen and progress indicator
@@ -214,7 +221,29 @@
 # We only check for 255 which means that the ksmserver process could not be 
 # started, any problems thereafter, e.g. ksmserver failing to initialize, 
 # will remain undetected.
-test -n "$KDEWM" && KDEWM="--windowmanager $KDEWM"
+
+USE_PROXY=0
+if test -n "$KDEWM"; then
+	if test -z `expr "$KDEWM" : '\(.*quartz-wm.*\)'`; then
+		USE_PROXY=1
+	fi
+	KDEWM="--windowmanager $KDEWM"
+else
+	KDEWM="--windowmanager kwin"
+	USE_PROXY=1
+fi
+
+if test "$USE_PROXY" = "1"; then
+	if test -x /usr/X11R6/bin/quartz-wm; then
+		/usr/X11R6/bin/quartz-wm --only-proxy >/dev/null 2>&1 &
+	else
+		AUTOCUTSEL=`which autocutsel 2>/dev/null`
+		if test -n "$AUTOCUTSEL" && test -x "$AUTOCUTSEL"; then
+			$AUTOCUTSEL >/dev/null 2>&1 &
+		fi
+	fi
+fi
+
 kwrapper ksmserver $KDEWM 
 if test $? -eq 255; then
   # Startup error
