diff -uNr kvim-6.2.14/src/auto/configure kvim-6.2.14-new/src/auto/configure
--- kvim-6.2.14/src/auto/configure	Sat Jun 14 13:38:44 2003
+++ kvim-6.2.14-new/src/auto/configure	Sun Feb  1 10:07:00 2004
@@ -1217,7 +1217,7 @@
 if test -z "$have_local_include" -a -d /usr/local/include; then
   tt=`echo "$CPPFLAGS" | sed -e 's+-I/usr/local/include ++g' -e 's+-I/usr/local/include$++g'`
   if test "$tt" = "$CPPFLAGS"; then
-    CPPFLAGS="$CPPFLAGS -I/usr/local/include"
+    CPPFLAGS="$CPPFLAGS"
   fi
 fi
 
@@ -1957,7 +1957,7 @@
 	  echo $ac_n "checking for Tcl library by myself""... $ac_c" 1>&6
 echo "configure:1959: checking for Tcl library by myself" >&5
 	  tcllib=`echo $tclinc | sed s/include/lib/g`
-	  for ext in .so .a ; do
+	  for ext in .dylib .so .a ; do
 	    for ver in "" $tclver ; do
 	      for try in $tcllib ; do
 		trylib=tcl$ver$ext
@@ -4791,7 +4791,7 @@
     gui_libs="`echo $x_libraries|sed 's%/^/^/*$%%'` `echo "$gui_XXX" | sed s/XXX/lib/g` `echo "$GUI_INC_LOC" | sed s/include/lib/` $GUI_LIB_LOC"
     GUI_LIB_LOC=
     for try in $gui_libs; do
-      for libtry in "$try"/libXm.a "$try"/libXm.so* "$try"/libXm.sl; do
+      for libtry in "$try"/libXm.a "$try"/libXm.so* "$try"/libXm.dylib "$try"/libXm.sl; do
 	if test -f "$libtry"; then
 	  GUI_LIB_LOC=$try
 	fi
diff -uNr kvim-6.2.14/src/configure kvim-6.2.14-new/src/configure
--- kvim-6.2.14/src/configure	Sat Jun 14 13:38:46 2003
+++ kvim-6.2.14-new/src/configure	Sun Feb  1 10:07:00 2004
@@ -1217,7 +1217,7 @@
 if test -z "$have_local_include" -a -d /usr/local/include; then
   tt=`echo "$CPPFLAGS" | sed -e 's+-I/usr/local/include ++g' -e 's+-I/usr/local/include$++g'`
   if test "$tt" = "$CPPFLAGS"; then
-    CPPFLAGS="$CPPFLAGS -I/usr/local/include"
+    CPPFLAGS="$CPPFLAGS"
   fi
 fi
 
@@ -1957,7 +1957,7 @@
 	  echo $ac_n "checking for Tcl library by myself""... $ac_c" 1>&6
 echo "configure:1959: checking for Tcl library by myself" >&5
 	  tcllib=`echo $tclinc | sed s/include/lib/g`
-	  for ext in .so .a ; do
+	  for ext in .dylib .so .a ; do
 	    for ver in "" $tclver ; do
 	      for try in $tcllib ; do
 		trylib=tcl$ver$ext
@@ -4791,7 +4791,7 @@
     gui_libs="`echo $x_libraries|sed 's%/^/^/*$%%'` `echo "$gui_XXX" | sed s/XXX/lib/g` `echo "$GUI_INC_LOC" | sed s/include/lib/` $GUI_LIB_LOC"
     GUI_LIB_LOC=
     for try in $gui_libs; do
-      for libtry in "$try"/libXm.a "$try"/libXm.so* "$try"/libXm.sl; do
+      for libtry in "$try"/libXm.a "$try"/libXm.so* "$try"/libXm.dylib "$try"/libXm.sl; do
 	if test -f "$libtry"; then
 	  GUI_LIB_LOC=$try
 	fi
