diff -uNr kdebase-3.3.91/startkde kdebase-3.3.91-new/startkde
--- kdebase-3.3.91/startkde	Mon Jan  3 18:36:08 2005
+++ kdebase-3.3.91-new/startkde	Wed Feb  2 20:49:29 2005
@@ -179,21 +179,24 @@
 
 echo 'startkde: Starting up...'  1>&2
 
-# run KPersonalizer before the session, if this is the first login
-if kreadconfig --file kpersonalizerrc --group General --key FirstLogin --default true --type bool; then
-    # start only dcopserver, don't start whole kdeinit (takes too long)
-    echo 'startkde: Running kpersonalizer...'  1>&2
-    dcopserver
-    kwin --lock &
-    kpersonalizer --before-session
-    # handle kpersonalizer restarts (language change)
-    while test $? -eq 1; do
-        kpersonalizer --r --before-session
-    done
-    dcopserver_shutdown
-    # shutdown will also make kwin quit, give it time to do so
-    sleep 1
-fi
+# on osx we skip this, the defaults are sane and most people
+# do bad things like "mac os menus" which makes it worse
+
+## run KPersonalizer before the session, if this is the first login
+#if kreadconfig --file kpersonalizerrc --group General --key FirstLogin --default true --type bool; then
+#    # start only dcopserver, don't start whole kdeinit (takes too long)
+#    echo 'startkde: Running kpersonalizer...'  1>&2
+#    dcopserver
+#    kwin --lock &
+#    kpersonalizer --before-session
+#    # handle kpersonalizer restarts (language change)
+#    while test $? -eq 1; do
+#        kpersonalizer --r --before-session
+#    done
+#    dcopserver_shutdown
+#    # shutdown will also make kwin quit, give it time to do so
+#    sleep 1
+#fi
 
 # the splashscreen and progress indicator
 splash=`kreadconfig --file ksplashrc --group KSplash --key Theme`
