--- kdeedu-3.2.90/kstars/kstars/indi/webcam/Makefile.am	Fri May  7 11:16:43 2004
+++ kdeedu-3.2.90-new/kstars/kstars/indi/webcam/Makefile.am	Thu Jun 24 21:24:18 2004
@@ -1,22 +1,13 @@
-if LINUX
-   libwebcam_linux = libwebcam_linux.la
-endif
-
-if BSD
-   libwebcam_bsd = libwebcam_null.la
-endif
-
 if NULL
    libwebcam_null = libwebcam_null.la
 endif
 
-noinst_LTLIBRARIES = libwebcam.la $(libwebcam_linux) $(libwebcam_bsd) $(libwebcam_null)
+noinst_LTLIBRARIES = libwebcam.la $(libwebcam_null)
 
-libwebcam_linux_la_SOURCES = PPort.cpp port.cpp vcvt_c.c QCamV4L.cpp ccvt_mmx_old.S
 libwebcam_null_la_SOURCES = dummy.cpp
 
 libwebcam_la_SOURCES = empty_file.cpp
-libwebcam_la_LIBADD = $(libwebcam_linux) $(libwebcam_bsd) $(libwebcam_null)
+libwebcam_la_LIBADD = $(libwebcam_null)
 
 empty_file.cpp:
 	echo > empty_file.cpp
--- kdeedu-3.2.90/kstars/kstars/indi/webcam/dummy.cpp	Wed Mar 31 05:07:19 2004
+++ kdeedu-3.2.90-new/kstars/kstars/indi/webcam/dummy.cpp	Thu Jun 24 21:57:41 2004
@@ -1,7 +1,6 @@
 #include <iostream>
 
 #include "ccvt.h"
-#include "QCamV4L.h"
 #include "../eventloop.h"
 
 using namespace std;
