diff -uNr kdegraphics-3.3.92/admin/ltmain.sh kdegraphics-3.3.92-new/admin/ltmain.sh
--- kdegraphics-3.3.92/admin/ltmain.sh	Mon Jan  3 09:53:11 2005
+++ kdegraphics-3.3.92-new/admin/ltmain.sh	Sat Feb  5 16:45:17 2005
@@ -4453,6 +4453,7 @@
 	# We should set the shlibpath_var
 	rpath=
 	for dir in $temp_rpath; do
+     if test "$dir" != "@FINKPREFIX@/lib"; then
 	  case $dir in
 	  [\\/]* | [A-Za-z]:[\\/]*)
 	    # Absolute path.
@@ -4463,6 +4464,7 @@
 	    rpath="$rpath\$thisdir/$dir:"
 	    ;;
 	  esac
+     fi
 	done
 	temp_rpath="$rpath"
       fi
diff -uNr kdegraphics-3.3.92/configure kdegraphics-3.3.92-new/configure
--- kdegraphics-3.3.92/configure	Fri Feb  4 06:29:17 2005
+++ kdegraphics-3.3.92-new/configure	Sat Feb  5 16:45:17 2005
@@ -8244,6 +8244,19 @@
     lt_cv_sys_max_cmd_len=8192;
     ;;
 
+  netbsd* | freebsd* | openbsd* | darwin* )
+    # This has been around since 386BSD, at least.  Likely further.
+    if test -x /sbin/sysctl; then
+      lt_cv_sys_max_cmd_len=`/sbin/sysctl -n kern.argmax`
+    elif test -x /usr/sbin/sysctl; then
+      lt_cv_sys_max_cmd_len=`/usr/sbin/sysctl -n kern.argmax`
+    else
+      lt_cv_sys_max_cmd_len=65536 # usable default for *BSD
+    fi
+    # And add a safety zone
+    lt_cv_sys_max_cmd_len=`expr $lt_cv_sys_max_cmd_len \/ 4`
+    ;;
+
  *)
     # If test is not a shell built-in, we'll probably end up computing a
     # maximum length that is only half of the actual maximum length, but
diff -uNr kdegraphics-3.3.92/kdvi/Makefile.in kdegraphics-3.3.92-new/kdvi/Makefile.in
--- kdegraphics-3.3.92/kdvi/Makefile.in	Fri Feb  4 06:29:02 2005
+++ kdegraphics-3.3.92-new/kdvi/Makefile.in	Sat Feb  5 17:08:03 2005
@@ -492,7 +492,7 @@
 # set the include path for X, qt and KDE
 INCLUDES = -I$(top_srcdir)/kviewshell \
 	-I$(top_builddir)/kviewshell \
-	$(all_includes) $(LIBFREETYPE_CFLAGS)
+	$(LIBFREETYPE_CFLAGS) $(all_includes)
 
 # claim, which subdirectories you want to install
 SUBDIRS = . pix
diff -uNr kdegraphics-3.3.92/kpovmodeler/Makefile.in kdegraphics-3.3.92-new/kpovmodeler/Makefile.in
--- kdegraphics-3.3.92/kpovmodeler/Makefile.in	Fri Feb  4 06:29:09 2005
+++ kdegraphics-3.3.92-new/kpovmodeler/Makefile.in	Sat Feb  5 17:08:03 2005
@@ -621,7 +621,7 @@
 xdg_appsdir = @xdg_appsdir@
 xdg_directorydir = @xdg_directorydir@
 xdg_menudir = @xdg_menudir@
-INCLUDES = $(LIBFREETYPE_CFLAGS) $(all_includes)
+INCLUDES = -I/usr/X11R6/include/freetype2 $(LIBFREETYPE_CFLAGS) $(all_includes)
 SUBDIRS = pics examples
 KDE_OPTIONS = nofinal
 
diff -uNr kdegraphics-3.3.92/ksvg/impl/libs/libtext2path/src/Makefile.in kdegraphics-3.3.92-new/ksvg/impl/libs/libtext2path/src/Makefile.in
--- kdegraphics-3.3.92/ksvg/impl/libs/libtext2path/src/Makefile.in	Fri Feb  4 06:29:12 2005
+++ kdegraphics-3.3.92-new/ksvg/impl/libs/libtext2path/src/Makefile.in	Sat Feb  5 17:08:03 2005
@@ -429,11 +429,11 @@
 xdg_menudir = @xdg_menudir@
 lib_LTLIBRARIES = libtext2path.la
 KDE_CXXFLAGS = $(USE_EXCEPTIONS)
-INCLUDES = $(FRIBIDI_CFLAGS) $(FREETYPE_CFLAGS) $(FONTCONFIG_CFLAGS) $(all_includes)
+INCLUDES = $(FREETYPE_CFLAGS) $(FRIBIDI_CFLAGS) $(FONTCONFIG_CFLAGS) $(all_includes)
 libtext2path_includedir = $(includedir)/libtext2path-0.1
 libtext2path_include_HEADERS = BezierPath.h Glyph.h GlyphTracer.h
 libtext2path_la_SOURCES = Affine.cpp Rectangle.cpp Font.cpp Glyph.cpp GlyphTracer.cpp Converter.cpp QtUnicode.cpp
-libtext2path_la_LDFLAGS = $(FRIBIDI_LIBS) $(FONTCONFIG_LIBS)
+libtext2path_la_LDFLAGS = $(FRIBIDI_LIBS) $(FONTCONFIG_LIBS) $(FREETYPE_LIBS)
 #>- all: all-am
 #>+ 1
 all: docs-am  all-am
diff -uNr kdegraphics-3.3.92/ksvg/plugin/backends/libart/Makefile.in kdegraphics-3.3.92-new/ksvg/plugin/backends/libart/Makefile.in
--- kdegraphics-3.3.92/ksvg/plugin/backends/libart/Makefile.in	Fri Feb  4 06:29:12 2005
+++ kdegraphics-3.3.92-new/ksvg/plugin/backends/libart/Makefile.in	Sat Feb  5 17:08:03 2005
@@ -436,7 +436,7 @@
 #>- libksvgrendererlibart_la_METASOURCES = AUTO
 kde_services_DATA = ksvglibartcanvas.desktop
 KDE_CXXFLAGS = $(USE_EXCEPTIONS)
-INCLUDES = $(LIBART_CFLAGS) $(FREETYPE_CFLAGS) $(FONTCONFIG_CFLAGS) -I$(top_srcdir)/ksvg/dom -I$(top_srcdir)/ksvg/impl -I$(top_srcdir)/ksvg/ecma -I$(top_srcdir)/ksvg/core -I$(top_srcdir)/ksvg/impl/libs/art_support -I$(top_srcdir)/ksvg/impl/libs/libtext2path/src $(KDE_INCLUDES) $(all_includes)
+INCLUDES = $(FREETYPE_CFLAGS) $(LIBART_CFLAGS) $(FONTCONFIG_CFLAGS) -I$(top_srcdir)/ksvg/dom -I$(top_srcdir)/ksvg/impl -I$(top_srcdir)/ksvg/ecma -I$(top_srcdir)/ksvg/core -I$(top_srcdir)/ksvg/impl/libs/art_support -I$(top_srcdir)/ksvg/impl/libs/libtext2path/src $(KDE_INCLUDES) $(all_includes)
 #>- all: all-am
 #>+ 1
 all: docs-am  all-am
diff -uNr kdegraphics-3.3.92/libkscan/img_canvas.cpp kdegraphics-3.3.92-new/libkscan/img_canvas.cpp
--- kdegraphics-3.3.92/libkscan/img_canvas.cpp	Tue Nov 16 13:07:25 2004
+++ kdegraphics-3.3.92-new/libkscan/img_canvas.cpp	Sat Feb  5 17:08:03 2005
@@ -700,7 +700,7 @@
                            double(image->width());
             kdDebug(29000) << "FIT WIDTH scrollbar to substract: " << sbWidth << endl;
         }
-        scale_factor = 100*used_xscaler;
+        scale_factor = (int)(100*used_xscaler);
         break;
     case FIT_HEIGHT:
         used_yscaler = used_xscaler = double(noSBSize.height())/double(image->height());
@@ -715,12 +715,12 @@
             kdDebug(29000) << "FIT HEIGHT scrollbar to substract: " << sbWidth << endl;
             // scale = int(100.0*(noSBSize.height() -sbWidth) / image->height());
         }
-        scale_factor = 100*used_xscaler;
+        scale_factor = (int)(100*used_xscaler);
 
         break;
     case ZOOM:
         used_xscaler = used_yscaler = double(getScaleFactor())/100.0;
-        scale_factor = 100*used_xscaler;
+        scale_factor = (int)(100*used_xscaler);
         break;
     default:
         break;
