--- kdelibs-3.3.91/admin/ltmain.sh	Mon Jan  3 09:53:16 2005
+++ kdelibs-3.3.91-new/admin/ltmain.sh	Fri Jan 14 15:04:17 2005
@@ -4453,6 +4453,7 @@
 	# We should set the shlibpath_var
 	rpath=
 	for dir in $temp_rpath; do
+     if test "$dir" != "@FINKPREFIX@/lib"; then
 	  case $dir in
 	  [\\/]* | [A-Za-z]:[\\/]*)
 	    # Absolute path.
@@ -4463,6 +4464,7 @@
 	    rpath="$rpath\$thisdir/$dir:"
 	    ;;
 	  esac
+     fi
 	done
 	temp_rpath="$rpath"
       fi
--- kdelibs-3.3.91/configure	Fri Jan  7 07:49:25 2005
+++ kdelibs-3.3.91-new/configure	Fri Jan 14 15:04:17 2005
@@ -8528,6 +8528,19 @@
     lt_cv_sys_max_cmd_len=8192;
     ;;
 
+  netbsd* | freebsd* | openbsd* | darwin* )
+    # This has been around since 386BSD, at least.  Likely further.
+    if test -x /sbin/sysctl; then
+      lt_cv_sys_max_cmd_len=`/sbin/sysctl -n kern.argmax`
+    elif test -x /usr/sbin/sysctl; then
+      lt_cv_sys_max_cmd_len=`/usr/sbin/sysctl -n kern.argmax`
+    else
+      lt_cv_sys_max_cmd_len=65536 # usable default for *BSD
+    fi
+    # And add a safety zone
+    lt_cv_sys_max_cmd_len=`expr $lt_cv_sys_max_cmd_len \/ 4`
+    ;;
+
  *)
     # If test is not a shell built-in, we'll probably end up computing a
     # maximum length that is only half of the actual maximum length, but
@@ -10171,10 +10184,21 @@
       archive_cmds_need_lc=no
       case "$host_os" in
       rhapsody* | darwin1.[012])
-	allow_undefined_flag='-undefined suppress'
+	allow_undefined_flag='-Wl,-undefined -Wl,suppress'
 	;;
       *) # Darwin 1.3 on
-	test -z ${LD_TWOLEVEL_NAMESPACE} && allow_undefined_flag='-flat_namespace -undefined suppress'
+	if test -z ${MACOSX_DEPLOYMENT_TARGET} ; then
+	  allow_undefined_flag='-Wl,-flat_namespace -Wl,-undefined -Wl,suppress'
+	else
+	  case ${MACOSX_DEPLOYMENT_TARGET} in
+	    10.[012])
+	      allow_undefined_flag='-Wl,-flat_namespace -Wl,-undefined -Wl,suppress'
+	      ;;
+	    10.*)
+	      allow_undefined_flag='-Wl,-undefined -Wl,dynamic_lookup'
+	      ;;
+	  esac
+	fi
 	;;
       esac
     	lt_int_apple_cc_single_mod=no
@@ -12024,7 +12048,18 @@
       allow_undefined_flag='-undefined suppress'
       ;;
     *) # Darwin 1.3 on
-      test -z ${LD_TWOLEVEL_NAMESPACE} && allow_undefined_flag='-flat_namespace -undefined suppress'
+	if test -z ${MACOSX_DEPLOYMENT_TARGET} ; then
+	  allow_undefined_flag='-Wl,-flat_namespace -Wl,-undefined -Wl,suppress'
+	else
+	  case ${MACOSX_DEPLOYMENT_TARGET} in
+	    10.[012])
+	      allow_undefined_flag='-Wl,-flat_namespace -Wl,-undefined -Wl,suppress'
+	      ;;
+	    10.*)
+	      allow_undefined_flag='-Wl,-undefined -Wl,dynamic_lookup'
+	      ;;
+	  esac
+	fi
       ;;
     esac
     output_verbose_link_cmd='echo'
@@ -13075,7 +13110,18 @@
       allow_undefined_flag_CXX='-undefined suppress'
       ;;
     *) # Darwin 1.3 on
-      test -z ${LD_TWOLEVEL_NAMESPACE} && allow_undefined_flag_CXX='-flat_namespace -undefined suppress'
+	if test -z ${MACOSX_DEPLOYMENT_TARGET} ; then
+	  allow_undefined_flag_CXX='-Wl,-flat_namespace -Wl,-undefined -Wl,suppress'
+	else
+	  case ${MACOSX_DEPLOYMENT_TARGET} in
+	    10.[012])
+	      allow_undefined_flag_CXX='-Wl,-flat_namespace -Wl,-undefined -Wl,suppress'
+	      ;;
+	    10.*)
+	      allow_undefined_flag_CXX='-Wl,-undefined -Wl,dynamic_lookup'
+	      ;;
+	  esac
+	fi
       ;;
     esac
     lt_int_apple_cc_single_mod=no
@@ -16898,7 +16944,18 @@
 	allow_undefined_flag_F77='-undefined suppress'
 	;;
       *) # Darwin 1.3 on
-	test -z ${LD_TWOLEVEL_NAMESPACE} && allow_undefined_flag_F77='-flat_namespace -undefined suppress'
+	if test -z ${MACOSX_DEPLOYMENT_TARGET} ; then
+	  allow_undefined_flag_F77='-Wl,-flat_namespace -Wl,-undefined -Wl,suppress'
+	else
+	  case ${MACOSX_DEPLOYMENT_TARGET} in
+	    10.[012])
+	      allow_undefined_flag_F77='-Wl,-flat_namespace -Wl,-undefined -Wl,suppress'
+	      ;;
+	    10.*)
+	      allow_undefined_flag_F77='-Wl,-undefined -Wl,dynamic_lookup'
+	      ;;
+	  esac
+	fi
 	;;
       esac
     	lt_int_apple_cc_single_mod=no
@@ -19180,7 +19237,18 @@
 	allow_undefined_flag_GCJ='-undefined suppress'
 	;;
       *) # Darwin 1.3 on
-	test -z ${LD_TWOLEVEL_NAMESPACE} && allow_undefined_flag_GCJ='-flat_namespace -undefined suppress'
+	if test -z ${MACOSX_DEPLOYMENT_TARGET} ; then
+	  allow_undefined_flag_GCJ='-Wl,-flat_namespace -Wl,-undefined -Wl,suppress'
+	else
+	  case ${MACOSX_DEPLOYMENT_TARGET} in
+	    10.[012])
+	      allow_undefined_flag_GCJ='-Wl,-flat_namespace -Wl,-undefined -Wl,suppress'
+	      ;;
+	    10.*)
+	      allow_undefined_flag_GCJ='-Wl,-undefined -Wl,dynamic_lookup'
+	      ;;
+	  esac
+	fi
 	;;
       esac
     	lt_int_apple_cc_single_mod=no
@@ -23040,6 +23108,9 @@
 	darwin*)
 	  libltdl_cv_sys_dlopen_deplibs=yes
 	  ;;
+	darwin*)
+	  libltdl_cv_sys_dlopen_deplibs=yes
+	  ;;
 	linux*)
 	  libltdl_cv_sys_dlopen_deplibs=yes
 	  ;;
@@ -43996,8 +44067,8 @@
 kde_save_libs="$LIBS"
 LIBS="$all_libraries $USER_LDFLAGS $kde_cups_libs"
 CFLAGS="$CFLAGS $all_includes $USER_INCLUDES"
-echo "$as_me:$LINENO: checking CUPS version >= 1.0120" >&5
-echo $ECHO_N "checking CUPS version >= 1.0120... $ECHO_C" >&6
+echo "$as_me:$LINENO: checking CUPS version >= 1.0121" >&5
+echo $ECHO_N "checking CUPS version >= 1.0121... $ECHO_C" >&6
 kde_save_link=$ac_link
 ac_link='${CC} $CFLAGS -o conftest conftest.c $LIBS 1>&5'
 cat > conftest.c <<EOF
@@ -44007,7 +44078,7 @@
 {
   if (CUPS_VERSION <
 EOF
-echo "1.0120" >> conftest.c
+echo "1.0121" >> conftest.c
 cat >> conftest.c <<EOF
 )
     printf("no\n");
@@ -46595,7 +46666,7 @@
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
   ac_check_lib_save_LIBS=$LIBS
-LIBS="-lqt-mt  $LIBS"
+LIBS="-lqt-mt -lXft $LIBS"
 cat >conftest.$ac_ext <<_ACEOF
 /* confdefs.h.  */
 _ACEOF
--- kdelibs-3.3.91/kdecore/Makefile.in	Fri Jan  7 07:48:59 2005
+++ kdelibs-3.3.91-new/kdecore/Makefile.in	Fri Jan 14 15:04:17 2005
@@ -653,7 +653,7 @@
 xdg_menudir = @xdg_menudir@
 INCLUDES = -I$(top_srcdir)/kdefx -I$(top_srcdir)/kdecore/network $(all_includes) $(LIBART_CFLAGS)
 @include_SVGICONS_TRUE@SVGICONS = svgicons
-@include_SVGICONS_TRUE@SVGICON_LIB = svgicons/libkdesvgicons.la
+@include_SVGICONS_TRUE@SVGICON_LIB = svgicons/libkdesvgicons.la $(LIBZ)
 SUBDIRS = malloc network $(SVGICONS) . kconfig_compiler tests
 AM_LDFLAGS = $(LDFLAGS_AS_NEEDED)
 lib_LTLIBRARIES = libkdefakes.la libkdecore.la
--- kdelibs-3.3.91/kdecore/kglobalsettings.cpp	Tue Oct 26 13:30:49 2004
+++ kdelibs-3.3.91-new/kdecore/kglobalsettings.cpp	Fri Jan 14 15:04:17 2005
@@ -460,7 +460,7 @@
       s_desktopPath->append('/');
 
     // Trash Path - TODO remove in KDE4 (kio_trash can't use it for interoperability reasons)
-    *s_trashPath = *s_desktopPath + i18n("Trash") + "/";
+    *s_trashPath = *s_desktopPath + "." + i18n("Trash") + "/";
     *s_trashPath = g.readPathEntry( "Trash" , *s_trashPath);
     *s_trashPath = QDir::cleanDirPath( *s_trashPath );
     if ( !s_trashPath->endsWith("/") )
--- kdelibs-3.3.91/kdecore/kstandarddirs.cpp	Mon Jan  3 09:53:18 2005
+++ kdelibs-3.3.91-new/kdecore/kstandarddirs.cpp	Fri Jan 14 15:04:17 2005
@@ -1020,7 +1020,7 @@
     if (!strcmp(type, "data"))
 	return "share/apps/";
     if (!strcmp(type, "html"))
-	return "share/doc/HTML/";
+	return "share/doc/kde/";
     if (!strcmp(type, "icon"))
 	return "share/icons/";
     if (!strcmp(type, "config"))
--- kdelibs-3.3.91/kdecore/network/kresolvermanager.cpp	Sat Dec  4 13:07:46 2004
+++ kdelibs-3.3.91-new/kdecore/network/kresolvermanager.cpp	Fri Jan 14 15:04:17 2005
@@ -31,6 +31,7 @@
 
 #ifdef HAVE_RES_INIT
 # include <sys/stat.h>
+# include <nameser.h>
 # include <resolv.h>
 # include <time.h>
 #endif
--- kdelibs-3.3.91/kdesu/su.cpp	Tue Apr 13 09:33:21 2004
+++ kdelibs-3.3.91-new/kdesu/su.cpp	Fri Jan 14 15:04:17 2005
@@ -74,13 +74,14 @@
 	setTerminal(true);
 
     QCStringList args;
+    args += "-c";
+    args += "staff";
     if ((m_Scheduler != SchedNormal) || (m_Priority > 50))
 	args += "root";
     else
 	args += m_User;
     args += "-c";
     args += QCString(__KDE_BINDIR) + "/kdesu_stub";
-    args += "-";
 
     QCString command = __PATH_SU;
     if (::access(__PATH_SU, X_OK) != 0)
--- kdelibs-3.3.91/kdeui/Makefile.in	Fri Jan  7 07:49:03 2005
+++ kdelibs-3.3.91-new/kdeui/Makefile.in	Fri Jan 14 15:04:18 2005
@@ -614,11 +614,11 @@
 lib_LTLIBRARIES = libkdeui.la libkspell.la
 #>- libkdeui_la_LDFLAGS = $(KDE_MT_LDFLAGS) -no-undefined -version-info 6:0:2
 #>+ 1
-libkdeui_la_LDFLAGS = $(KDE_MT_LDFLAGS) -no-undefined $(KDE_NO_UNDEFINED) -version-info 6:0:2
+libkdeui_la_LDFLAGS = $(KDE_MT_LDFLAGS) $(all_libraries) -no-undefined $(KDE_NO_UNDEFINED) -version-info 6:0:2
 libkdeui_la_LIBADD = ../kdecore/libkdecore.la
 #>- libkspell_la_LDFLAGS = $(KDE_MT_LDFLAGS) -version-info 6:0:2 -no-undefined
 #>+ 1
-libkspell_la_LDFLAGS = $(KDE_MT_LDFLAGS) -version-info 6:0:2 -no-undefined $(KDE_NO_UNDEFINED)
+libkspell_la_LDFLAGS = $(KDE_MT_LDFLAGS) $(all_libraries) -version-info 6:0:2 -no-undefined $(KDE_NO_UNDEFINED)
 libkspell_la_LIBADD = libkdeui.la
 libkspell_la_SOURCES = dummy.cpp
 picsdir = $(kde_datadir)/kdeui/pics
--- kdelibs-3.3.91/kinit/Makefile.in	Fri Jan  7 07:49:06 2005
+++ kdelibs-3.3.91-new/kinit/Makefile.in	Fri Jan 14 15:04:18 2005
@@ -596,7 +596,7 @@
 xdg_appsdir = @xdg_appsdir@
 xdg_directorydir = @xdg_directorydir@
 xdg_menudir = @xdg_menudir@
-INCLUDES = -I$(srcdir)/../libltdl/ $(all_includes) $(KDEINIT_XFT_INCLUDES)
+INCLUDES = -I$(srcdir)/../libltdl/ $(KDEINIT_XFT_INCLUDES) $(all_includes)
 SUBDIRS = . tests
 #>- lib_LTLIBRARIES = 
 #>+ 1
--- kdelibs-3.3.91/kio/kio/job.cpp	Mon Jan  3 09:53:25 2005
+++ kdelibs-3.3.91-new/kio/kio/job.cpp	Fri Jan 14 15:04:18 2005
@@ -3260,10 +3260,10 @@
                            f.close();
                            KSimpleConfig config( path );
                            config.setDesktopGroup();
-                           config.writePathEntry( QString::fromLatin1("URL"), (*it).uSource.url() );
-                           KURL urlName = (*it).uSource;
-                           urlName.setPass( "" );
-                           config.writeEntry( QString::fromLatin1("Name"), urlName.url() );
+                           KURL url = (*it).uSource;
+                           url.setPass( "" );
+                           config.writePathEntry( QString::fromLatin1("URL"), url.url() );
+                           config.writeEntry( QString::fromLatin1("Name"), url.url() );
                            config.writeEntry( QString::fromLatin1("Type"), QString::fromLatin1("Link") );
                            QString protocol = (*it).uSource.protocol();
                            if ( protocol == QString::fromLatin1("ftp") )
--- kdelibs-3.3.91/kio/kssl/kopenssl.cc	Tue May 11 04:31:45 2004
+++ kdelibs-3.3.91-new/kio/kssl/kopenssl.cc	Fri Jan 14 15:04:18 2005
@@ -322,6 +322,7 @@
              #elif defined(_AIX)
              << "libcrypto.a(libcrypto.so.0)"
 	     #elif defined(__APPLE__)
+	     << "libcrypto.0.9.7.dylib"
 	     << "libcrypto.dylib"
 	     << "libcrypto.0.9.dylib"
 	     #else
--- kdelibs-3.3.91/kio/misc/kpac/Makefile.in	Fri Jan  7 07:49:09 2005
+++ kdelibs-3.3.91-new/kio/misc/kpac/Makefile.in	Fri Jan 14 15:04:18 2005
@@ -533,7 +533,7 @@
                              downloader.cpp discovery.cpp proxyscout_skel.cpp
 
 kded_proxyscout_la_LDFLAGS = $(all_libraries) -module -avoid-version
-kded_proxyscout_la_LIBADD = $(LIB_KIO) $(top_builddir)/kjs/libkjs.la
+kded_proxyscout_la_LIBADD = $(LIB_KIO) $(top_builddir)/kjs/libkjs.la $(LIBRESOLV)
 kpac_dhcp_helper_SOURCES = kpac_dhcp_helper.c
 kpac_dhcp_helper_LDADD = $(LIBSOCKET)
 noinst_HEADERS = proxyscout.h script.h downloader.h discovery.h
