diff -uNr kdemultimedia-3.3.92/kioslave/audiocd/audiocd.cpp kdemultimedia-3.3.92-new/kioslave/audiocd/audiocd.cpp
--- kdemultimedia-3.3.92/kioslave/audiocd/audiocd.cpp	Thu Feb  3 17:48:25 2005
+++ kdemultimedia-3.3.92-new/kioslave/audiocd/audiocd.cpp	Mon Feb 14 15:19:11 2005
@@ -24,6 +24,8 @@
 
 #include <config.h>
 
+#undef VERSION
+
 extern "C"
 {
 #include <cdda_interface.h>
diff -uNr kdemultimedia-3.3.92/noatun/library/noatunarts/Makefile.am kdemultimedia-3.3.92-new/noatun/library/noatunarts/Makefile.am
--- kdemultimedia-3.3.92/noatun/library/noatunarts/Makefile.am	Thu Feb  3 17:48:27 2005
+++ kdemultimedia-3.3.92-new/noatun/library/noatunarts/Makefile.am	Mon Feb 14 15:18:30 2005
@@ -6,7 +6,7 @@
                            FFTScopes.cpp StereoEffectStack_impl.cpp \
                            StereoVolumeControl_impl.cpp Session_impl.cpp
 
-libnoatunarts_la_LDFLAGS = $(all_libraries) -avoid-version -no-undefined
+libnoatunarts_la_LDFLAGS = $(all_libraries) -no-undefined
 libnoatunarts_la_LIBADD =  -lkmedia2_idl -lsoundserver_idl -lartsflow
 libnoatunarts_la_METASOURCES = AUTO
 
diff -uNr kdemultimedia-3.3.92/noatun/modules/winskin/vis/Makefile.am kdemultimedia-3.3.92-new/noatun/modules/winskin/vis/Makefile.am
--- kdemultimedia-3.3.92/noatun/modules/winskin/vis/Makefile.am	Thu Feb  3 17:48:27 2005
+++ kdemultimedia-3.3.92-new/noatun/modules/winskin/vis/Makefile.am	Mon Feb 14 15:19:01 2005
@@ -14,7 +14,7 @@
 				realFFTFilter.cpp visQueue.cpp
 
 
-libwinskinvis_la_LDFLAGS=	$(all_libraries) -avoid-version \
+libwinskinvis_la_LDFLAGS=	$(all_libraries) \
 				-no-undefined
 
 libwinskinvis_la_LIBADD =	-lkmedia2_idl -lsoundserver_idl -lartsflow
diff -uNr kdemultimedia-3.3.92/oggvorbis_artsplugin/oggPlayObject_impl.h kdemultimedia-3.3.92-new/oggvorbis_artsplugin/oggPlayObject_impl.h
--- kdemultimedia-3.3.92/oggvorbis_artsplugin/oggPlayObject_impl.h	Sat Oct 26 21:10:28 2002
+++ kdemultimedia-3.3.92-new/oggvorbis_artsplugin/oggPlayObject_impl.h	Mon Feb 14 15:19:11 2005
@@ -1,7 +1,7 @@
 #ifndef OGGPLAYER_IMPL_H
 #define OGGPLAYER_IMPL_H "$Id$"
 
-#if defined(__GNU_LIBRARY__) && !defined(_SEM_SEMUN_UNDEFINED)
+#if (defined(__GNU_LIBRARY__) && !defined(_SEM_SEMUN_UNDEFINED)) || defined(__APPLE__)
 /* union semun is defined by including <sys/sem.h> */
 #else
 /* according to X/OPEN we have to define it ourselves */
