diff -uNr kdesdk-3.2.91/cervisia/cvsservice/Makefile.am kdesdk-3.2.91-new/cervisia/cvsservice/Makefile.am
--- kdesdk-3.2.91/cervisia/cvsservice/Makefile.am	2003-07-21 03:42:27.000000000 -0400
+++ kdesdk-3.2.91-new/cervisia/cvsservice/Makefile.am	2004-07-15 10:54:36.000000000 -0400
@@ -5,6 +5,8 @@
 #
 #
 
+CC=g++
+
 INCLUDES = $(all_includes)
 
 # cvs DCOP service
diff -uNr kdesdk-3.2.91/kbabel/kbabeldict/modules/dbsearchengine/configure.in.in kdesdk-3.2.91-new/kbabel/kbabeldict/modules/dbsearchengine/configure.in.in
--- kdesdk-3.2.91/kbabel/kbabeldict/modules/dbsearchengine/configure.in.in	2004-06-22 13:24:14.000000000 -0400
+++ kdesdk-3.2.91-new/kbabel/kbabeldict/modules/dbsearchengine/configure.in.in	2004-07-15 10:55:36.000000000 -0400
@@ -102,4 +102,7 @@
 
 AC_SUBST(LIB_DBIV) 
 
+# dunno why, dbsearchengine is breaking on OSX
+DBSEARCHENGINE=""
+
 AM_CONDITIONAL(include_DBSEARCHENGINE, test -n "$DBSEARCHENGINE")
diff -uNr kdesdk-3.2.91/kcachegrind/kcachegrind/sourceitem.h kdesdk-3.2.91-new/kcachegrind/kcachegrind/sourceitem.h
--- kdesdk-3.2.91/kcachegrind/kcachegrind/sourceitem.h	2004-05-22 16:58:52.000000000 -0400
+++ kdesdk-3.2.91-new/kcachegrind/kcachegrind/sourceitem.h	2004-07-15 10:54:36.000000000 -0400
@@ -25,6 +25,7 @@
 
 #include <qlistview.h>
 #include "tracedata.h"
+#undef fileno
 
 class SourceView;
 
diff -uNr kdesdk-3.2.91/umbrello/umbrello/classparser/Makefile.am kdesdk-3.2.91-new/umbrello/umbrello/classparser/Makefile.am
--- kdesdk-3.2.91/umbrello/umbrello/classparser/Makefile.am	2004-05-22 16:58:59.000000000 -0400
+++ kdesdk-3.2.91-new/umbrello/umbrello/classparser/Makefile.am	2004-07-15 10:54:36.000000000 -0400
@@ -1,4 +1,4 @@
-INCLUDES = -I../dialogs $(all_includes)
+INCLUDES = -I../dialogs -I@PREFIX@/lib/flex/include $(all_includes)
 
 noinst_LTLIBRARIES = libclassparser.la
 libclassparser_la_SOURCES = ast.cpp driver.cpp errors.cpp lexer.cpp lookup.cpp parser.cpp tree_parser.cpp urlutil.cpp ast_utils.cpp cpptree2uml.cpp
