diff -uNr kdesdk-3.3.92/cervisia/cvsservice/Makefile.am kdesdk-3.3.92-new/cervisia/cvsservice/Makefile.am
--- kdesdk-3.3.92/cervisia/cvsservice/Makefile.am	Fri Aug 27 11:24:38 2004
+++ kdesdk-3.3.92-new/cervisia/cvsservice/Makefile.am	Mon Feb 14 15:33:37 2005
@@ -30,7 +30,10 @@
 
 libcvsservice_la_LDFLAGS = $(all_libraries) -no-undefined -version-info 0:1
 libcvsservice_la_LIBADD = $(LIB_KDECORE)
-libcvsservice_la_SOURCES = cvsservice.stub cvsjob.stub repository.stub
+libcvsservice_la_SOURCES = cvsservice.stub cvsjob.stub repository.stub dummy.cpp
+
+dummy.cpp:
+	echo > dummy.cpp
 
 # let automoc handle all of the meta source files (moc)
 METASOURCES = AUTO
diff -uNr kdesdk-3.3.92/kcachegrind/kcachegrind/sourceitem.h kdesdk-3.3.92-new/kcachegrind/kcachegrind/sourceitem.h
--- kdesdk-3.3.92/kcachegrind/kcachegrind/sourceitem.h	Fri Apr  2 08:00:42 2004
+++ kdesdk-3.3.92-new/kcachegrind/kcachegrind/sourceitem.h	Mon Feb 14 15:34:18 2005
@@ -25,6 +25,7 @@
 
 #include <qlistview.h>
 #include "tracedata.h"
+#undef fileno
 
 class SourceView;
 
diff -uNr kdesdk-3.3.92/umbrello/umbrello/classparser/Makefile.am kdesdk-3.3.92-new/umbrello/umbrello/classparser/Makefile.am
--- kdesdk-3.3.92/umbrello/umbrello/classparser/Makefile.am	Wed Apr 14 14:44:53 2004
+++ kdesdk-3.3.92-new/umbrello/umbrello/classparser/Makefile.am	Mon Feb 14 15:34:10 2005
@@ -1,4 +1,4 @@
-INCLUDES = -I../dialogs $(all_includes)
+INCLUDES = -I../dialogs -I@FINKPREFIX@/lib/flex/include $(all_includes)
 
 noinst_LTLIBRARIES = libclassparser.la
 libclassparser_la_SOURCES = ast.cpp driver.cpp errors.cpp lexer.cpp lookup.cpp parser.cpp tree_parser.cpp urlutil.cpp ast_utils.cpp cpptree2uml.cpp
