diff -uNr koffice-1.4.0a/kexi/formeditor/Makefile.am koffice-1.4.0a-new/kexi/formeditor/Makefile.am
--- koffice-1.4.0a/kexi/formeditor/Makefile.am	2005-05-09 10:10:14.000000000 -0400
+++ koffice-1.4.0a-new/kexi/formeditor/Makefile.am	2005-06-29 15:23:28.000000000 -0400
@@ -1,4 +1,5 @@
 include $(top_srcdir)/kexi/Makefile.global
+USE_EXCEPTIONS = -fexceptions -fno-coalesce
 
 INCLUDES = -I$(top_srcdir)/kexi/widget \
   -I$(top_srcdir)/kexi/widget/propertyeditor \
diff -uNr koffice-1.4.0a/kexi/kexidb/drivers/configure.in.in koffice-1.4.0a-new/kexi/kexidb/drivers/configure.in.in
--- koffice-1.4.0a/kexi/kexidb/drivers/configure.in.in	2005-05-09 10:10:02.000000000 -0400
+++ koffice-1.4.0a-new/kexi/kexidb/drivers/configure.in.in	2005-06-29 20:24:12.000000000 -0400
@@ -142,8 +142,8 @@
 if test "$compile_pgsql_plugin" = "yes"; then
   AC_MSG_CHECKING([for PostgreSQL... Step 2 of 4])
   pgsql_libdirs="$ac_pgsql_libdir /usr/lib /usr/local/lib /usr/local/pgsql/lib /opt/postgres/lib"
-  AC_FIND_FILE(libpq.so, $pgsql_libdirs, pgsql_libdir)
-  if test ! -r $pgsql_libdir/libpq.so; then
+  AC_FIND_FILE(libpq.dylib, $pgsql_libdirs, pgsql_libdir)
+  if test ! -r $pgsql_libdir/libpq.dylib; then
     compile_pgsql_plugin="no"
     AC_MSG_RESULT([not found])
   else
@@ -160,8 +160,8 @@
 if test "$compile_pgsql_plugin" = "yes"; then
   AC_MSG_CHECKING([for PostgreSQL... Step 3 of 4])
   pqxx_incdirs="$ac_pqxx_incdir /usr/include /usr/local/pqxx/include /usr/local/include"
-  AC_FIND_FILE(pqxx/pqxx, $pqxx_incdirs, pqxx_incdir)
-  if test ! -r $pqxx_incdir/pqxx/pqxx; then
+  AC_FIND_FILE(pqxx, $pqxx_incdirs, pqxx_incdir)
+  if test ! -r $pqxx_incdir/pqxx; then
     compile_pgsql_plugin="no"
     AC_MSG_RESULT([not found])
   else
@@ -178,7 +178,7 @@
   AC_MSG_CHECKING([for PostgreSQL... Step 4 of 4])
   pqxx_libdirs="$ac_pqxx_libdir /usr/lib /usr/local/lib /usr/local/pqxx/lib"
   AC_FIND_FILE(libpqxx.so, $pqxx_libdirs, pqxx_libdir)
-  if test ! -r $pqxx_libdir/libpqxx.so; then
+  if test ! -r $pqxx_libdir/libpqxx.2.dylib; then
     compile_pgsql_plugin="no"
     AC_MSG_RESULT([not found])
   else
diff -uNr koffice-1.4.0a/kexi/kexidb/drivers/mySQL/Makefile.am koffice-1.4.0a-new/kexi/kexidb/drivers/mySQL/Makefile.am
--- koffice-1.4.0a/kexi/kexidb/drivers/mySQL/Makefile.am	2005-05-09 10:10:01.000000000 -0400
+++ koffice-1.4.0a-new/kexi/kexidb/drivers/mySQL/Makefile.am	2005-06-29 15:23:28.000000000 -0400
@@ -20,7 +20,8 @@
   $(LIB_QT) \
   $(MYSQL_LIBS) \
   -lmysqlclient \
-  ../../libkexidb.la
+  ../../libkexidb.la \
+  ../../parser/libkexidbparser.la
 
 kexidb_mysqldriver_la_LDFLAGS = $(all_libraries) -module $(KDE_PLUGIN) $(VER_INFO) -no-undefined
 
diff -uNr koffice-1.4.0a/kexi/kexidb/drivers/pqxx/Makefile.am koffice-1.4.0a-new/kexi/kexidb/drivers/pqxx/Makefile.am
--- koffice-1.4.0a/kexi/kexidb/drivers/pqxx/Makefile.am	2005-05-09 10:10:00.000000000 -0400
+++ koffice-1.4.0a-new/kexi/kexidb/drivers/pqxx/Makefile.am	2005-06-29 20:03:29.000000000 -0400
@@ -8,7 +8,7 @@
 
 kexidb_pqxxsqldriver_la_SOURCES = pqxxdriver.cpp pqxxcursor.cpp pqxxconnection.cpp pqxxkeywords.cpp
 
-kexidb_pqxxsqldriver_la_LIBADD = $(LIB_KPARTS) $(LIB_QT) -lpqxx ../../libkexidb.la
+kexidb_pqxxsqldriver_la_LIBADD = $(LIB_KPARTS) $(LIB_QT) -lpqxx.2 ../../libkexidb.la
 
 kexidb_pqxxsqldriver_la_LDFLAGS = $(all_libraries) -module $(KDE_PLUGIN) -L$(PQXX_LIBDIR) -L$(PG_LIBDIR) $(VER_INFO) -no-undefined
 
diff -uNr koffice-1.4.0a/kexi/main/Makefile.am koffice-1.4.0a-new/kexi/main/Makefile.am
--- koffice-1.4.0a/kexi/main/Makefile.am	2005-05-09 10:09:50.000000000 -0400
+++ koffice-1.4.0a-new/kexi/main/Makefile.am	2005-06-29 15:23:28.000000000 -0400
@@ -1,4 +1,5 @@
 include $(top_srcdir)/kexi/Makefile.global
+USE_EXCEPTIONS = -fexceptions -fno-coalesce
 
 lib_LTLIBRARIES = libkeximain.la
 libkeximain_la_SOURCES = \
diff -uNr koffice-1.4.0a/kexi/main/configure.in.in koffice-1.4.0a-new/kexi/main/configure.in.in
--- koffice-1.4.0a/kexi/main/configure.in.in	2005-05-09 10:09:50.000000000 -0400
+++ koffice-1.4.0a-new/kexi/main/configure.in.in	2005-06-29 20:04:23.000000000 -0400
@@ -20,7 +20,7 @@
 AC_FIND_FILE(knewstuff/downloaddialog.h, $kde_incdirs, knewstuff_incdir)
 ac_knewstuff_includes="$knewstuff_incdir"
 
-AC_FIND_FILE(libknewstuff.so, $kde_libdirs, knewstuff_libdir)
+AC_FIND_FILE(libknewstuff.dylib, $kde_libdirs, knewstuff_libdir)
 ac_knewstuff_libraries="$knewstuff_libdir"
 
 if test "$ac_knewstuff_includes" = NO || test "$ac_knewstuff_libraries" = NO; then
diff -uNr koffice-1.4.0a/kexi/main/startup/Makefile.am koffice-1.4.0a-new/kexi/main/startup/Makefile.am
--- koffice-1.4.0a/kexi/main/startup/Makefile.am	2005-05-09 10:09:49.000000000 -0400
+++ koffice-1.4.0a-new/kexi/main/startup/Makefile.am	2005-06-29 15:23:28.000000000 -0400
@@ -1,4 +1,5 @@
 include $(top_srcdir)/kexi/Makefile.global
+USE_EXCEPTIONS = -fexceptions -fno-coalesce
 
 noinst_LTLIBRARIES = libkeximainstartup.la
 
diff -uNr koffice-1.4.0a/kexi/widget/relations/Makefile.am koffice-1.4.0a-new/kexi/widget/relations/Makefile.am
--- koffice-1.4.0a/kexi/widget/relations/Makefile.am	2005-05-09 10:09:48.000000000 -0400
+++ koffice-1.4.0a-new/kexi/widget/relations/Makefile.am	2005-06-29 15:23:28.000000000 -0400
@@ -1,4 +1,5 @@
 include $(top_srcdir)/kexi/Makefile.global
+USE_EXCEPTIONS = -fexceptions -fno-coalesce
 
 lib_LTLIBRARIES = libkexirelationsview.la
 
