
--- dbus-0.60/qt/Makefile.in	2005-11-30 16:02:22.000000000 -0500
+++ dbus-0.60-new/qt/Makefile.in	2005-12-12 11:02:35.000000000 -0500
@@ -60,7 +60,7 @@
 LTLIBRARIES = $(lib_LTLIBRARIES)
 am__DEPENDENCIES_1 =
 @HAVE_QT_TRUE@libdbus_qt4_1_la_DEPENDENCIES = $(am__DEPENDENCIES_1) \
-@HAVE_QT_TRUE@	$(top_builddir)/dbus/libdbus-1.la
+@HAVE_QT_TRUE@	
 am__libdbus_qt4_1_la_SOURCES_DIST =  \
 	$(top_srcdir)/qt/qdbusconnection.cpp \
 	$(top_srcdir)/qt/qdbuserror.cpp \
@@ -388,7 +388,7 @@
 @HAVE_QT_TRUE@	$(top_srcdir)/qt/qdbusvariant.h
 
 @HAVE_QT_TRUE@CLEANFILES = qdbusserver.moc qdbusconnection.moc
-@HAVE_QT_TRUE@libdbus_qt4_1_la_LIBADD = $(DBUS_QT_LIBS) $(top_builddir)/dbus/libdbus-1.la
+@HAVE_QT_TRUE@libdbus_qt4_1_la_LIBADD = $(DBUS_QT_LIBS) -ldbus-1
 @HAVE_QT_TRUE@libdbus_qt4_1_la_LDFLAGS = -version-info 1:0 -no-undefined
 all: all-am
 
