diff -uNr libpqxx-2.5.2/configure libpqxx-2.5.2-new/configure
--- libpqxx-2.5.2/configure	Tue May 10 07:11:29 2005
+++ libpqxx-2.5.2-new/configure	Wed May 11 13:52:37 2005
@@ -20248,7 +20248,8 @@
 { echo "$as_me:$LINENO: using PostgreSQL libraries at $with_postgres_lib" >&5
 echo "$as_me: using PostgreSQL libraries at $with_postgres_lib" >&6;}
 
-
+with_postgres_include="@INSTPREFIX@/include/postgresql"
+with_postgres_lib="@INSTPREFIX@/lib"
 
 POSTGRES_LIB="-L${with_postgres_lib} -R${with_postgres_lib} -lpq"
 
@@ -21454,7 +21455,7 @@
 _ACEOF
 rm -f conftest.$ac_objext
 if { (eval echo "$as_me:$LINENO: \"$ac_compile\"") >&5
-  (eval $ac_compile) 2>conftest.er1
+  (eval $ac_compile -Werror) 2>conftest.er1
   ac_status=$?
   grep -v '^ *+' conftest.er1 >conftest.err
   rm -f conftest.er1
@@ -21503,7 +21504,7 @@
 _ACEOF
 rm -f conftest.$ac_objext
 if { (eval echo "$as_me:$LINENO: \"$ac_compile\"") >&5
-  (eval $ac_compile) 2>conftest.er1
+  (eval $ac_compile -Werror) 2>conftest.er1
   ac_status=$?
   grep -v '^ *+' conftest.er1 >conftest.err
   rm -f conftest.er1
diff -uNr libpqxx-2.5.2/src/Makefile.in libpqxx-2.5.2-new/src/Makefile.in
--- libpqxx-2.5.2/src/Makefile.in	Tue May 10 07:11:26 2005
+++ libpqxx-2.5.2-new/src/Makefile.in	Wed May 11 13:46:24 2005
@@ -214,9 +214,9 @@
 
 
 # Uncomment to build an unversioned library (version in soname)
-libpqxx_version = -release $(PQXXVERSION)
+#libpqxx_version = -release $(PQXXVERSION)
 # Uncomment to build a versioned library
-#libpqxx_version = -version-info $(ABI_CURRENT):$(ABI_REVISION):$(ABI_AGE)
+libpqxx_version = -version-info 7:2:5
 libpqxx_la_LDFLAGS = $(libpqxx_version) \
 	-rpath $(libdir) \
 	${POSTGRES_LIB}
