diff -uNr sqlite/configure sqlite-patched/configure
--- sqlite/configure	Sun Feb 29 10:18:31 2004
+++ sqlite-patched/configure	Wed Aug 25 14:42:15 2004
@@ -19982,7 +19982,7 @@
 
 fi
 if test "$found" = "no"; then
-  for dir in /usr/local /usr/X11 /usr/X11R6 /usr/pkg /usr/contrib /usr; do
+  for dir in /sw /usr/local /usr/X11 /usr/X11R6 /usr/pkg /usr/contrib /usr; do
     as_ac_File=`echo "ac_cv_file_$dir/include/tcl.h" | $as_tr_sh`
 echo "$as_me:$LINENO: checking for $dir/include/tcl.h" >&5
 echo $ECHO_N "checking for $dir/include/tcl.h... $ECHO_C" >&6
@@ -20156,7 +20156,7 @@
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
   ac_check_lib_save_LIBS=$LIBS
-LIBS="-lreadline  $LIBS"
+LIBS="-L/sw/lib -lreadline  $LIBS"
 cat >conftest.$ac_ext <<_ACEOF
 /* confdefs.h.  */
 _ACEOF
@@ -20218,7 +20218,7 @@
 #define HAVE_LIBREADLINE 1
 _ACEOF
 
-  LIBS="-lreadline $LIBS"
+  LIBS="-L/sw/lib -lreadline $LIBS"
 
 fi
 
@@ -20385,7 +20385,7 @@
 
 fi
 if test "$found" = "no"; then
-  for dir in /usr /usr/local /usr/local/readline /usr/contrib /mingw; do
+  for dir in /sw /usr /usr/local /usr/local/readline /usr/contrib /mingw; do
     as_ac_File=`echo "ac_cv_file_$dir/include/readline.h" | $as_tr_sh`
 echo "$as_me:$LINENO: checking for $dir/include/readline.h" >&5
 echo $ECHO_N "checking for $dir/include/readline.h... $ECHO_C" >&6
@@ -20435,7 +20435,7 @@
 fi
 
     if test "$found" = "yes"; then
-      TARGET_READLINE_INC="-I$dir/include/readline"
+      TARGET_READLINE_INC="-I$dir/include/readline -I$dir/include"
       break
     fi
   done
