--- /dev/null	Thu Nov 20 11:37:34 2003
+++ Makefile	Thu Nov 20 11:39:11 2003
@@ -0,0 +1,17 @@
+DEBUG = -g
+CXX = g++-3.3
+CPPFLAGS = -I%p/include
+CXXFLAGS = -Os -Wall --pipe -fvtable-thunks
+LDFLAGS = -L%p/lib
+LIBS = -lxbase
+
+all:  sample1 sample2 sample3 sample4 sample5 exfilter
+
+sample%%:
+	$(CXX) -o $@ $(CPPFLAGS) $(CXXFLAGS) $@.cpp $(LDFLAGS) $(LIBS)
+
+exfilter:
+	$(CXX) -o $@ $(CPPFLAGS) $(CXXFLAGS) $@.cpp $(LDFLAGS) $(LIBS)
+
+clean:
+	rm -rf sample1 sample2 sample3 sample4 sample5 exfilter
