diff -uNr xbsql-0.11/configure xbsql-0.11-patched/configure
--- xbsql-0.11/configure	Fri Mar 28 15:03:29 2003
+++ xbsql-0.11-patched/configure	Thu Nov 20 11:01:06 2003
@@ -6974,11 +6974,11 @@
 	echo "$as_me:6974: checking for (n)curses" >&5
 echo $ECHO_N "checking for (n)curses... $ECHO_C" >&6
 
-	if      test -f /usr/lib/libncurses.so
+	if      test -f @PREFIX@/lib/libncurses.dylib
 	then
 		curses=ncurses
 
-	elif    test -f /usr/lib/libcurses.so
+	elif    test -f @PREFIX@/lib/libcurses.so
 	then
 		curses=curses
 	else
diff -uNr xbsql-0.11/xbsql/xbsql.cpp xbsql-0.11-patched/xbsql/xbsql.cpp
--- xbsql-0.11/xbsql/xbsql.cpp	Fri Mar 28 14:07:31 2003
+++ xbsql-0.11-patched/xbsql/xbsql.cpp	Thu Nov 20 11:08:36 2003
@@ -16,6 +16,7 @@
 
 
 
+#include <sys/types.h>
 #ifndef 	_WIN32
 #include	<dirent.h>
 #include	<unistd.h>
@@ -26,7 +27,6 @@
 
 #include	<stdio.h>
 #include	<stdarg.h>
-#include	<sys/types.h>
 
 #include	<errno.h>
 #include	<ctype.h>
