diff -ruN -x '*~' auctex-11.82/fink/install auctex-11.82.patched/fink/install
--- auctex-11.82/fink/install	1969-12-31 19:00:00.000000000 -0500
+++ auctex-11.82.patched/fink/install	2005-12-19 14:04:31.000000000 -0500
@@ -0,0 +1,138 @@
+#! /bin/bash -e
+#
+# <% elisp %> emacs install script for auctex
+#
+# Copyright (C) 1997, 98, 99, 2000, 01, 02, 03, 04, 05
+# by Davide G. M. Salvetti for Debian version
+#
+# Copyright (C) 2006 Sebastien Maret for Fink version
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License as published by the
+# Free Software Foundation; either version 2 of the License, or (at your
+# option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to: The Free Software Foundation, Inc.,
+# 51 Franklin St, Fifth Floor, Boston, MA 02110-1301, USA.
+#
+# On Debian GNU/Linux System you can find a copy of the GNU General Public
+# License in "/usr/share/common-licenses/GPL".
+
+set -o posix
+
+FLAVOR=${1}
+
+INITSRC="auctex.el preview-latex.el tex-site.el"
+
+AUCTEXSRC="tex.el tex-buf.el tex-style.el latex.el tex-info.el \
+    texmathp.el multi-prompt.el tex-mik.el font-latex.el tex-font.el \
+    context.el context-en.el context-nl.el tex-fold.el \
+    toolbar-x.el tex-bar.el bib-cite.el tex-fptex.el"
+
+STYLESRC="style/prosper.el \
+    style/slides.el    style/foils.el     style/amstex.el \
+    style/article.el   style/book.el      style/letter.el \
+    style/report.el    style/amsart.el    style/amsbook.el \
+    style/epsf.el      style/psfig.el     style/latexinfo.el \
+    style/dutch.el     style/german.el    style/dk.el \
+    style/j-article.el style/j-book.el    style/j-report.el \
+    style/jarticle.el  style/jbook.el     style/jreport.el \
+    style/dinbrief.el  style/virtex.el    style/plfonts.el \
+    style/plhb.el      style/harvard.el	  style/swedish.el \
+    style/danish.el    style/slovak.el    style/czech.el \
+    style/amsmath.el   style/amstext.el   style/amsbsy.el \
+    style/amsopn.el    style/amsthm.el	  style/natbib.el \
+    style/index.el     style/makeidx.el   style/multind.el \
+    style/varioref.el  style/fancyref.el  style/mdwlist.el \
+    style/ngerman.el   style/graphicx.el  style/graphics.el \
+    style/verbatim.el  style/scrbase.el   style/scrartcl.el \
+    style/scrbook.el   style/scrreprt.el  style/scrlttr2.el \
+    style/scrpage2.el  style/captcont.el  style/subfigure.el \
+    style/paralist.el  style/booktabs.el  style/emp.el \
+    style/jura.el      style/alphanum.el  style/beamer.el \
+    style/alltt.el     style/italian.el   style/doc.el \
+    style/ltxdoc.el    style/ltx-base.el  style/units.el \
+    style/nicefrac.el  style/url.el       style/listings.el \
+    style/jurabib.el   style/csquotes.el  style/jsarticle.el \
+    style/jsbook.el    style/babel.el     style/dk-bib.el \
+    style/inputenc.el  style/frenchb.el   style/francais.el \
+    style/MinionPro.el"
+
+PREVIEWSRC="preview.el  prv-emacs.el"
+
+ELCC="${FLAVOR} -batch -q -no-site-file -no-init-file -l lpath.el"
+
+do_install () {
+    echo >&2 -n "auctex/install: Setting up for ${FLAVOR}... "
+    install -d @PREFIX@/share/${FLAVOR}/site-lisp
+    install -d @PREFIX@/share/${FLAVOR}/site-lisp/auctex
+    install -d @PREFIX@/share/${FLAVOR}/site-lisp/auctex/style
+    ln -s @PREFIX@/share/emacs/site-lisp/auctex/images \
+	@PREFIX@/share/${FLAVOR}/site-lisp/auctex \
+	>> @PREFIX@/share/${FLAVOR}/site-lisp/auctex//CompilationLog 2>&1
+    install -m 644 @PREFIX@/share/emacs/site-lisp/auctex.el \
+	@PREFIX@/share/emacs/site-lisp/preview-latex.el \
+	@PREFIX@/share/emacs/site-lisp/tex-site.el \
+	@PREFIX@/share/${FLAVOR}/site-lisp \
+	>> @PREFIX@/share/${FLAVOR}/site-lisp/auctex//CompilationLog 2>&1
+    install -m 644 @PREFIX@/share/emacs/site-lisp/auctex/lpath.el \
+	@PREFIX@/share/${FLAVOR}/site-lisp/auctex \
+	>> @PREFIX@/share/${FLAVOR}/site-lisp/auctex//CompilationLog 2>&1
+    (umask 022 && \
+	touch @PREFIX@/share/${FLAVOR}/site-lisp/auctex/style/.nosearch)
+    for i in ${STYLESRC}
+    do install -m 644 @PREFIX@/share/emacs/site-lisp/auctex/${i} \
+	@PREFIX@/share/${FLAVOR}/site-lisp/auctex/style
+    done
+    for i in ${AUCTEXSRC} ${PREVIEWSRC}
+    do install -m 644 @PREFIX@/share/emacs/site-lisp/auctex/${i} \
+	@PREFIX@/share/${FLAVOR}/site-lisp/auctex
+    done
+    cd @PREFIX@/share/${FLAVOR}/site-lisp
+#    ${ELCC} -f batch-byte-compile ${INITSRC} \
+#	>> @PREFIX@/share/${FLAVOR}/site-lisp/auctex//CompilationLog 2>&1
+    cd @PREFIX@/share/${FLAVOR}/site-lisp/auctex
+    ${ELCC} -f batch-byte-compile ${AUCTEXSRC} ${PREVIEWSRC} ${STYLESRC} \
+	>> @PREFIX@/share/${FLAVOR}/site-lisp/auctex//CompilationLog 2>&1
+#    for i in ${INITSRC}
+#    do rm -f @PREFIX@/share/${FLAVOR}/site-lisp/${i} \
+#	>> @PREFIX@/share/${FLAVOR}/site-lisp/auctex//CompilationLog 2>&1
+#    done
+    find @PREFIX@/share/${FLAVOR}/site-lisp/auctex/ -type f -name \*.el \
+	-print0 | xargs --null rm -f ,dummy, \
+	>> @PREFIX@/share/${FLAVOR}/site-lisp/auctex//CompilationLog 2>&1
+    rm -f @PREFIX@/share/${FLAVOR}/site-lisp/auctex/lpath.el \
+	@PREFIX@/share/${FLAVOR}/site-lisp/preview/lpath.el \
+	>> @PREFIX@/share/${FLAVOR}/site-lisp/auctex//CompilationLog 2>&1
+    gzip -9fq @PREFIX@/share/${FLAVOR}/site-lisp/auctex//CompilationLog
+    echo >&2 "done."
+    return 0
+}
+
+do_auto () {
+    LOGFILE="@PREFIX@/var/auctex/UpdateLog"
+    echo >&2 -n "update-auctex-elisp: "
+    echo >&2 "Further output will appear in: ${LOGFILE}.gz."
+    echo >&2 -n "auctex: "
+    echo >&2 -n "Waiting for update-auctex-elisp to terminate... "
+    @PREFIX@/sbin/update-auctex-elisp ${FLAVOR} > ${LOGFILE} 2>&1
+    gzip -9fq ${LOGFILE}
+    echo >&2 "done."
+    return 0
+}
+
+case "${FLAVOR}" in
+    (emacs) : ;;
+    (emacs21|emacs22)
+    do_install ${FLAVOR}
+    do_auto ${FLAVOR} ;;
+    (*) echo >&2 "auctex/install: Ignoring emacsen flavor ${FLAVOR}." ;;
+esac
+
+exit 0
diff -ruN -x '*~' auctex-11.82/fink/remove auctex-11.82.patched/fink/remove
--- auctex-11.82/fink/remove	1969-12-31 19:00:00.000000000 -0500
+++ auctex-11.82.patched/fink/remove	2005-12-19 13:54:13.000000000 -0500
@@ -0,0 +1,68 @@
+#! /bin/bash -e
+#
+# <% elisp %> emacs remove script for auctex
+#
+# Copyright (C) 1997, 98, 99, 2000, 01, 02, 03, 04, 05
+# by Davide G. M. Salvetti for Debian version
+#
+# Copyright (C) 2006 Sebastien Maret for Fink version
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License as published by the
+# Free Software Foundation; either version 2 of the License, or (at your
+# option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to: The Free Software Foundation, Inc.,
+# 51 Franklin St, Fifth Floor, Boston, MA 02110-1301, USA.
+#
+# On Debian GNU/Linux System you can find a copy of the GNU General Public
+# License in "/usr/share/common-licenses/GPL".
+
+set -o posix
+
+FLAVOR=${1}
+
+undo_install () {
+    echo >&2 -n "auctex/remove: Removing for ${FLAVOR}... "
+    rm -f @PREFIX@/share/${FLAVOR}/site-lisp/auctex//../tex-site.el
+    rm -f @PREFIX@/share/${FLAVOR}/site-lisp/auctex//../auctex.el
+    rm -f @PREFIX@/share/${FLAVOR}/site-lisp/auctex//../preview-latex.el
+    rm -f @PREFIX@/share/${FLAVOR}/site-lisp/auctex//CompilationLog{,.gz}
+    rm -f @PREFIX@/share/${FLAVOR}/site-lisp/auctex//*.elc
+    rm -f @PREFIX@/share/${FLAVOR}/site-lisp/auctex//style//*.elc
+    rm -f @PREFIX@/share/${FLAVOR}/site-lisp/auctex//style//.nosearch
+    rm -f @PREFIX@/share/${FLAVOR}/site-lisp/auctex/images
+    rm -f @PREFIX@/share/${FLAVOR}/site-lisp/preview//*.elc
+    test -d @PREFIX@/share/${FLAVOR}/site-lisp/auctex//style/ && \
+	rmdir @PREFIX@/share/${FLAVOR}/site-lisp/auctex//style/ || true
+    test -d @PREFIX@/share/${FLAVOR}/site-lisp/auctex/ && \
+	rmdir @PREFIX@/share/${FLAVOR}/site-lisp/auctex/ || true
+    test -d @PREFIX@/share/${FLAVOR}/site-lisp/preview/ && \
+	rmdir @PREFIX@/share/${FLAVOR}/site-lisp/preview/ || true
+    echo >&2 "done."
+    return 0
+}
+
+undo_auto () {
+    rm -rf @PREFIX@/var/lib/auctex//${FLAVOR}/
+    return 0
+}
+
+case "${FLAVOR}" in
+    (emacs) : ;;
+    (emacs21|emacs22)
+        undo_auto
+        undo_install
+    ;;
+    (*) echo >&2 "auctex/remove:" \
+	"Ignoring emacsen flavor: \"${FLAVOR}\""
+    ;;
+esac
+
+exit 0
diff -ruN -x '*~' auctex-11.82/fink/update-auctex-elisp auctex-11.82.patched/fink/update-auctex-elisp
--- auctex-11.82/fink/update-auctex-elisp	1969-12-31 19:00:00.000000000 -0500
+++ auctex-11.82.patched/fink/update-auctex-elisp	2005-12-18 12:53:02.000000000 -0500
@@ -0,0 +1,38 @@
+#! /bin/bash -e
+#
+# Update AUCTeX auto-loads.
+# Installed by the Debian auctex package.
+#
+# Copyright (C) 1997, 98, 99, 2000, 01, 02, 03, 04, 05
+# by Davide G. M. Salvetti.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License as published by the
+# Free Software Foundation; either version 2 of the License, or (at your
+# option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to: The Free Software Foundation, Inc.,
+# 51 Franklin St, Fifth Floor, Boston, MA 02110-1301, USA.
+#
+# On Debian GNU/Linux System you can find a copy of the GNU General Public
+# License in "/usr/share/common-licenses/GPL".
+
+set -o posix
+
+FLAVORS=${*:-"emacs21 emacs22"}
+
+for FLAVOR in ${FLAVORS}; do
+    if [ -x @PREFIX@/bin/${FLAVOR} ]; then
+	${FLAVOR} --no-site-file --no-init-file --batch \
+		  --load=@PREFIX@/share/${FLAVOR}/site-lisp/auctex//../tex-site.el \
+		  --funcall=TeX-auto-generate-global
+    fi
+done
+
+rm -f $(find @PREFIX@/var/lib/auctex/ -type f -name \*.el)
diff -ruN -x '*~' auctex-11.82/fink/update-auctex-elisp.8 auctex-11.82.patched/fink/update-auctex-elisp.8
--- auctex-11.82/fink/update-auctex-elisp.8	1969-12-31 19:00:00.000000000 -0500
+++ auctex-11.82.patched/fink/update-auctex-elisp.8	2005-12-18 12:53:02.000000000 -0500
@@ -0,0 +1,60 @@
+.\" Emacs, you know, this is an -*- nroff -*- source file.
+.\"
+.\" Man page for update-auctex-elisp.
+.\" Copyright (C) 1998, 99, 2000, 01, 04, 05
+.\" by Davide Giovanni Maria Salvetti.
+.\"
+.\" This manual page is free software; you can redistribute it and/or
+ \" modify it under the terms of the GNU General Public License as published
+ \" by the Free Software Foundation; either version 2 of the License, or (at
+ \" your option) any later version.
+.\"
+.\" This manual page is distributed in the hope that it will be useful,
+ \" but WITHOUT ANY WARRANTY; without even the implied warranty of
+ \" MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General
+ \" Public License for more details.
+.\"
+.\" You should have received a copy of the GNU General Public License along
+ \" with this manual page; if not, write to: The Free Software Foundation,
+ \" Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307, USA.
+.\"
+.\" On Debian GNU/Linux System you can find a copy of the GNU General
+ \" Public License in "/usr/share/common-licenses/GPL".
+.\"
+.TH Update\-AUCTeX\-elisp 8 "17 February 1998" "Debian Project" "Debian GNU/Linux"
+.SH NAME
+update\-auctex\-elisp \- update AUCTeX automatically generated TeX macros and
+LaTeX styles informations.
+.SH SYNOPSIS
+.B update\-auctex\-elisp
+.RI [ FLAVOR .\|.\|. ]
+.SH DESCRIPTION
+The
+.B update\-auctex\-elisp
+script updates the automatically generated TeX and LaTeX informations used by
+AUCTeX and stored in
+.B @PREFIX@/var/lib/auctex/.
+.PP
+It should be called after each major (La)TeX style files installation or
+upgrade.
+.PP
+If no arguments are specified, update\-auctex\-elisp caters to every supported
+Emacs flavor installed on the system; else it updates only the specified
+flavors, if they are installed.
+.PP
+On Debian GNU/Linux system
+.B update\-auctex\-elisp
+is called weekly by a
+.BR cron (8)
+driven mechanism, to ensure up-to-date informations.
+.SH BUGS
+None known.
+.SH FILES
+.B /var/lib/auctex/:
+the hierarchy containing the byte-compiled files.
+.SH SEE ALSO
+.BR emacs (1),
+.BR tex (1),
+.BR latex (1).
+.SH AUTHOR
+Davide G. M. Salvetti <salve@debian.org>.
diff -ruN -x '*~' auctex-11.82/fink/update-auctex-install auctex-11.82.patched/fink/update-auctex-install
--- auctex-11.82/fink/update-auctex-install	1969-12-31 19:00:00.000000000 -0500
+++ auctex-11.82.patched/fink/update-auctex-install	2005-12-18 12:53:02.000000000 -0500
@@ -0,0 +1,36 @@
+#! /bin/bash -e
+#
+# Update AUCTeX installation.
+# Installed by the Debian auctex package.
+#
+# Copyright (C) 1997, 98, 99, 2000, 01, 02, 03, 04, 05
+# by Davide G. M. Salvetti.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License as published by the
+# Free Software Foundation; either version 2 of the License, or (at your
+# option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to: The Free Software Foundation, Inc.,
+# 51 Franklin St, Fifth Floor, Boston, MA 02110-1301, USA.
+#
+# On Debian GNU/Linux System you can find a copy of the GNU General Public
+# License in "/usr/share/common-licenses/GPL".
+
+set -o posix
+
+FLAVORS=${*:-"emacs21 emacs22"}
+
+for FLAVOR in ${FLAVORS}; do
+    if [ -x @PREFIX@/bin/${FLAVOR} ]; then
+	@PREFIX@/lib/emacsen-common/packages//remove/auctex ${FLAVOR}
+	@PREFIX@/lib/emacsen-common/packages//install/auctex ${FLAVOR}
+    fi
+done
+
diff -ruN -x '*~' auctex-11.82/fink/update-auctex-install.8 auctex-11.82.patched/fink/update-auctex-install.8
--- auctex-11.82/fink/update-auctex-install.8	1969-12-31 19:00:00.000000000 -0500
+++ auctex-11.82.patched/fink/update-auctex-install.8	2005-12-18 12:53:02.000000000 -0500
@@ -0,0 +1,61 @@
+.\" Emacs, you know, this is an -*- nroff -*- source file.
+.\"
+.\" Man page for update-auctex-install.
+.\" Copyright (C) 2004, 05 by Davide Giovanni Maria Salvetti.
+.\"
+.\" This manual page is free software; you can redistribute it and/or
+ \" modify it under the terms of the GNU General Public License as published
+ \" by the Free Software Foundation; either version 2 of the License, or (at
+ \" your option) any later version.
+.\"
+.\" This manual page is distributed in the hope that it will be useful,
+ \" but WITHOUT ANY WARRANTY; without even the implied warranty of
+ \" MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General
+ \" Public License for more details.
+.\"
+.\" You should have received a copy of the GNU General Public License
+ \" along with this manual page; if not, write to: The Free Software
+ \" Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307,
+ \" USA.
+.\"
+.\" On Debian GNU/Linux System you can find a copy of the GNU General
+ \" Public License in "/usr/share/common-licenses/GPL".
+.\"
+.TH Update\-AUCTeX\-install 8 "19 May 2004" "Debian Project" "Debian GNU/Linux"
+.SH NAME
+update\-auctex\-install \- update AUCTeX installation.
+.SH SYNOPSIS
+.B update\-auctex\-install
+.RI [ FLAVOR .\|.\|. ]
+.SH DESCRIPTION
+The
+.B update\-auctex\-install
+script updates AUCTeX installation.
+.PP
+It should be called after each major (La)TeX installation or upgrade.
+.PP
+If no arguments are specified, update\-auctex\-install caters to every
+supported Emacs flavor installed on the system; else it updates only the
+specified flavors, if they are installed.
+.PP
+On Debian GNU/Linux system
+.B update\-auctex\-install
+is not called weekly by a
+.BR cron (8)
+driven mechanism, because it modifies files installed in the
+.B /usr/
+hierarchy, and that would not be policy compliant.  You may call it by hand,
+or have the maintainer of relevant packages add a call for it in their
+.B postinst
+package scripts.
+.SH BUGS
+None known.
+.SH FILES
+.B /usr/:
+the hierarchy containing AUCTeX installed files.
+.SH SEE ALSO
+.BR emacs (1),
+.BR tex (1),
+.BR latex (1).
+.SH AUTHOR
+Davide G. M. Salvetti <salve@debian.org>.
