diff -ruN -x '*~' dbus-0.60/fink/dbus.csh dbus-0.60.patched/fink/dbus.csh
--- dbus-0.60/fink/dbus.csh	1969-12-31 19:00:00.000000000 -0500
+++ dbus-0.60.patched/fink/dbus.csh	2006-01-24 23:26:18.000000000 -0500
@@ -0,0 +1,9 @@
+#!/bin/csh -e
+# dbus daemon start script for Fink
+# See dbus-launch man page for details
+
+# Test for an existing bus daemon, just to be safe
+if ( ! $?DBUS_SESSION_BUS_ADDRESS ) then
+    # If not found, launch a new one
+    eval `dbus-launch --csh-syntax`
+endif
diff -ruN -x '*~' dbus-0.60/fink/dbus.sh dbus-0.60.patched/fink/dbus.sh
--- dbus-0.60/fink/dbus.sh	1969-12-31 19:00:00.000000000 -0500
+++ dbus-0.60.patched/fink/dbus.sh	2006-01-24 23:26:30.000000000 -0500
@@ -0,0 +1,9 @@
+#!/bin/sh -e
+# dbus daemon start script
+# See dbus-lauch man page for details
+
+# Test for an existing bus daemon, just to be safe
+if test -z "$DBUS_SESSION_BUS_ADDRESS" ; then
+    # If not found, launch a new one
+    eval `dbus-launch --sh-syntax`
+fi
diff -ruN -x '*~' dbus-0.60/tools/dbus-launch.c dbus-0.60.patched/tools/dbus-launch.c
--- dbus-0.60/tools/dbus-launch.c	2005-11-30 14:32:27.000000000 -0500
+++ dbus-0.60.patched/tools/dbus-launch.c	2006-01-24 22:18:34.000000000 -0500
@@ -884,12 +884,12 @@
       else
 	{
 	  if (c_shell_syntax)
-	    printf ("setenv DBUS_SESSION_BUS_ADDRESS '%s'\n", bus_address);	
+	    printf ("setenv DBUS_SESSION_BUS_ADDRESS '%s';\n", bus_address);	
 	  else
 	    {
-	      printf ("DBUS_SESSION_BUS_ADDRESS='%s'\n", bus_address);
+	      printf ("DBUS_SESSION_BUS_ADDRESS='%s';\n", bus_address);
 	      if (bourne_shell_syntax)
-		printf ("export DBUS_SESSION_BUS_ADDRESS\n");
+		printf ("export DBUS_SESSION_BUS_ADDRESS;\n");
 	    }
 	  if (c_shell_syntax)
 	    printf ("set DBUS_SESSION_BUS_PID=%ld\n", (long) bus_pid);
