diff -ruN -x '*~' mozilla/.mozconfig mozilla.patched/.mozconfig
--- mozilla/.mozconfig	1969-12-31 19:00:00.000000000 -0500
+++ mozilla.patched/.mozconfig	2005-11-12 00:34:21.000000000 -0500
@@ -0,0 +1,45 @@
+# sh
+# Build configuration script
+#
+# See http://www.mozilla.org/build/unix.html for build instructions.
+#
+
+. $topsrcdir/browser/config/mozconfig
+
+# Options for client.mk.
+# environment variables to set
+#mk_add_options MOZILLA_OFFICIAL=1
+mk_add_options MOZ_MAKE_FLAGS=-j4
+
+# Options for 'configure' (same as command-line options).
+ac_add_options --enable-default-toolkit=gtk2
+ac_add_options --enable-xft
+#ac_add_options --disable-xft
+#ac_add_options --enable-freetype2
+ac_add_options --disable-freetype2
+ac_add_options --enable-xinerama
+ac_add_options --with-pthreads
+ac_add_options --disable-gnomevfs
+
+#ac_add_options --enable-official-branding
+#ac_add_options --disable-shared
+#ac_add_options --enable-static
+#ac_add_options --enable-prebinding
+ac_add_options --disable-prebinding
+ac_add_options --disable-debug
+ac_add_options --disable-tests
+ac_add_options --enable-strip
+ac_add_options --enable-optimize='-O2'
+ac_add_options --enable-macos-target=10.3
+ac_add_options --with-default-mozilla-five-home=@PREFIX@/lib/firefox
+ac_add_options --with-system-jpeg=@PREFIX@
+ac_add_options --with-system-zlib=@PREFIX@
+ac_add_options --with-system-png=@PREFIX@
+#ac_add_options --with-ft-prefix=@PREFIX@/lib/freetype2
+#ac_add_options --with-ft-exec-prefix=@PREFIX@/lib/freetype2/bin
+
+# SVG stuff
+#ac_add_options --enable-svg
+#ac_add_options --enable-svg-renderer-libart
+#mk_add_options MOZ_INTERNAL_LIBART_LGPL=1
+#MOZ_INTERNAL_LIBART_LGPL=1
diff -ruN -x '*~' mozilla/browser/components/build/Makefile.in mozilla.patched/browser/components/build/Makefile.in
--- mozilla/browser/components/build/Makefile.in	2004-07-15 16:43:54.000000000 -0400
+++ mozilla.patched/browser/components/build/Makefile.in	2005-11-12 00:34:21.000000000 -0500
@@ -75,6 +75,12 @@
 	$(MOZ_COMPONENT_LIBS) \
 	$(NULL)
 
+# fix for building on Mac/X11 (bug 257773)
+# dependencies for gkgfx
+ifeq ($(MOZ_WIDGET_TOOLKIT), gtk2)
+EXTRA_DSO_LDOPTS += $(MOZ_GTK2_LIBS)
+endif
+
 # Need to link to CoreFoundation for Mac Migrators (PList reading code)
 ifneq (,$(filter mac cocoa,$(MOZ_WIDGET_TOOLKIT)))
 EXTRA_DSO_LDOPTS += \
diff -ruN -x '*~' mozilla/browser/config/mozconfig mozilla.patched/browser/config/mozconfig
--- mozilla/browser/config/mozconfig	2004-07-07 04:31:10.000000000 -0400
+++ mozilla.patched/browser/config/mozconfig	2005-11-12 00:34:21.000000000 -0500
@@ -8,7 +8,7 @@
 
 ac_add_options --disable-ldap
 ac_add_options --disable-mailnews
-ac_add_options --enable-extensions=cookie,xml-rpc,xmlextras,pref,transformiix,universalchardet,webservices,inspector,gnomevfs,negotiateauth
+ac_add_options --enable-extensions=cookie,xml-rpc,xmlextras,pref,transformiix,universalchardet,webservices,inspector,negotiateauth
 ac_add_options --enable-crypto
 ac_add_options --disable-composer
 ac_add_options --enable-single-profile
diff -ruN -x '*~' mozilla/browser/locales/en-US/profile/bookmarks.html mozilla.patched/browser/locales/en-US/profile/bookmarks.html
--- mozilla/browser/locales/en-US/profile/bookmarks.html	2005-02-03 22:03:07.000000000 -0500
+++ mozilla.patched/browser/locales/en-US/profile/bookmarks.html	2005-11-12 00:34:21.000000000 -0500
@@ -12,6 +12,7 @@
     <DL><p>
         <DT><A HREF="http://www.mozilla.org/products/firefox/central.html" ICON="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAABGdBTUEAAK/INwWK6QAAABl0RVh0U29mdHdhcmUAQWRvYmUgSW1hZ2VSZWFkeXHJZTwAAAHWSURBVHjaYvz//z8DJQAggJiQOe/fv2fv7Oz8rays/N+VkfG/iYnJfyD/1+rVq7ffu3dPFpsBAAHEAHIBCJ85c8bN2Nj4vwsDw/8zQLwKiO8CcRoQu0DxqlWrdsHUwzBAAIGJmTNnPgYa9j8UqhFElwPxf2MIDeIrKSn9FwSJoRkAEEAM0DD4DzMAyPi/G+QKY4hh5WAXGf8PDQ0FGwJ22d27CjADAAIIrLmjo+MXA9R2kAHvGBA2wwx6B8W7od6CeQcggKCmCEL8bgwxYCbUIGTDVkHDBia+CuotgACCueD3TDQN75D4xmAvCoK9ARMHBzAw0AECiBHkAlC0Mdy7x9ABNA3obAZXIAa6iKEcGlMVQHwWyjYuL2d4v2cPg8vZswx7gHyAAAK7AOif7SAbOqCmn4Ha3AHFsIDtgPq/vLz8P4MSkJ2W9h8ggBjevXvHDo4FQUQg/kdypqCg4H8lUIACnQ/SOBMYI8bAsAJFPcj1AAEEjwVQqLpAbXmH5BJjqI0gi9DTAAgDBBCcAVLkgmQ7yKCZxpCQxqUZhAECCJ4XgMl493ug21ZD+aDAXH0WLM4A9MZPXJkJIIAwTAR5pQMalaCABQUULttBGCCAGCnNzgABBgAMJ5THwGvJLAAAAABJRU5ErkJggg==" LAST_CHARSET="ISO-8859-1" ID="rdf:#$GvPhC3">Getting Started</A>
         <DT><A HREF="http://fxfeeds.mozilla.org/" LAST_MODIFIED="1094668003" FEEDURL="http://fxfeeds.mozilla.org/rss20.xml" ID="rdf:#$HvPhC3">Latest Headlines</A>
+        <DT><A HREF="http://fink.sourceforge.net/" >Fink - Home</A>
     </DL><p>
     <HR>
     <DT><H3 ID="rdf:#$6wPhC3">Quick Searches</H3>
diff -ruN -x '*~' mozilla/config/add-chrome.pl mozilla.patched/config/add-chrome.pl
--- mozilla/config/add-chrome.pl	2004-02-20 17:14:12.000000000 -0500
+++ mozilla.patched/config/add-chrome.pl	2005-11-12 00:34:21.000000000 -0500
@@ -14,7 +14,7 @@
 my $jarFileName = $ARGV[4];
 
 my $win32 = ($^O =~ /((MS)?win32)|cygwin|os2/i) ? 1 : 0;
-my $macos = ($^O =~ /MacOS|darwin/i) ? 1 : 0;
+my $macos = 0;
 my $unix  = !($win32 || $macos) ? 1 : 0;
 
 sub foreignPlatformFile
diff -ruN -x '*~' mozilla/config/autoconf.mk.in mozilla.patched/config/autoconf.mk.in
--- mozilla/config/autoconf.mk.in	2004-10-31 00:29:54.000000000 -0400
+++ mozilla.patched/config/autoconf.mk.in	2005-11-12 00:34:21.000000000 -0500
@@ -32,14 +32,14 @@
 prefix		= @prefix@
 exec_prefix	= @exec_prefix@
 bindir		= @bindir@
-includedir	= @includedir@/$(MOZ_APP_NAME)-$(MOZ_APP_VERSION)
+includedir	= @includedir@/$(MOZ_APP_NAME)
 libdir		= @libdir@
 datadir		= @datadir@
 mandir		= @mandir@
-idldir		= @datadir@/idl/$(MOZ_APP_NAME)-$(MOZ_APP_VERSION)
+idldir		= @datadir@/idl/$(MOZ_APP_NAME)
 
-mozappdir	= $(libdir)/$(MOZ_APP_NAME)-$(MOZ_APP_VERSION)
-mredir		= $(libdir)/mre/mre-$(MOZ_APP_VERSION)
+mozappdir	= $(libdir)/$(MOZ_APP_NAME)
+mredir		= $(libdir)/mre/mre
 mrelibdir	= $(mredir)/lib
 
 DIST		= $(DEPTH)/dist
diff -ruN -x '*~' mozilla/config/config.mk mozilla.patched/config/config.mk
--- mozilla/config/config.mk	2004-10-12 00:11:11.000000000 -0400
+++ mozilla.patched/config/config.mk	2005-11-12 00:34:21.000000000 -0500
@@ -712,7 +712,7 @@
 ifeq ($(OS_ARCH),Darwin)
 ifdef USE_PREBINDING
 export LD_PREBIND=1
-export LD_SEG_ADDR_TABLE=$(shell cd $(topsrcdir); pwd)/config/prebind-address-table
+export LD_PREBIND_ALLOW_OVERLAP=1
 endif
 ifdef MACOS_SDK_DIR
 export NEXT_ROOT=$(MACOS_SDK_DIR)
@@ -765,7 +765,7 @@
 # Tell the linker where NSS is, if we're building crypto
 ifeq ($(OS_ARCH),Darwin)
 ifeq (,$(findstring crypto,$(MOZ_META_COMPONENTS)))
-MOZ_COMPONENTLIB_EXTRA_LIBS = $(foreach library, $(patsubst -l%, $(LIB_PREFIX)%$(DLL_SUFFIX), $(filter -l%, $(NSS_LIBS))), -dylib_file @executable_path/$(library):$(DIST)/bin/$(library))
+MOZ_COMPONENTLIB_EXTRA_LIBS = $(foreach library, $(patsubst -l%, $(LIB_PREFIX)%$(DLL_SUFFIX), $(filter -l%, $(NSS_LIBS))), -dylib_file @PREFIX@/lib/firefox/$(library):$(DIST)/bin/$(library))
 endif
 endif
 endif
diff -ruN -x '*~' mozilla/config/rules.mk mozilla.patched/config/rules.mk
--- mozilla/config/rules.mk	2004-08-11 15:44:17.000000000 -0400
+++ mozilla.patched/config/rules.mk	2005-11-12 00:34:21.000000000 -0500
@@ -421,7 +421,7 @@
 ifdef IS_COMPONENT
 EXTRA_DSO_LDOPTS	+= -bundle
 else
-EXTRA_DSO_LDOPTS	+= -dynamiclib -install_name @executable_path/$(SHARED_LIBRARY) -compatibility_version 1 -current_version 1
+EXTRA_DSO_LDOPTS	+= -dynamiclib -install_name \$(mozappdir)/$(SHARED_LIBRARY) -compatibility_version 1 -current_version 1
 endif
 endif
 endif
diff -ruN -x '*~' mozilla/configure mozilla.patched/configure
--- mozilla/configure	2004-10-31 02:17:58.000000000 -0500
+++ mozilla.patched/configure	2005-11-12 00:34:21.000000000 -0500
@@ -4996,7 +4996,7 @@
     ;;
 
 *-darwin*)
-    HOST_CFLAGS="$HOST_CFLAGS -DXP_UNIX -DXP_MACOSX -DNO_X11"
+    HOST_CFLAGS="$HOST_CFLAGS -DXP_UNIX"
     HOST_NSPR_MDCPUCFG='\"md/_darwin.cfg\"'
     HOST_OPTIMIZE_FLAGS="${HOST_OPTIMIZE_FLAGS=-O3}"
     ;;
@@ -5227,8 +5227,8 @@
 
 *-darwin*) 
     _PEDANTIC=
-    CFLAGS="$CFLAGS -fpascal-strings -no-cpp-precomp -fno-common"
-    CXXFLAGS="$CXXFLAGS -fpascal-strings -no-cpp-precomp -fno-common"
+    CFLAGS="$CFLAGS -no-cpp-precomp -fno-common"
+    CXXFLAGS="$CXXFLAGS -no-cpp-precomp -fno-common"
     DLL_SUFFIX=".dylib"
     DSO_LDOPTS=''
     STRIP="$STRIP -x -S"
diff -ruN -x '*~' mozilla/directory/c-sdk/config/config.mk mozilla.patched/directory/c-sdk/config/config.mk
--- mozilla/directory/c-sdk/config/config.mk	2003-02-23 10:54:18.000000000 -0500
+++ mozilla.patched/directory/c-sdk/config/config.mk	2005-11-12 00:34:21.000000000 -0500
@@ -60,10 +60,10 @@
 
 NFSPWD		= $(MOD_DEPTH)/config/nfspwd
 
-CFLAGS		= $(CC_ONLY_FLAGS) $(OPTIMIZER) $(OS_CFLAGS)\
-		  $(XP_DEFINE) $(DEFINES) $(INCLUDES) $(XCFLAGS)
-CCCFLAGS	= $(CCC_ONLY_FLAGS) $(OPTIMIZER) $(OS_CFLAGS)\
-		  $(XP_DEFINE) $(DEFINES) $(INCLUDES) $(XCFLAGS)
+CFLAGS		= $(CC_ONLY_FLAGS) $(OPTIMIZER) $(XP_DEFINE)\
+		  $(DEFINES) $(INCLUDES) $(XCFLAGS) $(OS_CFLAGS)
+CCCFLAGS	= $(CCC_ONLY_FLAGS) $(OPTIMIZER) $(XP_DEFINE)\
+		  $(DEFINES) $(INCLUDES) $(XCFLAGS) $(OS_CFLAGS)
 # For purify
 NOMD_CFLAGS	= $(CC_ONLY_FLAGS) $(OPTIMIZER) $(NOMD_OS_CFLAGS)\
 		  $(XP_DEFINE) $(DEFINES) $(INCLUDES) $(XCFLAGS)
diff -ruN -x '*~' mozilla/directory/c-sdk/configure mozilla.patched/directory/c-sdk/configure
--- mozilla/directory/c-sdk/configure	2004-09-14 17:10:26.000000000 -0400
+++ mozilla.patched/directory/c-sdk/configure	2005-11-12 00:34:21.000000000 -0500
@@ -3401,7 +3401,7 @@
 
         CPU_ARCH=ppc
     fi
-    DSO_LDOPTS='-dynamiclib -compatibility_version 1 -current_version 1 -all_load -install_name @executable_path/$@'
+    DSO_LDOPTS='-dynamiclib -compatibility_version 1 -current_version 1 -all_load -install_name @PREFIX@/lib/firefox/$@'
     # Use the standard preprocessor (cpp)
     CFLAGS="$CFLAGS -no-cpp-precomp"
     MKSHLIB='$(CC) -arch $(CPU_ARCH) $(DSO_LDOPTS) -o $@'
diff -ruN -x '*~' mozilla/directory/c-sdk/configure.in mozilla.patched/directory/c-sdk/configure.in
--- mozilla/directory/c-sdk/configure.in	2004-09-14 16:59:40.000000000 -0400
+++ mozilla.patched/directory/c-sdk/configure.in	2005-11-12 00:38:57.000000000 -0500
@@ -862,7 +862,7 @@
         AC_DEFINE(ppc)
         CPU_ARCH=ppc
     fi
-    DSO_LDOPTS='-dynamiclib -compatibility_version 1 -current_version 1 -all_load -install_name @executable_path/$@'
+    DSO_LDOPTS='-dynamiclib -compatibility_version 1 -current_version 1 -all_load -install_name @PREFIX@/lib/firefox/$@'
     # Use the standard preprocessor (cpp)
     CFLAGS="$CFLAGS -no-cpp-precomp"
     MKSHLIB='$(CC) -arch $(CPU_ARCH) $(DSO_LDOPTS) -o $@'
diff -ruN -x '*~' mozilla/fink/applications/firefox.desktop mozilla.patched/fink/applications/firefox.desktop
--- mozilla/fink/applications/firefox.desktop	1969-12-31 19:00:00.000000000 -0500
+++ mozilla.patched/fink/applications/firefox.desktop	2005-11-12 00:34:21.000000000 -0500
@@ -0,0 +1,10 @@
+[Desktop Entry]
+Encoding=UTF-8
+Name=Firefox Web Browser
+Comment=Browse the World Wide Web
+Exec=firefox
+Icon=firefox.xpm
+Terminal=false
+Type=Application
+Categories=Application;Network
+StartupNotify=true
diff -ruN -x '*~' mozilla/fink/prefs.js mozilla.patched/fink/prefs.js
--- mozilla/fink/prefs.js	1969-12-31 19:00:00.000000000 -0500
+++ mozilla.patched/fink/prefs.js	2005-11-12 00:34:21.000000000 -0500
@@ -0,0 +1,22 @@
+pref("general.useragent.vendorComment", "Fink Community Edition");
+
+// TrueType
+//pref("font.FreeType2.enable", true);
+//pref("font.freetype2.shared-library", "@PREFIX@/lib/libfreetype.6.dylib");
+//pref("font.freetype2.shared-library", "/usr/X11R6/lib/libfreetype.6.dylib");
+//pref("font.freetype2.autohinted", true);
+//pref("font.freetype2.unhinted", true);
+//pref("font.antialias.min", 10);
+//pref("font.embedded_bitmaps.max", 1000000);
+//pref("font.scale.tt_bitmap.dark_text.min", 64);
+//pref("font.scale.tt_bitmap.dark_text.gain", "0.0");
+//pref("font.directory.truetype.1", "/Library/Fonts");
+//pref("font.directory.truetype.2", "/Network/Library/Fonts");
+//pref("font.directory.truetype.3", "/System/Library/Fonts");
+//pref("font.directory.truetype.4", "/usr/X11R6/lib/X11/fonts/TTF");
+//pref("font.directory.truetype.5", "@PREFIX@/share/fonts/truetype");
+
+// AA with Bitmap scaling.
+//pref("font.scale.aa_bitmap.enable", true);
+//pref("font.scale.aa_bitmap.always", false);
+//pref("font.scale.aa_bitmap.min", 6);
diff -ruN -x '*~' mozilla/modules/libpref/src/nsPrefService.cpp mozilla.patched/modules/libpref/src/nsPrefService.cpp
--- mozilla/modules/libpref/src/nsPrefService.cpp	2004-07-26 03:52:43.000000000 -0400
+++ mozilla.patched/modules/libpref/src/nsPrefService.cpp	2005-11-12 00:34:22.000000000 -0500
@@ -747,6 +747,7 @@
 #elif defined(XP_BEOS)
       "beos.js"
 #endif
+      , "fink.js"
   };
 
   rv = pref_LoadPrefsInDir(defaultPrefDir, specialFiles, NS_ARRAY_LENGTH(specialFiles));
diff -ruN -x '*~' mozilla/nsprpub/configure mozilla.patched/nsprpub/configure
--- mozilla/nsprpub/configure	2004-09-14 17:10:26.000000000 -0400
+++ mozilla.patched/nsprpub/configure	2005-11-12 00:34:22.000000000 -0500
@@ -3305,7 +3305,7 @@
 
         CPU_ARCH=ppc
     fi
-    DSO_LDOPTS='-dynamiclib -compatibility_version 1 -current_version 1 -all_load -install_name @executable_path/$@ -headerpad_max_install_names'
+    DSO_LDOPTS='-dynamiclib -compatibility_version 1 -current_version 1 -all_load -install_name @PREFIX@/lib/firefox/$@ -headerpad_max_install_names'
     # Use the standard preprocessor (cpp)
     CFLAGS="$CFLAGS -no-cpp-precomp"
     MKSHLIB='$(CC) -arch $(CPU_ARCH) $(DSO_LDOPTS) -o $@'
@@ -3320,9 +3320,6 @@
 
     # Add Mac OS X support for loading CFM & CFBundle plugins
     if test -f /System/Library/Frameworks/Carbon.framework/Carbon; then
-        cat >> confdefs.h <<\EOF
-#define XP_MACOSX 1
-EOF
 
         OS_TARGET=MacOSX
 
@@ -4845,8 +4842,6 @@
 
 
 case $target in
-*-darwin*)
-    ;;
 *)
     echo $ac_n "checking for dlopen in -ldl""... $ac_c" 1>&6
 echo "configure:4853: checking for dlopen in -ldl" >&5
diff -ruN -x '*~' mozilla/nsprpub/configure.in mozilla.patched/nsprpub/configure.in
--- mozilla/nsprpub/configure.in	2004-09-14 16:59:41.000000000 -0400
+++ mozilla.patched/nsprpub/configure.in	2005-11-12 00:38:54.000000000 -0500
@@ -856,7 +856,7 @@
         AC_DEFINE(ppc)
         CPU_ARCH=ppc
     fi
-    DSO_LDOPTS='-dynamiclib -compatibility_version 1 -current_version 1 -all_load -install_name @executable_path/$@ -headerpad_max_install_names'
+    DSO_LDOPTS='-dynamiclib -compatibility_version 1 -current_version 1 -all_load -install_name @PREFIX@/lib/firefox/$@ -headerpad_max_install_names'
     # Use the standard preprocessor (cpp)
     CFLAGS="$CFLAGS -no-cpp-precomp"
     MKSHLIB='$(CC) -arch $(CPU_ARCH) $(DSO_LDOPTS) -o $@'
diff -ruN -x '*~' mozilla/nsprpub/pr/include/md/_darwin.h mozilla.patched/nsprpub/pr/include/md/_darwin.h
--- mozilla/nsprpub/pr/include/md/_darwin.h	2003-11-26 20:30:44.000000000 -0500
+++ mozilla.patched/nsprpub/pr/include/md/_darwin.h	2005-11-12 00:34:22.000000000 -0500
@@ -55,7 +55,8 @@
 
 #undef  HAVE_STACK_GROWING_UP
 #define HAVE_DLL
-#define USE_MACH_DYLD
+/* #define USE_MACH_DYLD */
+#define USE_DLFCN
 #define _PR_HAVE_SOCKADDR_LEN  
 #define _PR_STAT_HAS_ST_ATIMESPEC
 #define _PR_NO_LARGE_FILES
diff -ruN -x '*~' mozilla/nsprpub/pr/src/Makefile.in mozilla.patched/nsprpub/pr/src/Makefile.in
--- mozilla/nsprpub/pr/src/Makefile.in	2004-03-22 19:40:17.000000000 -0500
+++ mozilla.patched/nsprpub/pr/src/Makefile.in	2005-11-12 00:34:22.000000000 -0500
@@ -199,9 +199,9 @@
 endif
 endif
 
-ifeq ($(OS_TARGET),MacOSX)
-OS_LIBS		= -framework CoreServices -framework CoreFoundation
-endif
+#ifeq ($(OS_TARGET),MacOSX)
+#OS_LIBS		= -framework CoreServices -framework CoreFoundation
+#endif
 
 ifdef GC_LEAK_DETECTOR
 EXTRA_LIBS	= -L$(dist_libdir) -lboehm
diff -ruN -x '*~' mozilla/nsprpub/pr/src/linking/prlink.c mozilla.patched/nsprpub/pr/src/linking/prlink.c
--- mozilla/nsprpub/pr/src/linking/prlink.c	2004-01-20 14:46:21.000000000 -0500
+++ mozilla.patched/nsprpub/pr/src/linking/prlink.c	2005-11-12 00:34:22.000000000 -0500
@@ -164,7 +164,8 @@
 #if defined(SUNOS4) || defined(DARWIN) || defined(NEXTSTEP) \
     || defined(WIN16) || defined(XP_OS2) \
     || ((defined(OPENBSD) || defined(NETBSD)) && !defined(__ELF__))
-#define NEED_LEADING_UNDERSCORE
+/* On MacOSX 10.3, dlsym does not accept symbols with '_'. */
+/* #define NEED_LEADING_UNDERSCORE */
 #endif
 
 #ifdef XP_PC
diff -ruN -x '*~' mozilla/security/coreconf/Darwin.mk mozilla.patched/security/coreconf/Darwin.mk
--- mozilla/security/coreconf/Darwin.mk	2003-11-12 20:59:22.000000000 -0500
+++ mozilla.patched/security/coreconf/Darwin.mk	2005-11-12 00:34:22.000000000 -0500
@@ -68,7 +68,7 @@
 ARCH		= darwin
 
 # May override this with -bundle to create a loadable module.
-DSO_LDOPTS	= -dynamiclib -compatibility_version 1 -current_version 1 -install_name @executable_path/$(notdir $@) -headerpad_max_install_names
+DSO_LDOPTS	= -dynamiclib -compatibility_version 1 -current_version 1 -install_name @PREFIX@/lib/firefox/$(notdir $@) -headerpad_max_install_names
 
 MKSHLIB		= $(CC) -arch $(CPU_ARCH) $(DSO_LDOPTS)
 DLL_SUFFIX	= dylib
diff -ruN -x '*~' mozilla/security/nss/cmd/platlibs.mk mozilla.patched/security/nss/cmd/platlibs.mk
--- mozilla/security/nss/cmd/platlibs.mk	2004-10-15 17:13:50.000000000 -0400
+++ mozilla.patched/security/nss/cmd/platlibs.mk	2005-11-12 00:34:22.000000000 -0500
@@ -207,7 +207,7 @@
 endif
 
 ifeq ($(OS_ARCH), Darwin)
-EXTRA_SHARED_LIBS += -dylib_file @executable_path/libsoftokn3.dylib:$(DIST)/lib/libsoftokn3.dylib
+EXTRA_SHARED_LIBS += -dylib_file @PREFIX@/lib/firefox/libsoftokn3.dylib:$(DIST)/lib/libsoftokn3.dylib
 endif
 
 
diff -ruN -x '*~' mozilla/security/nss/lib/pki1/oiddata.h mozilla.patched/security/nss/lib/pki1/oiddata.h
--- mozilla/security/nss/lib/pki1/oiddata.h	2002-01-04 00:22:07.000000000 -0500
+++ mozilla.patched/security/nss/lib/pki1/oiddata.h	2005-11-12 00:34:22.000000000 -0500
@@ -39,6 +39,9 @@
 static const char OIDDATA_CVS_ID[] = "@(#) $RCSfile$ $Revision$ $Date$ $Name$ ; @(#) $RCSfile$ $Revision$ $Date$ $Name$";
 #endif /* DEBUG */
 
+/* added to fix building on Tiger/gcc4 (bug 284386) */
+/* fixed for 1.1, so remove this block from the patch */
+#include "pki1t.h"
 #ifndef NSSPKI1T_H
 #include "nsspki1t.h"
 #endif /* NSSPKI1T_H */
diff -ruN -x '*~' mozilla/security/nss/lib/pki1/oiddata.h.orig mozilla.patched/security/nss/lib/pki1/oiddata.h.orig
--- mozilla/security/nss/lib/pki1/oiddata.h.orig	1969-12-31 19:00:00.000000000 -0500
+++ mozilla.patched/security/nss/lib/pki1/oiddata.h.orig	2002-01-04 00:22:07.000000000 -0500
@@ -0,0 +1,217 @@
+/* THIS IS A GENERATED FILE */
+/* 
+ * The contents of this file are subject to the Mozilla Public
+ * License Version 1.1 (the "License"); you may not use this file
+ * except in compliance with the License. You may obtain a copy of
+ * the License at http://www.mozilla.org/MPL/
+ * 
+ * Software distributed under the License is distributed on an "AS
+ * IS" basis, WITHOUT WARRANTY OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * rights and limitations under the License.
+ * 
+ * The Original Code is the Netscape security libraries.
+ * 
+ * The Initial Developer of the Original Code is Netscape
+ * Communications Corporation.  Portions created by Netscape are 
+ * Copyright (C) 1994-2000 Netscape Communications Corporation.  All
+ * Rights Reserved.
+ * 
+ * Contributor(s):
+ * 
+ * Alternatively, the contents of this file may be used under the
+ * terms of the GNU General Public License Version 2 or later (the
+ * "GPL"), in which case the provisions of the GPL are applicable 
+ * instead of those above.  If you wish to allow use of your 
+ * version of this file only under the terms of the GPL and not to
+ * allow others to use your version of this file under the MPL,
+ * indicate your decision by deleting the provisions above and
+ * replace them with the notice and other provisions required by
+ * the GPL.  If you do not delete the provisions above, a recipient
+ * may use your version of this file under either the MPL or the
+ * GPL.
+ */
+
+#ifndef OIDDATA_H
+#define OIDDATA_H
+
+#ifdef DEBUG
+static const char OIDDATA_CVS_ID[] = "@(#) $RCSfile$ $Revision$ $Date$ $Name$ ; @(#) $RCSfile$ $Revision$ $Date$ $Name$";
+#endif /* DEBUG */
+
+#ifndef NSSPKI1T_H
+#include "nsspki1t.h"
+#endif /* NSSPKI1T_H */
+
+extern const NSSOID nss_builtin_oids[];
+extern const PRUint32 nss_builtin_oid_count;
+
+/*extern const nssAttributeTypeAliasTable nss_attribute_type_aliases[];*/
+/*extern const PRUint32 nss_attribute_type_alias_count;*/
+
+extern const NSSOID *NSS_OID_RFC1274_UID;
+extern const NSSOID *NSS_OID_RFC1274_EMAIL;
+extern const NSSOID *NSS_OID_RFC2247_DC;
+extern const NSSOID *NSS_OID_ANSIX9_DSA_SIGNATURE;
+extern const NSSOID *NSS_OID_ANSIX9_DSA_SIGNATURE_WITH_SHA1_DIGEST;
+extern const NSSOID *NSS_OID_X942_DIFFIE_HELMAN_KEY;
+extern const NSSOID *NSS_OID_PKCS1_RSA_ENCRYPTION;
+extern const NSSOID *NSS_OID_PKCS1_MD2_WITH_RSA_ENCRYPTION;
+extern const NSSOID *NSS_OID_PKCS1_MD4_WITH_RSA_ENCRYPTION;
+extern const NSSOID *NSS_OID_PKCS1_MD5_WITH_RSA_ENCRYPTION;
+extern const NSSOID *NSS_OID_PKCS1_SHA1_WITH_RSA_ENCRYPTION;
+extern const NSSOID *NSS_OID_PKCS5_PBE_WITH_MD2_AND_DES_CBC;
+extern const NSSOID *NSS_OID_PKCS5_PBE_WITH_MD5_AND_DES_CBC;
+extern const NSSOID *NSS_OID_PKCS5_PBE_WITH_SHA1_AND_DES_CBC;
+extern const NSSOID *NSS_OID_PKCS7;
+extern const NSSOID *NSS_OID_PKCS7_DATA;
+extern const NSSOID *NSS_OID_PKCS7_SIGNED_DATA;
+extern const NSSOID *NSS_OID_PKCS7_ENVELOPED_DATA;
+extern const NSSOID *NSS_OID_PKCS7_SIGNED_ENVELOPED_DATA;
+extern const NSSOID *NSS_OID_PKCS7_DIGESTED_DATA;
+extern const NSSOID *NSS_OID_PKCS7_ENCRYPTED_DATA;
+extern const NSSOID *NSS_OID_PKCS9_EMAIL_ADDRESS;
+extern const NSSOID *NSS_OID_PKCS9_UNSTRUCTURED_NAME;
+extern const NSSOID *NSS_OID_PKCS9_CONTENT_TYPE;
+extern const NSSOID *NSS_OID_PKCS9_MESSAGE_DIGEST;
+extern const NSSOID *NSS_OID_PKCS9_SIGNING_TIME;
+extern const NSSOID *NSS_OID_PKCS9_COUNTER_SIGNATURE;
+extern const NSSOID *NSS_OID_PKCS9_CHALLENGE_PASSWORD;
+extern const NSSOID *NSS_OID_PKCS9_UNSTRUCTURED_ADDRESS;
+extern const NSSOID *NSS_OID_PKCS9_EXTENDED_CERTIFICATE_ATTRIBUTES;
+extern const NSSOID *NSS_OID_PKCS9_SMIME_CAPABILITIES;
+extern const NSSOID *NSS_OID_PKCS9_FRIENDLY_NAME;
+extern const NSSOID *NSS_OID_PKCS9_LOCAL_KEY_ID;
+extern const NSSOID *NSS_OID_PKCS9_X509_CERT;
+extern const NSSOID *NSS_OID_PKCS9_SDSI_CERT;
+extern const NSSOID *NSS_OID_PKCS9_X509_CRL;
+extern const NSSOID *NSS_OID_PKCS12;
+extern const NSSOID *NSS_OID_PKCS12_PBE_IDS;
+extern const NSSOID *NSS_OID_PKCS12_PBE_WITH_SHA1_AND_128_BIT_RC4;
+extern const NSSOID *NSS_OID_PKCS12_PBE_WITH_SHA1_AND_40_BIT_RC4;
+extern const NSSOID *NSS_OID_PKCS12_PBE_WITH_SHA1_AND_3_KEY_TRIPLE_DES_CBC;
+extern const NSSOID *NSS_OID_PKCS12_PBE_WITH_SHA1_AND_2_KEY_TRIPLE_DES_CBC;
+extern const NSSOID *NSS_OID_PKCS12_PBE_WITH_SHA1_AND_128_BIT_RC2_CBC;
+extern const NSSOID *NSS_OID_PKCS12_PBE_WITH_SHA1_AND_40_BIT_RC2_CBC;
+extern const NSSOID *NSS_OID_PKCS12_KEY_BAG;
+extern const NSSOID *NSS_OID_PKCS12_PKCS8_SHROUDED_KEY_BAG;
+extern const NSSOID *NSS_OID_PKCS12_CERT_BAG;
+extern const NSSOID *NSS_OID_PKCS12_CRL_BAG;
+extern const NSSOID *NSS_OID_PKCS12_SECRET_BAG;
+extern const NSSOID *NSS_OID_PKCS12_SAFE_CONTENTS_BAG;
+extern const NSSOID *NSS_OID_MD2;
+extern const NSSOID *NSS_OID_MD4;
+extern const NSSOID *NSS_OID_MD5;
+extern const NSSOID *NSS_OID_RC2_CBC;
+extern const NSSOID *NSS_OID_RC4;
+extern const NSSOID *NSS_OID_DES_EDE3_CBC;
+extern const NSSOID *NSS_OID_RC5_CBC_PAD;
+extern const NSSOID *NSS_OID_X509_AUTH_INFO_ACCESS;
+extern const NSSOID *NSS_OID_PKIX_CPS_POINTER_QUALIFIER;
+extern const NSSOID *NSS_OID_PKIX_USER_NOTICE_QUALIFIER;
+extern const NSSOID *NSS_OID_EXT_KEY_USAGE_SERVER_AUTH;
+extern const NSSOID *NSS_OID_EXT_KEY_USAGE_CLIENT_AUTH;
+extern const NSSOID *NSS_OID_EXT_KEY_USAGE_CODE_SIGN;
+extern const NSSOID *NSS_OID_EXT_KEY_USAGE_EMAIL_PROTECTION;
+extern const NSSOID *NSS_OID_EXT_KEY_USAGE_IPSEC_END_SYSTEM;
+extern const NSSOID *NSS_OID_EXT_KEY_USAGE_IPSEC_TUNNEL;
+extern const NSSOID *NSS_OID_EXT_KEY_USAGE_IPSEC_USER;
+extern const NSSOID *NSS_OID_EXT_KEY_USAGE_TIME_STAMP;
+extern const NSSOID *NSS_OID_OCSP_RESPONDER;
+extern const NSSOID *NSS_OID_PKIX_REGCTRL_REGTOKEN;
+extern const NSSOID *NSS_OID_PKIX_REGCTRL_AUTHENTICATOR;
+extern const NSSOID *NSS_OID_PKIX_REGCTRL_PKIPUBINFO;
+extern const NSSOID *NSS_OID_PKIX_REGCTRL_PKI_ARCH_OPTIONS;
+extern const NSSOID *NSS_OID_PKIX_REGCTRL_OLD_CERT_ID;
+extern const NSSOID *NSS_OID_PKIX_REGCTRL_PROTOCOL_ENC_KEY;
+extern const NSSOID *NSS_OID_PKIX_REGINFO_UTF8_PAIRS;
+extern const NSSOID *NSS_OID_PKIX_REGINFO_CERT_REQUEST;
+extern const NSSOID *NSS_OID_OID_PKIX_OCSP;
+extern const NSSOID *NSS_OID_PKIX_OCSP_BASIC_RESPONSE;
+extern const NSSOID *NSS_OID_PKIX_OCSP_NONCE;
+extern const NSSOID *NSS_OID_PKIX_OCSP_RESPONSE;
+extern const NSSOID *NSS_OID_PKIX_OCSP_CRL;
+extern const NSSOID *NSS_OID_X509_OCSP_NO_CHECK;
+extern const NSSOID *NSS_OID_PKIX_OCSP_ARCHIVE_CUTOFF;
+extern const NSSOID *NSS_OID_PKIX_OCSP_SERVICE_LOCATOR;
+extern const NSSOID *NSS_OID_DES_ECB;
+extern const NSSOID *NSS_OID_DES_CBC;
+extern const NSSOID *NSS_OID_DES_OFB;
+extern const NSSOID *NSS_OID_DES_CFB;
+extern const NSSOID *NSS_OID_DES_MAC;
+extern const NSSOID *NSS_OID_ISO_SHA_WITH_RSA_SIGNATURE;
+extern const NSSOID *NSS_OID_DES_EDE;
+extern const NSSOID *NSS_OID_SHA1;
+extern const NSSOID *NSS_OID_BOGUS_DSA_SIGNATURE_WITH_SHA1_DIGEST;
+extern const NSSOID *NSS_OID_X520_COMMON_NAME;
+extern const NSSOID *NSS_OID_X520_SURNAME;
+extern const NSSOID *NSS_OID_X520_COUNTRY_NAME;
+extern const NSSOID *NSS_OID_X520_LOCALITY_NAME;
+extern const NSSOID *NSS_OID_X520_STATE_OR_PROVINCE_NAME;
+extern const NSSOID *NSS_OID_X520_ORGANIZATION_NAME;
+extern const NSSOID *NSS_OID_X520_ORGANIZATIONAL_UNIT_NAME;
+extern const NSSOID *NSS_OID_X520_TITLE;
+extern const NSSOID *NSS_OID_X520_NAME;
+extern const NSSOID *NSS_OID_X520_GIVEN_NAME;
+extern const NSSOID *NSS_OID_X520_INITIALS;
+extern const NSSOID *NSS_OID_X520_GENERATION_QUALIFIER;
+extern const NSSOID *NSS_OID_X520_DN_QUALIFIER;
+extern const NSSOID *NSS_OID_X500_RSA_ENCRYPTION;
+extern const NSSOID *NSS_OID_X509_SUBJECT_DIRECTORY_ATTR;
+extern const NSSOID *NSS_OID_X509_SUBJECT_DIRECTORY_ATTRIBUTES;
+extern const NSSOID *NSS_OID_X509_SUBJECT_KEY_ID;
+extern const NSSOID *NSS_OID_X509_KEY_USAGE;
+extern const NSSOID *NSS_OID_X509_PRIVATE_KEY_USAGE_PERIOD;
+extern const NSSOID *NSS_OID_X509_SUBJECT_ALT_NAME;
+extern const NSSOID *NSS_OID_X509_ISSUER_ALT_NAME;
+extern const NSSOID *NSS_OID_X509_BASIC_CONSTRAINTS;
+extern const NSSOID *NSS_OID_X509_CRL_NUMBER;
+extern const NSSOID *NSS_OID_X509_REASON_CODE;
+extern const NSSOID *NSS_OID_X509_HOLD_INSTRUCTION_CODE;
+extern const NSSOID *NSS_OID_X509_INVALID_DATE;
+extern const NSSOID *NSS_OID_X509_DELTA_CRL_INDICATOR;
+extern const NSSOID *NSS_OID_X509_ISSUING_DISTRIBUTION_POINT;
+extern const NSSOID *NSS_OID_X509_CERTIFICATE_ISSUER;
+extern const NSSOID *NSS_OID_X509_NAME_CONSTRAINTS;
+extern const NSSOID *NSS_OID_X509_CRL_DIST_POINTS;
+extern const NSSOID *NSS_OID_X509_CERTIFICATE_POLICIES;
+extern const NSSOID *NSS_OID_X509_POLICY_MAPPINGS;
+extern const NSSOID *NSS_OID_X509_AUTH_KEY_ID;
+extern const NSSOID *NSS_OID_X509_POLICY_CONSTRAINTS;
+extern const NSSOID *NSS_OID_X509_EXT_KEY_USAGE;
+extern const NSSOID *NSS_OID_MISSI_DSS_OLD;
+extern const NSSOID *NSS_OID_FORTEZZA_SKIPJACK;
+extern const NSSOID *NSS_OID_MISSI_KEA;
+extern const NSSOID *NSS_OID_MISSI_KEA_DSS_OLD;
+extern const NSSOID *NSS_OID_MISSI_DSS;
+extern const NSSOID *NSS_OID_MISSI_KEA_DSS;
+extern const NSSOID *NSS_OID_MISSI_ALT_KEY;
+extern const NSSOID *NSS_OID_NS_CERT_EXT_CERT_TYPE;
+extern const NSSOID *NSS_OID_NS_CERT_EXT_BASE_URL;
+extern const NSSOID *NSS_OID_NS_CERT_EXT_REVOCATION_URL;
+extern const NSSOID *NSS_OID_NS_CERT_EXT_CA_REVOCATION_URL;
+extern const NSSOID *NSS_OID_NS_CERT_EXT_CA_CRL_URL;
+extern const NSSOID *NSS_OID_NS_CERT_EXT_CA_CERT_URL;
+extern const NSSOID *NSS_OID_NS_CERT_EXT_CERT_RENEWAL_URL;
+extern const NSSOID *NSS_OID_NS_CERT_EXT_CA_POLICY_URL;
+extern const NSSOID *NSS_OID_NS_CERT_EXT_HOMEPAGE_URL;
+extern const NSSOID *NSS_OID_NS_CERT_EXT_ENTITY_LOGO;
+extern const NSSOID *NSS_OID_NS_CERT_EXT_USER_PICTURE;
+extern const NSSOID *NSS_OID_NS_CERT_EXT_SSL_SERVER_NAME;
+extern const NSSOID *NSS_OID_NS_CERT_EXT_COMMENT;
+extern const NSSOID *NSS_OID_NS_CERT_EXT_THAYES;
+extern const NSSOID *NSS_OID_NS_TYPE_GIF;
+extern const NSSOID *NSS_OID_NS_TYPE_JPEG;
+extern const NSSOID *NSS_OID_NS_TYPE_URL;
+extern const NSSOID *NSS_OID_NS_TYPE_HTML;
+extern const NSSOID *NSS_OID_NS_TYPE_CERT_SEQUENCE;
+extern const NSSOID *NSS_OID_NS_KEY_USAGE_GOVT_APPROVED;
+extern const NSSOID *NSS_OID_NETSCAPE_RECOVERY_REQUEST;
+extern const NSSOID *NSS_OID_NETSCAPE_SMIME_KEA;
+extern const NSSOID *NSS_OID_NETSCAPE_NICKNAME;
+extern const NSSOID *NSS_OID_VERISIGN_USER_NOTICES;
+extern const NSSOID *NSS_OID_NS_CERT_EXT_NETSCAPE_OK;
+extern const NSSOID *NSS_OID_NS_CERT_EXT_ISSUER_LOGO;
+extern const NSSOID *NSS_OID_NS_CERT_EXT_SUBJECT_LOGO;
+
+#endif /* OIDDATA_H */
diff -ruN -x '*~' mozilla/security/nss/lib/smime/config.mk mozilla.patched/security/nss/lib/smime/config.mk
--- mozilla/security/nss/lib/smime/config.mk	2004-10-15 17:13:56.000000000 -0400
+++ mozilla.patched/security/nss/lib/smime/config.mk	2005-11-12 00:34:22.000000000 -0500
@@ -70,7 +70,7 @@
 	$(NULL)
 
 ifeq ($(OS_ARCH), Darwin)
-EXTRA_SHARED_LIBS += -dylib_file @executable_path/libsoftokn3.dylib:$(DIST)/lib/libsoftokn3.dylib
+EXTRA_SHARED_LIBS += -dylib_file @PREFIX@/lib/firefox/libsoftokn3.dylib:$(DIST)/lib/libsoftokn3.dylib
 endif
 
 endif
diff -ruN -x '*~' mozilla/security/nss/lib/ssl/config.mk mozilla.patched/security/nss/lib/ssl/config.mk
--- mozilla/security/nss/lib/ssl/config.mk	2004-10-15 17:13:57.000000000 -0400
+++ mozilla.patched/security/nss/lib/ssl/config.mk	2005-11-12 00:34:22.000000000 -0500
@@ -75,7 +75,7 @@
 endif
 
 ifeq ($(OS_ARCH), Darwin)
-EXTRA_SHARED_LIBS += -dylib_file @executable_path/libsoftokn3.dylib:$(DIST)/lib/libsoftokn3.dylib
+EXTRA_SHARED_LIBS += -dylib_file @PREFIX@/lib/firefox/libsoftokn3.dylib:$(DIST)/lib/libsoftokn3.dylib
 endif
 
 ifeq ($(OS_TARGET),SunOS)
diff -ruN -x '*~' mozilla/themes/Makefile.in mozilla.patched/themes/Makefile.in
--- mozilla/themes/Makefile.in	2004-05-16 14:16:49.000000000 -0400
+++ mozilla.patched/themes/Makefile.in	2005-11-12 00:34:22.000000000 -0500
@@ -54,7 +54,9 @@
 
 # select classic as the default skin
 libs::
-	echo skin,install,select,classic/1.0 >> $(DIST)/bin/chrome/installed-chrome.txt
+# fix later
+#	echo skin,install,select,classic/1.0 >> $(DIST)/bin/chrome/installed-chrome.txt
 
 install::
-	echo skin,install,select,classic/1.0 >> $(DESTDIR)$(mozappdir)/chrome/installed-chrome.txt
+# fix later
+#	echo skin,install,select,classic/1.0 >> $(DESTDIR)$(mozappdir)/chrome/installed-chrome.txt
