diff -ruNX /Users/sylvain/Library/Application Support/diff/ignore libgsf-1.6.0/gsf/gsf-output-stdio.c libgsf-1.6.0-patched/gsf/gsf-output-stdio.c
--- libgsf-1.6.0/gsf/gsf-output-stdio.c	Wed Dec  4 17:02:03 2002
+++ libgsf-1.6.0-patched/gsf/gsf-output-stdio.c	Fri Jan 17 21:20:33 2003
@@ -19,6 +19,8 @@
  * USA
  */
 
+#undef _POSIX_SOURCE
+
 #include <gsf-config.h>
 #include <gsf/gsf-output-stdio.h>
 #include <gsf/gsf-output-impl.h>
diff -ruNX /Users/sylvain/Library/Application Support/diff/ignore libgsf-1.6.0/gsf/gsf-shared-memory.c libgsf-1.6.0-patched/gsf/gsf-shared-memory.c
--- libgsf-1.6.0/gsf/gsf-shared-memory.c	Thu Aug 15 23:17:45 2002
+++ libgsf-1.6.0-patched/gsf/gsf-shared-memory.c	Fri Jan 17 21:20:20 2003
@@ -23,6 +23,7 @@
 #include <gsf/gsf-impl-utils.h>
 
 #ifdef HAVE_MMAP
+#include <sys/types.h>
 #include <sys/mman.h>
 #endif
 
diff -ruNX /Users/sylvain/Library/Application Support/diff/ignore libgsf-1.6.0/gsf-gnome/gsf-input-bonobo.c libgsf-1.6.0-patched/gsf-gnome/gsf-input-bonobo.c
--- libgsf-1.6.0/gsf-gnome/gsf-input-bonobo.c	Wed Aug 28 06:58:41 2002
+++ libgsf-1.6.0-patched/gsf-gnome/gsf-input-bonobo.c	Fri Jan 17 21:19:39 2003
@@ -19,6 +19,8 @@
  * USA
  */
 
+#undef _POSIX_SOURCE
+
 #include <gsf-config.h>
 #include <gsf-gnome/gsf-input-bonobo.h>
 #include <gsf/gsf-input-impl.h>
diff -ruNX /Users/sylvain/Library/Application Support/diff/ignore libgsf-1.6.0/gsf-gnome/gsf-output-bonobo.c libgsf-1.6.0-patched/gsf-gnome/gsf-output-bonobo.c
--- libgsf-1.6.0/gsf-gnome/gsf-output-bonobo.c	Mon Sep 16 15:51:30 2002
+++ libgsf-1.6.0-patched/gsf-gnome/gsf-output-bonobo.c	Fri Jan 17 21:19:36 2003
@@ -19,6 +19,8 @@
  * USA
  */
 
+#undef _POSIX_SOURCE
+
 #include <gsf-config.h>
 #include <gsf-gnome/gsf-output-bonobo.h>
 #include <bonobo/bonobo-exception.h>
diff -ruNX /Users/sylvain/Library/Application Support/diff/ignore libgsf-1.6.0/gsf-gnome/gsf-shared-bonobo-stream.c libgsf-1.6.0-patched/gsf-gnome/gsf-shared-bonobo-stream.c
--- libgsf-1.6.0/gsf-gnome/gsf-shared-bonobo-stream.c	Mon Aug 19 04:44:28 2002
+++ libgsf-1.6.0-patched/gsf-gnome/gsf-shared-bonobo-stream.c	Fri Jan 17 21:19:16 2003
@@ -19,6 +19,8 @@
  * USA
  */
 
+#undef _POSIX_SOURCE
+
 #include <gsf-config.h>
 #include <gsf-gnome/gsf-shared-bonobo-stream.h>
 #include <gsf/gsf-impl-utils.h>
