diff -uNr dvdauthor-0.6.10/Makefile.in dvdauthor-0.6.10-new/Makefile.in
--- dvdauthor-0.6.10/Makefile.in	Thu Jan 15 14:42:47 2004
+++ dvdauthor-0.6.10-new/Makefile.in	Tue Mar 30 09:50:49 2004
@@ -6,7 +6,7 @@
 DOC = doc/spumux.1 doc/dvdauthor.1
 
 CC = @CC@
-CFLAGS = -Wall @CFLAGS@ @XML_CFLAGS@ @MAGICK_CFLAGS@ @FREETYPE_CFLAGS@ @FRIBIDI_CFLAGS@ -MD
+CFLAGS = -Wall @FREETYPE_CFLAGS@ @CFLAGS@ @XML_CFLAGS@ @MAGICK_CFLAGS@ @FRIBIDI_CFLAGS@ -MD
 # CFLAGS += -g
 
 BISON=bison
