diff -ruN easytag-0.27/configure easytag-0.27a/configure
--- easytag-0.27/configure	Sun Feb  2 17:00:35 2003
+++ easytag-0.27a/configure	Mon Feb  3 19:05:09 2003
@@ -1373,7 +1373,7 @@
 
 
 PACKAGE=easytag
-VERSION=0.27
+VERSION=0.27a
 
 ALL_LINGUAS="cs de es fr hu it ja nl pl ro ru sv uk"
 
diff -ruN easytag-0.27/src/about.c easytag-0.27a/src/about.c
--- easytag-0.27/src/about.c	Wed Jan 29 18:49:06 2003
+++ easytag-0.27a/src/about.c	Mon Feb  3 19:01:13 2003
@@ -24,6 +24,7 @@
 #include <stdio.h>
 #include <string.h>
 #include <id3.h>
+#include <errno.h>
 
 #include "about.h"
 #include "easytag.h"
diff -ruN easytag-0.27/src/misc.c easytag-0.27a/src/misc.c
--- easytag-0.27/src/misc.c	Sun Feb  2 15:41:26 2003
+++ easytag-0.27a/src/misc.c	Mon Feb  3 19:03:25 2003
@@ -1910,7 +1910,7 @@
 
     // Label of file content
     Label = gtk_label_new(_("Loaded File Content :"));
-    gtk_table_attach_defaults(GTK_TABLE(Table),Label,0,1,0,1);
+    gtk_table_attach(GTK_TABLE(Table),Label,0,1,0,1,GTK_FILL,GTK_FILL,0,2);
 
     // Content of the loaded file
     ScrollWindow = gtk_scrolled_window_new(NULL,NULL);
@@ -1930,7 +1930,7 @@
 
     // Label of current list
     Label = gtk_label_new(_("Files Name List :"));
-    gtk_table_attach_defaults(GTK_TABLE(Table),Label,1,2,0,1);
+    gtk_table_attach(GTK_TABLE(Table),Label,1,2,0,1,GTK_FILL,GTK_FILL,0,2);
 
     // List of current filenames
     ScrollWindow = gtk_scrolled_window_new(NULL,NULL);
