diff -uNr vtun/auth.c vtun-patched/auth.c
--- vtun/auth.c	Mon Dec 16 16:40:44 2002
+++ vtun-patched/auth.c	Tue Apr  6 10:54:27 2004
@@ -61,14 +61,9 @@
 /* Encryption and Decryption of the challenge key */
 #ifdef HAVE_SSL
 
-#ifndef __APPLE_CC__
 #include <openssl/md5.h>
 #include <openssl/blowfish.h>
 #include <openssl/rand.h>
-#else /* YAY - We're MAC OS */
-#include <sys/md5.h>
-#include <crypto/blowfish.h>
-#endif  /* __APPLE_CC__ */
 
 void gen_chal(char *buf)
 {
diff -uNr vtun/lfd_encrypt.c vtun-patched/lfd_encrypt.c
--- vtun/lfd_encrypt.c	Mon Dec 16 16:40:46 2002
+++ vtun-patched/lfd_encrypt.c	Tue Apr  6 10:54:46 2004
@@ -48,14 +48,9 @@
 
 #ifdef HAVE_SSL
 
-#ifndef __APPLE_CC__
 /* OpenSSL includes */
 #include <openssl/md5.h>
 #include <openssl/blowfish.h>
-#else /* YAY - We're MAC OS */
-#include <sys/md5.h>
-#include <crypto/blowfish.h>
-#endif  /* __APPLE_CC__ */
 
 #define ENC_BUF_SIZE VTUN_FRAME_SIZE + 16 
 #define ENC_KEY_SIZE 16
