diff -Naur rt-3.4.1-orig/bin/rt-crontool.in rt-3.4.1/bin/rt-crontool.in
--- rt-3.4.1-orig/bin/rt-crontool.in	2005-02-21 09:22:00.000000000 +0100
+++ rt-3.4.1/bin/rt-crontool.in	2005-02-21 09:48:21.000000000 +0100
@@ -47,7 +47,7 @@
 use strict;
 use Carp;
 
-use lib ("@LOCAL_LIB_PATH@", "@RT_LIB_PATH@");
+use lib ("@LOCAL_LIB_PATH@", "@RT_LIB_PATH@", "@PREFIX@/lib/perl5", "@PREFIX@/lib/perl5/darwin" );
 
 package RT;
 
diff -Naur rt-3.4.1-orig/bin/rt-mailgate.in rt-3.4.1/bin/rt-mailgate.in
--- rt-3.4.1-orig/bin/rt-mailgate.in	2005-02-21 09:22:00.000000000 +0100
+++ rt-3.4.1/bin/rt-mailgate.in	2005-02-21 09:45:16.000000000 +0100
@@ -52,6 +52,9 @@
 
 
 use strict;
+
+use lib ( "@PREFIX@/lib/perl5", "@PREFIX@/lib/perl5/darwin" );
+
 use Getopt::Long;
 use LWP::UserAgent;
 
diff -Naur rt-3.4.1-orig/bin/webmux.pl.in rt-3.4.1/bin/webmux.pl.in
--- rt-3.4.1-orig/bin/webmux.pl.in	2005-02-21 09:22:00.000000000 +0100
+++ rt-3.4.1/bin/webmux.pl.in	2005-02-21 09:45:08.000000000 +0100
@@ -60,7 +60,7 @@
 
 }
 
-use lib ( "@LOCAL_LIB_PATH@", "@RT_LIB_PATH@" );
+use lib ( "@LOCAL_LIB_PATH@", "@RT_LIB_PATH@", "@PREFIX@/lib/perl5", "@PREFIX@/lib/perl5/darwin" );
 use RT;
 
 package RT::Mason;
