--- FinkCommander/FinkController.m	2006-05-05 21:54:55.000000000 +0200
+++ FinkCommander.corr/FinkController.m	2006-09-22 11:09:05.000000000 +0200
@@ -1252,6 +1252,10 @@
 			[args insertObject: @"-f" atIndex: 1];
 			[args insertObject: @"-q0" atIndex: 1];
 		}
+		if (type == FINK && [cmd isEqualToString:@"cleanup"]){
+			args = [NSMutableArray arrayWithObjects: exe, cmd,
+				@"--srcs", @"--debs", @"--bl", @"--dpkg-status", nil];
+		}
     }else{
 		[self setLastCommand:exe];
 		args = [NSMutableArray arrayWithObjects: exe, @"--remove",
