Index: scribus/gdk-pixbuf-xlib-drawable.c
===================================================================
RCS file: /cvs/Scribus/scribus/gdk-pixbuf-xlib-drawable.c,v
retrieving revision 1.2.2.1
diff -u -p -r1.2.2.1 gdk-pixbuf-xlib-drawable.c
--- scribus/gdk-pixbuf-xlib-drawable.c	16 Nov 2004 22:37:39 -0000	1.2.2.1
+++ scribus/gdk-pixbuf-xlib-drawable.c	7 Mar 2005 00:04:29 -0000
@@ -1,3 +1,4 @@
+#ifndef QT_MAC
 /* -*- Mode: C; tab-width: 8; indent-tabs-mode: t; c-basic-offset: 8 -*- */
 /* GdkPixbuf library - convert X drawable information to RGB
  *
@@ -1136,3 +1137,4 @@ xlib_window_get_origin (Window w, int *x
 						  gdk_pixbuf_screen),
 				      0, 0, x, y, &child);
 }
+#endif
Index: scribus/gdk-pixbuf-xlib-private.h
===================================================================
RCS file: /cvs/Scribus/scribus/gdk-pixbuf-xlib-private.h,v
retrieving revision 1.3.2.1
diff -u -p -r1.3.2.1 gdk-pixbuf-xlib-private.h
--- scribus/gdk-pixbuf-xlib-private.h	16 Nov 2004 22:37:39 -0000	1.3.2.1
+++ scribus/gdk-pixbuf-xlib-private.h	7 Mar 2005 00:04:29 -0000
@@ -1,3 +1,4 @@
+#ifndef QT_MAC
 /* GdkPixbuf library - Xlib header file
  *
  * Authors: John Harper <john@dcs.warwick.ac.uk>
@@ -37,3 +38,4 @@ extern int gdk_pixbuf_screen;
 #define KSVG_BYTE_ORDER KSVG_LITTLE_ENDIAN
 #endif 
 #endif
+#endif
Index: scribus/gdk-pixbuf-xlib-render.c
===================================================================
RCS file: /cvs/Scribus/scribus/gdk-pixbuf-xlib-render.c,v
retrieving revision 1.1
diff -u -p -r1.1 gdk-pixbuf-xlib-render.c
--- scribus/gdk-pixbuf-xlib-render.c	1 Oct 2003 20:39:09 -0000	1.1
+++ scribus/gdk-pixbuf-xlib-render.c	7 Mar 2005 00:04:29 -0000
@@ -1,3 +1,4 @@
+#ifndef QT_MAC
 /* GdkPixbuf library - Rendering functions
  *
  * Copyright (C) 1999 The Free Software Foundation
@@ -378,3 +379,4 @@ gdk_pixbuf_xlib_render_pixmap_and_mask (
 			*mask_return = 0;
 	}
 }*/
+#endif
Index: scribus/gdk-pixbuf-xlib.c
===================================================================
RCS file: /cvs/Scribus/scribus/gdk-pixbuf-xlib.c,v
retrieving revision 1.1
diff -u -p -r1.1 gdk-pixbuf-xlib.c
--- scribus/gdk-pixbuf-xlib.c	1 Oct 2003 20:39:09 -0000	1.1
+++ scribus/gdk-pixbuf-xlib.c	7 Mar 2005 00:04:29 -0000
@@ -1,3 +1,4 @@
+#ifndef QT_MAC
 /* GdkPixbuf library - Initialization functions
  *
  * Author: John Harper <john@dcs.warwick.ac.uk>
@@ -61,3 +62,4 @@ gdk_pixbuf_xlib_init_with_depth (Display
     gdk_pixbuf_dpy = display;
     gdk_pixbuf_screen = screen_num;
 }
+#endif
Index: scribus/gdk-pixbuf-xlib.h
===================================================================
RCS file: /cvs/Scribus/scribus/gdk-pixbuf-xlib.h,v
retrieving revision 1.1
diff -u -p -r1.1 gdk-pixbuf-xlib.h
--- scribus/gdk-pixbuf-xlib.h	1 Oct 2003 20:39:09 -0000	1.1
+++ scribus/gdk-pixbuf-xlib.h	7 Mar 2005 00:04:29 -0000
@@ -1,3 +1,4 @@
+#ifndef QT_MAC
 /* GdkPixbuf library - Xlib header file
  *
  * Authors: John Harper <john@dcs.warwick.ac.uk>
@@ -78,3 +79,4 @@ GdkPixbuf *gdk_pixbuf_xlib_get_from_draw
 					      int width, int height);
 */
 #endif /* GDK_PIXBUF_XLIB_H */
+#endif
Index: scribus/gdk-pixbuf-xlibrgb.c
===================================================================
RCS file: /cvs/Scribus/scribus/gdk-pixbuf-xlibrgb.c,v
retrieving revision 1.1.2.2
diff -u -p -r1.1.2.2 gdk-pixbuf-xlibrgb.c
--- scribus/gdk-pixbuf-xlibrgb.c	16 Nov 2004 22:37:39 -0000	1.1.2.2
+++ scribus/gdk-pixbuf-xlibrgb.c	7 Mar 2005 00:04:34 -0000
@@ -1,3 +1,4 @@
+#ifndef QT_MAC
 /* -*- Mode: C++; tab-width: 2; indent-tabs-mode: nil; c-basic-offset: 2 -*-
  *
  * The contents of this file are subject to the Mozilla Public License
@@ -3722,3 +3723,4 @@ xlib_rgb_get_screen (void)
   
   return NULL;
 }
+#endif
\ No newline at end of file
Index: scribus/gdk-pixbuf-xlibrgb.h
===================================================================
RCS file: /cvs/Scribus/scribus/gdk-pixbuf-xlibrgb.h,v
retrieving revision 1.1
diff -u -p -r1.1 gdk-pixbuf-xlibrgb.h
--- scribus/gdk-pixbuf-xlibrgb.h	1 Oct 2003 20:39:09 -0000	1.1
+++ scribus/gdk-pixbuf-xlibrgb.h	7 Mar 2005 00:04:34 -0000
@@ -1,3 +1,4 @@
+#ifndef QT_MAC
 /* -*- Mode: C++; tab-width: 2; indent-tabs-mode: nil; c-basic-offset: 2 -*-
  *
  * The contents of this file are subject to the Mozilla Public License
@@ -192,3 +193,4 @@ xlib_rgb_get_screen (void);
 
 
 #endif /* __XLIB_RGB_H__ */
+#endif
Index: scribus/main.cpp
===================================================================
RCS file: /cvs/Scribus/scribus/main.cpp,v
retrieving revision 1.16.2.10
diff -u -p -r1.16.2.10 main.cpp
--- scribus/main.cpp	29 Jan 2005 00:29:47 -0000	1.16.2.10
+++ scribus/main.cpp	7 Mar 2005 00:04:35 -0000
@@ -61,6 +61,8 @@
 #define ARG_NOGUI_SHORT "-g"
 #define ARG_DISPLAY_SHORT "-d"
 
+#define ARG_PSN "-psn_"
+
 
 QString lang = "";
 bool showSplash = true;
@@ -104,6 +106,10 @@ int main(int argc, char *argv[])
 			showSplash = false;
 		} else if (arg == ARG_NOGUI || arg == ARG_NOGUI_SHORT) {
 			useGui = false;
+#ifdef QT_MAC
+		} else if (strncmp(arg,ARG_PSN,4) == 0) {
+		// QT/Mac expects the option -psn_blablabla
+#endif			
 		} else if ((arg == ARG_DISPLAY || arg==ARG_DISPLAY_SHORT) && ++i < argc) {
 		// allow setting of display, QT expect the
 		// option -display <display_name>
Index: scribus/scfonts.cpp
===================================================================
RCS file: /cvs/Scribus/scribus/scfonts.cpp,v
retrieving revision 1.15.2.10
diff -u -p -r1.15.2.10 scfonts.cpp
--- scribus/scfonts.cpp	30 Jan 2005 13:16:04 -0000	1.15.2.10
+++ scribus/scfonts.cpp	7 Mar 2005 00:04:35 -0000
@@ -19,8 +19,10 @@
 #include <qdict.h>
 #include <qmap.h>
 #include <qdir.h>
+#ifndef QT_MAC
 #include <X11/X.h>
 #include <X11/Xlib.h>
+#endif
 #include <qregexp.h>
 
 #include <cstdlib>
@@ -511,6 +513,7 @@ void SCFonts::AddFontconfigFonts()
 }
 
 #else
+#ifndef QT_MAC
 
 void SCFonts::AddXFontPath()
 {
@@ -576,6 +579,7 @@ void SCFonts::AddXFontServerPath()
 	}
 }
 #endif
+#endif
 
 /* Add an extra path to the X-Server's Fontpath
  * allowing a user to have extra Fonts installed
@@ -609,8 +613,10 @@ void SCFonts::GetFonts(QString pf)
 		AddScalableFonts(fpi.current());
 	AddFontconfigFonts();
 #else
+#ifndef QT_MAC
 	AddXFontPath();
 	AddXFontServerPath();
+#endif
 	for(QStrListIterator fpi(FontPath) ; fpi.current() ; ++fpi)
 		AddScalableFonts(fpi.current());
 #endif
Index: scribus/scfonts.h
===================================================================
RCS file: /cvs/Scribus/scribus/scfonts.h,v
retrieving revision 1.4.2.4
diff -u -p -r1.4.2.4 scfonts.h
--- scribus/scfonts.h	30 Jan 2005 13:16:04 -0000	1.4.2.4
+++ scribus/scfonts.h	7 Mar 2005 00:04:35 -0000
@@ -116,9 +116,11 @@ class SCFonts : public QDict<Foi>
 #ifdef HAVE_FONTCONFIG
 		void AddFontconfigFonts();
 #else
+#ifndef QT_MAC
 		void AddXFontServerPath();
 		void AddXFontPath();
 #endif
+#endif
 		QStrList FontPath;
 		QString ExtraPath;
 };
Index: scribus/scpainter.h
===================================================================
RCS file: /cvs/Scribus/scribus/scpainter.h,v
retrieving revision 1.7.2.2
diff -u -p -r1.7.2.2 scpainter.h
--- scribus/scpainter.h	16 Nov 2004 22:37:40 -0000	1.7.2.2
+++ scribus/scpainter.h	7 Mar 2005 00:04:36 -0000
@@ -155,8 +155,9 @@ private:
 /* Zoom Factor of the Painter */
 	double m_zoomFactor;
 	struct _ArtSVP *m_clipPath;
-
+#ifndef QT_MAC
 	GC gc;
+#endif
 };
 
 #endif
Index: scribus/scpainter.cpp
===================================================================
RCS file: /cvs/Scribus/scribus/scpainter.cpp,v
retrieving revision 1.19.2.5
diff -u -b -w -r1.19.2.5 scpainter.cpp
--- scribus/scpainter.cpp	6 Mar 2005 13:16:21 -0000	1.19.2.5
+++ scribus/scpainter.cpp	7 Mar 2005 09:37:10 -0000
@@ -53,8 +53,10 @@
 #include "art_rgb_svp.h"
 #include "art_render_pattern.h"
 
+#ifndef QT_MAC
 #include <X11/Xlib.h>
 #include <gdk-pixbuf-xlibrgb.h>
+#endif
 
 #include <math.h>
 #include "art_kmisc.h"
@@ -91,8 +93,10 @@
 	PLineJoin = Qt::RoundJoin;
 	fill_gradient = VGradient(VGradient::linear);
 	stroke_gradient = VGradient(VGradient::linear);
+#ifndef QT_MAC
 	xlib_rgb_init_with_depth( target->x11Display(), XScreenOfDisplay( target->x11Display(), target->x11Screen() ), target->x11Depth() );
 	gc = XCreateGC( target->x11Display(), target->handle(), 0, 0 );
+#endif
 	m_zoomFactor = 1;
 }
 
@@ -104,8 +108,10 @@
 		art_free( m_buffer );
 	if( m_path )
 		art_free( m_path );
+#ifndef QT_MAC
 	if( gc )
 		XFreeGC( m_target->x11Display(), gc );
+#endif
 }
 
 void ScPainter::resize( unsigned int w, unsigned int h )
@@ -129,8 +135,26 @@
 
 void ScPainter::end()
 {
+#ifndef QT_MAC
 	xlib_draw_rgb_32_image( m_target->handle(), gc, m_x, m_y, m_width, m_height, XLIB_RGB_DITHER_NONE, m_buffer, m_width * 4 );
 //	xlib_draw_rgb_32_image( m_target->handle(), gc, 0, 0, m_width, m_height, XLIB_RGB_DITHER_NONE, m_buffer, m_width * 4 );
+#else
+// Unfortunately Qt insists on ARGB and libart on RGBA :-(
+//        QImage qimg(m_buffer, m_width, m_height, 32, 0, 0, QImage::BigEndian);
+        QImage qimg(m_width, m_height, 32, QImage::BigEndian);
+        QRgb * bits = (QRgb *) qimg.bits();
+        int words = qimg.numBytes() / 4;
+        art_u8 * p = m_buffer;;
+        for (int i=0; i < words; ++i) {
+                art_u8 r = *p++;
+                art_u8 g = *p++;
+                art_u8 b = *p++;
+                art_u8 a = *p++;
+                *bits++ = qRgba(r,g,b,a);       
+        }
+        bitBlt(m_target, 0, 0, &qimg);
+
+#endif
 }
 
 void ScPainter::clear()
