Index: admin/Makefile.common
===================================================================
RCS file: /Volumes/src/cvs/od/proj/KDE-Darwin/admin/Makefile.common,v
retrieving revision 1.1.1.4
retrieving revision 1.5
diff -u -u -r1.1.1.4 -r1.5
Index: admin/acinclude.m4.in
===================================================================
RCS file: /Volumes/src/cvs/od/proj/KDE-Darwin/admin/acinclude.m4.in,v
retrieving revision 1.1.1.5
diff -u -u -r1.1.1.5 acinclude.m4.in
--- admin/acinclude.m4.in	29 Oct 2002 02:27:18 -0000	1.1.1.5
+++ admin/acinclude.m4.in	31 Oct 2002 18:47:42 -0000
@@ -704,13 +704,10 @@
      AC_CHECK_LIB(ipc, shmat, X_EXTRA_LIBS="$X_EXTRA_LIBS -lipc"))
    
    # darwin needs this to initialize the environment
-   AC_CHECK_HEADERS(crt_externs.h)
+   AC_CHECK_HEADERS([crt_externs.h sys/types.h])
    AC_CHECK_FUNC(_NSGetEnviron, [AC_DEFINE(HAVE_NSGETENVIRON, 1, [Define if your system needs _NSGetEnviron to set up the environment])])
  
-   # more headers that need to be explicitly included on darwin
-   AC_CHECK_HEADERS(sys/types.h stdint.h)
-
-   # darwin requires a poll emulation library
+   # darwin requires a poll.h emulation library
    AC_CHECK_LIB(poll, poll, LIB_POLL="-lpoll")
 
    # CoreAudio framework
@@ -719,11 +716,36 @@
      FRAMEWORK_COREAUDIO="-framework CoreAudio"
    ])
 
+   # CoreServices framework
+   AC_CHECK_HEADER(CoreServices/CoreServices.h, [
+     AC_DEFINE(HAVE_CORESERVICES, 1, [Define if you have the CoreServices API])
+     FRAMEWORK_CORESERVICES="-framework CoreServices"
+   ])
+
+   # Solaris 2.6 and others need -lresolv for res_init
+   AC_CHECK_FUNCS(res_init, , [
+     kde_libs_safe="$LIBS"
+     LIBS="$LIBS $X_EXTRA_LIBS -lresolv"
+     AC_TRY_LINK(
+[
+#include <resolv.h>
+],
+[ 
+res_init(); 
+],
+        LIBRESOLV="-lresolv"
+        X_EXTRA_LIBS="$X_EXTRA_LIBS $LIBRESOLV"
+        AC_DEFINE(HAVE_RES_INIT, 1, [Define if you have the res_init function])
+     )
+     LIBS=$kde_libs_safe
+   ])
+
    AC_CHECK_RES_INIT
-   AC_SUBST(LIB_POLL)
    AC_SUBST(FRAMEWORK_COREAUDIO)
    LIBSOCKET="$X_EXTRA_LIBS"
    AC_SUBST(LIBSOCKET)
+   AC_SUBST(FRAMEWORK_CORESERVICES)
+   AC_SUBST(LIB_POLL)
    AC_SUBST(LIBRESOLV)
    AC_SUBST(X_EXTRA_LIBS)
    AC_CHECK_LIB(ucb, killpg, [LIBUCB="-lucb"]) dnl for Solaris2.4
@@ -932,7 +954,7 @@
 kde_ldflags_safe="$LDFLAGS"
 kde_libs_safe="$LIBS"
 
-LDFLAGS="$LDFLAGS $X_LDFLAGS $USER_LDFLAGS"
+LDFLAGS="$LDFLAGS $USER_LDFLAGS $X_LDFLAGS"
 LIBS="-lXext -lX11 $LIBSOCKET"
 
 AC_TRY_LINK([
@@ -974,9 +996,9 @@
 kde_save_LDFLAGS="$LDFLAGS"
 kde_save_CFLAGS="$CFLAGS"
 kde_save_LIBS="$LIBS"
-LDFLAGS="$LDFLAGS $X_LDFLAGS $USER_LDFLAGS"
+LDFLAGS="$LDFLAGS $USER_LDFLAGS $X_LDFLAGS"
 CFLAGS="$CFLAGS -I$x_includes"
-LIBS="-lXinerama -lXext"
+LIBS="-lXinerama -lXext -lX11"
 
 if test "x$no_xinerama" = "xno"; then
 
@@ -1907,15 +1929,29 @@
    AC_REQUIRE([K_PATH_X])
 
 if test $kde_qtver = 3; then
+   AC_SUBST(LIB_ARTSKDE, "-lartskde")
+   AC_SUBST(LIB_DCOP, "-lDCOP")
+   AC_SUBST(LIB_KAB, "-lkab")
+   AC_SUBST(LIB_KABC, "-lkabc")
    AC_SUBST(LIB_KDECORE, "-lkdecore")
+   AC_SUBST(LIB_KDED, "@PREFIX@/lib/kded.la")
+   AC_SUBST(LIB_KDEFX, "-lkdefx")
+   AC_SUBST(LIB_KDEPRINT, "-lkdeprint")
+   AC_SUBST(LIB_KDESASL, "-lkdesasl")
+   AC_SUBST(LIB_KDESU, "-lkdesu")
    AC_SUBST(LIB_KDEUI, "-lkdeui")
-   AC_SUBST(LIB_KIO, "-lkio")
-   AC_SUBST(LIB_SMB, "-lsmb")
-   AC_SUBST(LIB_KAB, "-lkab")
    AC_SUBST(LIB_KHTML, "-lkhtml")
-   AC_SUBST(LIB_KSPELL, "-lkspell")
+   AC_SUBST(LIB_KIO, "-lkio")
    AC_SUBST(LIB_KPARTS, "-lkparts")
-   AC_SUBST(LIB_KDEPRINT, "-lkdeprint")
+   AC_SUBST(LIB_KJAVA, "-lkjava")
+   AC_SUBST(LIB_KJS, "-lkjs")
+   AC_SUBST(LIB_KMID, "-lkmid")
+   AC_SUBST(LIB_KSCRIPT, "-lkscript")
+   AC_SUBST(LIB_KSPELL, "-lkspell")
+   AC_SUBST(LIB_KTEXTEDITOR, "-lktexteditor")
+   AC_SUBST(LIB_SHELLSCRIPT, "-lshellscript")
+   AC_SUBST(LIB_SMB, "-lsmb")
+   AC_SUBST(LIB_VCARD, "-lvcard")
 # these are for backward compatibility
    AC_SUBST(LIB_KSYCOCA, "-lkio")
    AC_SUBST(LIB_KFILE, "-lkio")
@@ -2646,10 +2682,10 @@
       if test $kde_use_debug_code = "full"; then
         CFLAGS="-g3 $CFLAGS"
       else
-        CFLAGS="-g -O2 $CFLAGS"
+        CFLAGS="-g -O $CFLAGS"
       fi
     else
-      CFLAGS="-O2 $CFLAGS"
+      CFLAGS="-Os -frename-registers -mmultiple -mcpu=750 -mtune=750 $CFLAGS"
     fi
   fi
 
@@ -2679,7 +2715,7 @@
         if test "$kde_use_debug_code" = "full"; then
           CXXFLAGS="-g3 $CXXFLAGS"
         else
-          CXXFLAGS="-g -O2 $CXXFLAGS"
+          CXXFLAGS="-g -O $CXXFLAGS"
         fi
       fi
       KDE_CHECK_COMPILER_FLAG(fno-builtin,[CXXFLAGS="-fno-builtin $CXXFLAGS"])
@@ -2691,7 +2727,7 @@
       if test "$CXX" = "KCC"; then
         CXXFLAGS="+K3 $CXXFLAGS"
       else
-        CXXFLAGS="-O2 $CXXFLAGS"
+        CXXFLAGS="-Os -frename-registers -mmultiple -mcpu=750 -mtune=750 $CXXFLAGS"
       fi  
     fi
   fi
@@ -3188,9 +3224,9 @@
     ac_save_ldflags="$LDFLAGS"
     ac_save_cflags="$CFLAGS"
     if test "x$kde_use_qt_emb" != "xyes"; then
-      LDFLAGS="$LDFLAGS $X_LDFLAGS $USER_LDFLAGS $LDFLAGS $XPM_LDFLAGS $all_libraries -lXpm -lX11 -lXext $LIBZ $LIBSOCKET"
+      LDFLAGS="$LDFLAGS $USER_LDFLAGS $X_LDFLAGS $LDFLAGS $XPM_LDFLAGS $all_libraries -lXpm -lX11 -lXext $LIBZ $LIBSOCKET"
     else
-      LDFLAGS="$LDFLAGS $X_LDFLAGS $USER_LDFLAGS $LDFLAGS $XPM_LDFLAGS $all_libraries -lXpm $LIBZ $LIBSOCKET"
+      LDFLAGS="$LDFLAGS $USER_LDFLAGS $X_LDFLAGS $LDFLAGS $XPM_LDFLAGS $all_libraries -lXpm $LIBZ $LIBSOCKET"
     fi
     CFLAGS="$CFLAGS $X_INCLUDES $USER_INCLUDES"
     test -n "$XPM_INCLUDE" && CFLAGS="-I$XPM_INCLUDE $CFLAGS"
@@ -4625,6 +4661,83 @@
 	done
 ])
 
+AC_DEFUN(KDE_CHECK_LDAP,
+[
+  AC_MSG_CHECKING([for LDAP])
+
+  if test -n "$1"; then
+    ldap_extra_dirs="$1"
+  fi
+
+  ldap_incdirs="$ldap_extra_dirs/include /usr/include /usr/local/include/ $kde_extra_includes"
+  AC_FIND_FILE(ldap.h, $ldap_incdirs, ldap_incdir)
+  if test ! -r $ldap_incdir/ldap.h; then
+    ldap_incdir=no
+  fi
+
+  ldap_libdirs="$ldap_extra_dirs/lib /usr/lib /usr/local /usr/lib $kde_extra_libs"
+  for ext in la so sl dylib a; do
+    AC_FIND_FILE(libldap.$ext, $ldap_libdirs, ldap_libdir_test)
+    if test -r $ldap_libdir_test/libldap.$ext; then
+      ldap_libdir=$ldap_libdir_test
+      break
+    fi
+  done
+  if test -z "$ldap_libdir"; then
+    ldap_libdir=no
+  fi
+  if test x$ldap_libdir != xno && test x$ldap_incdir != xno; then
+    LDAP_INCS=-I$ldap_incdir
+    LDAP_LIBS=-L$ldap_libdir
+    # on MacOSX, we link against the framework instead of
+    # directly to the libraries, for forward-compatibility
+    if test -d "/System/Library/Frameworks/LDAP.framework"; then
+      LIBLDAP="-framework LDAP"
+    else
+      LIBLDAP="-lldap -llber -lresolv"
+    fi
+    if test "$USE_RPATH" = "yes"; then
+      LDAP_RPATH="-R $LDAPLIBDIR/lib"
+    fi
+
+    kde_safe_LIBS="$LIBS"
+    kde_safe_CFLAGS="$CFLAGS"
+    LIBS="$LIBS $all_libraries $LDAP_LIBS $LIBLDAP $KERBEROS_LIBS $X_EXTRA_LIBS"
+    CFLAGS="$CFLAGS $all_includes $LDAP_INCS $KERBEROS_INCS"
+    AC_LANG_SAVE
+    AC_LANG_C
+    AC_TRY_LINK(
+      [#include <lber.h>
+#include <ldap.h>
+#if LDAP_VENDOR_VERSION < 20000
+#error OpenLDAP version too old, please upgrade to version 2 or higher
+#endif
+],
+      [ LDAP *ldap; ],
+      [ ldap_old=no ],
+      [ ldap_old=yes ]
+    )
+    LIBS="$kde_safe_LIBS"
+    CFLAGS="$kde_safe_CFLAGS"
+
+    if test x$ldap_old = xno; then
+      AC_SUBST(LDAP_INCS)
+      AC_SUBST(LDAP_LIBS)
+      AC_SUBST(LDAP_RPATH)
+      AC_SUBST(LIBLDAP)
+      AC_DEFINE_UNQUOTED(HAVE_LIBLDAP,1,[Define if you have LDAP libraries])
+    else
+      LDAP_INCS=
+      LDAP_LIBS=
+      LDAP_RPATH=
+      LIBLDAP=
+    fi
+    AC_MSG_RESULT([libraries $ldap_libdir, headers $ldap_incdir])
+  else
+    AC_MSG_RESULT([not found])
+  fi
+])
+
 dnl KDE_CHEC_JAVA_DIR(onlyjre)
 AC_DEFUN(KDE_CHECK_JAVA_DIR,
 [
@@ -4659,6 +4772,7 @@
       KDE_JAVA_PREFIX(/usr/lib/IBMJava2*)
       KDE_JAVA_PREFIX(/usr/lib/IBMJava*)
       KDE_JAVA_PREFIX(/opt/java*)
+      KDE_JAVA_PREFIX(/Library/Java/Home)
     
       kde_cv_path="NONE"
       kde_save_IFS=$IFS
@@ -4710,10 +4824,20 @@
 dnl At this point kde_java_bindir and kde_java_includedir are either set or "no"
 if test "x$kde_java_bindir" != "xno"; then
 
+  case $host_os in
+    darwin*)
+      dnl Look for libjvm.dylib
+      kde_java_libjvmdir=`find $kde_java_bindir/.. -name libjvm.dylib | sed 's,libjvm.dylib,,'|head -n 1`
+      kd_java_libhpidir=''
+      dnl Darwin doesn't have libhpi
+      ;;
+    *)
   dnl Look for libjvm.so
   kde_java_libjvmdir=`find $kde_java_bindir/.. -name libjvm.so | sed 's,libjvm.so,,'|head -n 1`
   dnl Look for libhpi.so and avoid green threads
-  kde_java_libhpidir=`find $kde_java_bindir/.. -name libhpi.so | grep -v green | sed 's,libhpi.so,,' | head -n 1`
+  kde_java_libhpidir=`find $kde_java_bindir/.. -name libhpi.so | grep -v green | sed 's,libhpi.so,,' | tail -n 1`
+      ;;
+  esac
 
   dnl Now check everything's fine under there
   dnl the include dir is our flag for having the JDK
@@ -4742,17 +4866,29 @@
     JAVAC=
     jni_includes=
   fi
-
+  case $host_os in
+    darwin*)
+      if test ! -r "$kde_java_libjvmdir/libjvm.dylib"; then
+        AC_MSG_ERROR([libjvm.dylib not found under $kde_java_libjvmdir. Use --without-java.])
+      fi
+      dnl Darwin doesn't have libhpi, but it's not fatal
+      LIB_JVM="-framework JavaVM"
+      LIB_HPI=""
+      ;;
+    *)
   if test ! -r "$kde_java_libjvmdir/libjvm.so"; then
      AC_MSG_ERROR([libjvm.so not found under $kde_java_libjvmdir. Use --without-java.])
   fi 
+      LIB_JVM="-L$kde_java_libjvmdir -ljvm"
+      if test ! -r "$kde_java_libhpidir/libhpi.so"; then
+        AC_MSG_ERROR([libhpi.so not found under $kde_java_libhpidir. Use --without-java.])
+      fi
+      LIB_HPI="-L$kde_java_libhpidir -lhpi"
+      ;;
+  esac
 
   if test ! -x "$kde_java_bindir/java"; then
       AC_MSG_ERROR([java not found under $kde_java_bindir. javac was found though! Use --with-java or --without-java.])
-  fi
-
-  if test ! -r "$kde_java_libhpidir/libhpi.so"; then
-    AC_MSG_ERROR([libhpi.so not found under $kde_java_libhpidir. Use --without-java.])
   fi
 
   if test -n "$jni_includes"; then
Index: admin/detect-autoconf.sh
===================================================================
RCS file: /Volumes/src/cvs/od/proj/KDE-Darwin/admin/detect-autoconf.sh,v
retrieving revision 1.1.1.2
retrieving revision 1.3
diff -u -u -r1.1.1.2 -r1.3
--- admin/detect-autoconf.sh	29 Oct 2002 02:27:20 -0000	1.1.1.2
+++ admin/detect-autoconf.sh	29 Oct 2002 02:29:54 -0000	1.3
@@ -42,12 +42,12 @@
 checkAutomakeAclocal ()
 {
   if test -z "$UNSERMAKE"; then
-    if test -x "`$WHICH automake-1.5`" ; then
-      AUTOMAKE="`$WHICH automake-1.5`"
-      ACLOCAL="`$WHICH aclocal-1.5`"
-    elif test -x "`$WHICH automake-1.6`" ; then
+    if test -x "`$WHICH automake-1.6`" ; then
       AUTOMAKE="`$WHICH automake-1.6`"
       ACLOCAL="`$WHICH aclocal-1.6`"
+    elif test -x "`$WHICH automake-1.5`" ; then
+      AUTOMAKE="`$WHICH automake-1.5`"
+      ACLOCAL="`$WHICH aclocal-1.5`"
     fi
   else
      AUTOMAKE="$UNSERMAKE"
Index: admin/libtool.m4.in
===================================================================
RCS file: /Volumes/src/cvs/od/proj/KDE-Darwin/admin/libtool.m4.in,v
retrieving revision 1.1.1.3
retrieving revision 1.7
diff -u -u -r1.1.1.3 -r1.7
--- admin/libtool.m4.in	14 Aug 2002 13:18:45 -0000	1.1.1.3
+++ admin/libtool.m4.in	16 Sep 2002 14:31:34 -0000	1.7
@@ -609,6 +609,12 @@
     lt_cv_sys_max_cmd_len=-1;
     ;;
 
+  darwin*)
+    # this should be enough... the length test freaks out some versions
+    # of OSX's `expr`
+    lt_cv_sys_max_cmd_len=16384;
+    ;;
+
   *)
     # If test is not a shell built-in, we'll probably end up computing a
     # maximum length that is only half of the actual maximum length, but
@@ -1132,7 +1138,7 @@
 darwin* | rhapsody*)
   dynamic_linker="$host_os dyld"
   version_type=darwin
-  need_lib_prefix=no
+  need_lib_prefix=yes
   need_version=no
   # FIXME: Relying on posixy $() will cause problems for
   #        cross-compilation, but unfortunately the echo tests do not
@@ -1901,16 +1907,7 @@
   ;;
 
 darwin* | rhapsody*)
-  lt_cv_deplibs_check_method='file_magic Mach-O dynamically linked shared library'
-  lt_cv_file_magic_cmd='/usr/bin/file -L'
-  case "$host_os" in
-  rhapsody* | darwin1.[[012]])
-    lt_cv_file_magic_test_file=`/System/Library/Frameworks/System.framework/System`
-    ;;
-  *) # Darwin 1.3 on
-    lt_cv_file_magic_test_file='/usr/lib/libSystem.dylib'
-    ;;
-  esac
+  lt_cv_deplibs_check_method='pass_all'
   ;;
 
 freebsd*)
@@ -2560,6 +2557,11 @@
 	;;
     esac
     ;;
+  darwin*)
+    shared_flag='-dynamiclib'
+    _LT_AC_TAGVAR(archive_cmds, $1)='$CC -r -keep_private_externs -nostdlib -o ${lib}-master.o $libobjs && $CC -dynamiclib -install_name $rpath/$soname $predep_objects ${lib}-master.o $deplibs $postdep_objects $compiler_flags -o $lib'
+    output_verbose_link_cmd='$CC -dynamiclib $CFLAGS -v conftest.$objext 2>&1 | egrep "\-L"'
+    ;;
   dgux*)
     case $cc_basename in
       ec++)
@@ -4750,7 +4752,9 @@
 	_LT_AC_TAGVAR(allow_undefined_flag, $1)='-undefined suppress'
 	;;
       *) # Darwin 1.3 on
-	_LT_AC_TAGVAR(allow_undefined_flag, $1)='-flat_namespace -undefined suppress'
+	#_LT_AC_TAGVAR(allow_undefined_flag, $1)='-flat_namespace -undefined suppress'
+        # testing with twolevel namespaces
+	_LT_AC_TAGVAR(allow_undefined_flag, $1)=unsupported
 	;;
       esac
 
@@ -4758,7 +4762,7 @@
       #        cross-compilation, but unfortunately the echo tests do not
       #        yet detect zsh echo's removal of \ escapes.  Also zsh mangles
       #	       `"' quotes if we put them in here... so don't!
-      _LT_AC_TAGVAR(archive_cmds, $1)='$CC $(test .$module = .yes && echo -bundle || echo -dynamiclib) $allow_undefined_flag -o $lib $libobjs $deplibs$linker_flags -install_name $rpath/$soname $verstring'
+      _LT_AC_TAGVAR(archive_cmds, $1)='$CC -r -keep_private_externs -nostdlib -o ${lib}-master.o $libobjs && $CC $(test .$module = .yes && echo -bundle || echo -dynamiclib) $allow_undefined_flag -o $lib ${lib}-master.o $deplibs $linker_flags $(test x$module != xyes && echo -install_name $rpath/$soname $verstring)'
       # We need to add '_' to the symbols in $export_symbols first
       #_LT_AC_TAGVAR(archive_expsym_cmds, $1)="$_LT_AC_TAGVAR(archive_cmds, $1)"' && strip -s $export_symbols'
       _LT_AC_TAGVAR(hardcode_direct, $1)=yes
Index: admin/ltmain.sh
===================================================================
RCS file: /Volumes/src/cvs/od/proj/KDE-Darwin/admin/ltmain.sh,v
retrieving revision 1.1.1.2
retrieving revision 1.4
diff -u -u -r1.1.1.2 -r1.4
--- admin/ltmain.sh	10 Jul 2002 02:47:39 -0000	1.1.1.2
+++ admin/ltmain.sh	13 Jul 2002 04:33:25 -0000	1.4
@@ -3321,7 +3321,23 @@
 	if test -n "$export_symbols" && test -n "$archive_expsym_cmds"; then
 	  eval cmds=\"$archive_expsym_cmds\"
 	else
+          if test "x$verstring" = "x0.0"; then
+                tmp_verstring=
+          else
+                tmp_verstring="$verstring"
+          fi
+ 	  save_deplibs="$deplibs"
+ 	  for conv in $convenience; do
+            tmp_deplibs=
+            for test_deplib in $deplibs; do
+              if test "$test_deplib" != "$conv"; then
+                tmp_deplibs="$tmp_deplibs $test_deplib"
+              fi
+            done
+            deplibs="$tmp_deplibs"
+ 	  done
 	  eval cmds=\"$archive_cmds\"
+ 	  deplibs="$save_deplibs"
 	fi
 
 	if len=`expr "X$cmds" : ".*"` &&
@@ -3419,7 +3435,12 @@
 	  if test -n "$export_symbols" && test -n "$archive_expsym_cmds"; then
 	    eval cmds=\"$archive_expsym_cmds\"
 	  else
-	    eval cmds=\"$archive_cmds\"
+            save_deplibs="$deplibs"
+            for conv in $convenience; do
+              deplibs="${deplibs%$conv*} ${deplibs#*$conv}"
+            done
+            eval cmds=\"$archive_cmds\"
+            deplibs="$save_deplibs"
 	  fi
 
 	  # Append the command to remove the reloadable object files
@@ -3632,6 +3653,14 @@
 	compile_deplibs=`$echo "X $compile_deplibs" | $Xsed -e 's/ -lc / -framework System /'`
 	finalize_deplibs=`$echo "X $finalize_deplibs" | $Xsed -e 's/ -lc / -framework System /'`
 	;;
+      esac
+
+      case $host in
+      *darwin*)
+        # Don't allow lazy linking, it breaks C++ global constructors
+        compile_command="$compile_command ${wl}-bind_at_load"
+        finalize_command="$finalize_command ${wl}-bind_at_load"
+        ;;
       esac
 
       compile_command="$compile_command $compile_deplibs"
Index: admin/missing
===================================================================
RCS file: /Volumes/src/cvs/od/proj/KDE-Darwin/admin/missing,v
retrieving revision 1.1.1.3
retrieving revision 1.5
diff -u -u -r1.1.1.3 -r1.5
Index: Makefile.am.in
===================================================================
RCS file: /Volumes/src/cvs/od/proj/KDE-Darwin/arts/Makefile.am.in,v
retrieving revision 1.1.1.2
retrieving revision 1.3
diff -u -u -r1.1.1.2 -r1.3
--- Makefile.am.in	1 Oct 2002 01:47:59 -0000	1.1.1.2
+++ Makefile.am.in	1 Oct 2002 01:54:35 -0000	1.3
@@ -27,7 +27,7 @@
 	cd $(top_srcdir) && $(MAKE) -f admin/Makefile.common subdirs 
 
 AUTOMAKE_OPTIONS = foreign 1.5
-COMPILE_FIRST = libltdl mcop mcopidl flow mcop_mt soundserver artsc examples tests doc 
+COMPILE_FIRST = libltdl mcop mcopidl flow mcop_mt x11 soundserver artsc examples tests doc 
 EXTRA_DIST = admin
 
 dist-hook:
Index: configure.in.in
===================================================================
RCS file: /Volumes/src/cvs/od/proj/KDE-Darwin/arts/configure.in.in,v
retrieving revision 1.1.1.5
diff -u -u -r1.1.1.5 configure.in.in
--- configure.in.in	29 Oct 2002 02:31:10 -0000	1.1.1.5
+++ configure.in.in	31 Oct 2002 18:50:02 -0000
@@ -282,6 +282,19 @@
 fi
 ])
 
+dnl needed for mach realtime scheduling
+AC_DEFUN([AC_CHECK_MACH_REALTIME_SCHED],
+[
+  AC_CHECK_HEADERS([mach/mach.h mach/thread_policy.h sys/param.h sys/sysctl.h],
+    AC_CHECK_FUNCS(thread_policy_set,
+      arts_mach_realtime_sched=yes,
+      arts_mach_realtime_sched=no)
+  )
+  if test "x$arts_mach_realtime_sched" = "xyes"; then
+    AC_DEFINE(HAVE_MACH_REALTIME_SCHED,1, [Define if your system supports mach realtime scheduling])
+  fi
+])
+
 dnl Type of the ioctl function test - after some tries, it seems that this
 dnl not required for Linux vs. FreeBSD (for which this test was written), and
 dnl that only the Linux documentation claims that it has an "int" as second
@@ -432,6 +445,7 @@
 AC_CHECK_LIBAUDIOFILE
 AC_CHECK_SGILIBAUDIO
 AC_CHECK_REALTIME_SCHED
+AC_CHECK_MACH_REALTIME_SCHED
 AC_CHECK_GETDOMAINNAME
 AC_CHECK_IOCTL_TYPE
 AC_CHECK_X86_FLOAT_INT
@@ -604,6 +618,27 @@
 AC_CHECK_HEADERS(soundcard.h)
 AC_CHECK_LIB(ossaudio, _oss_ioctl, [LIBOSSAUDIO="-lossaudio"])
 AC_SUBST(LIBOSSAUDIO)
+
+dnl MacOSX has a similar library called liboss
+AC_MSG_CHECKING([for liboss])
+
+LIBOSS_CFLAGS=
+LIBOSS_LIBADD=
+LIBOSS_LDFLAGS=
+
+if test "x$PKG_CONFIG" != "xno"; then
+  LIBOSS_CFLAGS=""
+  LIBOSS_LIBADD="`$PKG_CONFIG --libs-only-l liboss`"
+  LIBOSS_LDFLAGS="`$PKG_CONFIG --libs-only-L liboss`"
+  AC_DEFINE(HAVE_LIBOSS,1,[Define if you have the liboss compatibility library])
+  AC_MSG_RESULT(yes)
+else
+  AC_MSG_RESULT(not installed)
+fi
+
+AC_SUBST(LIBOSS_CFLAGS)
+AC_SUBST(LIBOSS_LIBADD)
+AC_SUBST(LIBOSS_LDFLAGS)
 
 dnl Don't remove !
 dnl AC_OUTPUT(artsc/artsc-config)
Index: artsc/Makefile.am
===================================================================
RCS file: /Volumes/src/cvs/od/proj/KDE-Darwin/arts/artsc/Makefile.am,v
retrieving revision 1.1.1.3
retrieving revision 1.6
diff -u -u -r1.1.1.3 -r1.6
--- artsc/Makefile.am	14 Aug 2002 13:21:23 -0000	1.1.1.3
+++ artsc/Makefile.am	14 Aug 2002 14:05:27 -0000	1.6
@@ -9,21 +9,26 @@
 bin_SCRIPTS = artsc-config artsdsp
 
 libartsdsp_la_SOURCES = artsdsp.c
-libartsdsp_la_LDFLAGS = -no-undefined -module
+libartsdsp_la_LDFLAGS = -no-undefined -version-info 1:7 -module
 libartsdsp_la_LIBADD = libartsc.la
 
 libartsdsp_st_la_SOURCES = artsc.c artsdsp.c
-libartsdsp_st_la_LDFLAGS = -no-undefined -module
+libartsdsp_st_la_LDFLAGS = -no-undefined -version-info 1:7 -module
 libartsdsp_st_la_LIBADD = $(top_builddir)/libltdl/libltdlc.la
 
 libartsc_la_SOURCES = artsc.c
-libartsc_la_LDFLAGS = -no-undefined
+libartsc_la_LDFLAGS = -no-undefined -version-info 1:7
 libartsc_la_LIBADD = $(top_builddir)/libltdl/libltdlc.la $(LIBPTHREAD) $(USE_THREADS)
 
 libartscbackend_la_SOURCES = artscbackend.cc
-libartscbackend_la_LDFLAGS = -no-undefined -module $(KDE_RPATH)
-libartscbackend_la_LIBADD = $(FLOWLIBS) \
-    $(top_builddir)/soundserver/libsoundserver_idl.la
+libartscbackend_la_LDFLAGS = -no-undefined -version-info 1:7 -module $(KDE_RPATH)
+libartscbackend_la_LIBADD = \
+    $(top_builddir)/mcop/libmcop.la \
+    $(top_builddir)/flow/libartsflow.la \
+    $(top_builddir)/flow/libartsflow_idl.la \
+    $(top_builddir)/soundserver/libsoundserver_idl.la \
+    $(top_builddir)/soundserver/libkmedia2.la \
+    $(top_builddir)/soundserver/libkmedia2_idl.la
 
 artscincludedir = $(includedir)/artsc
 artscinclude_HEADERS = artsc.h
Index: flow/Makefile.am
===================================================================
RCS file: /Volumes/src/cvs/od/proj/KDE-Darwin/arts/flow/Makefile.am,v
retrieving revision 1.1.1.3
retrieving revision 1.12
diff -u -u -r1.1.1.3 -r1.12
--- flow/Makefile.am	1 Oct 2002 01:48:02 -0000	1.1.1.3
+++ flow/Makefile.am	23 Oct 2002 02:45:02 -0000	1.12
@@ -1,18 +1,20 @@
+AM_CFLAGS = -DGSL_WANT_GLIB_WRAPPER -DGSL_WANT_ARTS_THREADS
+AM_CXXFLAGS = -DGSL_WANT_GLIB_WRAPPER -DGSL_WANT_ARTS_THREADS
 
 SUBDIRS = mcopclass gsl gslpp
-INCLUDES = -I$(top_srcdir)/mcop $(all_includes)
+INCLUDES = -I$(top_srcdir)/mcop $(all_includes) $(LIBOSS_CFLAGS)
 
 ####### Files
 
 lib_LTLIBRARIES = libartsflow_idl.la libartsflow.la
 
 libartsflow_idl_la_SOURCES = artsflow.cc
-libartsflow_idl_la_LDFLAGS = -no-undefined -version-info 1:0
+libartsflow_idl_la_LDFLAGS = -no-undefined -version-info 1:7
 libartsflow_idl_la_LIBADD = $(top_builddir)/mcop/libmcop.la $(LIBPOSIX4)
 
-libartsflow_la_LIBADD = $(top_builddir)/mcop/libmcop.la libartsflow_idl.la $(top_builddir)/flow/gslpp/libgslpp.la $(LIBAUDIOFILE) $(LIBASOUND) $(LIBAUDIOIO) $(LIBOSSAUDIO) $(LIBAUDIONAS) $(LIBCSL) $(SGILIBAUDIO) -lm \
-  $(top_builddir)/flow/gsl/libgsl.la
-libartsflow_la_LDFLAGS = -no-undefined -version-info 1:0
+libartsflow_la_LIBADD = $(top_builddir)/mcop/libmcop.la libartsflow_idl.la $(top_builddir)/flow/gslpp/libgslpp.la $(LIBAUDIOFILE) $(LIBASOUND) $(LIBAUDIOIO) $(LIBOSSAUDIO) $(LIBOSS_LIBADD) $(LIBAUDIONAS) $(LIBCSL) $(SGILIBAUDIO) -lm \
+ $(top_builddir)/flow/gsl/libgsl.la $(LIB_POLL)
+libartsflow_la_LDFLAGS = -no-undefined -version-info 1:7 $(LIBOSS_LDFLAGS)
 libartsflow_la_COMPILE_FIRST = artsflow.h
 libartsflow_la_SOURCES =  synth_play_impl.cc \
   synthschedule.cc gslschedule.cc audiosubsys.cc \
Index: flow/audioiooss.cc
===================================================================
RCS file: /Volumes/src/cvs/od/proj/KDE-Darwin/arts/flow/audioiooss.cc,v
retrieving revision 1.1.1.3
diff -u -u -r1.1.1.3 audioiooss.cc
--- flow/audioiooss.cc	29 Oct 2002 02:31:12 -0000	1.1.1.3
+++ flow/audioiooss.cc	31 Oct 2002 18:50:02 -0000
@@ -28,10 +28,43 @@
 
 #if defined(HAVE_SYS_SOUNDCARD_H)
 	#include <sys/soundcard.h>
+	#include <sys/ioctl.h>
 	#define COMPILE_AUDIOIO_OSS 1
+	#define OSS_OPEN open
+	#define OSS_CLOSE close
+	#define OSS_IOCTL ioctl
 #elif defined(HAVE_SOUNDCARD_H)
 	#include <soundcard.h>
+	#include <sys/ioctl.h>
 	#define COMPILE_AUDIOIO_OSS 1
+	#define OSS_OPEN open
+	#define OSS_CLOSE close
+	#define OSS_IOCTL ioctl
+#endif
+
+#ifdef HAVE_LIBOSS
+	extern "C" {
+		extern int oss_ioctl (int,unsigned long,va_list);
+		extern int oss_open (const char *,int,va_list);
+		extern void oss_close (int);
+		static inline int OSS_IOCTL(int x, unsigned long y,...)
+		{
+			va_list l;
+			va_start(l,y);
+			return oss_ioctl(x,y,l);
+		}
+		static inline int OSS_OPEN(const char* x, int y,...)
+		{
+			va_list l;
+			va_start(l,y);
+			return oss_open(x,y,l);
+		}
+		static inline void OSS_CLOSE(int x) {oss_close(x);}
+	}
+	#define LIBOSS_INTERNAL 1
+	#define COMPILE_AUDIOIO_OSS 1
+	#include <liboss/soundcard.h>
+	#include <sys/ioctl.h>
 #endif
 
 /**
@@ -40,7 +73,6 @@
 #ifdef COMPILE_AUDIOIO_OSS
 
 #include <sys/types.h>
-#include <sys/ioctl.h>
 #include <sys/time.h>
 #include <sys/stat.h>
 
@@ -159,7 +191,7 @@
 		return false;
 	}
 
-	audio_fd = ::open(_deviceName.c_str(), mode, 0);
+	audio_fd = ::OSS_OPEN(_deviceName.c_str(), mode, 0);
 
 	if(audio_fd == -1)
 	{
@@ -174,7 +206,7 @@
 	 * check device capabilities
 	 */
 	int device_caps;
-	if(ioctl(audio_fd,SNDCTL_DSP_GETCAPS,&device_caps) == -1)
+	if(OSS_IOCTL(audio_fd,SNDCTL_DSP_GETCAPS,&device_caps) == -1)
             device_caps=0;
 
 	string caps = "";
@@ -189,7 +221,7 @@
 
 	int requestedFormat = (_format == 8)?AFMT_U8:AFMT_S16_LE;
 	int gotFormat = requestedFormat;
-	if (ioctl(audio_fd, SNDCTL_DSP_SETFMT, &gotFormat)==-1)  
+	if (OSS_IOCTL(audio_fd, SNDCTL_DSP_SETFMT, &gotFormat)==-1)  
 	{
 		_error = "SNDCTL_DSP_SETFMT failed - ";
 		_error += strerror(errno);
@@ -252,7 +284,7 @@
 
 	int requeststereo = stereo;
 
-	if (ioctl(audio_fd, SNDCTL_DSP_STEREO, &stereo)==-1)
+	if (OSS_IOCTL(audio_fd, SNDCTL_DSP_STEREO, &stereo)==-1)
 	{
 		_error = "SNDCTL_DSP_STEREO failed - ";
 		_error += strerror(errno);
@@ -271,7 +303,7 @@
 
 	int speed = _samplingRate;
 
-	if (ioctl(audio_fd, SNDCTL_DSP_SPEED, &speed)==-1)  
+	if (OSS_IOCTL(audio_fd, SNDCTL_DSP_SPEED, &speed)==-1)  
 	{
 		_error = "SNDCTL_DSP_SPEED failed - ";
 		_error += strerror(errno);
@@ -322,7 +354,7 @@
 	int size = _fragmentSize;
 	while(size > 1) { size /= 2; frag_arg++; }
 	frag_arg += (_fragmentCount << 16);
-	if(ioctl(audio_fd, SNDCTL_DSP_SETFRAGMENT, &frag_arg) == -1)
+	if(OSS_IOCTL(audio_fd, SNDCTL_DSP_SETFRAGMENT, &frag_arg) == -1)
 	{
 		char buffer[1024];
 		_error = "can't set requested fragments settings";
@@ -336,7 +368,7 @@
 	 * we asked for
 	 */
 	audio_buf_info info;
-	if(ioctl(audio_fd,SNDCTL_DSP_GETOSPACE, &info) == -1)
+	if(OSS_IOCTL(audio_fd,SNDCTL_DSP_GETOSPACE, &info) == -1)
 	{
 		_error = "can't retrieve fragment settings";
 		close();
@@ -393,7 +425,7 @@
 		if(param(direction) & 1) enable_bits |= PCM_ENABLE_INPUT;
 		if(param(direction) & 2) enable_bits |= PCM_ENABLE_OUTPUT;
 
-		if(ioctl(audio_fd,SNDCTL_DSP_SETTRIGGER, &enable_bits) == -1)
+		if(OSS_IOCTL(audio_fd,SNDCTL_DSP_SETTRIGGER, &enable_bits) == -1)
 		{
 			_error = "can't start sound i/o";
 
@@ -406,7 +438,7 @@
 
 void AudioIOOSS::close()
 {
-	::close(audio_fd);
+	::OSS_CLOSE(audio_fd);
 }
 
 void AudioIOOSS::setParam(AudioParam p, int& value)
@@ -431,12 +463,12 @@
 	switch(p)
 	{
 		case canRead:
-				ioctl(audio_fd, SNDCTL_DSP_GETISPACE, &info);
+				OSS_IOCTL(audio_fd, SNDCTL_DSP_GETISPACE, &info);
 				return info.bytes;
 			break;
 
 		case canWrite:
-				ioctl(audio_fd, SNDCTL_DSP_GETOSPACE, &info);
+				OSS_IOCTL(audio_fd, SNDCTL_DSP_GETOSPACE, &info);
 				return info.bytes;
 			break;
 
Index: flow/gsl/Makefile.am
===================================================================
RCS file: /Volumes/src/cvs/od/proj/KDE-Darwin/arts/flow/gsl/Makefile.am,v
retrieving revision 1.1.1.5
retrieving revision 1.9
diff -u -u -r1.1.1.5 -r1.9
--- flow/gsl/Makefile.am	29 Oct 2002 02:31:19 -0000	1.1.1.5
+++ flow/gsl/Makefile.am	29 Oct 2002 02:33:27 -0000	1.9
@@ -1,4 +1,3 @@
-
 EXTRA_DIST =
 CLEANFILES =
 MAINTAINERCLEANFILES =
@@ -15,11 +14,11 @@
 
 libgsl_la_SOURCES = $(GSL_C_SRC) gslglib.c gslglibhash.cc gslartsthreads.cc gslfilehash.c
 libgsl_la_LIBADD = $(top_builddir)/mcop/libmcop.la -lm $(LIBPOSIX4) $(GSL_LIBS) $(LIB_POLL)
-libgsl_la_LDFLAGS = -no-undefined
+libgsl_la_LDFLAGS = -no-undefined -version-info 1:7
 
 noinst_PROGRAMS = $(GSL_NOINST_PROGS)
 
-GSL_progs_ldadd = libgsl.la
+GSL_progs_ldadd = libgsl.la $(LIB_POLL)
 GSL_cc_dummy = dummy.cc
 $(srcdir)/dummy.cc: gslconfig.h
 
Index: flow/gsl/configure.in.in
===================================================================
RCS file: /Volumes/src/cvs/od/proj/KDE-Darwin/arts/flow/gsl/configure.in.in,v
retrieving revision 1.1.1.3
retrieving revision 1.3
diff -u -u -r1.1.1.3 -r1.3
Index: flow/gsl/gslconfig.h.in
===================================================================
RCS file: /Volumes/src/cvs/od/proj/KDE-Darwin/arts/flow/gsl/gslconfig.h.in,v
retrieving revision 1.1.1.3
retrieving revision 1.3
diff -u -u -r1.1.1.3 -r1.3
Index: flow/gsl/gsldatahandle-mad.c
===================================================================
RCS file: /Volumes/src/cvs/od/proj/KDE-Darwin/arts/flow/gsl/gsldatahandle-mad.c,v
retrieving revision 1.1.1.2
retrieving revision 1.3
diff -u -u -r1.1.1.2 -r1.3
--- flow/gsl/gsldatahandle-mad.c	14 Aug 2002 13:21:15 -0000	1.1.1.2
+++ flow/gsl/gsldatahandle-mad.c	14 Aug 2002 13:24:52 -0000	1.3
@@ -31,6 +31,10 @@
 #include <string.h>
 #include <errno.h>
 
+#ifndef EBADFD
+#define EBADFD 77
+#endif
+
 #if	GSL_HAVE_LIBMAD
 #include <mad.h>
 
Index: flow/gsl/gsldatahandle-vorbis.c
===================================================================
RCS file: /Volumes/src/cvs/od/proj/KDE-Darwin/arts/flow/gsl/gsldatahandle-vorbis.c,v
retrieving revision 1.1.1.3
retrieving revision 1.3
diff -u -u -r1.1.1.3 -r1.3
Index: gmcop/Makefile.am
===================================================================
RCS file: /Volumes/src/cvs/od/proj/KDE-Darwin/arts/gmcop/Makefile.am,v
retrieving revision 1.1.1.3
retrieving revision 1.4
diff -u -u -r1.1.1.3 -r1.4
Index: libltdl/ltdl.c
===================================================================
RCS file: /Volumes/src/cvs/od/proj/KDE-Darwin/arts/libltdl/ltdl.c,v
retrieving revision 1.1.1.2
retrieving revision 1.2
diff -u -u -r1.1.1.2 -r1.2
--- libltdl/ltdl.c	10 Jul 2002 02:52:29 -0000	1.1.1.2
+++ libltdl/ltdl.c	14 Sep 2002 16:28:08 -0000	1.2
@@ -1544,11 +1544,14 @@
   /* try to open the old library first; if it was dlpreopened,
      we want the preopened version of it, even if a dlopenable
      module is available */
+/*	HACKALERT: Why? Mac OS/Darwin can't open static libs, it is
+	a waste to even try. Disabled for kde.	*/
+#if 0
   if (old_name && tryall_dlopen(handle, old_name) == 0)
     {
       return 0;
     }
-
+#endif
   /* try to open the dynamic library */
   if (dlname)
     {
@@ -1567,6 +1570,28 @@
 	    }
 
 	  sprintf (filename, "%s/%s", libdir, dlname);
+	  error = (tryall_dlopen (handle, filename) != 0);
+	  LT_DLFREE (filename);
+
+	  if (!error)
+	    {
+	      return 0;
+	    }
+	}
+
+      /* then look for the same module, but with "lib" prepended */
+      if (installed && libdir)
+	{
+	  len	    = strlen (libdir) + 4 + strlen (dlname);
+	  filename  = LT_DLMALLOC (char, 1+ len);
+
+	  if (!filename)
+	    {
+	      MUTEX_SETERROR (LT_DLSTRERROR (NO_MEMORY));
+	      return 1;
+	    }
+
+	  sprintf (filename, "%s/%s%s", libdir, "lib", dlname);
 	  error = (tryall_dlopen (handle, filename) != 0);
 	  LT_DLFREE (filename);
 
Index: libltdl/ltdl.m4
===================================================================
RCS file: /Volumes/src/cvs/od/proj/KDE-Darwin/arts/libltdl/ltdl.m4,v
retrieving revision 1.1.1.2
retrieving revision 1.3
diff -u -u -r1.1.1.2 -r1.3
--- libltdl/ltdl.m4	10 Jul 2002 02:52:29 -0000	1.1.1.2
+++ libltdl/ltdl.m4	10 Jul 2002 03:01:23 -0000	1.3
@@ -103,6 +103,9 @@
 	osf*)
 	  libltdl_cv_sys_dlopen_deplibs=yes
 	  ;;
+	darwin*)
+	  libltdl_cv_sys_dlopen_deplibs=yes
+	  ;;
 	esac
 ])
 if test "$libltdl_cv_sys_dlopen_deplibs" != yes; then
Index: mcop/Makefile.am
===================================================================
RCS file: /Volumes/src/cvs/od/proj/KDE-Darwin/arts/mcop/Makefile.am,v
retrieving revision 1.1.1.2
retrieving revision 1.5
diff -u -u -r1.1.1.2 -r1.5
--- mcop/Makefile.am	10 Jul 2002 02:52:30 -0000	1.1.1.2
+++ mcop/Makefile.am	14 Aug 2002 14:05:28 -0000	1.5
@@ -17,7 +17,7 @@
 	 delayedreturn.cc thread.cc dynamicskeleton.cc
 
 libmcop_la_LIBADD = $(LIBSOCKET) $(top_builddir)/libltdl/libltdlc.la
-libmcop_la_LDFLAGS = -no-undefined -version-info 1:0
+libmcop_la_LDFLAGS = -no-undefined -version-info 1:7
 
 artsincludedir = $(includedir)/arts
 artsinclude_HEADERS = buffer.h common.h connection.h core.h dispatcher.h \
Index: mcop_mt/Makefile.am
===================================================================
RCS file: /Volumes/src/cvs/od/proj/KDE-Darwin/arts/mcop_mt/Makefile.am,v
retrieving revision 1.1.1.2
retrieving revision 1.5
diff -u -u -r1.1.1.2 -r1.5
--- mcop_mt/Makefile.am	10 Jul 2002 02:52:31 -0000	1.1.1.2
+++ mcop_mt/Makefile.am	14 Aug 2002 14:05:28 -0000	1.5
@@ -4,4 +4,4 @@
 
 libmcop_mt_la_SOURCES = threads_posix.cc
 libmcop_mt_la_LIBADD = $(top_builddir)/mcop/libmcop.la $(LIBPTHREAD) $(LIBPOSIX4)
-libmcop_mt_la_LDFLAGS = -no-undefined -version-info 1:0 $(USE_THREADS) 
+libmcop_mt_la_LDFLAGS = -no-undefined -version-info 1:7 $(USE_THREADS) 
Index: mcopidl/Makefile.am
===================================================================
RCS file: /Volumes/src/cvs/od/proj/KDE-Darwin/arts/mcopidl/Makefile.am,v
retrieving revision 1.1.1.3
retrieving revision 1.2
diff -u -u -r1.1.1.3 -r1.2
--- mcopidl/Makefile.am	14 Aug 2002 13:21:26 -0000	1.1.1.3
+++ mcopidl/Makefile.am	16 Sep 2002 14:31:34 -0000	1.2
@@ -1,4 +1,3 @@
-KDE_CXXFLAGS = $(NOOPT_CXXFLAGS) 
 INCLUDES = -I$(top_srcdir)/mcop $(all_includes)
 ####### Files
 
Index: qtmcop/Makefile.am
===================================================================
RCS file: /Volumes/src/cvs/od/proj/KDE-Darwin/arts/qtmcop/Makefile.am,v
retrieving revision 1.1.1.1
retrieving revision 1.5
diff -u -u -r1.1.1.1 -r1.5
--- qtmcop/Makefile.am	25 Jun 2002 01:45:59 -0000	1.1.1.1
+++ qtmcop/Makefile.am	14 Aug 2002 14:05:28 -0000	1.5
@@ -5,7 +5,7 @@
 
 libqtmcop_la_SOURCES = qiomanager.cc
 libqtmcop_la_LIBADD = $(top_builddir)/mcop/libmcop.la $(LIB_QT)
-libqtmcop_la_LDFLAGS = -no-undefined -version-info 1:0 $(KDE_RPATH) $(KDE_MT_LDFLAGS) $(QT_LDFLAGS) \
+libqtmcop_la_LDFLAGS = -no-undefined -version-info 1:7 $(KDE_RPATH) $(KDE_MT_LDFLAGS) $(QT_LDFLAGS) \
   $(X_LDFLAGS) $(USER_LDFLAGS)
 
 METASOURCES = qiomanager_p.moc
Index: soundserver/Makefile.am
===================================================================
RCS file: /Volumes/src/cvs/od/proj/KDE-Darwin/arts/soundserver/Makefile.am,v
retrieving revision 1.1.1.4
retrieving revision 1.7
diff -u -u -r1.1.1.4 -r1.7
--- soundserver/Makefile.am	14 Aug 2002 13:21:26 -0000	1.1.1.4
+++ soundserver/Makefile.am	14 Aug 2002 14:05:29 -0000	1.7
@@ -19,26 +19,29 @@
 lib_LTLIBRARIES = libkmedia2_idl.la libsoundserver_idl.la \
 	libartsgslplayobject.la libartswavplayobject.la libkmedia2.la
 
-libsoundserver_idl_la_LIBADD = libkmedia2_idl.la \
-                          $(top_builddir)/flow/libartsflow_idl.la
-libsoundserver_idl_la_LDFLAGS = -no-undefined -version-info 1:0
+libsoundserver_idl_la_LIBADD = libkmedia2_idl.la $(top_builddir)/x11/libx11globalcomm.la \
+                          $(top_builddir)/flow/libartsflow_idl.la $(top_builddir)/mcop/libmcop.la
+libsoundserver_idl_la_LDFLAGS = -no-undefined -version-info 1:7
 libsoundserver_idl_la_SOURCES = soundserver.cc
 libsoundserver_idl_la_COMPILE_FIRST = soundserver.h
 
 libkmedia2_idl_la_SOURCES = kmedia2.cc
 libkmedia2_idl_la_COMPILE_FIRST = kmedia2.h
-libkmedia2_idl_la_LIBADD  = $(top_builddir)/flow/libartsflow.la
-libkmedia2_idl_la_LDFLAGS = -no-undefined -version-info 1:0
+libkmedia2_idl_la_LIBADD  = $(top_builddir)/flow/libartsflow.la $(top_builddir)/mcop/libmcop.la \
+	$(top_builddir)/x11/libx11globalcomm.la $(top_builddir)/flow/libartsflow_idl.la
+libkmedia2_idl_la_LDFLAGS = -no-undefined -version-info 1:7
 
 libkmedia2_la_SOURCES = fileinputstream_impl.cc stdoutwriter_impl.cc
-libkmedia2_la_LIBADD  = libkmedia2_idl.la $(FLOWLIBS)
-libkmedia2_la_LDFLAGS = -no-undefined -version-info 1:0
+libkmedia2_la_LIBADD  = libkmedia2_idl.la $(FLOWLIBS) $(top_builddir)/flow/libartsflow_idl.la \
+	$(top_builddir)/mcop/libmcop.la $(top_builddir)/x11/libx11globalcomm.la
+libkmedia2_la_LDFLAGS = -no-undefined -version-info 1:7
 libkmedia2_la_COMPILE_FIRST = kmedia2.h ../flow/artsflow.h
 
 libartswavplayobject_la_SOURCES = wavplayobject_impl.cc
-libartswavplayobject_la_LIBADD  = $(top_builddir)/mcop/libmcop.la \
-                                  libsoundserver_idl.la $(FLOWLIBS)
-libartswavplayobject_la_LDFLAGS = -no-undefined -module
+libartswavplayobject_la_LIBADD  = $(top_builddir)/mcop/libmcop.la $(top_builddir)/x11/libx11globalcomm.la \
+                                  libsoundserver_idl.la $(FLOWLIBS) $(top_builddir)/flow/libartsflow_idl.la \
+				  libkmedia2_idl.la
+libartswavplayobject_la_LDFLAGS = -no-undefined -version-info 1:7 -module
 libartswavplayobject_la_COMPILE_FIRST = soundserver.h ../flow/artsflow.h
 
 libartsgslplayobject_la_SOURCES = gslplayobject_impl.cc
Index: soundserver/artsd.cc
===================================================================
RCS file: /Volumes/src/cvs/od/proj/KDE-Darwin/arts/soundserver/artsd.cc,v
retrieving revision 1.1.1.2
retrieving revision 1.4
diff -u -u -r1.1.1.2 -r1.4
--- soundserver/artsd.cc	10 Jul 2002 02:52:33 -0000	1.1.1.2
+++ soundserver/artsd.cc	20 Oct 2002 19:11:01 -0000	1.4
@@ -35,6 +35,13 @@
 #include "debug.h"
 #include "artsversion.h"
 
+#ifdef HAVE_MACH_REALTIME_SCHED
+  #include <mach/mach.h>
+  #include <mach/thread_policy.h>
+  #include <sys/param.h>
+  #include <sys/sysctl.h>
+#endif
+
 using namespace std;
 using namespace Arts;
 
@@ -232,6 +239,29 @@
 	cerr << endl;
 }
 
+#ifdef HAVE_MACH_REALTIME_SCHED
+static void mach_get_realtime_priority()
+{
+	struct thread_time_constraint_policy ttcpolicy;
+	int bus_speed, mib [2] = { CTL_HW, HW_BUS_FREQ };
+	size_t len;   
+
+	len = sizeof (bus_speed);
+	sysctl (mib, 2, &bus_speed, &len, NULL, 0);
+
+	/* Is it enough? */
+	ttcpolicy.period = bus_speed / 120;
+	ttcpolicy.computation = bus_speed / 2400;
+	ttcpolicy.constraint = bus_speed / 1200;
+	ttcpolicy.preemptible = 1;
+
+	thread_policy_set (mach_thread_self (),
+			   THREAD_TIME_CONSTRAINT_POLICY,
+			   (int*)&ttcpolicy,
+			   THREAD_TIME_CONSTRAINT_POLICY_COUNT);
+}
+#endif
+
 int main(int argc, char **argv)
 {
 	handleArgs(argc, argv);
@@ -244,7 +274,9 @@
 
 	if(cfgPort)			 TCPServer::setPort(cfgPort);
 
+#ifndef ARTS_NO_ALARM
 	CPUUsage	cpuUsage;
+#endif
 	Dispatcher	dispatcher(0,cfgServers);
 
 	initSignals();
@@ -298,6 +330,10 @@
 		cleanUnusedReferences();
 		if(!publishReferences(server,audioManager,false)) return 1;
 	}
+
+#ifdef HAVE_MACH_REALTIME_SCHED
+	mach_get_realtime_priority();
+#endif
 
 	/* warn if there was a problem with artswrapper */
 	char *wrapper = getenv("STARTED_THROUGH_ARTSWRAPPER");
Index: soundserver/soundserverv2_impl.cc
===================================================================
RCS file: /Volumes/src/cvs/od/proj/KDE-Darwin/arts/soundserver/soundserverv2_impl.cc,v
retrieving revision 1.1.1.1
retrieving revision 1.2
diff -u -u -r1.1.1.1 -r1.2
--- soundserver/soundserverv2_impl.cc	25 Jun 2002 01:45:59 -0000	1.1.1.1
+++ soundserver/soundserverv2_impl.cc	25 Jun 2002 01:46:40 -0000	1.2
@@ -374,9 +374,12 @@
 
 float SoundServerV2_impl::cpuUsage()
 {
+#ifndef ARTS_NO_ALARM
 	return CPUUsage::the()->usage() * 100.0;
+#else
+	return 5.0;
+#endif
 }
-
 
 #ifndef __SUNPRO_CC
 /* See bottom of simplesoundserver_impl.cc for the reason this is here.  */
Index: x11/Makefile.am
===================================================================
RCS file: /Volumes/src/cvs/od/proj/KDE-Darwin/arts/x11/Makefile.am,v
retrieving revision 1.1.1.2
retrieving revision 1.5
diff -u -u -r1.1.1.2 -r1.5
--- x11/Makefile.am	10 Jul 2002 02:52:34 -0000	1.1.1.2
+++ x11/Makefile.am	14 Aug 2002 14:05:29 -0000	1.5
@@ -4,7 +4,7 @@
 
 libx11globalcomm_la_SOURCES = x11globalcomm.cc x11globalcomm_impl.cc
 libx11globalcomm_la_LIBADD = $(top_builddir)/mcop/libmcop.la $(LIB_X11)
-libx11globalcomm_la_LDFLAGS = -no-undefined -module -version-info 1:0 $(X_LDFLAGS)
+libx11globalcomm_la_LDFLAGS = -no-undefined -module -version-info 1:7 $(X_LDFLAGS)
 
 DISTCLEANFILES = x11globalcomm.cc x11globalcomm.h \
                  x11globalcomm.mcoptype x11globalcomm.mcopclass
