diff -uNbr CardDesk/carddesk/applets/testapplet/Makefile.am CardDesk-new/carddesk/applets/testapplet/Makefile.am
--- CardDesk/carddesk/applets/testapplet/Makefile.am	Sat Jan 11 13:41:25 2003
+++ CardDesk-new/carddesk/applets/testapplet/Makefile.am	Sat Jan 11 16:55:11 2003
@@ -11,7 +11,7 @@
 libtestapplet_la_LDFLAGS = -version-info 0:0:0
 
 libtestapplet_la_METASOURCES = AUTO
-
+libtestapplet_la_LIBADD = $(LIB_QT) $(LIB_KDECORE) $(LIB_KDEUI) -lkdefx $(top_builddir)/carddesk/carddeskcore/libcarddeskcore.la
 libtestapplet_la_SOURCES = TestCardApplet.cpp plugin.cpp 
 
 
diff -uNbr CardDesk/carddesk/carddeskcore/Makefile.am CardDesk-new/carddesk/carddeskcore/Makefile.am
--- CardDesk/carddesk/carddeskcore/Makefile.am	Sat Jan 11 13:41:25 2003
+++ CardDesk-new/carddesk/carddeskcore/Makefile.am	Sat Jan 11 16:53:29 2003
@@ -9,7 +9,7 @@
 
 
 libcarddeskcore_la_LDFLAGS = -version-info 0:0:0
-
+libcarddeskcore_la_LIBADD = $(LIB_QT) $(LIB_KDECORE) $(LIB_KDEUI) -lkdefx
 libcarddeskcore_la_METASOURCES = AUTO
 
 libcarddeskcore_la_SOURCES = NewCardDialog.cpp PluginListBox.cpp PluginRoster.cpp LayoutSwitcher.cpp DragLogic.cpp Deck.cpp Const.cpp CardManager.cpp CardElements.cpp CardApplet.cpp Card.cpp 
