diff -uNbr kde-i18n-pt/messages/kdegraphics/Makefile.in kde-i18n-pt-new/messages/kdegraphics/Makefile.in
--- kde-i18n-pt/messages/kdegraphics/Makefile.in	Tue Sep 17 06:00:51 2002
+++ kde-i18n-pt-new/messages/kdegraphics/Makefile.in	Fri Oct  4 17:10:56 2002
@@ -242,8 +242,8 @@
 
 #>- POFILES = AUTO
 #>+ 2
-POFILES = kfile_ico.po kviewshell.po kghostview.po kviewtemplateplugin.po ksnapshot.po kdvi.po kfax.po kfile_pdf.po kfile_png.po kfile_tga.po kview.po kfile_xbm.po kviewscannerplugin.po kooka.po kuickshow.po desktop_kdegraphics.po libkscan.po kfile_ps.po kpovmodeler.po kfile_jpeg.po kpaint.po kviewpresenterplugin.po kiconedit.po kviewbrowserplugin.po kcmkmrml.po kruler.po kfile_tiff.po kcmkamera.po kview_scale.po kcoloredit.po kfile_bmp.po kfile_gif.po
-GMOFILES = kfile_ico.gmo kviewshell.gmo kghostview.gmo kviewtemplateplugin.gmo ksnapshot.gmo kdvi.gmo kfax.gmo kfile_pdf.gmo kfile_png.gmo kfile_tga.gmo kview.gmo kfile_xbm.gmo kviewscannerplugin.gmo kooka.gmo kuickshow.gmo desktop_kdegraphics.gmo libkscan.gmo kfile_ps.gmo kpovmodeler.gmo kfile_jpeg.gmo kpaint.gmo kviewpresenterplugin.gmo kiconedit.gmo kviewbrowserplugin.gmo kcmkmrml.gmo kruler.gmo kfile_tiff.gmo kcmkamera.gmo kview_scale.gmo kcoloredit.gmo kfile_bmp.gmo kfile_gif.gmo
+POFILES = kfile_ico.po kviewshell.po kghostview.po kviewtemplateplugin.po kdvi.po kfax.po kfile_pdf.po kfile_png.po kfile_tga.po kview.po kfile_xbm.po kviewscannerplugin.po kooka.po kuickshow.po desktop_kdegraphics.po libkscan.po kfile_ps.po kpovmodeler.po kfile_jpeg.po kpaint.po kviewpresenterplugin.po kiconedit.po kviewbrowserplugin.po kcmkmrml.po kruler.po kfile_tiff.po kcmkamera.po kview_scale.po kcoloredit.po kfile_bmp.po kfile_gif.po
+GMOFILES = kfile_ico.gmo kviewshell.gmo kghostview.gmo kviewtemplateplugin.gmo kdvi.gmo kfax.gmo kfile_pdf.gmo kfile_png.gmo kfile_tga.gmo kview.gmo kfile_xbm.gmo kviewscannerplugin.gmo kooka.gmo kuickshow.gmo desktop_kdegraphics.gmo libkscan.gmo kfile_ps.gmo kpovmodeler.gmo kfile_jpeg.gmo kpaint.gmo kviewpresenterplugin.gmo kiconedit.gmo kviewbrowserplugin.gmo kcmkmrml.gmo kruler.gmo kfile_tiff.gmo kcmkamera.gmo kview_scale.gmo kcoloredit.gmo kfile_bmp.gmo kfile_gif.gmo
 subdir = messages/kdegraphics
 mkinstalldirs = $(SHELL) $(top_srcdir)/mkinstalldirs
 CONFIG_HEADER = $(top_builddir)/config.h
@@ -377,7 +377,7 @@
 
 #>- DISTFILES = $(DIST_COMMON) $(DIST_SOURCES) $(TEXINFOS) $(EXTRA_DIST)
 #>+ 4
-KDE_DIST=kdvi.po kfile_tga.po kpaint.po kviewpresenterplugin.po kfile_ps.po ksnapshot.po kooka.po kviewscannerplugin.po desktop_kdegraphics.po kfile_tiff.po kuickshow.po kfile_pdf.po kfile_xbm.po kghostview.po kpovmodeler.po kviewshell.po kviewtemplateplugin.po kfax.po kview.po kfile_jpeg.po kviewbrowserplugin.po libkscan.po kfile_gif.po kview_scale.po kcoloredit.po kfile_ico.po kiconedit.po kruler.po kcmkamera.po kfile_png.po kfile_bmp.po kcmkmrml.po 
+KDE_DIST=kdvi.po kfile_tga.po kpaint.po kviewpresenterplugin.po kfile_ps.po kooka.po kviewscannerplugin.po desktop_kdegraphics.po kfile_tiff.po kuickshow.po kfile_pdf.po kfile_xbm.po kghostview.po kpovmodeler.po kviewshell.po kviewtemplateplugin.po kfax.po kview.po kfile_jpeg.po kviewbrowserplugin.po libkscan.po kfile_gif.po kview_scale.po kcoloredit.po kfile_ico.po kiconedit.po kruler.po kcmkamera.po kfile_png.po kfile_bmp.po kcmkmrml.po 
 
 #>- DISTFILES= $(DIST_COMMON) $(DIST_SOURCES) $(TEXINFOS) $(EXTRA_DIST) $(KDE_DIST)
 #>+ 1
@@ -526,9 +526,6 @@
 kviewtemplateplugin.gmo: kviewtemplateplugin.po
 	rm -f kviewtemplateplugin.gmo; $(GMSGFMT) -o kviewtemplateplugin.gmo $(srcdir)/kviewtemplateplugin.po
 	test ! -f kviewtemplateplugin.gmo || touch kviewtemplateplugin.gmo
-ksnapshot.gmo: ksnapshot.po
-	rm -f ksnapshot.gmo; $(GMSGFMT) -o ksnapshot.gmo $(srcdir)/ksnapshot.po
-	test ! -f ksnapshot.gmo || touch ksnapshot.gmo
 kdvi.gmo: kdvi.po
 	rm -f kdvi.gmo; $(GMSGFMT) -o kdvi.gmo $(srcdir)/kdvi.po
 	test ! -f kdvi.gmo || touch kdvi.gmo
@@ -613,12 +610,12 @@
 
 #>+ 3
 clean-nls:
-	-rm -f  kfile_ico.gmo kviewshell.gmo kghostview.gmo kviewtemplateplugin.gmo ksnapshot.gmo kdvi.gmo kfax.gmo kfile_pdf.gmo kfile_png.gmo kfile_tga.gmo kview.gmo kfile_xbm.gmo kviewscannerplugin.gmo kooka.gmo kuickshow.gmo desktop_kdegraphics.gmo libkscan.gmo kfile_ps.gmo kpovmodeler.gmo kfile_jpeg.gmo kpaint.gmo kviewpresenterplugin.gmo kiconedit.gmo kviewbrowserplugin.gmo kcmkmrml.gmo kruler.gmo kfile_tiff.gmo kcmkamera.gmo kview_scale.gmo kcoloredit.gmo kfile_bmp.gmo kfile_gif.gmo
+	-rm -f  kfile_ico.gmo kviewshell.gmo kghostview.gmo kviewtemplateplugin.gmo kdvi.gmo kfax.gmo kfile_pdf.gmo kfile_png.gmo kfile_tga.gmo kview.gmo kfile_xbm.gmo kviewscannerplugin.gmo kooka.gmo kuickshow.gmo desktop_kdegraphics.gmo libkscan.gmo kfile_ps.gmo kpovmodeler.gmo kfile_jpeg.gmo kpaint.gmo kviewpresenterplugin.gmo kiconedit.gmo kviewbrowserplugin.gmo kcmkmrml.gmo kruler.gmo kfile_tiff.gmo kcmkamera.gmo kview_scale.gmo kcoloredit.gmo kfile_bmp.gmo kfile_gif.gmo
 
 #>+ 8
 install-nls:
 	$(mkinstalldirs) $(DESTDIR)$(kde_locale)/$(KDE_LANG)/LC_MESSAGES
-	@for base in kfile_ico kviewshell kghostview kviewtemplateplugin ksnapshot kdvi kfax kfile_pdf kfile_png kfile_tga kview kfile_xbm kviewscannerplugin kooka kuickshow desktop_kdegraphics libkscan kfile_ps kpovmodeler kfile_jpeg kpaint kviewpresenterplugin kiconedit kviewbrowserplugin kcmkmrml kruler kfile_tiff kcmkamera kview_scale kcoloredit kfile_bmp kfile_gif ; do \
+	@for base in kfile_ico kviewshell kghostview kviewtemplateplugin kdvi kfax kfile_pdf kfile_png kfile_tga kview kfile_xbm kviewscannerplugin kooka kuickshow desktop_kdegraphics libkscan kfile_ps kpovmodeler kfile_jpeg kpaint kviewpresenterplugin kiconedit kviewbrowserplugin kcmkmrml kruler kfile_tiff kcmkamera kview_scale kcoloredit kfile_bmp kfile_gif ; do \
 	  echo $(INSTALL_DATA) $$base.gmo $(DESTDIR)$(kde_locale)/$(KDE_LANG)/LC_MESSAGES/$$base.mo ;\
 	  test ! -f $$base.gmo || $(INSTALL_DATA) $$base.gmo $(DESTDIR)$(kde_locale)/$(KDE_LANG)/LC_MESSAGES/$$base.mo ;\
 	done
@@ -630,7 +627,6 @@
 	rm -f $(DESTDIR)$(kde_locale)/$(KDE_LANG)/LC_MESSAGES/kviewshell.mo
 	rm -f $(DESTDIR)$(kde_locale)/$(KDE_LANG)/LC_MESSAGES/kghostview.mo
 	rm -f $(DESTDIR)$(kde_locale)/$(KDE_LANG)/LC_MESSAGES/kviewtemplateplugin.mo
-	rm -f $(DESTDIR)$(kde_locale)/$(KDE_LANG)/LC_MESSAGES/ksnapshot.mo
 	rm -f $(DESTDIR)$(kde_locale)/$(KDE_LANG)/LC_MESSAGES/kdvi.mo
 	rm -f $(DESTDIR)$(kde_locale)/$(KDE_LANG)/LC_MESSAGES/kfax.mo
 	rm -f $(DESTDIR)$(kde_locale)/$(KDE_LANG)/LC_MESSAGES/kfile_pdf.mo
