
===================================================================
diff -u -u -r1.1.1.5 -r1.2
--- ./noatun-plugins/Makefile.am	20 Nov 2002 01:27:49 -0000	1.1.1.5
+++ ./noatun-plugins/Makefile.am	1 Dec 2002 23:10:21 -0000	1.2
@@ -1,5 +1,6 @@
-if include_SDL
-SDL_SUBDIRS = blurscope synaescope tippercanoe tyler
-endif
+#if include_SDL
+#SDL_SUBDIRS = blurscope synaescope tippercanoe tyler
+#endif
+SDL_SUBDIRS =
 
 SUBDIRS = alarm noatunmadness charlatan alsaplayer pitchablespeed luckytag wavecapture lyrics ffrs $(SDL_SUBDIRS)
===================================================================
diff -u -u -r1.1.1.3 -r1.2
--- ./noatun-plugins/pitchablespeed/Makefile.am	14 Aug 2002 13:29:09 -0000	1.1.1.3
+++ ./noatun-plugins/pitchablespeed/Makefile.am	1 Dec 2002 23:10:21 -0000	1.2
@@ -2,7 +2,7 @@
 
 kde_module_LTLIBRARIES = noatunpitchablespeed.la
 noatunpitchablespeed_la_LDFLAGS = $(all_libraries) -module -avoid-version -no-undefined
-noatunpitchablespeed_la_LIBADD = $(LIB_KFILE) -lnoatun -lm -lartsmodules
+noatunpitchablespeed_la_LIBADD = $(LIB_KFILE) -lnoatun -lm -lartsmodules_shared
 
 noatunpitchablespeed_la_SOURCES = pitchablespeed.cpp speedtuner.ui
 noatunpitchablespeed_la_METASOURCES = AUTO
===================================================================
diff -u -u -r1.1.1.2 -r1.2
--- ./noatun-plugins/wavecapture/Makefile.am	14 Aug 2002 13:29:14 -0000	1.1.1.2
+++ ./noatun-plugins/wavecapture/Makefile.am	1 Dec 2002 23:10:21 -0000	1.2
@@ -5,7 +5,7 @@
 
 
 noatunwavecapture_la_LDFLAGS = $(all_libraries) -module -avoid-version -no-undefined
-noatunwavecapture_la_LIBADD = $(LIB_KFILE) -lnoatun -lm -lartsmodules
+noatunwavecapture_la_LIBADD = $(LIB_KFILE) -lnoatun -lm -lartsmodules_shared
 
 noatunwavecapture_la_SOURCES = wavecapture.cpp
 noatunwavecapture_la_METASOURCES = AUTO
