
===================================================================
diff -u -u -r1.1.1.7 acinclude.m4.in
--- admin/acinclude.m4.in	20 Nov 2002 00:45:10 -0000	1.1.1.7
+++ admin/acinclude.m4.in	24 Nov 2002 06:59:54 -0000
@@ -31,6 +31,9 @@
 dnl thanks to Raja R Harinath.
 dnl ------------------------------------------------------------------------
 dnl
+
+USER_LDFLAGS="-multiply_defined suppress $USER_LDFLAGS"
+
 ifdef([_AC_PATH_X_XMKMF],[],
    [AC_DEFUN([_AC_PATH_X_XMKMF],[AC_PATH_X_XMKMF])])
 ifdef([AC_OUTPUT_SUBDIRS],[],
@@ -366,7 +369,7 @@
   fi
 
 AC_MSG_CHECKING([for KDE libraries installed])
-ac_link='$LIBTOOL_SHELL --silent --mode=link ${CXX-g++} -o conftest $CXXFLAGS $all_includes $CPPFLAGS $LDFLAGS $all_libraries conftest.$ac_ext $LIBS -lkdecore $LIBQT $KDE_TEST_RPATH 1>&5'
+ac_link='$LIBTOOL_SHELL --mode=link ${CXX-g++} -o conftest $CXXFLAGS $all_includes $CPPFLAGS $LDFLAGS $all_libraries conftest.$ac_ext $LIBS -lkdecore $LIBQT $KDE_TEST_RPATH 1>&5'
 
 if AC_TRY_EVAL(ac_link) && test -s conftest; then
   AC_MSG_RESULT(yes)
@@ -704,13 +707,10 @@
      AC_CHECK_LIB(ipc, shmat, X_EXTRA_LIBS="$X_EXTRA_LIBS -lipc"))
    
    # darwin needs this to initialize the environment
-   AC_CHECK_HEADERS(crt_externs.h)
+   AC_CHECK_HEADERS([crt_externs.h sys/types.h])
    AC_CHECK_FUNC(_NSGetEnviron, [AC_DEFINE(HAVE_NSGETENVIRON, 1, [Define if your system needs _NSGetEnviron to set up the environment])])
  
-   # more headers that need to be explicitly included on darwin
-   AC_CHECK_HEADERS(sys/types.h stdint.h)
-
-   # darwin requires a poll emulation library
+   # darwin requires a poll.h emulation library
    AC_CHECK_LIB(poll, poll, LIB_POLL="-lpoll")
 
    # CoreAudio framework
@@ -719,11 +719,36 @@
      FRAMEWORK_COREAUDIO="-framework CoreAudio"
    ])
 
+   # CoreServices framework
+   AC_CHECK_HEADER(CoreServices/CoreServices.h, [
+     AC_DEFINE(HAVE_CORESERVICES, 1, [Define if you have the CoreServices API])
+     FRAMEWORK_CORESERVICES="-framework CoreServices"
+   ])
+
+   # Solaris 2.6 and others need -lresolv for res_init
+   AC_CHECK_FUNCS(res_init, , [
+     kde_libs_safe="$LIBS"
+     LIBS="$LIBS $X_EXTRA_LIBS -lresolv"
+     AC_TRY_LINK(
+[
+#include <resolv.h>
+],
+[ 
+res_init(); 
+],
+        LIBRESOLV="-lresolv"
+        X_EXTRA_LIBS="$X_EXTRA_LIBS $LIBRESOLV"
+        AC_DEFINE(HAVE_RES_INIT, 1, [Define if you have the res_init function])
+     )
+     LIBS=$kde_libs_safe
+   ])
+
    AC_CHECK_RES_INIT
-   AC_SUBST(LIB_POLL)
    AC_SUBST(FRAMEWORK_COREAUDIO)
    LIBSOCKET="$X_EXTRA_LIBS"
    AC_SUBST(LIBSOCKET)
+   AC_SUBST(FRAMEWORK_CORESERVICES)
+   AC_SUBST(LIB_POLL)
    AC_SUBST(LIBRESOLV)
    AC_SUBST(X_EXTRA_LIBS)
    AC_CHECK_LIB(ucb, killpg, [LIBUCB="-lucb"]) dnl for Solaris2.4
@@ -932,7 +957,7 @@
 kde_ldflags_safe="$LDFLAGS"
 kde_libs_safe="$LIBS"
 
-LDFLAGS="$LDFLAGS $X_LDFLAGS $USER_LDFLAGS"
+LDFLAGS="$LDFLAGS $USER_LDFLAGS $X_LDFLAGS"
 LIBS="-lXext -lX11 $LIBSOCKET"
 
 AC_TRY_LINK([
@@ -974,9 +999,9 @@
 kde_save_LDFLAGS="$LDFLAGS"
 kde_save_CFLAGS="$CFLAGS"
 kde_save_LIBS="$LIBS"
-LDFLAGS="$LDFLAGS $X_LDFLAGS $USER_LDFLAGS"
+LDFLAGS="$LDFLAGS $USER_LDFLAGS $X_LDFLAGS"
 CFLAGS="$CFLAGS -I$x_includes"
-LIBS="-lXinerama -lXext"
+LIBS="-lXinerama -lXext -lX11"
 
 if test "x$no_xinerama" = "xno"; then
 
@@ -1844,7 +1869,7 @@
  kde_cxxflags_safe="$CXXFLAGS"
 
  LIBS="-lkde-qt-addon $LIBQT $LIBS"
- CXXFLAGS="$CXXFLAGS -I$prefix/include -I$prefix/include/kde $all_includes"
+ CXXFLAGS="-isystem $prefix/include $CXXFLAGS $all_includes"
  LDFLAGS="$LDFLAGS $all_libraries $USER_LDFLAGS"
 
  AC_TRY_LINK([
@@ -1916,7 +1941,7 @@
    AC_SUBST(LIB_KHTML, "-lkhtml")
    AC_SUBST(LIB_KSPELL, "-lkspell")
    AC_SUBST(LIB_KPARTS, "-lkparts")
-   AC_SUBST(LIB_KDEPRINT, "-lkdeprint")
+   AC_SUBST(LIB_KDED, "@PREFIX@/lib/kded.la")
 # these are for backward compatibility
    AC_SUBST(LIB_KSYCOCA, "-lkio")
    AC_SUBST(LIB_KFILE, "-lkio")
@@ -2647,10 +2672,10 @@
       if test $kde_use_debug_code = "full"; then
         CFLAGS="-g3 $CFLAGS"
       else
-        CFLAGS="-g -O2 $CFLAGS"
+        CFLAGS="-g -O $CFLAGS"
       fi
     else
-      CFLAGS="-O2 $CFLAGS"
+      CFLAGS="-Os -frename-registers -mmultiple -mcpu=750 -mtune=750 $CFLAGS"
     fi
   fi
 
@@ -2680,7 +2705,7 @@
         if test "$kde_use_debug_code" = "full"; then
           CXXFLAGS="-g3 $CXXFLAGS"
         else
-          CXXFLAGS="-g -O2 $CXXFLAGS"
+          CXXFLAGS="-g -O $CXXFLAGS"
         fi
       fi
       KDE_CHECK_COMPILER_FLAG(fno-builtin,[CXXFLAGS="-fno-builtin $CXXFLAGS"])
@@ -2692,7 +2717,7 @@
       if test "$CXX" = "KCC"; then
         CXXFLAGS="+K3 $CXXFLAGS"
       else
-        CXXFLAGS="-O2 $CXXFLAGS"
+        CXXFLAGS="-Os -frename-registers -mmultiple -mcpu=750 -mtune=750 $CXXFLAGS"
       fi  
     fi
   fi
@@ -2832,7 +2857,7 @@
 AC_LIBTOOL_CXX
 
 LIBTOOL_SHELL="/bin/sh ./libtool"
-#  LIBTOOL="$LIBTOOL --silent"
+#  LIBTOOL="$LIBTOOL"
 KDE_PLUGIN="-avoid-version -module -no-undefined \$(KDE_RPATH) \$(KDE_MT_LDFLAGS)"
 AC_SUBST(KDE_PLUGIN)
 
@@ -2888,7 +2913,7 @@
 [
 AC_CANONICAL_SYSTEM
 AC_ARG_PROGRAM
-AM_INIT_AUTOMAKE($1, $2)
+AM_INIT_AUTOMAKE($1, $2, [foreign])
 AM_DISABLE_LIBRARIES
 AC_PREFIX_DEFAULT(${KDEDIR:-/usr/local/kde})
 AC_CHECK_COMPILERS
@@ -3189,9 +3214,9 @@
     ac_save_ldflags="$LDFLAGS"
     ac_save_cflags="$CFLAGS"
     if test "x$kde_use_qt_emb" != "xyes"; then
-      LDFLAGS="$LDFLAGS $X_LDFLAGS $USER_LDFLAGS $LDFLAGS $XPM_LDFLAGS $all_libraries -lXpm -lX11 -lXext $LIBZ $LIBSOCKET"
+      LDFLAGS="$LDFLAGS $USER_LDFLAGS $X_LDFLAGS $LDFLAGS $XPM_LDFLAGS $all_libraries -lXpm -lX11 -lXext $LIBZ $LIBSOCKET"
     else
-      LDFLAGS="$LDFLAGS $X_LDFLAGS $USER_LDFLAGS $LDFLAGS $XPM_LDFLAGS $all_libraries -lXpm $LIBZ $LIBSOCKET"
+      LDFLAGS="$LDFLAGS $USER_LDFLAGS $X_LDFLAGS $LDFLAGS $XPM_LDFLAGS $all_libraries -lXpm $LIBZ $LIBSOCKET"
     fi
     CFLAGS="$CFLAGS $X_INCLUDES $USER_INCLUDES"
     test -n "$XPM_INCLUDE" && CFLAGS="-I$XPM_INCLUDE $CFLAGS"
@@ -3418,13 +3443,23 @@
           AC_CHECK_LIB(pam, pam_start,
             [ AC_CHECK_HEADER(security/pam_appl.h,
                 [ use_pam=yes
-                  pam_service=kde ]) 
+		  pam_header=security/pam_appl.h
+                  pam_service=kde ],
+		[ AC_CHECK_HEADER(pam/pam_appl.h,
+			[ use_pam=yes
+			  pam_header=pam/pam_appl.h
+			  pam_service=kde ])
+		]) 
             ], , $LIBDL)
           ac_cv_path_pam="use_pam=$use_pam pam_service=$pam_service"
         ])
     ])
   eval "$ac_cv_path_pam"
 
+  if test "$pam_header" = "pam/pam_appl.h"; then
+    AC_DEFINE(PAM_HEADER_IN_PAM, 1, [Define if your pam headers are in pam/ instead of security/])
+  fi
+
   AC_MSG_CHECKING(for PAM)
   if test "x$use_pam" = xno; then
     AC_MSG_RESULT(no)
@@ -3434,17 +3469,18 @@
     AC_DEFINE(HAVE_PAM, 1, [Defines if you have PAM (Pluggable Authentication Modules)])
     PAMLIBS="$PAM_MISC_LIB -lpam $LIBDL"
 
+    AC_CHECK_HEADERS([pam/pam_appl.h])
+
     dnl test whether struct pam_message is const (Linux) or not (Sun)
     AC_MSG_CHECKING(for const pam_message)
-    AC_EGREP_HEADER([struct pam_message], security/pam_appl.h,
-      [ AC_EGREP_HEADER([const struct pam_message], security/pam_appl.h,
+    AC_EGREP_HEADER([struct pam_message], $pam_header,
+      [ AC_EGREP_HEADER([const struct pam_message], $pam_header,
                         [AC_MSG_RESULT([const: Linux-type PAM])],
                         [AC_MSG_RESULT([nonconst: Sun-type PAM])
                         AC_DEFINE(PAM_MESSAGE_NONCONST, 1, [Define if your PAM support takes non-const arguments (Solaris)])]
                         )],
       [AC_MSG_RESULT([not found - assume const, Linux-type PAM])])
   fi
-
   AC_SUBST(PAMLIBS)
 ])
 
@@ -4604,6 +4640,83 @@
      LIBS="$kde_save_LIBS"
 ])
 
+AC_DEFUN(KDE_CHECK_LDAP,
+[
+  AC_MSG_CHECKING([for LDAP])
+
+  if test -n "$1"; then
+    ldap_extra_dirs="$1"
+  fi
+
+  ldap_incdirs="$ldap_extra_dirs/include /usr/include /usr/local/include/ $kde_extra_includes"
+  AC_FIND_FILE(ldap.h, $ldap_incdirs, ldap_incdir)
+  if test ! -r $ldap_incdir/ldap.h; then
+    ldap_incdir=no
+  fi
+
+  ldap_libdirs="$ldap_extra_dirs/lib /usr/lib /usr/local /usr/lib $kde_extra_libs"
+  for ext in la so sl dylib a; do
+    AC_FIND_FILE(libldap.$ext, $ldap_libdirs, ldap_libdir_test)
+    if test -r $ldap_libdir_test/libldap.$ext; then
+      ldap_libdir=$ldap_libdir_test
+      break
+    fi
+  done
+  if test -z "$ldap_libdir"; then
+    ldap_libdir=no
+  fi
+  if test x$ldap_libdir != xno && test x$ldap_incdir != xno; then
+    LDAP_INCS=-I$ldap_incdir
+    LDAP_LIBS=-L$ldap_libdir
+    # on MacOSX, we link against the framework instead of
+    # directly to the libraries, for forward-compatibility
+    if test -d "/System/Library/Frameworks/LDAP.framework"; then
+      LIBLDAP="-framework LDAP"
+    else
+      LIBLDAP="-lldap -llber -lresolv"
+    fi
+    if test "$USE_RPATH" = "yes"; then
+      LDAP_RPATH="-R $LDAPLIBDIR/lib"
+    fi
+
+    kde_safe_LIBS="$LIBS"
+    kde_safe_CFLAGS="$CFLAGS"
+    LIBS="$LIBS $all_libraries $LDAP_LIBS $LIBLDAP $KERBEROS_LIBS $X_EXTRA_LIBS"
+    CFLAGS="$CFLAGS $all_includes $LDAP_INCS $KERBEROS_INCS"
+    AC_LANG_SAVE
+    AC_LANG_C
+    AC_TRY_LINK(
+      [#include <lber.h>
+#include <ldap.h>
+#if LDAP_VENDOR_VERSION < 20000
+#error OpenLDAP version too old, please upgrade to version 2 or higher
+#endif
+],
+      [ LDAP *ldap; ],
+      [ ldap_old=no ],
+      [ ldap_old=yes ]
+    )
+    LIBS="$kde_safe_LIBS"
+    CFLAGS="$kde_safe_CFLAGS"
+
+    if test x$ldap_old = xno; then
+      AC_SUBST(LDAP_INCS)
+      AC_SUBST(LDAP_LIBS)
+      AC_SUBST(LDAP_RPATH)
+      AC_SUBST(LIBLDAP)
+      AC_DEFINE_UNQUOTED(HAVE_LIBLDAP,1,[Define if you have LDAP libraries])
+    else
+      LDAP_INCS=
+      LDAP_LIBS=
+      LDAP_RPATH=
+      LIBLDAP=
+    fi
+    AC_MSG_RESULT([libraries $ldap_libdir, headers $ldap_incdir])
+  else
+    AC_MSG_RESULT([not found])
+  fi
+])
+
 AC_DEFUN(KDE_JAVA_PREFIX,
 [
 	dir=`dirname "$1"`
@@ -4660,6 +4773,7 @@
       KDE_JAVA_PREFIX(/usr/lib/IBMJava2*)
       KDE_JAVA_PREFIX(/usr/lib/IBMJava*)
       KDE_JAVA_PREFIX(/opt/java*)
+      KDE_JAVA_PREFIX(/Library/Java/Home)
     
       kde_cv_path="NONE"
       kde_save_IFS=$IFS
@@ -4711,10 +4825,20 @@
 dnl At this point kde_java_bindir and kde_java_includedir are either set or "no"
 if test "x$kde_java_bindir" != "xno"; then
 
+  case $host_os in
+    darwin*)
+      dnl Look for libjvm.dylib
+      kde_java_libjvmdir=`find $kde_java_bindir/.. -name libjvm.dylib | sed 's,libjvm.dylib,,'|head -n 1`
+      kd_java_libhpidir=''
+      dnl Darwin doesn't have libhpi
+      ;;
+    *)
   dnl Look for libjvm.so
   kde_java_libjvmdir=`find $kde_java_bindir/.. -name libjvm.so | sed 's,libjvm.so,,'|head -n 1`
   dnl Look for libhpi.so and avoid green threads
-  kde_java_libhpidir=`find $kde_java_bindir/.. -name libhpi.so | grep -v green | sed 's,libhpi.so,,' | head -n 1`
+  kde_java_libhpidir=`find $kde_java_bindir/.. -name libhpi.so | grep -v green | sed 's,libhpi.so,,' | tail -n 1`
+      ;;
+  esac
 
   dnl Now check everything's fine under there
   dnl the include dir is our flag for having the JDK
@@ -4743,19 +4867,31 @@
     JAVAC=
     jni_includes=
   fi
-
+  case $host_os in
+    darwin*)
+      if test ! -r "$kde_java_libjvmdir/libjvm.dylib"; then
+        AC_MSG_ERROR([libjvm.dylib not found under $kde_java_libjvmdir. Use --without-java.])
+      fi
+      dnl Darwin doesn't have libhpi, but it's not fatal
+      LIB_JVM="-framework JavaVM"
+      LIB_HPI=""
+      ;;
+    *)
   if test ! -r "$kde_java_libjvmdir/libjvm.so"; then
      AC_MSG_ERROR([libjvm.so not found under $kde_java_libjvmdir. Use --without-java.])
   fi 
+      LIB_JVM="-L$kde_java_libjvmdir -ljvm"
+      if test ! -r "$kde_java_libhpidir/libhpi.so"; then
+        AC_MSG_ERROR([libhpi.so not found under $kde_java_libhpidir. Use --without-java.])
+      fi
+      LIB_HPI="-L$kde_java_libhpidir -lhpi"
+      ;;
+  esac
 
   if test ! -x "$kde_java_bindir/java"; then
       AC_MSG_ERROR([java not found under $kde_java_bindir. javac was found though! Use --with-java or --without-java.])
   fi
 
-  if test ! -r "$kde_java_libhpidir/libhpi.so"; then
-    AC_MSG_ERROR([libhpi.so not found under $kde_java_libhpidir. Use --without-java.])
-  fi
-
   if test -n "$jni_includes"; then
     dnl Check for JNI version
     AC_LANG_SAVE
@@ -5119,8 +5255,7 @@
   LDFLAGS="$LDFLAGS -L$ssl_libdir $all_libraries"
   LIBS="$LIBS $LIBSSL -lRSAglue -lrsaref"
 
-  AC_TRY_LINK(,void RSAPrivateEncrypt(void);RSAPrivateEncrypt();,
-  ac_ssl_rsaref="yes"
+  AC_TRY_LINK(,void RSAPrivateEncrypt(void);RSAPrivateEncrypt();, ac_ssl_rsaref="yes"
   ,
   ac_ssl_rsaref="no"
   )
===================================================================
diff -u -u -r1.1.1.7 cvs.sh
--- admin/cvs.sh	20 Nov 2002 00:45:12 -0000	1.1.1.7
+++ admin/cvs.sh	24 Nov 2002 06:59:54 -0000
@@ -119,7 +119,7 @@
 fi
 
 echo "*** Creating Makefile templates"
-$AUTOMAKE || exit 1
+$AUTOMAKE --foreign || exit 1
 if test -z "$UNSERMAKE"; then
   echo "*** Postprocessing Makefile templates"
   perl -w admin/am_edit || exit 1
===================================================================
diff -u -u -r1.1.1.4 detect-autoconf.sh
--- admin/detect-autoconf.sh	20 Nov 2002 00:45:12 -0000	1.1.1.4
+++ admin/detect-autoconf.sh	24 Nov 2002 06:59:54 -0000
@@ -42,12 +42,15 @@
 checkAutomakeAclocal ()
 {
   if test -z "$UNSERMAKE"; then
-    if test -x "`$WHICH automake-1.5`" ; then
-      AUTOMAKE="`$WHICH automake-1.5`"
-      ACLOCAL="`$WHICH aclocal-1.5`"
+    if test -x "`$WHICH automake-1.7`" ; then
+      AUTOMAKE="`$WHICH automake-1.7`"
+      ACLOCAL="`$WHICH aclocal-1.7`"
     elif test -x "`$WHICH automake-1.6`" ; then
       AUTOMAKE="`$WHICH automake-1.6`"
       ACLOCAL="`$WHICH aclocal-1.6`"
+    elif test -x "`$WHICH automake-1.5`" ; then
+      AUTOMAKE="`$WHICH automake-1.5`"
+      ACLOCAL="`$WHICH aclocal-1.5`"
     fi
   else
      AUTOMAKE="$UNSERMAKE"
===================================================================
diff -u -u -r1.1.1.5 libtool.m4.in
--- admin/libtool.m4.in	20 Nov 2002 00:45:13 -0000	1.1.1.5
+++ admin/libtool.m4.in	24 Nov 2002 06:59:55 -0000
@@ -65,7 +65,7 @@
 LIBTOOL_DEPS="$ac_aux_dir/ltmain.sh"
 
 # Always use our own libtool.
-LIBTOOL='$(SHELL) $(top_builddir)/libtool --silent'
+LIBTOOL='$(SHELL) $(top_builddir)/libtool'
 AC_SUBST(LIBTOOL)dnl
 
 # Prevent multiple expansion
@@ -609,6 +609,12 @@
     lt_cv_sys_max_cmd_len=-1;
     ;;
 
+  darwin*)
+    # this should be enough... the length test freaks out some versions
+    # of OSX's `expr`
+    lt_cv_sys_max_cmd_len=16384;
+    ;;
+
   *)
     # If test is not a shell built-in, we'll probably end up computing a
     # maximum length that is only half of the actual maximum length, but
@@ -1908,7 +1914,8 @@
     lt_cv_file_magic_test_file=`/System/Library/Frameworks/System.framework/System`
     ;;
   *) # Darwin 1.3 on
-    lt_cv_file_magic_test_file='/usr/lib/libSystem.dylib'
+    lt_cv_deplibs_check_method='pass_all'
+    lt_cv_file_magic_cmd=''
     ;;
   esac
   ;;
@@ -2560,6 +2567,11 @@
 	;;
     esac
     ;;
+  darwin*)
+    shared_flag='-dynamiclib'
+    _LT_AC_TAGVAR(archive_cmds, $1)='$CC -r -keep_private_externs -nostdlib -o ${lib}-master.o $libobjs && $CC $(test .$module = .yes && echo -bundle || echo -dynamiclib) $predep_objects ${lib}-master.o $deplibs $postdep_objects $compiler_flags -o $lib $(test x$module != xyes && echo -install_name $rpath/$soname $verstring)'
+    output_verbose_link_cmd='$CC $(test .$module = .yes && echo -bundle || echo -dynamiclib) $CFLAGS -v conftest.$objext 2>&1 | egrep "\-L"'
+    ;;
   dgux*)
     case $cc_basename in
       ec++)
@@ -4750,7 +4762,9 @@
 	_LT_AC_TAGVAR(allow_undefined_flag, $1)='-undefined suppress'
 	;;
       *) # Darwin 1.3 on
-	_LT_AC_TAGVAR(allow_undefined_flag, $1)='-flat_namespace -undefined suppress'
+	#_LT_AC_TAGVAR(allow_undefined_flag, $1)='-flat_namespace -undefined suppress'
+        # testing with twolevel namespaces
+	_LT_AC_TAGVAR(allow_undefined_flag, $1)='-multiply_defined suppress'
 	;;
       esac
 
@@ -4758,7 +4772,7 @@
       #        cross-compilation, but unfortunately the echo tests do not
       #        yet detect zsh echo's removal of \ escapes.  Also zsh mangles
       #	       `"' quotes if we put them in here... so don't!
-      _LT_AC_TAGVAR(archive_cmds, $1)='$CC $(test .$module = .yes && echo -bundle || echo -dynamiclib) $allow_undefined_flag -o $lib $libobjs $deplibs$linker_flags -install_name $rpath/$soname $verstring'
+      _LT_AC_TAGVAR(archive_cmds, $1)='$CC -r -keep_private_externs -nostdlib -o ${lib}-master.o $libobjs && $CC $(test .$module = .yes && echo -bundle || echo -dynamiclib) $allow_undefined_flag -o $lib ${lib}-master.o $deplibs $linker_flags $(test x$module != xyes && echo -install_name $rpath/$soname $verstring)'
       # We need to add '_' to the symbols in $export_symbols first
       #_LT_AC_TAGVAR(archive_expsym_cmds, $1)="$_LT_AC_TAGVAR(archive_cmds, $1)"' && strip -s $export_symbols'
       _LT_AC_TAGVAR(hardcode_direct, $1)=yes
===================================================================
diff -u -u -r1.1.1.4 ltmain.sh
--- admin/ltmain.sh	20 Nov 2002 00:45:14 -0000	1.1.1.4
+++ admin/ltmain.sh	24 Nov 2002 06:59:56 -0000
@@ -2532,11 +2532,11 @@
 	eval libname=\"$libname_spec\"
 	;;
       *)
-	if test "$module" = no; then
-	  $echo "$modename: libtool library \`$output' must begin with \`lib'" 1>&2
-	  $echo "$help" 1>&2
-	  exit 1
-	fi
+#	if test "$module" = no; then
+#	  $echo "$modename: libtool library \`$output' must begin with \`lib'" 1>&2
+#	  $echo "$help" 1>&2
+#	  exit 1
+#	fi
 	if test "$need_lib_prefix" != no; then
 	  # Add the "lib" prefix for modules if required
 	  name=`$echo "X$outputname" | $Xsed -e 's/\.la$//'`
@@ -3321,7 +3321,23 @@
 	if test -n "$export_symbols" && test -n "$archive_expsym_cmds"; then
 	  eval cmds=\"$archive_expsym_cmds\"
 	else
+          if test "x$verstring" = "x0.0"; then
+                tmp_verstring=
+          else
+                tmp_verstring="$verstring"
+          fi
+ 	  save_deplibs="$deplibs"
+ 	  for conv in $convenience; do
+            tmp_deplibs=
+            for test_deplib in $deplibs; do
+              if test "$test_deplib" != "$conv"; then
+                tmp_deplibs="$tmp_deplibs $test_deplib"
+              fi
+            done
+            deplibs="$tmp_deplibs"
+ 	  done
 	  eval cmds=\"$archive_cmds\"
+ 	  deplibs="$save_deplibs"
 	fi
 
 	if len=`expr "X$cmds" : ".*"` &&
@@ -3419,7 +3435,12 @@
 	  if test -n "$export_symbols" && test -n "$archive_expsym_cmds"; then
 	    eval cmds=\"$archive_expsym_cmds\"
 	  else
-	    eval cmds=\"$archive_cmds\"
+            save_deplibs="$deplibs"
+            for conv in $convenience; do
+              deplibs="${deplibs%$conv*} ${deplibs#*$conv}"
+            done
+            eval cmds=\"$archive_cmds\"
+            deplibs="$save_deplibs"
 	  fi
 
 	  # Append the command to remove the reloadable object files
@@ -3634,6 +3655,14 @@
 	;;
       esac
 
+      case $host in
+      *darwin*)
+        # Don't allow lazy linking, it breaks C++ global constructors
+        compile_command="$compile_command ${wl}-bind_at_load -undefined suppress -flat_namespace"
+        finalize_command="$finalize_command ${wl}-bind_at_load"
+        ;;
+      esac
+
       compile_command="$compile_command $compile_deplibs"
       finalize_command="$finalize_command $finalize_deplibs"
 
@@ -4024,9 +4053,10 @@
 	    # fast_install is set to needless
 	    relink_command=
 	  fi
+          relink_command=
 	else
 	  link_command="$compile_var$compile_command$compile_rpath"
-	  relink_command="$finalize_var$finalize_command$finalize_rpath"
+	  #relink_command="$finalize_var$finalize_command$finalize_rpath"
 	fi
       fi
 
===================================================================
diff -u -u -r1.1.1.4 -r1.5
===================================================================
diff -u -u -r1.1.1.3 -r1.5
===================================================================
diff -N ./debian/konsole.links
===================================================================
diff -N ./doc/kaddressbook/index.cache.bz2
===================================================================
diff -u -u -r1.1.1.4 Makefile.am
--- ./kate/app/Makefile.am	12 Nov 2002 04:16:41 -0000	1.1.1.4
+++ ./kate/app/Makefile.am	24 Nov 2002 07:06:52 -0000
@@ -7,16 +7,15 @@
 	kateviewmanager.cpp kateviewspace.cpp \
 	kateIface.skel kategrepdialog.cpp katefiledialog.cpp katemailfilesdialog.cpp \
         kbookmarkhandler.cpp kactionselector.cpp katedockcontainer.cpp
-
-libkateinterfaces_la_LIBADD   = ../interfaces/libkateinterfacesprivate.la -lkscript ../lib/libkmultitabbar.la
-libkateinterfaces_la_LDFLAGS = $(all_libraries) $(KDE_RPATH) -module -avoid-version
+libkateinterfaces_la_LIBADD   = $(top_builddir)/kate/interfaces/libkateinterfacesprivate.la -lkscript $(top_builddir)/kate/lib/libkmultitabbar.la
+libkateinterfaces_la_LDFLAGS = $(all_libraries) $(KDE_RPATH) -avoid-version
 kate_la_SOURCES = katemain.cpp
-kate_la_LIBADD = libkateinterfaces.la
-kate_la_LDFLAGS = $(all_libraries) $(KDE_RPATH) -module -avoid-version 
+kate_la_LIBADD  = libkateinterfaces.la
+kate_la_LDFLAGS = $(all_libraries) $(KDE_RPATH) -avoid-version
 
 kwrite_la_SOURCES = kwritemain.cpp katefiledialog.cpp kwritedialogs.cpp
-kwrite_la_LIBADD   = -lktexteditor
-kwrite_la_LDFLAGS = $(all_libraries) $(KDE_RPATH) -module -avoid-version
+kwrite_la_LIBADD  = -lktexteditor $(LIB_QT) -lkdecore -lDCOP -lkdeui -lkio -lkparts
+kwrite_la_LDFLAGS = $(all_libraries) $(KDE_RPATH) -avoid-version
 
 kate_SOURCES = dummy_kate.cpp
 kate_LDADD = kate.la
===================================================================
diff -u -u -r1.1.1.3 -r1.6
===================================================================
diff -u -u -r1.1.1.1 -r1.2
--- ./kcheckpass/checkpass_pam.c	25 Jun 2002 02:03:44 -0000	1.1.1.1
+++ ./kcheckpass/checkpass_pam.c	21 Nov 2002 21:28:36 -0000	1.2
@@ -26,7 +26,12 @@
  *****************************************************************/
 #include <stdlib.h>
 #include <string.h>
+
+#ifdef HAVE_PAM_PAM_APPL_H
+#include <pam/pam_appl.h>
+#else
 #include <security/pam_appl.h>
+#endif
 
 static const char *PAM_username;
 static const char *PAM_password;
===================================================================
diff -u -u -r1.1.1.3 Makefile.am
--- ./kcontrol/access/Makefile.am	1 Oct 2002 02:33:58 -0000	1.1.1.3
+++ ./kcontrol/access/Makefile.am	24 Nov 2002 07:06:52 -0000
@@ -12,8 +12,8 @@
 lib_LTLIBRARIES = kaccess.la
 
 kaccess_la_SOURCES = kaccess.cpp main.cpp
-kaccess_la_LDFLAGS = $(all_libraries) -module -avoid-version
-kaccess_la_LIBADD = $(LIB_KDEUI)
+kaccess_la_LDFLAGS = $(all_libraries) -avoid-version
+kaccess_la_LIBADD = $(LIB_KDEUI) $(LIB_QT) -lkdecore -lDCOP
 
 kcm_access_la_SOURCES = kcmaccess.cpp 
 kcm_access_la_LDFLAGS  = $(all_libraries) -module -avoid-version -no-undefined
===================================================================
diff -u -u -r1.1.1.3 -r1.5
===================================================================
diff -u -u -r1.1.1.4 -r1.5
===================================================================
diff -u -u -r1.1.1.1 -r1.3
===================================================================
diff -u -u -r1.1.1.1 -r1.3
===================================================================
diff -u -u -r1.1.1.2 -r1.4
===================================================================
diff -u -u -r1.1.1.3 -r1.5
===================================================================
diff -u -u -r1.1.1.3 -r1.5
===================================================================
diff -u -u -r1.1.1.3 -r1.5
===================================================================
diff -u -u -r1.1.1.2 -r1.4
===================================================================
diff -u -u -r1.1.1.1 -r1.3
===================================================================
diff -u -u -r1.1.1.3 -r1.5
===================================================================
diff -u -u -r1.1.1.4 -r1.6
===================================================================
diff -u -u -r1.1.1.3 -r1.5
===================================================================
diff -u -u -r1.1.1.3 -r1.5
===================================================================
diff -u -u -r1.1.1.3 -r1.5
===================================================================
diff -u -u -r1.1.1.3 -r1.4
--- ./kcontrol/info/info.cpp	29 Oct 2002 03:03:00 -0000	1.1.1.3
+++ ./kcontrol/info/info.cpp	29 Oct 2002 03:39:29 -0000	1.4
@@ -471,6 +471,8 @@
 #include "info_svr4.cpp"
 #elif _AIX
 #include "info_aix.cpp"
+#elif defined(__APPLE__)
+#include "info_osx.cpp"
 #else
 #include "info_generic.cpp"	/* Default for unsupportet systems.... */
 #endif
===================================================================
diff -N ./kcontrol/info/info_osx.cpp
--- /dev/null	1 Jan 1970 00:00:00 -0000
+++ ./kcontrol/info/info_osx.cpp	25 Jun 2002 02:06:13 -0000	1.1
@@ -0,0 +1,163 @@
+/*
+ * kdebase3-3.0.1-3.patch,v 1.2 2002/06/24 01:48:12 ranger Exp
+ *
+ * info_fbsd.cpp is part of the KDE program kcminfo.  This displays
+ * various information about the system (hopefully a FreeBSD system)
+ * it's running on.
+ */
+
+#define INFO_CPU_AVAILABLE
+//#define INFO_IRQ_AVAILABLE
+//#define INFO_DMA_AVAILABLE
+//#define INFO_PCI_AVAILABLE
+//#define INFO_IOPORTS_AVAILABLE
+#define INFO_SOUND_AVAILABLE
+#define INFO_DEVICES_AVAILABLE
+#define INFO_SCSI_AVAILABLE
+#define INFO_PARTITIONS_AVAILABLE
+#define INFO_XSERVER_AVAILABLE
+
+/*
+ * all following functions should return TRUE, when the Information
+ * was filled into the lBox-Widget. Returning FALSE indicates that
+ * information was not available.
+ */
+
+#include <sys/types.h>
+#include <sys/sysctl.h>
+
+#include <fstab.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+#include <iostream.h>
+
+#include <qdict.h>
+#include <qfile.h>
+#include <qfontmetrics.h>
+#include <qptrlist.h>
+#include <qstring.h>
+#include <qtextstream.h>
+
+#include <kdebug.h>
+
+#include <mach/mach.h>
+#include <mach-o/arch.h>
+
+#ifdef HAVE_COREAUDIO
+#include <CoreAudio/CoreAudio.h>
+#endif
+
+#include <machine/limits.h>
+
+bool GetInfo_CPU (QListView *lBox)
+{
+
+	QString cpustring;
+
+        kern_return_t ret;
+        struct host_basic_info basic_info;
+        unsigned int count=HOST_BASIC_INFO_COUNT;
+
+        ret=host_info(mach_host_self(), HOST_BASIC_INFO, 
+                (host_info_t)&basic_info, &count);
+        if (ret != KERN_SUCCESS) {
+		kdDebug() << "unable to get host information from mach" << endl;
+		return false;
+        } else {
+		kdDebug() << "got Host Info: (" << basic_info.avail_cpus << ") CPUs available" << endl;
+        	const NXArchInfo *archinfo;
+        	archinfo=NXGetArchInfoFromCpuType(basic_info.cpu_type, basic_info.cpu_subtype);
+		new QListViewItem(lBox, i18n("Kernel is configured for %1 CPUs").arg(basic_info.max_cpus));
+		for (int i = 1; i <= basic_info.avail_cpus; i++) {
+			cpustring = i18n("CPU %1: %2").arg(i).arg(archinfo->description);
+			new QListViewItem(lBox, cpustring);
+		}
+		return true;
+	}
+	return false;
+}
+
+bool GetInfo_IRQ (QListView *)
+{
+	return false;
+}
+
+bool GetInfo_DMA (QListView *)
+{
+	return false;
+}
+
+bool GetInfo_PCI (QListView *)
+{
+	return false;
+}
+
+bool GetInfo_IO_Ports (QListView *)
+{
+	return false;
+}
+
+bool GetInfo_Sound (QListView *lBox)
+{
+#ifdef HAVE_COREAUDIO
+#define kMaxStringSize 1024
+	OSStatus status;
+	AudioDeviceID gOutputDeviceID;
+	unsigned long propertySize;
+	char deviceName[kMaxStringSize];
+	char manufacturer[kMaxStringSize];
+	propertySize = sizeof(gOutputDeviceID);
+	status = AudioHardwareGetProperty(kAudioHardwarePropertyDefaultOutputDevice, &propertySize, &gOutputDeviceID);
+	if (status) {
+		kdDebug() << "get default output device failed, status = " << (int)status << endl;
+		return false;
+	}
+
+	if (gOutputDeviceID != kAudioDeviceUnknown) {
+
+		propertySize = kMaxStringSize;
+
+		/* Device Name */
+		status = AudioDeviceGetProperty(gOutputDeviceID, 1, 0, kAudioDevicePropertyDeviceName, &propertySize, deviceName);
+		if (status) {
+			kdDebug() << "get device name failed, status = " << (int)status << endl;
+			return false;
+		}
+		new QListViewItem(lBox, i18n("Device Name: %1").arg(deviceName));
+
+		/* Manufacturer */
+		status = AudioDeviceGetProperty(gOutputDeviceID, 1, 0, kAudioDevicePropertyDeviceManufacturer, &propertySize, manufacturer);
+		if (status) {
+			kdDebug() << "get manufacturer failed, status = " << (int)status << endl;
+			return false;
+		}
+		new QListViewItem(lBox, i18n("Manufacturer: %1").arg(manufacturer));
+		return true;
+	} else {
+		return false;
+	}
+#else
+	return false;
+#endif
+}
+
+bool GetInfo_SCSI (QListView *lbox)
+{
+	return false;
+}
+
+bool GetInfo_Partitions (QListView *lbox)
+{
+	return false;
+}
+
+bool GetInfo_XServer_and_Video (QListView *lBox)
+{
+	return GetInfo_XServer_Generic( lBox );
+}
+
+bool GetInfo_Devices (QListView *lbox)
+{
+	return false;
+}
===================================================================
diff -u -u -r1.1.1.1 Makefile.am
--- ./kcontrol/info/Makefile.am	25 Jun 2002 02:03:45 -0000	1.1.1.1
+++ ./kcontrol/info/Makefile.am	24 Nov 2002 07:06:53 -0000
@@ -2,7 +2,7 @@
 
 kcm_info_la_SOURCES = main.cpp memory.cpp
 
-kcm_info_la_LDFLAGS = $(all_libraries) -module -avoid-version -no-undefined
+kcm_info_la_LDFLAGS = $(all_libraries) -module -avoid-version -no-undefined $(FRAMEWORK_COREAUDIO)
 kcm_info_la_LIBADD = $(LIBALIB) $(LIB_KDEUI) $(LIB_TRU64_MACH) $(LIBCFG) $(LIBODM) $(LIBKSTAT) $(LIBDEVINFO)
 
 INCLUDES= $(all_includes)
===================================================================
diff -u -u -r1.1.1.4 -r1.5
--- ./kcontrol/info/memory.cpp	1 Oct 2002 02:34:08 -0000	1.1.1.4
+++ ./kcontrol/info/memory.cpp	1 Oct 2002 03:01:46 -0000	1.5
@@ -431,6 +431,8 @@
 
 #ifdef __linux__
 #include "memory_linux.cpp"
+#elif defined(__APPLE__)
+#include "memory_osx.cpp"
 #elif sgi
 #include "memory_sgi.cpp"
 #elif defined(__svr4__) && defined(sun)
===================================================================
diff -N ./kcontrol/info/memory_osx.cpp
--- /dev/null	1 Jan 1970 00:00:00 -0000
+++ ./kcontrol/info/memory_osx.cpp	25 Jun 2002 02:06:13 -0000	1.1
@@ -0,0 +1,72 @@
+#include <unistd.h>
+#include <stdlib.h>
+#include <qfile.h>
+#include <mach/mach_init.h>
+#include <mach/mach_host.h>
+#include <mach/host_info.h>
+#include <sys/stat.h>
+#include <dirent.h>
+#include <kdebug.h>
+
+void KMemoryWidget::update()
+{
+
+	vm_statistics_data_t vm_info;
+	mach_msg_type_number_t info_count;
+	DIR *dirp;
+	struct dirent *dp;
+	off_t total;
+
+	info_count = HOST_VM_INFO_COUNT;
+	if (host_statistics(mach_host_self (), HOST_VM_INFO, (host_info_t)&vm_info, &info_count)) {
+		kdDebug() << "could not get memory statistics" << endl;
+		return;
+	}
+
+	Memory_Info[TOTAL_MEM]    = (vm_info.active_count + vm_info.inactive_count +
+		vm_info.free_count + vm_info.wire_count) * vm_page_size;
+	Memory_Info[FREE_MEM]     = vm_info.free_count * vm_page_size;
+	Memory_Info[SHARED_MEM]   = NO_MEMORY_INFO;
+	Memory_Info[BUFFER_MEM]   = NO_MEMORY_INFO;
+	Memory_Info[CACHED_MEM]   = NO_MEMORY_INFO;
+
+	dirp = opendir("/private/var/vm");
+	if (!dirp) {
+		kdDebug() << "unable to open /private/var/vm" << endl;
+		return;
+	}
+
+	total = 0;
+
+	while ((dp = readdir (dirp)) != NULL) {
+		struct stat sb;
+		char fname [MAXNAMLEN];
+
+		if (strncmp (dp->d_name, "swapfile", 8))
+			continue;
+
+		strcpy (fname, "/private/var/vm/");
+		strcat (fname, dp->d_name);
+		if (stat (fname, &sb) < 0)
+			continue;
+
+		total += sb.st_size;
+	}
+	closedir (dirp);
+
+	info_count = HOST_VM_INFO_COUNT;
+	if (host_statistics (mach_host_self (), HOST_VM_INFO,
+		(host_info_t) &vm_info, &info_count)) {
+			kdDebug() << "unable to get VM info" << endl;
+	}
+
+	Memory_Info[SWAP_MEM]     = total;
+	// off_t used = (vm_info.pageouts - vm_info.pageins) * vm_page_size;
+	Memory_Info[FREESWAP_MEM] = NO_MEMORY_INFO;
+
+	/* free = vm_info.free_count * vm_page_size;
+	   used = vm_info.active_count * vm_page_size;
+	   total = (vm_info.active_count + vm_info.inactive_count +
+		vm_info.free_count + vm_info.wire_count) * vm_page_size; */
+
+}
===================================================================
diff -u -u -r1.1.1.2 -r1.4
===================================================================
diff -u -u -r1.1.1.2 -r1.4
===================================================================
diff -u -u -r1.1.1.5 Makefile.am
--- ./kcontrol/kcontrol/Makefile.am	5 Nov 2002 00:15:25 -0000	1.1.1.5
+++ ./kcontrol/kcontrol/Makefile.am	24 Nov 2002 07:06:53 -0000
@@ -33,19 +33,19 @@
 	quickhelp.cpp moduletreeview.cpp moduleiconview.cpp \
 	kecdialog.cpp kcrootonly.cpp moduleIface.cpp moduleIface.skel
 
-kcontrol_la_LIBADD  = $(LIB_KDEUI) $(LIB_KSYCOCA)
-kcontrol_la_LDFLAGS = $(all_libraries) -module -avoid-version
+kcontrol_la_LIBADD  = $(LIB_KDEUI) $(LIB_KSYCOCA) $(LIB_QT) -lkdecore -lDCOP -lkdefx
+kcontrol_la_LDFLAGS = $(all_libraries) -avoid-version
 
 kcmshell_la_SOURCES = kcmshell.cpp moduleinfo.cpp modloader.cpp kecdialog.cpp \
 			proxywidget.cpp kcdialog.cpp global.cpp \
 			kcdialog.skel
-kcmshell_la_LIBADD = $(LIB_KDEUI) $(LIB_KSYCOCA)
-kcmshell_la_LDFLAGS = $(all_libraries) -module -avoid-version
+kcmshell_la_LIBADD = $(LIB_KDEUI) $(LIB_KSYCOCA) $(LIB_QT) -lkdecore -lDCOP
+kcmshell_la_LDFLAGS = $(all_libraries) -avoid-version
 
 
 kcminit_la_SOURCES = kcminit.cpp global.cpp
-kcminit_la_LIBADD   = $(LIB_KDEUI) $(LIB_KSYCOCA)
-kcminit_la_LDFLAGS = $(all_libraries) -module -avoid-version
+kcminit_la_LIBADD   = $(LIB_KDEUI) $(LIB_KSYCOCA) $(LIB_QT) -lkdecore -lDCOP
+kcminit_la_LDFLAGS = $(all_libraries) -avoid-version
 
 kcontrol_SOURCES = dummy.cpp
 kcontrol_LDADD = kcontrol.la
===================================================================
diff -u -u -r1.1.1.3 -r1.3
===================================================================
diff -u -u -r1.1.1.3 -r1.5
===================================================================
diff -u -u -r1.1.1.4 -r1.3
===================================================================
diff -u -u -r1.1.1.3 -r1.5
===================================================================
diff -u -u -r1.1.1.4 -r1.5
--- ./kcontrol/kfontinst/kfontinst/FontEngine.cpp	1 Oct 2002 02:34:18 -0000	1.1.1.4
+++ ./kcontrol/kfontinst/kfontinst/FontEngine.cpp	1 Oct 2002 03:01:46 -0000	1.5
@@ -26,6 +26,8 @@
 // (C) Craig Drummond, 2001
 ////////////////////////////////////////////////////////////////////////////////
 
+#include "config.h"
+
 #include "FontEngine.h"
 #if !defined KFI_THUMBNAIL && !defined KFI_METAINFO
 #include "KfiGlobal.h"
===================================================================
diff -u -u -r1.1.1.5 -r1.6
===================================================================
diff -u -u -r1.1.1.4 -r1.6
===================================================================
diff -u -u -r1.1.1.6 -r1.8
===================================================================
diff -u -u -r1.1.1.4 -r1.6
===================================================================
diff -u -u -r1.1.1.4 -r1.6
===================================================================
diff -u -u -r1.1.1.3 -r1.5
===================================================================
diff -u -u -r1.1.1.3 -r1.5
===================================================================
diff -u -u -r1.1.1.2 -r1.4
===================================================================
diff -u -u -r1.1.1.1 -r1.3
===================================================================
diff -u -u -r1.1.1.3 -r1.5
===================================================================
diff -u -u -r1.1.1.5 -r1.6
===================================================================
diff -u -u -r1.1.1.3 -r1.5
===================================================================
diff -u -u -r1.1.1.2 -r1.4
===================================================================
diff -u -u -r1.1.1.3 -r1.5
===================================================================
diff -u -u -r1.1.1.2 -r1.4
===================================================================
diff -u -u -r1.1.1.3 -r1.5
===================================================================
diff -u -u -r1.1.1.2 -r1.4
===================================================================
diff -u -u -r1.1.1.2 -r1.4
===================================================================
diff -u -u -r1.1.1.3 -r1.5
===================================================================
diff -u -u -r1.1.1.3 -r1.5
===================================================================
diff -u -u -r1.1.1.1 -r1.3
===================================================================
diff -u -u -r1.1.1.1 Makefile.am
--- ./kdeprint/kjobviewer/Makefile.am	25 Jun 2002 02:03:49 -0000	1.1.1.1
+++ ./kdeprint/kjobviewer/Makefile.am	24 Nov 2002 07:07:13 -0000
@@ -9,8 +9,8 @@
 
 kjobviewer_la_SOURCES = kjobviewer.cpp main.cpp
 kjobviewer_la_METASOURCES = AUTO
-kjobviewer_la_LIBADD = -lkdeprint_management $(LIB_KDEUI)
-kjobviewer_la_LDFLAGS = -module -avoid-version $(all_libraries) $(KDE_RPATH)
+kjobviewer_la_LIBADD = -lkdeprint_management $(LIB_KDEUI) $(LIB_QT) -lkdecore -lDCOP -lkdeprint
+kjobviewer_la_LDFLAGS = -avoid-version $(all_libraries) $(KDE_RPATH)
 
 noinst_HEADERS = kjobviewer.h
 
===================================================================
diff -u -u -r1.1.1.2 Makefile.am
--- ./kdeprint/kprinter/Makefile.am	10 Jul 2002 03:20:23 -0000	1.1.1.2
+++ ./kdeprint/kprinter/Makefile.am	24 Nov 2002 07:07:13 -0000
@@ -8,8 +8,8 @@
 kprinter_LDFLAGS = $(KDE_RPATH) $(all_libraries)
 
 kprinter_la_SOURCES = main.cpp printwrapper.cpp
-kprinter_la_LIBADD = -lkdeprint
-kprinter_la_LDFLAGS = -module -avoid-version $(all_libraries) $(KDE_RPATH)
+kprinter_la_LIBADD = -lkdeprint $(LIB_QT) -lkdecore -lDCOP -lkdeui -lkio
+kprinter_la_LDFLAGS = -avoid-version $(all_libraries) $(KDE_RPATH)
 kprinter_METASOURCES = AUTO
 
 noinst_HEADERS = printwrapper.h
===================================================================
diff -u -u -r1.1.1.3 Makefile.am
--- ./kdesktop/Makefile.am	1 Oct 2002 02:35:37 -0000	1.1.1.3
+++ ./kdesktop/Makefile.am	24 Nov 2002 07:07:14 -0000
@@ -1,10 +1,10 @@
 ## Makefile.am of kdebase/kdesktop
 
 INCLUDES = -I$(top_srcdir)/kcontrol/background -I$(top_srcdir)/libkonq $(all_includes)
-kdesktop_la_LDFLAGS  = $(KDE_RPATH) $(all_libraries) -module -avoid-version
+kdesktop_la_LDFLAGS  = $(KDE_RPATH) $(all_libraries) -avoid-version
 kdesktop_la_LIBADD   = $(top_builddir)/libkonq/libkonq.la $(top_builddir)/kcontrol/background/libbgnd.la 
 kdesktop_LDFLAGS  = $(all_libraries) $(KDE_RPATH)
-kdesktop_LDADD    = kdesktop.la
+kdesktop_LDADD = kdesktop.la
 
 SUBDIRS = . lock pics patterns programs init kwebdesktop
 
===================================================================
diff -u -u -r1.1.1.2 -r1.6
===================================================================
diff -u -u -r1.1.1.3 -r1.2
--- ./kdm/backend/client.c	29 Oct 2002 03:04:57 -0000	1.1.1.3
+++ ./kdm/backend/client.c	21 Nov 2002 21:28:42 -0000	1.2
@@ -38,6 +38,7 @@
  * user verification and session initiation.
  */
 
+#include <config.h>
 #include "dm.h"
 #include "dm_auth.h"
 #include "dm_error.h"
@@ -59,7 +60,11 @@
 # endif
 #endif
 #ifdef USE_PAM
-# include <security/pam_appl.h>
+#ifdef HAVE_PAM_PAM_APPL_H
+#include <pam/pam_appl.h>
+#else
+#include <security/pam_appl.h>
+#endif
 #elif defined(AIXV3) /* USE_PAM */
 # include <login.h>
 # include <usersec.h>
===================================================================
diff -u -u -r1.1.1.4 -r1.4
--- ./kdm/backend/dm.h	29 Oct 2002 03:04:57 -0000	1.1.1.4
+++ ./kdm/backend/dm.h	29 Oct 2002 03:39:31 -0000	1.4
@@ -41,6 +41,10 @@
 #ifndef _DM_H_
 #define _DM_H_ 1
 
+#ifdef __APPLE__
+#define CSRG_BASED
+#endif
+
 #include "greet.h"
 
 #include <X11/Xos.h>
===================================================================
diff -u -u -r1.1.1.7 -r1.7
===================================================================
diff -u -u -r1.1.1.1 Makefile.am
--- ./khotkeys/khotkeys/Makefile.am	25 Jun 2002 02:03:52 -0000	1.1.1.1
+++ ./khotkeys/khotkeys/Makefile.am	24 Nov 2002 07:07:15 -0000
@@ -4,11 +4,11 @@
 lib_LTLIBRARIES = khotkeys.la
 
 khotkeys_la_SOURCES = khotkeys.cpp main.cpp khotkeys.skel
-khotkeys_la_LIBADD   = ../shared/libshared.la $(LIB_KSYCOCA)
-khotkeys_la_LDFLAGS = $(all_libraries) -module -avoid-version
+khotkeys_la_LIBADD   = ../shared/libshared.la $(LIB_KSYCOCA) $(LIB_QT) -lkdecore -lDCOP
+khotkeys_la_LDFLAGS = $(all_libraries) -avoid-version
 
 khotkeys_SOURCES = dummy.cpp
-khotkeys_LDADD   = khotkeys.la
+khotkeys_LDADD = khotkeys.la
 khotkeys_LDFLAGS = $(all_libraries) $(KDE_RPATH)
 
 autostart_DATA = khotkeys.desktop
===================================================================
diff -u -u -r1.1.1.2 Makefile.am
--- ./kicker/Makefile.am	10 Jul 2002 03:21:29 -0000	1.1.1.2
+++ ./kicker/Makefile.am	24 Nov 2002 07:07:15 -0000
@@ -11,7 +11,7 @@
 	ui/libkicker_ui.la share/libkickermain.la $(LIB_KIO) 
 
 kicker_la_SOURCES = dummy.cpp
-kicker_la_LDFLAGS = $(KDE_PLUGIN) -no-undefined -module $(all_libraries)
+kicker_la_LDFLAGS = $(KDE_RPATH) -no-undefined -avoid-version $(all_libraries)
 
 kicker_LDADD = kicker.la
 kicker_SOURCES = dummy_la.cpp
===================================================================
diff -u -u -r1.1.1.1 Makefile.am
--- ./kicker/menuext/kdeprint/Makefile.am	25 Jun 2002 02:03:52 -0000	1.1.1.1
+++ ./kicker/menuext/kdeprint/Makefile.am	24 Nov 2002 07:07:15 -0000
@@ -4,7 +4,7 @@
 
 kickermenu_kdeprint_la_SOURCES = print_mnu.cpp
 kickermenu_kdeprint_la_LDFLAGS = $(all_libraries) -module -avoid-version
-kickermenu_kdeprint_la_LIBADD = $(LIB_KDEUI) $(LIB_KIO) -lkdeprint
+kickermenu_kdeprint_la_LIBADD = $(LIB_KDEUI) $(LIB_KIO) -lkdeprint $(LIB_QT) -lkdecore
 
 kickermenu_kdeprint_la_METASOURCES = AUTO
 
===================================================================
diff -u -u -r1.1.1.1 Makefile.am
--- ./kicker/menuext/konsole/Makefile.am	14 Aug 2002 14:30:12 -0000	1.1.1.1
+++ ./kicker/menuext/konsole/Makefile.am	24 Nov 2002 07:07:15 -0000
@@ -4,7 +4,7 @@
 
 kickermenu_konsole_la_SOURCES = konsole_mnu.cpp konsolebookmarkhandler.cpp konsolebookmarkmenu.cpp
 kickermenu_konsole_la_LDFLAGS = $(all_libraries) -module -avoid-version
-kickermenu_konsole_la_LIBADD = $(LIB_KDEUI) $(LIB_KIO)
+kickermenu_konsole_la_LIBADD = $(LIB_KDEUI) $(LIB_KIO) $(LIB_QT) -lkdecore
 
 kickermenu_konsole_la_METASOURCES = AUTO
 
===================================================================
diff -u -u -r1.1.1.1 Makefile.am
--- ./kicker/menuext/prefmenu/Makefile.am	10 Jul 2002 03:22:41 -0000	1.1.1.1
+++ ./kicker/menuext/prefmenu/Makefile.am	24 Nov 2002 07:07:15 -0000
@@ -3,7 +3,7 @@
 kde_module_LTLIBRARIES = kickermenu_prefmenu.la
 
 kickermenu_prefmenu_la_SOURCES = prefmenu.cpp
-kickermenu_prefmenu_la_LDFLAGS = $(all_libraries) -module -avoid-version
+kickermenu_prefmenu_la_LDFLAGS = $(all_libraries) -module -avoid-version -undefined suppress -flat_namespace
 kickermenu_prefmenu_la_LIBADD = $(LIB_KDEUI) $(top_builddir)/kicker/ui/libkicker_ui.la
 
 kickermenu_prefmenu_la_METASOURCES = AUTO
===================================================================
diff -u -u -r1.1.1.1 Makefile.am
--- ./kicker/menuext/recentdocs/Makefile.am	10 Jul 2002 03:22:42 -0000	1.1.1.1
+++ ./kicker/menuext/recentdocs/Makefile.am	24 Nov 2002 07:07:15 -0000
@@ -3,7 +3,7 @@
 kde_module_LTLIBRARIES = kickermenu_recentdocs.la
 
 kickermenu_recentdocs_la_SOURCES = recentdocsmenu.cpp
-kickermenu_recentdocs_la_LDFLAGS = $(all_libraries) -module -avoid-version
+kickermenu_recentdocs_la_LDFLAGS = $(all_libraries) -module -avoid-version -undefined suppress -flat_namespace
 kickermenu_recentdocs_la_LIBADD = $(LIB_KDEUI) $(top_builddir)/kicker/ui/libkicker_ui.la
 
 kickermenu_recentdocs_la_METASOURCES = AUTO
===================================================================
diff -u -u -r1.1.1.1 Makefile.am
--- ./kicker/proxy/Makefile.am	25 Jun 2002 02:03:52 -0000	1.1.1.1
+++ ./kicker/proxy/Makefile.am	24 Nov 2002 07:07:15 -0000
@@ -8,7 +8,7 @@
 appletproxy_la_LIBADD = $(LIB_KDEUI) ../share/libkickermain.la
 appletproxy_la_SOURCES = appletproxy.cpp
 appletproxy_la_METASOURCES = appletproxy.moc
-appletproxy_la_LDFLAGS = $(all_libraries) $(KDE_RPATH) -module -avoid-version
+appletproxy_la_LDFLAGS = $(all_libraries) $(KDE_RPATH) -avoid-version
 
 appletproxy_LDADD = appletproxy.la
 appletproxy_SOURCES = dummy.cpp
@@ -17,7 +17,7 @@
 extensionproxy_la_LIBADD = $(LIB_KDEUI) ../share/libkickermain.la
 extensionproxy_la_SOURCES = extensionproxy.cpp
 extensionproxy_la_METASOURCES = extensionproxy.moc
-extensionproxy_la_LDFLAGS = $(all_libraries) $(KDE_RPATH) -module -avoid-version
+extensionproxy_la_LDFLAGS = $(all_libraries) $(KDE_RPATH) -avoid-version
 
 extensionproxy_LDADD = extensionproxy.la
 extensionproxy_SOURCES = dummy.cpp
===================================================================
diff -u -u -r1.1.1.1 Makefile.am
--- ./kicker/share/Makefile.am	25 Jun 2002 02:03:52 -0000	1.1.1.1
+++ ./kicker/share/Makefile.am	24 Nov 2002 07:07:15 -0000
@@ -5,7 +5,7 @@
 libkickermain_la_SOURCES = pluginloader.cpp appletinfo.cpp panner.cpp fittslawframe.cpp menuinfo.cpp
 libkickermain_la_METASOURCES = AUTO
 
-libkickermain_la_LDFLAGS = $(all_libraries) -version-info 1:0:0 -module -no-undefined
+libkickermain_la_LDFLAGS = $(all_libraries) -version-info 1:0:0 -no-undefined
 libkickermain_la_LIBADD = $(LIB_KDECORE) $(LIB_KDEUI) $(top_builddir)/libkonq/libkonq.la
 
 messages:
===================================================================
diff -u -u -r1.1.1.1 Makefile.am
--- ./kicker/taskbar/Makefile.am	25 Jun 2002 02:03:52 -0000	1.1.1.1
+++ ./kicker/taskbar/Makefile.am	24 Nov 2002 07:07:15 -0000
@@ -4,7 +4,7 @@
 libtaskbar_la_SOURCES = taskcontainer.cpp taskbar.cpp taskbarcontainer.cpp
 libtaskbar_la_METASOURCES = AUTO
 
-libtaskbar_la_LDFLAGS = $(all_libraries) -version-info 1:0:0 -module -no-undefined
+libtaskbar_la_LDFLAGS = $(all_libraries) -version-info 1:0:0 -no-undefined
 libtaskbar_la_LIBADD = $(LIB_KDEUI) ../share/libkickermain.la ../taskmanager/libtaskmanager.la
 
 messages:
===================================================================
diff -u -u -r1.1.1.1 Makefile.am
--- ./kicker/taskmanager/Makefile.am	25 Jun 2002 02:03:52 -0000	1.1.1.1
+++ ./kicker/taskmanager/Makefile.am	24 Nov 2002 07:07:15 -0000
@@ -4,7 +4,7 @@
 libtaskmanager_la_SOURCES = tasklmbmenu.cpp taskrmbmenu.cpp taskmanager.cpp
 libtaskmanager_la_METASOURCES = AUTO
 
-libtaskmanager_la_LDFLAGS = $(all_libraries) -version-info 1:0:0 -module -no-undefined
+libtaskmanager_la_LDFLAGS = $(all_libraries) -version-info 1:0:0 -no-undefined
 libtaskmanager_la_LIBADD = $(LIB_KDECORE)
 
 messages:
===================================================================
diff -u -u -r1.1.1.4 -r1.6
--- ./kioslave/Makefile.am	29 Oct 2002 03:05:23 -0000	1.1.1.4
+++ ./kioslave/Makefile.am	29 Oct 2002 03:39:31 -0000	1.6
@@ -8,10 +8,14 @@
 SMB_SUBDIR=smb
 endif
 
+if include_kioslave_nfs
+NFS_SUBDIR=nfs
+endif
+
 if include_devicesdir
 DEVICES_SUBDIR=devices
 endif
 
-
-SUBDIRS = about cgi floppy filter fish info mac man nfs nntp pop3 imap4 smtp \
-  smbro sftp tar zip finger thumbnail $(LDAP_SUBDIR) $(SMB_SUBDIR) $(DEVICES_SUBDIR)
+SUBDIRS = about cgi floppy filter fish info mac man nntp pop3 imap4 smtp \
+	smbro sftp tar zip finger thumbnail $(LDAP_SUBDIR) $(SMB_SUBDIR) \
+	$(NFS_SUBDIR) $(DEVICES_SUBDIR)
===================================================================
diff -u -u -r1.1.1.2 -r1.2
--- ./kioslave/devices/configure.in.in	29 Oct 2002 03:05:23 -0000	1.1.1.2
+++ ./kioslave/devices/configure.in.in	11 Nov 2002 20:56:20 -0000	1.2
@@ -7,6 +7,9 @@
     Linux)
         DEVICESDIR=devices
         ;;
+    Darwin)
+        DEVICESDIR=devices
+        ;;
     *)
         DEVICESDIR=
         ;;
===================================================================
diff -u -u -r1.1.1.2 Makefile.am
--- ./kioslave/devices/kdedmodule/Makefile.am	29 Oct 2002 03:05:23 -0000	1.1.1.2
+++ ./kioslave/devices/kdedmodule/Makefile.am	24 Nov 2002 07:07:15 -0000
@@ -3,7 +3,7 @@
 INCLUDES = $(all_includes)
 kded_mountwatcher_la_SOURCES = mountwatcher.cpp mountwatcher.skel disklist.cpp disks.cpp
 kded_mountwatcher_la_LDFLAGS = $(all_libraries) -module -avoid-version
-kded_mountwatcher_la_LIBADD = $(LIB_KSYCOCA)
+kded_mountwatcher_la_LIBADD = $(LIB_KSYCOCA) $(LIB_QT) -lDCOP -lkdecore $(LIB_KDED)
 
 METASOURCES = AUTO
 
===================================================================
diff -u -u -r1.1.1.3 -r1.2
--- ./kioslave/fish/fish.cpp	29 Oct 2002 03:05:24 -0000	1.1.1.3
+++ ./kioslave/fish/fish.cpp	29 Oct 2002 18:27:45 -0000	1.2
@@ -103,7 +103,7 @@
 #define sendmimeType(x) mimeType(x)
 #endif
 
-#if defined(__FreeBSD__)
+#if defined(__FreeBSD__) || defined(__APPLE__)
 #define SIGCLD SIGCHLD
 #endif
 
===================================================================
diff -u -u -r1.1.1.2 -r1.5
--- ./kioslave/ldap/configure.in.in	10 Jul 2002 03:23:08 -0000	1.1.1.2
+++ ./kioslave/ldap/configure.in.in	20 Oct 2002 19:12:01 -0000	1.5
@@ -1,97 +1,13 @@
-AC_MSG_CHECKING(whether to compile LDAP support)
+AC_MSG_CHECKING([whether to compile LDAP support])
 AC_ARG_WITH(ldap,
-[  --without-ldap=PATH     Set path for LDAP files [default=check]],
-[ case "$withval" in
-  yes)
-    with_ldap=CHECK
-    ;;
-  esac ],
-[ with_ldap=CHECK ]
-)dnl
+  [  --without-ldap=PATH     Set path for LDAP files [default=check]],
+  [ ac_use_ldap="$withval" ],
+  [ ac_use_ldap=no ]
+)
+AC_MSG_RESULT($ac_use_ldap)
 
-if test "x$with_ldap" = "xCHECK" ; then
-  with_ldap=NOTFOUND
-  search_incs="$kde_includes /usr/include /usr/local/include"
-  AC_FIND_FILE(ldap.h, $search_incs, ldap_incdir)
-  if test -r $ldap_incdir/ldap.h ; then
-    test "x$ldap_incdir" != "x/usr/include" && LDAP_INCS="-I$ldap_incdir"
-    with_ldap=FOUND
-  fi
-  if test $with_ldap = FOUND ; then
-    with_ldap=NOTFOUND
-    for ext in la so sl a ; do
-      AC_FIND_FILE(libldap.$ext, $kde_libraries /usr/lib /usr/local/lib,
-        ldap_libdir)
-      if test -r $ldap_libdir/libldap.$ext ; then
-        if test "x$ldap_libdir" != "x/usr/lib" ; then
-          LDAP_LIBS="-L$ldap_libdir "
-          test "$USE_RPATH" = yes && LDAP_RPATH="-R $ldap_libdir"
-        fi
-        LDAP_LIBS="${LDAP_LIBS}-lldap -llber -lresolv"
-        with_ldap=FOUND
-        break
-      fi
-    done
-  fi
+if test "x$ac_use_ldap" != "no" ; then
+  KDE_CHECK_LDAP($ac_use_ldap)
 fi
 
-case "$with_ldap" in
-no) AC_MSG_RESULT(no) ;;
-NOTFOUND) AC_MSG_RESULT(searched but not found) ;;
-*)
-  if test "x$with_ldap" = "xFOUND" ; then
-    msg="incs=$ldap_incdir libs=$ldap_libdir"
-  else
-    msg="$with_ldap"
-    LDAP_ROOT="$with_ldap"
-    if test "x$LDAP_ROOT" != "x/usr" ; then
-      LDAP_INCS="-I${LDAP_ROOT}/include"
-      LDAP_LIBS="-L${LDAP_ROOT}/lib "
-      if test "$USE_RPATH" = "yes" ; then
-        LDAP_RPATH="-R ${LDAP_ROOT}/lib"
-      fi
-    fi
-    LDAP_LIBS="${LDAP_LIBS}-lldap -llber -lresolv"
-  fi
-  kde_safe_LIBS="$LIBS"
-  kde_safe_CFLAGS="$CFLAGS"
-  LIBS="$LIBS $all_libraries $LDAP_LIBS $KERBEROS_LIBS $X_EXTRA_LIBS"
-  CFLAGS="$CFLAGS $all_includes $LDAP_INCS $KERBEROS_INCS"
-  AC_LANG_SAVE
-  AC_LANG_C
-  AC_TRY_LINK(dnl
-  [
-  #include <lber.h>
-  #include <ldap.h>
-  #if LDAP_VENDOR_VERSION < 20000
-  #error OpenLDAP version too old, please upgrade to version 2 or higher
-  #endif
-  ],
-  [
-  LDAP *ldap;
-  ],
-  , with_ldap=no
-  )
-  AC_LANG_RESTORE
-  CFLAGS=$kde_safe_CFLAGS
-  LIBS=$kde_safe_LIBS
-  if test "$with_ldap" = "no" ; then
-    LDAP_INCS=
-    LDAP_LIBS=
-    LDAP_RPATH=
-    LDAP_SUBDIR=
-    AC_MSG_RESULT(no (but first try gave $msg))
-  else
-    AC_DEFINE_UNQUOTED(HAVE_LIBLDAP, 1, [Define if you have LDAP libraries])
-    LDAP_SUBDIR="ldap"
-    AC_MSG_RESULT($msg)
-  fi
-  ;;
-esac
-
-AC_SUBST(LDAP_INCS)
-AC_SUBST(LDAP_LIBS)
-AC_SUBST(LDAP_RPATH)
-
 AM_CONDITIONAL(include_kioslave_ldap, test -n "$LDAP_SUBDIR")
-
===================================================================
diff -u -u -r1.1.1.1 -r1.2
--- ./kioslave/ldap/Makefile.am	25 Jun 2002 02:03:52 -0000	1.1.1.1
+++ ./kioslave/ldap/Makefile.am	21 Oct 2002 01:34:23 -0000	1.2
@@ -9,7 +9,7 @@
 kde_module_LTLIBRARIES = kio_ldap.la
 
 kio_ldap_la_SOURCES = kio_ldap.cpp kldap.cpp kldapurl.cpp
-kio_ldap_la_LIBADD = $(LIB_KIO) $(LDAP_LIBS) $(KERBEROS_LIBS)
+kio_ldap_la_LIBADD = $(LIB_KIO) $(LDAP_LIBS) $(LIBLDAP) $(KERBEROS_LIBS)
 kio_ldap_la_LDFLAGS = $(all_libraries) $(KDE_RPATH) $(LDAP_RPATH) $(KRB_RPATH) -module $(KDE_PLUGIN)
 
 noinst_HEADERS = kio_ldap.h kldap.h kldapurl.h
===================================================================
diff -u -u -r1.1.1.4 -r1.3
===================================================================
diff -N ./kioslave/nfs/configure.in.in
--- /dev/null	1 Jan 1970 00:00:00 -0000
+++ ./kioslave/nfs/configure.in.in	12 Jul 2002 02:04:22 -0000	1.2
@@ -0,0 +1,7 @@
+AC_MSG_CHECKING(whether to compile NFS support)
+AC_ARG_WITH(nfs,
+[  --without-nfs           Disable NFS kioslave [default=no]],
+	with_nfs=$withval, with_nfs=yes
+)dnl
+AC_MSG_RESULT($withval)
+AM_CONDITIONAL(include_kioslave_nfs, test "$with_nfs" = "yes")
===================================================================
diff -u -u -r1.1.1.1 -r1.4
--- ./kioslave/nfs/README	25 Jun 2002 02:03:52 -0000	1.1.1.1
+++ ./kioslave/nfs/README	16 Sep 2002 14:52:42 -0000	1.4
@@ -1,3 +1,29 @@
 this is an ioslave for KDE 2 for NFS, version 2.
 
 Alex
+
+Darwin notes --
+
+It looks like there are C++ type issues with the Darwin headers
+(try building, you'll see the "too many arguments to function" stuff)
+
+The problem seems to be <rpc/clnt.h> having definitions that don't
+play nice with C++.  The client rpc handle does a bunch of void stuff
+in the struct which is fine in C but makes an empty arg list in C++.
+
+typedef struct {
+        AUTH    *cl_auth;                       /* authenticator */
+        struct clnt_ops {
+                enum clnt_stat  (*cl_call)();   /* call remote procedure */
+                void            (*cl_abort)();  /* abort a call */
+                void            (*cl_geterr)(); /* get specific error code */
+                bool_t          (*cl_freeres)(); /* frees results */
+                void            (*cl_destroy)();/* destroy this structure */
+                bool_t          (*cl_control)();/* the ioctl() of rpc */
+        } *cl_ops;
+        caddr_t                 cl_private;     /* private stuff */
+} CLIENT;
+
+
+The only solution is to probably reimplement clnt.h locally to fix
+these (wrapping the "#include <rpc/rpc.h>" in an extern "C" isn't enough).
===================================================================
diff -u -u -r1.1.1.4 Makefile.am
--- ./klipper/Makefile.am	12 Nov 2002 04:19:29 -0000	1.1.1.4
+++ ./klipper/Makefile.am	24 Nov 2002 07:07:16 -0000
@@ -6,12 +6,12 @@
 
 CLEANFILES = dummy.cpp
 
-klipper_la_LIBADD = $(LIB_KDEUI) $(LIB_KSYCOCA) 
+klipper_la_LIBADD = $(LIB_KDEUI) $(LIB_KSYCOCA) $(LIB_QT) -lkdecore -lDCOP
 klipper_la_SOURCES = main.cpp toplevel.cpp urlgrabber.cpp configdialog.cpp \
     toplevel.skel
 
 klipper_la_METASOURCES = AUTO
-klipper_la_LDFLAGS = $(all_libraries) -module -avoid-version
+klipper_la_LDFLAGS = $(all_libraries) -avoid-version
 
 klipper_LDADD = klipper.la $(LIB_KSYCOCA)
 klipper_SOURCES = dummy.cpp
===================================================================
diff -u -u -r1.1.1.3 Makefile.am
--- ./kmenuedit/Makefile.am	14 Aug 2002 14:27:55 -0000	1.1.1.3
+++ ./kmenuedit/Makefile.am	24 Nov 2002 07:07:16 -0000
@@ -7,11 +7,10 @@
 
 kmenuedit_la_SOURCES = basictab.cpp treeview.cpp desktopfileeditor.cpp menueditview.cpp kmenuedit.cpp main.cpp khotkeys.cpp
 
-kmenuedit_la_LIBADD = $(LIB_KDEUI) $(LIB_KIO)
-kmenuedit_la_LDFLAGS = $(all_libraries) -module -avoid-version
+kmenuedit_la_LIBADD = $(LIB_KDEUI) $(LIB_KIO) $(LIB_QT) -lkdecore -lDCOP
+kmenuedit_la_LDFLAGS = $(all_libraries) -avoid-version
 
 kmenuedit_SOURCES = dummy.cpp
-kmenuedit_LDADD   = kmenuedit.la
 kmenuedit_LDFLAGS = $(all_libraries) $(KDE_RPATH)
 
 noinst_HEADERS = kmenuedit.h menueditview.h treeview.h desktopfileeditor.h basictab.h khotkeys.h
===================================================================
diff -u -u -r1.1.1.4 Makefile.am
--- ./konqueror/Makefile.am	29 Oct 2002 03:05:38 -0000	1.1.1.4
+++ ./konqueror/Makefile.am	24 Nov 2002 07:07:16 -0000
@@ -43,7 +43,7 @@
 	konq_misc.h konq_openurlrequest.h konq_profiledlg.h konq_run.h \
 	konq_view.h konq_viewmgr.h version.h
 
-konqueror_la_LDFLAGS = $(all_libraries) -module -avoid-version
+konqueror_la_LDFLAGS = $(all_libraries) -avoid-version
 konqueror_la_LIBADD = ../libkonq/libkonq.la
 
 konqueror_SOURCES = dummy.cc
===================================================================
diff -u -u -r1.1.1.2 Makefile.am
--- ./konqueror/client/Makefile.am	14 Aug 2002 14:30:59 -0000	1.1.1.2
+++ ./konqueror/client/Makefile.am	24 Nov 2002 07:07:16 -0000
@@ -4,8 +4,8 @@
 lib_LTLIBRARIES = kfmclient.la
 
 INCLUDES= -I.. $(all_includes)
-kfmclient_la_LDFLAGS = $(all_libraries) -module -avoid-version
-kfmclient_la_LIBADD =  $(LIB_KIO)
+kfmclient_la_LDFLAGS = $(all_libraries) -avoid-version
+kfmclient_la_LIBADD =  $(LIB_KIO) $(LIB_QT) -lkdecore -lDCOP -lkdeui
 kfmclient_la_SOURCES = kfmclient.cc KonquerorIface.stub KDesktopIface.stub
 
 noinst_HEADERS = kfmclient.h
===================================================================
diff -u -u -r1.1.1.3 Makefile.am
--- ./konqueror/keditbookmarks/Makefile.am	29 Oct 2002 03:05:50 -0000	1.1.1.3
+++ ./konqueror/keditbookmarks/Makefile.am	24 Nov 2002 07:07:16 -0000
@@ -8,7 +8,7 @@
 kbookmarklistener_DIR = $(top_srcdir)/libkonq
 
 keditbookmarks_la_LIBADD  = $(top_builddir)/libkonq/libkonq.la $(LIB_KHTML)
-keditbookmarks_la_LDFLAGS = $(all_libraries) -module -avoid-version
+keditbookmarks_la_LDFLAGS = $(all_libraries) -avoid-version
 
 bin_PROGRAMS = keditbookmarks
 
===================================================================
diff -u -u -r1.1.1.2 Makefile.am
--- ./konqueror/sidebar/trees/Makefile.am	10 Jul 2002 03:24:11 -0000	1.1.1.2
+++ ./konqueror/sidebar/trees/Makefile.am	24 Nov 2002 07:07:16 -0000
@@ -3,15 +3,15 @@
 
 INCLUDES = -I$(top_srcdir)/libkonq -I$(top_srcdir)/konqueror -I$(srcdir)/.. $(all_includes)
 
-lib_LTLIBRARIES = libkonq_sidebar_tree.la
+lib_LTLIBRARIES = konq_sidebar_tree.la
 kde_module_LTLIBRARIES = konqsidebar_tree.la
 METASOURCES = AUTO
 
-libkonq_sidebar_tree_la_SOURCES = konq_sidebartree.cpp konq_sidebartreeitem.cpp konq_sidebartreetoplevelitem.cpp
-libkonq_sidebar_tree_la_LDFLAGS = $(all_libraries) -module -avoid-version -no-undefined
-libkonq_sidebar_tree_la_LIBADD = $(top_builddir)/libkonq/libkonq.la ../libkonqsidebarplugin.la
+konq_sidebar_tree_la_SOURCES = konq_sidebartree.cpp konq_sidebartreeitem.cpp konq_sidebartreetoplevelitem.cpp
+konq_sidebar_tree_la_LDFLAGS = $(all_libraries) -avoid-version -no-undefined
+konq_sidebar_tree_la_LIBADD = $(top_builddir)/libkonq/libkonq.la $(top_builddir)/konqueror/sidebar/libkonqsidebarplugin.la
 
 konqsidebar_tree_la_SOURCES = konqsidebar_tree.cpp
 konqsidebar_tree_la_LDFLAGS = $(all_libraries) -module -avoid-version -no-undefined
-konqsidebar_tree_la_LIBADD = $(top_builddir)/libkonq/libkonq.la ../libkonqsidebarplugin.la libkonq_sidebar_tree.la
+konqsidebar_tree_la_LIBADD = $(top_builddir)/libkonq/libkonq.la $(top_builddir)/konqueror/sidebar/libkonqsidebarplugin.la konq_sidebar_tree.la
 
===================================================================
diff -u -u -r1.1.1.1 Makefile.am
--- ./konqueror/sidebar/trees/bookmark_module/Makefile.am	25 Jun 2002 02:03:53 -0000	1.1.1.1
+++ ./konqueror/sidebar/trees/bookmark_module/Makefile.am	24 Nov 2002 07:07:16 -0000
@@ -6,5 +6,5 @@
 
 konq_sidebartree_bookmarks_la_SOURCES = bookmark_module.cpp bookmark_item.cpp
 konq_sidebartree_bookmarks_la_LDFLAGS = $(all_libraries) -module $(KDE_PLUGIN)
-konq_sidebartree_bookmarks_la_LIBADD = ../../libkonqsidebarplugin.la ../libkonq_sidebar_tree.la
+konq_sidebartree_bookmarks_la_LIBADD = $(top_builddir)/konqueror/sidebar/libkonqsidebarplugin.la $(top_builddir)/konqueror/sidebar/trees/konq_sidebar_tree.la
 
===================================================================
diff -u -u -r1.1.1.1 Makefile.am
--- ./konqueror/sidebar/trees/dirtree_module/Makefile.am	25 Jun 2002 02:03:53 -0000	1.1.1.1
+++ ./konqueror/sidebar/trees/dirtree_module/Makefile.am	24 Nov 2002 07:07:16 -0000
@@ -5,6 +5,6 @@
 METASOURCES = AUTO
 konq_sidebartree_dirtree_la_SOURCES = dirtree_module.cpp dirtree_item.cpp
 konq_sidebartree_dirtree_la_LDFLAGS = $(all_libraries) -module -avoid-version -no-undefined
-konq_sidebartree_dirtree_la_LIBADD = ../../libkonqsidebarplugin.la ../libkonq_sidebar_tree.la
+konq_sidebartree_dirtree_la_LIBADD = $(top_builddir)/konqueror/sidebar/libkonqsidebarplugin.la $(top_builddir)/konqueror/sidebar/trees/konq_sidebar_tree.la
 
 
===================================================================
diff -u -u -r1.1.1.2 -r1.3
--- ./konqueror/sidebar/trees/history_module/history_module.h	14 Aug 2002 14:30:58 -0000	1.1.1.2
+++ ./konqueror/sidebar/trees/history_module/history_module.h	14 Aug 2002 15:35:44 -0000	1.3
@@ -53,8 +53,7 @@
     bool sortsByName() const { return m_sortsByName; }
 
     static QString groupForURL( const KURL& url ) {
-	static const QString& misc = KGlobal::staticQString(i18n("Miscellaneous"));
-	return url.host().isEmpty() ? misc : url.host();
+	return url.host().isEmpty() ? KGlobal::staticQString(i18n("Miscellaneous")) : url.host();
     }
 
 public slots:
===================================================================
diff -u -u -r1.1.1.3 Makefile.am
--- ./konqueror/sidebar/trees/history_module/Makefile.am	1 Oct 2002 02:36:50 -0000	1.1.1.3
+++ ./konqueror/sidebar/trees/history_module/Makefile.am	24 Nov 2002 07:07:16 -0000
@@ -7,7 +7,7 @@
 konq_sidebartree_history_la_SOURCES = history_module.cpp history_item.cpp \
 	history_settings.cpp history_settings.skel
 konq_sidebartree_history_la_LDFLAGS = $(all_libraries) -module $(KDE_PLUGIN)
-konq_sidebartree_history_la_LIBADD =  ../../libkonqsidebarplugin.la ../libkonq_sidebar_tree.la
+konq_sidebartree_history_la_LIBADD =  $(top_builddir)/konqueror/sidebar/libkonqsidebarplugin.la $(top_builddir)/konqueror/sidebar/trees/konq_sidebar_tree.la
 
 kcm_history_la_SOURCES = kcmhistory.cpp history_dlg.ui history_settings.cpp history_settings.skel
 kcm_history_la_LDFLAGS = $(all_libraries) -module -avoid-version -no-undefined
===================================================================
diff -u -u -r1.1.1.8 -r1.6
===================================================================
diff -u -u -r1.1.1.8 -r1.6
===================================================================
diff -u -u -r1.1.1.6 -r1.7
===================================================================
diff -u -u -r1.1.1.8 -r1.6
===================================================================
diff -u -u -r1.1.1.8 -r1.6
===================================================================
diff -u -u -r1.1.1.7 -r1.6
===================================================================
diff -u -u -r1.1.1.7 -r1.6
===================================================================
diff -u -u -r1.1.1.6 -r1.4
===================================================================
diff -u -u -r1.1.1.4 -r1.4
===================================================================
diff -u -u -r1.1.1.3 -r1.4
===================================================================
diff -u -u -r1.1.1.3 -r1.3
--- ./konsole/konsole/konsole_grantpty.c	1 Oct 2002 02:36:59 -0000	1.1.1.3
+++ ./konsole/konsole/konsole_grantpty.c	22 Oct 2002 03:05:32 -0000	1.3
@@ -45,6 +45,13 @@
 
 int main (int argc, char *argv[])
 {
+
+  if (strcmp(getenv("DEBUG_ME"), "sleepinmain") == 0)
+  {
+    fprintf(stderr, "My PID is %d\n", (int) getpid());
+    kill(getpid(), SIGSTOP);
+  }
+
   char*         pty;
   struct stat   st;
   struct group* p;
===================================================================
diff -u -u -r1.1.1.6 -r1.5
===================================================================
diff -u -u -r1.1.1.6 -r1.5
===================================================================
diff -u -u -r1.1.1.5 -r1.3
===================================================================
diff -u -u -r1.1.1.4 Makefile.am
--- ./konsole/konsole/Makefile.am	29 Oct 2002 03:06:19 -0000	1.1.1.4
+++ ./konsole/konsole/Makefile.am	24 Nov 2002 07:07:16 -0000
@@ -29,17 +29,17 @@
 	konsoleiface.skel \
         sessioniface.skel
 
-libkonsolepart_la_LDFLAGS = $(all_libraries) $(KDE_PLUGIN)
-libkonsolepart_la_LIBADD = $(LIBUTIL) $(XTESTLIB) $(LIB_KPARTS)
+libkonsolepart_la_LDFLAGS = $(all_libraries) $(KDE_RPATH)
+libkonsolepart_la_LIBADD = $(LIBUTIL) $(XTESTLIB) $(LIB_KPARTS) $(LIB_QT) -lkdecore -lDCOP -lkdeui -lkio
 
 # kwrited kdeinit module
 kwrited_la_SOURCES = kwrited.cpp
-kwrited_la_LIBADD  = libkonsolepart.la
-kwrited_la_LDFLAGS = $(all_libraries) -module -avoid-version
+kwrited_la_LDFLAGS = $(all_libraries) -avoid-version
+kwrited_la_LIBADD = $(LIB_KPARTS) $(LIB_QT) -lkdecore -lDCOP libkonsolepart.la
 
 # kwrited executable
 kwrited_SOURCES	= dummy.cpp
-kwrited_LDADD   = kwrited.la 
+kwrited_LDADD = kwrited.la
 kwrited_LDFLAGS = $(all_libraries) $(KDE_RPATH)
 
 
@@ -47,8 +47,8 @@
 konsole_la_SOURCES = TEPty.cpp BlockArray.cpp main.cpp konsole.cpp schema.cpp session.cpp TEWidget.cpp TEmuVt102.cpp \
      TEScreen.cpp TEmulation.cpp TEHistory.cpp keytrans.cpp konsoleiface.skel sessioniface.skel \
      konsole_wcwidth.cpp konsolebookmarkhandler.cpp konsolebookmarkmenu.cpp konsole_child.cpp
-konsole_la_LDFLAGS = $(all_libraries) -module -avoid-version
-konsole_la_LIBADD = $(LIB_KDEUI) $(LIB_KIO) $(LIBUTIL) $(XTESTLIB)
+konsole_la_LDFLAGS = $(all_libraries) -avoid-version
+konsole_la_LIBADD = $(LIB_KDEUI) $(LIB_KIO) $(LIBUTIL) $(XTESTLIB) $(LIB_QT) -lkdecore -lDCOP
 
 # konsole executable
 konsole_SOURCES = dummy.cpp
===================================================================
diff -u -u -r1.1.1.4 -r1.3
===================================================================
diff -u -u -r1.1.1.4 -r1.3
===================================================================
diff -u -u -r1.1.1.5 -r1.4
===================================================================
diff -u -u -r1.1.1.5 -r1.4
===================================================================
diff -u -u -r1.1.1.5 -r1.8
===================================================================
diff -u -u -r1.1.1.4 -r1.3
===================================================================
diff -u -u -r1.1.1.3 -r1.3
===================================================================
diff -u -u -r1.1.1.3 -r1.3
===================================================================
diff -u -u -r1.1.1.5 -r1.4
===================================================================
diff -u -u -r1.1.1.5 -r1.4
===================================================================
diff -u -u -r1.1.1.8 -r1.6
===================================================================
diff -u -u -r1.1.1.8 -r1.6
===================================================================
diff -u -u -r1.1.1.7 -r1.5
===================================================================
diff -u -u -r1.1.1.8 -r1.6
===================================================================
diff -u -u -r1.1.1.8 -r1.6
===================================================================
diff -u -u -r1.1.1.8 -r1.6
===================================================================
diff -u -u -r1.1.1.8 -r1.6
===================================================================
diff -u -u -r1.1.1.7 -r1.5
===================================================================
diff -u -u -r1.1.1.2 Makefile.am
--- ./ksmserver/Makefile.am	10 Jul 2002 03:24:29 -0000	1.1.1.2
+++ ./ksmserver/Makefile.am	24 Nov 2002 07:07:36 -0000
@@ -28,9 +28,9 @@
 ksmserver_la_SOURCES = main.cpp server.cpp shutdown.cpp KSMServerInterface.skel
 ksmserver_SOURCES = dummy.cpp
 
-ksmserver_la_LDFLAGS = $(all_libraries) -avoid-version -module
+ksmserver_la_LDFLAGS = $(all_libraries) -avoid-version $(QT_LDFLAGS)
 ksmserver_LDFLAGS = $(all_libraries) $(KDE_RPATH)
-ksmserver_la_LIBADD = $(LIB_KDEUI)
+ksmserver_la_LIBADD = $(LIB_KDEUI) $(LIB_QT) -lkdecore -lDCOP
 ksmserver_LDADD = ksmserver.la
 
 picsdir = $(kde_datadir)/ksmserver/pics
===================================================================
diff -u -u -r1.1.1.4 configure.in.in
--- ./ksysguard/configure.in.in	29 Oct 2002 03:06:45 -0000	1.1.1.4
+++ ./ksysguard/configure.in.in	24 Nov 2002 07:07:36 -0000
@@ -1,3 +1,5 @@
+DO_NOT_COMPILE="$DO_NOT_COMPILE ksysguard"
+
 dnl Check whether ksysguardd can run; 
 AC_MSG_CHECKING([if ksysguardd can be compiled])
 case "$host" in 
===================================================================
diff -u -u -r1.1.1.2 -r1.3
===================================================================
diff -u -u -r1.1.1.1 -r1.2
--- ./ksysguard/gui/ksgrd/StyleEngine.h	25 Jun 2002 02:03:54 -0000	1.1.1.1
+++ ./ksysguard/gui/ksgrd/StyleEngine.h	16 Sep 2002 14:31:34 -0000	1.2
@@ -68,11 +68,12 @@
 	}
 	const QColor& getSensorColor(uint i)
 	{
-		static QColor dummy;
-		if (i < sensorColors.count())
+		if (i < sensorColors.count()) {
 			return *(sensorColors.at(i));
-		else
+		} else {
+			QColor dummy;
 			return dummy;
+		}
 	}
 	uint getSensorColorCount() const
 	{
===================================================================
diff -u -u -r1.1.1.3 -r1.3
===================================================================
diff -u -u -r1.1.1.4 -r1.3
--- ./ksysguard/gui/SensorDisplayLib/SensorDisplay.cc	1 Oct 2002 02:37:23 -0000	1.1.1.4
+++ ./ksysguard/gui/SensorDisplayLib/SensorDisplay.cc	1 Oct 2002 03:01:48 -0000	1.3
@@ -62,7 +62,6 @@
 	pausedWhileHidden = false;
 
 	timerOn();
-	QWhatsThis::add(this, "dummy");
 
 	frame = new QGroupBox(2, Qt::Vertical, "", this, "displayFrame"); 
 	Q_CHECK_PTR(frame);
===================================================================
diff -u -u -r1.1.1.5 -r1.2
--- ./ktip/tips	29 Oct 2002 03:06:51 -0000	1.1.1.5
+++ ./ktip/tips	31 Oct 2002 19:19:37 -0000	1.2
@@ -704,20 +704,6 @@
 </tip>
 
 
-<tip category="Win2Unix">
-<html>
-<p>
-You might wonder why there are very few (if any) files whose
-names end in <code>.exe</code> or <code>.bat</code> on UNIX
-systems. This is because filenames on UNIX do not need an
-extension. Executable files in KDE are represented by the gear icon
-in Konqueror. In the Konsole window, they are often colored red 
-(depending on your settings).
-</p>
-<p align="right"><em>Contributed by Carsten Niehaus</em></p>
-</html>
-</tip>
-
 <tip category="KDE|Win2Unix|General">
 <html>
 <p>
===================================================================
diff -u -u -r1.1.1.2 Makefile.am
--- ./kwin/Makefile.am	10 Jul 2002 03:24:51 -0000	1.1.1.2
+++ ./kwin/Makefile.am	24 Nov 2002 07:07:37 -0000
@@ -9,8 +9,8 @@
 kwin_la_SOURCES = workspace.cpp placement.cpp atoms.cpp  client.cpp main.cpp \
 	popupinfo.cpp tabbox.cpp options.cpp plugins.cpp events.cpp KWinInterface.skel \
 	killwindow.cpp kwinbutton.cpp
-kwin_la_LIBADD = $(LIB_KDEUI) $(LIBXINERAMA)
-kwin_la_LDFLAGS = $(all_libraries) -module -avoid-version
+kwin_la_LIBADD = $(LIB_KDEUI) $(LIBXINERAMA) $(LIB_QT) -lkdecore -lDCOP
+kwin_la_LDFLAGS = $(all_libraries) -avoid-version $(QT_LDFLAGS)
 
 include_HEADERS = KWinInterface.h
 
===================================================================
diff -u -u -r1.1.1.7 -r1.7
===================================================================
diff -u -u -r1.1.1.2 -r1.3
===================================================================
diff -u -u -r1.1.1.4 -r1.2
--- ./kwin/kcmkwin/kwinoptions/Makefile.am	1 Oct 2002 02:37:36 -0000	1.1.1.4
+++ ./kwin/kcmkwin/kwinoptions/Makefile.am	1 Oct 2002 12:06:42 -0000	1.2
@@ -10,7 +10,7 @@
 kcm_kwinoptions_la_LDFLAGS = -module -avoid-version $(all_libraries) -no-undefined
 kcm_kwinoptions_la_LIBADD = $(LIB_KDEUI)
 
-noinst_HEADERS = windows.h mouse.h 
+noinst_HEADERS = windows.h mouse.h main.h
 
 messages:
 	$(XGETTEXT) *.cpp -o $(podir)/kcmkwm.pot
===================================================================
diff -u -u -r1.1.1.5 Makefile.am
--- ./kxkb/Makefile.am	29 Oct 2002 03:07:04 -0000	1.1.1.5
+++ ./kxkb/Makefile.am	24 Nov 2002 07:07:37 -0000
@@ -19,8 +19,8 @@
 kcm_keyboard_la_LIBADD = $(XTESTLIB) $(LIB_KIO) -lxkbfile
 
 kxkb_la_SOURCES = kxkb.cpp rules.cpp extension.cpp pixmap.cpp
-kxkb_la_LDFLAGS = $(all_libraries) -module -avoid-version
-kxkb_la_LIBADD = -lX11 -lXext $(LIB_KDEUI) -lxkbfile
+kxkb_la_LDFLAGS = $(all_libraries) -avoid-version
+kxkb_la_LIBADD = -lX11 -lXext $(LIB_KDEUI) -lxkbfile $(LIB_QT) -lDCOP -lkdecore
 
 data_DATA = keyboard.desktop
 datadir = $(kde_appsdir)/Settings/Peripherals
===================================================================
diff -u -u -r1.1.1.2 -r1.4
===================================================================
diff -u -u -r1.1.1.6 -r1.4
===================================================================
diff -u -u -r1.1.1.5 -r1.3
===================================================================
diff -u -u -r1.1.1.4 -r1.6
===================================================================
diff -u -u -r1.1.1.1 Makefile.am
--- ./libkonq/favicons/Makefile.am	25 Jun 2002 02:03:55 -0000	1.1.1.1
+++ ./libkonq/favicons/Makefile.am	24 Nov 2002 07:07:40 -0000
@@ -2,7 +2,7 @@
 
 INCLUDES = $(all_includes)
 kded_favicons_la_SOURCES = favicons.cpp favicons.skel
-kded_favicons_la_LDFLAGS = $(all_libraries) -module -avoid-version
+kded_favicons_la_LDFLAGS = $(all_libraries) -module -avoid-version $(QT_LDFLAGS) -undefined suppress -flat_namespace
 kded_favicons_la_LIBADD = $(LIB_KSYCOCA)
 
 METASOURCES = AUTO
===================================================================
diff -u -u -r1.1.1.3 -r1.4
--- ./nsplugins/pluginscan.cpp	29 Oct 2002 03:07:09 -0000	1.1.1.3
+++ ./nsplugins/pluginscan.cpp	29 Oct 2002 03:39:40 -0000	1.4
@@ -443,6 +443,7 @@
         paths.append("/usr/lib/netscape/plugins");
         paths.append("/usr/lib/netscape/plugins-libc5");
         paths.append("/usr/lib/netscape/plugins-libc6");
+        paths.append("@PREFIX@/lib/mozilla/plugins");
         paths.append("/usr/lib/mozilla/plugins");
         paths.append("$MOZILLA_HOME/plugins");
         config->writeEntry( "scanPaths", paths );
