
--- debian/kasteroids.install	5 Nov 2002 02:15:00 -0000	1.1.1.3
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,10 +0,0 @@
-debian/tmp/usr/games/kasteroids
-debian/tmp/usr/share/man/man6/kasteroids*
-debian/tmp/usr/share/icons/crystalsvg/48x48/apps/kasteroids.png
-debian/tmp/usr/share/icons/crystalsvg/32x32/apps/kasteroids.png
-debian/tmp/usr/share/icons/crystalsvg/16x16/apps/kasteroids.png
-debian/tmp/usr/share/applnk/Games/Arcade/kasteroids.desktop
-debian/tmp/usr/share/apps/kasteroids/*
-debian/tmp/usr/share/doc/kdegames/kasteroids/*
-debian/tmp/usr/share/doc/kde/HTML/en/kasteroids/*
-debian/icons/kasteroids.xpm /usr/share/pixmaps
--- debian/katomic.install	5 Nov 2002 02:15:00 -0000	1.1.1.3
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,8 +0,0 @@
-debian/tmp/usr/share/apps/katomic/*
-debian/tmp/usr/share/icons/crystalsvg/32x32/apps/katomic.png
-debian/tmp/usr/share/applnk/Games/TacticStrategy/katomic.desktop
-debian/tmp/usr/games/katomic
-debian/tmp/usr/share/man/man6/katomic*
-debian/tmp/usr/share/doc/kdegames/katomic/*
-debian/tmp/usr/share/doc/kde/HTML/en/katomic/*
-debian/icons/katomic.xpm /usr/share/pixmaps
--- debian/kbackgammon.install	4 Jan 2003 18:36:36 -0000	1.1.1.4
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,12 +0,0 @@
-debian/tmp/usr/games/kbackgammon
-debian/tmp/usr/share/man/man6/kbackgammon*
-debian/tmp/usr/share/applnk/Games/Board/kbackgammon.desktop
-debian/tmp/usr/share/apps/kbackgammon/*
-debian/tmp/usr/share/icons/crystalsvg/48x48/apps/kbackgammon.png
-debian/tmp/usr/share/icons/crystalsvg/32x32/apps/kbackgammon.png
-debian/tmp/usr/share/icons/crystalsvg/16x16/apps/kbackgammon.png
-debian/tmp/usr/share/icons/crystalsvg/48x48/apps/kbackgammon_engine.png
-debian/tmp/usr/share/icons/crystalsvg/32x32/apps/kbackgammon_engine.png
-debian/tmp/usr/share/icons/crystalsvg/16x16/apps/kbackgammon_engine.png
-debian/tmp/usr/share/doc/kde/HTML/en/kbackgammon/*
-debian/icons/kbackgammon.xpm /usr/share/pixmaps
--- debian/kbattleship.install	5 Nov 2002 02:15:00 -0000	1.1.1.3
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,7 +0,0 @@
-debian/tmp/usr/share/apps/kbattleship/*
-debian/tmp/usr/share/icons/crystalsvg/32x32/apps/kbattleship.png
-debian/tmp/usr/share/icons/crystalsvg/16x16/apps/kbattleship.png
-debian/tmp/usr/share/applnk/Games/Board/kbattleship.desktop
-debian/tmp/usr/games/kbattleship
-debian/tmp/usr/share/doc/kde/HTML/en/kbattleship/*
-debian/icons/kbattleship.xpm /usr/share/pixmaps
--- debian/kblackbox.install	5 Nov 2002 02:15:00 -0000	1.1.1.3
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,10 +0,0 @@
-debian/tmp/usr/share/apps/kblackbox/*
-debian/tmp/usr/share/icons/crystalsvg/48x48/apps/kblackbox.png
-debian/tmp/usr/share/icons/crystalsvg/32x32/apps/kblackbox.png
-debian/tmp/usr/share/icons/crystalsvg/16x16/apps/kblackbox.png
-debian/tmp/usr/share/applnk/Games/Board/kblackbox.desktop
-debian/tmp/usr/games/kblackbox
-debian/tmp/usr/share/man/man6/kblackbox*
-debian/tmp/usr/share/doc/kdegames/kblackbox/*
-debian/tmp/usr/share/doc/kde/HTML/en/kblackbox/*
-debian/icons/kblackbox.xpm /usr/share/pixmaps
--- debian/kbounce.install	5 Nov 2002 02:15:00 -0000	1.1.1.3
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,10 +0,0 @@
-debian/tmp/usr/share/apps/kbounce/*
-debian/tmp/usr/share/applnk/Games/Arcade/kbounce.desktop
-debian/tmp/usr/share/icons/locolor/16x16/apps/kbounce.png
-debian/tmp/usr/share/icons/locolor/32x32/apps/kbounce.png
-debian/tmp/usr/share/icons/crystalsvg/16x16/apps/kbounce.png
-debian/tmp/usr/share/icons/crystalsvg/32x32/apps/kbounce.png
-debian/tmp/usr/games/kbounce
-debian/tmp/usr/share/man/man6/kbounce*
-debian/tmp/usr/share/doc/kde/HTML/en/kbounce/*
-debian/icons/kbounce.xpm /usr/share/pixmaps
--- debian/kdecarddecks.install	25 Jun 2002 02:34:24 -0000	1.1.1.1
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1 +0,0 @@
-debian/tmp/usr/share/apps/carddecks/*
--- debian/kenolaba.install	5 Nov 2002 02:15:00 -0000	1.1.1.3
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,8 +0,0 @@
-debian/tmp/usr/share/apps/kenolaba/*
-debian/tmp/usr/share/icons/crystalsvg/16x16/apps/kenolaba.png
-debian/tmp/usr/share/icons/crystalsvg/32x32/apps/kenolaba.png
-debian/tmp/usr/share/icons/crystalsvg/48x48/apps/kenolaba.png
-debian/tmp/usr/share/applnk/Games/Board/kenolaba.desktop
-debian/tmp/usr/games/kenolaba
-debian/tmp/usr/share/doc/kde/HTML/en/kenolaba/*
-debian/icons/kenolaba.xpm /usr/share/pixmaps
--- debian/kjumpingcube.install	5 Nov 2002 02:15:00 -0000	1.1.1.3
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,9 +0,0 @@
-debian/tmp/usr/games/kjumpingcube
-debian/tmp/usr/share/icons/crystalsvg/32x32/apps/kjumpingcube.png
-debian/tmp/usr/share/icons/crystalsvg/16x16/apps/kjumpingcube.png
-debian/tmp/usr/share/man/man6/kjumpingcube*
-debian/tmp/usr/share/apps/kjumpingcube/*
-debian/tmp/usr/share/applnk/Games/TacticStrategy/kjumpingcube.desktop
-debian/tmp/usr/share/doc/kdegames/kjumpingcube/*
-debian/tmp/usr/share/doc/kde/HTML/en/kjumpingcube/*
-debian/icons/kjumpingcube.xpm /usr/share/pixmaps
--- debian/klines.install	5 Nov 2002 02:15:00 -0000	1.1.1.3
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,7 +0,0 @@
-debian/tmp/usr/share/apps/klines/*
-debian/tmp/usr/share/applnk/Games/TacticStrategy/klines.desktop
-debian/tmp/usr/games/klines
-debian/tmp/usr/share/doc/kde/HTML/en/klines/*
-debian/tmp/usr/share/icons/crystalsvg/16x16/apps/klines.png
-debian/tmp/usr/share/icons/crystalsvg/32x32/apps/klines.png
-debian/icons/klines.xpm /usr/share/pixmaps
--- debian/kmahjongg.install	5 Nov 2002 02:15:00 -0000	1.1.1.3
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,10 +0,0 @@
-debian/tmp/usr/share/apps/kmahjongg/*
-debian/tmp/usr/share/icons/crystalsvg/48x48/apps/kmahjongg.png
-debian/tmp/usr/share/icons/crystalsvg/32x32/apps/kmahjongg.png
-debian/tmp/usr/share/icons/crystalsvg/16x16/apps/kmahjongg.png
-debian/tmp/usr/share/applnk/Games/Board/kmahjongg.desktop
-debian/tmp/usr/games/kmahjongg
-debian/tmp/usr/share/man/man6/kmahjongg*
-debian/tmp/usr/share/doc/kdegames/kmahjongg/*
-debian/tmp/usr/share/doc/kde/HTML/en/kmahjongg/*
-debian/icons/kmahjongg.xpm /usr/share/pixmaps
--- debian/kmines.install	5 Nov 2002 02:15:00 -0000	1.1.1.3
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,10 +0,0 @@
-debian/tmp/usr/share/icons/crystalsvg/48x48/apps/kmines.png
-debian/tmp/usr/share/icons/crystalsvg/32x32/apps/kmines.png
-debian/tmp/usr/share/icons/crystalsvg/16x16/apps/kmines.png
-debian/tmp/usr/share/applnk/Games/TacticStrategy/kmines.desktop
-debian/tmp/usr/share/apps/kmines/*
-debian/tmp/usr/games/kmines
-debian/tmp/usr/share/man/man6/kmines*
-debian/tmp/usr/share/doc/kdegames/kmines/*
-debian/tmp/usr/share/doc/kde/HTML/en/kmines/*
-debian/icons/kmines.xpm /usr/share/pixmaps
--- debian/konquest.install	5 Nov 2002 02:15:00 -0000	1.1.1.3
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,10 +0,0 @@
-debian/tmp/usr/share/apps/konquest/*
-debian/tmp/usr/share/icons/crystalsvg/48x48/apps/konquest.png
-debian/tmp/usr/share/icons/crystalsvg/32x32/apps/konquest.png
-debian/tmp/usr/share/icons/crystalsvg/16x16/apps/konquest.png
-debian/tmp/usr/share/applnk/Games/TacticStrategy/konquest.desktop
-debian/tmp/usr/games/konquest
-debian/tmp/usr/share/man/man6/konquest*
-debian/tmp/usr/share/doc/kdegames/konquest/*
-debian/tmp/usr/share/doc/kde/HTML/en/konquest/*
-debian/icons/konquest.xpm /usr/share/pixmaps
--- debian/kpat.install	5 Nov 2002 02:15:00 -0000	1.1.1.3
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,10 +0,0 @@
-debian/tmp/usr/share/icons/crystalsvg/48x48/apps/kpat.png
-debian/tmp/usr/share/icons/crystalsvg/32x32/apps/kpat.png
-debian/tmp/usr/share/icons/crystalsvg/16x16/apps/kpat.png
-debian/tmp/usr/share/apps/kpat/*
-debian/tmp/usr/share/applnk/Games/Card/kpat.desktop
-debian/tmp/usr/games/kpat
-debian/tmp/usr/share/man/man6/kpat*
-debian/tmp/usr/share/doc/kdegames/kpat/*
-debian/tmp/usr/share/doc/kde/HTML/en/kpat/*
-debian/icons/kpat.xpm /usr/share/pixmaps
--- debian/kpoker.install	5 Nov 2002 02:15:00 -0000	1.1.1.3
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,10 +0,0 @@
-debian/tmp/usr/share/apps/kpoker/*
-debian/tmp/usr/share/icons/crystalsvg/48x48/apps/kpoker.png
-debian/tmp/usr/share/icons/crystalsvg/32x32/apps/kpoker.png
-debian/tmp/usr/share/icons/crystalsvg/16x16/apps/kpoker.png
-debian/tmp/usr/share/applnk/Games/Card/kpoker.desktop
-debian/tmp/usr/games/kpoker
-debian/tmp/usr/share/man/man6/kpoker*
-debian/tmp/usr/share/doc/kdegames/kpoker/*
-debian/tmp/usr/share/doc/kde/HTML/en/kpoker/*
-debian/icons/kpoker.xpm /usr/share/pixmaps
--- debian/kreversi.install	5 Nov 2002 02:15:00 -0000	1.1.1.3
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,10 +0,0 @@
-debian/tmp/usr/games/kreversi
-debian/tmp/usr/share/man/man6/kreversi*
-debian/tmp/usr/share/doc/kdegames/kreversi/*
-debian/tmp/usr/share/doc/kde/HTML/en/kreversi/*
-debian/tmp/usr/share/applnk/Games/Board/kreversi.desktop
-debian/tmp/usr/share/apps/kreversi/*
-debian/tmp/usr/share/icons/crystalsvg/48x48/apps/kreversi.png
-debian/tmp/usr/share/icons/crystalsvg/32x32/apps/kreversi.png
-debian/tmp/usr/share/icons/crystalsvg/16x16/apps/kreversi.png
-debian/icons/kreversi.xpm /usr/share/pixmaps
--- debian/ksame.install	5 Nov 2002 02:15:00 -0000	1.1.1.3
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,10 +0,0 @@
-debian/tmp/usr/games/ksame
-debian/tmp/usr/share/man/man6/ksame*
-debian/tmp/usr/share/doc/kdegames/ksame/*
-debian/tmp/usr/share/doc/kde/HTML/en/ksame/*
-debian/tmp/usr/share/icons/crystalsvg/48x48/apps/ksame.png
-debian/tmp/usr/share/icons/crystalsvg/32x32/apps/ksame.png
-debian/tmp/usr/share/icons/crystalsvg/16x16/apps/ksame.png
-debian/tmp/usr/share/apps/ksame/*
-debian/tmp/usr/share/applnk/Games/TacticStrategy/ksame.desktop
-debian/icons/ksame.xpm /usr/share/pixmaps
--- debian/kshisen.install	5 Nov 2002 02:15:00 -0000	1.1.1.3
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,10 +0,0 @@
-debian/tmp/usr/share/apps/kshisen/*
-debian/tmp/usr/share/icons/crystalsvg/48x48/apps/kshisen.png
-debian/tmp/usr/share/icons/crystalsvg/32x32/apps/kshisen.png
-debian/tmp/usr/share/icons/crystalsvg/16x16/apps/kshisen.png
-debian/tmp/usr/share/applnk/Games/Board/kshisen.desktop
-debian/tmp/usr/games/kshisen
-debian/tmp/usr/share/man/man6/kshisen*
-debian/tmp/usr/share/doc/kdegames/kshisen/*
-debian/tmp/usr/share/doc/kde/HTML/en/kshisen/*
-debian/icons/kshisen.xpm /usr/share/pixmaps
--- debian/ksirtet.install	10 Dec 2002 20:31:49 -0000	1.1.1.4
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,10 +0,0 @@
-debian/tmp/usr/share/applnk/Games/Arcade/ksirtet.desktop
-debian/tmp/usr/share/icons/crystalsvg/48x48/apps/ksirtet.png
-debian/tmp/usr/share/icons/crystalsvg/32x32/apps/ksirtet.png
-debian/tmp/usr/share/icons/crystalsvg/16x16/apps/ksirtet.png
-debian/tmp/usr/share/apps/ksirtet/*
-debian/tmp/usr/games/ksirtet
-debian/tmp/usr/share/man/man6/ksirtet*
-debian/tmp/usr/share/doc/kdegames/ksirtet/*
-debian/tmp/usr/share/doc/kde/HTML/en/ksirtet/*
-debian/icons/ksirtet.xpm /usr/share/pixmaps
--- debian/ksmiletris.install	5 Nov 2002 02:15:00 -0000	1.1.1.3
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,10 +0,0 @@
-debian/tmp/usr/share/apps/ksmiletris/*
-debian/tmp/usr/share/icons/crystalsvg/48x48/apps/ksmiletris.png
-debian/tmp/usr/share/icons/crystalsvg/32x32/apps/ksmiletris.png
-debian/tmp/usr/share/icons/crystalsvg/16x16/apps/ksmiletris.png
-debian/tmp/usr/share/applnk/Games/Arcade/ksmiletris.desktop
-debian/tmp/usr/games/ksmiletris
-debian/tmp/usr/share/man/man6/ksmiletris*
-debian/tmp/usr/share/doc/kdegames/ksmiletris/*
-debian/tmp/usr/share/doc/kde/HTML/en/ksmiletris/*
-debian/icons/ksmiletris.xpm /usr/share/pixmaps
--- debian/ksnake.install	5 Nov 2002 02:15:00 -0000	1.1.1.3
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,10 +0,0 @@
-debian/tmp/usr/share/apps/ksnake/*
-debian/tmp/usr/share/icons/crystalsvg/48x48/apps/ksnake.png
-debian/tmp/usr/share/icons/crystalsvg/32x32/apps/ksnake.png
-debian/tmp/usr/share/icons/crystalsvg/16x16/apps/ksnake.png
-debian/tmp/usr/share/applnk/Games/Arcade/ksnake.desktop
-debian/tmp/usr/games/ksnake
-debian/tmp/usr/share/man/man6/ksnake*
-debian/tmp/usr/share/doc/kdegames/ksnake/*
-debian/tmp/usr/share/doc/kde/HTML/en/ksnake/*
-debian/icons/ksnake.xpm /usr/share/pixmaps
--- debian/ksokoban.install	5 Nov 2002 02:15:00 -0000	1.1.1.3
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,9 +0,0 @@
-debian/tmp/usr/share/applnk/Games/TacticStrategy/ksokoban.desktop
-debian/tmp/usr/share/icons/crystalsvg/48x48/apps/ksokoban.png
-debian/tmp/usr/share/icons/crystalsvg/32x32/apps/ksokoban.png
-debian/tmp/usr/share/icons/crystalsvg/16x16/apps/ksokoban.png
-debian/tmp/usr/games/ksokoban
-debian/tmp/usr/share/man/man6/ksokoban*
-debian/tmp/usr/share/doc/kdegames/ksokoban/*
-debian/tmp/usr/share/doc/kde/HTML/en/ksokoban/*
-debian/icons/ksokoban.xpm /usr/share/pixmaps
--- debian/kspaceduel.install	5 Nov 2002 02:15:00 -0000	1.1.1.3
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,11 +0,0 @@
-debian/tmp/usr/share/apps/kspaceduel/*
-debian/tmp/usr/share/icons/crystalsvg/32x32/apps/kspaceduel.png
-debian/tmp/usr/share/icons/crystalsvg/16x16/apps/kspaceduel.png
-debian/tmp/usr/share/icons/locolor/32x32/apps/kspaceduel.png
-debian/tmp/usr/share/icons/locolor/16x16/apps/kspaceduel.png
-debian/tmp/usr/share/applnk/Games/Arcade/kspaceduel.desktop
-debian/tmp/usr/games/kspaceduel
-debian/tmp/usr/share/man/man6/kspaceduel*
-debian/tmp/usr/share/doc/kdegames/kspaceduel/*
-debian/tmp/usr/share/doc/kde/HTML/en/kspaceduel/*
-debian/icons/kspaceduel.xpm /usr/share/pixmaps
--- debian/ktron.install	5 Nov 2002 02:15:00 -0000	1.1.1.3
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,9 +0,0 @@
-debian/tmp/usr/games/ktron
-debian/tmp/usr/share/icons/crystalsvg/32x32/apps/ktron.png
-debian/tmp/usr/share/icons/crystalsvg/16x16/apps/ktron.png
-debian/tmp/usr/share/man/man6/ktron*
-debian/tmp/usr/share/doc/kdegames/ktron/*
-debian/tmp/usr/share/doc/kde/HTML/en/ktron/*
-debian/tmp/usr/share/applnk/Games/Arcade/ktron.desktop
-debian/tmp/usr/share/apps/ktron/*
-debian/icons/ktron.xpm /usr/share/pixmaps
--- debian/ktuberling.install	10 Dec 2002 20:31:49 -0000	1.1.1.4
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,9 +0,0 @@
-debian/tmp/usr/games/ktuberling
-debian/tmp/usr/share/icons/crystalsvg/32x32/apps/ktuberling.png
-debian/tmp/usr/share/man/man6/ktuberling*
-debian/tmp/usr/share/doc/kdegames/ktuberling/*
-debian/tmp/usr/share/doc/kde/HTML/en/ktuberling/*
-debian/tmp/usr/share/applnk/Toys/ktuberling.desktop
-debian/tmp/usr/share/applnk/Games/Kidsgames/ktuberling.desktop
-debian/tmp/usr/share/apps/ktuberling/*
-debian/icons/ktuberling.xpm /usr/share/pixmaps
--- debian/kwin4.install	5 Nov 2002 02:15:00 -0000	1.1.1.3
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,11 +0,0 @@
-debian/tmp/usr/games/kwin4
-debian/tmp/usr/games/kwin4proc
-debian/tmp/usr/share/man/man6/kwin4*
-debian/tmp/usr/share/man/man6/kproc4*
-debian/tmp/usr/share/apps/kwin4/*
-debian/tmp/usr/share/icons/crystalsvg/48x48/apps/kwin4.png
-debian/tmp/usr/share/icons/crystalsvg/32x32/apps/kwin4.png
-debian/tmp/usr/share/icons/crystalsvg/16x16/apps/kwin4.png
-debian/tmp/usr/share/applnk/Games/Board/kwin4.desktop
-debian/tmp/usr/share/doc/kde/HTML/en/kwin4/*
-debian/icons/kwin4.xpm /usr/share/pixmaps
--- debian/libkdegames-dev.install	15 Jan 2003 02:27:01 -0000	1.1.1.3
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,5 +0,0 @@
-debian/tmp/usr/include/kde/*.h
-debian/tmp/usr/include/kde/kgame
-debian/tmp/usr/lib/libkdegames.a
-debian/tmp/usr/lib/libkdegames.so
-debian/tmp/usr/lib/libkdegames.la
--- debian/libkdegames1.install	15 Jan 2003 02:27:01 -0000	1.1.1.4
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,7 +0,0 @@
-debian/tmp/usr/lib/libkdegames.so.*
-debian/tmp/usr/share/apps/kdegames/pics/star.png
-debian/tmp/usr/share/icons/crystalsvg/16x16/actions/roll.png
-debian/tmp/usr/share/icons/crystalsvg/16x16/actions/highscore.png
-debian/tmp/usr/share/icons/crystalsvg/22x22/actions/roll.png
-debian/tmp/usr/share/icons/crystalsvg/32x32/actions/roll.png
-debian/tmp/usr/share/icons/crystalsvg/32x32/actions/highscore.png
--- debian/lskat.install	5 Nov 2002 02:15:00 -0000	1.1.1.3
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,12 +0,0 @@
-debian/tmp/usr/games/lskat
-debian/tmp/usr/games/lskatproc
-debian/tmp/usr/share/man/man6/lskat*
-debian/tmp/usr/share/doc/kdegames/lskat/*
-debian/tmp/usr/share/doc/kde/HTML/en/lskat/*
-debian/tmp/usr/share/apps/lskat/*
-debian/tmp/usr/share/icons/crystalsvg/48x48/apps/lskat.png
-debian/tmp/usr/share/icons/crystalsvg/32x32/apps/lskat.png
-debian/tmp/usr/share/icons/locolor/32x32/apps/lskat.png
-debian/tmp/usr/share/icons/locolor/16x16/apps/lskat.png
-debian/tmp/usr/share/applnk/Games/Card/lskat.desktop
-debian/icons/lskat.xpm /usr/share/pixmaps
--- debian/icons/kasteroids.xpm	10 Dec 2002 20:31:49 -0000	1.1.1.2
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,55 +0,0 @@
-/* XPM */
-static char *kasteroids[] = {
-/* columns rows colors chars-per-pixel */
-"32 32 17 1",
-"  c black",
-". c #191919",
-"X c gray20",
-"o c #00007f",
-"O c #7f7f00",
-"+ c #4c4c4c",
-"@ c #666667",
-"# c gray50",
-"$ c blue",
-"% c red",
-"& c yellow",
-"* c gray60",
-"= c #b2b2b2",
-"- c gray80",
-"; c gray90",
-": c white",
-"> c None",
-/* pixels */
-">>>>>    >>>  .>>>>>>>>>>>>>>>>>",
-">>> .+@@# >> ;-.>>>>>>>>>>>>>>>>",
-">> .X+##** > ;%= >>>>>>>>>>>>>>>",
-"> .X+@*#== >> &%= >>>>>>>>>>>>>>",
-">.X++##*== >>>.=%.>>>>>>>>>>>>>>",
-" X++@#**- >>>>.  .>>>>>>>>>>>>>>",
-" +@+***=- >>>>>>>>>>>>>>>>>>>>>>",
-" XO@**O= >>>>>>>>>   >>>>>>>>>>>",
-" @+*#*# >>>>>>>>>>.;-.>>>>>>>>>>",
-"> +#@  >>>>>>>>>>> -%= >>>>>>>>>",
-">>   >>>>>>>>>>>>>> &%=.>>>>>>>>",
-">>>>>>>>>>>>>>>>>>>>.=O >>>>>>>>",
-"  .>>>>>>>>>>>>>>>>>>   >>>>>>>>",
-" ;-.>>>>>>>>>>>>>>>>>>>>   >>>>>",
-" ;%= .>>>>>>>>>>>>>>>>>> -: >>>>",
-"> &%=.>>>>>>>>>>     >>> +-: >>>",
-">>.=O >>>>>>>>> @#--= >>> *=: >>",
-">>> . >>>>>>>>> *=;;;* > # +=: >",
-">>>>>>>   >>>>> =-=:;-= *=- *=: ",
-">>>>>>> ;- >>>> #==$$#-= -=# #@ ",
-">>>>>>> -%=.>>> #== o:$-= # >  >",
-">>>>>>>>.&%& >>> #- oo:#== >>>>>",
-">>>>>>>>> =% >>>> #-.oo$$-= >>>>",
-">>>>>>>>>> . >>>>> #- o$$*== >>>",
-">>>>>>>>>>>>>   > # #= oo -# >>>",
-">>>>>>>>>>>>> -: *=- *=  -#  >>>",
-">>>>>>>>>>>>> @=: ==# *=-# -# >>",
-">>>>>>>>>>>>>> #=: # > ## -* >>>",
-">>>>>>>>>>>>>>> @=: >>>   # >>>>",
-">>>>>>>>>>>>>>>> #=: >>>>> >>>>>",
-">>>>>>>>>>>>>>>>> *+ >>>>>>>>>>>",
-">>>>>>>>>>>>>>>>>>  >>>>>>>>>>>>"
-};
--- debian/icons/katomic.xpm	10 Dec 2002 20:31:49 -0000	1.1.1.2
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,43 +0,0 @@
-/* XPM */
-static char *katomic[] = {
-/* columns rows colors chars-per-pixel */
-"32 32 5 1",
-"  c black",
-". c cyan",
-"X c red",
-"o c white",
-"O c None",
-/* pixels */
-"OOOOOOOOOO...OOOOOOOOOOOOOOOOOOO",
-"OOOOOOOOOO...OOOOOOOOOOOOOOOOOOO",
-"OOOOOOOOOO...OOOOOOOOOOOOOOOOOOO",
-"OOOOOOOOOOOooOOOOOOOOOOOOOOOOOOO",
-"OOOOOOOOOOOooOOOOOOOOOOOOOOOOOOO",
-"OOOOOOOOOOOooOOOOOOOOOOOOOOOOOOO",
-"OOOOOOOOOOOooOOOOOOOOOOOOOOOOOOO",
-"OOOOOOOOOOOooOOOOOOOOOOOOOOOOOOO",
-"OOOOOOOOOOOooOOOOOOOOOOOOOOOOOOO",
-"OOOOOOOOOOOOooOOOOOOOOOOOOOOOOOO",
-"OOOOOOOOOOOOooOOOOOOOOOOOOOOOOOO",
-"OOOOOOOOOOOXXXXXOOOOOOOOOOOOOOOO",
-"OOOOOOOOOOXXXXXXXOOOOOOOOOOOOOOO",
-"OOOOOOOOOXXXXXXXXXOOOOOOOOOO...O",
-"OOOOOOOOOXXXXXXXXXoooooooooo...O",
-"OOOOOOOOOXXXXXXXXXoooooooooo...O",
-"OOOOOOOOOXXXXXXXXXOOOOOOOOOOOOOO",
-"OOOOOOOOOXXXXXXXXXOOOOOOOOOOOOOO",
-"OOOOOOOOOoXXXXXXXOOOOOOOOOOOOOOO",
-"OOOOOOOOoooXXXXXoOOOOOOOOOOOOOOO",
-"OOOOOOOoooOOOOOooOOOOOOOOOOOOOOO",
-"OOOOOOoooOOOOOOOooOOOOOOOOOOOOOO",
-"OOOOOoooOOOOOOOOooOOOOOOOOOOOOOO",
-"OOOOoooOOOOOOOOOOooOOOOOOOOOOOOO",
-"OO...oOOOOOOOOOOOooOOOOOOOOOOOOO",
-"OO...OOOOOOOOOOOOOooOOOOOOOOOOOO",
-"OO...OOOOOOOOOOOOO...OOOOOOOOOOO",
-"OOOOOOOOOOOOOOOOOO...OOOOOOOOOOO",
-"OOOOOOOOOOOOOOOOOO...OOOOOOOOOOO",
-"OOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOO",
-"OOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOO",
-"OOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOO"
-};
--- debian/icons/kbackgammon.xpm	10 Dec 2002 20:31:49 -0000	1.1.1.2
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,53 +0,0 @@
-/* XPM */
-static char *kbackgammon[] = {
-/* columns rows colors chars-per-pixel */
-"32 32 15 1",
-"  c black",
-". c #191919",
-"X c gray20",
-"o c #7f0000",
-"O c #7f7f00",
-"+ c #4c4c4c",
-"@ c #666667",
-"# c gray50",
-"$ c red",
-"% c yellow",
-"& c gray60",
-"* c #b2b2b2",
-"= c gray80",
-"- c gray90",
-"; c None",
-/* pixels */
-";;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;",
-";;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;",
-";;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;",
-".o.oo.o.o o..o.oo.o..o..o..oo.o.",
-"o.oo@oo.O$+ooO#oo.oo#+oo@Oo.X@$o",
-"o. #** .=*@ @==oo. #=* .==@ +*=o",
-"oX *== .==# #*=ooX &==..==#.#*=o",
-"o+ =*=.+==& &=-oo+ =*=.X=*& &=-o",
-"oO.-=%+@==% *==ooO.==-X@==*.*==o",
-"o&.-==+@-=- =--oo#.-*-+O==- %--o",
-"o&X-=-@&==-.=%-oo*+-=-@&-==.-=-o",
-"o=O==-#%==-X=--oo=@%==#%===O==-o",
-"o-#--=**==-O---oo=O-==%*==-O=--o",
-"o==%==-%-%-*=--oo-*-=-=-%--*=%-o",
-"o===%--=-==-=--oo-==--==----=--o",
-"o-==--==--=%=%-oo%==-%==---%=--o",
-"o-%===--=--%==-oo-===%--==---=%o",
-"o--=%==-=--==--oo-%==---%=-%==-o",
-"o---=-%-=%==%=-oo%--=-==-*===*-o",
-"o=--O-==-&==-&-oo=--&%=--&===&-o",
-"o=-=O--==@-=-O=oo-%-+---=@-%=#=o",
-"o-%=.==-%+-=-O*oo===X--%*O-=%@*o",
-"o=-- ===#O==-X&oo=== ===&O==-X&o",
-"o=== %==@X==-.@oo=== *==@ =*-.#o",
-"o=**.#==+ -== +o$***.#==+ ===.Oo",
-"o==&.@*=X *=* Xoo*=& @*=+ ***.Xo",
-"o*=#.+==. &*&..oo*=@ +*=. &=& .o",
-"o&#ooo&Oo.@#O.ooo##+.o##.o@#+o.o",
-".o.ooo.oo.oo.o.ooo.oo.oo.o.ooo.o",
-";;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;",
-";;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;",
-";;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;"
-};
--- debian/icons/kbattleship.xpm	25 Jun 2002 02:34:24 -0000	1.1.1.1
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,54 +0,0 @@
-/* XPM */
-static char *kbattleship[] = {
-/* columns rows colors chars-per-pixel */
-"32 32 16 1",
-"  c #00007f",
-". c #007f7f",
-"X c #7f007f",
-"o c #7f7f00",
-"O c #4c4c4c",
-"+ c #666667",
-"@ c gray50",
-"# c blue",
-"$ c cyan",
-"% c red",
-"& c yellow",
-"* c gray60",
-"= c #b2b2b2",
-"- c gray80",
-"; c gray90",
-": c white",
-/* pixels */
-";;----*====*****=;-;-;;-;;-;;;;-",
-"-========-==**=.=====-=-===-===-",
-"******=--$--=****=*=****+**=****",
-"*@***==$=====******.*@@+@**.#@@@",
-"#**========****@*@@.@***@+@+*.*@",
-"*.*=-====*====@@.@+@@oo+@@+@@@#@",
-"@@*-====*+@**@++@@@@@+o+++***.@@",
-"OO+.*@@@@@.***@**X@@****+@#@*@+@",
-"O++@=****@@***@%@**=***=@@***@#@",
-".@+*o*&oo@++@*@@@&%*@@***=***@@.",
-"+@@@*@***@++@@**%**%%*%-*****@+ ",
-"***@@@@..@@@@@@%%*&==&*******@+@",
-"@@*@@@@@*@*@@@*&---&&=&****=*@.+",
-"@@.**@@@@@.@@@%=&;;;;-*%%@**#+#+",
-"+@@***@@@@@@@o&&;&;;&;;&%*@+ +++",
-"@@@**.@@@@**@@=-;--;;-;;%*X.@@ O",
-"*@.**@#*@@*@@&-&-;;-;--;&*%@+X+.",
-"****@**@@@@@%=&--;;;;--&=*@@+++X",
-"@*@***@@@@@@*&-;;;--;--;&&X+@#.@",
-".*@@**@@*@*=%=&--;;&-;&;*%%*++.X",
-"**@***.@@@@@%-&-&&-&=&*%@@++++.X",
-"*=*=**=*@@@+@*@@@X@@@X+XX ++ @@.",
-"@=***@@@@ +X@ +++X+X++.+++#++#X.",
-".X..+++#@++. ++#++.+X+X.+++++@.X",
-" +.X.++ +X.++#++.XX+++@.+ .+ ++ ",
-" ++ +  +.XX.O+ +++.. #**-*X.+ .+",
-"@+#+.X@++. + .X. +.X++@*=-=. .X.",
-" ++OX.+  +++++.X@ X.@ #@=---=X.X",
-" +++.++ X.O +.X.+ +.X.X+*--;:*.X",
-"++  +  +.X @ X.+ +.X.+.+ .=-:;-.",
-".XX.++@ X. @+++  ++ +  +.X+=;::-",
-"X..+  +@.++ + X.@ ++ +@ +.+.=;;;"
-};
--- debian/icons/kblackbox.xpm	10 Dec 2002 20:31:49 -0000	1.1.1.2
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,50 +0,0 @@
-/* XPM */
-static char *kblackbox[] = {
-/* columns rows colors chars-per-pixel */
-"32 32 12 1",
-"  c black",
-". c #191919",
-"X c gray20",
-"o c #00007f",
-"O c #7f7f00",
-"+ c #4c4c4c",
-"@ c #666667",
-"# c gray50",
-"$ c yellow",
-"% c gray60",
-"& c #b2b2b2",
-"* c None",
-/* pixels */
-"*************       ************",
-"********* .. ++%+++X       *****",
-"***** .X.+@++@+X@+@@++@+++#     ",
-"*    X++++XXX+X+@++@+@@++X+@++X ",
-" XXXX++X++++++XXX+X@@XX++X+++XX ",
-" X++XXXXXX@+X@+++++XX+++++@@X.X ",
-" X+XXX+XX+XX++XXXX+++X@+++X+X.. ",
-" X+XX+XXXXXXX@XXX+XXXXXX+X.X.X. ",
-"* ++X.OOXXXX+#+XXX+XXXX+XX.X.. *",
-"* ++XOX+OOOO+++XXXXXXXXXXXXXXX *",
-"* XO$.X++.+@OOO+OXXXXXX+XX..XX *",
-"* ++&$O.+X+XX.$OO@XXXXX+@XX@XX *",
-"** X+O$+XXXXXXO$O%$XXXXX+X.XX  *",
-"** +X+XXXXXX+X#$OO$OXXX++XX+. **",
-"** XXX+XXXXX++XO+$$%@X+XXXXXX **",
-"** XXX+XX+XXXX.OOOO$$XXXX.X.X **",
-"*** XXXXX@XXXX#OO%$$$o.XXXX+. **",
-"*** ++XXXXXXoO$OO$$OOXXXX.+X. **",
-"*** X+XXXX.X.$$O$OOXXXXXX.XX ***",
-"*** XXXXXXo$$OOXoXX+XXX+.X.X ***",
-"**** +X+@XXO+XXX+XXXXXXX.XX ****",
-"**** +X.@X+XXXXX++XXXXXX+XX ****",
-"**** XX+X+O@++X+XXXXXXXXXX *****",
-"****  XXXX#$O+XX+XXX+@XXXX *****",
-"******  +XXO#$XXXXXXX+X.X ******",
-"********  XX.XX+@+XXXXXXX ******",
-"**********  X++X++++X+X+ *******",
-"************  XXXXXXX+XX *******",
-"**************  ++XXX+X ********",
-"****************  XX+++ ********",
-"******************  +X *********",
-"********************  .*********"
-};
--- debian/icons/kbounce.xpm	10 Dec 2002 20:31:49 -0000	1.1.1.2
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,49 +0,0 @@
-/* XPM */
-static char *kbounce[] = {
-/* columns rows colors chars-per-pixel */
-"32 32 11 1",
-"  c black",
-". c #191919",
-"X c gray20",
-"o c gray50",
-"O c red",
-"+ c gray60",
-"@ c #b2b2b2",
-"# c gray80",
-"$ c gray90",
-"% c white",
-"& c None",
-/* pixels */
-"&&&&&&&&&&&&$%%%%$$$$&&&&&&&&&&&",
-"&&&&&&&&&&$%%%%%%%%%#@O+&&&&&&&&",
-"&&&&&&&&$%%%%%%%%$%@OOOOO&&&&&&&",
-"&&&&&&&%%%%%%%%%%$+OOOOOOO+&&&&&",
-"&&&&&&%%%%%%%%%%$OOOOOOOOOO&&&&&",
-"&&&&&%%%%%%%%%%$+OOOOOOOOOOO&&&&",
-"&&&&%%%%%%%%%%%@OOOOOOOOOOOOO&&&",
-"&&&$%%%%%%%%%%$OOOOOOOOOOOOOO+&&",
-"&&&%%%%%%%%%%$OOOOOOOOOOOOOOOO&&",
-"&&$%%%%%%%%%$@OOOOOOOOOOOOOOOOO&",
-"&&%%%%%%%%%%$OOOOOOOOOOOOOOOOOO&",
-"&&%%%%%%%%%$OOOOOOOOOOOOOOOOOOO&",
-"&$%%%%%%%%%OOOOOOOOOOOOOOOOOOOO&",
-"&%%%%%%%%%#OOOOOOOOOOOOOOOOOOOO&",
-"&%%%%%%%%$+OOOOOOOOOOOOOOOOOOOO&",
-"&%%%%%%%%#OOOOOOOOOOOOOOOOOOOOO+",
-"&%%%%%%%%#OOOOOOOOOOOOOOOOOOOOO+",
-"&$%%%%%%#OOOOOOOOOOOOOOOOOOOOOOO",
-"&$%%%%%$+OOOOOOOOOOOOOOOOOOOOOO&",
-"&$%%%%%$OOOOOOOOOOOOOOOOOOOOOOO&",
-"&&%%%%%#OOOOOOOOOOOOOOOOOOOOOOo&",
-"&&$%%%%OOOOOOOOOOOOOOOOOOOOOOO&&",
-"&&&%%%%OOOOOOOOOOOOOOOOOOOOOOO&&",
-"&&&&%%$OOOOOOOOOOOOOOOOOOOOOO&&&",
-"&&&&$%@OOOOOOOOOOOOOOOOOOOOO+&&&",
-"&&&&&%+OOOOOOOOOOOOOOOOOOOOO&&&&",
-"&&&&&&#OOOOOOOOOOOOOOOOOOOo&&&&&",
-"&&&&&&&OOOOOOOOOOOOOOOOOOO&&&&&&",
-"&&&&&&&&OOOOOOOOOOOOOOOo&&&&&&&&",
-"&&&&&&&&&&OOOOOOOOOOOO&&&&&&&&&&",
-"&&&&&&&&&&&&&+O++Oo&&&&&&&&&&&&&",
-"&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&"
-};
--- debian/icons/kenolaba.xpm	10 Dec 2002 20:31:49 -0000	1.1.1.2
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,52 +0,0 @@
-/* XPM */
-static char *kenolaba[] = {
-/* columns rows colors chars-per-pixel */
-"32 32 14 1",
-"  c black",
-". c #191919",
-"X c gray20",
-"o c #7f7f00",
-"O c #666667",
-"+ c gray50",
-"@ c red",
-"# c yellow",
-"$ c gray60",
-"% c #b2b2b2",
-"& c gray80",
-"* c gray90",
-"= c white",
-"- c None",
-/* pixels */
-"-------  .    . .   ..   -------",
-"------- ***&**&*&*&*&&&& -------",
-"------ &**&**&**&&*&&&&$%X------",
-"------ **&%$$$$$$%$%%$$$$ ------",
-"----- &*%%@%$$$&#+$$+%&o$% -----",
-"----- *&**#@$+&*&@$+%*=@+$ -----",
-"----.%*%*&@@$+#&@@$+o*#@+$% ----",
-"----.**&@@o%%$$@@%%$$@@++$$ ----",
-"----.***&&**%%$$#$%%$oO++$$.----",
-"--- %***&****&%$$$&@+++#&@+% ---",
-"--- ***&&=*===%$+**%@$$**@$$ ---",
-"-- &**&%&*=***&$+%&@o$$*#@$$% --",
-"--.***&%*******%+@@$$$+@@o$$$ --",
-"- &***&&&&*&&*%$$oo$o$o$+++%$% -",
-"- =*&$$%&&#$$%$#$$$+o$$$+%&@$$ -",
-" $*&$$+$**@$$$$$#$%o+$oo$*=@+$+ ",
-" X*&%$$o%%#$+$$$+$%oo$$%$%&@+$+.",
-"-.$&%$$+@#@o+$+#$$*&$%#*%@@o$% -",
-"- O&%%++$$$+++$$#&*#&**&#%&$$+ -",
-"-- &#%&&o$$$++$$%&&$#*&&%&#$$ --",
-"--.%&%**#$$$o$+$$**%@=*%%*$$+ --",
-"--- *$#%@$+$$$+o+#%##*&%#%+$ ---",
-"--- &&#@#+o%%$++$#@o&*%%*&$+ ---",
-"---- &%$+++$o$$#$$&***$#&$$ ----",
-"---- %&%%&#$o%%&$+$&*&%$%$$ ----",
-"---- +&%**@$o+**%#oo#*&#+$O ----",
-"----- &#%%o#$$#*@#+$$%$@$$.-----",
-"----- $&$#@$$o+@@++++##o$$.-----",
-"------ &%+$o$$$$%$#+$%%+% ------",
-"------.$$$$$$+$$$$$$$$$$+ ------",
-"------- $$$$$$$%$$$$$$$$ -------",
-"-------   .  .      .    -------"
-};
--- debian/icons/kjumpingcube.xpm	10 Dec 2002 20:31:49 -0000	1.1.1.2
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,46 +0,0 @@
-/* XPM */
-static char *kjumpingcube[] = {
-/* columns rows colors chars-per-pixel */
-"32 32 8 1",
-"  c black",
-". c #191919",
-"X c gray20",
-"o c #00007f",
-"O c #7f0000",
-"+ c blue",
-"@ c red",
-"# c None",
-/* pixels */
-"################################",
-"################################",
-"##                            ##",
-"## ++++++++ @@@@@@@@ @@@@@@@@ X#",
-"## +  +o+oo @  O@OOO @@O@O@OO XX",
-"## +  o+o+o @  @O@@O @O@O@O@O XX",
-"## ++o+o+oo @@O  O@O @@O  @OO XX",
-"## +o+o+o+o @O@  @OO @O@  @OO XX",
-"## ++o+o  o @O@O@  O @@O@O@OO XX",
-"## +o+o+  o @@O@O  O @O@O@O@O XX",
-"## +ooooooo @OOOOOOO @OOOOOOO XX",
-"##                            XX",
-"## ++++++++ @@@@@@@@ ++++++++ XX",
-"## +  o+o+o @  @@  O +  o+o+o XX",
-"## +  +o+oo @  O@  O +  +oo+o XX",
-"## ++oo+o+o @@O@O@OO +o+  +oo XX",
-"## ++o+oo+o @@O@@@OO ++o  +oo XX",
-"## +o+o+  o @  OO  O +o+o+  o XX",
-"## ++o+o  o @  @@  O +o++o  o XX",
-"## +ooooooo @OOOOOOO +ooooooo XX",
-"##                            XX",
-"## @@@@@@@@ ++++++++ ++++++++ XX",
-"## @@O@O@OO +  o+o+o ++o+oo+o XX",
-"## @O@O@O@O +  +o+oo +o+o++oo XX",
-"## @@O  @OO ++o  o+o ++o  +oo XX",
-"## @O@  O@O +o+  +oo +o+  o+o XX",
-"## @@O@O@OO +o+o+  o ++o+o+oo XX",
-"## @O@O@O@O ++o+o  o +o+o+o+o XX",
-"## @OOOOOOO +ooooooo +ooooooo XX",
-"##                            XX",
-"###XXXXXXXXXXXXXXXXXXXXXXXXXXXXX",
-"####XXXXX.XXXXXXXXXXX.XXXXXXX.XX"
-};
--- debian/icons/klines.xpm	10 Dec 2002 20:31:49 -0000	1.1.1.2
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,57 +0,0 @@
-/* XPM */
-static char *klines[] = {
-/* columns rows colors chars-per-pixel */
-"32 32 19 1",
-"  c black",
-". c #191919",
-"X c gray20",
-"o c #00007f",
-"O c #007f00",
-"+ c #7f0000",
-"@ c #7f7f00",
-"# c #4c4c4c",
-"$ c #666667",
-"% c gray50",
-"& c green",
-"* c red",
-"= c yellow",
-"- c gray60",
-"; c #b2b2b2",
-": c gray80",
-"> c gray90",
-", c white",
-"< c None",
-/* pixels */
-"<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<",
-"<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<",
-"<<<;=<<<<<<<<<<<<<<<<<<<<<<<<<<<",
-"<<@#:<<<<<<<<<<<<<<<<<<<OO<<<<<<",
-"<<X@=<<<<<<<<<<<<<<<<<<&<<&<<<<<",
-"<@.;=<<<<<<<<<<<<<<<<<<<<OO<<<<<",
-"<X+===<<<<<<<<<<<<<<<<<<&<<<<<<<",
-"@X=-=:=<<<<<<<<<<<<;::;&&<<<<<<<",
-"@@=< ==-OO<<<<<<&&&XO#:O&&&<<<<<",
-"=<<<=@+@&&&O<<<&,;&O.%$&&&&&&<<<",
-"<<<<=-..O&&O<<<&&&&&&..O&&&&&&<<",
-"<<<<<=@..O@=<<<O&&OO&OOO&&&&&&<<",
-"<<<<<<=@X O==-&@#OOOOO&&&&&O&&&<",
-"<<<<<<<<=@%@@@@&&OO&&&&&&&O&&&&<",
-"<<<<<<<<<==<OOO&OO&&&&&&&&OOO&&&",
-"<<<<<<<<<<<<&O&&O&&&&&&&&&&OOO&&",
-"<<<<<<<<<<<<&&&&O&&&&&&&&&&&OOO&",
-"<<<<<<<<<<<<&&&&&O&&&&&&&&&&&OO&",
-"<<<<<<<<<<<<O&&&&&&&&&&&&&&OOO&&",
-"<<<<<<<<<<<<<O&OO&&&&&&&&&&XO&&<",
-"<<<<<<<<<<<<<X.@@#O&&&&&O&&O&&&<",
-"<<<<<<<<<<<<<=@@#@@O&&OOOOO.O+@<",
-"<<<<<<<<<<<<<XX@=@X@XOXO&&@=*=<<",
-"<<<<<<<<<<<<<<O$@+@@=*@OOO@@=<<<",
-"<<<<<<<<<<<<<<<=@@X@@@@OO..O<<<<",
-"<<<<<<<<<<<<<<<<<<OOOOOOOOO<<<<<",
-"<<<<<<<<<<<<<<<<<<OOOOOOO<<<<<<<",
-"<<<<<<<<<<<<&-&<<&&<<<<<&&<<&&-&",
-"<<<<<<<<<<<<&>&&O&OO<<<&O&&&&&>&",
-"<<<<<<<<<<<<&&&&O&O&<<<OOOOO&&&&",
-"<<<<<<<<<<<<<O&OO&O<<<<<&OO&O&O<",
-"<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<"
-};
--- debian/icons/kmahjongg.xpm	10 Dec 2002 20:31:49 -0000	1.1.1.2
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,52 +0,0 @@
-/* XPM */
-static char *kmahjongg[] = {
-/* columns rows colors chars-per-pixel */
-"32 32 14 1",
-"  c black",
-". c #191919",
-"X c #00007f",
-"o c #007f00",
-"O c #7f0000",
-"+ c #7f7f00",
-"@ c blue",
-"# c red",
-"$ c magenta",
-"% c yellow",
-"& c gray80",
-"* c gray90",
-"= c white",
-"- c None",
-/* pixels */
-"--------.  .  .--------   .  .--",
-"--------.&*****.------- ****** -",
-"--------.*&&&*&# ------ ***&%&+.",
-"--------.&&#%#&+ ------ &%ooo*# ",
-"--------.*&*##&% ------ *&%**&+O",
-"--------.&%##&&# ------ **ooo*+ ",
-"  .. . -.*&*##&+ ------ &%*&&&# ",
-".*&**** .&&##&*+ ------ **ooo&+ ",
-" *&%*&&+.*%*&#&# ------ *&&o&*+.",
-" **@%@&#.*&&&&&+   .  - *%&&&%#.",
-" *&&@@&%  #+++# ******.=.+#+#++.",
-" =%@@&&#  ..  . &&&*%&+ - . .   ",
-" *&%@@&+**&&%*  &*#*#*+ --------",
-".*&@@**+&&&&&*+ &%&##%#.--------",
-" *&&@X*#&*ooo%+.&&##*&+ --------",
-" *%*&&&+&%*&&&#.%&*##*# --------",
-" .#+#+#+&&ooo&+ *&##&&+.--------",
-" **&***O&&&*&&+ *&&*#&#.--------",
-" *&&&&&+&%ooo&#.**%**&+ --------",
-".**$$%*+***o%&+..+#+#+#.--------",
-" *&$$$&#*&&&*&+**&**&.......----",
-" **%*$&+.++##+#**%&*&+&&*&*& ---",
-".&&$&%*#  ..  .&*$$%&+=&&%&&+ --",
-".*&$$$&+ -----.*&$$$%#*&@*@&+ --",
-" *%*$$&+ ----- *&*&$*#**&@@&+.--",
-".*&**&*# -----.*&$%*&+*%@@&&#.--",
-"- +#+++# ----- *%$$$%+*&&@@*+ --",
-"--. .... ----- *&*$$*#**@@&%# --",
-"-------------- *&*&%*+*%*@@&+.--",
-"--------------- +##++#*&&%&&+.--",
-"----------------  ..  .+#+++# --",
-"----------------------- ..  . --"
-};
--- debian/icons/kmines.xpm	10 Dec 2002 20:31:49 -0000	1.1.1.2
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,54 +0,0 @@
-/* XPM */
-static char *kmines[] = {
-/* columns rows colors chars-per-pixel */
-"32 32 16 1",
-"  c black",
-". c #191919",
-"X c #007f7f",
-"o c #4c4c4c",
-"O c #666667",
-"+ c gray50",
-"@ c blue",
-"# c cyan",
-"$ c red",
-"% c yellow",
-"& c gray60",
-"* c #b2b2b2",
-"= c gray80",
-"- c gray90",
-"; c white",
-": c None",
-/* pixels */
-"::::::::::::::::::::::   :::::::",
-":::::::::::::::::::: . = :::::::",
-"::::::::::::::::::  *%.= :::::::",
-"::::::::::::::::  %*$$ = :::::::",
-"::::::::::::::: *%$$$$ = :::::::",
-"::::::::::::::::  $$$$.= :::::::",
-"::::::::::::::::::  $$ = :::::::",
-"::::::::::::::::::::   = :::::::",
-"::::::::::::::::::::::.= :::::::",
-":::::::::::::::::::::: = :::::::",
-":::::::::::::::::::::: = :::::::",
-"::::::::::::::   :::::.= :::::::",
-"::::::::::::  +*@  ::: = :::::::",
-"::::::::::  &*=@=@+  : = :::::::",
-"::::::::  &**@*=@@=*+ .=   :::::",
-"::::::  &-++*=@@**=  + = =&  :::",
-"::::  +===;;++**=  +=* = =*&+  :",
-"::  +====X@=;;&+ &=**= - &&*==+ ",
-"  +*=*=X@=*@**;; ;;**=   *==*;; ",
-" =;;*#@*=@@=*=+& =*;;*=*==*;;OO ",
-" ===;;*@@*=*++;;&+*=*;;==;;OOOO ",
-" =====;;*=+&;;*=**@&=**;;OOOOOO ",
-" ====*=*++;;*===@*=@+&*oOO+O+OO ",
-"  +==*=*+;;;=*@*=@@*=*OOO+OOOOO ",
-"::  &**=+-**;;*@@==*;;+oO+OO+O  ",
-"::::  &*&;****;;**;;O+OO+OO+  ::",
-"::::::  &-******;;O++++oOO  ::::",
-"::::::::  +**&&*+++++O+o  ::::::",
-"::::::::::  +***+++O++  ::::::::",
-"::::::::::::  +&++++  ::::::::::",
-"::::::::::::::  ++  ::::::::::::",
-"::::::::::::::::  ::::::::::::::"
-};
--- debian/icons/konquest.xpm	10 Dec 2002 20:31:49 -0000	1.1.1.2
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,55 +0,0 @@
-/* XPM */
-static char *konquest[] = {
-/* columns rows colors chars-per-pixel */
-"32 32 17 1",
-"  c black",
-". c #191919",
-"X c gray20",
-"o c #00007f",
-"O c #007f7f",
-"+ c #7f007f",
-"@ c #4c4c4c",
-"# c #666667",
-"$ c gray50",
-"% c blue",
-"& c cyan",
-"* c magenta",
-"= c gray60",
-"- c #b2b2b2",
-"; c gray80",
-": c gray90",
-"> c white",
-/* pixels */
-"......X...........o.X..X.....X..",
-"..XX.....XX.X..X.X..............",
-".......X..............X.......X.",
-".X...o....X..@....X..XX.X.Xo.X..",
-"...X.....X..........X..........X",
-"..X..XXX..X..X...X..@..X.X.oX...",
-".....X..X.X.X...X..X..X..X......",
-"X.X...........X............XX..X",
-".......o..............X..X.X....",
-".XXXX.XX.X......@..............X",
-".OoXo@O@.o..XoX....X.X..X.......",
-".$X#$=#@##X@XX.@..o.X. X......X.",
-".@X+@=-$$==-$++oO..XXo........Xo",
-".X$$+$o$o$$=;;=$##@...X.........",
-"o X$$=#$##o#$;;;;;=OO@o.X...X...",
-"...@##-+@=$#o-;::>::;-$@..XX ...",
-"..o..X@-*-;;:>>>>>:;:;;;$oXX@..X",
-". ..+XX@#===>>>>>::;;::;;-#oX...",
-". ... o.@#=;-;:::-+-;=:::;;O@X..",
-"........X@@@$-===@@--==:-;;-$Xo.",
-".........oX@O+$;---=$O$o----;Oo.",
-"... .......X@@#o$+$O=-==&==--$@X",
-".  .  . ...X.X.@@@@$%$#$+$O$$#X.",
-"...X....... .X.Xo#XX@X@@$=--$$#.",
-".... ........ Xo..Xo@Xo.o.@.oX.X",
-".. .. ......X.. X....XXXO@..X...",
-"...... .. .   ... .....  X......",
-"..... ............ ......  ..o..",
-".... . . ..... ... .........  ..",
-".   ... ... . .. . ...... . . ..",
-"...  ........ . . .X.... ... . .",
-"..o..............o........X..X.."
-};
--- debian/icons/kpat.xpm	10 Dec 2002 20:31:49 -0000	1.1.1.2
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,47 +0,0 @@
-/* XPM */
-static char *kpat[] = {
-/* columns rows colors chars-per-pixel */
-"32 32 9 1",
-"  c black",
-". c #191919",
-"X c #00007f",
-"o c blue",
-"O c red",
-"+ c #b2b2b2",
-"@ c gray80",
-"# c white",
-"$ c None",
-/* pixels */
-"$       $$$$       $$$$       $$",
-" ####### $$ ####### $$ ####### $",
-" #O###O#  $ # ### #  $ #O###O#  ",
-" #######  $ #######  $ #######  ",
-" #######  $ #######  $ #######  ",
-" ###O###  $ ### ###  $ ###O###  ",
-" #######  $ #######  $ #######  ",
-" #######  $ #######  $ #######  ",
-" #O###O#  $ # ### #  $ #O###O#  ",
-" #######  $ #######  $ #######  ",
-"          $$         $          ",
-" @+@@@+@  $$$       $$ @@+@++@  ",
-" #######  $$$$$$$$$$$$ #######  ",
-" #O###O#  $$   .   $$$ # ### #  ",
-" #######  $ @o@o@o@ $$ #######  ",
-"          $ o@o@o@o  $          ",
-" @+@+@+@  $ +o@o@o@. $ #######  ",
-" #######  $ o@o@o@o  $ #######  ",
-" # ### #  $ @o@o@o@. $ #O###O#  ",
-" #######  $.X@o@o@o  $ #######  ",
-"          $ @o@o@o@. $$         ",
-" @+@+@@+  $ o@o@o@o  $$$       $",
-" #######  $ @o@o@o@  $$$$$$$$$$$",
-" #O###O#  $$    . .  $$$$$$$$$$$",
-" #######  $$$       $$$$$$$$$$$$",
-"          $$$$$$$$$$$$$$$$$$$$$$",
-" @+@+@@+  $$$$$$$$$$$$$$$$$$$$$$",
-" #######  $$$$$$$$$$$$$$$$$$$$$$",
-" # ### #  $$$$$$$$$$$$$$$$$$$$$$",
-" #######  $$$$$$$$$$$$$$$$$$$$$$",
-"$         $$$$$$$$$$$$$$$$$$$$$$",
-"$$       $$$$$$$$$$$$$$$$$$$$$$$"
-};
--- debian/icons/kpoker.xpm	10 Dec 2002 20:31:49 -0000	1.1.1.2
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,57 +0,0 @@
-/* XPM */
-static char *kpoker[] = {
-/* columns rows colors chars-per-pixel */
-"32 32 19 1",
-"  c black",
-". c #191919",
-"X c gray20",
-"o c #007f00",
-"O c #007f7f",
-"+ c #7f0000",
-"@ c #7f7f00",
-"# c #4c4c4c",
-"$ c #666667",
-"% c gray50",
-"& c red",
-"* c magenta",
-"= c yellow",
-"- c gray60",
-"; c #b2b2b2",
-": c gray80",
-"> c gray90",
-", c white",
-"< c None",
-/* pixels */
-"<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<",
-"<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<",
-"<<<<<<<<<<<<<  .<<<<<<<. .<<<<<<",
-"<<<<<<<<<<<       <<<  =::  <<<<",
-"<<<<<<<<<< .  .   .< >:::==: <<<",
-"<<<<<<<<<<          >==:::>::.<<",
-"<<<<<<<<<< ..   ..X$>>:>=:=>:.<<",
-"<<<<   <<< XXXXXXX.$>=:==::>@ <<",
-"<<  =>:. < .#.... .@:>::>>;&@.<<",
-"< :>:::=:  X#... .X$:>=;=-%=%.<<",
-".:=>:=::=: .# .. ..$=>>=;=%; <<<",
-" :=:::==:>$.#......X=:>>;:& <<<<",
-" :>=>::::=#.#...  .$->:&;> <<<<<",
-".:>==:=::-X.#. ..%:,;-@:-:, <<<<",
-".=>:::=-@=$.X.$%,,,;-%&*;;,,.<<<",
-" ::>:=;-;&$#%:,,,;#+;;>+#+;,, <<",
-".==:=;;@@->,,>::.@-o+-$>;X@:,, <",
-".->>:;=:,,,,--%@$$%%-@.;#O@#->, ",
-"< -::=,,,:%$-;XX%+%X$:;-:-#$%,> ",
-"<<   +>,,%@%#-@-X-X@$@@#$%;,,:: ",
-"<<<<< >:,,%@#;->-+@$&#$:,,>:;,> ",
-"<<<<< >::,,%$%--:%&$-,,,::;,,:> ",
-"<<<<<.:>::,,#%&&;->:>,:;,,,;:>; ",
-"<<<<<< :,::,;$>;%:>>:;>,:;::;-  ",
-"<<<<<<< :,::::-:,,::,>:::;-   <<",
-"<<<<<<<< :,:::>:;>>,;>;-   <<<<<",
-"<<<<<<<<< ;,::;>,:;;-   <<<<<<<<",
-"<<<<<<<<<< >>,::;-   <<<<<<<<<<<",
-"<<<<<<<<<<< :;-   <<<<<<<<<<<<<<",
-"<<<<<<<<<<<<   <<<<<<<<<<<<<<<<<",
-"<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<",
-"<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<"
-};
--- debian/icons/kreversi.xpm	10 Dec 2002 20:31:49 -0000	1.1.1.2
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,58 +0,0 @@
-/* XPM */
-static char *kreversi[] = {
-/* columns rows colors chars-per-pixel */
-"32 32 20 1",
-"  c black",
-". c #191919",
-"X c gray20",
-"o c #00007f",
-"O c #007f7f",
-"+ c #7f0000",
-"@ c #7f007f",
-"# c #7f7f00",
-"$ c #4c4c4c",
-"% c #666667",
-"& c gray50",
-"* c blue",
-"= c red",
-"- c yellow",
-"; c gray60",
-": c #b2b2b2",
-"> c gray80",
-", c gray90",
-"< c white",
-"1 c None",
-/* pixels */
-"11111111111111111111111111111111",
-"1  . .  ..     . .  ..  . ..  . ",
-"1 ,,,,,,>,:&,,>:;&:,>#,,,:=::,, ",
-"1 ,,,,,,,,:%,>=-===%>&>,=:===#, ",
-"1.,>,,>>>>-%,=:<,==X;&>#:,<=++& ",
-"1.>>>>>>>>:%>=:<>=++X&>=:<,==+X.",
-"1 >,,>>>>>:%>==:==++.%>====+++. ",
-"1 >>>>>>,>:%>====++X+&>=====+.X ",
-"1 <,>>,>>,:%>#=++++.&#>#$=++..% ",
-"1 ,,,>>>-,:#>>+++. +>&>>+X+. X> ",
-"1 :::::::::%>>>%X$%>,#>>>%X$#>>.",
-"1 &#%#%%#%%%;%##%%%%%$&#%%%#%%% ",
-"1 ,,,::>,,,%,,,>;>,,,%>:>>&:>>:.",
-"1 >>-==##>>#,>:===#;,#>:%***o&: ",
-"1 >;::=@+%>%,:=>:==+;&>*;,>*oX&.",
-"1.:=,,===.;%:=,,<=++$&>*>,<*oo$ ",
-"1 &=,<==+.%%>=;<:=++X%:**>O*o.X ",
-"1.====++++%%>=====+.+%:oo**oo.$ ",
-"1 ;==++++ &#>===++++$&:$oooo. % ",
-"1 >%++++ X>%,%++++..&&>#ooX. $: ",
-"1 >:%+ .$>>#,>#X+.X&>&>:;%$$&:: ",
-"1.&%%#%%%%%$&#&&%%&##$%##%%$%%% ",
-"1.>>,>&>>>>%>::>&:>>>%::::-:::: ",
-"1.>:***oX%>%,:****o%:%:;;;;;;;; ",
-"1 :*:,>*oo;%>*;,>ooo;&:;;;;;#;;+",
-"1 &*>,,*oo$%>*>,<*oo+#::;&;;:;& ",
-"1 ;**:&*o.X%:*&:*oo.$%:#;;&&&;; ",
-"1 ;****oo.X$:o***oo $&:;-&;;;&; ",
-"1 :$oooo. %%>ooooo. #&:;;;&;&;;.",
-"1 >&Xo...X:%>;o.X. $;%:;;#;&;#& ",
-"1 :::%X$;::&>:;%$$;::&:;;;;;;;;.",
-"1.           .          ..      "
-};
--- debian/icons/ksame.xpm	10 Dec 2002 20:31:49 -0000	1.1.1.2
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,58 +0,0 @@
-/* XPM */
-static char *ksame[] = {
-/* columns rows colors chars-per-pixel */
-"32 32 20 1",
-"  c black",
-". c #191919",
-"X c gray20",
-"o c #00007f",
-"O c #007f7f",
-"+ c #7f0000",
-"@ c #7f007f",
-"# c #7f7f00",
-"$ c #4c4c4c",
-"% c #666667",
-"& c gray50",
-"* c blue",
-"= c cyan",
-"- c red",
-"; c yellow",
-": c gray60",
-"> c #b2b2b2",
-", c gray80",
-"< c gray90",
-"1 c white",
-/* pixels */
-"   .    .    .  . . . . . . . . ",
-" %$%%%%%%$%%$%%$%$$%$%$%%$$%$%% ",
-".$%$$%$%$%$%$%$%%@%%&%$%%%-%%%% ",
-" %$%%$%$%$%$%$%$O:>=*&%%:>:--$% ",
-" %$%$%$%$%$%$%$%*11>*O*:,<,---$.",
-" $%$%$%$%$%$%$%$*<1,*OO->1>-%+$ ",
-".%$%$%$%$%$%$%%$**:*&%o----&+X$.",
-" $%$%$%$%$%$%$$%:=*O*Oo#---#++$.",
-".$%$%$%$%$%$%$%$@OOooXX%&++$+X$.",
-" %$%%%%$$%$%$%$%%$%XX$$%$%+$X%% ",
-".%%>>--&%$%$%$%$%>,;;:#&:>---%% ",
-" &-,<:---%$%$%$%;<<>;;#>><>--+% ",
-" %-11:+%+$$%$%%$;<1;;&##,1,-%+&.",
-".%-:---#X%%$%$$$;,,;;#$--:--++$.",
-" %----@++$%%$%$%;;;##%$----@++$.",
-" %%-%#+XX%$$%$%$%;:###$#@##+XX$.",
-".%%++++$$$%@%&%$$$+++$$%$+++X$%.",
-".%%:>:-&%%:==*%%%&,-:-#%&>:--%% ",
-".$>,<--+@&,<:**O->1>-@->><>--+$ ",
-".%-<1:-%$*<1>*%O:,<:-#+#><>-&+$.",
-".%-->-#++*>::*Oo-->--%+-->&--+$.",
-" %:----$$O=**O$o:---#+X-----$+$.",
-".$#-@++.X%*OOoXX#-&+@+X$&-+$+X$.",
-" %%@++XX%%$ooXX%&++X+X$%$++..$% ",
-".$&>:=:*%%&=*:*%#:<;;&%%:;;>;&%.",
-" %*,,**%*:,,>*%O><<;;;#><1>;##% ",
-".%*<1>O*o><1=*O*;<1,#:#;<1,;##% ",
-" %=>>*O&o*=>*&%o;;,;;#$;;,;;:#% ",
-".%*O*O&oo*OOOoO.;;;###$;;;;##X$.",
-".%&*O*oOo%:**$oo&:&;##X#&;&###$.",
-".%%$OoXX$%Oo$XX$%##+XX$%+##.X$% ",
-"       .. .  . . .     . .  . . "
-};
--- debian/icons/kshisen.xpm	10 Dec 2002 20:31:49 -0000	1.1.1.2
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,57 +0,0 @@
-/* XPM */
-static char *kshisen[] = {
-/* columns rows colors chars-per-pixel */
-"32 32 19 1",
-"  c black",
-". c #191919",
-"X c gray20",
-"o c #007f00",
-"O c #7f0000",
-"+ c #7f007f",
-"@ c #7f7f00",
-"# c #4c4c4c",
-"$ c #666667",
-"% c gray50",
-"& c green",
-"* c red",
-"= c magenta",
-"- c yellow",
-"; c gray60",
-": c #b2b2b2",
-"> c gray80",
-", c gray90",
-"< c white",
-/* pixels */
-"        .. .  .. .... ..     .. ",
-" oo&o&&oo&o&&o&ooo&oo&ooo&o&ooo ",
-".ooo&ooooo@,,,,:o-,,,,$&o&oooo& ",
-" &&ooo&oo@>,*O->*,o,@@;ooooo&oo ",
-".oo&oo&o&X::%*;:@<@,@%;&ooooooo.",
-" &oo&oooo@-;***-@,,@@,%oo&ooooo.",
-" ooo&o&oo@-,>O>:@,@>$:;oo&o&o&o.",
-" &&ooo&oo@:,:*::*,o,&$-oooooooo.",
-" &oo&ooo&@>;*:*-@-o,;o:oooooooo.",
-" oo&o&&oo#-;>,;:@>;,:;-o&oo&&oo ",
-" &ooooooo@*@*@%&*%@*@@ooooooooo.",
-".oo&->>-:@@>>->@@:>>>>#@>>>::oo ",
-" oo@<:>>,@>;>>@>@<,;<,%%-:,--$o.",
-" &@;,>>>>@-@:-X:O<:+><;-:%-O:oo.",
-" &o*>+;+;@>>>o:>*+>;=%;;:-,->oo.",
-" oo@>-;+>*:,:@>-@:,>>:-@,:,->oo.",
-".oo*>;%=;@>$;-o:*<::>,;:-:>@>oo.",
-".&o->%>::%-&:>o:@:o@@>@-:*-*;@o.",
-" oo@>>:+>*>@:>#:@,>@><;;-:>->oo ",
-" &o*;--::@*;;--%*;:>::*@:;-;;oo.",
-" &o@@@OO@o@*@*@@@@@*-@@@*@@@@oo.",
-" oooo&o&oo:,,,,>@,>>,,@@<>>>>@o ",
-".&o&ooo&o@>-;-*:@,::>:*;<;*-,oo.",
-" ooo&ooooO>@:-@:@,;;:;-;>*@*>oo ",
-" oo&o&ooo@:,>>->@,;;+>*:>**@>oo.",
-".oooooooo@>->-:-*>>%+>-@,,*:>@o ",
-".&o&o&&oo@>*--*:@,*>>+-;,;*;>&o ",
-" oooooooo@>-::@:@,>>+,*;-*:*:oo ",
-" oooooooo@;>>,->*>>:>>@%:>>>:@o ",
-" o&oo&&oo@O@@*%o@@@**@@@@@*@@oo ",
-" &o&ooooo&ooo@ooooo@oooo@oooooo ",
-"   ..  . ..     . .   .  . .    "
-};
--- debian/icons/ksirtet.xpm	10 Dec 2002 20:31:49 -0000	1.1.1.2
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,61 +0,0 @@
-/* XPM */
-static char *ksirtet[] = {
-/* columns rows colors chars-per-pixel */
-"32 32 23 1",
-"  c black",
-". c #191919",
-"X c gray20",
-"o c #00007f",
-"O c #007f00",
-"+ c #007f7f",
-"@ c #7f0000",
-"# c #7f007f",
-"$ c #7f7f00",
-"% c #4c4c4c",
-"& c #666667",
-"* c gray50",
-"= c blue",
-"- c green",
-"; c cyan",
-": c red",
-"> c yellow",
-", c gray60",
-"< c #b2b2b2",
-"1 c gray80",
-"2 c gray90",
-"3 c white",
-"4 c None",
-/* pixels */
-"44444444444. 4444444444444444444",
-"4444444444 ,$ 444444444444444444",
-"444444444 >,,*.44444444444444444",
-"44444444.-,-->-.4444444 44444444",
-"4444444 +O$>>,.+ 44444.: 4444444",
-"444444 ,>OO1,.O>$ 444 *:: 444444",
-"44444.,>>,$.-$,-<,.4 :::*:.44444",
-"4444.*>,<>, O,>,>--.@,:,::@ 4444",
-"444.$O$1>>.@:.$>>-%$:@:*,:..4444",
-"44.,1,O,<O:,,:.,,X:::@#:@. 44444",
-"4.$11>>O.:,::<:$O*:,,:@X@ 444444",
-" >2>><,.%@<<,,:@%:<:,::X.4444444",
-" .>1<-.,1$@1,:@,:X:<,:X==.444444",
-"4.X-$.>12>%@,@:,,:@@@#*,<+ 44444",
-"44 .X>32221$:<<<<:*@.;1;;==.4444",
-"444 $&222>$$$@<::<:@#o<=,<o& 444",
-"44 >3$X12$$12$@,1*@,:@+<=+o,=.44",
-"4 <223$$$>2111$@@:<:,:#%o*;,;= 4",
-" <22232$$2221>1$@<<,:<:.%1=,;<*.",
-".$2222$,>%2223*&*@1:,,#o+o1;,=o ",
-".$$32$>23>%22$*;1=X1:#o11+o<,oo ",
-"4 $$><2322,X,=21;2+&#&;1;1=+=o.4",
-"44 %$222222$o12221;..+21<;,oo 44",
-"444 $$2221$X.+2221o*1o<;<,oo 444",
-"4444 $$22$$  o%12o,21,o<<=o.4444",
-"44444 $$$%.44 oo=;1;;2;+oo 44444",
-"444444.%$ 4444 o%12221<oo 444444",
-"4444444  444444.o+222;oo 4444444",
-"4444444444444444 oo<1oo 44444444",
-"44444444444444444 oooo.444444444",
-"444444444444444444 oX 4444444444",
-"4444444444444444444..44444444444"
-};
--- debian/icons/ksmiletris.xpm	10 Dec 2002 20:31:49 -0000	1.1.1.2
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,54 +0,0 @@
-/* XPM */
-static char *ksmiletris[] = {
-/* columns rows colors chars-per-pixel */
-"32 32 16 1",
-"  c black",
-". c #191919",
-"X c gray20",
-"o c #7f0000",
-"O c #7f007f",
-"+ c #7f7f00",
-"@ c #4c4c4c",
-"# c #666667",
-"$ c gray50",
-"% c yellow",
-"& c gray60",
-"* c #b2b2b2",
-"= c gray80",
-"- c gray90",
-"; c white",
-": c None",
-/* pixels */
-":::::::::::::       ::::::::::::",
-"::::::::::. .%=%%==%  .:::::::::",
-"::::::::. %%=%%=%=%*%*%o :::::::",
-"::::::: =%==%==%%%*%*%&&%.::::::",
-":::::: %%=%%=%%=*%*%%*%&%* :::::",
-"::::: %==%--==-%%=%**%*%&%& ::::",
-"::::.==%----=%=%*%*%%%%&%&%$ :::",
-"::: %%=;-;;&&%=%%*%$#**%&%&%&.::",
-"::: %==---&+X+*%%*+X@+%&%&&%$.::",
-"::.%=%----+XX+%=*%+XX+&%&%%&%$ :",
-"::.*%-----+XXX%*%*@X.o%*%&%&&%.:",
-"::.%=%--=-#X o*%*%+X.X&%&%&%+& :",
-": %=%==%%-+  X%*%*+  @%&&%&$%%$ ",
-": =%=%=-=%*@@%*%*%%Xo&&%%&%%*&& ",
-": *=%%=%%=%%*%*&*%&%&%&%&%%&&%% ",
-":.%%%=%==%*%%*%%%&%*%&%&%&&%&+& ",
-": %=*%*%%&%**%*%&%&%%&%%&%&+%%$.",
-":.*%%*%*+%*%%&%&%&%&&%&&++%&&&+.",
-":.%**%%%O%%*&%&%&%*%%&%*#&%$&%&.",
-": %*%**% $%*%&%&%&%&&%&+.+&%+*+.",
-":: %%*%&+.$%&%*%&%%&%&+.&%$%*+o:",
-":: **%*%%.+$%&%&&&&%%+@ %*+&%$.:",
-":: %%*%*%& @*%%%%&%*&o %$&%&%$ :",
-":::.&%&%*%+ .+&&&%$+..+*$%+&$ ::",
-":::.&&%&*%&%@... ...++$%+*%$+.::",
-":::: %&%%&%&*++++#++*$%*$%&+.:::",
-"::::: *%*%&%%&%*&%%&%&$++$&.::::",
-":::::: &+&%&&%$+&&*+%&%&*%.:::::",
-"::::::: &%$%$%%$%&%&$+*$+.::::::",
-"::::::::  +*+&$%$++&%&% .:::::::",
-"::::::::::. o&+*&+%$X  :::::::::",
-"::::::::::::: ......::::::::::::"
-};
--- debian/icons/ksnake.xpm	10 Dec 2002 20:31:49 -0000	1.1.1.2
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,55 +0,0 @@
-/* XPM */
-static char *ksnake[] = {
-/* columns rows colors chars-per-pixel */
-"32 32 17 1",
-"  c black",
-". c #191919",
-"X c gray20",
-"o c #007f00",
-"O c #007f7f",
-"+ c #7f0000",
-"@ c #4c4c4c",
-"# c #666667",
-"$ c gray50",
-"% c cyan",
-"& c red",
-"* c gray60",
-"= c #b2b2b2",
-"- c gray80",
-"; c gray90",
-": c white",
-"> c None",
-/* pixels */
-">>>> >         > >>>>>>>>>>>>>>>",
-">>   > .  . .  >  >>>>>>>>>>>>>>",
-">>> > @-%.$=OO. > >>>>>>>>>>>>>>",
-">> >> @%O.O*$O@.  > >>>>>>>>>>>>",
-">  > @*X@@$#.X:. > > >>>>>>>>>>>",
-" > > #;X;=-*#:;X  > > >>>>>>>>>>",
-">   .@-:;O*::;=OOX > >>>>>>>>>>>",
-"    O%O*O%OO**$%%O. > >>>>>>>>>>",
-">  O%%%O%O%OOO%O*%O > >>>> >   >",
-"  O%OO=%O%O%%%OO%%O > >>> >  >  ",
-"  %%$$%#$O%=OX.O%%O.> >> > >   >",
-"> .X%%%%%%OOOOo$%O. > >>> > OO  ",
-">   .X....OOO%O%OO.> >>  > XOO..",
-" > +&@. . @O*%%%%.     > > OO. >",
-"> .=$ > > O%%%%%O >  >  > OO. > ",
-"> .&& > > O%%O*O.      > O%O >  ",
-">     >> @O%%%%O >    > O%%. > >",
-">    > > O%**O%  >   > O%O. > >>",
-" >    > O%%%%%O. >   > @%O > > >",
-">       OO*%%O. > > > X%*X > > >",
-">> > > O%%%OOO. > > > O%%X  > >>",
-"> >    O%*=%%. > >> > XO%%O. >  ",
-">> >  O*%*%%O        > O%$OX > >",
-"  >  o%%%%%O. > >>>>.> .O%%O  > ",
-"> > XO%%%O%o.          .X%%*. > ",
-" > .O%%*%*O@XOO@OOO@O@OXOO%O..> ",
-" > .O%%%%%%%%%%%%*%%%%%%%%O%. > ",
-" >> O$%O*%OOO%*%%%O%%*O%%*%X.>> ",
-"> >> .OO%O%%O%O%O%O%%O%%OX..>> >",
-" > >  .XXX.X@XOX@XXXXXX..   > > ",
-"> >  >.                   >  > >",
-">> >   > >>>>>>>>>>>>>> >   > >>"
-};
--- debian/icons/ksokoban.xpm	10 Dec 2002 20:31:49 -0000	1.1.1.2
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,54 +0,0 @@
-/* XPM */
-static char *ksokoban[] = {
-/* columns rows colors chars-per-pixel */
-"32 32 16 1",
-"  c black",
-". c #191919",
-"X c #7f0000",
-"o c #7f007f",
-"O c #7f7f00",
-"+ c #4c4c4c",
-"@ c #666667",
-"# c gray50",
-"$ c red",
-"% c yellow",
-"& c gray60",
-"* c #b2b2b2",
-"= c gray80",
-"- c gray90",
-"; c white",
-": c None",
-/* pixels */
-":::::::::::::::: :::::::::::::::",
-":::::::::::::::.&.::::::::::::::",
-":::::::::::::: *-X :::::::::::::",
-"::::::::::::: #-@#X.::::::::::::",
-":::::::::::::.=*X&$.::::::::::::",
-"::::::::::::.*=@oO#+ :::::::::::",
-"::::::::::: &*-+$#$$X ::::::::::",
-":::::::::::X*=&O&O##@ ::::::::::",
-":::::::::: =*-&o@#$#@X :::::::::",
-":::::::::.&===$O$$#O$@X ::::::::",
-":::::::::.**--+$#OOo#Oo.::::::::",
-":::::::: ==*;#X#@$#$o@O+ :::::::",
-"::::::: #===-X#$$@X#@@X$X ::::::",
-":::::::.=*=-=$O$#$#$$@#X+.::::::",
-":::::: ***=-#$##$#$O$##XXO.:::::",
-"::::: *==*=-$O$$$O$o#$O#XoX ::::",
-":::::X**==--$O#$#$#$Oo@#X@$.::::",
-":::: ***=--=$$$$$O$@$@O$#XOX :::",
-"::: #=*===;$&$$#$@$@@$@$#X@@+ ::",
-"::: --==**;$$$$&$$@$$#XO$@$@X.::",
-":::: +--*=*$$$$$#$O$#$Oo@$@+.:::",
-"::::: X@--*$%$&$O$$$$@$#X@X ::::",
-"::::::.XX@O@#@+@#@@@@@XX@X :::::",
-"::::::: XX+XX+Xo+XX@+X@XX.::::::",
-":::::::: X+X+XX@X@X@+X+X.:::::::",
-"::::::::: X@X+@X+X@XX+X ::::::::",
-"::::::::::.X@oO+oO+O@X :::::::::",
-":::::::::::.OXX++XX@X ::::::::::",
-"::::::::::::.@@XoX@X :::::::::::",
-":::::::::::::.X+O@X.::::::::::::",
-"::::::::::::::  @  :::::::::::::",
-":::::::::::::::: :::::::::::::::"
-};
--- debian/icons/kspaceduel.xpm	10 Dec 2002 20:31:49 -0000	1.1.1.2
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,53 +0,0 @@
-/* XPM */
-static char *kspaceduel[] = {
-/* columns rows colors chars-per-pixel */
-"32 32 15 1",
-"  c black",
-". c #191919",
-"X c #00007f",
-"o c #7f7f00",
-"O c #4c4c4c",
-"+ c #666667",
-"@ c gray50",
-"# c blue",
-"$ c red",
-"% c yellow",
-"& c gray60",
-"* c #b2b2b2",
-"= c gray80",
-"- c gray90",
-"; c None",
-/* pixels */
-";;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;",
-";;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;",
-";;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;",
-";;;;;;;;;;;;;;;;;;;;;;;;;;;;  ;;",
-";;;;;;;;;;;;;;;;;;;;;;;;;;; $$ ;",
-";;;;;;;;;;;;;;;;;;;;;;;;;;; $$ ;",
-";;;;;;;;;;;;;;;;;;;;;;;;;;;;  ;;",
-";;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;",
-";;;;;;;;;;;;;;;;;;;  ;;;;;;;;;;;",
-";;;;;;;;;;;;;;;;;; $$ ;;;;;;;;;;",
-";;;;;;;;;;;;;;;;;; $$ ;;;;;;;;;;",
-";; ;;;;;;;;;;;;;;;;  ;;;;;;;;;;;",
-"; # ;;;;;;;;;;;;;;;;;;;;;;;;;;;;",
-" ### ;;;;;;;;;;;;;;;;;;;;;;;;;;;",
-" ###X ;;;;;;;;;;;;;;;;;;;;;;;;;;",
-"; #X## ;;;;;;;;;;;;;;;;;;;;;;;;;",
-";; ###X ;;;;   ;;;;;;;;;;;;;;;;;",
-";;; #X## ;; $% ;;;;;;;;;;;;;;;;;",
-";;;; #### .$%& ;;;;;;;;;;;;;;;;;",
-";;;;; ##  o$%= ;;;;;;;;;;;;;;;;;",
-";;;;;;   $o&- ;;;;;;;;;;;;;;;;;;",
-";;;;;;; o%%= ;;;;;;;;;;;;;;;;;;;",
-";;;;;; $$%=.  ;;;;;;;;;;;;;;;;;;",
-";;;;  O%*=  ## ;;;;;;;;;;;;;;;;;",
-";;;; +@@=  #### ;;;;;;;;;;;;;;;;",
-";;;;; &* ;; ##X# ;;;;;;;;;;;;;;;",
-";;;;;; * ;;; X### ;;;;;;;;;;;;;;",
-";;;;;;;  ;;;; ##X# ;;;;;;;;;;;;;",
-";;;;;;;;;;;;;; X### ;;;;;;;;;;;;",
-";;;;;;;;;;;;;;; ### ;;;;;;;;;;;;",
-";;;;;;;;;;;;;;;; # ;;;;;;;;;;;;;",
-";;;;;;;;;;;;;;;;; ;;;;;;;;;;;;;;"
-};
--- debian/icons/ktron.xpm	10 Dec 2002 20:31:49 -0000	1.1.1.2
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,47 +0,0 @@
-/* XPM */
-static char *ktron[] = {
-/* columns rows colors chars-per-pixel */
-"32 32 9 1",
-"  c black",
-". c #191919",
-"X c gray20",
-"o c #00007f",
-"O c #7f0000",
-"+ c #4c4c4c",
-"@ c #666667",
-"# c blue",
-"$ c red",
-/* pixels */
-"                                ",
-"                                ",
-"  XXXXXX ################ XXXX+@",
-"  XXXXXX #o##o#oo#o#o#o#o XXXX@+",
-"  XX.XXX ##ooooooooooo#oo XXXX+@",
-"  XXXXXX #oo          ##o XXX.@+",
-"         ##o XX.XXXXX #oo XXXX++",
-"  ########oo XXXXXXXX ##o XX.X@@",
-"  ###o#o#o#o XXXXXXXX #oo XX.X++",
-"  #ooooooooo XXXXX.XX ##o XXXX+@",
-"  ##o             XXX #oo XXXX@+",
-"  #oo $$$$$$$$$$$ XXX ##o XXXX+@",
-"  #oo $O$O$O$OO$O XXX ##o XXXX++",
-"  ##o $$OOOOOO$$O XXX #oo X.XX+@",
-"  #oo $OO     $OO XXX #oo XXXX+@",
-"  ##o $$O XXX $$O XXX ##o XXXX@+",
-"  #oo $$O X.X $$O XXX ##o XXXX++",
-"  ##o $OO XXX $OO XXX     XXXX@+",
-"  #oo     XXX $$O             +@",
-"  ##o XXXXXXX $$$$$$$$$$$$$$O @+",
-"  ##o XXXXXXX $O$O$O$O$O$O$OO +@",
-"  #oo XXXXX.X OOOOOOOOOOOO$$O @+",
-"  ##o XXXXXXX             $OO +@",
-"      XX.XXXXXXXXXXXXXXXX $$O @+",
-"  XX.XXXXXXXXXXXXXXXXXXXX $$O +@",
-"  XXXXXXXX.XXXX           $OO @+",
-"  XXXXX.XXXXXXX $$$$$$$$$$$$O +@",
-"  XXXXXX.XXXX.X $OO$$OO$$O$OO ++",
-"  XXXXXXXX.XXXX OOOOOOOOOOOOO @+",
-"  XXX.XXXXXXXXX               +@",
-"+@+@+@+++@++++@@+@+@@@+@+@+@@+@+",
-"@++@@+@+@+@@@+++@+@+++@++@@++@+@"
-};
--- debian/icons/ktuberling.xpm	25 Jun 2002 02:34:24 -0000	1.1.1.1
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,58 +0,0 @@
-/* XPM */
-static char *ktuberling[] = {
-/* columns rows colors chars-per-pixel */
-"32 32 20 1",
-"  c #191919",
-". c gray20",
-"X c #00007f",
-"o c #007f00",
-"O c #007f7f",
-"+ c #7f0000",
-"@ c #7f7f00",
-"# c #4c4c4c",
-"$ c #666667",
-"% c gray50",
-"& c blue",
-"* c green",
-"= c cyan",
-"- c red",
-"; c yellow",
-": c gray60",
-"> c #b2b2b2",
-", c gray80",
-"< c gray90",
-"1 c white",
-/* pixels */
-"1111111111111<,>:>1111111111<111",
-"111111111111<%*$*O,11<1111111111",
-"111111111111,o$o@:%:<<1111111111",
-"111111111111:o*%%@:@%>1111111111",
-"111111111111:@:%%@@>>%<111111111",
-"11111111111<%%%:%:%:>@<<11111111",
-"1111111<11<%:# @@::++:@<11111111",
-"11111,::11:$:@%%%-%%:%%>11111111",
-"11111:::<1>#:%;:@:@::@@>11111111",
-"1111<:>,:><%-::%-->%@%$<11111111",
-"11111>>>>11%$:%%:--::%>1<1111111",
-"11111<::111:@:@-%:@%:-,111111111",
-"1111<1$#11>.$@@:@-:%@:$:,<111111",
-"111111>.1< .$:+:>:>@%@+# .>11111",
-"111111< $#$>#:%%:>%%:%:11$.11111",
-"1111111,.#<,@%@:%%@%@%>>: %11111",
-"11111111111<.$%@%%-:@:1,$%111111",
-"111111111111%@%%@::@#$,><1111111",
-"1111111111<1<$#$@%:;$><111111111",
-"1111111111111<###$$$>11<11111111",
-"11111111111111> :##<111111111111",
-"111111111111<1%#<:#1111111111111",
-"11111111111111$$1>.1111111111111",
-"1111111111111<%%1>.1111111111111",
-"11111111111<11%$1>.<<<1111111111",
-"111111111<,,>,%X,:.,,,,,<1111111",
-"1111111<,>:::,::>#O>::=:><<11111",
-"1111111:&:=>:&:>,$%:,&,&>>111111",
-"1111111:O::>:>><:>,>>>::>><11111",
-"1111111<%%:&>,<>O<1<,&:::><11111",
-"11111111,%:&O:>%:>>>::=>>,111111",
-"11111111<<,:::&::>&:>>,<<<111111"
-};
--- debian/icons/kwin4.xpm	10 Dec 2002 20:31:49 -0000	1.1.1.2
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,53 +0,0 @@
-/* XPM */
-static char *kwin4[] = {
-/* columns rows colors chars-per-pixel */
-"32 32 15 1",
-"  c black",
-". c #191919",
-"X c gray20",
-"o c #00007f",
-"O c #007f7f",
-"+ c #7f0000",
-"@ c #7f007f",
-"# c #7f7f00",
-"$ c #4c4c4c",
-"% c #666667",
-"& c gray50",
-"* c blue",
-"= c red",
-"- c yellow",
-"; c gray60",
-/* pixels */
-"                                ",
-"                     .          ",
-"  *oooooooooooooooooooooooooo*. ",
-"  o*oooooooooooooooooooooooo**  ",
-"  **oooooooooooooooooooooooo**  ",
-"  o**ooooooooooooooooooooooo**  ",
-"  o*oooooooo@ooooooooooooooo**  ",
-"  **ooooooo@+=+ooooooooooooo**  ",
-"  **ooooooo+==@ooooooooooooo**  ",
-"  o*oooooo@@==+ooooooooooooo**  ",
-"  ***oooooo++=+ooooooooooooo**  ",
-"  o*oo$o$ooo+@Xoooo$oooooooo**  ",
-"  o*o$#-%oo@=+@oo$-#%ooooooo**  ",
-"  **o&---Xo====o$&---$oooooo**  ",
-"  **o#--#$o+==+Xo%---ooooooo**  ",
-"  o*o%-;&oo+@=@oo$-##ooooooo**  ",
-"  **oX$%Xooo@@o@o$%%%Xoooooo**  ",
-"  o**@+=+oX@+=+oo%#-&ooooooo**  ",
-"  o*o+===@o===+oo#--#ooooooo**. ",
-"  **@@===oo+===X$%---$oooooo**  ",
-"  **o@++@oo+=+@oo$#-&ooooooo**  ",
-"  o*oo$@XoooX@ooo$@$$ooX$$oo**  ",
-"  **o%-#&oo+=+@oo++=+oo$;-$o**  ",
-"  o*o%---oo===+oo@===o$&---o**  ",
-"  **o#---oo+==+oo+===$o#---o**. ",
-"  o*o%#-#oo@+=@oo@++@oo###&o**  ",
-" .***oo$$oooo@ooooo@oooo$ooo**  ",
-" .**oooooooooooooooooooooooo**. ",
-".O****************************o.",
-"**.X************************.o**",
-"*o.**oooooooooooooooooooooo**..*",
-"****oooooooooooooooooooooooo****"
-};
--- debian/icons/lskat.xpm	10 Dec 2002 20:31:49 -0000	1.1.1.2
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,51 +0,0 @@
-/* XPM */
-static char *lskat[] = {
-/* columns rows colors chars-per-pixel */
-"32 32 13 1",
-"  c black",
-". c #191919",
-"X c gray20",
-"o c #00007f",
-"O c #007f7f",
-"+ c #7f0000",
-"@ c #7f007f",
-"# c #7f7f00",
-"$ c #4c4c4c",
-"% c #666667",
-"& c red",
-"* c yellow",
-"= c white",
-/* pixels */
-"oXoXoXoXXoXooXoXoXoXoXo.oXo.oXo.",
-"ooXo.oXo%+X&&+XoXooXXoXooooXXoXo",
-"XooXoXo#&o&@o&ooXoXooXoXX%XoXooX",
-"oXXoXoo&#oX&O+o$oXXoXo.oX+oXXXoo",
-"XooXoXoX+&$Xo#Xo+Xo++@#o&#+XooXo",
-"ooXoXoXo$+@$X&@+ooXo+$&o@&oo.ooX",
-"oXXXoXo+o==========$@o&oo#o.XoXo",
-"Xooo.oO&&.+=&+&==+=+&#&O$&$XoXoX",
-"ooXooXoXo *====   =oXooXoooXoXoX",
-"oXoXXoXoX= === == =.ooXooXXoXo.o",
-"XoXooXoXo= =&  &  =oXoo.o.oXoXoX",
-"oXoXXoXoX=*=&**&  =XoXXoXoXooXoo",
-"oXXoXoXoo=* = **==========o.XXoo",
-"Ooo.oXo..=**=***====== ===XooXoX",
-"o.ooXo.oo=* = =& =   == ==oXo.Xo",
-"oX==========  = == ==&========oX",
-"o.&&========= ====  =&&&===&==oX",
-"oo=========== &&==  &=&&==&&==Xo",
-"oX============================o.",
-"Oo==========XooX==  ===&===&==Xo",
-"oX====&=====XoXo== ===&&=&&&==o.",
-".o====&&====oXoX========&&====Xo",
-"o.=====&====Xo.o==  ==&&==&&==oo",
-"oo==========oXoo== ===&&==&&==oX",
-"oX=========&oXXX==============oX",
-"Oo=========&.ooXoXoo===&===&=&.o",
-"oX========&=ooXo.ooX==&&===&==oX",
-".o=========&oXoXoX.o=========&oo",
-"oXoXoXooXoXooXXoXoXooXo.oXoXX.oX",
-"oo.o.oXXoXoXXooXoXoXXoXoXoXoooXo",
-"oXo.oXooXoo.oXXooXXooXo.oXoXXooX",
-"OoOoXooXoXoOoooXooXooXoOooXooXoo"
-};
--- kbackgammon/Makefile.am	10 Dec 2002 20:31:41 -0000	1.1.1.4
+++ kbackgammon/Makefile.am	10 Dec 2002 20:56:04 -0000	1.7
@@ -9,6 +9,9 @@
 			kbgstatus.cpp
 kbackgammon_LDFLAGS = $(all_libraries) $(KDE_RPATH)
 
+kbackgammon_main.cpp: main.cpp
+	cat main.cpp > kbackgammon_main.cpp
+
 noinst_HEADERS = version.h kbg.h kbgboard.h kbgtextview.h kbgstatus.h
 
 SUBDIRS = pics sounds icons engines
@@ -22,9 +25,3 @@
 messages: rc.cpp	
 	LIST=`find . -name \*.h -o -name \*.hh -o -name \*.H -o -name \*.hxx -o -name \*.hpp -o -name \*.cpp -o -name \*.cc -o -name \*.cxx -o -name \*.ecpp -o -name \*.C`; \
 	$(XGETTEXT) $$LIST -o $(podir)/kbackgammon.pot
-
-dummy.cpp:
-	echo > dummy.cpp
-
-install-data-local:
-	rm -rf $(DESTDIR)$(kde_appsdir)/Games/kbackgammon.desktop
--- kbackgammon/engines/Makefile.am	25 Jun 2002 02:34:25 -0000	1.1.1.1
+++ kbackgammon/engines/Makefile.am	25 Jun 2002 02:35:57 -0000	1.2
@@ -3,7 +3,7 @@
 libkbgengines_la_SOURCES = dummy.cpp
 libkbgengines_la_LIBADD  = offline/libkbgoffline.la gnubg/libkbggnubg.la \
                            generic/libkbggeneric.la fibs/libkbgfibs.la   \
-			   nextgen/libkbgnextgen.la
+			   nextgen/libkbgnextgen.la $(LIB_KDECORE) $(LIB_QT)
 
 INCLUDES= $(all_includes)
 
--- kolf/Makefile.am	29 Oct 2002 15:53:13 -0000	1.1.1.3
+++ kolf/Makefile.am	1 Dec 2002 17:02:07 -0000	1.7
@@ -1,17 +1,17 @@
 SUBDIRS = objects sounds courses pics graphics
 INCLUDES = -I$(top_srcdir) -I$(top_srcdir)/libkdegames $(all_includes)
 
-lib_LTLIBRARIES = libkolf.la
+lib_LTLIBRARIES = libkolf_shared.la
 bin_PROGRAMS = kolf
 
-libkolf_la_SOURCES = \
+libkolf_shared_la_SOURCES = \
 	kolf.cpp game.cpp canvasitem.cpp ball.cpp newgame.cpp config.cpp \
 	scoreboard.cpp editor.cpp pluginloader.cpp object.cpp vector.cpp \
 	printdialogpage.cpp kcomboboxdialog.cpp kvolumecontrol.cpp \
 	floater.cpp slope.cpp
 
-libkolf_la_LDFLAGS = $(all_libraries) $(KDE_RPATH) -export-dynamic
-libkolf_la_LIBADD = $(LIB_KDEGAMES) $(LIB_KDEPRINT) $(LIB_KIO) -lartskde
+libkolf_shared_la_LDFLAGS = $(all_libraries) $(KDE_RPATH) -export-dynamic -avoid-version
+libkolf_shared_la_LIBADD = $(LIB_KDEGAMES) $(LIB_KDEPRINT) $(LIB_KIO) -lartskde -lsoundserver_idl -lmcop -lartsflow_idl -lkdeprint
 
 # the installed header files
 kolfinclude_HEADERS = \
@@ -24,17 +24,16 @@
 # kdeinited lib
 kde_module_LTLIBRARIES = kolf.la
 kolf_la_SOURCES = main.cpp
-kolf_la_LDFLAGS = $(KDE_RPATH) $(all_libraries) -module -avoid-version
-kolf_la_LIBADD = libkolf.la
-
+kolf_la_LDFLAGS = $(KDE_RPATH) $(all_libraries) -avoid-version -module
+kolf_la_LIBADD = libkolf_shared.la
 
 # the empty "stub" app that has nothing
-kolf_SOURCES = dummy.cpp
+kolf_SOURCES = kolf_main.cpp
 kolf_LDFLAGS = $(all_libraries) $(KDE_RPATH)
-kolf_LDADD   = kolf.la
+kolf_LDADD   = libkolf_shared.la
 
-dummy.cpp:
-	echo > dummy.cpp
+kolf_main.cpp: main.cpp
+	cat main.cpp > kolf_main.cpp
 
 EXTRA_DIST = kolf.desktop
 
--- kpat/freecell-solver/freecell.c	25 Jun 2002 02:34:26 -0000	1.1.1.1
+++ kpat/freecell-solver/freecell.c	25 Jun 2002 02:35:57 -0000	1.2
@@ -12,7 +12,9 @@
 #include <stdio.h>
 #include <math.h>
 #if FCS_STATE_STORAGE==FCS_STATE_STORAGE_LIBREDBLACK_TREE
+#ifndef __APPLE__
 #include <search.h>
+#endif
 #endif
 
 
--- kpat/freecell-solver/intrface.c	25 Jun 2002 02:34:26 -0000	1.1.1.1
+++ kpat/freecell-solver/intrface.c	25 Jun 2002 02:35:57 -0000	1.2
@@ -16,7 +16,9 @@
 #include <sys/stat.h>
 #include <fcntl.h>
 #if FCS_STATE_STORAGE==FCS_STATE_STORAGE_LIBREDBLACK_TREE
+#ifndef __APPLE__
 #include <search.h>
+#endif
 #endif
 
 
--- kpat/freecell-solver/md5.h	14 Aug 2002 16:24:51 -0000	1.1.1.2
+++ kpat/freecell-solver/md5.h	14 Aug 2002 18:49:24 -0000	1.3
@@ -14,7 +14,11 @@
 #elif defined(__osf__)
 typedef unsigned int u_int32_t;
 #elif !defined(__linux__) && !defined(__NetBSD__) && !defined(__FreeBSD__)
+# ifdef __APPLE__
+# include <sys/types.h>
+# else
 typedef uint32_t u_int32_t;
+# endif
 #endif  
 
 
--- kpat/freecell-solver/scans.c	25 Jun 2002 02:34:26 -0000	1.1.1.1
+++ kpat/freecell-solver/scans.c	25 Jun 2002 02:35:57 -0000	1.2
@@ -13,7 +13,9 @@
 #include <stdio.h>
 #include <math.h>
 #if FCS_STATE_STORAGE==FCS_STATE_STORAGE_LIBREDBLACK_TREE
+#ifndef __APPLE__
 #include <search.h>
+#endif
 #endif
 
 #include "config.h"
--- libksirtet/lib/mp_interface.cpp	10 Jul 2002 04:33:13 -0000	1.1.1.1
+++ libksirtet/lib/mp_interface.cpp	13 Jul 2002 18:11:52 -0000	1.2
@@ -1,5 +1,6 @@
 #include "mp_interface.h"
 #include "mp_interface.moc"
+#include "mp_board.moc"
 
 #include <qpainter.h>
 #include <qlayout.h>
