Index: admin/Makefile.common
===================================================================
RCS file: /Volumes/src/cvs/od/proj/KDE-Darwin/admin/Makefile.common,v
retrieving revision 1.1.1.4
retrieving revision 1.5
diff -u -u -r1.1.1.4 -r1.5
Index: admin/acinclude.m4.in
===================================================================
RCS file: /Volumes/src/cvs/od/proj/KDE-Darwin/admin/acinclude.m4.in,v
retrieving revision 1.1.1.5
retrieving revision 1.18
diff -u -u -r1.1.1.5 -r1.18
--- admin/acinclude.m4.in	29 Oct 2002 02:27:18 -0000	1.1.1.5
+++ admin/acinclude.m4.in	29 Oct 2002 02:29:54 -0000	1.18
@@ -704,13 +704,10 @@
      AC_CHECK_LIB(ipc, shmat, X_EXTRA_LIBS="$X_EXTRA_LIBS -lipc"))
    
    # darwin needs this to initialize the environment
-   AC_CHECK_HEADERS(crt_externs.h)
+   AC_CHECK_HEADERS([crt_externs.h sys/types.h])
    AC_CHECK_FUNC(_NSGetEnviron, [AC_DEFINE(HAVE_NSGETENVIRON, 1, [Define if your system needs _NSGetEnviron to set up the environment])])
  
-   # more headers that need to be explicitly included on darwin
-   AC_CHECK_HEADERS(sys/types.h stdint.h)
-
-   # darwin requires a poll emulation library
+   # darwin requires a poll.h emulation library
    AC_CHECK_LIB(poll, poll, LIB_POLL="-lpoll")
 
    # CoreAudio framework
@@ -719,11 +716,36 @@
      FRAMEWORK_COREAUDIO="-framework CoreAudio"
    ])
 
+   # CoreServices framework
+   AC_CHECK_HEADER(CoreServices/CoreServices.h, [
+     AC_DEFINE(HAVE_CORESERVICES, 1, [Define if you have the CoreServices API])
+     FRAMEWORK_CORESERVICES="-framework CoreServices"
+   ])
+
+   # Solaris 2.6 and others need -lresolv for res_init
+   AC_CHECK_FUNCS(res_init, , [
+     kde_libs_safe="$LIBS"
+     LIBS="$LIBS $X_EXTRA_LIBS -lresolv"
+     AC_TRY_LINK(
+[
+#include <resolv.h>
+],
+[ 
+res_init(); 
+],
+        LIBRESOLV="-lresolv"
+        X_EXTRA_LIBS="$X_EXTRA_LIBS $LIBRESOLV"
+        AC_DEFINE(HAVE_RES_INIT, 1, [Define if you have the res_init function])
+     )
+     LIBS=$kde_libs_safe
+   ])
+
    AC_CHECK_RES_INIT
-   AC_SUBST(LIB_POLL)
    AC_SUBST(FRAMEWORK_COREAUDIO)
    LIBSOCKET="$X_EXTRA_LIBS"
    AC_SUBST(LIBSOCKET)
+   AC_SUBST(FRAMEWORK_CORESERVICES)
+   AC_SUBST(LIB_POLL)
    AC_SUBST(LIBRESOLV)
    AC_SUBST(X_EXTRA_LIBS)
    AC_CHECK_LIB(ucb, killpg, [LIBUCB="-lucb"]) dnl for Solaris2.4
@@ -932,7 +954,7 @@
 kde_ldflags_safe="$LDFLAGS"
 kde_libs_safe="$LIBS"
 
-LDFLAGS="$LDFLAGS $X_LDFLAGS $USER_LDFLAGS"
+LDFLAGS="$LDFLAGS $USER_LDFLAGS $X_LDFLAGS"
 LIBS="-lXext -lX11 $LIBSOCKET"
 
 AC_TRY_LINK([
@@ -974,9 +996,9 @@
 kde_save_LDFLAGS="$LDFLAGS"
 kde_save_CFLAGS="$CFLAGS"
 kde_save_LIBS="$LIBS"
-LDFLAGS="$LDFLAGS $X_LDFLAGS $USER_LDFLAGS"
+LDFLAGS="$LDFLAGS $USER_LDFLAGS $X_LDFLAGS"
 CFLAGS="$CFLAGS -I$x_includes"
-LIBS="-lXinerama -lXext"
+LIBS="-lXinerama -lXext -lX11"
 
 if test "x$no_xinerama" = "xno"; then
 
@@ -1907,15 +1929,29 @@
    AC_REQUIRE([K_PATH_X])
 
 if test $kde_qtver = 3; then
+   AC_SUBST(LIB_ARTSKDE, "-lartskde")
+   AC_SUBST(LIB_DCOP, "-lDCOP")
+   AC_SUBST(LIB_KAB, "-lkab")
+   AC_SUBST(LIB_KABC, "-lkabc")
    AC_SUBST(LIB_KDECORE, "-lkdecore")
+   AC_SUBST(LIB_KDED, "@PREFIX@/lib/kded.la")
+   AC_SUBST(LIB_KDEFX, "-lkdefx")
+   AC_SUBST(LIB_KDEPRINT, "-lkdeprint")
+   AC_SUBST(LIB_KDESASL, "-lkdesasl")
+   AC_SUBST(LIB_KDESU, "-lkdesu")
    AC_SUBST(LIB_KDEUI, "-lkdeui")
-   AC_SUBST(LIB_KIO, "-lkio")
-   AC_SUBST(LIB_SMB, "-lsmb")
-   AC_SUBST(LIB_KAB, "-lkab")
    AC_SUBST(LIB_KHTML, "-lkhtml")
-   AC_SUBST(LIB_KSPELL, "-lkspell")
+   AC_SUBST(LIB_KIO, "-lkio")
    AC_SUBST(LIB_KPARTS, "-lkparts")
-   AC_SUBST(LIB_KDEPRINT, "-lkdeprint")
+   AC_SUBST(LIB_KJAVA, "-lkjava")
+   AC_SUBST(LIB_KJS, "-lkjs")
+   AC_SUBST(LIB_KMID, "-lkmid")
+   AC_SUBST(LIB_KSCRIPT, "-lkscript")
+   AC_SUBST(LIB_KSPELL, "-lkspell")
+   AC_SUBST(LIB_KTEXTEDITOR, "-lktexteditor")
+   AC_SUBST(LIB_SHELLSCRIPT, "-lshellscript")
+   AC_SUBST(LIB_SMB, "-lsmb")
+   AC_SUBST(LIB_VCARD, "-lvcard")
 # these are for backward compatibility
    AC_SUBST(LIB_KSYCOCA, "-lkio")
    AC_SUBST(LIB_KFILE, "-lkio")
@@ -2646,10 +2682,10 @@
       if test $kde_use_debug_code = "full"; then
         CFLAGS="-g3 $CFLAGS"
       else
-        CFLAGS="-g -O2 $CFLAGS"
+        CFLAGS="-g -O $CFLAGS"
       fi
     else
-      CFLAGS="-O2 $CFLAGS"
+      CFLAGS="-Os -frename-registers -mmultiple -mcpu=750 -mtune=750 $CFLAGS"
     fi
   fi
 
@@ -2679,7 +2715,7 @@
         if test "$kde_use_debug_code" = "full"; then
           CXXFLAGS="-g3 $CXXFLAGS"
         else
-          CXXFLAGS="-g -O2 $CXXFLAGS"
+          CXXFLAGS="-g -O $CXXFLAGS"
         fi
       fi
       KDE_CHECK_COMPILER_FLAG(fno-builtin,[CXXFLAGS="-fno-builtin $CXXFLAGS"])
@@ -2691,7 +2727,7 @@
       if test "$CXX" = "KCC"; then
         CXXFLAGS="+K3 $CXXFLAGS"
       else
-        CXXFLAGS="-O2 $CXXFLAGS"
+        CXXFLAGS="-Os -frename-registers -mmultiple -mcpu=750 -mtune=750 $CXXFLAGS"
       fi  
     fi
   fi
@@ -3188,9 +3224,9 @@
     ac_save_ldflags="$LDFLAGS"
     ac_save_cflags="$CFLAGS"
     if test "x$kde_use_qt_emb" != "xyes"; then
-      LDFLAGS="$LDFLAGS $X_LDFLAGS $USER_LDFLAGS $LDFLAGS $XPM_LDFLAGS $all_libraries -lXpm -lX11 -lXext $LIBZ $LIBSOCKET"
+      LDFLAGS="$LDFLAGS $USER_LDFLAGS $X_LDFLAGS $LDFLAGS $XPM_LDFLAGS $all_libraries -lXpm -lX11 -lXext $LIBZ $LIBSOCKET"
     else
-      LDFLAGS="$LDFLAGS $X_LDFLAGS $USER_LDFLAGS $LDFLAGS $XPM_LDFLAGS $all_libraries -lXpm $LIBZ $LIBSOCKET"
+      LDFLAGS="$LDFLAGS $USER_LDFLAGS $X_LDFLAGS $LDFLAGS $XPM_LDFLAGS $all_libraries -lXpm $LIBZ $LIBSOCKET"
     fi
     CFLAGS="$CFLAGS $X_INCLUDES $USER_INCLUDES"
     test -n "$XPM_INCLUDE" && CFLAGS="-I$XPM_INCLUDE $CFLAGS"
@@ -4625,6 +4661,83 @@
 	done
 ])
 
+AC_DEFUN(KDE_CHECK_LDAP,
+[
+  AC_MSG_CHECKING([for LDAP])
+
+  if test -n "$1"; then
+    ldap_extra_dirs="$1"
+  fi
+
+  ldap_incdirs="$ldap_extra_dirs/include /usr/include /usr/local/include/ $kde_extra_includes"
+  AC_FIND_FILE(ldap.h, $ldap_incdirs, ldap_incdir)
+  if test ! -r $ldap_incdir/ldap.h; then
+    ldap_incdir=no
+  fi
+
+  ldap_libdirs="$ldap_extra_dirs/lib /usr/lib /usr/local /usr/lib $kde_extra_libs"
+  for ext in la so sl dylib a; do
+    AC_FIND_FILE(libldap.$ext, $ldap_libdirs, ldap_libdir_test)
+    if test -r $ldap_libdir_test/libldap.$ext; then
+      ldap_libdir=$ldap_libdir_test
+      break
+    fi
+  done
+  if test -z "$ldap_libdir"; then
+    ldap_libdir=no
+  fi
+  if test x$ldap_libdir != xno && test x$ldap_incdir != xno; then
+    LDAP_INCS=-I$ldap_incdir
+    LDAP_LIBS=-L$ldap_libdir
+    # on MacOSX, we link against the framework instead of
+    # directly to the libraries, for forward-compatibility
+    if test -d "/System/Library/Frameworks/LDAP.framework"; then
+      LIBLDAP="-framework LDAP"
+    else
+      LIBLDAP="-lldap -llber -lresolv"
+    fi
+    if test "$USE_RPATH" = "yes"; then
+      LDAP_RPATH="-R $LDAPLIBDIR/lib"
+    fi
+
+    kde_safe_LIBS="$LIBS"
+    kde_safe_CFLAGS="$CFLAGS"
+    LIBS="$LIBS $all_libraries $LDAP_LIBS $LIBLDAP $KERBEROS_LIBS $X_EXTRA_LIBS"
+    CFLAGS="$CFLAGS $all_includes $LDAP_INCS $KERBEROS_INCS"
+    AC_LANG_SAVE
+    AC_LANG_C
+    AC_TRY_LINK(
+      [#include <lber.h>
+#include <ldap.h>
+#if LDAP_VENDOR_VERSION < 20000
+#error OpenLDAP version too old, please upgrade to version 2 or higher
+#endif
+],
+      [ LDAP *ldap; ],
+      [ ldap_old=no ],
+      [ ldap_old=yes ]
+    )
+    LIBS="$kde_safe_LIBS"
+    CFLAGS="$kde_safe_CFLAGS"
+
+    if test x$ldap_old = xno; then
+      AC_SUBST(LDAP_INCS)
+      AC_SUBST(LDAP_LIBS)
+      AC_SUBST(LDAP_RPATH)
+      AC_SUBST(LIBLDAP)
+      AC_DEFINE_UNQUOTED(HAVE_LIBLDAP,1,[Define if you have LDAP libraries])
+    else
+      LDAP_INCS=
+      LDAP_LIBS=
+      LDAP_RPATH=
+      LIBLDAP=
+    fi
+    AC_MSG_RESULT([libraries $ldap_libdir, headers $ldap_incdir])
+  else
+    AC_MSG_RESULT([not found])
+  fi
+])
+
 dnl KDE_CHEC_JAVA_DIR(onlyjre)
 AC_DEFUN(KDE_CHECK_JAVA_DIR,
 [
@@ -4659,6 +4772,7 @@
       KDE_JAVA_PREFIX(/usr/lib/IBMJava2*)
       KDE_JAVA_PREFIX(/usr/lib/IBMJava*)
       KDE_JAVA_PREFIX(/opt/java*)
+      KDE_JAVA_PREFIX(/Library/Java/Home)
     
       kde_cv_path="NONE"
       kde_save_IFS=$IFS
@@ -4710,10 +4824,20 @@
 dnl At this point kde_java_bindir and kde_java_includedir are either set or "no"
 if test "x$kde_java_bindir" != "xno"; then
 
+  case $host_os in
+    darwin*)
+      dnl Look for libjvm.dylib
+      kde_java_libjvmdir=`find $kde_java_bindir/.. -name libjvm.dylib | sed 's,libjvm.dylib,,'|head -n 1`
+      kd_java_libhpidir=''
+      dnl Darwin doesn't have libhpi
+      ;;
+    *)
   dnl Look for libjvm.so
   kde_java_libjvmdir=`find $kde_java_bindir/.. -name libjvm.so | sed 's,libjvm.so,,'|head -n 1`
   dnl Look for libhpi.so and avoid green threads
-  kde_java_libhpidir=`find $kde_java_bindir/.. -name libhpi.so | grep -v green | sed 's,libhpi.so,,' | head -n 1`
+  kde_java_libhpidir=`find $kde_java_bindir/.. -name libhpi.so | grep -v green | sed 's,libhpi.so,,' | tail -n 1`
+      ;;
+  esac
 
   dnl Now check everything's fine under there
   dnl the include dir is our flag for having the JDK
@@ -4742,17 +4866,29 @@
     JAVAC=
     jni_includes=
   fi
-
+  case $host_os in
+    darwin*)
+      if test ! -r "$kde_java_libjvmdir/libjvm.dylib"; then
+        AC_MSG_ERROR([libjvm.dylib not found under $kde_java_libjvmdir. Use --without-java.])
+      fi
+      dnl Darwin doesn't have libhpi, but it's not fatal
+      LIB_JVM="-framework JavaVM"
+      LIB_HPI=""
+      ;;
+    *)
   if test ! -r "$kde_java_libjvmdir/libjvm.so"; then
      AC_MSG_ERROR([libjvm.so not found under $kde_java_libjvmdir. Use --without-java.])
   fi 
+      LIB_JVM="-L$kde_java_libjvmdir -ljvm"
+      if test ! -r "$kde_java_libhpidir/libhpi.so"; then
+        AC_MSG_ERROR([libhpi.so not found under $kde_java_libhpidir. Use --without-java.])
+      fi
+      LIB_HPI="-L$kde_java_libhpidir -lhpi"
+      ;;
+  esac
 
   if test ! -x "$kde_java_bindir/java"; then
       AC_MSG_ERROR([java not found under $kde_java_bindir. javac was found though! Use --with-java or --without-java.])
-  fi
-
-  if test ! -r "$kde_java_libhpidir/libhpi.so"; then
-    AC_MSG_ERROR([libhpi.so not found under $kde_java_libhpidir. Use --without-java.])
   fi
 
   if test -n "$jni_includes"; then
Index: admin/detect-autoconf.sh
===================================================================
RCS file: /Volumes/src/cvs/od/proj/KDE-Darwin/admin/detect-autoconf.sh,v
retrieving revision 1.1.1.2
retrieving revision 1.3
diff -u -u -r1.1.1.2 -r1.3
--- admin/detect-autoconf.sh	29 Oct 2002 02:27:20 -0000	1.1.1.2
+++ admin/detect-autoconf.sh	29 Oct 2002 02:29:54 -0000	1.3
@@ -42,12 +42,12 @@
 checkAutomakeAclocal ()
 {
   if test -z "$UNSERMAKE"; then
-    if test -x "`$WHICH automake-1.5`" ; then
-      AUTOMAKE="`$WHICH automake-1.5`"
-      ACLOCAL="`$WHICH aclocal-1.5`"
-    elif test -x "`$WHICH automake-1.6`" ; then
+    if test -x "`$WHICH automake-1.6`" ; then
       AUTOMAKE="`$WHICH automake-1.6`"
       ACLOCAL="`$WHICH aclocal-1.6`"
+    elif test -x "`$WHICH automake-1.5`" ; then
+      AUTOMAKE="`$WHICH automake-1.5`"
+      ACLOCAL="`$WHICH aclocal-1.5`"
     fi
   else
      AUTOMAKE="$UNSERMAKE"
Index: admin/libtool.m4.in
===================================================================
RCS file: /Volumes/src/cvs/od/proj/KDE-Darwin/admin/libtool.m4.in,v
retrieving revision 1.1.1.3
retrieving revision 1.7
diff -u -u -r1.1.1.3 -r1.7
--- admin/libtool.m4.in	14 Aug 2002 13:18:45 -0000	1.1.1.3
+++ admin/libtool.m4.in	16 Sep 2002 14:31:34 -0000	1.7
@@ -609,6 +609,12 @@
     lt_cv_sys_max_cmd_len=-1;
     ;;
 
+  darwin*)
+    # this should be enough... the length test freaks out some versions
+    # of OSX's `expr`
+    lt_cv_sys_max_cmd_len=16384;
+    ;;
+
   *)
     # If test is not a shell built-in, we'll probably end up computing a
     # maximum length that is only half of the actual maximum length, but
@@ -1132,7 +1138,7 @@
 darwin* | rhapsody*)
   dynamic_linker="$host_os dyld"
   version_type=darwin
-  need_lib_prefix=no
+  need_lib_prefix=yes
   need_version=no
   # FIXME: Relying on posixy $() will cause problems for
   #        cross-compilation, but unfortunately the echo tests do not
@@ -1901,16 +1907,7 @@
   ;;
 
 darwin* | rhapsody*)
-  lt_cv_deplibs_check_method='file_magic Mach-O dynamically linked shared library'
-  lt_cv_file_magic_cmd='/usr/bin/file -L'
-  case "$host_os" in
-  rhapsody* | darwin1.[[012]])
-    lt_cv_file_magic_test_file=`/System/Library/Frameworks/System.framework/System`
-    ;;
-  *) # Darwin 1.3 on
-    lt_cv_file_magic_test_file='/usr/lib/libSystem.dylib'
-    ;;
-  esac
+  lt_cv_deplibs_check_method='pass_all'
   ;;
 
 freebsd*)
@@ -2560,6 +2557,11 @@
 	;;
     esac
     ;;
+  darwin*)
+    shared_flag='-dynamiclib'
+    _LT_AC_TAGVAR(archive_cmds, $1)='$CC -r -keep_private_externs -nostdlib -o ${lib}-master.o $libobjs && $CC -dynamiclib -install_name $rpath/$soname $predep_objects ${lib}-master.o $deplibs $postdep_objects $compiler_flags -o $lib'
+    output_verbose_link_cmd='$CC -dynamiclib $CFLAGS -v conftest.$objext 2>&1 | egrep "\-L"'
+    ;;
   dgux*)
     case $cc_basename in
       ec++)
@@ -4750,7 +4752,9 @@
 	_LT_AC_TAGVAR(allow_undefined_flag, $1)='-undefined suppress'
 	;;
       *) # Darwin 1.3 on
-	_LT_AC_TAGVAR(allow_undefined_flag, $1)='-flat_namespace -undefined suppress'
+	#_LT_AC_TAGVAR(allow_undefined_flag, $1)='-flat_namespace -undefined suppress'
+        # testing with twolevel namespaces
+	_LT_AC_TAGVAR(allow_undefined_flag, $1)=unsupported
 	;;
       esac
 
@@ -4758,7 +4762,7 @@
       #        cross-compilation, but unfortunately the echo tests do not
       #        yet detect zsh echo's removal of \ escapes.  Also zsh mangles
       #	       `"' quotes if we put them in here... so don't!
-      _LT_AC_TAGVAR(archive_cmds, $1)='$CC $(test .$module = .yes && echo -bundle || echo -dynamiclib) $allow_undefined_flag -o $lib $libobjs $deplibs$linker_flags -install_name $rpath/$soname $verstring'
+      _LT_AC_TAGVAR(archive_cmds, $1)='$CC -r -keep_private_externs -nostdlib -o ${lib}-master.o $libobjs && $CC $(test .$module = .yes && echo -bundle || echo -dynamiclib) $allow_undefined_flag -o $lib ${lib}-master.o $deplibs $linker_flags $(test x$module != xyes && echo -install_name $rpath/$soname $verstring)'
       # We need to add '_' to the symbols in $export_symbols first
       #_LT_AC_TAGVAR(archive_expsym_cmds, $1)="$_LT_AC_TAGVAR(archive_cmds, $1)"' && strip -s $export_symbols'
       _LT_AC_TAGVAR(hardcode_direct, $1)=yes
Index: admin/ltmain.sh
===================================================================
RCS file: /Volumes/src/cvs/od/proj/KDE-Darwin/admin/ltmain.sh,v
retrieving revision 1.1.1.2
retrieving revision 1.4
diff -u -u -r1.1.1.2 -r1.4
--- admin/ltmain.sh	10 Jul 2002 02:47:39 -0000	1.1.1.2
+++ admin/ltmain.sh	13 Jul 2002 04:33:25 -0000	1.4
@@ -3321,7 +3321,23 @@
 	if test -n "$export_symbols" && test -n "$archive_expsym_cmds"; then
 	  eval cmds=\"$archive_expsym_cmds\"
 	else
+          if test "x$verstring" = "x0.0"; then
+                tmp_verstring=
+          else
+                tmp_verstring="$verstring"
+          fi
+ 	  save_deplibs="$deplibs"
+ 	  for conv in $convenience; do
+            tmp_deplibs=
+            for test_deplib in $deplibs; do
+              if test "$test_deplib" != "$conv"; then
+                tmp_deplibs="$tmp_deplibs $test_deplib"
+              fi
+            done
+            deplibs="$tmp_deplibs"
+ 	  done
 	  eval cmds=\"$archive_cmds\"
+ 	  deplibs="$save_deplibs"
 	fi
 
 	if len=`expr "X$cmds" : ".*"` &&
@@ -3419,7 +3435,12 @@
 	  if test -n "$export_symbols" && test -n "$archive_expsym_cmds"; then
 	    eval cmds=\"$archive_expsym_cmds\"
 	  else
-	    eval cmds=\"$archive_cmds\"
+            save_deplibs="$deplibs"
+            for conv in $convenience; do
+              deplibs="${deplibs%$conv*} ${deplibs#*$conv}"
+            done
+            eval cmds=\"$archive_cmds\"
+            deplibs="$save_deplibs"
 	  fi
 
 	  # Append the command to remove the reloadable object files
@@ -3632,6 +3653,14 @@
 	compile_deplibs=`$echo "X $compile_deplibs" | $Xsed -e 's/ -lc / -framework System /'`
 	finalize_deplibs=`$echo "X $finalize_deplibs" | $Xsed -e 's/ -lc / -framework System /'`
 	;;
+      esac
+
+      case $host in
+      *darwin*)
+        # Don't allow lazy linking, it breaks C++ global constructors
+        compile_command="$compile_command ${wl}-bind_at_load"
+        finalize_command="$finalize_command ${wl}-bind_at_load"
+        ;;
       esac
 
       compile_command="$compile_command $compile_deplibs"
Index: admin/missing
===================================================================
RCS file: /Volumes/src/cvs/od/proj/KDE-Darwin/admin/missing,v
retrieving revision 1.1.1.3
retrieving revision 1.5
diff -u -u -r1.1.1.3 -r1.5
Index: debian/kasteroids.install
===================================================================
RCS file: debian/kasteroids.install
diff -N debian/kasteroids.install
--- debian/kasteroids.install	29 Oct 2002 15:52:04 -0000	1.1.1.2
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,10 +0,0 @@
-debian/tmp/usr/games/kasteroids
-debian/tmp/usr/share/man/man6/kasteroids*
-debian/tmp/usr/share/icons/hicolor/48x48/apps/kasteroids.png
-debian/tmp/usr/share/icons/hicolor/32x32/apps/kasteroids.png
-debian/tmp/usr/share/icons/hicolor/16x16/apps/kasteroids.png
-debian/tmp/usr/share/applnk/Games/Arcade/kasteroids.desktop
-debian/tmp/usr/share/apps/kasteroids/*
-debian/tmp/usr/share/doc/kdegames/kasteroids/*
-debian/tmp/usr/share/doc/kde/HTML/en/kasteroids/*
-debian/icons/kasteroids.xpm /usr/share/pixmaps
Index: debian/katomic.install
===================================================================
RCS file: debian/katomic.install
diff -N debian/katomic.install
--- debian/katomic.install	29 Oct 2002 15:52:04 -0000	1.1.1.2
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,8 +0,0 @@
-debian/tmp/usr/share/apps/katomic/*
-debian/tmp/usr/share/icons/hicolor/32x32/apps/katomic.png
-debian/tmp/usr/share/applnk/Games/TacticStrategy/katomic.desktop
-debian/tmp/usr/games/katomic
-debian/tmp/usr/share/man/man6/katomic*
-debian/tmp/usr/share/doc/kdegames/katomic/*
-debian/tmp/usr/share/doc/kde/HTML/en/katomic/*
-debian/icons/katomic.xpm /usr/share/pixmaps
Index: debian/kbackgammon.install
===================================================================
RCS file: debian/kbackgammon.install
diff -N debian/kbackgammon.install
--- debian/kbackgammon.install	29 Oct 2002 15:52:04 -0000	1.1.1.2
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,14 +0,0 @@
-debian/tmp/usr/games/kbackgammon
-debian/tmp/usr/lib/kbackgammon.la
-debian/tmp/usr/lib/kbackgammon.so
-debian/tmp/usr/share/man/man6/kbackgammon*
-debian/tmp/usr/share/applnk/Games/Board/kbackgammon.desktop
-debian/tmp/usr/share/apps/kbackgammon/*
-debian/tmp/usr/share/icons/hicolor/48x48/apps/kbackgammon.png
-debian/tmp/usr/share/icons/hicolor/32x32/apps/kbackgammon.png
-debian/tmp/usr/share/icons/hicolor/16x16/apps/kbackgammon.png
-debian/tmp/usr/share/icons/hicolor/48x48/apps/kbackgammon_engine.png
-debian/tmp/usr/share/icons/hicolor/32x32/apps/kbackgammon_engine.png
-debian/tmp/usr/share/icons/hicolor/16x16/apps/kbackgammon_engine.png
-debian/tmp/usr/share/doc/kde/HTML/en/kbackgammon/*
-debian/icons/kbackgammon.xpm /usr/share/pixmaps
Index: debian/kbattleship.install
===================================================================
RCS file: debian/kbattleship.install
diff -N debian/kbattleship.install
--- debian/kbattleship.install	29 Oct 2002 15:52:04 -0000	1.1.1.2
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,7 +0,0 @@
-debian/tmp/usr/share/apps/kbattleship/*
-debian/tmp/usr/share/icons/hicolor/32x32/apps/kbattleship.png
-debian/tmp/usr/share/icons/hicolor/16x16/apps/kbattleship.png
-debian/tmp/usr/share/applnk/Games/Board/kbattleship.desktop
-debian/tmp/usr/games/kbattleship
-debian/tmp/usr/share/doc/kde/HTML/en/kbattleship/*
-debian/icons/kbattleship.xpm /usr/share/pixmaps
Index: debian/kblackbox.install
===================================================================
RCS file: debian/kblackbox.install
diff -N debian/kblackbox.install
--- debian/kblackbox.install	29 Oct 2002 15:52:04 -0000	1.1.1.2
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,10 +0,0 @@
-debian/tmp/usr/share/apps/kblackbox/*
-debian/tmp/usr/share/icons/hicolor/48x48/apps/kblackbox.png
-debian/tmp/usr/share/icons/hicolor/32x32/apps/kblackbox.png
-debian/tmp/usr/share/icons/hicolor/16x16/apps/kblackbox.png
-debian/tmp/usr/share/applnk/Games/Board/kblackbox.desktop
-debian/tmp/usr/games/kblackbox
-debian/tmp/usr/share/man/man6/kblackbox*
-debian/tmp/usr/share/doc/kdegames/kblackbox/*
-debian/tmp/usr/share/doc/kde/HTML/en/kblackbox/*
-debian/icons/kblackbox.xpm /usr/share/pixmaps
Index: debian/kbounce.install
===================================================================
RCS file: debian/kbounce.install
diff -N debian/kbounce.install
--- debian/kbounce.install	29 Oct 2002 15:52:04 -0000	1.1.1.2
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,10 +0,0 @@
-debian/tmp/usr/share/apps/kbounce/*
-debian/tmp/usr/share/applnk/Games/Arcade/kbounce.desktop
-debian/tmp/usr/share/icons/locolor/16x16/apps/kbounce.png
-debian/tmp/usr/share/icons/locolor/32x32/apps/kbounce.png
-debian/tmp/usr/share/icons/hicolor/16x16/apps/kbounce.png
-debian/tmp/usr/share/icons/hicolor/32x32/apps/kbounce.png
-debian/tmp/usr/games/kbounce
-debian/tmp/usr/share/man/man6/kbounce*
-debian/tmp/usr/share/doc/kde/HTML/en/kbounce/*
-debian/icons/kbounce.xpm /usr/share/pixmaps
Index: debian/kdecarddecks.install
===================================================================
RCS file: debian/kdecarddecks.install
diff -N debian/kdecarddecks.install
--- debian/kdecarddecks.install	25 Jun 2002 02:34:24 -0000	1.1.1.1
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1 +0,0 @@
-debian/tmp/usr/share/apps/carddecks/*
Index: debian/kenolaba.install
===================================================================
RCS file: debian/kenolaba.install
diff -N debian/kenolaba.install
--- debian/kenolaba.install	29 Oct 2002 15:52:04 -0000	1.1.1.2
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,8 +0,0 @@
-debian/tmp/usr/share/apps/kenolaba/*
-debian/tmp/usr/share/icons/hicolor/16x16/apps/kenolaba.png
-debian/tmp/usr/share/icons/hicolor/32x32/apps/kenolaba.png
-debian/tmp/usr/share/icons/hicolor/48x48/apps/kenolaba.png
-debian/tmp/usr/share/applnk/Games/Board/kenolaba.desktop
-debian/tmp/usr/games/kenolaba
-debian/tmp/usr/share/doc/kde/HTML/en/kenolaba/*
-debian/icons/kenolaba.xpm /usr/share/pixmaps
Index: debian/kjumpingcube.install
===================================================================
RCS file: debian/kjumpingcube.install
diff -N debian/kjumpingcube.install
--- debian/kjumpingcube.install	29 Oct 2002 15:52:04 -0000	1.1.1.2
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,9 +0,0 @@
-debian/tmp/usr/games/kjumpingcube
-debian/tmp/usr/share/icons/hicolor/32x32/apps/kjumpingcube.png
-debian/tmp/usr/share/icons/hicolor/16x16/apps/kjumpingcube.png
-debian/tmp/usr/share/man/man6/kjumpingcube*
-debian/tmp/usr/share/apps/kjumpingcube/*
-debian/tmp/usr/share/applnk/Games/TacticStrategy/kjumpingcube.desktop
-debian/tmp/usr/share/doc/kdegames/kjumpingcube/*
-debian/tmp/usr/share/doc/kde/HTML/en/kjumpingcube/*
-debian/icons/kjumpingcube.xpm /usr/share/pixmaps
Index: debian/klines.install
===================================================================
RCS file: debian/klines.install
diff -N debian/klines.install
--- debian/klines.install	29 Oct 2002 15:52:04 -0000	1.1.1.2
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,7 +0,0 @@
-debian/tmp/usr/share/apps/klines/*
-debian/tmp/usr/share/applnk/Games/TacticStrategy/klines.desktop
-debian/tmp/usr/games/klines
-debian/tmp/usr/share/doc/kde/HTML/en/klines/*
-debian/tmp/usr/share/icons/hicolor/16x16/apps/klines.png
-debian/tmp/usr/share/icons/hicolor/32x32/apps/klines.png
-debian/icons/klines.xpm /usr/share/pixmaps
Index: debian/kmahjongg.install
===================================================================
RCS file: debian/kmahjongg.install
diff -N debian/kmahjongg.install
--- debian/kmahjongg.install	29 Oct 2002 15:52:04 -0000	1.1.1.2
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,10 +0,0 @@
-debian/tmp/usr/share/apps/kmahjongg/*
-debian/tmp/usr/share/icons/hicolor/48x48/apps/kmahjongg.png
-debian/tmp/usr/share/icons/hicolor/32x32/apps/kmahjongg.png
-debian/tmp/usr/share/icons/hicolor/16x16/apps/kmahjongg.png
-debian/tmp/usr/share/applnk/Games/Board/kmahjongg.desktop
-debian/tmp/usr/games/kmahjongg
-debian/tmp/usr/share/man/man6/kmahjongg*
-debian/tmp/usr/share/doc/kdegames/kmahjongg/*
-debian/tmp/usr/share/doc/kde/HTML/en/kmahjongg/*
-debian/icons/kmahjongg.xpm /usr/share/pixmaps
Index: debian/kmines.install
===================================================================
RCS file: debian/kmines.install
diff -N debian/kmines.install
--- debian/kmines.install	29 Oct 2002 15:52:04 -0000	1.1.1.2
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,10 +0,0 @@
-debian/tmp/usr/share/icons/hicolor/48x48/apps/kmines.png
-debian/tmp/usr/share/icons/hicolor/32x32/apps/kmines.png
-debian/tmp/usr/share/icons/hicolor/16x16/apps/kmines.png
-debian/tmp/usr/share/applnk/Games/TacticStrategy/kmines.desktop
-debian/tmp/usr/share/apps/kmines/*
-debian/tmp/usr/games/kmines
-debian/tmp/usr/share/man/man6/kmines*
-debian/tmp/usr/share/doc/kdegames/kmines/*
-debian/tmp/usr/share/doc/kde/HTML/en/kmines/*
-debian/icons/kmines.xpm /usr/share/pixmaps
Index: debian/konquest.install
===================================================================
RCS file: debian/konquest.install
diff -N debian/konquest.install
--- debian/konquest.install	29 Oct 2002 15:52:04 -0000	1.1.1.2
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,10 +0,0 @@
-debian/tmp/usr/share/apps/konquest/*
-debian/tmp/usr/share/icons/hicolor/48x48/apps/konquest.png
-debian/tmp/usr/share/icons/hicolor/32x32/apps/konquest.png
-debian/tmp/usr/share/icons/hicolor/16x16/apps/konquest.png
-debian/tmp/usr/share/applnk/Games/TacticStrategy/konquest.desktop
-debian/tmp/usr/games/konquest
-debian/tmp/usr/share/man/man6/konquest*
-debian/tmp/usr/share/doc/kdegames/konquest/*
-debian/tmp/usr/share/doc/kde/HTML/en/konquest/*
-debian/icons/konquest.xpm /usr/share/pixmaps
Index: debian/kpat.install
===================================================================
RCS file: debian/kpat.install
diff -N debian/kpat.install
--- debian/kpat.install	29 Oct 2002 15:52:04 -0000	1.1.1.2
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,10 +0,0 @@
-debian/tmp/usr/share/icons/hicolor/48x48/apps/kpat.png
-debian/tmp/usr/share/icons/hicolor/32x32/apps/kpat.png
-debian/tmp/usr/share/icons/hicolor/16x16/apps/kpat.png
-debian/tmp/usr/share/apps/kpat/*
-debian/tmp/usr/share/applnk/Games/Card/kpat.desktop
-debian/tmp/usr/games/kpat
-debian/tmp/usr/share/man/man6/kpat*
-debian/tmp/usr/share/doc/kdegames/kpat/*
-debian/tmp/usr/share/doc/kde/HTML/en/kpat/*
-debian/icons/kpat.xpm /usr/share/pixmaps
Index: debian/kpoker.install
===================================================================
RCS file: debian/kpoker.install
diff -N debian/kpoker.install
--- debian/kpoker.install	29 Oct 2002 15:52:04 -0000	1.1.1.2
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,10 +0,0 @@
-debian/tmp/usr/share/apps/kpoker/*
-debian/tmp/usr/share/icons/hicolor/48x48/apps/kpoker.png
-debian/tmp/usr/share/icons/hicolor/32x32/apps/kpoker.png
-debian/tmp/usr/share/icons/hicolor/16x16/apps/kpoker.png
-debian/tmp/usr/share/applnk/Games/Card/kpoker.desktop
-debian/tmp/usr/games/kpoker
-debian/tmp/usr/share/man/man6/kpoker*
-debian/tmp/usr/share/doc/kdegames/kpoker/*
-debian/tmp/usr/share/doc/kde/HTML/en/kpoker/*
-debian/icons/kpoker.xpm /usr/share/pixmaps
Index: debian/kreversi.install
===================================================================
RCS file: debian/kreversi.install
diff -N debian/kreversi.install
--- debian/kreversi.install	29 Oct 2002 15:52:04 -0000	1.1.1.2
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,10 +0,0 @@
-debian/tmp/usr/games/kreversi
-debian/tmp/usr/share/man/man6/kreversi*
-debian/tmp/usr/share/doc/kdegames/kreversi/*
-debian/tmp/usr/share/doc/kde/HTML/en/kreversi/*
-debian/tmp/usr/share/applnk/Games/Board/kreversi.desktop
-debian/tmp/usr/share/apps/kreversi/*
-debian/tmp/usr/share/icons/hicolor/48x48/apps/kreversi.png
-debian/tmp/usr/share/icons/hicolor/32x32/apps/kreversi.png
-debian/tmp/usr/share/icons/hicolor/16x16/apps/kreversi.png
-debian/icons/kreversi.xpm /usr/share/pixmaps
Index: debian/ksame.install
===================================================================
RCS file: debian/ksame.install
diff -N debian/ksame.install
--- debian/ksame.install	29 Oct 2002 15:52:04 -0000	1.1.1.2
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,10 +0,0 @@
-debian/tmp/usr/games/ksame
-debian/tmp/usr/share/man/man6/ksame*
-debian/tmp/usr/share/doc/kdegames/ksame/*
-debian/tmp/usr/share/doc/kde/HTML/en/ksame/*
-debian/tmp/usr/share/icons/hicolor/48x48/apps/ksame.png
-debian/tmp/usr/share/icons/hicolor/32x32/apps/ksame.png
-debian/tmp/usr/share/icons/hicolor/16x16/apps/ksame.png
-debian/tmp/usr/share/apps/ksame/*
-debian/tmp/usr/share/applnk/Games/TacticStrategy/ksame.desktop
-debian/icons/ksame.xpm /usr/share/pixmaps
Index: debian/kshisen.install
===================================================================
RCS file: debian/kshisen.install
diff -N debian/kshisen.install
--- debian/kshisen.install	29 Oct 2002 15:52:04 -0000	1.1.1.2
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,10 +0,0 @@
-debian/tmp/usr/share/apps/kshisen/*
-debian/tmp/usr/share/icons/hicolor/48x48/apps/kshisen.png
-debian/tmp/usr/share/icons/hicolor/32x32/apps/kshisen.png
-debian/tmp/usr/share/icons/hicolor/16x16/apps/kshisen.png
-debian/tmp/usr/share/applnk/Games/Board/kshisen.desktop
-debian/tmp/usr/games/kshisen
-debian/tmp/usr/share/man/man6/kshisen*
-debian/tmp/usr/share/doc/kdegames/kshisen/*
-debian/tmp/usr/share/doc/kde/HTML/en/kshisen/*
-debian/icons/kshisen.xpm /usr/share/pixmaps
Index: debian/ksirtet.install
===================================================================
RCS file: debian/ksirtet.install
diff -N debian/ksirtet.install
--- debian/ksirtet.install	29 Oct 2002 15:52:04 -0000	1.1.1.2
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,20 +0,0 @@
-debian/tmp/usr/share/applnk/Games/Arcade/ksirtet.desktop
-debian/tmp/usr/share/applnk/Games/Arcade/kfouleggs.desktop
-debian/tmp/usr/share/icons/hicolor/48x48/apps/ksirtet.png
-debian/tmp/usr/share/icons/hicolor/32x32/apps/ksirtet.png
-debian/tmp/usr/share/icons/hicolor/16x16/apps/ksirtet.png
-debian/tmp/usr/share/apps/kfouleggs/*
-debian/tmp/usr/share/apps/ksirtet/*
-debian/tmp/usr/games/ksirtet
-debian/tmp/usr/games/kfouleggs
-debian/tmp/usr/share/man/man6/ksirtet*
-debian/tmp/usr/share/man/man6/kfouleggs*
-debian/tmp/usr/share/doc/kdegames/ksirtet/*
-debian/tmp/usr/share/doc/kde/HTML/en/ksirtet/*
-debian/tmp/usr/share/doc/kdegames/kfouleggs/*
-debian/tmp/usr/share/doc/kde/HTML/en/kfouleggs/*
-debian/tmp/usr/lib/libksirtetbase.so.0.0.1
-debian/tmp/usr/lib/libksirtetbase.so.0
-debian/tmp/usr/lib/libksirtetcommon.so.0.0.1
-debian/tmp/usr/lib/libksirtetcommon.so.0
-debian/icons/ksirtet.xpm /usr/share/pixmaps
Index: debian/ksmiletris.install
===================================================================
RCS file: debian/ksmiletris.install
diff -N debian/ksmiletris.install
--- debian/ksmiletris.install	29 Oct 2002 15:52:04 -0000	1.1.1.2
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,10 +0,0 @@
-debian/tmp/usr/share/apps/ksmiletris/*
-debian/tmp/usr/share/icons/hicolor/48x48/apps/ksmiletris.png
-debian/tmp/usr/share/icons/hicolor/32x32/apps/ksmiletris.png
-debian/tmp/usr/share/icons/hicolor/16x16/apps/ksmiletris.png
-debian/tmp/usr/share/applnk/Games/Arcade/ksmiletris.desktop
-debian/tmp/usr/games/ksmiletris
-debian/tmp/usr/share/man/man6/ksmiletris*
-debian/tmp/usr/share/doc/kdegames/ksmiletris/*
-debian/tmp/usr/share/doc/kde/HTML/en/ksmiletris/*
-debian/icons/ksmiletris.xpm /usr/share/pixmaps
Index: debian/ksnake.install
===================================================================
RCS file: debian/ksnake.install
diff -N debian/ksnake.install
--- debian/ksnake.install	29 Oct 2002 15:52:04 -0000	1.1.1.2
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,10 +0,0 @@
-debian/tmp/usr/share/apps/ksnake/*
-debian/tmp/usr/share/icons/hicolor/48x48/apps/ksnake.png
-debian/tmp/usr/share/icons/hicolor/32x32/apps/ksnake.png
-debian/tmp/usr/share/icons/hicolor/16x16/apps/ksnake.png
-debian/tmp/usr/share/applnk/Games/Arcade/ksnake.desktop
-debian/tmp/usr/games/ksnake
-debian/tmp/usr/share/man/man6/ksnake*
-debian/tmp/usr/share/doc/kdegames/ksnake/*
-debian/tmp/usr/share/doc/kde/HTML/en/ksnake/*
-debian/icons/ksnake.xpm /usr/share/pixmaps
Index: debian/ksokoban.install
===================================================================
RCS file: debian/ksokoban.install
diff -N debian/ksokoban.install
--- debian/ksokoban.install	29 Oct 2002 15:52:04 -0000	1.1.1.2
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,9 +0,0 @@
-debian/tmp/usr/share/applnk/Games/TacticStrategy/ksokoban.desktop
-debian/tmp/usr/share/icons/hicolor/48x48/apps/ksokoban.png
-debian/tmp/usr/share/icons/hicolor/32x32/apps/ksokoban.png
-debian/tmp/usr/share/icons/hicolor/16x16/apps/ksokoban.png
-debian/tmp/usr/games/ksokoban
-debian/tmp/usr/share/man/man6/ksokoban*
-debian/tmp/usr/share/doc/kdegames/ksokoban/*
-debian/tmp/usr/share/doc/kde/HTML/en/ksokoban/*
-debian/icons/ksokoban.xpm /usr/share/pixmaps
Index: debian/kspaceduel.install
===================================================================
RCS file: debian/kspaceduel.install
diff -N debian/kspaceduel.install
--- debian/kspaceduel.install	29 Oct 2002 15:52:04 -0000	1.1.1.2
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,11 +0,0 @@
-debian/tmp/usr/share/apps/kspaceduel/*
-debian/tmp/usr/share/icons/hicolor/32x32/apps/kspaceduel.png
-debian/tmp/usr/share/icons/hicolor/16x16/apps/kspaceduel.png
-debian/tmp/usr/share/icons/locolor/32x32/apps/kspaceduel.png
-debian/tmp/usr/share/icons/locolor/16x16/apps/kspaceduel.png
-debian/tmp/usr/share/applnk/Games/Arcade/kspaceduel.desktop
-debian/tmp/usr/games/kspaceduel
-debian/tmp/usr/share/man/man6/kspaceduel*
-debian/tmp/usr/share/doc/kdegames/kspaceduel/*
-debian/tmp/usr/share/doc/kde/HTML/en/kspaceduel/*
-debian/icons/kspaceduel.xpm /usr/share/pixmaps
Index: debian/ktron.install
===================================================================
RCS file: debian/ktron.install
diff -N debian/ktron.install
--- debian/ktron.install	29 Oct 2002 15:52:04 -0000	1.1.1.2
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,9 +0,0 @@
-debian/tmp/usr/games/ktron
-debian/tmp/usr/share/icons/hicolor/32x32/apps/ktron.png
-debian/tmp/usr/share/icons/hicolor/16x16/apps/ktron.png
-debian/tmp/usr/share/man/man6/ktron*
-debian/tmp/usr/share/doc/kdegames/ktron/*
-debian/tmp/usr/share/doc/kde/HTML/en/ktron/*
-debian/tmp/usr/share/applnk/Games/Arcade/ktron.desktop
-debian/tmp/usr/share/apps/ktron/*
-debian/icons/ktron.xpm /usr/share/pixmaps
Index: debian/ktuberling.install
===================================================================
RCS file: debian/ktuberling.install
diff -N debian/ktuberling.install
--- debian/ktuberling.install	29 Oct 2002 15:52:04 -0000	1.1.1.2
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,8 +0,0 @@
-debian/tmp/usr/games/ktuberling
-debian/tmp/usr/share/icons/hicolor/32x32/apps/ktuberling.png
-debian/tmp/usr/share/man/man6/ktuberling*
-debian/tmp/usr/share/doc/kdegames/ktuberling/*
-debian/tmp/usr/share/doc/kde/HTML/en/ktuberling/*
-debian/tmp/usr/share/applnk/Games/Kidsgames/ktuberling.desktop
-debian/tmp/usr/share/apps/ktuberling/*
-debian/icons/ktuberling.xpm /usr/share/pixmaps
Index: debian/kwin4.install
===================================================================
RCS file: debian/kwin4.install
diff -N debian/kwin4.install
--- debian/kwin4.install	29 Oct 2002 15:52:04 -0000	1.1.1.2
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,11 +0,0 @@
-debian/tmp/usr/games/kwin4
-debian/tmp/usr/games/kwin4proc
-debian/tmp/usr/share/man/man6/kwin4*
-debian/tmp/usr/share/man/man6/kproc4*
-debian/tmp/usr/share/apps/kwin4/*
-debian/tmp/usr/share/icons/hicolor/48x48/apps/kwin4.png
-debian/tmp/usr/share/icons/hicolor/32x32/apps/kwin4.png
-debian/tmp/usr/share/icons/hicolor/16x16/apps/kwin4.png
-debian/tmp/usr/share/applnk/Games/Board/kwin4.desktop
-debian/tmp/usr/share/doc/kde/HTML/en/kwin4/*
-debian/icons/kwin4.xpm /usr/share/pixmaps
Index: debian/libkdegames-dev.install
===================================================================
RCS file: debian/libkdegames-dev.install
diff -N debian/libkdegames-dev.install
--- debian/libkdegames-dev.install	25 Jun 2002 02:34:24 -0000	1.1.1.1
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,7 +0,0 @@
-debian/tmp/usr/include/kde
-debian/tmp/usr/lib/libkdegames.a
-debian/tmp/usr/lib/libkdegames.so
-debian/tmp/usr/lib/libkdegames.la
-debian/tmp/usr/lib/libkdehighscores.a
-debian/tmp/usr/lib/libkdehighscores.so
-debian/tmp/usr/lib/libkdehighscores.la
Index: debian/libkdegames1.install
===================================================================
RCS file: debian/libkdegames1.install
diff -N debian/libkdegames1.install
--- debian/libkdegames1.install	29 Oct 2002 15:52:04 -0000	1.1.1.2
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,10 +0,0 @@
-debian/tmp/usr/lib/libkdegames.so.1.0.1
-debian/tmp/usr/lib/libkdegames.so.1
-debian/tmp/usr/lib/libkdehighscores.so.1.0.0
-debian/tmp/usr/lib/libkdehighscores.so.1
-debian/tmp/usr/share/apps/kdegames/pics/star.png
-debian/tmp/usr/share/icons/hicolor/16x16/actions/roll.png
-debian/tmp/usr/share/icons/hicolor/16x16/actions/highscore.png
-debian/tmp/usr/share/icons/hicolor/22x22/actions/roll.png
-debian/tmp/usr/share/icons/hicolor/32x32/actions/roll.png
-debian/tmp/usr/share/icons/hicolor/32x32/actions/highscore.png
Index: debian/lskat.install
===================================================================
RCS file: debian/lskat.install
diff -N debian/lskat.install
--- debian/lskat.install	29 Oct 2002 15:52:04 -0000	1.1.1.2
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,12 +0,0 @@
-debian/tmp/usr/games/lskat
-debian/tmp/usr/games/lskatproc
-debian/tmp/usr/share/man/man6/lskat*
-debian/tmp/usr/share/doc/kdegames/lskat/*
-debian/tmp/usr/share/doc/kde/HTML/en/lskat/*
-debian/tmp/usr/share/apps/lskat/*
-debian/tmp/usr/share/icons/hicolor/48x48/apps/lskat.png
-debian/tmp/usr/share/icons/hicolor/32x32/apps/lskat.png
-debian/tmp/usr/share/icons/locolor/32x32/apps/lskat.png
-debian/tmp/usr/share/icons/locolor/16x16/apps/lskat.png
-debian/tmp/usr/share/applnk/Games/Card/lskat.desktop
-debian/icons/lskat.xpm /usr/share/pixmaps
Index: debian/icons/kasteroids.xpm
===================================================================
RCS file: debian/icons/kasteroids.xpm
diff -N debian/icons/kasteroids.xpm
--- debian/icons/kasteroids.xpm	25 Jun 2002 02:34:24 -0000	1.1.1.1
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,55 +0,0 @@
-/* XPM */
-static char *kasteroids[] = {
-/* columns rows colors chars-per-pixel */
-"32 32 17 1",
-"  c opaque",
-". c #191919",
-"X c gray20",
-"o c #00007f",
-"O c #7f7f00",
-"+ c #4c4c4c",
-"@ c #666667",
-"# c gray50",
-"$ c blue",
-"% c red",
-"& c yellow",
-"* c gray60",
-"= c #b2b2b2",
-"- c gray80",
-"; c gray90",
-": c white",
-"> c None",
-/* pixels */
-">>>>>    >>>  .>>>>>>>>>>>>>>>>>",
-">>> .+@@# >> ;-.>>>>>>>>>>>>>>>>",
-">> .X+##** > ;%= >>>>>>>>>>>>>>>",
-"> .X+@*#== >> &%= >>>>>>>>>>>>>>",
-">.X++##*== >>>.=%.>>>>>>>>>>>>>>",
-" X++@#**- >>>>.  .>>>>>>>>>>>>>>",
-" +@+***=- >>>>>>>>>>>>>>>>>>>>>>",
-" XO@**O= >>>>>>>>>   >>>>>>>>>>>",
-" @+*#*# >>>>>>>>>>.;-.>>>>>>>>>>",
-"> +#@  >>>>>>>>>>> -%= >>>>>>>>>",
-">>   >>>>>>>>>>>>>> &%=.>>>>>>>>",
-">>>>>>>>>>>>>>>>>>>>.=O >>>>>>>>",
-"  .>>>>>>>>>>>>>>>>>>   >>>>>>>>",
-" ;-.>>>>>>>>>>>>>>>>>>>>   >>>>>",
-" ;%= .>>>>>>>>>>>>>>>>>> -: >>>>",
-"> &%=.>>>>>>>>>>     >>> +-: >>>",
-">>.=O >>>>>>>>> @#--= >>> *=: >>",
-">>> . >>>>>>>>> *=;;;* > # +=: >",
-">>>>>>>   >>>>> =-=:;-= *=- *=: ",
-">>>>>>> ;- >>>> #==$$#-= -=# #@ ",
-">>>>>>> -%=.>>> #== o:$-= # >  >",
-">>>>>>>>.&%& >>> #- oo:#== >>>>>",
-">>>>>>>>> =% >>>> #-.oo$$-= >>>>",
-">>>>>>>>>> . >>>>> #- o$$*== >>>",
-">>>>>>>>>>>>>   > # #= oo -# >>>",
-">>>>>>>>>>>>> -: *=- *=  -#  >>>",
-">>>>>>>>>>>>> @=: ==# *=-# -# >>",
-">>>>>>>>>>>>>> #=: # > ## -* >>>",
-">>>>>>>>>>>>>>> @=: >>>   # >>>>",
-">>>>>>>>>>>>>>>> #=: >>>>> >>>>>",
-">>>>>>>>>>>>>>>>> *+ >>>>>>>>>>>",
-">>>>>>>>>>>>>>>>>>  >>>>>>>>>>>>"
-};
Index: debian/icons/katomic.xpm
===================================================================
RCS file: debian/icons/katomic.xpm
diff -N debian/icons/katomic.xpm
--- debian/icons/katomic.xpm	25 Jun 2002 02:34:24 -0000	1.1.1.1
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,43 +0,0 @@
-/* XPM */
-static char *katomic[] = {
-/* columns rows colors chars-per-pixel */
-"32 32 5 1",
-"  c opaque",
-". c cyan",
-"X c red",
-"o c white",
-"O c None",
-/* pixels */
-"OOOOOOOOOO...OOOOOOOOOOOOOOOOOOO",
-"OOOOOOOOOO...OOOOOOOOOOOOOOOOOOO",
-"OOOOOOOOOO...OOOOOOOOOOOOOOOOOOO",
-"OOOOOOOOOOOooOOOOOOOOOOOOOOOOOOO",
-"OOOOOOOOOOOooOOOOOOOOOOOOOOOOOOO",
-"OOOOOOOOOOOooOOOOOOOOOOOOOOOOOOO",
-"OOOOOOOOOOOooOOOOOOOOOOOOOOOOOOO",
-"OOOOOOOOOOOooOOOOOOOOOOOOOOOOOOO",
-"OOOOOOOOOOOooOOOOOOOOOOOOOOOOOOO",
-"OOOOOOOOOOOOooOOOOOOOOOOOOOOOOOO",
-"OOOOOOOOOOOOooOOOOOOOOOOOOOOOOOO",
-"OOOOOOOOOOOXXXXXOOOOOOOOOOOOOOOO",
-"OOOOOOOOOOXXXXXXXOOOOOOOOOOOOOOO",
-"OOOOOOOOOXXXXXXXXXOOOOOOOOOO...O",
-"OOOOOOOOOXXXXXXXXXoooooooooo...O",
-"OOOOOOOOOXXXXXXXXXoooooooooo...O",
-"OOOOOOOOOXXXXXXXXXOOOOOOOOOOOOOO",
-"OOOOOOOOOXXXXXXXXXOOOOOOOOOOOOOO",
-"OOOOOOOOOoXXXXXXXOOOOOOOOOOOOOOO",
-"OOOOOOOOoooXXXXXoOOOOOOOOOOOOOOO",
-"OOOOOOOoooOOOOOooOOOOOOOOOOOOOOO",
-"OOOOOOoooOOOOOOOooOOOOOOOOOOOOOO",
-"OOOOOoooOOOOOOOOooOOOOOOOOOOOOOO",
-"OOOOoooOOOOOOOOOOooOOOOOOOOOOOOO",
-"OO...oOOOOOOOOOOOooOOOOOOOOOOOOO",
-"OO...OOOOOOOOOOOOOooOOOOOOOOOOOO",
-"OO...OOOOOOOOOOOOO...OOOOOOOOOOO",
-"OOOOOOOOOOOOOOOOOO...OOOOOOOOOOO",
-"OOOOOOOOOOOOOOOOOO...OOOOOOOOOOO",
-"OOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOO",
-"OOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOO",
-"OOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOO"
-};
Index: debian/icons/kbackgammon.xpm
===================================================================
RCS file: debian/icons/kbackgammon.xpm
diff -N debian/icons/kbackgammon.xpm
--- debian/icons/kbackgammon.xpm	25 Jun 2002 02:34:24 -0000	1.1.1.1
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,53 +0,0 @@
-/* XPM */
-static char *kbackgammon[] = {
-/* columns rows colors chars-per-pixel */
-"32 32 15 1",
-"  c opaque",
-". c #191919",
-"X c gray20",
-"o c #7f0000",
-"O c #7f7f00",
-"+ c #4c4c4c",
-"@ c #666667",
-"# c gray50",
-"$ c red",
-"% c yellow",
-"& c gray60",
-"* c #b2b2b2",
-"= c gray80",
-"- c gray90",
-"; c None",
-/* pixels */
-";;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;",
-";;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;",
-";;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;",
-".o.oo.o.o o..o.oo.o..o..o..oo.o.",
-"o.oo@oo.O$+ooO#oo.oo#+oo@Oo.X@$o",
-"o. #** .=*@ @==oo. #=* .==@ +*=o",
-"oX *== .==# #*=ooX &==..==#.#*=o",
-"o+ =*=.+==& &=-oo+ =*=.X=*& &=-o",
-"oO.-=%+@==% *==ooO.==-X@==*.*==o",
-"o&.-==+@-=- =--oo#.-*-+O==- %--o",
-"o&X-=-@&==-.=%-oo*+-=-@&-==.-=-o",
-"o=O==-#%==-X=--oo=@%==#%===O==-o",
-"o-#--=**==-O---oo=O-==%*==-O=--o",
-"o==%==-%-%-*=--oo-*-=-=-%--*=%-o",
-"o===%--=-==-=--oo-==--==----=--o",
-"o-==--==--=%=%-oo%==-%==---%=--o",
-"o-%===--=--%==-oo-===%--==---=%o",
-"o--=%==-=--==--oo-%==---%=-%==-o",
-"o---=-%-=%==%=-oo%--=-==-*===*-o",
-"o=--O-==-&==-&-oo=--&%=--&===&-o",
-"o=-=O--==@-=-O=oo-%-+---=@-%=#=o",
-"o-%=.==-%+-=-O*oo===X--%*O-=%@*o",
-"o=-- ===#O==-X&oo=== ===&O==-X&o",
-"o=== %==@X==-.@oo=== *==@ =*-.#o",
-"o=**.#==+ -== +o$***.#==+ ===.Oo",
-"o==&.@*=X *=* Xoo*=& @*=+ ***.Xo",
-"o*=#.+==. &*&..oo*=@ +*=. &=& .o",
-"o&#ooo&Oo.@#O.ooo##+.o##.o@#+o.o",
-".o.ooo.oo.oo.o.ooo.oo.oo.o.ooo.o",
-";;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;",
-";;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;",
-";;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;"
-};
Index: debian/icons/kbattleship.xpm
===================================================================
RCS file: debian/icons/kbattleship.xpm
diff -N debian/icons/kbattleship.xpm
--- debian/icons/kbattleship.xpm	25 Jun 2002 02:34:24 -0000	1.1.1.1
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,54 +0,0 @@
-/* XPM */
-static char *kbattleship[] = {
-/* columns rows colors chars-per-pixel */
-"32 32 16 1",
-"  c #00007f",
-". c #007f7f",
-"X c #7f007f",
-"o c #7f7f00",
-"O c #4c4c4c",
-"+ c #666667",
-"@ c gray50",
-"# c blue",
-"$ c cyan",
-"% c red",
-"& c yellow",
-"* c gray60",
-"= c #b2b2b2",
-"- c gray80",
-"; c gray90",
-": c white",
-/* pixels */
-";;----*====*****=;-;-;;-;;-;;;;-",
-"-========-==**=.=====-=-===-===-",
-"******=--$--=****=*=****+**=****",
-"*@***==$=====******.*@@+@**.#@@@",
-"#**========****@*@@.@***@+@+*.*@",
-"*.*=-====*====@@.@+@@oo+@@+@@@#@",
-"@@*-====*+@**@++@@@@@+o+++***.@@",
-"OO+.*@@@@@.***@**X@@****+@#@*@+@",
-"O++@=****@@***@%@**=***=@@***@#@",
-".@+*o*&oo@++@*@@@&%*@@***=***@@.",
-"+@@@*@***@++@@**%**%%*%-*****@+ ",
-"***@@@@..@@@@@@%%*&==&*******@+@",
-"@@*@@@@@*@*@@@*&---&&=&****=*@.+",
-"@@.**@@@@@.@@@%=&;;;;-*%%@**#+#+",
-"+@@***@@@@@@@o&&;&;;&;;&%*@+ +++",
-"@@@**.@@@@**@@=-;--;;-;;%*X.@@ O",
-"*@.**@#*@@*@@&-&-;;-;--;&*%@+X+.",
-"****@**@@@@@%=&--;;;;--&=*@@+++X",
-"@*@***@@@@@@*&-;;;--;--;&&X+@#.@",
-".*@@**@@*@*=%=&--;;&-;&;*%%*++.X",
-"**@***.@@@@@%-&-&&-&=&*%@@++++.X",
-"*=*=**=*@@@+@*@@@X@@@X+XX ++ @@.",
-"@=***@@@@ +X@ +++X+X++.+++#++#X.",
-".X..+++#@++. ++#++.+X+X.+++++@.X",
-" +.X.++ +X.++#++.XX+++@.+ .+ ++ ",
-" ++ +  +.XX.O+ +++.. #**-*X.+ .+",
-"@+#+.X@++. + .X. +.X++@*=-=. .X.",
-" ++OX.+  +++++.X@ X.@ #@=---=X.X",
-" +++.++ X.O +.X.+ +.X.X+*--;:*.X",
-"++  +  +.X @ X.+ +.X.+.+ .=-:;-.",
-".XX.++@ X. @+++  ++ +  +.X+=;::-",
-"X..+  +@.++ + X.@ ++ +@ +.+.=;;;"
-};
Index: debian/icons/kblackbox.xpm
===================================================================
RCS file: debian/icons/kblackbox.xpm
diff -N debian/icons/kblackbox.xpm
--- debian/icons/kblackbox.xpm	25 Jun 2002 02:34:24 -0000	1.1.1.1
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,50 +0,0 @@
-/* XPM */
-static char *kblackbox[] = {
-/* columns rows colors chars-per-pixel */
-"32 32 12 1",
-"  c opaque",
-". c #191919",
-"X c gray20",
-"o c #00007f",
-"O c #7f7f00",
-"+ c #4c4c4c",
-"@ c #666667",
-"# c gray50",
-"$ c yellow",
-"% c gray60",
-"& c #b2b2b2",
-"* c None",
-/* pixels */
-"*************       ************",
-"********* .. ++%+++X       *****",
-"***** .X.+@++@+X@+@@++@+++#     ",
-"*    X++++XXX+X+@++@+@@++X+@++X ",
-" XXXX++X++++++XXX+X@@XX++X+++XX ",
-" X++XXXXXX@+X@+++++XX+++++@@X.X ",
-" X+XXX+XX+XX++XXXX+++X@+++X+X.. ",
-" X+XX+XXXXXXX@XXX+XXXXXX+X.X.X. ",
-"* ++X.OOXXXX+#+XXX+XXXX+XX.X.. *",
-"* ++XOX+OOOO+++XXXXXXXXXXXXXXX *",
-"* XO$.X++.+@OOO+OXXXXXX+XX..XX *",
-"* ++&$O.+X+XX.$OO@XXXXX+@XX@XX *",
-"** X+O$+XXXXXXO$O%$XXXXX+X.XX  *",
-"** +X+XXXXXX+X#$OO$OXXX++XX+. **",
-"** XXX+XXXXX++XO+$$%@X+XXXXXX **",
-"** XXX+XX+XXXX.OOOO$$XXXX.X.X **",
-"*** XXXXX@XXXX#OO%$$$o.XXXX+. **",
-"*** ++XXXXXXoO$OO$$OOXXXX.+X. **",
-"*** X+XXXX.X.$$O$OOXXXXXX.XX ***",
-"*** XXXXXXo$$OOXoXX+XXX+.X.X ***",
-"**** +X+@XXO+XXX+XXXXXXX.XX ****",
-"**** +X.@X+XXXXX++XXXXXX+XX ****",
-"**** XX+X+O@++X+XXXXXXXXXX *****",
-"****  XXXX#$O+XX+XXX+@XXXX *****",
-"******  +XXO#$XXXXXXX+X.X ******",
-"********  XX.XX+@+XXXXXXX ******",
-"**********  X++X++++X+X+ *******",
-"************  XXXXXXX+XX *******",
-"**************  ++XXX+X ********",
-"****************  XX+++ ********",
-"******************  +X *********",
-"********************  .*********"
-};
Index: debian/icons/kbounce.xpm
===================================================================
RCS file: debian/icons/kbounce.xpm
diff -N debian/icons/kbounce.xpm
--- debian/icons/kbounce.xpm	25 Jun 2002 02:34:24 -0000	1.1.1.1
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,49 +0,0 @@
-/* XPM */
-static char *kbounce[] = {
-/* columns rows colors chars-per-pixel */
-"32 32 11 1",
-"  c opaque",
-". c #191919",
-"X c gray20",
-"o c gray50",
-"O c red",
-"+ c gray60",
-"@ c #b2b2b2",
-"# c gray80",
-"$ c gray90",
-"% c white",
-"& c None",
-/* pixels */
-"&&&&&&&&&&&&$%%%%$$$$&&&&&&&&&&&",
-"&&&&&&&&&&$%%%%%%%%%#@O+&&&&&&&&",
-"&&&&&&&&$%%%%%%%%$%@OOOOO&&&&&&&",
-"&&&&&&&%%%%%%%%%%$+OOOOOOO+&&&&&",
-"&&&&&&%%%%%%%%%%$OOOOOOOOOO&&&&&",
-"&&&&&%%%%%%%%%%$+OOOOOOOOOOO&&&&",
-"&&&&%%%%%%%%%%%@OOOOOOOOOOOOO&&&",
-"&&&$%%%%%%%%%%$OOOOOOOOOOOOOO+&&",
-"&&&%%%%%%%%%%$OOOOOOOOOOOOOOOO&&",
-"&&$%%%%%%%%%$@OOOOOOOOOOOOOOOOO&",
-"&&%%%%%%%%%%$OOOOOOOOOOOOOOOOOO&",
-"&&%%%%%%%%%$OOOOOOOOOOOOOOOOOOO&",
-"&$%%%%%%%%%OOOOOOOOOOOOOOOOOOOO&",
-"&%%%%%%%%%#OOOOOOOOOOOOOOOOOOOO&",
-"&%%%%%%%%$+OOOOOOOOOOOOOOOOOOOO&",
-"&%%%%%%%%#OOOOOOOOOOOOOOOOOOOOO+",
-"&%%%%%%%%#OOOOOOOOOOOOOOOOOOOOO+",
-"&$%%%%%%#OOOOOOOOOOOOOOOOOOOOOOO",
-"&$%%%%%$+OOOOOOOOOOOOOOOOOOOOOO&",
-"&$%%%%%$OOOOOOOOOOOOOOOOOOOOOOO&",
-"&&%%%%%#OOOOOOOOOOOOOOOOOOOOOOo&",
-"&&$%%%%OOOOOOOOOOOOOOOOOOOOOOO&&",
-"&&&%%%%OOOOOOOOOOOOOOOOOOOOOOO&&",
-"&&&&%%$OOOOOOOOOOOOOOOOOOOOOO&&&",
-"&&&&$%@OOOOOOOOOOOOOOOOOOOOO+&&&",
-"&&&&&%+OOOOOOOOOOOOOOOOOOOOO&&&&",
-"&&&&&&#OOOOOOOOOOOOOOOOOOOo&&&&&",
-"&&&&&&&OOOOOOOOOOOOOOOOOOO&&&&&&",
-"&&&&&&&&OOOOOOOOOOOOOOOo&&&&&&&&",
-"&&&&&&&&&&OOOOOOOOOOOO&&&&&&&&&&",
-"&&&&&&&&&&&&&+O++Oo&&&&&&&&&&&&&",
-"&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&"
-};
Index: debian/icons/kenolaba.xpm
===================================================================
RCS file: debian/icons/kenolaba.xpm
diff -N debian/icons/kenolaba.xpm
--- debian/icons/kenolaba.xpm	25 Jun 2002 02:34:24 -0000	1.1.1.1
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,52 +0,0 @@
-/* XPM */
-static char *kenolaba[] = {
-/* columns rows colors chars-per-pixel */
-"32 32 14 1",
-"  c opaque",
-". c #191919",
-"X c gray20",
-"o c #7f7f00",
-"O c #666667",
-"+ c gray50",
-"@ c red",
-"# c yellow",
-"$ c gray60",
-"% c #b2b2b2",
-"& c gray80",
-"* c gray90",
-"= c white",
-"- c None",
-/* pixels */
-"-------  .    . .   ..   -------",
-"------- ***&**&*&*&*&&&& -------",
-"------ &**&**&**&&*&&&&$%X------",
-"------ **&%$$$$$$%$%%$$$$ ------",
-"----- &*%%@%$$$&#+$$+%&o$% -----",
-"----- *&**#@$+&*&@$+%*=@+$ -----",
-"----.%*%*&@@$+#&@@$+o*#@+$% ----",
-"----.**&@@o%%$$@@%%$$@@++$$ ----",
-"----.***&&**%%$$#$%%$oO++$$.----",
-"--- %***&****&%$$$&@+++#&@+% ---",
-"--- ***&&=*===%$+**%@$$**@$$ ---",
-"-- &**&%&*=***&$+%&@o$$*#@$$% --",
-"--.***&%*******%+@@$$$+@@o$$$ --",
-"- &***&&&&*&&*%$$oo$o$o$+++%$% -",
-"- =*&$$%&&#$$%$#$$$+o$$$+%&@$$ -",
-" $*&$$+$**@$$$$$#$%o+$oo$*=@+$+ ",
-" X*&%$$o%%#$+$$$+$%oo$$%$%&@+$+.",
-"-.$&%$$+@#@o+$+#$$*&$%#*%@@o$% -",
-"- O&%%++$$$+++$$#&*#&**&#%&$$+ -",
-"-- &#%&&o$$$++$$%&&$#*&&%&#$$ --",
-"--.%&%**#$$$o$+$$**%@=*%%*$$+ --",
-"--- *$#%@$+$$$+o+#%##*&%#%+$ ---",
-"--- &&#@#+o%%$++$#@o&*%%*&$+ ---",
-"---- &%$+++$o$$#$$&***$#&$$ ----",
-"---- %&%%&#$o%%&$+$&*&%$%$$ ----",
-"---- +&%**@$o+**%#oo#*&#+$O ----",
-"----- &#%%o#$$#*@#+$$%$@$$.-----",
-"----- $&$#@$$o+@@++++##o$$.-----",
-"------ &%+$o$$$$%$#+$%%+% ------",
-"------.$$$$$$+$$$$$$$$$$+ ------",
-"------- $$$$$$$%$$$$$$$$ -------",
-"-------   .  .      .    -------"
-};
Index: debian/icons/kjumpingcube.xpm
===================================================================
RCS file: debian/icons/kjumpingcube.xpm
diff -N debian/icons/kjumpingcube.xpm
--- debian/icons/kjumpingcube.xpm	25 Jun 2002 02:34:24 -0000	1.1.1.1
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,46 +0,0 @@
-/* XPM */
-static char *kjumpingcube[] = {
-/* columns rows colors chars-per-pixel */
-"32 32 8 1",
-"  c opaque",
-". c #191919",
-"X c gray20",
-"o c #00007f",
-"O c #7f0000",
-"+ c blue",
-"@ c red",
-"# c None",
-/* pixels */
-"################################",
-"################################",
-"##                            ##",
-"## ++++++++ @@@@@@@@ @@@@@@@@ X#",
-"## +  +o+oo @  O@OOO @@O@O@OO XX",
-"## +  o+o+o @  @O@@O @O@O@O@O XX",
-"## ++o+o+oo @@O  O@O @@O  @OO XX",
-"## +o+o+o+o @O@  @OO @O@  @OO XX",
-"## ++o+o  o @O@O@  O @@O@O@OO XX",
-"## +o+o+  o @@O@O  O @O@O@O@O XX",
-"## +ooooooo @OOOOOOO @OOOOOOO XX",
-"##                            XX",
-"## ++++++++ @@@@@@@@ ++++++++ XX",
-"## +  o+o+o @  @@  O +  o+o+o XX",
-"## +  +o+oo @  O@  O +  +oo+o XX",
-"## ++oo+o+o @@O@O@OO +o+  +oo XX",
-"## ++o+oo+o @@O@@@OO ++o  +oo XX",
-"## +o+o+  o @  OO  O +o+o+  o XX",
-"## ++o+o  o @  @@  O +o++o  o XX",
-"## +ooooooo @OOOOOOO +ooooooo XX",
-"##                            XX",
-"## @@@@@@@@ ++++++++ ++++++++ XX",
-"## @@O@O@OO +  o+o+o ++o+oo+o XX",
-"## @O@O@O@O +  +o+oo +o+o++oo XX",
-"## @@O  @OO ++o  o+o ++o  +oo XX",
-"## @O@  O@O +o+  +oo +o+  o+o XX",
-"## @@O@O@OO +o+o+  o ++o+o+oo XX",
-"## @O@O@O@O ++o+o  o +o+o+o+o XX",
-"## @OOOOOOO +ooooooo +ooooooo XX",
-"##                            XX",
-"###XXXXXXXXXXXXXXXXXXXXXXXXXXXXX",
-"####XXXXX.XXXXXXXXXXX.XXXXXXX.XX"
-};
Index: debian/icons/klines.xpm
===================================================================
RCS file: debian/icons/klines.xpm
diff -N debian/icons/klines.xpm
--- debian/icons/klines.xpm	25 Jun 2002 02:34:24 -0000	1.1.1.1
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,57 +0,0 @@
-/* XPM */
-static char *klines[] = {
-/* columns rows colors chars-per-pixel */
-"32 32 19 1",
-"  c opaque",
-". c #191919",
-"X c gray20",
-"o c #00007f",
-"O c #007f00",
-"+ c #7f0000",
-"@ c #7f7f00",
-"# c #4c4c4c",
-"$ c #666667",
-"% c gray50",
-"& c green",
-"* c red",
-"= c yellow",
-"- c gray60",
-"; c #b2b2b2",
-": c gray80",
-"> c gray90",
-", c white",
-"< c None",
-/* pixels */
-"<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<",
-"<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<",
-"<<<;=<<<<<<<<<<<<<<<<<<<<<<<<<<<",
-"<<@#:<<<<<<<<<<<<<<<<<<<OO<<<<<<",
-"<<X@=<<<<<<<<<<<<<<<<<<&<<&<<<<<",
-"<@.;=<<<<<<<<<<<<<<<<<<<<OO<<<<<",
-"<X+===<<<<<<<<<<<<<<<<<<&<<<<<<<",
-"@X=-=:=<<<<<<<<<<<<;::;&&<<<<<<<",
-"@@=< ==-OO<<<<<<&&&XO#:O&&&<<<<<",
-"=<<<=@+@&&&O<<<&,;&O.%$&&&&&&<<<",
-"<<<<=-..O&&O<<<&&&&&&..O&&&&&&<<",
-"<<<<<=@..O@=<<<O&&OO&OOO&&&&&&<<",
-"<<<<<<=@X O==-&@#OOOOO&&&&&O&&&<",
-"<<<<<<<<=@%@@@@&&OO&&&&&&&O&&&&<",
-"<<<<<<<<<==<OOO&OO&&&&&&&&OOO&&&",
-"<<<<<<<<<<<<&O&&O&&&&&&&&&&OOO&&",
-"<<<<<<<<<<<<&&&&O&&&&&&&&&&&OOO&",
-"<<<<<<<<<<<<&&&&&O&&&&&&&&&&&OO&",
-"<<<<<<<<<<<<O&&&&&&&&&&&&&&OOO&&",
-"<<<<<<<<<<<<<O&OO&&&&&&&&&&XO&&<",
-"<<<<<<<<<<<<<X.@@#O&&&&&O&&O&&&<",
-"<<<<<<<<<<<<<=@@#@@O&&OOOOO.O+@<",
-"<<<<<<<<<<<<<XX@=@X@XOXO&&@=*=<<",
-"<<<<<<<<<<<<<<O$@+@@=*@OOO@@=<<<",
-"<<<<<<<<<<<<<<<=@@X@@@@OO..O<<<<",
-"<<<<<<<<<<<<<<<<<<OOOOOOOOO<<<<<",
-"<<<<<<<<<<<<<<<<<<OOOOOOO<<<<<<<",
-"<<<<<<<<<<<<&-&<<&&<<<<<&&<<&&-&",
-"<<<<<<<<<<<<&>&&O&OO<<<&O&&&&&>&",
-"<<<<<<<<<<<<&&&&O&O&<<<OOOOO&&&&",
-"<<<<<<<<<<<<<O&OO&O<<<<<&OO&O&O<",
-"<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<"
-};
Index: debian/icons/kmahjongg.xpm
===================================================================
RCS file: debian/icons/kmahjongg.xpm
diff -N debian/icons/kmahjongg.xpm
--- debian/icons/kmahjongg.xpm	25 Jun 2002 02:34:24 -0000	1.1.1.1
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,52 +0,0 @@
-/* XPM */
-static char *kmahjongg[] = {
-/* columns rows colors chars-per-pixel */
-"32 32 14 1",
-"  c opaque",
-". c #191919",
-"X c #00007f",
-"o c #007f00",
-"O c #7f0000",
-"+ c #7f7f00",
-"@ c blue",
-"# c red",
-"$ c magenta",
-"% c yellow",
-"& c gray80",
-"* c gray90",
-"= c white",
-"- c None",
-/* pixels */
-"--------.  .  .--------   .  .--",
-"--------.&*****.------- ****** -",
-"--------.*&&&*&# ------ ***&%&+.",
-"--------.&&#%#&+ ------ &%ooo*# ",
-"--------.*&*##&% ------ *&%**&+O",
-"--------.&%##&&# ------ **ooo*+ ",
-"  .. . -.*&*##&+ ------ &%*&&&# ",
-".*&**** .&&##&*+ ------ **ooo&+ ",
-" *&%*&&+.*%*&#&# ------ *&&o&*+.",
-" **@%@&#.*&&&&&+   .  - *%&&&%#.",
-" *&&@@&%  #+++# ******.=.+#+#++.",
-" =%@@&&#  ..  . &&&*%&+ - . .   ",
-" *&%@@&+**&&%*  &*#*#*+ --------",
-".*&@@**+&&&&&*+ &%&##%#.--------",
-" *&&@X*#&*ooo%+.&&##*&+ --------",
-" *%*&&&+&%*&&&#.%&*##*# --------",
-" .#+#+#+&&ooo&+ *&##&&+.--------",
-" **&***O&&&*&&+ *&&*#&#.--------",
-" *&&&&&+&%ooo&#.**%**&+ --------",
-".**$$%*+***o%&+..+#+#+#.--------",
-" *&$$$&#*&&&*&+**&**&.......----",
-" **%*$&+.++##+#**%&*&+&&*&*& ---",
-".&&$&%*#  ..  .&*$$%&+=&&%&&+ --",
-".*&$$$&+ -----.*&$$$%#*&@*@&+ --",
-" *%*$$&+ ----- *&*&$*#**&@@&+.--",
-".*&**&*# -----.*&$%*&+*%@@&&#.--",
-"- +#+++# ----- *%$$$%+*&&@@*+ --",
-"--. .... ----- *&*$$*#**@@&%# --",
-"-------------- *&*&%*+*%*@@&+.--",
-"--------------- +##++#*&&%&&+.--",
-"----------------  ..  .+#+++# --",
-"----------------------- ..  . --"
-};
Index: debian/icons/kmines.xpm
===================================================================
RCS file: debian/icons/kmines.xpm
diff -N debian/icons/kmines.xpm
--- debian/icons/kmines.xpm	25 Jun 2002 02:34:24 -0000	1.1.1.1
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,54 +0,0 @@
-/* XPM */
-static char *kmines[] = {
-/* columns rows colors chars-per-pixel */
-"32 32 16 1",
-"  c opaque",
-". c #191919",
-"X c #007f7f",
-"o c #4c4c4c",
-"O c #666667",
-"+ c gray50",
-"@ c blue",
-"# c cyan",
-"$ c red",
-"% c yellow",
-"& c gray60",
-"* c #b2b2b2",
-"= c gray80",
-"- c gray90",
-"; c white",
-": c None",
-/* pixels */
-"::::::::::::::::::::::   :::::::",
-":::::::::::::::::::: . = :::::::",
-"::::::::::::::::::  *%.= :::::::",
-"::::::::::::::::  %*$$ = :::::::",
-"::::::::::::::: *%$$$$ = :::::::",
-"::::::::::::::::  $$$$.= :::::::",
-"::::::::::::::::::  $$ = :::::::",
-"::::::::::::::::::::   = :::::::",
-"::::::::::::::::::::::.= :::::::",
-":::::::::::::::::::::: = :::::::",
-":::::::::::::::::::::: = :::::::",
-"::::::::::::::   :::::.= :::::::",
-"::::::::::::  +*@  ::: = :::::::",
-"::::::::::  &*=@=@+  : = :::::::",
-"::::::::  &**@*=@@=*+ .=   :::::",
-"::::::  &-++*=@@**=  + = =&  :::",
-"::::  +===;;++**=  +=* = =*&+  :",
-"::  +====X@=;;&+ &=**= - &&*==+ ",
-"  +*=*=X@=*@**;; ;;**=   *==*;; ",
-" =;;*#@*=@@=*=+& =*;;*=*==*;;OO ",
-" ===;;*@@*=*++;;&+*=*;;==;;OOOO ",
-" =====;;*=+&;;*=**@&=**;;OOOOOO ",
-" ====*=*++;;*===@*=@+&*oOO+O+OO ",
-"  +==*=*+;;;=*@*=@@*=*OOO+OOOOO ",
-"::  &**=+-**;;*@@==*;;+oO+OO+O  ",
-"::::  &*&;****;;**;;O+OO+OO+  ::",
-"::::::  &-******;;O++++oOO  ::::",
-"::::::::  +**&&*+++++O+o  ::::::",
-"::::::::::  +***+++O++  ::::::::",
-"::::::::::::  +&++++  ::::::::::",
-"::::::::::::::  ++  ::::::::::::",
-"::::::::::::::::  ::::::::::::::"
-};
Index: debian/icons/konquest.xpm
===================================================================
RCS file: debian/icons/konquest.xpm
diff -N debian/icons/konquest.xpm
--- debian/icons/konquest.xpm	25 Jun 2002 02:34:24 -0000	1.1.1.1
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,55 +0,0 @@
-/* XPM */
-static char *konquest[] = {
-/* columns rows colors chars-per-pixel */
-"32 32 17 1",
-"  c opaque",
-". c #191919",
-"X c gray20",
-"o c #00007f",
-"O c #007f7f",
-"+ c #7f007f",
-"@ c #4c4c4c",
-"# c #666667",
-"$ c gray50",
-"% c blue",
-"& c cyan",
-"* c magenta",
-"= c gray60",
-"- c #b2b2b2",
-"; c gray80",
-": c gray90",
-"> c white",
-/* pixels */
-"......X...........o.X..X.....X..",
-"..XX.....XX.X..X.X..............",
-".......X..............X.......X.",
-".X...o....X..@....X..XX.X.Xo.X..",
-"...X.....X..........X..........X",
-"..X..XXX..X..X...X..@..X.X.oX...",
-".....X..X.X.X...X..X..X..X......",
-"X.X...........X............XX..X",
-".......o..............X..X.X....",
-".XXXX.XX.X......@..............X",
-".OoXo@O@.o..XoX....X.X..X.......",
-".$X#$=#@##X@XX.@..o.X. X......X.",
-".@X+@=-$$==-$++oO..XXo........Xo",
-".X$$+$o$o$$=;;=$##@...X.........",
-"o X$$=#$##o#$;;;;;=OO@o.X...X...",
-"...@##-+@=$#o-;::>::;-$@..XX ...",
-"..o..X@-*-;;:>>>>>:;:;;;$oXX@..X",
-". ..+XX@#===>>>>>::;;::;;-#oX...",
-". ... o.@#=;-;:::-+-;=:::;;O@X..",
-"........X@@@$-===@@--==:-;;-$Xo.",
-".........oX@O+$;---=$O$o----;Oo.",
-"... .......X@@#o$+$O=-==&==--$@X",
-".  .  . ...X.X.@@@@$%$#$+$O$$#X.",
-"...X....... .X.Xo#XX@X@@$=--$$#.",
-".... ........ Xo..Xo@Xo.o.@.oX.X",
-".. .. ......X.. X....XXXO@..X...",
-"...... .. .   ... .....  X......",
-"..... ............ ......  ..o..",
-".... . . ..... ... .........  ..",
-".   ... ... . .. . ...... . . ..",
-"...  ........ . . .X.... ... . .",
-"..o..............o........X..X.."
-};
Index: debian/icons/kpat.xpm
===================================================================
RCS file: debian/icons/kpat.xpm
diff -N debian/icons/kpat.xpm
--- debian/icons/kpat.xpm	25 Jun 2002 02:34:24 -0000	1.1.1.1
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,47 +0,0 @@
-/* XPM */
-static char *kpat[] = {
-/* columns rows colors chars-per-pixel */
-"32 32 9 1",
-"  c opaque",
-". c #191919",
-"X c #00007f",
-"o c blue",
-"O c red",
-"+ c #b2b2b2",
-"@ c gray80",
-"# c white",
-"$ c None",
-/* pixels */
-"$       $$$$       $$$$       $$",
-" ####### $$ ####### $$ ####### $",
-" #O###O#  $ # ### #  $ #O###O#  ",
-" #######  $ #######  $ #######  ",
-" #######  $ #######  $ #######  ",
-" ###O###  $ ### ###  $ ###O###  ",
-" #######  $ #######  $ #######  ",
-" #######  $ #######  $ #######  ",
-" #O###O#  $ # ### #  $ #O###O#  ",
-" #######  $ #######  $ #######  ",
-"          $$         $          ",
-" @+@@@+@  $$$       $$ @@+@++@  ",
-" #######  $$$$$$$$$$$$ #######  ",
-" #O###O#  $$   .   $$$ # ### #  ",
-" #######  $ @o@o@o@ $$ #######  ",
-"          $ o@o@o@o  $          ",
-" @+@+@+@  $ +o@o@o@. $ #######  ",
-" #######  $ o@o@o@o  $ #######  ",
-" # ### #  $ @o@o@o@. $ #O###O#  ",
-" #######  $.X@o@o@o  $ #######  ",
-"          $ @o@o@o@. $$         ",
-" @+@+@@+  $ o@o@o@o  $$$       $",
-" #######  $ @o@o@o@  $$$$$$$$$$$",
-" #O###O#  $$    . .  $$$$$$$$$$$",
-" #######  $$$       $$$$$$$$$$$$",
-"          $$$$$$$$$$$$$$$$$$$$$$",
-" @+@+@@+  $$$$$$$$$$$$$$$$$$$$$$",
-" #######  $$$$$$$$$$$$$$$$$$$$$$",
-" # ### #  $$$$$$$$$$$$$$$$$$$$$$",
-" #######  $$$$$$$$$$$$$$$$$$$$$$",
-"$         $$$$$$$$$$$$$$$$$$$$$$",
-"$$       $$$$$$$$$$$$$$$$$$$$$$$"
-};
Index: debian/icons/kpoker.xpm
===================================================================
RCS file: debian/icons/kpoker.xpm
diff -N debian/icons/kpoker.xpm
--- debian/icons/kpoker.xpm	25 Jun 2002 02:34:24 -0000	1.1.1.1
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,57 +0,0 @@
-/* XPM */
-static char *kpoker[] = {
-/* columns rows colors chars-per-pixel */
-"32 32 19 1",
-"  c opaque",
-". c #191919",
-"X c gray20",
-"o c #007f00",
-"O c #007f7f",
-"+ c #7f0000",
-"@ c #7f7f00",
-"# c #4c4c4c",
-"$ c #666667",
-"% c gray50",
-"& c red",
-"* c magenta",
-"= c yellow",
-"- c gray60",
-"; c #b2b2b2",
-": c gray80",
-"> c gray90",
-", c white",
-"< c None",
-/* pixels */
-"<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<",
-"<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<",
-"<<<<<<<<<<<<<  .<<<<<<<. .<<<<<<",
-"<<<<<<<<<<<       <<<  =::  <<<<",
-"<<<<<<<<<< .  .   .< >:::==: <<<",
-"<<<<<<<<<<          >==:::>::.<<",
-"<<<<<<<<<< ..   ..X$>>:>=:=>:.<<",
-"<<<<   <<< XXXXXXX.$>=:==::>@ <<",
-"<<  =>:. < .#.... .@:>::>>;&@.<<",
-"< :>:::=:  X#... .X$:>=;=-%=%.<<",
-".:=>:=::=: .# .. ..$=>>=;=%; <<<",
-" :=:::==:>$.#......X=:>>;:& <<<<",
-" :>=>::::=#.#...  .$->:&;> <<<<<",
-".:>==:=::-X.#. ..%:,;-@:-:, <<<<",
-".=>:::=-@=$.X.$%,,,;-%&*;;,,.<<<",
-" ::>:=;-;&$#%:,,,;#+;;>+#+;,, <<",
-".==:=;;@@->,,>::.@-o+-$>;X@:,, <",
-".->>:;=:,,,,--%@$$%%-@.;#O@#->, ",
-"< -::=,,,:%$-;XX%+%X$:;-:-#$%,> ",
-"<<   +>,,%@%#-@-X-X@$@@#$%;,,:: ",
-"<<<<< >:,,%@#;->-+@$&#$:,,>:;,> ",
-"<<<<< >::,,%$%--:%&$-,,,::;,,:> ",
-"<<<<<.:>::,,#%&&;->:>,:;,,,;:>; ",
-"<<<<<< :,::,;$>;%:>>:;>,:;::;-  ",
-"<<<<<<< :,::::-:,,::,>:::;-   <<",
-"<<<<<<<< :,:::>:;>>,;>;-   <<<<<",
-"<<<<<<<<< ;,::;>,:;;-   <<<<<<<<",
-"<<<<<<<<<< >>,::;-   <<<<<<<<<<<",
-"<<<<<<<<<<< :;-   <<<<<<<<<<<<<<",
-"<<<<<<<<<<<<   <<<<<<<<<<<<<<<<<",
-"<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<",
-"<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<"
-};
Index: debian/icons/kreversi.xpm
===================================================================
RCS file: debian/icons/kreversi.xpm
diff -N debian/icons/kreversi.xpm
--- debian/icons/kreversi.xpm	25 Jun 2002 02:34:24 -0000	1.1.1.1
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,58 +0,0 @@
-/* XPM */
-static char *kreversi[] = {
-/* columns rows colors chars-per-pixel */
-"32 32 20 1",
-"  c opaque",
-". c #191919",
-"X c gray20",
-"o c #00007f",
-"O c #007f7f",
-"+ c #7f0000",
-"@ c #7f007f",
-"# c #7f7f00",
-"$ c #4c4c4c",
-"% c #666667",
-"& c gray50",
-"* c blue",
-"= c red",
-"- c yellow",
-"; c gray60",
-": c #b2b2b2",
-"> c gray80",
-", c gray90",
-"< c white",
-"1 c None",
-/* pixels */
-"11111111111111111111111111111111",
-"1  . .  ..     . .  ..  . ..  . ",
-"1 ,,,,,,>,:&,,>:;&:,>#,,,:=::,, ",
-"1 ,,,,,,,,:%,>=-===%>&>,=:===#, ",
-"1.,>,,>>>>-%,=:<,==X;&>#:,<=++& ",
-"1.>>>>>>>>:%>=:<>=++X&>=:<,==+X.",
-"1 >,,>>>>>:%>==:==++.%>====+++. ",
-"1 >>>>>>,>:%>====++X+&>=====+.X ",
-"1 <,>>,>>,:%>#=++++.&#>#$=++..% ",
-"1 ,,,>>>-,:#>>+++. +>&>>+X+. X> ",
-"1 :::::::::%>>>%X$%>,#>>>%X$#>>.",
-"1 &#%#%%#%%%;%##%%%%%$&#%%%#%%% ",
-"1 ,,,::>,,,%,,,>;>,,,%>:>>&:>>:.",
-"1 >>-==##>>#,>:===#;,#>:%***o&: ",
-"1 >;::=@+%>%,:=>:==+;&>*;,>*oX&.",
-"1.:=,,===.;%:=,,<=++$&>*>,<*oo$ ",
-"1 &=,<==+.%%>=;<:=++X%:**>O*o.X ",
-"1.====++++%%>=====+.+%:oo**oo.$ ",
-"1 ;==++++ &#>===++++$&:$oooo. % ",
-"1 >%++++ X>%,%++++..&&>#ooX. $: ",
-"1 >:%+ .$>>#,>#X+.X&>&>:;%$$&:: ",
-"1.&%%#%%%%%$&#&&%%&##$%##%%$%%% ",
-"1.>>,>&>>>>%>::>&:>>>%::::-:::: ",
-"1.>:***oX%>%,:****o%:%:;;;;;;;; ",
-"1 :*:,>*oo;%>*;,>ooo;&:;;;;;#;;+",
-"1 &*>,,*oo$%>*>,<*oo+#::;&;;:;& ",
-"1 ;**:&*o.X%:*&:*oo.$%:#;;&&&;; ",
-"1 ;****oo.X$:o***oo $&:;-&;;;&; ",
-"1 :$oooo. %%>ooooo. #&:;;;&;&;;.",
-"1 >&Xo...X:%>;o.X. $;%:;;#;&;#& ",
-"1 :::%X$;::&>:;%$$;::&:;;;;;;;;.",
-"1.           .          ..      "
-};
Index: debian/icons/ksame.xpm
===================================================================
RCS file: debian/icons/ksame.xpm
diff -N debian/icons/ksame.xpm
--- debian/icons/ksame.xpm	25 Jun 2002 02:34:24 -0000	1.1.1.1
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,58 +0,0 @@
-/* XPM */
-static char *ksame[] = {
-/* columns rows colors chars-per-pixel */
-"32 32 20 1",
-"  c opaque",
-". c #191919",
-"X c gray20",
-"o c #00007f",
-"O c #007f7f",
-"+ c #7f0000",
-"@ c #7f007f",
-"# c #7f7f00",
-"$ c #4c4c4c",
-"% c #666667",
-"& c gray50",
-"* c blue",
-"= c cyan",
-"- c red",
-"; c yellow",
-": c gray60",
-"> c #b2b2b2",
-", c gray80",
-"< c gray90",
-"1 c white",
-/* pixels */
-"   .    .    .  . . . . . . . . ",
-" %$%%%%%%$%%$%%$%$$%$%$%%$$%$%% ",
-".$%$$%$%$%$%$%$%%@%%&%$%%%-%%%% ",
-" %$%%$%$%$%$%$%$O:>=*&%%:>:--$% ",
-" %$%$%$%$%$%$%$%*11>*O*:,<,---$.",
-" $%$%$%$%$%$%$%$*<1,*OO->1>-%+$ ",
-".%$%$%$%$%$%$%%$**:*&%o----&+X$.",
-" $%$%$%$%$%$%$$%:=*O*Oo#---#++$.",
-".$%$%$%$%$%$%$%$@OOooXX%&++$+X$.",
-" %$%%%%$$%$%$%$%%$%XX$$%$%+$X%% ",
-".%%>>--&%$%$%$%$%>,;;:#&:>---%% ",
-" &-,<:---%$%$%$%;<<>;;#>><>--+% ",
-" %-11:+%+$$%$%%$;<1;;&##,1,-%+&.",
-".%-:---#X%%$%$$$;,,;;#$--:--++$.",
-" %----@++$%%$%$%;;;##%$----@++$.",
-" %%-%#+XX%$$%$%$%;:###$#@##+XX$.",
-".%%++++$$$%@%&%$$$+++$$%$+++X$%.",
-".%%:>:-&%%:==*%%%&,-:-#%&>:--%% ",
-".$>,<--+@&,<:**O->1>-@->><>--+$ ",
-".%-<1:-%$*<1>*%O:,<:-#+#><>-&+$.",
-".%-->-#++*>::*Oo-->--%+-->&--+$.",
-" %:----$$O=**O$o:---#+X-----$+$.",
-".$#-@++.X%*OOoXX#-&+@+X$&-+$+X$.",
-" %%@++XX%%$ooXX%&++X+X$%$++..$% ",
-".$&>:=:*%%&=*:*%#:<;;&%%:;;>;&%.",
-" %*,,**%*:,,>*%O><<;;;#><1>;##% ",
-".%*<1>O*o><1=*O*;<1,#:#;<1,;##% ",
-" %=>>*O&o*=>*&%o;;,;;#$;;,;;:#% ",
-".%*O*O&oo*OOOoO.;;;###$;;;;##X$.",
-".%&*O*oOo%:**$oo&:&;##X#&;&###$.",
-".%%$OoXX$%Oo$XX$%##+XX$%+##.X$% ",
-"       .. .  . . .     . .  . . "
-};
Index: debian/icons/kshisen.xpm
===================================================================
RCS file: debian/icons/kshisen.xpm
diff -N debian/icons/kshisen.xpm
--- debian/icons/kshisen.xpm	25 Jun 2002 02:34:24 -0000	1.1.1.1
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,57 +0,0 @@
-/* XPM */
-static char *kshisen[] = {
-/* columns rows colors chars-per-pixel */
-"32 32 19 1",
-"  c opaque",
-". c #191919",
-"X c gray20",
-"o c #007f00",
-"O c #7f0000",
-"+ c #7f007f",
-"@ c #7f7f00",
-"# c #4c4c4c",
-"$ c #666667",
-"% c gray50",
-"& c green",
-"* c red",
-"= c magenta",
-"- c yellow",
-"; c gray60",
-": c #b2b2b2",
-"> c gray80",
-", c gray90",
-"< c white",
-/* pixels */
-"        .. .  .. .... ..     .. ",
-" oo&o&&oo&o&&o&ooo&oo&ooo&o&ooo ",
-".ooo&ooooo@,,,,:o-,,,,$&o&oooo& ",
-" &&ooo&oo@>,*O->*,o,@@;ooooo&oo ",
-".oo&oo&o&X::%*;:@<@,@%;&ooooooo.",
-" &oo&oooo@-;***-@,,@@,%oo&ooooo.",
-" ooo&o&oo@-,>O>:@,@>$:;oo&o&o&o.",
-" &&ooo&oo@:,:*::*,o,&$-oooooooo.",
-" &oo&ooo&@>;*:*-@-o,;o:oooooooo.",
-" oo&o&&oo#-;>,;:@>;,:;-o&oo&&oo ",
-" &ooooooo@*@*@%&*%@*@@ooooooooo.",
-".oo&->>-:@@>>->@@:>>>>#@>>>::oo ",
-" oo@<:>>,@>;>>@>@<,;<,%%-:,--$o.",
-" &@;,>>>>@-@:-X:O<:+><;-:%-O:oo.",
-" &o*>+;+;@>>>o:>*+>;=%;;:-,->oo.",
-" oo@>-;+>*:,:@>-@:,>>:-@,:,->oo.",
-".oo*>;%=;@>$;-o:*<::>,;:-:>@>oo.",
-".&o->%>::%-&:>o:@:o@@>@-:*-*;@o.",
-" oo@>>:+>*>@:>#:@,>@><;;-:>->oo ",
-" &o*;--::@*;;--%*;:>::*@:;-;;oo.",
-" &o@@@OO@o@*@*@@@@@*-@@@*@@@@oo.",
-" oooo&o&oo:,,,,>@,>>,,@@<>>>>@o ",
-".&o&ooo&o@>-;-*:@,::>:*;<;*-,oo.",
-" ooo&ooooO>@:-@:@,;;:;-;>*@*>oo ",
-" oo&o&ooo@:,>>->@,;;+>*:>**@>oo.",
-".oooooooo@>->-:-*>>%+>-@,,*:>@o ",
-".&o&o&&oo@>*--*:@,*>>+-;,;*;>&o ",
-" oooooooo@>-::@:@,>>+,*;-*:*:oo ",
-" oooooooo@;>>,->*>>:>>@%:>>>:@o ",
-" o&oo&&oo@O@@*%o@@@**@@@@@*@@oo ",
-" &o&ooooo&ooo@ooooo@oooo@oooooo ",
-"   ..  . ..     . .   .  . .    "
-};
Index: debian/icons/ksirtet.xpm
===================================================================
RCS file: debian/icons/ksirtet.xpm
diff -N debian/icons/ksirtet.xpm
--- debian/icons/ksirtet.xpm	25 Jun 2002 02:34:24 -0000	1.1.1.1
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,61 +0,0 @@
-/* XPM */
-static char *ksirtet[] = {
-/* columns rows colors chars-per-pixel */
-"32 32 23 1",
-"  c opaque",
-". c #191919",
-"X c gray20",
-"o c #00007f",
-"O c #007f00",
-"+ c #007f7f",
-"@ c #7f0000",
-"# c #7f007f",
-"$ c #7f7f00",
-"% c #4c4c4c",
-"& c #666667",
-"* c gray50",
-"= c blue",
-"- c green",
-"; c cyan",
-": c red",
-"> c yellow",
-", c gray60",
-"< c #b2b2b2",
-"1 c gray80",
-"2 c gray90",
-"3 c white",
-"4 c None",
-/* pixels */
-"44444444444. 4444444444444444444",
-"4444444444 ,$ 444444444444444444",
-"444444444 >,,*.44444444444444444",
-"44444444.-,-->-.4444444 44444444",
-"4444444 +O$>>,.+ 44444.: 4444444",
-"444444 ,>OO1,.O>$ 444 *:: 444444",
-"44444.,>>,$.-$,-<,.4 :::*:.44444",
-"4444.*>,<>, O,>,>--.@,:,::@ 4444",
-"444.$O$1>>.@:.$>>-%$:@:*,:..4444",
-"44.,1,O,<O:,,:.,,X:::@#:@. 44444",
-"4.$11>>O.:,::<:$O*:,,:@X@ 444444",
-" >2>><,.%@<<,,:@%:<:,::X.4444444",
-" .>1<-.,1$@1,:@,:X:<,:X==.444444",
-"4.X-$.>12>%@,@:,,:@@@#*,<+ 44444",
-"44 .X>32221$:<<<<:*@.;1;;==.4444",
-"444 $&222>$$$@<::<:@#o<=,<o& 444",
-"44 >3$X12$$12$@,1*@,:@+<=+o,=.44",
-"4 <223$$$>2111$@@:<:,:#%o*;,;= 4",
-" <22232$$2221>1$@<<,:<:.%1=,;<*.",
-".$2222$,>%2223*&*@1:,,#o+o1;,=o ",
-".$$32$>23>%22$*;1=X1:#o11+o<,oo ",
-"4 $$><2322,X,=21;2+&#&;1;1=+=o.4",
-"44 %$222222$o12221;..+21<;,oo 44",
-"444 $$2221$X.+2221o*1o<;<,oo 444",
-"4444 $$22$$  o%12o,21,o<<=o.4444",
-"44444 $$$%.44 oo=;1;;2;+oo 44444",
-"444444.%$ 4444 o%12221<oo 444444",
-"4444444  444444.o+222;oo 4444444",
-"4444444444444444 oo<1oo 44444444",
-"44444444444444444 oooo.444444444",
-"444444444444444444 oX 4444444444",
-"4444444444444444444..44444444444"
-};
Index: debian/icons/ksmiletris.xpm
===================================================================
RCS file: debian/icons/ksmiletris.xpm
diff -N debian/icons/ksmiletris.xpm
--- debian/icons/ksmiletris.xpm	25 Jun 2002 02:34:24 -0000	1.1.1.1
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,54 +0,0 @@
-/* XPM */
-static char *ksmiletris[] = {
-/* columns rows colors chars-per-pixel */
-"32 32 16 1",
-"  c opaque",
-". c #191919",
-"X c gray20",
-"o c #7f0000",
-"O c #7f007f",
-"+ c #7f7f00",
-"@ c #4c4c4c",
-"# c #666667",
-"$ c gray50",
-"% c yellow",
-"& c gray60",
-"* c #b2b2b2",
-"= c gray80",
-"- c gray90",
-"; c white",
-": c None",
-/* pixels */
-":::::::::::::       ::::::::::::",
-"::::::::::. .%=%%==%  .:::::::::",
-"::::::::. %%=%%=%=%*%*%o :::::::",
-"::::::: =%==%==%%%*%*%&&%.::::::",
-":::::: %%=%%=%%=*%*%%*%&%* :::::",
-"::::: %==%--==-%%=%**%*%&%& ::::",
-"::::.==%----=%=%*%*%%%%&%&%$ :::",
-"::: %%=;-;;&&%=%%*%$#**%&%&%&.::",
-"::: %==---&+X+*%%*+X@+%&%&&%$.::",
-"::.%=%----+XX+%=*%+XX+&%&%%&%$ :",
-"::.*%-----+XXX%*%*@X.o%*%&%&&%.:",
-"::.%=%--=-#X o*%*%+X.X&%&%&%+& :",
-": %=%==%%-+  X%*%*+  @%&&%&$%%$ ",
-": =%=%=-=%*@@%*%*%%Xo&&%%&%%*&& ",
-": *=%%=%%=%%*%*&*%&%&%&%&%%&&%% ",
-":.%%%=%==%*%%*%%%&%*%&%&%&&%&+& ",
-": %=*%*%%&%**%*%&%&%%&%%&%&+%%$.",
-":.*%%*%*+%*%%&%&%&%&&%&&++%&&&+.",
-":.%**%%%O%%*&%&%&%*%%&%*#&%$&%&.",
-": %*%**% $%*%&%&%&%&&%&+.+&%+*+.",
-":: %%*%&+.$%&%*%&%%&%&+.&%$%*+o:",
-":: **%*%%.+$%&%&&&&%%+@ %*+&%$.:",
-":: %%*%*%& @*%%%%&%*&o %$&%&%$ :",
-":::.&%&%*%+ .+&&&%$+..+*$%+&$ ::",
-":::.&&%&*%&%@... ...++$%+*%$+.::",
-":::: %&%%&%&*++++#++*$%*$%&+.:::",
-"::::: *%*%&%%&%*&%%&%&$++$&.::::",
-":::::: &+&%&&%$+&&*+%&%&*%.:::::",
-"::::::: &%$%$%%$%&%&$+*$+.::::::",
-"::::::::  +*+&$%$++&%&% .:::::::",
-"::::::::::. o&+*&+%$X  :::::::::",
-"::::::::::::: ......::::::::::::"
-};
Index: debian/icons/ksnake.xpm
===================================================================
RCS file: debian/icons/ksnake.xpm
diff -N debian/icons/ksnake.xpm
--- debian/icons/ksnake.xpm	25 Jun 2002 02:34:24 -0000	1.1.1.1
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,55 +0,0 @@
-/* XPM */
-static char *ksnake[] = {
-/* columns rows colors chars-per-pixel */
-"32 32 17 1",
-"  c opaque",
-". c #191919",
-"X c gray20",
-"o c #007f00",
-"O c #007f7f",
-"+ c #7f0000",
-"@ c #4c4c4c",
-"# c #666667",
-"$ c gray50",
-"% c cyan",
-"& c red",
-"* c gray60",
-"= c #b2b2b2",
-"- c gray80",
-"; c gray90",
-": c white",
-"> c None",
-/* pixels */
-">>>> >         > >>>>>>>>>>>>>>>",
-">>   > .  . .  >  >>>>>>>>>>>>>>",
-">>> > @-%.$=OO. > >>>>>>>>>>>>>>",
-">> >> @%O.O*$O@.  > >>>>>>>>>>>>",
-">  > @*X@@$#.X:. > > >>>>>>>>>>>",
-" > > #;X;=-*#:;X  > > >>>>>>>>>>",
-">   .@-:;O*::;=OOX > >>>>>>>>>>>",
-"    O%O*O%OO**$%%O. > >>>>>>>>>>",
-">  O%%%O%O%OOO%O*%O > >>>> >   >",
-"  O%OO=%O%O%%%OO%%O > >>> >  >  ",
-"  %%$$%#$O%=OX.O%%O.> >> > >   >",
-"> .X%%%%%%OOOOo$%O. > >>> > OO  ",
-">   .X....OOO%O%OO.> >>  > XOO..",
-" > +&@. . @O*%%%%.     > > OO. >",
-"> .=$ > > O%%%%%O >  >  > OO. > ",
-"> .&& > > O%%O*O.      > O%O >  ",
-">     >> @O%%%%O >    > O%%. > >",
-">    > > O%**O%  >   > O%O. > >>",
-" >    > O%%%%%O. >   > @%O > > >",
-">       OO*%%O. > > > X%*X > > >",
-">> > > O%%%OOO. > > > O%%X  > >>",
-"> >    O%*=%%. > >> > XO%%O. >  ",
-">> >  O*%*%%O        > O%$OX > >",
-"  >  o%%%%%O. > >>>>.> .O%%O  > ",
-"> > XO%%%O%o.          .X%%*. > ",
-" > .O%%*%*O@XOO@OOO@O@OXOO%O..> ",
-" > .O%%%%%%%%%%%%*%%%%%%%%O%. > ",
-" >> O$%O*%OOO%*%%%O%%*O%%*%X.>> ",
-"> >> .OO%O%%O%O%O%O%%O%%OX..>> >",
-" > >  .XXX.X@XOX@XXXXXX..   > > ",
-"> >  >.                   >  > >",
-">> >   > >>>>>>>>>>>>>> >   > >>"
-};
Index: debian/icons/ksokoban.xpm
===================================================================
RCS file: debian/icons/ksokoban.xpm
diff -N debian/icons/ksokoban.xpm
--- debian/icons/ksokoban.xpm	25 Jun 2002 02:34:24 -0000	1.1.1.1
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,54 +0,0 @@
-/* XPM */
-static char *ksokoban[] = {
-/* columns rows colors chars-per-pixel */
-"32 32 16 1",
-"  c opaque",
-". c #191919",
-"X c #7f0000",
-"o c #7f007f",
-"O c #7f7f00",
-"+ c #4c4c4c",
-"@ c #666667",
-"# c gray50",
-"$ c red",
-"% c yellow",
-"& c gray60",
-"* c #b2b2b2",
-"= c gray80",
-"- c gray90",
-"; c white",
-": c None",
-/* pixels */
-":::::::::::::::: :::::::::::::::",
-":::::::::::::::.&.::::::::::::::",
-":::::::::::::: *-X :::::::::::::",
-"::::::::::::: #-@#X.::::::::::::",
-":::::::::::::.=*X&$.::::::::::::",
-"::::::::::::.*=@oO#+ :::::::::::",
-"::::::::::: &*-+$#$$X ::::::::::",
-":::::::::::X*=&O&O##@ ::::::::::",
-":::::::::: =*-&o@#$#@X :::::::::",
-":::::::::.&===$O$$#O$@X ::::::::",
-":::::::::.**--+$#OOo#Oo.::::::::",
-":::::::: ==*;#X#@$#$o@O+ :::::::",
-"::::::: #===-X#$$@X#@@X$X ::::::",
-":::::::.=*=-=$O$#$#$$@#X+.::::::",
-":::::: ***=-#$##$#$O$##XXO.:::::",
-"::::: *==*=-$O$$$O$o#$O#XoX ::::",
-":::::X**==--$O#$#$#$Oo@#X@$.::::",
-":::: ***=--=$$$$$O$@$@O$#XOX :::",
-"::: #=*===;$&$$#$@$@@$@$#X@@+ ::",
-"::: --==**;$$$$&$$@$$#XO$@$@X.::",
-":::: +--*=*$$$$$#$O$#$Oo@$@+.:::",
-"::::: X@--*$%$&$O$$$$@$#X@X ::::",
-"::::::.XX@O@#@+@#@@@@@XX@X :::::",
-"::::::: XX+XX+Xo+XX@+X@XX.::::::",
-":::::::: X+X+XX@X@X@+X+X.:::::::",
-"::::::::: X@X+@X+X@XX+X ::::::::",
-"::::::::::.X@oO+oO+O@X :::::::::",
-":::::::::::.OXX++XX@X ::::::::::",
-"::::::::::::.@@XoX@X :::::::::::",
-":::::::::::::.X+O@X.::::::::::::",
-"::::::::::::::  @  :::::::::::::",
-":::::::::::::::: :::::::::::::::"
-};
Index: debian/icons/kspaceduel.xpm
===================================================================
RCS file: debian/icons/kspaceduel.xpm
diff -N debian/icons/kspaceduel.xpm
--- debian/icons/kspaceduel.xpm	25 Jun 2002 02:34:24 -0000	1.1.1.1
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,53 +0,0 @@
-/* XPM */
-static char *kspaceduel[] = {
-/* columns rows colors chars-per-pixel */
-"32 32 15 1",
-"  c opaque",
-". c #191919",
-"X c #00007f",
-"o c #7f7f00",
-"O c #4c4c4c",
-"+ c #666667",
-"@ c gray50",
-"# c blue",
-"$ c red",
-"% c yellow",
-"& c gray60",
-"* c #b2b2b2",
-"= c gray80",
-"- c gray90",
-"; c None",
-/* pixels */
-";;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;",
-";;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;",
-";;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;",
-";;;;;;;;;;;;;;;;;;;;;;;;;;;;  ;;",
-";;;;;;;;;;;;;;;;;;;;;;;;;;; $$ ;",
-";;;;;;;;;;;;;;;;;;;;;;;;;;; $$ ;",
-";;;;;;;;;;;;;;;;;;;;;;;;;;;;  ;;",
-";;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;",
-";;;;;;;;;;;;;;;;;;;  ;;;;;;;;;;;",
-";;;;;;;;;;;;;;;;;; $$ ;;;;;;;;;;",
-";;;;;;;;;;;;;;;;;; $$ ;;;;;;;;;;",
-";; ;;;;;;;;;;;;;;;;  ;;;;;;;;;;;",
-"; # ;;;;;;;;;;;;;;;;;;;;;;;;;;;;",
-" ### ;;;;;;;;;;;;;;;;;;;;;;;;;;;",
-" ###X ;;;;;;;;;;;;;;;;;;;;;;;;;;",
-"; #X## ;;;;;;;;;;;;;;;;;;;;;;;;;",
-";; ###X ;;;;   ;;;;;;;;;;;;;;;;;",
-";;; #X## ;; $% ;;;;;;;;;;;;;;;;;",
-";;;; #### .$%& ;;;;;;;;;;;;;;;;;",
-";;;;; ##  o$%= ;;;;;;;;;;;;;;;;;",
-";;;;;;   $o&- ;;;;;;;;;;;;;;;;;;",
-";;;;;;; o%%= ;;;;;;;;;;;;;;;;;;;",
-";;;;;; $$%=.  ;;;;;;;;;;;;;;;;;;",
-";;;;  O%*=  ## ;;;;;;;;;;;;;;;;;",
-";;;; +@@=  #### ;;;;;;;;;;;;;;;;",
-";;;;; &* ;; ##X# ;;;;;;;;;;;;;;;",
-";;;;;; * ;;; X### ;;;;;;;;;;;;;;",
-";;;;;;;  ;;;; ##X# ;;;;;;;;;;;;;",
-";;;;;;;;;;;;;; X### ;;;;;;;;;;;;",
-";;;;;;;;;;;;;;; ### ;;;;;;;;;;;;",
-";;;;;;;;;;;;;;;; # ;;;;;;;;;;;;;",
-";;;;;;;;;;;;;;;;; ;;;;;;;;;;;;;;"
-};
Index: debian/icons/ktron.xpm
===================================================================
RCS file: debian/icons/ktron.xpm
diff -N debian/icons/ktron.xpm
--- debian/icons/ktron.xpm	25 Jun 2002 02:34:24 -0000	1.1.1.1
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,47 +0,0 @@
-/* XPM */
-static char *ktron[] = {
-/* columns rows colors chars-per-pixel */
-"32 32 9 1",
-"  c opaque",
-". c #191919",
-"X c gray20",
-"o c #00007f",
-"O c #7f0000",
-"+ c #4c4c4c",
-"@ c #666667",
-"# c blue",
-"$ c red",
-/* pixels */
-"                                ",
-"                                ",
-"  XXXXXX ################ XXXX+@",
-"  XXXXXX #o##o#oo#o#o#o#o XXXX@+",
-"  XX.XXX ##ooooooooooo#oo XXXX+@",
-"  XXXXXX #oo          ##o XXX.@+",
-"         ##o XX.XXXXX #oo XXXX++",
-"  ########oo XXXXXXXX ##o XX.X@@",
-"  ###o#o#o#o XXXXXXXX #oo XX.X++",
-"  #ooooooooo XXXXX.XX ##o XXXX+@",
-"  ##o             XXX #oo XXXX@+",
-"  #oo $$$$$$$$$$$ XXX ##o XXXX+@",
-"  #oo $O$O$O$OO$O XXX ##o XXXX++",
-"  ##o $$OOOOOO$$O XXX #oo X.XX+@",
-"  #oo $OO     $OO XXX #oo XXXX+@",
-"  ##o $$O XXX $$O XXX ##o XXXX@+",
-"  #oo $$O X.X $$O XXX ##o XXXX++",
-"  ##o $OO XXX $OO XXX     XXXX@+",
-"  #oo     XXX $$O             +@",
-"  ##o XXXXXXX $$$$$$$$$$$$$$O @+",
-"  ##o XXXXXXX $O$O$O$O$O$O$OO +@",
-"  #oo XXXXX.X OOOOOOOOOOOO$$O @+",
-"  ##o XXXXXXX             $OO +@",
-"      XX.XXXXXXXXXXXXXXXX $$O @+",
-"  XX.XXXXXXXXXXXXXXXXXXXX $$O +@",
-"  XXXXXXXX.XXXX           $OO @+",
-"  XXXXX.XXXXXXX $$$$$$$$$$$$O +@",
-"  XXXXXX.XXXX.X $OO$$OO$$O$OO ++",
-"  XXXXXXXX.XXXX OOOOOOOOOOOOO @+",
-"  XXX.XXXXXXXXX               +@",
-"+@+@+@+++@++++@@+@+@@@+@+@+@@+@+",
-"@++@@+@+@+@@@+++@+@+++@++@@++@+@"
-};
Index: debian/icons/ktuberling.xpm
===================================================================
RCS file: debian/icons/ktuberling.xpm
diff -N debian/icons/ktuberling.xpm
--- debian/icons/ktuberling.xpm	25 Jun 2002 02:34:24 -0000	1.1.1.1
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,58 +0,0 @@
-/* XPM */
-static char *ktuberling[] = {
-/* columns rows colors chars-per-pixel */
-"32 32 20 1",
-"  c #191919",
-". c gray20",
-"X c #00007f",
-"o c #007f00",
-"O c #007f7f",
-"+ c #7f0000",
-"@ c #7f7f00",
-"# c #4c4c4c",
-"$ c #666667",
-"% c gray50",
-"& c blue",
-"* c green",
-"= c cyan",
-"- c red",
-"; c yellow",
-": c gray60",
-"> c #b2b2b2",
-", c gray80",
-"< c gray90",
-"1 c white",
-/* pixels */
-"1111111111111<,>:>1111111111<111",
-"111111111111<%*$*O,11<1111111111",
-"111111111111,o$o@:%:<<1111111111",
-"111111111111:o*%%@:@%>1111111111",
-"111111111111:@:%%@@>>%<111111111",
-"11111111111<%%%:%:%:>@<<11111111",
-"1111111<11<%:# @@::++:@<11111111",
-"11111,::11:$:@%%%-%%:%%>11111111",
-"11111:::<1>#:%;:@:@::@@>11111111",
-"1111<:>,:><%-::%-->%@%$<11111111",
-"11111>>>>11%$:%%:--::%>1<1111111",
-"11111<::111:@:@-%:@%:-,111111111",
-"1111<1$#11>.$@@:@-:%@:$:,<111111",
-"111111>.1< .$:+:>:>@%@+# .>11111",
-"111111< $#$>#:%%:>%%:%:11$.11111",
-"1111111,.#<,@%@:%%@%@%>>: %11111",
-"11111111111<.$%@%%-:@:1,$%111111",
-"111111111111%@%%@::@#$,><1111111",
-"1111111111<1<$#$@%:;$><111111111",
-"1111111111111<###$$$>11<11111111",
-"11111111111111> :##<111111111111",
-"111111111111<1%#<:#1111111111111",
-"11111111111111$$1>.1111111111111",
-"1111111111111<%%1>.1111111111111",
-"11111111111<11%$1>.<<<1111111111",
-"111111111<,,>,%X,:.,,,,,<1111111",
-"1111111<,>:::,::>#O>::=:><<11111",
-"1111111:&:=>:&:>,$%:,&,&>>111111",
-"1111111:O::>:>><:>,>>>::>><11111",
-"1111111<%%:&>,<>O<1<,&:::><11111",
-"11111111,%:&O:>%:>>>::=>>,111111",
-"11111111<<,:::&::>&:>>,<<<111111"
-};
Index: debian/icons/kwin4.xpm
===================================================================
RCS file: debian/icons/kwin4.xpm
diff -N debian/icons/kwin4.xpm
--- debian/icons/kwin4.xpm	25 Jun 2002 02:34:24 -0000	1.1.1.1
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,53 +0,0 @@
-/* XPM */
-static char *kwin4[] = {
-/* columns rows colors chars-per-pixel */
-"32 32 15 1",
-"  c opaque",
-". c #191919",
-"X c gray20",
-"o c #00007f",
-"O c #007f7f",
-"+ c #7f0000",
-"@ c #7f007f",
-"# c #7f7f00",
-"$ c #4c4c4c",
-"% c #666667",
-"& c gray50",
-"* c blue",
-"= c red",
-"- c yellow",
-"; c gray60",
-/* pixels */
-"                                ",
-"                     .          ",
-"  *oooooooooooooooooooooooooo*. ",
-"  o*oooooooooooooooooooooooo**  ",
-"  **oooooooooooooooooooooooo**  ",
-"  o**ooooooooooooooooooooooo**  ",
-"  o*oooooooo@ooooooooooooooo**  ",
-"  **ooooooo@+=+ooooooooooooo**  ",
-"  **ooooooo+==@ooooooooooooo**  ",
-"  o*oooooo@@==+ooooooooooooo**  ",
-"  ***oooooo++=+ooooooooooooo**  ",
-"  o*oo$o$ooo+@Xoooo$oooooooo**  ",
-"  o*o$#-%oo@=+@oo$-#%ooooooo**  ",
-"  **o&---Xo====o$&---$oooooo**  ",
-"  **o#--#$o+==+Xo%---ooooooo**  ",
-"  o*o%-;&oo+@=@oo$-##ooooooo**  ",
-"  **oX$%Xooo@@o@o$%%%Xoooooo**  ",
-"  o**@+=+oX@+=+oo%#-&ooooooo**  ",
-"  o*o+===@o===+oo#--#ooooooo**. ",
-"  **@@===oo+===X$%---$oooooo**  ",
-"  **o@++@oo+=+@oo$#-&ooooooo**  ",
-"  o*oo$@XoooX@ooo$@$$ooX$$oo**  ",
-"  **o%-#&oo+=+@oo++=+oo$;-$o**  ",
-"  o*o%---oo===+oo@===o$&---o**  ",
-"  **o#---oo+==+oo+===$o#---o**. ",
-"  o*o%#-#oo@+=@oo@++@oo###&o**  ",
-" .***oo$$oooo@ooooo@oooo$ooo**  ",
-" .**oooooooooooooooooooooooo**. ",
-".O****************************o.",
-"**.X************************.o**",
-"*o.**oooooooooooooooooooooo**..*",
-"****oooooooooooooooooooooooo****"
-};
Index: debian/icons/lskat.xpm
===================================================================
RCS file: debian/icons/lskat.xpm
diff -N debian/icons/lskat.xpm
--- debian/icons/lskat.xpm	25 Jun 2002 02:34:24 -0000	1.1.1.1
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,51 +0,0 @@
-/* XPM */
-static char *lskat[] = {
-/* columns rows colors chars-per-pixel */
-"32 32 13 1",
-"  c opaque",
-". c #191919",
-"X c gray20",
-"o c #00007f",
-"O c #007f7f",
-"+ c #7f0000",
-"@ c #7f007f",
-"# c #7f7f00",
-"$ c #4c4c4c",
-"% c #666667",
-"& c red",
-"* c yellow",
-"= c white",
-/* pixels */
-"oXoXoXoXXoXooXoXoXoXoXo.oXo.oXo.",
-"ooXo.oXo%+X&&+XoXooXXoXooooXXoXo",
-"XooXoXo#&o&@o&ooXoXooXoXX%XoXooX",
-"oXXoXoo&#oX&O+o$oXXoXo.oX+oXXXoo",
-"XooXoXoX+&$Xo#Xo+Xo++@#o&#+XooXo",
-"ooXoXoXo$+@$X&@+ooXo+$&o@&oo.ooX",
-"oXXXoXo+o==========$@o&oo#o.XoXo",
-"Xooo.oO&&.+=&+&==+=+&#&O$&$XoXoX",
-"ooXooXoXo *====   =oXooXoooXoXoX",
-"oXoXXoXoX= === == =.ooXooXXoXo.o",
-"XoXooXoXo= =&  &  =oXoo.o.oXoXoX",
-"oXoXXoXoX=*=&**&  =XoXXoXoXooXoo",
-"oXXoXoXoo=* = **==========o.XXoo",
-"Ooo.oXo..=**=***====== ===XooXoX",
-"o.ooXo.oo=* = =& =   == ==oXo.Xo",
-"oX==========  = == ==&========oX",
-"o.&&========= ====  =&&&===&==oX",
-"oo=========== &&==  &=&&==&&==Xo",
-"oX============================o.",
-"Oo==========XooX==  ===&===&==Xo",
-"oX====&=====XoXo== ===&&=&&&==o.",
-".o====&&====oXoX========&&====Xo",
-"o.=====&====Xo.o==  ==&&==&&==oo",
-"oo==========oXoo== ===&&==&&==oX",
-"oX=========&oXXX==============oX",
-"Oo=========&.ooXoXoo===&===&=&.o",
-"oX========&=ooXo.ooX==&&===&==oX",
-".o=========&oXoXoX.o=========&oo",
-"oXoXoXooXoXooXXoXoXooXo.oXoXX.oX",
-"oo.o.oXXoXoXXooXoXoXXoXoXoXoooXo",
-"oXo.oXooXoo.oXXooXXooXo.oXoXXooX",
-"OoOoXooXoXoOoooXooXooXoOooXooXoo"
-};
Index: kbackgammon/Makefile.am
===================================================================
RCS file: /Volumes/src/cvs/od/proj/KDE-Darwin/kdegames/kbackgammon/Makefile.am,v
retrieving revision 1.1.1.3
retrieving revision 1.4
diff -u -u -r1.1.1.3 -r1.4
--- kbackgammon/Makefile.am	14 Aug 2002 16:26:28 -0000	1.1.1.3
+++ kbackgammon/Makefile.am	14 Aug 2002 18:49:23 -0000	1.4
@@ -7,7 +7,8 @@
 CLEANFILES = dummy.cpp
 
 kbackgammon_la_LIBADD = $(LIB_KDEUI) $(LIB_KSYCOCA) -lkdeprint \
-			./engines/libkbgengines.la
+			$(top_builddir)/kbackgammon/engines/libkbgengines.la \
+			$(LIB_KDECORE) $(LIB_QT) $(top_builddir)/libkdegames/libkdegames.la
 kbackgammon_la_SOURCES = main.cpp kbg.cpp kbgboard.cpp kbgtextview.cpp \
 			kbgstatus.cpp
 
Index: kbackgammon/engines/Makefile.am
===================================================================
RCS file: /Volumes/src/cvs/od/proj/KDE-Darwin/kdegames/kbackgammon/engines/Makefile.am,v
retrieving revision 1.1.1.1
retrieving revision 1.2
diff -u -u -r1.1.1.1 -r1.2
--- kbackgammon/engines/Makefile.am	25 Jun 2002 02:34:25 -0000	1.1.1.1
+++ kbackgammon/engines/Makefile.am	25 Jun 2002 02:35:57 -0000	1.2
@@ -3,7 +3,7 @@
 libkbgengines_la_SOURCES = dummy.cpp
 libkbgengines_la_LIBADD  = offline/libkbgoffline.la gnubg/libkbggnubg.la \
                            generic/libkbggeneric.la fibs/libkbgfibs.la   \
-			   nextgen/libkbgnextgen.la
+			   nextgen/libkbgnextgen.la $(LIB_KDECORE) $(LIB_QT)
 
 INCLUDES= $(all_includes)
 
Index: kolf/Makefile.am
===================================================================
RCS file: /Volumes/src/cvs/od/proj/KDE-Darwin/kdegames/kolf/Makefile.am,v
retrieving revision 1.1.1.3
retrieving revision 1.4
diff -u -u -r1.1.1.3 -r1.4
--- kolf/Makefile.am	29 Oct 2002 15:53:13 -0000	1.1.1.3
+++ kolf/Makefile.am	29 Oct 2002 16:17:03 -0000	1.4
@@ -11,7 +11,8 @@
 	floater.cpp slope.cpp
 
 libkolf_la_LDFLAGS = $(all_libraries) $(KDE_RPATH) -export-dynamic
-libkolf_la_LIBADD = $(LIB_KDEGAMES) $(LIB_KDEPRINT) $(LIB_KIO) -lartskde
+libkolf_la_LIBADD = $(LIB_KDEGAMES) $(LIB_KDEPRINT) $(LIB_KIO) -lartskde \
+	-lartsflow_idl -lsoundserver_idl -lmcop
 
 # the installed header files
 kolfinclude_HEADERS = \
@@ -22,16 +23,16 @@
 
 
 # kdeinited lib
-kde_module_LTLIBRARIES = kolf.la
-kolf_la_SOURCES = main.cpp
-kolf_la_LDFLAGS = $(KDE_RPATH) $(all_libraries) -module -avoid-version
-kolf_la_LIBADD = libkolf.la
+kde_module_LTLIBRARIES = kolfinit.la
+kolfinit_la_SOURCES = main.cpp
+kolfinit_la_LDFLAGS = $(KDE_RPATH) $(all_libraries) -module -avoid-version
+kolfinit_la_LIBADD = libkolf.la
 
 
 # the empty "stub" app that has nothing
 kolf_SOURCES = dummy.cpp
 kolf_LDFLAGS = $(all_libraries) $(KDE_RPATH)
-kolf_LDADD   = kolf.la
+kolf_LDADD   = kolfinit.la
 
 dummy.cpp:
 	echo > dummy.cpp
Index: kpat/freecell-solver/freecell.c
===================================================================
RCS file: /Volumes/src/cvs/od/proj/KDE-Darwin/kdegames/kpat/freecell-solver/freecell.c,v
retrieving revision 1.1.1.1
retrieving revision 1.2
diff -u -u -r1.1.1.1 -r1.2
--- kpat/freecell-solver/freecell.c	25 Jun 2002 02:34:26 -0000	1.1.1.1
+++ kpat/freecell-solver/freecell.c	25 Jun 2002 02:35:57 -0000	1.2
@@ -12,7 +12,9 @@
 #include <stdio.h>
 #include <math.h>
 #if FCS_STATE_STORAGE==FCS_STATE_STORAGE_LIBREDBLACK_TREE
+#ifndef __APPLE__
 #include <search.h>
+#endif
 #endif
 
 
Index: kpat/freecell-solver/intrface.c
===================================================================
RCS file: /Volumes/src/cvs/od/proj/KDE-Darwin/kdegames/kpat/freecell-solver/intrface.c,v
retrieving revision 1.1.1.1
retrieving revision 1.2
diff -u -u -r1.1.1.1 -r1.2
--- kpat/freecell-solver/intrface.c	25 Jun 2002 02:34:26 -0000	1.1.1.1
+++ kpat/freecell-solver/intrface.c	25 Jun 2002 02:35:57 -0000	1.2
@@ -16,7 +16,9 @@
 #include <sys/stat.h>
 #include <fcntl.h>
 #if FCS_STATE_STORAGE==FCS_STATE_STORAGE_LIBREDBLACK_TREE
+#ifndef __APPLE__
 #include <search.h>
+#endif
 #endif
 
 
Index: kpat/freecell-solver/md5.h
===================================================================
RCS file: /Volumes/src/cvs/od/proj/KDE-Darwin/kdegames/kpat/freecell-solver/md5.h,v
retrieving revision 1.1.1.2
retrieving revision 1.3
diff -u -u -r1.1.1.2 -r1.3
--- kpat/freecell-solver/md5.h	14 Aug 2002 16:24:51 -0000	1.1.1.2
+++ kpat/freecell-solver/md5.h	14 Aug 2002 18:49:24 -0000	1.3
@@ -14,7 +14,11 @@
 #elif defined(__osf__)
 typedef unsigned int u_int32_t;
 #elif !defined(__linux__) && !defined(__NetBSD__) && !defined(__FreeBSD__)
+# ifdef __APPLE__
+# include <sys/types.h>
+# else
 typedef uint32_t u_int32_t;
+# endif
 #endif  
 
 
Index: kpat/freecell-solver/scans.c
===================================================================
RCS file: /Volumes/src/cvs/od/proj/KDE-Darwin/kdegames/kpat/freecell-solver/scans.c,v
retrieving revision 1.1.1.1
retrieving revision 1.2
diff -u -u -r1.1.1.1 -r1.2
--- kpat/freecell-solver/scans.c	25 Jun 2002 02:34:26 -0000	1.1.1.1
+++ kpat/freecell-solver/scans.c	25 Jun 2002 02:35:57 -0000	1.2
@@ -13,7 +13,9 @@
 #include <stdio.h>
 #include <math.h>
 #if FCS_STATE_STORAGE==FCS_STATE_STORAGE_LIBREDBLACK_TREE
+#ifndef __APPLE__
 #include <search.h>
+#endif
 #endif
 
 #include "config.h"
Index: libksirtet/lib/mp_interface.cpp
===================================================================
RCS file: /Volumes/src/cvs/od/proj/KDE-Darwin/kdegames/libksirtet/lib/mp_interface.cpp,v
retrieving revision 1.1.1.1
retrieving revision 1.2
diff -u -u -r1.1.1.1 -r1.2
--- libksirtet/lib/mp_interface.cpp	10 Jul 2002 04:33:13 -0000	1.1.1.1
+++ libksirtet/lib/mp_interface.cpp	13 Jul 2002 18:11:52 -0000	1.2
@@ -1,5 +1,6 @@
 #include "mp_interface.h"
 #include "mp_interface.moc"
+#include "mp_board.moc"
 
 #include <qpainter.h>
 #include <qlayout.h>
