Index: atlantik/libatlantikclient/klatencytimer.cpp
===================================================================
RCS file: /home/kde/kdegames/atlantik/libatlantikclient/klatencytimer.cpp,v
retrieving revision 1.2
diff -u -r1.2 klatencytimer.cpp
--- atlantik/libatlantikclient/klatencytimer.cpp	24 Jul 2003 18:33:29 -0000	1.2
+++ atlantik/libatlantikclient/klatencytimer.cpp	3 Nov 2003 20:13:16 -0000
@@ -17,6 +17,7 @@
  *  Boston, MA 02111-1307, USA.
  */
 
+#include "config.h"
 #include <sys/types.h>
 #include <sys/socket.h>
 #include <netinet/in.h>
