Index: kdecore/kpty.cpp
===================================================================
RCS file: /home/kde/kdelibs/kdecore/kpty.cpp,v
retrieving revision 1.9
diff -u -r1.9 kpty.cpp
--- kdecore/kpty.cpp	29 Sep 2003 08:52:52 -0000	1.9
+++ kdecore/kpty.cpp	10 Nov 2003 18:33:56 -0000
@@ -225,6 +225,7 @@
       struct group* p = getgrnam(TTY_GROUP);    /* posix */
       gid_t gid = p ? p->gr_gid : getgid ();    /* posix */
 
+#if !defined(__APPLE__)
       if (fchown(d->slaveFd, (uid_t) -1, gid) < 0)
       {
          int e = errno;
@@ -239,6 +240,7 @@
          kdWarning(175) << "Cannot chmod " << d->ttyName << endl
                         << "Reason " << strerror(e) << endl;
       }
+#endif
       goto gotpty;
     }
 #endif
