Index: configure.in.in
===================================================================
RCS file: /home/kde/kdemultimedia/configure.in.in,v
retrieving revision 1.49
diff -u -u -r1.49 configure.in.in
--- configure.in.in	20 Oct 2003 08:56:30 -0000	1.49
+++ configure.in.in	12 Nov 2003 17:34:41 -0000
@@ -197,3 +197,4 @@
 
 AC_PATH_CDPARANOIA
 
+DO_NOT_COMPILE="$DO_NOT_COMPILE kscd"
Index: audiofile_artsplugin/Makefile.am
===================================================================
RCS file: /home/kde/kdemultimedia/audiofile_artsplugin/Makefile.am,v
retrieving revision 1.4
diff -u -u -r1.4 Makefile.am
--- audiofile_artsplugin/Makefile.am	9 Mar 2002 10:50:19 -0000	1.4
+++ audiofile_artsplugin/Makefile.am	12 Nov 2003 17:34:43 -0000
@@ -2,7 +2,7 @@
 
 noinst_HEADERS = audiofilePlayObjectI.h
 
-lib_LTLIBRARIES = libaudiofilearts.la
+kde_module_LTLIBRARIES = libaudiofilearts.la
 libaudiofilearts_la_COMPILE_FIRST = audiofilearts.h
 libaudiofilearts_la_SOURCES = audiofilearts.cc audiofilePlayObjectI.cpp
 libaudiofilearts_la_LDFLAGS = $(all_libraries) -module -avoid-version -no-undefined
Index: kmid/Makefile.am
===================================================================
RCS file: /home/kde/kdemultimedia/kmid/Makefile.am,v
retrieving revision 1.58
diff -u -u -r1.58 Makefile.am
--- kmid/Makefile.am	4 Mar 2003 16:01:28 -0000	1.58
+++ kmid/Makefile.am	12 Nov 2003 17:34:43 -0000
@@ -1,17 +1,25 @@
 INCLUDES = $(all_includes)
 
 kde_module_LTLIBRARIES = libkmidpart.la
+lib_LTLIBRARIES = libkmidcommon.la
 
-libkmidpart_la_LDFLAGS = $(all_libraries) $(KDE_PLUGIN) -module
-libkmidpart_la_LIBADD = $(LIB_KFILE) $(LIB_KPARTS) $(LIB_KIO) -lkmid
+libkmidcommon_la_LDFLAGS = $(all_libraries) $(KDE_RPATH) -no-undefined
+libkmidcommon_la_LIBADD = $(LIB_KFILE) $(LIB_KPARTS) $(LIB_KIO) -lkmid
 
-libkmidpart_la_SOURCES = kmidframe.cpp kmidclient.cpp kdisptext.cpp \
+libkmidcommon_la_SOURCES = kmidframe.cpp kmidclient.cpp kdisptext.cpp \
 	qslidertime.cpp klcdnumber.cpp ktrianglebutton.cpp \
 	midicfgdlg.cpp collectdlg.cpp randomlist.cpp \
 	channelview.cpp channel.cpp channel3d.cpp channel4d.cpp \
 	channelcfgdlg.cpp instrname.cpp rhythmview.cpp \
 	songlist.cpp slman.cpp kmid_part.cpp kmidIface.skel
 
+libkmidpart_la_LDFLAGS = $(all_libraries) $(KDE_PLUGIN) -module -avoid-version
+libkmidpart_la_LIBADD  = libkmidcommon.la
+libkmidpart_la_SOURCES = dummy.cpp
+
+dummy.cpp:
+	echo > dummy.cpp
+
 noinst_HEADERS = kmidframe.h kmidclient.h kdisptext.h \
         qslidertime.h klcdnumber.h ktrianglebutton.h \
         midicfgdlg.h collectdlg.h randomlist.h \
@@ -21,7 +29,7 @@
 
 kmid_LDFLAGS = $(all_libraries) $(KDE_RPATH)
 kmid_SOURCES = main.cpp
-kmid_LDADD   = libkmidpart.la
+kmid_LDADD   = libkmidcommon.la
 
 bin_PROGRAMS = kmid
 METASOURCES = AUTO
Index: mpg123_artsplugin/Makefile.am
===================================================================
RCS file: /home/kde/kdemultimedia/mpg123_artsplugin/Makefile.am,v
retrieving revision 1.10
diff -u -u -r1.10 Makefile.am
--- mpg123_artsplugin/Makefile.am	28 May 2002 12:59:17 -0000	1.10
+++ mpg123_artsplugin/Makefile.am	12 Nov 2003 17:34:43 -0000
@@ -4,7 +4,7 @@
 
 noinst_HEADERS = mpg123PlayObject_impl.h
 
-lib_LTLIBRARIES = libmpg123arts.la
+kde_module_LTLIBRARIES = libmpg123arts.la
 libmpg123arts_la_COMPILE_FIRST = mpg123arts.h
 libmpg123arts_la_SOURCES = mpg123arts.cc mpg123PlayObject_impl.cpp dxhead.c
 libmpg123arts_la_LDFLAGS = $(all_libraries) -module -avoid-version -no-undefined
Index: mpg123_artsplugin/mpg123/Makefile.am
===================================================================
RCS file: /home/kde/kdemultimedia/mpg123_artsplugin/mpg123/Makefile.am,v
retrieving revision 1.12
diff -u -u -r1.12 Makefile.am
--- mpg123_artsplugin/mpg123/Makefile.am	27 Jul 2002 09:29:23 -0000	1.12
+++ mpg123_artsplugin/mpg123/Makefile.am	12 Nov 2003 17:34:43 -0000
@@ -27,7 +27,6 @@
 AM_CFLAGS = $(COMMON_CFLAGS) $(ARCH_CFLAGS)
 
 lib_LTLIBRARIES = libmpg123.la
-libmpg123_la_LDFLAGS = -module -avoid-version
 libmpg123_la_LIBADD = ./$(MPG123_PLAT_LIB)
 libmpg123_la_DEPENDENCIES = ./$(MPG123_PLAT_LIB)
 
Index: noatun/library/noatunarts/Makefile.am
===================================================================
RCS file: /home/kde/kdemultimedia/noatun/library/noatunarts/Makefile.am,v
retrieving revision 1.8
diff -u -u -r1.8 Makefile.am
--- noatun/library/noatunarts/Makefile.am	19 Jan 2003 20:24:24 -0000	1.8
+++ noatun/library/noatunarts/Makefile.am	12 Nov 2003 17:34:43 -0000
@@ -1,14 +1,23 @@
 INCLUDES= -I$(kde_includes)/arts $(all_includes)
 KDE_OPTIONS = nofinal
 
-lib_LTLIBRARIES = libnoatunarts.la
-libnoatunarts_la_SOURCES = noatunarts.cc fft.c Equalizer_impl.cpp \
+kde_module_LTLIBRARIES = libnoatunarts.la
+lib_LTLIBRARIES = libnoatunartscommon.la
+
+libnoatunartscommon_la_SOURCES = noatunarts.cc fft.c Equalizer_impl.cpp \
                            FFTScopes.cpp StereoEffectStack_impl.cpp \
                            StereoVolumeControl_impl.cpp Session_impl.cpp
 
-libnoatunarts_la_LDFLAGS = $(all_libraries) -module -avoid-version -no-undefined
-libnoatunarts_la_LIBADD =  -lkmedia2_idl -lsoundserver_idl -lartsflow
-libnoatunarts_la_METASOURCES = AUTO
+libnoatunartscommon_la_LDFLAGS = $(all_libraries) -no-undefined
+libnoatunartscommon_la_LIBADD =  -lkmedia2_idl -lsoundserver_idl -lartsflow
+libnoatunartscommon_la_METASOURCES = AUTO
+
+libnoatunarts_la_LDFLAGS = $(all_libraries) -avoid-version
+libnoatunarts_la_LIBADD  = libnoatunartscommon.la
+libnoatunarts_la_SOURCES = dummy.cpp
+
+dummy.cpp:
+	echo > dummy.cpp
 
 noatunarts.mcoptype: noatunarts.h
 noatunarts.mcopclass: noatunarts.h
Index: noatun/modules/artseffects/Makefile.am
===================================================================
RCS file: /home/kde/kdemultimedia/noatun/modules/artseffects/Makefile.am,v
retrieving revision 1.18
diff -u -u -r1.18 Makefile.am
--- noatun/modules/artseffects/Makefile.am	12 Jul 2003 20:47:18 -0000	1.18
+++ noatun/modules/artseffects/Makefile.am	12 Nov 2003 17:34:43 -0000
@@ -1,5 +1,5 @@
 INCLUDES= -I$(top_builddir)/arts/gui/common -I$(top_srcdir)/arts/gui/common -I$(kde_includes)/arts $(all_includes)
-lib_LTLIBRARIES = libartseffects.la
+kde_module_LTLIBRARIES = libartseffects.la
 #libartseffectsui.la
 
 libartseffects_la_SOURCES = artseffects.cc extrastereo_impl.cc
Index: noatun/modules/winskin/Makefile.am
===================================================================
RCS file: /home/kde/kdemultimedia/noatun/modules/winskin/Makefile.am,v
retrieving revision 1.23
diff -u -u -r1.23 Makefile.am
--- noatun/modules/winskin/Makefile.am	9 Apr 2003 12:32:40 -0000	1.23
+++ noatun/modules/winskin/Makefile.am	12 Nov 2003 17:34:43 -0000
@@ -40,7 +40,7 @@
 					-module -avoid-version -no-undefined 
 
 noatun_winskin_la_LIBADD		= $(top_builddir)/noatun/library/libnoatun.la \
-					$(top_builddir)/noatun/modules/winskin/vis/libwinskinvis.la
+					$(top_builddir)/noatun/modules/winskin/vis/libwinskinviscommon.la
 
 noatun_winskin_la_METASOURCES	=	AUTO
 
Index: noatun/modules/winskin/vis/Makefile.am
===================================================================
RCS file: /home/kde/kdemultimedia/noatun/modules/winskin/vis/Makefile.am,v
retrieving revision 1.6
diff -u -u -r1.6 Makefile.am
--- noatun/modules/winskin/vis/Makefile.am	29 Nov 2002 16:38:36 -0000	1.6
+++ noatun/modules/winskin/vis/Makefile.am	12 Nov 2003 17:34:43 -0000
@@ -8,20 +8,25 @@
 winskinvis.cc winskinvis.h :	$(srcdir)/winskinvis.idl
 	$(MCOPIDL) -t -I$(kde_includes)/arts $(srcdir)/winskinvis.idl
 
-lib_LTLIBRARIES		=	libwinskinvis.la
-libwinskinvis_la_SOURCES=	winskinvis.cc \
+lib_LTLIBRARIES		=	libwinskinviscommon.la
+kde_module_LTLIBRARIES	=	libwinskinvis.la
+libwinskinviscommon_la_SOURCES=	winskinvis.cc \
 				winSkinFFT_impl.cpp realFFT.cpp \
 				realFFTFilter.cpp visQueue.cpp
 
 
-libwinskinvis_la_LDFLAGS=	$(all_libraries) -module -avoid-version \
-				-no-undefined
-
-libwinskinvis_la_LIBADD =	-lkmedia2_idl -lsoundserver_idl -lartsflow
-libwinskinvis_la_COMPILE_FIRST = winskinvis.cc
-libwinskinvis_la_METASOURCES =	AUTO
+libwinskinviscommon_la_LDFLAGS=	$(all_libraries) -no-undefined
 
+libwinskinviscommon_la_LIBADD =	-lkmedia2_idl -lsoundserver_idl -lartsflow
+libwinskinviscommon_la_COMPILE_FIRST = winskinvis.cc
+libwinskinviscommon_la_METASOURCES =	AUTO
+
+libwinskinvis_la_LDFLAGS	= $(all_libraries) -avoid-version -module
+libwinskinvis_la_LIBADD		= libwinskinviscommon.la
+libwinskinvis_la_SOURCES	= dummy.cpp
 
+dummy.cpp:
+	echo > dummy.cpp
 
 mcoptypedir		=	$(libdir)/mcop
 mcoptype_DATA		=	winskinvis.mcoptype winskinvis.mcopclass
Index: oggvorbis_artsplugin/Makefile.am
===================================================================
RCS file: /home/kde/kdemultimedia/oggvorbis_artsplugin/Makefile.am,v
retrieving revision 1.4
diff -u -u -r1.4 Makefile.am
--- oggvorbis_artsplugin/Makefile.am	28 May 2002 12:59:17 -0000	1.4
+++ oggvorbis_artsplugin/Makefile.am	12 Nov 2003 17:34:43 -0000
@@ -4,7 +4,7 @@
 
 noinst_HEADERS = oggPlayObject_impl.h
 
-lib_LTLIBRARIES = liboggarts.la
+kde_module_LTLIBRARIES = liboggarts.la
 liboggarts_la_COMPILE_FIRST = oggarts.h
 liboggarts_la_SOURCES = oggarts.cc oggPlayObject_impl.cpp
 liboggarts_la_LDFLAGS = $(all_libraries) -module -avoid-version -no-undefined
Index: oggvorbis_artsplugin/oggPlayObject_impl.h
===================================================================
RCS file: /home/kde/kdemultimedia/oggvorbis_artsplugin/oggPlayObject_impl.h,v
retrieving revision 1.4
diff -u -u -r1.4 oggPlayObject_impl.h
--- oggvorbis_artsplugin/oggPlayObject_impl.h	27 Oct 2002 01:10:28 -0000	1.4
+++ oggvorbis_artsplugin/oggPlayObject_impl.h	12 Nov 2003 17:34:43 -0000
@@ -1,7 +1,7 @@
 #ifndef OGGPLAYER_IMPL_H
 #define OGGPLAYER_IMPL_H "$Id$"
 
-#if defined(__GNU_LIBRARY__) && !defined(_SEM_SEMUN_UNDEFINED)
+#if (defined(__GNU_LIBRARY__) && !defined(_SEM_SEMUN_UNDEFINED)) || defined(__APPLE__)
 /* union semun is defined by including <sys/sem.h> */
 #else
 /* according to X/OPEN we have to define it ourselves */
