
===================================================================
diff -u -u -r1.1.1.7 -r1.25
--- admin/acinclude.m4.in	20 Nov 2002 00:45:10 -0000	1.1.1.7
+++ admin/acinclude.m4.in	29 Nov 2002 18:19:37 -0000	1.25
@@ -31,6 +31,9 @@
 dnl thanks to Raja R Harinath.
 dnl ------------------------------------------------------------------------
 dnl
+
+USER_LDFLAGS="-multiply_defined suppress $USER_LDFLAGS"
+
 ifdef([_AC_PATH_X_XMKMF],[],
    [AC_DEFUN([_AC_PATH_X_XMKMF],[AC_PATH_X_XMKMF])])
 ifdef([AC_OUTPUT_SUBDIRS],[],
@@ -366,7 +369,7 @@
   fi
 
 AC_MSG_CHECKING([for KDE libraries installed])
-ac_link='$LIBTOOL_SHELL --silent --mode=link ${CXX-g++} -o conftest $CXXFLAGS $all_includes $CPPFLAGS $LDFLAGS $all_libraries conftest.$ac_ext $LIBS -lkdecore $LIBQT $KDE_TEST_RPATH 1>&5'
+ac_link='$LIBTOOL_SHELL --mode=link ${CXX-g++} -o conftest $CXXFLAGS $all_includes $CPPFLAGS $LDFLAGS $all_libraries conftest.$ac_ext $LIBS -lkdecore $LIBQT $KDE_TEST_RPATH 1>&5'
 
 if AC_TRY_EVAL(ac_link) && test -s conftest; then
   AC_MSG_RESULT(yes)
@@ -704,13 +707,10 @@
      AC_CHECK_LIB(ipc, shmat, X_EXTRA_LIBS="$X_EXTRA_LIBS -lipc"))
    
    # darwin needs this to initialize the environment
-   AC_CHECK_HEADERS(crt_externs.h)
+   AC_CHECK_HEADERS([crt_externs.h sys/types.h])
    AC_CHECK_FUNC(_NSGetEnviron, [AC_DEFINE(HAVE_NSGETENVIRON, 1, [Define if your system needs _NSGetEnviron to set up the environment])])
  
-   # more headers that need to be explicitly included on darwin
-   AC_CHECK_HEADERS(sys/types.h stdint.h)
-
-   # darwin requires a poll emulation library
+   # darwin requires a poll.h emulation library
    AC_CHECK_LIB(poll, poll, LIB_POLL="-lpoll")
 
    # CoreAudio framework
@@ -719,11 +719,36 @@
      FRAMEWORK_COREAUDIO="-framework CoreAudio"
    ])
 
+   # CoreServices framework
+   AC_CHECK_HEADER(CoreServices/CoreServices.h, [
+     AC_DEFINE(HAVE_CORESERVICES, 1, [Define if you have the CoreServices API])
+     FRAMEWORK_CORESERVICES="-framework CoreServices"
+   ])
+
+   # Solaris 2.6 and others need -lresolv for res_init
+   AC_CHECK_FUNCS(res_init, , [
+     kde_libs_safe="$LIBS"
+     LIBS="$LIBS $X_EXTRA_LIBS -lresolv"
+     AC_TRY_LINK(
+[
+#include <resolv.h>
+],
+[ 
+res_init(); 
+],
+        LIBRESOLV="-lresolv"
+        X_EXTRA_LIBS="$X_EXTRA_LIBS $LIBRESOLV"
+        AC_DEFINE(HAVE_RES_INIT, 1, [Define if you have the res_init function])
+     )
+     LIBS=$kde_libs_safe
+   ])
+
    AC_CHECK_RES_INIT
-   AC_SUBST(LIB_POLL)
    AC_SUBST(FRAMEWORK_COREAUDIO)
    LIBSOCKET="$X_EXTRA_LIBS"
    AC_SUBST(LIBSOCKET)
+   AC_SUBST(FRAMEWORK_CORESERVICES)
+   AC_SUBST(LIB_POLL)
    AC_SUBST(LIBRESOLV)
    AC_SUBST(X_EXTRA_LIBS)
    AC_CHECK_LIB(ucb, killpg, [LIBUCB="-lucb"]) dnl for Solaris2.4
@@ -932,7 +957,7 @@
 kde_ldflags_safe="$LDFLAGS"
 kde_libs_safe="$LIBS"
 
-LDFLAGS="$LDFLAGS $X_LDFLAGS $USER_LDFLAGS"
+LDFLAGS="$LDFLAGS $USER_LDFLAGS $X_LDFLAGS"
 LIBS="-lXext -lX11 $LIBSOCKET"
 
 AC_TRY_LINK([
@@ -974,9 +999,9 @@
 kde_save_LDFLAGS="$LDFLAGS"
 kde_save_CFLAGS="$CFLAGS"
 kde_save_LIBS="$LIBS"
-LDFLAGS="$LDFLAGS $X_LDFLAGS $USER_LDFLAGS"
+LDFLAGS="$LDFLAGS $USER_LDFLAGS $X_LDFLAGS"
 CFLAGS="$CFLAGS -I$x_includes"
-LIBS="-lXinerama -lXext"
+LIBS="-lXinerama -lXext -lX11"
 
 if test "x$no_xinerama" = "xno"; then
 
@@ -1844,7 +1869,7 @@
  kde_cxxflags_safe="$CXXFLAGS"
 
  LIBS="-lkde-qt-addon $LIBQT $LIBS"
- CXXFLAGS="$CXXFLAGS -I$prefix/include -I$prefix/include/kde $all_includes"
+ CXXFLAGS="-isystem $prefix/include $CXXFLAGS $all_includes"
  LDFLAGS="$LDFLAGS $all_libraries $USER_LDFLAGS"
 
  AC_TRY_LINK([
@@ -1917,6 +1942,7 @@
    AC_SUBST(LIB_KSPELL, "-lkspell")
    AC_SUBST(LIB_KPARTS, "-lkparts")
    AC_SUBST(LIB_KDEPRINT, "-lkdeprint")
+   AC_SUBST(LIB_KDED, "-lkded_main")
 # these are for backward compatibility
    AC_SUBST(LIB_KSYCOCA, "-lkio")
    AC_SUBST(LIB_KFILE, "-lkio")
@@ -2156,6 +2182,7 @@
 [
 	KDE_CHECK_FUNC_EXT(mkstemp, [
 #include <stdlib.h>
+#include <unistd.h>
 ],
 	[
 mkstemp("/tmp/aaaXXXXXX");
@@ -2168,6 +2195,7 @@
 [
 	KDE_CHECK_FUNC_EXT(mkstemps, [
 #include <stdlib.h>
+#include <unistd.h>
 ],
 	[
 mkstemps("/tmp/aaaXXXXXX", 6);
@@ -2647,10 +2675,10 @@
       if test $kde_use_debug_code = "full"; then
         CFLAGS="-g3 $CFLAGS"
       else
-        CFLAGS="-g -O2 $CFLAGS"
+        CFLAGS="-g -O $CFLAGS"
       fi
     else
-      CFLAGS="-O2 $CFLAGS"
+      CFLAGS="-Os -frename-registers -mmultiple -mcpu=750 -mtune=750 $CFLAGS"
     fi
   fi
 
@@ -2680,7 +2708,7 @@
         if test "$kde_use_debug_code" = "full"; then
           CXXFLAGS="-g3 $CXXFLAGS"
         else
-          CXXFLAGS="-g -O2 $CXXFLAGS"
+          CXXFLAGS="-g -O $CXXFLAGS"
         fi
       fi
       KDE_CHECK_COMPILER_FLAG(fno-builtin,[CXXFLAGS="-fno-builtin $CXXFLAGS"])
@@ -2692,7 +2720,7 @@
       if test "$CXX" = "KCC"; then
         CXXFLAGS="+K3 $CXXFLAGS"
       else
-        CXXFLAGS="-O2 $CXXFLAGS"
+        CXXFLAGS="-Os -frename-registers -mmultiple -mcpu=750 -mtune=750 $CXXFLAGS"
       fi  
     fi
   fi
@@ -2832,7 +2860,7 @@
 AC_LIBTOOL_CXX
 
 LIBTOOL_SHELL="/bin/sh ./libtool"
-#  LIBTOOL="$LIBTOOL --silent"
+#  LIBTOOL="$LIBTOOL"
 KDE_PLUGIN="-avoid-version -module -no-undefined \$(KDE_RPATH) \$(KDE_MT_LDFLAGS)"
 AC_SUBST(KDE_PLUGIN)
 
@@ -2888,7 +2916,7 @@
 [
 AC_CANONICAL_SYSTEM
 AC_ARG_PROGRAM
-AM_INIT_AUTOMAKE($1, $2)
+AM_INIT_AUTOMAKE($1, $2, [foreign])
 AM_DISABLE_LIBRARIES
 AC_PREFIX_DEFAULT(${KDEDIR:-/usr/local/kde})
 AC_CHECK_COMPILERS
@@ -3189,9 +3217,9 @@
     ac_save_ldflags="$LDFLAGS"
     ac_save_cflags="$CFLAGS"
     if test "x$kde_use_qt_emb" != "xyes"; then
-      LDFLAGS="$LDFLAGS $X_LDFLAGS $USER_LDFLAGS $LDFLAGS $XPM_LDFLAGS $all_libraries -lXpm -lX11 -lXext $LIBZ $LIBSOCKET"
+      LDFLAGS="$LDFLAGS $USER_LDFLAGS $X_LDFLAGS $LDFLAGS $XPM_LDFLAGS $all_libraries -lXpm -lX11 -lXext $LIBZ $LIBSOCKET"
     else
-      LDFLAGS="$LDFLAGS $X_LDFLAGS $USER_LDFLAGS $LDFLAGS $XPM_LDFLAGS $all_libraries -lXpm $LIBZ $LIBSOCKET"
+      LDFLAGS="$LDFLAGS $USER_LDFLAGS $X_LDFLAGS $LDFLAGS $XPM_LDFLAGS $all_libraries -lXpm $LIBZ $LIBSOCKET"
     fi
     CFLAGS="$CFLAGS $X_INCLUDES $USER_INCLUDES"
     test -n "$XPM_INCLUDE" && CFLAGS="-I$XPM_INCLUDE $CFLAGS"
@@ -3418,13 +3446,23 @@
           AC_CHECK_LIB(pam, pam_start,
             [ AC_CHECK_HEADER(security/pam_appl.h,
                 [ use_pam=yes
-                  pam_service=kde ]) 
+		  pam_header=security/pam_appl.h
+                  pam_service=kde ],
+		[ AC_CHECK_HEADER(pam/pam_appl.h,
+			[ use_pam=yes
+			  pam_header=pam/pam_appl.h
+			  pam_service=kde ])
+		]) 
             ], , $LIBDL)
           ac_cv_path_pam="use_pam=$use_pam pam_service=$pam_service"
         ])
     ])
   eval "$ac_cv_path_pam"
 
+  if test "$pam_header" = "pam/pam_appl.h"; then
+    AC_DEFINE(PAM_HEADER_IN_PAM, 1, [Define if your pam headers are in pam/ instead of security/])
+  fi
+
   AC_MSG_CHECKING(for PAM)
   if test "x$use_pam" = xno; then
     AC_MSG_RESULT(no)
@@ -3434,17 +3472,18 @@
     AC_DEFINE(HAVE_PAM, 1, [Defines if you have PAM (Pluggable Authentication Modules)])
     PAMLIBS="$PAM_MISC_LIB -lpam $LIBDL"
 
+    AC_CHECK_HEADERS([pam/pam_appl.h])
+
     dnl test whether struct pam_message is const (Linux) or not (Sun)
     AC_MSG_CHECKING(for const pam_message)
-    AC_EGREP_HEADER([struct pam_message], security/pam_appl.h,
-      [ AC_EGREP_HEADER([const struct pam_message], security/pam_appl.h,
+    AC_EGREP_HEADER([struct pam_message], $pam_header,
+      [ AC_EGREP_HEADER([const struct pam_message], $pam_header,
                         [AC_MSG_RESULT([const: Linux-type PAM])],
                         [AC_MSG_RESULT([nonconst: Sun-type PAM])
                         AC_DEFINE(PAM_MESSAGE_NONCONST, 1, [Define if your PAM support takes non-const arguments (Solaris)])]
                         )],
       [AC_MSG_RESULT([not found - assume const, Linux-type PAM])])
   fi
-
   AC_SUBST(PAMLIBS)
 ])
 
@@ -4604,6 +4643,83 @@
      LIBS="$kde_save_LIBS"
 ])
 
+AC_DEFUN(KDE_CHECK_LDAP,
+[
+  AC_MSG_CHECKING([for LDAP])
+
+  if test -n "$1"; then
+    ldap_extra_dirs="$1"
+  fi
+
+  ldap_incdirs="$ldap_extra_dirs/include /usr/include /usr/local/include/ $kde_extra_includes"
+  AC_FIND_FILE(ldap.h, $ldap_incdirs, ldap_incdir)
+  if test ! -r $ldap_incdir/ldap.h; then
+    ldap_incdir=no
+  fi
+
+  ldap_libdirs="$ldap_extra_dirs/lib /usr/lib /usr/local /usr/lib $kde_extra_libs"
+  for ext in la so sl dylib a; do
+    AC_FIND_FILE(libldap.$ext, $ldap_libdirs, ldap_libdir_test)
+    if test -r $ldap_libdir_test/libldap.$ext; then
+      ldap_libdir=$ldap_libdir_test
+      break
+    fi
+  done
+  if test -z "$ldap_libdir"; then
+    ldap_libdir=no
+  fi
+  if test x$ldap_libdir != xno && test x$ldap_incdir != xno; then
+    LDAP_INCS=-I$ldap_incdir
+    LDAP_LIBS=-L$ldap_libdir
+    # on MacOSX, we link against the framework instead of
+    # directly to the libraries, for forward-compatibility
+    if test -d "/System/Library/Frameworks/LDAP.framework"; then
+      LIBLDAP="-framework LDAP"
+    else
+      LIBLDAP="-lldap -llber -lresolv"
+    fi
+    if test "$USE_RPATH" = "yes"; then
+      LDAP_RPATH="-R $LDAPLIBDIR/lib"
+    fi
+
+    kde_safe_LIBS="$LIBS"
+    kde_safe_CFLAGS="$CFLAGS"
+    LIBS="$LIBS $all_libraries $LDAP_LIBS $LIBLDAP $KERBEROS_LIBS $X_EXTRA_LIBS"
+    CFLAGS="$CFLAGS $all_includes $LDAP_INCS $KERBEROS_INCS"
+    AC_LANG_SAVE
+    AC_LANG_C
+    AC_TRY_LINK(
+      [#include <lber.h>
+#include <ldap.h>
+#if LDAP_VENDOR_VERSION < 20000
+#error OpenLDAP version too old, please upgrade to version 2 or higher
+#endif
+],
+      [ LDAP *ldap; ],
+      [ ldap_old=no ],
+      [ ldap_old=yes ]
+    )
+    LIBS="$kde_safe_LIBS"
+    CFLAGS="$kde_safe_CFLAGS"
+
+    if test x$ldap_old = xno; then
+      AC_SUBST(LDAP_INCS)
+      AC_SUBST(LDAP_LIBS)
+      AC_SUBST(LDAP_RPATH)
+      AC_SUBST(LIBLDAP)
+      AC_DEFINE_UNQUOTED(HAVE_LIBLDAP,1,[Define if you have LDAP libraries])
+    else
+      LDAP_INCS=
+      LDAP_LIBS=
+      LDAP_RPATH=
+      LIBLDAP=
+    fi
+    AC_MSG_RESULT([libraries $ldap_libdir, headers $ldap_incdir])
+  else
+    AC_MSG_RESULT([not found])
+  fi
+])
+
 AC_DEFUN(KDE_JAVA_PREFIX,
 [
 	dir=`dirname "$1"`
@@ -4660,6 +4776,7 @@
       KDE_JAVA_PREFIX(/usr/lib/IBMJava2*)
       KDE_JAVA_PREFIX(/usr/lib/IBMJava*)
       KDE_JAVA_PREFIX(/opt/java*)
+      KDE_JAVA_PREFIX(/Library/Java/Home)
     
       kde_cv_path="NONE"
       kde_save_IFS=$IFS
@@ -4711,10 +4828,20 @@
 dnl At this point kde_java_bindir and kde_java_includedir are either set or "no"
 if test "x$kde_java_bindir" != "xno"; then
 
+  case $host_os in
+    darwin*)
+      dnl Look for libjvm.dylib
+      kde_java_libjvmdir=`find $kde_java_bindir/.. -name libjvm.dylib | sed 's,libjvm.dylib,,'|head -n 1`
+      kd_java_libhpidir=''
+      dnl Darwin doesn't have libhpi
+      ;;
+    *)
   dnl Look for libjvm.so
   kde_java_libjvmdir=`find $kde_java_bindir/.. -name libjvm.so | sed 's,libjvm.so,,'|head -n 1`
   dnl Look for libhpi.so and avoid green threads
-  kde_java_libhpidir=`find $kde_java_bindir/.. -name libhpi.so | grep -v green | sed 's,libhpi.so,,' | head -n 1`
+  kde_java_libhpidir=`find $kde_java_bindir/.. -name libhpi.so | grep -v green | sed 's,libhpi.so,,' | tail -n 1`
+      ;;
+  esac
 
   dnl Now check everything's fine under there
   dnl the include dir is our flag for having the JDK
@@ -4743,19 +4870,31 @@
     JAVAC=
     jni_includes=
   fi
-
+  case $host_os in
+    darwin*)
+      if test ! -r "$kde_java_libjvmdir/libjvm.dylib"; then
+        AC_MSG_ERROR([libjvm.dylib not found under $kde_java_libjvmdir. Use --without-java.])
+      fi
+      dnl Darwin doesn't have libhpi, but it's not fatal
+      LIB_JVM="-framework JavaVM"
+      LIB_HPI=""
+      ;;
+    *)
   if test ! -r "$kde_java_libjvmdir/libjvm.so"; then
      AC_MSG_ERROR([libjvm.so not found under $kde_java_libjvmdir. Use --without-java.])
   fi 
+      LIB_JVM="-L$kde_java_libjvmdir -ljvm"
+      if test ! -r "$kde_java_libhpidir/libhpi.so"; then
+        AC_MSG_ERROR([libhpi.so not found under $kde_java_libhpidir. Use --without-java.])
+      fi
+      LIB_HPI="-L$kde_java_libhpidir -lhpi"
+      ;;
+  esac
 
   if test ! -x "$kde_java_bindir/java"; then
       AC_MSG_ERROR([java not found under $kde_java_bindir. javac was found though! Use --with-java or --without-java.])
   fi
 
-  if test ! -r "$kde_java_libhpidir/libhpi.so"; then
-    AC_MSG_ERROR([libhpi.so not found under $kde_java_libhpidir. Use --without-java.])
-  fi
-
   if test -n "$jni_includes"; then
     dnl Check for JNI version
     AC_LANG_SAVE
@@ -5119,8 +5258,7 @@
   LDFLAGS="$LDFLAGS -L$ssl_libdir $all_libraries"
   LIBS="$LIBS $LIBSSL -lRSAglue -lrsaref"
 
-  AC_TRY_LINK(,void RSAPrivateEncrypt(void);RSAPrivateEncrypt();,
-  ac_ssl_rsaref="yes"
+  AC_TRY_LINK(,void RSAPrivateEncrypt(void);RSAPrivateEncrypt();, ac_ssl_rsaref="yes"
   ,
   ac_ssl_rsaref="no"
   )
===================================================================
diff -u -u -r1.1.1.6 -r1.2
--- admin/am_edit	29 Oct 2002 02:27:18 -0000	1.1.1.6
+++ admin/am_edit	27 Nov 2002 19:43:04 -0000	1.2
@@ -1034,7 +1034,7 @@
             if ($allidls !~ /$source\_kidl/) {
                 
                 $dep_lines .= "$source.kidl: $sourcedir$source.h \$(DCOP_DEPENDENCIES)\n";
-                $dep_lines .= "\t\$(DCOPIDL) $sourcedir$source.h > $source.kidl || ( rm -f $source.kidl ; /bin/false )\n";
+                $dep_lines .= "\t\$(DCOPIDL) $sourcedir$source.h > $source.kidl || ( rm -f $source.kidl ; false )\n";
                 
                 $allidls .= $source . "_kidl ";
             }
===================================================================
diff -u -u -r1.1.1.8 -r1.3
--- admin/cvs.sh	26 Nov 2002 21:12:30 -0000	1.1.1.8
+++ admin/cvs.sh	26 Nov 2002 21:14:13 -0000	1.3
@@ -119,7 +119,7 @@
 fi
 
 echo "*** Creating Makefile templates"
-$AUTOMAKE || exit 1
+$AUTOMAKE --foreign || exit 1
 if test -z "$UNSERMAKE"; then
   echo "*** Postprocessing Makefile templates"
   perl -w admin/am_edit || exit 1
===================================================================
diff -u -u -r1.1.1.5 -r1.5
--- admin/detect-autoconf.sh	26 Nov 2002 21:12:30 -0000	1.1.1.5
+++ admin/detect-autoconf.sh	26 Nov 2002 21:14:14 -0000	1.5
@@ -52,12 +52,15 @@
 checkAutomakeAclocal ()
 {
   if test -z "$UNSERMAKE"; then
-    if test -x "`$WHICH automake-1.5`" ; then
-      AUTOMAKE="`$WHICH automake-1.5`"
-      ACLOCAL="`$WHICH aclocal-1.5`"
+    if test -x "`$WHICH automake-1.7`" ; then
+      AUTOMAKE="`$WHICH automake-1.7`"
+      ACLOCAL="`$WHICH aclocal-1.7`"
     elif test -x "`$WHICH automake-1.6`" ; then
       AUTOMAKE="`$WHICH automake-1.6`"
       ACLOCAL="`$WHICH aclocal-1.6`"
+    elif test -x "`$WHICH automake-1.5`" ; then
+      AUTOMAKE="`$WHICH automake-1.5`"
+      ACLOCAL="`$WHICH aclocal-1.5`"
     fi
   else
      AUTOMAKE="$UNSERMAKE"
===================================================================
diff -u -u -r1.1.1.5 -r1.11
--- admin/libtool.m4.in	20 Nov 2002 00:45:13 -0000	1.1.1.5
+++ admin/libtool.m4.in	24 Nov 2002 07:11:18 -0000	1.11
@@ -65,7 +65,7 @@
 LIBTOOL_DEPS="$ac_aux_dir/ltmain.sh"
 
 # Always use our own libtool.
-LIBTOOL='$(SHELL) $(top_builddir)/libtool --silent'
+LIBTOOL='$(SHELL) $(top_builddir)/libtool'
 AC_SUBST(LIBTOOL)dnl
 
 # Prevent multiple expansion
@@ -609,6 +609,12 @@
     lt_cv_sys_max_cmd_len=-1;
     ;;
 
+  darwin*)
+    # this should be enough... the length test freaks out some versions
+    # of OSX's `expr`
+    lt_cv_sys_max_cmd_len=16384;
+    ;;
+
   *)
     # If test is not a shell built-in, we'll probably end up computing a
     # maximum length that is only half of the actual maximum length, but
@@ -1908,7 +1914,8 @@
     lt_cv_file_magic_test_file=`/System/Library/Frameworks/System.framework/System`
     ;;
   *) # Darwin 1.3 on
-    lt_cv_file_magic_test_file='/usr/lib/libSystem.dylib'
+    lt_cv_deplibs_check_method='pass_all'
+    lt_cv_file_magic_cmd=''
     ;;
   esac
   ;;
@@ -2560,6 +2567,11 @@
 	;;
     esac
     ;;
+  darwin*)
+    shared_flag='-dynamiclib'
+    _LT_AC_TAGVAR(archive_cmds, $1)='$CC -r -keep_private_externs -nostdlib -o ${lib}-master.o $libobjs && $CC $(test .$module = .yes && echo -bundle || echo -dynamiclib) $predep_objects ${lib}-master.o $deplibs $postdep_objects $compiler_flags -o $lib $(test x$module != xyes && echo -install_name $rpath/$soname $verstring)'
+    output_verbose_link_cmd='$CC $(test .$module = .yes && echo -bundle || echo -dynamiclib) $CFLAGS -v conftest.$objext 2>&1 | egrep "\-L"'
+    ;;
   dgux*)
     case $cc_basename in
       ec++)
@@ -4750,7 +4762,9 @@
 	_LT_AC_TAGVAR(allow_undefined_flag, $1)='-undefined suppress'
 	;;
       *) # Darwin 1.3 on
-	_LT_AC_TAGVAR(allow_undefined_flag, $1)='-flat_namespace -undefined suppress'
+	#_LT_AC_TAGVAR(allow_undefined_flag, $1)='-flat_namespace -undefined suppress'
+        # testing with twolevel namespaces
+	_LT_AC_TAGVAR(allow_undefined_flag, $1)='-multiply_defined suppress'
 	;;
       esac
 
@@ -4758,7 +4772,7 @@
       #        cross-compilation, but unfortunately the echo tests do not
       #        yet detect zsh echo's removal of \ escapes.  Also zsh mangles
       #	       `"' quotes if we put them in here... so don't!
-      _LT_AC_TAGVAR(archive_cmds, $1)='$CC $(test .$module = .yes && echo -bundle || echo -dynamiclib) $allow_undefined_flag -o $lib $libobjs $deplibs$linker_flags -install_name $rpath/$soname $verstring'
+      _LT_AC_TAGVAR(archive_cmds, $1)='$CC -r -keep_private_externs -nostdlib -o ${lib}-master.o $libobjs && $CC $(test .$module = .yes && echo -bundle || echo -dynamiclib) $allow_undefined_flag -o $lib ${lib}-master.o $deplibs $linker_flags $(test x$module != xyes && echo -install_name $rpath/$soname $verstring)'
       # We need to add '_' to the symbols in $export_symbols first
       #_LT_AC_TAGVAR(archive_expsym_cmds, $1)="$_LT_AC_TAGVAR(archive_cmds, $1)"' && strip -s $export_symbols'
       _LT_AC_TAGVAR(hardcode_direct, $1)=yes
===================================================================
diff -u -u -r1.1.1.4 -r1.6
--- admin/ltmain.sh	20 Nov 2002 00:45:14 -0000	1.1.1.4
+++ admin/ltmain.sh	24 Nov 2002 07:11:19 -0000	1.6
@@ -2532,11 +2532,11 @@
 	eval libname=\"$libname_spec\"
 	;;
       *)
-	if test "$module" = no; then
-	  $echo "$modename: libtool library \`$output' must begin with \`lib'" 1>&2
-	  $echo "$help" 1>&2
-	  exit 1
-	fi
+#	if test "$module" = no; then
+#	  $echo "$modename: libtool library \`$output' must begin with \`lib'" 1>&2
+#	  $echo "$help" 1>&2
+#	  exit 1
+#	fi
 	if test "$need_lib_prefix" != no; then
 	  # Add the "lib" prefix for modules if required
 	  name=`$echo "X$outputname" | $Xsed -e 's/\.la$//'`
@@ -3321,7 +3321,23 @@
 	if test -n "$export_symbols" && test -n "$archive_expsym_cmds"; then
 	  eval cmds=\"$archive_expsym_cmds\"
 	else
+          if test "x$verstring" = "x0.0"; then
+                tmp_verstring=
+          else
+                tmp_verstring="$verstring"
+          fi
+ 	  save_deplibs="$deplibs"
+ 	  for conv in $convenience; do
+            tmp_deplibs=
+            for test_deplib in $deplibs; do
+              if test "$test_deplib" != "$conv"; then
+                tmp_deplibs="$tmp_deplibs $test_deplib"
+              fi
+            done
+            deplibs="$tmp_deplibs"
+ 	  done
 	  eval cmds=\"$archive_cmds\"
+ 	  deplibs="$save_deplibs"
 	fi
 
 	if len=`expr "X$cmds" : ".*"` &&
@@ -3419,7 +3435,12 @@
 	  if test -n "$export_symbols" && test -n "$archive_expsym_cmds"; then
 	    eval cmds=\"$archive_expsym_cmds\"
 	  else
-	    eval cmds=\"$archive_cmds\"
+            save_deplibs="$deplibs"
+            for conv in $convenience; do
+              deplibs="${deplibs%$conv*} ${deplibs#*$conv}"
+            done
+            eval cmds=\"$archive_cmds\"
+            deplibs="$save_deplibs"
 	  fi
 
 	  # Append the command to remove the reloadable object files
@@ -3634,6 +3655,14 @@
 	;;
       esac
 
+      case $host in
+      *darwin*)
+        # Don't allow lazy linking, it breaks C++ global constructors
+        compile_command="$compile_command ${wl}-bind_at_load -undefined suppress -flat_namespace"
+        finalize_command="$finalize_command ${wl}-bind_at_load"
+        ;;
+      esac
+
       compile_command="$compile_command $compile_deplibs"
       finalize_command="$finalize_command $finalize_deplibs"
 
@@ -4024,9 +4053,10 @@
 	    # fast_install is set to needless
 	    relink_command=
 	  fi
+          relink_command=
 	else
 	  link_command="$compile_var$compile_command$compile_rpath"
-	  relink_command="$finalize_var$finalize_command$finalize_rpath"
+	  #relink_command="$finalize_var$finalize_command$finalize_rpath"
 	fi
       fi
 
===================================================================
diff -u -u -r1.1.1.4 -r1.5
===================================================================
diff -u -u -r1.1.1.3 -r1.5
===================================================================
diff -u -u -r1.1.1.10 -r1.6
===================================================================
diff -u -u -r1.1.1.6 -r1.2
===================================================================
diff -u -u -r1.1.1.2 -r1.3
--- ./kmailcvt/filter_plain.cxx	14 Aug 2002 19:13:56 -0000	1.1.1.2
+++ ./kmailcvt/filter_plain.cxx	14 Aug 2002 19:36:48 -0000	1.3
@@ -24,7 +24,6 @@
 #include <kfiledialog.h>
 #include <kdirlister.h>
 #include <qregexp.h>
-#include <libgen.h>
 #include <qdir.h>
 
 #include "filter_plain.hxx"
===================================================================
diff -u -u -r1.1.1.5 -r1.5
--- ./kpf/src/WebServer.cpp	27 Nov 2002 02:21:16 -0000	1.1.1.5
+++ ./kpf/src/WebServer.cpp	27 Nov 2002 02:26:50 -0000	1.5
@@ -234,7 +234,7 @@
 #ifdef __osf__
         (size_t) sizeof(on)
 #else
-        reinterpret_cast<socklen_t>(sizeof(on))
+	(socklen_t) sizeof(on)
 #endif
       );
 
@@ -249,7 +249,7 @@
 #ifdef __osf__
         (size_t) sizeof(on)
 #else
-        reinterpret_cast<socklen_t>(sizeof(on))
+        (socklen_t) sizeof(on)
 #endif
       );
 
===================================================================
diff -u -u -r1.1.1.3 -r1.4
--- ./krdc/rfbproto.h	20 Nov 2002 02:57:22 -0000	1.1.1.3
+++ ./krdc/rfbproto.h	20 Nov 2002 03:11:12 -0000	1.4
@@ -43,6 +43,9 @@
  *      messages have to be explained by comments.
  */
 
+#ifndef RFBPROTO_H
+#define RFBPROTO_H
+
 #include "vnctypes.h"
 
 /*****************************************************************************
@@ -833,3 +836,5 @@
     rfbPointerEventMsg pe;
     rfbClientCutTextMsg cct;
 } rfbClientToServerMsg;
+
+#endif
===================================================================
diff -u -u -r1.1.1.2 -r1.4
--- ./krdc/vnctypes.h	14 Aug 2002 19:12:45 -0000	1.1.1.2
+++ ./krdc/vnctypes.h	17 Aug 2002 17:34:19 -0000	1.4
@@ -30,7 +30,6 @@
 
 #include <X11/Xmd.h>
 
-
 typedef struct {
   int shareDesktop; /* bool */
   int viewOnly; /* bool */
===================================================================
diff -u -u -r1.1.1.2 -r1.4
--- ./krdc/vncviewer.h	14 Aug 2002 19:12:46 -0000	1.1.1.2
+++ ./krdc/vncviewer.h	17 Aug 2002 17:34:19 -0000	1.4
@@ -177,4 +177,5 @@
 #if(defined __cplusplus)
 }
 #endif
-#endif
+
+#endif /* VNCVIEWER_H */
===================================================================
diff -u -u -r1.1.1.4 Makefile.am
--- ./krfb/kinetd/Makefile.am	29 Oct 2002 17:41:27 -0000	1.1.1.4
+++ ./krfb/kinetd/Makefile.am	29 Nov 2002 18:36:42 -0000
@@ -5,15 +5,14 @@
 
 kded_kinetd_la_SOURCES = kinetd.cpp kinetd.skel
 kded_kinetd_la_LDFLAGS  = $(all_libraries) -module -avoid-version
-kded_kinetd_la_LIBADD   = ../srvloc/libsrvloc.la $(LIB_KIO) 
+kded_kinetd_la_LIBADD   = $(top_builddir)/krfb/srvloc/libsrvloc.la $(LIB_KIO) $(LIB_KDED) -lkdeui -lDCOP
 
 # Services
 kde_servicetypes_DATA = kinetdmodule.desktop
 kdeddir = $(kde_servicesdir)/kded
 kded_DATA = kinetd.desktop
 
-EXTRA_DIST = $(kded_kinetd_la_SOURCES)\
-             $(kded_DATA) \
+EXTRA_DIST = $(kded_DATA) \
              $(kde_servicetypes_DATA) \
 	     eventsrc README.debugging
 
===================================================================
diff -u -u -r1.1.1.2 -r1.3
--- ./krfb/libvncserver/rfb.h	29 Oct 2002 17:41:39 -0000	1.1.1.2
+++ ./krfb/libvncserver/rfb.h	29 Oct 2002 17:58:56 -0000	1.3
@@ -31,6 +31,8 @@
 {
 #endif
 
+typedef unsigned int socklen_t;
+
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
===================================================================
diff -N ./krfb/srvloc/inet_ntop.c
--- /dev/null	1 Jan 1970 00:00:00 -0000
+++ ./krfb/srvloc/inet_ntop.c	23 Aug 2002 21:19:36 -0000	1.1
@@ -0,0 +1,195 @@
+/*
+ * Copyright (c) 1996-1999 by Internet Software Consortium.
+ *
+ * Permission to use, copy, modify, and distribute this software for any
+ * purpose with or without fee is hereby granted, provided that the above
+ * copyright notice and this permission notice appear in all copies.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS" AND INTERNET SOFTWARE CONSORTIUM DISCLAIMS
+ * ALL WARRANTIES WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES
+ * OF MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL INTERNET SOFTWARE
+ * CONSORTIUM BE LIABLE FOR ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL
+ * DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR
+ * PROFITS, WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS
+ * ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS
+ * SOFTWARE.
+ */
+
+#if defined(LIBC_SCCS) && !defined(lint)
+static const char rcsid[] = "$Id$";
+#endif /* LIBC_SCCS and not lint */
+
+/*#include "port_before.h"*/
+
+#include <sys/param.h>
+#include <sys/types.h>
+#include <sys/socket.h>
+
+#include <netinet/in.h>
+#include <arpa/inet.h>
+#include <arpa/nameser.h>
+
+#include <errno.h>
+#include <stdio.h>
+#include <string.h>
+
+/*#include "port_after.h"*/
+
+#define NS_INT16SZ      2
+#define NS_IN6ADDRSZ   16
+ 
+#ifdef SPRINTF_CHAR
+# define SPRINTF(x) strlen(sprintf/**/x)
+#else
+# define SPRINTF(x) ((size_t)sprintf x)
+#endif
+
+/*
+ * WARNING: Don't even consider trying to compile this on a system where
+ * sizeof(int) < 4.  sizeof(int) > 4 is fine; all the world's not a VAX.
+ */
+
+static const char *inet_ntop4 __P((const u_char *src, char *dst, size_t size));
+static const char *inet_ntop6 __P((const u_char *src, char *dst, size_t size));
+
+/* char *
+ * inet_ntop(af, src, dst, size)
+ *	convert a network format address to presentation format.
+ * return:
+ *	pointer to presentation format address (`dst'), or NULL (see errno).
+ * author:
+ *	Paul Vixie, 1996.
+ */
+const char *
+inet_ntop(int af, const void *src, char *dst, size_t size)
+{
+	switch (af) {
+	case AF_INET:
+		return (inet_ntop4(src, dst, size));
+	case AF_INET6:
+		return (inet_ntop6(src, dst, size));
+	default:
+		errno = EAFNOSUPPORT;
+		return (NULL);
+	}
+	/* NOTREACHED */
+}
+
+/* const char *
+ * inet_ntop4(src, dst, size)
+ *	format an IPv4 address
+ * return:
+ *	`dst' (as a const)
+ * notes:
+ *	(1) uses no statics
+ *	(2) takes a u_char* not an in_addr as input
+ * author:
+ *	Paul Vixie, 1996.
+ */
+static const char *
+inet_ntop4(const u_char *src, char *dst, size_t size)
+{
+	static const char fmt[] = "%u.%u.%u.%u";
+	char tmp[sizeof "255.255.255.255"];
+
+	if (SPRINTF((tmp, fmt, src[0], src[1], src[2], src[3])) >= size) {
+		errno = ENOSPC;
+		return (NULL);
+	}
+	strcpy(dst, tmp);
+	return (dst);
+}
+
+/* const char *
+ * inet_ntop6(src, dst, size)
+ *	convert IPv6 binary address into presentation (printable) format
+ * author:
+ *	Paul Vixie, 1996.
+ */
+static const char *
+inet_ntop6(const u_char *src, char *dst, size_t size)
+{
+	/*
+	 * Note that int32_t and int16_t need only be "at least" large enough
+	 * to contain a value of the specified size.  On some systems, like
+	 * Crays, there is no such thing as an integer variable with 16 bits.
+	 * Keep this in mind if you think this function should have been coded
+	 * to use pointer overlays.  All the world's not a VAX.
+	 */
+	char tmp[sizeof "ffff:ffff:ffff:ffff:ffff:ffff:255.255.255.255"], *tp;
+	struct { int base, len; } best, cur;
+	u_int words[NS_IN6ADDRSZ / NS_INT16SZ];
+	int i;
+
+	/*
+	 * Preprocess:
+	 *	Copy the input (bytewise) array into a wordwise array.
+	 *	Find the longest run of 0x00's in src[] for :: shorthanding.
+	 */
+	memset(words, '\0', sizeof words);
+	for (i = 0; i < NS_IN6ADDRSZ; i++)
+		words[i / 2] |= (src[i] << ((1 - (i % 2)) << 3));
+	best.base = -1;
+	cur.base = -1;
+	for (i = 0; i < (NS_IN6ADDRSZ / NS_INT16SZ); i++) {
+		if (words[i] == 0) {
+			if (cur.base == -1)
+				cur.base = i, cur.len = 1;
+			else
+				cur.len++;
+		} else {
+			if (cur.base != -1) {
+				if (best.base == -1 || cur.len > best.len)
+					best = cur;
+				cur.base = -1;
+			}
+		}
+	}
+	if (cur.base != -1) {
+		if (best.base == -1 || cur.len > best.len)
+			best = cur;
+	}
+	if (best.base != -1 && best.len < 2)
+		best.base = -1;
+
+	/*
+	 * Format the result.
+	 */
+	tp = tmp;
+	for (i = 0; i < (NS_IN6ADDRSZ / NS_INT16SZ); i++) {
+		/* Are we inside the best run of 0x00's? */
+		if (best.base != -1 && i >= best.base &&
+		    i < (best.base + best.len)) {
+			if (i == best.base)
+				*tp++ = ':';
+			continue;
+		}
+		/* Are we following an initial run of 0x00s or any real hex? */
+		if (i != 0)
+			*tp++ = ':';
+		/* Is this address an encapsulated IPv4? */
+		if (i == 6 && best.base == 0 &&
+		    (best.len == 6 || (best.len == 5 && words[5] == 0xffff))) {
+			if (!inet_ntop4(src+12, tp, sizeof tmp - (tp - tmp)))
+				return (NULL);
+			tp += strlen(tp);
+			break;
+		}
+		tp += SPRINTF((tp, "%x", words[i]));
+	}
+	/* Was it a trailing run of 0x00's? */
+	if (best.base != -1 && (best.base + best.len) == 
+	    (NS_IN6ADDRSZ / NS_INT16SZ))
+		*tp++ = ':';
+	*tp++ = '\0';
+
+	/*
+	 * Check for overflow, copy, and we're done.
+	 */
+	if ((size_t)(tp - tmp) > size) {
+		errno = ENOSPC;
+		return (NULL);
+	}
+	strcpy(dst, tmp);
+	return (dst);
+}
===================================================================
diff -u -u -r1.1.1.3 -r1.6
--- ./krfb/srvloc/kinetaddr.cpp	29 Oct 2002 17:41:41 -0000	1.1.1.3
+++ ./krfb/srvloc/kinetaddr.cpp	31 Oct 2002 19:24:26 -0000	1.6
@@ -28,6 +28,8 @@
 
 #include <limits.h>
 #include <string.h>
+#include <sys/types.h>
+#include <sys/socket.h>
 
 #include <kdebug.h>
 #include <klocale.h>
===================================================================
diff -u -u -r1.1.1.6 Makefile.am
--- ./ksirc/Makefile.am	5 Nov 2002 02:44:24 -0000	1.1.1.6
+++ ./ksirc/Makefile.am	29 Nov 2002 18:36:42 -0000
@@ -17,17 +17,16 @@
 testview_LDADD = kstextview.lo $(LIB_KDEUI)
 testview_LDFLAGS = $(all_libraries) $(KDE_RPATH)
 
-lib_LTLIBRARIES = ksirc.la
+lib_LTLIBRARIES = ksirc.la libksirc_main.la
 
 noinst_PROGRAMS = getdate
 
 # Which sources should be compiled for ksirc.
 
-ksirc_la_SOURCES = \
+libksirc_main_la_SOURCES = \
 	toplevel.cpp \
 	charSelector.cpp \
 	chanButtons.cpp \
-	ksirc.cpp \
 	alistbox.cpp \
 	ahistlineedit.cpp \
 	iocontroller.cpp \
@@ -57,8 +56,7 @@
         colorpicker.cpp
 
 # the library search path
-ksirc_la_LDFLAGS = $(all_libraries) -module -avoid-version
-ksirc_la_COMPILE_FIRST = cdate.h
+libksirc_main_la_LDFLAGS = $(all_libraries) -avoid-version
 
 all_ksirc_libraries = \
 	./KSPrefs/libksprefs.la \
@@ -68,10 +66,17 @@
 
 # the libraries to link against. Be aware of the order. First the libraries,
 # that depend on the following ones.
-ksirc_la_LIBADD   = $(all_ksirc_libraries)
+libksirc_main_la_LIBADD   = $(all_ksirc_libraries) -lkdecore -lkdeui $(LIB_QT)
 
-ksirc_SOURCES = dummy.cpp
-ksirc_LDADD = ksirc.la
+ksirc_la_SOURCES = ksirc_la_main.cpp
+ksirc_la_LIBADD  = libksirc_main.la
+ksirc_la_LDFLAGS = $(all_libraries) $(KDE_RPATH) -module -avoid-version
+
+ksirc_la_main.cpp: ksirc.cpp cdate.h
+	cat ksirc.cpp > ksirc_la_main.cpp
+
+ksirc_SOURCES = ksirc.cpp
+ksirc_LDADD = libksirc_main.la
 ksirc_LDFLAGS = $(all_libraries) $(KDE_RPATH)
 
 kdoc:
===================================================================
diff -u -u -r1.1.1.2 -r1.2
===================================================================
diff -u -u -r1.1.1.2 -r1.2
===================================================================
diff -u -u -r1.1.1.1 -r1.2
--- ./ktalkd/ktalkd/machines/answmach.cpp	25 Jun 2002 03:12:42 -0000	1.1.1.1
+++ ./ktalkd/ktalkd/machines/answmach.cpp	25 Jun 2002 03:14:13 -0000	1.2
@@ -32,6 +32,10 @@
  *
  */
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include "answmach.h"
 
 #include <stdio.h>
===================================================================
diff -u -u -r1.1.1.1 -r1.2
--- ./ktalkd/ktalkd/machines/forwmach.cpp	25 Jun 2002 03:12:42 -0000	1.1.1.1
+++ ./ktalkd/ktalkd/machines/forwmach.cpp	25 Jun 2002 03:14:13 -0000	1.2
@@ -32,6 +32,9 @@
  *
  */
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
 #include "../includ.h"
 #include "forwmach.h"
 #include <stdio.h>
===================================================================
diff -u -u -r1.1.1.1 kxmlrpcdaemon.cpp
--- ./kxmlrpc/kxmlrpcdaemon.cpp	25 Jun 2002 03:12:42 -0000	1.1.1.1
+++ ./kxmlrpc/kxmlrpcdaemon.cpp	29 Nov 2002 18:36:42 -0000
@@ -358,7 +358,8 @@
    { 0, 0, 0 }
 };
 
-int main(int argc, char **argv)
+extern "C" int kdemain(int, char**);
+int kdemain(int argc, char **argv)
 {
     KAboutData about("kxmlrpcd", I18N_NOOP("KXmlRpc Daemon"),
                      version, description, KAboutData::License_BSD);
===================================================================
diff -u -u -r1.1.1.2 Makefile.am
--- ./kxmlrpc/Makefile.am	3 Oct 2002 00:44:41 -0000	1.1.1.2
+++ ./kxmlrpc/Makefile.am	29 Nov 2002 18:36:42 -0000
@@ -1,21 +1,31 @@
 # Makefile for kxmlrpcd.
 # Written by Kurt Granroth using tons of KDE Makefiles as examples
 
+METASOURCES      = AUTO
 INCLUDES         = $(all_includes)
 
 bin_PROGRAMS     = kxmlrpcd
-kxmlrpcd_LDADD   = kxmlrpcd.la
+kxmlrpcd_LDADD   = libkxmlrpcd_main.la
 kxmlrpcd_LDFLAGS = $(all_libraries) $(KDE_RPATH)
-kxmlrpcd_SOURCES = dummy.cpp
+kxmlrpcd_SOURCES = kxmlrpcd_main.cpp
 
-lib_LTLIBRARIES        = kxmlrpcd.la
+kxmlrpcd_main.cpp: stub_main.cpp
+	cat stub_main.cpp > kxmlrpcd_main.cpp
+
+lib_LTLIBRARIES        = kxmlrpcd.la libkxmlrpcd_main.la
 kde_module_LTLIBRARIES = kcm_xmlrpcd.la
 
-kxmlrpcd_la_LIBADD  = $(LIB_KSYCOCA)
-kxmlrpcd_la_LDFLAGS = $(all_libraries) -module -avoid-version 
-kxmlrpcd_la_SOURCES = kxmlrpcserver.cpp kxmlrpcparser.cpp \
+libkxmlrpcd_main_la_LIBADD  = $(LIB_KSYCOCA) $(LIB_QT) -lkdecore -lDCOP
+libkxmlrpcd_main_la_LDFLAGS = $(all_libraries) -avoid-version 
+libkxmlrpcd_main_la_SOURCES = kxmlrpcserver.cpp kxmlrpcparser.cpp \
 	kxmlrpcdaemon.cpp kxmlrpcutil.cpp
-kxmlrpcd_la_METASOURCES = AUTO
+
+kxmlrpcd_la_LIBADD  = libkxmlrpcd_main.la
+kxmlrpcd_la_LDFLAGS = $(all_libraries) -avoid-version -module
+kxmlrpcd_la_SOURCES = kxmlrpcd_la_main.cpp
+
+kxmlrpcd_la_main.cpp: stub_main.cpp
+	cat stub_main.cpp > kxmlrpcd_la_main.cpp
 
 kcm_xmlrpcd_la_LIBADD = $(LIB_KDECORE)
 kcm_xmlrpcd_la_LDFLAGS = $(all_libraries) $(KDE_PLUGIN) -module
@@ -28,6 +38,3 @@
 
 messages: rc.cpp
 	$(XGETTEXT) *.cpp -o $(podir)/kxmlrpcd.pot
-
-dummy.cpp:
-	echo > dummy.cpp
===================================================================
diff -N ./kxmlrpc/stub_main.cpp
--- /dev/null	1 Jan 1970 00:00:00 -0000
+++ ./kxmlrpc/stub_main.cpp	29 Nov 2002 18:36:42 -0000
@@ -0,0 +1,28 @@
+/*
+ *
+ * Copyright (c) 2001 Nick Hudson <skrll@netbsd.org>
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+ * AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN
+ * AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+extern "C" int kdemain(int, char* []);
+
+int main( int argc, char* argv[] )
+{
+        return kdemain(argc, argv);
+}
+
===================================================================
diff -u -u -r1.1.1.4 -r1.5
--- ./libkdenetwork/Makefile.am	29 Oct 2002 17:41:58 -0000	1.1.1.4
+++ ./libkdenetwork/Makefile.am	29 Oct 2002 17:58:57 -0000	1.5
@@ -43,9 +43,8 @@
 
 libkdenetwork_la_LDFLAGS = $(all_libraries) -no-undefined -version-info 2:0
 noinst_HEADERS = kpgpbase.h
+libkdenetwork_la_LIBADD = $(LIB_KDEUI) $(LIB_POLL)
 include_HEADERS = cryptplugwrapper.h cryptplug.h cryptplugwrapperlist.h
-
-libkdenetwork_la_LIBADD = $(LIB_KDEUI)
 
 METASOURCES = AUTO
 
===================================================================
diff -u -u -r1.1.1.1 -r1.2
--- ./mimelib/mimelib/config.h	25 Jun 2002 03:12:43 -0000	1.1.1.1
+++ ./mimelib/mimelib/config.h	25 Jun 2002 03:14:13 -0000	1.2
@@ -35,7 +35,7 @@
 #   define DW_WIN32
 #endif
 
-#if defined(__unix__) || defined(__unix) || defined(_AIX) || defined(__NetBSD__)
+#if defined(__unix__) || defined(__unix) || defined(_AIX) || defined(__NetBSD__) || defined(__MACH__)
 #   define DW_UNIX
 #endif
 
