
--- /dev/null	1 Jan 1970 00:00:00 -0000
+++ Makefile.cvs	25 Jun 2002 03:20:49 -0000	1.1.1.1
@@ -0,0 +1,15 @@
+
+all:
+	@echo "This Makefile is only for the CVS repository"
+	@echo "This will be deleted before making the distribution"
+	@echo ""
+	@if test ! -d admin; then \
+	   echo "Please recheckout this module!" ;\
+	   echo "for cvs: use checkout once and after that update again" ;\
+	   echo "for cvsup: checkout kde-common from cvsup and" ;\
+	   echo "   link kde-common/admin to ./admin" ;\
+	   exit 1 ;\
+	fi
+	$(MAKE) -f admin/Makefile.common cvs
+
+.SILENT:
--- cervisia/Makefile.am	5 Nov 2002 03:00:09 -0000	1.1.1.3
+++ cervisia/Makefile.am	1 Jun 2003 14:59:19 -0000	1.4
@@ -4,9 +4,9 @@
 AM_CPPFLAGS = -UQT_NO_COMPAT -D_BSD_SOURCE
 
 bin_PROGRAMS = cervisia
-lib_LTLIBRARIES = libcervisia.la
+lib_LTLIBRARIES = libcervisiapart_shared.la cervisiapart.la cervisia.la
 
-libcervisia_la_SOURCES = updateview.cpp protocolview.cpp watchdlg.cpp \
+libcervisiapart_shared_la_SOURCES = cervisiashell.cpp updateview.cpp protocolview.cpp watchdlg.cpp \
 		   settingsdlg.cpp changelogdlg.cpp historydlg.cpp repositorydlg.cpp \
 		   commitdlg.cpp checkoutdlg.cpp updatedlg.cpp \
 		   tagdlg.cpp mergedlg.cpp \
@@ -14,12 +14,26 @@
 		   annotatedlg.cpp annotateview.cpp \
 		   cvsprogressdlg.cpp tiplabel.cpp listview.cpp \
 		   cvsdir.cpp repositories.cpp misc.cpp cervisiapart.cpp qttableview.cpp
-libcervisia_la_LDFLAGS = $(all_libraries) -version-info 1:0:0 -module -no-undefined
-libcervisia_la_LIBADD = $(LIB_KFILE) $(LIB_KPARTS)
+libcervisiapart_shared_la_LDFLAGS = $(all_libraries) -version-info 1:0:0 -no-undefined
+libcervisiapart_shared_la_LIBADD = $(LIB_KFILE) $(LIB_KPARTS) $(LIB_QT) $(LIB_KDECORE) $(LIB_KDEUI)
 
-cervisia_SOURCES = main.cpp cervisiashell.cpp
+cervisiapart_la_SOURCES = cervisiapart_la_dummy.cpp
+cervisiapart_la_LDFLAGS = $(all_libraries) -avoid-version -module
+cervisiapart_la_LIBADD  = libcervisiapart_shared.la
+
+cervisiapart_la_dummy.cpp:
+	echo > cervisiapart_la_dummy.cpp
+
+cervisia_la_SOURCES = cervisia_la_main.cpp
+cervisia_la_LDFLAGS = $(all_libraries) $(KDE_RPATH) -module -avoid-version
+cervisia_la_LIBADD  = libcervisiapart_shared.la
+
+cervisia_la_main.cpp: main.cpp
+	cat main.cpp > cervisia_la_main.cpp
+
+cervisia_SOURCES = main.cpp
 cervisia_LDFLAGS = $(all_libraries) $(KDE_RPATH)
-cervisia_LDADD = libcervisia.la
+cervisia_LDADD = libcervisiapart_shared.la
 
 man_MANS = cervisia.1
 
--- cervisia/updateview.cpp	3 Mar 2003 03:53:43 -0000	1.1.1.5
+++ cervisia/updateview.cpp	14 Jun 2003 02:31:23 -0000	1.9
@@ -313,7 +313,7 @@
 	    time_t time;
 
 	    oldLocale = setlocale(LC_TIME, "C");
-	    strptime(timestamp.local8Bit(), "%c" , &tmp);
+	    strptime(timestamp.latin1(), "%c" , &tmp);
 	    setlocale(LC_TIME, oldLocale);
 	    time = mktime(&tmp);
 	    updateEntriesItem(name, status, isdir, isbin, rev, tagdate, time);
--- kbabel/catalogmanager/Makefile.am	5 Nov 2002 03:00:24 -0000	1.1.1.3
+++ kbabel/catalogmanager/Makefile.am	1 Jun 2003 14:59:19 -0000	1.4
@@ -6,7 +6,7 @@
 
 # this is the program that gets installed.  it's name is used for all
 # of the other Makefile.am variables
-lib_LTLIBRARIES = libcatalogmanager.la
+lib_LTLIBRARIES = catalogmanager.la libcatalogmanager_main.la
 bin_PROGRAMS = catalogmanager
 
 # set the include path for X, qt and KDE
@@ -14,21 +14,29 @@
 
 
 # which sources should be compiled for kbabel
-libcatalogmanager_la_SOURCES =	catalogmanageriface.skel \
+libcatalogmanager_main_la_SOURCES =	catalogmanageriface.skel \
 			prefwidgets.cpp catalogmanagerview.cpp \
 			catalogmanager.cpp cmdedit.cpp finddialog.cpp settings.cpp \
-			catmanlistitem.cpp roughtransdlg.cpp
+			catmanlistitem.cpp roughtransdlg.cpp main.cpp
 
+libcatalogmanager_main_la_LIBADD =  ../common/libkbabelcommon.la ../kbabeldict/libkbabeldict_main.la $(LIB_KIO)
+libcatalogmanager_main_la_LDFLAGS = $(all_libraries) -version-info 1:0:0 -no-undefined
 
-libcatalogmanager_la_LIBADD =  ../common/libkbabelcommon.la ../kbabeldict/libkbabeldict.la $(LIB_KIO)
-libcatalogmanager_la_LDFLAGS = $(all_libraries) -module -version-info 1:0:0 -no-undefined
+catalogmanager_la_SOURCES = catalogmanager_la_main.cpp
+catalogmanager_la_LDFLAGS = $(all_libraries) -avoid-version -module
+catalogmanager_la_LIBADD  = libcatalogmanager_main.la
 
+catalogmanager_la_main.cpp: stub_main.cpp
+	cat stub_main.cpp > catalogmanager_la_main.cpp
 
-catalogmanager_SOURCES = main.cpp
+catalogmanager_SOURCES = catalogmanager_main.cpp
 
 # the libraries to link against.
-catalogmanager_LDADD   = libcatalogmanager.la
+catalogmanager_LDADD   = libcatalogmanager_main.la
 catalogmanager_LDFLAGS = $(all_libraries) $(KDE_RPATH)
+
+catalogmanager_main.cpp: stub_main.cpp
+	cat stub_main.cpp > catalogmanager_main.cpp
 
 # these are the headers for your project
 noinst_HEADERS   = catalogmanageriface.h prefwidgets.h catalogmanager.h cmdedit.h\
--- kbabel/catalogmanager/main.cpp	29 Oct 2002 18:25:01 -0000	1.1.1.4
+++ kbabel/catalogmanager/main.cpp	1 Dec 2002 17:02:09 -0000	1.2
@@ -152,8 +152,8 @@
    {0,0,0}
 };
 
-
-int main(int argc, char **argv)
+extern "C" int kdemain(int, char**);
+int kdemain(int argc, char **argv)
 {
     KLocale::setMainCatalogue("kbabel");
     KAboutData about("catalogmanager",I18N_NOOP("KBabel - Catalog Manager"),VERSION,
--- /dev/null	1 Jan 1970 00:00:00 -0000
+++ kbabel/catalogmanager/stub_main.cpp	29 Nov 2002 19:01:22 -0000	1.1
@@ -0,0 +1,28 @@
+/*
+ *
+ * Copyright (c) 2001 Nick Hudson <skrll@netbsd.org>
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+ * AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN
+ * AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+extern "C" int kdemain(int, char* []);
+
+int main( int argc, char* argv[] )
+{
+        return kdemain(argc, argv);
+}
+
--- kbabel/common/Makefile.am	29 Oct 2002 18:25:03 -0000	1.1.1.5
+++ kbabel/common/Makefile.am	29 Oct 2002 18:31:05 -0000	1.4
@@ -11,7 +11,7 @@
 lib_LTLIBRARIES = libkbabelcommon.la
 
 # set the include path for X, qt and KDE
-INCLUDES         = $(all_includes)
+INCLUDES         = $(all_includes) -I$(top_srcdir)/kbabel/kbabel
 
 # which sources should be compiled 
 libkbabelcommon_la_SOURCES = catalog.cpp catalogitem.cpp editcmd.cpp\
--- kbabel/kbabel/Makefile.am	10 Jul 2002 05:29:11 -0000	1.1.1.2
+++ kbabel/kbabel/Makefile.am	7 May 2003 00:34:56 -0000	1.3
@@ -6,7 +6,7 @@
 
 # this is the program that gets installed.  it's name is used for all
 # of the other Makefile.am variables
-lib_LTLIBRARIES = libkbabel.la
+lib_LTLIBRARIES = kbabel.la libkbabel_main.la
 bin_PROGRAMS = kbabel
 
 # set the include path for X, qt and KDE
@@ -14,7 +14,7 @@
 
 
 # which sources should be compiled for kbabel
-libkbabel_la_SOURCES =	kbabel.cpp kbabelview.cpp kbabelview2.cpp\
+libkbabel_main_la_SOURCES = main.cpp kbabel.cpp kbabelview.cpp kbabelview2.cpp\
 			kbabelpref.cpp kbabeliface.skel \
 			kbcatalog.cpp \
 			headereditor.cpp prefwidgets.cpp\
@@ -23,15 +23,24 @@
 			finddialog.cpp spelldlg.cpp roughtransdlg.cpp settings.cpp
 
 
-libkbabel_la_LIBADD =  ../common/libkbabelcommon.la ../kbabeldict/libkbabeldict.la $(LIB_KIO) $(LIB_KSPELL)
-libkbabel_la_LDFLAGS = $(all_libraries) -module -version-info 1:0:0 -no-undefined
+libkbabel_main_la_LIBADD =  ../common/libkbabelcommon.la ../kbabeldict/libkbabeldict_main.la $(LIB_KIO) $(LIB_KSPELL)
+libkbabel_main_la_LDFLAGS = $(all_libraries) -version-info 1:0:0 -no-undefined
 
+kbabel_la_SOURCES = kbabel_la_main.cpp
+kbabel_la_LIBADD  = libkbabel_main.la
+kbabel_la_LDFLAGS = $(all_libraries) -module -avoid-version
 
-kbabel_SOURCES = main.cpp
+kbabel_la_main.cpp: stub_main.cpp
+	cat stub_main.cpp > kbabel_la_main.cpp
+
+kbabel_SOURCES = kbabel_main.cpp
 
 # the libraries to link against.
-kbabel_LDADD   = libkbabel.la
+kbabel_LDADD   = libkbabel_main.la
 kbabel_LDFLAGS = $(all_libraries) $(KDE_RPATH)
+
+kbabel_main.cpp: stub_main.cpp
+	cat stub_main.cpp > kbabel_main.cpp
 
 # these are the headers for your project
 noinst_HEADERS   = kbabel.h kbabelview.h kbabeliface.h kbabelpref.h\
--- kbabel/kbabel/main.cpp	7 May 2003 00:28:56 -0000	1.1.1.7
+++ kbabel/kbabel/main.cpp	7 May 2003 00:31:36 -0000	1.4
@@ -408,7 +408,8 @@
 };
 
 
-int main(int argc, char **argv)
+extern "C" int kdemain(int, char**);
+int kdemain(int argc, char **argv)
 {
     KAboutData about("kbabel",I18N_NOOP("KBabel"),VERSION,
        I18N_NOOP("An advanced PO file editor"),KAboutData::License_GPL,
--- /dev/null	1 Jan 1970 00:00:00 -0000
+++ kbabel/kbabel/stub_main.cpp	29 Nov 2002 19:01:23 -0000	1.1
@@ -0,0 +1,28 @@
+/*
+ *
+ * Copyright (c) 2001 Nick Hudson <skrll@netbsd.org>
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+ * AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN
+ * AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+extern "C" int kdemain(int, char* []);
+
+int main( int argc, char* argv[] )
+{
+        return kdemain(argc, argv);
+}
+
--- /dev/null	1 Jan 1970 00:00:00 -0000
+++ kbabel/kbabel/version.h	17 Aug 2002 17:46:03 -0000	1.1
@@ -0,0 +1,3 @@
+#undef VERSION
+#define VERSION "0.5.5"
+
--- kbabel/kbabeldict/Makefile.am	29 Oct 2002 18:25:10 -0000	1.1.1.4
+++ kbabel/kbabeldict/Makefile.am	1 Jun 2003 14:59:20 -0000	1.3
@@ -8,7 +8,7 @@
 
 # this is the program that gets installed.  it's name is used for all
 # of the other Makefile.am variables
-lib_LTLIBRARIES = libkbabeldict.la libkbabeldictplugin.la
+lib_LTLIBRARIES = kbabeldict.la libkbabeldictplugin.la libkbabeldictplugin_shared.la libkbabeldict_main.la
 bin_PROGRAMS = kbabeldict
 
 # set the include path for X, qt and KDE
@@ -16,21 +16,34 @@
 
 
 # which sources should be compiled for kbabeldict
-libkbabeldictplugin_la_SOURCES = searchengine.cpp 
-libkbabeldictplugin_la_LIBADD =  $(LIB_KDECORE)
-libkbabeldictplugin_la_LDFLAGS = $(all_libraries) -module -version-info 1:0:0 -no-undefined
+libkbabeldictplugin_shared_la_SOURCES = searchengine.cpp 
+libkbabeldictplugin_shared_la_LIBADD =  $(LIB_KDECORE)
+libkbabeldictplugin_shared_la_LDFLAGS = $(all_libraries) -version-info 1:0:0 -no-undefined
+
+libkbabeldictplugin_la_SOURCES = libkbabeldictplugin_la_dummy.cpp
+libkbabeldictplugin_la_LIBADD  = libkbabeldictplugin_shared.la
+libkbabeldictplugin_la_LDFLAGS = $(all_libraries) -avoid-version -no-undefined
 
-libkbabeldict_la_SOURCES = kbabeldictbox.cpp \
+libkbabeldictplugin_la_dummy.cpp:
+	echo > libkbabeldictplugin_la_dummy.cpp
+
+libkbabeldict_main_la_SOURCES = kbabeldictbox.cpp \
             kbabeldictiface.skel dictionarymenu.cpp dictchooser.cpp \
-            aboutmoduledlg.cpp 
-libkbabeldict_la_LIBADD =  ../common/libkbabelcommon.la libkbabeldictplugin.la $(LIB_KDEUI)
-libkbabeldict_la_LDFLAGS = $(all_libraries) -module -version-info 1:0:0 -no-undefined
+            aboutmoduledlg.cpp kbabeldictview.cpp kbabeldict.cpp kbabelsplash.cpp
+libkbabeldict_main_la_LIBADD =  ../common/libkbabelcommon.la libkbabeldictplugin_shared.la $(LIB_KDEUI)
+libkbabeldict_main_la_LDFLAGS = $(all_libraries) -version-info 1:0:0 -no-undefined
+
+kbabeldict_la_SOURCES = kbabeldict_main.cpp
+kbabeldict_la_LDFLAGS = $(all_libraries) -module -avoid-version
+kbabeldict_la_LIBADD  = libkbabeldict_main.la
 
+kbabeldict_main.cpp:
+	cat main.cpp > kbabeldict_main.cpp
 
-kbabeldict_SOURCES = main.cpp kbabeldictview.cpp kbabeldict.cpp kbabelsplash.cpp
+kbabeldict_SOURCES = main.cpp
 
 # the libraries to link against.
-kbabeldict_LDADD   = libkbabeldict.la
+kbabeldict_LDADD   = libkbabeldict_main.la
 kbabeldict_LDFLAGS = $(all_libraries) $(KDE_RPATH)
 
 # these are the headers for your project
@@ -42,7 +55,7 @@
 # let automoc handle all of the meta source files (moc)
 METASOURCES = AUTO
 
-KDE_OPTIONS = nofinal
+#KDE_OPTIONS = nofinal
 #rcdir = $(kde_datadir)/kbabel
 #rc_DATA = kbabelui.rc
 
--- kbabel/kbabeldict/modules/dbsearchengine/configure.in.in	15 May 2003 03:17:47 -0000	1.1.1.4
+++ kbabel/kbabeldict/modules/dbsearchengine/configure.in.in	15 May 2003 14:09:27 -0000	1.2
@@ -77,15 +77,15 @@
    AC_DEFINE_UNQUOTED(HAVE_DB_DB_H, 1, [DB 4 header location] )
    ;;
  db4-db)
-   LIB_DBIV='-ldb4'
+   LIB_DBIV='-ldb-4'
    kde_db_header=db.h
-   AC_MSG_RESULT("flag is -ldb4 and header is db.h")   
+   AC_MSG_RESULT("flag is -ldb-4 and header is db.h")   
    AC_DEFINE_UNQUOTED(HAVE_DB_DB_H, 1, [DB 4 header location] )
    ;;
  db4)
-   LIB_DBIV='-ldb4'
+   LIB_DBIV='-ldb-4'
    kde_db_header=db4/db.h
-   AC_MSG_RESULT(-ldb4) 
+   AC_MSG_RESULT(-ldb-4)
    AC_DEFINE_UNQUOTED(HAVE_DB4_DB_H, 1, [DB 4 header location] )
 esac
 
--- kmtrace/Makefile.am	10 Jul 2002 05:30:08 -0000	1.1.1.2
+++ kmtrace/Makefile.am	1 Jun 2003 14:59:20 -0000	1.2
@@ -40,7 +40,7 @@
 	$(LN_S) $< $@
 
 lib_LTLIBRARIES = libktrace.la
-libktrace_la_LDFLAGS = -version-info 1:0 
+libktrace_la_LDFLAGS = -version-info 1:0 -no-undefined
 libktrace_la_SOURCES = ksotrace.cpp ktrace.c
 libktrace_la_LIBADD = $(LIBDL)
 
--- kompare/komparepart/Makefile.am	10 Jul 2002 05:30:35 -0000	1.1.1.2
+++ kompare/komparepart/Makefile.am	1 Jun 2003 14:59:21 -0000	1.4
@@ -25,7 +25,8 @@
 # let automoc handle all of the meta source files (moc)
 METASOURCES = AUTO
 
-kde_module_LTLIBRARIES = libkomparepart.la
+lib_LTLIBRARIES = libkomparepart.la
+kde_module_LTLIBRARIES = komparepart.la
 
 # the Part's source, library search path, and link libraries
 libkomparepart_la_SOURCES = \
@@ -46,9 +47,16 @@
 	komparesaveoptionsbase.ui \
 	komparesaveoptionswidget.cpp
 
-libkomparepart_la_LDFLAGS = -module $(KDE_PLUGIN) $(all_libraries)
+libkomparepart_la_LDFLAGS = $(KDE_RPATH) $(all_libraries) -no-undefined
 libkomparepart_la_LIBADD  = $(LIB_KPARTS) $(LIB_KFILE) \
 	$(top_builddir)/kompare/libdiff2/libdiff2.la
+
+komparepart_la_SOURCES = dummy.cpp
+komparepart_la_LDFLAGS = $(all_libraries) $(KDE_PLUGIN) -module
+komparepart_la_LIBADD  = libkomparepart.la
+
+dummy.cpp:
+	echo > dummy.cpp
 
 # this is where the desktop file will go
 partdesktopdir   = $(kde_servicesdir)
--- kompare/komparepart/komparepart.desktop	7 May 2003 00:28:31 -0000	1.1.1.4
+++ kompare/komparepart/komparepart.desktop	7 May 2003 00:31:36 -0000	1.3
@@ -16,6 +16,6 @@
 Name[zu]=Ingxenye ye-Kompare
 MimeType=text/x-diff
 ServiceTypes=KParts/ReadOnlyPart,KParts/ReadWritePart,Kompare/ViewPart
-X-KDE-Library=libkomparepart
+X-KDE-Library=komparepart
 Type=Service
 Icon=kompare
--- kompare/libdiff2/Makefile.am	4 Jan 2003 20:24:03 -0000	1.1.1.2
+++ kompare/libdiff2/Makefile.am	1 Jun 2003 14:59:21 -0000	1.5
@@ -21,6 +21,6 @@
 	difference.cpp \
 	diffhunk.cpp
 
-libdiff2_la_LDFLAGS = $(all_libraries)
+libdiff2_la_LDFLAGS = $(all_libraries) -no-undefined
 libdiff2_la_LIBADD  = $(LIB_KPARTS) $(LIB_KFILE)
 
