diff -uNbr kmplayer/kmplayer/Makefile.am kmplayer-new/kmplayer/Makefile.am
--- kmplayer/kmplayer/Makefile.am	Sat Feb 15 10:24:45 2003
+++ kmplayer-new/kmplayer/Makefile.am	Tue Feb 18 09:49:22 2003
@@ -2,33 +2,42 @@
 
 SUBDIRS=
 
-lib_LTLIBRARIES=kdeinit_kmplayer.la
+lib_LTLIBRARIES=libkparts_kmplayer_shared.la kdeinit_kmplayer.la libkdeinit_kmplayer_shared.la
 
 kde_module_LTLIBRARIES = kparts_kmplayer.la
 
-kparts_kmplayer_la_SOURCES= kmplayer_part.cpp kmplayerview.cpp configdialog.ui kmplayerconfig.cpp
-
-kparts_kmplayer_la_LDFLAGS= $(all_libraries) -module $(KDE_PLUGIN) -lartskde -lm
-kparts_kmplayer_la_LIBADD= $(LIB_KFILE) $(LIB_KPARTS) $(LIB_KIO) -lkmediaplayer
-kparts_kmplayer_la_METASOURCES= kmplayer_part.moc kmplayerview.moc configdialog.moc kmplayerconfig.moc
-
-kdeinit_kmplayer_la_SOURCES= main.cpp kmplayerapp.cpp kmplayerdoc.cpp
-
-kdeinit_kmplayer_la_LIBADD= kparts_kmplayer.la
-kdeinit_kmplayer_la_LDFLAGS= $(all_libraries) $(KDE_RPATH) -module -avoid-version
-kdeinit_kmplayer_la_METASOURCES= kmplayer.moc kmplayerdoc.moc
+libkparts_kmplayer_shared_la_SOURCES= kmplayer_part.cpp kmplayerview.cpp configdialog.ui kmplayerconfig.cpp
+libkparts_kmplayer_shared_la_LDFLAGS= $(all_libraries) -avoid-version
+libkparts_kmplayer_shared_la_LIBADD= $(LIB_KFILE) $(LIB_KPARTS) $(LIB_KIO) $(LIB_KDECORE) $(LIB_KDEUI) -lkmediaplayer -lartskde -lm $(LIB_QT) -lDCOP -lartsflow_idl -lsoundserver_idl
+libkparts_kmplayer_shared_la_METASOURCES= kmplayer_part.moc kmplayerview.moc configdialog.moc kmplayerconfig.moc
+
+kparts_kmplayer_la_SOURCES = dummy.cpp
+kparts_kmplayer_la_LDFLAGS = $(all_libraries) -module $(KDE_PLUGIN) -avoid-version
+kparts_kmplayer_la_LIBADD  = libkparts_kmplayer_shared.la
+
+libkdeinit_kmplayer_shared_la_SOURCES= kmplayerapp.cpp kmplayerdoc.cpp
+libkdeinit_kmplayer_shared_la_LIBADD= libkparts_kmplayer_shared.la
+libkdeinit_kmplayer_shared_la_LDFLAGS= $(all_libraries) $(KDE_RPATH) -avoid-version -no-undefined
+libkdeinit_kmplayer_shared_la_METASOURCES= kmplayer.moc kmplayerdoc.moc
+
+kdeinit_kmplayer_la_SOURCES = main_la.cpp
+kdeinit_kmplayer_la_LIBADD  = libkdeinit_kmplayer_shared.la
+kdeinit_kmplayer_la_LDFLAGS = $(all_libraries) $(KDE_RPATH) -module -avoid-version
 
 bin_PROGRAMS= kmplayer
 
-kmplayer_LDADD= kdeinit_kmplayer.la
+kmplayer_LDADD= libkdeinit_kmplayer_shared.la
 kmplayer_LDFLAGS= $(all_libraries) $(KDE_RPATH)
-kmplayer_SOURCES= dummy.cpp
+kmplayer_SOURCES= main.cpp
 
 
-CLEANFILES = dummy.cpp
+CLEANFILES = dummy.cpp ain_la.cpp
 
 dummy.cpp:
 	echo > dummy.cpp
+
+main_la.cpp:
+	cat main.cpp > main_la.cpp
 
 applnk_DATA = kmplayer.desktop
 applnkdir = $(kde_appsdir)/Multimedia
