
--- example/Makefile.am	25 Apr 2003 15:43:28 -0000	1.1.1.3
+++ example/Makefile.am	9 Jul 2003 14:40:21 -0000	1.4
@@ -9,19 +9,12 @@
 libexamplepart_la_LIBADD = $(LIB_KOFFICEUI)
 
 ## The kdeinit loadable module
-lib_LTLIBRARIES = example.la
+bin_PROGRAMS = 
+lib_LTLIBRARIES =
+kdeinit_LTLIBRARIES = example.la
 example_la_SOURCES = main.cc
-example_la_LDFLAGS = $(all_libraries) -module -avoid-version
+example_la_LDFLAGS = $(all_libraries) $(KDE_RPATH) -module -avoid-version
 example_la_LIBADD = $(LIB_KOFFICEUI)
-
-## The executable
-bin_PROGRAMS = example
-example_SOURCES = dummy.cc
-example_LDFLAGS = $(all_libraries) $(KDE_RPATH)
-example_LDADD = example.la
-
-dummy.cc:
-	echo > dummy.cc
 
 METASOURCES = AUTO
 noinst_HEADERS =  example_part.h \
--- filters/karbon/ai/Makefile.am	25 Apr 2003 15:43:29 -0000	1.1.1.3
+++ filters/karbon/ai/Makefile.am	26 Apr 2003 02:07:12 -0000	1.3
@@ -2,7 +2,7 @@
 
 libkarbonaiimport_la_LIBADD = \
 	$(LIB_KOFFICEUI) \
-	../../../karbon/libkarbonpart.la
+	../../../karbon/libkarbonpart_shared.la
 
 libkarbonaiimport_la_LDFLAGS = \
 	$(all_libraries) \
--- filters/karbon/eps/Makefile.am	25 Apr 2003 15:43:29 -0000	1.1.1.3
+++ filters/karbon/eps/Makefile.am	26 Apr 2003 02:07:13 -0000	1.3
@@ -9,7 +9,7 @@
 libkarbonepsexport_la_LDFLAGS = $(KDE_PLUGIN)
 libkarbonepsexport_la_LIBADD  = \
 	$(LIB_KOFFICEUI) \
-	../../../karbon/libkarbonpart.la
+	../../../karbon/libkarbonpart_shared.la
 
 libkarbonepsimport_la_SOURCES = epsimport.cc pscommentlexer.cc
 libkarbonepsimport_la_LDFLAGS = $(KDE_PLUGIN)
--- filters/karbon/kontour/Makefile.am	25 Apr 2003 15:43:30 -0000	1.1.1.2
+++ filters/karbon/kontour/Makefile.am	27 Apr 2003 19:04:23 -0000	1.2
@@ -11,7 +11,7 @@
 
 libkarbonkontourimport_la_SOURCES = kontourimport.cpp
 libkarbonkontourimport_la_LDFLAGS = -module $(KDE_PLUGIN)
-libkarbonkontourimport_la_LIBADD = $(KOFFICE_LIBS) ../../../karbon/libkarbonpart.la
+libkarbonkontourimport_la_LIBADD = $(KOFFICE_LIBS) ../../../karbon/libkarbonpart_shared.la
 
 METASOURCES = AUTO
 
--- filters/karbon/msod/Makefile.am	25 Apr 2003 15:43:30 -0000	1.1.1.1
+++ filters/karbon/msod/Makefile.am	27 Apr 2003 19:04:23 -0000	1.2
@@ -2,7 +2,7 @@
 
 INCLUDES= -I$(srcdir) $(KOFFICE_INCLUDES) $(all_includes)
 libkarbonmsodimport_la_LDFLAGS	= -module -avoid-version -no-undefined $(all_libraries)
-libkarbonmsodimport_la_LIBADD = $(KOFFICE_LIBS)
+libkarbonmsodimport_la_LIBADD = $(KOFFICE_LIBS) $(LIBZ)
 
 ####### Files
 
--- filters/karbon/oodraw/Makefile.am	25 Apr 2003 15:43:30 -0000	1.1.1.1
+++ filters/karbon/oodraw/Makefile.am	27 Apr 2003 19:04:23 -0000	1.2
@@ -14,7 +14,7 @@
 liboodrawimport_la_SOURCES = oodrawimport.cc
 liboodrawimport_la_LDFLAGS = $(all_libraries) -module -avoid-version -no-undefined
 liboodrawimport_la_LIBADD  = ../../liboofilter/liboofilter.la \
-							../../../karbon/libkarbonpart.la \
+							../../../karbon/libkarbonpart_shared.la \
 							$(KOFFICE_LIBS)
 
 METASOURCES = AUTO
--- filters/karbon/png/Makefile.am	25 Apr 2003 15:43:30 -0000	1.1.1.2
+++ filters/karbon/png/Makefile.am	26 Apr 2003 02:07:13 -0000	1.3
@@ -3,7 +3,7 @@
 libkarbonpngexport_la_LDFLAGS = $(KDE_PLUGIN)
 libkarbonpngexport_la_LIBADD = \
 	$(LIB_KOFFICEUI) \
-	../../../karbon/libkarbonpart.la
+	../../../karbon/libkarbonpart_shared.la
 
 INCLUDES = \
 	$(KOFFICE_INCLUDES) $(KOPAINTER_INCLUDES) \
--- filters/karbon/svg/Makefile.am	25 Apr 2003 15:43:31 -0000	1.1.1.3
+++ filters/karbon/svg/Makefile.am	26 Apr 2003 02:07:14 -0000	1.3
@@ -3,12 +3,12 @@
 libkarbonsvgexport_la_LDFLAGS = $(KDE_PLUGIN)
 libkarbonsvgexport_la_LIBADD = \
 	$(LIB_KOFFICEUI) \
-	../../../karbon/libkarbonpart.la
+	../../../karbon/libkarbonpart_shared.la
 
 libkarbonsvgimport_la_LDFLAGS = $(KDE_PLUGIN)
 libkarbonsvgimport_la_LIBADD = \
 	$(LIB_KOFFICEUI) \
-	../../../karbon/libkarbonpart.la
+	../../../karbon/libkarbonpart_shared.la
 
 INCLUDES = \
 	$(KOFFICE_INCLUDES) \
--- filters/karbon/wmf/Makefile.am	5 Jul 2003 18:31:01 -0000	1.1.1.3
+++ filters/karbon/wmf/Makefile.am	6 Jul 2003 14:11:25 -0000	1.3
@@ -12,12 +12,11 @@
 kde_module_LTLIBRARIES = libwmfimport.la libwmfexport.la
 
 libwmfimport_la_LDFLAGS = -module $(KDE_PLUGIN)
-libwmfimport_la_LIBADD = $(KOFFICE_LIBS) $(LIB_KOWMF) ../../../karbon/libkarbonpart.la
-#libwmfimport_la_LIBADD = $(KOFFICE_LIBS) $(LIB_KOWMF) ../../../karbon/libkarbonpart.la
+libwmfimport_la_LIBADD = $(KOFFICE_LIBS) $(LIB_KOWMF) ../../../karbon/libkarbonpart_shared.la
 libwmfimport_la_SOURCES = wmfimport.cc wmfimportparser.cc
 
 libwmfexport_la_LDFLAGS = -module $(KDE_PLUGIN)
-libwmfexport_la_LIBADD = $(KOFFICE_LIBS) $(LIB_KOWMF) ../../../karbon/libkarbonpart.la
+libwmfexport_la_LIBADD = $(KOFFICE_LIBS) $(LIB_KOWMF) ../../../karbon/libkarbonpart_shared.la
 libwmfexport_la_SOURCES = wmfexport.cc
 
 noinst_HEADERS = wmfimport.h wmfimportparser.h wmfexport.h 
--- filters/karbon/xcf/Makefile.am	25 Apr 2003 15:43:31 -0000	1.1.1.1
+++ filters/karbon/xcf/Makefile.am	27 Apr 2003 19:04:24 -0000	1.2
@@ -3,7 +3,7 @@
 libkarbonxcfexport_la_LDFLAGS = $(KDE_PLUGIN)
 libkarbonxcfexport_la_LIBADD = \
 	$(LIB_KOFFICEUI) \
-	../../../karbon/libkarbonpart.la
+	../../../karbon/libkarbonpart_shared.la
 
 INCLUDES = \
 	$(KOFFICE_INCLUDES) $(KOPAINTER_INCLUDES) \
--- filters/kspread/csv/Makefile.am	25 Apr 2003 15:43:38 -0000	1.1.1.2
+++ filters/kspread/csv/Makefile.am	26 Apr 2003 02:07:15 -0000	1.3
@@ -9,11 +9,11 @@
 
 libcsvimport_la_SOURCES = csvimport.cc xmltree.cc csvdialog.cpp dialogui.ui
 libcsvimport_la_LDFLAGS	= $(all_libraries) -module -avoid-version -no-undefined 
-libcsvimport_la_LIBADD  = $(KOFFICE_LIBS) ../../../kspread/libkspreadpart.la
+libcsvimport_la_LIBADD  = $(KOFFICE_LIBS) ../../../kspread/libkspreadpart_shared.la
 
 libcsvexport_la_SOURCES = csvexport.cc csvexportdialog.cpp exportdialogui.ui
 libcsvexport_la_LDFLAGS = $(all_libraries) -module -avoid-version -no-undefined
-libcsvexport_la_LIBADD  = $(KOFFICE_LIBS) ../../../kspread/libkspreadpart.la
+libcsvexport_la_LIBADD  = $(KOFFICE_LIBS) ../../../kspread/libkspreadpart_shared.la
 
 METASOURCES = AUTO
 
--- filters/kspread/gnumeric/Makefile.am	25 Apr 2003 15:43:38 -0000	1.1.1.2
+++ filters/kspread/gnumeric/Makefile.am	27 Apr 2003 19:04:24 -0000	1.5
@@ -4,15 +4,15 @@
 	$(KOTEXT_INCLUDES) $(KOSCRIPT_INCLUDES) $(all_includes)
 
 ####### Files
-kde_module_LTLIBRARIES = libgnumericexport.la libgnumericimport.la 
+kde_module_LTLIBRARIES = libgnumericexport.la libgnumericimport.la
 
 libgnumericexport_la_SOURCES = gnumericexport.cc
 libgnumericexport_la_LDFLAGS = $(all_libraries) -module -avoid-version -no-undefined
-libgnumericexport_la_LIBADD  = $(KOFFICE_LIBS) ../../../kspread/libkspreadpart.la
+libgnumericexport_la_LIBADD  = $(KOFFICE_LIBS) ../../../kspread/libkspreadpart_shared.la $(LIBZ)
 
 libgnumericimport_la_SOURCES = gnumericimport.cc
 libgnumericimport_la_LDFLAGS = $(all_libraries) -module -avoid-version -no-undefined
-libgnumericimport_la_LIBADD  = $(KOFFICE_LIBS) ../../../kspread/libkspreadpart.la
+libgnumericimport_la_LIBADD  = $(KOFFICE_LIBS) ../../../kspread/libkspreadpart_shared.la $(LIBZ)
 
 METASOURCES = AUTO
 
--- filters/kspread/html/Makefile.am	25 Apr 2003 15:43:39 -0000	1.1.1.2
+++ filters/kspread/html/Makefile.am	26 Apr 2003 02:07:16 -0000	1.3
@@ -9,7 +9,7 @@
 
 libkspreadhtmlexport_la_SOURCES = htmlexport.cc
 libkspreadhtmlexport_la_LDFLAGS = $(all_libraries) -module -avoid-version -no-undefined
-libkspreadhtmlexport_la_LIBADD  = $(KOFFICE_LIBS) ../../../kspread/libkspreadpart.la
+libkspreadhtmlexport_la_LIBADD  = $(KOFFICE_LIBS) ../../../kspread/libkspreadpart_shared.la
 
 METASOURCES = AUTO
 
--- filters/kspread/opencalc/Makefile.am	25 Apr 2003 15:43:40 -0000	1.1.1.1
+++ filters/kspread/opencalc/Makefile.am	27 Apr 2003 19:04:24 -0000	1.2
@@ -8,11 +8,11 @@
 
 libopencalcexport_la_SOURCES = opencalcexport.cc opencalcstyleexport.cc
 libopencalcexport_la_LDFLAGS = $(all_libraries) -module -avoid-version -no-undefined
-libopencalcexport_la_LIBADD  = $(KOFFICE_LIBS) ../../../kspread/libkspreadpart.la
+libopencalcexport_la_LIBADD  = $(KOFFICE_LIBS) ../../../kspread/libkspreadpart_shared.la
 
 libopencalcimport_la_SOURCES = opencalcimport.cc
 libopencalcimport_la_LDFLAGS = $(all_libraries) -module -avoid-version -no-undefined
-libopencalcimport_la_LIBADD  = $(KOFFICE_LIBS) ../../../kspread/libkspreadpart.la
+libopencalcimport_la_LIBADD  = $(KOFFICE_LIBS) ../../../kspread/libkspreadpart_shared.la
 
 METASOURCES = AUTO
 
--- filters/kspread/qpro/Makefile.am	25 Apr 2003 15:43:41 -0000	1.1.1.2
+++ filters/kspread/qpro/Makefile.am	26 Apr 2003 02:07:16 -0000	1.3
@@ -13,7 +13,7 @@
 libqproimport_la_SOURCES = qproimport.cc qproformula.cc
 
 libqproimport_la_LDFLAGS = $(all_libraries) -module -avoid-version -no-undefined
-libqproimport_la_LIBADD  = $(KOFFICE_LIBS) ../../../kspread/libkspreadpart.la \
+libqproimport_la_LIBADD  = $(KOFFICE_LIBS) ../../../kspread/libkspreadpart_shared.la \
 	libqpro/src/libqpro.la
 
 METASOURCES = AUTO
--- filters/kword/Makefile.am	5 Jul 2003 18:31:12 -0000	1.1.1.3
+++ filters/kword/Makefile.am	9 Jul 2003 14:40:22 -0000	1.2
@@ -2,5 +2,6 @@
 MSWORD_SUBDIR=msword
 endif
 
+# removed pdf, broken on OSX currently
 SUBDIRS = libexport abiword amipro applixword ascii docbook html latex mswrite \
-	  oowriter palmdoc rtf wml wordperfect pdf $(MSWORD_SUBDIR)
+	  oowriter palmdoc rtf wml wordperfect $(MSWORD_SUBDIR)
--- filters/kword/latex/import/parser/Makefile.am	25 Apr 2003 15:43:49 -0000	1.1.1.5
+++ filters/kword/latex/import/parser/Makefile.am	26 Apr 2003 02:07:18 -0000	1.6
@@ -1,7 +1,7 @@
 ####### General stuff
 
 INCLUDES= -I$(srcdir) $(all_includes)
-liblatexparser_la_LDFLAGS = $(all_libraries) -module -avoid-version -no-undefined
+liblatexparser_la_LDFLAGS = $(all_libraries) -avoid-version -no-undefined
 liblatexparser_la_LIBADD  = -lm $(KOFFICE_LIBS)
 
 ####### Files
--- filters/kword/mswrite/kword_mswrite_export.desktop	5 Jul 2003 18:31:18 -0000	1.1.1.2
+++ filters/kword/mswrite/kword_mswrite_export.desktop	9 Jul 2003 14:40:22 -0000	1.2
@@ -3,18 +3,23 @@
 Type=Service
 Name=KWord Microsoft Write Export Filter
 Name[bg]=Филтър за импорт от Microsoft Write към KWord
+Name[ca]=Filtre d'exportació Microsoft Write per a KWord
 Name[cs]=Exportní filtr formátu MS Write pro KWord
+Name[cy]=Hidl Allforio Microsoft Write KWord
 Name[da]=KWord Microsoft Write-eksportfilter
 Name[el]=Φίλτρο εξαγωγής σε Microsoft Write του KWord
 Name[es]=Filtro de importación a Microsoft Write de KWord
 Name[fa]=KWord برای Microsoft Write صافی صدور
 Name[fr]=Filtre d'exportation MS-Write de KWord
 Name[hu]=KWord Microsoft Write exportszűrő
+Name[it]=Filtro di esportazione Microsoft Write per KWord
 Name[lv]=KWord Microsoft Write Eksporta Filtrs
 Name[pt]=Filtro de Exportação de Microsoft Write para o KWord
 Name[pt_BR]=Filtro de Exportação Microsoft Write do KWord
 Name[ru]=Фильтр экспорта KWord в Microsoft Write
+Name[sk]=Microsoft Write filter pre export z KWordu
 Name[sv]=Kword-exportfilter för Microsoft Write
+Name[tr]=KWord Microsoft Write Alma Filtresi 
 Name[wa]=Passete MS-Write di rexhowe po KWord
 X-KDE-Export=application/x-mswrite
 X-KDE-Import=application/x-kword
--- filters/kword/mswrite/kword_mswrite_import.desktop	5 Jul 2003 18:31:18 -0000	1.1.1.6
+++ filters/kword/mswrite/kword_mswrite_import.desktop	9 Jul 2003 14:40:22 -0000	1.2
@@ -6,6 +6,7 @@
 Name[bg]=Филтър за импорт от Microsoft Write към KWord
 Name[ca]=Filtre d'importació Microsoft Write per a KWord
 Name[cs]=Importní filtr Microsoft Write pro KWord
+Name[cy]=Hidl Mewnforio Microsoft Write KWord 
 Name[da]=KWord Microsoft Write-importfilter
 Name[de]=KWord-Import für MS Write
 Name[el]=Φίλτρο εισαγωγής Microsoft Write του KWord
--- filters/kword/mswrite/libmswrite.cpp	5 Jul 2003 18:31:18 -0000	1.1.1.2
+++ filters/kword/mswrite/libmswrite.cpp	9 Jul 2003 14:40:23 -0000	1.2
@@ -328,6 +328,14 @@
 	}
 
 
+	Generator::Generator ()
+	{
+	}
+
+	Generator::~Generator ()
+	{
+	}
+
 	bool Generator::processText (const Byte *string, bool willReachEndOfParagraph)
 	{
 		//
--- filters/kword/mswrite/libmswrite.h	5 Jul 2003 18:31:19 -0000	1.1.1.2
+++ filters/kword/mswrite/libmswrite.h	9 Jul 2003 14:40:23 -0000	1.2
@@ -86,7 +86,7 @@
 		FormatInfo &operator= (const FormatInfo &rhs);	// TODO
 	public:
 		FormatInfo ();
-		~FormatInfo ();
+		virtual ~FormatInfo ();
 
 		void setType (const enum FormatInfoPageTypes t)	{	m_type = t;	}
 
@@ -129,6 +129,9 @@
 	class Generator
 	{
 	public:
+		Generator ();
+		virtual ~Generator ();
+
 		/**
 		 *
 		 * Document block (you must implement these signals)
@@ -251,7 +254,7 @@
 
 	public:
 		InternalParser ();
-		~InternalParser ();
+		virtual ~InternalParser ();
 
 		bool parse (void);
 	};
@@ -274,7 +277,7 @@
 		
 	public:
 		InternalGenerator ();
-		~InternalGenerator ();
+		virtual ~InternalGenerator ();
 
 		bool writeDocumentBegin (const Word format, const PageLayout *pageLayout);
 		bool writeDocumentEnd (const Word format, const PageLayout *pageLayout);
--- filters/kword/mswrite/libmswrite_defs.cpp	5 Jul 2003 18:31:19 -0000	1.1.1.2
+++ filters/kword/mswrite/libmswrite_defs.cpp	9 Jul 2003 14:40:23 -0000	1.2
@@ -23,7 +23,7 @@
 
 namespace MSWrite
 {
-	const char *Version = "0.51pre 2003/06/08";
+	const char *Version = "0.51pre 2003/07/09";
 }
 
 // end of libmswrite_defs.cpp
--- filters/kword/mswrite/libmswrite_defs.h	5 Jul 2003 18:31:19 -0000	1.1.1.2
+++ filters/kword/mswrite/libmswrite_defs.h	9 Jul 2003 14:40:24 -0000	1.2
@@ -434,6 +434,14 @@
 	class MemoryDevice : public Device
 	{
 	public:
+		MemoryDevice ()
+		{
+		}
+
+		virtual ~MemoryDevice ()
+		{
+		}
+
 		bool read (Byte * /*buf*/, const DWord /*numBytes*/)
 		{
 			error (Error::InternalError, "memory device not reading from memory?\n");
@@ -502,6 +510,10 @@
 			setVal (val);
 		}
 
+		~UseThisMuchPrefixSize ()
+		{
+		}
+
 		bool operator== (const UseThisMuchPrefixSize &rhs)
 		{
 			return m_val == rhs.m_val;
@@ -573,6 +585,15 @@
 				return biggest / 8 + 1;	// account for fractional byte
 			else
 				return biggest / 8;
+		}
+
+	public:
+		UseThisMuch ()
+		{
+		}
+
+		virtual ~UseThisMuch ()
+		{
 		}
 	};
 	
--- filters/kword/mswrite/list.h	5 Jul 2003 18:31:19 -0000	1.1.1.2
+++ filters/kword/mswrite/list.h	9 Jul 2003 14:40:24 -0000	1.2
@@ -61,6 +61,10 @@
 			m_data = data;
 		}
 
+		~ListElement ()
+		{
+		}
+
 		dtype &getData (void) const
 		{
 			return m_data;
@@ -96,7 +100,7 @@
 	class ListIterator
 	{
 	private:
-		bool m_foward;
+		bool m_forward;
 		ListElement <dtype> *m_upto;
 
 		void setPtr (ListElement <dtype> *ptr)
@@ -105,14 +109,14 @@
 		}
 
 	public:
-		ListIterator (const bool foward = true)
+		ListIterator (const bool forward = true)
 		{
-			m_foward = foward;
+			m_forward = forward;
 		}
 
 		ListIterator <dtype> &prev (void)
 		{
-			if (m_foward)
+			if (m_forward)
 				m_upto = m_upto->m_prev;
 			else
 				m_upto = m_upto->m_next;
@@ -120,6 +124,10 @@
 			return *this;
 		}
 
+		~ListIterator ()
+		{
+		}
+		
 		ListIterator <dtype> &operator-- (void)
 		{
 			return prev ();
@@ -132,7 +140,7 @@
 
 		ListIterator <dtype> &next (void)
 		{
-			if (m_foward)
+			if (m_forward)
 				m_upto = m_upto->m_next;
 			else
 				m_upto = m_upto->m_prev;
@@ -185,7 +193,7 @@
 			m_good = true;
 		}
 
-		~List ()
+		virtual ~List ()
 		{
 			killself ();
 		}
@@ -308,10 +316,10 @@
 		typedef ListIterator <dtype> Iterator;
 		friend class ListIterator <dtype>;
 
-		ListIterator <dtype> begin (const bool foward = true) const
+		ListIterator <dtype> begin (const bool forward = true) const
 		{
-			ListIterator <dtype> ret (foward);
-			if (foward)
+			ListIterator <dtype> ret (forward);
+			if (forward)
 				ret.setPtr (m_head);
 			else
 				ret.setPtr (m_tail);
--- filters/kword/mswrite/mswriteexport.cc	5 Jul 2003 18:31:19 -0000	1.1.1.2
+++ filters/kword/mswrite/mswriteexport.cc	9 Jul 2003 14:40:25 -0000	1.4
@@ -1,4 +1,4 @@
-/* $Id$ */
+/* $Id$ */
 
 /* This file is part of the KDE project
    Copyright (C) 2002-2003 Clarence Dang <dang@kde.org>
@@ -408,9 +408,13 @@
 	// In Write the header/footer must be the same for every page except that
 	// you can choose to not display it on the first page
 	//
-	// This filter aims to be as lossless as possible so if we can't
-	// accomodate the types of headers/footers found in KWord, we at least
-	// print out the paragraphs in the body
+	// /*This filter aims to be as lossless as possible so if we can't
+	// accommodate the types of headers/footers found in KWord, we at least
+	// print out the paragraphs in the body*/
+	//
+	//    Not anymore. Dumping headers & footers in the body didn't
+	//    turn out to be that useful, nor was it expected by users.
+	//
 	bool doPageInfo (int headerType, int footerType)
 	{
 		kdDebug (30509) << "doPageInfo (headerType=" << headerType
@@ -1063,6 +1067,11 @@
 
 						// convert twip measurements that aren't in 72dpi
 						int defaultDpi = wmf.defaultDpi ();
+						if (defaultDpi <= 0)
+						{
+							kdWarning (30509) << "Invalid defaultDPI: " << defaultDpi << endl;
+							defaultDpi = 1440;
+						}
 						imageActualWidth = width * 1440 / defaultDpi;
 						imageActualHeight = height * 1440 / defaultDpi;
 
@@ -1247,7 +1256,7 @@
 
 		MSWrite::FormatParaProperty paraProp; paraProp = *paraPropIn;
 		paraProp.setIsObject (true);
-		paraProp.setLeftIndent (0);	// not neccessary but...
+		paraProp.setLeftIndent (0);	// not necessary but...
 		if (!m_generator->writeParaInfoBegin (&paraProp, NULL, &image))
 			return false;
 
@@ -1897,7 +1906,7 @@
 
 KoFilter::ConversionStatus MSWriteExport::convert (const QCString &from, const QCString &to)
 {
-	kdDebug (30509) << "MSWriteExport $Date$ using LibMSWrite "
+	kdDebug (30509) << "MSWriteExport $Date$ using LibMSWrite "
 			  				<< MSWrite::Version << endl;
 
 	if (to != "application/x-mswrite" || from != "application/x-kword")
--- filters/kword/mswrite/mswriteimport.cc	5 Jul 2003 18:31:19 -0000	1.1.1.5
+++ filters/kword/mswrite/mswriteimport.cc	9 Jul 2003 14:40:26 -0000	1.2
@@ -1,5 +1,3 @@
-/* $Id$ */
-
 /* This file is part of the KDE project
    Copyright (C) 2001-2003 Clarence Dang <dang@kde.org>
 
@@ -1093,7 +1091,7 @@
 
 		m_formatOutput += "<FONT name=\"";
 			m_formatOutput += (const char *) (charProperty->getFont ()->getName ());
-			m_formatOutput += "\"/>";;
+			m_formatOutput += "\"/>";
 			m_formatOutput += "<SIZE value=\"";
 			m_formatOutput += QString::number (charProperty->getFontSize ());
 			m_formatOutput += "\"/>";
@@ -1348,7 +1346,7 @@
 
 KoFilter::ConversionStatus MSWriteImport::convert (const QCString &from, const QCString &to)
 {
-	kdDebug (30509) << "MSWriteImport $Date$ using LibMSWrite "
+	kdDebug (30509) << "MSWriteImport $Date$ using LibMSWrite "
 			  				<< MSWrite::Version << endl;
 
 	if (to != "application/x-kword" || from != "application/x-mswrite")
--- filters/kword/mswrite/structures.cpp	5 Jul 2003 18:31:20 -0000	1.1.1.2
+++ filters/kword/mswrite/structures.cpp	9 Jul 2003 14:40:26 -0000	1.2
@@ -23,7 +23,7 @@
  * structures.cpp - Process Internal Microsoft(r) Write file structures
  * This file has ugly reading/writing code that couldn't be generated.
  * It performs some higher-level sanity checks.
- * 
+ *
  */
 
 #include "structures.h"
@@ -39,6 +39,19 @@
 	{
 	}
 
+	PageLayout &PageLayout::operator= (const PageLayout &rhs)
+	{
+		if (this == &rhs)
+			return *this;
+
+		PageLayoutGenerated::operator= (rhs);
+		NeedsHeader::operator= (rhs);
+
+		m_numModified = rhs.m_numModified;
+
+		return *this;
+	}
+
 	bool PageLayout::readFromDevice (void)
 	{
 	CHECK_DEVICE;
@@ -69,7 +82,7 @@
 	#ifdef DEBUG_PAGELAYOUT
 		Dump (magic102);
 		Dump (magic512);
-		
+
 		Dump (pageHeight);
 		Dump (pageWidth);
 		Dump (pageNumberStart);
@@ -79,7 +92,7 @@
 		Dump (textWidth);
 
 		Dump (magic256);
-		
+
 		Dump (headerFromTop);
 		Dump (footerFromTop);
 
@@ -109,7 +122,7 @@
 		//UpdateModifiedCount (unknown);	// no reliable default for unknown
 		UpdateModifiedCount (zero2);
 		#undef UpdateModifiedCount
-		
+
 		return true;
 	}
 
@@ -145,6 +158,18 @@
 		delete [] m_name;
 	}
 
+	Font &Font::operator= (const Font &rhs)
+	{
+		if (this == &rhs)
+			return *this;
+
+		FontGenerated::operator= (rhs);
+
+		setName (rhs.getName ());
+
+		return *this;
+	}
+
 	bool Font::readFromDevice (void)
 	{
 	CHECK_DEVICE;
@@ -228,11 +253,38 @@
 	FormatCharProperty::~FormatCharProperty ()
 	{
 	}
-	
+
+	FormatCharProperty &FormatCharProperty::operator= (const FormatCharProperty &rhs)
+	{
+		if (this == &rhs)
+			return *this;
+
+		FormatCharPropertyGenerated::operator= (rhs);
+
+		m_afterEndCharByte = rhs.m_afterEndCharByte;
+		m_fontTable = rhs.m_fontTable;
+		m_font = rhs.m_font;
+
+		return *this;
+	}
+
+	bool FormatCharProperty::operator== (FormatCharProperty &rhs)
+	{
+		DWord numDataBytes;
+
+		if ((numDataBytes = getNumDataBytes ()) != rhs.getNumDataBytes ())
+			return false;
+
+		writeToArray ();
+		rhs.writeToArray ();
+
+		return memcmp (m_data + sizeof (m_numDataBytes), rhs.m_data + sizeof (m_numDataBytes), numDataBytes) == 0;
+	}
+
 	bool FormatCharProperty::readFromDevice (void)
 	{
 	CHECK_DEVICE;
-	
+
 	#ifdef DEBUG_CHAR
 		m_device->debug ("\nFormatCharProperty::readFromDevice\n");
 	#endif
@@ -259,7 +311,7 @@
 
 		return FormatCharPropertyGenerated::writeToDevice ();
 	}
-	
+
 	bool FormatCharProperty::updateFont (void)
 	{
 		Font *fontPtr = m_fontTable->getFont (getFontCode ());
@@ -291,6 +343,16 @@
 	{
 	}
 
+	FormatParaPropertyTabulator &FormatParaPropertyTabulator::operator= (const FormatParaPropertyTabulator &rhs)
+	{
+		if (this == &rhs)
+			return *this;
+
+		FormatParaPropertyTabulatorGenerated::operator= (rhs);
+
+		return *this;
+	}
+
 	bool FormatParaPropertyTabulator::readFromDevice (void)
 	{
 	CHECK_DEVICE;
@@ -328,13 +390,39 @@
 																m_numTabulators (0),
 																m_addedTooManyTabs (false)
 	{
-		// m_afterEndCharByte = 
+		// m_afterEndCharByte =
 	}
 
 	FormatParaProperty::~FormatParaProperty ()
 	{
 	}
 
+	FormatParaProperty &FormatParaProperty::operator= (const FormatParaProperty &rhs)
+	{
+		if (this == &rhs)
+			return *this;
+
+		FormatParaPropertyGenerated::operator= (rhs);
+
+		m_afterEndCharByte = rhs.m_afterEndCharByte;
+		m_leftMargin = rhs.m_leftMargin, m_rightMargin = rhs.m_rightMargin;
+		m_numTabulators = rhs.m_numTabulators;
+		m_addedTooManyTabs = rhs.m_addedTooManyTabs;
+
+		return *this;
+	}
+
+	bool FormatParaProperty::operator== (FormatParaProperty &rhs)
+	{
+		DWord numDataBytes;
+		if ((numDataBytes = getNumDataBytes ()) != rhs.getNumDataBytes ())
+			return false;
+
+		writeToArray ();
+		rhs.writeToArray ();
+
+		return memcmp (m_data + sizeof (m_numDataBytes), rhs.m_data + sizeof (m_numDataBytes), numDataBytes) == 0;
+	}
 
 	bool FormatParaProperty::readFromDevice (void)
 	{
@@ -343,7 +431,7 @@
 	#ifdef DEBUG_PARA
 		m_device->debug ("\nFormatParaProperty::readFromDevice\n");
 	#endif
-		
+
 	#ifdef CHECK_INTERNAL
 		if (m_leftMargin == 0xFFFF && m_rightMargin == 0xFFFF)
 			ErrorAndQuit (Error::InternalError, "FormatParaProperty::readFromDevice call not setup\n");
@@ -405,9 +493,9 @@
 
 		if (m_numTabulators > 12)
 			m_device->error (Error::Warn, "should not have more than 12 tabulators since you can only access 12 tabs via the GUI\n");
-		
+
 		// --- indents adjusted in updateIndents() called by FormatInfoPage::add() ---
-		
+
 		if (!FormatParaPropertyGenerated::writeToDevice ())
 			return false;
 
@@ -428,6 +516,33 @@
 		delete [] m_externalImage;
 	}
 
+	Image &Image::operator= (const Image &rhs)
+	{
+		if (this == &rhs)
+			return *this;
+
+		ImageGenerated::operator= (rhs);
+
+		m_externalImageSize = rhs.m_externalImageSize;
+		m_externalImageUpto = rhs.m_externalImageUpto;
+
+		delete [] m_externalImage;
+		m_externalImage = new Byte [m_externalImageSize];
+		if (!m_externalImage)
+			return *this;  // TODO: error check
+
+		if (rhs.m_externalImage)
+			memcpy (m_externalImage, rhs.m_externalImage, m_externalImageUpto);
+
+		m_originalWidth = rhs.m_originalWidth;
+		m_originalHeight = rhs.m_originalHeight;
+
+		m_displayedWidth = rhs.m_displayedWidth;
+		m_displayedHeight = rhs.m_displayedHeight;
+
+		return *this;
+	}
+
 	// returns how many bytes are needed for each scanline (with byte padding)
 	int Image::getBytesPerScanLine (const int width, const int bitsPerPixel, const int padBytes)
 	{
@@ -539,10 +654,10 @@
 				m_device->error (Error::Warn, "m_MFP_height != m_bmh->getHeight() * 2.64\n");
 
 #undef MSWrite_fabs
-				
+
 			if (m_width)
 				m_device->error (Error::Warn, "m_width should not be set for BMPs\n");
-			
+
 			if (m_height)
 				m_device->error (Error::Warn, "m_height should not be set for BMPs\n");
 
@@ -550,13 +665,13 @@
 			//
 			// read image
 			//
-			
+
 			Byte *internalData = new Byte [getNumDataBytes ()];
 			if (!internalData)
 				ErrorAndQuit (Error::OutOfMemory, "could not allocate memory for internal BMP image\n");
 			if (!m_device->readInternal (internalData, getNumDataBytes ()))
 				ErrorAndQuit (Error::FileError, "could not read internal BMP\n");
-		
+
 			// infoHeader
 			BMP_BitmapInfoHeader infoHeader;
 			infoHeader.setWidth (m_bmh->getWidth ());
@@ -713,8 +828,8 @@
 
 			// not used by WMFs
 			m_horizontalScalingRel1000 = m_verticalScalingRel1000 = 1000;
-			
-			
+
+
 			// write header
 			setNumDataBytes (m_externalImageSize);
 			if (!ImageGenerated::writeToDevice ())
@@ -748,7 +863,7 @@
 			BMP_BitmapInfoHeader infoHeader;
 			infoHeader.setDevice (m_device);
 			if (!infoHeader.readFromDevice ()) return false;
-			
+
 			// write out each scanline
 			// to .WRI (padded to 2) vs input BMP (padded to 4
 			Word scanLineWRILength = getBytesPerScanLine (infoHeader.getWidth (), infoHeader.getBitsPerPixel (), 2);
@@ -758,7 +873,7 @@
 				ErrorAndQuit (Error::Warn, "infoHeader width != m_originalWidth\n");
 			if (infoHeader.getHeight () != Word (Twip2Point (m_originalHeight)))
 				ErrorAndQuit (Error::Warn, "infoHeader.height != m_originalHeight\n");
-				
+
 			m_bmh->setWidth (infoHeader.getWidth ());
 			m_bmh->setHeight (infoHeader.getHeight ());
 			m_bmh->setWidthBytes (scanLineWRILength);
@@ -792,20 +907,20 @@
 			// finish reading from m_externalImage
 			m_device->setCache (NULL);
 
-		
+
 			//
 			// set image dimensions
 			//
-			
+
 			m_MFP_width = Word (Twip2Point (m_originalWidth) * 2.64);
 			m_MFP_height = Word (Twip2Point (m_originalHeight) * 2.64);
-			
+
 			// BMPs don't use
 			m_width = 0, m_height = 0;
-			
+
 			m_horizontalScalingRel1000 = Word (m_displayedWidth * 1.38889 * 1000.0 / m_originalWidth);
 			m_verticalScalingRel1000 = Word (m_displayedHeight * 1.38889 * 1000.0 / m_originalHeight);
-			
+
 
 			// write header
 			setNumDataBytes (infoHeader.getHeight () * scanLineBMPLength);
@@ -841,6 +956,27 @@
 		delete [] m_externalObject;
 	}
 
+	OLE &OLE::operator= (const OLE &rhs)
+	{
+		if (this == &rhs)
+			return *this;
+
+		OLEGenerated::operator= (rhs);
+
+		m_externalObjectSize = rhs.m_externalObjectSize;
+		m_externalObjectUpto = rhs.m_externalObjectUpto;
+
+		delete [] m_externalObject;
+		m_externalObject = new Byte [m_externalObjectSize];
+		if (!m_externalObject)
+			return *this;  // TODO: error check
+
+		if (rhs.m_externalObject)
+			memcpy (m_externalObject, rhs.m_externalObject, m_externalObjectUpto);
+
+		return *this;
+	}
+
 	bool OLE::readFromDevice (void)
 	{
 	CHECK_DEVICE;
@@ -900,7 +1036,7 @@
 	#ifdef DEBUG_OBJECT
 		m_device->debug ("\n>>>> OLE::writeToDevice <<<<\n");
 	#endif
-		
+
 	#ifdef DEBUG_OBJECT
 		Dump (zero);
 
@@ -934,11 +1070,11 @@
 		// write header
 		if (!OLEGenerated::writeToDevice ())
 			return false;
-		
+
 		// write data
 		if (!m_device->writeInternal (m_externalObject, m_externalObjectSize))
 			return false;
-		
+
 		return true;
 	}
 
--- filters/kword/mswrite/structures.h	5 Jul 2003 18:31:20 -0000	1.1.1.2
+++ filters/kword/mswrite/structures.h	9 Jul 2003 14:40:26 -0000	1.2
@@ -65,7 +65,9 @@
 
 	public:
 		PageLayout ();
-		~PageLayout ();
+		virtual ~PageLayout ();
+
+		PageLayout &operator= (const PageLayout &rhs);
 
 		bool getIsModified (void) const	{	return m_numModified > 0;	}
 
@@ -161,19 +163,9 @@
 		};
 
 		Font (const Byte *name = NULL, const Byte family = DontCare);
-		~Font ();
-
-		Font &operator= (const Font &rhs)
-		{
-			if (this == &rhs)
-				return *this;
+		virtual ~Font ();
 
-			FontGenerated::operator= (rhs);
-
-			setName (rhs.getName ());
-
-			return *this;
-		}
+		Font &operator= (const Font &rhs);
 
 		Byte *getName (void) const	{	return m_name;	}
 		Byte *setName (const Byte *s)
@@ -258,21 +250,9 @@
 
 	public:
 		FormatCharProperty ();
-		~FormatCharProperty ();
+		virtual ~FormatCharProperty ();
 
-		FormatCharProperty &operator= (const FormatCharProperty &rhs)
-		{
-			if (this == &rhs)
-				return *this;
-
-			FormatCharPropertyGenerated::operator= (rhs);
-
-			m_afterEndCharByte = rhs.m_afterEndCharByte;
-			m_fontTable = rhs.m_fontTable;
-			m_font = rhs.m_font;
-
-			return *this;
-		}
+		FormatCharProperty &operator= (const FormatCharProperty &rhs);
 
 		const Font *getFont (void) const
 		{
@@ -310,18 +290,7 @@
 		void setIsSubscript (void)	{	setPosition (252);	/* common value */	}
 
 		// compares if the contents are the same (not if it points to the same place etc. etc.)
-		bool operator== (FormatCharProperty &rhs)
-		{
-			DWord numDataBytes;
-			
-			if ((numDataBytes = getNumDataBytes ()) != rhs.getNumDataBytes ())
-				return false;
-			
-			writeToArray ();
-			rhs.writeToArray ();
-			
-			return memcmp (m_data + sizeof (m_numDataBytes), rhs.m_data + sizeof (m_numDataBytes), numDataBytes) == 0;
-		}
+		bool operator== (FormatCharProperty &rhs);
 	};
 
 	
@@ -335,8 +304,10 @@
 
 	public:
 		FormatParaPropertyTabulator ();
-		~FormatParaPropertyTabulator ();
+		virtual ~FormatParaPropertyTabulator ();
 
+		FormatParaPropertyTabulator &operator= (const FormatParaPropertyTabulator &rhs);
+		
 		// you can use getType()/setType() if you really want to...
 		bool getIsNormal (void) const	{	return m_type == 0;	}
 		void setIsNormal (const bool yes = true)	{	m_type = (yes ? 0 : 3);	}
@@ -401,21 +372,9 @@
 
 	public:
 		FormatParaProperty ();
-		~FormatParaProperty ();
+		virtual ~FormatParaProperty ();
 
-		FormatParaProperty &operator= (const FormatParaProperty &rhs)
-		{
-			if (this == &rhs)
-				return *this;
-
-			FormatParaPropertyGenerated::operator= (rhs);
-			
-			m_afterEndCharByte = rhs.m_afterEndCharByte;
-			m_leftMargin = rhs.m_leftMargin, m_rightMargin = rhs.m_rightMargin;
-			m_numTabulators = rhs.m_numTabulators;
-			
-			return *this;
-		}
+		FormatParaProperty &operator= (const FormatParaProperty &rhs);
 		
 		// convenience functions
 		Byte getAlign (void) const	{	return getAlignment ();	}
@@ -507,17 +466,7 @@
 		}
 
 		// compares if the contents are the same (not if it points to the same place etc. etc.)
-		bool operator== (FormatParaProperty &rhs)
-		{
-			DWord numDataBytes;
-			if ((numDataBytes = getNumDataBytes ()) != rhs.getNumDataBytes ())
-				return false;
-			
-			writeToArray ();
-			rhs.writeToArray ();
-			
-			return memcmp (m_data + sizeof (m_numDataBytes), rhs.m_data + sizeof (m_numDataBytes), numDataBytes) == 0;
-		}
+		bool operator== (FormatParaProperty &rhs);
 	};
 	
 
@@ -548,34 +497,9 @@
 		bool writeToDevice (void);	friend class InternalParser;
 	public:
 		Image ();
-		~Image ();
-
-		Image &operator= (const Image &rhs)
-		{
-			if (this == &rhs)
-				return *this;
-
-			ImageGenerated::operator= (rhs);
-
-			m_externalImageSize = rhs.m_externalImageSize;
-			m_externalImageUpto = rhs.m_externalImageUpto;
-
-			delete [] m_externalImage;
-			m_externalImage = new Byte [m_externalImageSize];
-			if (!m_externalImage)
-				return *this;	// TODO: error check
-
-			if (rhs.m_externalImage)
-				memcpy (m_externalImage, rhs.m_externalImage, m_externalImageUpto);
+		virtual ~Image ();
 
-			m_originalWidth = rhs.m_originalWidth;
-			m_originalHeight = rhs.m_originalHeight;
-
-			m_displayedWidth = rhs.m_displayedWidth;
-			m_displayedHeight = rhs.m_displayedHeight;
-			
-			return *this;
-		}
+		Image &operator= (const Image &rhs);
 		
 		bool getIsWMF (void) const	{	return m_mappingMode != 0xE3;	}
 		bool getIsBMP (void) const	{	return m_mappingMode == 0xE3;	}
@@ -658,28 +582,9 @@
 
 	public:
 		OLE ();
-		~OLE ();
-
-		OLE &operator= (const OLE &rhs)
-		{
-			if (this == &rhs)
-				return *this;
+		virtual ~OLE ();
 
-			OLEGenerated::operator= (rhs);
-
-			m_externalObjectSize = rhs.m_externalObjectSize;
-			m_externalObjectUpto = rhs.m_externalObjectUpto;
-
-			delete [] m_externalObject;
-			m_externalObject = new Byte [m_externalObjectSize];
-			if (!m_externalObject)
-				return *this;	// TODO: error check
-				
-			if (rhs.m_externalObject)
-				memcpy (m_externalObject, rhs.m_externalObject, m_externalObjectUpto);
-
-			return *this;
-		}
+		OLE &operator= (const OLE &rhs);
 
 		Byte *getExternalObject (void) const	{	return m_externalObject;	}
 		DWord getExternalObjectSize (void) const	{	return m_externalObjectSize;	}
--- filters/kword/mswrite/structures_generated.cpp	5 Jul 2003 18:31:20 -0000	1.1.1.2
+++ filters/kword/mswrite/structures_generated.cpp	9 Jul 2003 14:40:27 -0000	1.2
@@ -21,7 +21,7 @@
 
 /*
  * structures_generated - Internal MS Write file structures
- * This file was generated by make_structures_generated 0.51pre 2003/06/08.
+ * This file was generated by make_structures_generated 0.51pre 2003/07/09.
  * ALL CHANGES MADE TO THIS FILE "WILL BE LOST IN TIME LIKE TEARS IN RAIN"!
  */
 
@@ -30,8 +30,8 @@
 #include <string.h>
 
 #include "structures_generated.h"
-#include "structures_private.h"	// for foward references to embedded structures
-#include "structures.h"	// for foward references to embedded structures
+#include "structures_private.h"	// for forward references to embedded structures
+#include "structures.h"	// for forward references to embedded structures
 
 namespace MSWrite
 {
@@ -431,7 +431,7 @@
 		// --- allocate memory for embedded structures ---
 		for (int i = 0; i < 2; i++)
 		{
-			m_sed [i] = new SectionDescriptor;	// foward ref to structures*.cpp
+			m_sed [i] = new SectionDescriptor;	// forward ref to structures*.cpp
 			if (!m_sed [i]) return;
 		}
 		
@@ -1160,7 +1160,7 @@
 		// --- allocate memory for embedded structures ---
 		for (int i = 0; i < 14; i++)
 		{
-			m_tab [i] = new FormatParaPropertyTabulator;	// foward ref to structures*.cpp
+			m_tab [i] = new FormatParaPropertyTabulator;	// forward ref to structures*.cpp
 			if (!m_tab [i]) return;
 		}
 		
@@ -1968,7 +1968,7 @@
 		m_bmh = NULL;
 		
 		// --- allocate memory for embedded structures ---
-		m_bmh = new BitmapHeader;	// foward ref to structures*.cpp
+		m_bmh = new BitmapHeader;	// forward ref to structures*.cpp
 		if (!m_bmh) return;
 		
 		// --- set defaults for variables --
--- filters/kword/mswrite/structures_generated.h	5 Jul 2003 18:31:20 -0000	1.1.1.2
+++ filters/kword/mswrite/structures_generated.h	9 Jul 2003 14:40:27 -0000	1.5
@@ -21,7 +21,7 @@
 
 /*
  * structures_generated - Internal MS Write file structures
- * This file was generated by make_structures_generated 0.51pre 2003/06/08.
+ * This file was generated by make_structures_generated 0.51pre 2003/07/09.
  * ALL CHANGES MADE TO THIS FILE "WILL BE LOST IN TIME LIKE TEARS IN RAIN"!
  */
 
@@ -32,7 +32,7 @@
 
 namespace MSWrite
 {
-	// $Id$
+	// $Id$
 	// 
 	//    This file is part of the LibMSWrite Library
 	//    Copyright (C) 2001-2003 Clarence Dang <clarencedang@users.sourceforge.net>
--- filters/kword/mswrite/structures_private.cpp	5 Jul 2003 18:31:20 -0000	1.1.1.2
+++ filters/kword/mswrite/structures_private.cpp	9 Jul 2003 14:40:27 -0000	1.2
@@ -24,7 +24,7 @@
  *                        - that are hidden/abstracted away from the user
  * This file has ugly reading/writing code that couldn't be generated.
  * It performs some higher-level sanity checks.
- * 
+ *
  * Don't use any of these classes as it would defeat the purpose of LibMSWrite.
  */
 
@@ -33,6 +33,27 @@
 
 namespace MSWrite
 {
+	Header::Header ()
+	{
+	}
+
+	Header::~Header ()
+	{
+	}
+
+	Header &Header::operator= (const Header &rhs)
+	{
+		if (this == &rhs)
+			return *this;
+
+		HeaderGenerated::operator= (rhs);
+
+		m_numCharBytes = rhs.m_numCharBytes;
+		m_pageCharInfo = rhs.m_pageCharInfo;
+
+		return *this;
+	}
+
 	bool Header::readFromDevice (void)
 	{
 	CHECK_DEVICE;
@@ -125,6 +146,44 @@
 	}
 
 
+	SectionDescriptor::SectionDescriptor ()
+	{
+	}
+
+	SectionDescriptor::~SectionDescriptor ()
+	{
+	}
+
+	SectionDescriptor &SectionDescriptor::operator= (const SectionDescriptor &rhs)
+	{
+		if (this == &rhs)
+			return *this;
+
+		SectionDescriptorGenerated::operator= (rhs);
+
+		return *this;
+	}
+
+
+	SectionTable::SectionTable ()
+	{
+	}
+
+	SectionTable::~SectionTable ()
+	{
+	}
+
+	SectionTable &SectionTable::operator= (const SectionTable &rhs)
+	{
+		if (this == &rhs)
+			return *this;
+
+		SectionTableGenerated::operator= (rhs);
+		NeedsHeader::operator= (rhs);
+
+		return *this;
+	}
+
 	// this function really only does some sanity checking
 	// in fact, a SectionTable is quite useless in Write...
 	// ...so you actually don't have to call it :)
@@ -319,7 +378,7 @@
 
 					// let's not cause problems later with uninitialised Font::m_name's
 					m_fontList.erase (it);
-								 
+
 					// better quit just in case (it is better to have fewer fonts than risk errors)
 					break;
 				}
@@ -410,13 +469,13 @@
 	DWord FontTable::findFont (const Font *want) const
 	{
 		DWord code = 0;
-		
+
 		List <Font>::Iterator it;
 		for (it = m_fontList.begin (); it != m_fontList.end (); it++)
 		{
 			if ((*it) == *want)
 				return code;
-			
+
 			code++;
 		}
 
@@ -440,7 +499,7 @@
 		#ifdef DEBUG_FONT
 			m_device->debug ("\t\t\t\tAlready in list, fontCode=", e);
 		#endif
-			
+
 			// return fontCode
 			return e;
 		}
@@ -450,7 +509,7 @@
 		#ifdef DEBUG_FONT
 			m_device->debug ("\t\t\t\tNew font, adding to list\n");
 		#endif
-			
+
 			if (!m_fontList.addToBack (*input))
 				ErrorAndQuit (Error::OutOfMemory, "could not allocate memory for next font element\n");
 			return m_fontList.getNumElements () - 1;
@@ -458,6 +517,24 @@
 	}
 
 
+	PagePointer::PagePointer ()
+	{
+	}
+
+	PagePointer::~PagePointer ()
+	{
+	}
+
+	PagePointer &PagePointer::operator= (const PagePointer &rhs)
+	{
+		if (this == &rhs)
+			return *this;
+
+		PagePointerGenerated::operator= (rhs);
+
+		return *this;
+	}
+
 	bool PagePointer::readFromDevice (void)
 	{
 	CHECK_DEVICE;
@@ -492,6 +569,22 @@
 	{
 	}
 
+	PageTable &PageTable::operator= (const PageTable &rhs)
+	{
+		if (this == &rhs)
+			return *this;
+
+		PageTableGenerated::operator= (rhs);
+		NeedsHeader::operator= (rhs);
+
+		m_pagePointerList = rhs.m_pagePointerList;
+		m_pageNumberStart = rhs.m_pageNumberStart;
+
+		m_pageTableIterator = rhs.m_pageTableIterator;
+
+		return *this;
+	}
+
 	bool PageTable::readFromDevice (void)
 	{
 	CHECK_DEVICE;
@@ -556,7 +649,7 @@
 					m_device->error (Error::Warn, "pages don't follow each other\n");
 
 				if (pp.getFirstCharByte () <= lastCharByte)
-					ErrorAndQuit (Error::InvalidFormat, "pageTable is not going foward?\n");
+					ErrorAndQuit (Error::InvalidFormat, "pageTable is not going forward?\n");
 			}
 
 			// for checking, on next iteration
@@ -615,7 +708,20 @@
 	FormatPointer::~FormatPointer ()
 	{
 	}
-	
+
+	FormatPointer &FormatPointer::operator= (const FormatPointer &rhs)
+	{
+		if (this == &rhs)
+			return *this;
+
+		FormatPointerGenerated::operator= (rhs);
+
+		m_afterEndCharByte = rhs.m_afterEndCharByte;
+		m_formatProperty = rhs.m_formatProperty;
+
+		return *this;
+	}
+
 	bool FormatPointer::readFromDevice (void)
 	{
 	CHECK_DEVICE;
@@ -636,7 +742,7 @@
 	bool FormatPointer::writeToDevice (void)
 	{
 	CHECK_DEVICE;
-	
+
 	#ifdef DEBUG_FORMATINFO
 		m_device->debug ("\t\t\t\tafterEndCharByte: ", m_afterEndCharByte);
 		m_device->debug ("\t\t\t\tformatPropertyOffset: ", m_formatPropertyOffset);
@@ -646,7 +752,7 @@
 
 		if (!FormatPointerGenerated::writeToDevice ())
 			return false;
-			
+
 		return true;
 	}
 
@@ -684,7 +790,7 @@
 		else	// if (m_type == CharType)
 			m_device->debug ("FormatInfoPage::readFromDevice (CharType)\n");
 	#endif
-		
+
 		if (!FormatInfoPageGenerated::readFromDevice ())
 			return false;
 
@@ -694,7 +800,7 @@
 		m_device->debug ("number of FormatPointers on this page: ", FormatInfoPageGenerated::m_numFormatPointers);
 		m_device->debug ("byte number of first character covered by this page: ", m_firstCharByte);
 	#endif
-	
+
 		return true;
 	}
 
@@ -708,7 +814,7 @@
 		else	// if (m_type == CharType)
 			m_device->debug ("FormatInfoPage::writeToDevice (CharType)\n");
 	#endif
-		
+
 		FormatInfoPageGenerated::m_firstCharBytePlus128 = m_firstCharByte + 128;
 
 	#ifdef DEBUG_FORMATINFO
@@ -725,16 +831,16 @@
 	bool FormatInfoPage::writeToArray (void)
 	{
 	CHECK_DEVICE;
-	
+
 	#ifdef DEBUG_FORMATINFO
 		if (m_type == ParaType)
 			m_device->debug ("\tFormatInfoPage::writeToArray (ParaType)\n");
 		else	// if (m_type == CharType)
 			m_device->debug ("\tFormatInfoPage::writeToArray (CharType)\n");
 	#endif
-		
+
 		MemoryDevice memdev;
-		
+
 		m_formatPointerPos = 0;
 		m_formatPropertyPos = 123;
 
@@ -763,7 +869,7 @@
 			if (m_type == ParaType)
 			{
 				FormatParaProperty *prop = &m_formatParaProperty [i];
-					
+
 				m_formatPropertyPos -= (sizeof (Byte) + prop->getNumDataBytes ());
 				memdev.setCache (m_packedStructs + m_formatPropertyPos);
 					prop->setDevice (&memdev);	// device was set to m_device to keep operator== happy in FormatInfoPage::add
@@ -784,10 +890,10 @@
 				m_device->debug ("\t\t\tWrote at Offset=", m_formatPropertyPos);
 			#endif
 		}
-		
+
 		if (!FormatInfoPageGenerated::writeToArray ())
 			return false;
-		
+
 		return true;
 	}
 
@@ -834,7 +940,7 @@
 			m_device->debug ("\tChar FormatPointer #", m_formatPointerUpto);
 	#endif
 
-		
+
 		//
 		// read Format Pointer
 		//
@@ -854,7 +960,7 @@
 
 		// check that it continues on from the last
 		if (formatPointerAfterEndCharByte <= m_nextCharByte)
-			m_device->error (Error::Warn, "FormatPointer afterEndCharByte does not go foward\n");
+			m_device->error (Error::Warn, "FormatPointer afterEndCharByte does not go forward\n");
 		m_nextCharByte = formatPointerAfterEndCharByte;
 
 		// check whether it is the last one (EOF-wise)
@@ -880,7 +986,7 @@
 			}
 		}
 
-		
+
 		//
 		// read in corresponding Format Property
 		//
@@ -944,7 +1050,7 @@
 
 				m_formatParaProperty->setDevice (m_device);
 				m_formatParaProperty->setMargins (m_leftMargin, m_rightMargin);
-				
+
 				// corresponding paraProperty structure exists?
 				if (currentPropertyOffset != 0xFFFF)
 				{
@@ -983,7 +1089,7 @@
 		// 	This is also true for FormatProperties because m_numFormatPointer >= m_numProperty
 		// 	so MaxElements is also the upper limit on the number of properties
 		const int MaxElements = 123 / FormatPointer::s_size;
-			
+
 	#ifdef DEBUG_FORMATINFO
 		if (m_type == ParaType)
 			m_device->debug (">>>> FormatInfoPage::add (ParaType) <<<<\n");
@@ -1004,7 +1110,7 @@
 		#ifdef DEBUG_FORMATINFO
 			m_device->debug ("\tFIRST CALL!  Allocating memory for FormatPointer[] and FormatProperty[]\n");
 		#endif
-			
+
 			// allocate memory for FormatPointers
 			m_formatPointer = new FormatPointer [MaxElements + 1];	// lazy memory allocation (+1 in case we can merge the last one)
 			if (!m_formatPointer)
@@ -1032,7 +1138,7 @@
 
 			m_numProperty = 0;
 		}
-		
+
 
 		//
 		// Setup pointers
@@ -1040,13 +1146,13 @@
 
 		FormatParaProperty inpp;
 		FormatCharProperty incp;
-		
+
 		if (m_type == ParaType)
 		{
 		#ifdef DEBUG_FORMATINFO
 			m_device->debug ("\tTrying to add ParaProperty, #numDataBytes=", ((FormatParaProperty *) property)->getNumDataBytes ());
 		#endif
-			
+
 			inpp = *((FormatParaProperty *) property);
 			inpp.setDevice (m_device);
 			inpp.setAfterEndCharByte (m_device->tellInternal () - 128);
@@ -1058,7 +1164,7 @@
 		#ifdef DEBUG_FORMATINFO
 			m_device->debug ("\tTrying to add CharProperty, #numDataBytes=", ((FormatCharProperty *) property)->getNumDataBytes ());
 		#endif
-			
+
 			incp = *((FormatCharProperty *) property);
 			incp.setDevice (m_device);
 			incp.setAfterEndCharByte (m_device->tellInternal () - 128);
@@ -1066,15 +1172,15 @@
 			if (!incp.updateFontCode ()) return false;
 		}
 
-		
+
 		//
 		// Attempt to add Property to current page:
-		// 
+		//
 		// Since MS programmers tried so hard to save space, we will do the same :)
 		// This results in having to store variable-length structures and the use of other nasty
 		// tricks such as having multiple FormatPointers pointing to the same FormatProperty.
 		// Details follow in code :)
-		// 
+		//
 
 		// In Write, character formatting blocks can extend across paragraphs
 		// In LibMSWrite they don't (for convenience?) so here we merge FormatPointers that have been split up
@@ -1088,7 +1194,7 @@
 		#ifdef DEBUG_FORMATINFO
 			m_device->debug ("\tIs it the same property as last? ");
 		#endif
-			
+
 			// simply extend last FormatPointer
 			FormatCharProperty *lastProp = (FormatCharProperty *) m_formatPointer [m_numFormatPointers - 1].getFormatProperty ();
 			if (incp == *lastProp)
@@ -1097,9 +1203,9 @@
 				m_device->debug ("Yes, extending FormatPointer #", m_numFormatPointers - 1);
 				m_device->debug ("\t\tFrom ", m_formatPointer [m_numFormatPointers - 1].getAfterEndCharByte ());
 			#endif
-				
+
 				m_formatPointer [m_numFormatPointers - 1].setAfterEndCharByte (incp.getAfterEndCharByte ());
-				
+
 			#ifdef DEBUG_FORMATINFO
 				m_device->debug ("\t\tTo: ", incp.getAfterEndCharByte ());
 			#endif
@@ -1114,7 +1220,7 @@
 		#ifdef DEBUG_FORMATINFO
 			m_device->debug ("\tSame as earlier FormatPointer? ");
 		#endif
-		
+
 		// points to same property as an earlier formatPointer?
 		for (int i = 0; i < m_numFormatPointers - (checkedLastFormatPointer ? 1 : 0); i++)
 		{
@@ -1161,7 +1267,7 @@
 					#endif
 						return false;	// not enough room
 					}
-					
+
 				#ifdef DEBUG_FORMATINFO
 					m_device->debug ("Yes, same property as ", i);
 				#endif
@@ -1178,7 +1284,7 @@
 		#ifdef DEBUG_FORMATINFO
 			m_device->debug ("No, outputting full FormatPointer & Property\n");
 		#endif
-		
+
 		// ok we can't compress, must output a full FormatPointer & FormatProperty
 		if (m_type == ParaType)
 		{
@@ -1251,19 +1357,37 @@
 			return true;
 		}
 	}
-	
-	
+
+
+	BMP_BitmapFileHeader::BMP_BitmapFileHeader ()
+	{
+	}
+
+	BMP_BitmapFileHeader::~BMP_BitmapFileHeader ()
+	{
+	}
+
+	BMP_BitmapFileHeader &BMP_BitmapFileHeader::operator= (const BMP_BitmapFileHeader &rhs)
+	{
+		if (this == &rhs)
+			return *this;
+
+		BMP_BitmapFileHeaderGenerated::operator= (rhs);
+
+		return *this;
+	}
+
 	bool BMP_BitmapFileHeader::readFromDevice (void)
 	{
 	CHECK_DEVICE;
-	
+
 	#ifdef DEBUG_IMAGE
 		m_device->debug ("\n<<<< BMP_BitmapFileHeader::readFromDevice >>>>\n");
 	#endif
-	
+
 		if (!BMP_BitmapFileHeaderGenerated::readFromDevice ())
 			return false;
-		
+
 	#ifdef DEBUG_IMAGE
 		Dump (magic);
 		Dump (totalBytes);
@@ -1271,10 +1395,10 @@
 		Dump (zero [1]);
 		Dump (actualImageOffset);
 	#endif
-		
+
 		return true;
 	}
-	
+
 	bool BMP_BitmapFileHeader::writeToDevice (void)
 	{
 	CHECK_DEVICE;
@@ -1291,14 +1415,32 @@
 	}
 
 
+	BMP_BitmapInfoHeader::BMP_BitmapInfoHeader ()
+	{
+	}
+
+	BMP_BitmapInfoHeader::~BMP_BitmapInfoHeader ()
+	{
+	}
+
+	BMP_BitmapInfoHeader &BMP_BitmapInfoHeader::operator= (const BMP_BitmapInfoHeader &rhs)
+	{
+		if (this == &rhs)
+			return *this;
+
+		BMP_BitmapInfoHeaderGenerated::operator= (rhs);
+
+		return *this;
+	}
+
 	bool BMP_BitmapInfoHeader::readFromDevice (void)
 	{
 	CHECK_DEVICE;
-	
+
 	#ifdef DEBUG_IMAGE
 		m_device->debug ("\n<<<< BMP_BitmapInfoHeader::readFromDevice >>>>\n");
 	#endif
-	
+
 		if (!BMP_BitmapInfoHeaderGenerated::readFromDevice ())
 			return false;
 
@@ -1315,24 +1457,61 @@
 		Dump (coloursUsed);
 		Dump (coloursImportant);
 	#endif
-		
+
 		return true;
 	}
-	
+
 	bool BMP_BitmapInfoHeader::writeToDevice (void)
 	{
 	CHECK_DEVICE;
-	
+
 	#ifdef DEBUG_IMAGE
 		m_device->debug ("\n<<<< BMP_BitmapInfoHeader::writeToDevice >>>>\n");
 	#endif
 
 		if (!BMP_BitmapInfoHeaderGenerated::writeToDevice ())
 			return false;
-			
+
 		return true;
 	}
-	
+
+
+	BMP_BitmapColourIndex::BMP_BitmapColourIndex ()
+	{
+	}
+
+	BMP_BitmapColourIndex::~BMP_BitmapColourIndex ()
+	{
+	}
+
+	BMP_BitmapColourIndex &BMP_BitmapColourIndex::operator= (const BMP_BitmapColourIndex &rhs)
+	{
+		if (this == &rhs)
+			return *this;
+
+		BMP_BitmapColourIndexGenerated::operator= (rhs);
+
+		return *this;
+	}
+
+
+	BitmapHeader::BitmapHeader ()
+	{
+	}
+
+	BitmapHeader::~BitmapHeader ()
+	{
+	}
+
+	BitmapHeader &BitmapHeader::operator= (const BitmapHeader &rhs)
+	{
+		if (this == &rhs)
+			return *this;
+
+		BitmapHeaderGenerated::operator= (rhs);
+
+		return *this;
+	}
 
 	bool BitmapHeader::readFromDevice (void)
 	{
@@ -1354,7 +1533,7 @@
 		Dump (bitsPerPixel);
 		Dump (zero2);
 	#endif
-		
+
 		return true;
 	}
 
@@ -1373,6 +1552,24 @@
 	}
 
 
+	WMFHeader::WMFHeader ()
+	{
+	}
+
+	WMFHeader::~WMFHeader ()
+	{
+	}
+
+	WMFHeader &WMFHeader::operator= (const WMFHeader &rhs)
+	{
+		if (this == &rhs)
+			return *this;
+
+		WMFHeaderGenerated::operator= (rhs);
+
+		return *this;
+	}
+
 	bool WMFHeader::readFromDevice (void)
 	{
 	CHECK_DEVICE;
@@ -1380,7 +1577,7 @@
 	#ifdef DEBUG_IMAGE
 		m_device->debug ("\n<<<< WMFHeader::readFromDevice >>>>\n");
 	#endif
-		
+
 		if (!WMFHeaderGenerated::readFromDevice ())
 			return false;
 
@@ -1393,7 +1590,7 @@
 		Dump (maxRecordSize);
 		Dump (zero);
 	#endif
-				
+
 		return true;
 	}
 
@@ -1404,7 +1601,7 @@
 	#ifdef DEBUG_IMAGE
 		m_device->debug ("\n>>>> WMFHeader::writeToDevice <<<<\n");
 	#endif
-		
+
 	#ifdef DEBUG_IMAGE
 		Dump (fieldType);
 		Dump (headerSize);
--- filters/kword/mswrite/structures_private.h	5 Jul 2003 18:31:20 -0000	1.1.1.2
+++ filters/kword/mswrite/structures_private.h	9 Jul 2003 14:40:28 -0000	1.2
@@ -37,18 +37,10 @@
 		bool writeToDevice (void);	friend class InternalGenerator;
 
 	public:
-		Header &operator= (const Header &rhs)
-		{
-			if (this == &rhs)
-				return *this;
+		Header ();
+		virtual ~Header ();
 
-			HeaderGenerated::operator= (rhs);
-
-			m_numCharBytes = rhs.m_numCharBytes;
-			m_pageCharInfo = rhs.m_pageCharInfo;
-
-			return *this;
-		}
+		Header &operator= (const Header &rhs);
 
 		Word getFormat (void) const	{	return getMagic ();	}
 		void setFormat (const Word f)	{	setMagic (f);	}
@@ -91,18 +83,29 @@
 
 			return *this;
 		}
-		
+
 		void setHeader (Header *header)	{	m_header = header;	}
 	};
 
 
 	class SectionDescriptor : public SectionDescriptorGenerated
 	{
+	public:
+		SectionDescriptor ();
+		virtual ~SectionDescriptor ();
+
+		SectionDescriptor &operator= (const SectionDescriptor &rhs);
 	};
 
 
 	class SectionTable : public SectionTableGenerated, public NeedsHeader
 	{
+	public:
+		SectionTable ();
+		virtual ~SectionTable ();
+
+		SectionTable &operator= (const SectionTable &rhs);
+
 	private:
 		bool readFromDevice (void);	friend class InternalParser;
 		bool writeToDevice (const bool needed);	friend class InternalGenerator;
@@ -120,7 +123,7 @@
 
 	public:
 		FontTable ();
-		~FontTable ();
+		virtual ~FontTable ();
 
 		FontTable &operator= (const FontTable &rhs);
 
@@ -129,16 +132,22 @@
 		DWord addFont (const Font *input);
 	};
 
-	
+
 	class PagePointer : public PagePointerGenerated
 	{
 	private:
 		friend class PageTable;
 			bool readFromDevice (void);
 			bool writeToDevice (void);
+
+	public:
+		PagePointer ();
+		virtual ~PagePointer ();
+
+		PagePointer &operator= (const PagePointer &rhs);
 	};
 
-	
+
 	class PageTable : public PageTableGenerated, public NeedsHeader
 	{
 	private:
@@ -157,24 +166,9 @@
 
 	public:
 		PageTable ();
-		~PageTable ();
-
-		PageTable &operator= (const PageTable &rhs)
-		{
-			if (this == &rhs)
-				return *this;
+		virtual ~PageTable ();
 
-			PageTableGenerated::operator= (rhs);
-			NeedsHeader::operator= (rhs);
-
-			m_pagePointerList = rhs.m_pagePointerList;
-			m_pageNumberStart = rhs.m_pageNumberStart;
-
-			m_pageTableIterator = rhs.m_pageTableIterator;
-
-			return *this;
-		}
-		
+		PageTable &operator= (const PageTable &rhs);
 
 		PagePointer *begin (void)
 		{
@@ -187,7 +181,7 @@
 			m_pageTableIterator = m_pageTableIterator.next ();
 			return &(*m_pageTableIterator);
 		}
-		
+
 		bool end (void) /*const*/
 		{
 			return m_pageTableIterator == m_pagePointerList.end ();
@@ -199,7 +193,7 @@
 		}
 	};
 
-	
+
 	class FormatPointer : public FormatPointerGenerated
 	{
 	private:
@@ -211,7 +205,7 @@
 
 			//
 			// for exporting
-			// 
+			//
 			const void *m_formatProperty;
 
 			const void *getFormatProperty (void) const	{	return m_formatProperty;	}
@@ -219,7 +213,9 @@
 
 	public:
 		FormatPointer ();
-		~FormatPointer ();
+		virtual ~FormatPointer ();
+
+		FormatPointer &operator= (const FormatPointer &rhs);
 
 		DWord getAfterEndCharByte (void) const	{	return m_afterEndCharByte;	}
 		void setAfterEndCharByte (const DWord val)	{	m_afterEndCharByte = val;	}
@@ -240,7 +236,7 @@
 		DWord m_firstCharByte;	// FormatInfoPageGenerated::m_firstCharBytePlus128 - 128
 		enum FormatInfoPageTypes m_type;	// CharInfo or ParaInfo?
 
-		
+
 		//
 		// uncompressed data
 		//
@@ -254,7 +250,7 @@
 		FormatParaProperty *m_formatParaProperty;
 			Word m_leftMargin, m_rightMargin;	// for ParaProperty
 
-			
+
 		//
 		// Iterator helpers
 		//
@@ -275,7 +271,7 @@
 
 	public:
 		FormatInfoPage ();
-		~FormatInfoPage ();
+		virtual ~FormatInfoPage ();
 
 		void setMargins (const Word leftMargin, const Word rightMargin)
 		{
@@ -305,6 +301,11 @@
 	class BMP_BitmapFileHeader : public BMP_BitmapFileHeaderGenerated
 	{
 	public:
+		BMP_BitmapFileHeader ();
+		virtual ~BMP_BitmapFileHeader ();
+
+		BMP_BitmapFileHeader &operator= (const BMP_BitmapFileHeader &rhs);
+
 		bool readFromDevice (void);
 		bool writeToDevice (void);
 	};
@@ -313,6 +314,11 @@
 	class BMP_BitmapInfoHeader : public BMP_BitmapInfoHeaderGenerated
 	{
 	public:
+		BMP_BitmapInfoHeader ();
+		virtual ~BMP_BitmapInfoHeader ();
+
+		BMP_BitmapInfoHeader &operator= (const BMP_BitmapInfoHeader &rhs);
+
 		bool readFromDevice (void);
 		bool writeToDevice (void);
 	};
@@ -320,6 +326,11 @@
 
 	class BMP_BitmapColourIndex : public BMP_BitmapColourIndexGenerated
 	{
+	public:
+		BMP_BitmapColourIndex ();
+		virtual ~BMP_BitmapColourIndex ();
+
+		BMP_BitmapColourIndex &operator= (const BMP_BitmapColourIndex &rhs);
 	};
 
 
@@ -328,6 +339,11 @@
 	//private:
 	//	friend class ImageGenerated;
 	public:
+		BitmapHeader ();
+		virtual ~BitmapHeader ();
+
+		BitmapHeader &operator= (const BitmapHeader &rhs);
+
 		bool readFromDevice (void);
 		bool writeToDevice (void);
 	};
@@ -338,6 +354,11 @@
 	//private:
 	//	friend class Image;
 	public:
+		WMFHeader ();
+		virtual ~WMFHeader ();
+
+		WMFHeader &operator= (const WMFHeader &rhs);
+
 		bool readFromDevice (void);
 		bool writeToDevice (void);
 	};
--- interfaces/Makefile.am	20 Jun 2002 04:43:14 -0000	1.1.1.1
+++ interfaces/Makefile.am	1 Jun 2003 14:59:32 -0000	1.2
@@ -3,7 +3,7 @@
 lib_LTLIBRARIES = libkochart.la
 
 libkochart_la_SOURCES = koChart.cc
-libkochart_la_LDFLAGS = $(all_libraries) -version-info 1:0:0
+libkochart_la_LDFLAGS = $(all_libraries) -version-info 1:0:0 -no-undefined
 libkochart_la_LIBADD = $(LIB_KOFFICECORE)
 METASOURCES = AUTO
 
--- karbon/Makefile.am	5 Jul 2003 18:31:46 -0000	1.1.1.6
+++ karbon/Makefile.am	9 Jul 2003 14:40:31 -0000	1.11
@@ -12,22 +12,24 @@
 
 SUBDIRS = commands visitors core shapes widgets dialogs dockers render tools . plugins data pics
 
-bin_PROGRAMS = karbon
+bin_PROGRAMS = 
+lib_LTLIBRARIES =
 
-lib_LTLIBRARIES = karbon.la
+noinst_LTLIBRARIES = libkarbonpart_shared.la
 kde_module_LTLIBRARIES = libkarbonbase.la libkarbonpart.la
+kdeinit_LTLIBRARIES = karbon.la
 
-libkarbonpart_la_SOURCES = karbon_factory.cc 	karbon_resourceserver.cc 	karbon_part.cc 	karbon_part_iface.cc 	karbon_part_iface.skel 	karbon_view.cc 	karbon_view_iface.cc 	karbon_view_iface.skel karbon_drag.cpp
+libkarbonpart_shared_la_SOURCES = karbon_factory.cc 	karbon_resourceserver.cc 	karbon_part.cc 	karbon_part_iface.cc 	karbon_part_iface.skel 	karbon_view.cc 	karbon_view_iface.cc 	karbon_view_iface.skel karbon_drag.cpp
 
 libkarbonbase_la_SOURCES = \
 	karbon_view_base.cc \
 	karbon_part_base.cc
 
-libkarbonbase_la_LDFLAGS = $(KDE_PLUGIN)
+libkarbonbase_la_LDFLAGS = $(all_libraries) $(KDE_RPATH) -no-undefined
 libkarbonbase_la_LIBADD = $(LIB_KOFFICEUI)
 
-libkarbonpart_la_LDFLAGS = $(KDE_PLUGIN) $(LIBFREETYPE_RPATH)
-libkarbonpart_la_LIBADD = $(LIB_KOFFICEUI) $(LIB_KOPAINTER) \
+libkarbonpart_shared_la_LDFLAGS = $(all_libraries) $(KDE_RPATH) -avoid-version $(LIBFREETYPE_RPATH) -no-undefined
+libkarbonpart_shared_la_LIBADD = $(LIB_KOFFICEUI) $(LIB_KOPAINTER) \
 	dockers/libkarbondockers.la \
 	widgets/libkarbonwidgets.la \
 	render/libvpainter.la \
@@ -43,18 +45,18 @@
 	$(FONTCONFIG_LIBS) \
 	$(LIBFREETYPE_LIBS)
 
-libkarbonpart_la_METASOURCES = AUTO
+libkarbonpart_shared_la_METASOURCES = AUTO
 
-karbon_la_SOURCES = main.cc
-karbon_la_LDFLAGS = $(all_libraries) -module -avoid-version
-karbon_la_LIBADD = libkarbonpart.la
-
-karbon_SOURCES = dummy.cc
-karbon_LDFLAGS = $(all_libraries) $(KDE_RPATH)
-karbon_LDADD = karbon.la
+libkarbonpart_la_SOURCES = dummy.cc
+libkarbonpart_la_LDFLAGS = $(all_libraries) $(KDE_PLUGIN) -module
+libkarbonpart_la_LIBADD  = libkarbonpart_shared.la
 
 dummy.cc:
 	echo > dummy.cc
+
+karbon_la_SOURCES = main.cc
+karbon_la_LDFLAGS = $(all_libraries) $(KDE_RPATH) -module -avoid-version
+karbon_la_LIBADD = libkarbonpart_shared.la
 
 noinst_HEADERS = \
 	karbon_factory.h \
--- karbon/configure.in.in	5 Jul 2003 18:31:45 -0000	1.1.1.5
+++ karbon/configure.in.in	6 Jul 2003 14:11:33 -0000	1.3
@@ -1,5 +1,5 @@
 
-KDE_FIND_PATH(freetype-config, FREETYPE_CONFIG, [${prefix}/bin ${exec_prefix}/bin /usr/local/bin /opt/local/bin], [
+KDE_FIND_PATH(freetype-config, FREETYPE_CONFIG, [/usr/X11R6/bin ${prefix}/bin ${exec_prefix}/bin /usr/local/bin /opt/local/bin], [
   AC_MSG_WARN([Could not find libfreetype anywhere, check http://www.freetype.org/])
 ])
 
--- karbon/dockers/Makefile.am	5 Jul 2003 18:31:48 -0000	1.1.1.3
+++ karbon/dockers/Makefile.am	6 Jul 2003 14:11:35 -0000	1.3
@@ -20,7 +20,6 @@
 	vtransformdocker.h
 
 libkarbondockers_la_SOURCES = \
-	dummy.cc \
 	vcolordocker.cc \
 	vdocumentdocker.cc \
 	vdocker.cc \
@@ -29,10 +28,3 @@
 	vtransformdocker.cc
 
 libkarbondockers_la_METASOURCES = AUTO
-
-DISTCLEANFILES = \
-	dummy.cc
-
-dummy.cc:
-	echo > dummy.cc
-
--- karbon/main.cc	5 Jul 2003 18:31:46 -0000	1.1.1.4
+++ karbon/main.cc	6 Jul 2003 21:05:00 -0000	1.2
@@ -34,7 +34,7 @@
 		KCmdLineLastOption
 	};
 
-int main( int argc, char* argv[] )
+extern "C" int kdemain( int argc, char* argv[] )
 {
 	KCmdLineArgs::init( argc, argv, KarbonFactory::aboutData() );
 	KCmdLineArgs::addCmdLineOptions( options );
--- karbon/plugins/flattenpath/Makefile.am	25 Apr 2003 15:44:10 -0000	1.1.1.1
+++ karbon/plugins/flattenpath/Makefile.am	27 Apr 2003 19:04:26 -0000	1.2
@@ -4,7 +4,7 @@
 
 karbon_flattenpathplugin_la_SOURCES = flattenpathplugin.cc
 karbon_flattenpathplugin_la_LIBADD = $(LIB_KPARTS) $(LIB_KOFFICEUI) \
-				../../libkarbonpart.la
+				../../libkarbonpart_shared.la
 
 karbon_flattenpathplugin_la_LDFLAGS = $(all_libraries) -module $(KDE_PLUGIN)
 
--- karbon/plugins/imagetool/Makefile.am	25 Apr 2003 15:44:10 -0000	1.1.1.1
+++ karbon/plugins/imagetool/Makefile.am	27 Apr 2003 19:04:26 -0000	1.2
@@ -4,7 +4,7 @@
 
 karbon_imagetoolplugin_la_SOURCES = imagetoolplugin.cc
 karbon_imagetoolplugin_la_LIBADD = $(LIB_KPARTS) $(LIB_KOFFICEUI) \
-				../../libkarbonpart.la
+				../../libkarbonpart_shared.la
 
 karbon_imagetoolplugin_la_LDFLAGS = $(all_libraries) -module $(KDE_PLUGIN)
 
--- karbon/plugins/insertknots/Makefile.am	25 Apr 2003 15:44:10 -0000	1.1.1.1
+++ karbon/plugins/insertknots/Makefile.am	27 Apr 2003 19:04:26 -0000	1.2
@@ -4,7 +4,7 @@
 
 karbon_insertknotsplugin_la_SOURCES = insertknotsplugin.cc
 karbon_insertknotsplugin_la_LIBADD = $(LIB_KPARTS) \
-					../../libkarbonpart.la
+					../../libkarbonpart_shared.la
 
 karbon_insertknotsplugin_la_LDFLAGS = $(all_libraries) -module $(KDE_PLUGIN)
 
--- karbon/plugins/roundcorners/Makefile.am	25 Apr 2003 15:44:10 -0000	1.1.1.1
+++ karbon/plugins/roundcorners/Makefile.am	27 Apr 2003 19:04:26 -0000	1.2
@@ -4,7 +4,7 @@
 
 karbon_roundcornersplugin_la_SOURCES = roundcornersplugin.cc
 karbon_roundcornersplugin_la_LIBADD = $(LIB_KPARTS) \
-					../../libkarbonpart.la
+					../../libkarbonpart_shared.la
 
 karbon_roundcornersplugin_la_LDFLAGS = $(all_libraries) -module $(KDE_PLUGIN)
 
--- karbon/plugins/shadoweffect/Makefile.am	25 Apr 2003 15:44:10 -0000	1.1.1.1
+++ karbon/plugins/shadoweffect/Makefile.am	27 Apr 2003 19:04:27 -0000	1.2
@@ -4,7 +4,7 @@
 
 karbon_shadoweffectplugin_la_SOURCES = shadoweffectplugin.cc vshadowdecorator.cc 
 karbon_shadoweffectplugin_la_LIBADD = $(LIB_KPARTS) $(LIB_KOFFICEUI) \
-					../../libkarbonpart.la
+					../../libkarbonpart_shared.la
 
 karbon_shadoweffectplugin_la_LDFLAGS = $(all_libraries) -module $(KDE_PLUGIN)
 
--- karbon/plugins/whirlpinch/Makefile.am	5 Jul 2003 18:31:49 -0000	1.1.1.2
+++ karbon/plugins/whirlpinch/Makefile.am	6 Jul 2003 14:11:37 -0000	1.3
@@ -4,7 +4,7 @@
 
 karbon_whirlpinchplugin_la_SOURCES = whirlpinchplugin.cc
 karbon_whirlpinchplugin_la_LIBADD = $(LIB_KPARTS) $(LIB_KOPAINTER) \
-				../../libkarbonpart.la
+				../../libkarbonpart_shared.la
 
 
 karbon_whirlpinchplugin_la_LDFLAGS = $(all_libraries) -module $(KDE_PLUGIN)
--- karbon/plugins/zoomtool/Makefile.am	25 Apr 2003 15:44:10 -0000	1.1.1.1
+++ karbon/plugins/zoomtool/Makefile.am	27 Apr 2003 19:04:27 -0000	1.2
@@ -4,7 +4,7 @@
 
 karbon_zoomtoolplugin_la_SOURCES = zoomtoolplugin.cc
 karbon_zoomtoolplugin_la_LIBADD = $(LIB_KPARTS) $(LIB_KOFFICEUI) \
-				../../libkarbonpart.la
+				../../libkarbonpart_shared.la
 
 karbon_zoomtoolplugin_la_LDFLAGS = $(all_libraries) -module $(KDE_PLUGIN)
 
--- karbon/widgets/Makefile.am	5 Jul 2003 18:31:54 -0000	1.1.1.3
+++ karbon/widgets/Makefile.am	6 Jul 2003 14:11:41 -0000	1.3
@@ -26,7 +26,6 @@
 	vtypebuttonbox.h
 
 libkarbonwidgets_la_SOURCES = \
-	dummy.cc \
 	vcanvas.cc \
 	vcolorslider.cc \
 	vcolordlg.cc \
@@ -42,10 +41,3 @@
 
 libkarbonwidgets_la_METASOURCES = \
 	AUTO
-
-DISTCLEANFILES = \
-	dummy.cc
-
-dummy.cc:
-	echo > dummy.cc
-
--- kchart/Makefile.am	25 Apr 2003 15:44:18 -0000	1.1.1.3
+++ kchart/Makefile.am	9 Jul 2003 14:40:31 -0000	1.7
@@ -2,6 +2,7 @@
 
 ## The part
 kde_module_LTLIBRARIES = libkchartpart.la
+
 libkchartpart_la_SOURCES = kchart_part.cc \
 	kchart_view.cc kchart_factory.cc \
 	kchartBarConfigDialog.cc \
@@ -33,25 +34,18 @@
 	KChartParamsIface.cc kchartHeaderFooterConfigPage.cc \
 	kchartLine3dConfigPage.cc kchartParameterPolarConfigPage.cc kchartPageLayout.cc
 
-libkchartpart_la_LDFLAGS = $(KDE_PLUGIN)
+libkchartpart_la_LDFLAGS = $(all_libraries) $(KDE_PLUGIN) -module
 libkchartpart_la_LIBADD = $(top_builddir)/interfaces/libkochart.la \
                           $(LIB_KOFFICEUI) \
 			  kdchart/libkdchart.la			
 
 ## The kdeinit loadable module
-lib_LTLIBRARIES = kchart.la
+bin_PROGRAMS =
+lib_LTLIBRARIES =
+kdeinit_LTLIBRARIES = kchart.la
 kchart_la_SOURCES = main.cc
 kchart_la_LDFLAGS = $(all_libraries) -avoid-version -module
 kchart_la_LIBADD = $(LIB_KOFFICECORE)
-
-## The executable
-bin_PROGRAMS = kchart
-kchart_SOURCES = dummy.cc
-kchart_LDFLAGS = $(all_libraries) $(KDE_RPATH)
-kchart_LDADD = kchart.la
-
-dummy.cc:
-	echo > dummy.cc
 
 noinst_HEADERS =  kchart_part.h \
 		kchart_view.h kchart_factory.h \
--- kchart/kdchart/Makefile.am	20 Jun 2002 04:49:58 -0000	1.1.1.2
+++ kchart/kdchart/Makefile.am	1 Jun 2003 14:59:33 -0000	1.5
@@ -1,5 +1,7 @@
 lib_LTLIBRARIES = libkdchart.la
 
+libkdchart_la_LIBADD  = $(LIB_QT)
+libkdchart_la_LDFLAGS = $(all_libraries) -avoid-version -no-undefined
 libkdchart_la_SOURCES = KDChart.cpp 		\
 			KDChartAreaPainter.cpp	\
 			KDChartAxesPainter.cpp	\
--- kchart/main.cc	25 Apr 2003 15:44:18 -0000	1.1.1.3
+++ kchart/main.cc	9 Jul 2003 14:40:32 -0000	1.2
@@ -28,7 +28,7 @@
 	KCmdLineLastOption
 };
 
-int main( int argc, char **argv )
+extern "C" int kdemain( int argc, char **argv )
 {
     KCmdLineArgs::init( argc, argv, newKChartAboutData());
     KCmdLineArgs::addCmdLineOptions( options );
--- kformula/Makefile.am	25 Apr 2003 15:45:03 -0000	1.1.1.4
+++ kformula/Makefile.am	9 Jul 2003 14:40:32 -0000	1.5
@@ -12,19 +12,12 @@
 libkformulapart_la_METASOURCES = AUTO
 
 ## The kdeinit loadable module
-lib_LTLIBRARIES = kformulamain.la
-kformulamain_la_SOURCES = main.cc
-kformulamain_la_LDFLAGS = $(all_libraries) -module -avoid-version
-kformulamain_la_LIBADD = $(LIB_KOFFICECORE)
-
-## The executable
-bin_PROGRAMS = kformula
-kformula_SOURCES = dummy.cc
-kformula_LDFLAGS = $(all_libraries) $(KDE_RPATH)
-kformula_LDADD = kformulamain.la
-
-dummy.cc:
-	echo > dummy.cc
+bin_PROGRAMS =
+lib_LTLIBRARIES =
+kdeinit_LTLIBRARIES = kformula.la
+kformula_la_SOURCES = main.cc
+kformula_la_LDFLAGS = $(all_libraries) -module -avoid-version
+kformula_la_LIBADD = $(LIB_KOFFICECORE)
 
 noinst_HEADERS =  kformula_doc.h \
 		  kformula_view.h kformula_factory.h \
--- kformula/main.cc	25 Apr 2003 15:45:03 -0000	1.1.1.3
+++ kformula/main.cc	6 Jul 2003 21:05:01 -0000	1.2
@@ -31,7 +31,7 @@
         KCmdLineLastOption
 };
 
-int main( int argc, char **argv )
+extern "C" int kdemain( int argc, char **argv )
 {
     KCmdLineArgs::init( argc, argv, newKFormulaAboutData() );
     KCmdLineArgs::addCmdLineOptions( options );
--- kivio/kiviopart/Makefile.am	5 Jul 2003 18:32:14 -0000	1.1.1.5
+++ kivio/kiviopart/Makefile.am	9 Jul 2003 14:40:33 -0000	1.9
@@ -1,7 +1,9 @@
 SUBDIRS = ui kiviosdk tools pics tiles stencils .
 
 kde_module_LTLIBRARIES = libkiviopart.la
-libkiviopart_la_LIBADD = \
+lib_LTLIBRARIES = libkiviopart_shared.la
+
+libkiviopart_shared_la_LIBADD = \
 		$(top_builddir)/kivio/kiviopart/tools/libtools.la\
 		$(top_builddir)/kivio/kiviopart/kiviosdk/libkiviosdk.la\
 		$(top_builddir)/kivio/kiviopart/ui/libui.la \
@@ -16,8 +18,8 @@
 	-I$(top_srcdir)/lib/kformula\
 	$(all_includes)
 
-libkiviopart_la_METASOURCES = AUTO
-libkiviopart_la_SOURCES = kivio_canvas.cpp kivio_doc.cpp kivio_factory.cpp kivio_tabbar.cpp 	kivio_view.cpp kivio_map.cpp kivio_page.cpp kivio_dlg_pageshow.cpp kivio_zoomaction.cpp 	kivio_icon_view.cpp kivio_config.cpp kivio_spawner_drag.cpp kivio_intra_stencil_data.cpp	kivio_stackbar.cpp stencilbarbutton.cpp stencilbardockmanager.cpp stencilbarmovemanager.cpp	tooldockbase.cpp tooldockmanager.cpp stencil_text_dlg.cpp kivio_grid_data.cpp kivio_guidelines.cpp	kivio_stencil_geometry_panel.cpp kivio_viewmanager_panel.cpp KIvioViewIface.skel	KIvioViewIface.cc KIvioDocIface.skel KIvioDocIface.cc KIvioMapIface.skel KIvioMapIface.cc	KIvioPageIface.skel KIvioPageIface.cc kivio_command.cc KIvioLayerIface.skel KIvioLayerIface.cc	kivioglobal.cpp kiviostencilsetaction.cpp kivioarrowheadaction.cpp KIvioStencilIface.skel	KIvioStencilIface.cc kiviodragobject.cpp
+libkiviopart_shared_la_METASOURCES = AUTO
+libkiviopart_shared_la_SOURCES = kivio_canvas.cpp kivio_doc.cpp kivio_factory.cpp kivio_tabbar.cpp 	kivio_view.cpp kivio_map.cpp kivio_page.cpp kivio_dlg_pageshow.cpp kivio_zoomaction.cpp 	kivio_icon_view.cpp kivio_config.cpp kivio_spawner_drag.cpp kivio_intra_stencil_data.cpp	kivio_stackbar.cpp stencilbarbutton.cpp stencilbardockmanager.cpp stencilbarmovemanager.cpp	tooldockbase.cpp tooldockmanager.cpp stencil_text_dlg.cpp kivio_grid_data.cpp kivio_guidelines.cpp	kivio_stencil_geometry_panel.cpp kivio_viewmanager_panel.cpp KIvioViewIface.skel	KIvioViewIface.cc KIvioDocIface.skel KIvioDocIface.cc KIvioMapIface.skel KIvioMapIface.cc	KIvioPageIface.skel KIvioPageIface.cc kivio_command.cc KIvioLayerIface.skel KIvioLayerIface.cc	kivioglobal.cpp kiviostencilsetaction.cpp kivioarrowheadaction.cpp KIvioStencilIface.skel	KIvioStencilIface.cc kiviodragobject.cpp
 
 EXTRA_DIST = \
 		main.cpp\
@@ -95,17 +97,18 @@
 		kiviodragobject.cpp
 
 
-lib_LTLIBRARIES = kivio.la
+bin_PROGRAMS =
+kdeinit_LTLIBRARIES = kivio.la
+
 kivio_la_SOURCES = main.cpp
 kivio_la_LDFLAGS = $(all_libraries) -module -avoid-version
 kivio_la_LIBADD = ../../lib/kofficecore/libkofficecore.la
 
-libkiviopart_la_LDFLAGS = -avoid-version -module $(all_libraries) -no-undefined
+libkiviopart_shared_la_LDFLAGS = -avoid-version $(all_libraries) -no-undefined
 
-bin_PROGRAMS = kivio
-kivio_SOURCES = dummy.cpp
-kivio_LDFLAGS = $(all_libraries) $(KDE_RPATH)
-kivio_LDADD = kivio.la
+libkiviopart_la_SOURCES = dummy.cpp
+libkiviopart_la_LIBADD  = libkiviopart_shared.la
+libkiviopart_la_LDFLAGS = -avoid-version $(all_libraries) $(KDE_PLUGIN) -module
 
 dummy.cpp:
 	echo > dummy.cpp
--- kivio/kiviopart/main.cpp	25 Apr 2003 15:45:13 -0000	1.1.1.4
+++ kivio/kiviopart/main.cpp	6 Jul 2003 21:05:02 -0000	1.2
@@ -30,7 +30,7 @@
   KCmdLineLastOption
 };
 
-int main( int argc, char **argv )
+extern "C" int kdemain( int argc, char **argv )
 {
   KCmdLineArgs::init( argc, argv, newKivioAboutData() );
   KCmdLineArgs::addCmdLineOptions( options );
--- kivio/plugins/kivioconnectortool/Makefile.am	5 Jul 2003 18:32:27 -0000	1.1.1.5
+++ kivio/plugins/kivioconnectortool/Makefile.am	6 Jul 2003 14:11:50 -0000	1.5
@@ -1,5 +1,6 @@
 SUBDIRS = straight_connector
-lib_LTLIBRARIES =libkivioconnectortool.la
+
+kde_module_LTLIBRARIES =libkivioconnectortool.la
 
 
 INCLUDES = -I$(top_srcdir)/kivio/kiviopart\
@@ -23,7 +24,7 @@
 		kivio_connector_cursor1.xpm
 
 libkivioconnectortool_la_LDFLAGS = -module $(KDE_PLUGIN)
-libkivioconnectortool_la_LIBADD = $(all_libraries) $(LIB_KOPAINTER) $(top_builddir)/kivio/kiviopart/libkiviopart.la
+libkivioconnectortool_la_LIBADD = $(all_libraries) $(LIB_KOPAINTER) $(top_builddir)/kivio/kiviopart/libkiviopart_shared.la
 
 rc_DATA = kivioconnectortool.rc
 rcdir = $(kde_datadir)/kivio/kpartplugins
--- kivio/plugins/kivioconnectortool/straight_connector/Makefile.am	5 Jul 2003 18:32:27 -0000	1.1.1.5
+++ kivio/plugins/kivioconnectortool/straight_connector/Makefile.am	6 Jul 2003 14:11:52 -0000	1.8
@@ -6,7 +6,7 @@
 straight_connector_la_LDFLAGS = -module $(KDE_PLUGIN) $(all_libraries)
 
 # Hack to make it work (dave m)
-straight_connector_la_LIBADD = $(LIB_KOPAINTER) $(top_builddir)/kivio/kiviopart/libkiviopart.la
+straight_connector_la_LIBADD = $(LIB_KOPAINTER) $(top_builddir)/kivio/kiviopart/libkiviopart_shared.la
 
 INCLUDES = -I$(top_srcdir)/kivio/kiviopart\
 	-I$(top_srcdir)/kivio/kiviopart/kiviosdk\
--- kivio/plugins/kivioselecttool/Makefile.am	5 Jul 2003 18:32:27 -0000	1.1.1.5
+++ kivio/plugins/kivioselecttool/Makefile.am	6 Jul 2003 14:11:55 -0000	1.5
@@ -1,5 +1,5 @@
-SUBDIRS = select_pics
-lib_LTLIBRARIES =libkivioselecttool.la
+SUBDIRS = select_pics 
+kde_module_LTLIBRARIES =libkivioselecttool.la
 
 
 INCLUDES = -I$(top_srcdir)/kivio/kiviopart\
@@ -18,7 +18,7 @@
 		tool_select.h\
 		kivioselecttool.rc
 libkivioselecttool_la_LDFLAGS = -module $(KDE_PLUGIN)
-libkivioselecttool_la_LIBADD = $(LIB_KOTEXT) $(LIB_KOPAINTER) $(all_libraries) $(top_builddir)/kivio/kiviopart/libkiviopart.la
+libkivioselecttool_la_LIBADD = $(LIB_KOTEXT) $(LIB_KOPAINTER) $(all_libraries) $(top_builddir)/kivio/kiviopart/libkiviopart_shared.la
 
 rc_DATA = kivioselecttool.rc
 rcdir = $(kde_datadir)/kivio/kpartplugins
--- kivio/plugins/kiviosmlconnector/Makefile.am	5 Jul 2003 18:32:27 -0000	1.1.1.1
+++ kivio/plugins/kiviosmlconnector/Makefile.am	9 Jul 2003 14:40:33 -0000	1.2
@@ -10,7 +10,7 @@
 METASOURCES = AUTO
 libkiviosmlconnector_la_SOURCES = plugin.cpp tool_connector.cpp
 libkiviosmlconnector_la_LDFLAGS = -module $(KDE_PLUGIN) $(all_libraries)
-libkiviosmlconnector_la_LIBADD = $(LIB_KOPAINTER) $(top_builddir)/kivio/kiviopart/libkiviopart.la
+libkiviosmlconnector_la_LIBADD = $(LIB_KOPAINTER) $(top_builddir)/kivio/kiviopart/libkiviopart_shared.la
 
 
 
--- kivio/plugins/kiviosmlconnector/sml_connector/Makefile.am	5 Jul 2003 18:32:27 -0000	1.1.1.1
+++ kivio/plugins/kiviosmlconnector/sml_connector/Makefile.am	9 Jul 2003 14:40:34 -0000	1.2
@@ -9,7 +9,7 @@
 
 sml_connector_la_SOURCES = sml_connector.cpp
 sml_connector_la_LDFLAGS = -module $(KDE_PLUGIN) $(all_libraries)
-sml_connector_la_LIBADD = $(LIB_KOPAINTER) $(top_builddir)/kivio/kiviopart/libkiviopart.la
+sml_connector_la_LIBADD = $(LIB_KOPAINTER) $(top_builddir)/kivio/kiviopart/libkiviopart_shared.la
 
 stencildir = $(kde_datadir)/kivio/autoloadStencils
 stencil_DATA = sml_connector.xpm
--- kivio/plugins/kiviotexttool/Makefile.am	5 Jul 2003 18:32:27 -0000	1.1.1.5
+++ kivio/plugins/kiviotexttool/Makefile.am	9 Jul 2003 14:40:35 -0000	1.6
@@ -1,4 +1,4 @@
-lib_LTLIBRARIES =libkiviotexttool.la
+kde_module_LTLIBRARIES =libkiviotexttool.la
 
 
 INCLUDES = -I$(top_srcdir)/kivio/kiviopart\
@@ -21,7 +21,7 @@
 		Text.sml\
 		Text.xpm
 libkiviotexttool_la_LDFLAGS = -module $(KDE_PLUGIN)
-libkiviotexttool_la_LIBADD = $(all_libraries) $(LIB_KOPAINTER) $(top_builddir)/kivio/kiviopart/libkiviopart.la
+libkiviotexttool_la_LIBADD = $(all_libraries) $(LIB_KOPAINTER) $(top_builddir)/kivio/kiviopart/libkiviopart_shared.la
 
 rc_DATA = kiviotexttool.rc
 rcdir = $(kde_datadir)/kivio/kpartplugins
--- kivio/plugins/kiviozoomtool/Makefile.am	5 Jul 2003 18:32:27 -0000	1.1.1.5
+++ kivio/plugins/kiviozoomtool/Makefile.am	9 Jul 2003 14:40:35 -0000	1.6
@@ -1,5 +1,5 @@
-SUBDIRS = zoom_pics
-lib_LTLIBRARIES =libkiviozoomtool.la
+SUBDIRS = zoom_pics 
+kde_module_LTLIBRARIES =libkiviozoomtool.la
 
 INCLUDES = -I$(top_srcdir)/kivio/kiviopart\
 	-I$(top_srcdir)/kivio/kiviopart/kiviosdk\
@@ -17,7 +17,7 @@
 		tool_zoom.cpp\
 		tool_zoom.h
 libkiviozoomtool_la_LDFLAGS = -module $(KDE_PLUGIN)
-libkiviozoomtool_la_LIBADD = $(LIB_KOTEXT) $(all_libraries) $(LIB_KOPAINTER) $(top_builddir)/kivio/kiviopart/libkiviopart.la
+libkiviozoomtool_la_LIBADD = $(LIB_KOTEXT) $(all_libraries) $(LIB_KOPAINTER) $(top_builddir)/kivio/kiviopart/libkiviopart_shared.la
 
 rc_DATA = kiviozoomtool.rc
 rcdir = $(kde_datadir)/kivio/kpartplugins
--- koshell/Makefile.am	25 Apr 2003 15:45:37 -0000	1.1.1.2
+++ koshell/Makefile.am	9 Jul 2003 14:40:36 -0000	1.8
@@ -2,19 +2,19 @@
 INCLUDES  = $(KOFFICE_INCLUDES) $(all_includes)
 
 ## The kdeinit loadable module
-lib_LTLIBRARIES = koshell.la
-koshell_la_SOURCES = koshell_main.cc koshell_shell.cc
-koshell_la_LDFLAGS = $(all_libraries) -avoid-version -module
-koshell_la_LIBADD = $(LIB_KOFFICEUI)
+bin_PROGRAMS =
+lib_LTLIBRARIES =
+kdeinit_LTLIBRARIES = koshell.la
+noinst_LTLIBRARIES = libkoshell_shared.la
 
-## The executable
-bin_PROGRAMS = koshell
-koshell_SOURCES = dummy.cc
-koshell_LDFLAGS = $(all_libraries) $(KDE_RPATH)
-koshell_LDADD = koshell.la
+libkoshell_shared_la_SOURCES = koshell_shell.cc
+libkoshell_shared_la_LDFLAGS = $(all_libraries) -avoid-version -no-undefined
+libkoshell_shared_la_LIBADD = $(LIB_KOFFICEUI)
 
-dummy.cc:
-	echo > dummy.cc
+koshell_la_SOURCES = koshell_main.cc
+koshell_la_LDFLAGS = $(all_libraries) $(KDE_RPATH) -module -avoid-version
+koshell_la_LIBADD  = libkoshell_shared.la
+koshell_la_DEPENDENCIES = libkoshell_shared.la
 
 noinst_HEADERS = koshell_shell.h
 
--- koshell/koshell_main.cc	25 Apr 2003 15:45:36 -0000	1.1.1.6
+++ koshell/koshell_main.cc	6 Jul 2003 21:05:03 -0000	1.2
@@ -30,7 +30,7 @@
 static const char* description=I18N_NOOP("KOffice Workspace");
 static const char* version=VERSION;
 
-int main( int argc, char **argv )
+extern "C" int kdemain( int argc, char **argv )
 {
   KAboutData * aboutData = new KAboutData( "koshell", I18N_NOOP("KOffice Workspace"),
      version, description, KAboutData::License_GPL,
--- kpresenter/Makefile.am	25 Apr 2003 15:45:57 -0000	1.1.1.4
+++ kpresenter/Makefile.am	9 Jul 2003 14:40:36 -0000	1.6
@@ -54,19 +54,12 @@
 libkpresenterpart_la_METASOURCES = AUTO
 
 ## The kdeinit loadable module
-lib_LTLIBRARIES = kpresenter.la
+bin_PROGRAMS =
+lib_LTLIBRARIES =
+kdeinit_LTLIBRARIES = kpresenter.la
 kpresenter_la_SOURCES = main.cc
-kpresenter_la_LDFLAGS = $(all_libraries) -avoid-version -module
+kpresenter_la_LDFLAGS = $(all_libraries) -avoid-version $(KDE_RPATH) -module
 kpresenter_la_LIBADD = $(LIB_KOFFICECORE)
-
-## The executable
-bin_PROGRAMS = kpresenter
-kpresenter_SOURCES = dummy.cc
-kpresenter_LDFLAGS = $(all_libraries) $(KDE_RPATH)
-kpresenter_LDADD = kpresenter.la
-
-dummy.cc:
-	echo > dummy.cc
 
 noinst_HEADERS = koPointArray.h kprcanvas.h kprpage.h kpresenter_doc.h kpresenter_view.h \
 		backdia.h styledia.h \
--- kpresenter/kpresenter_view.cc	5 Jul 2003 18:32:37 -0000	1.1.1.7
+++ kpresenter/kpresenter_view.cc	6 Jul 2003 14:12:02 -0000	1.11
@@ -4276,7 +4276,7 @@
     //(Laurent) deselect object when we change page.
     //otherwise you can change object properties on other page
     deSelectAllObjects();
-    m_pKPresenterDoc->repaint( FALSE );
+    m_pKPresenterDoc->repaint( (bool)FALSE );
 
     m_pKPresenterDoc->displayActivePage( page );
 
--- kpresenter/main.cc	5 Jul 2003 18:32:42 -0000	1.1.1.4
+++ kpresenter/main.cc	6 Jul 2003 21:05:04 -0000	1.2
@@ -32,7 +32,7 @@
 	KCmdLineLastOption
 };
 
-int main( int argc, char **argv )
+extern "C" int kdemain( int argc, char **argv )
 {
     KCmdLineArgs::init( argc, argv, newKPresenterAboutData() );
     KCmdLineArgs::addCmdLineOptions( options );
--- kspread/Makefile.am	25 Apr 2003 15:46:34 -0000	1.1.1.4
+++ kspread/Makefile.am	9 Jul 2003 14:40:37 -0000	1.8
@@ -3,10 +3,10 @@
 	$(KOSCRIPT_INCLUDES) -I$(top_srcdir)/interfaces \
 	 -I$(top_srcdir)/kchart/kdchart $(all_includes)
 
-kde_module_LTLIBRARIES = libkspreadpart.la
+kde_module_LTLIBRARIES = libkspreadpart.la libkspreadpart_shared.la
 
 ## The part
-libkspreadpart_la_SOURCES = kspread_locale.cc \
+libkspreadpart_shared_la_SOURCES = kspread_locale.cc \
 	kspread_dlg_formula.cc digest.cc \
 	kspread_functions.cc kspread_dlg_csv.cc \
 	kspread_canvas.cc kspread_functions_helper.cc \
@@ -45,26 +45,27 @@
 	kspread_selection.cc kspread_dlg_paperlayout.cc kspread_format.cc \
         kspread_sheetprint.cc
 
-libkspreadpart_la_LDFLAGS = $(all_libraries) $(KDE_PLUGIN)
-libkspreadpart_la_LIBADD = $(top_builddir)/interfaces/libkochart.la \
+libkspreadpart_shared_la_LDFLAGS = $(all_libraries) $(KDE_RPATH) -avoid-version -no-undefined
+libkspreadpart_shared_la_LIBADD = $(top_builddir)/interfaces/libkochart.la \
 	$(LIB_KOSCRIPT) $(LIB_KOFFICEUI) $(LIB_KOTEXT) -lkspell
 
+libkspreadpart_la_LDFLAGS = $(all_libraries) $(KDE_PLUGIN)
+libkspreadpart_la_LIBADD  = libkspreadpart_shared.la
+libkspreadpart_la_SOURCES = dummy.cpp
+libkspreadpart_la_DEPENDENCIES = libkspreadpart_shared.la
+
+dummy.cpp:
+	echo > dummy.cpp
+
 METASOURCES = AUTO
 
 ## The kdeinit loadable module
-lib_LTLIBRARIES =  kspread.la
+bin_PROGRAMS =
+lib_LTLIBRARIES =
+kdeinit_LTLIBRARIES =  kspread.la
 kspread_la_SOURCES = main.cc
 kspread_la_LDFLAGS = $(all_libraries) -avoid-version -module
-kspread_la_LIBADD = $(LIB_KOFFICECORE)
-
-## The executable
-bin_PROGRAMS = kspread
-kspread_SOURCES = dummy.cc
-kspread_LDFLAGS = $(all_libraries) $(KDE_RPATH)
-kspread_LDADD = kspread.la
-
-dummy.cc:
-	echo > dummy.cc
+kspread_la_LIBADD = $(LIB_KOFFICECORE) libkspreadpart_shared.la
 
 noinst_HEADERS = kspread_doc.h kspread_functions.h kspread_functions_helper.h \
 	kspread_undo.h kspread_view.h kspread_tabbar.h \
--- kspread/main.cc	25 Apr 2003 15:46:34 -0000	1.1.1.3
+++ kspread/main.cc	6 Jul 2003 21:05:04 -0000	1.2
@@ -30,7 +30,7 @@
         KCmdLineLastOption
 };
 
-int main( int argc, char **argv )
+extern "C" int kdemain( int argc, char **argv )
 {
 
     KCmdLineArgs::init( argc, argv, newKSpreadAboutData() );
--- kspread/plugins/calculator/Makefile.am	25 Apr 2003 15:46:36 -0000	1.1.1.2
+++ kspread/plugins/calculator/Makefile.am	26 Apr 2003 02:07:38 -0000	1.3
@@ -9,7 +9,7 @@
 			main.cpp \
 			stats.cpp
 libkspreadcalc_la_LDFLAGS = -avoid-version -module -no-undefined
-libkspreadcalc_la_LIBADD = $(all_libraries) ../../libkspreadpart.la
+libkspreadcalc_la_LIBADD = $(all_libraries) ../../libkspreadpart_shared.la
 
 METASOURCES = AUTO
 
--- kugar/kudesigner/Makefile.am	5 Jul 2003 18:33:16 -0000	1.1.1.5
+++ kugar/kudesigner/Makefile.am	6 Jul 2003 21:05:05 -0000	1.5
@@ -4,30 +4,24 @@
 
 
 ## The kdeinit loadable module
-lib_LTLIBRARIES = libkudesignercore.la kudesigner.la
+lib_LTLIBRARIES = libkudesignercore.la
 libkudesignercore_la_SOURCES = kudesigner_view.cpp kudesigner_factory.cpp kudesigner_doc.cpp
-libkudesignercore_la_LDFLAGS = -avoid-version -module $(all_libraries) $(KDE_RPATH)
+libkudesignercore_la_LDFLAGS = -avoid-version $(all_libraries) $(KDE_RPATH)
 libkudesignercore_la_LIBADD = ../kudesigner_lib/libkudesignercommon.la $(LIB_KOFFICEUI) $(LIB_KPARTS) $(LIB_KOFFICECORE) $(LIB_KIO) $(LIB_KDEPRINT)
 
 ## The part
 kde_module_LTLIBRARIES = libkudesignerpart.la
 libkudesignerpart_la_SOURCES = kudesigner_part.cpp
-libkudesignerpart_la_LDFLAGS = $(all_libraries) $(KDE_PLUGIN) $(KDE_RPATH)
+libkudesignerpart_la_LDFLAGS = $(all_libraries) $(KDE_PLUGIN) $(KDE_RPATH) -module
 libkudesignerpart_la_LIBADD = libkudesignercore.la $(LIB_KOFFICEUI) $(LIB_KPARTS) $(LIB_KOFFICECORE)
 
+bin_PROGRAMS =
+kdeinit_LTLIBRARIES = kudesigner.la
+
 ## The kdeinit loadable module
 kudesigner_la_SOURCES = main.cpp
-kudesigner_la_LDFLAGS = $(all_libraries) -module -avoid-version
+kudesigner_la_LDFLAGS = $(all_libraries) -avoid-version $(KDE_PLUGIN) -module
 kudesigner_la_LIBADD = $(LIB_KOFFICECORE)
-
-## The executable
-bin_PROGRAMS = kudesigner
-kudesigner_SOURCES = dummy.cc
-kudesigner_LDFLAGS = $(all_libraries) $(KDE_RPATH)
-kudesigner_LDADD = kudesigner.la
-
-dummy.cc:
-	echo > dummy.cc
 
 appsdir = $(kde_appsdir)/Office
 apps_DATA = kudesigner.desktop
--- kugar/kudesigner/main.cpp	5 Jul 2003 18:33:16 -0000	1.1.1.5
+++ kugar/kudesigner/main.cpp	9 Jul 2003 14:40:38 -0000	1.2
@@ -32,7 +32,7 @@
 	KCmdLineLastOption
 };
 
-int main( int argc, char **argv )
+extern "C" int kdemain( int argc, char **argv )
 {
     KLocale::setMainCatalogue("kugar");
     KCmdLineArgs::init( argc, argv, newKudesignerAboutData() );
--- kugar/part/Makefile.am	25 Apr 2003 15:46:43 -0000	1.1.1.2
+++ kugar/part/Makefile.am	9 Jul 2003 14:40:39 -0000	1.6
@@ -2,7 +2,7 @@
 
 kde_module_LTLIBRARIES = libkugarpart.la
 libkugarpart_la_SOURCES = kugar_part.cpp kugar_factory.cpp kugar_view.cpp
-libkugarpart_la_LDFLAGS = $(all_libraries) -module -avoid-version
+libkugarpart_la_LDFLAGS = $(all_libraries) $(KDE_PLUGIN) -module -avoid-version
 libkugarpart_la_LIBADD = ../lib/libkugar.la $(LIB_KPARTS) $(LIB_KDEPRINT) $(LIB_KOFFICECORE)
 
 libkugarpart_la_METASOURCES = AUTO
@@ -26,16 +26,9 @@
 
 
 ## The kdeinit loadable module
-lib_LTLIBRARIES = kugar.la
+bin_PROGRAMS =
+lib_LTLIBRARIES =
+kdeinit_LTLIBRARIES = kugar.la
 kugar_la_SOURCES = main.cpp
 kugar_la_LDFLAGS = $(all_libraries) -module -avoid-version
 kugar_la_LIBADD = $(LIB_KOFFICECORE)
-
-## The executable
-bin_PROGRAMS = kugar
-kugar_SOURCES = dummy.cc
-kugar_LDFLAGS = $(all_libraries) $(KDE_RPATH)
-kugar_LDADD = kugar.la
-
-dummy.cc:
-	echo > dummy.cc
--- kugar/part/main.cpp	5 Jul 2003 18:33:21 -0000	1.1.1.2
+++ kugar/part/main.cpp	9 Jul 2003 14:40:39 -0000	1.2
@@ -28,7 +28,7 @@
   // INSERT YOUR COMMANDLINE OPTIONS HERE
 };
 
-int main(int argc, char *argv[])
+extern "C" int kdemain(int argc, char *argv[])
 {
 
 	KCmdLineArgs::init( argc, argv, newKugarAboutData() );
--- kword/Makefile.am	5 Jul 2003 18:33:34 -0000	1.1.1.5
+++ kword/Makefile.am	6 Jul 2003 21:05:06 -0000	1.6
@@ -1,3 +1,4 @@
+KDE_OPTIONS = nofinal
 
 KDE_CXXFLAGS = $(USE_RTTI)
 INCLUDES = $(KOFFICE_INCLUDES) -I$(top_srcdir)/lib/kformula \
@@ -48,7 +49,7 @@
 libkwordpart_la_LIBADD = $(LIB_KOFFICEUI) $(LIB_KOTEXT) ../lib/kformula/libkformula.la -lkspell $(LIBKOSPELL) ./libkwmailmerge_interface.la
 METASOURCES = AUTO
 
-lib_LTLIBRARIES = kword.la libkwmailmerge_interface.la
+lib_LTLIBRARIES = libkwmailmerge_interface.la
 
 ## The mailmerge interface library
 libkwmailmerge_interface_la_SOURCES = mailmerge_interface.cc mailmerge_interface.skel
@@ -61,18 +62,11 @@
 kwmailmerge_servicetypedir=$(kde_servicetypesdir)
 
 ## The kdeinit loadable module
+bin_PROGRAMS =
+kdeinit_LTLIBRARIES = kword.la
 kword_la_SOURCES = main.cc
-kword_la_LDFLAGS = $(all_libraries) -avoid-version -module
-kword_la_LIBADD = ../lib/kofficecore/libkofficecore.la
-
-## The executable
-bin_PROGRAMS = kword
-kword_SOURCES = dummy.cc
-kword_LDFLAGS = $(all_libraries) $(KDE_RPATH)
-kword_LDADD = kword.la
-
-dummy.cc:
-	echo > dummy.cc
+kword_la_LDFLAGS = $(all_libraries) -avoid-version $(KDE_PLUGIN) -module
+kword_la_LIBADD = $(LIB_KOFFICECORE)
 
 kdelnk_DATA = kword.desktop
 kdelnkdir = $(kde_appsdir)/Office
--- kword/main.cc	25 Apr 2003 15:46:57 -0000	1.1.1.3
+++ kword/main.cc	6 Jul 2003 21:05:06 -0000	1.2
@@ -28,7 +28,7 @@
 	KCmdLineLastOption
 };
 
-int main( int argc, char **argv )
+extern "C" int kdemain( int argc, char **argv )
 {
     KCmdLineArgs::init( argc, argv, newKWordAboutData());
     KCmdLineArgs::addCmdLineOptions( options );
