Index: example/Makefile.am
===================================================================
RCS file: /home/kde/koffice/example/Makefile.am,v
retrieving revision 1.25
diff -u -u -r1.25 Makefile.am
--- example/Makefile.am	13 Dec 2002 18:55:15 -0000	1.25
+++ example/Makefile.am	9 Nov 2003 04:04:39 -0000
@@ -8,20 +8,13 @@
 libexamplepart_la_LDFLAGS = $(KDE_PLUGIN)
 libexamplepart_la_LIBADD = $(LIB_KOFFICEUI)
 
-## The kdeinit loadable module
-lib_LTLIBRARIES = example.la
+bin_PROGRAMS =
+lib_LTLIBRARIES =
+kdeinit_LTLIBRARIES = example.la
+
 example_la_SOURCES = main.cc
 example_la_LDFLAGS = $(all_libraries) -module -avoid-version
 example_la_LIBADD = $(LIB_KOFFICEUI)
-
-## The executable
-bin_PROGRAMS = example
-example_SOURCES = dummy.cc
-example_LDFLAGS = $(all_libraries) $(KDE_RPATH)
-example_LDADD = example.la
-
-dummy.cc:
-	echo > dummy.cc
 
 METASOURCES = AUTO
 noinst_HEADERS =  example_part.h \
Index: example/main.cc
===================================================================
RCS file: /home/kde/koffice/example/main.cc,v
retrieving revision 1.19
diff -u -u -r1.19 main.cc
--- example/main.cc	15 May 2003 12:31:41 -0000	1.19
+++ example/main.cc	9 Nov 2003 04:04:39 -0000
@@ -32,7 +32,7 @@
 	KCmdLineLastOption
 };
 
-int main( int argc, char **argv )
+extern "C" int kdemain( int argc, char **argv )
 {
     KCmdLineArgs::init( argc, argv, newExampleAboutData() );
     KCmdLineArgs::addCmdLineOptions( options );
Index: filters/Makefile.am
===================================================================
RCS file: /home/kde/koffice/filters/Makefile.am,v
retrieving revision 1.45
diff -u -u -r1.45 Makefile.am
--- filters/Makefile.am	14 Mar 2003 19:18:21 -0000	1.45
+++ filters/Makefile.am	9 Nov 2003 04:04:39 -0000
@@ -27,9 +27,4 @@
 KUGARDIR = kugar
 endif
 
-if compile_filter_KRITA
-KRITADIR = krita
-endif
-
-
-SUBDIRS = generic_wrapper liboofilter $(KSPREADDIR) $(KWORDDIR) $(KPRESENTERDIR) olefilters $(XSLTFILTERDIR) $(KFORMULADIR) $(KARBONDIR) $(KUGARDIR) $(KRITADIR)
+SUBDIRS = generic_wrapper liboofilter $(KSPREADDIR) $(KWORDDIR) $(KPRESENTERDIR) olefilters $(XSLTFILTERDIR) $(KFORMULADIR) $(KARBONDIR) $(KUGARDIR)
Index: filters/karbon/ai/Makefile.am
===================================================================
RCS file: /home/kde/koffice/filters/karbon/ai/Makefile.am,v
retrieving revision 1.16
diff -u -u -r1.16 Makefile.am
--- filters/karbon/ai/Makefile.am	18 Sep 2002 19:27:21 -0000	1.16
+++ filters/karbon/ai/Makefile.am	9 Nov 2003 04:04:39 -0000
@@ -2,7 +2,7 @@
 
 libkarbonaiimport_la_LIBADD = \
 	$(LIB_KOFFICEUI) \
-	../../../karbon/libkarbonpart.la
+	../../../karbon/libkarboncommon.la
 
 libkarbonaiimport_la_LDFLAGS = \
 	$(all_libraries) \
Index: filters/karbon/eps/Makefile.am
===================================================================
RCS file: /home/kde/koffice/filters/karbon/eps/Makefile.am,v
retrieving revision 1.15
diff -u -u -r1.15 Makefile.am
--- filters/karbon/eps/Makefile.am	21 Jun 2003 20:27:47 -0000	1.15
+++ filters/karbon/eps/Makefile.am	9 Nov 2003 04:04:39 -0000
@@ -7,7 +7,7 @@
 	epsexportdlg.cc 
 
 libkarbonepsexport_la_LDFLAGS = $(KDE_PLUGIN)
-libkarbonepsexport_la_LIBADD  = $(LIB_KOFFICEUI) ../../../karbon/libkarbonpart.la
+libkarbonepsexport_la_LIBADD  = $(LIB_KOFFICEUI) ../../../karbon/libkarboncommon.la
 
 libkarbonepsimport_la_SOURCES = epsimport.cc pscommentlexer.cc
 libkarbonepsimport_la_LDFLAGS = $(KDE_PLUGIN)
Index: filters/karbon/kontour/Makefile.am
===================================================================
RCS file: /home/kde/koffice/filters/karbon/kontour/Makefile.am,v
retrieving revision 1.3
diff -u -u -r1.3 Makefile.am
--- filters/karbon/kontour/Makefile.am	23 Mar 2003 15:10:16 -0000	1.3
+++ filters/karbon/kontour/Makefile.am	9 Nov 2003 04:04:39 -0000
@@ -11,7 +11,7 @@
 
 libkarbonkontourimport_la_SOURCES = kontourimport.cpp
 libkarbonkontourimport_la_LDFLAGS = -module $(KDE_PLUGIN)
-libkarbonkontourimport_la_LIBADD = $(KOFFICE_LIBS) ../../../karbon/libkarbonpart.la
+libkarbonkontourimport_la_LIBADD = $(KOFFICE_LIBS) ../../../karbon/libkarboncommon.la
 
 METASOURCES = AUTO
 
Index: filters/karbon/oodraw/Makefile.am
===================================================================
RCS file: /home/kde/koffice/filters/karbon/oodraw/Makefile.am,v
retrieving revision 1.1
diff -u -u -r1.1 Makefile.am
--- filters/karbon/oodraw/Makefile.am	30 Mar 2003 10:19:35 -0000	1.1
+++ filters/karbon/oodraw/Makefile.am	9 Nov 2003 04:04:39 -0000
@@ -14,7 +14,7 @@
 liboodrawimport_la_SOURCES = oodrawimport.cc
 liboodrawimport_la_LDFLAGS = $(all_libraries) -module -avoid-version -no-undefined
 liboodrawimport_la_LIBADD  = ../../liboofilter/liboofilter.la \
-							../../../karbon/libkarbonpart.la \
+							../../../karbon/libkarboncommon.la \
 							$(KOFFICE_LIBS)
 
 METASOURCES = AUTO
Index: filters/karbon/png/Makefile.am
===================================================================
RCS file: /home/kde/koffice/filters/karbon/png/Makefile.am,v
retrieving revision 1.2
diff -u -u -r1.2 Makefile.am
--- filters/karbon/png/Makefile.am	8 Sep 2002 18:52:36 -0000	1.2
+++ filters/karbon/png/Makefile.am	9 Nov 2003 04:04:39 -0000
@@ -3,7 +3,7 @@
 libkarbonpngexport_la_LDFLAGS = $(KDE_PLUGIN)
 libkarbonpngexport_la_LIBADD = \
 	$(LIB_KOFFICEUI) \
-	../../../karbon/libkarbonpart.la
+	../../../karbon/libkarboncommon.la
 
 INCLUDES = \
 	$(KOFFICE_INCLUDES) $(KOPAINTER_INCLUDES) \
Index: filters/karbon/svg/Makefile.am
===================================================================
RCS file: /home/kde/koffice/filters/karbon/svg/Makefile.am,v
retrieving revision 1.7
diff -u -u -r1.7 Makefile.am
--- filters/karbon/svg/Makefile.am	13 Oct 2003 21:35:48 -0000	1.7
+++ filters/karbon/svg/Makefile.am	9 Nov 2003 04:04:39 -0000
@@ -3,12 +3,12 @@
 libkarbonsvgexport_la_LDFLAGS = $(KDE_PLUGIN)
 libkarbonsvgexport_la_LIBADD = \
 	$(LIB_KOFFICEUI) \
-	../../../karbon/libkarbonpart.la
+	../../../karbon/libkarboncommon.la
 
 libkarbonsvgimport_la_LDFLAGS = $(KDE_PLUGIN)
 libkarbonsvgimport_la_LIBADD = \
 	$(LIB_KOFFICEUI) \
-	../../../karbon/libkarbonpart.la
+	../../../karbon/libkarboncommon.la
 
 INCLUDES = \
 	$(KOFFICE_INCLUDES) \
Index: filters/karbon/wmf/Makefile.am
===================================================================
RCS file: /home/kde/koffice/filters/karbon/wmf/Makefile.am,v
retrieving revision 1.4
diff -u -u -r1.4 Makefile.am
--- filters/karbon/wmf/Makefile.am	17 Apr 2003 19:23:31 -0000	1.4
+++ filters/karbon/wmf/Makefile.am	9 Nov 2003 04:04:39 -0000
@@ -12,12 +12,12 @@
 kde_module_LTLIBRARIES = libwmfimport.la libwmfexport.la
 
 libwmfimport_la_LDFLAGS = -module $(KDE_PLUGIN)
-libwmfimport_la_LIBADD = $(KOFFICE_LIBS) $(LIB_KOWMF) ../../../karbon/libkarbonpart.la
-#libwmfimport_la_LIBADD = $(KOFFICE_LIBS) $(LIB_KOWMF) ../../../karbon/libkarbonpart.la
+libwmfimport_la_LIBADD = $(KOFFICE_LIBS) $(LIB_KOWMF) ../../../karbon/libkarboncommon.la
+#libwmfimport_la_LIBADD = $(KOFFICE_LIBS) $(LIB_KOWMF) ../../../karbon/libkarboncommon.la
 libwmfimport_la_SOURCES = wmfimport.cc wmfimportparser.cc
 
 libwmfexport_la_LDFLAGS = -module $(KDE_PLUGIN)
-libwmfexport_la_LIBADD = $(KOFFICE_LIBS) $(LIB_KOWMF) ../../../karbon/libkarbonpart.la
+libwmfexport_la_LIBADD = $(KOFFICE_LIBS) $(LIB_KOWMF) ../../../karbon/libkarboncommon.la
 libwmfexport_la_SOURCES = wmfexport.cc
 
 noinst_HEADERS = wmfimport.h wmfimportparser.h wmfexport.h 
Index: filters/karbon/xcf/Makefile.am
===================================================================
RCS file: /home/kde/koffice/filters/karbon/xcf/Makefile.am,v
retrieving revision 1.1
diff -u -u -r1.1 Makefile.am
--- filters/karbon/xcf/Makefile.am	14 Sep 2002 12:14:14 -0000	1.1
+++ filters/karbon/xcf/Makefile.am	9 Nov 2003 04:04:39 -0000
@@ -3,7 +3,7 @@
 libkarbonxcfexport_la_LDFLAGS = $(KDE_PLUGIN)
 libkarbonxcfexport_la_LIBADD = \
 	$(LIB_KOFFICEUI) \
-	../../../karbon/libkarbonpart.la
+	../../../karbon/libkarboncommon.la
 
 INCLUDES = \
 	$(KOFFICE_INCLUDES) $(KOPAINTER_INCLUDES) \
Index: filters/kontour/latex/launcher/Makefile.am
===================================================================
RCS file: /home/kde/koffice/filters/kontour/latex/launcher/Makefile.am,v
retrieving revision 1.1
diff -u -u -r1.1 Makefile.am
--- filters/kontour/latex/launcher/Makefile.am	15 Jul 2001 15:09:54 -0000	1.1
+++ filters/kontour/latex/launcher/Makefile.am	9 Nov 2003 04:04:39 -0000
@@ -2,9 +2,11 @@
 
 INCLUDES= -I.. $(KOFFICE_INCLUDES) $(all_includes)
  
-bin_PROGRAMS = texlauncher
-
-texlauncher_SOURCES = texlauncher.cc
-texlauncher_LDADD = $(KOFFICE_LIBS) ../libtexgraphexport.la
+bin_PROGRAMS =
+lib_LTLIBRARIES =
+kdeinit_LTLIBRARIES = texlauncher.la
 
+texlauncher_la_SOURCES = texlauncher.cc
+texlauncher_la_LIBADD  = $(KOFFICE_LIBS) ../libtexgraphexport.la
+texlauncher_la_LDFLAGS = $(all_libraries) -module -avoid-version
 
Index: filters/kontour/latex/launcher/texlauncher.cc
===================================================================
RCS file: /home/kde/koffice/filters/kontour/latex/launcher/texlauncher.cc,v
retrieving revision 1.1
diff -u -u -r1.1 texlauncher.cc
--- filters/kontour/latex/launcher/texlauncher.cc	15 Jul 2001 15:09:54 -0000	1.1
+++ filters/kontour/latex/launcher/texlauncher.cc	9 Nov 2003 04:04:39 -0000
@@ -6,7 +6,7 @@
 #include <koFilter.h>
 #include <koStore.h>
 
-int main(int argc, char* argv[])
+extern "C" int kdemain(int argc, char* argv[])
 {
 	kdDebug() << argv[0] << endl;
 	kdDebug() << argv[1] << endl;
Index: filters/kspread/csv/Makefile.am
===================================================================
RCS file: /home/kde/koffice/filters/kspread/csv/Makefile.am,v
retrieving revision 1.22
diff -u -u -r1.22 Makefile.am
--- filters/kspread/csv/Makefile.am	1 Oct 2002 04:55:09 -0000	1.22
+++ filters/kspread/csv/Makefile.am	9 Nov 2003 04:04:39 -0000
@@ -9,11 +9,11 @@
 
 libcsvimport_la_SOURCES = csvimport.cc xmltree.cc csvdialog.cpp dialogui.ui
 libcsvimport_la_LDFLAGS	= $(all_libraries) -module -avoid-version -no-undefined 
-libcsvimport_la_LIBADD  = $(KOFFICE_LIBS) ../../../kspread/libkspreadpart.la
+libcsvimport_la_LIBADD  = $(KOFFICE_LIBS) ../../../kspread/libkspreadcommon.la
 
 libcsvexport_la_SOURCES = csvexport.cc csvexportdialog.cpp exportdialogui.ui
 libcsvexport_la_LDFLAGS = $(all_libraries) -module -avoid-version -no-undefined
-libcsvexport_la_LIBADD  = $(KOFFICE_LIBS) ../../../kspread/libkspreadpart.la
+libcsvexport_la_LIBADD  = $(KOFFICE_LIBS) ../../../kspread/libkspreadcommon.la
 
 METASOURCES = AUTO
 
Index: filters/kspread/excel/Makefile.am
===================================================================
RCS file: /home/kde/koffice/filters/kspread/excel/Makefile.am,v
retrieving revision 1.2
diff -u -u -r1.2 Makefile.am
--- filters/kspread/excel/Makefile.am	13 Dec 2002 18:53:27 -0000	1.2
+++ filters/kspread/excel/Makefile.am	9 Nov 2003 04:04:39 -0000
@@ -9,7 +9,7 @@
 
 libkspreadexcelexport_la_SOURCES = excelexport.cc
 libkspreadexcelexport_la_LDFLAGS = $(all_libraries) -module -avoid-version -no-undefined
-libkspreadexcelexport_la_LIBADD  = $(KOFFICE_LIBS) ../../../kspread/libkspreadpart.la ../libkspreadexport/libkspreadexport.la
+libkspreadexcelexport_la_LIBADD  = $(KOFFICE_LIBS) ../../../kspread/libkspreadcommon.la ../libkspreadexport/libkspreadexport.la
 
 METASOURCES = AUTO
 
Index: filters/kspread/gnumeric/Makefile.am
===================================================================
RCS file: /home/kde/koffice/filters/kspread/gnumeric/Makefile.am,v
retrieving revision 1.7
diff -u -u -r1.7 Makefile.am
--- filters/kspread/gnumeric/Makefile.am	1 Oct 2002 04:55:09 -0000	1.7
+++ filters/kspread/gnumeric/Makefile.am	9 Nov 2003 04:04:39 -0000
@@ -8,11 +8,11 @@
 
 libgnumericexport_la_SOURCES = gnumericexport.cc
 libgnumericexport_la_LDFLAGS = $(all_libraries) -module -avoid-version -no-undefined
-libgnumericexport_la_LIBADD  = $(KOFFICE_LIBS) ../../../kspread/libkspreadpart.la
+libgnumericexport_la_LIBADD  = $(KOFFICE_LIBS) ../../../kspread/libkspreadcommon.la
 
 libgnumericimport_la_SOURCES = gnumericimport.cc
 libgnumericimport_la_LDFLAGS = $(all_libraries) -module -avoid-version -no-undefined
-libgnumericimport_la_LIBADD  = $(KOFFICE_LIBS) ../../../kspread/libkspreadpart.la
+libgnumericimport_la_LIBADD  = $(KOFFICE_LIBS) ../../../kspread/libkspreadcommon.la
 
 METASOURCES = AUTO
 
Index: filters/kspread/html/Makefile.am
===================================================================
RCS file: /home/kde/koffice/filters/kspread/html/Makefile.am,v
retrieving revision 1.7
diff -u -u -r1.7 Makefile.am
--- filters/kspread/html/Makefile.am	1 Oct 2002 04:55:09 -0000	1.7
+++ filters/kspread/html/Makefile.am	9 Nov 2003 04:04:39 -0000
@@ -9,7 +9,7 @@
 
 libkspreadhtmlexport_la_SOURCES = htmlexport.cc
 libkspreadhtmlexport_la_LDFLAGS = $(all_libraries) -module -avoid-version -no-undefined
-libkspreadhtmlexport_la_LIBADD  = $(KOFFICE_LIBS) ../../../kspread/libkspreadpart.la
+libkspreadhtmlexport_la_LIBADD  = $(KOFFICE_LIBS) ../../../kspread/libkspreadcommon.la
 
 METASOURCES = AUTO
 
Index: filters/kspread/libkspreadexport/Makefile.am
===================================================================
RCS file: /home/kde/koffice/filters/kspread/libkspreadexport/Makefile.am,v
retrieving revision 1.1
diff -u -u -r1.1 Makefile.am
--- filters/kspread/libkspreadexport/Makefile.am	1 Dec 2002 23:11:06 -0000	1.1
+++ filters/kspread/libkspreadexport/Makefile.am	9 Nov 2003 04:04:39 -0000
@@ -5,7 +5,7 @@
 
 libkspreadexport_la_LDFLAGS = $(all_libraries) -module -version-info 1:0:0 -no-undefined
 
-libkspreadexport_la_LIBADD = $(KOFFICE_LIBS) ../../../kspread/libkspreadpart.la
+libkspreadexport_la_LIBADD = $(KOFFICE_LIBS) ../../../kspread/libkspreadcommon.la
 
 ####### Files
 
Index: filters/kspread/opencalc/Makefile.am
===================================================================
RCS file: /home/kde/koffice/filters/kspread/opencalc/Makefile.am,v
retrieving revision 1.5
diff -u -u -r1.5 Makefile.am
--- filters/kspread/opencalc/Makefile.am	4 Feb 2003 19:50:33 -0000	1.5
+++ filters/kspread/opencalc/Makefile.am	9 Nov 2003 04:04:39 -0000
@@ -8,11 +8,11 @@
 
 libopencalcexport_la_SOURCES = opencalcexport.cc opencalcstyleexport.cc
 libopencalcexport_la_LDFLAGS = $(all_libraries) -module -avoid-version -no-undefined
-libopencalcexport_la_LIBADD  = $(KOFFICE_LIBS) ../../../kspread/libkspreadpart.la
+libopencalcexport_la_LIBADD  = $(KOFFICE_LIBS) ../../../kspread/libkspreadcommon.la
 
 libopencalcimport_la_SOURCES = opencalcimport.cc
 libopencalcimport_la_LDFLAGS = $(all_libraries) -module -avoid-version -no-undefined
-libopencalcimport_la_LIBADD  = $(KOFFICE_LIBS) ../../../kspread/libkspreadpart.la
+libopencalcimport_la_LIBADD  = $(KOFFICE_LIBS) ../../../kspread/libkspreadcommon.la
 
 METASOURCES = AUTO
 
Index: filters/kspread/qpro/Makefile.am
===================================================================
RCS file: /home/kde/koffice/filters/kspread/qpro/Makefile.am,v
retrieving revision 1.8
diff -u -u -r1.8 Makefile.am
--- filters/kspread/qpro/Makefile.am	1 Oct 2002 04:55:09 -0000	1.8
+++ filters/kspread/qpro/Makefile.am	9 Nov 2003 04:04:39 -0000
@@ -13,7 +13,7 @@
 libqproimport_la_SOURCES = qproimport.cc qproformula.cc
 
 libqproimport_la_LDFLAGS = $(all_libraries) -module -avoid-version -no-undefined
-libqproimport_la_LIBADD  = $(KOFFICE_LIBS) ../../../kspread/libkspreadpart.la \
+libqproimport_la_LIBADD  = $(KOFFICE_LIBS) ../../../kspread/libkspreadcommon.la \
 	libqpro/src/libqpro.la
 
 METASOURCES = AUTO
Index: karbon/Makefile.am
===================================================================
RCS file: /home/kde/koffice/karbon/Makefile.am,v
retrieving revision 1.63
diff -u -u -r1.63 Makefile.am
--- karbon/Makefile.am	16 Oct 2003 07:55:10 -0000	1.63
+++ karbon/Makefile.am	9 Nov 2003 04:04:40 -0000
@@ -12,22 +12,21 @@
 
 SUBDIRS = commands visitors core shapes widgets dialogs dockers render tools . plugins data pics
 
-bin_PROGRAMS = karbon
-
-lib_LTLIBRARIES = karbon.la
-kde_module_LTLIBRARIES = libkarbonbase.la libkarbonpart.la
-
-libkarbonpart_la_SOURCES = karbon_factory.cc 	karbon_resourceserver.cc 	karbon_part.cc 	karbon_part_iface.cc 	karbon_part_iface.skel 	karbon_view.cc 	karbon_view_iface.cc 	karbon_view_iface.skel karbon_drag.cpp
+bin_PROGRAMS = 
+lib_LTLIBRARIES = libkarboncommon.la libkarbonbase.la
+kdeinit_LTLIBRARIES = karbon.la
+kde_module_LTLIBRARIES = libkarbonpart.la
 
 libkarbonbase_la_SOURCES = \
 	karbon_view_base.cc \
 	karbon_part_base.cc
 
-libkarbonbase_la_LDFLAGS = $(KDE_PLUGIN)
+libkarbonbase_la_LDFLAGS = $(KDE_RPATH)
 libkarbonbase_la_LIBADD = $(LIB_KOFFICEUI)
 
-libkarbonpart_la_LDFLAGS = $(KDE_PLUGIN) $(LIBFREETYPE_RPATH)
-libkarbonpart_la_LIBADD = $(LIB_KOFFICEUI) $(LIB_KOPAINTER) \
+libkarboncommon_la_SOURCES = karbon_factory.cc 	karbon_resourceserver.cc 	karbon_part.cc 	karbon_part_iface.cc 	karbon_part_iface.skel 	karbon_view.cc 	karbon_view_iface.cc 	karbon_view_iface.skel karbon_drag.cpp
+libkarboncommon_la_LDFLAGS = $(KDE_RPATH) $(LIBFREETYPE_RPATH)
+libkarboncommon_la_LIBADD  = $(LIB_KOFFICEUI) $(LIB_KOPAINTER) \
 	dockers/libkarbondockers.la \
 	widgets/libkarbonwidgets.la \
 	render/libvpainter.la \
@@ -43,18 +42,17 @@
 	$(LIBFONTCONFIG_LIBS) \
 	$(LIBFREETYPE_LIBS)
 
+libkarbonpart_la_SOURCES = dummy.cpp
+libkarbonpart_la_LDFLAGS = $(KDE_PLUGIN) $(LIBFREETYPE_RPATH)
+libkarbonpart_la_LIBADD = libkarboncommon.la
 libkarbonpart_la_METASOURCES = AUTO
 
 karbon_la_SOURCES = main.cc
 karbon_la_LDFLAGS = $(all_libraries) -module -avoid-version
-karbon_la_LIBADD = libkarbonpart.la
-
-karbon_SOURCES = dummy.cc
-karbon_LDFLAGS = $(all_libraries) $(KDE_RPATH)
-karbon_LDADD = karbon.la
+karbon_la_LIBADD = libkarboncommon.la
 
-dummy.cc:
-	echo > dummy.cc
+dummy.cpp:
+	echo > dummy.cpp
 
 noinst_HEADERS = \
 	karbon_factory.h \
Index: karbon/main.cc
===================================================================
RCS file: /home/kde/koffice/karbon/main.cc,v
retrieving revision 1.13
diff -u -u -r1.13 main.cc
--- karbon/main.cc	15 May 2003 12:31:42 -0000	1.13
+++ karbon/main.cc	9 Nov 2003 04:04:40 -0000
@@ -34,7 +34,7 @@
 		KCmdLineLastOption
 	};
 
-int main( int argc, char* argv[] )
+extern "C" int kdemain( int argc, char* argv[] )
 {
 	KCmdLineArgs::init( argc, argv, KarbonFactory::aboutData() );
 	KCmdLineArgs::addCmdLineOptions( options );
Index: karbon/plugins/flattenpath/Makefile.am
===================================================================
RCS file: /home/kde/koffice/karbon/plugins/flattenpath/Makefile.am,v
retrieving revision 1.5
diff -u -u -r1.5 Makefile.am
--- karbon/plugins/flattenpath/Makefile.am	23 Mar 2003 15:08:23 -0000	1.5
+++ karbon/plugins/flattenpath/Makefile.am	9 Nov 2003 04:04:40 -0000
@@ -4,7 +4,7 @@
 
 karbon_flattenpathplugin_la_SOURCES = flattenpathplugin.cc
 karbon_flattenpathplugin_la_LIBADD = $(LIB_KPARTS) $(LIB_KOFFICEUI) \
-				../../libkarbonpart.la
+				../../libkarboncommon.la
 
 karbon_flattenpathplugin_la_LDFLAGS = $(all_libraries) -module $(KDE_PLUGIN)
 
Index: karbon/plugins/imagetool/Makefile.am
===================================================================
RCS file: /home/kde/koffice/karbon/plugins/imagetool/Makefile.am,v
retrieving revision 1.3
diff -u -u -r1.3 Makefile.am
--- karbon/plugins/imagetool/Makefile.am	23 Mar 2003 15:08:23 -0000	1.3
+++ karbon/plugins/imagetool/Makefile.am	9 Nov 2003 04:04:40 -0000
@@ -4,7 +4,7 @@
 
 karbon_imagetoolplugin_la_SOURCES = imagetoolplugin.cc
 karbon_imagetoolplugin_la_LIBADD = $(LIB_KPARTS) $(LIB_KOFFICEUI) \
-				../../libkarbonpart.la
+				../../libkarboncommon.la
 
 karbon_imagetoolplugin_la_LDFLAGS = $(all_libraries) -module $(KDE_PLUGIN)
 
Index: karbon/plugins/insertknots/Makefile.am
===================================================================
RCS file: /home/kde/koffice/karbon/plugins/insertknots/Makefile.am,v
retrieving revision 1.5
diff -u -u -r1.5 Makefile.am
--- karbon/plugins/insertknots/Makefile.am	23 Mar 2003 15:08:23 -0000	1.5
+++ karbon/plugins/insertknots/Makefile.am	9 Nov 2003 04:04:40 -0000
@@ -4,7 +4,7 @@
 
 karbon_insertknotsplugin_la_SOURCES = insertknotsplugin.cc
 karbon_insertknotsplugin_la_LIBADD = $(LIB_KPARTS) \
-					../../libkarbonpart.la
+					../../libkarboncommon.la
 
 karbon_insertknotsplugin_la_LDFLAGS = $(all_libraries) -module $(KDE_PLUGIN)
 
Index: karbon/plugins/roundcorners/Makefile.am
===================================================================
RCS file: /home/kde/koffice/karbon/plugins/roundcorners/Makefile.am,v
retrieving revision 1.5
diff -u -u -r1.5 Makefile.am
--- karbon/plugins/roundcorners/Makefile.am	23 Mar 2003 15:08:23 -0000	1.5
+++ karbon/plugins/roundcorners/Makefile.am	9 Nov 2003 04:04:40 -0000
@@ -4,7 +4,7 @@
 
 karbon_roundcornersplugin_la_SOURCES = roundcornersplugin.cc
 karbon_roundcornersplugin_la_LIBADD = $(LIB_KPARTS) \
-					../../libkarbonpart.la
+					../../libkarboncommon.la
 
 karbon_roundcornersplugin_la_LDFLAGS = $(all_libraries) -module $(KDE_PLUGIN)
 
Index: karbon/plugins/shadoweffect/Makefile.am
===================================================================
RCS file: /home/kde/koffice/karbon/plugins/shadoweffect/Makefile.am,v
retrieving revision 1.3
diff -u -u -r1.3 Makefile.am
--- karbon/plugins/shadoweffect/Makefile.am	23 Mar 2003 15:08:23 -0000	1.3
+++ karbon/plugins/shadoweffect/Makefile.am	9 Nov 2003 04:04:40 -0000
@@ -4,7 +4,7 @@
 
 karbon_shadoweffectplugin_la_SOURCES = shadoweffectplugin.cc vshadowdecorator.cc 
 karbon_shadoweffectplugin_la_LIBADD = $(LIB_KPARTS) $(LIB_KOFFICEUI) \
-					../../libkarbonpart.la
+					../../libkarboncommon.la
 
 karbon_shadoweffectplugin_la_LDFLAGS = $(all_libraries) -module $(KDE_PLUGIN)
 
Index: karbon/plugins/whirlpinch/Makefile.am
===================================================================
RCS file: /home/kde/koffice/karbon/plugins/whirlpinch/Makefile.am,v
retrieving revision 1.7
diff -u -u -r1.7 Makefile.am
--- karbon/plugins/whirlpinch/Makefile.am	6 May 2003 09:57:55 -0000	1.7
+++ karbon/plugins/whirlpinch/Makefile.am	9 Nov 2003 04:04:40 -0000
@@ -4,7 +4,7 @@
 
 karbon_whirlpinchplugin_la_SOURCES = whirlpinchplugin.cc
 karbon_whirlpinchplugin_la_LIBADD = $(LIB_KPARTS) $(LIB_KOPAINTER) \
-				../../libkarbonpart.la
+				../../libkarboncommon.la
 
 
 karbon_whirlpinchplugin_la_LDFLAGS = $(all_libraries) -module $(KDE_PLUGIN)
Index: karbon/plugins/zoomtool/Makefile.am
===================================================================
RCS file: /home/kde/koffice/karbon/plugins/zoomtool/Makefile.am,v
retrieving revision 1.5
diff -u -u -r1.5 Makefile.am
--- karbon/plugins/zoomtool/Makefile.am	23 Mar 2003 15:08:24 -0000	1.5
+++ karbon/plugins/zoomtool/Makefile.am	9 Nov 2003 04:04:40 -0000
@@ -4,7 +4,7 @@
 
 karbon_zoomtoolplugin_la_SOURCES = zoomtoolplugin.cc
 karbon_zoomtoolplugin_la_LIBADD = $(LIB_KPARTS) $(LIB_KOFFICEUI) \
-				../../libkarbonpart.la
+				../../libkarboncommon.la
 
 karbon_zoomtoolplugin_la_LDFLAGS = $(all_libraries) -module $(KDE_PLUGIN)
 
Index: kchart/Makefile.am
===================================================================
RCS file: /home/kde/koffice/kchart/Makefile.am,v
retrieving revision 1.100
diff -u -u -r1.100 Makefile.am
--- kchart/Makefile.am	11 Sep 2003 08:56:33 -0000	1.100
+++ kchart/Makefile.am	9 Nov 2003 04:04:41 -0000
@@ -1,5 +1,8 @@
 INCLUDES  = $(KOFFICE_INCLUDES) -I$(srcdir)/kdchart -I$(top_srcdir)/interfaces -I$(interfacedir) $(all_includes)
 
+bin_PROGRAMS =
+lib_LTLIBRARIES =
+
 ## The part
 kde_module_LTLIBRARIES = libkchartpart.la
 libkchartpart_la_SOURCES = kchart_part.cc \
@@ -39,19 +42,10 @@
 			  kdchart/libkdchart.la			
 
 ## The kdeinit loadable module
-lib_LTLIBRARIES = kchart.la
+kdeinit_LTLIBRARIES = kchart.la
 kchart_la_SOURCES = main.cc
 kchart_la_LDFLAGS = $(all_libraries) -avoid-version -module
 kchart_la_LIBADD = $(LIB_KOFFICECORE)
-
-## The executable
-bin_PROGRAMS = kchart
-kchart_SOURCES = dummy.cc
-kchart_LDFLAGS = $(all_libraries) $(KDE_RPATH)
-kchart_LDADD = kchart.la
-
-dummy.cc:
-	echo > dummy.cc
 
 noinst_HEADERS =  kchart_part.h \
 		kchart_view.h kchart_factory.h \
Index: kchart/main.cc
===================================================================
RCS file: /home/kde/koffice/kchart/main.cc,v
retrieving revision 1.18
diff -u -u -r1.18 main.cc
--- kchart/main.cc	23 Jan 2003 09:10:23 -0000	1.18
+++ kchart/main.cc	9 Nov 2003 04:04:41 -0000
@@ -28,7 +28,7 @@
 	KCmdLineLastOption
 };
 
-int main( int argc, char **argv )
+extern "C" int kdemain( int argc, char **argv )
 {
     KCmdLineArgs::init( argc, argv, newKChartAboutData());
     KCmdLineArgs::addCmdLineOptions( options );
Index: kformula/Makefile.am
===================================================================
RCS file: /home/kde/koffice/kformula/Makefile.am,v
retrieving revision 1.71
diff -u -u -r1.71 Makefile.am
--- kformula/Makefile.am	18 Jan 2003 20:01:07 -0000	1.71
+++ kformula/Makefile.am	9 Nov 2003 04:04:41 -0000
@@ -11,20 +11,13 @@
 libkformulapart_la_LIBADD = $(LIB_KOFFICEUI) $(LIB_KFORMULA) $(LIB_KOTEXT)
 libkformulapart_la_METASOURCES = AUTO
 
-## The kdeinit loadable module
-lib_LTLIBRARIES = kformulamain.la
-kformulamain_la_SOURCES = main.cc
-kformulamain_la_LDFLAGS = $(all_libraries) -module -avoid-version
-kformulamain_la_LIBADD = $(LIB_KOFFICECORE)
+bin_PROGRAMS =
+lib_LTLIBRARIES =
+kdeinit_LTLIBRARIES = kformula.la
 
-## The executable
-bin_PROGRAMS = kformula
-kformula_SOURCES = dummy.cc
-kformula_LDFLAGS = $(all_libraries) $(KDE_RPATH)
-kformula_LDADD = kformulamain.la
-
-dummy.cc:
-	echo > dummy.cc
+kformula_la_SOURCES = main.cc
+kformula_la_LDFLAGS = $(all_libraries) -module -avoid-version
+kformula_la_LIBADD = $(LIB_KOFFICECORE)
 
 noinst_HEADERS =  kformula_doc.h \
 		  kformula_view.h kformula_factory.h \
Index: kformula/main.cc
===================================================================
RCS file: /home/kde/koffice/kformula/main.cc,v
retrieving revision 1.11
diff -u -u -r1.11 main.cc
--- kformula/main.cc	23 Jan 2003 09:10:23 -0000	1.11
+++ kformula/main.cc	9 Nov 2003 04:04:41 -0000
@@ -31,7 +31,7 @@
         KCmdLineLastOption
 };
 
-int main( int argc, char **argv )
+extern "C" int kdemain( int argc, char **argv )
 {
     KCmdLineArgs::init( argc, argv, newKFormulaAboutData() );
     KCmdLineArgs::addCmdLineOptions( options );
Index: kivio/kiviopart/Makefile.am
===================================================================
RCS file: /home/kde/koffice/kivio/kiviopart/Makefile.am,v
retrieving revision 1.43
diff -u -u -r1.43 Makefile.am
--- kivio/kiviopart/Makefile.am	30 Jun 2003 22:04:53 -0000	1.43
+++ kivio/kiviopart/Makefile.am	9 Nov 2003 04:04:41 -0000
@@ -11,18 +11,19 @@
 
 
 kde_module_LTLIBRARIES = libkiviopart.la
+lib_LTLIBRARIES = libkiviocommon.la
 
-libkiviopart_la_LDFLAGS = -avoid-version -module $(all_libraries) -no-undefined
+libkiviocommon_la_LDFLAGS = $(all_libraries) -no-undefined
 
-libkiviopart_la_LIBADD = \
+libkiviocommon_la_LIBADD = \
 	$(top_builddir)/kivio/kiviopart/tools/libtools.la\
 	$(top_builddir)/kivio/kiviopart/kiviosdk/libkiviosdk.la\
 	$(top_builddir)/kivio/kiviopart/ui/libui.la \
 	$(LIB_KOFFICEUI) $(LIB_KOPAINTER) $(LIB_KOTEXT) $(LIBPYTHON) $(PYTHONLIB)
 
-libkiviopart_la_METASOURCES = AUTO
+libkiviocommon_la_METASOURCES = AUTO
 
-libkiviopart_la_SOURCES = kivio_canvas.cpp kivio_doc.cpp kivio_factory.cpp \
+libkiviocommon_la_SOURCES = kivio_canvas.cpp kivio_doc.cpp kivio_factory.cpp \
 	kivio_tabbar.cpp kivio_view.cpp kivio_map.cpp kivio_page.cpp \
 	kivio_dlg_pageshow.cpp kivio_zoomaction.cpp kivio_icon_view.cpp \
 	kivio_config.cpp kivio_spawner_drag.cpp kivio_intra_stencil_data.cpp \
@@ -36,32 +37,22 @@
 	kiviostencilsetaction.cpp kivioarrowheadaction.cpp KIvioStencilIface.skel \
 	KIvioStencilIface.cc kiviodragobject.cpp
 
+libkiviopart_la_LDFLAGS = -avoid-version $(all_libraries) -module
+libkiviopart_la_LIBADD  = libkiviocommon.la
+libkiviopart_la_SOURCES = dummy.cpp
 
-lib_LTLIBRARIES = kivio.la
+dummy.cpp:
+	echo > dummy.cpp
 
-kivio_la_SOURCES = main.cpp
+bin_PROGRAMS =
+kdeinit_LTLIBRARIES = kivio.la
 
+kivio_la_SOURCES = main.cpp
 kivio_la_LDFLAGS = $(all_libraries) -module -avoid-version
-
 kivio_la_LIBADD = ../../lib/kofficecore/libkofficecore.la
 
-
-bin_PROGRAMS = kivio
-
-kivio_SOURCES = dummy.cpp
-
-kivio_LDFLAGS = $(all_libraries) $(KDE_RPATH)
-
-kivio_LDADD = kivio.la
-
-
-dummy.cpp:
-	echo > dummy.cpp
-
-
 kdelnk_DATA = kivio.desktop
 kdelnkdir = $(kde_appsdir)/Office
-
 
 rc_DATA = kivio.rc kivio_readonly.rc
 rcdir = $(kde_datadir)/kivio
Index: kivio/kiviopart/main.cpp
===================================================================
RCS file: /home/kde/koffice/kivio/kiviopart/main.cpp,v
retrieving revision 1.13
diff -u -u -r1.13 main.cpp
--- kivio/kiviopart/main.cpp	26 Jan 2003 13:26:21 -0000	1.13
+++ kivio/kiviopart/main.cpp	9 Nov 2003 04:04:41 -0000
@@ -30,7 +30,7 @@
   KCmdLineLastOption
 };
 
-int main( int argc, char **argv )
+extern "C" int kdemain( int argc, char **argv )
 {
   KCmdLineArgs::init( argc, argv, newKivioAboutData() );
   KCmdLineArgs::addCmdLineOptions( options );
Index: kivio/plugins/kivioconnectortool/Makefile.am
===================================================================
RCS file: /home/kde/koffice/kivio/plugins/kivioconnectortool/Makefile.am,v
retrieving revision 1.16
diff -u -u -r1.16 Makefile.am
--- kivio/plugins/kivioconnectortool/Makefile.am	19 Jul 2003 13:56:05 -0000	1.16
+++ kivio/plugins/kivioconnectortool/Makefile.am	9 Nov 2003 04:04:41 -0000
@@ -23,7 +23,7 @@
 		kivio_connector_cursor1.xpm
 
 libkivioconnectortool_la_LDFLAGS = -module $(KDE_PLUGIN)
-libkivioconnectortool_la_LIBADD = $(all_libraries) $(LIB_KOPAINTER) $(top_builddir)/kivio/kiviopart/libkiviopart.la
+libkivioconnectortool_la_LIBADD = $(all_libraries) $(LIB_KOPAINTER) $(top_builddir)/kivio/kiviopart/libkiviocommon.la
 
 rc_DATA = kivioconnectortool.rc
 rcdir = $(kde_datadir)/kivio/kpartplugins
Index: kivio/plugins/kivioconnectortool/straight_connector/Makefile.am
===================================================================
RCS file: /home/kde/koffice/kivio/plugins/kivioconnectortool/straight_connector/Makefile.am,v
retrieving revision 1.23
diff -u -u -r1.23 Makefile.am
--- kivio/plugins/kivioconnectortool/straight_connector/Makefile.am	6 May 2003 20:34:41 -0000	1.23
+++ kivio/plugins/kivioconnectortool/straight_connector/Makefile.am	9 Nov 2003 04:04:41 -0000
@@ -6,7 +6,7 @@
 straight_connector_la_LDFLAGS = -module $(KDE_PLUGIN) $(all_libraries)
 
 # Hack to make it work (dave m)
-straight_connector_la_LIBADD = $(LIB_KOPAINTER) $(top_builddir)/kivio/kiviopart/libkiviopart.la
+straight_connector_la_LIBADD = $(LIB_KOPAINTER) $(top_builddir)/kivio/kiviopart/libkiviocommon.la
 
 INCLUDES = -I$(top_srcdir)/kivio/kiviopart\
 	-I$(top_srcdir)/kivio/kiviopart/kiviosdk\
Index: kivio/plugins/kivioselecttool/Makefile.am
===================================================================
RCS file: /home/kde/koffice/kivio/plugins/kivioselecttool/Makefile.am,v
retrieving revision 1.12
diff -u -u -r1.12 Makefile.am
--- kivio/plugins/kivioselecttool/Makefile.am	6 May 2003 20:34:41 -0000	1.12
+++ kivio/plugins/kivioselecttool/Makefile.am	9 Nov 2003 04:04:41 -0000
@@ -18,7 +18,7 @@
 		tool_select.h\
 		kivioselecttool.rc
 libkivioselecttool_la_LDFLAGS = -module $(KDE_PLUGIN)
-libkivioselecttool_la_LIBADD = $(LIB_KOTEXT) $(LIB_KOPAINTER) $(all_libraries) $(top_builddir)/kivio/kiviopart/libkiviopart.la
+libkivioselecttool_la_LIBADD = $(LIB_KOTEXT) $(LIB_KOPAINTER) $(all_libraries) $(top_builddir)/kivio/kiviopart/libkiviocommon.la
 
 rc_DATA = kivioselecttool.rc
 rcdir = $(kde_datadir)/kivio/kpartplugins
Index: kivio/plugins/kiviosmlconnector/Makefile.am
===================================================================
RCS file: /home/kde/koffice/kivio/plugins/kiviosmlconnector/Makefile.am,v
retrieving revision 1.4
diff -u -u -r1.4 Makefile.am
--- kivio/plugins/kiviosmlconnector/Makefile.am	6 May 2003 20:34:41 -0000	1.4
+++ kivio/plugins/kiviosmlconnector/Makefile.am	9 Nov 2003 04:04:41 -0000
@@ -10,7 +10,7 @@
 METASOURCES = AUTO
 libkiviosmlconnector_la_SOURCES = plugin.cpp tool_connector.cpp
 libkiviosmlconnector_la_LDFLAGS = -module $(KDE_PLUGIN) $(all_libraries)
-libkiviosmlconnector_la_LIBADD = $(LIB_KOPAINTER) $(top_builddir)/kivio/kiviopart/libkiviopart.la
+libkiviosmlconnector_la_LIBADD = $(LIB_KOPAINTER) $(top_builddir)/kivio/kiviopart/libkiviocommon.la
 
 
 
Index: kivio/plugins/kiviosmlconnector/sml_connector/Makefile.am
===================================================================
RCS file: /home/kde/koffice/kivio/plugins/kiviosmlconnector/sml_connector/Makefile.am,v
retrieving revision 1.5
diff -u -u -r1.5 Makefile.am
--- kivio/plugins/kiviosmlconnector/sml_connector/Makefile.am	6 May 2003 20:34:41 -0000	1.5
+++ kivio/plugins/kiviosmlconnector/sml_connector/Makefile.am	9 Nov 2003 04:04:41 -0000
@@ -9,7 +9,7 @@
 
 sml_connector_la_SOURCES = sml_connector.cpp
 sml_connector_la_LDFLAGS = -module $(KDE_PLUGIN) $(all_libraries)
-sml_connector_la_LIBADD = $(LIB_KOPAINTER) $(top_builddir)/kivio/kiviopart/libkiviopart.la
+sml_connector_la_LIBADD = $(LIB_KOPAINTER) $(top_builddir)/kivio/kiviopart/libkiviocommon.la
 
 stencildir = $(kde_datadir)/kivio/autoloadStencils
 stencil_DATA = sml_connector.xpm
Index: kivio/plugins/kiviotexttool/Makefile.am
===================================================================
RCS file: /home/kde/koffice/kivio/plugins/kiviotexttool/Makefile.am,v
retrieving revision 1.12
diff -u -u -r1.12 Makefile.am
--- kivio/plugins/kiviotexttool/Makefile.am	6 May 2003 20:34:41 -0000	1.12
+++ kivio/plugins/kiviotexttool/Makefile.am	9 Nov 2003 04:04:41 -0000
@@ -21,7 +21,7 @@
 		Text.sml\
 		Text.xpm
 libkiviotexttool_la_LDFLAGS = -module $(KDE_PLUGIN)
-libkiviotexttool_la_LIBADD = $(all_libraries) $(LIB_KOPAINTER) $(top_builddir)/kivio/kiviopart/libkiviopart.la
+libkiviotexttool_la_LIBADD = $(all_libraries) $(LIB_KOPAINTER) $(top_builddir)/kivio/kiviopart/libkiviocommon.la
 
 rc_DATA = kiviotexttool.rc
 rcdir = $(kde_datadir)/kivio/kpartplugins
Index: kivio/plugins/kiviozoomtool/Makefile.am
===================================================================
RCS file: /home/kde/koffice/kivio/plugins/kiviozoomtool/Makefile.am,v
retrieving revision 1.13
diff -u -u -r1.13 Makefile.am
--- kivio/plugins/kiviozoomtool/Makefile.am	6 May 2003 20:34:41 -0000	1.13
+++ kivio/plugins/kiviozoomtool/Makefile.am	9 Nov 2003 04:04:41 -0000
@@ -17,7 +17,7 @@
 		tool_zoom.cpp\
 		tool_zoom.h
 libkiviozoomtool_la_LDFLAGS = -module $(KDE_PLUGIN)
-libkiviozoomtool_la_LIBADD = $(LIB_KOTEXT) $(all_libraries) $(LIB_KOPAINTER) $(top_builddir)/kivio/kiviopart/libkiviopart.la
+libkiviozoomtool_la_LIBADD = $(LIB_KOTEXT) $(all_libraries) $(LIB_KOPAINTER) $(top_builddir)/kivio/kiviopart/libkiviocommon.la
 
 rc_DATA = kiviozoomtool.rc
 rcdir = $(kde_datadir)/kivio/kpartplugins
Index: koshell/Makefile.am
===================================================================
RCS file: /home/kde/koffice/koshell/Makefile.am,v
retrieving revision 1.26
diff -u -u -r1.26 Makefile.am
--- koshell/Makefile.am	8 Dec 2002 21:12:27 -0000	1.26
+++ koshell/Makefile.am	9 Nov 2003 04:04:41 -0000
@@ -1,20 +1,13 @@
 
 INCLUDES  = $(KOFFICE_INCLUDES) $(all_includes)
 
-## The kdeinit loadable module
-lib_LTLIBRARIES = koshell.la
+bin_PROGRAMS =
+lib_LTLIBRARIES =
+kdeinit_LTLIBRARIES = koshell.la
+
 koshell_la_SOURCES = koshell_main.cc koshell_shell.cc
 koshell_la_LDFLAGS = $(all_libraries) -avoid-version -module
 koshell_la_LIBADD = $(LIB_KOFFICEUI)
-
-## The executable
-bin_PROGRAMS = koshell
-koshell_SOURCES = dummy.cc
-koshell_LDFLAGS = $(all_libraries) $(KDE_RPATH)
-koshell_LDADD = koshell.la
-
-dummy.cc:
-	echo > dummy.cc
 
 noinst_HEADERS = koshell_shell.h
 
Index: koshell/koshell_main.cc
===================================================================
RCS file: /home/kde/koffice/koshell/koshell_main.cc,v
retrieving revision 1.26
diff -u -u -r1.26 koshell_main.cc
--- koshell/koshell_main.cc	20 Jan 2003 14:58:49 -0000	1.26
+++ koshell/koshell_main.cc	9 Nov 2003 04:04:41 -0000
@@ -30,7 +30,7 @@
 static const char* description=I18N_NOOP("KOffice Workspace");
 static const char* version=VERSION;
 
-int main( int argc, char **argv )
+extern "C" int kdemain( int argc, char **argv )
 {
   KAboutData * aboutData = new KAboutData( "koshell", I18N_NOOP("KOffice Workspace"),
      version, description, KAboutData::License_GPL,
Index: kpresenter/Makefile.am
===================================================================
RCS file: /home/kde/koffice/kpresenter/Makefile.am,v
retrieving revision 1.208
diff -u -u -r1.208 Makefile.am
--- kpresenter/Makefile.am	11 Jul 2003 11:23:51 -0000	1.208
+++ kpresenter/Makefile.am	9 Nov 2003 04:04:41 -0000
@@ -53,20 +53,13 @@
 libkpresenterpart_la_LIBADD = autoformEdit/libaf.la $(LIB_KOFFICEUI) $(LIB_KOTEXT) -lkspell $(LIBKOSPELL) -lartskde -lkdefx
 libkpresenterpart_la_METASOURCES = AUTO
 
-## The kdeinit loadable module
-lib_LTLIBRARIES = kpresenter.la
+lib_LTLIBRARIES =
+bin_PROGRAMS =
+kdeinit_LTLIBRARIES = kpresenter.la
+
 kpresenter_la_SOURCES = main.cc
 kpresenter_la_LDFLAGS = $(all_libraries) -avoid-version -module
 kpresenter_la_LIBADD = $(LIB_KOFFICECORE)
-
-## The executable
-bin_PROGRAMS = kpresenter
-kpresenter_SOURCES = dummy.cc
-kpresenter_LDFLAGS = $(all_libraries) $(KDE_RPATH)
-kpresenter_LDADD = kpresenter.la
-
-dummy.cc:
-	echo > dummy.cc
 
 noinst_HEADERS = koPointArray.h kprcanvas.h kprpage.h kpresenter_doc.h kpresenter_view.h \
 		backdia.h styledia.h \
Index: kpresenter/main.cc
===================================================================
RCS file: /home/kde/koffice/kpresenter/main.cc,v
retrieving revision 1.24
diff -u -u -r1.24 main.cc
--- kpresenter/main.cc	25 Apr 2003 13:52:27 -0000	1.24
+++ kpresenter/main.cc	9 Nov 2003 04:04:41 -0000
@@ -32,7 +32,7 @@
 	KCmdLineLastOption
 };
 
-int main( int argc, char **argv )
+extern "C" int kdemain( int argc, char **argv )
 {
     KCmdLineArgs::init( argc, argv, newKPresenterAboutData() );
     KCmdLineArgs::addCmdLineOptions( options );
Index: kspread/Makefile.am
===================================================================
RCS file: /home/kde/koffice/kspread/Makefile.am,v
retrieving revision 1.193
diff -u -u -r1.193 Makefile.am
--- kspread/Makefile.am	13 Oct 2003 21:35:48 -0000	1.193
+++ kspread/Makefile.am	9 Nov 2003 04:04:41 -0000
@@ -3,10 +3,11 @@
 	$(KOSCRIPT_INCLUDES) -I$(top_srcdir)/interfaces \
 	 -I$(top_srcdir)/kchart/kdchart $(all_includes)
 
+lib_LTLIBRARIES = libkspreadcommon.la
 kde_module_LTLIBRARIES = libkspreadpart.la
 
 ## The part
-libkspreadpart_la_SOURCES = kspread_locale.cc \
+libkspreadcommon_la_SOURCES = kspread_locale.cc \
 	kspread_dlg_formula.cc digest.cc \
 	kspread_functions.cc kspread_dlg_csv.cc \
 	kspread_canvas.cc kspread_functions_helper.cc \
@@ -45,26 +46,25 @@
 	kspread_selection.cc kspread_dlg_paperlayout.cc kspread_format.cc \
         kspread_sheetprint.cc
 
-libkspreadpart_la_LDFLAGS = $(all_libraries) $(KDE_PLUGIN)
-libkspreadpart_la_LIBADD = $(top_builddir)/interfaces/libkochart.la \
+libkspreadcommon_la_LDFLAGS = $(all_libraries) $(KDE_RPATH) -no-undefined
+libkspreadcommon_la_LIBADD = $(top_builddir)/interfaces/libkochart.la \
 	$(LIB_KOSCRIPT) $(LIB_KOFFICEUI) $(LIB_KOTEXT) -lkspell
 
+libkspreadpart_la_LDFLAGS = $(all_libraries) $(KDE_PLUGIN)
+libkspreadpart_la_LIBADD  = libkspreadcommon.la
+libkspreadpart_la_SOURCES = dummy.cpp
+
+dummy.cpp:
+	echo > dummy.cpp
+
 METASOURCES = AUTO
 
-## The kdeinit loadable module
-lib_LTLIBRARIES =  kspread.la
+bin_PROGRAMS =
+kdeinit_LTLIBRARIES = kspread.la
+
 kspread_la_SOURCES = main.cc
 kspread_la_LDFLAGS = $(all_libraries) -avoid-version -module
 kspread_la_LIBADD = $(LIB_KOFFICECORE)
-
-## The executable
-bin_PROGRAMS = kspread
-kspread_SOURCES = dummy.cc
-kspread_LDFLAGS = $(all_libraries) $(KDE_RPATH)
-kspread_LDADD = kspread.la
-
-dummy.cc:
-	echo > dummy.cc
 
 noinst_HEADERS = kspread_doc.h kspread_functions.h kspread_functions_helper.h \
 	kspread_undo.h kspread_view.h kspread_tabbar.h \
Index: kspread/main.cc
===================================================================
RCS file: /home/kde/koffice/kspread/main.cc,v
retrieving revision 1.19
diff -u -u -r1.19 main.cc
--- kspread/main.cc	23 Jan 2003 09:10:24 -0000	1.19
+++ kspread/main.cc	9 Nov 2003 04:04:41 -0000
@@ -30,7 +30,7 @@
         KCmdLineLastOption
 };
 
-int main( int argc, char **argv )
+extern "C" int kdemain( int argc, char **argv )
 {
 
     KCmdLineArgs::init( argc, argv, newKSpreadAboutData() );
Index: kspread/plugins/calculator/Makefile.am
===================================================================
RCS file: /home/kde/koffice/kspread/plugins/calculator/Makefile.am,v
retrieving revision 1.15
diff -u -u -r1.15 Makefile.am
--- kspread/plugins/calculator/Makefile.am	1 Oct 2002 04:53:42 -0000	1.15
+++ kspread/plugins/calculator/Makefile.am	9 Nov 2003 04:04:41 -0000
@@ -9,7 +9,7 @@
 			main.cpp \
 			stats.cpp
 libkspreadcalc_la_LDFLAGS = -avoid-version -module -no-undefined
-libkspreadcalc_la_LIBADD = $(all_libraries) ../../libkspreadpart.la
+libkspreadcalc_la_LIBADD = $(all_libraries) ../../libkspreadcommon.la
 
 METASOURCES = AUTO
 
Index: kugar/kudesigner/Makefile.am
===================================================================
RCS file: /home/kde/koffice/kugar/kudesigner/Makefile.am,v
retrieving revision 1.19
diff -u -u -r1.19 Makefile.am
--- kugar/kudesigner/Makefile.am	30 May 2003 20:07:52 -0000	1.19
+++ kugar/kudesigner/Makefile.am	9 Nov 2003 04:04:42 -0000
@@ -2,11 +2,12 @@
 
 INCLUDES = -I$(srcdir)/../kudesigner_lib -I$(srcdir)/../lib $(KOFFICE_INCLUDES) $(all_includes)
 
+bin_PROGRAMS =
 
 ## The kdeinit loadable module
-lib_LTLIBRARIES = libkudesignercore.la kudesigner.la
+lib_LTLIBRARIES = libkudesignercore.la
 libkudesignercore_la_SOURCES = kudesigner_view.cpp kudesigner_factory.cpp kudesigner_doc.cpp
-libkudesignercore_la_LDFLAGS = -avoid-version -module $(all_libraries) $(KDE_RPATH)
+libkudesignercore_la_LDFLAGS = -avoid-version $(all_libraries) $(KDE_RPATH)
 libkudesignercore_la_LIBADD = ../kudesigner_lib/libkudesignercommon.la $(LIB_KOFFICEUI) $(LIB_KPARTS) $(LIB_KOFFICECORE) $(LIB_KIO) $(LIB_KDEPRINT)
 
 ## The part
@@ -15,19 +16,10 @@
 libkudesignerpart_la_LDFLAGS = $(all_libraries) $(KDE_PLUGIN) $(KDE_RPATH)
 libkudesignerpart_la_LIBADD = libkudesignercore.la $(LIB_KOFFICEUI) $(LIB_KPARTS) $(LIB_KOFFICECORE)
 
-## The kdeinit loadable module
+kdeinit_LTLIBRARIES = kudesigner.la
 kudesigner_la_SOURCES = main.cpp
 kudesigner_la_LDFLAGS = $(all_libraries) -module -avoid-version
 kudesigner_la_LIBADD = $(LIB_KOFFICECORE)
-
-## The executable
-bin_PROGRAMS = kudesigner
-kudesigner_SOURCES = dummy.cc
-kudesigner_LDFLAGS = $(all_libraries) $(KDE_RPATH)
-kudesigner_LDADD = kudesigner.la
-
-dummy.cc:
-	echo > dummy.cc
 
 appsdir = $(kde_appsdir)/Office
 apps_DATA = kudesigner.desktop
Index: kugar/kudesigner/main.cpp
===================================================================
RCS file: /home/kde/koffice/kugar/kudesigner/main.cpp,v
retrieving revision 1.11
diff -u -u -r1.11 main.cpp
--- kugar/kudesigner/main.cpp	28 May 2003 20:35:18 -0000	1.11
+++ kugar/kudesigner/main.cpp	9 Nov 2003 04:04:42 -0000
@@ -32,7 +32,7 @@
 	KCmdLineLastOption
 };
 
-int main( int argc, char **argv )
+extern "C" int kdemain( int argc, char **argv )
 {
     KLocale::setMainCatalogue("kugar");
     KCmdLineArgs::init( argc, argv, newKudesignerAboutData() );
Index: kugar/part/Makefile.am
===================================================================
RCS file: /home/kde/koffice/kugar/part/Makefile.am,v
retrieving revision 1.7
diff -u -u -r1.7 Makefile.am
--- kugar/part/Makefile.am	25 Nov 2002 21:08:11 -0000	1.7
+++ kugar/part/Makefile.am	9 Nov 2003 04:04:42 -0000
@@ -24,18 +24,11 @@
 
 CLEANFILES = kugarpart_meta_unload.cpp
 
+bin_PROGRAMS =
+lib_LTLIBRARIES =
 
-## The kdeinit loadable module
-lib_LTLIBRARIES = kugar.la
+kdeinit_LTLIBRARIES = kugar.la
 kugar_la_SOURCES = main.cpp
 kugar_la_LDFLAGS = $(all_libraries) -module -avoid-version
 kugar_la_LIBADD = $(LIB_KOFFICECORE)
 
-## The executable
-bin_PROGRAMS = kugar
-kugar_SOURCES = dummy.cc
-kugar_LDFLAGS = $(all_libraries) $(KDE_RPATH)
-kugar_LDADD = kugar.la
-
-dummy.cc:
-	echo > dummy.cc
Index: kugar/part/main.cpp
===================================================================
RCS file: /home/kde/koffice/kugar/part/main.cpp,v
retrieving revision 1.3
diff -u -u -r1.3 main.cpp
--- kugar/part/main.cpp	15 May 2003 12:31:45 -0000	1.3
+++ kugar/part/main.cpp	9 Nov 2003 04:04:42 -0000
@@ -28,7 +28,7 @@
   // INSERT YOUR COMMANDLINE OPTIONS HERE
 };
 
-int main(int argc, char *argv[])
+extern "C" int kdemain(int argc, char *argv[])
 {
 
 	KCmdLineArgs::init( argc, argv, newKugarAboutData() );
Index: kugar/shell/Makefile.am
===================================================================
RCS file: /home/kde/koffice/kugar/shell/Makefile.am,v
retrieving revision 1.5
diff -u -u -r1.5 Makefile.am
--- kugar/shell/Makefile.am	29 Mar 2001 12:32:02 -0000	1.5
+++ kugar/shell/Makefile.am	9 Nov 2003 04:04:42 -0000
@@ -1,9 +1,13 @@
 INCLUDES = -I$(srcdir)/../lib $(KOFFICE_INCLUDES) $(all_includes)
 
-bin_PROGRAMS = kugar
-kugar_SOURCES = main.cpp report.cpp
-kugar_LDFLAGS = $(all_libraries) $(KDE_RPATH)
-kugar_LDADD = ../lib/libkugar.la $(LIB_KIO) $(LIB_KDEPRINT)
+bin_PROGRAMS =
+lib_LTLIBRARIES =
+
+kdeinit_LTLIBRARIES = kugar.la
+
+kugar_la_SOURCES = main.cpp report.cpp
+kugar_la_LDFLAGS = $(all_libraries) $(KDE_RPATH) -module -avoid-version
+kugar_la_LIBADD = ../lib/libkugar.la $(LIB_KIO) $(LIB_KDEPRINT)
 
 kugar_METASOURCES = AUTO
 
Index: kugar/shell/main.cpp
===================================================================
RCS file: /home/kde/koffice/kugar/shell/main.cpp,v
retrieving revision 1.9
diff -u -u -r1.9 main.cpp
--- kugar/shell/main.cpp	21 Apr 2003 11:11:18 -0000	1.9
+++ kugar/shell/main.cpp	9 Nov 2003 04:04:42 -0000
@@ -16,7 +16,7 @@
 #include "report.h"
 
 
-int main(int argc,char **argv)
+extern "C" int kdemain(int argc,char **argv)
 {
 	KAboutData about("kugar", I18N_NOOP("Kugar"), VERSION,
 		I18N_NOOP("A template driven report viewer for XML data."),
Index: kword/Makefile.am
===================================================================
RCS file: /home/kde/koffice/kword/Makefile.am,v
retrieving revision 1.206
diff -u -u -r1.206 Makefile.am
--- kword/Makefile.am	22 Sep 2003 21:13:47 -0000	1.206
+++ kword/Makefile.am	9 Nov 2003 04:04:42 -0000
@@ -46,10 +46,11 @@
 
 
 libkwordpart_la_LDFLAGS = $(all_libraries) $(KDE_PLUGIN)
-libkwordpart_la_LIBADD = $(LIB_KOFFICEUI) $(LIB_KOTEXT) ../lib/kformula/libkformula.la -lkspell $(LIBKOSPELL) ./libkwmailmerge_interface.la
+libkwordpart_la_LIBADD = $(LIB_KOFFICEUI) $(LIB_KOTEXT) ../lib/kformula/libkformula.la -lkspell $(LIBKOSPELL) libkwmailmerge_interface.la
 METASOURCES = AUTO
 
-lib_LTLIBRARIES = kword.la libkwmailmerge_interface.la
+lib_LTLIBRARIES = libkwmailmerge_interface.la
+bin_PROGRAMS =
 
 ## The mailmerge interface library
 libkwmailmerge_interface_la_SOURCES = mailmerge_interface.cc mailmerge_interface.skel
@@ -61,19 +62,10 @@
 kwmailmerge_servicetype_DATA=kwmailmerge.desktop
 kwmailmerge_servicetypedir=$(kde_servicetypesdir)
 
-## The kdeinit loadable module
+kdeinit_LTLIBRARIES = kword.la
 kword_la_SOURCES = main.cc
 kword_la_LDFLAGS = $(all_libraries) -avoid-version -module
 kword_la_LIBADD = ../lib/kofficecore/libkofficecore.la
-
-## The executable
-bin_PROGRAMS = kword
-kword_SOURCES = dummy.cc
-kword_LDFLAGS = $(all_libraries) $(KDE_RPATH)
-kword_LDADD = kword.la
-
-dummy.cc:
-	echo > dummy.cc
 
 kdelnk_DATA = kword.desktop
 kdelnkdir = $(kde_appsdir)/Office
Index: kword/main.cc
===================================================================
RCS file: /home/kde/koffice/kword/main.cc,v
retrieving revision 1.23
diff -u -u -r1.23 main.cc
--- kword/main.cc	23 Jan 2003 09:10:24 -0000	1.23
+++ kword/main.cc	9 Nov 2003 04:04:42 -0000
@@ -28,7 +28,7 @@
 	KCmdLineLastOption
 };
 
-int main( int argc, char **argv )
+extern "C" int kdemain( int argc, char **argv )
 {
     KCmdLineArgs::init( argc, argv, newKWordAboutData());
     KCmdLineArgs::addCmdLineOptions( options );
Index: lib/configure.in.in
===================================================================
RCS file: /home/kde/koffice/lib/configure.in.in,v
retrieving revision 1.23
diff -u -u -r1.23 configure.in.in
--- lib/configure.in.in	9 Apr 2003 17:59:46 -0000	1.23
+++ lib/configure.in.in	9 Nov 2003 04:04:42 -0000
@@ -14,7 +14,7 @@
 AC_SUBST(LIB_KWMF, '$(top_builddir)/lib/kwmf/libkwmf.la')
 AC_SUBST(LIB_KOWMF, '$(top_builddir)/lib/kwmf/libkowmf.la')
 AC_SUBST(LIB_KFORMULA, '$(top_builddir)/lib/kformula/libkformula.la')
-AC_SUBST(LIB_KOSCRIPT, '$(top_builddir)/lib/koscript/libkoscript.la')
+AC_SUBST(LIB_KOSCRIPT, '$(top_builddir)/lib/koscript/libkoscriptcommon.la')
 
 AC_SUBST(interfacedir, '$(top_srcdir)/lib/interfaces')
 AC_SUBST(KOFFICE_LIBS, '$(LIB_KOFFICEUI) $(LIB_KOFFICECORE) $(LIB_KSTORE)')
Index: lib/koscript/Makefile.am
===================================================================
RCS file: /home/kde/koffice/lib/koscript/Makefile.am,v
retrieving revision 1.8
diff -u -u -r1.8 Makefile.am
--- lib/koscript/Makefile.am	18 May 2002 07:58:18 -0000	1.8
+++ lib/koscript/Makefile.am	9 Nov 2003 04:04:42 -0000
@@ -4,19 +4,22 @@
 
 ####### Files
 
-lib_LTLIBRARIES = libkoscript.la
-libkoscript_la_SOURCES = koscript_locale.cc \
+lib_LTLIBRARIES = libkoscriptcommon.la
+bin_PROGRAMS =
+
+kdeinit_LTLIBRARIES = koscript.la
+
+libkoscriptcommon_la_SOURCES = koscript_locale.cc \
 		  koscript_eval.cc koscript_func.cc koscript_property.cc koscript_method.cc \
 		  koscript_context.cc koscript_value.cc koscript_parser.cc koscript_parsenode.cc yacc.cc \
                   scanner.cc koscript.cc koscript_util.cc koscript_struct.cc
-libkoscript_la_METASOURCES = AUTO
-libkoscript_la_LDFLAGS = -version-info 2:0:0 -no-undefined $(all_libraries)
-libkoscript_la_LIBADD = $(LIB_KDEUI)
+libkoscriptcommon_la_METASOURCES = AUTO
+libkoscriptcommon_la_LDFLAGS = -version-info 2:0:0 -no-undefined $(all_libraries)
+libkoscriptcommon_la_LIBADD = $(LIB_KDEUI)
 
-bin_PROGRAMS = koscript
-koscript_SOURCES = main.cc
-koscript_LDADD = ./libkoscript.la $(all_libraries)
-koscript_LDFLAGS = $(KDE_RPATH)
+koscript_la_SOURCES = main.cc
+koscript_la_LIBADD = libkoscriptcommon.la
+koscript_la_LDFLAGS = $(KDE_RPATH) $(all_libraries) -module -avoid-version
 
 noinst_HEADERS = koscript_locale.h \
 		 koscript_eval.h koscript_func.h koscript_context.h koscript_value.h \
Index: lib/koscript/main.cc
===================================================================
RCS file: /home/kde/koffice/lib/koscript/main.cc,v
retrieving revision 1.12
diff -u -u -r1.12 main.cc
--- lib/koscript/main.cc	23 Jan 2003 09:10:24 -0000	1.12
+++ lib/koscript/main.cc	9 Nov 2003 04:04:42 -0000
@@ -36,7 +36,7 @@
   KCmdLineLastOption
 };
 
-int main( int argc, char** argv )
+extern "C" int kdemain( int argc, char** argv )
 {
   const char *appName = (argc > 1) ? argv[1] : "koscript";
   KCmdLineArgs::init(argc, argv, appName,
Index: tools/converter/Makefile.am
===================================================================
RCS file: /home/kde/koffice/tools/converter/Makefile.am,v
retrieving revision 1.2
diff -u -u -r1.2 Makefile.am
--- tools/converter/Makefile.am	29 May 2002 07:37:33 -0000	1.2
+++ tools/converter/Makefile.am	9 Nov 2003 04:04:42 -0000
@@ -1,9 +1,12 @@
 INCLUDES = $(KOFFICE_INCLUDES) $(all_includes)
 
-bin_PROGRAMS = koconverter
-koconverter_SOURCES = koconverter.cc
-koconverter_LDFLAGS = $(all_libraries) $(KDE_RPATH)
-koconverter_LDADD = $(top_builddir)/lib/kofficecore/libkofficecore.la
+bin_PROGRAMS =
+lib_LTLIBRARIES =
+
+kdeinit_LTLIBRARIES = koconverter.la
+koconverter_la_SOURCES = koconverter.cc
+koconverter_la_LDFLAGS = $(all_libraries) $(KDE_RPATH) -module -avoid-version
+koconverter_la_LIBADD = $(top_builddir)/lib/kofficecore/libkofficecore.la
 
 METASOURCES = AUTO
 
Index: tools/converter/koconverter.cc
===================================================================
RCS file: /home/kde/koffice/tools/converter/koconverter.cc,v
retrieving revision 1.8
diff -u -u -r1.8 koconverter.cc
--- tools/converter/koconverter.cc	13 Oct 2003 21:35:48 -0000	1.8
+++ tools/converter/koconverter.cc	9 Nov 2003 04:04:42 -0000
@@ -58,7 +58,7 @@
     //kdDebug() << "ProgressObject::slotProgress " << progress << endl;
 }
 
-int main( int argc, char **argv )
+extern "C" int kdemain( int argc, char **argv )
 {
     KAboutData aboutData( "koconverter", I18N_NOOP("KOConverter"), "1.1",
                           I18N_NOOP("KOffice Document Converter"),
Index: tools/kthesaurus/Makefile.am
===================================================================
RCS file: /home/kde/koffice/tools/kthesaurus/Makefile.am,v
retrieving revision 1.6
diff -u -u -r1.6 Makefile.am
--- tools/kthesaurus/Makefile.am	9 Jun 2002 01:22:54 -0000	1.6
+++ tools/kthesaurus/Makefile.am	9 Nov 2003 04:04:42 -0000
@@ -1,11 +1,13 @@
 
 INCLUDES = $(all_includes)
 
-bin_PROGRAMS = kthesaurus
+bin_PROGRAMS =
+lib_LTLIBRARIES =
+kdeinit_LTLIBRARIES = kthesaurus.la
 
-kthesaurus_LDFLAGS = $(all_libraries) $(KDE_RPATH)
-kthesaurus_SOURCES = main.cc
-kthesaurus_LDADD = $(LIB_KSYCOCA)
+kthesaurus_la_LDFLAGS = $(all_libraries) $(KDE_RPATH) -module -avoid-version
+kthesaurus_la_SOURCES = main.cc
+kthesaurus_la_LIBADD = $(LIB_KSYCOCA)
  
 appsdir = $(kde_appsdir)/Office
 apps_DATA = KThesaurus.desktop
Index: tools/kthesaurus/main.cc
===================================================================
RCS file: /home/kde/koffice/tools/kthesaurus/main.cc,v
retrieving revision 1.7
diff -u -u -r1.7 main.cc
--- tools/kthesaurus/main.cc	23 Jan 2003 09:10:24 -0000	1.7
+++ tools/kthesaurus/main.cc	9 Nov 2003 04:04:42 -0000
@@ -32,7 +32,7 @@
 	KCmdLineLastOption
 };
 
-int main(int argc, char **argv)
+extern "C" int kdemain(int argc, char **argv)
 {
 
 	KAboutData aboutData("kthesaurus", I18N_NOOP("KThesaurus"), "1.0",
