diff -uNbr kvirc-3.0.0-beta1/admin/acinclude.m4.in kvirc-3.0.0-beta1-patched/admin/acinclude.m4.in
--- kvirc-3.0.0-beta1/admin/acinclude.m4.in	Sun May 26 21:29:34 2002
+++ kvirc-3.0.0-beta1-patched/admin/acinclude.m4.in	Sat Dec 21 18:28:41 2002
@@ -4,6 +4,9 @@
 
 AC_DEFUN(AC_SS_INIT_VARIABLES,
 [
+	FLAT="-undefined suppress -flat_namespace"
+	AC_SUBST(FLAT)
+
 	SS_BUILD_FLAGS=""
 	SS_BUILD_NUMBER=""
 
@@ -95,6 +98,7 @@
 AC_DEFUN(AC_SS_FIND_FILE_PATH,
 [
 	$3="FAILED"
+	for ext in so dylib; do
 	for a_dir in $2; do
 		for a_file in $1; do
 			if test -r "$a_dir/$a_file"; then
@@ -103,12 +107,14 @@
 			fi
 		done
 	done	
+	done	
 ])
 
 AC_DEFUN(AC_SS_FIND_FILE_PATH_EXT,
 [
 	$4="FAILED"
 
+	for ext in so dylib; do
 	for a_dir in $2; do
 		for a_semidir in $3; do
 			for a_file in $1; do
@@ -119,6 +125,7 @@
 			done
 		done
 	done
+	done
 ])
 
 dnl # // ##################################################################################################################
@@ -1033,7 +1040,7 @@
 		AC_MSG_CHECKING([for the Qt header files])
 
 		ss_qt_include_test_dirs="$SS_QT_GENERAL_DIR/include /include /usr/include /usr/local/include /usr/qt/include /usr/local/qt/include /usr/X11R6/include /usr/lib/qt/include /usr/lib/qt /usr/include/X11"
-		test -n "$QTDIR" && ss_qt_include_test_dirs="$QTDIR/include $QTDIR $ss_qt_include_test_dirs"
+		test -n "$QTDIR" && ss_qt_include_test_dirs="$QTDIR/include/qt $QTDIR/include $QTDIR $ss_qt_include_test_dirs"
 		ss_qt_include_test_dirs="$ss_qt_include_test_dirs /usr/include/qt /usr/local/include/qt /include/qt /usr/X11R6/include/qt /usr/build/qt/include"
 		ss_qt_include_test_dirs="$ss_qt_include_test_dirs /usr/include/qt$SS_QT_MAJOR_VERSION_REQUIRED /usr/local/include/qt$SS_QT_MAJOR_VERSION_REQUIRED"
 		ss_qt_include_test_dirs="$ss_qt_include_test_dirs /include/qt$SS_QT_MAJOR_VERSION_REQUIRED /usr/X11R6/include/qt$SS_QT_MAJOR_VERSION_REQUIRED"
@@ -1083,7 +1090,9 @@
 			ss_qt_library_test_dirs="$SS_QT_GENERAL_DIR/lib $ss_qt_library_test_dirs"
 		fi
 
-		AC_SS_FIND_FILE_PATH(lib$SS_QTLIB_NAME-mt.so.$SS_QT_MAJOR_VERSION_REQUIRED.$SS_QT_MINOR_VERSION_REQUIRED.$SS_QT_PATCH_LEVEL_REQUIRED libqt.so.$SS_QT_MAJOR_VERSION_REQUIRED.$SS_QT_MINOR_VERSION_REQUIRED lib$SS_QTLIB_NAME-mt.so.$SS_QT_MAJOR_VERSION_REQUIRED lib$SS_QTLIB_NAME-mt.so lib$SS_QTLIB_NAME.so.$SS_QT_MAJOR_VERSION_REQUIRED.$SS_QT_MINOR_VERSION_REQUIRED.$SS_QT_PATCH_LEVEL_REQUIRED libqt.so.$SS_QT_MAJOR_VERSION_REQUIRED.$SS_QT_MINOR_VERSION_REQUIRED lib$SS_QTLIB_NAME.so.$SS_QT_MAJOR_VERSION_REQUIRED lib$SS_QTLIB_NAME.a lib$SS_QTLIB_NAME.so,$ss_qt_library_test_dirs,SS_QT_LIBDIR)
+		for ext in so dylib; do
+			AC_SS_FIND_FILE_PATH(lib$SS_QTLIB_NAME-mt.$ext.$SS_QT_MAJOR_VERSION_REQUIRED.$SS_QT_MINOR_VERSION_REQUIRED.$SS_QT_PATCH_LEVEL_REQUIRED libqt.$ext.$SS_QT_MAJOR_VERSION_REQUIRED.$SS_QT_MINOR_VERSION_REQUIRED lib$SS_QTLIB_NAME-mt.$ext.$SS_QT_MAJOR_VERSION_REQUIRED lib$SS_QTLIB_NAME-mt.$ext lib$SS_QTLIB_NAME.$ext.$SS_QT_MAJOR_VERSION_REQUIRED.$SS_QT_MINOR_VERSION_REQUIRED.$SS_QT_PATCH_LEVEL_REQUIRED libqt.$ext.$SS_QT_MAJOR_VERSION_REQUIRED.$SS_QT_MINOR_VERSION_REQUIRED lib$SS_QTLIB_NAME.$ext.$SS_QT_MAJOR_VERSION_REQUIRED lib$SS_QTLIB_NAME.a lib$SS_QTLIB_NAME.$ext,$ss_qt_library_test_dirs,SS_QT_LIBDIR)
+		done
 
 		if test "$SS_QT_LIBDIR" = "FAILED"; then
 			AC_MSG_RESULT([FAILED])
@@ -1163,7 +1172,9 @@
 	AC_MSG_CHECKING([for the multithreaded version of Qt])
 
 	if test "$SS_QT_NOMT" = "NOTGIVEN"; then
-		AC_SS_FIND_FILE_PATH(lib$SS_QTLIB_NAME-mt.so.$SS_QT_MAJOR_VERSION_REQUIRED.$SS_QT_MINOR_VERSION_REQUIRED.$SS_QT_PATCH_LEVEL_REQUIRED libqt.so.$SS_QT_MAJOR_VERSION_REQUIRED.$SS_QT_MINOR_VERSION_REQUIRED lib$SS_QTLIB_NAME-mt.so.$SS_QT_MAJOR_VERSION_REQUIRED lib$SS_QTLIB_NAME-mt.so,$SS_QT_LIBDIR,SS_QTMTLIBDIR)
+		for ext in so dylib; do
+			AC_SS_FIND_FILE_PATH(lib$SS_QTLIB_NAME-mt.$ext.$SS_QT_MAJOR_VERSION_REQUIRED.$SS_QT_MINOR_VERSION_REQUIRED.$SS_QT_PATCH_LEVEL_REQUIRED libqt.$ext.$SS_QT_MAJOR_VERSION_REQUIRED.$SS_QT_MINOR_VERSION_REQUIRED lib$SS_QTLIB_NAME-mt.$ext.$SS_QT_MAJOR_VERSION_REQUIRED lib$SS_QTLIB_NAME-mt.$ext,$SS_QT_LIBDIR,SS_QTMTLIBDIR)
+		done
 		if test "$SS_QTMTLIBDIR" = "FAILED"; then
 			AC_MSG_RESULT([not found])
 		else
@@ -1371,7 +1382,9 @@
 			ss_kde_library_test_dirs="$SS_KDE_GENERAL_DIR/lib $ss_kde_library_test_dirs"
 		fi
 
-		AC_SS_FIND_FILE_PATH(libkdecore.so.3.0.0 libkdecore.so.3.0 libkdecore.a libkdecore.so,$ss_kde_library_test_dirs,SS_KDE_LIBDIR)
+		for ext in so dylib; do
+			AC_SS_FIND_FILE_PATH(libkdecore.$ext.3.0.0 libkdecore.$ext.3.0 libkdecore.a libkdecore.$ext,$ss_kde_library_test_dirs,SS_KDE_LIBDIR)
+		done
 
 		if test "$SS_KDE_LIBDIR" = "FAILED"; then
 			AC_MSG_RESULT("FAILED")
@@ -1402,7 +1415,7 @@
 		ss_save_LIBS="$LIBS"
 		CPPFLAGS="-I$SS_X_INCDIR -I$SS_QT_INCDIR -I$SS_KDE_INCDIR $CPPFLAGS $X_CFLAGS"
 		CXXFLAGS="-O2 -Wall -g $CXXFLAGS"
-		LIBS="-L$SS_KDE_LIBDIR -L$SS_X_LIBDIR -L$SS_QT_LIBDIR $X_ $X_EXTRA_LIBS $X_LIBS $LIBS -lkdecore -l$SS_QTLIB_NAME $SS_XFT_LIBLINK -lXext -lX11 -Wl,-rpath $SS_KDE_LIBDIR"
+		LIBS="-L$SS_KDE_LIBDIR -L$SS_X_LIBDIR -L$SS_QT_LIBDIR $X_ $X_EXTRA_LIBS $X_LIBS $LIBS -lkdecore -l$SS_QTLIB_NAME $SS_XFT_LIBLINK -lXext -lX11"
 		if test "$SS_LINK_TO_LIBDL" = "yes"; then
 			LIBS="$LIBS -ldl";
 		fi
@@ -1541,7 +1554,7 @@
 	SS_CPPFLAGS="$SS_X_CPPFLAGS -D_REENTRANT -DREENTRANT"
 	SS_INCDIRS="-I$SS_QT_INCDIR -I$SS_X_INCDIR"
 	SS_LDFLAGS="$SS_X_LDFLAGS"
-	SS_RPATH="-rpath $SS_QT_LIBDIR -rpath $SS_X_LIBDIR -rpath $libdir"
+	SS_RPATH="-L$SS_QT_LIBDIR -L$SS_X_LIBDIR -L$libdir"
 	SS_LIBDIRS="-L$SS_TOPSRCDIR/src/kvilib/build/ -L$SS_QT_LIBDIR -L$SS_X_LIBDIR"
 	SS_LIBLINK="-l$SS_QTLIB_NAME -lXext -lX11 $SS_THREAD_LIBLINK"
 
@@ -1571,7 +1584,7 @@
 
 	if test "$SS_KDE_REQUESTED" = "yes"; then
 		SS_INCDIRS="$SS_INCDIRS -I$SS_KDE_INCDIR"
-		SS_RPATH="$SS_RPATH -rpath $SS_KDE_LIBDIR"
+		SS_RPATH="$SS_RPATH -L$SS_KDE_LIBDIR"
 		SS_LIBDIRS="$SS_LIBDIRS -L$SS_KDE_LIBDIR"
 		SS_LIBLINK="-lkdecore -lkdeui -lkparts $SS_LIBLINK"
 		AC_DEFINE(COMPILE_KDE_SUPPORT)
@@ -1589,7 +1602,7 @@
 	if test -n "$SS_OTHER_LIBDIRS"; then
 		SS_LIBDIRS="$SS_LIBDIRS $SS_OTHER_LIBDIRS"
 		$SS_OTHER_LIBDIRS=`echo "$SS_OTHER_LIBDIRS" | sed -e s/-L//g`
-		SS_RPATH="$SS_RPATH -rpath $SS_OTHER_LIBDIRS"
+		SS_RPATH="$SS_RPATH -L$SS_OTHER_LIBDIRS"
 	fi
 
 	if test -n "$SS_OTHER_INCDIRS"; then
diff -uNbr kvirc-3.0.0-beta1/admin/libtool.m4.in kvirc-3.0.0-beta1-patched/admin/libtool.m4.in
--- kvirc-3.0.0-beta1/admin/libtool.m4.in	Thu Apr 11 20:41:06 2002
+++ kvirc-3.0.0-beta1-patched/admin/libtool.m4.in	Sat Dec 21 14:21:30 2002
@@ -453,6 +453,22 @@
   rm -rf conftest*
   ;;
 
+*-*-linux*)
+  # Test if the compiler is 64bit
+  echo 'int i;' > conftest.$ac_ext
+  if AC_TRY_EVAL(ac_compile); then
+    case `/usr/bin/file conftest.$ac_objext` in
+    *"ELF 32"*)
+      LINUX_64_MODE="32"
+      ;;
+    *"ELF 64"*)
+      LINUX_64_MODE="64"
+      ;;
+    esac
+  fi
+  rm -rf conftest*
+  ;;
+  
 *-*-sco3.2v5*)
   # On SCO OpenServer 5, we need -belf to get full-featured binaries.
   SAVE_CFLAGS="$CFLAGS"
@@ -593,6 +609,12 @@
     lt_cv_sys_max_cmd_len=-1;
     ;;
 
+  darwin*)
+    # this should be enough... the length test freaks out some versions
+    # of OSX's `expr`
+    lt_cv_sys_max_cmd_len=16384;
+    ;;
+
   *)
     # If test is not a shell built-in, we'll probably end up computing a
     # maximum length that is only half of the actual maximum length, but
@@ -1114,6 +1136,7 @@
   ;;
 
 darwin* | rhapsody*)
+
   dynamic_linker="$host_os dyld"
   version_type=darwin
   need_lib_prefix=no
@@ -1251,8 +1274,17 @@
   library_names_spec='${libname}${release}.so$versuffix ${libname}${release}.so$major $libname.so'
   soname_spec='${libname}${release}.so$major'
   finish_cmds='PATH="\$PATH:/sbin" ldconfig -n $libdir'
+  libsuff=
+  if test "x$LINUX_64_MODE" = x64; then
+    # Some platforms are per default 64-bit, so there's no /lib64
+    if test -d /lib64; then
+      libsuff=64
+    fi
+  fi
   shlibpath_var=LD_LIBRARY_PATH
   shlibpath_overrides_runpath=no
+  sys_lib_dlsearch_path_spec="/lib${libsuff} /usr/lib${libsuff}"
+  sys_lib_search_path_spec="/lib${libsuff} /usr/lib${libsuff} /usr/local/lib${libsuff}"
   # This implies no fast_install, which is unacceptable.
   # Some rework will be needed to allow for fast_install
   # before this can be enabled.
@@ -1883,7 +1915,8 @@
     lt_cv_file_magic_test_file=`/System/Library/Frameworks/System.framework/System`
     ;;
   *) # Darwin 1.3 on
-    lt_cv_file_magic_test_file='/usr/lib/libSystem.dylib'
+    lt_cv_deplibs_check_method='pass_all'
+    lt_cv_file_magic_cmd=''
     ;;
   esac
   ;;
@@ -1943,7 +1976,7 @@
 # This must be Linux ELF.
 linux*)
   case $host_cpu in
-  alpha* | hppa* | i*86 | ia64* | m68* | mips | mipsel | powerpc* | sparc* | s390* | sh*)
+  alpha* | hppa* | i*86 | ia64* | m68* | mips | mipsel | powerpc* | sparc* | s390* | sh* | x86_64* )
     lt_cv_deplibs_check_method=pass_all ;;
   *)
     # glibc up to 2.1.1 does not perform some relocations on ARM
@@ -2438,20 +2471,8 @@
       exp_sym_flag='-Bexport'
       no_entry_flag=""
     else
-      aix_use_runtimelinking=no
-
-      # Test if we are trying to use run time linking or normal
-      # AIX style linking. If -brtl is somewhere in LDFLAGS, we
-      # need to do runtime linking.
-      case $host_os in aix4.[[23]]|aix4.[[23]].*|aix5*)
-	for ld_flag in $LDFLAGS; do
-	  if (test $ld_flag = "-brtl" || test $ld_flag = "-Wl,-brtl"); then
+      # KDE requires run time linking.  Make it the default.
 	    aix_use_runtimelinking=yes
-	    break
-	  fi
-	done
-      esac
-
       exp_sym_flag='-bexport'
       no_entry_flag='-bnoentry'
     fi
@@ -2497,16 +2518,15 @@
 	shared_flag='-G'
       else
 	if test "$aix_use_runtimelinking" = yes; then
-	  shared_flag='${wl}-G'
+	  shared_flag='-qmkshrobj ${wl}-G'
 	else
-	  shared_flag='${wl}-bM:SRE'
+	  shared_flag='-qmkshrobj'
 	fi
       fi
     fi
 
-    # It seems that -bexpall does not export symbols beginning with
-    # underscore (_), so it is better to generate a list of symbols to export.
-    _LT_AC_TAGVAR(always_export_symbols, $1)=yes
+    # Let the compiler handle the export list.
+    _LT_AC_TAGVAR(always_export_symbols, $1)=no
     if test "$aix_use_runtimelinking" = yes; then
       # Warning - without using the other runtime loading flags (-brtl),
       # -berok will link without error, but may produce a broken library.
@@ -2515,7 +2535,8 @@
       _LT_AC_SYS_LIBPATH_AIX
       _LT_AC_TAGVAR(hardcode_libdir_flag_spec, $1)='${wl}-blibpath:$libdir:'"$aix_libpath"
 
-      _LT_AC_TAGVAR(archive_expsym_cmds, $1)="\$CC"' -o $output_objdir/$soname $libobjs $deplibs $compiler_flags `if test "x${allow_undefined_flag}" != "x"; then echo "${wl}${allow_undefined_flag}"; else :; fi` '"\${wl}$no_entry_flag \${wl}$exp_sym_flag:\$export_symbols $shared_flag"
+      _LT_AC_TAGVAR(archive_cmds, $1)="\$CC"' -o $output_objdir/$soname $libobjs $deplibs $compiler_flags `if test "x${allow_undefined_flag}" != "x"; then echo "${wl}${allow_undefined_flag}"; else :; fi` '" $shared_flag"
+      _LT_AC_TAGVAR(archive_expsym_cmds, $1)="\$CC"' -o $output_objdir/$soname $libobjs $deplibs $compiler_flags `if test "x${allow_undefined_flag}" != "x"; then echo "${wl}${allow_undefined_flag}"; else :; fi` '"\${wl}$exp_sym_flag:\$export_symbols $shared_flag"
      else
       if test "$host_cpu" = ia64; then
 	_LT_AC_TAGVAR(hardcode_libdir_flag_spec, $1)='${wl}-R $libdir:/usr/lib:/lib'
@@ -2547,6 +2568,20 @@
 	;;
     esac
     ;;
+  darwin*)
+    shared_flag='-dynamiclib'
+    BUILD=`gcc -v 2>&1 | grep 'GCC version'`
+    BUILD=`expr "$BUILD" : '.*GCC version \(.*\), based.*'`
+    test -n "$BUILD" || BUILD=0
+    if test "$BUILD" -ge 1173; then
+      export MACOSX_DEPLOYMENT_TARGET=10.2
+      _LT_AC_TAGVAR(archive_cmds, $1)='$CC $(test .$module = .yes && echo -bundle || echo -dynamiclib -single_module) $predep_objects $libobjs $deplibs $postdep_objects $compiler_flags -o $lib $(test .$module != .yes && echo -install_name $rpath/$soname $verstring)'
+    else
+      # this may cause problems with some libraries!
+      _LT_AC_TAGVAR(archive_cmds, $1)='$CC -r -keep_private_externs -nostdlib -o ${lib}-master.o $libobjs && $CC $(test .$module = .yes && echo -bundle || echo -dynamiclib) $predep_objects ${lib}-master.o $deplibs $postdep_objects $compiler_flags -o $lib $(test x$module != xyes && echo -install_name $rpath/$soname $verstring)'
+    fi
+    output_verbose_link_cmd='$CC $(test .$module = .yes && echo -bundle || echo -dynamiclib) $CFLAGS -v conftest.$objext 2>&1 | egrep "\-L"'
+    ;;
   dgux*)
     case $cc_basename in
       ec++)
@@ -3074,7 +3109,7 @@
        fi
        ;;
 
-    *.$objext)
+    *.$objext|*.$libext)
        # This assumes that the test object file only shows up
        # once in the compiler output.
        if test "$p" = "conftest.$objext"; then
@@ -4611,20 +4646,9 @@
 	else
 	  _LT_AC_TAGVAR(export_symbols_cmds, $1)='$NM -BCpg $libobjs $convenience | awk '\''{ if (((\[$]2 == "T") || (\[$]2 == "D") || (\[$]2 == "B")) && ([substr](\[$]3,1,1) != ".")) { print \[$]3 } }'\'' | sort -u > $export_symbols'
 	fi
-	aix_use_runtimelinking=no
 
-	# Test if we are trying to use run time linking or normal
-	# AIX style linking. If -brtl is somewhere in LDFLAGS, we
-	# need to do runtime linking.
-	case $host_os in aix4.[[23]]|aix4.[[23]].*|aix5*)
-	  for ld_flag in $LDFLAGS; do
-  	  if (test $ld_flag = "-brtl" || test $ld_flag = "-Wl,-brtl"); then
+	# KDE requires run time linking.  Make it the default.
   	    aix_use_runtimelinking=yes
-  	    break
-  	  fi
-	  done
-	esac
-
 	exp_sym_flag='-bexport'
 	no_entry_flag='-bnoentry'
       fi
@@ -4670,16 +4694,15 @@
 	  shared_flag='-G'
 	else
   	if test "$aix_use_runtimelinking" = yes; then
-	    shared_flag='${wl}-G'
+	    shared_flag='-qmkshrobj ${wl}-G'
 	  else
-	    shared_flag='${wl}-bM:SRE'
+	    shared_flag='-qmkshrobj'
   	fi
 	fi
       fi
 
-      # It seems that -bexpall does not export symbols beginning with
-      # underscore (_), so it is better to generate a list of symbols to export.
-      _LT_AC_TAGVAR(always_export_symbols, $1)=yes
+      # Let the compiler handle the export list.
+      _LT_AC_TAGVAR(always_export_symbols, $1)=no
       if test "$aix_use_runtimelinking" = yes; then
 	# Warning - without using the other runtime loading flags (-brtl),
 	# -berok will link without error, but may produce a broken library.
@@ -4687,7 +4710,8 @@
        # Determine the default libpath from the value encoded in an empty executable.
        _LT_AC_SYS_LIBPATH_AIX
        _LT_AC_TAGVAR(hardcode_libdir_flag_spec, $1)='${wl}-blibpath:$libdir:'"$aix_libpath"
-	_LT_AC_TAGVAR(archive_expsym_cmds, $1)="\$CC"' -o $output_objdir/$soname $libobjs $deplibs $compiler_flags `if test "x${allow_undefined_flag}" != "x"; then echo "${wl}${allow_undefined_flag}"; else :; fi` '"\${wl}$no_entry_flag \${wl}$exp_sym_flag:\$export_symbols $shared_flag"
+	_LT_AC_TAGVAR(archive_cmds, $1)="\$CC"' -o $output_objdir/$soname $libobjs $deplibs $compiler_flags `if test "x${allow_undefined_flag}" != "x"; then echo "${wl}${allow_undefined_flag}"; else :; fi` '" $shared_flag"
+	_LT_AC_TAGVAR(archive_expsym_cmds, $1)="\$CC"' -o $output_objdir/$soname $libobjs $deplibs $compiler_flags `if test "x${allow_undefined_flag}" != "x"; then echo "${wl}${allow_undefined_flag}"; else :; fi` '"\${wl}$exp_sym_flag:\$export_symbols $shared_flag"
        else
 	if test "$host_cpu" = ia64; then
 	  _LT_AC_TAGVAR(hardcode_libdir_flag_spec, $1)='${wl}-R $libdir:/usr/lib:/lib'
@@ -4748,7 +4772,9 @@
 	_LT_AC_TAGVAR(allow_undefined_flag, $1)='-undefined suppress'
 	;;
       *) # Darwin 1.3 on
-	_LT_AC_TAGVAR(allow_undefined_flag, $1)='-flat_namespace -undefined suppress'
+	#_LT_AC_TAGVAR(allow_undefined_flag, $1)='-flat_namespace -undefined suppress'
+        # testing with twolevel namespaces
+	_LT_AC_TAGVAR(allow_undefined_flag, $1)='-multiply_defined suppress'
 	;;
       esac
 
@@ -4756,7 +4782,16 @@
       #        cross-compilation, but unfortunately the echo tests do not
       #        yet detect zsh echo's removal of \ escapes.  Also zsh mangles
       #	       `"' quotes if we put them in here... so don't!
-      _LT_AC_TAGVAR(archive_cmds, $1)='$CC $(test .$module = .yes && echo -bundle || echo -dynamiclib) $allow_undefined_flag -o $lib $libobjs $deplibs$linker_flags -install_name $rpath/$soname $verstring'
+      BUILD=`gcc -v 2>&1 | grep 'GCC version'`
+      BUILD=`expr "$BUILD" : '.*GCC version \(.*\), based.*'`
+      test -n "$BUILD" || BUILD=0
+      if test "$BUILD" -ge 1173; then
+        export MACOSX_DEPLOYMENT_TARGET=10.2
+        _LT_AC_TAGVAR(archive_cmds, $1)='$CC $(test .$module = .yes && echo -bundle || echo -dynamiclib -single_module) $allow_undefined_flag -o $lib $libobjs $deplibs $linker_flags $(test .$module != .yes && echo -install_name $rpath/$soname $verstring)'
+      else
+        # this may cause problems with some libraries!
+        _LT_AC_TAGVAR(archive_cmds, $1)='$CC -r -keep_private_externs -nostdlib -o ${lib}-master.o $libobjs && $CC $(test .$module = .yes && echo -bundle || echo -dynamiclib) $allow_undefined_flag -o $lib ${lib}-master.o $deplibs $linker_flags $(test x$module != xyes && echo -install_name $rpath/$soname $verstring)'
+      fi
       # We need to add '_' to the symbols in $export_symbols first
       #_LT_AC_TAGVAR(archive_expsym_cmds, $1)="$_LT_AC_TAGVAR(archive_cmds, $1)"' && strip -s $export_symbols'
       _LT_AC_TAGVAR(hardcode_direct, $1)=yes
diff -uNbr kvirc-3.0.0-beta1/src/kvilib/net/kvi_sockettype.h kvirc-3.0.0-beta1-patched/src/kvilib/net/kvi_sockettype.h
--- kvirc-3.0.0-beta1/src/kvilib/net/kvi_sockettype.h	Thu Apr 11 21:27:30 2002
+++ kvirc-3.0.0-beta1-patched/src/kvilib/net/kvi_sockettype.h	Sat Dec 21 14:29:19 2002
@@ -38,6 +38,7 @@
 
 #else
 
+	typedef int    socklen_t;
 	typedef int    kvi_socket_t;
 
 #endif
diff -uNbr kvirc-3.0.0-beta1/src/modules/Makefile.am kvirc-3.0.0-beta1-patched/src/modules/Makefile.am
--- kvirc-3.0.0-beta1/src/modules/Makefile.am	Sat Apr 27 08:04:44 2002
+++ kvirc-3.0.0-beta1-patched/src/modules/Makefile.am	Sat Dec 21 18:51:35 2002
@@ -9,7 +9,6 @@
 		editor \
 		file \
 		dockwidget \
-		gnutella \
 		help \
 		ident \
 		iograph \
diff -uNbr kvirc-3.0.0-beta1/src/modules/about/Makefile.am kvirc-3.0.0-beta1-patched/src/modules/about/Makefile.am
--- kvirc-3.0.0-beta1/src/modules/about/Makefile.am	Mon Jun 24 21:21:43 2002
+++ kvirc-3.0.0-beta1-patched/src/modules/about/Makefile.am	Sat Dec 21 18:33:42 2002
@@ -7,7 +7,7 @@
 
 pluglib_LTLIBRARIES      = libkviabout.la
 
-libkviabout_la_LDFLAGS = -module -avoid-version $(SS_LDFLAGS) $(SS_LIBDIRS)
+libkviabout_la_LDFLAGS = -module -avoid-version $(SS_LDFLAGS) $(SS_LIBDIRS) $(FLAT)
 
 libkviabout_la_SOURCES = libkviabout.cpp aboutdialog.cpp
 
diff -uNbr kvirc-3.0.0-beta1/src/modules/chan/Makefile.am kvirc-3.0.0-beta1-patched/src/modules/chan/Makefile.am
--- kvirc-3.0.0-beta1/src/modules/chan/Makefile.am	Sun Feb  3 05:24:00 2002
+++ kvirc-3.0.0-beta1-patched/src/modules/chan/Makefile.am	Sat Dec 21 18:33:42 2002
@@ -7,7 +7,7 @@
 
 pluglib_LTLIBRARIES      = libkvichan.la
 
-libkvichan_la_LDFLAGS = -module -avoid-version $(SS_LDFLAGS) $(SS_LIBDIRS)
+libkvichan_la_LDFLAGS = -module -avoid-version $(SS_LDFLAGS) $(SS_LIBDIRS) $(FLAT)
 
 libkvichan_la_SOURCES = libkvichan.cpp
 libkvichan_la_LIBADD  = $(SS_LIBLINK) -lkvilib
diff -uNbr kvirc-3.0.0-beta1/src/modules/channelsjoin/Makefile.am kvirc-3.0.0-beta1-patched/src/modules/channelsjoin/Makefile.am
--- kvirc-3.0.0-beta1/src/modules/channelsjoin/Makefile.am	Wed Nov  7 00:14:52 2001
+++ kvirc-3.0.0-beta1-patched/src/modules/channelsjoin/Makefile.am	Sat Dec 21 18:33:42 2002
@@ -7,7 +7,7 @@
 
 pluglib_LTLIBRARIES      = libkvichannelsjoin.la
 
-libkvichannelsjoin_la_LDFLAGS = -module -avoid-version $(SS_LDFLAGS) $(SS_LIBDIRS)
+libkvichannelsjoin_la_LDFLAGS = -module -avoid-version $(SS_LDFLAGS) $(SS_LIBDIRS) $(FLAT)
 
 libkvichannelsjoin_la_SOURCES = libkvichannelsjoin.cpp channelsjoinwindow.cpp
 libkvichannelsjoin_la_LIBADD  = $(SS_LIBLINK) -lkvilib
diff -uNbr kvirc-3.0.0-beta1/src/modules/clock/Makefile.am kvirc-3.0.0-beta1-patched/src/modules/clock/Makefile.am
--- kvirc-3.0.0-beta1/src/modules/clock/Makefile.am	Sat Sep 22 08:50:49 2001
+++ kvirc-3.0.0-beta1-patched/src/modules/clock/Makefile.am	Sat Dec 21 18:33:42 2002
@@ -7,7 +7,7 @@
 
 pluglib_LTLIBRARIES      = libkviclock.la
 
-libkviclock_la_LDFLAGS = -module -avoid-version $(SS_LDFLAGS) $(SS_LIBDIRS)
+libkviclock_la_LDFLAGS = -module -avoid-version $(SS_LDFLAGS) $(SS_LIBDIRS) $(FLAT)
 
 libkviclock_la_SOURCES = libkviclock.cpp
 libkviclock_la_LIBADD  = $(SS_LIBLINK)  -lkvilib
diff -uNbr kvirc-3.0.0-beta1/src/modules/config/Makefile.am kvirc-3.0.0-beta1-patched/src/modules/config/Makefile.am
--- kvirc-3.0.0-beta1/src/modules/config/Makefile.am	Thu Jan 31 20:47:03 2002
+++ kvirc-3.0.0-beta1-patched/src/modules/config/Makefile.am	Sat Dec 21 18:33:42 2002
@@ -7,7 +7,7 @@
 
 pluglib_LTLIBRARIES      = libkviconfig.la
 
-libkviconfig_la_LDFLAGS = -module -avoid-version $(SS_LDFLAGS) $(SS_LIBDIRS)
+libkviconfig_la_LDFLAGS = -module -avoid-version $(SS_LDFLAGS) $(SS_LIBDIRS) $(FLAT)
 
 libkviconfig_la_SOURCES = libkviconfig.cpp
 libkviconfig_la_LIBADD  = $(SS_LIBLINK) -lkvilib
diff -uNbr kvirc-3.0.0-beta1/src/modules/dcc/Makefile.am kvirc-3.0.0-beta1-patched/src/modules/dcc/Makefile.am
--- kvirc-3.0.0-beta1/src/modules/dcc/Makefile.am	Sat Sep 22 08:50:49 2001
+++ kvirc-3.0.0-beta1-patched/src/modules/dcc/Makefile.am	Sat Dec 21 18:33:42 2002
@@ -7,7 +7,7 @@
 
 pluglib_LTLIBRARIES      = libkvidcc.la
 
-libkvidcc_la_LDFLAGS = -module -avoid-version $(SS_LDFLAGS) $(SS_LIBDIRS)
+libkvidcc_la_LDFLAGS = -module -avoid-version $(SS_LDFLAGS) $(SS_LIBDIRS) $(FLAT)
 
 libkvidcc_la_SOURCES = adpcmcodec.cpp \
 										broker.cpp \
diff -uNbr kvirc-3.0.0-beta1/src/modules/dialog/Makefile.am kvirc-3.0.0-beta1-patched/src/modules/dialog/Makefile.am
--- kvirc-3.0.0-beta1/src/modules/dialog/Makefile.am	Sat Sep 22 08:50:49 2001
+++ kvirc-3.0.0-beta1-patched/src/modules/dialog/Makefile.am	Sat Dec 21 18:33:42 2002
@@ -7,7 +7,7 @@
 
 pluglib_LTLIBRARIES      = libkvidialog.la
 
-libkvidialog_la_LDFLAGS = -module -avoid-version $(SS_LDFLAGS) $(SS_LIBDIRS)
+libkvidialog_la_LDFLAGS = -module -avoid-version $(SS_LDFLAGS) $(SS_LIBDIRS) $(FLAT)
 
 libkvidialog_la_SOURCES = libkvidialog.cpp
 
diff -uNbr kvirc-3.0.0-beta1/src/modules/dirbrowser/Makefile.am kvirc-3.0.0-beta1-patched/src/modules/dirbrowser/Makefile.am
--- kvirc-3.0.0-beta1/src/modules/dirbrowser/Makefile.am	Sat Sep 22 08:50:49 2001
+++ kvirc-3.0.0-beta1-patched/src/modules/dirbrowser/Makefile.am	Sat Dec 21 18:33:42 2002
@@ -4,7 +4,7 @@
 
 pluglib_LTLIBRARIES      = libkvidirbrowser.la
 
-libkvidirbrowser_la_LDFLAGS = -module -avoid-version $(SS_LDFLAGS) $(SS_LIBDIRS)
+libkvidirbrowser_la_LDFLAGS = -module -avoid-version $(SS_LDFLAGS) $(SS_LIBDIRS) $(FLAT)
 
 libkvidirbrowser_la_SOURCES = libkvidirbrowser.cpp dirbrowserwindow.cpp dirbrowserwidget.cpp
 
diff -uNbr kvirc-3.0.0-beta1/src/modules/dockwidget/Makefile.am kvirc-3.0.0-beta1-patched/src/modules/dockwidget/Makefile.am
--- kvirc-3.0.0-beta1/src/modules/dockwidget/Makefile.am	Sat Sep 22 08:50:49 2001
+++ kvirc-3.0.0-beta1-patched/src/modules/dockwidget/Makefile.am	Sat Dec 21 18:33:42 2002
@@ -7,7 +7,7 @@
 
 pluglib_LTLIBRARIES      = libkvidockwidget.la
 
-libkvidockwidget_la_LDFLAGS = -module -avoid-version $(SS_LDFLAGS) $(SS_LIBDIRS)
+libkvidockwidget_la_LDFLAGS = -module -avoid-version $(SS_LDFLAGS) $(SS_LIBDIRS) $(FLAT)
 
 libkvidockwidget_la_SOURCES = libkvidockwidget.cpp
 libkvidockwidget_la_LIBADD  = $(SS_LIBLINK) -lkvilib
diff -uNbr kvirc-3.0.0-beta1/src/modules/editor/Makefile.am kvirc-3.0.0-beta1-patched/src/modules/editor/Makefile.am
--- kvirc-3.0.0-beta1/src/modules/editor/Makefile.am	Wed Feb 27 19:02:00 2002
+++ kvirc-3.0.0-beta1-patched/src/modules/editor/Makefile.am	Sat Dec 21 18:33:42 2002
@@ -7,7 +7,7 @@
 
 pluglib_LTLIBRARIES      = libkvieditor.la
 
-libkvieditor_la_LDFLAGS = -module -avoid-version $(SS_LDFLAGS) $(SS_LIBDIRS)
+libkvieditor_la_LDFLAGS = -module -avoid-version $(SS_LDFLAGS) $(SS_LIBDIRS) $(FLAT)
 
 libkvieditor_la_SOURCES = libkvieditor.cpp editor.cpp editorwindow.cpp scripteditor.cpp tableview.cpp
 libkvieditor_la_LIBADD  = $(SS_LIBLINK) -lkvilib
diff -uNbr kvirc-3.0.0-beta1/src/modules/file/Makefile.am kvirc-3.0.0-beta1-patched/src/modules/file/Makefile.am
--- kvirc-3.0.0-beta1/src/modules/file/Makefile.am	Thu Nov  8 23:09:04 2001
+++ kvirc-3.0.0-beta1-patched/src/modules/file/Makefile.am	Sat Dec 21 18:33:42 2002
@@ -7,7 +7,7 @@
 
 pluglib_LTLIBRARIES      = libkvifile.la
 
-libkvifile_la_LDFLAGS = -module -avoid-version $(SS_LDFLAGS) $(SS_LIBDIRS)
+libkvifile_la_LDFLAGS = -module -avoid-version $(SS_LDFLAGS) $(SS_LIBDIRS) $(FLAT)
 
 libkvifile_la_SOURCES = libkvifile.cpp
 libkvifile_la_LIBADD  = $(SS_LIBLINK) -lkvilib
diff -uNbr kvirc-3.0.0-beta1/src/modules/gnutella/Makefile.am kvirc-3.0.0-beta1-patched/src/modules/gnutella/Makefile.am
--- kvirc-3.0.0-beta1/src/modules/gnutella/Makefile.am	Sat Sep 22 08:50:49 2001
+++ kvirc-3.0.0-beta1-patched/src/modules/gnutella/Makefile.am	Sat Dec 21 18:33:42 2002
@@ -7,7 +7,7 @@
 
 pluglib_LTLIBRARIES      = libkvignutella.la
 
-libkvignutella_la_LDFLAGS = -module -avoid-version $(SS_LDFLAGS) $(SS_LIBDIRS)
+libkvignutella_la_LDFLAGS = -module -avoid-version $(SS_LDFLAGS) $(SS_LIBDIRS) $(FLAT)
 
 libkvignutella_la_SOURCES = gnutelladcache.cpp gnutellanode.cpp \
 			gnutellaoptions.cpp gnutellarouting.cpp gnutellashares.cpp gnutellasearch.cpp \
diff -uNbr kvirc-3.0.0-beta1/src/modules/help/Makefile.am kvirc-3.0.0-beta1-patched/src/modules/help/Makefile.am
--- kvirc-3.0.0-beta1/src/modules/help/Makefile.am	Sat Sep 22 08:50:49 2001
+++ kvirc-3.0.0-beta1-patched/src/modules/help/Makefile.am	Sat Dec 21 18:33:42 2002
@@ -7,7 +7,7 @@
 
 pluglib_LTLIBRARIES      = libkvihelp.la
 
-libkvihelp_la_LDFLAGS = -module -avoid-version $(SS_LDFLAGS) $(SS_LIBDIRS)
+libkvihelp_la_LDFLAGS = -module -avoid-version $(SS_LDFLAGS) $(SS_LIBDIRS) $(FLAT)
 
 libkvihelp_la_SOURCES = libkvihelp.cpp helpwidget.cpp helpwindow.cpp
 libkvihelp_la_LIBADD  = $(SS_LIBLINK) -lkvilib
diff -uNbr kvirc-3.0.0-beta1/src/modules/ident/Makefile.am kvirc-3.0.0-beta1-patched/src/modules/ident/Makefile.am
--- kvirc-3.0.0-beta1/src/modules/ident/Makefile.am	Tue Oct  2 18:32:34 2001
+++ kvirc-3.0.0-beta1-patched/src/modules/ident/Makefile.am	Sat Dec 21 18:33:42 2002
@@ -7,7 +7,7 @@
 
 pluglib_LTLIBRARIES      = libkviident.la
 
-libkviident_la_LDFLAGS = -module -avoid-version $(SS_LDFLAGS) $(SS_LIBDIRS)
+libkviident_la_LDFLAGS = -module -avoid-version $(SS_LDFLAGS) $(SS_LIBDIRS) $(FLAT)
 
 libkviident_la_SOURCES = libkviident.cpp
 libkviident_la_LIBADD  = $(SS_LIBLINK)  -lkvilib
diff -uNbr kvirc-3.0.0-beta1/src/modules/iograph/Makefile.am kvirc-3.0.0-beta1-patched/src/modules/iograph/Makefile.am
--- kvirc-3.0.0-beta1/src/modules/iograph/Makefile.am	Sat Sep 22 08:50:49 2001
+++ kvirc-3.0.0-beta1-patched/src/modules/iograph/Makefile.am	Sat Dec 21 18:33:42 2002
@@ -7,7 +7,7 @@
 
 pluglib_LTLIBRARIES      = libkviiograph.la
 
-libkviiograph_la_LDFLAGS = -module -avoid-version $(SS_LDFLAGS) $(SS_LIBDIRS)
+libkviiograph_la_LDFLAGS = -module -avoid-version $(SS_LDFLAGS) $(SS_LIBDIRS) $(FLAT)
 
 libkviiograph_la_SOURCES = libkviiograph.cpp
 libkviiograph_la_LIBADD  = $(SS_LIBLINK) -lkvilib
diff -uNbr kvirc-3.0.0-beta1/src/modules/lamerizer/Makefile.am kvirc-3.0.0-beta1-patched/src/modules/lamerizer/Makefile.am
--- kvirc-3.0.0-beta1/src/modules/lamerizer/Makefile.am	Sun Feb 17 22:06:41 2002
+++ kvirc-3.0.0-beta1-patched/src/modules/lamerizer/Makefile.am	Sat Dec 21 18:33:42 2002
@@ -7,7 +7,7 @@
 
 pluglib_LTLIBRARIES      = libkvilamerizer.la
 
-libkvilamerizer_la_LDFLAGS = -module -avoid-version $(SS_LDFLAGS) $(SS_LIBDIRS)
+libkvilamerizer_la_LDFLAGS = -module -avoid-version $(SS_LDFLAGS) $(SS_LIBDIRS) $(FLAT)
 
 libkvilamerizer_la_SOURCES = libkvilamerizer.cpp
 libkvilamerizer_la_LIBADD  = $(SS_LIBLINK) -lkvilib
diff -uNbr kvirc-3.0.0-beta1/src/modules/links/Makefile.am kvirc-3.0.0-beta1-patched/src/modules/links/Makefile.am
--- kvirc-3.0.0-beta1/src/modules/links/Makefile.am	Sat Sep 22 08:50:49 2001
+++ kvirc-3.0.0-beta1-patched/src/modules/links/Makefile.am	Sat Dec 21 18:33:42 2002
@@ -7,7 +7,7 @@
 
 pluglib_LTLIBRARIES      = libkvilinks.la
 
-libkvilinks_la_LDFLAGS = -module -avoid-version $(SS_LDFLAGS) $(SS_LIBDIRS)
+libkvilinks_la_LDFLAGS = -module -avoid-version $(SS_LDFLAGS) $(SS_LIBDIRS) $(FLAT)
 
 libkvilinks_la_SOURCES = libkvilinks.cpp linkswindow.cpp
 libkvilinks_la_LIBADD  = $(SS_LIBLINK) -lkvilib
diff -uNbr kvirc-3.0.0-beta1/src/modules/list/Makefile.am kvirc-3.0.0-beta1-patched/src/modules/list/Makefile.am
--- kvirc-3.0.0-beta1/src/modules/list/Makefile.am	Fri Oct 12 11:01:34 2001
+++ kvirc-3.0.0-beta1-patched/src/modules/list/Makefile.am	Sat Dec 21 18:33:42 2002
@@ -7,7 +7,7 @@
 
 pluglib_LTLIBRARIES      = libkvilist.la
 
-libkvilist_la_LDFLAGS = -module -avoid-version $(SS_LDFLAGS) $(SS_LIBDIRS)
+libkvilist_la_LDFLAGS = -module -avoid-version $(SS_LDFLAGS) $(SS_LIBDIRS) $(FLAT)
 
 libkvilist_la_SOURCES = libkvilist.cpp listwindow.cpp
 libkvilist_la_LIBADD  = $(SS_LIBLINK) -lkvilib
diff -uNbr kvirc-3.0.0-beta1/src/modules/log/Makefile.am kvirc-3.0.0-beta1-patched/src/modules/log/Makefile.am
--- kvirc-3.0.0-beta1/src/modules/log/Makefile.am	Sat Sep 22 08:50:49 2001
+++ kvirc-3.0.0-beta1-patched/src/modules/log/Makefile.am	Sat Dec 21 18:33:42 2002
@@ -7,7 +7,7 @@
 
 pluglib_LTLIBRARIES      = libkvilog.la
 
-libkvilog_la_LDFLAGS = -module -avoid-version $(SS_LDFLAGS) $(SS_LIBDIRS)
+libkvilog_la_LDFLAGS = -module -avoid-version $(SS_LDFLAGS) $(SS_LIBDIRS) $(FLAT)
 
 libkvilog_la_SOURCES = libkvilog.cpp
 libkvilog_la_LIBADD  = $(SS_LIBLINK) -lkvilib
diff -uNbr kvirc-3.0.0-beta1/src/modules/logview/Makefile.am kvirc-3.0.0-beta1-patched/src/modules/logview/Makefile.am
--- kvirc-3.0.0-beta1/src/modules/logview/Makefile.am	Thu Apr 25 21:43:53 2002
+++ kvirc-3.0.0-beta1-patched/src/modules/logview/Makefile.am	Sat Dec 21 18:33:42 2002
@@ -7,7 +7,7 @@
 
 pluglib_LTLIBRARIES      = libkvilogview.la
 
-libkvilogview_la_LDFLAGS = -module -avoid-version $(SS_LDFLAGS) $(SS_LIBDIRS)
+libkvilogview_la_LDFLAGS = -module -avoid-version $(SS_LDFLAGS) $(SS_LIBDIRS) $(FLAT)
 
 libkvilogview_la_SOURCES = libkvilogview.cpp logviewwidget.cpp logviewmdiwindow.cpp
 libkvilogview_la_LIBADD  = $(SS_LIBLINK) -lkvilib
diff -uNbr kvirc-3.0.0-beta1/src/modules/mp3player/Makefile.am kvirc-3.0.0-beta1-patched/src/modules/mp3player/Makefile.am
--- kvirc-3.0.0-beta1/src/modules/mp3player/Makefile.am	Sun Nov  4 09:41:10 2001
+++ kvirc-3.0.0-beta1-patched/src/modules/mp3player/Makefile.am	Sat Dec 21 18:33:42 2002
@@ -7,7 +7,7 @@
 
 pluglib_LTLIBRARIES      = libkvimp3player.la
 
-libkvimp3player_la_LDFLAGS = -module -avoid-version $(SS_LDFLAGS) $(SS_LIBDIRS)
+libkvimp3player_la_LDFLAGS = -module -avoid-version $(SS_LDFLAGS) $(SS_LIBDIRS) $(FLAT)
 
 libkvimp3player_la_SOURCES = libkvimp3player.cpp
 libkvimp3player_la_LIBADD  = $(SS_LIBLINK)  -lkvilib
diff -uNbr kvirc-3.0.0-beta1/src/modules/objects/Makefile.am kvirc-3.0.0-beta1-patched/src/modules/objects/Makefile.am
--- kvirc-3.0.0-beta1/src/modules/objects/Makefile.am	Tue Feb 19 23:34:06 2002
+++ kvirc-3.0.0-beta1-patched/src/modules/objects/Makefile.am	Sat Dec 21 18:33:42 2002
@@ -7,7 +7,7 @@
 
 pluglib_LTLIBRARIES      = libkviobjects.la
 
-libkviobjects_la_LDFLAGS = -module -avoid-version $(SS_LDFLAGS) $(SS_LIBDIRS)
+libkviobjects_la_LDFLAGS = -module -avoid-version $(SS_LDFLAGS) $(SS_LIBDIRS) $(FLAT)
 
 libkviobjects_la_SOURCES =  libkviobjects.cpp \
 			    class_button.cpp \
diff -uNbr kvirc-3.0.0-beta1/src/modules/offer/Makefile.am kvirc-3.0.0-beta1-patched/src/modules/offer/Makefile.am
--- kvirc-3.0.0-beta1/src/modules/offer/Makefile.am	Thu Mar 28 09:01:56 2002
+++ kvirc-3.0.0-beta1-patched/src/modules/offer/Makefile.am	Sat Dec 21 18:33:42 2002
@@ -7,7 +7,7 @@
 
 pluglib_LTLIBRARIES      = libkvioffer.la
 
-libkvioffer_la_LDFLAGS = -module -avoid-version $(SS_LDFLAGS) $(SS_LIBDIRS)
+libkvioffer_la_LDFLAGS = -module -avoid-version $(SS_LDFLAGS) $(SS_LIBDIRS) $(FLAT)
 
 libkvioffer_la_SOURCES = libkvioffer.cpp offerwindow.cpp
 libkvioffer_la_LIBADD  = $(SS_LIBLINK) -lkvilib
diff -uNbr kvirc-3.0.0-beta1/src/modules/options/Makefile.am kvirc-3.0.0-beta1-patched/src/modules/options/Makefile.am
--- kvirc-3.0.0-beta1/src/modules/options/Makefile.am	Thu Jun 20 14:54:36 2002
+++ kvirc-3.0.0-beta1-patched/src/modules/options/Makefile.am	Sat Dec 21 18:33:42 2002
@@ -9,7 +9,7 @@
 
 pluglib_LTLIBRARIES      = libkvioptions.la
 
-libkvioptions_la_LDFLAGS = -module -avoid-version $(SS_LDFLAGS) $(SS_LIBDIRS)
+libkvioptions_la_LDFLAGS = -module -avoid-version $(SS_LDFLAGS) $(SS_LIBDIRS) $(FLAT)
 
 libkvioptions_la_SOURCES = libkvioptions.cpp \
 											container.cpp \
diff -uNbr kvirc-3.0.0-beta1/src/modules/reguser/Makefile.am kvirc-3.0.0-beta1-patched/src/modules/reguser/Makefile.am
--- kvirc-3.0.0-beta1/src/modules/reguser/Makefile.am	Sat Sep 22 08:50:49 2001
+++ kvirc-3.0.0-beta1-patched/src/modules/reguser/Makefile.am	Sat Dec 21 18:33:42 2002
@@ -7,7 +7,7 @@
 
 pluglib_LTLIBRARIES      = libkvireguser.la
 
-libkvireguser_la_LDFLAGS = -module -avoid-version $(SS_LDFLAGS) $(SS_LIBDIRS)
+libkvireguser_la_LDFLAGS = -module -avoid-version $(SS_LDFLAGS) $(SS_LIBDIRS) $(FLAT)
 
 libkvireguser_la_SOURCES = libkvireguser.cpp edituser.cpp
 libkvireguser_la_LIBADD  = $(SS_LIBLINK) -lkvilib
diff -uNbr kvirc-3.0.0-beta1/src/modules/rijndael/Makefile.am kvirc-3.0.0-beta1-patched/src/modules/rijndael/Makefile.am
--- kvirc-3.0.0-beta1/src/modules/rijndael/Makefile.am	Fri Apr 12 14:53:18 2002
+++ kvirc-3.0.0-beta1-patched/src/modules/rijndael/Makefile.am	Sat Dec 21 18:33:42 2002
@@ -9,7 +9,7 @@
 
 pluglib_LTLIBRARIES      = libkvirijndael.la
 
-libkvirijndael_la_LDFLAGS = -module -avoid-version $(SS_LDFLAGS) $(SS_LIBDIRS)
+libkvirijndael_la_LDFLAGS = -module -avoid-version $(SS_LDFLAGS) $(SS_LIBDIRS) $(FLAT)
 
 libkvirijndael_la_SOURCES = libkvirijndael.cpp rijndael.cpp
 libkvirijndael_la_LIBADD  = $(SS_LIBLINK) -lkvilib
diff -uNbr kvirc-3.0.0-beta1/src/modules/scriptcenter/Makefile.am kvirc-3.0.0-beta1-patched/src/modules/scriptcenter/Makefile.am
--- kvirc-3.0.0-beta1/src/modules/scriptcenter/Makefile.am	Fri Jan 11 07:51:53 2002
+++ kvirc-3.0.0-beta1-patched/src/modules/scriptcenter/Makefile.am	Sat Dec 21 18:33:42 2002
@@ -7,7 +7,7 @@
 
 pluglib_LTLIBRARIES      = libkviscriptcenter.la
 
-libkviscriptcenter_la_LDFLAGS = -module -avoid-version $(SS_LDFLAGS) $(SS_LIBDIRS)
+libkviscriptcenter_la_LDFLAGS = -module -avoid-version $(SS_LDFLAGS) $(SS_LIBDIRS) $(FLAT)
 
 libkviscriptcenter_la_SOURCES = libkviscriptcenter.cpp \
 							scriptcenterwidget.cpp \
diff -uNbr kvirc-3.0.0-beta1/src/modules/setup/Makefile.am kvirc-3.0.0-beta1-patched/src/modules/setup/Makefile.am
--- kvirc-3.0.0-beta1/src/modules/setup/Makefile.am	Sat Oct  6 15:12:17 2001
+++ kvirc-3.0.0-beta1-patched/src/modules/setup/Makefile.am	Sat Dec 21 18:33:42 2002
@@ -7,7 +7,7 @@
 
 pluglib_LTLIBRARIES      = libkvisetup.la
 
-libkvisetup_la_LDFLAGS = -module -avoid-version $(SS_LDFLAGS) $(SS_LIBDIRS)
+libkvisetup_la_LDFLAGS = -module -avoid-version $(SS_LDFLAGS) $(SS_LIBDIRS) $(FLAT)
 
 libkvisetup_la_SOURCES = libkvisetup.cpp setupwizard.cpp dirview.cpp
 libkvisetup_la_LIBADD  = $(SS_LIBLINK) -lkvilib
diff -uNbr kvirc-3.0.0-beta1/src/modules/snd/Makefile.am kvirc-3.0.0-beta1-patched/src/modules/snd/Makefile.am
--- kvirc-3.0.0-beta1/src/modules/snd/Makefile.am	Wed May  1 20:24:24 2002
+++ kvirc-3.0.0-beta1-patched/src/modules/snd/Makefile.am	Sat Dec 21 18:33:42 2002
@@ -8,7 +8,7 @@
 
 pluglib_LTLIBRARIES      = libkvisnd.la
 
-libkvisnd_la_LDFLAGS = -module -avoid-version $(SS_LDFLAGS) $(SS_LIBDIRS)
+libkvisnd_la_LDFLAGS = -module -avoid-version $(SS_LDFLAGS) $(SS_LIBDIRS) $(FLAT)
 
 libkvisnd_la_SOURCES = libkvisnd.cpp
 libkvisnd_la_LIBADD  = $(SS_LIBLINK) -lkvilib
diff -uNbr kvirc-3.0.0-beta1/src/modules/socketspy/Makefile.am kvirc-3.0.0-beta1-patched/src/modules/socketspy/Makefile.am
--- kvirc-3.0.0-beta1/src/modules/socketspy/Makefile.am	Sat Sep 22 08:50:50 2001
+++ kvirc-3.0.0-beta1-patched/src/modules/socketspy/Makefile.am	Sat Dec 21 18:33:42 2002
@@ -7,7 +7,7 @@
 
 pluglib_LTLIBRARIES      = libkvisocketspy.la
 
-libkvisocketspy_la_LDFLAGS = -module -avoid-version $(SS_LDFLAGS) $(SS_LIBDIRS)
+libkvisocketspy_la_LDFLAGS = -module -avoid-version $(SS_LDFLAGS) $(SS_LIBDIRS) $(FLAT)
 
 libkvisocketspy_la_SOURCES = libkvisocketspy.cpp socketspywindow.cpp
 libkvisocketspy_la_LIBADD  = $(SS_LIBLINK) -lkvilib
diff -uNbr kvirc-3.0.0-beta1/src/modules/spaste/Makefile.am kvirc-3.0.0-beta1-patched/src/modules/spaste/Makefile.am
--- kvirc-3.0.0-beta1/src/modules/spaste/Makefile.am	Wed May  1 21:40:51 2002
+++ kvirc-3.0.0-beta1-patched/src/modules/spaste/Makefile.am	Sat Dec 21 18:33:42 2002
@@ -10,7 +10,7 @@
 #ARTS_C_FLAGS = -L/usr/lib -ldl -lartsc -lpthread
 pluglib_LTLIBRARIES      = libkvispaste.la
 
-libkvispaste_la_LDFLAGS = -module -avoid-version $(SS_LDFLAGS) $(SS_LIBDIRS)
+libkvispaste_la_LDFLAGS = -module -avoid-version $(SS_LDFLAGS) $(SS_LIBDIRS) $(FLAT)
 
 libkvispaste_la_SOURCES = libkvispaste.cpp controller.cpp
 libkvispaste_la_LIBADD  = $(SS_LIBLINK) -lkvilib
diff -uNbr kvirc-3.0.0-beta1/src/modules/str/Makefile.am kvirc-3.0.0-beta1-patched/src/modules/str/Makefile.am
--- kvirc-3.0.0-beta1/src/modules/str/Makefile.am	Wed Jan  2 15:43:06 2002
+++ kvirc-3.0.0-beta1-patched/src/modules/str/Makefile.am	Sat Dec 21 18:33:42 2002
@@ -7,7 +7,7 @@
 
 pluglib_LTLIBRARIES      = libkvistr.la
 
-libkvistr_la_LDFLAGS = -module -avoid-version $(SS_LDFLAGS) $(SS_LIBDIRS)
+libkvistr_la_LDFLAGS = -module -avoid-version $(SS_LDFLAGS) $(SS_LIBDIRS) $(FLAT)
 
 libkvistr_la_SOURCES = libkvistr.cpp
 libkvistr_la_LIBADD  = $(SS_LIBLINK) -lkvilib
diff -uNbr kvirc-3.0.0-beta1/src/modules/system/Makefile.am kvirc-3.0.0-beta1-patched/src/modules/system/Makefile.am
--- kvirc-3.0.0-beta1/src/modules/system/Makefile.am	Fri Nov 16 20:56:31 2001
+++ kvirc-3.0.0-beta1-patched/src/modules/system/Makefile.am	Sat Dec 21 18:33:42 2002
@@ -7,7 +7,7 @@
 
 pluglib_LTLIBRARIES      = libkvisystem.la
 
-libkvisystem_la_LDFLAGS = -module -avoid-version $(SS_LDFLAGS) $(SS_LIBDIRS)
+libkvisystem_la_LDFLAGS = -module -avoid-version $(SS_LDFLAGS) $(SS_LIBDIRS) $(FLAT)
 
 libkvisystem_la_SOURCES = libkvisystem.cpp
 libkvisystem_la_LIBADD  = $(SS_LIBLINK) -lkvilib
diff -uNbr kvirc-3.0.0-beta1/src/modules/term/Makefile.am kvirc-3.0.0-beta1-patched/src/modules/term/Makefile.am
--- kvirc-3.0.0-beta1/src/modules/term/Makefile.am	Sat Sep 22 08:50:50 2001
+++ kvirc-3.0.0-beta1-patched/src/modules/term/Makefile.am	Sat Dec 21 18:33:43 2002
@@ -7,7 +7,7 @@
 
 pluglib_LTLIBRARIES      = libkviterm.la
 
-libkviterm_la_LDFLAGS = -module -avoid-version $(SS_LDFLAGS) $(SS_LIBDIRS)
+libkviterm_la_LDFLAGS = -module -avoid-version $(SS_LDFLAGS) $(SS_LIBDIRS) $(FLAT)
 
 libkviterm_la_SOURCES = libkviterm.cpp termwidget.cpp termwindow.cpp
 libkviterm_la_LIBADD  = $(SS_LIBLINK) -lkvilib
diff -uNbr kvirc-3.0.0-beta1/src/modules/tip/Makefile.am kvirc-3.0.0-beta1-patched/src/modules/tip/Makefile.am
--- kvirc-3.0.0-beta1/src/modules/tip/Makefile.am	Sat Sep 22 08:50:50 2001
+++ kvirc-3.0.0-beta1-patched/src/modules/tip/Makefile.am	Sat Dec 21 18:33:43 2002
@@ -7,7 +7,7 @@
 
 pluglib_LTLIBRARIES      = libkvitip.la
 
-libkvitip_la_LDFLAGS = -module -avoid-version $(SS_LDFLAGS) $(SS_LIBDIRS)
+libkvitip_la_LDFLAGS = -module -avoid-version $(SS_LDFLAGS) $(SS_LIBDIRS) $(FLAT)
 
 libkvitip_la_SOURCES = libkvitip.cpp
 libkvitip_la_LIBADD  = $(SS_LIBLINK) -lkvilib
diff -uNbr kvirc-3.0.0-beta1/src/modules/tracker/Makefile.am kvirc-3.0.0-beta1-patched/src/modules/tracker/Makefile.am
--- kvirc-3.0.0-beta1/src/modules/tracker/Makefile.am	Sat Dec  8 12:21:46 2001
+++ kvirc-3.0.0-beta1-patched/src/modules/tracker/Makefile.am	Sat Dec 21 18:33:43 2002
@@ -7,7 +7,7 @@
 
 pluglib_LTLIBRARIES      = libkvitracker.la
 
-libkvitracker_la_LDFLAGS = -module -avoid-version $(SS_LDFLAGS) $(SS_LIBDIRS)
+libkvitracker_la_LDFLAGS = -module -avoid-version $(SS_LDFLAGS) $(SS_LIBDIRS) $(FLAT)
 
 libkvitracker_la_SOURCES = libkvitracker.cpp trackeduser.cpp
 libkvitracker_la_LIBADD  = $(SS_LIBLINK) -lkvilib
diff -uNbr kvirc-3.0.0-beta1/src/modules/url/Makefile.am kvirc-3.0.0-beta1-patched/src/modules/url/Makefile.am
--- kvirc-3.0.0-beta1/src/modules/url/Makefile.am	Sun Jan  6 18:34:45 2002
+++ kvirc-3.0.0-beta1-patched/src/modules/url/Makefile.am	Sat Dec 21 18:33:43 2002
@@ -7,7 +7,7 @@
 
 pluglib_LTLIBRARIES      = libkviurl.la
 
-libkviurl_la_LDFLAGS = -module -avoid-version $(SS_LDFLAGS) $(SS_LIBDIRS)
+libkviurl_la_LDFLAGS = -module -avoid-version $(SS_LDFLAGS) $(SS_LIBDIRS) $(FLAT)
 
 libkviurl_la_SOURCES = libkviurl.cpp
 libkviurl_la_LIBADD  = $(SS_LIBLINK) -lkvilib
diff -uNbr kvirc-3.0.0-beta1/src/modules/window/Makefile.am kvirc-3.0.0-beta1-patched/src/modules/window/Makefile.am
--- kvirc-3.0.0-beta1/src/modules/window/Makefile.am	Sat Sep 22 08:50:50 2001
+++ kvirc-3.0.0-beta1-patched/src/modules/window/Makefile.am	Sat Dec 21 18:33:43 2002
@@ -7,7 +7,7 @@
 
 pluglib_LTLIBRARIES      = libkviwindow.la
 
-libkviwindow_la_LDFLAGS = -module -avoid-version $(SS_LDFLAGS) $(SS_LIBDIRS)
+libkviwindow_la_LDFLAGS = -module -avoid-version $(SS_LDFLAGS) $(SS_LIBDIRS) $(FLAT)
 
 libkviwindow_la_SOURCES = libkviwindow.cpp
 libkviwindow_la_LIBADD  = $(SS_LIBLINK) -lkvilib
