diff -uNbr kxmleditor-0.8.1/part/Makefile.am kxmleditor-0.8.1-new/part/Makefile.am
--- kxmleditor-0.8.1/part/Makefile.am	Fri Jul 12 15:02:34 2002
+++ kxmleditor-0.8.1-new/part/Makefile.am	Sun Feb 16 12:26:09 2003
@@ -16,7 +16,7 @@
 	kxmleditorpart.cpp kxmleditorfactory.cpp \
 	dcopiface_part_ro.skel dcopiface_part_ro.cpp
 libkxmleditorpart_la_LDFLAGS = $(all_libraries) -version-info 1:0:0 -module
-libkxmleditorpart_la_LIBADD   = $(LIB_KPARTS) $(LIB_KFILE) $(LIB_KDEUI) $(LIB_KDECORE) $(LIB_QT) $(LIBSOCKET)
+libkxmleditorpart_la_LIBADD   = $(LIB_KPARTS) $(LIB_KFILE) $(LIB_KDEUI) $(LIB_KDECORE) $(LIB_QT) $(LIBSOCKET) -lDCOP
 
 SUBDIRS = pics icons
 
