diff -uNbr mosfet-imagepart-0.2.1/background/Makefile.am mosfet-imagepart-0.2.1-new/background/Makefile.am
--- mosfet-imagepart-0.2.1/background/Makefile.am	Sun Feb  9 18:31:48 2003
+++ mosfet-imagepart-0.2.1-new/background/Makefile.am	Mon Feb 17 20:46:57 2003
@@ -2,15 +2,19 @@
 noinst_HEADERS = backgnd.h bgdialogs.h bgsettings.h bgrender.h dialog.h
 
 bin_PROGRAMS = mosfetwallpaper
-lib_LTLIBRARIES = mosfetwallpaper.la
+lib_LTLIBRARIES = mosfetwallpaper.la libmosfetwallpaper_shared.la
 
-mosfetwallpaper_la_SOURCES =  bgrender.cc bgsettings.cc bgdialogs.cpp \
+libmosfetwallpaper_shared_la_SOURCES =  bgrender.cc bgsettings.cc bgdialogs.cpp \
 	backgnd.cpp dialog.cpp main.cpp
-mosfetwallpaper_la_LIBADD = $(LIB_KFILE)
-mosfetwallpaper_la_LDFLAGS = $(all_libraries) -module -avoid-version
+libmosfetwallpaper_shared_la_LIBADD = $(LIB_KFILE) $(LIB_QT) -lkdeui -lkdecore -lDCOP -lkdefx
+libmosfetwallpaper_shared_la_LDFLAGS = $(all_libraries) -avoid-version $(QT_LDFLAGS)
+
+mosfetwallpaper_la_SOURCES = dummy_la.cpp
+mosfetwallpaper_la_LIBADD  = libmosfetwallpaper_shared.la
+mosfetwallpaper_la_LDFLAGS = $(all_libraries) -module -avoid-version $(QT_LDFLAGS)
 
 mosfetwallpaper_SOURCES = dummy.cpp
-mosfetwallpaper_LDADD   = mosfetwallpaper.la
+mosfetwallpaper_LDADD   = libmosfetwallpaper_shared.la
 mosfetwallpaper_LDFLAGS = $(all_libraries) $(KDE_RPATH)
 
 METASOURCES = AUTO
@@ -19,4 +23,7 @@
 
 dummy.cpp:
 	echo > dummy.cpp
+
+dummy_la.cpp:
+	echo > dummy_la.cpp
 
