diff -uNbr slicker/slicker/share/Makefile.am slicker-mac/slicker/share/Makefile.am
--- slicker/slicker/share/Makefile.am	Thu Feb  6 13:17:37 2003
+++ slicker-mac/slicker/share/Makefile.am	Sat Feb 15 17:38:16 2003
@@ -1,4 +1,4 @@
-noinst_LTLIBRARIES = libslicker_share.la
+lib_LTLIBRARIES = libslicker_share.la
 libslicker_share_la_SOURCES = plugininfo.cpp
 
 # set the include path for X, qt and KDE
