Index: slicker/Makefile.am
===================================================================
RCS file: /cvsroot/slicker/slicker/slicker/Makefile.am,v
retrieving revision 1.3
diff -u -r1.3 Makefile.am
--- slicker/Makefile.am	6 Feb 2003 18:17:32 -0000	1.3
+++ slicker/Makefile.am	12 Feb 2003 18:41:56 -0000
@@ -11,7 +11,7 @@
 slicker_la_SOURCES = dummy.cpp
 slicker_la_LDFLAGS = $(KDE_PLUGIN) -no-undefined -module $(all_libraries)
 
-slicker_LDADD = slicker.la
+slicker_LDADD = share/libslicker_share.la carddeskcore/libcarddeskcore.la cardhandler/libslicker_cardhandler.la taskbar/libslicker_taskbar.la slider/libslicker_slider.la core/libslicker_core.la
 slicker_SOURCES = dummy_la.cpp
 slicker_LDFLAGS = $(all_libraries) $(KDE_RPATH)
 
Index: slicker/applets/cards/clockcard/Makefile.am
===================================================================
RCS file: /cvsroot/slicker/slicker/slicker/applets/cards/clockcard/Makefile.am,v
retrieving revision 1.1
diff -u -r1.1 Makefile.am
--- slicker/applets/cards/clockcard/Makefile.am	6 Feb 2003 18:33:09 -0000	1.1
+++ slicker/applets/cards/clockcard/Makefile.am	12 Feb 2003 18:41:56 -0000
@@ -9,7 +9,7 @@
 
 
 libclockcardapplet_la_LDFLAGS = -version-info 0:0:0
-
+libclockcardapplet_la_LIBADD  = $(top_builddir)/slicker/carddeskcore/libcarddeskcore.la
 libclockcardapplet_la_METASOURCES = AUTO
 
 libclockcardapplet_la_SOURCES = settings.cpp datepicker.cpp conf.ui clock.cpp 
Index: slicker/applets/cards/testapplet/Makefile.am
===================================================================
RCS file: /cvsroot/slicker/slicker/slicker/applets/cards/testapplet/Makefile.am,v
retrieving revision 1.1
diff -u -r1.1 Makefile.am
--- slicker/applets/cards/testapplet/Makefile.am	6 Feb 2003 18:33:09 -0000	1.1
+++ slicker/applets/cards/testapplet/Makefile.am	12 Feb 2003 18:41:56 -0000
@@ -9,7 +9,7 @@
 
 
 libtestapplet_la_LDFLAGS = -version-info 0:0:0
-
+libtestapplet_la_LIBADD  = $(top_builddir)/slicker/carddeskcore/libcarddeskcore.la
 libtestapplet_la_METASOURCES = AUTO
 
 libtestapplet_la_SOURCES = TestCardApplet.cpp plugin.cpp 
Index: slicker/applets/cards/traytestapplet/Makefile.am
===================================================================
RCS file: /cvsroot/slicker/slicker/slicker/applets/cards/traytestapplet/Makefile.am,v
retrieving revision 1.1
diff -u -r1.1 Makefile.am
--- slicker/applets/cards/traytestapplet/Makefile.am	6 Feb 2003 18:33:10 -0000	1.1
+++ slicker/applets/cards/traytestapplet/Makefile.am	12 Feb 2003 18:41:56 -0000
@@ -9,7 +9,7 @@
 
 
 libtraytestapplet_la_LDFLAGS = -version-info 0:0:0
-
+libtraytestapplet_la_LIBADD  = $(top_builddir)/slicker/carddeskcore/libcarddeskcore.la
 libtraytestapplet_la_METASOURCES = AUTO
 
 libtraytestapplet_la_SOURCES = TrayTestApplet.cpp plugin.cpp 
Index: slicker/carddeskcore/Makefile.am
===================================================================
RCS file: /cvsroot/slicker/slicker/slicker/carddeskcore/Makefile.am,v
retrieving revision 1.1
diff -u -r1.1 Makefile.am
--- slicker/carddeskcore/Makefile.am	6 Feb 2003 18:17:34 -0000	1.1
+++ slicker/carddeskcore/Makefile.am	12 Feb 2003 18:41:56 -0000
@@ -8,8 +8,8 @@
 LDFLAGS =  
 
 
-libcarddeskcore_la_LDFLAGS = -version-info 0:0:0
-
+libcarddeskcore_la_LDFLAGS = -version-info 0:0:0 $(QT_LDFLAGS)
+libcarddeskcore_la_LIBADD  = $(LIB_QT) -lkdecore -lkdefx -lkdeui
 libcarddeskcore_la_METASOURCES = AUTO
 
 libcarddeskcore_la_SOURCES = NewCardDialog.cpp PluginListBox.cpp PluginRoster.cpp LayoutSwitcher.cpp DragLogic.cpp Deck.cpp Const.cpp CardManager.cpp CardElements.cpp CardApplet.cpp Card.cpp 
