diff -ruN qt-gui-1.0.4.orig/configure qt-gui-1.0.4/configure
--- qt-gui-1.0.4.orig/configure	Wed Nov 28 14:40:59 2001
+++ qt-gui-1.0.4/configure	Mon Apr  1 12:37:16 2002
@@ -1216,7 +1216,7 @@
 echo "${ECHO_T}yes" >&6
 
 # expand $ac_aux_dir to an absolute path
-am_aux_dir=`CDPATH=:; cd $ac_aux_dir && pwd`
+am_aux_dir=`CDPATH=:; cd $ac_aux_dir` # removed cause zsh sucks -&&  pwd`-
 
 test x"${MISSING+set}" = xset || MISSING="\${SHELL} $am_aux_dir/missing"
 # Use eval to expand $SHELL
@@ -5822,7 +5822,7 @@
       fi
 
       for qt_dir in $qt_library_dirs; do
-        if test -r $qt_dir/libqt-mt.so.2 || test -r $qt_dir/libqt-mt.so.3; then
+        if test -r $qt_dir/libqt-mt.2.dylib || test -r $qt_dir/libqt-mt.3.dylib; then
           ac_qt_libname=-lqt-mt
           ac_qt_libdir=$qt_dir
           break
@@ -5831,7 +5831,7 @@
 
       if test -z "$qt_libdir"; then
         for qt_dir in $qt_library_dirs; do
-          if test -r $qt_dir/libqt.so.2 || test -r $qt_dir/libqt.so.3; then
+          if test -r $qt_dir/libqt.2.dylib || test -r $qt_dir/libqt.3.dylib; then
             ac_qt_libname=-lqt
             ac_qt_libdir=$qt_dir
             break
