diff -ruN console-1.0.4.orig/configure console-1.0.4/configure
--- console-1.0.4.orig/configure	Wed Nov 28 14:41:04 2001
+++ console-1.0.4/configure	Mon Apr  1 21:34:52 2002
@@ -1206,7 +1206,7 @@
 echo "${ECHO_T}yes" >&6
 
 # expand $ac_aux_dir to an absolute path
-am_aux_dir=`CDPATH=:; cd $ac_aux_dir && pwd`
+am_aux_dir=`CDPATH=:; cd $ac_aux_dir `#zsh sux! && pwd`
 
 test x"${MISSING+set}" = xset || MISSING="\${SHELL} $am_aux_dir/missing"
 # Use eval to expand $SHELL
