diff -uNr xsp-1.0.6/test/2.0/treeview/Makefile.in xsp-1.0.6-new/test/2.0/treeview/Makefile.in
--- xsp-1.0.6/test/2.0/treeview/Makefile.in	Wed Feb  2 19:51:08 2005
+++ xsp-1.0.6-new/test/2.0/treeview/Makefile.in	Sun Mar 20 12:32:27 2005
@@ -350,12 +350,12 @@
 	$(GMCS) -r:System.Web.dll -t:library -out:$@ $^
 
 dist-hook:
-	mkdir $(distdir)/TreeLineImages
-	cp -p $(srcdir)/TreeLineImages/*gif $(distdir)/TreeLineImages
+	mkdir -p $(DESTDIR)$(distdir)/TreeLineImages
+	cp -p $(srcdir)/TreeLineImages/*gif $(DESTDIR)$(distdir)/TreeLineImages
 
 install-data-hook:
-	mkdir $(datadir)/doc/xsp/test/2.0/treeview/TreeLineImages
-	cp $(srcdir)/TreeLineImages/*gif $(datadir)/doc/xsp/test/2.0/treeview/TreeLineImages
+	mkdir -p $(DESTDIR)$(datadir)/doc/xsp/test/2.0/treeview/TreeLineImages
+	cp $(srcdir)/TreeLineImages/*gif $(DESTDIR)$(datadir)/doc/xsp/test/2.0/treeview/TreeLineImages
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
 .NOEXPORT:
