diff -uNbr kio_fish-1.1.2/aclocal.m4 kio_fish-1.1.2-new/aclocal.m4
--- kio_fish-1.1.2/aclocal.m4	Thu Mar 21 22:05:52 2002
+++ kio_fish-1.1.2-new/aclocal.m4	Thu Jun  6 21:37:54 2002
@@ -10219,9 +10219,9 @@
 # Some tools Automake needs.
 AC_REQUIRE([AM_SANITY_CHECK])dnl
 AC_REQUIRE([AC_ARG_PROGRAM])dnl
-AM_MISSING_PROG(ACLOCAL, aclocal)
+AM_MISSING_PROG(ACLOCAL, aclocal-1.5)
 AM_MISSING_PROG(AUTOCONF, autoconf)
-AM_MISSING_PROG(AUTOMAKE, automake)
+AM_MISSING_PROG(AUTOMAKE, automake-1.5)
 AM_MISSING_PROG(AUTOHEADER, autoheader)
 AM_MISSING_PROG(MAKEINFO, makeinfo)
 AM_MISSING_PROG(AMTAR, tar)
diff -uNbr kio_fish-1.1.2/admin/Makefile.common kio_fish-1.1.2-new/admin/Makefile.common
--- kio_fish-1.1.2/admin/Makefile.common	Thu Mar 21 18:26:38 2002
+++ kio_fish-1.1.2-new/admin/Makefile.common	Thu Jun  6 21:37:54 2002
@@ -2,6 +2,23 @@
 SHELL=/bin/sh
 
 cvs:
+	if test -x /usr/bin/autoconf2.50 ; then \
+	   AUTOCONF="/usr/bin/autoconf2.50" ;\
+	else \
+	   AUTOCONF="autoconf" ;\
+	fi; \
+	AUTOCONF_VERSION=`$$AUTOCONF --version` ;\
+	if test "x$$AUTOCONF_VERSION" = "xAutoconf version 2.13"; then \
+	   echo "*** YOU'RE USING AUTOCONF 2.13. We suggest updating." ;\
+	   echo "*** autoconf 2.5x works best for now, we will drop" ;\
+	   echo "*** support for autoconf 2.13 soon."; \
+	   test ! -f admin/libtool.m4.in || mv admin/libtool.m4.in admin/new-libtool.m4.in ;\
+	   test ! -f admin/ltmain.sh || mv admin/ltmain.sh admin/new-ltmain.sh ;\
+	   for file in libtool.m4.in ltcf-c.sh ltcf-cxx.sh ltcf-gcj.sh ltconfig ltmain.sh; do \
+		rm -f admin/$$file ;\
+		ln -s old-$$file admin/$$file ;\
+	   done ;\
+	fi 
 	@if grep '$$(top_srcdir)/acinclude.m4:' Makefile.am >/dev/null; then \
 	  echo "*** Creating acinclude.m4" ;\
 	  rm -f acinclude.m4 configure.files ;\
@@ -17,7 +34,7 @@
 	  $(MAKE) -f Makefile.am top_srcdir=. configure.in ;\
 	fi
 	@echo "*** Creating aclocal.m4"
-	@aclocal
+	@aclocal-1.5
 	@echo "*** Creating configure"
 	@autoconf
 	@if test -r configure.in.in ; then \
@@ -35,7 +52,7 @@
 		autoheader ;\
 	fi
 	@echo "*** Creating Makefile templates"
-	@automake
+	@automake-1.5
 	@echo "*** Postprocessing Makefile templates"
 	@perl admin/am_edit
 	@if egrep "^cvs-local:" Makefile.am >/dev/null; then \
@@ -54,9 +71,9 @@
 	@if test -r configure.in.in; then \
 	  $(MAKE) -f Makefile.am top_srcdir=. subdirs configure.in; \
 	fi ;\
-	aclocal; \
+	aclocal-1.5; \
 	autoheader; \
-	automake --foreign --include-deps; \
+	automake-1.5 --foreign --include-deps; \
 	perl admin/am_edit; \
 	autoconf; \
 	if test -r configure.in.in ; then \
@@ -213,7 +230,7 @@
 	   modulename=`pwd`; modulename=`basename $$modulename`; \
 	fi ;\
 	if test -z "$$VERSION" || test "$$VERSION" = "@VERSION@"; then \
-	     VERSION="\"2.97\"";  \
+	     VERSION="\"3.0.1\"";  \
 	fi ;\
 	if test -n "$$kde_use_qt_param"; then \
 	      sed -e "s#^dnl KDE_USE_QT#KDE_USE_QT($$kde_use_qt_param)#" configure.in.new > configure.in \
@@ -264,7 +281,9 @@
 	done ;\
 	rm -f _SUBDIRS ;\
 	for i in $$dirs; do \
+	if [ $$i != "kdm" ]; then \
 	echo $$i >> ./_SUBDIRS; \
+        fi \
 	done
 	@if test -r subdirs && diff subdirs _SUBDIRS > /dev/null; then \
 	  rm -f _SUBDIRS; \
diff -uNbr kio_fish-1.1.2/admin/acinclude.m4.in kio_fish-1.1.2-new/admin/acinclude.m4.in
--- kio_fish-1.1.2/admin/acinclude.m4.in	Thu Mar 21 18:26:38 2002
+++ kio_fish-1.1.2-new/admin/acinclude.m4.in	Thu Jun  6 21:38:05 2002
@@ -693,6 +693,13 @@
    AC_CHECK_FUNC(shmat, ,
      AC_CHECK_LIB(ipc, shmat, X_EXTRA_LIBS="$X_EXTRA_LIBS -lipc"))
    
+   # darwin needs this to initialize the environment
+   AC_CHECK_HEADERS(crt_externs.h)
+   AC_CHECK_FUNC(_NSGetEnviron, [AC_DEFINE(HAVE_NSGETENVIRON, 1, [Define if your system needs _NSGetEnviron to set up the environment])])
+
+   # darwin requires a poll.h emulation library
+   AC_CHECK_LIB(poll, poll, LIB_POLL="-lpoll")
+
    # Solaris 2.6 and others need -lresolv for res_init
    AC_CHECK_FUNCS(res_init, , [
      kde_libs_safe="$LIBS"
@@ -1526,6 +1533,10 @@
   kde_includes=${prefix}/include
   ac_kde_includes=$prefix/include
 
+  if test "${libdir}" != NONE; then
+      kde_libraries=${libdir}
+      ac_kde_libraries=$libdir
+  else
   if test "${exec_prefix}" != NONE; then
     kde_libraries=${exec_prefix}/lib
     ac_kde_libraries=$exec_prefix/lib
@@ -1533,6 +1544,7 @@
     kde_libraries=${prefix}/lib
     ac_kde_libraries=$prefix/lib
   fi
+  fi
 else
   ac_kde_includes=
   ac_kde_libraries=
@@ -1553,7 +1565,7 @@
 
 if test -z "$1"; then
 
-kde_incdirs="/usr/lib/kde/include /usr/local/kde/include /usr/local/include /usr/kde/include /usr/include/kde /usr/include /opt/kde3/include /opt/kde/include $x_includes $qt_includes"
+kde_incdirs="/usr/lib/kde/include /usr/local/kde/include /usr/local/include /usr/kde/include /usr/include/kde /usr/include /opt/kde3/include /opt/kde/include $x_includes $qt_includes $includedir"
 test -n "$KDEDIR" && kde_incdirs="$KDEDIR/include $KDEDIR/include/kde $KDEDIR $kde_incdirs"
 kde_incdirs="$ac_kde_includes $kde_incdirs"
 AC_FIND_FILE($kde_check_header, $kde_incdirs, kde_incdir)
@@ -1567,7 +1579,7 @@
 
 kde_libdirs="/usr/lib/kde/lib /usr/local/kde/lib /usr/kde/lib /usr/lib/kde /usr/lib/kde3 /usr/lib /usr/X11R6/lib /usr/local/lib /opt/kde3/lib /opt/kde/lib /usr/X11R6/kde/lib"
 test -n "$KDEDIR" && kde_libdirs="$KDEDIR/lib $KDEDIR $kde_libdirs"
-kde_libdirs="$ac_kde_libraries $kde_libdirs"
+kde_libdirs="$ac_kde_libraries $libdir $kde_libdirs"
 AC_FIND_FILE($kde_check_lib, $kde_libdirs, kde_libdir)
 ac_kde_libraries="$kde_libdir"
 
@@ -1608,7 +1620,12 @@
   AC_MSG_RESULT([will be installed in $ac_kde_prefix and $ac_kde_exec_prefix])
  fi
 
+ if test "$libdir" = NONE; then
  kde_libraries="${ac_kde_exec_prefix}/lib"
+ else
+  kde_libraries="${libdir}"
+ fi
+
  kde_includes=${ac_kde_prefix}/include
 
 else
@@ -1826,6 +1843,32 @@
 if test $kde_qtver = 3; then
    LIB_KDECORE='-lkdecore'
    AC_SUBST(LIB_KDECORE)
+   LIB_KDEFX='-lkdefx'
+   AC_SUBST(LIB_KDEFX)
+   LIB_ARTSKDE='-lartskde'
+   AC_SUBST(LIB_ARTSKDE)
+   LIB_KABC='-lkabc'
+   AC_SUBST(LIB_KABC)
+   LIB_KDESASL='-lkdesasl'
+   AC_SUBST(LIB_KDESASL)
+   LIB_KDESU='-lkdesu'
+   AC_SUBST(LIB_KDESU)
+   LIB_KJAVA='-lkjava'
+   AC_SUBST(LIB_KJAVA)
+   LIB_KJS='-lkjs'
+   AC_SUBST(LIB_KJS)
+   LIB_KMID='-lkmid'
+   AC_SUBST(LIB_KMID)
+   LIB_KSCRIPT='-lkscript'
+   AC_SUBST(LIB_KSCRIPT)
+   LIB_KTEXTEDITOR='-lktexteditor'
+   AC_SUBST(LIB_KTEXTEDITOR)
+   LIB_SHELLSCRIPT='-lshellscript'
+   AC_SUBST(LIB_SHELLSCRIPT)
+   LIB_VCARD='-lvcard'
+   AC_SUBST(LIB_VCARD)
+   LIB_DCOP='-lDCOP'
+   AC_SUBST(LIB_DCOP)
    LIB_KDEUI='-lkdeui'
    AC_SUBST(LIB_KDEUI)
    LIB_KIO='-lkio'
@@ -1842,6 +1885,8 @@
    AC_SUBST(LIB_KPARTS)
    LIB_KDEPRINT='-lkdeprint'
    AC_SUBST(LIB_KDEPRINT)
+   LIB_KDED='@PREFIX@/lib/kded.la'
+   AC_SUBST(LIB_KDED)
 # these are for backward compatibility
    LIB_KSYCOCA='-lkio'
    AC_SUBST(LIB_KSYCOCA)
@@ -4158,7 +4203,7 @@
 AC_DEFUN(KDE_CREATE_SUBDIRSLIST,
 [
 
-DO_NOT_COMPILE="$DO_NOT_COMPILE CVS debian bsd-port admin"
+DO_NOT_COMPILE="$DO_NOT_COMPILE CVS debian bsd-port admin kdm ksysguard"
 
 if test ! -s $srcdir/subdirs; then
   dnl Note: Makefile.common creates subdirs, so this is just a fallback
@@ -4369,7 +4414,7 @@
 
   if test "x$prefix" = "xNONE"; then
     prefix=$ac_default_prefix
-    ac_configure_args="$ac_configure_args --prefix $prefix"
+    ac_configure_args="$ac_configure_args --prefix=$prefix"
   fi
   KDE_FAST_CONFIGURE
   KDE_CONF_FILES
diff -uNbr kio_fish-1.1.2/admin/am_edit kio_fish-1.1.2-new/admin/am_edit
--- kio_fish-1.1.2/admin/am_edit	Thu Mar 21 18:26:38 2002
+++ kio_fish-1.1.2-new/admin/am_edit	Thu Jun  6 21:37:54 2002
@@ -285,7 +285,8 @@
         appendLines ("$cleantarget-metasources:\n\t-rm -f $cleanMoc\n");
         $target_adds{"$cleantarget-am"} .= "$cleantarget-metasources ";
     }
-    tag_DIST();
+    
+    tag_DIST() unless ($kdeopts{"noautodist"});
 
     if ($idl_output) {
         appendLines ("$cleantarget-idl:\n\t-rm -f $idl_output\n");
@@ -430,7 +431,7 @@
             $MakefileData =~ s/$lookup/$1 --tag=CXX $2/;
         }
 
-        $lookup = '(\n[^#].*\$\(LIBTOOL\) --mode=compile) (\$\(CXX\).*\$\(KCXXFLAGS\))';
+        $lookup = '(\n[^#].*\$\(LIBTOOL\) --mode=compile)\s+(\$\(CXX\)\s+)';
         if ($MakefileData =~ m/$lookup/ ) {
             $MakefileData =~ s/$lookup/$1 --tag=CXX $2/;
         }
@@ -490,6 +491,7 @@
     %kdeopts = ();
     $kdeopts{"foreign"} = 0;
     $kdeopts{"qtonly"} = 0;
+    $kdeopts{"noautodist"} = 0;
     $kdeopts{"foreign-libtool"} = $foreign_libtool;
     $kdeopts{"nofinal"} = !$use_final; # default
 
diff -uNbr kio_fish-1.1.2/admin/conf.change.pl kio_fish-1.1.2-new/admin/conf.change.pl
--- kio_fish-1.1.2/admin/conf.change.pl	Thu Mar 21 18:26:38 2002
+++ kio_fish-1.1.2-new/admin/conf.change.pl	Thu Jun  6 21:37:54 2002
@@ -6,11 +6,27 @@
 # autoconf function (which are GPL) into our LGPL acinclude.m4.in
 # written by Michael Matz <matz@kde.org>
 # adapted by Dirk Mueller <mueller@kde.org>
+#
+#   This file is free software; you can redistribute it and/or
+#   modify it under the terms of the GNU Library General Public
+#   License as published by the Free Software Foundation; either
+#   version 2 of the License, or (at your option) any later version.
+
+#   This library is distributed in the hope that it will be useful,
+#   but WITHOUT ANY WARRANTY; without even the implied warranty of
+#   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+#   Library General Public License for more details.
+
+#   You should have received a copy of the GNU Library General Public License
+#   along with this library; see the file COPYING.LIB.  If not, write to
+#   the Free Software Foundation, Inc., 59 Temple Place - Suite 330,
+#   Boston, MA 02111-1307, USA.
 
 # we have to change two places
 # 1. the splitting of the substitutions into chunks of 90 (or even 48 in
 #    later autoconf's
 # 2. the big main loop which patches all Makefile.in's
+
 use File::Basename;
 
 my $ac_aux_dir = dirname($0);
diff -uNbr kio_fish-1.1.2/admin/config.pl kio_fish-1.1.2-new/admin/config.pl
--- kio_fish-1.1.2/admin/config.pl	Thu Mar 21 18:26:38 2002
+++ kio_fish-1.1.2-new/admin/config.pl	Thu Jun  6 21:37:54 2002
@@ -12,10 +12,21 @@
 #
 # written by Michael Matz <matz@kde.org>
 # adapted by Dirk Mueller <mueller@kde.org>
-#
-# the first part was done by looking at the config.status files generated
-# by configure.
-# 
+
+#   This file is free software; you can redistribute it and/or
+#   modify it under the terms of the GNU Library General Public
+#   License as published by the Free Software Foundation; either
+#   version 2 of the License, or (at your option) any later version.
+
+#   This library is distributed in the hope that it will be useful,
+#   but WITHOUT ANY WARRANTY; without even the implied warranty of
+#   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+#   Library General Public License for more details.
+
+#   You should have received a copy of the GNU Library General Public License
+#   along with this library; see the file COPYING.LIB.  If not, write to
+#   the Free Software Foundation, Inc., 59 Temple Place - Suite 330,
+#   Boston, MA 02111-1307, USA.
 
 my $ac_subs=$ARGV[0];
 my $ac_sacfiles = $ARGV[1];
diff -uNbr kio_fish-1.1.2/admin/debianrules kio_fish-1.1.2-new/admin/debianrules
--- kio_fish-1.1.2/admin/debianrules	Thu Mar 21 18:26:38 2002
+++ kio_fish-1.1.2-new/admin/debianrules	Thu Jun  6 21:37:54 2002
@@ -7,34 +7,36 @@
 $origPwd = `pwd`;
 chomp $origPwd;
 
+$kde_confdir		=	"/etc/kde3";
+$kde_prefix		=	"/usr";
+$kde_bindir		=	"$kde_prefix/bin";
+$kde_includedir		=	"$kde_prefix/include/kde";
+$kde_libdir		=	"$kde_prefix/lib";
+$kde_cgidir		=	"$kde_prefix/lib/cgi-bin";
+$kde_moduledir		=	"$kde_prefix/lib/kde3";
+$kde_appsdir		=	"$kde_prefix/share/applnk";
+$kde_datadir		=	"$kde_prefix/share/apps";
+$kde_htmldir		=	"$kde_prefix/share/doc/kde/HTML";
+$kde_icondir		=	"$kde_prefix/share/icons";
+$kde_locale		=	"$kde_prefix/share/locale";
+$kde_mimedir		=	"$kde_prefix/share/mimelnk";
+$kde_servicesdir	=	"$kde_prefix/share/services";
+$kde_servicetypesdir	=	"$kde_prefix/share/servicetypes";
+$kde_sounddir		=	"$kde_prefix/share/sounds";
+$kde_templatesdir	=	"$kde_prefix/share/templates";
+$kde_wallpaperdir	=	"$kde_prefix/share/wallpapers";
 
-$QTDIR="/usr/share/qt";
-$kde_prefix="/usr";
-$kde_htmldir="$kde_prefix/share/doc/kde/HTML";
-$kde_icondir="$kde_prefix/share/icons";
-$kde_sounddir="$kde_prefix/share/sounds";
-$kde_datadir="$kde_prefix/share/apps";
-$kde_locale="$kde_prefix/share/locale";
-$kde_confdir="/etc/kde3";
-$kde_cgidir="$kde_prefix/lib/cgi-bin";
-$kde_appsdir="$kde_prefix/share/applnk";
-$kde_mimedir="$kde_prefix/share/mimelnk";
-$kde_wallpaperdir="$kde_prefix/share/wallpapers";
-$kde_bindir="$kde_prefix/bin";
-$kde_servicesdir="$kde_prefix/share/services";
-$kde_servicetypesdir="$kde_prefix/share/servicetypes";
-$kde_templatesdir="$kde_prefix/share/templates";
-$kde_includedir="$kde_prefix/include/kde";
-$kde_libdir="$kde_prefix/lib";
-$kde_moduledir="$kde_prefix/lib/kde3";
-$mandir="$kde_prefix/share/man"; 
-$infodir="$kde_prefix/share/info";
-$IDL="$kde_bindir/cuteidl";
-$DCOPIDL="$kde_bindir/dcopidl";
-$DCOPIDL2CPP="$kde_bindir/dcopidl2cpp";
-$KDB2HTML="$kde_bindir/kdb2html";
-$MCOPIDL="$kde_bindir/mcopidl";
-$ARTSCCONFIG="$kde_bindir/artsc-config";
+$mandir			=	"$kde_prefix/share/man";
+$infodir		=	"$kde_prefix/share/info";
+$sysconfdir		=	"/etc";
+
+$QTDIR			=	"/usr/share/qt";
+$IDL			=	"$kde_bindir/cuteidl";
+$DCOPIDL		=	"$kde_bindir/dcopidl";
+$DCOPIDL2CPP		=	"$kde_bindir/dcopidl2cpp";
+$KDB2HTML		=	"$kde_bindir/kdb2html";
+$MCOPIDL		=	"$kde_bindir/mcopidl";
+$ARTSCCONFIG		=	"$kde_bindir/artsc-config";
 
 if (defined $ENV{DEB_BUILD_OPTIONS} &&
     $ENV{DEB_BUILD_OPTIONS} =~ /\bdebug\b/) {
@@ -47,37 +49,66 @@
 }
 
 if (@ARGV && $ARGV[0] eq 'echodirs') {
-  print STDOUT "export kde_htmldir=$kde_htmldir\n";
+  print STDOUT "export kde_confdir=$kde_confdir\n";
+  print STDOUT "export kde_prefix=$kde_prefix\n";
+  print STDOUT "export kde_bindir=$kde_bindir\n";
+  print STDOUT "export kde_includedir=$kde_includedir\n";
+  print STDOUT "export kde_libdir=$kde_libdir\n";
+  print STDOUT "export kde_cgidir=$kde_cgidir\n";
+  print STDOUT "export kde_moduledir=$kde_moduledir\n";
   print STDOUT "export kde_appsdir=$kde_appsdir\n";
-  print STDOUT "export kde_icondir=$kde_icondir\n";
-  print STDOUT "export kde_sounddir=$kde_sounddir\n";
   print STDOUT "export kde_datadir=$kde_datadir\n";
-  print STDOUT "export kde_cgidir=$kde_cgidir\n";
+  print STDOUT "export kde_htmldir=$kde_htmldir\n";
+  print STDOUT "export kde_icondir=$kde_icondir\n";
   print STDOUT "export kde_locale=$kde_locale\n";
-  print STDOUT "export kde_confdir=$kde_confdir\n";
   print STDOUT "export kde_mimedir=$kde_mimedir\n";
-  print STDOUT "export kde_wallpaperdir=$kde_wallpaperdir\n";
-  print STDOUT "export kde_bindir=$kde_bindir\n";
   print STDOUT "export kde_servicesdir=$kde_servicesdir\n";
   print STDOUT "export kde_servicetypesdir=$kde_servicetypesdir\n";
+  print STDOUT "export kde_sounddir=$kde_sounddir\n";
   print STDOUT "export kde_templatesdir=$kde_templatesdir\n";
-  print STDOUT "export kde_includedir=$kde_includedir\n";
-  print STDOUT "export kde_libdir=$kde_libdir\n";
-  print STDOUT "export kde_moduledir=$kde_moduledir\n";
+  print STDOUT "export kde_wallpaperdir=$kde_wallpaperdir\n";
+
   print STDOUT "export mandir=$mandir\n";
   print STDOUT "export infodir=$infodir\n";
+  print STDOUT "export sysconfdir=$sysconfdir\n";
+
   print STDOUT "export QTDIR=$QTDIR\n";
-  print STDOUT "export kde_prefix=$kde_prefix\n";
-  print STDOUT "export INSTALL_DATA=install -p -c -m 644\n";
   print STDOUT "export IDL=$IDL\n";
   print STDOUT "export DCOPIDL=$DCOPIDL\n";
   print STDOUT "export DCOPIDL2CPP=$DCOPIDL2CPP\n";
   print STDOUT "export KDB2HTML=$KDB2HTML\n";
   print STDOUT "export MCOPIDL=$MCOPIDL\n";
   print STDOUT "export ARTSCCONFIG=$ARTSCCONFIG\n";
-  print STDOUT "configkde=$enable_debug $enable_final --disable-rpath --prefix=\$(kde_prefix) --libexecdir=\$(kde_bindir) --libdir=\$(kde_libdir) --includedir=\$(kde_includedir) --with-qt-includes=/usr/include/qt --mandir=\$(mandir) --infodir=\$(infodir)\n";
+  print STDOUT "export INSTALL_DATA=install -p -c -m 644\n";
+
+  print STDOUT "configkde=$enable_debug $enable_final --disable-rpath --prefix=\$(kde_prefix) --libexecdir=\$(kde_bindir) --sysconfdir=\$(sysconfdir) --libdir=\$(kde_libdir) --includedir=\$(kde_includedir) --with-qt-includes=/usr/include/qt --mandir=\$(mandir) --infodir=\$(infodir)\n";
   print STDOUT "configkdevelop=$enable_debug $enable_final --disable-rpath --enable-docbase --enable-kdoc2 --libdir=\$(kde_libdir) --includedir=\$(kde_includedir) --with-qt-includes=/usr/include/qt --mandir=\$(mandir) --with-kdelibsdoc-dir=/usr/share/doc/kdelibs3-doc/html\n";
   print STDOUT "configkdepim=$enable_debug $enable_final --disable-rpath --with-extra-includes=/usr/include/libpisock --enable-shared --prefix=\$(kde_prefix) --libexecdir=\$(kde_bindir) --with-qt-includes=/usr/include/qt --mandir=\$(mandir) --infodir=\$(infodir)\n";
+
   exit
 }
 
+if (@ARGV && $ARGV[0] eq 'echoglobals') {
+	print STDOUT "[Directories]\n";
+	print STDOUT "dir_config=$kde_confdir\n";
+	print STDOUT "dir_tmp=/tmp\n";
+	print STDOUT "dir_socket=/tmp\n";
+	print STDOUT "dir_exe=$kde_bindir\n";
+	print STDOUT "dir_lib=$kde_libdir\n";
+	print STDOUT "dir_cgi=$kde_cgidir\n";
+	print STDOUT "dir_module=$kde_moduledir\n";
+	print STDOUT "dir_apps=$kde_appsdir\n";
+	print STDOUT "dir_data=$kde_datadir\n";
+	print STDOUT "dir_html=$kde_htmldir\n";
+	print STDOUT "dir_icon=$kde_icondir\n";
+	print STDOUT "dir_locale=$kde_locale\n";
+	print STDOUT "dir_mime=$kde_mimedir\n";
+	print STDOUT "dir_services=$kde_servicesdir\n";
+	print STDOUT "dir_servicetypes=$kde_servicetypesdir\n";
+	print STDOUT "dir_sound=$kde_sounddir\n";
+	print STDOUT "dir_templates=$kde_templatesdir\n";
+	print STDOUT "dir_wallpaper=$kde_wallpaperdir\n";
+	print STDOUT "[General]\n";
+	print STDOUT "TerminalApplication=x-terminal-emulator\n";
+	exit
+}
diff -uNbr kio_fish-1.1.2/admin/depcomp kio_fish-1.1.2-new/admin/depcomp
--- kio_fish-1.1.2/admin/depcomp	Thu Mar 21 18:26:38 2002
+++ kio_fish-1.1.2-new/admin/depcomp	Thu Jun  6 21:37:54 2002
@@ -18,6 +18,11 @@
 # Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
 # 02111-1307, USA.
 
+# As a special exception to the GNU General Public License, if you
+# distribute this file as part of a program that contains a
+# configuration script generated by Autoconf, you may include it under
+# the same distribution terms that you use for the rest of that program.
+
 # Originally written by Alexandre Oliva <oliva@dcc.unicamp.br>.
 
 if test -z "$depmode" || test -z "$source" || test -z "$object"; then
@@ -50,10 +55,12 @@
 case "$depmode" in
 gcc3)
 ## gcc 3 implements dependency tracking that does exactly what
-## we want.  Yay!
-  if "$@" -MT "$object" -MF "$tmpdepfile" -MD -MP; then :
-  else
+## we want.  Yay!  Note: for some reason libtool 1.4 doesn't like
+## it if -MD -MP comes after the -MF stuff.  Hmm.
+  "$@" -MT "$object" -MD -MP -MF "$tmpdepfile"
     stat=$?
+  if test $stat -eq 0; then :
+  else
     rm -f "$tmpdepfile"
     exit $stat
   fi
@@ -73,9 +80,10 @@
   if test -z "$gccflag"; then
     gccflag=-MD,
   fi
-  if "$@" -Wp,"$gccflag$tmpdepfile"; then :
-  else
+  "$@" -Wp,"$gccflag$tmpdepfile"
     stat=$?
+  if test $stat -eq 0; then :
+  else
     rm -f "$tmpdepfile"
     exit $stat
   fi
@@ -129,16 +137,21 @@
     # Clip off the initial element (the dependent). Don't try to be
     # clever and replace this with sed code, as IRIX sed won't handle
     # lines with more than a fixed number of characters (4096 in
-    # IRIX 6.2 sed, 8192 in IRIX 6.5).
+    # IRIX 6.2 sed, 8192 in IRIX 6.5).  We also remove comment lines;
+    # the IRIX cc adds comments like `#:fec' to the end of the
+    # dependency line.
     tr ' ' '
-' < "$tmpdepfile" | sed 's/^[^\.]*\.o://' | tr '
+' < "$tmpdepfile" \
+    | sed -e 's/^.*\.o://' -e 's/#.*$//' -e '/^$/ d' | \
+    tr '
 ' ' ' >> $depfile
+    echo >> $depfile
 
+    # The second pass generates a dummy entry for each header file.
     tr ' ' '
-' < "$tmpdepfile" | \
-## Some versions of the HPUX 10.20 sed can't process this invocation
-## correctly.  Breaking it into two sed invocations is a workaround.
-      sed -e 's/^\\$//' -e '/^$/d' -e '/:$/d' | sed -e 's/$/ :/' >> "$depfile"
+' < "$tmpdepfile" \
+   | sed -e 's/^.*\.o://' -e 's/#.*$//' -e '/^$/ d' -e 's/$/:/' \
+   >> $depfile
   else
     # The sourcefile does not contain any dependencies, so just
     # store a dummy comment line, to avoid errors with the Makefile
@@ -150,8 +163,12 @@
 
 aix)
   # The C for AIX Compiler uses -M and outputs the dependencies
-  # in a .u file.
-  tmpdepfile=`echo "$object" | sed 's/\(.*\)\..*$/\1.u/'`
+  # in a .u file.  This file always lives in the current directory.
+  # Also, the AIX compiler puts `$object:' at the start of each line;
+  # $object doesn't have directory information.
+  stripped=`echo "$object" | sed -e 's,^.*/,,' -e 's/\(.*\)\..*$/\1/'`
+  tmpdepfile="$stripped.u"
+  outname="$stripped.o"
   if test "$libtool" = yes; then
     "$@" -Wc,-M
   else
@@ -166,21 +183,11 @@
   fi
 
   if test -f "$tmpdepfile"; then
-    echo "$object : \\" > "$depfile"
-
-    # Clip off the initial element (the dependent). Don't try to be
-    # clever and replace this with sed code, as IRIX sed won't handle
-    # lines with more than a fixed number of characters (4096 in
-    # IRIX 6.2 sed, 8192 in IRIX 6.5).
-    tr ' ' '
-' < "$tmpdepfile" | sed 's/^[^\.]*\.o://' | tr '
-' ' ' >> $depfile
-
-    tr ' ' '
-' < "$tmpdepfile" | \
-## Some versions of the HPUX 10.20 sed can't process this invocation
-## correctly.  Breaking it into two sed invocations is a workaround.
-      sed -e 's/^\\$//' -e '/^$/d' -e '/:$/d' | sed -e 's/$/ :/' >> "$depfile"
+    # Each line is of the form `foo.o: dependent.h'.
+    # Do two passes, one to just change these to
+    # `$object: dependent.h' and one to simply `dependent.h:'.
+    sed -e "s,^$outname:,$object :," < "$tmpdepfile" > "$depfile"
+    sed -e "s,^$outname: \(.*\)$,\1:," < "$tmpdepfile" >> "$depfile"
   else
     # The sourcefile does not contain any dependencies, so just
     # store a dummy comment line, to avoid errors with the Makefile
@@ -189,6 +196,43 @@
   fi
   rm -f "$tmpdepfile"
   ;;
+
+tru64)
+   # The Tru64 AIX compiler uses -MD to generate dependencies as a side
+   # effect.  `cc -MD -o foo.o ...' puts the dependencies into `foo.o.d'.
+   # At least on Alpha/Redhat 6.1, Compaq CCC V6.2-504 seems to put 
+   # dependencies in `foo.d' instead, so we check for that too.
+   # Subdirectories are respected.
+
+   tmpdepfile1="$object.d"
+   tmpdepfile2=`echo "$object" | sed -e 's/.o$/.d/'` 
+   if test "$libtool" = yes; then
+      "$@" -Wc,-MD
+   else
+      "$@" -MD
+   fi
+
+   stat=$?
+   if test $stat -eq 0; then :
+   else
+      rm -f "$tmpdepfile1" "$tmpdepfile2"
+      exit $stat
+   fi
+
+   if test -f "$tmpdepfile1"; then
+      tmpdepfile="$tmpdepfile1"
+   else
+      tmpdepfile="$tmpdepfile2"
+   fi
+   if test -f "$tmpdepfile"; then
+      sed -e "s,^.*\.[a-z]*:,$object:," < "$tmpdepfile" > "$depfile"
+      # That's a space and a tab in the [].
+      sed -e 's,^.*\.[a-z]*:[ 	]*,,' -e 's,$,:,' < "$tmpdepfile" >> "$depfile"
+   else
+      echo "#dummy" > "$depfile"
+   fi
+   rm -f "$tmpdepfile"
+   ;;
 
 #nosideeffect)
   # This comment above is used by automake to tell side-effect
diff -uNbr kio_fish-1.1.2/admin/libtool.m4.in kio_fish-1.1.2-new/admin/libtool.m4.in
--- kio_fish-1.1.2/admin/libtool.m4.in	Thu Mar 21 18:26:38 2002
+++ kio_fish-1.1.2-new/admin/libtool.m4.in	Thu Jun  6 21:37:54 2002
@@ -593,6 +593,12 @@
     lt_cv_sys_max_cmd_len=-1;
     ;;
 
+  darwin*)
+    # this should be enough... the length test freaks out some versions
+    # of OSX's `expr`
+    lt_cv_sys_max_cmd_len=16384;
+    ;;
+
   *)
     # If test is not a shell built-in, we'll probably end up computing a
     # maximum length that is only half of the actual maximum length, but
@@ -1876,16 +1882,7 @@
   ;;
 
 darwin* | rhapsody*)
-  lt_cv_deplibs_check_method='file_magic Mach-O dynamically linked shared library'
-  lt_cv_file_magic_cmd='/usr/bin/file -L'
-  case "$host_os" in
-  rhapsody* | darwin1.[[012]])
-    lt_cv_file_magic_test_file=`/System/Library/Frameworks/System.framework/System`
-    ;;
-  *) # Darwin 1.3 on
-    lt_cv_file_magic_test_file='/usr/lib/libSystem.dylib'
-    ;;
-  esac
+  lt_cv_deplibs_check_method='pass_all'
   ;;
 
 freebsd*)
@@ -1943,7 +1940,7 @@
 # This must be Linux ELF.
 linux*)
   case $host_cpu in
-  alpha* | hppa* | i*86 | ia64* | m68* | mips | mipsel | powerpc* | sparc* | s390* | sh*)
+  alpha* | hppa* | i*86 | ia64* | m68* | mips | mipsel | powerpc* | sparc* | s390* | sh* | x86_64* )
     lt_cv_deplibs_check_method=pass_all ;;
   *)
     # glibc up to 2.1.1 does not perform some relocations on ARM
@@ -2438,20 +2435,8 @@
       exp_sym_flag='-Bexport'
       no_entry_flag=""
     else
-      aix_use_runtimelinking=no
-
-      # Test if we are trying to use run time linking or normal
-      # AIX style linking. If -brtl is somewhere in LDFLAGS, we
-      # need to do runtime linking.
-      case $host_os in aix4.[[23]]|aix4.[[23]].*|aix5*)
-	for ld_flag in $LDFLAGS; do
-	  if (test $ld_flag = "-brtl" || test $ld_flag = "-Wl,-brtl"); then
+      # KDE requires run time linking.  Make it the default.
 	    aix_use_runtimelinking=yes
-	    break
-	  fi
-	done
-      esac
-
       exp_sym_flag='-bexport'
       no_entry_flag='-bnoentry'
     fi
@@ -2497,16 +2482,15 @@
 	shared_flag='-G'
       else
 	if test "$aix_use_runtimelinking" = yes; then
-	  shared_flag='${wl}-G'
+	  shared_flag='-qmkshrobj ${wl}-G'
 	else
-	  shared_flag='${wl}-bM:SRE'
+	  shared_flag='-qmkshrobj'
 	fi
       fi
     fi
 
-    # It seems that -bexpall does not export symbols beginning with
-    # underscore (_), so it is better to generate a list of symbols to export.
-    _LT_AC_TAGVAR(always_export_symbols, $1)=yes
+    # Let the compiler handle the export list.
+    _LT_AC_TAGVAR(always_export_symbols, $1)=no
     if test "$aix_use_runtimelinking" = yes; then
       # Warning - without using the other runtime loading flags (-brtl),
       # -berok will link without error, but may produce a broken library.
@@ -2515,7 +2499,8 @@
       _LT_AC_SYS_LIBPATH_AIX
       _LT_AC_TAGVAR(hardcode_libdir_flag_spec, $1)='${wl}-blibpath:$libdir:'"$aix_libpath"
 
-      _LT_AC_TAGVAR(archive_expsym_cmds, $1)="\$CC"' -o $output_objdir/$soname $libobjs $deplibs $compiler_flags `if test "x${allow_undefined_flag}" != "x"; then echo "${wl}${allow_undefined_flag}"; else :; fi` '"\${wl}$no_entry_flag \${wl}$exp_sym_flag:\$export_symbols $shared_flag"
+      _LT_AC_TAGVAR(archive_cmds, $1)="\$CC"' -o $output_objdir/$soname $libobjs $deplibs $compiler_flags `if test "x${allow_undefined_flag}" != "x"; then echo "${wl}${allow_undefined_flag}"; else :; fi` '" $shared_flag"
+      _LT_AC_TAGVAR(archive_expsym_cmds, $1)="\$CC"' -o $output_objdir/$soname $libobjs $deplibs $compiler_flags `if test "x${allow_undefined_flag}" != "x"; then echo "${wl}${allow_undefined_flag}"; else :; fi` '"\${wl}$exp_sym_flag:\$export_symbols $shared_flag"
      else
       if test "$host_cpu" = ia64; then
 	_LT_AC_TAGVAR(hardcode_libdir_flag_spec, $1)='${wl}-R $libdir:/usr/lib:/lib'
@@ -2547,6 +2532,11 @@
 	;;
     esac
     ;;
+  darwin*)
+    shared_flag='-dynamiclib'
+    _LT_AC_TAGVAR(archive_cmds, $1)='$CC -r -keep_private_externs -nostdlib -o ${lib}-master.o $libobjs && $CC -dynamiclib -install_name $rpath/$soname $predep_objects ${lib}-master.o $deplibs $postdep_objects $compiler_flags -o $lib'
+    output_verbose_link_cmd='$CC -dynamiclib $CFLAGS -v conftest.$objext 2>&1 | egrep "\-L"'
+    ;;
   dgux*)
     case $cc_basename in
       ec++)
@@ -4611,20 +4601,9 @@
 	else
 	  _LT_AC_TAGVAR(export_symbols_cmds, $1)='$NM -BCpg $libobjs $convenience | awk '\''{ if (((\[$]2 == "T") || (\[$]2 == "D") || (\[$]2 == "B")) && ([substr](\[$]3,1,1) != ".")) { print \[$]3 } }'\'' | sort -u > $export_symbols'
 	fi
-	aix_use_runtimelinking=no
 
-	# Test if we are trying to use run time linking or normal
-	# AIX style linking. If -brtl is somewhere in LDFLAGS, we
-	# need to do runtime linking.
-	case $host_os in aix4.[[23]]|aix4.[[23]].*|aix5*)
-	  for ld_flag in $LDFLAGS; do
-  	  if (test $ld_flag = "-brtl" || test $ld_flag = "-Wl,-brtl"); then
+	# KDE requires run time linking.  Make it the default.
   	    aix_use_runtimelinking=yes
-  	    break
-  	  fi
-	  done
-	esac
-
 	exp_sym_flag='-bexport'
 	no_entry_flag='-bnoentry'
       fi
@@ -4670,16 +4649,15 @@
 	  shared_flag='-G'
 	else
   	if test "$aix_use_runtimelinking" = yes; then
-	    shared_flag='${wl}-G'
+	    shared_flag='-qmkshrobj ${wl}-G'
 	  else
-	    shared_flag='${wl}-bM:SRE'
+	    shared_flag='-qmkshrobj'
   	fi
 	fi
       fi
 
-      # It seems that -bexpall does not export symbols beginning with
-      # underscore (_), so it is better to generate a list of symbols to export.
-      _LT_AC_TAGVAR(always_export_symbols, $1)=yes
+      # Let the compiler handle the export list.
+      _LT_AC_TAGVAR(always_export_symbols, $1)=no
       if test "$aix_use_runtimelinking" = yes; then
 	# Warning - without using the other runtime loading flags (-brtl),
 	# -berok will link without error, but may produce a broken library.
@@ -4687,7 +4665,8 @@
        # Determine the default libpath from the value encoded in an empty executable.
        _LT_AC_SYS_LIBPATH_AIX
        _LT_AC_TAGVAR(hardcode_libdir_flag_spec, $1)='${wl}-blibpath:$libdir:'"$aix_libpath"
-	_LT_AC_TAGVAR(archive_expsym_cmds, $1)="\$CC"' -o $output_objdir/$soname $libobjs $deplibs $compiler_flags `if test "x${allow_undefined_flag}" != "x"; then echo "${wl}${allow_undefined_flag}"; else :; fi` '"\${wl}$no_entry_flag \${wl}$exp_sym_flag:\$export_symbols $shared_flag"
+	_LT_AC_TAGVAR(archive_cmds, $1)="\$CC"' -o $output_objdir/$soname $libobjs $deplibs $compiler_flags `if test "x${allow_undefined_flag}" != "x"; then echo "${wl}${allow_undefined_flag}"; else :; fi` '" $shared_flag"
+	_LT_AC_TAGVAR(archive_expsym_cmds, $1)="\$CC"' -o $output_objdir/$soname $libobjs $deplibs $compiler_flags `if test "x${allow_undefined_flag}" != "x"; then echo "${wl}${allow_undefined_flag}"; else :; fi` '"\${wl}$exp_sym_flag:\$export_symbols $shared_flag"
        else
 	if test "$host_cpu" = ia64; then
 	  _LT_AC_TAGVAR(hardcode_libdir_flag_spec, $1)='${wl}-R $libdir:/usr/lib:/lib'
@@ -4748,7 +4727,9 @@
 	_LT_AC_TAGVAR(allow_undefined_flag, $1)='-undefined suppress'
 	;;
       *) # Darwin 1.3 on
-	_LT_AC_TAGVAR(allow_undefined_flag, $1)='-flat_namespace -undefined suppress'
+	#_LT_AC_TAGVAR(allow_undefined_flag, $1)='-flat_namespace -undefined suppress'
+        # testing with twolevel namespaces
+	_LT_AC_TAGVAR(allow_undefined_flag, $1)=unsupported
 	;;
       esac
 
@@ -4756,7 +4737,7 @@
       #        cross-compilation, but unfortunately the echo tests do not
       #        yet detect zsh echo's removal of \ escapes.  Also zsh mangles
       #	       `"' quotes if we put them in here... so don't!
-      _LT_AC_TAGVAR(archive_cmds, $1)='$CC $(test .$module = .yes && echo -bundle || echo -dynamiclib) $allow_undefined_flag -o $lib $libobjs $deplibs$linker_flags -install_name $rpath/$soname $verstring'
+      _LT_AC_TAGVAR(archive_cmds, $1)='$CC -r -keep_private_externs -nostdlib -o ${lib}-master.o $libobjs && $CC $(test .$module = .yes && echo -bundle || echo -dynamiclib) -install_name $rpath/$soname $allow_undefined_flag -o $lib ${lib}-master.o $deplibs $linker_flags $verstring'
       # We need to add '_' to the symbols in $export_symbols first
       #_LT_AC_TAGVAR(archive_expsym_cmds, $1)="$_LT_AC_TAGVAR(archive_cmds, $1)"' && strip -s $export_symbols'
       _LT_AC_TAGVAR(hardcode_direct, $1)=yes
diff -uNbr kio_fish-1.1.2/admin/ltmain.sh kio_fish-1.1.2-new/admin/ltmain.sh
--- kio_fish-1.1.2/admin/ltmain.sh	Thu Mar 21 18:26:38 2002
+++ kio_fish-1.1.2-new/admin/ltmain.sh	Thu Jun  6 21:37:54 2002
@@ -828,6 +828,7 @@
     old_convenience=
     deplibs=
     old_deplibs=
+    add_flags=
     compiler_flags=
     linker_flags=
     dllsearchpath=
@@ -1392,6 +1393,7 @@
 	  arg="\"$arg\""
 	  ;;
 	esac
+	add_flags="$add_flags $arg"
 	;;
 
       *.$objext)
@@ -1526,6 +1528,7 @@
 	  arg="\"$arg\""
 	  ;;
 	esac
+	add_flags="$add_flags $arg"
 	;;
       esac # arg
 
@@ -1723,7 +1726,7 @@
 	    $echo "$modename: warning: \`-l' is ignored for archives/objects" 1>&2
 	    continue
 	  fi
-	  if test "$pass" = conv; then
+	  if test "$pass" = conv && test "$allow_undefined" = yes; then
 	    deplibs="$deplib $deplibs"
 	    continue
 	  fi
@@ -1885,16 +1888,9 @@
 	if test "$pass" = conv; then
 	  # Only check for convenience libraries
 	  deplibs="$lib $deplibs"
-	  if test -z "$libdir"; then
-	    if test -z "$old_library"; then
-	      $echo "$modename: cannot find name of link library for \`$lib'" 1>&2
-	      exit 1
-	    fi
-	    # It is a libtool convenience library, so add in its objects.
-	    convenience="$convenience $ladir/$objdir/$old_library"
-	    old_convenience="$old_convenience $ladir/$objdir/$old_library"
 	    tmp_libs=
 	    for deplib in $dependency_libs; do
+	    #echo "Adding $deplib to \$deplibs"
 	      deplibs="$deplib $deplibs"
               if test "X$duplicate_deps" = "Xyes" ; then
 	        case "$tmp_libs " in
@@ -1903,6 +1899,14 @@
               fi
 	      tmp_libs="$tmp_libs $deplib"
 	    done
+	  if test -z "$libdir"; then
+	    if test -z "$old_library"; then
+	      $echo "$modename: cannot find name of link library for \`$lib'" 1>&2
+	      exit 1
+	    fi
+	    # It is a libtool convenience library, so add in its objects.
+	    convenience="$convenience $ladir/$objdir/$old_library"
+	    old_convenience="$old_convenience $ladir/$objdir/$old_library"
 	  elif test "$linkmode" != prog && test "$linkmode" != lib; then
 	    $echo "$modename: \`$lib' is not a convenience library" 1>&2
 	    exit 1
@@ -3307,11 +3311,31 @@
 	  $run eval '(cd $output_objdir && $rm ${realname}U && $mv $realname ${realname}U)' || exit $?
 	fi
 
+	# Add all flags from the command line.  We here create a library,
+	# but those flags were only added to compile_command and
+	# finalize_command, which are only used when creating executables.
+	# So do it by hand here.
+	compiler_flags="$compiler_flags $add_flags"
+	# Only add it to commands which use CC, instead of LD, i.e.
+	# only to $compiler_flags
+	#linker_flags="$linker_flags $add_flags"
+
 	# Do each of the archive commands.
 	if test -n "$export_symbols" && test -n "$archive_expsym_cmds"; then
 	  eval cmds=\"$archive_expsym_cmds\"
 	else
+ 	  save_deplibs="$deplibs"
+ 	  for conv in $convenience; do
+            tmp_deplibs=
+            for test_deplib in $deplibs; do
+              if test "$test_deplib" != "$conv"; then
+                tmp_deplibs="$tmp_deplibs $test_deplib"
+              fi
+            done
+            deplibs="$tmp_deplibs"
+ 	  done
 	  eval cmds=\"$archive_cmds\"
+ 	  deplibs="$save_deplibs"
 	fi
 
 	if len=`expr "X$cmds" : ".*"` &&
@@ -3624,6 +3648,14 @@
 	;;
       esac
 
+      case $host in
+      *darwin*)
+        # Don't allow lazy linking, it breaks C++ global constructors
+        compile_command="$compile_command ${wl}-bind_at_load"
+        finalize_command="$finalize_command ${wl}-bind_at_load"
+        ;;
+      esac
+
       compile_command="$compile_command $compile_deplibs"
       finalize_command="$finalize_command $finalize_deplibs"
 
@@ -3909,6 +3941,17 @@
 	finalize_command=`$echo "X$finalize_command" | $Xsed -e "s% @SYMFILE@%%"`
       fi
 
+      # AIX runtime linking requires linking programs with -Wl,-brtl and libs with -Wl,-G
+      # Also add -bnolibpath to the beginning of the link line, to clear the hardcoded runpath.
+      # Otherwise, things like the -L path to libgcc.a are accidentally hardcoded by ld.
+      # This does not apply on AIX for ia64, which uses a SysV linker.
+      case "$host" in
+        ia64-*-aix5*) ;;
+        *-*-aix4* | *-*-aix5*)
+                   compile_command=`$echo "X$compile_command $wl-brtl" | $Xsed -e "s/\$CC/\$CC $wl-bnolibpath/1"`
+                   finalize_command=`$echo "X$finalize_command $wl-brtl" | $Xsed -e "s/\$CC/\$CC $wl-bnolibpath/1"` ;;
+      esac
+
       if test "$need_relink" = no || test "$build_libtool_libs" != yes; then
 	# Replace the output file specification.
 	compile_command=`$echo "X$compile_command" | $Xsed -e 's%@OUTPUT@%'"$output"'%g'`
@@ -4265,7 +4308,8 @@
 	  oldobjs="$libobjs_save"
 	  build_libtool_libs=no
 	else
-	  oldobjs="$oldobjs$old_deplibs $non_pic_objects"
+	  #oldobjs="$oldobjs$old_deplibs $non_pic_objects"
+	  oldobjs="$old_deplibs $non_pic_objects"
 	fi
 	addlibs="$old_convenience"
       fi
@@ -4306,6 +4350,8 @@
 	  oldobjs="$oldobjs "`find $xdir -name \*.${objext} -print -o -name \*.lo -print | $NL2SP`
 	done
       fi
+
+      compiler_flags="$compiler_flags $add_flags"
 
       # Do each command in the archive commands.
       if test -n "$old_archive_from_new_cmds" && test "$build_libtool_libs" = yes; then
diff -uNbr kio_fish-1.1.2/admin/missing kio_fish-1.1.2-new/admin/missing
--- kio_fish-1.1.2/admin/missing	Thu Mar 21 18:26:38 2002
+++ kio_fish-1.1.2-new/admin/missing	Thu Jun  6 21:37:54 2002
@@ -87,7 +87,7 @@
     exit 1
     ;;
 
-  aclocal)
+  aclocal*)
     echo 1>&2 "\
 WARNING: \`$1' is missing on your system.  You should only need it if
          you modified \`acinclude.m4' or \`${configure_ac}'.  You might want
@@ -124,7 +124,7 @@
     touch $touch_files
     ;;
 
-  automake)
+  automake*)
     echo 1>&2 "\
 WARNING: \`$1' is missing on your system.  You should only need it if
          you modified \`Makefile.am', \`acinclude.m4' or \`${configure_ac}'.
diff -uNbr kio_fish-1.1.2/configure kio_fish-1.1.2-new/configure
--- kio_fish-1.1.2/configure	Thu Mar 21 22:06:12 2002
+++ kio_fish-1.1.2-new/configure	Thu Jun  6 21:37:54 2002
@@ -41,10 +41,6 @@
   exit 0
 fi
 
-# The HP-UX ksh and POSIX shell print the target directory to stdout
-# if CDPATH is set.
-if test "X${CDPATH+set}" = Xset; then CDPATH=:; export CDPATH; fi
-
 if test -z "$ECHO"; then
 if test "X${echo_test_string+set}" != Xset; then
 # find a string as large as possible, as long as the shell can cope with it
@@ -226,9 +222,6 @@
 '
 IFS=" 	$as_nl"
 
-# CDPATH.
-$as_unset CDPATH || test "${CDPATH+set}" != set || { CDPATH=:; export CDPATH; }
-
 # Name of the host.
 # hostname on some systems (SVR3.2, Linux) returns a bogus exit status,
 # so uname gets run too.
@@ -1439,7 +1432,7 @@
 echo "${ECHO_T}yes" >&6
 
 # expand $ac_aux_dir to an absolute path
-am_aux_dir=`CDPATH=:; cd $ac_aux_dir && pwd`
+am_aux_dir=`cd $ac_aux_dir && pwd`
 
 test x"${MISSING+set}" = xset || MISSING="\${SHELL} $am_aux_dir/missing"
 # Use eval to expand $SHELL
@@ -1547,7 +1540,7 @@
 rmdir .deps 2>/dev/null
 
 # test to see if srcdir already configured
-if test "`CDPATH=:; cd $srcdir && pwd`" != "`pwd`" &&
+if test "`cd $srcdir && pwd`" != "`pwd`" &&
    test -f $srcdir/config.status; then
   { { echo "$as_me:1552: error: source directory already configured; run \"make distclean\" there first" >&5
 echo "$as_me: error: source directory already configured; run \"make distclean\" there first" >&2;}
@@ -1579,11 +1572,11 @@
 
 # Some tools Automake needs.
 
-ACLOCAL=${ACLOCAL-"${am_missing_run}aclocal"}
+ACLOCAL=${ACLOCAL-"${am_missing_run}aclocal-1.5"}
 
 AUTOCONF=${AUTOCONF-"${am_missing_run}autoconf"}
 
-AUTOMAKE=${AUTOMAKE-"${am_missing_run}automake"}
+AUTOMAKE=${AUTOMAKE-"${am_missing_run}automake-1.5"}
 
 AUTOHEADER=${AUTOHEADER-"${am_missing_run}autoheader"}
 
@@ -5481,6 +5474,10 @@
     lt_cv_sys_max_cmd_len=-1;
     ;;
 
+  darwin*)
+    lt_cv_sys_max_cmd_len=16384;
+    ;;
+
   *)
     # If test is not a shell built-in, we'll probably end up computing a
     # maximum length that is only half of the actual maximum length, but
@@ -8596,10 +8593,6 @@
 # Sed that helps us avoid accidentally triggering echo(1) options like -n.
 Xsed="sed -e s/^X//"
 
-# The HP-UX ksh and POSIX shell print the target directory to stdout
-# if CDPATH is set.
-if test "X\${CDPATH+set}" = Xset; then CDPATH=:; export CDPATH; fi
-
 # The names of the tagged configurations supported by this script.
 available_tags=
 
@@ -21485,9 +21478,6 @@
 as_nl='
 '
 IFS=" 	$as_nl"
-
-# CDPATH.
-$as_unset CDPATH || test "${CDPATH+set}" != set || { CDPATH=:; export CDPATH; }
 
 exec 6>&1
 
diff -uNbr kio_fish-1.1.2/fish/Makefile.am kio_fish-1.1.2-new/fish/Makefile.am
--- kio_fish-1.1.2/fish/Makefile.am	Tue Jun  4 06:25:39 2002
+++ kio_fish-1.1.2-new/fish/Makefile.am	Thu Jun  6 21:37:54 2002
@@ -15,6 +15,7 @@
 
 $(srcdir)/fish.cpp: fishcode.h
 
+	$(INSTALL) -d $(kde_servicesdir)
 fishcode.h: fish.pl
 	if echo | md5sum > /dev/null 2>&1 ; then \
 	SUM=`md5sum $(srcdir)/fish.pl | cut -d ' ' -f 1`; else \
diff -uNbr kio_fish-1.1.2/fish/fish.cpp kio_fish-1.1.2-new/fish/fish.cpp
--- kio_fish-1.1.2/fish/fish.cpp	Tue Jun  4 07:18:23 2002
+++ kio_fish-1.1.2-new/fish/fish.cpp	Thu Jun  6 21:37:54 2002
@@ -33,6 +33,7 @@
 #include <qregexp.h>
 
 #include <stdlib.h>
+#include <util.h>
 #ifdef HAVE_PTY_H
 #include <pty.h>
 #endif
