diff -uNbr kxine-0.4/admin/Makefile.common kxine-0.4-new/admin/Makefile.common
--- kxine-0.4/admin/Makefile.common	Sun Jul  7 15:46:33 2002
+++ kxine-0.4-new/admin/Makefile.common	Tue Aug 13 10:50:44 2002
@@ -20,12 +20,11 @@
 	    exit 1; \
 	  fi; \
 	fi; \
-	echo ADMINDIR is $$admindir; \
 	MAKE=$(MAKE) $(SHELL) $$admindir/cvs.sh $@
 
 package-merge:
 	@MAKE=$(MAKE) POFILES=$(POFILES) PACKAGE=$(PACKAGE) \
-	  $(SHELL) ../admin/cvs.sh package-merge
+	  $(SHELL) admin/cvs.sh package-merge
 
 configure.in: configure.files $(shell test -f configure.files && cat configure.files) subdirs
 configure.files: subdirs
diff -uNbr kxine-0.4/admin/acinclude.m4.in kxine-0.4-new/admin/acinclude.m4.in
--- kxine-0.4/admin/acinclude.m4.in	Sun Jul  7 15:46:33 2002
+++ kxine-0.4-new/admin/acinclude.m4.in	Tue Aug 13 10:50:44 2002
@@ -699,13 +699,10 @@
      AC_CHECK_LIB(ipc, shmat, X_EXTRA_LIBS="$X_EXTRA_LIBS -lipc"))
    
    # darwin needs this to initialize the environment
-   AC_CHECK_HEADERS(crt_externs.h)
+   AC_CHECK_HEADERS([crt_externs.h sys/types.h])
    AC_CHECK_FUNC(_NSGetEnviron, [AC_DEFINE(HAVE_NSGETENVIRON, 1, [Define if your system needs _NSGetEnviron to set up the environment])])
  
-   # more headers that need to be explicitly included on darwin
-   AC_CHECK_HEADERS(sys/types.h stdint.h)
-
-   # darwin requires a poll emulation library
+   # darwin requires a poll.h emulation library
    AC_CHECK_LIB(poll, poll, LIB_POLL="-lpoll")
 
    # CoreAudio framework
@@ -736,6 +733,8 @@
    AC_SUBST(FRAMEWORK_COREAUDIO)
    LIBSOCKET="$X_EXTRA_LIBS"
    AC_SUBST(LIBSOCKET)
+   AC_SUBST(FRAMEWORK_COREAUDIO)
+   AC_SUBST(LIB_POLL)
    AC_SUBST(LIBRESOLV)
    AC_SUBST(X_EXTRA_LIBS)
    AC_CHECK_LIB(ucb, killpg, [LIBUCB="-lucb"]) dnl for Solaris2.4
@@ -905,7 +904,7 @@
 kde_ldflags_safe="$LDFLAGS"
 kde_libs_safe="$LIBS"
 
-LDFLAGS="$LDFLAGS $X_LDFLAGS $USER_LDFLAGS"
+LDFLAGS="$LDFLAGS $USER_LDFLAGS $X_LDFLAGS"
 LIBS="-lXext -lX11 $LIBSOCKET"
 
 AC_TRY_LINK([
@@ -947,9 +946,9 @@
 kde_save_LDFLAGS="$LDFLAGS"
 kde_save_CFLAGS="$CFLAGS"
 kde_save_LIBS="$LIBS"
-LDFLAGS="$LDFLAGS $X_LDFLAGS $USER_LDFLAGS"
+LDFLAGS="$LDFLAGS $USER_LDFLAGS $X_LDFLAGS"
 CFLAGS="$CFLAGS -I$x_includes"
-LIBS="-lXinerama -lXext"
+LIBS="-lXinerama -lXext -lX11"
 
 if test "x$no_xinerama" = "xno"; then
 
@@ -1846,15 +1845,29 @@
    AC_REQUIRE([K_PATH_X])
 
 if test $kde_qtver = 3; then
+   AC_SUBST(LIB_ARTSKDE, "-lartskde")
+   AC_SUBST(LIB_DCOP, "-lDCOP")
+   AC_SUBST(LIB_KAB, "-lkab")
+   AC_SUBST(LIB_KABC, "-lkabc")
    AC_SUBST(LIB_KDECORE, "-lkdecore")
+   AC_SUBST(LIB_KDED, "@PREFIX@/lib/kded.la")
+   AC_SUBST(LIB_KDEFX, "-lkdefx")
+   AC_SUBST(LIB_KDEPRINT, "-lkdeprint")
+   AC_SUBST(LIB_KDESASL, "-lkdesasl")
+   AC_SUBST(LIB_KDESU, "-lkdesu")
    AC_SUBST(LIB_KDEUI, "-lkdeui")
-   AC_SUBST(LIB_KIO, "-lkio")
-   AC_SUBST(LIB_SMB, "-lsmb")
-   AC_SUBST(LIB_KAB, "-lkab")
    AC_SUBST(LIB_KHTML, "-lkhtml")
-   AC_SUBST(LIB_KSPELL, "-lkspell")
+   AC_SUBST(LIB_KIO, "-lkio")
    AC_SUBST(LIB_KPARTS, "-lkparts")
-   AC_SUBST(LIB_KDEPRINT, "-lkdeprint")
+   AC_SUBST(LIB_KJAVA, "-lkjava")
+   AC_SUBST(LIB_KJS, "-lkjs")
+   AC_SUBST(LIB_KMID, "-lkmid")
+   AC_SUBST(LIB_KSCRIPT, "-lkscript")
+   AC_SUBST(LIB_KSPELL, "-lkspell")
+   AC_SUBST(LIB_KTEXTEDITOR, "-lktexteditor")
+   AC_SUBST(LIB_SHELLSCRIPT, "-lshellscript")
+   AC_SUBST(LIB_SMB, "-lsmb")
+   AC_SUBST(LIB_VCARD, "-lvcard")
 # these are for backward compatibility
    AC_SUBST(LIB_KSYCOCA, "-lkio")
    AC_SUBST(LIB_KFILE, "-lkio")
@@ -2544,10 +2557,10 @@
       if test $kde_use_debug_code = "full"; then
         CFLAGS="-g3 $CFLAGS"
       else
-        CFLAGS="-g -O2 $CFLAGS"
+        CFLAGS="-g -Os $CFLAGS"
       fi
     else
-      CFLAGS="-O2 $CFLAGS"
+      CFLAGS="-Os $CFLAGS"
     fi
   fi
 
@@ -2577,7 +2590,7 @@
         if test "$kde_use_debug_code" = "full"; then
           CXXFLAGS="-g3 $CXXFLAGS"
         else
-          CXXFLAGS="-g -O2 $CXXFLAGS"
+          CXXFLAGS="-g -Os $CXXFLAGS"
         fi
       fi
       KDE_CHECK_COMPILER_FLAG(fno-builtin,[CXXFLAGS="-fno-builtin $CXXFLAGS"])
@@ -2589,7 +2602,7 @@
       if test "$CXX" = "KCC"; then
         CXXFLAGS="+K3 $CXXFLAGS"
       else
-        CXXFLAGS="-O2 $CXXFLAGS"
+        CXXFLAGS="-Os $CXXFLAGS"
       fi  
     fi
   fi
@@ -3070,9 +3083,9 @@
     ac_save_ldflags="$LDFLAGS"
     ac_save_cflags="$CFLAGS"
     if test "x$kde_use_qt_emb" != "xyes"; then
-      LDFLAGS="$LDFLAGS $X_LDFLAGS $USER_LDFLAGS $LDFLAGS $XPM_LDFLAGS $all_libraries -lXpm -lX11 -lXext $LIBZ $LIBSOCKET"
+      LDFLAGS="$LDFLAGS $USER_LDFLAGS $X_LDFLAGS $LDFLAGS $XPM_LDFLAGS $all_libraries -lXpm -lX11 -lXext $LIBZ $LIBSOCKET"
     else
-      LDFLAGS="$LDFLAGS $X_LDFLAGS $USER_LDFLAGS $LDFLAGS $XPM_LDFLAGS $all_libraries -lXpm $LIBZ $LIBSOCKET"
+      LDFLAGS="$LDFLAGS $USER_LDFLAGS $X_LDFLAGS $LDFLAGS $XPM_LDFLAGS $all_libraries -lXpm $LIBZ $LIBSOCKET"
     fi
     CFLAGS="$CFLAGS $X_INCLUDES $USER_INCLUDES"
     test -n "$XPM_INCLUDE" && CFLAGS="-I$XPM_INCLUDE $CFLAGS"
@@ -4117,7 +4130,7 @@
 AC_DEFUN(KDE_CREATE_SUBDIRSLIST,
 [
 
-DO_NOT_COMPILE="$DO_NOT_COMPILE CVS debian bsd-port admin"
+DO_NOT_COMPILE="$DO_NOT_COMPILE CVS debian bsd-port admin kdm ksysguard kscd"
 
 if test ! -s $srcdir/subdirs; then
   dnl Note: Makefile.common creates subdirs, so this is just a fallback
@@ -4347,6 +4360,15 @@
     prefix=$ac_default_prefix
     ac_configure_args="$ac_configure_args --prefix=$prefix"
   fi
+  # And delete superflous '/' to make compares easier
+  prefix=`echo "$prefix" | sed 's,//*,/,g' | sed -e 's,/$,,'`
+  exec_prefix=`echo "$exec_prefix" | sed 's,//*,/,g' | sed -e 's,/$,,'`
+  # For unsermake we let top_builddir be absolute, instead of a
+  # ../.. cascade to make dependencies for better from subdirectories
+  # This doesn't change the behaviour of automake, as there the
+  # top_builddir line is hardcoded into the Makefile.in's.
+  top_builddir=`pwd`
+  AC_SUBST(top_builddir)
   KDE_FAST_CONFIGURE
   KDE_CONF_FILES
 ])
@@ -4510,7 +4532,7 @@
 [
 	dir=`dirname "$1"`
 	base=`basename "$1"`
-	list=`ls -1 $dir`
+	list=`ls -1 $dir 2> /dev/null`
 	for entry in $list; do 
 		if test -d $dir/$entry/bin; then
 			case $entry in
@@ -4561,6 +4583,7 @@
       KDE_JAVA_PREFIX(/usr/lib/IBMJava2*)
       KDE_JAVA_PREFIX(/usr/lib/IBMJava*)
       KDE_JAVA_PREFIX(/opt/java*)
+      KDE_JAVA_PREFIX(/Library/Java/Home)
     
       kde_cv_path="NONE"
       kde_save_IFS=$IFS
@@ -4612,10 +4635,20 @@
 dnl At this point kde_java_bindir and kde_java_includedir are either set or "no"
 if test "x$kde_java_bindir" != "xno"; then
 
+  case $host_os in
+    darwin*)
+      dnl Look for libjvm.dylib
+      kde_java_libjvmdir=`find $kde_java_bindir/.. -name libjvm.dylib | sed 's,libjvm.dylib,,'|head -n 1`
+      kd_java_libhpidir=''
+      dnl Darwin doesn't have libhpi
+      ;;
+    *)
   dnl Look for libjvm.so
   kde_java_libjvmdir=`find $kde_java_bindir/.. -name libjvm.so | sed 's,libjvm.so,,'|head -n 1`
   dnl Look for libhpi.so and avoid green threads
   kde_java_libhpidir=`find $kde_java_bindir/.. -name libhpi.so | grep -v green | sed 's,libhpi.so,,' | tail -n 1`
+      ;;
+  esac
 
   dnl Now check everything's fine under there
   dnl the include dir is our flag for having the JDK
@@ -4643,17 +4676,29 @@
     JAVAC=
     jni_includes=
   fi
-
+  case $host_os in
+    darwin*)
+      if test ! -r "$kde_java_libjvmdir/libjvm.dylib"; then
+        AC_MSG_ERROR([libjvm.dylib not found under $kde_java_libjvmdir. Use --without-java.])
+      fi
+      dnl Darwin doesn't have libhpi, but it's not fatal
+      LIB_JVM="-framework JavaVM"
+      LIB_HPI=""
+      ;;
+    *)
   if test ! -r "$kde_java_libjvmdir/libjvm.so"; then
      AC_MSG_ERROR([libjvm.so not found under $kde_java_libjvmdir. Use --without-java.])
   fi 
+      LIB_JVM="-L$kde_java_libjvmdir -ljvm"
+      if test ! -r "$kde_java_libhpidir/libhpi.so"; then
+        AC_MSG_ERROR([libhpi.so not found under $kde_java_libhpidir. Use --without-java.])
+      fi
+      LIB_HPI="-L$kde_java_libhpidir -lhpi"
+      ;;
+  esac
 
   if test ! -x "$kde_java_bindir/java"; then
       AC_MSG_ERROR([java not found under $kde_java_bindir. javac was found though! Use --with-java or --without-java.])
-  fi
-
-  if test ! -r "$kde_java_libhpidir/libhpi.so"; then
-    AC_MSG_ERROR([libhpi.so not found under $kde_java_libhpidir. Use --without-java.])
   fi
 
   if test -n "$jni_includes"; then
diff -uNbr kxine-0.4/admin/am_edit kxine-0.4-new/admin/am_edit
--- kxine-0.4/admin/am_edit	Sun Jul  7 16:33:54 2002
+++ kxine-0.4-new/admin/am_edit	Tue Aug 13 12:35:54 2002
@@ -692,9 +692,9 @@
         {
             my ($dir, $hFile, $cppFile) = split ("\035", $globalmocs{$mocFile}, 3);
             if (defined ($cppFile)) {
-                $mocs .= " $mocFile.moc" if exists $shash{$cppFile};
+                $mocs .= " $mocFile.moc " if exists $shash{$cppFile};
             } else {
-		$sourcelist{$cxxsuffix} .= $mocFile . ".moc." . $cxxsuffix;
+		$sourcelist{$cxxsuffix} .= $mocFile . ".moc." . $cxxsuffix . " ";
 		$moc_cpp_added = 1;
 	    }
         }
@@ -1939,7 +1939,7 @@
     $MakefileData =~ s/\034/\\\n\t/g;    # Restore continuation lines
     # Append our $progId line, _below_ the "generated by automake" line
     # because automake-1.6 relies on the first line to be his own.
-    my $progIdLine = "\# $progId - " . '$Revision$ '."\n";
+    my $progIdLine = "\# $progId - " . '$Revision$ '."\n";
     if ( !( $MakefileData =~ s/^(.*generated .*by automake.*\n)/$1$progIdLine/ ) ) {
         warn "automake line not found in $makefile\n";
 	# Fallback: first line
diff -uNbr kxine-0.4/admin/cvs.sh kxine-0.4-new/admin/cvs.sh
--- kxine-0.4/admin/cvs.sh	Sun Jul  7 15:46:33 2002
+++ kxine-0.4-new/admin/cvs.sh	Tue Aug 13 10:50:44 2002
@@ -250,7 +250,7 @@
    modulename=`pwd`; modulename=`basename $modulename`
 fi
 if test -z "$VERSION" || test "$VERSION" = "@VERSION@"; then
-     VERSION="\"3.0\""
+     VERSION="\"3.0.6\""
 fi
 if test -n "$kde_use_qt_param"; then
       sed -e "s#^dnl KDE_USE_QT#KDE_USE_QT($kde_use_qt_param)#" \
@@ -398,10 +398,10 @@
 	echo -e 'i18n("_: NAME OF TRANSLATORS\\n"\n"Your names")\ni18n("_: EMAIL OF TRANSLATORS\\n"\n"Your emails")' > _translatorinfo.cpp
    else echo " " > _translatorinfo.cpp
    fi
-   perl -e '$mes=0; while (<STDIN>) { if (/^messages:/) { $mes=1; print $_; next; } if ($mes) { if (/$\\(XGETTEXT\)/ && / -o/) { s/ -o \$\(podir\)/ _translatorinfo.cpp -o \$\(podir\)/ } print $_; } else { print $_; } }' < Makefile.am > _transMakefile
+   perl -e '$mes=0; while (<STDIN>) { next if (/^(if|else|endif)\s/); if (/^messages:/) { $mes=1; print $_; next; } if ($mes) { if (/$\\(XGETTEXT\)/ && / -o/) { s/ -o \$\(podir\)/ _translatorinfo.cpp -o \$\(podir\)/ } print $_; } else { print $_; } }' < Makefile.am > _transMakefile
 
    $MAKE -s -f _transMakefile podir=$podir EXTRACTRC="$EXTRACTRC" PREPARETIPS="$PREPARETIPS" \
-	XGETTEXT="${XGETTEXT:-xgettext} -C -ki18n -ktr2i18n -kI18N_NOOP -ktranslate -kaliasLocale -x /opt/kde3/include/kde.pot" \
+	XGETTEXT="${XGETTEXT:-xgettext} -C -ki18n -ktr2i18n -kI18N_NOOP -ktranslate -kaliasLocale -x ${includedir:-$KDEDIR/include}/kde.pot" \
 	messages 
    ) 2>&1 | grep -v '^make\[1\]' > $tmpname
    test -s $tmpname && { echo $subdir ; cat "$tmpname"; }
diff -uNbr kxine-0.4/admin/detect-autoconf.sh kxine-0.4-new/admin/detect-autoconf.sh
--- kxine-0.4/admin/detect-autoconf.sh	Sun Jul  7 15:46:33 2002
+++ kxine-0.4-new/admin/detect-autoconf.sh	Tue Aug 13 10:50:44 2002
@@ -42,12 +42,12 @@
 checkAutomakeAclocal ()
 {
   if test -z "$UNSERMAKE"; then
-    if test -x "`$WHICH automake-1.5`" ; then
-      AUTOMAKE="`$WHICH automake-1.5`"
-      ACLOCAL="`$WHICH aclocal-1.5`"
-    elif test -x "`$WHICH automake-1.6`" ; then
+    if test -x "`$WHICH automake-1.6`" ; then
       AUTOMAKE="`$WHICH automake-1.6`"
       ACLOCAL="`$WHICH aclocal-1.6`"
+    elif test -x "`$WHICH automake-1.5`" ; then
+      AUTOMAKE="`$WHICH automake-1.5`"
+      ACLOCAL="`$WHICH aclocal-1.5`"
     fi
   else
      AUTOMAKE="$UNSERMAKE"
diff -uNbr kxine-0.4/admin/libtool.m4.in kxine-0.4-new/admin/libtool.m4.in
--- kxine-0.4/admin/libtool.m4.in	Sun Jul  7 15:46:33 2002
+++ kxine-0.4-new/admin/libtool.m4.in	Tue Aug 13 10:50:44 2002
@@ -609,6 +609,12 @@
     lt_cv_sys_max_cmd_len=-1;
     ;;
 
+  darwin*)
+    # this should be enough... the length test freaks out some versions
+    # of OSX's `expr`
+    lt_cv_sys_max_cmd_len=16384;
+    ;;
+
   *)
     # If test is not a shell built-in, we'll probably end up computing a
     # maximum length that is only half of the actual maximum length, but
@@ -1901,16 +1907,7 @@
   ;;
 
 darwin* | rhapsody*)
-  lt_cv_deplibs_check_method='file_magic Mach-O dynamically linked shared library'
-  lt_cv_file_magic_cmd='/usr/bin/file -L'
-  case "$host_os" in
-  rhapsody* | darwin1.[[012]])
-    lt_cv_file_magic_test_file=`/System/Library/Frameworks/System.framework/System`
-    ;;
-  *) # Darwin 1.3 on
-    lt_cv_file_magic_test_file='/usr/lib/libSystem.dylib'
-    ;;
-  esac
+  lt_cv_deplibs_check_method='pass_all'
   ;;
 
 freebsd*)
@@ -2560,6 +2557,11 @@
 	;;
     esac
     ;;
+  darwin*)
+    shared_flag='-dynamiclib'
+    _LT_AC_TAGVAR(archive_cmds, $1)='$CC -r -keep_private_externs -nostdlib -o ${lib}-master.o $libobjs && $CC -dynamiclib -install_name $rpath/$soname $predep_objects ${lib}-master.o $deplibs $postdep_objects $compiler_flags -o $lib'
+    output_verbose_link_cmd='$CC -dynamiclib $CFLAGS -v conftest.$objext 2>&1 | egrep "\-L"'
+    ;;
   dgux*)
     case $cc_basename in
       ec++)
@@ -4750,7 +4752,9 @@
 	_LT_AC_TAGVAR(allow_undefined_flag, $1)='-undefined suppress'
 	;;
       *) # Darwin 1.3 on
-	_LT_AC_TAGVAR(allow_undefined_flag, $1)='-flat_namespace -undefined suppress'
+	#_LT_AC_TAGVAR(allow_undefined_flag, $1)='-flat_namespace -undefined suppress'
+        # testing with twolevel namespaces
+	_LT_AC_TAGVAR(allow_undefined_flag, $1)=unsupported
 	;;
       esac
 
@@ -4758,7 +4762,7 @@
       #        cross-compilation, but unfortunately the echo tests do not
       #        yet detect zsh echo's removal of \ escapes.  Also zsh mangles
       #	       `"' quotes if we put them in here... so don't!
-      _LT_AC_TAGVAR(archive_cmds, $1)='$CC $(test .$module = .yes && echo -bundle || echo -dynamiclib) $allow_undefined_flag -o $lib $libobjs $deplibs$linker_flags -install_name $rpath/$soname $verstring'
+      _LT_AC_TAGVAR(archive_cmds, $1)='$CC -r -keep_private_externs -nostdlib -o ${lib}-master.o $libobjs && $CC $(test .$module = .yes && echo -bundle || echo -dynamiclib) $allow_undefined_flag -o $lib ${lib}-master.o $deplibs $linker_flags $(test x$module != xyes && echo -install_name $rpath/$soname $verstring)'
       # We need to add '_' to the symbols in $export_symbols first
       #_LT_AC_TAGVAR(archive_expsym_cmds, $1)="$_LT_AC_TAGVAR(archive_cmds, $1)"' && strip -s $export_symbols'
       _LT_AC_TAGVAR(hardcode_direct, $1)=yes
diff -uNbr kxine-0.4/admin/ltmain.sh kxine-0.4-new/admin/ltmain.sh
--- kxine-0.4/admin/ltmain.sh	Sun Jul  7 15:46:33 2002
+++ kxine-0.4-new/admin/ltmain.sh	Tue Aug 13 10:50:44 2002
@@ -3321,7 +3321,23 @@
 	if test -n "$export_symbols" && test -n "$archive_expsym_cmds"; then
 	  eval cmds=\"$archive_expsym_cmds\"
 	else
+          if test "x$verstring" = "x0.0"; then
+                tmp_verstring=
+          else
+                tmp_verstring="$verstring"
+          fi
+ 	  save_deplibs="$deplibs"
+ 	  for conv in $convenience; do
+            tmp_deplibs=
+            for test_deplib in $deplibs; do
+              if test "$test_deplib" != "$conv"; then
+                tmp_deplibs="$tmp_deplibs $test_deplib"
+              fi
+            done
+            deplibs="$tmp_deplibs"
+ 	  done
 	  eval cmds=\"$archive_cmds\"
+ 	  deplibs="$save_deplibs"
 	fi
 
 	if len=`expr "X$cmds" : ".*"` &&
@@ -3419,7 +3435,12 @@
 	  if test -n "$export_symbols" && test -n "$archive_expsym_cmds"; then
 	    eval cmds=\"$archive_expsym_cmds\"
 	  else
+            save_deplibs="$deplibs"
+            for conv in $convenience; do
+              deplibs="${deplibs%$conv*} ${deplibs#*$conv}"
+            done
 	    eval cmds=\"$archive_cmds\"
+            deplibs="$save_deplibs"
 	  fi
 
 	  # Append the command to remove the reloadable object files
@@ -3632,6 +3653,14 @@
 	compile_deplibs=`$echo "X $compile_deplibs" | $Xsed -e 's/ -lc / -framework System /'`
 	finalize_deplibs=`$echo "X $finalize_deplibs" | $Xsed -e 's/ -lc / -framework System /'`
 	;;
+      esac
+
+      case $host in
+      *darwin*)
+        # Don't allow lazy linking, it breaks C++ global constructors
+        compile_command="$compile_command ${wl}-bind_at_load"
+        finalize_command="$finalize_command ${wl}-bind_at_load"
+        ;;
       esac
 
       compile_command="$compile_command $compile_deplibs"
diff -uNbr kxine-0.4/admin/missing kxine-0.4-new/admin/missing
--- kxine-0.4/admin/missing	Sun Jul  7 15:46:33 2002
+++ kxine-0.4-new/admin/missing	Tue Aug 13 10:50:44 2002
@@ -106,7 +106,7 @@
     exit 1
     ;;
 
-  aclocal)
+  aclocal*)
     echo 1>&2 "\
 WARNING: \`$1' is missing on your system.  You should only need it if
          you modified \`acinclude.m4' or \`${configure_ac}'.  You might want
@@ -143,7 +143,7 @@
     touch $touch_files
     ;;
 
-  automake)
+  automake*)
     echo 1>&2 "\
 WARNING: \`$1' is missing on your system.  You should only need it if
          you modified \`Makefile.am', \`acinclude.m4' or \`${configure_ac}'.
diff -uNbr kxine-0.4/admin/mkinstalldirs kxine-0.4-new/admin/mkinstalldirs
--- kxine-0.4/admin/mkinstalldirs	Sun Jul  7 16:33:54 2002
+++ kxine-0.4-new/admin/mkinstalldirs	Tue Aug 13 10:50:44 2002
@@ -4,7 +4,7 @@
 # Created: 1993-05-16
 # Public domain
 
-# $Id$
+# $Id$
 
 errstatus=0
 
diff -uNbr kxine-0.4/kxine/kxine.cpp kxine-0.4-new/kxine/kxine.cpp
--- kxine-0.4/kxine/kxine.cpp	Sun Jul 21 11:04:21 2002
+++ kxine-0.4-new/kxine/kxine.cpp	Tue Aug 13 11:07:31 2002
@@ -260,7 +260,7 @@
 void
 Kxine::frameAdjustSize ()
 {
-  cerr << "frameAdjustSize received..." << endl;
+  std::cerr << "frameAdjustSize received..." << endl;
   if (!isFullScreen)
     {
 #if 0
@@ -279,7 +279,7 @@
         QSize (videoSizeHint.width() + frameSize.width() - videoSize.width(),
                videoSizeHint.height() + frameSize.height() - videoSize.height());
       resize (newFrameSize);
-      cerr << "... and executed" << endl;
+      std::cerr << "... and executed" << endl;
     }
 
 }
diff -uNbr kxine-0.4/kxine/kxinemenu.cpp kxine-0.4-new/kxine/kxinemenu.cpp
--- kxine-0.4/kxine/kxinemenu.cpp	Sun Jul 14 12:40:20 2002
+++ kxine-0.4-new/kxine/kxinemenu.cpp	Tue Aug 13 11:07:45 2002
@@ -183,7 +183,7 @@
       emit zoomChanged (1.0);
       break;
     }
-  cerr << "emitting adjustSize" << endl;
+  std::cerr << "emitting adjustSize" << endl;
   emit frameAdjustSize ();
 }
 
diff -uNbr kxine-0.4/kxine/kxinevideo.cpp kxine-0.4-new/kxine/kxinevideo.cpp
--- kxine-0.4/kxine/kxinevideo.cpp	Sun Jul 21 10:39:34 2002
+++ kxine-0.4-new/kxine/kxinevideo.cpp	Tue Aug 13 11:07:21 2002
@@ -79,7 +79,7 @@
     homedir = "/tmp";
 
   QString xinerc = QString("%1/%2").arg(homedir).arg(".xine/config");
-  cerr << static_cast <const char*> (xinerc.local8Bit()) << endl;
+  std::cerr << static_cast <const char*> (xinerc.local8Bit()) << endl;
 
   xineconf = xine_config_file_init (const_cast <char *> (static_cast <const char*> (
                                      xinerc.local8Bit())));
@@ -266,16 +266,16 @@
     case XINE_EVENT_MOUSE_BUTTON:
       break;
     default:
-      cerr << "received xine event with type " << ev->type << endl;
+      std::cerr << "received xine event with type " << ev->type << endl;
     }
 }
 
 void
 KxineVideo::setVideoSize (const QSize & newSize)
 {
-  cerr << "videoSize: " << videoSize.width() << " x " << videoSize.height ();
+  std::cerr << "videoSize: " << videoSize.width() << " x " << videoSize.height ();
   videoSize = newSize;
-  cerr << " -> " << videoSize.width() << " x " << videoSize.height () << endl;
+  std::cerr << " -> " << videoSize.width() << " x " << videoSize.height () << endl;
   
 }
 
@@ -327,9 +327,9 @@
 void
 KxineVideo::zoomChanged (double newzoom)
 {
-  cerr << "zoomChanged " << sizeMultiple;
+  std::cerr << "zoomChanged " << sizeMultiple;
   sizeMultiple = newzoom;
-  cerr << " -> " << sizeMultiple << endl; 
+  std::cerr << " -> " << sizeMultiple << endl; 
 }
 
 bool
@@ -402,7 +402,7 @@
     retval = QSize (videoSize.width () * sizeMultiple,
                     videoSize.height () * sizeMultiple);
   
-  cerr << "hinting size " << retval.width () << " x " << retval.height () << endl;
+  std::cerr << "hinting size " << retval.width () << " x " << retval.height () << endl;
   return retval;
 }
 
@@ -541,11 +541,11 @@
 {
   if (xine_is_stream_seekable (xine))
     {
-      cerr << "Seeking to " << pos << "..." << flush;
+      std::cerr << "Seeking to " << pos << "..." << flush;
       // seeking is done by ``playing'' at the new location -_-
       xine_play (xine, const_cast<char *> (static_cast<const char *> (
                  mrl.local8Bit ())), 0, pos);
-      cerr << "...done" << endl;
+      std::cerr << "...done" << endl;
  
     }
 }
@@ -579,7 +579,7 @@
   if (vol > 100)
     vol = 100;
 
-  cerr << "setting volume to " << vol << endl;
+  std::cerr << "setting volume to " << vol << endl;
 
   if(mixer != -1)
     {
@@ -598,7 +598,7 @@
 void
 KxineVideo::xineOpenAutoplayInputPlugin(const QString& aip)
 {
-  cerr << "open aip " << aip << endl;
+  std::cerr << "open aip " << aip << endl;
   int num;
   char **mrls =xine_get_autoplay_mrls (xine, const_cast<char *> (
                                       static_cast<const char*> (aip.local8Bit())), &num);
@@ -607,7 +607,7 @@
       xineLoadMRL (*mrls);
     }
   if (num > 1)
-    cerr << "Kxine: warning: got more than one MRL for plugin " << aip << endl;
+    std::cerr << "Kxine: warning: got more than one MRL for plugin " << aip << endl;
 }
 
 void
@@ -625,7 +625,7 @@
   qt_remove_postselect_handler (&KxineVideo::x11Lock);
   x11Unlock ();
   xine_exit (xine);
-  cerr << "xine_exit returned" << endl;
+  std::cerr << "xine_exit returned" << endl;
 }
 
 void
