diff -uNr libapt-pkg-perl-0.1.17/Makefile.PL libapt-pkg-perl-0.1.17-patched/Makefile.PL
--- libapt-pkg-perl-0.1.17/Makefile.PL	2004-10-13 20:50:02.000000000 -0400
+++ libapt-pkg-perl-0.1.17-patched/Makefile.PL	2005-08-31 15:57:45.000000000 -0400
@@ -7,8 +7,10 @@
 WriteMakefile(
     NAME	 => 'AptPkg',
     VERSION_FROM => 'AptPkg.pm',
-    CC		 => 'c++',
-    LD		 => 'c++',
+    CC		 => 'g++-3.3',
+    LD		 => 'g++-3.3',
+    LDDLFLAGS	 => '-bundle -undefined dynamic_lookup -L@PREFIX@/lib -lapt-pkg',
+    CCFLAGS	 => '-I@PREFIX@/include',
     OBJECT	 => '$(O_FILES)',
     LIBS	 => ['-lapt-pkg'],
     XSOPT	 => '-C++',
diff -uNr libapt-pkg-perl-0.1.17/t/00_config.t libapt-pkg-perl-0.1.17-patched/t/00_config.t
--- libapt-pkg-perl-0.1.17/t/00_config.t	2003-06-09 08:26:56.000000000 -0400
+++ libapt-pkg-perl-0.1.17-patched/t/00_config.t	2005-08-31 15:53:37.000000000 -0400
@@ -93,7 +93,7 @@
 print 'not ' unless $c->init;
 print "ok 17\n";
 
-print 'not ' unless $c->get_dir('Dir::Etc') eq '/etc/apt/';
+print 'not ' unless $c->get_dir('Dir::Etc') eq '@PREFIX@/etc/apt/';
 print "ok 18\n";
 
 # hash access
