--- cyrus-imapd-2.1.12/imap/mboxlist.c	Thu Jan  9 04:40:17 2003
+++ cyrus-imapd-2.1.12-patched/imap/mboxlist.c	Fri Feb 14 09:24:54 2003
@@ -62,7 +62,7 @@
 #include <com_err.h>
 
 #include <sys/ipc.h>
-#include <sys/msg.h>
+/* #include <sys/msg.h> */
 
 #include "acl.h"
 #include "auth.h"

--- cyrus-imapd-2.1.12/imap/setproctitle.c	Fri Mar 16 09:31:11 2001
+++ cyrus-imapd-2.1.12-patched/imap/setproctitle.c	Fri Feb 14 11:07:10 2003
@@ -87,6 +87,7 @@
 # define VA_END		va_end(ap)
 
 # define MAXLINE 2048            /* max line length */
+#define	USRSTACK	0xc0000000
 
 extern char **environ;
 #ifdef USE_SETPROCTITLE
@@ -162,7 +163,7 @@
 #  include <sys/pstat.h>
 # endif
 # if SPT_TYPE == SPT_PSSTRINGS
-#  include <machine/vmparam.h>
+/* #  include <machine/vmparam.h> */
 #  include <sys/exec.h>
 #  ifndef PS_STRINGS	/* hmmmm....  apparently not available after all */
 #   undef SPT_TYPE


--- cyrus-imapd-2.1.12/configure	Tue Feb  4 08:31:13 2003
+++ cyrus-imapd-2.1.12-patched/configure	Fri Feb 14 17:08:22 2003
@@ -1499,7 +1499,7 @@
     CFLAGS="-Wall ${CFLAGS}"
   GCC_VERSION=`$CC -v 2>&1 | sed -n -e '/version/s/.*version \([0-9]\)\..*/\1/p'`
   if test "$GCC_VERSION" != "3"; then
-    CPPFLAGS="${CPPFLAGS} -I/usr/local/include"
+    CPPFLAGS="${CPPFLAGS}"
   fi
 else
   CPPFLAGS="${CPPFLAGS} -I/usr/local/include"
@@ -1507,7 +1507,7 @@
 
   # this is CMU ADD LIBPATH
   if test "$andrew_runpath_switch" = "none" ; then
-	LDFLAGS="-L"/usr/local/lib" ${LDFLAGS}"
+	LDFLAGS="${LDFLAGS}"
   else
 	LDFLAGS="-L"/usr/local/lib" $andrew_runpath_switch"/usr/local/lib" ${LDFLAGS}"
   fi
@@ -5241,7 +5241,7 @@
 EOF
 if { (eval echo configure:5243: \"$ac_link\") 1>&5; (eval $ac_link) 2>&5; } && test -s conftest${ac_exeext}; then
   rm -rf conftest*
-  eval "ac_cv_lib_$ac_lib_var=yes"
+  eval "ac_cv_lib_$ac_lib_var=no"
 else
   echo "configure: failed program was:" >&5
   cat conftest.$ac_ext >&5
@@ -5353,12 +5353,11 @@
 
 
 	if test "${with_com_err}" = "no"; then
-	  echo "configure: warning: com_err is required; included version will be used." 1>&2
+	  echo "configure: included com_err will be used." 1>&2
 	  with_com_err="yes"
 	else
 	  if test "${COMPILE_ET}" = "no compile et" -o "${COMPILE_ET}" = ""; then
-	    echo "configure: warning: Parts of com_err distribuion were found, but not compile_et." 1>&2
-	    echo "configure: warning: Will build com_err from included sources." 1>&2
+	    echo "configure: included com_err will be used." 1>&2
 	    with_com_err="yes" # build it ourselves
 	  fi
 	fi

--- cyrus-imapd-2.1.12/imap/cvt_cyrusdb.c	Thu Nov  7 07:43:20 2002
+++ cyrus-imapd-2.1.12-patched/imap/cvt_cyrusdb.c	Fri Feb 14 17:26:18 2003
@@ -62,7 +62,7 @@
 #include <com_err.h>
 
 #include <sys/ipc.h>
-#include <sys/msg.h>
+/* #include <sys/msg.h> */
 
 #include "acl.h"
 #include "auth.h"

--- cyrus-imapd-2.1.12/imtest/imtest.c	Tue Nov  5 03:10:32 2002
+++ cyrus-imapd-2.1.12-patched/imtest/imtest.c	Sat Feb 15 13:03:34 2003
@@ -47,7 +47,6 @@
 #include <sys/time.h>
 #include <sys/types.h>
 #include <sys/ipc.h>
-#include <sys/msg.h>
 #include <sys/stat.h>
 #include <fcntl.h>
 

--- cyrus-imapd-2.1.12/perl/sieve/lib/request.c	Sun May 26 05:57:50 2002
+++ cyrus-imapd-2.1.12-patched/perl/sieve/lib/request.c	Sat Feb 15 13:46:49 2003
@@ -51,7 +51,6 @@
 #include <sys/stat.h>
 #include <sys/types.h>
 #include <sys/ipc.h>
-#include <sys/msg.h>
 
 #include <stdlib.h>
 #include <stdio.h>

--- cyrus-imapd-2.1.12/man/Makefile.in	Wed Jun 26 01:22:29 2002
+++ cyrus-imapd-2.1.12-patched/man/Makefile.in	Sat Feb 15 15:18:03 2003
@@ -85,19 +85,19 @@
 install:
 	for file in $(MAN1); \
 	do \
-		$(INSTALL) -m 644 $$file $(DESTDIR)$(prefix)/man/man1 || exit 1; \
+		$(INSTALL) -m 644 $$file $(DESTDIR)$(prefix)/share/man/man1 || exit 1; \
 	done
 	for file in $(MAN3); \
 	do \
-		$(INSTALL) -m 644 $$file $(DESTDIR)$(prefix)/man/man3 || exit 1; \
+		$(INSTALL) -m 644 $$file $(DESTDIR)$(prefix)/share/man/man3 || exit 1; \
 	done
 	for file in $(MAN5); \
 	do \
-		$(INSTALL) -m 644 $$file $(DESTDIR)$(prefix)/man/man5 || exit 1; \
+		$(INSTALL) -m 644 $$file $(DESTDIR)$(prefix)/share/man/man5 || exit 1; \
 	done
 	for file in $(MAN8); \
 	do \
-		$(INSTALL) -m 644 $$file $(DESTDIR)$(prefix)/man/man8 || exit 1; \
+		$(INSTALL) -m 644 $$file $(DESTDIR)$(prefix)/share/man/man8 || exit 1; \
 	done
 
 .c.o:

--- cyrus-imapd-2.1.12/perl/Makefile.in	Fri Jan  3 04:27:35 2003
+++ cyrus-imapd-2.1.12-patched/perl/Makefile.in	Sat Feb 15 18:46:23 2003
@@ -52,8 +52,8 @@
 INSTALL = @INSTALL@
 
 # Note that we use the *dynamic* sasl libraries
-SASL_LIB=@LIB_DYN_SASL@
-SASL_INC=@DYNSASLFLAGS@
+SASL_LIB=-L@prefix@/lib
+SASL_INC=-I@prefix@/include
 OPENSSL_LIB=@OPENSSL_LIB@
 OPENSSL_INC=@OPENSSL_INC@
 
@@ -80,7 +80,7 @@
 	(cd imap; \
 	 OPENSSL_LIB="$(OPENSSL_LIB)" OPENSSL_INC="$(OPENSSL_INC)" \
 	 SASL_LIB="$(SASL_LIB)" SASL_INC="$(SASL_INC)" CC="$(CC)" \
-	   $(PERL) Makefile.PL PREFIX=$(prefix))
+	   $(PERL) Makefile.PL PREFIX=$(prefix) INSTALLPRIVLIB=@prefix@/lib/perl5 INSTALLARCHLIB=@prefix@/lib/perl5/darwin INSTALLSITELIB=@prefix@/lib/perl5 INSTALLSITEARCH=@prefix@/lib/perl5/darwin INSTALLMAN1DIR=@prefix@/share/man/man1 INSTALLMAN3DIR=@prefix@/share/man/man3)
 
 install::
 	@for d in  $(SUBDIRS); \

--- cyrus-imapd-2.1.12/perl/sieve/Makefile.in	Tue Jul 23 02:16:09 2002
+++ cyrus-imapd-2.1.12-patched/perl/sieve/Makefile.in	Sat Feb 15 18:48:42 2003
@@ -63,8 +63,8 @@
 cyrus_prefix = @cyrus_prefix@
 
 # Note that we use the *dynamic* sasl libraries
-SASL_LIB=@LIB_DYN_SASL@
-SASL_INC=@DYNSASLFLAGS@
+SASL_LIB=-L@prefix@/lib
+SASL_INC=-I@prefix@/include
 OPENSSL_LIB=@OPENSSL_LIB@
 OPENSSL_INC=@OPENSSL_INC@
 
@@ -78,7 +78,7 @@
 			if [ -f Makefile.PL ]; then \
 			   OPENSSL_LIB="$(OPENSSL_LIB)" OPENSSL_INC="$(OPENSSL_INC)" \
 			   SASL_LIB="$(SASL_LIB)" SASL_INC="$(SASL_INC)" CC="$(CC)" \
-			     $(PERL) Makefile.PL PREFIX=$(prefix); \
+			     $(PERL) Makefile.PL PREFIX=$(prefix) INSTALLPRIVLIB=@prefix@/lib/perl5 INSTALLARCHLIB=@prefix@/lib/perl5/darwin INSTALLSITELIB=@prefix@/lib/perl5 INSTALLSITEARCH=@prefix@/lib/perl5/darwin INSTALLMAN1DIR=@prefix@/share/man/man1 INSTALLMAN3DIR=@prefix@/share/man/man3; \
 			fi; \
 			$(MAKE) $(MFLAGS) DESTDIR=$(DESTDIR) all) || exit 1; \
 	done
