--- cyrus-imapd-2.1.13/configure	Fri May  2 05:08:14 2003
+++ cyrus-imapd-2.1.13-patched/configure	Mon May  5 22:47:02 2003
@@ -1499,7 +1499,7 @@
     CFLAGS="-Wall ${CFLAGS}"
   GCC_VERSION=`$CC -v 2>&1 | sed -n -e '/version/s/.*version \([0-9]\)\..*/\1/p'`
   if test "$GCC_VERSION" != "3"; then
-    CPPFLAGS="${CPPFLAGS} -I/usr/local/include"
+    CPPFLAGS="${CPPFLAGS}"
   fi
 else
   CPPFLAGS="${CPPFLAGS} -I/usr/local/include"
@@ -1507,7 +1507,7 @@
 
   # this is CMU ADD LIBPATH
   if test "$andrew_runpath_switch" = "none" ; then
-	LDFLAGS="-L"/usr/local/lib" ${LDFLAGS}"
+	LDFLAGS="${LDFLAGS}"
   else
 	LDFLAGS="-L"/usr/local/lib" $andrew_runpath_switch"/usr/local/lib" ${LDFLAGS}"
   fi
@@ -2841,7 +2841,7 @@
 	    BDB_LIBADD=""
 	fi
 
-        for dbname in db-4.1 db4.1 db41 db-4.0 db4.0 db-4 db40 db4 db-3.3 db3.3 db33 db-3.2 db3.2 db32 db-3.1 db3.1 db31 db-3 db30 db3 db
+        for dbname in db-4.1
           do
             echo $ac_n "checking for db_create in -l$dbname""... $ac_c" 1>&6
 echo "configure:2848: checking for db_create in -l$dbname" >&5
@@ -5306,7 +5306,7 @@
 EOF
 if { (eval echo configure:5308: \"$ac_link\") 1>&5; (eval $ac_link) 2>&5; } && test -s conftest${ac_exeext}; then
   rm -rf conftest*
-  eval "ac_cv_lib_$ac_lib_var=yes"
+  eval "ac_cv_lib_$ac_lib_var=no"
 else
   echo "configure: failed program was:" >&5
   cat conftest.$ac_ext >&5
@@ -5422,8 +5422,7 @@
 	  with_com_err="yes"
 	else
 	  if test "${COMPILE_ET}" = "no compile et" -o "${COMPILE_ET}" = ""; then
-	    echo "configure: warning: Parts of com_err distribuion were found, but not compile_et." 1>&2
-	    echo "configure: warning: Will build com_err from included sources." 1>&2
+            echo "configure: included com_err will be used." 1>&2
 	    with_com_err="yes" # build it ourselves
 	  fi
 	fi

--- cyrus-imapd-2.1.13/imap/config.c	Tue Apr 15 06:31:36 2003
+++ cyrus-imapd-2.1.13-patched/imap/config.c	Tue May  6 00:30:02 2003
@@ -68,7 +68,7 @@
 #include "hash.h"
 #include "prot.h" /* for PROT_BUFSIZE */
 
-#define CONFIG_FILENAME "/etc/imapd.conf"
+#define CONFIG_FILENAME "@prefix@/etc/imapd.conf"
 
 #define CONFIGHASHSIZE 200 /* > 2x # of options */
 static struct hash_table confighash;

--- cyrus-imapd-2.1.13/imap/cvt_cyrusdb.c	Tue Apr 15 06:15:12 2003
+++ cyrus-imapd-2.1.13-patched/imap/cvt_cyrusdb.c	Tue May  6 00:46:22 2003
@@ -62,7 +62,6 @@
 #include <com_err.h>
 
 #include <sys/ipc.h>
-#include <sys/msg.h>
 
 #include "acl.h"
 #include "auth.h"

--- cyrus-imapd-2.1.13/imap/krbck.c	Fri Feb 14 07:15:26 2003
+++ cyrus-imapd-2.1.13-patched/imap/krbck.c	Tue May  6 00:28:30 2003
@@ -59,7 +59,7 @@
 
 #define EXIT_CHOKE (int) 10
 
-#define CONFIG_FILENAME "/etc/imapd.conf"
+#define CONFIG_FILENAME "@prefix@/etc/imapd.conf"
 #define MAX_FILENAME_LEN 255
 
 static int  evalcheck(char*, int, char*);

--- cyrus-imapd-2.1.13/imap/mboxlist.c	Thu May  1 06:33:20 2003
+++ cyrus-imapd-2.1.13-patched/imap/mboxlist.c	Tue May  6 00:51:14 2003
@@ -62,7 +62,6 @@
 #include <com_err.h>
 
 #include <sys/ipc.h>
-#include <sys/msg.h>
 
 #include "acl.h"
 #include "auth.h"

--- cyrus-imapd-2.1.13/imap/setproctitle.c	Fri Feb 14 07:15:30 2003
+++ cyrus-imapd-2.1.13-patched/imap/setproctitle.c	Tue May  6 00:49:52 2003
@@ -87,6 +87,7 @@
 # define VA_END		va_end(ap)
 
 # define MAXLINE 2048            /* max line length */
+#define	 USRSTACK  	0xc0000000
 
 extern char **environ;
 #ifdef USE_SETPROCTITLE
@@ -162,7 +163,7 @@
 #  include <sys/pstat.h>
 # endif
 # if SPT_TYPE == SPT_PSSTRINGS
-#  include <machine/vmparam.h>
+/*#  include <machine/vmparam.h> */
 #  include <sys/exec.h>
 #  ifndef PS_STRINGS	/* hmmmm....  apparently not available after all */
 #   undef SPT_TYPE

--- cyrus-imapd-2.1.13/imtest/imtest.c	Fri Feb 14 07:15:33 2003
+++ cyrus-imapd-2.1.13-patched/imtest/imtest.c	Tue May  6 00:44:32 2003
@@ -47,7 +47,6 @@
 #include <sys/time.h>
 #include <sys/types.h>
 #include <sys/ipc.h>
-#include <sys/msg.h>
 #include <sys/stat.h>
 #include <fcntl.h>
 

--- cyrus-imapd-2.1.13/master/conf/normal.conf	Wed Oct  2 06:46:01 2002
+++ cyrus-imapd-2.1.13-patched/master/conf/normal.conf	Tue May  6 00:24:55 2003
@@ -19,10 +19,10 @@
 
   # at least one LMTP is required for delivery
 #  lmtp		cmd="lmtpd" listen="lmtp" prefork=0
-  lmtpunix	cmd="lmtpd" listen="/var/imap/socket/lmtp" prefork=0
+  lmtpunix	cmd="lmtpd" listen="@prefix@/var/imap/socket/lmtp" prefork=0
 
   # this is only necessary if using notifications
-#  notify	cmd="notifyd" listen="/var/imap/socket/notify" proto="udp" prefork=1
+#  notify	cmd="notifyd" listen="@prefix@/var/imap/socket/notify" proto="udp" prefork=1
 }
 
 EVENTS {

--- cyrus-imapd-2.1.13/master/masterconf.c	Fri Feb 14 07:15:45 2003
+++ cyrus-imapd-2.1.13-patched/master/masterconf.c	Tue May  6 00:26:44 2003
@@ -58,7 +58,7 @@
 
 #include "masterconf.h"
 
-#define CONFIG_FILENAME "/etc/cyrus.conf"
+#define CONFIG_FILENAME "@prefix@/etc/cyrus.conf"
 
 struct configlist {
     char *key;

--- cyrus-imapd-2.1.13/tools/mkimap	Fri Sep 28 02:02:34 2001
+++ cyrus-imapd-2.1.13-patched/tools/mkimap	Tue May  6 00:22:13 2003
@@ -59,8 +59,8 @@
 __END__
 require 5;
 
-$imapdconf = shift || "/etc/imapd.conf";
-$sievedir = "/usr/sieve";
+$imapdconf = shift || "@prefix@/etc/imapd.conf";
+$sievedir = "@prefix@/var/imap/sieve";
 $nosievedir = 0;
 $hashispool = 0;
 
--- cyrus-imapd-2.1.13/imapd.conf	Tue May  6 00:19:19 2003
+++ cyrus-imapd-2.1.13-patched/imapd.conf	Fri Mar  7 13:44:25 2003
@@ -0,0 +1,371 @@
+#
+# Sample configurations file for Cyrus IMAPd
+# Most lines in this file are commented; in this case the default is used. 
+# The commented lines (usually) contain the default value
+
+# The pathname of the IMAP configuration directory
+#
+configdirectory: @prefix@/var/imap
+
+# The partition name used by default for new mailboxes
+#
+#defaultpartition: default
+
+# The directory for the different partitions
+#
+partition-default: @prefix@/var/spool/imap
+
+# Use the UNIX separator character '/' for delimiting levels of
+# mailbox hierarchy.  The default is to use the netnews separator
+# character '.'.
+#unixhierarchysep: no
+
+# Use the alternate IMAP namespace, where personal folders reside at
+# the same level in the hierarchy as INBOX.
+#
+#altnamespace: no
+
+# If using the alternate IMAP namespace, the prefix for the other
+# users namespace.  The hierarchy delimiter will be automatically
+# appended.
+#
+#userprefix: Other Users
+
+# If using the alternate IMAP namespace, the prefix for the shared
+# namespace.  The hierarchy delimiter will be automatically appended.
+#
+#sharedprefix: Shared Folders
+
+# The umask value used by various Cyrus IMAP programs
+#
+#umask: 077
+
+# This is the hostname visible in the greeting messages of the POP,
+# IMAP and LMTP daemons. If it is unset, then the result returned from
+# gethostname(2) is used.
+#
+#servername: <result returned by gethostname(2)>
+
+# Whether to allow anonymous logins
+#
+#allowanonymouslogin: no
+
+# Allow the use of the SASL PLAIN mechanism.
+#
+#allowplaintext: yes
+
+# The percent  of  quota  utilization  over  which  the server generates
+# warnings.
+#
+#quotawarn: 90
+
+# The length of the IMAP server's inactivity autologout timer, in minutes.
+# The  minimum  value  is  30,  the default.
+#
+#timeout: 30
+
+# The interval (in seconds) for polling the mailbox for changes while
+# running the IDLE command.  This option is used when idled can not be
+# contacted or when polling is used exclusively.  The minimum value is
+# 1.  A value of 0 will disable polling (and disable IDLE if polling
+# is the only method available).
+#
+#imapidlepoll: 60
+
+# If enabled, the server responds to an ID command with a parameter
+# list containing: version, vendor, support-url, os, os-version,
+# command, arguments, environment.  Otherwise the server returns NIL.
+#
+#imapidresponse: yes
+
+# Set the length of the POP server's inactivity autologout timer, in
+# minutes.  The minimum value is 10, the default.
+#
+#poptimeout: 10
+
+# Set the minimum amount of time the server forces users to wait between 
+# successive POP logins, in  minutes.  The default is 0.
+#
+#popminpoll: 0
+
+# The number of days advertised as being the minimum a message may be
+# left on the POP server before it is deleted (via the CAPA command,
+# defined in the POP3 Extension Mechanism, which some clients may
+# support).  "NEVER", the default, may be specified with a negative
+# number.  The Cyrus POP3 server never deletes mail, no matter what
+# the value of this parameter is.  However, if a site implements a
+# less liberal policy, it needs to change this parameter accordingly.
+#
+#popexpiretime: 0
+
+# The list of userids with administrative rights.  Separate each userid
+# with a space.  We recommend that administrator userids be separate from 
+# standard userids.  Sites using Kerberos authentication may use separate
+# "admin" instances.
+#
+#admins: <none>
+
+# A list of users and groups that are allowed to proxy for other
+# users, seperated by spaces.  Any user listed in this will be allowed
+# to login for any other user: use with caution.
+#
+#proxyservers: <none>
+
+# The Access Control List (ACL) placed on a newly-created (non-user)
+# mailbox that does not have a parent mailbox.
+#
+#defaultacl: anyone lrs
+
+# The pathname of the news spool directory.  Only used if the partition-news
+# configuration option is set.
+#
+#newsspool: <no default>
+
+# Prefix to be prepended to newsgroup names to make the corresponding IMAP
+# mailbox names.
+#
+#newsprefix: <none>
+
+# If nonzero, normal users may create their own IMAP accounts by creating
+# the mailbox INBOX.  The user's quota is set to the value if it is positive,
+# otherwise the user has unlimited quota.
+#
+#autocreatequota: 0
+
+# Include notations in the protocol telemetry logs indicating the number
+# of seconds since the last command or response.
+#
+#logtimestamps: no
+
+# Number of seconds to pause after a successful plaintext login.  For systems
+# that support strong authentication, this permits users to perceive a cost
+# of using plaintext passwords.
+#
+#plaintextloginpause: 0
+
+# The pathname of srvtab file containing the server's private key.
+# This option is passed to the SASL library and overrides its default
+# setting.
+#
+#srvtab: /etc/srvtab
+
+# The list of remote realms whose users may log in using  cross-realm
+# authentications.   Seperate each realm name by a space.  This option is
+# only used when the server is compiled with Kerberos authentication.
+#
+#loginrealms: <none>
+
+# If enabled, any authentication identity which has a rights on a user's
+# INBOX may log in as that user.  This option is only used when the server
+# is compiled with Kerberos authentication.
+#
+#loginuseacl: no
+
+# If enabled, lmtpd attempts to only write one copy of a message per
+# partition and create hard links, resulting in a potentially large
+# disk savings.
+#
+#singleinstancestore: yes
+
+# If enabled, lmtpd will suppress delivery of a message to a mailbox
+# if a message with the same message-id (or resent-message-id) is
+# recorded as having already been delivered to the mailbox.  Records
+# the mailbox and message-id/resent-message-id of all successful
+# deliveries.
+# Note that duplicate delivery suppression is required for Sieve.
+# Disabling duplicate delivery suppression will also disable Sieve,
+# and as such should only be disabled for performance reasons.
+#
+#duplicatesuppression: yes
+
+# If enabled, lmtpd rejects messages with 8-bit characters in the
+# headers.  Otherwise, 8-bit characters are changed to `X'.  (A proper
+# soultion to non-ASCII characters in headers is offered by RFC 2047
+# and its predecessors.)
+#
+#reject8bit: no
+
+# Maximum incoming LMTP message size.  If set, lmtpd will reject
+# messages larger than maxmessagesize bytes.  The default is to allow
+# messages of any size.
+#
+#maxmessagesize: <unlimited>
+
+# Userid used to deliver messages to shared folders.  For example, if
+# set to "bb", email sent to "bb+shared.blah" would be delivered to
+# the "shared.blah" folder.  By default, an email address of
+# "+shared.blah" would be used.
+#postuser: <none>
+
+# If enabled at compile time, this specifies a URL to reply when
+# Netscape asks the server where the mail administration HTTP server
+# is.  The default is a site at CMU with a hopefully informative
+# message; administrators should set this to a local resource with
+# some information of greater use.
+#
+#netscapeurl: http://andrew2.andrew.cmu.edu/cyrus/imapd/netscapeadmin.html
+
+# Notifyd(8) method to use for "MAIL" notifications.  If not set,
+# "MAIL" notifications are disabled.
+#
+#mailnotifier: <no default>
+
+# Notifyd(8) method to use for "SIEVE" notifications.  If not set,
+# "SIEVE" notifications are disabled.
+#
+# This method is only used when no method is specified in the script.
+#
+#sievenotifier: <no default>
+
+# If enabled, lmtpd will look for Sieve scripts in user's home
+# directories: ~user/.sieve.
+#
+sieveusehomedir: false
+
+# If sieveusehomedir is false, this directory is searched for Sieve scripts.
+# The active Sieve script is s called "default", placed in the users sieve
+# sieve directory (ie. /var/imap/sieve/u/user).
+#
+sievedir: @prefix@/var/imap/sieve
+
+# The pathname of the sendmail executable.  Sieve uses sendmail for
+# sending rejections, redirects and vaca- tion responses.
+#
+#sendmail: /usr/sbin/sendmail
+
+# Username that is used as the 'From' address in rejection MDNs
+# produced by sieve.
+#
+#postmaster: postmaster
+
+# If enabled, the partitions will also be hashed, in addition to the hashing
+# done on configuration directories.  This is recommended if one partition has
+# a very bushy mailbox tree.
+#
+#hashimapspool: false
+
+# Maximum size (in kilobytes) any sieve script can be, enforced at
+# submission by timsieved(8).
+#
+#sieve_maxscriptsize: 32
+
+# Maximum number of sieve scripts any user may have, enforced at
+# submission by timsieved(8).
+#
+#sieve_maxscripts: 5
+
+# Maximum SSF (security strength factor) that the server will allow a
+# client to negotiate.
+#
+#sasl_maximum_layer: 256
+
+# The minimum SSF that the server will allow a client to negotiate.  A
+# value of 1 requires integrity protection; any higher value requires
+# some amount of encryption.
+#
+#sasl_minimum_layer: 0
+
+# The mechanism used by the server to verify plaintext passwords.  Possible
+# values include "auxprop", "saslauthd", and "pwcheck"
+#
+sasl_pwcheck_method: saslauthd
+
+# If enabled,  the SASL library will automatically create authentication
+# secrets when given a plaintext password.  See the SASL documentation.
+#
+#sasl_auto_transition: no
+
+# Location of the opiekeys file
+#
+#sasl_opiekeys: /etc/opiekeys
+
+# File containing the global certificate used for ALL services (imap,
+# pop3, lmtp).
+#
+#tls_cert_file: @prefix@/etc/server.crt
+
+# File containing the private key belonging to the global server
+# certificate.
+#
+#tls_key_file: @prefix@/etc/server.key
+
+# File containing the certificate used for imap ONLY.  If not
+# specified, the global certificate is used.  A value of "disabled"
+# will disable SSL/TLS for imap.
+#
+#tls_imap_cert_file: <none>
+
+# File containing the private key belonging to the imap-specific
+# server certificate.  If not specified, the global private key is
+# used.  A value of "disabled" will disable SSL/TLS for imap.
+#
+#tls_imap_key_file: <none>
+
+# File containing the certificate used for pop3 ONLY.  If not
+# specified, the global certificate is used.  A value of "disabled"
+# will disable SSL/TLS for pop3.
+#
+#tls_pop3_cert_file: <none>
+
+# File containing the private key belonging to the pop3-specific
+# server certificate.  If not specified, the global private key is
+# used.  A value of "disabled" will disable SSL/TLS for pop3.
+#
+#tls_pop3_key_file: <none>
+
+# File containing the certificate used for lmtp ONLY.  If not
+# specified, the global certificate is used.  A value of "disabled"
+# will disable TLS for lmtp.
+#
+#tls_lmtp_cert_file: <none>
+
+# File containing the private key belonging to the lmtp-specific
+# server certificate.  If not specified, the global private key is
+# used.  A value of "disabled" will disable TLS for lmtp.
+#
+#tls_lmtp_key_file: <none>
+
+# File containing the certificate used for sieve ONLY.  If not
+# specified, the global certificate is used.  A value of "disabled"
+# will disable TLS for sieve.
+#
+#tls_sieve_cert_file: <none>
+
+# File containing the private key belonging to the sieve-specific
+# server certificate.  If not specified, the global private key is
+# used.  A value of "disabled" will disable TLS for sieve.
+#
+#tls_sieve_key_file: <none>
+
+# File containing one or more Certificate Authority (CA) certificates.
+#
+#tls_ca_file: <none>
+
+# Path to directory with certificates of CAs.
+#
+#tls_ca_path: <none>
+
+# The length of time (in minutes) that a TLS session will be cached
+# for later reuse.  The maximum value is 1440 (24 hours), the default.
+# A value of 0 will disable session caching.
+#
+#tls_session_timeout: 1440
+
+# The right that a user needs to delete a mailbox.
+#
+#deleteright: c
+
+# Unix domain socket that lmtpd listens on, used by deliver(8).
+#
+#lmtpsocket: /var/imap/socket/lmtp
+
+# Unix domain socket that idled listens on.
+#
+#idlesocket: /var/imap/socket/idle
+
+# Unix domain socket that the new mail notification daemon listens on.
+#
+#notifysocket: /var/imap/socket/notify
+
+#
+# EOF

--- cyrus-imapd-2.1.13/Makefile.in	Fri May  2 05:05:22 2003
+++ cyrus-imapd-2.1.13-patched/Makefile.in	Sun May 11 00:14:28 2003
@@ -42,8 +42,8 @@
 # OUT OF OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
 # 
 
-SUBDIRS = man @PRE_SUBDIRS@ acap lib @SERVER_SUBDIRS@ \
-	imtest @SNMP_SUBDIRS@ @EXTRA_SUBDIRS@
+SUBDIRS = man et  sieve acap lib master imap \
+	imtest   timsieved notifyd
 DISTSUBDIRS = doc
 
 srcdir = @srcdir@
@@ -95,10 +95,10 @@
 	$(srcdir)/install-sh -d ${DESTDIR}$(exec_prefix)/bin
 	$(srcdir)/install-sh -d ${DESTDIR}$(exec_prefix)/lib
 	$(srcdir)/install-sh -d ${DESTDIR}$(prefix)/include/cyrus
-	$(srcdir)/install-sh -d ${DESTDIR}$(prefix)/man/man1
-	$(srcdir)/install-sh -d ${DESTDIR}$(prefix)/man/man3
-	$(srcdir)/install-sh -d ${DESTDIR}$(prefix)/man/man5
-	$(srcdir)/install-sh -d ${DESTDIR}$(prefix)/man/man8
+	$(srcdir)/install-sh -d ${DESTDIR}$(prefix)/share/man/man1
+	$(srcdir)/install-sh -d ${DESTDIR}$(prefix)/share/man/man3
+	$(srcdir)/install-sh -d ${DESTDIR}$(prefix)/share/man/man5
+	$(srcdir)/install-sh -d ${DESTDIR}$(prefix)/share/man/man8
 	@for d in  $(SUBDIRS); \
 	do \
 		(cd $$d; echo "### Making" install "in" `pwd`;	\

--- cyrus-imapd-2.1.13/man/Makefile.in	Wed Jun 26 01:22:29 2002
+++ cyrus-imapd-2.1.13-patched/man/Makefile.in	Sun May 11 00:43:33 2003
@@ -63,8 +63,7 @@
 cyrus_prefix = @cyrus_prefix@
 
 MAN1 = $(srcdir)/imtest.1 $(srcdir)/pop3test.1 $(srcdir)/lmtptest.1 \
-	$(srcdir)/smtptest.1 $(srcdir)/sivtest.1 $(srcdir)/mupdatetest.1 \
-	$(srcdir)/installsieve.1 $(srcdir)/sieveshell.1
+	$(srcdir)/smtptest.1 $(srcdir)/sivtest.1 $(srcdir)/mupdatetest.1
 MAN3 = $(srcdir)/imclient.3
 MAN5 = $(srcdir)/imapd.conf.5 $(srcdir)/krb.equiv.5 $(srcdir)/cyrus.conf.5
 MAN8 = $(srcdir)/arbitron.8 $(srcdir)/collectnews.8 $(srcdir)/deliver.8 \
@@ -85,19 +84,19 @@
 install:
 	for file in $(MAN1); \
 	do \
-		$(INSTALL) -m 644 $$file $(DESTDIR)$(prefix)/man/man1 || exit 1; \
+		$(INSTALL) -m 644 $$file $(DESTDIR)$(prefix)/share/man/man1 || exit 1; \
 	done
 	for file in $(MAN3); \
 	do \
-		$(INSTALL) -m 644 $$file $(DESTDIR)$(prefix)/man/man3 || exit 1; \
+		$(INSTALL) -m 644 $$file $(DESTDIR)$(prefix)/share/man/man3 || exit 1; \
 	done
 	for file in $(MAN5); \
 	do \
-		$(INSTALL) -m 644 $$file $(DESTDIR)$(prefix)/man/man5 || exit 1; \
+		$(INSTALL) -m 644 $$file $(DESTDIR)$(prefix)/share/man/man5 || exit 1; \
 	done
 	for file in $(MAN8); \
 	do \
-		$(INSTALL) -m 644 $$file $(DESTDIR)$(prefix)/man/man8 || exit 1; \
+		$(INSTALL) -m 644 $$file $(DESTDIR)$(prefix)/share/man/man8 || exit 1; \
 	done
 
 .c.o:

--- cyrus-imapd-2.1.13/cyrus-imapd	Mon May 12 11:25:51 2003
+++ cyrus-imapd-2.1.13-patched/cyrus-imapd	Fri May 16 01:00:57 2003
@@ -0,0 +1,22 @@
+#!/bin/sh
+#
+# Start/Stop script for Cyrus IMAP Server
+
+case "$1" in
+    start)
+        echo "Starting Cyrus IMAP Server"
+        @prefix@/bin/master_imapd &
+        ;;
+
+    stop)
+        echo "Stopping Cyrus IMAP Server"
+        killall master_imapd
+        ;;
+
+    *)
+        echo "Start/Stop script for Cyrus IMAP Server"
+        echo "Usage: $0 {start|stop}"
+        exit 1
+        ;;
+
+esac
