--- Pmw/find_site.py.orig	Mon Feb  4 11:09:27 2002
+++ Pmw/find_site.py	Mon Feb  4 11:09:27 2002
@@ -0,0 +1,8 @@
+# a utility script to find Python's site-packages directory and print it.
+# used for installing Python megawidgets.
+import sys, os
+for x in sys.path:
+   if os.path.basename(x) == 'site-packages':
+       print x;raise SystemExit, 0
+else:
+   raise SystemExit, 1

--- Pmw/Pmw_1_1/lib/PmwMenuBar.py	Sun Aug 18 04:19:11 2002
+++ Pmw-patched/Pmw_1_1/lib/PmwMenuBar.py	Mon Mar 10 11:28:41 2003
@@ -185,7 +185,7 @@
             for menuName in self._menuInfo.keys():
                 if self._menuInfo[menuName][0] is None:
                     menubutton = self.component(menuName + '-button')
-                    underline = string.atoi(menubutton.cget('underline'))
+                    underline = int(menubutton.cget('underline'))
                     if underline != -1:
                         label = menubutton.cget(textKey)
                         if underline < len(label):
@@ -198,7 +198,7 @@
             if end is not None:
                 for item in range(end + 1):
                     if menu.type(item) not in ('separator', 'tearoff'):
-                        underline = string.atoi(
+                        underline = int(
                             menu.entrycget(item, 'underline'))
                         if underline != -1:
                             label = menu.entrycget(item, textKey)

--- Pmw/Pmw_1_1/lib/PmwBalloon.py	Sun Aug 18 04:19:11 2002
+++ Pmw-patched/Pmw_1_1/lib/PmwBalloon.py	Mon Mar 10 11:46:07 2003
@@ -349,8 +349,8 @@
             y = bottomrel + widget.winfo_rooty()
         y = y + self['yoffset']
 
-        edges = (string.atoi(self.cget('hull_highlightthickness')) +
-            string.atoi(self.cget('hull_borderwidth'))) * 2
+        edges = (int(self.cget('hull_highlightthickness')) +
+            int(self.cget('hull_borderwidth'))) * 2
         if x + self._label.winfo_reqwidth() + edges > screenWidth:
             x = screenWidth - self._label.winfo_reqwidth() - edges
 
