--- qtella-0.6.2/configure	2003-07-02 07:49:32.000000000 +1000
+++ qtella-0.6.2-patched/configure	2003-07-26 10:44:14.000000000 +1000
@@ -3200,7 +3200,7 @@
 	do
 		for i in $QTLIBS
 		do
-			if test -f $dir/lib$i.so
+			if test -f $dir/lib$i.dylib
 			then
 				QTLIBDIR=$dir
 				echo "$as_me:$LINENO: result: found in $QTLIBDIR" >&5
@@ -3456,7 +3456,7 @@
 	do
 		for i in $OGGLIBS
 		do
-			if test -f $dir/lib$i.so
+			if test -f $dir/lib$i.dylib
 			then
 				OGGLIBDIR=$dir
 				echo "$as_me:$LINENO: result: found in $OGGLIBDIR" >&5
@@ -3534,7 +3534,7 @@
 	do
 		for i in $ID3LIBS
 		do
-			if test -f $dir/lib$i.so
+			if test -f $dir/lib$i.dylib
 			then
 				ID3LIBDIR=$dir
 				echo "$as_me:$LINENO: result: found in $ID3LIBDIR" >&5
