--- qtella-0.5.4/configure	Thu Oct  3 07:22:33 2002
+++ qtella-0.5.4-patched/configure	Mon Oct  7 21:43:48 2002
@@ -1507,7 +1507,7 @@
 	do
 		for i in $QTLIBS
 		do
-			if test -f $dir/lib$i.so
+			if test -f $dir/lib$i.dylib
 			then
 				QTLIBDIR=$dir
 				echo "$ac_t"""found in $QTLIBDIR"" 1>&6

--- qtella-0.5.4/src/HostCacheConnection.cpp	Sat Sep 21 19:36:37 2002
+++ qtella-0.5.4-patched/src/HostCacheConnection.cpp	Tue Oct  8 09:34:01 2002
@@ -12,6 +12,7 @@
 #include <qlineedit.h>
 #include <string>
 #include <vector>
+#include <iostream>
 
 #define TIMEOUT 2

--- qtella-0.5.4/src/SHA1Storage.cpp	Thu Oct  3 01:27:57 2002
+++ qtella-0.5.4-patched/src/SHA1Storage.cpp	Thu Oct 10 22:43:35 2002
@@ -100,7 +100,7 @@
 QString *SHA1Storage::get(SharedFile *sf)
 { 
       std::string n = sf -> completeFilename();
-      n += std::string::traits_type::eos();
+      n += std::string::traits_type::eof();
       QString fileName(n.data());
       QString *result = _cache[fileName];
       if (!result) {

