--- qtella-0.6.1/configure	Mon Feb  3 10:58:09 2003
+++ qtella-0.6.1-patched/configure	Sat Feb  8 11:00:34 2003
@@ -3192,7 +3192,7 @@
 	do
 		for i in $QTLIBS
 		do
-			if test -f $dir/lib$i.so
+			if test -f $dir/lib$i.dylib
 			then
 				QTLIBDIR=$dir
 				echo "$as_me:$LINENO: result: \"found in $QTLIBDIR\"" >&5
@@ -3331,7 +3331,7 @@
 
 	for dir in $with_kde_libs $with_kde_dir $KDEDIR /opt/kde3 /opt/kde2 /opt/kde /usr/lib /usr/local/lib
 	do
-		if test -f $dir/libkdecore.so
+		if test -f $dir/libkdecore.dylib
 		then
 			KDELIBDIR=$dir
 			echo "$as_me:$LINENO: result: \"found in $KDELIBDIR\"" >&5
@@ -3452,7 +3452,7 @@
 	do
 		for i in $OGGLIBS
 		do
-			if test -f $dir/lib$i.so
+			if test -f $dir/lib$i.dylib
 			then
 				OGGLIBDIR=$dir
 				echo "$as_me:$LINENO: result: \"found in $OGGLIBDIR\"" >&5
@@ -3529,7 +3529,7 @@
 	do
 		for i in $ID3LIBS
 		do
-			if test -f $dir/lib$i.so
+			if test -f $dir/lib$i.dylib
 			then
 				ID3LIBDIR=$dir
 				echo "$as_me:$LINENO: result: \"found in $ID3LIBDIR\"" >&5
