--- xcdroast-0.98alpha10/Makefile	2002-04-23 19:23:10.000000000 +1000
+++ xcdroast-0.98alpha10-patched/Makefile	2002-08-07 23:33:30.000000000 +1000
@@ -6,7 +6,7 @@
 # Installation prefix directory:
 
 #PREFIX=/usr
-PREFIX=/usr/local
+PREFIX=@prefix@
 PRE_LIBDIR=$(PREFIX)/lib/xcdroast-0.98
 
 # put here the prefix where you want xcdroast to look for 
@@ -21,8 +21,8 @@
 #INSTALL=/usr/sbin/install
 
 # for osX
-#EXTRA_CFLAGS= -g -traditional-cpp
-#EXTRA_LDFLAGS= -framework CoreAudio
+EXTRA_CFLAGS= -g -traditional-cpp
+EXTRA_LDFLAGS= -framework CoreAudio -framework IOKit -framework CoreFoundation
 
 # activate large file support
 EXTRA_CFLAGS+= -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -DUSE_LARGEFILES
@@ -78,15 +78,15 @@
 
 install:
 	$(INSTALL) -m 0755 -d $(DESTDIR)/$(PREFIX)/bin 
-	$(INSTALL) -m 0755 -d $(DESTDIR)/$(PREFIX)/man 
-	$(INSTALL) -m 0755 -d $(DESTDIR)/$(PREFIX)/man/man1
+	$(INSTALL) -m 0755 -d $(DESTDIR)/$(PREFIX)/share/man 
+	$(INSTALL) -m 0755 -d $(DESTDIR)/$(PREFIX)/share/man/man1
 	$(INSTALL) -m 0755 -d $(DESTDIR)/$(PRE_LIBDIR)
 	$(INSTALL) -m 0755 -d $(DESTDIR)/$(PRE_LIBDIR)/bin
 	$(INSTALL) -m 0755 -d $(DESTDIR)/$(PRE_LIBDIR)/icons
 	$(INSTALL) -m 0755 -d $(DESTDIR)/$(PRE_LIBDIR)/lang
 	$(INSTALL) -m 0755 -d $(DESTDIR)/$(PRE_LIBDIR)/sound
 	$(INSTALL) -m 0755 ./xcdroast $(DESTDIR)/$(PREFIX)/bin
-	$(INSTALL) -m 0644 ./xcdroast.1 $(DESTDIR)/$(PREFIX)/man/man1
+	$(INSTALL) -m 0644 ./xcdroast.1 $(DESTDIR)/$(PREFIX)/share/man/man1
 	$(INSTALL) -m 0755 ./lib/bin/* $(DESTDIR)/$(PRE_LIBDIR)/bin
 	$(INSTALL) -m 0644 ./lib/icons/* $(DESTDIR)/$(PRE_LIBDIR)/icons
 	$(INSTALL) -m 0644 ./lib/lang/* $(DESTDIR)/$(PRE_LIBDIR)/lang
