diff -ruN mutt-1.4/configure mutt-1.4-patched/configure
--- mutt-1.4/configure	2002-05-29 05:31:03.000000000 -0400
+++ mutt-1.4-patched/configure	2004-12-28 13:23:35.000000000 -0500
@@ -562,7 +562,7 @@
 ac_ext=c
 # CFLAGS is not in ac_cpp because -g, -O, etc. are not valid cpp options.
 ac_cpp='$CPP $CPPFLAGS'
-ac_compile='${CC-cc} -c $CFLAGS $CPPFLAGS conftest.$ac_ext 1>&5'
+ac_compile='${CC-cc} -c -no-cpp-precomp $CFLAGS $CPPFLAGS conftest.$ac_ext 1>&5'
 ac_link='${CC-cc} -o conftest${ac_exeext} $CFLAGS $CPPFLAGS $LDFLAGS conftest.$ac_ext $LIBS 1>&5'
 cross_compiling=$ac_cv_prog_cc_cross
 
@@ -5696,7 +5696,7 @@
 
 
 if test x$GCC = xyes -a $mutt_cv_warnings = yes; then
-        CFLAGS="-Wall -pedantic $CFLAGS"
+        CFLAGS="-Wall $CFLAGS"
 fi
 
 # Check whether --enable-nfs-fix or --disable-nfs-fix was given.
diff -ruN mutt-1.4/regex.c mutt-1.4-patched/regex.c
--- mutt-1.4/regex.c	2001-01-08 18:09:30.000000000 -0500
+++ mutt-1.4-patched/regex.c	2004-12-28 13:25:07.000000000 -0500
@@ -74,8 +74,8 @@
 /* For platform which support the ISO C amendement 1 functionality we
    support user defined character classes.  */
 #if defined _LIBC || (defined HAVE_WCTYPE_H && defined HAVE_WCHAR_H)
-# include <wctype.h>
-# include <wchar.h>
+#include <wchar.h>
+#include <wctype.h>
 #endif
 
 /* This is for other GNU distributions with internationalized messages.  */
