diff -ru apache_mod_hfs_apple-3/Makefile apache_mod_hfs_apple-3-patched/Makefile
--- apache_mod_hfs_apple-3/Makefile	2002-03-09 21:44:48.000000000 +0100
+++ apache_mod_hfs_apple-3-patched/Makefile	2002-08-05 14:28:05.000000000 +0200
@@ -21,15 +21,17 @@
 SRCFILES = Makefile $(MODULE_SRC) $(OTHER_SRC) $(HEADERS)
 #NEXTSTEP_INSTALLDIR := $(shell apxs -q LIBEXECDIR)
 #NEXTSTEP_INSTALLDIR = $(DSTROOT)
+PREFIX = @PREFIX@
 
 MORE_FLAGS = -Wc,"-DDEBUG=1 -traditional-cpp -Wno-four-char-constants"
+MORE_FLAGS = -Wc,"-DDEBUG=1 -traditional-cpp -Wno-four-char-constants -I$(PREFIX)/include" 
 
 MAKEFILEDIR = $(MAKEFILEPATH)/pb_makefiles
 include $(MAKEFILEDIR)/platform.make
 include $(MAKEFILEDIR)/commands-$(OS).make
 
 all build $(MODULE): $(MODULE_SRC) $(OTHER_SRC)
-	apxs -c -S LDFLAGS_SHLIB="-bundle -bundle_loader $(shell which httpd)" $(MORE_FLAGS) -o $(MODULE) $(MODULE_SRC) $(OTHER_SRC)
+	$(PREFIX)/sbin/apxs -c -S LDFLAGS_SHLIB="-bundle -bundle_loader $(PREFIX)/sbin/httpd" $(MORE_FLAGS) -o $(MODULE) $(MODULE_SRC) $(OTHER_SRC)
 
 installhdrs:
 	@echo "Installing header files..."
