diff -ruN cdrdao/dao/Makefile.in cdrdao.test/dao/Makefile.in
--- cdrdao/dao/Makefile.in	Sun Jun 30 15:38:53 2002
+++ cdrdao.test/dao/Makefile.in	Tue Sep 10 22:31:09 2002
@@ -67,14 +67,14 @@
 all: $(PRG)
 
 install: all installdirs
-	$(INSTALL_PROGRAM) $(PRG) $(bindir)
-	$(INSTALL_DATA) cdrdao.man $(man1dir)/cdrdao.1
-	$(INSTALL_DATA)	cdrdao.drivers $(DRIVER_TABLE_FILE)
+	$(INSTALL_PROGRAM) $(PRG) $(DESTDIR)$(bindir)
+	$(INSTALL_DATA) cdrdao.man $(DESTDIR)$(man1dir)/cdrdao.1
+	$(INSTALL_DATA)	cdrdao.drivers $(DESTDIR)$(DRIVER_TABLE_FILE)
 
 installdirs:
-	$(srcdir)/../mkinstalldirs $(bindir)
-	$(srcdir)/../mkinstalldirs $(man1dir)
-	$(srcdir)/../mkinstalldirs $(CDRDAO_DATA_DIR)
+	$(srcdir)/../mkinstalldirs $(DESTDIR)$(bindir)
+	$(srcdir)/../mkinstalldirs $(DESTDIR)$(man1dir)
+	$(srcdir)/../mkinstalldirs $(DESTDIR)$(CDRDAO_DATA_DIR)
 
 clean:
 	rm -f *.o $(PRG)
diff -ruN cdrdao/utils/Makefile.in cdrdao.test/utils/Makefile.in
--- cdrdao/utils/Makefile.in	Tue Sep 10 22:24:50 2002
+++ cdrdao.test/utils/Makefile.in	Tue Sep 10 22:32:49 2002
@@ -56,9 +56,9 @@
 	$(INSTALL_PROGRAM) @utils_targets@ $(bindir)
 
 installdirs:
-	$(srcdir)/../mkinstalldirs $(bindir)
-#	$(srcdir)/../mkinstalldirs $(man1dir)
-#	$(srcdir)/../mkinstalldirs $(CDRDAO_DATA_DIR)
+	$(srcdir)/../mkinstalldirs $(DESTDIR)$(bindir)
+#	$(srcdir)/../mkinstalldirs $(DESTDIR)$(man1dir)
+#	$(srcdir)/../mkinstalldirs $(DESTDIR)$(CDRDAO_DATA_DIR)
 
 clean:
 	rm -f *.o toc2cue toc2mp3
diff -ruN cdrdao/xdao/CdDevice.cc cdrdao.test/xdao/CdDevice.cc
--- cdrdao/xdao/CdDevice.cc	Tue Jun 25 08:08:23 2002
+++ cdrdao.test/xdao/CdDevice.cc	Tue Sep 10 22:31:09 2002
@@ -22,7 +22,11 @@
 #include <stddef.h>
 #include <stdio.h>
 #include <unistd.h>
+
+#ifdef HAVE_MALLOC_H
 #include <malloc.h>
+#endif
+
 #include <errno.h>
 #include <ctype.h>
 #include <assert.h>
diff -ruN cdrdao/xdao/Makefile.in cdrdao.test/xdao/Makefile.in
--- cdrdao/xdao/Makefile.in	Thu Jun 13 16:12:42 2002
+++ cdrdao.test/xdao/Makefile.in	Tue Sep 10 22:31:09 2002
@@ -71,32 +71,32 @@
 all: $(PRG)
 
 install: all installdirs
-	$(INSTALL_PROGRAM) $(PRG) $(bindir)
-	$(INSTALL_DATA) xcdrdao.man $(man1dir)/xcdrdao.1
-	$(INSTALL_DATA) gcdmaster.png $(PIXMAPS_DIR)/gcdmaster.png
-	$(INSTALL_DATA) pixmap_audiocd.png $(PIXMAPS_DIR)/pixmap_audiocd.png
-	$(INSTALL_DATA) pixmap_copycd.png $(PIXMAPS_DIR)/pixmap_copycd.png
-	$(INSTALL_DATA) pixmap_datacd.png $(PIXMAPS_DIR)/pixmap_datacd.png
-	$(INSTALL_DATA) pixmap_dumpcd.png $(PIXMAPS_DIR)/pixmap_dumpcd.png
-	$(INSTALL_DATA) pixmap_help.png $(PIXMAPS_DIR)/pixmap_help.png
-	$(INSTALL_DATA) pixmap_mixedcd.png $(PIXMAPS_DIR)/pixmap_mixedcd.png
-	$(INSTALL_DATA) pixmap_open.png $(PIXMAPS_DIR)/pixmap_open.png
-	$(INSTALL_DATA) pixmap_play-start.xpm $(PIXMAPS_DIR)/pixmap_play-start.xpm
-	$(INSTALL_DATA) pixmap_play-pause.xpm $(PIXMAPS_DIR)/pixmap_play-pause.xpm
-	$(INSTALL_DATA) pixmap_play-stop.xpm $(PIXMAPS_DIR)/pixmap_play-stop.xpm
-	$(INSTALL_DATA) pixmap_cursor-tool.xpm $(PIXMAPS_DIR)/pixmap_cursor-tool.xpm
-	$(INSTALL_DATA) pixmap_zoom-tool.xpm $(PIXMAPS_DIR)/pixmap_zoom-tool.xpm
-	$(INSTALL_DATA) pixmap_zoom-in.xpm $(PIXMAPS_DIR)/pixmap_zoom-in.xpm
-	$(INSTALL_DATA) pixmap_zoom-out.xpm $(PIXMAPS_DIR)/pixmap_zoom-out.xpm
-	$(INSTALL_DATA) pixmap_zoom-selection.xpm $(PIXMAPS_DIR)/pixmap_zoom-selection.xpm
-	$(INSTALL_DATA) pixmap_zoom-fit.xpm $(PIXMAPS_DIR)/pixmap_zoom-fit.xpm
-	$(INSTALL_DATA) gcdmaster.desktop $(datadir)/gnome/apps/Applications/gcdmaster.desktop
+	$(INSTALL_PROGRAM) $(PRG) $(DESTDIR)$(bindir)
+	$(INSTALL_DATA) xcdrdao.man $(DESTDIR)$(man1dir)/xcdrdao.1
+	$(INSTALL_DATA) gcdmaster.png $(DESTDIR)$(PIXMAPS_DIR)/gcdmaster.png
+	$(INSTALL_DATA) pixmap_audiocd.png $(DESTDIR)$(PIXMAPS_DIR)/pixmap_audiocd.png
+	$(INSTALL_DATA) pixmap_copycd.png $(DESTDIR)$(PIXMAPS_DIR)/pixmap_copycd.png
+	$(INSTALL_DATA) pixmap_datacd.png $(DESTDIR)$(PIXMAPS_DIR)/pixmap_datacd.png
+	$(INSTALL_DATA) pixmap_dumpcd.png $(DESTDIR)$(PIXMAPS_DIR)/pixmap_dumpcd.png
+	$(INSTALL_DATA) pixmap_help.png $(DESTDIR)$(PIXMAPS_DIR)/pixmap_help.png
+	$(INSTALL_DATA) pixmap_mixedcd.png $(DESTDIR)$(PIXMAPS_DIR)/pixmap_mixedcd.png
+	$(INSTALL_DATA) pixmap_open.png $(DESTDIR)$(PIXMAPS_DIR)/pixmap_open.png
+	$(INSTALL_DATA) pixmap_play-start.xpm $(DESTDIR)$(PIXMAPS_DIR)/pixmap_play-start.xpm
+	$(INSTALL_DATA) pixmap_play-pause.xpm $(DESTDIR)$(PIXMAPS_DIR)/pixmap_play-pause.xpm
+	$(INSTALL_DATA) pixmap_play-stop.xpm $(DESTDIR)$(PIXMAPS_DIR)/pixmap_play-stop.xpm
+	$(INSTALL_DATA) pixmap_cursor-tool.xpm $(DESTDIR)$(PIXMAPS_DIR)/pixmap_cursor-tool.xpm
+	$(INSTALL_DATA) pixmap_zoom-tool.xpm $(DESTDIR)$(PIXMAPS_DIR)/pixmap_zoom-tool.xpm
+	$(INSTALL_DATA) pixmap_zoom-in.xpm $(DESTDIR)$(PIXMAPS_DIR)/pixmap_zoom-in.xpm
+	$(INSTALL_DATA) pixmap_zoom-out.xpm $(DESTDIR)$(PIXMAPS_DIR)/pixmap_zoom-out.xpm
+	$(INSTALL_DATA) pixmap_zoom-selection.xpm $(DESTDIR)$(PIXMAPS_DIR)/pixmap_zoom-selection.xpm
+	$(INSTALL_DATA) pixmap_zoom-fit.xpm $(DESTDIR)$(PIXMAPS_DIR)/pixmap_zoom-fit.xpm
+	$(INSTALL_DATA) gcdmaster.desktop $(DESTDIR)$(datadir)/gnome/apps/Applications/gcdmaster.desktop
 
 installdirs:
-	$(srcdir)/../mkinstalldirs $(bindir)
-	$(srcdir)/../mkinstalldirs $(man1dir)
-	$(srcdir)/../mkinstalldirs $(PIXMAPS_DIR)
-	$(srcdir)/../mkinstalldirs $(datadir)/gnome/apps/Applications
+	$(srcdir)/../mkinstalldirs $(DESTDIR)$(bindir)
+	$(srcdir)/../mkinstalldirs $(DESTDIR)$(man1dir)
+	$(srcdir)/../mkinstalldirs $(DESTDIR)$(PIXMAPS_DIR)
+	$(srcdir)/../mkinstalldirs $(DESTDIR)$(datadir)/gnome/apps/Applications
 
 uninstall:
 	rm -f $(bindir)/$(PRG) $(man1dir)/xcdrdao.1
