diff -ruN cdrtools-1.11/DEFAULTS/Defaults.darwin cdrtools-1.11.new/DEFAULTS/Defaults.darwin
--- cdrtools-1.11/DEFAULTS/Defaults.darwin	Thu Feb 15 14:56:32 2001
+++ cdrtools-1.11.new/DEFAULTS/Defaults.darwin	Mon Oct 28 18:34:48 2002
@@ -34,5 +34,5 @@
 #
 DEFINSMODEF=	444
 DEFINSMODEX=	755
-DEFINSUSR=	bin
-DEFINSGRP=	bin
+DEFINSUSR=	root
+DEFINSGRP=	wheel
diff -ruN cdrtools-1.11/cdrecord/defaults.c cdrtools-1.11.new/cdrecord/defaults.c
--- cdrtools-1.11/cdrecord/defaults.c	Wed Aug 14 13:52:00 2002
+++ cdrtools-1.11.new/cdrecord/defaults.c	Mon Oct 28 18:59:51 2002
@@ -53,7 +53,7 @@
 	 * don't contribute work and don't give support, they are causing extra
 	 * work for me and this way slow down the cdrecord development.
 	 */
-	return (defltopen("/etc/default/cdrecord"));
+	return (defltopen("@FINKPREFIX@/etc/default/cdrecord"));
 }
 
 EXPORT void
diff -ruN cdrtools-1.11/doc/cdrecord.man cdrtools-1.11.new/doc/cdrecord.man
--- cdrtools-1.11/doc/cdrecord.man	Mon Apr 16 18:02:12 2001
+++ cdrtools-1.11.new/doc/cdrecord.man	Mon Oct 28 19:03:35 2002
@@ -71,7 +71,7 @@
 
 
 
-     If a file /etc/default/cdrecord exists, the parameter to the
+     If a file @FINKPREFIX@/etc/default/cdrecord exists, the parameter to the
      ddddeeeevvvv==== option may also be a drive name label in said file (see
      FILES section).
 
@@ -386,7 +386,7 @@
           If the argument to the ddddeeeevvvv==== option does not contain the
           characters  ',',  '/', '@' or ':', it is interpreted as
           an  label  name  that  may  be  found   in   the   file
-          /etc/default/cdrecord (see FILES section).
+          @FINKPREFIX@/etc/default/cdrecord (see FILES section).
 
 
 
@@ -930,7 +930,7 @@
 
 
           suitable to the open call of the SCSI transport library
-          or a label in the file /etc/default/cdrecord.
+          or a label in the file @FINKPREFIX@/etc/default/cdrecord.
 
      CDR_SPEED
           Sets the default speed value for writing (see  also   ----
@@ -942,16 +942,16 @@
 
 
 FFFFIIIILLLLEEEESSSS
-     /etc/default/cdrecord
+     @FINKPREFIX@/etc/default/cdrecord
           Default values can be set for the following options  in
-          /etc/default/cdrecord.   For  example:  CDR_FIFOSIZE=8m
+          @FINKPREFIX@/etc/default/cdrecord.   For  example:  CDR_FIFOSIZE=8m
           or CDR_SPEED=2
 
           CDR_DEVICE
                This may either hold a device identifier  that  is
                suitable  to  the  open call of the SCSI transport
                library    or    a    label    in     the     file
-               /etc/default/cdrecord  that  allows  to identify a
+               @FINKPREFIX@/etc/default/cdrecord  that  allows  to identify a
                specific drive on the system.
 
           CDR_SPEED
diff -ruN cdrtools-1.11/readcd/defaults.c cdrtools-1.11.new/readcd/defaults.c
--- cdrtools-1.11/readcd/defaults.c	Wed Aug 14 13:52:00 2002
+++ cdrtools-1.11.new/readcd/defaults.c	Mon Oct 28 18:59:51 2002
@@ -53,7 +53,7 @@
 	 * don't contribute work and don't give support, they are causing extra
 	 * work for me and this way slow down the cdrecord development.
 	 */
-	return (defltopen("/etc/default/cdrecord"));
+	return (defltopen("@FINKPREFIX@/etc/default/cdrecord"));
 }
 
 EXPORT void
