diff -ruN cdrtools-1.11/DEFAULTS/Defaults.darwin cdrtools-1.11.new/DEFAULTS/Defaults.darwin
--- cdrtools-1.11/DEFAULTS/Defaults.darwin	Thu Feb 15 14:56:32 2001
+++ cdrtools-1.11.new/DEFAULTS/Defaults.darwin	Fri Nov  1 14:38:58 2002
@@ -34,5 +34,5 @@
 #
 DEFINSMODEF=	444
 DEFINSMODEX=	755
-DEFINSUSR=	bin
-DEFINSGRP=	bin
+DEFINSUSR=	root
+DEFINSGRP=	wheel
diff -ruN cdrtools-1.11/DEFAULTS_ENG/Defaults.darwin cdrtools-1.11.new/DEFAULTS_ENG/Defaults.darwin
--- cdrtools-1.11/DEFAULTS_ENG/Defaults.darwin	Thu Feb 15 14:56:47 2001
+++ cdrtools-1.11.new/DEFAULTS_ENG/Defaults.darwin	Fri Nov  1 14:38:58 2002
@@ -36,5 +36,5 @@
 #
 DEFINSMODEF=	444
 DEFINSMODEX=	755
-DEFINSUSR=	bin
-DEFINSGRP=	bin
+DEFINSUSR=	root
+DEFINSGRP=	wheel
