diff -ruN --exclude='*~' dvd+rw-tools-6.1/growisofs.c dvd+rw-tools-6.1.new/growisofs.c
--- dvd+rw-tools-6.1/growisofs.c	2006-01-26 16:16:54.000000000 -0500
+++ dvd+rw-tools-6.1.new/growisofs.c	2006-05-23 12:04:52.000000000 -0400
@@ -372,7 +372,7 @@
 # ifndef _GNU_SOURCE
 #  define _GNU_SOURCE
 # endif
-#elif defined(__OpenBSD__) || defined(__NetBSD__) || defined(__FreeBSD__)
+#elif defined(__OpenBSD__) || defined(__NetBSD__) || defined(__FreeBSD__) || defined(__APPLE__)
 # define off64_t	off_t
 # define stat64		stat
 # define fstat64	fstat
@@ -764,7 +764,7 @@
     else				goto open_poor_man;
 }
 
-#elif defined(__OpenBSD__) || defined(__NetBSD__)
+#elif defined(__OpenBSD__) || defined(__NetBSD__) || defined(__APPLE__)
 
 char *setup_fds (char *device)
 { uid_t uid=getuid();
diff -ruN --exclude='*~' dvd+rw-tools-6.1/transport.hxx dvd+rw-tools-6.1.new/transport.hxx
--- dvd+rw-tools-6.1/transport.hxx	2006-01-26 03:02:32.000000000 -0500
+++ dvd+rw-tools-6.1.new/transport.hxx	2006-05-23 12:12:19.000000000 -0400
@@ -355,7 +355,7 @@
     {	return ioctl (fd,CDROM_MEDIA_CHANGED,CDSL_CURRENT) == 0;   }
 };
 
-#elif defined(__OpenBSD__) || defined(__NetBSD__)
+#elif defined(__OpenBSD__) || defined(__NetBSD__) || defined(__APPLE__)
 
 #include <sys/ioctl.h>
 #include <sys/scsiio.h>
