diff -ruN sh-utils-2.0/src/uname.c sh-utils-2.0-patched/src/uname.c
--- sh-utils-2.0/src/uname.c	1999-03-31 07:36:01.000000000 +0200
+++ sh-utils-2.0-patched/src/uname.c	2002-07-06 01:30:52.000000000 +0200
@@ -180,9 +180,10 @@
   if (uname (&name) == -1)
     error (1, errno, _("cannot get system name"));
 
-#if defined (HAVE_SYSINFO) && defined (SI_ARCHITECTURE)
-  if (sysinfo (SI_ARCHITECTURE, processor, sizeof (processor)) == -1)
-    error (1, errno, _("cannot get processor type"));
+#if defined(__ppc__)
+  strcpy (processor, "powerpc");
+#elif defined(__i386__)
+  strcpy (processor, "i386");
 #else
   strcpy (processor, "unknown");
 #endif
