diff -ruN uptimed-0.2.0.orig/configure uptimed-0.2.0/configure
--- uptimed-0.2.0.orig/configure	Mon May 13 05:08:52 2002
+++ uptimed-0.2.0/configure	Tue Jul 30 19:05:26 2002
@@ -3994,7 +3994,7 @@
     ;;
 
   darwin* | rhapsody*)
-    allow_undefined_flag='-undefined suppress'
+    allow_undefined_flag='-undefined suppress -flat_namespace'
     # FIXME: Relying on posixy $() will cause problems for
     #        cross-compilation, but unfortunately the echo tests do not
     #        yet detect zsh echo's removal of \ escapes.
diff -ruN uptimed-0.2.0.orig/etc/Makefile.am uptimed-0.2.0/etc/Makefile.am
--- uptimed-0.2.0.orig/etc/Makefile.am	Fri Mar 15 23:37:20 2002
+++ uptimed-0.2.0/etc/Makefile.am	Tue Jul 30 19:37:49 2002
@@ -1,3 +1,3 @@
-sysconf_DATA = uptimed.conf-dist
+sysconf_DATA = uptimed.conf
 
-EXTRA_DIST = uptimed.conf-dist rc.uptimed
+EXTRA_DIST = uptimed.conf rc.uptimed
diff -ruN uptimed-0.2.0.orig/etc/Makefile.in uptimed-0.2.0/etc/Makefile.in
--- uptimed-0.2.0.orig/etc/Makefile.in	Mon May 13 05:10:57 2002
+++ uptimed-0.2.0/etc/Makefile.in	Tue Jul 30 19:39:08 2002
@@ -74,9 +74,9 @@
 STRIP = @STRIP@
 VERSION = @VERSION@
 
-sysconf_DATA = uptimed.conf-dist
+sysconf_DATA = uptimed.conf
 
-EXTRA_DIST = uptimed.conf-dist rc.uptimed
+EXTRA_DIST = uptimed.conf rc.uptimed
 mkinstalldirs = $(SHELL) $(top_srcdir)/mkinstalldirs
 CONFIG_HEADER = ../config.h
 CONFIG_CLEAN_FILES = 
