diff -ruN kcdlabel-2.10-KDE3/kcdlabel/audiocd.h kcdlabel-2.10-KDE3.new/kcdlabel/audiocd.h
--- kcdlabel-2.10-KDE3/kcdlabel/audiocd.h	2002-11-14 10:26:59.000000000 -0500
+++ kcdlabel-2.10-KDE3.new/kcdlabel/audiocd.h	2002-12-24 10:23:33.000000000 -0500
@@ -29,7 +29,13 @@
 #include <vector>
 #include <qobject.h>
 
+#ifdef __LINUX__
 #include <linux/cdrom.h>		// specific to Linux !!! --> must be encapsulated in DEFINE IFDEF to enable compilation on other Un*x platforms...
+#endif /* __LINUX__ */
+
+#if defined( __APPLE__ )
+#include "linuxcdrom.h"
+#endif /* __APPLE__ */
 
 /**Access all the audio CD functions and compute CDDB DiscID
   *@author Pascal 'PeP' Panneels
diff -ruN kcdlabel-2.10-KDE3/kcdlabel/linuxcdrom.h kcdlabel-2.10-KDE3.new/kcdlabel/linuxcdrom.h
--- kcdlabel-2.10-KDE3/kcdlabel/linuxcdrom.h	1969-12-31 19:00:00.000000000 -0500
+++ kcdlabel-2.10-KDE3.new/kcdlabel/linuxcdrom.h	2002-12-24 10:37:02.000000000 -0500
@@ -0,0 +1,30 @@
+/*
+ * -- <linux/cdrom.h>
+ * General header file for linux CD-ROM drivers
+ * Copyright (C) 1992         David Giller, rafetmad@oxy.edu
+ *               1994, 1995   Eberhard Moenkeberg, emoenke@gwdg.de
+ *               1996         David van Leeuwen, david@tm.tno.nl
+ *               1997, 1998   Erik Andersen, andersee@debian.org
+ *               1998-2000    Jens Axboe, axboe@suse.de
+ *
+ *               Excerps taken by Chris Zubrzycki, beren12@users.sourceforge.net
+ */
+
+#ifndef _LINUX_CDROM_H
+#define _LINUX_CDROM_H
+
+
+#ifndef __u8
+# define __u8  unsigned char
+#endif
+
+
+/* Address in MSF format */
+struct cdrom_msf0
+{
+        __u8    minute;
+        __u8    second;
+        __u8    frame;
+};
+
+#endif /* _LINUX_CDROM_H */
