diff -rubN MPlayer-1.0pre3/configure MPlayer-1.0pre3.new/configure
--- MPlayer-1.0pre3/configure	Mon Dec  8 16:33:31 2003
+++ MPlayer-1.0pre3.new/configure	Tue Dec 23 11:34:06 2003
@@ -257,6 +257,7 @@
 
 Miscellaneous options:
   --enable-runtime-cpudetection    Enable runtime CPU detection [disable]
+  --disable-cpu-optimization    Disable gcc CPU build optimization [enable]
   --cc=COMPILER          use this C compiler to build MPlayer [gcc]
   --target=PLATFORM      target platform (i386-linux, arm-linux, etc)
   --enable-static        build a statically linked binary. Set further linking
@@ -337,6 +338,7 @@
 test "$CC" && _cc="$CC"
 _as=auto
 _runtime_cpudetection=no
+_cpu_optimization=yes
 for ac_option do
   case "$ac_option" in
   --target=*)
@@ -376,6 +378,12 @@
   --disable-runtime-cpudetection)
     _runtime_cpudetection=no
     ;;
+  --disable-cpu-optimization)
+    _cpu_optimization=no
+    ;;
+  --enable-cpu-optimization)
+    _cpu_optimization=yes
+    ;;
   --install-path=*)
     _install=`echo $ac_option | cut -d '=' -f 2 | sed 's/\/$//'`"/install"
     ;;
@@ -883,6 +891,7 @@
 
     echocheck "GCC & CPU optimization abilities"
 
+    if test "$_cpu_optimization" = yes; then
     if test -n "$proc"; then
         case "$proc" in
 	    601) _march='-mcpu=601' _mcpu='-mtune=601' ;;
@@ -909,6 +918,7 @@
     else
         echores "none"
     fi
+    fi
 
     ;;
 
@@ -1209,6 +1219,8 @@
   --with-extralibdir=*) ;;
   --enable-runtime-cpudetection) ;;
   --disable-runtime-cpudetection) ;;
+  --enable-cpu-optimization) ;;
+  --disable-cpu-optimization) ;;
   --install-path=*) ;;
   --with-install=*) ;;
 
diff -rubN MPlayer-1.0pre3/etc/example.conf MPlayer-1.0pre3.new/etc/example.conf
--- MPlayer-1.0pre3/etc/example.conf	Sun Nov 30 11:36:05 2003
+++ MPlayer-1.0pre3.new/etc/example.conf	Thu Dec 18 12:43:41 2003
@@ -8,13 +8,13 @@
 ## more flexibly here. See below.
 ##
 
-# vo=xv			# To specify default video driver (see -vo help for
+# vo=sdl		# To specify default video driver (see -vo help for
 			# list)
 
-# ao=oss		# To specify default audio driver (see -ao help for
+# ao=sdl		# To specify default audio driver (see -ao help for
 			# list)
 
-fs=yes			# Enlarges movie window to your desktop's size.
+fs=no			# Enlarges movie window to your desktop's size.
 			# Used by drivers: all
 
 # vm=no			# Tries to change to a different videomode
@@ -131,4 +131,4 @@
 ## Delete this default :)
 ##
 
-include = /home/gabucino/.mplayer/i_did_not_RTFM_carefully_enough...
+#include = /home/gabucino/.mplayer/i_did_not_RTFM_carefully_enough...
