diff -uNr dbus-0.31/configure dbus-0.31-new/configure
--- dbus-0.31/configure	Mon Mar  7 15:10:01 2005
+++ dbus-0.31-new/configure	Wed Mar  9 18:03:24 2005
@@ -26469,16 +26469,8 @@
 echo "${ECHO_T}not found" >&6
 fi
 
-if (! kde-config >& /dev/null); then
-    have_qt=no
-else
-    kdelibs=`kde-config --install lib --expandvars 2>/dev/null`
-    if test -z $kdelibs -o ! -f $kdelibs/libkdecore.la; then
-        have_qt=no
-    else
-        DBUS_QT_LIBS="$kdelibs/libkdecore.la"
-    fi
-fi
+have_qt=yes
+DBUS_QT_LIBS="-lqt-mt"
 
 if test x$have_qt = xno ; then
     { echo "$as_me:$LINENO: WARNING: Qt development libraries not found" >&5
diff -uNr dbus-0.31/qt/Makefile.in dbus-0.31-new/qt/Makefile.in
--- dbus-0.31/qt/Makefile.in	Mon Mar  7 15:09:55 2005
+++ dbus-0.31-new/qt/Makefile.in	Wed Mar  9 18:02:23 2005
@@ -60,8 +60,7 @@
 libLTLIBRARIES_INSTALL = $(INSTALL)
 LTLIBRARIES = $(lib_LTLIBRARIES)
 am__DEPENDENCIES_1 =
-@HAVE_QT_TRUE@libdbus_qt_1_la_DEPENDENCIES = $(am__DEPENDENCIES_1) \
-@HAVE_QT_TRUE@	$(top_builddir)/dbus/libdbus-1.la
+@HAVE_QT_TRUE@libdbus_qt_1_la_DEPENDENCIES = $(am__DEPENDENCIES_1)
 am__libdbus_qt_1_la_SOURCES_DIST = dbus-qthread.cpp \
 	$(top_srcdir)/qt/message.cpp $(top_srcdir)/qt/connection.cpp \
 	$(top_srcdir)/qt/integrator.cpp $(top_srcdir)/qt/server.cpp \
@@ -337,7 +336,7 @@
 @HAVE_QT_TRUE@	$(top_srcdir)/qt/server.h
 
 @HAVE_QT_TRUE@CLEANFILES = connection.moc integrator.moc server.moc
-@HAVE_QT_TRUE@libdbus_qt_1_la_LIBADD = $(DBUS_QT_LIBS) $(top_builddir)/dbus/libdbus-1.la
+@HAVE_QT_TRUE@libdbus_qt_1_la_LIBADD = $(DBUS_QT_LIBS) -ldbus-1
 @HAVE_QT_TRUE@libdbus_qt_1_la_LDFLAGS = -version-info 1:0
 all: all-am
 
