--- dbus-0.31/glib/Makefile.in	Mon Mar  7 15:09:50 2005
+++ dbus-0.31-new/glib/Makefile.in	Thu Mar 10 12:37:39 2005
@@ -61,8 +61,7 @@
 libLTLIBRARIES_INSTALL = $(INSTALL)
 LTLIBRARIES = $(lib_LTLIBRARIES) $(noinst_LTLIBRARIES)
 am__DEPENDENCIES_1 =
-libdbus_glib_1_la_DEPENDENCIES = $(am__DEPENDENCIES_1) \
-	$(top_builddir)/dbus/libdbus-1.la
+libdbus_glib_1_la_DEPENDENCIES = $(am__DEPENDENCIES_1)
 am_libdbus_glib_1_la_OBJECTS = dbus-glib.lo dbus-gmain.lo \
 	dbus-gmarshal.lo dbus-gobject.lo dbus-gproxy.lo dbus-gtest.lo \
 	dbus-gthread.lo dbus-gutils.lo dbus-gvalue.lo
@@ -341,7 +340,7 @@
 	dbus-gvalue.c				\
 	dbus-gvalue.h
 
-libdbus_glib_1_la_LIBADD = $(DBUS_GLIB_LIBS) $(top_builddir)/dbus/libdbus-1.la
+libdbus_glib_1_la_LIBADD = $(DBUS_GLIB_LIBS) -ldbus-1
 libdbus_glib_1_la_LDFLAGS = -export-symbols-regex "^[^_].*" -version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE)
 
 # convenience lib used here and by dbus-viewer
