diff -uNr libgdiplus-1.1.4/Makefile.in libgdiplus-1.1.4-new/Makefile.in
--- libgdiplus-1.1.4/Makefile.in	Mon Jan 31 21:05:44 2005
+++ libgdiplus-1.1.4-new/Makefile.in	Tue Feb 22 15:20:16 2005
@@ -182,7 +182,7 @@
 target_cpu = @target_cpu@
 target_os = @target_os@
 target_vendor = @target_vendor@
-SUBDIRS = libpixman cairo src tests
+SUBDIRS = src tests
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = libgdiplus.pc 
 DISTCLEANFILES = libgdiplus.pc
diff -uNr libgdiplus-1.1.4/configure libgdiplus-1.1.4-new/configure
--- libgdiplus-1.1.4/configure	Mon Jan 31 21:05:45 2005
+++ libgdiplus-1.1.4-new/configure	Tue Feb 22 14:59:24 2005
@@ -20954,13 +20954,13 @@
 fi
 
   if test x$with_libungif != xno && test -z "$LIBUNGIF"; then
-    echo "$as_me:$LINENO: checking for DGifOpenFileName in -lungif" >&5
-echo $ECHO_N "checking for DGifOpenFileName in -lungif... $ECHO_C" >&6
+    echo "$as_me:$LINENO: checking for DGifOpenFileName in -lgif" >&5
+echo $ECHO_N "checking for DGifOpenFileName in -lgif... $ECHO_C" >&6
 if test "${ac_cv_lib_ungif_DGifOpenFileName+set}" = set; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
   ac_check_lib_save_LIBS=$LIBS
-LIBS="-lungif  $LIBS"
+LIBS="-lgif  $LIBS"
 cat >conftest.$ac_ext <<_ACEOF
 /* confdefs.h.  */
 _ACEOF
@@ -21157,7 +21157,7 @@
 
 fi
 if test $ac_cv_header_gif_lib_h = yes; then
-  UNGIF='ungif'; LIBUNGIF='-lungif'
+  UNGIF='ungif'; LIBUNGIF='-lgif'
 else
   { echo "$as_me:$LINENO: WARNING: *** GIF loader will not be built (ungiflibrary not found) ***" >&5
 echo "$as_me: WARNING: *** GIF loader will not be built (ungiflibrary not found) ***" >&2;}
diff -uNr libgdiplus-1.1.4/src/gifcodec.c libgdiplus-1.1.4-new/src/gifcodec.c
--- libgdiplus-1.1.4/src/gifcodec.c	Wed Jan 26 20:33:54 2005
+++ libgdiplus-1.1.4-new/src/gifcodec.c	Tue Feb 22 15:49:15 2005
@@ -422,7 +422,7 @@
 #ifdef WORDS_BIGENDIAN
 					*ptr_red++ = (color & 0x0000ff00) >> 8;
 					*ptr_green++ = (color & 0x00ff0000) >> 16;
-					*ptr_blue++ =  (color & 0xff0000000) >> 24;
+					*ptr_blue++ =  (color & 0xff000000) >> 24;
 #else
 					*ptr_red++ = (color & 0x00ff0000) >> 16;
 					*ptr_green++ = (color & 0x0000ff00) >> 8;
