diff -uNr libgdiplus-1.1.13.4/Makefile.in libgdiplus-1.1.13.4-new/Makefile.in
--- libgdiplus-1.1.13.4/Makefile.in	Tue Feb 28 18:04:40 2006
+++ libgdiplus-1.1.13.4-new/Makefile.in	Thu Mar 16 17:07:46 2006
@@ -190,7 +190,7 @@
 target_cpu = @target_cpu@
 target_os = @target_os@
 target_vendor = @target_vendor@
-SUBDIRS = cairo src tests
+SUBDIRS = src tests
 #libpixman
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = libgdiplus.pc 
diff -uNr libgdiplus-1.1.13.4/configure libgdiplus-1.1.13.4-new/configure
--- libgdiplus-1.1.13.4/configure	Tue Feb 28 18:04:41 2006
+++ libgdiplus-1.1.13.4-new/configure	Thu Mar 16 17:10:33 2006
@@ -19605,12 +19605,8 @@
 
 subdirs="$subdirs cairo"
 
-CAIRO_LIBS='$(top_builddir)/cairo/src/libcairo.la'
-CAIRO_CFLAGS='-I$(top_builddir)/cairo/pixman/src -I$(top_srcdir)/cairo/pixman/src -I$(top_builddir)/cairo/src -I$(top_srcdir)/cairo/src'
-
-cat >>confdefs.h <<\_ACEOF
-#define USE_INCLUDED_CAIRO 1
-_ACEOF
+CAIRO_LIBS="`pkg-config --libs cairo`"
+CAIRO_CFLAGS="`pkg-config --cflags cairo`"
 
 
 # Activate this when we support linking to an already exiting Cairo installation
@@ -19994,7 +19990,7 @@
 #define _REENTRANT 1
 _ACEOF
 
-		CFLAGS="-pthreads $CFLAGS"
+		CFLAGS=" $CFLAGS"
 		;;
 
 	*-*-darwin*)
@@ -20004,7 +20000,7 @@
 _ACEOF
 
 		CFLAGS="-no-cpp-precomp $CFLAGS"
-		LDFLAGS="-flat_namespace -undefined suppress $LDFLAGS"
+		LDFLAGS="-Wl,-framework,ApplicationServices $LDFLAGS"
 		;;
 
 	*)
@@ -20027,7 +20023,7 @@
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
   ac_save_CFLAGS=$CFLAGS
-   CFLAGS="$CFLAGS -pthread"
+   CFLAGS="$CFLAGS"
 
 
    ac_ext=c
@@ -20096,7 +20092,7 @@
 if test $myapp_cv_gcc_pthread = yes
 then
    myapp_threads=""
-   CFLAGS="$CFLAGS -pthread"
+   CFLAGS="$CFLAGS"
 else
    echo "$as_me:$LINENO: checking for pthread_create in -lpthread" >&5
 echo $ECHO_N "checking for pthread_create in -lpthread... $ECHO_C" >&6
