diff -uNr libgdiplus-1.1.9.2/configure libgdiplus-1.1.9.2-new/configure
--- libgdiplus-1.1.9.2/configure	2005-10-05 12:53:16.000000000 -0400
+++ libgdiplus-1.1.9.2-new/configure	2005-10-19 22:34:11.000000000 -0400
@@ -19994,7 +19994,7 @@
 #define _REENTRANT 1
 _ACEOF
 
-		CFLAGS="-pthreads $CFLAGS"
+		CFLAGS="$CFLAGS"
 		;;
 
 	*-*-darwin*)
@@ -20004,7 +20004,7 @@
 _ACEOF
 
 		CFLAGS="-no-cpp-precomp $CFLAGS"
-		LDFLAGS="-flat_namespace -undefined suppress $LDFLAGS"
+		LDFLAGS="-Xlinker -framework -Xlinker ApplicationServices $LDFLAGS"
 		;;
 
 	*)
@@ -20027,7 +20027,7 @@
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
   ac_save_CFLAGS=$CFLAGS
-   CFLAGS="$CFLAGS -pthread"
+   CFLAGS="$CFLAGS"
 
 
    ac_ext=c
@@ -20096,7 +20096,7 @@
 if test $myapp_cv_gcc_pthread = yes
 then
    myapp_threads=""
-   CFLAGS="$CFLAGS -pthread"
+   CFLAGS="$CFLAGS"
 else
    echo "$as_me:$LINENO: checking for pthread_create in -lpthread" >&5
 echo $ECHO_N "checking for pthread_create in -lpthread... $ECHO_C" >&6
@@ -21187,13 +21187,13 @@
 fi
 
   if test x$with_libungif != xno && test -z "$LIBUNGIF"; then
-    echo "$as_me:$LINENO: checking for DGifOpenFileName in -lungif" >&5
-echo $ECHO_N "checking for DGifOpenFileName in -lungif... $ECHO_C" >&6
+    echo "$as_me:$LINENO: checking for DGifOpenFileName in -lgif" >&5
+echo $ECHO_N "checking for DGifOpenFileName in -lgif... $ECHO_C" >&6
 if test "${ac_cv_lib_ungif_DGifOpenFileName+set}" = set; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
   ac_check_lib_save_LIBS=$LIBS
-LIBS="-lungif  $LIBS"
+LIBS="-lgif  $LIBS"
 cat >conftest.$ac_ext <<_ACEOF
 /* confdefs.h.  */
 _ACEOF
@@ -21390,7 +21390,7 @@
 
 fi
 if test $ac_cv_header_gif_lib_h = yes; then
-  UNGIF='ungif'; LIBUNGIF='-lungif'
+  UNGIF='gif'; LIBUNGIF='-lgif'
 else
   { echo "$as_me:$LINENO: WARNING: *** GIF loader will not be built (ungiflibrary not found) ***" >&5
 echo "$as_me: WARNING: *** GIF loader will not be built (ungiflibrary not found) ***" >&2;}
