--- libmath++-0.0.3/examples/Makefile.am	2002-05-06 07:51:59.000000000 -0500
+++ libmath++-0.0.3-patched/examples/Makefile.am	2004-11-02 11:16:29.000000000 -0600
@@ -1,6 +1,6 @@
 #include $(top_srcdir)/niceprint.mak
 
-CXXFLAGS = -pedantic -ansi -Wall -Wno-long-long
+CXXFLAGS = -Wall -Wno-long-long
 
 LIBS = ../math++/libmath++.la
 
--- libmath++-0.0.3/math++/Makefile.am	2002-07-01 22:28:02.000000000 -0500
+++ libmath++-0.0.3-patched/math++/Makefile.am	2004-11-02 11:16:35.000000000 -0600
@@ -1,6 +1,6 @@
 #include $(top_srcdir)/niceprint.mak
 
-CXXFLAGS = -pedantic -ansi -Wall -Wno-long-long
+CXXFLAGS = -Wall -Wno-long-long
 
 lib_LTLIBRARIES = libmath++.la
 
