diff -uNr poll-emulator-1.4/Makefile poll-emulator-1.4-new/Makefile
--- poll-emulator-1.4/Makefile	2002-09-30 07:48:00.000000000 -0400
+++ poll-emulator-1.4-new/Makefile	2005-05-14 14:21:58.000000000 -0400
@@ -18,21 +18,21 @@
 # Platform-specific bits
 #
 # For GNU CC on *BSD. Should work for FreeBSD, NetBSD, OpenBSD and BSD/OS
-LINK_SHARED	= $(CC) -shared
-SHLIB_EXT	= so
-SHLIB_NOVER	= $(LIB_NAME).$(SHLIB_EXT)
-SHLIB		= $(LIB_NAME).$(SHLIB_EXT).$(VERSION)
-SHLIB_INSTALLED = $(LIBDIR)/$(LIB_NAME).$(SHLIB_EXT).$(MAJOR)
+#LINK_SHARED	= $(CC) -shared
+#SHLIB_EXT	= so
+#SHLIB_NOVER	= $(LIB_NAME).$(SHLIB_EXT)
+#SHLIB		= $(LIB_NAME).$(SHLIB_EXT).$(VERSION)
+#SHLIB_INSTALLED = $(LIBDIR)/$(LIB_NAME).$(SHLIB_EXT).$(MAJOR)
 
 # Benjamin Reed <ranger@befunk.com>:
 # On Mac OS X, comment out the above lines, and uncomment these instead.
-#LINK_SHARED	= $(CC) -install_name $(PREFIX)/lib/$(SHLIB) \
-#			-compatibility_version $(COMPAT_VERSION) \
-#			-current_version $(VERSION) -dynamiclib
-#SHLIB_EXT	= dylib
-#SHLIB_NOVER	= $(LIB_NAME).$(SHLIB_EXT)
-#SHLIB		= $(LIB_NAME).$(VERSION).$(SHLIB_EXT)
-#SHLIB_INSTALLED= $(LIBDIR)/$(LIB_NAME).$(MAJOR).$(SHLIB_EXT)
+LINK_SHARED	= $(CC) -install_name $(PREFIX)/lib/$(LIB_NAME).1.$(SHLIB_EXT) \
+			-compatibility_version $(COMPAT_VERSION) \
+			-current_version $(VERSION) -dynamiclib
+SHLIB_EXT	= dylib
+SHLIB_NOVER	= $(LIB_NAME).$(SHLIB_EXT)
+SHLIB		= $(LIB_NAME).$(VERSION).$(SHLIB_EXT)
+SHLIB_INSTALLED= $(LIBDIR)/$(LIB_NAME).$(MAJOR).$(SHLIB_EXT)
 
 # If you have a BSD-compatible install(1), use:
 INSTALL		= install -c
@@ -46,7 +46,7 @@
 MAJOR		= 1
 MINOR		= 3
 VERSION		= $(MAJOR).$(MINOR)
-COMPAT_VERSION	= $(MAJOR)
+COMPAT_VERSION	= 1.1
 CC		= cc
 LIB_NAME	= libpoll
 LIB		= $(LIB_NAME).a
diff -uNr poll-emulator-1.4/poll.c poll-emulator-1.4-new/poll.c
--- poll-emulator-1.4/poll.c	2003-01-01 14:02:29.000000000 -0500
+++ poll-emulator-1.4-new/poll.c	2005-05-14 14:23:02.000000000 -0400
@@ -6,7 +6,7 @@
 	poll - select(2)-based poll() emulation function for BSD systems.
 
   SYNOPSIS
-	#include "poll.h"
+	#include "pollemu.h"
 
 	struct pollfd
 	{
@@ -15,7 +15,7 @@
 	    short   revents;
 	}
 
-	int poll (struct pollfd *pArray, unsigned long n_fds, int timeout)
+	int pollemu (struct pollfd *pArray, unsigned long n_fds, int timeout)
 
   DESCRIPTION
 
@@ -76,7 +76,7 @@
 #include <sys/time.h>                        /* time definitions */
 #include <assert.h>                          /* assertion macros */
 #include <string.h>                          /* string functions */
-#include "poll.h"                            /* this package */
+#include "pollemu.h"                         /* this package */
 
 /*---------------------------------------------------------------------------*\
 				  Macros
@@ -255,7 +255,7 @@
 			     Public Functions
 \*---------------------------------------------------------------------------*/
 
-int poll
+int pollemu
 
 #if __STDC__ > 0
 	(struct pollfd *pArray, unsigned long n_fds, int timeout)
diff -uNr poll-emulator-1.4/poll.h poll-emulator-1.4-new/poll.h
--- poll-emulator-1.4/poll.h	2002-05-11 07:32:30.000000000 -0400
+++ poll-emulator-1.4-new/poll.h	2005-05-14 14:23:28.000000000 -0400
@@ -6,7 +6,7 @@
 	poll - select(2)-based poll() emulation function for BSD systems.
 
   SYNOPSIS
-	#include "poll.h"
+	#include "pollemu.h"
 
 	struct pollfd
 	{
@@ -15,7 +15,7 @@
 	    short   revents;
 	}
 
-	int poll (struct pollfd *pArray, unsigned long n_fds, int timeout)
+	int pollemu (struct pollfd *pArray, unsigned long n_fds, int timeout)
 
   DESCRIPTION
 
@@ -89,9 +89,9 @@
 #endif
 
 #if (__STDC__ > 0) || defined(__cplusplus)
-extern int poll (struct pollfd *pArray, unsigned long n_fds, int timeout);
+extern int pollemu (struct pollfd *pArray, unsigned long n_fds, int timeout);
 #else
-extern int poll();
+extern int pollemu();
 #endif
 
 #ifdef __cplusplus
