--- MPlayer-1.0pre7try2/libavcodec/libpostproc/Makefile	2005-04-16 16:41:13.000000000 -0400
+++ MPlayer-1.0pre7try2-new/libavcodec/libpostproc/Makefile	2006-02-16 08:04:07.000000000 -0500
@@ -4,8 +4,9 @@
 VPATH=$(SRC_PATH)/libavcodec/libpostproc
 
 ifeq ($(SHARED_PP),yes)
-SPPLIB = $(SLIBPREF)postproc$(SLIBSUF)
 SPPVERSION = 0.0.1
+SPPLIB = $(SLIBPREF)postproc.0$(SLIBSUF)
+SPPLIBINSTALL = $(libdir)/$(SPPLIB)
 endif
 PPLIB = $(LIBPREF)postproc$(LIBSUF)
 
@@ -40,7 +41,8 @@
 	$(CC) -c $(CFLAGS) -fomit-frame-pointer -fPIC -DPIC -I.. -I../.. -o $@ $<
 
 $(SPPLIB): $(SPPOBJS)
-	$(CC) -shared -Wl,-soname,$(SPPLIB).0 \
+	$(CC) -dynamiclib -install_name $(SPPLIBINSTALL) \
+	-compatibility_version 1 -current_version 1.0.2 \
 	-o $(SPPLIB) $(SPPOBJS)
 endif
 
--- MPlayer-1.0pre7try2/configure	2005-04-13 07:46:35.000000000 -0400
+++ MPlayer-1.0pre7try2-new/configure	2006-02-16 08:10:36.000000000 -0500
@@ -605,7 +605,7 @@
       cc_version="v. ?.??, bad"
       cc_verc_fail=yes
       ;;
-    2.95.[2-9]|2.95.[2-9][-.]*|3.[0-9]|3.[0-9].[0-9])
+    2.95.[2-9]|2.95.[2-9][-.]*|[3-4].*)
       _cc_major=`echo $cc_version | cut -d '.' -f 1`
       _cc_minor=`echo $cc_version | cut -d '.' -f 2`
       _cc_mini=`echo $cc_version | cut -d '.' -f 3`
@@ -6698,7 +6698,7 @@
 INSTALL = $_install
 # OPTFLAGS = -O4 $_profile $_debug $_march $_mcpu $_pipe -fomit-frame-pointer -ffast-math
 EXTRA_INC = $_inc_extra $_inc_gtk
-OPTFLAGS = -I../libvo -I../../libvo $_inc_x11 $CFLAGS \$(EXTRA_INC)
+OPTFLAGS = -faltivec -maltivec -fno-common -I../libvo -I../../libvo $_inc_x11 $CFLAGS \$(EXTRA_INC)
 STRIPBINARIES = $_stripbinaries
 CHARSET = $_charset
 HELP_FILE = $_mp_help
@@ -6754,7 +6754,7 @@
 LIBPREF=lib
 LIBSUF=.a
 SLIBPREF=lib
-SLIBSUF=.so
+SLIBSUF=.dylib
 
 # video output
 X_LIB = $_ld_gl $_ld_dga $_ld_xv $_ld_xvmc $_ld_vm $_ld_xinerama $_ld_x11 $_ld_sock
