diff -uNr gtk-sharp-0.93/Makefile.in gtk-sharp-0.93-new/Makefile.in
--- gtk-sharp-0.93/Makefile.in	Mon May 31 15:23:58 2004
+++ gtk-sharp-0.93-new/Makefile.in	Wed Jun  2 16:45:02 2004
@@ -116,7 +116,7 @@
 am__include = @am__include@
 am__quote = @am__quote@
 install_sh = @install_sh@
-SUBDIRS = sources generator parser glib pango atk gdk gtk glade art gnome gda gnomedb gtkhtml gconf rsvg vte sample doc
+SUBDIRS = sources generator parser glib pango atk gdk gtk glade art gnome gda gnomedb gtkhtml gconf rsvg vte sample
 
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = gtk-sharp.pc
