--- libgdiplus-1.1.5/Makefile.in	Fri Mar 18 08:59:48 2005
+++ libgdiplus-1.1.5-new/Makefile.in	Thu Mar 24 14:39:08 2005
@@ -182,7 +182,7 @@
 target_cpu = @target_cpu@
 target_os = @target_os@
 target_vendor = @target_vendor@
-SUBDIRS = libpixman cairo src tests
+SUBDIRS = libpixman cairo src tests
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = libgdiplus.pc 
 DISTCLEANFILES = libgdiplus.pc
--- libgdiplus-1.1.5/configure	Fri Mar 18 08:59:50 2005
+++ libgdiplus-1.1.5-new/configure	Thu Mar 24 14:55:50 2005
@@ -19390,7 +19390,7 @@
 _ACEOF
 
 else
-	if pkg-config --exact-version 0.3.0 cairo; then
+	if pkg-config --atleast-version 0.3.0 cairo; then
 		echo Cairo installation OK
 	else
 		{ { echo "$as_me:$LINENO: error: \"Did not find Cairo == 0.3.0\"" >&5
@@ -20954,13 +20954,13 @@
 fi
 
   if test x$with_libungif != xno && test -z "$LIBUNGIF"; then
-    echo "$as_me:$LINENO: checking for DGifOpenFileName in -lungif" >&5
-echo $ECHO_N "checking for DGifOpenFileName in -lungif... $ECHO_C" >&6
+    echo "$as_me:$LINENO: checking for DGifOpenFileName in -lgif" >&5
+echo $ECHO_N "checking for DGifOpenFileName in -lgif... $ECHO_C" >&6
 if test "${ac_cv_lib_ungif_DGifOpenFileName+set}" = set; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
   ac_check_lib_save_LIBS=$LIBS
-LIBS="-lungif  $LIBS"
+LIBS="-lgif  $LIBS"
 cat >conftest.$ac_ext <<_ACEOF
 /* confdefs.h.  */
 _ACEOF
@@ -21157,7 +21157,7 @@
 
 fi
 if test $ac_cv_header_gif_lib_h = yes; then
-  UNGIF='ungif'; LIBUNGIF='-lungif'
+  UNGIF='ungif'; LIBUNGIF='-lgif'
 else
   { echo "$as_me:$LINENO: WARNING: *** GIF loader will not be built (ungiflibrary not found) ***" >&5
 echo "$as_me: WARNING: *** GIF loader will not be built (ungiflibrary not found) ***" >&2;}
