diff -uNr arts-1.3.92/x11/Makefile.am arts-1.3.92-new/x11/Makefile.am
--- arts-1.3.92/x11/Makefile.am	Mon Jan  3 09:24:21 2005
+++ arts-1.3.92-new/x11/Makefile.am	Wed Feb 23 13:35:23 2005
@@ -6,7 +6,7 @@
 
 libx11globalcomm_la_SOURCES = x11globalcomm.cc x11globalcomm_impl.cc
 libx11globalcomm_la_LIBADD = $(top_builddir)/mcop/libmcop.la $(LIB_X11)
-libx11globalcomm_la_LDFLAGS = -no-undefined -module -version-info 1:0 $(X_LDFLAGS) $(all_libraries)
+libx11globalcomm_la_LDFLAGS = -no-undefined -module -version-info 1:0 $(all_libraries) $(X_LDFLAGS)
 
 DISTCLEANFILES = x11globalcomm.cc x11globalcomm.h \
                  x11globalcomm.mcoptype x11globalcomm.mcopclass
