diff -uNr kdebase-3.4.0/kcontrol/kfontinst/configure.in.in kdebase-3.4.0-new/kcontrol/kfontinst/configure.in.in
--- kdebase-3.4.0/kcontrol/kfontinst/configure.in.in	2005-02-23 06:25:11.000000000 -0500
+++ kdebase-3.4.0-new/kcontrol/kfontinst/configure.in.in	2005-03-08 11:52:40.000000000 -0500
@@ -56,7 +56,7 @@
             done
             LIBFONTCONFIG_RPATH=`echo $LIBFONTCONFIG_RPATH | sed -e "s/-L/-R/g"`
             LIBFONTCONFIG_CFLAGS="`$PKGCONFIG fontconfig --cflags`"
-            KFI_FOUND_FONTCONFIG=1
+            KFI_FOUND_FONTCONFIG=0
         fi
     fi
 
@@ -74,7 +74,7 @@
             done
             LIBFONTCONFIG_RPATH=`echo $LIBFONTCONFIG_RPATH | sed -e "s/-L/-R/g"`
             LIBFONTCONFIG_CFLAGS="`$FONTCONFIG_CONFIG --cflags`"
-            KFI_FOUND_FONTCONFIG=1
+            KFI_FOUND_FONTCONFIG=0
         fi
     fi
 
diff -uNr kdebase-3.4.0/kdeprint/kprinter/printwrapper.cpp kdebase-3.4.0-new/kdeprint/kprinter/printwrapper.cpp
--- kdebase-3.4.0/kdeprint/kprinter/printwrapper.cpp	2004-11-26 07:07:25.000000000 -0500
+++ kdebase-3.4.0-new/kdeprint/kprinter/printwrapper.cpp	2005-03-08 11:52:41.000000000 -0500
@@ -137,10 +137,6 @@
 {
 	KCmdLineArgs	*args = KCmdLineArgs::parsedArgs();
 
-#if defined(HAVE_SIGACTION) && !defined(HAVE_SIGSET)
-	struct sigaction action;
-#endif /* HAVE_SIGACTION && !HAVE_SIGSET*/
-
 	// read variables from command line
 	QString	printer = args->getOption("d");
 	QString	title = args->getOption("t");
@@ -325,6 +321,10 @@
 
 		// print from stdin
 
+#if defined(HAVE_SIGACTION) && !defined(HAVE_SIGSET)
+	struct sigaction action;
+#endif /* HAVE_SIGACTION && !HAVE_SIGSET*/
+
 #  if defined(HAVE_SIGSET)
 		sigset(SIGHUP, signal_handler);
 		sigset(SIGINT, signal_handler);
diff -uNr kdebase-3.4.0/kdesktop/init.cc kdebase-3.4.0-new/kdesktop/init.cc
--- kdebase-3.4.0/kdesktop/init.cc	2005-02-03 18:18:56.000000000 -0500
+++ kdebase-3.4.0-new/kdesktop/init.cc	2005-03-08 11:52:41.000000000 -0500
@@ -42,6 +42,9 @@
 #include <errno.h>
 #include <ksimpleconfig.h>
 
+#include <sys/attr.h>
+#include <sys/paths.h>
+
 // for multihead
 extern int kdesktop_screen_number;
 
@@ -210,21 +213,33 @@
     const QString trashDesktopPath = desktopPath + "/trash.desktop";
     const bool trashDesktopExists = QFile::exists( trashDesktopPath );
     const bool installNewTrashi18n = newRelease && trashDesktopExists; // not if deleted by user
-    if ( emptyDesktop || firstTimeWithNewTrash || installNewTrashi18n ) {
-        QString oldIcon, oldEmptyIcon;
-        if ( trashDesktopExists ) {
-            KDesktopFile trashDesktop( trashDesktopPath, true );
-            oldIcon = trashDesktop.readIcon();
-            oldEmptyIcon = trashDesktop.readEntry( "EmptyIcon" );
-        }
-        copyFile( locate( "data", "kdesktop/directory.trash" ), trashDesktopPath );
-        if ( trashDesktopExists ) {
-            KDesktopFile trashDesktop( trashDesktopPath );
-            trashDesktop.writeEntry( "Icon", oldIcon );
-            trashDesktop.writeEntry( "EmptyIcon", oldEmptyIcon );
-            trashDesktop.sync();
-        }
-    }
+//    if ( emptyDesktop || firstTimeWithNewTrash || installNewTrashi18n ) {
+//        QString oldIcon, oldEmptyIcon;
+//        if ( trashDesktopExists ) {
+//            KDesktopFile trashDesktop( trashDesktopPath, true );
+//            oldIcon = trashDesktop.readIcon();
+//            oldEmptyIcon = trashDesktop.readEntry( "EmptyIcon" );
+//        }
+//        copyFile( locate( "data", "kdesktop/directory.trash" ), trashDesktopPath );
+//        if ( trashDesktopExists ) {
+//            KDesktopFile trashDesktop( trashDesktopPath );
+//            trashDesktop.writeEntry( "Icon", oldIcon );
+//            trashDesktop.writeEntry( "EmptyIcon", oldEmptyIcon );
+//            trashDesktop.sync();
+//        }
+//        fileinfobuf finfo;
+//        struct attrlist alist;
+//        int result = getattrlist(trashDesktopPath, &alist, &finfo, sizeof(fileinfobuf), 0);
+//        if (result != 0) {
+//          kdDebug() << "couldn't get attribute list for " << trashDesktopPath << endl;
+//        } else {
+//          finfo.data.finder.fdFlags |= kFinfoIsInvisible;
+//          result = setattrlist(trashDesktopPath, &alist, &finfo.data, sizeof(finfo.data), 0);
+//          if (result != 0) {
+//            kdDebug() << "couldn't set attribute list for " << trashDesktopPath << endl;
+//          }
+//        }
+//    }
 
     if ( firstTimeWithNewTrash ) { // migrate pre-kde-3.4 trash contents
         QByteArray packedArgs;
diff -uNr kdebase-3.4.0/kdesktop/init.h kdebase-3.4.0-new/kdesktop/init.h
--- kdebase-3.4.0/kdesktop/init.h	1999-10-31 17:53:12.000000000 -0500
+++ kdebase-3.4.0-new/kdesktop/init.h	2005-03-08 11:52:41.000000000 -0500
@@ -27,4 +27,18 @@
  */
 void testLocalInstallation();
 
+typedef struct fileinfobuf {
+   u_int32_t info_length;
+   union {
+     struct {
+       u_int32_t type;
+       u_int32_t creator;
+       u_int16_t fdFlags;
+       u_int16_t location;
+       u_int32_t padding[4];
+     } finder;
+     off_t rsrcForkSize;
+   } data;
+} fileinfobuf;
+
 #endif
diff -uNr kdebase-3.4.0/kdm/configure.in.in kdebase-3.4.0-new/kdm/configure.in.in
--- kdebase-3.4.0/kdm/configure.in.in	2005-02-23 06:25:59.000000000 -0500
+++ kdebase-3.4.0-new/kdm/configure.in.in	2005-03-10 09:19:26.000000000 -0500
@@ -53,6 +53,7 @@
 AC_CHECK_FUNCS([getttyent])
 case $host_os in
 	linux*) ac_cv_func_getutxent=no;;
+	darwin*) ac_cv_func_getutxent=no;;
 	*) AC_CHECK_FUNC([getutxent]);;
 esac
 if test $ac_cv_func_getutxent = yes; then
diff -uNr kdebase-3.4.0/kioslave/info/kde-info2html.conf kdebase-3.4.0-new/kioslave/info/kde-info2html.conf
--- kdebase-3.4.0/kioslave/info/kde-info2html.conf	2004-02-24 06:30:11.000000000 -0500
+++ kdebase-3.4.0-new/kioslave/info/kde-info2html.conf	2005-03-08 11:52:41.000000000 -0500
@@ -25,6 +25,7 @@
 
 #-- location of info files.
 our @INFODIR = (
+	    "@FINKPREFIX@/share/info",
 	    "/usr/share/info",
 	    "/usr/info",
 	    "/usr/lib/info",
diff -uNr kdebase-3.4.0/kioslave/man/kio_man.cpp kdebase-3.4.0-new/kioslave/man/kio_man.cpp
--- kdebase-3.4.0/kioslave/man/kio_man.cpp	2004-11-15 21:10:34.000000000 -0500
+++ kdebase-3.4.0-new/kioslave/man/kio_man.cpp	2005-03-08 11:52:41.000000000 -0500
@@ -50,7 +50,7 @@
 
 MANProtocol *MANProtocol::_self = 0;
 
-#define SGML2ROFF_DIRS "/usr/lib/sgml"
+#define SGML2ROFF_DIRS "@FINKPREFIX@/share/sgml"
 
 /*
  * Drop trailing ".section[.gz]" from name
@@ -862,6 +862,7 @@
                         "/usr/sunpc/man",
                         "/usr/ncd/man",
                         "/usr/newsprint/man",
+                        "@FINKPREFIX@/share/man",
                         NULL };
 
 
diff -uNr kdebase-3.4.0/libkonq/Makefile.am kdebase-3.4.0-new/libkonq/Makefile.am
--- kdebase-3.4.0/libkonq/Makefile.am	2004-11-21 05:07:01.000000000 -0500
+++ kdebase-3.4.0-new/libkonq/Makefile.am	2005-03-08 11:52:41.000000000 -0500
@@ -21,7 +21,7 @@
 
 lib_LTLIBRARIES = libkonq.la
 libkonq_la_LDFLAGS = $(all_libraries) -version-info 6:0:2 -no-undefined
-libkonq_la_LIBADD = $(LIB_KPARTS)
+libkonq_la_LIBADD = $(LIB_KPARTS) $(LIBZ)
 
 libkonq_la_SOURCES = konq_popupmenu.cc knewmenu.cc \
    konq_xmlguiclient.cc\
diff -uNr kdebase-3.4.0/startkde kdebase-3.4.0-new/startkde
--- kdebase-3.4.0/startkde	2005-03-04 06:20:04.000000000 -0500
+++ kdebase-3.4.0-new/startkde	2005-03-08 11:52:41.000000000 -0500
@@ -3,6 +3,11 @@
 #  DEFAULT KDE STARTUP SCRIPT ( KDE-3.4 )
 #
 
+source "@FINKPREFIX@/bin/init.sh"
+
+[ -x @FINKPREFIX@/bin/remap-bad-apple-keys.pl ] && @FINKPREFIX@/bin/remap-bad-apple-keys.pl
+nohup @FINKPREFIX@/bin/esd -noterminate -nobeeps -as 2 >/dev/null 2>&1 &
+
 # When the X server dies we get a HUP signal from xinit. We must ignore it
 # because we still need to do some cleanup.
 trap 'echo GOT SIGHUP' HUP
@@ -179,20 +184,23 @@
 
 echo 'startkde: Starting up...'  1>&2
 
-# run KPersonalizer before the session, if this is the first login
-if kreadconfig --file kpersonalizerrc --group General --key FirstLogin --default true --type bool; then
-    # start only dcopserver, don't start whole kdeinit (takes too long)
-    echo 'startkde: Running kpersonalizer...'  1>&2
-    dcopserver
-    kwin --lock &
-    kpersonalizer --before-session
-    # handle kpersonalizer restarts (language change)
-    while test $? -eq 1; do
-        kpersonalizer --r --before-session
-    done
-    dcopquit kwin
-    dcopserver_shutdown --wait
-fi
+# on osx we skip this, the defaults are sane and most people
+# do bad things like "mac os menus" which makes it worse
+
+## run KPersonalizer before the session, if this is the first login
+#if kreadconfig --file kpersonalizerrc --group General --key FirstLogin --default true --type bool; then
+#    # start only dcopserver, don't start whole kdeinit (takes too long)
+#    echo 'startkde: Running kpersonalizer...'  1>&2
+#    dcopserver
+#    kwin --lock &
+#    kpersonalizer --before-session
+#    # handle kpersonalizer restarts (language change)
+#    while test $? -eq 1; do
+#        kpersonalizer --r --before-session
+#    done
+#    dcopquit kwin
+#    dcopserver_shutdown --wait
+#fi
 
 # the splashscreen and progress indicator
 splash=`kreadconfig --file ksplashrc --group KSplash --key Theme`
@@ -226,7 +234,29 @@
 # We only check for 255 which means that the ksmserver process could not be 
 # started, any problems thereafter, e.g. ksmserver failing to initialize, 
 # will remain undetected.
-test -n "$KDEWM" && KDEWM="--windowmanager $KDEWM"
+
+USE_PROXY=0
+if test -n "$KDEWM"; then
+  if test -z `expr "$KDEWM" : '\(.*quartz-wm.*\)'`; then
+    USE_PROXY=1
+  fi
+  KDEWM="--windowmanager $KDEWM"
+else
+  KDEWM="--windowmanager kwin"
+  USE_PROXY=1
+fi
+
+if test "$USE_PROXY" = "1"; then
+  if test -x /usr/X11R6/bin/quartz-wm; then
+    /usr/X11R6/bin/quartz-wm --only-proxy >/dev/null 2>&1 &
+  else
+    AUTOCUTSEL=`which autocutsel 2>/dev/null`
+    if test -n "$AUTOCUTSEL" && test -x "$AUTOCUTSEL"; then
+      $AUTOCUTSEL >/dev/null 2>&1 &
+    fi
+  fi
+fi
+
 kwrapper ksmserver $KDEWM 
 if test $? -eq 255; then
   # Startup error
