diff -uNr kdeedu-3.2.91/kig/configure.in.in kdeedu-3.2.91-new/kig/configure.in.in
--- kdeedu-3.2.91/kig/configure.in.in	2004-06-25 05:09:55.000000000 -0400
+++ kdeedu-3.2.91-new/kig/configure.in.in	2004-07-03 15:47:06.000000000 -0400
@@ -102,9 +102,10 @@
     __regina_py_save_libs="$LIBS"
     __regina_py_ok=0
     for pyver in python python2.3 python2.2; do
-      for incdir in "/usr/include/$pyver" "/usr/local/include/$pyver" \
+      for incdir in "$prefix/$pyver/include " \
+          "/usr/include/$pyver" "/usr/local/include/$pyver" \
           "/usr/$pyver/include" "/usr/local/$pyver/include" \
-          "$prefix/include/$pyver" "$prefix/$pyver/include"; do
+          "$prefix/include/$pyver"; do
         CXXFLAGS="$__regina_py_save_cxxflags -I$incdir"
         LDFLAGS="$__regina_py_save_ldflags -shared"
 
@@ -126,9 +127,9 @@
             AC_MSG_RESULT([yes])
 	
             # Check for -lpython.
-            for pylib in "$pyver" python python2.3 python2.2; do
-              for pylibdir in "/usr/lib/" "/usr/local/lib" "/usr/lib/$pyver/config/" \
-		"/usr/local/lib/$pyver/config"; do
+            for pylib in "$pyver" python2.3 python2.2 python; do
+              for pylibdir in "$prefix/lib/" "$prefix/lib/$pyver/config/" "/usr/lib/" \
+              "/usr/local/lib" "/usr/lib/$pyver/config/" "/usr/local/lib/$pyver/config"; do
 		for extralibs in "" "-lpthread -lm -lutil -ldl"; do
                   AC_MSG_CHECKING([for $pylibdir/lib$pylib and $extralibs with $incdir/Python.h])
 		  LDFLAGS="$__regina_py_save_ldflags -L$pylibdir"
diff -uNr kdeedu-3.2.91/kstars/kstars/indi/Makefile.am kdeedu-3.2.91-new/kstars/kstars/indi/Makefile.am
--- kdeedu-3.2.91/kstars/kstars/indi/Makefile.am	2004-06-25 05:10:04.000000000 -0400
+++ kdeedu-3.2.91-new/kstars/kstars/indi/Makefile.am	2004-07-03 22:17:03.000000000 -0400
@@ -1,4 +1,4 @@
-bin_PROGRAMS = indiserver lx200generic celestrongps fliccd v4ldriver v4lphilips
+bin_PROGRAMS = indiserver lx200generic celestrongps fliccd
 
 liblilxml_a_SOURCES = lilxml.c
 libindicom_a_SOURCES = indicom.c fitsrw.c
@@ -17,12 +17,6 @@
 fliccd_SOURCES =  eventloop.c fli_ccd.c indidrivermain.c
 fliccd_LDADD = fli/libfli.la lzo/libminilzo.la libindicom.a liblilxml.a -lm
 
-v4ldriver_SOURCES = eventloop.c indidrivermain.c v4ldriver.cpp
-v4ldriver_LDADD = libindicom.a liblilxml.a webcam/libwebcam.la lzo/libminilzo.la
-
-v4lphilips_SOURCES = eventloop.c indidrivermain.c v4lphilips.cpp
-v4lphilips_LDADD = libindicom.a liblilxml.a webcam/libwebcam.la lzo/libminilzo.la
-
 appdatadir = $(kde_datadir)/kstars
 appdata_DATA = drivers.xml
 
diff -uNr kdeedu-3.2.91/kstars/kstars/indi/webcam/Makefile.am kdeedu-3.2.91-new/kstars/kstars/indi/webcam/Makefile.am
--- kdeedu-3.2.91/kstars/kstars/indi/webcam/Makefile.am	2004-06-27 14:15:23.000000000 -0400
+++ kdeedu-3.2.91-new/kstars/kstars/indi/webcam/Makefile.am	2004-07-03 22:11:55.000000000 -0400
@@ -1,23 +1,9 @@
-if LINUX
-   libwebcam_linux = libwebcam_linux.la
-endif
-
-if BSD
-   libwebcam_bsd = libwebcam_bsd.la
-endif
+libwebcam_files = dummy.cpp
 
-if NULL
-   libwebcam_null = libwebcam_null.la
+if LINUX
+	libwebcam_files = PPort.cpp port.cpp QCamV4L.cpp philipsV4L.cpp ccvt_c2.c ccvt_misc.c
 endif
 
-noinst_LTLIBRARIES = libwebcam.la $(libwebcam_linux) $(libwebcam_bsd) $(libwebcam_null)
-
-libwebcam_linux_la_SOURCES = PPort.cpp port.cpp QCamV4L.cpp philipsV4L.cpp ccvt_c2.c ccvt_misc.c
-libwebcam_bsd_la_SOURCES = dummy.cpp
-libwebcam_null_la_SOURCES = dummy.cpp
-
-libwebcam_la_SOURCES = empty_file.cpp
-libwebcam_la_LIBADD = $(libwebcam_linux) $(libwebcam_bsd) $(libwebcam_null)
+noinst_LTLIBRARIES = libwebcam.la
 
-empty_file.cpp:
-	echo > empty_file.cpp
+libwebcam_la_SOURCES = $(libwebcam_files)
