diff -uNr kdeedu-3.2.90/kstars/kstars/indi/webcam/Makefile.am kdeedu-3.2.90-new/kstars/kstars/indi/webcam/Makefile.am
--- kdeedu-3.2.90/kstars/kstars/indi/webcam/Makefile.am	Fri May  7 11:16:43 2004
+++ kdeedu-3.2.90-new/kstars/kstars/indi/webcam/Makefile.am	Thu Jun 24 21:15:17 2004
@@ -10,13 +10,13 @@
    libwebcam_null = libwebcam_null.la
 endif
 
-noinst_LTLIBRARIES = libwebcam.la $(libwebcam_linux) $(libwebcam_bsd) $(libwebcam_null)
+noinst_LTLIBRARIES = libwebcam.la $(libwebcam_null)
 
 libwebcam_linux_la_SOURCES = PPort.cpp port.cpp vcvt_c.c QCamV4L.cpp ccvt_mmx_old.S
 libwebcam_null_la_SOURCES = dummy.cpp
 
 libwebcam_la_SOURCES = empty_file.cpp
-libwebcam_la_LIBADD = $(libwebcam_linux) $(libwebcam_bsd) $(libwebcam_null)
+libwebcam_la_LIBADD = $(libwebcam_null)
 
 empty_file.cpp:
 	echo > empty_file.cpp
