diff -uNr kdegraphics-3.4.0/kdvi/Makefile.am kdegraphics-3.4.0-new/kdvi/Makefile.am
--- kdegraphics-3.4.0/kdvi/Makefile.am	Thu Feb  3 17:44:57 2005
+++ kdegraphics-3.4.0-new/kdvi/Makefile.am	Mon Feb 28 13:26:48 2005
@@ -1,7 +1,7 @@
 # set the include path for X, qt and KDE
 INCLUDES= -I$(top_srcdir)/kviewshell \
 	-I$(top_builddir)/kviewshell \
-	$(all_includes) $(LIBFREETYPE_CFLAGS)
+	$(LIBFREETYPE_CFLAGS) $(all_includes)
 # claim, which subdirectories you want to install
 SUBDIRS = . pix
 
diff -uNr kdegraphics-3.4.0/ksvg/impl/libs/libtext2path/src/Makefile.am kdegraphics-3.4.0-new/ksvg/impl/libs/libtext2path/src/Makefile.am
--- kdegraphics-3.4.0/ksvg/impl/libs/libtext2path/src/Makefile.am	Fri Mar 26 15:21:56 2004
+++ kdegraphics-3.4.0-new/ksvg/impl/libs/libtext2path/src/Makefile.am	Mon Feb 28 13:26:48 2005
@@ -1,10 +1,10 @@
 lib_LTLIBRARIES = libtext2path.la
 
 KDE_CXXFLAGS = $(USE_EXCEPTIONS)
-INCLUDES = $(FRIBIDI_CFLAGS) $(FREETYPE_CFLAGS) $(FONTCONFIG_CFLAGS) $(all_includes)
+INCLUDES = $(FREETYPE_CFLAGS) $(FRIBIDI_CFLAGS) $(FONTCONFIG_CFLAGS) $(all_includes)
 
 libtext2path_includedir=$(includedir)/libtext2path-0.1
 libtext2path_include_HEADERS = BezierPath.h Glyph.h GlyphTracer.h
 
 libtext2path_la_SOURCES = Affine.cpp Rectangle.cpp Font.cpp Glyph.cpp GlyphTracer.cpp Converter.cpp QtUnicode.cpp
-libtext2path_la_LDFLAGS = $(FRIBIDI_LIBS) $(FONTCONFIG_LIBS)
+libtext2path_la_LDFLAGS = $(FRIBIDI_LIBS) $(FONTCONFIG_LIBS) $(FREETYPE_LIBS)
diff -uNr kdegraphics-3.4.0/ksvg/plugin/backends/libart/Makefile.am kdegraphics-3.4.0-new/ksvg/plugin/backends/libart/Makefile.am
--- kdegraphics-3.4.0/ksvg/plugin/backends/libart/Makefile.am	Mon Jan  3 09:53:15 2005
+++ kdegraphics-3.4.0-new/ksvg/plugin/backends/libart/Makefile.am	Mon Feb 28 13:26:48 2005
@@ -8,4 +8,4 @@
 kde_services_DATA = ksvglibartcanvas.desktop
 
 KDE_CXXFLAGS = $(USE_EXCEPTIONS)
-INCLUDES = $(LIBART_CFLAGS) $(FREETYPE_CFLAGS) $(FONTCONFIG_CFLAGS) -I$(top_srcdir)/ksvg/dom -I$(top_srcdir)/ksvg/impl -I$(top_srcdir)/ksvg/ecma -I$(top_srcdir)/ksvg/core -I$(top_srcdir)/ksvg/impl/libs/art_support -I$(top_srcdir)/ksvg/impl/libs/libtext2path/src $(KDE_INCLUDES) $(all_includes)
+INCLUDES = $(FREETYPE_CFLAGS) $(LIBART_CFLAGS) $(FONTCONFIG_CFLAGS) -I$(top_srcdir)/ksvg/dom -I$(top_srcdir)/ksvg/impl -I$(top_srcdir)/ksvg/ecma -I$(top_srcdir)/ksvg/core -I$(top_srcdir)/ksvg/impl/libs/art_support -I$(top_srcdir)/ksvg/impl/libs/libtext2path/src $(KDE_INCLUDES) $(all_includes)
diff -uNr kdegraphics-3.4.0/libkscan/img_canvas.cpp kdegraphics-3.4.0-new/libkscan/img_canvas.cpp
--- kdegraphics-3.4.0/libkscan/img_canvas.cpp	Tue Nov 16 13:07:25 2004
+++ kdegraphics-3.4.0-new/libkscan/img_canvas.cpp	Mon Feb 28 13:26:48 2005
@@ -700,7 +700,7 @@
                            double(image->width());
             kdDebug(29000) << "FIT WIDTH scrollbar to substract: " << sbWidth << endl;
         }
-        scale_factor = 100*used_xscaler;
+        scale_factor = (int)(100*used_xscaler);
         break;
     case FIT_HEIGHT:
         used_yscaler = used_xscaler = double(noSBSize.height())/double(image->height());
@@ -715,12 +715,12 @@
             kdDebug(29000) << "FIT HEIGHT scrollbar to substract: " << sbWidth << endl;
             // scale = int(100.0*(noSBSize.height() -sbWidth) / image->height());
         }
-        scale_factor = 100*used_xscaler;
+        scale_factor = (int)(100*used_xscaler);
 
         break;
     case ZOOM:
         used_xscaler = used_yscaler = double(getScaleFactor())/100.0;
-        scale_factor = 100*used_xscaler;
+        scale_factor = (int)(100*used_xscaler);
         break;
     default:
         break;
