Index: korganizer/plugins/Makefile.am
===================================================================
RCS file: /home/kde/kdepim/korganizer/plugins/Makefile.am,v
retrieving revision 1.9
diff -u -u -r1.9 Makefile.am
--- korganizer/plugins/Makefile.am	27 Sep 2003 21:44:32 -0000	1.9
+++ korganizer/plugins/Makefile.am	6 Nov 2003 14:21:05 -0000
@@ -2,5 +2,5 @@
 exchange_SUBDIR=exchange
 endif
 
-SUBDIRS=birthdays datenums holidays projectview webexport hebrew $(exchange_SUBDIR) 
+SUBDIRS=birthdays datenums projectview webexport hebrew $(exchange_SUBDIR) 
 
Index: libical/src/libical/Makefile.am
===================================================================
RCS file: /home/kde/kdepim/libical/src/libical/Makefile.am,v
retrieving revision 1.24
diff -u -u -r1.24 Makefile.am
--- libical/src/libical/Makefile.am	23 Jul 2003 05:37:15 -0000	1.24
+++ libical/src/libical/Makefile.am	6 Nov 2003 14:21:06 -0000
@@ -26,7 +26,7 @@
 DESIGNDATA = $(top_srcdir)/libical/design-data
 ICALSCRIPTS = $(top_srcdir)/libical/scripts
 
-noinst_LTLIBRARIES = libical.la
+kde_module_LTLIBRARIES = libical.la
 
 AM_YFLAGS = -d -v -t -pical_yy
 AM_LFLAGS = -Pical_yy -olex.yy.c
Index: libical/src/libicalss/Makefile.am
===================================================================
RCS file: /home/kde/kdepim/libical/src/libicalss/Makefile.am,v
retrieving revision 1.15
diff -u -u -r1.15 Makefile.am
--- libical/src/libicalss/Makefile.am	30 Aug 2003 20:03:29 -0000	1.15
+++ libical/src/libicalss/Makefile.am	6 Nov 2003 14:21:06 -0000
@@ -1,5 +1,5 @@
 
-noinst_LTLIBRARIES = libicalss.la
+kde_module_LTLIBRARIES = libicalss.la
 
 AM_YFLAGS =-d -v -p ss
 AM_LFLAGS = -Pss
Index: kpilot/conduits/malconduit/mal-conduit.cc
===================================================================
RCS file: /home/kde/kdepim/kpilot/conduits/malconduit/mal-conduit.cc,v
retrieving revision 1.17
diff -u -u -r1.17 mal-conduit.cc
--- kpilot/conduits/malconduit/mal-conduit.cc	13 Oct 2003 13:43:38 -0000	1.17
+++ kpilot/conduits/malconduit/mal-conduit.cc	6 Nov 2003 15:23:05 -0000
@@ -34,6 +34,7 @@
 
 #include "options.h"
 
+#include <stdarg.h>
 #include <qregexp.h>
 #include <kconfig.h>
 #include <kdebug.h>
