Index: cervisia/cvsservice/Makefile.am
===================================================================
RCS file: /home/kde/kdesdk/cervisia/cvsservice/Makefile.am,v
retrieving revision 1.11
diff -u -u -r1.11 Makefile.am
--- cervisia/cvsservice/Makefile.am	18 Jul 2003 21:55:21 -0000	1.11
+++ cervisia/cvsservice/Makefile.am	7 Nov 2003 02:28:01 -0000
@@ -5,6 +5,8 @@
 #
 #
 
+CC=g++-3.3
+
 INCLUDES = $(all_includes)
 
 # cvs DCOP service
Index: kcachegrind/kcachegrind/sourceitem.h
===================================================================
RCS file: /home/kde/kdesdk/kcachegrind/kcachegrind/sourceitem.h,v
retrieving revision 1.2
diff -u -u -r1.2 sourceitem.h
--- kcachegrind/kcachegrind/sourceitem.h	30 Aug 2003 22:47:25 -0000	1.2
+++ kcachegrind/kcachegrind/sourceitem.h	7 Nov 2003 02:28:01 -0000
@@ -25,6 +25,7 @@
 
 #include <qlistview.h>
 #include "tracedata.h"
+#undef fileno
 
 class SourceItem: public QListViewItem
 {
