diff -uNr kdevelop-3.1.1/admin/libtool.m4.in kdevelop-3.1.1-new/admin/libtool.m4.in
--- kdevelop-3.1.1/admin/libtool.m4.in	Fri Apr  2 10:39:50 2004
+++ kdevelop-3.1.1-new/admin/libtool.m4.in	Fri Oct 22 07:43:50 2004
@@ -672,6 +672,19 @@
     lt_cv_sys_max_cmd_len=8192;
     ;;
 
+  netbsd* | freebsd* | openbsd* | darwin* )
+    # This has been around since 386BSD, at least.  Likely further.
+    if test -x /sbin/sysctl; then
+      lt_cv_sys_max_cmd_len=`/sbin/sysctl -n kern.argmax`
+    elif test -x /usr/sbin/sysctl; then
+      lt_cv_sys_max_cmd_len=`/usr/sbin/sysctl -n kern.argmax`
+    else
+      lt_cv_sys_max_cmd_len=65536 # usable default for *BSD
+    fi
+    # And add a safety zone
+    lt_cv_sys_max_cmd_len=`expr $lt_cv_sys_max_cmd_len \/ 4`
+    ;;
+
  *)
     # If test is not a shell built-in, we'll probably end up computing a
     # maximum length that is only half of the actual maximum length, but
@@ -2560,10 +2573,21 @@
     _LT_AC_TAGVAR(archive_cmds_need_lc, $1)=no
     case "$host_os" in
     rhapsody* | darwin1.[[012]])
-      _LT_AC_TAGVAR(allow_undefined_flag, $1)='-undefined suppress'
+      _LT_AC_TAGVAR(allow_undefined_flag, $1)='-Wl,-undefined -Wl,suppress'
       ;;
     *) # Darwin 1.3 on
-      test -z ${LD_TWOLEVEL_NAMESPACE} && _LT_AC_TAGVAR(allow_undefined_flag, $1)='-flat_namespace -undefined suppress'
+      if test -z ${MACOSX_DEPLOYMENT_TARGET} ; then
+        _LT_AC_TAGVAR(allow_undefined_flag, $1)='-Wl,-flat_namespace -Wl,-undefined -Wl,suppress'
+      else
+        case ${MACOSX_DEPLOYMENT_TARGET} in
+          10.[012])
+            _LT_AC_TAGVAR(allow_undefined_flag, $1)='-Wl,-flat_namespace -Wl,-undefined -Wl,suppress'
+            ;;
+          10.*)
+            _LT_AC_TAGVAR(allow_undefined_flag, $1)='-Wl,-undefined -Wl,dynamic_lookup'
+            ;;
+        esac
+      fi
       ;;
     esac
     output_verbose_link_cmd='echo'
@@ -2872,10 +2896,21 @@
     _LT_AC_TAGVAR(archive_cmds_need_lc, $1)=no
     case "$host_os" in
     rhapsody* | darwin1.[[012]])
-      _LT_AC_TAGVAR(allow_undefined_flag, $1)='-undefined suppress'
+      _LT_AC_TAGVAR(allow_undefined_flag, $1)='-Wl,-undefined -Wl,suppress'
       ;;
     *) # Darwin 1.3 on
-      test -z ${LD_TWOLEVEL_NAMESPACE} && _LT_AC_TAGVAR(allow_undefined_flag, $1)='-flat_namespace -undefined suppress'
+      if test -z ${MACOSX_DEPLOYMENT_TARGET} ; then
+        _LT_AC_TAGVAR(allow_undefined_flag, $1)='-Wl,-flat_namespace -Wl,-undefined -Wl,suppress'
+      else
+        case ${MACOSX_DEPLOYMENT_TARGET} in
+          10.[012])
+            _LT_AC_TAGVAR(allow_undefined_flag, $1)='-Wl,-flat_namespace -Wl,-undefined -Wl,suppress'
+            ;;
+          10.*)
+            _LT_AC_TAGVAR(allow_undefined_flag, $1)='-Wl,-undefined -Wl,dynamic_lookup'
+            ;;
+        esac
+      fi
       ;;
     esac
     lt_int_apple_cc_single_mod=no
@@ -5312,10 +5347,21 @@
       _LT_AC_TAGVAR(archive_cmds_need_lc, $1)=no
       case "$host_os" in
       rhapsody* | darwin1.[[012]])
-	_LT_AC_TAGVAR(allow_undefined_flag, $1)='-undefined suppress'
+	_LT_AC_TAGVAR(allow_undefined_flag, $1)='-Wl,-undefined -Wl,suppress'
 	;;
       *) # Darwin 1.3 on
-	test -z ${LD_TWOLEVEL_NAMESPACE} && _LT_AC_TAGVAR(allow_undefined_flag, $1)='-flat_namespace -undefined suppress'
+      if test -z ${MACOSX_DEPLOYMENT_TARGET} ; then
+        _LT_AC_TAGVAR(allow_undefined_flag, $1)='-Wl,-flat_namespace -Wl,-undefined -Wl,suppress'
+      else
+        case ${MACOSX_DEPLOYMENT_TARGET} in
+          10.[012])
+            _LT_AC_TAGVAR(allow_undefined_flag, $1)='-Wl,-flat_namespace -Wl,-undefined -Wl,suppress'
+            ;;
+          10.*)
+            _LT_AC_TAGVAR(allow_undefined_flag, $1)='-Wl,-undefined -Wl,dynamic_lookup'
+            ;;
+        esac
+      fi
 	;;
       esac
     	lt_int_apple_cc_single_mod=no
diff -uNr kdevelop-3.1.1/admin/ltmain.sh kdevelop-3.1.1-new/admin/ltmain.sh
--- kdevelop-3.1.1/admin/ltmain.sh	Tue Jun 15 12:47:06 2004
+++ kdevelop-3.1.1-new/admin/ltmain.sh	Fri Oct 22 07:43:51 2004
@@ -4397,6 +4397,7 @@
 	# We should set the shlibpath_var
 	rpath=
 	for dir in $temp_rpath; do
+     if test "$dir" != "@FINKPREFIX@/lib"; then
 	  case $dir in
 	  [\\/]* | [A-Za-z]:[\\/]*)
 	    # Absolute path.
@@ -4407,6 +4408,7 @@
 	    rpath="$rpath\$thisdir/$dir:"
 	    ;;
 	  esac
+     fi
 	done
 	temp_rpath="$rpath"
       fi
diff -uNr kdevelop-3.1.1/languages/cpp/Makefile.am kdevelop-3.1.1-new/languages/cpp/Makefile.am
--- kdevelop-3.1.1/languages/cpp/Makefile.am	Thu Aug  5 15:27:58 2004
+++ kdevelop-3.1.1-new/languages/cpp/Makefile.am	Fri Oct 22 07:43:50 2004
@@ -3,12 +3,14 @@
 SUBDIRS = templates subclassing_template newclass_templates file_templates app_templates compiler pcsimporter debugger doc
 
 INCLUDES = -I$(top_srcdir)/lib/catalog -I$(top_srcdir)/lib/compat -I$(top_srcdir)/lib/cppparser -I$(top_srcdir)/lib/external_interfaces -I$(top_srcdir)/lib/interfaces -I$(top_srcdir)/lib/util -I$(top_builddir)/bdb/build_unix -I$(top_srcdir)/bdb/include $(all_includes)
+LDFLAGS = -L$(top_builddir)/bdb/build_unix
 
+noinst_LTLIBRARIES = libkdevcppcommon.la
 kde_module_LTLIBRARIES = libkdevcppsupport.la
-libkdevcppsupport_la_LDFLAGS = $(all_libraries) $(KDE_PLUGIN) -L$(top_srcdir)/bdb/dist
-libkdevcppsupport_la_LIBADD = $(top_builddir)/bdb/build_unix/libdb.a $(top_builddir)/lib/libkdevelop.la $(top_builddir)/lib/catalog/libkdevcatalog.la $(top_builddir)/lib/cppparser/libkdevcppparser.la
 
-libkdevcppsupport_la_SOURCES = cppsupportpart.cpp ccconfigwidget.cpp \
+libkdevcppcommon_la_LDFLAGS = $(all_libraries) $(KDE_RPATH) -L$(top_srcdir)/bdb/dist
+libkdevcppcommon_la_LIBADD = -ldb $(top_builddir)/lib/libkdevelop.la $(top_builddir)/lib/catalog/libkdevcatalog.la $(top_builddir)/lib/cppparser/libkdevcppparser.la
+libkdevcppcommon_la_SOURCES = cppsupportpart.cpp ccconfigwidget.cpp \
 	ccconfigwidgetbase.ui kdevdriver.cpp cppcodecompletion.cpp problemreporter.cpp \
 	configproblemreporter.ui backgroundparser.cpp ast_utils.cpp store_walker.cpp \
 	KDevCppSupportIface.cpp KDevCppSupportIface.skel cppsupportfactory.cpp tag_creator.cpp \
@@ -18,6 +20,13 @@
 	addmethoddialog.cpp cppsupport_utils.cpp createpcsdialogbase.ui createpcsdialog.cpp \
 	qtdesignerintegration.cpp implementationwidgetbase.ui implementationwidget.cpp
 
+libkdevcppsupport_la_LDFLAGS = $(libkdevcppcommon_la_LDFLAGS) $(KDE_PLUGIN)
+libkdevcppsupport_la_LIBADD  = libkdevcppcommon.la
+libkdevcppsupport_la_SOURCES = dummy.cpp
+
+dummy.cpp:
+	echo > dummy.cpp
+
 METASOURCES = AUTO
 
 servicedir = $(kde_servicesdir)
@@ -37,6 +46,6 @@
 bin_PROGRAMS = r++
 r___LDFLAGS = $(all_libraries)
 r___SOURCES = main.cpp
-r___DEPENDENCIES = libkdevcppsupport.la
-r___LDADD = libkdevcppsupport.la $(top_builddir)/lib/catalog/libkdevcatalog.la $(top_builddir)/lib/cppparser/libkdevcppparser.la
+r___DEPENDENCIES = libkdevcppcommon.la
+r___LDADD = libkdevcppcommon.la $(top_builddir)/lib/catalog/libkdevcatalog.la $(top_builddir)/lib/cppparser/libkdevcppparser.la
 noinst_HEADERS = qtdesignerintegration.h implementationwidget.h
diff -uNr kdevelop-3.1.1/languages/java/Makefile.am kdevelop-3.1.1-new/languages/java/Makefile.am
--- kdevelop-3.1.1/languages/java/Makefile.am	Thu Aug  5 15:28:06 2004
+++ kdevelop-3.1.1-new/languages/java/Makefile.am	Fri Oct 22 07:43:50 2004
@@ -6,10 +6,11 @@
 SUBDIRS = file_templates app_templates debugger doc
 
 INCLUDES = -I$(top_srcdir)/lib/antlr -I$(top_srcdir)/lib/interfaces -I$(top_srcdir)/lib/external_interfaces -I$(top_srcdir)/lib/util -I$(top_srcdir)/lib/catalog -I$(top_srcdir)/lib/compat -I$(top_builddir)/bdb/build_unix -I$(top_srcdir)/bdb/include $(all_includes)
+LDFLAGS = -L$(top_builddir)/bdb/build_unix
 
 kde_module_LTLIBRARIES = libkdevjavasupport.la
 libkdevjavasupport_la_LDFLAGS = $(all_libraries) $(KDE_PLUGIN)
-libkdevjavasupport_la_LIBADD = $(top_builddir)/bdb/build_unix/libdb.a $(top_builddir)/lib/libkdevelop.la $(top_builddir)/lib/antlr/src/libantlr.la $(top_builddir)/lib/catalog/libkdevcatalog.la
+libkdevjavasupport_la_LIBADD = -ldb $(top_builddir)/lib/libkdevelop.la $(top_builddir)/lib/antlr/src/libantlr.la $(top_builddir)/lib/catalog/libkdevcatalog.la
 
 libkdevjavasupport_la_SOURCES = JavaLexer.cpp KDevJavaSupportIface.cpp javasupportfactory.cpp \
 JavaRecognizer.cpp backgroundparser.cpp configproblemreporter.ui  javasupportpart.cpp \
diff -uNr kdevelop-3.1.1/languages/pascal/Makefile.am kdevelop-3.1.1-new/languages/pascal/Makefile.am
--- kdevelop-3.1.1/languages/pascal/Makefile.am	Thu Aug  5 15:28:08 2004
+++ kdevelop-3.1.1-new/languages/pascal/Makefile.am	Fri Oct 22 07:43:50 2004
@@ -1,11 +1,13 @@
 KDE_CXXFLAGS = $(USE_EXCEPTIONS)
 
 INCLUDES = -I$(top_srcdir)/lib/antlr -I$(top_srcdir)/lib/catalog -I$(top_srcdir)/lib/interfaces -I$(top_srcdir)/lib/util -I$(top_srcdir)/lib/external_interfaces -I$(top_builddir)/bdb/build_unix -I$(top_srcdir)/bdb/include $(all_includes)
+LDFLAGS = -L$(top_builddir)/bdb/build_unix
+
 SUBDIRS = file_templates app_templates compiler doc
 
 kde_module_LTLIBRARIES = libkdevpascalsupport.la
-libkdevpascalsupport_la_LDFLAGS = $(LEXLIB) $(all_libraries) $(KDE_PLUGIN)
-libkdevpascalsupport_la_LIBADD = $(top_builddir)/bdb/build_unix/libdb.a $(top_builddir)/lib/libkdevelop.la $(top_builddir)/lib/antlr/src/libantlr.la $(top_builddir)/lib/catalog/libkdevcatalog.la
+libkdevpascalsupport_la_LDFLAGS = $(all_libraries) $(KDE_PLUGIN)
+libkdevpascalsupport_la_LIBADD = -ldb $(top_builddir)/lib/libkdevelop.la $(top_builddir)/lib/antlr/src/libantlr.la $(top_builddir)/lib/catalog/libkdevcatalog.la
 
 libkdevpascalsupport_la_SOURCES = pascalsupport_part.cpp PascalLexer.cpp PascalParser.cpp PascalStoreWalker.cpp backgroundparser.cpp configproblemreporter.ui problemreporter.cpp
 
diff -uNr kdevelop-3.1.1/languages/pascal/PascalLexer.cpp kdevelop-3.1.1-new/languages/pascal/PascalLexer.cpp
--- kdevelop-3.1.1/languages/pascal/PascalLexer.cpp	Sun Nov 30 04:51:25 2003
+++ kdevelop-3.1.1-new/languages/pascal/PascalLexer.cpp	Fri Oct 22 07:43:50 2004
@@ -1,4 +1,7 @@
 /* $ANTLR 2.7.2: "pascal.g" -> "PascalLexer.cpp"$ */
+
+#include "PascalTokenTypes.hpp"
+
 #include "PascalLexer.hpp"
 #include <antlr/CharBuffer.hpp>
 #include <antlr/TokenStreamException.hpp>
diff -uNr kdevelop-3.1.1/languages/pascal/PascalParser.cpp kdevelop-3.1.1-new/languages/pascal/PascalParser.cpp
--- kdevelop-3.1.1/languages/pascal/PascalParser.cpp	Sun Nov 30 04:51:25 2003
+++ kdevelop-3.1.1-new/languages/pascal/PascalParser.cpp	Fri Oct 22 07:43:50 2004
@@ -1,4 +1,7 @@
 /* $ANTLR 2.7.2: "pascal.g" -> "PascalParser.cpp"$ */
+
+#include "PascalTokenTypes.hpp"
+
 #include "PascalParser.hpp"
 #include <antlr/NoViableAltException.hpp>
 #include <antlr/SemanticException.hpp>
diff -uNr kdevelop-3.1.1/languages/pascal/PascalStoreWalker.cpp kdevelop-3.1.1-new/languages/pascal/PascalStoreWalker.cpp
--- kdevelop-3.1.1/languages/pascal/PascalStoreWalker.cpp	Sun Nov 30 04:51:25 2003
+++ kdevelop-3.1.1-new/languages/pascal/PascalStoreWalker.cpp	Fri Oct 22 07:43:50 2004
@@ -1,4 +1,5 @@
 /* $ANTLR 2.7.2: "pascal.tree.g" -> "PascalStoreWalker.cpp"$ */
+#include "PascalStoreWalkerTokenTypes.hpp"
 #include "PascalStoreWalker.hpp"
 #include <antlr/Token.hpp>
 #include <antlr/AST.hpp>
diff -uNr kdevelop-3.1.1/languages/pascal/backgroundparser.cpp kdevelop-3.1.1-new/languages/pascal/backgroundparser.cpp
--- kdevelop-3.1.1/languages/pascal/backgroundparser.cpp	Sun Nov 30 04:51:25 2003
+++ kdevelop-3.1.1-new/languages/pascal/backgroundparser.cpp	Fri Oct 22 07:43:50 2004
@@ -9,6 +9,8 @@
  *                                                                         *
  ***************************************************************************/
 
+#include "PascalTokenTypes.hpp"
+
 #include <strstream>
 
 #include "backgroundparser.h"
diff -uNr kdevelop-3.1.1/languages/pascal/pascalsupport_part.cpp kdevelop-3.1.1-new/languages/pascal/pascalsupport_part.cpp
--- kdevelop-3.1.1/languages/pascal/pascalsupport_part.cpp	Tue Jun 22 13:28:07 2004
+++ kdevelop-3.1.1-new/languages/pascal/pascalsupport_part.cpp	Fri Oct 22 07:43:50 2004
@@ -7,6 +7,10 @@
  *   the Free Software Foundation; either version 2 of the License, or     *
  *   (at your option) any later version.                                   *
  ***************************************************************************/
+
+#include "PascalTokenTypes.hpp"
+#include "PascalStoreWalkerTokenTypes.hpp"
+
 #include <fstream>
 
 #include <qdir.h>
diff -uNr kdevelop-3.1.1/lib/catalog/Makefile.am kdevelop-3.1.1-new/lib/catalog/Makefile.am
--- kdevelop-3.1.1/lib/catalog/Makefile.am	Tue Jun 22 13:28:08 2004
+++ kdevelop-3.1.1-new/lib/catalog/Makefile.am	Fri Oct 22 07:43:50 2004
@@ -1,6 +1,8 @@
 INCLUDES = $(all_includes)
 lib_LTLIBRARIES = libkdevcatalog.la
 libkdevcatalog_la_SOURCES = tag.cpp
+libkdevcatalog_la_LIBADD  = $(LIB_QT) $(LIB_X11)
+libkdevcatalog_la_LDFLAGS = $(all_libraries) -no-undefined
 
 kdevcatalogincludedir = $(includedir)/kdevelop/catalog
 kdevcataloginclude_HEADERS = gcatalog.h gcatalog.tcc catalog.h tag.h
diff -uNr kdevelop-3.1.1/lib/cppparser/Makefile.am kdevelop-3.1.1-new/lib/cppparser/Makefile.am
--- kdevelop-3.1.1/lib/cppparser/Makefile.am	Sat Jan 17 07:57:54 2004
+++ kdevelop-3.1.1-new/lib/cppparser/Makefile.am	Fri Oct 22 07:43:50 2004
@@ -1,7 +1,8 @@
 INCLUDES = $(all_includes)
 AM_CXXFLAGS = -DKDEVELOP_BGPARSER
 lib_LTLIBRARIES = libkdevcppparser.la
-libkdevcppparser_la_LDFLAGS = $(all_libraries)
+libkdevcppparser_la_LIBADD  = $(LIB_QT) $(LIB_KDECORE)
+libkdevcppparser_la_LDFLAGS = $(all_libraries) $(KDE_RPATH) -no-undefined
 libkdevcppparser_la_SOURCES = ast.cpp driver.cpp errors.cpp lexer.cpp lookup.cpp parser.cpp tree_parser.cpp
 
 kdevcppparserincludedir = $(includedir)/kdevelop/cppparser
diff -uNr kdevelop-3.1.1/parts/doctreeview/htdigindex.cpp kdevelop-3.1.1-new/parts/doctreeview/htdigindex.cpp
--- kdevelop-3.1.1/parts/doctreeview/htdigindex.cpp	Tue Jun 22 13:28:10 2004
+++ kdevelop-3.1.1-new/parts/doctreeview/htdigindex.cpp	Fri Oct 22 07:43:50 2004
@@ -597,7 +597,7 @@
     KDialogBase::done( 0 );
 }
 
-int main(int argc, char *argv[])
+main(int argc, char *argv[])
 {
 #if 0
     static KCmdLineOptions options[] = {
diff -uNr kdevelop-3.1.1/src/Makefile.am kdevelop-3.1.1-new/src/Makefile.am
--- kdevelop-3.1.1/src/Makefile.am	Sat Jul 17 15:14:15 2004
+++ kdevelop-3.1.1-new/src/Makefile.am	Fri Oct 22 07:43:50 2004
@@ -1,16 +1,18 @@
 INCLUDES = -I$(top_srcdir)/lib -I$(top_srcdir)/lib/external_interfaces -I$(top_srcdir)/lib/interfaces $(KDEVMDI_INCLUDES) -I$(top_srcdir)/lib/util -I$(top_srcdir)/lib/widgets -I$(top_srcdir)/lib/structure $(all_includes)
 
-bin_PROGRAMS = kdevelop kdevassistant
+bin_PROGRAMS =
+lib_LTLIBRARIES =
+kdeinit_LTLIBRARIES = kdevelop.la kdevassistant.la
 
-kdevelop_SOURCES = core.cpp partcontroller.cpp projectmanager.cpp \
+kdevelop_la_SOURCES = core.cpp partcontroller.cpp projectmanager.cpp \
 	plugincontroller.cpp api.cpp partselectwidget.cpp settingswidget.ui toplevel.cpp \
 	statusbar.cpp editorproxy.cpp debugger.cpp documentationpart.cpp projectsession.cpp \
 	mainwindowshare.cpp generalinfowidgetbase.ui generalinfowidget.cpp languageselectwidget.cpp \
 	newmainwindow.cpp mimewarningdialog.ui main.cpp splashscreen.cpp
 
-kdevelop_METASOURCES = AUTO
-kdevelop_LDFLAGS = $(all_libraries) $(KDE_RPATH)
-kdevelop_LDADD =  $(top_builddir)/lib/widgets/libwidgets.la $(top_builddir)/lib/libkdevelop.la $(LIB_KDEVMDI) $(LIB_KDEUI) $(LIB_KPARTS) $(LIB_KHTML) -lktexteditor
+kdevelop_la_METASOURCES = AUTO
+kdevelop_la_LDFLAGS     = $(all_libraries) $(KDE_RPATH) $(KDE_PLUGIN) -module
+kdevelop_la_LIBADD      = $(top_builddir)/lib/widgets/libwidgets.la $(top_builddir)/lib/libkdevelop.la $(LIB_KDEVMDI) $(LIB_KDEUI) $(LIB_KPARTS) $(LIB_KHTML) -lktexteditor
 
 rcdir = $(kde_datadir)/kdevelop
 rc_DATA = kdevelopui.rc eventsrc
@@ -21,13 +23,14 @@
 
 SUBDIRS = profiles
 noinst_HEADERS = languageselectwidget.h newmainwindow.h
-kdevassistant_LDFLAGS = $(all_libraries) $(KDE_RPATH)
-kdevassistant_SOURCES = api.cpp core.cpp debugger.cpp documentationpart.cpp \
+
+kdevassistant_la_LDFLAGS = $(all_libraries) $(KDE_RPATH) $(KDE_PLUGIN) -module
+kdevassistant_la_SOURCES = api.cpp core.cpp debugger.cpp documentationpart.cpp \
 		editorproxy.cpp generalinfowidget.cpp languageselectwidget.cpp mainwindowshare.cpp \
 		newmainwindow.cpp partcontroller.cpp partselectwidget.cpp plugincontroller.cpp \
 		projectmanager.cpp projectsession.cpp settingswidget.ui statusbar.cpp toplevel.cpp \
 		generalinfowidgetbase.ui main_assistant.cpp mimewarningdialog.ui splashscreen.cpp
-kdevassistant_LDADD = $(top_builddir)/lib/widgets/libwidgets.la \
+kdevassistant_la_LIBADD = $(top_builddir)/lib/widgets/libwidgets.la \
 	$(top_builddir)/lib/libkdevelop.la $(LIB_KDEVMDI) -lktexteditor \
 	$(LIB_KHTML) $(LIB_KPARTS) $(LIB_KDEUI)
 
diff -uNr kdevelop-3.1.1/src/main.cpp kdevelop-3.1.1-new/src/main.cpp
--- kdevelop-3.1.1/src/main.cpp	Sat Jul 17 15:14:15 2004
+++ kdevelop-3.1.1-new/src/main.cpp	Fri Oct 22 07:43:50 2004
@@ -25,7 +25,7 @@
     { 0,0,0 }
 };
 
-int main(int argc, char *argv[])
+extern "C" int kdemain(int argc, char *argv[])
 {
   static const char description[] = I18N_NOOP("The KDevelop Integrated Development Environment");
   KAboutData aboutData("kdevelop", I18N_NOOP("KDevelop"),
diff -uNr kdevelop-3.1.1/src/main_assistant.cpp kdevelop-3.1.1-new/src/main_assistant.cpp
--- kdevelop-3.1.1/src/main_assistant.cpp	Sat Jul 17 15:14:15 2004
+++ kdevelop-3.1.1-new/src/main_assistant.cpp	Fri Oct 22 07:44:20 2004
@@ -25,7 +25,7 @@
     { 0,0,0 }
 };
 
-int main(int argc, char *argv[])
+extern "C" int kdemain(int argc, char *argv[])
 {
   static const char description[] = I18N_NOOP("The KDevelop Integrated Development Environment:\nassistant and documentation viewer");
   KAboutData aboutData("kdevassistant", I18N_NOOP("KDevelop Assistant"),
