diff -Naur mod_perl-2.0.2-orig/Makefile.PL mod_perl-2.0.2/Makefile.PL
--- mod_perl-2.0.2-orig/Makefile.PL	2005-12-14 13:05:10.000000000 +0100
+++ mod_perl-2.0.2/Makefile.PL	2005-12-14 13:05:50.000000000 +0100
@@ -78,7 +78,6 @@
     macro         => {
         MODPERL_SRC           => $code->path,
         MODPERL_MAKEFILE      => basename($build->default_file('makefile')),
-        PERL                  => $build->perl_config('perlpath'),
         MOD_INSTALL           => ModPerl::BuildMM::mod_install(),
         MODPERL_AP_INCLUDEDIR =>
             $build->ap_destdir($build->install_headers_dir),
@@ -732,9 +732,6 @@
     # ModPerl::MM::add_dep(\$string, pure_install => 'aprext_install')
     #    if BUILD_APREXT;
 
-    ModPerl::MM::add_dep(\$string, pure_install => 'nuke_Apache__test')
-          if $apache_test_install;
-
     ModPerl::MM::add_dep_after(\$string, qw(install doc_install modperl_banner));
 
     # glue_pods target must come first
