diff -urN subversion-1.0.1.orig/ac-helpers/ltmain.sh subversion-1.0.1/ac-helpers/ltmain.sh
--- subversion-1.0.1.orig/ac-helpers/ltmain.sh	Sat Dec 20 12:02:12 2003
+++ subversion-1.0.1/ac-helpers/ltmain.sh	Thu Dec 25 20:21:10 2003
@@ -1395,6 +1395,8 @@
 	lib=
 	found=no
 	case $deplib in
+	-ldb-4.1*) continue ;;
+	libdb-4.1*) continue ;;
 	-l*)
 	  if test $linkmode = oldlib && test $linkmode = obj; then
 	    $echo "$modename: warning: \`-l' is ignored for archives/objects: $deplib" 1>&2
@@ -3649,7 +3651,7 @@
 # if CDPATH is set.
 if test \"\${CDPATH+set}\" = set; then CDPATH=:; export CDPATH; fi
 
-relink_command=\"$relink_command\"
+relink_command=""
 
 # This environment variable determines our operation mode.
 if test \"\$libtool_install_magic\" = \"$magic\"; then
@@ -4020,7 +4022,7 @@
 libdir='$install_libdir'"
 	  if test "$installed" = no && test $need_relink = yes; then
 	    $echo >> $output "\
-relink_command=\"$relink_command\""
+relink_command=\"\""
 	  fi
 	done
       fi

diff -Naur subversion-1.1.0-rc2/subversion/bindings/swig/perl/native/Makefile.PL.in subversion-1.1.0-rc2-patched/subversion/bindings/swig/perl/native/Makefile.PL.in
--- subversion-1.1.0-rc2/subversion/bindings/swig/perl/native/Makefile.PL.in	2004-08-09 23:30:24.000000000 +0200
+++ subversion-1.1.0-rc2-patched/subversion/bindings/swig/perl/native/Makefile.PL.in	2004-08-11 10:44:28.000000000 +0200
@@ -23,7 +23,9 @@
                map {"$svnlib_builddir/libsvn_$_/.libs"} (@modules, qw/diff subr
                                                                       ra_local
                                                                       ra_svn
-                                                                      ra_dav/));
+                                                                      ra_dav
+                                                                      fs_base
+                                                                      fs_fs/));
 my @ldmodules = map {"-lsvn_$_-1"} (@modules, qw/diff subr/);
 
 my $apr_shlib_path_var = '@SVN_APR_SHLIB_PATH_VAR@';
