diff -urN subversion-1.0.1.orig/ac-helpers/ltmain.sh subversion-1.0.1/ac-helpers/ltmain.sh
--- subversion-1.0.1.orig/ac-helpers/ltmain.sh	Sat Dec 20 12:02:12 2003
+++ subversion-1.0.1/ac-helpers/ltmain.sh	Thu Dec 25 20:21:10 2003
@@ -1395,6 +1395,8 @@
 	lib=
 	found=no
 	case $deplib in
+	-ldb-4.1*) continue ;;
+	libdb-4.1*) continue ;;
 	-l*)
 	  if test $linkmode = oldlib && test $linkmode = obj; then
 	    $echo "$modename: warning: \`-l' is ignored for archives/objects: $deplib" 1>&2
@@ -3649,7 +3651,7 @@
 # if CDPATH is set.
 if test \"\${CDPATH+set}\" = set; then CDPATH=:; export CDPATH; fi
 
-relink_command=\"$relink_command\"
+relink_command=""
 
 # This environment variable determines our operation mode.
 if test \"\$libtool_install_magic\" = \"$magic\"; then
@@ -4020,7 +4022,7 @@
 libdir='$install_libdir'"
 	  if test "$installed" = no && test $need_relink = yes; then
 	    $echo >> $output "\
-relink_command=\"$relink_command\""
+relink_command=\"\""
 	  fi
 	done
       fi

diff -urN subversion-1.1.0-rc1.orig/build.conf subversion-1.1.0-rc1/build.conf
--- subversion-1.1.0-rc1.orig/build.conf	Tue Jul 20 12:00:00 2004
+++ subversion-1.1.0-rc1/build.conf	Tue Jul 25 20:00:00 2004
@@ -277,6 +277,7 @@
 #
 
 [swig_client]
+undefined-lib-symbols = yes
 type = swig
 path = subversion/bindings/swig
 sources = svn_client.i
@@ -285,6 +286,7 @@
 description = Subversion client library bindings
 
 [swig_delta]
+undefined-lib-symbols = yes
 type = swig
 path = subversion/bindings/swig
 sources = svn_delta.i
@@ -293,6 +295,7 @@
 description = Subversion delta library bindings
 
 [swig_fs]
+undefined-lib-symbols = yes
 type = swig
 path = subversion/bindings/swig
 sources = svn_fs.i
@@ -301,6 +304,7 @@
 description = Subversion FS library bindings
 
 [swig_ra]
+undefined-lib-symbols = yes
 type = swig
 path = subversion/bindings/swig
 sources = svn_ra.i
@@ -309,6 +313,7 @@
 description = Subversion RA library bindings
 
 [swig_repos]
+undefined-lib-symbols = yes
 type = swig
 path = subversion/bindings/swig
 sources = svn_repos.i
@@ -317,6 +322,7 @@
 description = Subversion repository library bindings
 
 [swig_wc]
+undefined-lib-symbols = yes
 type = swig
 path = subversion/bindings/swig
 sources = svn_wc.i
@@ -325,6 +331,7 @@
 description = Subversion WC library bindings
 
 [swig_core]
+undefined-lib-symbols = yes
 type = swig
 path = subversion/bindings/swig
 sources = core.i
@@ -341,6 +348,7 @@
 
 # SWIG utility library for Python modules
 [libsvn_swig_py]
+undefined-lib-symbols = yes
 type = swig_lib
 lang = python
 path = subversion/bindings/swig/python/libsvn_swig_py
