diff -ruN VFlib3-3.6.13/doc/Makefile.in VFlib3-3.6.13.new/doc/Makefile.in
--- VFlib3-3.6.13/doc/Makefile.in	2001-10-11 22:10:53.000000000 -0400
+++ VFlib3-3.6.13.new/doc/Makefile.in	2003-05-16 10:50:30.000000000 -0400
@@ -40,7 +40,7 @@
 	   install-info $(DOC_INFO) \
 	     --info-dir=$(infodir) \
 	     --section='Miscellaneous' \
-	     --item="* VFlib 3.6: (VFlib-36.info).		A font library VFlib."; \
+	     --menuentry"* VFlib 3.6: (VFlib-36.info).		A font library VFlib."; \
 	 else \
 	   echo "NOTICE: No 'install-info' program!"; \
 	   echo "        Edit $(infodir)/dir file by yourself."; \
