diff -ruN yahoo-transport-2/Makefile yahoo-transport-2-patched/Makefile
--- yahoo-transport-2/Makefile	Sun Sep 28 20:40:44 2003
+++ yahoo-transport-2-patched/Makefile	Tue Dec  9 13:35:47 2003
@@ -1,10 +1,13 @@
 
+
 #  $Id$
 
-include ../platform-settings
+include ./platform-settings
 
-CFLAGS:=$(CFLAGS) -I../jabberd  `glib-config --cflags`
-LIBS:=$(LIBS) `glib-config --libs glib`
+CFLAGS:=$(CFLAGS) -I@FINKPREFIX@/include/glib-1.2 -I@FINKPREFIX@/lib/glib/include 
+#-I../jabberd  `glib-config --cflags`
+LIBS:=$(LIBS)
+# `glib-config --libs glib`
 
 YAHOO_OBJECTS=yahoo-transport.o yahoo-session.o yahoo-phandler.o yahoo.o \
 	crypt.o gaim-sha.o yahoo-presence.o yahoo-server.o md5.o yahoo-message.o \
diff -ruN yahoo-transport-2/platform-settings yahoo-transport-2-patched/platform-settings
--- yahoo-transport-2/platform-settings	Wed Dec 31 19:00:00 1969
+++ yahoo-transport-2-patched/platform-settings	Tue Dec  9 12:51:21 2003
@@ -0,0 +1,13 @@
+#!/bin/sh
+CC=cc
+CFLAGS= -g -Wall -I. -I.. -I@FINKPREFIX@/include/jabber -I@FINKPREFIX@/include/ -DWITH_IPV6 -I@FINKPREFIX@/include -fPIC -DTMZONE
+CCFLAGS= -g -Wall -I. -I.. -DWITH_IPV6 -I/sw/include -fPIC -DTMZONE
+MCFLAGS=-bundle -undefined suppress -flat_namespace
+LDFLAGS=-L@FINKPREFIX@/lib -L@FINKPREFIX@/lib
+LIBS= -lpth -ldl
+SLIBS= @FINKPREFIX@/lib/libpth.a
+XLDFLAGS=
+PSUBDIR=
+PLINK=
+JHOME=@FINKPREFIX@/lib/jabber
+__CYGWIN__=
diff -ruN yahoo-transport-2/yahoo-transport.h yahoo-transport-2-patched/yahoo-transport.h
--- yahoo-transport-2/yahoo-transport.h	Wed Oct  1 08:42:54 2003
+++ yahoo-transport-2-patched/yahoo-transport.h	Tue Dec  9 13:31:05 2003
@@ -36,7 +36,7 @@
  * --------------------------------------------------------------------------*/
 
 
-#include "jabberd.h"
+#include "@FINKPREFIX@/include/jabber/jabberd.h"
 #include <sys/utsname.h>
 #include <stdio.h>
 #include <unistd.h>
