diff -uNr mono-1.1.8.2/data/config.in mono-1.1.8.2-new/data/config.in
--- mono-1.1.8.2/data/config.in	2005-06-13 10:00:53.000000000 -0400
+++ mono-1.1.8.2-new/data/config.in	2005-07-17 19:06:34.000000000 -0400
@@ -9,6 +9,10 @@
 	<dllmap dll="db2cli" target="libdb2_36@libsuffix@"/>
 	<dllmap dll="msvcrt" target="@LIBC@"/>
 	<dllmap dll="MonoPosixHelper" target="libMonoPosixHelper@libsuffix@"/>
+	<dllmap dll="libgtk-win32-2.0-0.dll" target="libgtk-x11-2.0@libsuffix@"/>
+	<dllmap dll="glib-2.0" target="libglib-2.0@libsuffix@"/>
+	<dllmap dll="gnomevfs-2" target="libgnomevfs-2@libsuffix@"/>
+	<dllmap dll="gtksourceview-1.0" target="libgtksourceview-1.0@libsuffix@"/>
 	<dllmap dll="sqlite" target="@SQLITE@"/>
 	<dllmap dll="sqlite3" target="@SQLITE3@"/>
 	<dllmap dll="libX11" target="@X11@"/>
diff -uNr mono-1.1.8.2/mono/metadata/Makefile.in mono-1.1.8.2-new/mono/metadata/Makefile.in
--- mono-1.1.8.2/mono/metadata/Makefile.in	2005-06-30 19:35:03.000000000 -0400
+++ mono-1.1.8.2-new/mono/metadata/Makefile.in	2005-07-17 19:06:34.000000000 -0400
@@ -68,7 +68,7 @@
 am_libmonoruntime_static_la_OBJECTS = $(am__objects_1)
 libmonoruntime_static_la_OBJECTS =  \
 	$(am_libmonoruntime_static_la_OBJECTS)
-libmonoruntime_la_LIBADD =
+libmonoruntime_la_LIBADD = $(ICU_LIBS)
 am_libmonoruntime_la_OBJECTS = reflection.lo object.lo icall.lo \
 	decimal.lo boehm-gc.lo null-gc.lo gc.lo marshal.lo monitor.lo \
 	threads.lo threadpool.lo file-io.lo socket-io.lo exception.lo \
diff -uNr mono-1.1.8.2/mono/metadata/loader.c mono-1.1.8.2-new/mono/metadata/loader.c
--- mono-1.1.8.2/mono/metadata/loader.c	2005-06-13 10:11:20.000000000 -0400
+++ mono-1.1.8.2-new/mono/metadata/loader.c	2005-07-17 19:06:34.000000000 -0400
@@ -736,6 +736,19 @@
 		}
 
 		if (!gmodule) {
+			full_name = g_module_build_path ("@PREFIX@/lib", file_name);
+			mono_trace (G_LOG_LEVEL_INFO, MONO_TRACE_DLLIMPORT,
+					"DllImport loading library: '@PREFIX@/lib/%s'.", full_name);
+			gmodule = g_module_open (full_name, G_MODULE_BIND_LAZY);
+			if (!gmodule) {
+				mono_trace (G_LOG_LEVEL_INFO, MONO_TRACE_DLLIMPORT,
+						"DllImport error loading library '%s'.",
+						g_module_error ());
+			}
+			g_free (full_name);
+		}
+
+		if (!gmodule) {
 			full_name = g_module_build_path (".", file_name);
 			mono_trace (G_LOG_LEVEL_INFO, MONO_TRACE_DLLIMPORT,
 					"DllImport loading library: '%s'.", full_name);
