--- apt-0.6.41/apt-inst/makefile	2005-08-01 12:05:41.000000000 -0400
+++ apt-0.6.41-new/apt-inst/makefile	2005-09-09 12:30:35.000000000 -0400
@@ -12,9 +12,9 @@
 # The library name
 LIBRARY=apt-inst
 LIBEXT=$(GLIBC_VER)$(LIBSTDCPP_VER)
-MAJOR=1.1
+MAJOR=0.6.40
 MINOR=0
-SLIBS=$(PTHREADLIB) -lapt-pkg
+SLIBS=$(PTHREADLIB) -lapt-pkg $(INTLLIBS)
 APT_DOMAIN:=libapt-inst$(MAJOR)
 
 # Source code for the contributed non-core things
--- apt-0.6.41/apt-pkg/acquire-item.cc	2005-08-23 04:13:13.000000000 -0400
+++ apt-0.6.41-new/apt-pkg/acquire-item.cc	2005-09-09 12:30:35.000000000 -0400
@@ -708,12 +708,14 @@
 // ---------------------------------------------------------------------
 /* This just sets up the initial fetch environment and queues the first
    possibilitiy */
+
+// FIXME: when fink has a way to trust stuff, set Trusted(true) back to false  ;)
 pkgAcqArchive::pkgAcqArchive(pkgAcquire *Owner,pkgSourceList *Sources,
 			     pkgRecords *Recs,pkgCache::VerIterator const &Version,
 			     string &StoreFilename) :
                Item(Owner), Version(Version), Sources(Sources), Recs(Recs), 
                StoreFilename(StoreFilename), Vf(Version.FileList()), 
-	       Trusted(false)
+	       Trusted(true)
 {
    Retries = _config->FindI("Acquire::Retries",0);
 
--- apt-0.6.41/apt-pkg/contrib/cdromutl.cc	2005-08-01 12:05:41.000000000 -0400
+++ apt-0.6.41-new/apt-pkg/contrib/cdromutl.cc	2005-09-09 12:30:35.000000000 -0400
@@ -23,13 +23,13 @@
     
 #include <sys/wait.h>
 #include <sys/errno.h>
-#include <sys/statvfs.h>
+#include <sys/mount.h>
 #include <dirent.h>
 #include <fcntl.h>
 #include <sys/stat.h>
 #include <unistd.h>
 #include <stdio.h>
-									/*}}}*/
+
 
 // IsMounted - Returns true if the mount point is mounted		/*{{{*/
 // ---------------------------------------------------------------------
@@ -185,8 +185,8 @@
    // Some stats from the fsys
    if (_config->FindB("Debug::identcdrom",false) == false)
    {
-      struct statvfs Buf;
-      if (statvfs(CD.c_str(),&Buf) != 0)
+      struct statfs Buf;
+      if (statfs(CD.c_str(),&Buf) != 0)
 	 return _error->Errno("statfs",_("Failed to stat the cdrom"));
       
       // We use a kilobyte block size to advoid overflow
--- apt-0.6.41/apt-pkg/deb/debindexfile.cc	2005-08-01 12:05:46.000000000 -0400
+++ apt-0.6.41-new/apt-pkg/deb/debindexfile.cc	2005-09-09 12:30:35.000000000 -0400
@@ -444,3 +444,9 @@
 }
 
 									/*}}}*/
+void init_deb2()
+{
+  (void)_apt_Src;
+  (void)_apt_Pkg;
+  (void)_apt_Status;
+}
--- apt-0.6.41/apt-pkg/deb/debsystem.cc	2005-08-23 04:13:13.000000000 -0400
+++ apt-0.6.41-new/apt-pkg/deb/debsystem.cc	2005-09-09 12:30:35.000000000 -0400
@@ -27,6 +27,111 @@
 #include <dirent.h>
 #include <errno.h>
 									/*}}}*/
+/* FINK LOCAL begin */
+#include <sys/utsname.h>
+#include <CoreFoundation/CoreFoundation.h>
+#include <fstream>
+#include <sys/stat.h>
+
+extern void init_deb2();
+extern void init_deb3();
+
+#define FINKSTATUSFILE "/tmp/finkaptstatus"
+
+struct versionrevision {
+  unsigned long epoch;
+  const char *version;
+  const char *revision;
+};  
+
+struct versionrevision darwin_version = {0,NULL,NULL};
+struct versionrevision macosx_version = {0,NULL,NULL};
+
+using namespace std;
+
+static void finkinit()
+{
+  Boolean status;
+  SInt32 errorCode;
+  CFURLRef fileURL = NULL;
+  CFDataRef resourceData = NULL;
+  CFPropertyListRef propertyList = NULL;
+  CFStringRef string;
+  static char buffer[256];	// This is static, to ensure the buffer stays around
+
+  static struct utsname ver;	// This is static, to ensure the buffer stays around
+  
+  /* Determine system version */
+  /* TODO - should maybe check if this is really Darwin? */
+  if (!uname(&ver)) {
+    darwin_version.version = ver.release;
+  }
+
+  /* Check whether this is Mac OS X, and which version of it */
+
+  fileURL = CFURLCreateWithFileSystemPath( NULL, 	
+		CFSTR("/System/Library/CoreServices/SystemVersion.plist"),
+		kCFURLPOSIXPathStyle,				
+		false );
+  if (!fileURL)
+    goto BAIL;
+  
+  /* Read the XML */
+  status = CFURLCreateDataAndPropertiesFromResource(
+		NULL,
+		fileURL,
+		&resourceData,
+		NULL,		
+		NULL,
+		&errorCode);
+  if (!status || errorCode != 0)
+    goto BAIL;
+  
+  /* Reconstitute the dictionary using the XML data. */
+  propertyList = CFPropertyListCreateFromXMLData( NULL,
+		resourceData,
+		kCFPropertyListImmutable,
+		&string);
+  if (!propertyList)
+    goto BAIL;
+  
+  /* Try to read the system version from it. */
+  status = CFDictionaryGetValueIfPresent( 
+		(CFDictionaryRef) propertyList,
+		(const void *) CFSTR("ProductVersion"),
+		(const void**) &string);
+  if (!status)
+    goto BAIL;
+  
+  /* Convert into a C string */
+  status = CFStringGetCString( string,
+		buffer,
+		sizeof(buffer),
+		kCFStringEncodingISOLatin1);
+  if (!status)
+    goto BAIL;
+  
+  /* Finally link the buffer into the macosx_version struct. */
+  macosx_version.version = buffer;
+  
+BAIL:
+  // Release all of the CF objects we're responsible for.
+  if (fileURL)
+    CFRelease(fileURL);
+  if (resourceData)
+    CFRelease(resourceData);
+  if (propertyList)
+    CFRelease(propertyList);
+}
+
+void initDebSystem()
+{
+  finkinit();
+  (void)debSys;
+  init_deb2();
+  init_deb3();
+}
+/* FINK LOCAL end */
 
 debSystem debSys;
 
@@ -49,6 +154,8 @@
 debSystem::~debSystem()
 {
    delete StatusFile;
+   delete FinkStatusFile;
+   unlink(FINKSTATUSFILE);
 }
 									/*}}}*/
 // System::Lock - Get the lock						/*{{{*/
@@ -162,8 +269,8 @@
       which is yet to be determined. The functions in pkgcachegen should
       be the only users of these */
    Cnf.CndSet("Dir::State::userstatus","status.user"); // Defunct
-   Cnf.CndSet("Dir::State::status","/var/lib/dpkg/status");
-   Cnf.CndSet("Dir::Bin::dpkg","/usr/bin/dpkg");
+   Cnf.CndSet("Dir::State::status","@PREFIX@/var/lib/dpkg/status");
+   Cnf.CndSet("Dir::Bin::dpkg","@PREFIX@/bin/dpkg");
 
    if (StatusFile) {
      delete StatusFile;
@@ -191,9 +298,9 @@
 signed debSystem::Score(Configuration const &Cnf)
 {
    signed Score = 0;
-   if (FileExists(Cnf.FindFile("Dir::State::status","/var/lib/dpkg/status")) == true)
+   if (FileExists(Cnf.FindFile("Dir::State::status","@PREFIX@/var/lib/dpkg/status")) == true)
        Score += 10;
-   if (FileExists(Cnf.FindFile("Dir::Bin::dpkg","/usr/bin/dpkg")) == true)
+   if (FileExists(Cnf.FindFile("Dir::Bin::dpkg","@PREFIX@/bin/dpkg")) == true)
       Score += 10;
    if (FileExists("/etc/debian_version") == true)
       Score += 10;
@@ -208,6 +315,44 @@
    if (StatusFile == 0)
       StatusFile = new debStatusIndex(_config->FindFile("Dir::State::status"));
    List.push_back(StatusFile);
+/* FINK LOCAL begin */
+
+   if (FinkStatusFile == 0) {
+      struct stat unused_sbuf;
+      int sys_ok=0;
+      unlink(FINKSTATUSFILE);
+      if ( 0 == stat("@PREFIX@/bin/fink-virtual-pkgs",&unused_sbuf)) {
+          if ( 0 == system("@PREFIX@/bin/fink-virtual-pkgs --apt")) sys_ok=1;
+      }    
+      if (stat(FINKSTATUSFILE, &unused_sbuf) || !sys_ok) {
+	  std::ofstream finkstatus(FINKSTATUSFILE);
+      if(macosx_version.version != 0)
+      {
+        finkstatus << "Package: macosx" << endl;
+        finkstatus << "Status: install ok installed" << endl;      
+        finkstatus << "Priority: optional" << endl;
+        finkstatus << "Section: base" << endl;
+        finkstatus << "Maintainer: None" << endl;
+        finkstatus << "Source: macosx" << endl;
+        finkstatus << "Version: " << macosx_version.version << endl;
+        finkstatus << "Description: Pseudo package representing Mac OS X" << endl;
+        finkstatus << " Pseudo package representing Mac OS X" << endl << endl;
+      }
+      finkstatus << "Package: darwin" << endl;
+      finkstatus << "Status: install ok installed" << endl;
+      finkstatus << "Priority: optional" << endl;
+      finkstatus << "Section: base" << endl;
+      finkstatus << "Maintainer: None" << endl;
+      finkstatus << "Source: darwin" << endl;
+      finkstatus << "Version: " << darwin_version.version  << endl;
+      finkstatus << "Description: Pseudo package representing Darwin" << endl;
+      finkstatus << " Pseudo package representing Darwin" << endl << endl;
+      finkstatus.close();
+      }		
+      FinkStatusFile = new debStatusIndex(FINKSTATUSFILE);
+   }
+   List.push_back(FinkStatusFile);
+/* FINK LOCAL end */
    return true;
 }
 									/*}}}*/
@@ -223,6 +368,10 @@
    {
       Found = StatusFile;
       return true;
+   }  else if ((FinkStatusFile != 0) && (FinkStatusFile->FindInCache(*File.Cache()) == File))
+   {
+      Found = FinkStatusFile;
+      return true;
    }
    
    return false;
--- apt-0.6.41/apt-pkg/deb/debsystem.h	2005-08-01 12:05:41.000000000 -0400
+++ apt-0.6.41-new/apt-pkg/deb/debsystem.h	2005-09-09 12:30:35.000000000 -0400
@@ -25,6 +25,7 @@
    bool CheckUpdates();
    
    debStatusIndex *StatusFile;
+   debStatusIndex *FinkStatusFile;
    
    public:
 
--- apt-0.6.41/apt-pkg/deb/debversion.cc	2005-08-01 12:05:41.000000000 -0400
+++ apt-0.6.41-new/apt-pkg/deb/debversion.cc	2005-09-09 12:30:35.000000000 -0400
@@ -24,6 +24,11 @@
 
 debVersioningSystem debVS;
 
+void init_deb3()
+{
+  (void)debVS;
+}
+
 // debVS::debVersioningSystem - Constructor				/*{{{*/
 // ---------------------------------------------------------------------
 /* */
--- apt-0.6.41/apt-pkg/deb/dpkgpm.cc	2005-08-19 04:04:07.000000000 -0400
+++ apt-0.6.41-new/apt-pkg/deb/dpkgpm.cc	2005-09-09 12:30:35.000000000 -0400
@@ -26,6 +26,10 @@
 #include <errno.h>
 #include <stdio.h>
 #include <sstream>
+
+#ifdef __APPLE__
+typedef sig_t sighandler_t;
+#endif
 #include <map>
 
 #include <config.h>
--- apt-0.6.41/apt-pkg/init.cc	2005-08-01 12:05:41.000000000 -0400
+++ apt-0.6.41-new/apt-pkg/init.cc	2005-09-09 12:30:35.000000000 -0400
@@ -15,6 +15,8 @@
 #include <apti18n.h>
 #include <config.h>
 #include <sys/stat.h>
+
+extern void initDebSystem();
 									/*}}}*/
 
 #define Stringfy_(x) # x
@@ -40,7 +42,7 @@
    else
       Cnf.Set("APT::Architecture",COMMON_OS "-" COMMON_CPU);
    Cnf.Set("APT::Build-Essential::", "build-essential");
-   Cnf.Set("Dir","/");
+   Cnf.Set("Dir","@PREFIX@/");
    
    // State   
    Cnf.Set("Dir::State","var/lib/apt/");
@@ -69,7 +71,7 @@
    Cnf.Set("Dir::Etc::main","apt.conf");
    Cnf.Set("Dir::Etc::parts","apt.conf.d");
    Cnf.Set("Dir::Etc::preferences","preferences");
-   Cnf.Set("Dir::Bin::methods","/usr/lib/apt/methods");
+   Cnf.Set("Dir::Bin::methods","@PREFIX@/lib/apt/methods");
 	      
    bool Res = true;
    
@@ -110,6 +112,8 @@
 /* */
 bool pkgInitSystem(Configuration &Cnf,pkgSystem *&Sys)
 {
+   initDebSystem();
+
    Sys = 0;
    string Label = Cnf.Find("Apt::System","");
    if (Label.empty() == false)
--- apt-0.6.41/apt-pkg/makefile	2005-08-19 04:04:07.000000000 -0400
+++ apt-0.6.41-new/apt-pkg/makefile	2005-09-09 12:30:35.000000000 -0400
@@ -13,7 +13,7 @@
 # methods/makefile - FIXME
 LIBRARY=apt-pkg
 LIBEXT=$(GLIBC_VER)$(LIBSTDCPP_VER)
-MAJOR=3.10
+MAJOR=0.6.40
 MINOR=0
 SLIBS=$(PTHREADLIB) $(INTLLIBS)
 APT_DOMAIN:=libapt-pkg$(MAJOR)
--- apt-0.6.41/buildlib/environment.mak.in	2005-08-01 12:05:41.000000000 -0400
+++ apt-0.6.41-new/buildlib/environment.mak.in	2005-09-09 12:30:35.000000000 -0400
@@ -7,14 +7,14 @@
 CC = @CC@
 CPPFLAGS+= @CPPFLAGS@ @DEFS@ -D_REENTRANT -Wall
 CXX = @CXX@
-CXXFLAGS+= @CXXFLAGS@
+CXXFLAGS+= @CXXFLAGS@ -I@PREFIX@/include
 NUM_PROCS = @NUM_PROCS@
 GLIBC_VER = @GLIBC_VER@
 LIBSTDCPP_VER = @LIBSTDCPP_VER@
 
 # Linker stuff
-PICFLAGS+= -fPIC -DPIC
-LFLAGS+= @LDFLAGS@
+PICFLAGS+= -fno-common -DPIC
+LFLAGS+= @LDFLAGS@ -framework CoreFoundation
 LEFLAGS+= 
 SOCKETLIBS:= @SOCKETLIBS@
 AR:=@AR@
@@ -62,11 +62,12 @@
 
 # Shared library things
 HOST_OS = @host_os@
-ifneq ($(words $(filter linux-gnu gnu% %gnu,$(HOST_OS))),0)
-   SONAME_MAGIC=-Wl,-soname -Wl,
-   LFLAGS_SO=
-else
-   # Do not know how to create shared libraries here.
-   ONLYSTATICLIBS = yes
-endif
-	
+#ifneq ($(words $(filter linux-gnu gnu% %gnu,$(HOST_OS))),0)
+#   SONAME_MAGIC=-Wl,-soname -Wl,
+#   LFLAGS_SO=
+#else
+#   # Do not know how to create shared libraries here.
+#   ONLYSTATICLIBS = yes
+#endif
+SONAME_MAGIC=-install_name @PREFIX@/lib/
+LFLAGS_SO=-dynamiclib
--- apt-0.6.41/buildlib/library.mak	2005-08-01 12:05:41.000000000 -0400
+++ apt-0.6.41-new/buildlib/library.mak	2005-09-09 12:30:35.000000000 -0400
@@ -16,11 +16,11 @@
 # See defaults.mak for information about LOCAL
 
 # Some local definitions
-LOCAL := lib$(LIBRARY)$(LIBEXT).so.$(MAJOR).$(MINOR)
+LOCAL := lib$(LIBRARY)$(LIBEXT).$(MAJOR).$(MINOR).dylib
 $(LOCAL)-OBJS := $(addprefix $(OBJ)/,$(addsuffix .opic,$(notdir $(basename $(SOURCE)))))
 $(LOCAL)-DEP := $(addprefix $(DEP)/,$(addsuffix .opic.d,$(notdir $(basename $(SOURCE)))))
 $(LOCAL)-HEADERS := $(addprefix $(INCLUDE)/,$(HEADERS))
-$(LOCAL)-SONAME := lib$(LIBRARY)$(LIBEXT).so.$(MAJOR)
+$(LOCAL)-SONAME := lib$(LIBRARY)$(LIBEXT).$(MAJOR).dylib
 $(LOCAL)-SLIBS := $(SLIBS)
 $(LOCAL)-LIBRARY := $(LIBRARY)
 
@@ -29,7 +29,7 @@
 
 # Install the command hooks
 headers: $($(LOCAL)-HEADERS)
-library: $(LIB)/lib$(LIBRARY).so $(LIB)/lib$(LIBRARY)$(LIBEXT).so.$(MAJOR)
+library: $(LIB)/lib$(LIBRARY).dylib $(LIB)/lib$(LIBRARY)$(LIBEXT).$(MAJOR).dylib
 clean: clean/$(LOCAL)
 veryclean: veryclean/$(LOCAL)
 
@@ -41,21 +41,23 @@
 clean/$(LOCAL):
 	-rm -f $($(@F)-OBJS) $($(@F)-DEP)
 veryclean/$(LOCAL): clean/$(LOCAL)
-	-rm -f $($(@F)-HEADERS) $(LIB)/lib$($(@F)-LIBRARY)*.so*
+	-rm -f $($(@F)-HEADERS) $(LIB)/lib$($(@F)-LIBRARY)*.dylib
 
 # Build rules for the two symlinks
-.PHONY: $(LIB)/lib$(LIBRARY)$(LIBEXT).so.$(MAJOR) $(LIB)/lib$(LIBRARY).so
-$(LIB)/lib$(LIBRARY)$(LIBEXT).so.$(MAJOR): $(LIB)/lib$(LIBRARY)$(LIBEXT).so.$(MAJOR).$(MINOR)
+.PHONY: $(LIB)/lib$(LIBRARY)$(LIBEXT).$(MAJOR).dylib $(LIB)/lib$(LIBRARY).dylib
+$(LIB)/lib$(LIBRARY)$(LIBEXT).$(MAJOR).dylib: $(LIB)/lib$(LIBRARY)$(LIBEXT).$(MAJOR).$(MINOR).dylib
 	ln -sf $(<F) $@
-$(LIB)/lib$(LIBRARY).so: $(LIB)/lib$(LIBRARY)$(LIBEXT).so.$(MAJOR).$(MINOR)
+$(LIB)/lib$(LIBRARY).dylib: $(LIB)/lib$(LIBRARY)$(LIBEXT).$(MAJOR).$(MINOR).dylib
 	ln -sf $(<F) $@
 
 # The binary build rule
-$(LIB)/lib$(LIBRARY)$(LIBEXT).so.$(MAJOR).$(MINOR): $($(LOCAL)-HEADERS) $($(LOCAL)-OBJS)
-	-rm -f $(LIB)/lib$($(@F)-LIBRARY)*.so* 2> /dev/null
+$(LIB)/lib$(LIBRARY)$(LIBEXT).$(MAJOR).$(MINOR).dylib: $($(LOCAL)-HEADERS) $($(LOCAL)-OBJS)
+	-rm -f $(LIB)/lib$($(@F)-LIBRARY)*.dylib 2> /dev/null
 	echo Building shared library $@
 	$(CXX) $(CXXFLAGS) $(LDFLAGS) $(PICFLAGS) $(LFLAGS) $(LFLAGS_SO)\
-	   -o $@ $(SONAME_MAGIC)$($(@F)-SONAME) -shared \
+	   -o $@ $(SONAME_MAGIC)$($(@F)-SONAME) \
+	   -compatibility_version $(MAJOR) \
+	   -current_version $(MAJOR) \
 	   $(filter %.opic,$^) \
 	   $($(@F)-SLIBS) 
 
--- apt-0.6.41/buildlib/ostable	2005-08-01 12:05:41.000000000 -0400
+++ apt-0.6.41-new/buildlib/ostable	2005-09-09 12:30:35.000000000 -0400
@@ -16,6 +16,7 @@
 hp-hpux[^-]*	    hp-ux
 sun-solaris[^-]*    solaris
 [^-]*-openbsd[^-]*  openbsd
+[^-]*-darwin[^-]*   darwin
 
 # Catch all
 .*	unknown
--- apt-0.6.41/cmdline/makefile	2005-08-01 12:05:41.000000000 -0400
+++ apt-0.6.41-new/cmdline/makefile	2005-09-09 12:30:35.000000000 -0400
@@ -7,42 +7,42 @@
 
 # The apt-cache program
 PROGRAM=apt-cache
-SLIBS = -lapt-pkg
+SLIBS = $(INTLLIBS) -lapt-pkg
 LIB_MAKES = apt-pkg/makefile
 SOURCE = apt-cache.cc
 include $(PROGRAM_H)
 
 # The apt-get program
 PROGRAM=apt-get
-SLIBS = -lapt-pkg
+SLIBS = $(INTLLIBS) -lapt-pkg
 LIB_MAKES = apt-pkg/makefile
 SOURCE = apt-get.cc acqprogress.cc
 include $(PROGRAM_H)
 
 # The apt-config program
 PROGRAM=apt-config
-SLIBS = -lapt-pkg
+SLIBS = $(INTLLIBS) -lapt-pkg
 LIB_MAKES = apt-pkg/makefile
 SOURCE = apt-config.cc
 include $(PROGRAM_H)
 
 # The apt-cdrom program
 PROGRAM=apt-cdrom
-SLIBS = -lapt-pkg
+SLIBS = $(INTLLIBS) -lapt-pkg
 LIB_MAKES = apt-pkg/makefile
 SOURCE = apt-cdrom.cc 
 include $(PROGRAM_H)
 
 # The apt-sortpkgs program
 PROGRAM=apt-sortpkgs
-SLIBS = -lapt-pkg
+SLIBS = $(INTLLIBS) -lapt-pkg
 LIB_MAKES = apt-pkg/makefile
 SOURCE = apt-sortpkgs.cc
 include $(PROGRAM_H)
 
 # The apt-extracttemplates program
 PROGRAM=apt-extracttemplates
-SLIBS = -lapt-pkg -lapt-inst
+SLIBS = $(INTLLIBS) -lapt-pkg -lapt-inst
 LIB_MAKES = apt-pkg/makefile
 SOURCE = apt-extracttemplates.cc 
 include $(PROGRAM_H)
--- apt-0.6.41/configure	2005-09-05 17:01:37.000000000 -0400
+++ apt-0.6.41-new/configure	2005-09-09 12:30:36.000000000 -0400
@@ -1296,8 +1296,8 @@
 
 ac_ext=c
 ac_cpp='$CPP $CPPFLAGS'
-ac_compile='$CC -c $CFLAGS $CPPFLAGS conftest.$ac_ext >&5'
-ac_link='$CC -o conftest$ac_exeext $CFLAGS $CPPFLAGS $LDFLAGS conftest.$ac_ext $LIBS >&5'
+ac_compile='$CC -c $CFLAGS $CPPFLAGS -I@PREFIX@/include conftest.$ac_ext >&5'
+ac_link='$CC -o conftest$ac_exeext $CFLAGS $CPPFLAGS -I@PREFIX@/include $LDFLAGS conftest.$ac_ext $LIBS >&5'
 ac_compiler_gnu=$ac_cv_c_compiler_gnu
 
 
@@ -1443,8 +1443,8 @@
 
 ac_ext=c
 ac_cpp='$CPP $CPPFLAGS'
-ac_compile='$CC -c $CFLAGS $CPPFLAGS conftest.$ac_ext >&5'
-ac_link='$CC -o conftest$ac_exeext $CFLAGS $CPPFLAGS $LDFLAGS conftest.$ac_ext $LIBS >&5'
+ac_compile='$CC -c $CFLAGS $CPPFLAGS -I@PREFIX@/include conftest.$ac_ext >&5'
+ac_link='$CC -o conftest$ac_exeext $CFLAGS $CPPFLAGS -I@PREFIX@/include $LDFLAGS conftest.$ac_ext $LIBS >&5'
 ac_compiler_gnu=$ac_cv_c_compiler_gnu
 if test -n "$ac_tool_prefix"; then
   # Extract the first word of "${ac_tool_prefix}gcc", so it can be a program name with args.
@@ -2361,8 +2361,8 @@
 rm -f conftest.err conftest.$ac_objext conftest.$ac_ext
 ac_ext=c
 ac_cpp='$CPP $CPPFLAGS'
-ac_compile='$CC -c $CFLAGS $CPPFLAGS conftest.$ac_ext >&5'
-ac_link='$CC -o conftest$ac_exeext $CFLAGS $CPPFLAGS $LDFLAGS conftest.$ac_ext $LIBS >&5'
+ac_compile='$CC -c $CFLAGS $CPPFLAGS -I@PREFIX@/include conftest.$ac_ext >&5'
+ac_link='$CC -o conftest$ac_exeext $CFLAGS $CPPFLAGS -I@PREFIX@/include $LDFLAGS conftest.$ac_ext $LIBS >&5'
 ac_compiler_gnu=$ac_cv_c_compiler_gnu
 
 
@@ -2439,8 +2439,8 @@
 
 ac_ext=cc
 ac_cpp='$CXXCPP $CPPFLAGS'
-ac_compile='$CXX -c $CXXFLAGS $CPPFLAGS conftest.$ac_ext >&5'
-ac_link='$CXX -o conftest$ac_exeext $CXXFLAGS $CPPFLAGS $LDFLAGS conftest.$ac_ext $LIBS >&5'
+ac_compile='$CXX -c $CXXFLAGS $CPPFLAGS -I@PREFIX@/include conftest.$ac_ext >&5'
+ac_link='$CXX -o conftest$ac_exeext $CXXFLAGS $CPPFLAGS -I@PREFIX@/include $LDFLAGS conftest.$ac_ext $LIBS >&5'
 ac_compiler_gnu=$ac_cv_cxx_compiler_gnu
 if test -n "$ac_tool_prefix"; then
   for ac_prog in $CCC g++ c++ gpp aCC CC cxx cc++ cl FCC KCC RCC xlC_r xlC
@@ -2784,14 +2784,14 @@
 
 ac_ext=c
 ac_cpp='$CPP $CPPFLAGS'
-ac_compile='$CC -c $CFLAGS $CPPFLAGS conftest.$ac_ext >&5'
-ac_link='$CC -o conftest$ac_exeext $CFLAGS $CPPFLAGS $LDFLAGS conftest.$ac_ext $LIBS >&5'
+ac_compile='$CC -c $CFLAGS $CPPFLAGS -I@PREFIX@/include conftest.$ac_ext >&5'
+ac_link='$CC -o conftest$ac_exeext $CFLAGS $CPPFLAGS -I@PREFIX@/include $LDFLAGS conftest.$ac_ext $LIBS >&5'
 ac_compiler_gnu=$ac_cv_c_compiler_gnu
 
 ac_ext=c
 ac_cpp='$CPP $CPPFLAGS'
-ac_compile='$CC -c $CFLAGS $CPPFLAGS conftest.$ac_ext >&5'
-ac_link='$CC -o conftest$ac_exeext $CFLAGS $CPPFLAGS $LDFLAGS conftest.$ac_ext $LIBS >&5'
+ac_compile='$CC -c $CFLAGS $CPPFLAGS -I@PREFIX@/include conftest.$ac_ext >&5'
+ac_link='$CC -o conftest$ac_exeext $CFLAGS $CPPFLAGS -I@PREFIX@/include $LDFLAGS conftest.$ac_ext $LIBS >&5'
 ac_compiler_gnu=$ac_cv_c_compiler_gnu
 echo "$as_me:$LINENO: checking how to run the C preprocessor" >&5
 echo $ECHO_N "checking how to run the C preprocessor... $ECHO_C" >&6
@@ -3018,8 +3018,8 @@
 
 ac_ext=c
 ac_cpp='$CPP $CPPFLAGS'
-ac_compile='$CC -c $CFLAGS $CPPFLAGS conftest.$ac_ext >&5'
-ac_link='$CC -o conftest$ac_exeext $CFLAGS $CPPFLAGS $LDFLAGS conftest.$ac_ext $LIBS >&5'
+ac_compile='$CC -c $CFLAGS $CPPFLAGS -I@PREFIX@/include conftest.$ac_ext >&5'
+ac_link='$CC -o conftest$ac_exeext $CFLAGS $CPPFLAGS -I@PREFIX@/include $LDFLAGS conftest.$ac_ext $LIBS >&5'
 ac_compiler_gnu=$ac_cv_c_compiler_gnu
 
 if test -n "$ac_tool_prefix"; then
@@ -8629,7 +8629,7 @@
 fi
 
 
-                    ac_config_files="$ac_config_files environment.mak:buildlib/environment.mak.in makefile:buildlib/makefile.in"
+                    ac_config_files="$ac_config_files environment.mak:buildlib/environment.mak.in makefile.wrap:buildlib/makefile.in"
           ac_config_commands="$ac_config_commands default"
 cat >confcache <<\_ACEOF
 # This file is a shell script that caches the results of configure
@@ -9173,7 +9173,7 @@
   case "$ac_config_target" in
   # Handling of arguments.
   "environment.mak" ) CONFIG_FILES="$CONFIG_FILES environment.mak:buildlib/environment.mak.in" ;;
-  "makefile" ) CONFIG_FILES="$CONFIG_FILES makefile:buildlib/makefile.in" ;;
+  "makefile.wrap" ) CONFIG_FILES="$CONFIG_FILES makefile.wrap:buildlib/makefile.in" ;;
   "default-1" ) CONFIG_COMMANDS="$CONFIG_COMMANDS default-1" ;;
   "default" ) CONFIG_COMMANDS="$CONFIG_COMMANDS default" ;;
   "include/config.h" ) CONFIG_HEADERS="$CONFIG_HEADERS include/config.h:buildlib/config.h.in" ;;
@@ -10018,7 +10018,7 @@
         ;;
       esac
     done ;;
-    default ) make -s dirs ;;
+    default ) make -f makefile.wrap -s dirs ;;
   esac
 done
 _ACEOF
--- apt-0.6.41/configure.in	2005-08-23 04:13:13.000000000 -0400
+++ apt-0.6.41-new/configure.in	2005-09-09 12:30:36.000000000 -0400
@@ -200,4 +200,4 @@
 AC_SUBST(USE_NLS)
 AC_PATH_PROG(BASH, bash)
 
-AC_OUTPUT(environment.mak:buildlib/environment.mak.in makefile:buildlib/makefile.in,make -s dirs)
+AC_OUTPUT(environment.mak:buildlib/environment.mak.in makefile.wrap:buildlib/makefile.in,make -f makefile.wrap -s dirs)
--- apt-0.6.41/doc/apt-cache.8	2005-09-05 17:01:54.000000000 -0400
+++ apt-0.6.41-new/doc/apt-cache.8	2005-09-09 12:30:36.000000000 -0400
@@ -42,7 +42,7 @@
 
 .TP
 gencaches
-gencaches performs the same operation as \fBapt\-get check\fR\&. It builds the source and package caches from the sources in \fB\fIsources\&.list\fR\fR(5) and from \fI/var/lib/dpkg/status\fR\&.
+gencaches performs the same operation as \fBapt\-get check\fR\&. It builds the source and package caches from the sources in \fB\fIsources\&.list\fR\fR(5) and from \fI@PREFIX@/var/lib/dpkg/status\fR\&.
 
 .TP
 showpkg \fIpkg(s)\fR
@@ -54,7 +54,7 @@
 .nf
 
 Package: libreadline2
-Versions: 2\&.1\-12(/var/state/apt/lists/foo_Packages),
+Versions: 2\&.1\-12(@PREFIX@/var/state/apt/lists/foo_Packages),
 Reverse Depends: 
   libreadlineg2,libreadline2
   libreadline2\-altdev,libreadline2
@@ -223,15 +223,15 @@
 .SH "FILES"
 
 .TP
-\fI/etc/apt/sources\&.list\fR
+\fI@PREFIX@/etc/apt/sources\&.list\fR
 Locations to fetch packages from\&. Configuration Item: Dir::Etc::SourceList\&.
 
 .TP
-\fI/var/lib/apt/lists/\fR
+\fI@PREFIX@/var/lib/apt/lists/\fR
 Storage area for state information for each package resource specified in \fB\fIsources\&.list\fR\fR(5) Configuration Item: Dir::State::Lists\&.
 
 .TP
-\fI/var/lib/apt/lists/partial/\fR
+\fI@PREFIX@/var/lib/apt/lists/partial/\fR
 Storage area for state information in transit\&. Configuration Item: Dir::State::Lists (implicit partial)\&.
 
 .SH "SEE ALSO"
@@ -247,7 +247,7 @@
 .SH "BUGS"
 
 .PP
-APT bug page: \fIhttp://bugs.debian.org/src:apt\fR\&. If you wish to report a bug in APT, please see \fI/usr/share/doc/debian/bug\-reporting\&.txt\fR or the \fB\fBreportbug\fR\fR(1) command\&.
+APT bug page: \fIhttp://bugs.debian.org/src:apt\fR\&. If you wish to report a bug in APT, please see \fI@PREFIX@/share/doc/debian/bug\-reporting\&.txt\fR or the \fB\fBreportbug\fR\fR(1) command\&.
 
 .SH AUTHORS
 Jason Gunthorpe, APT team.
--- apt-0.6.41/doc/apt-cdrom.8	2005-09-05 17:01:57.000000000 -0400
+++ apt-0.6.41-new/doc/apt-cdrom.8	2005-09-09 12:30:36.000000000 -0400
@@ -43,7 +43,7 @@
 add
 add is used to add a new disc to the source list\&. It will unmount the CDROM device, prompt for a disk to be inserted and then procceed to scan it and copy the index files\&. If the disc does not have a proper \fIdisk\fR directory you will be prompted for a descriptive title\&.
 
-APT uses a CDROM ID to track which disc is currently in the drive and maintains a database of these IDs in \fI/var/lib/apt/cdroms\&.list\fR 
+APT uses a CDROM ID to track which disc is currently in the drive and maintains a database of these IDs in \fI@PREFIX@/var/lib/apt/cdroms\&.list\fR 
 
 .TP
 ident
@@ -108,7 +108,7 @@
 .SH "BUGS"
 
 .PP
-APT bug page: \fIhttp://bugs.debian.org/src:apt\fR\&. If you wish to report a bug in APT, please see \fI/usr/share/doc/debian/bug\-reporting\&.txt\fR or the \fB\fBreportbug\fR\fR(1) command\&.
+APT bug page: \fIhttp://bugs.debian.org/src:apt\fR\&. If you wish to report a bug in APT, please see \fI@PREFIX@/share/doc/debian/bug\-reporting\&.txt\fR or the \fB\fBreportbug\fR\fR(1) command\&.
 
 .SH AUTHORS
 Jason Gunthorpe, APT team.
--- apt-0.6.41/doc/apt-config.8	2005-09-05 17:02:02.000000000 -0400
+++ apt-0.6.41-new/doc/apt-config.8	2005-09-09 12:30:36.000000000 -0400
@@ -31,7 +31,7 @@
 .SH "DESCRIPTION"
 
 .PP
-\fBapt\-config\fR is an internal program used by various portions of the APT suite to provide consistent configurability\&. It accesses the main configuration file \fI/etc/apt/apt\&.conf\fR in a manner that is easy to use by scripted applications\&.
+\fBapt\-config\fR is an internal program used by various portions of the APT suite to provide consistent configurability\&. It accesses the main configuration file \fI@PREFIX@/etc/apt/apt\&.conf\fR in a manner that is easy to use by scripted applications\&.
 
 .PP
 Unless the \fB\-h\fR, or \fB\-\-help\fR option is given one of the commands below must be present\&.
@@ -92,7 +92,7 @@
 .SH "BUGS"
 
 .PP
-APT bug page: \fIhttp://bugs.debian.org/src:apt\fR\&. If you wish to report a bug in APT, please see \fI/usr/share/doc/debian/bug\-reporting\&.txt\fR or the \fB\fBreportbug\fR\fR(1) command\&.
+APT bug page: \fIhttp://bugs.debian.org/src:apt\fR\&. If you wish to report a bug in APT, please see \fI@PREFIX@/share/doc/debian/bug\-reporting\&.txt\fR or the \fB\fBreportbug\fR\fR(1) command\&.
 
 .SH AUTHORS
 Jason Gunthorpe, APT team.
--- apt-0.6.41/doc/apt-get.8	2005-09-05 17:01:55.000000000 -0400
+++ apt-0.6.41-new/doc/apt-get.8	2005-09-09 12:30:36.000000000 -0400
@@ -38,11 +38,11 @@
 
 .TP
 update
-update is used to resynchronize the package index files from their sources\&. The indexes of available packages are fetched from the location(s) specified in \fI/etc/apt/sources\&.list\fR\&. For example, when using a Debian archive, this command retrieves and scans the \fIPackages\&.gz\fR files, so that information about new and updated packages is available\&. An update should always be performed before an upgrade or dist\-upgrade\&. Please be aware that the overall progress meter will be incorrect as the size of the package files cannot be known in advance\&.
+update is used to resynchronize the package index files from their sources\&. The indexes of available packages are fetched from the location(s) specified in \fI@PREFIX@/etc/apt/sources\&.list\fR\&. For example, when using a Debian archive, this command retrieves and scans the \fIPackages\&.gz\fR files, so that information about new and updated packages is available\&. An update should always be performed before an upgrade or dist\-upgrade\&. Please be aware that the overall progress meter will be incorrect as the size of the package files cannot be known in advance\&.
 
 .TP
 upgrade
-upgrade is used to install the newest versions of all packages currently installed on the system from the sources enumerated in \fI/etc/apt/sources\&.list\fR\&. Packages currently installed with new versions available are retrieved and upgraded; under no circumstances are currently installed packages removed, or packages not already installed retrieved and installed\&. New versions of currently installed packages that cannot be upgraded without changing the install status of another package will be left at their current version\&. An update must be performed first so that \fBapt\-get\fR knows that new versions of packages are available\&.
+upgrade is used to install the newest versions of all packages currently installed on the system from the sources enumerated in \fI@PREFIX@/etc/apt/sources\&.list\fR\&. Packages currently installed with new versions available are retrieved and upgraded; under no circumstances are currently installed packages removed, or packages not already installed retrieved and installed\&. New versions of currently installed packages that cannot be upgraded without changing the install status of another package will be left at their current version\&. An update must be performed first so that \fBapt\-get\fR knows that new versions of packages are available\&.
 
 .TP
 dselect\-upgrade
@@ -50,11 +50,11 @@
 
 .TP
 dist\-upgrade
-dist\-upgrade in addition to performing the function of upgrade, also intelligently handles changing dependencies with new versions of packages; \fBapt\-get\fR has a "smart" conflict resolution system, and it will attempt to upgrade the most important packages at the expense of less important ones if necessary\&. The \fI/etc/apt/sources\&.list\fR file contains a list of locations from which to retrieve desired package files\&. See also \fB\fBapt_preferences\fR\fR(5) for a mechanism for overriding the general settings for individual packages\&.
+dist\-upgrade in addition to performing the function of upgrade, also intelligently handles changing dependencies with new versions of packages; \fBapt\-get\fR has a "smart" conflict resolution system, and it will attempt to upgrade the most important packages at the expense of less important ones if necessary\&. The \fI@PREFIX@/etc/apt/sources\&.list\fR file contains a list of locations from which to retrieve desired package files\&. See also \fB\fBapt_preferences\fR\fR(5) for a mechanism for overriding the general settings for individual packages\&.
 
 .TP
 install
-install is followed by one or more packages desired for installation\&. Each package is a package name, not a fully qualified filename (for instance, in a Debian GNU/Linux system, libc6 would be the argument provided, not libc6_1\&.9\&.6\-2\&.deb) All packages required by the package(s) specified for installation will also be retrieved and installed\&. The \fI/etc/apt/sources\&.list\fR file is used to locate the desired packages\&. If a hyphen is appended to the package name (with no intervening space), the identified package will be removed if it is installed\&. Similarly a plus sign can be used to designate a package to install\&. These latter features may be used to override decisions made by apt\-get's conflict resolution system\&.
+install is followed by one or more packages desired for installation\&. Each package is a package name, not a fully qualified filename (for instance, in a Debian GNU/Linux system, libc6 would be the argument provided, not libc6_1\&.9\&.6\-2\&.deb) All packages required by the package(s) specified for installation will also be retrieved and installed\&. The \fI@PREFIX@/etc/apt/sources\&.list\fR file is used to locate the desired packages\&. If a hyphen is appended to the package name (with no intervening space), the identified package will be removed if it is installed\&. Similarly a plus sign can be used to designate a package to install\&. These latter features may be used to override decisions made by apt\-get's conflict resolution system\&.
 
 A specific version of a package can be selected for installation by following the package name with an equals and the version of the package to select\&. This will cause that version to be located and selected for install\&. Alternatively a specific distribution can be selected by following the package name with a slash and the version of the distribution or the Archive name (stable, testing, unstable)\&.
 
@@ -86,7 +86,7 @@
 
 .TP
 clean
-clean clears out the local repository of retrieved package files\&. It removes everything but the lock file from \fI/var/cache/apt/archives/\fR and \fI/var/cache/apt/archives/partial/\fR\&. When APT is used as a \fB\fBdselect\fR\fR(8) method, clean is run automatically\&. Those who do not use dselect will likely want to run apt\-get clean from time to time to free up disk space\&.
+clean clears out the local repository of retrieved package files\&. It removes everything but the lock file from \fI@PREFIX@/var/cache/apt/archives/\fR and \fI@PREFIX@/var/cache/apt/archives/partial/\fR\&. When APT is used as a \fB\fBdselect\fR\fR(8) method, clean is run automatically\&. Those who do not use dselect will likely want to run apt\-get clean from time to time to free up disk space\&.
 
 .TP
 autoclean
@@ -165,7 +165,7 @@
 
 .TP
 \fB\-\-list\-cleanup\fR
-This option defaults to on, use \-\-no\-list\-cleanup to turn it off\&. When on \fBapt\-get\fR will automatically manage the contents of \fI/var/lib/apt/lists\fR to ensure that obsolete files are erased\&. The only reason to turn it off is if you frequently change your source list\&. Configuration Item: APT::Get::List\-Cleanup\&.
+This option defaults to on, use \-\-no\-list\-cleanup to turn it off\&. When on \fBapt\-get\fR will automatically manage the contents of \fI@PREFIX@/var/lib/apt/lists\fR to ensure that obsolete files are erased\&. The only reason to turn it off is if you frequently change your source list\&. Configuration Item: APT::Get::List\-Cleanup\&.
 
 .TP
 \fB\-t\fR, \fB\-\-target\-release\fR, \fB\-\-default\-release\fR
@@ -214,41 +214,41 @@
 .SH "FILES"
 
 .TP
-\fI/etc/apt/sources\&.list\fR
+\fI@PREFIX@/etc/apt/sources\&.list\fR
 Locations to fetch packages from\&. Configuration Item: Dir::Etc::SourceList\&.
 
 .TP
-\fI/etc/apt/apt\&.conf\fR
+\fI@PREFIX@/etc/apt/apt\&.conf\fR
 APT configuration file\&. Configuration Item: Dir::Etc::Main\&.
 
 .TP
-\fI/etc/apt/apt\&.conf\&.d/\fR
+\fI@PREFIX@/etc/apt/apt\&.conf\&.d/\fR
 APT configuration file fragments Configuration Item: Dir::Etc::Parts\&.
 
 .TP
-\fI/etc/apt/preferences\fR
+\fI@PREFIX@/etc/apt/preferences\fR
 Version preferences file\&. This is where you would specify "pinning", i\&.e\&. a preference to get certain packages from a separate source or from a different version of a distribution\&. Configuration Item: Dir::Etc::Preferences\&.
 
 .TP
-\fI/var/cache/apt/archives/\fR
+\fI@PREFIX@/var/cache/apt/archives/\fR
 Storage area for retrieved package files\&. Configuration Item: Dir::Cache::Archives\&.
 
 .TP
-\fI/var/cache/apt/archives/partial/\fR
+\fI@PREFIX@/var/cache/apt/archives/partial/\fR
 Storage area for package files in transit\&. Configuration Item: Dir::Cache::Archives (implicit partial)\&.
 
 .TP
-\fI/var/lib/apt/lists/\fR
+\fI@PREFIX@/var/lib/apt/lists/\fR
 Storage area for state information for each package resource specified in \fB\fIsources\&.list\fR\fR(5) Configuration Item: Dir::State::Lists\&.
 
 .TP
-\fI/var/lib/apt/lists/partial/\fR
+\fI@PREFIX@/var/lib/apt/lists/partial/\fR
 Storage area for state information in transit\&. Configuration Item: Dir::State::Lists (implicit partial)\&.
 
 .SH "SEE ALSO"
 
 .PP
-\fB\fBapt\-cache\fR\fR(8), \fB\fBapt\-cdrom\fR\fR(8), \fB\fBdpkg\fR\fR(8), \fB\fBdselect\fR\fR(8), \fB\fIsources\&.list\fR\fR(5), \fB\fIapt\&.conf\fR\fR(5), \fB\fBapt\-config\fR\fR(8), The APT User's guide in /usr/share/doc/apt/, \fB\fBapt_preferences\fR\fR(5), the APT Howto\&.
+\fB\fBapt\-cache\fR\fR(8), \fB\fBapt\-cdrom\fR\fR(8), \fB\fBdpkg\fR\fR(8), \fB\fBdselect\fR\fR(8), \fB\fIsources\&.list\fR\fR(5), \fB\fIapt\&.conf\fR\fR(5), \fB\fBapt\-config\fR\fR(8), The APT User's guide in @PREFIX@/share/doc/apt/, \fB\fBapt_preferences\fR\fR(5), the APT Howto\&.
 
 .SH "DIAGNOSTICS"
 
@@ -258,7 +258,7 @@
 .SH "BUGS"
 
 .PP
-APT bug page: \fIhttp://bugs.debian.org/src:apt\fR\&. If you wish to report a bug in APT, please see \fI/usr/share/doc/debian/bug\-reporting\&.txt\fR or the \fB\fBreportbug\fR\fR(1) command\&.
+APT bug page: \fIhttp://bugs.debian.org/src:apt\fR\&. If you wish to report a bug in APT, please see \fI@PREFIX@/share/doc/debian/bug\-reporting\&.txt\fR or the \fB\fBreportbug\fR\fR(1) command\&.
 
 .SH AUTHORS
 Jason Gunthorpe, APT team.
--- apt-0.6.41/doc/apt_preferences.5	2005-09-05 17:02:03.000000000 -0400
+++ apt-0.6.41-new/doc/apt_preferences.5	2005-09-09 12:30:36.000000000 -0400
@@ -23,7 +23,7 @@
 .SH "DESCRIPTION"
 
 .PP
-The APT preferences file \fI/etc/apt/preferences\fR can be used to control which versions of packages will be selected for installation\&.
+The APT preferences file \fI@PREFIX@/etc/apt/preferences\fR can be used to control which versions of packages will be selected for installation\&.
 
 .PP
 Several versions of a package may be available for installation when the \fB\fIsources\&.list\fR\fR(5) file contains references to more than one distribution (for example, stable and testing)\&. APT assigns a priority to each version that is available\&. Subject to dependency constraints, \fBapt\-get\fR selects the version with the highest priority for installation\&. The APT preferences file overrides the priorities that APT assigns to package versions by default, thus giving the user control over which one is selected for installation\&.
@@ -34,7 +34,7 @@
 .SS "APT's Default Priority Assignments"
 
 .PP
-If there is no preferences file or if there is no entry in the file that applies to a particular version then the priority assigned to that version is the priority of the distribution to which that version belongs\&. It is possible to single out a distribution, "the target release", which receives a higher priority than other distributions do by default\&. The target release can be set on the \fBapt\-get\fR command line or in the APT configuration file \fI/etc/apt/apt\&.conf\fR\&. For example, 
+If there is no preferences file or if there is no entry in the file that applies to a particular version then the priority assigned to that version is the priority of the distribution to which that version belongs\&. It is possible to single out a distribution, "the target release", which receives a higher priority than other distributions do by default\&. The target release can be set on the \fBapt\-get\fR command line or in the APT configuration file \fI@PREFIX@/etc/apt/apt\&.conf\fR\&. For example, 
 
 .nf
 
@@ -291,7 +291,7 @@
  
 
 .PP
-All of the \fIPackages\fR and \fIRelease\fR files retrieved from locations listed in the \fB\fIsources\&.list\fR\fR(5) file are stored in the directory \fI/var/lib/apt/lists\fR, or in the file named by the variable Dir::State::Lists in the \fIapt\&.conf\fR file\&. For example, the file \fIdebian\&.lcs\&.mit\&.edu_debian_dists_unstable_contrib_binary\-i386_Release\fR contains the \fIRelease\fR file retrieved from the site debian\&.lcs\&.mit\&.edu for binary\-i386 architecture files from the contrib component of the unstable distribution\&.
+All of the \fIPackages\fR and \fIRelease\fR files retrieved from locations listed in the \fB\fIsources\&.list\fR\fR(5) file are stored in the directory \fI@PREFIX@/var/lib/apt/lists\fR, or in the file named by the variable Dir::State::Lists in the \fIapt\&.conf\fR file\&. For example, the file \fIdebian\&.lcs\&.mit\&.edu_debian_dists_unstable_contrib_binary\-i386_Release\fR contains the \fIRelease\fR file retrieved from the site debian\&.lcs\&.mit\&.edu for binary\-i386 architecture files from the contrib component of the unstable distribution\&.
 
 .SS "Optional Lines in an APT Preferences Record"
 
@@ -397,7 +397,7 @@
 .SH "BUGS"
 
 .PP
-APT bug page: \fIhttp://bugs.debian.org/src:apt\fR\&. If you wish to report a bug in APT, please see \fI/usr/share/doc/debian/bug\-reporting\&.txt\fR or the \fB\fBreportbug\fR\fR(1) command\&.
+APT bug page: \fIhttp://bugs.debian.org/src:apt\fR\&. If you wish to report a bug in APT, please see \fI@PREFIX@/share/doc/debian/bug\-reporting\&.txt\fR or the \fB\fBreportbug\fR\fR(1) command\&.
 
 .SH AUTHOR
 APT team.
--- apt-0.6.41/doc/sources.list.5	2005-09-05 17:02:00.000000000 -0400
+++ apt-0.6.41-new/doc/sources.list.5	2005-09-09 12:30:36.000000000 -0400
@@ -23,7 +23,7 @@
 .SH "DESCRIPTION"
 
 .PP
-The package resource list is used to locate archives of the package distribution system in use on the system\&. At this time, this manual page documents only the packaging system used by the Debian GNU/Linux system\&. This control file is located in \fI/etc/apt/sources\&.list\fR
+The package resource list is used to locate archives of the package distribution system in use on the system\&. At this time, this manual page documents only the packaging system used by the Debian GNU/Linux system\&. This control file is located in \fI@PREFIX@/etc/apt/sources\&.list\fR
 
 .PP
 The source list is designed to support any number of active sources and a variety of source media\&. The file lists one source per line, with the most preferred source listed first\&. The format of each line is: type uri args The first item, type determines the format for args  uri is a Universal Resource Identifier (URI), which is a superset of the more specific and well\-known Universal Resource Locator, or URL\&. The rest of the line can be marked as a comment by using a #\&.
@@ -159,7 +159,7 @@
 .SH "BUGS"
 
 .PP
-APT bug page: \fIhttp://bugs.debian.org/src:apt\fR\&. If you wish to report a bug in APT, please see \fI/usr/share/doc/debian/bug\-reporting\&.txt\fR or the \fB\fBreportbug\fR\fR(1) command\&.
+APT bug page: \fIhttp://bugs.debian.org/src:apt\fR\&. If you wish to report a bug in APT, please see \fI@PREFIX@/share/doc/debian/bug\-reporting\&.txt\fR or the \fB\fBreportbug\fR\fR(1) command\&.
 
 .SH AUTHORS
 Jason Gunthorpe, APT team.
--- apt-0.6.41/dselect/install	2005-08-01 12:05:41.000000000 -0400
+++ apt-0.6.41-new/dselect/install	2005-09-09 12:30:36.000000000 -0400
@@ -6,8 +6,8 @@
 # Get the configuration from /etc/apt/apt.conf
 CLEAN="prompt"
 OPTS="-f"
-APTGET="/usr/bin/apt-get"
-DPKG="/usr/bin/dpkg"
+APTGET="@PREFIX@/bin/apt-get"
+DPKG="@PREFIX@/bin/dpkg"
 DPKG_OPTS="--admindir=$1"
 APT_OPT0="-oDir::State::status=$1/status"
 APT_OPT1="-oDPkg::Options::=$DPKG_OPTS"
--- apt-0.6.41/dselect/setup	2005-08-01 12:05:41.000000000 -0400
+++ apt-0.6.41-new/dselect/setup	2005-09-09 12:30:36.000000000 -0400
@@ -23,15 +23,17 @@
 my $vardir=$ARGV[0];
 my $method=$ARGV[1];
 my $option=$ARGV[2];
-my $config_file = '/etc/apt/sources.list';
+my $config_file = '@PREFIX@/etc/apt/sources.list';
 
-my $boldon=`setterm -bold on`;
-my $boldoff=`setterm -bold off`;
+my $boldon=`setterm -bold on 2>/dev/null`;
+my $boldoff=`setterm -bold off 2>/dev/null`;
+$boldon = "" unless defined $boldon;
+$boldoff = "" unless defined $boldon;
 
 my @known_types           = ('deb');
 my @known_access         = ('http', 'ftp', 'file');
-my @typical_distributions = ('stable', 'unstable', 'testing', 'non-US');
-my @typical_components    = ('main', 'contrib', 'non-free');
+my @typical_distributions = ('@DIST@/release', '@DIST@/current');
+my @typical_components    = ('main', 'crypto');
 
 my %known_access           = map {($_,$_)} @known_access;
 my %typical_distributions  = map {($_,$_)} @typical_distributions;
@@ -118,9 +120,9 @@
   }
 
   $type         = 'deb';
-  $urn          = "http://http.us.debian.org/debian" unless $urn;
-  $distribution = "stable" unless $distribution;
-  $components   = "main contrib non-free" unless $components;
+  $urn          = "http://us.dl.sourceforge.net/fink/direct_download" unless $urn;
+  $distribution = "@DIST@/release" unless $distribution;
+  $components   = "main" unless $components;
 
     
   $rec->{'Type'} = 'deb';
@@ -222,19 +224,13 @@
   print "\t$boldon Set up a list of distribution source locations $boldoff \n";
   print "\n";
 
-  print " Please give the base URL of the debian distribution.\n";
+  print " Please give the base URL of the Fink distribution.\n";
   print " The access schemes I know about are:$boldon ";
   print join (' ', @known_access), "$boldoff\n";
-#  print " The mirror scheme is special  that it does not specify the\n";
-#  print " location of a debian archive but specifies the location\n";
-#  print " of a list of mirrors to use to access the archive.\n";
   print "\n";
   print " For example:\n";
-  print "              file:/mnt/debian,\n";
-  print "              ftp://ftp.debian.org/debian,\n";
-  print "              http://ftp.de.debian.org/debian,\n";
-#  print " and the special mirror scheme,\n";
-#  print "              mirror:http://www.debian.org/archivemirrors \n";
+  print "              file:@PREFIX@/fink,\n";
+  print "              http://us.dl.sourceforge.net/fink/direct_download\n";
   print "\n";
 
   my $index = 0;
@@ -269,7 +265,11 @@
     print "-" x 72, "\n";
     &print_config('Config' => \@Oldconfig);
     print "-" x 72, "\n";
-    print "$boldon Do you wish to overwrite it? [y/N]$boldoff ";
+    print "$boldon In most cases, this file was installed by Fink or by apt,"
+        ." and$boldoff\n";
+    print "$boldon should NOT be changed.  " .
+        "Do you wish to change (overwrite) it?[y/N]$boldoff ";
+
     my $answer = <STDIN>;
     chomp ($answer);
     $answer =~ s/\s+/ /og;
--- apt-0.6.41/dselect/update	2005-08-01 12:05:43.000000000 -0400
+++ apt-0.6.41-new/dselect/update	2005-09-09 12:30:36.000000000 -0400
@@ -7,13 +7,13 @@
 # Get the configuration from /etc/apt/apt.conf
 CLEAN="prompt"
 OPTS="-f"
-APTGET="/usr/bin/apt-get"
-APTCACHE="/usr/bin/apt-cache"
-DPKG="/usr/bin/dpkg"
+APTGET="@PREFIX@/bin/apt-get"
+APTCACHE="@PREFIX@/bin/apt-cache"
+DPKG="@PREFIX@/bin/dpkg"
 DPKG_OPTS="--admindir=$1"
 APT_OPT0="-oDir::State::status=$1/status"
 APT_OPT1="-oDPkg::Options::=$DPKG_OPTS"
-CACHEDIR="/var/cache/apt"
+CACHEDIR="@PREFIX@/var/cache/apt"
 PROMPT="false"
 RES=`apt-config shell CLEAN DSelect::Clean OPTS DSelect::UpdateOptions \
 		      DPKG Dir::Bin::dpkg/f APTGET Dir::Bin::apt-get/f \
--- apt-0.6.41/methods/makefile	2005-08-19 04:04:07.000000000 -0400
+++ apt-0.6.41-new/methods/makefile	2005-09-09 12:30:36.000000000 -0400
@@ -7,61 +7,61 @@
 BIN := $(BIN)/methods
 
 # FIXME..
-LIB_APT_PKG_MAJOR = 3.10
+LIB_APT_PKG_MAJOR = 0.6.40
 APT_DOMAIN := libapt-pkg$(LIB_APT_PKG_MAJOR)
 
 # The file method
 PROGRAM=file
-SLIBS = -lapt-pkg 
+SLIBS = $(INTLLIBS) -lapt-pkg 
 LIB_MAKES = apt-pkg/makefile
 SOURCE = file.cc
 include $(PROGRAM_H)
 
 # The copy method
 PROGRAM=copy
-SLIBS = -lapt-pkg 
+SLIBS = $(INTLLIBS) -lapt-pkg 
 LIB_MAKES = apt-pkg/makefile
 SOURCE = copy.cc
 include $(PROGRAM_H)
 
 # The gzip method
 PROGRAM=gzip
-SLIBS = -lapt-pkg 
+SLIBS = $(INTLLIBS) -lapt-pkg 
 LIB_MAKES = apt-pkg/makefile
 SOURCE = gzip.cc
 include $(PROGRAM_H)
 
 # The gpgv method
 PROGRAM=gpgv
-SLIBS = -lapt-pkg
+SLIBS = $(INTLLIBS) -lapt-pkg
 LIB_MAKES = apt-pkg/makefile
 SOURCE = gpgv.cc
 include $(PROGRAM_H)
 
 # The cdrom method
 PROGRAM=cdrom
-SLIBS = -lapt-pkg 
+SLIBS = $(INTLLIBS) -lapt-pkg 
 LIB_MAKES = apt-pkg/makefile
 SOURCE = cdrom.cc
 include $(PROGRAM_H)
 
 # The http method
 PROGRAM=http
-SLIBS = -lapt-pkg $(SOCKETLIBS)
+SLIBS = $(INTLLIBS) -lapt-pkg $(SOCKETLIBS)
 LIB_MAKES = apt-pkg/makefile
 SOURCE = http.cc rfc2553emu.cc connect.cc
 include $(PROGRAM_H)
 
 # The ftp method
 PROGRAM=ftp
-SLIBS = -lapt-pkg $(SOCKETLIBS)
+SLIBS = $(INTLLIBS) -lapt-pkg $(SOCKETLIBS)
 LIB_MAKES = apt-pkg/makefile
 SOURCE = ftp.cc rfc2553emu.cc connect.cc
 include $(PROGRAM_H)
 
 # The rsh method
 PROGRAM=rsh
-SLIBS = -lapt-pkg
+SLIBS = $(INTLLIBS) -lapt-pkg
 LIB_MAKES = apt-pkg/makefile
 SOURCE = rsh.cc
 include $(PROGRAM_H)
--- apt-0.6.41/methods/rfc2553emu.h	2005-08-01 12:05:41.000000000 -0400
+++ apt-0.6.41-new/methods/rfc2553emu.h	2005-09-09 12:30:37.000000000 -0400
@@ -26,6 +26,11 @@
 #include <sys/types.h>
 #include <sys/socket.h>
 
+// Always use full emulation on Darwin:
+//  netdb.h has the structures and constants, but getnameinfo() is missing
+//  and getaddrinfo() seems to be broken
+#ifndef __APPLE__
+
 // Autosense getaddrinfo
 #if defined(AI_PASSIVE) && defined(EAI_NONAME)
 #define HAVE_GETADDRINFO
@@ -36,6 +41,8 @@
 #define HAVE_GETNAMEINFO
 #endif
 
+#endif /* __APPLE__ */
+
 // getaddrinfo support?
 #ifndef HAVE_GETADDRINFO
   // Renamed to advoid type clashing.. (for debugging)
@@ -101,6 +108,9 @@
   #define NI_NAMEREQD (1<<3)
   #define NI_DATAGRAM (1<<4)
   #endif
+  #ifndef NI_DATAGRAM
+  #define NI_DATAGRAM NI_DGRAM
+  #endif
 
   #define sockaddr_storage sockaddr_in
 #endif
--- apt-0.6.41/patch_flush	1969-12-31 19:00:00.000000000 -0500
+++ apt-0.6.41-new/patch_flush	2005-09-09 12:30:37.000000000 -0400
@@ -0,0 +1,11 @@
+#!/bin/sh
+set -e
+
+files=`find . -name '*.cc' -print | xargs grep -l 'flush;'`
+
+for i in $files ; do
+  sed 's/<< flush;/<< flush, fflush(NULL);/g' <$i >$i.tmp
+  mv $i.tmp $i
+done
+
+exit 0
--- apt-0.6.41/cmdline/apt-get.cc	2005-08-23 04:13:13.000000000 -0400
+++ apt-0.6.41-new/cmdline/apt-get.cc	2005-10-03 11:13:16.000000000 -0400
@@ -51,7 +51,7 @@
 #include <termios.h>
 #include <sys/ioctl.h>
 #include <sys/stat.h>
-#include <sys/statvfs.h>
+#include <sys/mount.h>
 #include <signal.h>
 #include <unistd.h>
 #include <stdio.h>
@@ -113,6 +113,8 @@
 /* Returns true on a Yes.*/
 bool YnPrompt(bool Default=true)
 {
+   fflush(NULL);
+   
    if (_config->FindB("APT::Get::Assume-Yes",false) == true)
    {
       c1out << _("Y") << endl;
@@ -331,7 +333,20 @@
 		  if (Cache[Targ].CandidateVerIter(Cache).end() == true)
 		  {
 		     if (Targ->ProvidesList == 0)
-			out << _("but it is not installable");
+		     {
+			out << _("but it is not installable. For Fink users, ");
+			out << _("this often means that you have attempted ");
+			out << _("to install a package from the binary distribution ");
+			out << _("that depends on a package (or has a subdependency) ");
+			out << _("that is not part of the binary distribution.  The ");
+			out << _("omitted package may be a virtual package, which ");
+			out << _("must be installed manually instead of with fink, ");
+			out << _("or one that has a \"Restrictive\" license that ");
+			out << _("prohibits binary distribution.  See:\n");
+			out << _("  <http://fink.sourceforge.net/faq/usage-fink.php#bindist>\n");
+			out << _("  <http://fink.sourceforge.net/doc/users-guide/packages.php#bin-exceptions>\n");
+			out << _("  <http://fink.sourceforge.net/doc/usage-general.php#virtpackage>\n");
+		     }
 		     else
 			out << _("but it is a virtual package");
 		  }		  
@@ -638,13 +653,15 @@
       return false;
    
    // Nothing is broken
-   if (DCache->BrokenCount() == 0 || AllowBroken == true)
+   // FINK LOCAL added APT::Get::Ignore-Breakage test
+   if (DCache->BrokenCount() == 0 || AllowBroken == true
+       || _config->FindB("APT::Get::Ignore-Breakage") == true)
       return true;
 
    // Attempt to fix broken things
    if (_config->FindB("APT::Get::Fix-Broken",false) == true)
    {
-      c1out << _("Correcting dependencies...") << flush;
+      c1out << _("Correcting dependencies...") << flush, fflush(NULL);
       if (pkgFixBroken(*DCache) == false || DCache->BrokenCount() != 0)
       {
 	 c1out << _(" failed.") << endl;
@@ -747,7 +764,9 @@
    Stats(c1out,Cache);
 
    // Sanity check
-   if (Cache->BrokenCount() != 0)
+   // FINK LOCAL added APT::Get::Ignore-Breakage test
+   if (Cache->BrokenCount() != 0
+       && _config->FindB("APT::Get::Ignore-Breakage",false) == false)
    {
       ShowBroken(c1out,Cache,false);
       return _error->Error(_("Internal error, InstallPackages was called with broken packages!"));
@@ -838,10 +857,10 @@
    if (_config->FindB("APT::Get::Print-URIs") == false &&
        _config->FindB("APT::Get::Download",true) == true)
    {
-      struct statvfs Buf;
+      struct statfs Buf;
       string OutputDir = _config->FindDir("Dir::Cache::Archives");
-      if (statvfs(OutputDir.c_str(),&Buf) != 0)
-	 return _error->Errno("statvfs",_("Couldn't determine free space in %s"),
+      if (statfs(OutputDir.c_str(),&Buf) != 0)
+	 return _error->Errno("statfs",_("Couldn't determine free space in %s"),
 			      OutputDir.c_str());
       if (unsigned(Buf.f_bfree) < (FetchBytes - FetchPBytes)/Buf.f_bsize)
 	 return _error->Error(_("You don't have enough free space in %s."),
@@ -866,7 +885,7 @@
 	       _("You are about to do something potentially harmful.\n"
 		 "To continue type in the phrase '%s'\n"
 		 " ?] "),Prompt);
-      c2out << flush;
+      c2out << flush, fflush(NULL);
       if (AnalPrompt(Prompt) == false)
       {
 	 c2out << _("Abort.") << endl;
@@ -1144,7 +1163,9 @@
       ExpectedInst++;
    
    // Install it with autoinstalling enabled.
-   if (State.InstBroken() == true && BrokenFix == false)
+   // FINK LOCAL added APT::Get::Ignore-Breakage test
+   if (State.InstBroken() == true && BrokenFix == false
+      && _config->FindB("APT::Get::Ignore-Breakage") == false)
       Cache.MarkInstall(Pkg,true);
    return true;
 }
@@ -1521,7 +1542,9 @@
    /* If we are in the Broken fixing mode we do not attempt to fix the
       problems. This is if the user invoked install without -f and gave
       packages */
-   if (BrokenFix == true && Cache->BrokenCount() != 0)
+   // FINK LOCAL added APT::Get::Ignore-Breakage test
+   if (BrokenFix == true && Cache->BrokenCount() != 0
+       && _config->FindB("APT::Get::Ignore-Breakage") == false)
    {
       c1out << _("You might want to run `apt-get -f install' to correct these:") << endl;
       ShowBroken(c1out,Cache,false);
@@ -1531,11 +1554,13 @@
    
    // Call the scored problem resolver
    Fix.InstallProtect();
-   if (Fix.Resolve(true) == false)
+   // FINK LOCAL added APT::Get::Ignore-Breakage test
+   if (_config->FindB("APT::Get::Ignore-Breakage") == false && Fix.Resolve(true) == false)
       _error->Discard();
 
    // Now we check the state of the packages,
-   if (Cache->BrokenCount() != 0)
+   // FINK LOCAL added APT::Get::Ignore-Breakage test
+   if (Cache->BrokenCount() != 0 && _config->FindB("APT::Get::Ignore-Breakage") == false)
    {
       c1out << 
        _("Some packages could not be installed. This may mean that you have\n" 
@@ -1921,10 +1946,10 @@
    double DebBytes = Fetcher.TotalNeeded();
 
    // Check for enough free space
-   struct statvfs Buf;
+   struct statfs Buf;
    string OutputDir = ".";
-   if (statvfs(OutputDir.c_str(),&Buf) != 0)
-      return _error->Errno("statvfs",_("Couldn't determine free space in %s"),
+   if (statfs(OutputDir.c_str(),&Buf) != 0)
+      return _error->Errno("statfs",_("Couldn't determine free space in %s"),
 			   OutputDir.c_str());
    if (unsigned(Buf.f_bfree) < (FetchBytes - FetchPBytes)/Buf.f_bsize)
       return _error->Error(_("You don't have enough free space in %s"),
@@ -2419,6 +2444,8 @@
    _config->Set("APT::Get::Simulate",false);
    _config->Set("APT::Get::Assume-Yes",false);
    _config->Set("APT::Get::Fix-Broken",false);
+   // FINK LOCAL added APT::Get::Ignore-Breakage
+   _config->Set("APT::Get::Ignore-Breakage",false);
    _config->Set("APT::Get::Force-Yes",false);
    _config->Set("APT::Get::List-Cleanup",true);
 }
@@ -2457,6 +2484,8 @@
       {'y',"yes","APT::Get::Assume-Yes",0},
       {'y',"assume-yes","APT::Get::Assume-Yes",0},      
       {'f',"fix-broken","APT::Get::Fix-Broken",0},
+      // FINK LOCAL added APT::Get::Ignore-Breakage
+      {0,"ignore-breakage","APT::Get::Ignore-Breakage",0},
       {'u',"show-upgraded","APT::Get::Show-Upgraded",0},
       {'m',"ignore-missing","APT::Get::Fix-Missing",0},
       {'t',"target-release","APT::Default-Release",CommandLine::HasArg},
@@ -2520,6 +2549,22 @@
       ShowHelp(CmdL);
       return 0;
    }
+
+   /* FINK LOCAL begin */
+   if (_config->FindB("APT::Get::Ignore-Breakage",false) == true) {
+     if (_config->FindB("APT::Get::Print-URIs",false) == false &&
+	 _config->FindB("APT::Get::Download-Only",false) == false) {
+       _error->Error("--ignore-breakage can only be used with --print-uris or --download-only");
+       _error->DumpErrors();
+       return 100;
+     }
+     if (strcmp(CmdL.FileList[0],"install") != 0) {
+       _error->Error("--ignore-breakage can only be used with apt-get install");
+       _error->DumpErrors();
+       return 100;
+     }
+   }
+   /* FINK LOCAL end */
    
    // Deal with stdout not being a tty
    if (!isatty(STDOUT_FILENO) && _config->FindI("quiet",0) < 1)
--- apt-0.6.41/ftparchive/contents.cc	2005-08-01 12:05:43.000000000 -0400
+++ apt-0.6.41-new/ftparchive/contents.cc	2005-10-03 12:01:11.000000000 -0400
@@ -41,7 +41,6 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
-#include <malloc.h>    
 									/*}}}*/
 
 // GenContents::~GenContents - Free allocated memory			/*{{{*/
--- apt-0.6.41/ftparchive/makefile	2005-08-01 12:05:41.000000000 -0400
+++ apt-0.6.41-new/ftparchive/makefile	2005-10-03 12:03:24.000000000 -0400
@@ -8,7 +8,7 @@
 # The apt-ftparchive program
 ifdef BDBLIB
 PROGRAM=apt-ftparchive
-SLIBS = -lapt-pkg -lapt-inst $(BDBLIB)
+SLIBS = -lapt-pkg -lapt-inst $(BDBLIB) $(INTLLIBS)
 LIB_MAKES = apt-pkg/makefile apt-inst/makefile
 SOURCE = apt-ftparchive.cc cachedb.cc writer.cc contents.cc override.cc \
          multicompress.cc
--- apt-0.6.41/buildlib/sgml_manpage.mak	2005-08-01 12:05:41.000000000 -0400
+++ apt-0.6.41-new/buildlib/sgml_manpage.mak	2005-10-03 13:47:16.000000000 -0400
@@ -11,7 +11,7 @@
 # See defaults.mak for information about LOCAL
 
 # Some local definitions
-ifdef DOCBOOK2MAN
+ifdef DOCBOOK2MANx
 
 LOCAL := sgml-manpage-$(firstword $(SOURCE))
 $(LOCAL)-LIST := $(SOURCE)
