Index: Makefile
===================================================================
RCS file: /cvsroot/fink/3rdparty/apt/Makefile,v
retrieving revision 1.1.1.1
retrieving revision 1.2
diff -u -r1.1.1.1 -r1.2
--- Makefile	17 Mar 2006 01:58:06 -0000	1.1.1.1
+++ Makefile	17 Mar 2006 02:25:34 -0000	1.2
@@ -15,7 +15,6 @@
 	$(MAKE) -C apt-inst $@
 	$(MAKE) -C methods $@
 	$(MAKE) -C cmdline $@
-	$(MAKE) -C ftparchive $@
 	$(MAKE) -C dselect $@
 	$(MAKE) -C doc $@
 	$(MAKE) -C po $@
Index: configure
===================================================================
RCS file: /cvsroot/fink/3rdparty/apt/configure,v
retrieving revision 1.1.1.1
retrieving revision 1.2
diff -u -r1.1.1.1 -r1.2
--- configure	17 Mar 2006 03:41:14 -0000	1.1.1.1
+++ configure	17 Mar 2006 04:06:39 -0000	1.2
@@ -8683,7 +8683,7 @@
 fi
 
 
-                    ac_config_files="$ac_config_files environment.mak:buildlib/environment.mak.in makefile:buildlib/makefile.in"
+                    ac_config_files="$ac_config_files environment.mak:buildlib/environment.mak.in makefile.wrap:buildlib/makefile.in"
           ac_config_commands="$ac_config_commands default"
 cat >confcache <<\_ACEOF
 # This file is a shell script that caches the results of configure
@@ -9227,7 +9227,7 @@
   case "$ac_config_target" in
   # Handling of arguments.
   "environment.mak" ) CONFIG_FILES="$CONFIG_FILES environment.mak:buildlib/environment.mak.in" ;;
-  "makefile" ) CONFIG_FILES="$CONFIG_FILES makefile:buildlib/makefile.in" ;;
+  "makefile.wrap" ) CONFIG_FILES="$CONFIG_FILES makefile.wrap:buildlib/makefile.in" ;;
   "default-1" ) CONFIG_COMMANDS="$CONFIG_COMMANDS default-1" ;;
   "default" ) CONFIG_COMMANDS="$CONFIG_COMMANDS default" ;;
   "include/config.h" ) CONFIG_HEADERS="$CONFIG_HEADERS include/config.h:buildlib/config.h.in" ;;
@@ -10072,7 +10072,7 @@
         ;;
       esac
     done ;;
-    default ) make -s dirs ;;
+    default ) make -f makefile.wrap -s dirs ;;
   esac
 done
 _ACEOF
Index: configure.in
===================================================================
RCS file: /cvsroot/fink/3rdparty/apt/configure.in,v
retrieving revision 1.1.1.1
retrieving revision 1.2
diff -u -r1.1.1.1 -r1.2
--- configure.in	17 Mar 2006 01:58:06 -0000	1.1.1.1
+++ configure.in	17 Mar 2006 02:25:34 -0000	1.2
@@ -200,4 +200,4 @@
 AC_SUBST(USE_NLS)
 AC_PATH_PROG(BASH, bash)
 
-AC_OUTPUT(environment.mak:buildlib/environment.mak.in makefile:buildlib/makefile.in,make -s dirs)
+AC_OUTPUT(environment.mak:buildlib/environment.mak.in makefile.wrap:buildlib/makefile.in,make -f makefile.wrap -s dirs)
Index: ftparchive-conf
===================================================================
RCS file: ftparchive-conf
diff -N ftparchive-conf
--- /dev/null	1 Jan 1970 00:00:00 -0000
+++ ftparchive-conf	17 Mar 2006 02:25:34 -0000	1.1
@@ -0,0 +1 @@
+Dir::Bin::gzip "@PREFIX@/bin/gzip";
Index: patch_flush
===================================================================
RCS file: patch_flush
diff -N patch_flush
--- /dev/null	1 Jan 1970 00:00:00 -0000
+++ patch_flush	17 Mar 2006 02:25:34 -0000	1.1
@@ -0,0 +1,11 @@
+#!/bin/sh
+set -e
+
+files=`find . -name '*.cc' -print | xargs grep -l 'flush;'`
+
+for i in $files ; do
+  sed 's/<< flush;/<< flush, fflush(NULL);/g' <$i >$i.tmp
+  mv $i.tmp $i
+done
+
+exit 0
Index: apt-inst/makefile
===================================================================
RCS file: /cvsroot/fink/3rdparty/apt/apt-inst/makefile,v
retrieving revision 1.1.1.1
retrieving revision 1.2
diff -u -r1.1.1.1 -r1.2
--- apt-inst/makefile	17 Mar 2006 01:58:07 -0000	1.1.1.1
+++ apt-inst/makefile	17 Mar 2006 02:25:36 -0000	1.2
@@ -12,9 +12,10 @@
 # The library name
 LIBRARY=apt-inst
 LIBEXT=$(GLIBC_VER)$(LIBSTDCPP_VER)
+COMPAT=1
 MAJOR=1.1
 MINOR=0
-SLIBS=$(PTHREADLIB) -lapt-pkg
+SLIBS=$(PTHREADLIB) -lapt-pkg $(INTLLIBS)
 APT_DOMAIN:=libapt-inst$(MAJOR)
 
 # Source code for the contributed non-core things
Index: apt-inst/contrib/extracttar.cc
===================================================================
RCS file: /cvsroot/fink/3rdparty/apt/apt-inst/contrib/extracttar.cc,v
retrieving revision 1.1.1.1
retrieving revision 1.4
diff -u -r1.1.1.1 -r1.4
--- apt-inst/contrib/extracttar.cc	17 Mar 2006 01:58:07 -0000	1.1.1.1
+++ apt-inst/contrib/extracttar.cc	17 Mar 2006 04:19:21 -0000	1.4
@@ -150,6 +150,18 @@
    return true;
 }
 									/*}}}*/
+
+// Handle the ridiculous way that tar stores large numbers
+static bool TarUIDToNum(const char *Str, unsigned long &Res, unsigned Len) {
+	switch (*Str) {
+		case '\200':
+			Res = ntohl(*((unsigned long *)(Str + Len - sizeof(unsigned long))));
+			return true;
+		default:
+			return StrToNum(Str+1, Res, Len-1, 8);
+	}
+}
+
 // ExtractTar::Go - Perform extraction					/*{{{*/
 // ---------------------------------------------------------------------
 /* This reads each 512 byte block from the archive and extracts the header
@@ -197,8 +209,8 @@
       // Decode all of the fields
       pkgDirStream::Item Itm;
       if (StrToNum(Tar->Mode,Itm.Mode,sizeof(Tar->Mode),8) == false ||
-	  StrToNum(Tar->UserID,Itm.UID,sizeof(Tar->UserID),8) == false ||
-	  StrToNum(Tar->GroupID,Itm.GID,sizeof(Tar->GroupID),8) == false ||
+	  TarUIDToNum(Tar->UserID,Itm.UID,sizeof(Tar->UserID)) == false ||
+	  TarUIDToNum(Tar->GroupID,Itm.GID,sizeof(Tar->GroupID)) == false ||
 	  StrToNum(Tar->Size,Itm.Size,sizeof(Tar->Size),8) == false ||
 	  StrToNum(Tar->MTime,Itm.MTime,sizeof(Tar->MTime),8) == false ||
 	  StrToNum(Tar->Major,Itm.Major,sizeof(Tar->Major),8) == false ||
Index: apt-pkg/acquire-item.cc
===================================================================
RCS file: /cvsroot/fink/3rdparty/apt/apt-pkg/acquire-item.cc,v
retrieving revision 1.1.1.1
retrieving revision 1.3
diff -u -r1.1.1.1 -r1.3
--- apt-pkg/acquire-item.cc	17 Mar 2006 01:58:07 -0000	1.1.1.1
+++ apt-pkg/acquire-item.cc	17 Mar 2006 04:19:21 -0000	1.3
@@ -739,12 +739,14 @@
 // ---------------------------------------------------------------------
 /* This just sets up the initial fetch environment and queues the first
    possibilitiy */
+
+// FIXME: when fink has a way to trust stuff, set Trusted(true) back to false  ;)
 pkgAcqArchive::pkgAcqArchive(pkgAcquire *Owner,pkgSourceList *Sources,
 			     pkgRecords *Recs,pkgCache::VerIterator const &Version,
 			     string &StoreFilename) :
                Item(Owner), Version(Version), Sources(Sources), Recs(Recs), 
                StoreFilename(StoreFilename), Vf(Version.FileList()), 
-	       Trusted(false)
+	       Trusted(true)
 {
    Retries = _config->FindI("Acquire::Retries",0);
 
Index: apt-pkg/init.cc
===================================================================
RCS file: /cvsroot/fink/3rdparty/apt/apt-pkg/init.cc,v
retrieving revision 1.1.1.1
retrieving revision 1.3
diff -u -r1.1.1.1 -r1.3
--- apt-pkg/init.cc	17 Mar 2006 01:58:08 -0000	1.1.1.1
+++ apt-pkg/init.cc	17 Mar 2006 04:19:21 -0000	1.3
@@ -15,6 +15,8 @@
 #include <apti18n.h>
 #include <config.h>
 #include <sys/stat.h>
+
+extern void initDebSystem();
 									/*}}}*/
 
 #define Stringfy_(x) # x
@@ -40,7 +42,7 @@
    else
       Cnf.Set("APT::Architecture",COMMON_OS "-" COMMON_CPU);
    Cnf.Set("APT::Build-Essential::", "build-essential");
-   Cnf.Set("Dir","/");
+   Cnf.Set("Dir","@PREFIX@/");
    
    // State   
    Cnf.Set("Dir::State","var/lib/apt/");
@@ -70,7 +72,7 @@
    Cnf.Set("Dir::Etc::main","apt.conf");
    Cnf.Set("Dir::Etc::parts","apt.conf.d");
    Cnf.Set("Dir::Etc::preferences","preferences");
-   Cnf.Set("Dir::Bin::methods","/usr/lib/apt/methods");
+   Cnf.Set("Dir::Bin::methods","@PREFIX@/lib/apt/methods");
    
    bool Res = true;
    
@@ -111,6 +113,8 @@
 /* */
 bool pkgInitSystem(Configuration &Cnf,pkgSystem *&Sys)
 {
+   initDebSystem();
+
    Sys = 0;
    string Label = Cnf.Find("Apt::System","");
    if (Label.empty() == false)
Index: apt-pkg/makefile
===================================================================
RCS file: /cvsroot/fink/3rdparty/apt/apt-pkg/makefile,v
retrieving revision 1.1.1.1
retrieving revision 1.2
diff -u -r1.1.1.1 -r1.2
--- apt-pkg/makefile	17 Mar 2006 01:58:08 -0000	1.1.1.1
+++ apt-pkg/makefile	17 Mar 2006 02:25:36 -0000	1.2
@@ -13,6 +13,7 @@
 # methods/makefile - FIXME
 LIBRARY=apt-pkg
 LIBEXT=$(GLIBC_VER)$(LIBSTDCPP_VER)
+COMPAT=3
 MAJOR=3.11
 MINOR=0
 SLIBS=$(PTHREADLIB) $(INTLLIBS)
Index: apt-pkg/contrib/cdromutl.cc
===================================================================
RCS file: /cvsroot/fink/3rdparty/apt/apt-pkg/contrib/cdromutl.cc,v
retrieving revision 1.1.1.1
retrieving revision 1.3
diff -u -r1.1.1.1 -r1.3
--- apt-pkg/contrib/cdromutl.cc	17 Mar 2006 01:58:10 -0000	1.1.1.1
+++ apt-pkg/contrib/cdromutl.cc	17 Mar 2006 04:19:21 -0000	1.3
@@ -23,13 +23,13 @@
     
 #include <sys/wait.h>
 #include <sys/errno.h>
-#include <sys/statvfs.h>
+#include <sys/mount.h>
 #include <dirent.h>
 #include <fcntl.h>
 #include <sys/stat.h>
 #include <unistd.h>
 #include <stdio.h>
-									/*}}}*/
+
 
 // IsMounted - Returns true if the mount point is mounted		/*{{{*/
 // ---------------------------------------------------------------------
@@ -185,8 +185,8 @@
    // Some stats from the fsys
    if (_config->FindB("Debug::identcdrom",false) == false)
    {
-      struct statvfs Buf;
-      if (statvfs(CD.c_str(),&Buf) != 0)
+      struct statfs Buf;
+      if (statfs(CD.c_str(),&Buf) != 0)
 	 return _error->Errno("statfs",_("Failed to stat the cdrom"));
       
       // We use a kilobyte block size to advoid overflow
Index: apt-pkg/deb/debindexfile.cc
===================================================================
RCS file: /cvsroot/fink/3rdparty/apt/apt-pkg/deb/debindexfile.cc,v
retrieving revision 1.1.1.1
retrieving revision 1.3
diff -u -r1.1.1.1 -r1.3
--- apt-pkg/deb/debindexfile.cc	17 Mar 2006 01:58:11 -0000	1.1.1.1
+++ apt-pkg/deb/debindexfile.cc	17 Mar 2006 04:19:21 -0000	1.3
@@ -444,3 +444,9 @@
 }
 
 									/*}}}*/
+void init_deb2()
+{
+  (void)_apt_Src;
+  (void)_apt_Pkg;
+  (void)_apt_Status;
+}
Index: apt-pkg/deb/debsystem.cc
===================================================================
RCS file: /cvsroot/fink/3rdparty/apt/apt-pkg/deb/debsystem.cc,v
retrieving revision 1.1.1.1
retrieving revision 1.3
diff -u -r1.1.1.1 -r1.3
--- apt-pkg/deb/debsystem.cc	17 Mar 2006 01:58:11 -0000	1.1.1.1
+++ apt-pkg/deb/debsystem.cc	17 Mar 2006 04:19:21 -0000	1.3
@@ -27,6 +27,111 @@
 #include <dirent.h>
 #include <errno.h>
 									/*}}}*/
+/* FINK LOCAL begin */
+#include <sys/utsname.h>
+#include <CoreFoundation/CoreFoundation.h>
+#include <fstream>
+#include <sys/stat.h>
+
+extern void init_deb2();
+extern void init_deb3();
+
+#define FINKSTATUSFILE "/tmp/finkaptstatus"
+
+struct versionrevision {
+  unsigned long epoch;
+  const char *version;
+  const char *revision;
+};  
+
+struct versionrevision darwin_version = {0,NULL,NULL};
+struct versionrevision macosx_version = {0,NULL,NULL};
+
+using namespace std;
+
+static void finkinit()
+{
+  Boolean status;
+  SInt32 errorCode;
+  CFURLRef fileURL = NULL;
+  CFDataRef resourceData = NULL;
+  CFPropertyListRef propertyList = NULL;
+  CFStringRef string;
+  static char buffer[256];	// This is static, to ensure the buffer stays around
+
+  static struct utsname ver;	// This is static, to ensure the buffer stays around
+  
+  /* Determine system version */
+  /* TODO - should maybe check if this is really Darwin? */
+  if (!uname(&ver)) {
+    darwin_version.version = ver.release;
+  }
+
+  /* Check whether this is Mac OS X, and which version of it */
+
+  fileURL = CFURLCreateWithFileSystemPath( NULL, 	
+		CFSTR("/System/Library/CoreServices/SystemVersion.plist"),
+		kCFURLPOSIXPathStyle,				
+		false );
+  if (!fileURL)
+    goto BAIL;
+  
+  /* Read the XML */
+  status = CFURLCreateDataAndPropertiesFromResource(
+		NULL,
+		fileURL,
+		&resourceData,
+		NULL,		
+		NULL,
+		&errorCode);
+  if (!status || errorCode != 0)
+    goto BAIL;
+  
+  /* Reconstitute the dictionary using the XML data. */
+  propertyList = CFPropertyListCreateFromXMLData( NULL,
+		resourceData,
+		kCFPropertyListImmutable,
+		&string);
+  if (!propertyList)
+    goto BAIL;
+  
+  /* Try to read the system version from it. */
+  status = CFDictionaryGetValueIfPresent( 
+		(CFDictionaryRef) propertyList,
+		(const void *) CFSTR("ProductVersion"),
+		(const void**) &string);
+  if (!status)
+    goto BAIL;
+  
+  /* Convert into a C string */
+  status = CFStringGetCString( string,
+		buffer,
+		sizeof(buffer),
+		kCFStringEncodingISOLatin1);
+  if (!status)
+    goto BAIL;
+  
+  /* Finally link the buffer into the macosx_version struct. */
+  macosx_version.version = buffer;
+  
+BAIL:
+  // Release all of the CF objects we're responsible for.
+  if (fileURL)
+    CFRelease(fileURL);
+  if (resourceData)
+    CFRelease(resourceData);
+  if (propertyList)
+    CFRelease(propertyList);
+}
+
+void initDebSystem()
+{
+  finkinit();
+  (void)debSys;
+  init_deb2();
+  init_deb3();
+}
+/* FINK LOCAL end */
 
 debSystem debSys;
 
@@ -49,6 +154,8 @@
 debSystem::~debSystem()
 {
    delete StatusFile;
+   delete FinkStatusFile;
+   unlink(FINKSTATUSFILE);
 }
 									/*}}}*/
 // System::Lock - Get the lock						/*{{{*/
@@ -162,8 +269,8 @@
       which is yet to be determined. The functions in pkgcachegen should
       be the only users of these */
    Cnf.CndSet("Dir::State::userstatus","status.user"); // Defunct
-   Cnf.CndSet("Dir::State::status","/var/lib/dpkg/status");
-   Cnf.CndSet("Dir::Bin::dpkg","/usr/bin/dpkg");
+   Cnf.CndSet("Dir::State::status","@PREFIX@/var/lib/dpkg/status");
+   Cnf.CndSet("Dir::Bin::dpkg","@PREFIX@/bin/dpkg");
 
    if (StatusFile) {
      delete StatusFile;
@@ -191,9 +298,9 @@
 signed debSystem::Score(Configuration const &Cnf)
 {
    signed Score = 0;
-   if (FileExists(Cnf.FindFile("Dir::State::status","/var/lib/dpkg/status")) == true)
+   if (FileExists(Cnf.FindFile("Dir::State::status","@PREFIX@/var/lib/dpkg/status")) == true)
        Score += 10;
-   if (FileExists(Cnf.FindFile("Dir::Bin::dpkg","/usr/bin/dpkg")) == true)
+   if (FileExists(Cnf.FindFile("Dir::Bin::dpkg","@PREFIX@/bin/dpkg")) == true)
       Score += 10;
    if (FileExists("/etc/debian_version") == true)
       Score += 10;
@@ -208,6 +315,44 @@
    if (StatusFile == 0)
       StatusFile = new debStatusIndex(_config->FindFile("Dir::State::status"));
    List.push_back(StatusFile);
+/* FINK LOCAL begin */
+
+   if (FinkStatusFile == 0) {
+      struct stat unused_sbuf;
+      int sys_ok=0;
+      unlink(FINKSTATUSFILE);
+      if ( 0 == stat("@PREFIX@/bin/fink-virtual-pkgs",&unused_sbuf)) {
+          if ( 0 == system("@PREFIX@/bin/fink-virtual-pkgs --apt")) sys_ok=1;
+      }    
+      if (stat(FINKSTATUSFILE, &unused_sbuf) || !sys_ok) {
+	  std::ofstream finkstatus(FINKSTATUSFILE);
+      if(macosx_version.version != 0)
+      {
+        finkstatus << "Package: macosx" << endl;
+        finkstatus << "Status: install ok installed" << endl;      
+        finkstatus << "Priority: optional" << endl;
+        finkstatus << "Section: base" << endl;
+        finkstatus << "Maintainer: None" << endl;
+        finkstatus << "Source: macosx" << endl;
+        finkstatus << "Version: " << macosx_version.version << endl;
+        finkstatus << "Description: Pseudo package representing Mac OS X" << endl;
+        finkstatus << " Pseudo package representing Mac OS X" << endl << endl;
+      }
+      finkstatus << "Package: darwin" << endl;
+      finkstatus << "Status: install ok installed" << endl;
+      finkstatus << "Priority: optional" << endl;
+      finkstatus << "Section: base" << endl;
+      finkstatus << "Maintainer: None" << endl;
+      finkstatus << "Source: darwin" << endl;
+      finkstatus << "Version: " << darwin_version.version  << endl;
+      finkstatus << "Description: Pseudo package representing Darwin" << endl;
+      finkstatus << " Pseudo package representing Darwin" << endl << endl;
+      finkstatus.close();
+      }		
+      FinkStatusFile = new debStatusIndex(FINKSTATUSFILE);
+   }
+   List.push_back(FinkStatusFile);
+/* FINK LOCAL end */
    return true;
 }
 									/*}}}*/
@@ -223,6 +368,10 @@
    {
       Found = StatusFile;
       return true;
+   }  else if ((FinkStatusFile != 0) && (FinkStatusFile->FindInCache(*File.Cache()) == File))
+   {
+      Found = FinkStatusFile;
+      return true;
    }
    
    return false;
Index: apt-pkg/deb/debsystem.h
===================================================================
RCS file: /cvsroot/fink/3rdparty/apt/apt-pkg/deb/debsystem.h,v
retrieving revision 1.1.1.1
retrieving revision 1.3
diff -u -r1.1.1.1 -r1.3
--- apt-pkg/deb/debsystem.h	17 Mar 2006 01:58:11 -0000	1.1.1.1
+++ apt-pkg/deb/debsystem.h	17 Mar 2006 04:19:21 -0000	1.3
@@ -25,6 +25,7 @@
    bool CheckUpdates();
    
    debStatusIndex *StatusFile;
+   debStatusIndex *FinkStatusFile;
    
    public:
 
Index: apt-pkg/deb/debversion.cc
===================================================================
RCS file: /cvsroot/fink/3rdparty/apt/apt-pkg/deb/debversion.cc,v
retrieving revision 1.1.1.1
retrieving revision 1.3
diff -u -r1.1.1.1 -r1.3
--- apt-pkg/deb/debversion.cc	17 Mar 2006 01:58:11 -0000	1.1.1.1
+++ apt-pkg/deb/debversion.cc	17 Mar 2006 04:19:21 -0000	1.3
@@ -24,6 +24,11 @@
 
 debVersioningSystem debVS;
 
+void init_deb3()
+{
+  (void)debVS;
+}
+
 // debVS::debVersioningSystem - Constructor				/*{{{*/
 // ---------------------------------------------------------------------
 /* */
Index: apt-pkg/deb/dpkgpm.cc
===================================================================
RCS file: /cvsroot/fink/3rdparty/apt/apt-pkg/deb/dpkgpm.cc,v
retrieving revision 1.1.1.1
retrieving revision 1.3
diff -u -r1.1.1.1 -r1.3
--- apt-pkg/deb/dpkgpm.cc	17 Mar 2006 01:58:11 -0000	1.1.1.1
+++ apt-pkg/deb/dpkgpm.cc	17 Mar 2006 04:19:21 -0000	1.3
@@ -26,6 +26,10 @@
 #include <errno.h>
 #include <stdio.h>
 #include <sstream>
+
+#ifdef __APPLE__
+typedef sig_t sighandler_t;
+#endif
 #include <map>
 
 #include <config.h>
Index: buildlib/environment.mak.in
===================================================================
RCS file: /cvsroot/fink/3rdparty/apt/buildlib/environment.mak.in,v
retrieving revision 1.1.1.1
retrieving revision 1.2
diff -u -r1.1.1.1 -r1.2
--- buildlib/environment.mak.in	17 Mar 2006 01:58:12 -0000	1.1.1.1
+++ buildlib/environment.mak.in	17 Mar 2006 02:25:38 -0000	1.2
@@ -7,14 +7,14 @@
 CC = @CC@
 CPPFLAGS+= @CPPFLAGS@ @DEFS@ -D_REENTRANT -Wall
 CXX = @CXX@
-CXXFLAGS+= @CXXFLAGS@
+CXXFLAGS+= @CXXFLAGS@ -I@PREFIX@/include
 NUM_PROCS = @NUM_PROCS@
 GLIBC_VER = @GLIBC_VER@
 LIBSTDCPP_VER = @LIBSTDCPP_VER@
 
 # Linker stuff
-PICFLAGS+= -fPIC -DPIC
-LFLAGS+= @LDFLAGS@
+PICFLAGS+= -fno-common -DPIC
+LFLAGS+= @LDFLAGS@ -framework CoreFoundation
 LEFLAGS+= 
 SOCKETLIBS:= @SOCKETLIBS@
 AR:=@AR@
@@ -62,11 +62,12 @@
 
 # Shared library things
 HOST_OS = @host_os@
-ifneq ($(words $(filter linux-gnu gnu% %gnu,$(HOST_OS))),0)
-   SONAME_MAGIC=-Wl,-soname -Wl,
-   LFLAGS_SO=
-else
-   # Do not know how to create shared libraries here.
-   ONLYSTATICLIBS = yes
-endif
-	
+#ifneq ($(words $(filter linux-gnu gnu% %gnu,$(HOST_OS))),0)
+#   SONAME_MAGIC=-Wl,-soname -Wl,
+#   LFLAGS_SO=
+#else
+#   # Do not know how to create shared libraries here.
+#   ONLYSTATICLIBS = yes
+#endif
+SONAME_MAGIC=-install_name @PREFIX@/lib/
+LFLAGS_SO=-dynamiclib
Index: buildlib/library.mak
===================================================================
RCS file: /cvsroot/fink/3rdparty/apt/buildlib/library.mak,v
retrieving revision 1.1.1.1
retrieving revision 1.2
diff -u -r1.1.1.1 -r1.2
--- buildlib/library.mak	17 Mar 2006 01:58:12 -0000	1.1.1.1
+++ buildlib/library.mak	17 Mar 2006 02:25:38 -0000	1.2
@@ -16,11 +16,11 @@
 # See defaults.mak for information about LOCAL
 
 # Some local definitions
-LOCAL := lib$(LIBRARY)$(LIBEXT).so.$(MAJOR).$(MINOR)
+LOCAL := lib$(LIBRARY)$(LIBEXT).$(MAJOR).$(MINOR).dylib
 $(LOCAL)-OBJS := $(addprefix $(OBJ)/,$(addsuffix .opic,$(notdir $(basename $(SOURCE)))))
 $(LOCAL)-DEP := $(addprefix $(DEP)/,$(addsuffix .opic.d,$(notdir $(basename $(SOURCE)))))
 $(LOCAL)-HEADERS := $(addprefix $(INCLUDE)/,$(HEADERS))
-$(LOCAL)-SONAME := lib$(LIBRARY)$(LIBEXT).so.$(MAJOR)
+$(LOCAL)-SONAME := lib$(LIBRARY)$(LIBEXT).$(COMPAT).dylib
 $(LOCAL)-SLIBS := $(SLIBS)
 $(LOCAL)-LIBRARY := $(LIBRARY)
 
@@ -29,7 +29,7 @@
 
 # Install the command hooks
 headers: $($(LOCAL)-HEADERS)
-library: $(LIB)/lib$(LIBRARY).so $(LIB)/lib$(LIBRARY)$(LIBEXT).so.$(MAJOR)
+library: $(LIB)/lib$(LIBRARY).dylib $(LIB)/lib$(LIBRARY)$(LIBEXT).$(MAJOR).dylib
 clean: clean/$(LOCAL)
 veryclean: veryclean/$(LOCAL)
 
@@ -41,21 +41,23 @@
 clean/$(LOCAL):
 	-rm -f $($(@F)-OBJS) $($(@F)-DEP)
 veryclean/$(LOCAL): clean/$(LOCAL)
-	-rm -f $($(@F)-HEADERS) $(LIB)/lib$($(@F)-LIBRARY)*.so*
+	-rm -f $($(@F)-HEADERS) $(LIB)/lib$($(@F)-LIBRARY)*.dylib
 
 # Build rules for the two symlinks
-.PHONY: $(LIB)/lib$(LIBRARY)$(LIBEXT).so.$(MAJOR) $(LIB)/lib$(LIBRARY).so
-$(LIB)/lib$(LIBRARY)$(LIBEXT).so.$(MAJOR): $(LIB)/lib$(LIBRARY)$(LIBEXT).so.$(MAJOR).$(MINOR)
+.PHONY: $(LIB)/lib$(LIBRARY)$(LIBEXT).$(MAJOR).dylib $(LIB)/lib$(LIBRARY).dylib
+$(LIB)/lib$(LIBRARY)$(LIBEXT).$(MAJOR).dylib: $(LIB)/lib$(LIBRARY)$(LIBEXT).$(MAJOR).$(MINOR).dylib
 	ln -sf $(<F) $@
-$(LIB)/lib$(LIBRARY).so: $(LIB)/lib$(LIBRARY)$(LIBEXT).so.$(MAJOR).$(MINOR)
+$(LIB)/lib$(LIBRARY).dylib: $(LIB)/lib$(LIBRARY)$(LIBEXT).$(MAJOR).$(MINOR).dylib
 	ln -sf $(<F) $@
 
 # The binary build rule
-$(LIB)/lib$(LIBRARY)$(LIBEXT).so.$(MAJOR).$(MINOR): $($(LOCAL)-HEADERS) $($(LOCAL)-OBJS)
-	-rm -f $(LIB)/lib$($(@F)-LIBRARY)*.so* 2> /dev/null
+$(LIB)/lib$(LIBRARY)$(LIBEXT).$(MAJOR).$(MINOR).dylib: $($(LOCAL)-HEADERS) $($(LOCAL)-OBJS)
+	-rm -f $(LIB)/lib$($(@F)-LIBRARY)*.dylib 2> /dev/null
 	echo Building shared library $@
 	$(CXX) $(CXXFLAGS) $(LDFLAGS) $(PICFLAGS) $(LFLAGS) $(LFLAGS_SO)\
-	   -o $@ $(SONAME_MAGIC)$($(@F)-SONAME) -shared \
+	   -o $@ $(SONAME_MAGIC)$($(@F)-SONAME) \
+	   -compatibility_version $(COMPAT) \
+	   -current_version $(MAJOR).$(MINOR) \
 	   $(filter %.opic,$^) \
 	   $($(@F)-SLIBS) 
 
Index: buildlib/ostable
===================================================================
RCS file: /cvsroot/fink/3rdparty/apt/buildlib/ostable,v
retrieving revision 1.1.1.1
retrieving revision 1.2
diff -u -r1.1.1.1 -r1.2
--- buildlib/ostable	17 Mar 2006 01:58:12 -0000	1.1.1.1
+++ buildlib/ostable	17 Mar 2006 02:25:38 -0000	1.2
@@ -16,6 +16,7 @@
 hp-hpux[^-]*	    hp-ux
 sun-solaris[^-]*    solaris
 [^-]*-openbsd[^-]*  openbsd
+[^-]*-darwin[^-]*   darwin
 
 # Catch all
 .*	unknown
Index: buildlib/sgml_manpage.mak
===================================================================
RCS file: /cvsroot/fink/3rdparty/apt/buildlib/sgml_manpage.mak,v
retrieving revision 1.1.1.1
retrieving revision 1.2
diff -u -r1.1.1.1 -r1.2
--- buildlib/sgml_manpage.mak	17 Mar 2006 01:58:12 -0000	1.1.1.1
+++ buildlib/sgml_manpage.mak	17 Mar 2006 02:25:38 -0000	1.2
@@ -11,7 +11,7 @@
 # See defaults.mak for information about LOCAL
 
 # Some local definitions
-ifdef DOCBOOK2MAN
+ifdef DOCBOOK2MANx
 
 LOCAL := sgml-manpage-$(firstword $(SOURCE))
 $(LOCAL)-LIST := $(SOURCE)
Index: cmdline/apt-get.cc
===================================================================
RCS file: /cvsroot/fink/3rdparty/apt/cmdline/apt-get.cc,v
retrieving revision 1.1.1.1
retrieving revision 1.3
diff -u -r1.1.1.1 -r1.3
--- cmdline/apt-get.cc	17 Mar 2006 01:58:13 -0000	1.1.1.1
+++ cmdline/apt-get.cc	17 Mar 2006 04:19:21 -0000	1.3
@@ -53,7 +53,7 @@
 #include <termios.h>
 #include <sys/ioctl.h>
 #include <sys/stat.h>
-#include <sys/statvfs.h>
+#include <sys/mount.h>
 #include <signal.h>
 #include <unistd.h>
 #include <stdio.h>
@@ -115,6 +115,8 @@
 /* Returns true on a Yes.*/
 bool YnPrompt(bool Default=true)
 {
+   fflush(NULL);
+   
    if (_config->FindB("APT::Get::Assume-Yes",false) == true)
    {
       c1out << _("Y") << endl;
@@ -333,7 +335,20 @@
 		  if (Cache[Targ].CandidateVerIter(Cache).end() == true)
 		  {
 		     if (Targ->ProvidesList == 0)
-			out << _("but it is not installable");
+		     {
+			out << _("but it is not installable. For Fink users, ");
+			out << _("this often means that you have attempted ");
+			out << _("to install a package from the binary distribution ");
+			out << _("that depends on a package (or has a subdependency) ");
+			out << _("that is not part of the binary distribution.  The ");
+			out << _("omitted package may be a virtual package, which ");
+			out << _("must be installed manually instead of with fink, ");
+			out << _("or one that has a \"Restrictive\" license that ");
+			out << _("prohibits binary distribution.  See:\n");
+			out << _("  <http://fink.sourceforge.net/faq/usage-fink.php#bindist>\n");
+			out << _("  <http://fink.sourceforge.net/doc/users-guide/packages.php#bin-exceptions>\n");
+			out << _("  <http://fink.sourceforge.net/doc/usage-general.php#virtpackage>\n");
+		     }
 		     else
 			out << _("but it is a virtual package");
 		  }		  
@@ -640,13 +655,15 @@
       return false;
    
    // Nothing is broken
-   if (DCache->BrokenCount() == 0 || AllowBroken == true)
+   // FINK LOCAL added APT::Get::Ignore-Breakage test
+   if (DCache->BrokenCount() == 0 || AllowBroken == true
+       || _config->FindB("APT::Get::Ignore-Breakage") == true)
       return true;
 
    // Attempt to fix broken things
    if (_config->FindB("APT::Get::Fix-Broken",false) == true)
    {
-      c1out << _("Correcting dependencies...") << flush;
+      c1out << _("Correcting dependencies...") << flush, fflush(NULL);
       if (pkgFixBroken(*DCache) == false || DCache->BrokenCount() != 0)
       {
 	 c1out << _(" failed.") << endl;
@@ -749,7 +766,9 @@
    Stats(c1out,Cache);
 
    // Sanity check
-   if (Cache->BrokenCount() != 0)
+   // FINK LOCAL added APT::Get::Ignore-Breakage test
+   if (Cache->BrokenCount() != 0
+       && _config->FindB("APT::Get::Ignore-Breakage",false) == false)
    {
       ShowBroken(c1out,Cache,false);
       return _error->Error(_("Internal error, InstallPackages was called with broken packages!"));
@@ -840,10 +859,10 @@
    if (_config->FindB("APT::Get::Print-URIs") == false &&
        _config->FindB("APT::Get::Download",true) == true)
    {
-      struct statvfs Buf;
+      struct statfs Buf;
       string OutputDir = _config->FindDir("Dir::Cache::Archives");
-      if (statvfs(OutputDir.c_str(),&Buf) != 0)
-	 return _error->Errno("statvfs",_("Couldn't determine free space in %s"),
+      if (statfs(OutputDir.c_str(),&Buf) != 0)
+	 return _error->Errno("statfs",_("Couldn't determine free space in %s"),
 			      OutputDir.c_str());
       if (unsigned(Buf.f_bfree) < (FetchBytes - FetchPBytes)/Buf.f_bsize)
 	 return _error->Error(_("You don't have enough free space in %s."),
@@ -868,7 +887,7 @@
 	       _("You are about to do something potentially harmful.\n"
 		 "To continue type in the phrase '%s'\n"
 		 " ?] "),Prompt);
-      c2out << flush;
+      c2out << flush, fflush(NULL);
       if (AnalPrompt(Prompt) == false)
       {
 	 c2out << _("Abort.") << endl;
@@ -1146,7 +1165,9 @@
       ExpectedInst++;
    
    // Install it with autoinstalling enabled.
-   if (State.InstBroken() == true && BrokenFix == false)
+   // FINK LOCAL added APT::Get::Ignore-Breakage test
+   if (State.InstBroken() == true && BrokenFix == false
+      && _config->FindB("APT::Get::Ignore-Breakage") == false)
       Cache.MarkInstall(Pkg,true);
    return true;
 }
@@ -1541,7 +1562,9 @@
    /* If we are in the Broken fixing mode we do not attempt to fix the
       problems. This is if the user invoked install without -f and gave
       packages */
-   if (BrokenFix == true && Cache->BrokenCount() != 0)
+   // FINK LOCAL added APT::Get::Ignore-Breakage test
+   if (BrokenFix == true && Cache->BrokenCount() != 0
+       && _config->FindB("APT::Get::Ignore-Breakage") == false)
    {
       c1out << _("You might want to run `apt-get -f install' to correct these:") << endl;
       ShowBroken(c1out,Cache,false);
@@ -1551,11 +1574,13 @@
    
    // Call the scored problem resolver
    Fix.InstallProtect();
-   if (Fix.Resolve(true) == false)
+   // FINK LOCAL added APT::Get::Ignore-Breakage test
+   if (_config->FindB("APT::Get::Ignore-Breakage") == false && Fix.Resolve(true) == false)
       _error->Discard();
 
    // Now we check the state of the packages,
-   if (Cache->BrokenCount() != 0)
+   // FINK LOCAL added APT::Get::Ignore-Breakage test
+   if (Cache->BrokenCount() != 0 && _config->FindB("APT::Get::Ignore-Breakage") == false)
    {
       c1out << 
        _("Some packages could not be installed. This may mean that you have\n" 
@@ -1965,10 +1990,10 @@
    double DebBytes = Fetcher.TotalNeeded();
 
    // Check for enough free space
-   struct statvfs Buf;
+   struct statfs Buf;
    string OutputDir = ".";
-   if (statvfs(OutputDir.c_str(),&Buf) != 0)
-      return _error->Errno("statvfs",_("Couldn't determine free space in %s"),
+   if (statfs(OutputDir.c_str(),&Buf) != 0)
+      return _error->Errno("statfs",_("Couldn't determine free space in %s"),
 			   OutputDir.c_str());
    if (unsigned(Buf.f_bfree) < (FetchBytes - FetchPBytes)/Buf.f_bsize)
       return _error->Error(_("You don't have enough free space in %s"),
@@ -2464,6 +2489,8 @@
    _config->Set("APT::Get::Simulate",false);
    _config->Set("APT::Get::Assume-Yes",false);
    _config->Set("APT::Get::Fix-Broken",false);
+   // FINK LOCAL added APT::Get::Ignore-Breakage
+   _config->Set("APT::Get::Ignore-Breakage",false);
    _config->Set("APT::Get::Force-Yes",false);
    _config->Set("APT::Get::List-Cleanup",true);
 }
@@ -2502,6 +2529,8 @@
       {'y',"yes","APT::Get::Assume-Yes",0},
       {'y',"assume-yes","APT::Get::Assume-Yes",0},      
       {'f',"fix-broken","APT::Get::Fix-Broken",0},
+      // FINK LOCAL added APT::Get::Ignore-Breakage
+      {0,"ignore-breakage","APT::Get::Ignore-Breakage",0},
       {'u',"show-upgraded","APT::Get::Show-Upgraded",0},
       {'m',"ignore-missing","APT::Get::Fix-Missing",0},
       {'t',"target-release","APT::Default-Release",CommandLine::HasArg},
@@ -2565,6 +2594,22 @@
       ShowHelp(CmdL);
       return 0;
    }
+
+   /* FINK LOCAL begin */
+   if (_config->FindB("APT::Get::Ignore-Breakage",false) == true) {
+     if (_config->FindB("APT::Get::Print-URIs",false) == false &&
+	 _config->FindB("APT::Get::Download-Only",false) == false) {
+       _error->Error("--ignore-breakage can only be used with --print-uris or --download-only");
+       _error->DumpErrors();
+       return 100;
+     }
+     if (strcmp(CmdL.FileList[0],"install") != 0) {
+       _error->Error("--ignore-breakage can only be used with apt-get install");
+       _error->DumpErrors();
+       return 100;
+     }
+   }
+   /* FINK LOCAL end */
    
    // Deal with stdout not being a tty
    if (!isatty(STDOUT_FILENO) && _config->FindI("quiet",0) < 1)
Index: cmdline/makefile
===================================================================
RCS file: /cvsroot/fink/3rdparty/apt/cmdline/makefile,v
retrieving revision 1.1.1.1
retrieving revision 1.2
diff -u -r1.1.1.1 -r1.2
--- cmdline/makefile	17 Mar 2006 01:58:13 -0000	1.1.1.1
+++ cmdline/makefile	17 Mar 2006 02:25:39 -0000	1.2
@@ -7,42 +7,42 @@
 
 # The apt-cache program
 PROGRAM=apt-cache
-SLIBS = -lapt-pkg
+SLIBS = $(INTLLIBS) -lapt-pkg
 LIB_MAKES = apt-pkg/makefile
 SOURCE = apt-cache.cc
 include $(PROGRAM_H)
 
 # The apt-get program
 PROGRAM=apt-get
-SLIBS = -lapt-pkg
+SLIBS = $(INTLLIBS) -lapt-pkg
 LIB_MAKES = apt-pkg/makefile
 SOURCE = apt-get.cc acqprogress.cc
 include $(PROGRAM_H)
 
 # The apt-config program
 PROGRAM=apt-config
-SLIBS = -lapt-pkg
+SLIBS = $(INTLLIBS) -lapt-pkg
 LIB_MAKES = apt-pkg/makefile
 SOURCE = apt-config.cc
 include $(PROGRAM_H)
 
 # The apt-cdrom program
 PROGRAM=apt-cdrom
-SLIBS = -lapt-pkg
+SLIBS = $(INTLLIBS) -lapt-pkg
 LIB_MAKES = apt-pkg/makefile
 SOURCE = apt-cdrom.cc 
 include $(PROGRAM_H)
 
 # The apt-sortpkgs program
 PROGRAM=apt-sortpkgs
-SLIBS = -lapt-pkg
+SLIBS = $(INTLLIBS) -lapt-pkg
 LIB_MAKES = apt-pkg/makefile
 SOURCE = apt-sortpkgs.cc
 include $(PROGRAM_H)
 
 # The apt-extracttemplates program
 PROGRAM=apt-extracttemplates
-SLIBS = -lapt-pkg -lapt-inst
+SLIBS = $(INTLLIBS) -lapt-pkg -lapt-inst
 LIB_MAKES = apt-pkg/makefile
 SOURCE = apt-extracttemplates.cc 
 include $(PROGRAM_H)
Index: dselect/install
===================================================================
RCS file: /cvsroot/fink/3rdparty/apt/dselect/install,v
retrieving revision 1.1.1.1
retrieving revision 1.2
diff -u -r1.1.1.1 -r1.2
--- dselect/install	17 Mar 2006 01:58:22 -0000	1.1.1.1
+++ dselect/install	17 Mar 2006 02:25:41 -0000	1.2
@@ -6,8 +6,8 @@
 # Get the configuration from /etc/apt/apt.conf
 CLEAN="prompt"
 OPTS="-f"
-APTGET="/usr/bin/apt-get"
-DPKG="/usr/bin/dpkg"
+APTGET="@PREFIX@/bin/apt-get"
+DPKG="@PREFIX@/bin/dpkg"
 DPKG_OPTS="--admindir=$1"
 APT_OPT0="-oDir::State::status=$1/status"
 APT_OPT1="-oDPkg::Options::=$DPKG_OPTS"
Index: dselect/setup
===================================================================
RCS file: /cvsroot/fink/3rdparty/apt/dselect/setup,v
retrieving revision 1.1.1.1
retrieving revision 1.2
diff -u -r1.1.1.1 -r1.2
--- dselect/setup	17 Mar 2006 01:58:22 -0000	1.1.1.1
+++ dselect/setup	17 Mar 2006 02:25:41 -0000	1.2
@@ -23,15 +23,17 @@
 my $vardir=$ARGV[0];
 my $method=$ARGV[1];
 my $option=$ARGV[2];
-my $config_file = '/etc/apt/sources.list';
+my $config_file = '@PREFIX@/etc/apt/sources.list';
 
-my $boldon=`setterm -bold on`;
-my $boldoff=`setterm -bold off`;
+my $boldon=`setterm -bold on 2>/dev/null`;
+my $boldoff=`setterm -bold off 2>/dev/null`;
+$boldon = "" unless defined $boldon;
+$boldoff = "" unless defined $boldon;
 
 my @known_types           = ('deb');
 my @known_access         = ('http', 'ftp', 'file');
-my @typical_distributions = ('stable', 'unstable', 'testing', 'non-US');
-my @typical_components    = ('main', 'contrib', 'non-free');
+my @typical_distributions = ('@DIST@/release', '@DIST@/current');
+my @typical_components    = ('main', 'crypto');
 
 my %known_access           = map {($_,$_)} @known_access;
 my %typical_distributions  = map {($_,$_)} @typical_distributions;
@@ -118,9 +120,9 @@
   }
 
   $type         = 'deb';
-  $urn          = "http://http.us.debian.org/debian" unless $urn;
-  $distribution = "stable" unless $distribution;
-  $components   = "main contrib non-free" unless $components;
+  $urn          = "http://us.dl.sourceforge.net/fink/direct_download" unless $urn;
+  $distribution = "@DIST@/release" unless $distribution;
+  $components   = "main" unless $components;
 
     
   $rec->{'Type'} = 'deb';
@@ -222,19 +224,13 @@
   print "\t$boldon Set up a list of distribution source locations $boldoff \n";
   print "\n";
 
-  print " Please give the base URL of the debian distribution.\n";
+  print " Please give the base URL of the Fink distribution.\n";
   print " The access schemes I know about are:$boldon ";
   print join (' ', @known_access), "$boldoff\n";
-#  print " The mirror scheme is special  that it does not specify the\n";
-#  print " location of a debian archive but specifies the location\n";
-#  print " of a list of mirrors to use to access the archive.\n";
   print "\n";
   print " For example:\n";
-  print "              file:/mnt/debian,\n";
-  print "              ftp://ftp.debian.org/debian,\n";
-  print "              http://ftp.de.debian.org/debian,\n";
-#  print " and the special mirror scheme,\n";
-#  print "              mirror:http://www.debian.org/archivemirrors \n";
+  print "              file:@PREFIX@/fink,\n";
+  print "              http://us.dl.sourceforge.net/fink/direct_download\n";
   print "\n";
 
   my $index = 0;
@@ -269,7 +265,11 @@
     print "-" x 72, "\n";
     &print_config('Config' => \@Oldconfig);
     print "-" x 72, "\n";
-    print "$boldon Do you wish to overwrite it? [y/N]$boldoff ";
+    print "$boldon In most cases, this file was installed by Fink or by apt,"
+        ." and$boldoff\n";
+    print "$boldon should NOT be changed.  " .
+        "Do you wish to change (overwrite) it?[y/N]$boldoff ";
+
     my $answer = <STDIN>;
     chomp ($answer);
     $answer =~ s/\s+/ /og;
Index: dselect/update
===================================================================
RCS file: /cvsroot/fink/3rdparty/apt/dselect/update,v
retrieving revision 1.1.1.1
retrieving revision 1.2
diff -u -r1.1.1.1 -r1.2
--- dselect/update	17 Mar 2006 01:58:22 -0000	1.1.1.1
+++ dselect/update	17 Mar 2006 02:25:41 -0000	1.2
@@ -7,13 +7,13 @@
 # Get the configuration from /etc/apt/apt.conf
 CLEAN="prompt"
 OPTS="-f"
-APTGET="/usr/bin/apt-get"
-APTCACHE="/usr/bin/apt-cache"
-DPKG="/usr/bin/dpkg"
+APTGET="@PREFIX@/bin/apt-get"
+APTCACHE="@PREFIX@/bin/apt-cache"
+DPKG="@PREFIX@/bin/dpkg"
 DPKG_OPTS="--admindir=$1"
 APT_OPT0="-oDir::State::status=$1/status"
 APT_OPT1="-oDPkg::Options::=$DPKG_OPTS"
-CACHEDIR="/var/cache/apt"
+CACHEDIR="@PREFIX@/var/cache/apt"
 PROMPT="false"
 RES=`apt-config shell CLEAN DSelect::Clean OPTS DSelect::UpdateOptions \
 		      DPKG Dir::Bin::dpkg/f APTGET Dir::Bin::apt-get/f \
Index: fink/apt.info
===================================================================
RCS file: fink/apt.info
diff -N fink/apt.info
--- /dev/null	1 Jan 1970 00:00:00 -0000
+++ fink/apt.info	17 Mar 2006 04:06:40 -0000	1.2
@@ -0,0 +1,155 @@
+Package: apt
+Version: 0.6.43.3
+Revision: 3
+GCC: 3.3
+Depends: libapt3-shlibs (>= %v-%r), gzip, libftw0-shlibs
+BuildDepends: fink-prebinding, libftw0, libiconv-dev, gettext-dev
+#Source: mirror:sourceforge:fink/%n_%v.tar.gz
+Source: http://debian.ethz.ch/debian/pool/main/a/%n/%n_%v.tar.gz
+Source-MD5: 7d607342ffce8ba8705ae7797cb1a15c
+SourceDirectory: %n-%v
+PatchScript: <<
+#!/bin/sh -ex
+	DISTRIBUTION=`grep -i -E '^Distribution:' %p/etc/fink.conf | awk '{ print $2 }'`
+	sed -e 's|@PREFIX@|%p|g' -e "s|@DIST@|${DISTRIBUTION}|g" <%a/%n.patch | patch -p0
+	sh patch_flush
+	for i in `grep -rl '#ifdef __GNUG__' .` ; do perl -pi.bak -e 's/#ifdef __GNUG__/#if defined(__GNUG__) && !defined(__APPLE_CC__)/' $i; done
+	find doc -type f | xargs perl -pi -e 's,/usr,%p,g; s,/var,%p/var,g; s,/etc/(?!fstab),\@PREFIX\@/etc/$1,g;'
+<<
+NoSetCPPFLAGS: true
+SetLIBS: -L%p/lib -undefined dynamic_lookup
+NoSetLDFLAGS: true
+SetCXXFLAGS: -O2 -DEMULATE_MMAP -D__USE_MISC -fconstant-cfstrings -g
+CompileScript: <<
+#!/bin/sh -ex
+
+	export CPPFLAGS="-I%p/include/ftw"
+	./configure %c
+	make -f makefile.wrap NOISY=1 GLIBC_VER= LIBSTDCPP_VER= INTLLIBS="-L%p/lib -lintl"
+<<
+InstallScript: <<
+#!/bin/sh -ex
+
+	install -d -m 755 %i/bin
+	install -d -m 755 %i/lib
+	install -d -m 755 %i/lib/apt/methods
+	install -d -m 755 %i/lib/dpkg/methods/apt
+	install -d -m 755 %i/include/apt-pkg
+	install -d -m 755 %i/share/man/man{1,5,8}
+	install -d -m 755 %i/etc/apt/apt.conf.d
+	install -d -m 755 %i/var/cache/apt/archives/partial
+	install -d -m 755 %i/var/lib/apt/lists/partial
+
+	# binaries
+	install -c -m 755 bin/apt-* %i/bin/
+
+	# libraries
+	install -c -m 644 bin/libapt-pkg.3.11.0.dylib %i/lib/
+	pushd %i/lib/ && ln -s libapt-pkg.3.11.0.dylib libapt-pkg.3.dylib && ln -s libapt-pkg.3.11.0.dylib libapt-pkg.dylib && popd
+	install -c -m 644 bin/libapt-inst.1.1.0.dylib %i/lib/
+	pushd %i/lib/ && ln -s libapt-inst.1.1.0.dylib libapt-inst.1.dylib && ln -s libapt-inst.1.1.0.dylib libapt-inst.dylib && popd
+
+	# methods
+	install -c -m 755 bin/methods/* %i/lib/apt/methods/
+	install -c -m 755 scripts/dselect/* %i/lib/dpkg/methods/apt/
+
+	# man pages
+	for ext in 1 5 8; do
+		for file in doc/*.$ext docs/*.$ext; do
+			install -c -m 644 $file %i/share/man/man$ext/
+		done
+	done
+
+	# headers
+	install -c -m 644 include/apt-pkg/*.h %i/include/apt-pkg/
+<<
+SplitOff: <<
+	Package: libapt3-shlibs
+	Depends: dpkg, fink (>= 0.20.6-1), gettext
+	Replaces: apt-shlibs
+	Files: lib/lib*.3*.dylib
+	Shlibs: <<
+		%p/lib/libapt-pkg.3.dylib 3.0.0 libapt3-shlibs (>= 0.6.40.1-1)
+		%p/lib/libapt-inst.3.dylib 3.0.0 libapt3-shlibs (>= 0.6.40.1-1)
+	<<
+	DocFiles: COPYING* AUTHORS
+<<
+SplitOff2: <<
+	Package: libapt3-dev
+	Depends: libapt3-shlibs (>= %v-%r)
+	BuildDependsOnly: true
+	Recommends: libiconv-dev, libgettext3-dev
+	Conflicts: apt-dev
+	Replaces: apt-dev
+	Files: lib/*.dylib include
+	DocFiles: COPYING* AUTHORS
+<<
+DocFiles: COPYING* AUTHORS
+PostInstScript: <<
+if [ ! -f %p/var/lib/dpkg/cmethopt ]; then
+	echo "apt apt" >%p/var/lib/dpkg/cmethopt
+	chmod 644 %p/var/lib/dpkg/cmethopt
+else
+	read a b <%p/var/lib/dpkg/cmethopt
+	if [ "$a" != "apt" -o "$b" != "apt" ]; then
+		echo
+		echo "dselect is not set up to use apt as its access method. Downloading binary"
+		echo -n "package will likely not work. Do you want to use apt instead?"
+		read answer
+		answer=`echo $answer | sed 's/^[yY].*$/y/'`
+		if [ -z "$answer" -o "x$answer" = "xy" ]; then
+			echo "apt apt" >%p/var/lib/dpkg/cmethopt
+			chmod 644 %p/var/lib/dpkg/cmethopt
+		fi
+	fi
+fi
+
+rm -f %p/var/cache/apt/pkgcache.bin %p/var/cache/apt/srcpkgcache.bin
+<<
+#
+Description: Advanced front-end for dpkg
+DescPort: <<
+There are three troublemakers when porting apt:
+- It was written for Linux/ELF/glibc.
+- It was written to maintain a distribution that is in charge of the
+  system and hardcodes paths like /usr/bin, /usr/lib and /var/lib.
+- There is no install target in the Makefiles because the Debian
+  packaging files take care of what goes where.
+There's also the usual trouble like not recognizing Darwin and
+depending on a case-sensitive file system. All of this amounts to a
+big, bad patch.
+
+The patch also fixes some potential and some real crashing bugs.
+
+Oh, one more thing: mmap() is broken for non-trivial uses in OS X
+10.1. Luckily apt uses a wrapper class that can be equipped with a
+workaround (malloc() + read() + write()...).
+
+The -fno-rtti works around a bug in GCC 3.1 on Jaguar.
+
+Ben Hines - added patch to make apt recognize macosx and darwin pkgs
+
+The sources.list file is now supplied by the fink program rather than
+by apt-get.
+
+The type of the fifth parameter for getsocklen keeps changing. The
+original code uses unsigned int, which was never correct for darwin,
+so we patch several choices. Through OS X 10.3, it was int; starting
+in 10.4 it is socklen_t, a symbol that first appeared in 10.3 but was
+an int. So starting in 10.3 we can just use socklen_t.
+See: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=323716
+
+dmacks - Implemented "--ignore-breakage" cmdline flag to 'apt-get install'
+(first appears in: 10.2-gcc3.3/0.5.4-41, 10.3/0.5.4-51)
+
+rangerrick -- Updated everything to 0.6.41
+<<
+DescPackaging: <<
+Previous versions by Christoph Pfisterer.
+
+We don't do anything with docbook2man even though it's detected,
+we just patch and install the man pages directly.
+<<
+License: GPL
+Maintainer: Fink Core Group <fink-core@lists.sourceforge.net>
+Homepage: http://packages.qa.debian.org/a/apt.html
Index: ftparchive/contents.cc
===================================================================
RCS file: /cvsroot/fink/3rdparty/apt/ftparchive/contents.cc,v
retrieving revision 1.1.1.1
retrieving revision 1.3
diff -u -r1.1.1.1 -r1.3
--- ftparchive/contents.cc	17 Mar 2006 01:58:22 -0000	1.1.1.1
+++ ftparchive/contents.cc	17 Mar 2006 04:19:22 -0000	1.3
@@ -41,7 +41,6 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
-#include <malloc.h>    
 									/*}}}*/
 
 // GenContents::~GenContents - Free allocated memory			/*{{{*/
Index: ftparchive/makefile
===================================================================
RCS file: /cvsroot/fink/3rdparty/apt/ftparchive/makefile,v
retrieving revision 1.1.1.1
retrieving revision 1.2
diff -u -r1.1.1.1 -r1.2
--- ftparchive/makefile	17 Mar 2006 01:58:22 -0000	1.1.1.1
+++ ftparchive/makefile	17 Mar 2006 02:25:41 -0000	1.2
@@ -8,7 +8,7 @@
 # The apt-ftparchive program
 ifdef BDBLIB
 PROGRAM=apt-ftparchive
-SLIBS = -lapt-pkg -lapt-inst $(BDBLIB)
+SLIBS = -lapt-pkg -lapt-inst $(BDBLIB) $(INTLLIBS) -lftw
 LIB_MAKES = apt-pkg/makefile apt-inst/makefile
 SOURCE = apt-ftparchive.cc cachedb.cc writer.cc contents.cc override.cc \
          multicompress.cc
Index: methods/makefile
===================================================================
RCS file: /cvsroot/fink/3rdparty/apt/methods/makefile,v
retrieving revision 1.1.1.1
retrieving revision 1.2
diff -u -r1.1.1.1 -r1.2
--- methods/makefile	17 Mar 2006 01:58:23 -0000	1.1.1.1
+++ methods/makefile	17 Mar 2006 02:25:41 -0000	1.2
@@ -7,61 +7,62 @@
 BIN := $(BIN)/methods
 
 # FIXME..
+COMPAT=3
 LIB_APT_PKG_MAJOR = 3.11
 APT_DOMAIN := libapt-pkg$(LIB_APT_PKG_MAJOR)
 
 # The file method
 PROGRAM=file
-SLIBS = -lapt-pkg 
+SLIBS = $(INTLLIBS) -lapt-pkg 
 LIB_MAKES = apt-pkg/makefile
 SOURCE = file.cc
 include $(PROGRAM_H)
 
 # The copy method
 PROGRAM=copy
-SLIBS = -lapt-pkg 
+SLIBS = $(INTLLIBS) -lapt-pkg 
 LIB_MAKES = apt-pkg/makefile
 SOURCE = copy.cc
 include $(PROGRAM_H)
 
 # The gzip method
 PROGRAM=gzip
-SLIBS = -lapt-pkg 
+SLIBS = $(INTLLIBS) -lapt-pkg 
 LIB_MAKES = apt-pkg/makefile
 SOURCE = gzip.cc
 include $(PROGRAM_H)
 
 # The gpgv method
 PROGRAM=gpgv
-SLIBS = -lapt-pkg
+SLIBS = $(INTLLIBS) -lapt-pkg
 LIB_MAKES = apt-pkg/makefile
 SOURCE = gpgv.cc
 include $(PROGRAM_H)
 
 # The cdrom method
 PROGRAM=cdrom
-SLIBS = -lapt-pkg 
+SLIBS = $(INTLLIBS) -lapt-pkg 
 LIB_MAKES = apt-pkg/makefile
 SOURCE = cdrom.cc
 include $(PROGRAM_H)
 
 # The http method
 PROGRAM=http
-SLIBS = -lapt-pkg $(SOCKETLIBS)
+SLIBS = $(INTLLIBS) -lapt-pkg $(SOCKETLIBS)
 LIB_MAKES = apt-pkg/makefile
 SOURCE = http.cc rfc2553emu.cc connect.cc
 include $(PROGRAM_H)
 
 # The ftp method
 PROGRAM=ftp
-SLIBS = -lapt-pkg $(SOCKETLIBS)
+SLIBS = $(INTLLIBS) -lapt-pkg $(SOCKETLIBS)
 LIB_MAKES = apt-pkg/makefile
 SOURCE = ftp.cc rfc2553emu.cc connect.cc
 include $(PROGRAM_H)
 
 # The rsh method
 PROGRAM=rsh
-SLIBS = -lapt-pkg
+SLIBS = $(INTLLIBS) -lapt-pkg
 LIB_MAKES = apt-pkg/makefile
 SOURCE = rsh.cc
 include $(PROGRAM_H)
Index: methods/rfc2553emu.h
===================================================================
RCS file: /cvsroot/fink/3rdparty/apt/methods/rfc2553emu.h,v
retrieving revision 1.1.1.1
retrieving revision 1.3
diff -u -r1.1.1.1 -r1.3
--- methods/rfc2553emu.h	17 Mar 2006 01:58:23 -0000	1.1.1.1
+++ methods/rfc2553emu.h	17 Mar 2006 04:19:22 -0000	1.3
@@ -26,6 +26,11 @@
 #include <sys/types.h>
 #include <sys/socket.h>
 
+
+
+
+
+
 // Autosense getaddrinfo
 #if defined(AI_PASSIVE) && defined(EAI_NONAME)
 #define HAVE_GETADDRINFO
@@ -36,6 +41,8 @@
 #define HAVE_GETNAMEINFO
 #endif
 
+
+
 // getaddrinfo support?
 #ifndef HAVE_GETADDRINFO
   // Renamed to advoid type clashing.. (for debugging)
@@ -101,6 +108,9 @@
   #define NI_NAMEREQD (1<<3)
   #define NI_DATAGRAM (1<<4)
   #endif
+  #ifndef NI_DATAGRAM
+  #define NI_DATAGRAM NI_DGRAM
+  #endif
 
   #define sockaddr_storage sockaddr_in
 #endif
