diff -rc ../rxvt-unicode-2.4.orig/src/ptytty.C ./src/ptytty.C
*** ../rxvt-unicode-2.4.orig/src/ptytty.C	Sat Mar 27 00:25:06 2004
--- ./src/ptytty.C	Sat Mar 27 00:25:55 2004
***************
*** 55,60 ****
--- 55,62 ----
  # include <sys/stropts.h>      /* for I_PUSH */
  #endif
  
+ #include <util.h>
+ 
  #ifdef DEBUG_TTY
  # define D_TTY(x)		fprintf x ; fputc('\n', stderr) ; fflush(stderr)
  #else
diff -rc ../rxvt-unicode-2.4.orig/src/rxvtc.C ./src/rxvtc.C
*** ../rxvt-unicode-2.4.orig/src/rxvtc.C	Sat Mar 27 00:25:06 2004
--- ./src/rxvtc.C	Sat Mar 27 00:43:22 2004
***************
*** 7,12 ****
--- 7,13 ----
  #include <signal.h>
  #include <sys/socket.h>
  #include <sys/un.h>
+ #include <limits.h>
  
  struct client : rxvt_connection {
    client ();
