diff -Naur gnome-vfs-2.4.2-old/Makefile.in gnome-vfs-2.4.2/Makefile.in
--- gnome-vfs-2.4.2-old/Makefile.in	Tue Jan 27 13:29:31 2004
+++ gnome-vfs-2.4.2/Makefile.in	Tue Jan 27 14:07:56 2004
@@ -191,7 +191,7 @@
 
 SUBDIRS_GNOME = monikers doc schemas
 
-SUBDIRS =  	libgnomevfs					modules						schemas						devel-docs					doc						programs					test						monikers					po
+SUBDIRS =  	libgnomevfs					modules						schemas						devel-docs					doc						programs					monikers					po
 
 
 EXTRA_DIST =  	HACKING						TODO						autogen.sh					gnome-vfs.spec					gnome-vfs-2.0.pc.in				gnome-vfs-module-2.0.pc.in			intltool-extract.in				intltool-merge.in				intltool-update.in
diff -Naur gnome-vfs-2.4.2-old/configure gnome-vfs-2.4.2/configure
--- gnome-vfs-2.4.2-old/configure	Tue Jan 27 13:29:31 2004
+++ gnome-vfs-2.4.2/configure	Tue Jan 27 14:05:46 2004
@@ -22448,6 +22448,8 @@
 if eval "test \"\${$as_ac_var+set}\" = set"; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
+  ac_check_lib_save_LIBS=$LIBS
+  LIBS="-lintl  $LIBS"
   cat >conftest.$ac_ext <<_ACEOF
 /* confdefs.h.  */
 _ACEOF
diff -Naur gnome-vfs-2.4.2-old/libgnomevfs/gnome-vfs-configuration.c gnome-vfs-2.4.2/libgnomevfs/gnome-vfs-configuration.c
--- gnome-vfs-2.4.2-old/libgnomevfs/gnome-vfs-configuration.c	Tue Jan 27 13:29:31 2004
+++ gnome-vfs-2.4.2/libgnomevfs/gnome-vfs-configuration.c	Tue Jan 27 14:07:56 2004
@@ -24,6 +24,7 @@
 #include <config.h>
 #include "gnome-vfs-configuration.h"
 
+#include <sys/types.h>
 #include <dirent.h>
 #include <errno.h>
 #include <glib/ghash.h>
diff -Naur gnome-vfs-2.4.2-old/ltmain.sh gnome-vfs-2.4.2/ltmain.sh
--- gnome-vfs-2.4.2-old/ltmain.sh	Tue Jan 27 13:29:31 2004
+++ gnome-vfs-2.4.2/ltmain.sh	Tue Jan 27 14:07:56 2004
@@ -5145,10 +5145,10 @@
 
 # Directory that this library needs to be installed in:
 libdir='$install_libdir'"
-	  if test "$installed" = no && test "$need_relink" = yes; then
-	    $echo >> $output "\
-relink_command=\"$relink_command\""
-	  fi
+#	  if test "$installed" = no && test "$need_relink" = yes; then
+#	    $echo >> $output "\
+#relink_command=\"$relink_command\""
+#	  fi
 	done
       fi
 
diff -Naur gnome-vfs-2.4.2-old/modules/file-method.c gnome-vfs-2.4.2/modules/file-method.c
--- gnome-vfs-2.4.2-old/modules/file-method.c	Tue Jan 27 13:29:31 2004
+++ gnome-vfs-2.4.2/modules/file-method.c	Tue Jan 27 14:07:56 2004
@@ -36,6 +36,7 @@
 #include <libgnomevfs/gnome-vfs-utils.h>
 #include <libgnomevfs/gnome-vfs-mime.h>
 #include <libgnomevfs/gnome-vfs-monitor-private.h>
+#include <sys/types.h>
 #include <dirent.h>
 #include <errno.h>
 #include <fcntl.h>
@@ -46,7 +47,6 @@
 #include <stdlib.h>
 #include <string.h>
 #include <sys/stat.h>
-#include <sys/types.h>
 #include <unistd.h>
 #include <utime.h>
 #include <string.h>
diff -Naur gnome-vfs-2.4.2-old/modules/fstype.c gnome-vfs-2.4.2/modules/fstype.c
--- gnome-vfs-2.4.2-old/modules/fstype.c	Tue Jan 27 13:29:31 2004
+++ gnome-vfs-2.4.2/modules/fstype.c	Tue Jan 27 14:07:56 2004
@@ -363,7 +363,7 @@
   char *p;
 
   if (S_ISLNK (statp->st_mode))
-    p = dirname (relpath);
+    p = g_path_get_dirname (relpath);
   else
     p = relpath;
 
