diff -Naur gnome2-user-docs-2.4.0-old/gnome2-user-guide/C/Makefile.in gnome2-user-docs-2.4.0/gnome2-user-guide/C/Makefile.in
--- gnome2-user-docs-2.4.0-old/gnome2-user-guide/C/Makefile.in	Sun Sep  7 21:28:51 2003
+++ gnome2-user-docs-2.4.0/gnome2-user-guide/C/Makefile.in	Wed Oct 15 18:23:01 2003
@@ -299,7 +299,7 @@
 	for file in $(omffile); do \
 		$(INSTALL_DATA) $$file.out $(DESTDIR)$(omf_dest_dir)/$$file; \
 	done
-	-scrollkeeper-update -p $(scrollkeeper_localstate_dir) -o $(DESTDIR)$(omf_dest_dir)
+#	-scrollkeeper-update -p $(scrollkeeper_localstate_dir) -o $(DESTDIR)$(omf_dest_dir)
 
 uninstall-local-omf:
 	-for file in $(srcdir)/*.omf; do \
@@ -374,7 +374,7 @@
 	for file in $(extraomf); do \
 		$(INSTALL_DATA) $$file.out $(DESTDIR)$(omf_dest_dir)/$$file; \
 	done
-	-scrollkeeper-update -p $(scrollkeeper_localstate_dir) -o $(DESTDIR)$(omf_dest_dir)
+#	-scrollkeeper-update -p $(scrollkeeper_localstate_dir) -o $(DESTDIR)$(omf_dest_dir)
 
 dist-hook: app-dist-hook
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
diff -Naur gnome2-user-docs-2.4.0-old/introduction-to-gnome/C/Makefile.in gnome2-user-docs-2.4.0/introduction-to-gnome/C/Makefile.in
--- gnome2-user-docs-2.4.0-old/introduction-to-gnome/C/Makefile.in	Sun Sep  7 21:28:51 2003
+++ gnome2-user-docs-2.4.0/introduction-to-gnome/C/Makefile.in	Wed Oct 15 18:23:01 2003
@@ -300,7 +300,7 @@
 	for file in $(omffile); do \
 		$(INSTALL_DATA) $$file.out $(DESTDIR)$(omf_dest_dir)/$$file; \
 	done
-	-scrollkeeper-update -p $(scrollkeeper_localstate_dir) -o $(DESTDIR)$(omf_dest_dir)
+#	-scrollkeeper-update -p $(scrollkeeper_localstate_dir) -o $(DESTDIR)$(omf_dest_dir)
 
 uninstall-local-omf:
 	-for file in $(srcdir)/*.omf; do \
diff -Naur gnome2-user-docs-2.4.0-old/omf.make gnome2-user-docs-2.4.0/omf.make
--- gnome2-user-docs-2.4.0-old/omf.make	Tue May 28 16:21:29 2002
+++ gnome2-user-docs-2.4.0/omf.make	Wed Oct 15 18:23:01 2003
@@ -42,7 +42,7 @@
 	for file in $(omffile); do \
 		$(INSTALL_DATA) $$file.out $(DESTDIR)$(omf_dest_dir)/$$file; \
 	done
-	-scrollkeeper-update -p $(scrollkeeper_localstate_dir) -o $(DESTDIR)$(omf_dest_dir)
+#	-scrollkeeper-update -p $(scrollkeeper_localstate_dir) -o $(DESTDIR)$(omf_dest_dir)
 
 uninstall-local-omf:
 	-for file in $(srcdir)/*.omf; do \
