diff -Naur librsvg-2.4.0-old/configure librsvg-2.4.0/configure
--- librsvg-2.4.0-old/configure	Mon Sep  8 08:35:23 2003
+++ librsvg-2.4.0/configure	Fri Oct 17 15:55:06 2003
@@ -8815,11 +8815,11 @@
             echo "$as_me:$LINENO: checking LIBRSVG_CFLAGS" >&5
 echo $ECHO_N "checking LIBRSVG_CFLAGS... $ECHO_C" >&6
             LIBRSVG_CFLAGS=`$PKG_CONFIG --cflags "\
+	pangoft2 >= $PANGOFT2_REQUIRED \
 	gdk-pixbuf-2.0 >= $GDK_PIXBUF_REQUIRED \
 	glib-2.0 >= $GLIB_REQUIRED \
 	libart-2.0 >= $LIBART_REQUIRED \
-	libxml-2.0 >= $LIBXML_REQUIRED \
-	pangoft2 >= $PANGOFT2_REQUIRED"`
+	libxml-2.0 >= $LIBXML_REQUIRED"`
             echo "$as_me:$LINENO: result: $LIBRSVG_CFLAGS" >&5
 echo "${ECHO_T}$LIBRSVG_CFLAGS" >&6
 
diff -Naur librsvg-2.4.0-old/ltmain.sh librsvg-2.4.0/ltmain.sh
--- librsvg-2.4.0-old/ltmain.sh	Sat Feb  8 09:47:44 2003
+++ librsvg-2.4.0/ltmain.sh	Fri Oct 17 16:00:28 2003
@@ -4050,10 +4050,10 @@
 
 # Directory that this library needs to be installed in:
 libdir='$install_libdir'"
-	  if test "$installed" = no && test $need_relink = yes; then
-	    $echo >> $output "\
-relink_command=\"$relink_command\""
-	  fi
+#	  if test "$installed" = no && test $need_relink = yes; then
+#	    $echo >> $output "\
+#relink_command=\"$relink_command\""
+#	  fi
 	done
       fi
 
