diff -Naur vte-0.11.10-old/src/dumpkeys.c vte-0.11.10/src/dumpkeys.c
--- vte-0.11.10-old/src/dumpkeys.c	Wed Apr 23 22:50:54 2003
+++ vte-0.11.10/src/dumpkeys.c	Mon Oct 27 12:49:57 2003
@@ -18,11 +18,11 @@
 
 #ident "$Id$"
 #include "../config.h"
+#include <sys/types.h>
 #ifdef HAVE_SYS_SELECT_H
 #include <sys/select.h>
 #endif
 #include <sys/time.h>
-#include <sys/types.h>
 #include <unistd.h>
 #include <fcntl.h>
 #include <signal.h>
