diff -Naur gcalctool-4.3.3-old/gcalctool/calctool.c gcalctool-4.3.3/gcalctool/calctool.c
--- gcalctool-4.3.3-old/gcalctool/calctool.c	Tue Jul 29 10:33:19 2003
+++ gcalctool-4.3.3/gcalctool/calctool.c	Sat Oct 18 11:54:47 2003
@@ -21,7 +21,7 @@
 
 #include <stdio.h>
 #include <string.h>
-#include <malloc.h>
+#include <sys/malloc.h>
 #include <sys/types.h>
 #include "calctool.h"
 #include <gdk/gdkkeysyms.h>
