diff -uNr gnome-games-2.4.0/aisleriot/Makefile.in gnome-games-2.4.0-new/aisleriot/Makefile.in
--- gnome-games-2.4.0/aisleriot/Makefile.in	Tue Sep  9 03:12:21 2003
+++ gnome-games-2.4.0-new/aisleriot/Makefile.in	Sat Nov  1 15:30:36 2003
@@ -184,11 +184,12 @@
 	card.c
 
 
-sol_LDADD = $(GNOME_GAMES_LIBS) \
+sol_LDADD = \
 	../gdk-card-image/libgdkcardimage.la \
 	../libgames-support/libgames-support.la \
 	@UCB_LDFLAGS@ \
-	$(GUILE_LIBS) 
+	$(GUILE_LIBS) \
+	$(GNOME_GAMES_LIBS)
 
 
 types_DATA = \
diff -uNr gnome-games-2.4.0/aisleriot/menu.c gnome-games-2.4.0-new/aisleriot/menu.c
--- gnome-games-2.4.0/aisleriot/menu.c	Mon Jul 28 05:24:49 2003
+++ gnome-games-2.4.0-new/aisleriot/menu.c	Sat Nov  1 15:27:41 2003
@@ -23,6 +23,7 @@
 
 #include <string.h>
 #include <gnome.h>
+#include <sys/types.h>
 
 #include "sol.h"
 #include "menu.h"
diff -uNr gnome-games-2.4.0/blackjack/src/Makefile.in gnome-games-2.4.0-new/blackjack/src/Makefile.in
--- gnome-games-2.4.0/blackjack/src/Makefile.in	Tue Sep  9 03:12:22 2003
+++ gnome-games-2.4.0-new/blackjack/src/Makefile.in	Sat Nov  1 15:30:57 2003
@@ -179,10 +179,11 @@
 	chips.cpp
 
 
-blackjack_LDADD = $(GNOME_GAMES_LIBS) \
+blackjack_LDADD = \
 	../../gdk-card-image/libgdkcardimage.la \
 	../../libgames-support/libgames-support.la \
-	@UCB_LDFLAGS@
+	@UCB_LDFLAGS@ \
+	$(GNOME_GAMES_LIBS)
 
 mkinstalldirs = $(SHELL) $(top_srcdir)/mkinstalldirs
 CONFIG_HEADER = ../../config.h
diff -uNr gnome-games-2.4.0/gataxx/Makefile.in gnome-games-2.4.0-new/gataxx/Makefile.in
--- gnome-games-2.4.0/gataxx/Makefile.in	Tue Sep  9 03:12:12 2003
+++ gnome-games-2.4.0-new/gataxx/Makefile.in	Sat Nov  1 15:31:16 2003
@@ -160,8 +160,9 @@
 	properties.h
 
 
-gataxx_LDADD = $(GNOME_GAMES_LIBS) \
-	../libgames-support/libgames-support.la
+gataxx_LDADD = \
+	../libgames-support/libgames-support.la \
+	$(GNOME_GAMES_LIBS)
 
 
 appicon_DATA = gataxx.png
diff -uNr gnome-games-2.4.0/gdk-card-image/Makefile.in gnome-games-2.4.0-new/gdk-card-image/Makefile.in
--- gnome-games-2.4.0/gdk-card-image/Makefile.in	Tue Sep  9 03:12:17 2003
+++ gnome-games-2.4.0-new/gdk-card-image/Makefile.in	Sat Nov  1 15:31:36 2003
@@ -179,7 +179,7 @@
 libgdkcardimageincludedir = $(includedir)/gdkcardimage
 libgdkcardimageinclude_HEADERS = gdk-card-image.h
 libgdkcardimage_la_LDFLAGS = -version-info 0:0:0
-libgdkcardimage_la_LIBADD = $(GNOME_GAMES_LIBS) @UCB_LDFLAGS@
+libgdkcardimage_la_LIBADD = @UCB_LDFLAGS@ $(GNOME_GAMES_LIBS)
 mkinstalldirs = $(SHELL) $(top_srcdir)/mkinstalldirs
 CONFIG_HEADER = ../config.h
 CONFIG_CLEAN_FILES = 
diff -uNr gnome-games-2.4.0/glines/Makefile.in gnome-games-2.4.0-new/glines/Makefile.in
--- gnome-games-2.4.0/glines/Makefile.in	Tue Sep  9 03:12:15 2003
+++ gnome-games-2.4.0-new/glines/Makefile.in	Sat Nov  1 15:31:47 2003
@@ -162,9 +162,9 @@
 
 
 glines_LDADD = \
-	$(GNOME_GAMES_LIBS) \
 	../libgames-support/libgames-support.la \
-	$(INTLLIBS)
+	$(INTLLIBS) \
+	$(GNOME_GAMES_LIBS)
 
 
 pixmap_DATA = ball.png classic.png dots.png pulse.png gumball.png 
diff -uNr gnome-games-2.4.0/gnect/src/Makefile.in gnome-games-2.4.0-new/gnect/src/Makefile.in
--- gnome-games-2.4.0/gnect/src/Makefile.in	Tue Sep  9 03:12:02 2003
+++ gnome-games-2.4.0-new/gnect/src/Makefile.in	Sat Nov  1 15:32:05 2003
@@ -184,8 +184,8 @@
 
 gnect_LDADD = \
 	../../libgames-support/libgames-support.la         \
-	$(GNOME_GAMES_LIBS)     \
-	$(INTLLIBS)
+	$(INTLLIBS) \
+	$(GNOME_GAMES_LIBS)
 
 mkinstalldirs = $(SHELL) $(top_srcdir)/mkinstalldirs
 CONFIG_HEADER = ../../config.h
diff -uNr gnome-games-2.4.0/gnect/src/adjmtrx.c gnome-games-2.4.0-new/gnect/src/adjmtrx.c
--- gnome-games-2.4.0/gnect/src/adjmtrx.c	Wed Jun 18 15:09:38 2003
+++ gnome-games-2.4.0-new/gnect/src/adjmtrx.c	Sat Nov  1 15:27:41 2003
@@ -21,10 +21,13 @@
 */
 
 
+#include <config.h>
 #include <stdio.h>
 #include <string.h>
 #include <stdlib.h>
-#include <malloc.h>
+#ifdef HAVE_MALLOC_H
+#  include <malloc.h>
+#endif
 
 #include "connect4.h"
 #include "con4vals.h"
diff -uNr gnome-games-2.4.0/gnect/src/bintree.c gnome-games-2.4.0-new/gnect/src/bintree.c
--- gnome-games-2.4.0/gnect/src/bintree.c	Wed Jun 18 15:09:38 2003
+++ gnome-games-2.4.0-new/gnect/src/bintree.c	Sat Nov  1 15:27:41 2003
@@ -21,10 +21,13 @@
 */
 
 
+#include <config.h>
 #include <stdio.h>
 #include <string.h>
 #include <stdlib.h>
-#include <malloc.h>
+#ifdef HAVE_MALLOC_H
+#  include <malloc.h>
+#endif
 
 #include "connect4.h"
 #include "con4vals.h"
diff -uNr gnome-games-2.4.0/gnect/src/connect4.c gnome-games-2.4.0-new/gnect/src/connect4.c
--- gnome-games-2.4.0/gnect/src/connect4.c	Wed Jun 18 15:09:38 2003
+++ gnome-games-2.4.0-new/gnect/src/connect4.c	Sat Nov  1 15:27:41 2003
@@ -24,10 +24,13 @@
 */
 
 
+#include <config.h>
 #include <stdio.h>
 #include <string.h>
 #include <stdlib.h>
-#include <malloc.h>
+#ifdef HAVE_MALLOC_H
+#  include <malloc.h>
+#endif
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <fcntl.h>
diff -uNr gnome-games-2.4.0/gnect/src/evaluate.c gnome-games-2.4.0-new/gnect/src/evaluate.c
--- gnome-games-2.4.0/gnect/src/evaluate.c	Wed Jun 18 15:09:38 2003
+++ gnome-games-2.4.0-new/gnect/src/evaluate.c	Sat Nov  1 15:27:41 2003
@@ -22,10 +22,13 @@
 
 
 
+#include <config.h>
 #include <stdio.h>
 #include <string.h>
 #include <stdlib.h>
-#include <malloc.h>
+#ifdef HAVE_MALLOC_H
+#  include <malloc.h>
+#endif
 #include <time.h>
 #include <math.h>
 
diff -uNr gnome-games-2.4.0/gnect/src/heurist.c gnome-games-2.4.0-new/gnect/src/heurist.c
--- gnome-games-2.4.0/gnect/src/heurist.c	Wed Jun 18 15:09:38 2003
+++ gnome-games-2.4.0-new/gnect/src/heurist.c	Sat Nov  1 15:27:41 2003
@@ -21,10 +21,13 @@
 */
 
 
+#include <config.h>
 #include <stdio.h>
 #include <string.h>
 #include <stdlib.h>
-#include <malloc.h>
+#ifdef HAVE_MALLOC_H
+#  include <malloc.h>
+#endif
 
 #include "connect4.h"
 #include "con4vals.h"
diff -uNr gnome-games-2.4.0/gnect/src/ia_main.c gnome-games-2.4.0-new/gnect/src/ia_main.c
--- gnome-games-2.4.0/gnect/src/ia_main.c	Wed Jun 18 15:09:38 2003
+++ gnome-games-2.4.0-new/gnect/src/ia_main.c	Sat Nov  1 15:27:41 2003
@@ -21,10 +21,13 @@
 */
 
 
+#include <config.h>
 #include <stdio.h>
 #include <string.h>
 #include <stdlib.h>
-#include <malloc.h>
+#ifdef HAVE_MALLOC_H
+#  include <malloc.h>
+#endif
 #include <time.h>
 
 #include <gtk/gtk.h>
diff -uNr gnome-games-2.4.0/gnect/src/pbsolver.c gnome-games-2.4.0-new/gnect/src/pbsolver.c
--- gnome-games-2.4.0/gnect/src/pbsolver.c	Wed Jun 18 15:09:38 2003
+++ gnome-games-2.4.0-new/gnect/src/pbsolver.c	Sat Nov  1 15:27:42 2003
@@ -21,10 +21,13 @@
 */
 
 
+#include <config.h>
 #include <stdio.h>
 #include <string.h>
 #include <stdlib.h>
-#include <malloc.h>
+#ifdef HAVE_MALLOC_H
+#  include <malloc.h>
+#endif
 
 #include "connect4.h"
 #include "con4vals.h"
diff -uNr gnome-games-2.4.0/gnibbles/Makefile.in gnome-games-2.4.0-new/gnibbles/Makefile.in
--- gnome-games-2.4.0/gnibbles/Makefile.in	Tue Sep  9 03:12:19 2003
+++ gnome-games-2.4.0-new/gnibbles/Makefile.in	Sat Nov  1 15:32:13 2003
@@ -179,9 +179,9 @@
 
 
 gnibbles_LDADD = \
-	$(GNOME_GAMES_LIBS) \
 	../libgames-support/libgames-support.la \
-	$(INTLLIBS)
+	$(INTLLIBS) \
+	$(GNOME_GAMES_LIBS)
 
 
 pixmap_DATA = \
diff -uNr gnome-games-2.4.0/gnobots2/Makefile.in gnome-games-2.4.0-new/gnobots2/Makefile.in
--- gnome-games-2.4.0/gnobots2/Makefile.in	Tue Sep  9 03:12:18 2003
+++ gnome-games-2.4.0-new/gnobots2/Makefile.in	Sat Nov  1 15:32:20 2003
@@ -186,9 +186,9 @@
 
 
 gnobots2_LDADD = \
-	$(GNOME_GAMES_LIBS) \
 	../libgames-support/libgames-support.la \
-	$(INTLLIBS)
+	$(INTLLIBS) \
+	$(GNOME_GAMES_LIBS)
 
 
 appicon_DATA = gnome-gnobots2.png
diff -uNr gnome-games-2.4.0/gnome-stones/Makefile.in gnome-games-2.4.0-new/gnome-stones/Makefile.in
--- gnome-games-2.4.0/gnome-stones/Makefile.in	Tue Sep  9 03:12:11 2003
+++ gnome-games-2.4.0-new/gnome-stones/Makefile.in	Sat Nov  1 15:32:30 2003
@@ -189,10 +189,10 @@
 
 
 gnome_stones_LDADD = \
-	$(GNOME_GAMES_LIBS)			\
 	../libgames-support/libgames-support.la \
 	$(ESD_LIBS)				\
-	$(INTLLIBS)
+	$(INTLLIBS) \
+	$(GNOME_GAMES_LIBS)
 
 
 gnome_stones_LDFLAGS = -export-dynamic
diff -uNr gnome-games-2.4.0/gnometris/Makefile.in gnome-games-2.4.0-new/gnometris/Makefile.in
--- gnome-games-2.4.0/gnometris/Makefile.in	Tue Sep  9 03:12:24 2003
+++ gnome-games-2.4.0-new/gnometris/Makefile.in	Sat Nov  1 15:32:37 2003
@@ -170,9 +170,9 @@
 
 
 gnometris_LDADD = \
-	$(GNOME_GAMES_LIBS) \
 	../libgames-support/libgames-support.la \
-	$(INTLLIBS)
+	$(INTLLIBS) \
+	$(GNOME_GAMES_LIBS)
 
 
 Games_in_files = gnometris.desktop.in
diff -uNr gnome-games-2.4.0/gnomine/Makefile.in gnome-games-2.4.0-new/gnomine/Makefile.in
--- gnome-games-2.4.0/gnomine/Makefile.in	Tue Sep  9 03:12:07 2003
+++ gnome-games-2.4.0-new/gnomine/Makefile.in	Sat Nov  1 15:32:45 2003
@@ -159,9 +159,9 @@
 
 
 gnomine_LDADD = \
-	$(GNOME_GAMES_LIBS) \
 	../libgames-support/libgames-support.la \
-	$(INTLLIBS)
+	$(INTLLIBS) \
+	$(GNOME_GAMES_LIBS)
 
 
 schemadir = @GCONF_SCHEMA_FILE_DIR@
diff -uNr gnome-games-2.4.0/gnotravex/Makefile.in gnome-games-2.4.0-new/gnotravex/Makefile.in
--- gnome-games-2.4.0/gnotravex/Makefile.in	Tue Sep  9 03:12:13 2003
+++ gnome-games-2.4.0-new/gnotravex/Makefile.in	Sat Nov  1 15:32:55 2003
@@ -159,9 +159,9 @@
 
 
 gnotravex_LDADD = \
-	$(GNOME_GAMES_LIBS) \
 	../libgames-support/libgames-support.la \
-	$(INTLLIBS) 
+	$(INTLLIBS) \
+	$(GNOME_GAMES_LIBS)
 
 
 sharedpixmap_DATA = gnome-gnotravex.png
diff -uNr gnome-games-2.4.0/gnotski/Makefile.in gnome-games-2.4.0-new/gnotski/Makefile.in
--- gnome-games-2.4.0/gnotski/Makefile.in	Tue Sep  9 03:12:14 2003
+++ gnome-games-2.4.0-new/gnotski/Makefile.in	Sat Nov  1 15:33:03 2003
@@ -157,8 +157,8 @@
 
 
 gnotski_LDADD = \
-	$(GNOME_GAMES_LIBS) \
-	$(INTLLIBS) 
+	$(INTLLIBS) \
+	$(GNOME_GAMES_LIBS)
 
 
 pixmap_DATA = gnotski.png gnotski-icon.png
diff -uNr gnome-games-2.4.0/gtali/Makefile.in gnome-games-2.4.0-new/gtali/Makefile.in
--- gnome-games-2.4.0/gtali/Makefile.in	Tue Sep  9 03:12:09 2003
+++ gnome-games-2.4.0-new/gtali/Makefile.in	Sat Nov  1 15:33:12 2003
@@ -163,9 +163,9 @@
 
 
 gtali_LDADD = \
-	$(GNOME_GAMES_LIBS) \
 	../libgames-support/libgames-support.la \
-	$(INTLLIBS)
+	$(INTLLIBS) \
+	$(GNOME_GAMES_LIBS)
 
 
 schemadir = @GCONF_SCHEMA_FILE_DIR@
diff -uNr gnome-games-2.4.0/iagno/Makefile.in gnome-games-2.4.0-new/iagno/Makefile.in
--- gnome-games-2.4.0/iagno/Makefile.in	Tue Sep  9 03:12:16 2003
+++ gnome-games-2.4.0-new/iagno/Makefile.in	Sat Nov  1 15:33:23 2003
@@ -164,9 +164,9 @@
 
 
 iagno_LDADD = \
-	$(GNOME_GAMES_LIBS) \
 	../libgames-support/libgames-support.la \
-	$(INTLLIBS)
+	$(INTLLIBS) \
+	$(GNOME_GAMES_LIBS)
 
 
 pixmap_DATA = \
diff -uNr gnome-games-2.4.0/iagno/network.c gnome-games-2.4.0-new/iagno/network.c
--- gnome-games-2.4.0/iagno/network.c	Wed Jun 18 11:54:41 2003
+++ gnome-games-2.4.0-new/iagno/network.c	Sat Nov  1 15:27:42 2003
@@ -26,6 +26,10 @@
 #include "gnothello.h"
 #include "network.h"
 
+#if defined(__APPLE__) && !defined(socklen_t)
+#define socklen_t int
+#endif
+
 /* (('g'<<8)+'n' */
 #define GAME_PORT "26478"
 
diff -uNr gnome-games-2.4.0/ltmain.sh gnome-games-2.4.0-new/ltmain.sh
--- gnome-games-2.4.0/ltmain.sh	Mon May 26 04:00:52 2003
+++ gnome-games-2.4.0-new/ltmain.sh	Sat Nov  1 15:27:42 2003
@@ -2972,6 +2972,11 @@
 	if test -n "$export_symbols" && test -n "$archive_expsym_cmds"; then
 	  eval cmds=\"$archive_expsym_cmds\"
 	else
+	  if test "x$verstring" = "x0.0"; then
+	    tmp_verstring=
+	  else
+	    tmp_verstring="$verstring"
+	  fi
 	  save_deplibs="$deplibs"
 	  for conv in $convenience; do
 	    tmp_deplibs=
diff -uNr gnome-games-2.4.0/mahjongg/Makefile.in gnome-games-2.4.0-new/mahjongg/Makefile.in
--- gnome-games-2.4.0/mahjongg/Makefile.in	Tue Sep  9 03:12:08 2003
+++ gnome-games-2.4.0-new/mahjongg/Makefile.in	Sat Nov  1 15:33:32 2003
@@ -169,9 +169,9 @@
 
 
 mahjongg_LDADD = \
-	$(GNOME_GAMES_LIBS)				\
 	../libgames-support/libgames-support.la		\
-	$(INTLLIBS)
+	$(INTLLIBS) \
+	$(GNOME_GAMES_LIBS)
 
 
 pixmap_DATA = default.png seta.png
diff -uNr gnome-games-2.4.0/omf.make gnome-games-2.4.0-new/omf.make
--- gnome-games-2.4.0/omf.make	Tue May  7 18:08:52 2002
+++ gnome-games-2.4.0-new/omf.make	Sat Nov  1 15:27:42 2003
@@ -42,7 +42,7 @@
 	for file in $(omffile); do \
 		$(INSTALL_DATA) $$file.out $(DESTDIR)$(omf_dest_dir)/$$file; \
 	done
-	-scrollkeeper-update -p $(scrollkeeper_localstate_dir) -o $(DESTDIR)$(omf_dest_dir)
+#	-scrollkeeper-update -p $(scrollkeeper_localstate_dir) -o $(DESTDIR)$(omf_dest_dir)
 
 uninstall-local-omf:
 	-for file in $(srcdir)/*.omf; do \
