diff -Naur gnome-themes-extras-0.4-old/Amaranth/icons/scalable/mimetypes/Makefile.in gnome-themes-extras-0.4/Amaranth/icons/scalable/mimetypes/Makefile.in
--- gnome-themes-extras-0.4-old/Amaranth/icons/scalable/mimetypes/Makefile.in	Sun Nov 23 11:13:30 2003
+++ gnome-themes-extras-0.4/Amaranth/icons/scalable/mimetypes/Makefile.in	Sun Nov 23 11:14:30 2003
@@ -312,7 +312,7 @@
 
 install-data-local:
 	mkdir -p $(DESTDIR)$(icondir)
-	cd $(icondir) && \
+	cd $(DESTDIR)$(icondir) && \
 	ln -sf gnome-mime-audio.svg gnome-mime-application-x-ogg.svg && \
 	ln -sf gnome-compressed.svg gnome-mime-application-x-stuffit.svg && \
 	ln -sf gnome-compressed.svg gnome-mime-application-x-tgz.svg
