diff -Naur gnome-applets-2.4.0-old/charpick/charpick.c gnome-applets-2.4.0/charpick/charpick.c
--- gnome-applets-2.4.0-old/charpick/charpick.c	Wed Sep  3 19:51:46 2003
+++ gnome-applets-2.4.0/charpick/charpick.c	Fri Oct 24 13:58:43 2003
@@ -3,9 +3,11 @@
  */
 
 #include <config.h>
+/*
 #ifndef sun
   #include <tgmath.h>
 #endif
+*/
 #include <panel-applet.h>
 #include <egg-screen-help.h>
 #include "charpick.h"
diff -Naur gnome-applets-2.4.0-old/configure gnome-applets-2.4.0/configure
--- gnome-applets-2.4.0-old/configure	Mon Sep  8 20:00:28 2003
+++ gnome-applets-2.4.0/configure	Fri Oct 24 13:58:43 2003
@@ -11932,6 +11932,9 @@
 	echo "warning: ${host} is not supported by battstat_applet, not building" >&2
 	build_battstat_applet=no
 	;;
+  *darwin*)
+	build_battstat_applet=no
+	;;
 esac
 
 
diff -Naur gnome-applets-2.4.0-old/gkb-new/preset.c gnome-applets-2.4.0/gkb-new/preset.c
--- gnome-applets-2.4.0-old/gkb-new/preset.c	Sun Jun 29 12:25:11 2003
+++ gnome-applets-2.4.0/gkb-new/preset.c	Fri Oct 24 13:58:43 2003
@@ -29,6 +29,7 @@
  * USA
  */
 
+#include <sys/types.h>
 #include <dirent.h>
 #include "gkb.h"
 
diff -Naur gnome-applets-2.4.0-old/multiload/Makefile.in gnome-applets-2.4.0/multiload/Makefile.in
--- gnome-applets-2.4.0-old/multiload/Makefile.in	Mon Sep  8 20:07:07 2003
+++ gnome-applets-2.4.0/multiload/Makefile.in	Fri Oct 24 13:58:43 2003
@@ -148,7 +148,7 @@
 multiload_applet_2_SOURCES =  			global.h				linux-proc.h				load-graph.h				cpuload.c				linux-proc.c				load-graph.c				loadavg.c				main.c				memload.c				netload.c				properties.c				swapload.c
 
 
-multiload_applet_2_LDADD =  	$(top_builddir)/screen-exec/libscreen-exec.la					$(GNOME_APPLETS_LIBS) 					$(GTOP_APPLETS_LIBS)
+multiload_applet_2_LDADD =  	$(top_builddir)/screen-exec/libscreen-exec.la					$(GNOME_APPLETS_LIBS) 					$(GTOP_APPLETS_LIBS)	$(X_LIBS)
 
 
 pixmaps = gnome-cpu.png
diff -Naur gnome-applets-2.4.0-old/omf.make gnome-applets-2.4.0/omf.make
--- gnome-applets-2.4.0-old/omf.make	Mon Sep  8 19:59:38 2003
+++ gnome-applets-2.4.0/omf.make	Fri Oct 24 13:58:43 2003
@@ -45,7 +45,7 @@
 	for file in $(omffile); do \
 		$(INSTALL_DATA) $$file.out $(DESTDIR)$(omf_dest_dir)/$$file; \
 	done
-	-scrollkeeper-update -p $(scrollkeeper_localstate_dir) -o $(DESTDIR)$(omf_dest_dir)
+	#-scrollkeeper-update -p $(scrollkeeper_localstate_dir) -o $(DESTDIR)$(omf_dest_dir)
 
 uninstall-local-omf:
 	-for file in $(srcdir)/*.omf; do \
