diff -Naur gnome-games-2.4.0-old/aisleriot/menu.c gnome-games-2.4.0/aisleriot/menu.c
--- gnome-games-2.4.0-old/aisleriot/menu.c	Mon Jul 28 05:24:49 2003
+++ gnome-games-2.4.0/aisleriot/menu.c	Fri Oct 17 11:02:47 2003
@@ -23,6 +23,7 @@
 
 #include <string.h>
 #include <gnome.h>
+#include <sys/types.h>
 
 #include "sol.h"
 #include "menu.h"
diff -Naur gnome-games-2.4.0-old/gnect/src/adjmtrx.c gnome-games-2.4.0/gnect/src/adjmtrx.c
--- gnome-games-2.4.0-old/gnect/src/adjmtrx.c	Wed Jun 18 15:09:38 2003
+++ gnome-games-2.4.0/gnect/src/adjmtrx.c	Fri Oct 17 10:59:25 2003
@@ -21,10 +21,13 @@
 */
 
 
+#include <config.h>
 #include <stdio.h>
 #include <string.h>
 #include <stdlib.h>
-#include <malloc.h>
+#ifdef HAVE_MALLOC_H
+#  include <malloc.h>
+#endif
 
 #include "connect4.h"
 #include "con4vals.h"
diff -Naur gnome-games-2.4.0-old/gnect/src/bintree.c gnome-games-2.4.0/gnect/src/bintree.c
--- gnome-games-2.4.0-old/gnect/src/bintree.c	Wed Jun 18 15:09:38 2003
+++ gnome-games-2.4.0/gnect/src/bintree.c	Fri Oct 17 10:59:25 2003
@@ -21,10 +21,13 @@
 */
 
 
+#include <config.h>
 #include <stdio.h>
 #include <string.h>
 #include <stdlib.h>
-#include <malloc.h>
+#ifdef HAVE_MALLOC_H
+#  include <malloc.h>
+#endif
 
 #include "connect4.h"
 #include "con4vals.h"
diff -Naur gnome-games-2.4.0-old/gnect/src/connect4.c gnome-games-2.4.0/gnect/src/connect4.c
--- gnome-games-2.4.0-old/gnect/src/connect4.c	Wed Jun 18 15:09:38 2003
+++ gnome-games-2.4.0/gnect/src/connect4.c	Fri Oct 17 10:59:25 2003
@@ -24,10 +24,13 @@
 */
 
 
+#include <config.h>
 #include <stdio.h>
 #include <string.h>
 #include <stdlib.h>
-#include <malloc.h>
+#ifdef HAVE_MALLOC_H
+#  include <malloc.h>
+#endif
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <fcntl.h>
diff -Naur gnome-games-2.4.0-old/gnect/src/evaluate.c gnome-games-2.4.0/gnect/src/evaluate.c
--- gnome-games-2.4.0-old/gnect/src/evaluate.c	Wed Jun 18 15:09:38 2003
+++ gnome-games-2.4.0/gnect/src/evaluate.c	Fri Oct 17 10:59:25 2003
@@ -22,10 +22,13 @@
 
 
 
+#include <config.h>
 #include <stdio.h>
 #include <string.h>
 #include <stdlib.h>
-#include <malloc.h>
+#ifdef HAVE_MALLOC_H
+#  include <malloc.h>
+#endif
 #include <time.h>
 #include <math.h>
 
diff -Naur gnome-games-2.4.0-old/gnect/src/heurist.c gnome-games-2.4.0/gnect/src/heurist.c
--- gnome-games-2.4.0-old/gnect/src/heurist.c	Wed Jun 18 15:09:38 2003
+++ gnome-games-2.4.0/gnect/src/heurist.c	Fri Oct 17 10:59:25 2003
@@ -21,10 +21,13 @@
 */
 
 
+#include <config.h>
 #include <stdio.h>
 #include <string.h>
 #include <stdlib.h>
-#include <malloc.h>
+#ifdef HAVE_MALLOC_H
+#  include <malloc.h>
+#endif
 
 #include "connect4.h"
 #include "con4vals.h"
diff -Naur gnome-games-2.4.0-old/gnect/src/ia_main.c gnome-games-2.4.0/gnect/src/ia_main.c
--- gnome-games-2.4.0-old/gnect/src/ia_main.c	Wed Jun 18 15:09:38 2003
+++ gnome-games-2.4.0/gnect/src/ia_main.c	Fri Oct 17 10:59:25 2003
@@ -21,10 +21,13 @@
 */
 
 
+#include <config.h>
 #include <stdio.h>
 #include <string.h>
 #include <stdlib.h>
-#include <malloc.h>
+#ifdef HAVE_MALLOC_H
+#  include <malloc.h>
+#endif
 #include <time.h>
 
 #include <gtk/gtk.h>
diff -Naur gnome-games-2.4.0-old/gnect/src/pbsolver.c gnome-games-2.4.0/gnect/src/pbsolver.c
--- gnome-games-2.4.0-old/gnect/src/pbsolver.c	Wed Jun 18 15:09:38 2003
+++ gnome-games-2.4.0/gnect/src/pbsolver.c	Fri Oct 17 10:59:25 2003
@@ -21,10 +21,13 @@
 */
 
 
+#include <config.h>
 #include <stdio.h>
 #include <string.h>
 #include <stdlib.h>
-#include <malloc.h>
+#ifdef HAVE_MALLOC_H
+#  include <malloc.h>
+#endif
 
 #include "connect4.h"
 #include "con4vals.h"
diff -Naur gnome-games-2.4.0-old/iagno/network.c gnome-games-2.4.0/iagno/network.c
--- gnome-games-2.4.0-old/iagno/network.c	Wed Jun 18 11:54:41 2003
+++ gnome-games-2.4.0/iagno/network.c	Fri Oct 17 10:59:25 2003
@@ -26,6 +26,10 @@
 #include "gnothello.h"
 #include "network.h"
 
+#if defined(__APPLE__) && !defined(socklen_t)
+#define socklen_t int
+#endif
+
 /* (('g'<<8)+'n' */
 #define GAME_PORT "26478"
 
diff -Naur gnome-games-2.4.0-old/ltmain.sh gnome-games-2.4.0/ltmain.sh
--- gnome-games-2.4.0-old/ltmain.sh	Mon May 26 04:00:52 2003
+++ gnome-games-2.4.0/ltmain.sh	Fri Oct 17 10:59:25 2003
@@ -2972,6 +2972,11 @@
 	if test -n "$export_symbols" && test -n "$archive_expsym_cmds"; then
 	  eval cmds=\"$archive_expsym_cmds\"
 	else
+	  if test "x$verstring" = "x0.0"; then
+	    tmp_verstring=
+	  else
+	    tmp_verstring="$verstring"
+	  fi
 	  save_deplibs="$deplibs"
 	  for conv in $convenience; do
 	    tmp_deplibs=
diff -Naur gnome-games-2.4.0-old/omf.make gnome-games-2.4.0/omf.make
--- gnome-games-2.4.0-old/omf.make	Tue May  7 18:08:52 2002
+++ gnome-games-2.4.0/omf.make	Fri Oct 17 10:59:25 2003
@@ -42,7 +42,7 @@
 	for file in $(omffile); do \
 		$(INSTALL_DATA) $$file.out $(DESTDIR)$(omf_dest_dir)/$$file; \
 	done
-	-scrollkeeper-update -p $(scrollkeeper_localstate_dir) -o $(DESTDIR)$(omf_dest_dir)
+#	-scrollkeeper-update -p $(scrollkeeper_localstate_dir) -o $(DESTDIR)$(omf_dest_dir)
 
 uninstall-local-omf:
 	-for file in $(srcdir)/*.omf; do \
