diff -Naur libglade-0.17-old/Makefile.in libglade-0.17/Makefile.in
--- libglade-0.17-old/Makefile.in	Thu Sep  6 23:00:35 2001
+++ libglade-0.17/Makefile.in	Fri Oct 24 10:40:15 2003
@@ -167,7 +167,7 @@
 bin_SCRIPTS = libglade-config libglade-xgettext
 noinst_PROGRAMS = test-libglade
 test_libglade_LDADD = $(THE_LIBS) $(XML_LIBS)
-SUBDIRS = intl macros glade doc
+SUBDIRS = intl macros glade
 
 m4datadir = $(datadir)/aclocal
 m4data_DATA = libglade.m4
diff -Naur libglade-0.17-old/doc/Makefile.in libglade-0.17/doc/Makefile.in
--- libglade-0.17-old/doc/Makefile.in	Thu Sep  6 23:14:11 2001
+++ libglade-0.17/doc/Makefile.in	Fri Oct 24 10:39:57 2003
@@ -351,14 +351,14 @@
 	if test "$$installfiles" = '$(srcdir)/html/*.html'; \
 	then echo '-- Nothing to install' ; \
 	else \
+	  echo '-- Fixing Crossreferences' ; \
+	  gtkdoc-fixxref --module=$(DOC_MODULE) --module-dir=html --html-dir=$(HTML_DIR); \
 	  for i in $$installfiles; do \
 	    echo '-- Installing '$$i ; \
 	    $(INSTALL_DATA) $$i $(DESTDIR)$(TARGET_DIR); \
 	  done; \
 	  echo '-- Installing $(srcdir)/html/index.sgml' ; \
 	  $(INSTALL_DATA) $(srcdir)/html/index.sgml $(TARGET_DIR); \
-	  echo '-- Fixing Crossreferences' ; \
-	  gtkdoc-fixxref --module=$(DOC_MODULE) --html-dir=$(HTML_DIR) || true; \
 	fi)
 
 uninstall-local:
