diff -Naur linc-1.0.3-old/Makefile.in linc-1.0.3/Makefile.in
--- linc-1.0.3-old/Makefile.in	Fri Jun 20 08:27:27 2003
+++ linc-1.0.3/Makefile.in	Tue Oct 14 10:22:38 2003
@@ -98,7 +98,7 @@
 VERSION = @VERSION@
 WARN_CFLAGS = @WARN_CFLAGS@
 
-SUBDIRS = include src test docs
+SUBDIRS = include src docs
 
 m4datadir = $(datadir)/aclocal
 m4data_DATA = linc.m4
diff -Naur linc-1.0.3-old/config.h.in linc-1.0.3/config.h.in
--- linc-1.0.3-old/config.h.in	Fri Jun 20 08:24:38 2003
+++ linc-1.0.3/config.h.in	Tue Oct 14 10:20:14 2003
@@ -10,7 +10,9 @@
  * Copyright 2001, Red Hat, Inc., Ximian, Inc.,
  *                 Sun Microsystems, Inc.
  */
+#if !defined (__APPLE__)
 #undef socklen_t
+#endif
 #undef HAVE_SOCKADDR_SA_LEN
 #undef HAVE_INET_PTON
 #undef HAVE_GETADDRINFO
diff -Naur linc-1.0.3-old/config.h.in.orig linc-1.0.3/config.h.in.orig
--- linc-1.0.3-old/config.h.in.orig	Wed Dec 31 19:00:00 1969
+++ linc-1.0.3/config.h.in.orig	Tue Oct 14 10:20:14 2003
@@ -0,0 +1,148 @@
+/* config.h.in.  Generated from configure.in by autoheader.  */
+/*
+ * acconfig.h: This file is part of the linc library.
+ *
+ * Authors:
+ *    Elliot Lee     (sopwith@redhat.com)
+ *    Michael Meeks  (michael@ximian.com)
+ *    Mark McLouglin (mark@skynet.ie) & others
+ *
+ * Copyright 2001, Red Hat, Inc., Ximian, Inc.,
+ *                 Sun Microsystems, Inc.
+ */
+#undef socklen_t
+#undef HAVE_SOCKADDR_SA_LEN
+#undef HAVE_INET_PTON
+#undef HAVE_GETADDRINFO
+#undef HAVE_GETNAMEINFO
+
+/* Define to 1 if you have the <arpa/nameser.h> header file. */
+#undef HAVE_ARPA_NAMESER_H
+
+/* Define to 1 if you have the `basename' function. */
+#undef HAVE_BASENAME
+
+/* Define to 1 if you have the <dlfcn.h> header file. */
+#undef HAVE_DLFCN_H
+
+/* Define to 1 if you have the <endian.h> header file. */
+#undef HAVE_ENDIAN_H
+
+/* Define to 1 if you have the <fcntl.h> header file. */
+#undef HAVE_FCNTL_H
+
+/* Define to 1 if you have the <inttypes.h> header file. */
+#undef HAVE_INTTYPES_H
+
+/* Define to 1 if you have the `nsl' library (-lnsl). */
+#undef HAVE_LIBNSL
+
+/* Define to 1 if you have the `resolv' library (-lresolv). */
+#undef HAVE_LIBRESOLV
+
+/* Define to 1 if you have the `socket' library (-lsocket). */
+#undef HAVE_LIBSOCKET
+
+/* Define to 1 if you have the <linux/irda.h> header file. */
+#undef HAVE_LINUX_IRDA_H
+
+/* Define to 1 if you have the <machine/endian.h> header file. */
+#undef HAVE_MACHINE_ENDIAN_H
+
+/* Define to 1 if you have the <machine/types.h> header file. */
+#undef HAVE_MACHINE_TYPES_H
+
+/* Define to 1 if you have the <memory.h> header file. */
+#undef HAVE_MEMORY_H
+
+/* Define to 1 if you have the <netinet/in.h> header file. */
+#undef HAVE_NETINET_IN_H
+
+/* Define to 1 if you have the <netinet/tcp.h> header file. */
+#undef HAVE_NETINET_TCP_H
+
+/* Define to 1 if you have the `poll' function. */
+#undef HAVE_POLL
+
+/* Define to 1 if you have the <resolv.h> header file. */
+#undef HAVE_RESOLV_H
+
+/* Define to 1 if you have the <stddef.h> header file. */
+#undef HAVE_STDDEF_H
+
+/* Define to 1 if you have the <stdint.h> header file. */
+#undef HAVE_STDINT_H
+
+/* Define to 1 if you have the <stdlib.h> header file. */
+#undef HAVE_STDLIB_H
+
+/* Define to 1 if you have the <strings.h> header file. */
+#undef HAVE_STRINGS_H
+
+/* Define to 1 if you have the <string.h> header file. */
+#undef HAVE_STRING_H
+
+/* Define to 1 if you have the <sys/endian.h> header file. */
+#undef HAVE_SYS_ENDIAN_H
+
+/* Define to 1 if you have the <sys/isa_defs.h> header file. */
+#undef HAVE_SYS_ISA_DEFS_H
+
+/* Define to 1 if you have the <sys/machine.h> header file. */
+#undef HAVE_SYS_MACHINE_H
+
+/* Define to 1 if you have the <sys/poll.h> header file. */
+#undef HAVE_SYS_POLL_H
+
+/* Define to 1 if you have the <sys/stat.h> header file. */
+#undef HAVE_SYS_STAT_H
+
+/* Define to 1 if you have the <sys/types.h> header file. */
+#undef HAVE_SYS_TYPES_H
+
+/* Define to 1 if you have the <sys/un.h> header file. */
+#undef HAVE_SYS_UN_H
+
+/* Define to 1 if you have the <unistd.h> header file. */
+#undef HAVE_UNISTD_H
+
+/* Define to 1 if you have the <wchar.h> header file. */
+#undef HAVE_WCHAR_H
+
+/* Define to 1 if you have the <wcstr.h> header file. */
+#undef HAVE_WCSTR_H
+
+/* Define to 1 if you have the <wctype.h> header file. */
+#undef HAVE_WCTYPE_H
+
+/* Define to the address where bug reports for this package should be sent. */
+#undef PACKAGE_BUGREPORT
+
+/* Define to the full name of this package. */
+#undef PACKAGE_NAME
+
+/* Define to the full name and version of this package. */
+#undef PACKAGE_STRING
+
+/* Define to the one symbol short name of this package. */
+#undef PACKAGE_TARNAME
+
+/* Define to the version of this package. */
+#undef PACKAGE_VERSION
+
+/* Define to 1 if you have the ANSI C header files. */
+#undef STDC_HEADERS
+
+/* Define to 1 if your processor stores words with the most significant byte
+   first (like Motorola and SPARC, unlike Intel and VAX). */
+#undef WORDS_BIGENDIAN
+
+/* Define to empty if `const' does not conform to ANSI C. */
+#undef const
+
+/* Define as `__inline' if that's what the C compiler calls it, or to nothing
+   if it is not supported. */
+#undef inline
+
+/* Define to `unsigned' if <sys/types.h> does not define. */
+#undef size_t
diff -Naur linc-1.0.3-old/include/linc/linc-protocol.h linc-1.0.3/include/linc/linc-protocol.h
--- linc-1.0.3-old/include/linc/linc-protocol.h	Fri Jun 20 07:31:21 2003
+++ linc-1.0.3/include/linc/linc-protocol.h	Tue Oct 14 10:20:14 2003
@@ -14,6 +14,10 @@
 
 #include <glib/gmacros.h>
 
+#if defined(__APPLE__) && !defined(socklen_t)
+#  define socklen_t int
+#endif
+
 G_BEGIN_DECLS
 
 #include <linc/linc-types.h>
diff -Naur linc-1.0.3-old/src/linc-protocols.c linc-1.0.3/src/linc-protocols.c
--- linc-1.0.3-old/src/linc-protocols.c	Fri Jun 20 07:31:13 2003
+++ linc-1.0.3/src/linc-protocols.c	Tue Oct 14 10:20:14 2003
@@ -11,12 +11,17 @@
  */
 #include <config.h>
 #include "linc-compat.h"
+#include <sys/types.h>
 #include <dirent.h>
 #include <linc/linc-protocol.h>
 #include <linc/linc-connection.h>
 
 #include "linc-private.h"
 #include "linc-debug.h"
+
+#if defined(__APPLE__) && !defined(RES_USE_INET6)
+#  undef AF_INET6
+#endif
 
 #undef LOCAL_DEBUG
 
