diff -Naur gnome-desktop-2.0.0/configure gnome-desktop-new/configure
--- gnome-desktop-2.0.0/configure	2002-06-10 01:33:32.000000000 -0400
+++ gnome-desktop-new/configure	2002-06-13 07:14:48.000000000 -0400
@@ -5240,7 +5240,7 @@
     # FIXME: Relying on posixy $() will cause problems for
     #        cross-compilation, but unfortunately the echo tests do not
     #        yet detect zsh echo's removal of \ escapes.
-    archive_cmds='$nonopt $(test "x$module" = xyes && echo -bundle || echo -dynamiclib) $allow_undefined_flag -o $lib $libobjs $deplibs$linker_flags -install_name $rpath/$soname $verstring'
+    archive_cmds='$nonopt $(test x$module = xyes && echo -bundle || echo -dynamiclib) $allow_undefined_flag -o $lib $libobjs $deplibs$linker_flags $(test x$module = xyes || echo -install_name $rpath/$soname $verstring)'
     # We need to add '_' to the symbols in $export_symbols first
     #archive_expsym_cmds="$archive_cmds"' && strip -s $export_symbols'
     hardcode_direct=yes
diff -Naur gnome-desktop-2.0.0/libgnome-desktop/gnome-desktop-item.c gnome-desktop-new/libgnome-desktop/gnome-desktop-item.c
--- gnome-desktop-2.0.0/libgnome-desktop/gnome-desktop-item.c	2002-06-03 16:24:18.000000000 -0400
+++ gnome-desktop-new/libgnome-desktop/gnome-desktop-item.c	2002-06-13 07:35:05.000000000 -0400
@@ -902,7 +902,11 @@
 static const char *
 lookup_best_locale (const GnomeDesktopItem *item, const char *key)
 {
+#ifdef HAVE_LC_MESSAGES
 	const GList *list = gnome_i18n_get_language_list ("LC_MESSAGES");
+#else
+	const GList *list = gnome_i18n_get_language_list ("LC_ALL");
+#endif
 	while (list != NULL) {
 		const char *ret;
 		const char *locale = list->data;
@@ -2384,7 +2388,11 @@
 static const char *
 get_language (void)
 {
+#ifdef HAVE_LC_MESSAGES
 	const GList *list = gnome_i18n_get_language_list ("LC_MESSAGES");
+#else
+	const GList *list = gnome_i18n_get_language_list ("LC_ALL");
+#endif
 	while (list != NULL) {
 		const char *lang = list->data;
 		/* find first without encoding  */
diff -Naur gnome-desktop-2.0.0/libgnome-desktop/gnome-ditem-edit.c gnome-desktop-new/libgnome-desktop/gnome-ditem-edit.c
--- gnome-desktop-2.0.0/libgnome-desktop/gnome-ditem-edit.c	2002-06-03 20:02:04.000000000 -0400
+++ gnome-desktop-new/libgnome-desktop/gnome-ditem-edit.c	2002-06-13 07:34:32.000000000 -0400
@@ -37,6 +37,7 @@
 #include <libgnomeui/gnome-uidefs.h>
 #include <libgnomeui/gnome-icon-entry.h>
 
+#include "config.h"
 #include "gnome-desktop-item.h"
 #include "gnome-ditem-edit.h"
 
@@ -557,7 +558,11 @@
 	 * and en_US as the second. gnome-desktop-item handles all this correctly
 	 * and needs a function to return to us the locale it's using as default.
 	 */
+#ifdef HAVE_LC_MESSAGES
 	langs = gnome_i18n_get_language_list ("LC_MESSAGES");
+#else
+	langs = gnome_i18n_get_language_list ("LC_ALL");
+#endif
 	tmp = langs ? langs->data : NULL;
 	if ((tmp && !strcmp (tmp, lang)) || (!tmp && !strcmp (lang, "C"))) {
 		gtk_entry_set_text (
@@ -1028,7 +1033,11 @@
 	const GList *list;
 	const GList *l;
 
+#ifdef HAVE_LC_MESSAGES
 	list = gnome_i18n_get_language_list ("LC_MESSAGES");
+#else
+	list = gnome_i18n_get_language_list ("LC_ALL");
+#endif
 
 	for (l = list; l; l = l->next)
 		if (!strchr (l->data, '.'))
diff -Naur gnome-desktop-2.0.0/libgnome-desktop/gnome-theme-parser.c gnome-desktop-new/libgnome-desktop/gnome-theme-parser.c
--- gnome-desktop-2.0.0/libgnome-desktop/gnome-theme-parser.c	2002-05-16 14:12:46.000000000 -0400
+++ gnome-desktop-new/libgnome-desktop/gnome-theme-parser.c	2002-06-13 07:31:15.000000000 -0400
@@ -21,6 +21,7 @@
 #include <locale.h>
 #include <stdlib.h>
 
+#include "config.h"
 #include "gnome-theme-parser.h"
 
 typedef struct _GnomeThemeFileSection GnomeThemeFileSection;
@@ -763,7 +764,11 @@
 {
   char *p, *lang;
 
+#ifdef HAVE_LC_MESSAGES
   lang = g_strdup (setlocale (LC_MESSAGES, NULL));
+#else
+  lang = g_strdup (setlocale (LC_ALL, NULL));
+#endif
   
   if (lang)
     {
diff -Naur gnome-desktop-2.0.0/ltmain.sh gnome-desktop-new/ltmain.sh
--- gnome-desktop-2.0.0/ltmain.sh	2002-06-10 01:33:26.000000000 -0400
+++ gnome-desktop-new/ltmain.sh	2002-06-13 07:15:01.000000000 -0400
@@ -1777,7 +1777,7 @@
 
 	  if test "$installed" = no; then
 	    notinst_deplibs="$notinst_deplibs $lib"
-	    need_relink=yes
+	    # need_relink=yes
 	  fi
 
 	  if test -n "$old_archive_from_expsyms_cmds"; then
