diff -Naur libgnomecanvas-2.0.0/configure libgnomecanvas-new/configure
--- libgnomecanvas-2.0.0/configure	2002-06-07 04:30:14.000000000 -0400
+++ libgnomecanvas-new/configure	2002-06-12 03:20:43.000000000 -0400
@@ -5137,7 +5137,7 @@
     # FIXME: Relying on posixy $() will cause problems for
     #        cross-compilation, but unfortunately the echo tests do not
     #        yet detect zsh echo's removal of \ escapes.
-    archive_cmds='$nonopt $(test "x$module" = xyes && echo -bundle || echo -dynamiclib) $allow_undefined_flag -o $lib $libobjs $deplibs$linker_flags -install_name $rpath/$soname $verstring'
+    archive_cmds='$nonopt $(test x$module = xyes && echo -bundle || echo -dynamiclib) $allow_undefined_flag -o $lib $libobjs $deplibs$linker_flags $(test x$module = xyes || echo -install_name $rpath/$soname $verstring)'
     # We need to add '_' to the symbols in $export_symbols first
     #archive_expsym_cmds="$archive_cmds"' && strip -s $export_symbols'
     hardcode_direct=yes
diff -Naur libgnomecanvas-2.0.0/demos/Makefile.in libgnomecanvas-new/demos/Makefile.in
--- libgnomecanvas-2.0.0/demos/Makefile.in	2002-06-07 04:30:58.000000000 -0400
+++ libgnomecanvas-new/demos/Makefile.in	2002-06-12 03:20:43.000000000 -0400
@@ -113,7 +113,7 @@
 INCLUDES =  	-I$(top_srcdir) 					-I$(top_builddir)					$(WARN_CFLAGS)						$(LIBGNOMECANVAS_CFLAGS)			        -DGNOMECANVASLIBDIR=\""$(libdir)"\" 		        -DGNOMECANVASDATADIR=\""$(datadir)"\" 		        -DGNOMECANVASPIXMAPDIR=\""$(datadir)/pixmaps"\"	        -DGNOMECANVASBINDIR=\""$(bindir)"\" 		        -DGNOMECANVASLOCALSTATEDIR=\""$(localstatedir)"\" 	        -DGNOMECANVASLOCALEDIR=\""$(gnomelocaledir)"\" 		-DGTK_VERSION=\""$(GTK_VERSION)"\"			-DG_LOG_DOMAIN=\"GnomeCanvas\"				-DVERSION=\"$(VERSION)\"
 
 
-LDADD =  	$(top_builddir)/libgnomecanvas/libgnomecanvas-2.la 	$(INTLLIBS)
+LDADD =  	$(top_builddir)/libgnomecanvas/libgnomecanvas-2.la 	$(LIBGNOMECANVAS_LIBS)	$(INTLLIBS)
 
 
 noinst_PROGRAMS = canvas_demo
diff -Naur libgnomecanvas-2.0.0/libgnomecanvas/libgnomecanvas-2.0.pc.in libgnomecanvas-new/libgnomecanvas/libgnomecanvas-2.0.pc.in
--- libgnomecanvas-2.0.0/libgnomecanvas/libgnomecanvas-2.0.pc.in	2001-10-30 12:28:46.000000000 -0500
+++ libgnomecanvas-new/libgnomecanvas/libgnomecanvas-2.0.pc.in	2002-06-12 03:21:12.000000000 -0400
@@ -5,7 +5,7 @@
 
 Name: libgnomecanvas-2.0
 Description: libgnomecanvas
-Requires: libart-2.0 pango pangoft2 gtk+-2.0
+Requires: gtk+-2.0 libart-2.0 pango pangoft2
 Version: @VERSION@
 Libs: -L${libdir} -lgnomecanvas-2
 Cflags: -I${includedir}/libgnomecanvas-2.0
diff -Naur libgnomecanvas-2.0.0/ltmain.sh libgnomecanvas-new/ltmain.sh
--- libgnomecanvas-2.0.0/ltmain.sh	2002-06-07 04:30:06.000000000 -0400
+++ libgnomecanvas-new/ltmain.sh	2002-06-12 03:20:43.000000000 -0400
@@ -1741,7 +1741,7 @@
 
 	  if test "$installed" = no; then
 	    notinst_deplibs="$notinst_deplibs $lib"
-	    need_relink=yes
+	    # need_relink=yes
 	  fi
 
 	  if test -n "$old_archive_from_expsyms_cmds"; then
