diff -Naur libgnomeui-2.0.0/configure libgnomeui-new/configure
--- libgnomeui-2.0.0/configure	2002-06-07 05:12:38.000000000 -0400
+++ libgnomeui-new/configure	2002-06-11 14:03:45.000000000 -0400
@@ -5144,7 +5144,7 @@
     # FIXME: Relying on posixy $() will cause problems for
     #        cross-compilation, but unfortunately the echo tests do not
     #        yet detect zsh echo's removal of \ escapes.
-    archive_cmds='$nonopt $(test "x$module" = xyes && echo -bundle || echo -dynamiclib) $allow_undefined_flag -o $lib $libobjs $deplibs$linker_flags -install_name $rpath/$soname $verstring'
+    archive_cmds='$nonopt $(test x$module = xyes && echo -bundle || echo -dynamiclib) $allow_undefined_flag -o $lib $libobjs $deplibs$linker_flags $(test x$module = xyes || echo -install_name $rpath/$soname $verstring)'
     # We need to add '_' to the symbols in $export_symbols first
     #archive_expsym_cmds="$archive_cmds"' && strip -s $export_symbols'
     hardcode_direct=yes
diff -Naur libgnomeui-2.0.0/demos/Makefile.in libgnomeui-new/demos/Makefile.in
--- libgnomeui-2.0.0/demos/Makefile.in	2002-06-07 05:42:55.000000000 -0400
+++ libgnomeui-new/demos/Makefile.in	2002-06-12 08:48:36.000000000 -0400
@@ -144,11 +144,11 @@
 X_EXTRA_LIBS = @X_EXTRA_LIBS@
 X_PRE_LIBS = @X_PRE_LIBS@
 mdi_demo_OBJECTS =  mdi_demo.$(OBJEXT)
-mdi_demo_LDADD = $(LDADD)
+mdi_demo_LDADD = $(LDADD) $(LIBGNOMEUI_LIBS)
 mdi_demo_DEPENDENCIES =  $(top_builddir)/libgnomeui/libgnomeui-2.la
 mdi_demo_LDFLAGS = 
 animator_demo_OBJECTS =  animator_demo.$(OBJEXT)
-animator_demo_LDADD = $(LDADD)
+animator_demo_LDADD = $(LDADD) $(LIBGNOMEUI_LIBS)
 animator_demo_DEPENDENCIES =  $(top_builddir)/libgnomeui/libgnomeui-2.la
 animator_demo_LDFLAGS = 
 COMPILE = $(CC) $(DEFS) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS)
diff -Naur libgnomeui-2.0.0/doc/reference/Makefile.in libgnomeui-new/doc/reference/Makefile.in
--- libgnomeui-2.0.0/doc/reference/Makefile.in	2002-06-07 05:42:57.000000000 -0400
+++ libgnomeui-new/doc/reference/Makefile.in	2002-06-13 02:06:16.000000000 -0400
@@ -162,7 +162,7 @@
 # CFLAGS and LDFLAGS for compiling scan program. Only needed if your app/lib
 # contains GtkObjects/GObjects and you want to document signals and properties.
 GTKDOC_CFLAGS = -I$(top_builddir) -I$(top_srcdir) $(LIBGNOMEUI_CFLAGS) $(CFLAGS)
-GTKDOC_LIBS = $(top_builddir)/libgnomeui/libgnomeui-2.la $(LDFLAGS)
+GTKDOC_LIBS = $(top_builddir)/libgnomeui/.libs/libgnomeui-2.a $(LDFLAGS) $(LIBGNOMEUI_LIBS) -L/usr/X11R6/lib -lSM -lICE
 
 GTKDOC_CC = $(LIBTOOL) --mode=compile $(CC)
 GTKDOC_LD = $(LIBTOOL) --mode=link $(CC)
diff -Naur libgnomeui-2.0.0/libgnomeui/Makefile.in libgnomeui-new/libgnomeui/Makefile.in
--- libgnomeui-2.0.0/libgnomeui/Makefile.in	2002-06-07 05:42:52.000000000 -0400
+++ libgnomeui-new/libgnomeui/Makefile.in	2002-06-12 08:46:27.000000000 -0400
@@ -208,7 +208,7 @@
 PROGRAMS =  $(bin_PROGRAMS)
 
 gnome_segv2_OBJECTS =  gnome_segv.$(OBJEXT)
-gnome_segv2_LDADD = $(LDADD)
+gnome_segv2_LDADD = $(LDADD) $(LIBGNOMEUI_LIBS)
 gnome_segv2_DEPENDENCIES =  libgnomeui-2.la
 gnome_segv2_LDFLAGS = 
 COMPILE = $(CC) $(DEFS) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS)
diff -Naur libgnomeui-2.0.0/libgnomeui/gnome-dateedit.c libgnomeui-new/libgnomeui/gnome-dateedit.c
--- libgnomeui-2.0.0/libgnomeui/gnome-dateedit.c	2002-06-03 15:40:48.000000000 -0400
+++ libgnomeui-new/libgnomeui/gnome-dateedit.c	2002-06-12 09:05:11.000000000 -0400
@@ -48,6 +48,29 @@
 
 #include "libgnomeui-access.h"
 
+static char *
+strtok_r (char *s, const char *delim, char **save_ptr)
+{
+	char *token;
+
+	if (s == NULL)
+		s = *save_ptr;
+
+	s += strspn (s, delim);
+	if (*s == '\0')
+		return NULL;
+
+	token = s;
+	s = strpbrk (token, delim);
+	if (s == NULL) {
+		*save_ptr = strchr (token, '\0');
+	} else {
+		*s = '\0';
+		*save_ptr = s + 1;
+	}
+	return token;
+}
+
 struct _GnomeDateEditPrivate {
 	GtkWidget *date_entry;
 	GtkWidget *date_button;
diff -Naur libgnomeui-2.0.0/ltmain.sh libgnomeui-new/ltmain.sh
--- libgnomeui-2.0.0/ltmain.sh	2002-06-07 05:12:27.000000000 -0400
+++ libgnomeui-new/ltmain.sh	2002-06-11 14:03:07.000000000 -0400
@@ -1741,7 +1741,7 @@
 
 	  if test "$installed" = no; then
 	    notinst_deplibs="$notinst_deplibs $lib"
-	    need_relink=yes
+	    # need_relink=yes
 	  fi
 
 	  if test -n "$old_archive_from_expsyms_cmds"; then
diff -Naur libgnomeui-2.0.0/test-gnome/Makefile.in libgnomeui-new/test-gnome/Makefile.in
--- libgnomeui-2.0.0/test-gnome/Makefile.in	2002-06-07 05:42:55.000000000 -0400
+++ libgnomeui-new/test-gnome/Makefile.in	2002-06-12 08:47:31.000000000 -0400
@@ -162,19 +162,19 @@
 X_EXTRA_LIBS = @X_EXTRA_LIBS@
 X_PRE_LIBS = @X_PRE_LIBS@
 test_gnome_OBJECTS =  testgnome.$(OBJEXT)
-test_gnome_LDADD = $(LDADD)
+test_gnome_LDADD = $(LDADD) $(LIBGNOMEUI_LIBS)
 test_gnome_DEPENDENCIES =  $(top_builddir)/libgnomeui/libgnomeui-2.la
 test_gnome_LDFLAGS = 
 test_druid_OBJECTS =  testdruid.$(OBJEXT)
-test_druid_LDADD = $(LDADD)
+test_druid_LDADD = $(LDADD) $(LIBGNOMEUI_LIBS)
 test_druid_DEPENDENCIES =  $(top_builddir)/libgnomeui/libgnomeui-2.la
 test_druid_LDFLAGS = 
 test_entry_OBJECTS =  testentry.$(OBJEXT)
-test_entry_LDADD = $(LDADD)
+test_entry_LDADD = $(LDADD) $(LIBGNOMEUI_LIBS)
 test_entry_DEPENDENCIES =  $(top_builddir)/libgnomeui/libgnomeui-2.la
 test_entry_LDFLAGS = 
 test_iconlist_OBJECTS =  testiconlist.$(OBJEXT)
-test_iconlist_LDADD = $(LDADD)
+test_iconlist_LDADD = $(LDADD) $(LIBGNOMEUI_LIBS)
 test_iconlist_DEPENDENCIES =  $(top_builddir)/libgnomeui/libgnomeui-2.la
 test_iconlist_LDFLAGS = 
 COMPILE = $(CC) $(DEFS) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS)
