--- feResource.cc_ori	2004-04-28 18:40:45.000000000 +0200
+++ feResource.cc	2004-04-28 18:51:29.000000000 +0200
@@ -83,31 +83,32 @@
 {
   {"SearchPath",    's',     feResPath,  NULL,
    "$SINGULARPATH;"
+   "%r/share/singular/LIB;"
+   "%d/LIB;"
    "%b/LIB;"
    "%r/LIB;"
    "%r/../LIB;"
-   "%d/LIB;"
    "%d/../LIB"
    ""},
-  {"Singular",  'S',    feResBinary,"SINGULAR_EXECUTABLE",  "%d/"S_UNAME"/Singular"EXT,""},
-  {"BinDir",    'b',    feResDir,   "SINGULAR_BIN_DIR",     "%d/"S_UNAME            ""},
+  {"Singular",  'S',    feResBinary,"SINGULAR_EXECUTABLE",  "%b/Singular"EXT,""},
+  {"BinDir",    'b',    feResDir,   "SINGULAR_BIN_DIR",     "%r/bin"            ""},
   {"RootDir",   'r',    feResDir,   "SINGULAR_ROOT_DIR",    "%b/..",                ""},
   {"DefaultDir",'d',    feResDir,   "SINGULAR_DEFAULT_DIR",  SINGULAR_DEFAULT_DIR,  ""},
-  {"InfoFile",  'i',    feResFile,  "SINGULAR_INFO_FILE",   "%r/info/singular.hlp", ""},
-  {"IdxFile",   'x',    feResFile,  "SINGULAR_IDX_FILE",    "%r/doc/singular.idx",  ""},
-  {"HtmlDir",   'h',    feResDir,   "SINGULAR_HTML_DIR",    "%r/html",              ""},
+  {"InfoFile",  'i',    feResFile,  "SINGULAR_INFO_FILE",   "%r/share/info/singular.info", ""},
+  {"IdxFile",   'x',    feResFile,  "SINGULAR_IDX_FILE",    "%r/share/singular/singular.idx",  ""},
+  {"HtmlDir",   'h',    feResDir,   "SINGULAR_HTML_DIR",    "%r/share/doc/singular/html",              ""},
   {"ManualUrl", 'u',    feResUrl,   "SINGULAR_URL",         "http://www.singular.uni-kl.de/Manual/"S_VERSION1,    ""},
-  {"ExDir",      'm',   feResDir,   "SINGULAR_EXAMPLES_DIR","%r/examples",              ""},
+  {"ExDir",      'm',   feResDir,   "SINGULAR_EXAMPLES_DIR","%r/share/doc/singular/examples",              ""},
 #if !defined(macintosh)
-  {"netscape",  'N',    feResBinary,"NETSCAPE",             "%b/netscape",          ""},
+  {"netscape",  'N',    feResBinary,"NETSCAPE",             "%b/mozilla",          ""},
   {"info",      'I',    feResBinary,"INFO",                 "%b/info",              ""},
   {"tkinfo",    'T',    feResBinary,"TKINFO",               "%b/tkinfo",            ""},
 #ifdef ix86_Win
-  {"rxvt",     'X',    feResBinary,"RXVT",                "%b/rxvt.exe",             ""},
+  {"rxvt",     'X',    feResBinary,"RXVT",                "%b/rxvt",             ""},
 #else
-  {"xterm",     'X',    feResBinary,"XTERM",                "%b/xterm",             ""},
+  {"xterm",     'X',    feResBinary,"XTERM",                "/usr/X11R6/bin/xterm",             ""},
 #endif
-  {"Path",      'p',    feResPath,  NULL,                   "%b;$PATH",         ""},
+  {"Path",      'p',    feResPath,  NULL,                   "$PATH",         ""},
 #endif // ! defined(macintosh)
 
 #ifdef ESINGULAR
@@ -115,11 +116,11 @@
   {"xemacs",    'A',    feResBinary, "ESINGULAR_EMACS",      "%b/xemacs"EXT,              ""},
   {"SingularEmacs",'M',feResBinary, "ESINGULAR_SINGULAR",    "%b/Singular"EXT,           ""},
   {"EmacsLoad",'l',    feResFile,   "ESINGULAR_EMACS_LOAD",  "%e/.emacs-singular",             ""},
-  {"EmacsDir",  'e',    feResDir,   "ESINGULAR_EMACS_DIR",   "%r/emacs",             ""},
+  {"EmacsDir",  'e',    feResDir,   "ESINGULAR_EMACS_DIR",   "%r/share/singular/emacs",             ""},
 #elif defined(TSINGULAR)
   {"SingularXterm",'M',feResBinary, "TSINGULAR_SINGULAR",    "%b/Singular"EXT,           ""},
 #else
-  {"EmacsDir",  'e',    feResDir,   "SINGULAR_EMACS_DIR",   "%r/emacs",             ""},
+  {"EmacsDir",  'e',    feResDir,   "SINGULAR_EMACS_DIR",   "%r/share/singular/emacs",             ""},
 #endif
   {NULL, 0, feResUndef, NULL, NULL, NULL}, // must be the last record
 };
