--- xc/programs/Xserver/hw/darwin/bundle/quartzCursor.c	Tue Aug 20 17:56:15 2002
+++ xc/programs/Xserver/hw/darwin/bundle/quartzCursor.c.fixed	Tue Oct 15 13:40:18 2002
@@ -63,6 +63,7 @@
 
 // Add a color entry to a ctab
 #define CTAB_ENTER(ctab, index, r, g, b) \
+    ctab->ctTable[index].value = index; \
     ctab->ctTable[index].rgb.red = r; \
     ctab->ctTable[index].rgb.green = g; \
     ctab->ctTable[index].rgb.blue = b
