diff -u -r1.43 darwin.cf
--- config/cf/darwin.cf	11 Nov 2003 01:29:46 -0000	1.43
+++ config/cf/darwin.cf	14 Nov 2003 20:48:37 -0000
@@ -155,8 +155,8 @@
 #endif
 
 /* we have a gcc compatible compiler, but its not called gcc */
-#define CcCmd                   /usr/bin/cc
-#define CplusplusCmd            /usr/bin/c++
+#define CcCmd                   gcc-3.3
+#define CplusplusCmd            g++-3.3
 
 /* default flags to pass to cc */
 #ifndef DefaultCCOptions
