diff -uNr qt-x11-free-3.3.4/Makefile qt-x11-free-3.3.4-new/Makefile
--- qt-x11-free-3.3.4/Makefile	Mon Jan  3 11:08:00 2005
+++ qt-x11-free-3.3.4-new/Makefile	Fri Jan 28 17:31:11 2005
@@ -11,7 +11,7 @@
 install: FORCE
 	@$(MAKE) qt.install
 
-all: symlinks src-qmake src-moc sub-src sub-tools sub-tutorial sub-examples
+all: symlinks src-qmake src-moc sub-src sub-tools
 	@echo
 	@echo "The Qt library is now built in ./lib"
 	@echo "The Qt examples are built in the directories in ./examples"
diff -uNr qt-x11-free-3.3.4/configure qt-x11-free-3.3.4-new/configure
--- qt-x11-free-3.3.4/configure	Mon Oct 18 13:17:00 2004
+++ qt-x11-free-3.3.4-new/configure	Fri Jan 28 17:31:11 2005
@@ -1046,8 +1046,8 @@
 [ -z "$QT_INSTALL_DATA" ]         && QT_INSTALL_DATA=$QT_INSTALL_PREFIX
 # default PREFIX/translations
 [ -z "$QT_INSTALL_TRANSLATIONS" ] && QT_INSTALL_TRANSLATIONS=$QT_INSTALL_PREFIX/translations
-# default PREFIX/etc/settings
-[ -z "$QT_INSTALL_SYSCONF" ]      && QT_INSTALL_SYSCONF=$QT_INSTALL_PREFIX/etc/settings
+# default PREFIX/etc/qt
+[ -z "$QT_INSTALL_SYSCONF" ]      && QT_INSTALL_SYSCONF=$QT_INSTALL_PREFIX/etc/qt
 
 # generate qconfig.cpp
 [ -d $outpath/src/tools ] || mkdir -p $outpath/src/tools
