diff -uNbr cheesetracker-0.5.5/interface_gtk/interface_popups.cpp cheesetracker-0.5.5-patched/interface_gtk/interface_popups.cpp
--- cheesetracker-0.5.5/interface_gtk/interface_popups.cpp	Wed Apr 10 21:57:05 2002
+++ cheesetracker-0.5.5-patched/interface_gtk/interface_popups.cpp	Mon Oct 28 01:09:31 2002
@@ -21,6 +21,8 @@
 #include "../icons/help.xpm"
 #include "../icons/ctsplash.xpm"
 
+using std::string;
+
 /************************
 	WINDOW FILE POPUP
 	**************************/
diff -uNbr cheesetracker-0.5.5/interface_gtk/properties.h cheesetracker-0.5.5-patched/interface_gtk/properties.h
--- cheesetracker-0.5.5/interface_gtk/properties.h	Wed Apr 10 13:39:58 2002
+++ cheesetracker-0.5.5-patched/interface_gtk/properties.h	Mon Oct 28 01:19:25 2002
@@ -19,9 +19,12 @@
 #define PROPERTIES_H
 
 #include "typedefs.h"
-#include <hash_map>
+#include <ext/hash_map>
 #include <stdio.h>
 
+using __gnu_cxx::hash_map;
+using __gnu_cxx::hash;
+using std::string;
 
 /**
   *@author Juan Sebastian Linietsky
diff -uNbr cheesetracker-0.5.5/trackercore/file_format_manager.cpp cheesetracker-0.5.5-patched/trackercore/file_format_manager.cpp
--- cheesetracker-0.5.5/trackercore/file_format_manager.cpp	Wed Apr 10 01:57:30 2002
+++ cheesetracker-0.5.5-patched/trackercore/file_format_manager.cpp	Mon Oct 28 00:27:45 2002
@@ -16,6 +16,10 @@
  ***************************************************************************/
 
 #include "file_format_manager.h"
+#include <iostream>
+
+using std::cout;
+using std::endl;
 
 File_Format_Manager::File_Format_Manager(){
 }
diff -uNbr cheesetracker-0.5.5/trackercore/file_format_manager.h cheesetracker-0.5.5-patched/trackercore/file_format_manager.h
--- cheesetracker-0.5.5/trackercore/file_format_manager.h	Wed Apr 10 01:57:30 2002
+++ cheesetracker-0.5.5-patched/trackercore/file_format_manager.h	Mon Oct 28 00:25:30 2002
@@ -22,6 +22,8 @@
 #include "loader.h"
 #include "saver.h"
 
+using std::list;
+
 /**Collection of loaders/savers/etc
   *@author Juan Linietsky
   */
diff -uNbr cheesetracker-0.5.5/trackercore/instrument.h cheesetracker-0.5.5-patched/trackercore/instrument.h
--- cheesetracker-0.5.5/trackercore/instrument.h	Sun Mar  3 22:52:38 2002
+++ cheesetracker-0.5.5-patched/trackercore/instrument.h	Mon Oct 28 00:29:43 2002
@@ -22,6 +22,7 @@
 #include "note.h"
 #include "envelope.h"
 
+using std::string;
 
 /******************************
  instrument.h
diff -uNbr cheesetracker-0.5.5/trackercore/loader_it.cpp cheesetracker-0.5.5-patched/trackercore/loader_it.cpp
--- cheesetracker-0.5.5/trackercore/loader_it.cpp	Tue Apr 30 20:11:51 2002
+++ cheesetracker-0.5.5-patched/trackercore/loader_it.cpp	Mon Oct 28 00:32:00 2002
@@ -16,6 +16,10 @@
  ***************************************************************************/
 
 #include "loader_it.h"
+#include <iostream>
+
+using std::cout;
+using std::endl;
 
 const char * Loader_IT::IT_Version[]={
 		
diff -uNbr cheesetracker-0.5.5/trackercore/loader_it_patterns.cpp cheesetracker-0.5.5-patched/trackercore/loader_it_patterns.cpp
--- cheesetracker-0.5.5/trackercore/loader_it_patterns.cpp	Wed May  9 09:43:01 2001
+++ cheesetracker-0.5.5-patched/trackercore/loader_it_patterns.cpp	Mon Oct 28 00:33:17 2002
@@ -16,8 +16,10 @@
  ***************************************************************************/
 
 #include "loader_it.h"
+#include <iostream>
 
-
+using std::cout;
+using std::endl;
 
 int Loader_IT::load_patterns() {
 
diff -uNbr cheesetracker-0.5.5/trackercore/loader_it_samples.cpp cheesetracker-0.5.5-patched/trackercore/loader_it_samples.cpp
--- cheesetracker-0.5.5/trackercore/loader_it_samples.cpp	Wed Apr 10 23:16:38 2002
+++ cheesetracker-0.5.5-patched/trackercore/loader_it_samples.cpp	Mon Oct 28 00:34:01 2002
@@ -16,6 +16,10 @@
  ***************************************************************************/
 #include "loader_it.h"
 #include "player_data.h"
+#include <iostream>
+
+using std::cout;
+using std::endl;
 
 int Loader_IT::load_sample_info(IT_Sample *p_sample) {
 
diff -uNbr cheesetracker-0.5.5/trackercore/loader_wav.cpp cheesetracker-0.5.5-patched/trackercore/loader_wav.cpp
--- cheesetracker-0.5.5/trackercore/loader_wav.cpp	Wed Jan 30 01:47:16 2002
+++ cheesetracker-0.5.5-patched/trackercore/loader_wav.cpp	Mon Oct 28 00:36:14 2002
@@ -2,6 +2,9 @@
 
 #include <iostream>
 
+using std::cout;
+using std::endl;
+
 struct format_chunk {
 
    short int channels;
diff -uNbr cheesetracker-0.5.5/trackercore/loader_xm.h cheesetracker-0.5.5-patched/trackercore/loader_xm.h
--- cheesetracker-0.5.5/trackercore/loader_xm.h	Thu Apr 11 01:50:38 2002
+++ cheesetracker-0.5.5-patched/trackercore/loader_xm.h	Mon Oct 28 00:35:40 2002
@@ -22,6 +22,9 @@
 #include "song.h"
 #include "file_reader.h"
 #include <list>
+
+using std::list;
+
 /**
   *@author red
   */
diff -uNbr cheesetracker-0.5.5/trackercore/mixer_procedure.h cheesetracker-0.5.5-patched/trackercore/mixer_procedure.h
--- cheesetracker-0.5.5/trackercore/mixer_procedure.h	Tue Apr 30 03:11:28 2002
+++ cheesetracker-0.5.5-patched/trackercore/mixer_procedure.h	Mon Oct 28 00:39:05 2002
@@ -19,7 +19,11 @@
 #define MIXER_PROCEDURE_H
 
 #include  "sample_data.h"
-#include <hash_map>
+#include <iostream>
+#include <ext/hash_map>
+using __gnu_cxx::hash_map;
+using __gnu_cxx::hash;
+using std::string;
 
 /**Mixer Procedure
 
diff -uNbr cheesetracker-0.5.5/trackercore/mixer_software_control.cpp cheesetracker-0.5.5-patched/trackercore/mixer_software_control.cpp
--- cheesetracker-0.5.5/trackercore/mixer_software_control.cpp	Tue Apr 30 21:09:44 2002
+++ cheesetracker-0.5.5-patched/trackercore/mixer_software_control.cpp	Mon Oct 28 00:41:25 2002
@@ -17,6 +17,10 @@
 
 #include "mixer_software.h"
 #include <stdio.h>
+#include <iostream>
+
+using std::cout;
+using std::endl;
 
 void Mixer_Software::setup_voice(int p_voice_index,Sample_Data *p_sample_data,Sint32 p_start_index) {
 
diff -uNbr cheesetracker-0.5.5/trackercore/pattern.cpp cheesetracker-0.5.5-patched/trackercore/pattern.cpp
--- cheesetracker-0.5.5/trackercore/pattern.cpp	Wed May  9 09:43:01 2001
+++ cheesetracker-0.5.5-patched/trackercore/pattern.cpp	Mon Oct 28 00:28:39 2002
@@ -16,7 +16,10 @@
  ***************************************************************************/
 
 #include "pattern.h"
+#include <iostream>
 
+using std::cout;
+using std::endl;
 
 Note Pattern::empty_note;
 Mutex_Lock *Pattern::data_lock=NULL;
diff -uNbr cheesetracker-0.5.5/trackercore/pattern.h cheesetracker-0.5.5-patched/trackercore/pattern.h
--- cheesetracker-0.5.5/trackercore/pattern.h	Wed May  9 09:43:05 2001
+++ cheesetracker-0.5.5-patched/trackercore/pattern.h	Mon Oct 28 00:24:28 2002
@@ -24,6 +24,9 @@
 #include "note.h"
 #include "mutex_lock.h"
 
+using std::map;
+using std::vector;
+
 #define PATTERN_WIDTH 64
 #define MIN_PATTERN_LENGTH 32
 #define MAX_PATTERN_LENGTH 200
diff -uNbr cheesetracker-0.5.5/trackercore/player_data_notes.cpp cheesetracker-0.5.5-patched/trackercore/player_data_notes.cpp
--- cheesetracker-0.5.5/trackercore/player_data_notes.cpp	Wed Apr 10 23:16:38 2002
+++ cheesetracker-0.5.5-patched/trackercore/player_data_notes.cpp	Mon Oct 28 00:31:25 2002
@@ -17,6 +17,10 @@
 
 #include "player_data.h"
 #include <stdio.h>
+#include <iostream>
+
+using std::cout;
+using std::endl;
 
 void Player_Data::process_new_note(int p_track,Uint8 p_note) { // if there's really a new note....
 
diff -uNbr cheesetracker-0.5.5/trackercore/sample.h cheesetracker-0.5.5-patched/trackercore/sample.h
--- cheesetracker-0.5.5/trackercore/sample.h	Wed May  9 09:43:05 2001
+++ cheesetracker-0.5.5-patched/trackercore/sample.h	Mon Oct 28 00:24:57 2002
@@ -22,6 +22,8 @@
 #include "typedefs.h"
 #include "sample_data.h"
 
+using std::string;
+
 /**
   *@author Juan Linietsky
   */
diff -uNbr cheesetracker-0.5.5/trackercore/sample_data.cpp cheesetracker-0.5.5-patched/trackercore/sample_data.cpp
--- cheesetracker-0.5.5/trackercore/sample_data.cpp	Wed Apr 10 04:54:31 2002
+++ cheesetracker-0.5.5-patched/trackercore/sample_data.cpp	Mon Oct 28 00:35:07 2002
@@ -16,6 +16,10 @@
  ***************************************************************************/
 
 #include "sample_data.h"
+#include <iostream>
+
+using std::cout;
+using std::endl;
 
 Sample_Data::Sample_Data(){
 
diff -uNbr cheesetracker-0.5.5/trackercore/saver_it.cpp cheesetracker-0.5.5-patched/trackercore/saver_it.cpp
--- cheesetracker-0.5.5/trackercore/saver_it.cpp	Wed Apr 10 04:25:27 2002
+++ cheesetracker-0.5.5-patched/trackercore/saver_it.cpp	Mon Oct 28 00:37:26 2002
@@ -16,6 +16,7 @@
  ***************************************************************************/
 
 #include "saver_it.h"
+#include <cctype>
 
 Saver_IT::Saver_IT(){
 
diff -uNbr cheesetracker-0.5.5/trackercore/saver_wav.cpp cheesetracker-0.5.5-patched/trackercore/saver_wav.cpp
--- cheesetracker-0.5.5/trackercore/saver_wav.cpp	Wed Jan 30 01:34:40 2002
+++ cheesetracker-0.5.5-patched/trackercore/saver_wav.cpp	Mon Oct 28 00:42:21 2002
@@ -13,7 +13,10 @@
  ***************************************************************************/
 
 #include "saver_wav.h"
+#include <iostream>
 
+using std::cout;
+using std::endl;
 
 int Saver_Wav::save(char *p_filename) {
 
diff -uNbr cheesetracker-0.5.5/trackercore/sound_driver.h cheesetracker-0.5.5-patched/trackercore/sound_driver.h
--- cheesetracker-0.5.5/trackercore/sound_driver.h	Mon Mar  4 15:59:45 2002
+++ cheesetracker-0.5.5-patched/trackercore/sound_driver.h	Mon Oct 28 00:39:57 2002
@@ -18,8 +18,10 @@
 #ifndef SOUND_DRIVER_H
 #define SOUND_DRIVER_H
 
-
 #include "mixer.h"
+#include <iostream>
+
+using std::string;
 
 /**sound driver base class
   *@author Juan Linietsky
diff -uNbr cheesetracker-0.5.5/trackercore/sound_driver_diskwrite.cpp cheesetracker-0.5.5-patched/trackercore/sound_driver_diskwrite.cpp
--- cheesetracker-0.5.5/trackercore/sound_driver_diskwrite.cpp	Mon Mar  4 17:27:27 2002
+++ cheesetracker-0.5.5-patched/trackercore/sound_driver_diskwrite.cpp	Mon Oct 28 00:43:05 2002
@@ -16,6 +16,10 @@
  ***************************************************************************/
 
 #include "sound_driver_diskwrite.h"
+#include <iostream>
+
+using std::cout;
+using std::endl;
 
 int Sound_Driver_DiskWrite::get_mix_frequency() {
 
diff -uNbr cheesetracker-0.5.5/trackercore/sound_driver_manager.h cheesetracker-0.5.5-patched/trackercore/sound_driver_manager.h
--- cheesetracker-0.5.5/trackercore/sound_driver_manager.h	Wed Apr 10 21:49:39 2002
+++ cheesetracker-0.5.5-patched/trackercore/sound_driver_manager.h	Mon Oct 28 00:40:33 2002
@@ -22,6 +22,8 @@
 #include "sound_driver.h"
 #include <vector>
 
+using std::vector;
+
 /**
   *@author Juan Linietsky
   */
diff -uNbr cheesetracker-0.5.5/waveout/sound_driver_oss.cpp cheesetracker-0.5.5-patched/waveout/sound_driver_oss.cpp
--- cheesetracker-0.5.5/waveout/sound_driver_oss.cpp	Wed Apr 10 23:46:44 2002
+++ cheesetracker-0.5.5-patched/waveout/sound_driver_oss.cpp	Fri Nov  8 15:49:30 2002
@@ -15,8 +15,17 @@
  *                                                                         *
  ***************************************************************************/
 
+#define LIBOSS_IOCTL OSS_IOCTL
+#define LIBOSS_OPEN  OSS_OPEN
+#define LIBOSS_CLOSE OSS_CLOSE
+#define LIBOSS_WRITE OSS_WRITE
+
 #include "sound_driver_oss.h"
 #include "math.h"
+#include <iostream>
+
+using std::cout;
+using std::endl;
 
 #define AUDIO_DEVICE "/dev/dsp"
 
@@ -36,7 +45,7 @@
 
 	}
 
-	if ( (sound_fd=open(AUDIO_DEVICE,O_WRONLY))<0 ) {
+	if ( (sound_fd=OSS_OPEN(AUDIO_DEVICE,O_WRONLY))<0 ) {
 
 		return SOUND_DRIVER_ERROR_OPENING_DEVICE;
 	}
@@ -53,7 +62,7 @@
 
 	fragment_size=(numfrags<<16)|fragsize;
 	
-	if( ioctl(sound_fd,SNDCTL_DSP_SETFRAGMENT,&fragment_size)<0 ) {
+	if( OSS_IOCTL(sound_fd,SNDCTL_DSP_SETFRAGMENT,&fragment_size)<0 ) {
 
 		finish();
 		return SOUND_DRIVER_ERROR_CONFIGURING_DEVICE;
@@ -61,7 +70,7 @@
 
 
 /* Ask device for supported formats */
-	if(ioctl(sound_fd,SNDCTL_DSP_GETFMTS,&supported_formats)<0) {
+	if(OSS_IOCTL(sound_fd,SNDCTL_DSP_GETFMTS,&supported_formats)<0) {
 
 		finish();
 		return SOUND_DRIVER_ERROR_CONFIGURING_DEVICE;
@@ -83,7 +92,7 @@
 		}
 	}
 
-	if ( (ioctl(sound_fd,SNDCTL_DSP_SETFMT,&play_precision)<0 ) && ( orig_precision!=play_precision )) {
+	if ( (OSS_IOCTL(sound_fd,SNDCTL_DSP_SETFMT,&play_precision)<0 ) && ( orig_precision!=play_precision )) {
 
 		finish();
 		return SOUND_DRIVER_ERROR_CONFIGURING_DEVICE;
@@ -91,7 +100,7 @@
 
 	orig_stereo=play_stereo=(mix_stereo)?2:1;
 
-	if ( ioctl(sound_fd,SNDCTL_DSP_CHANNELS,&play_stereo)<0 ) {
+	if ( OSS_IOCTL(sound_fd,SNDCTL_DSP_CHANNELS,&play_stereo)<0 ) {
 
 		finish();
 		return SOUND_DRIVER_ERROR_CONFIGURING_DEVICE;
@@ -99,7 +108,7 @@
 
 	play_rate=mix_frequency;
 
-	if ( (ioctl(sound_fd,SNDCTL_DSP_SPEED,&play_rate)<0) ) {
+	if ( (OSS_IOCTL(sound_fd,SNDCTL_DSP_SPEED,&play_rate)<0) ) {
 
 		finish();
 		return SOUND_DRIVER_ERROR_CONFIGURING_DEVICE;
@@ -107,9 +116,9 @@
 
 
 	/* This call fails on Linux/PPC */
-	if( (ioctl(sound_fd,SNDCTL_DSP_GETOSPACE,&buffinf)<0) ) {
+	if( (OSS_IOCTL(sound_fd,SNDCTL_DSP_GETOSPACE,&buffinf)<0) ) {
 
-		ioctl(sound_fd,SNDCTL_DSP_GETBLKSIZE,&buffinf.fragsize);
+		OSS_IOCTL(sound_fd,SNDCTL_DSP_GETBLKSIZE,&buffinf.fragsize);
 	}
 
 	
@@ -143,7 +152,7 @@
 
 bool Sound_Driver_OSS::finish() {
 
-	if ( sound_fd>=0 ) close(sound_fd);
+	if ( sound_fd>=0 ) OSS_CLOSE(sound_fd);
 
 	if (audiobuffer!=NULL) free(audiobuffer);
 	audiobuffer=NULL;	
@@ -163,7 +172,7 @@
 	
 	for (;;) {
 
-		if ((ioctl(sound_fd,SNDCTL_DSP_GETOSPACE,&buffinf)<0)) {
+		if ((OSS_IOCTL(sound_fd,SNDCTL_DSP_GETOSPACE,&buffinf)<0)) {
 
 			buffinf.fragments--;
 			buffinf.fragsize = buffinf.bytes = buffersize;
@@ -173,7 +182,7 @@
 
 		done=mixer->write_bytes(audiobuffer,buffinf.fragsize>buffinf.bytes?buffinf.bytes:buffinf.fragsize);
 
-                write(sound_fd,audiobuffer,done);
+                OSS_WRITE(sound_fd,audiobuffer,done);
 	}		
 
 	return true;	
diff -uNbr cheesetracker-0.5.5/waveout/sound_driver_oss.h cheesetracker-0.5.5-patched/waveout/sound_driver_oss.h
--- cheesetracker-0.5.5/waveout/sound_driver_oss.h	Wed May  9 09:43:05 2001
+++ cheesetracker-0.5.5-patched/waveout/sound_driver_oss.h	Mon Oct 28 00:44:40 2002
@@ -22,7 +22,7 @@
 
 #include <sound_driver.h>
 #include <fcntl.h>
-#include <sys/soundcard.h>
+#include <liboss/soundcard.h>
 #include <sys/ioctl.h>
 
 #include "mutex_lock_pthreads.h"
