diff -uNbr esound-0.2.27/Makefile.in esound-0.2.27-new/Makefile.in
--- esound-0.2.27/Makefile.in	Fri Jun  7 07:25:59 2002
+++ esound-0.2.27-new/Makefile.in	Thu Jun 13 20:52:48 2002
@@ -113,7 +113,7 @@
 lib_LTLIBRARIES = libesd.la $(libesddsp_la)
 
 libesd_la_LDFLAGS = -version-info $(ESD_VERSION_INFO)
-libesd_la_LIBADD = $(AUDIOFILE_LIBS) $(SOUND_LIBS)
+libesd_la_LIBADD = $(AUDIOFILE_LIBS)
 
 libesd_la_SOURCES =  	esdlib.c		esdmgr.c		esdfile.c		esd_config.c		audio.c			genrand.c		genrand.h		util.c
 
@@ -134,7 +134,7 @@
 bin_PROGRAMS =  	esd			esdcat			esdctl			esdfilt			esdmon			esdrec			esdsample		esdloop			$(ESDPLAY)
 
 
-esd_SOURCES =  	esd.c 			clients.c		filter.c		mix.c			players.c		proto.c			samples.c		$(getopt_src)   	util.c			esd-server.h
+esd_SOURCES =  	esd_audio.c	esd.c 			clients.c		filter.c		mix.c			players.c		proto.c			samples.c		$(getopt_src)   	util.c			esd-server.h
 
 
 esdcat_SOURCES =  	esdcat.c
diff -uNbr esound-0.2.27/acconfig.h esound-0.2.27-new/acconfig.h
--- esound-0.2.27/acconfig.h	Tue Apr 23 02:33:07 2002
+++ esound-0.2.27-new/acconfig.h	Thu Jun 13 20:50:23 2002
@@ -9,6 +9,7 @@
 #undef DRIVER_NEWALSA
 #undef DRIVER_ALSA_09
 #undef DRIVER_DART
+#undef DRIVER_COREAUDIO
 #undef DRIVER_NONE
 #undef HAVE_INET_ATON
 #undef HAVE_NANOSLEEP
diff -uNbr esound-0.2.27/audio.c esound-0.2.27-new/audio.c
--- esound-0.2.27/audio.c	Tue Jun  4 07:40:44 2002
+++ esound-0.2.27-new/audio.c	Thu Jun 13 20:50:23 2002
@@ -20,6 +20,8 @@
 /*******************************************************************/
 /* returns audio_fd for use by main prog - platform dependent */
 
+#if defined(INSIDE_ESD)
+
 /* ALSA before OSS as ALSA is OSS compatible */
 #if defined(DRIVER_ALSA) || defined(DRIVER_NEWALSA) 
 #  include "audio_alsa.c"
@@ -41,9 +43,16 @@
 #  include "audio_mklinux.c"
 #elif defined(DRIVER_DART)
 #  include "audio_dart.c"
+#elif defined(DRIVER_COREAUDIO)
+#  include "audio_coreaudio.c"
 #else
 #  include "audio_none.c"
 #endif
+
+#else
+#define DRIVER_NONE 1
+#include "audio_none.c"
+#endif /* INSIDE_ESD */
 
 /*******************************************************************/
 /* display available devices */
diff -uNbr esound-0.2.27/audio_coreaudio.c esound-0.2.27-new/audio_coreaudio.c
--- esound-0.2.27/audio_coreaudio.c	Wed Dec 31 16:00:00 1969
+++ esound-0.2.27-new/audio_coreaudio.c	Thu Jun 13 20:50:23 2002
@@ -0,0 +1,60 @@
+
+/* for Darwin
+ * 
+ * Everything important removed so this can build.
+ */
+
+#define ARCH_esd_audio_devices
+const char *esd_audio_devices()
+{
+    return "darwin coreaudio API only";
+}
+
+#define ARCH_esd_audio_open
+int esd_audio_open()
+{
+  return 0;
+}
+
+/*
+ * This is called to reset the device status.
+ * (before calls esd_audio_open again)
+ */
+#define ARCH_esd_audio_close
+void esd_audio_close()
+{
+}
+
+
+#define ARCH_esd_audio_pause
+void esd_audio_pause()
+{
+  return;
+}
+
+
+#define ARCH_esd_audio_write
+/*******************************************************************/
+/* dump a buffer to the sound device */
+int esd_audio_write( void *buffer, int buf_size )
+{
+  return (buf_size);
+}
+
+#define ARCH_esd_audio_read
+/*******************************************************************/
+/* read a chunk from the sound device */
+int esd_audio_read( void *buffer, int buf_size )
+{
+  return -1;
+}
+
+
+#define ARCH_esd_audio_flush
+/*******************************************************************/
+/* flush the audio buffer */
+void esd_audio_flush()
+{
+  /* nothing needed */
+  return;
+}
diff -uNbr esound-0.2.27/config.h.in esound-0.2.27-new/config.h.in
--- esound-0.2.27/config.h.in	Tue Jun  4 07:41:36 2002
+++ esound-0.2.27-new/config.h.in	Thu Jun 13 20:50:23 2002
@@ -43,6 +43,7 @@
 #undef DRIVER_NEWALSA
 #undef DRIVER_ALSA_09
 #undef DRIVER_DART
+#undef DRIVER_COREAUDIO
 #undef DRIVER_NONE
 #undef HAVE_INET_ATON
 #undef HAVE_NANOSLEEP
diff -uNbr esound-0.2.27/configure esound-0.2.27-new/configure
--- esound-0.2.27/configure	Fri Jun  7 07:25:47 2002
+++ esound-0.2.27-new/configure	Thu Jun 13 20:50:50 2002
@@ -6889,6 +6889,14 @@
 EOF
 
       ;;
+   darwin*)
+      found_sound=yes
+      cat >> confdefs.h <<\EOF
+#define DRIVER_COREAUDIO 1
+EOF
+
+      SOUND_LIBS=
+      ;;
    esac
 
    if test "${ac_cv_header_dmedia_audio_h}" = "yes"; then
diff -uNbr esound-0.2.27/configure.in esound-0.2.27-new/configure.in
--- esound-0.2.27/configure.in	Fri Jun  7 07:20:06 2002
+++ esound-0.2.27-new/configure.in	Thu Jun 13 20:50:23 2002
@@ -156,6 +156,11 @@
       found_sound=yes
       AC_DEFINE(DRIVER_DART)
       ;;
+   darwin*)
+      found_sound=yes
+      AC_DEFINE(DRIVER_COREAUDIO)
+      SOUND_LIBS=
+      ;;
    esac
 
    if test "${ac_cv_header_dmedia_audio_h}" = "yes"; then
diff -uNbr esound-0.2.27/esd.c esound-0.2.27-new/esd.c
--- esound-0.2.27/esd.c	Tue Jun  4 07:39:26 2002
+++ esound-0.2.27-new/esd.c	Thu Jun 13 20:50:23 2002
@@ -21,6 +21,13 @@
 #define gethostbyname2(host, family) gethostbyname((host))
 #endif /* HAVE_GETHOSTBYNAME2 */
 
+#if defined (__APPLE__)
+#include <mach/mach.h>
+#include <mach/thread_policy.h>
+#include <sys/param.h>
+#include <sys/sysctl.h>
+#endif
+
 /*******************************************************************/
 /* esd.c - prototypes */
 void set_audio_buffer( void *buf, esd_format_t format, int magl, int magr,
@@ -514,6 +521,29 @@
 
     int default_format = ESD_BITS16 | ESD_STEREO;
     /* end test scaffolding parameters */
+
+#if defined (__APPLE__)
+    /* get realtime priority */
+    {
+	struct thread_time_constraint_policy ttcpolicy;
+	int bus_speed, mib [2] = { CTL_HW, HW_BUS_FREQ };
+	size_t len;   
+
+	len = sizeof (bus_speed);
+	sysctl (mib, 2, &bus_speed, &len, NULL, 0);
+
+	/* Is it enough? */
+	ttcpolicy.period = bus_speed / 120;
+	ttcpolicy.computation = bus_speed / 2400;
+	ttcpolicy.constraint = bus_speed / 1200;
+	ttcpolicy.preemptible = 1;
+
+	thread_policy_set (mach_thread_self (),
+			   THREAD_TIME_CONSTRAINT_POLICY,
+			   (int*)&ttcpolicy,
+			   THREAD_TIME_CONSTRAINT_POLICY_COUNT);
+    }
+#endif
 
     programname = *argv;
 
diff -uNbr esound-0.2.27/esd.conf esound-0.2.27-new/esd.conf
--- esound-0.2.27/esd.conf	Tue Jun  4 07:30:47 2002
+++ esound-0.2.27-new/esd.conf	Thu Jun 13 20:50:23 2002
@@ -1,4 +1,4 @@
 [esd]
 auto_spawn=1
 spawn_options=-terminate -nobeeps -as 2
-spawn_wait_ms=100
+spawn_wait_ms=999
diff -uNbr esound-0.2.27/esd_audio.c esound-0.2.27-new/esd_audio.c
--- esound-0.2.27/esd_audio.c	Wed Dec 31 16:00:00 1969
+++ esound-0.2.27-new/esd_audio.c	Thu Jun 13 20:50:23 2002
@@ -0,0 +1,4 @@
+/* dirty trickery to reduce shared library dependencies */
+
+#define INSIDE_ESD
+#include "audio.c"
diff -uNbr esound-0.2.27/esd_config.c esound-0.2.27-new/esd_config.c
--- esound-0.2.27/esd_config.c	Tue Jun  4 06:32:46 2002
+++ esound-0.2.27-new/esd_config.c	Thu Jun 13 20:50:23 2002
@@ -16,7 +16,7 @@
 int esd_no_spawn=1; /* If we can't find even the system config file,
 		       things are screwed up - don't try to make things
 		       worse. */
-int esd_spawn_wait_ms=100; /* Time to wait trying to connect to an
+int esd_spawn_wait_ms=999; /* Time to wait trying to connect to an
 			      autospawned ESD, in milliseconds. */
 char esd_spawn_options[LINEBUF_SIZE] = "-terminate -nobeeps -as 2";
 
diff -uNbr esound-0.2.27/esdlib.c esound-0.2.27-new/esdlib.c
--- esound-0.2.27/esdlib.c	Wed Jun  5 01:05:24 2002
+++ esound-0.2.27-new/esdlib.c	Thu Jun 13 20:50:23 2002
@@ -671,8 +671,8 @@
 	FD_ZERO (&fdset);
 	FD_SET (esd_pipe[0], &fdset);
 	
-	timeout.tv_sec = 0;
-	timeout.tv_usec = esd_spawn_wait_ms * 1000;
+	timeout.tv_sec = (esd_spawn_wait_ms * 1000) / 1000000;
+	timeout.tv_usec = (esd_spawn_wait_ms * 1000) % 1000000;
 	
 	ret = select (esd_pipe[0] + 1, &fdset, NULL, NULL, &timeout);
 	
