diff -uNbr libvorbis-1.0/doc/Makefile.in libvorbis-1.0-new/doc/Makefile.in
--- libvorbis-1.0/doc/Makefile.in	Fri Jul 19 10:12:33 2002
+++ libvorbis-1.0-new/doc/Makefile.in	Sat May 17 23:07:30 2003
@@ -94,7 +94,7 @@
 
 SUBDIRS = vorbisfile vorbisenc
 
-docdir = $(datadir)/doc/$(PACKAGE)-$(VERSION)
+docdir = $(datadir)/doc/$(PACKAGE)
 
 doc_DATA = components.png 	draft-moffitt-vorbis-rtp-00.txt 	eightphase.png 	evenlsp.png 	floor1-1.png 	floor1-2.png 	floor1-3.png 	floor1-4.png 	floor1_inverse_dB_table.html 	floorval.png 	fourphase.png 	framing.html 	helper.html 	hufftree-under.png 	hufftree.png 	index.html 	lspmap.png 	oddlsp.png 	oggstream.html 	programming.html 	residue-pack.png 	residue2.png 	squarepolar.png 	stereo.html 	stream.png 	v-comment.html 	vorbis-clip.txt 	vorbis-errors.txt 	vorbis-fidelity.html 	vorbis-ogg.html 	vorbis-spec-bitpack.html 	vorbis-spec-codebook.html 	vorbis-spec-floor0.html 	vorbis-spec-floor1.html 	vorbis-spec-intro.html 	vorbis-spec-ref.html 	vorbis-spec-res.html 	vorbis.html 	vorbisword2.png 	wait.png 	white-ogg.png 	white-xifish.png 	window1.png 	window2.png
 
diff -uNbr libvorbis-1.0/doc/vorbisenc/Makefile.in libvorbis-1.0-new/doc/vorbisenc/Makefile.in
--- libvorbis-1.0/doc/vorbisenc/Makefile.in	Fri Jul 19 10:12:35 2002
+++ libvorbis-1.0-new/doc/vorbisenc/Makefile.in	Sat May 17 23:07:30 2003
@@ -92,7 +92,7 @@
 
 AUTOMAKE_OPTIONS = foreign
 
-docdir = $(datadir)/doc/$(PACKAGE)-$(VERSION)/vorbisenc
+docdir = $(datadir)/doc/$(PACKAGE)/vorbisenc
 
 doc_DATA = index.html overview.html reference.html style.css 	 vorbis_encode_ctl.html vorbis_encode_init.html vorbis_info.html 	vorbis_encode_init_vbr.html
 
diff -uNbr libvorbis-1.0/doc/vorbisfile/Makefile.in libvorbis-1.0-new/doc/vorbisfile/Makefile.in
--- libvorbis-1.0/doc/vorbisfile/Makefile.in	Fri Jul 19 10:12:34 2002
+++ libvorbis-1.0-new/doc/vorbisfile/Makefile.in	Sat May 17 23:07:30 2003
@@ -92,7 +92,7 @@
 
 AUTOMAKE_OPTIONS = foreign
 
-docdir = $(datadir)/doc/$(PACKAGE)-$(VERSION)/vorbisfile
+docdir = $(datadir)/doc/$(PACKAGE)/vorbisfile
 
 doc_DATA = OggVorbis_File.html chaining_example_c.html chainingexample.html 	datastructures.html decoding.html example.html exampleindex.html 	fileinfo.html index.html initialization.html ov_bitrate.html 	ov_bitrate_instant.html ov_callbacks.html ov_clear.html 	ov_comment.html ov_info.html ov_open.html ov_open_callbacks.html 	ov_pcm_seek.html ov_pcm_tell.html ov_pcm_total.html ov_raw_seek.html 	ov_raw_tell.html ov_raw_total.html ov_read.html ov_seekable.html 	ov_serialnumber.html ov_streams.html ov_time_seek.html 	ov_time_tell.html ov_time_total.html overview.html reference.html 	seekexample.html seeking.html seeking_test_c.html seekingexample.html 	style.css vorbis_comment.html vorbis_info.html 	vorbisfile_example_c.html ov_read_float.html ov_test.html 	ov_test_open.html ov_test_callbacks.html
 
diff -uNbr libvorbis-1.0/lib/Makefile.in libvorbis-1.0-new/lib/Makefile.in
--- libvorbis-1.0/lib/Makefile.in	Fri Jul 19 10:12:26 2002
+++ libvorbis-1.0-new/lib/Makefile.in	Sat May 17 23:26:04 2003
@@ -131,9 +131,9 @@
 window.lo lsp.lo lpc.lo analysis.lo synthesis.lo psy.lo info.lo \
 floor1.lo floor0.lo res0.lo mapping0.lo registry.lo codebook.lo \
 sharedbook.lo lookup.lo bitrate.lo
-libvorbisfile_la_LIBADD = 
+libvorbisfile_la_LIBADD = libvorbis.la
 libvorbisfile_la_OBJECTS =  vorbisfile.lo
-libvorbisenc_la_LIBADD = 
+libvorbisenc_la_LIBADD = libvorbis.la
 libvorbisenc_la_OBJECTS =  vorbisenc.lo
 barkmel_OBJECTS =  barkmel.$(OBJEXT)
 barkmel_LDADD = $(LDADD)
