diff -uNbr mad-0.14.2b/libid3tag/Makefile.in mad-0.14.2b-new/libid3tag/Makefile.in
--- mad-0.14.2b/libid3tag/Makefile.in	Fri Nov  9 02:02:58 2001
+++ mad-0.14.2b-new/libid3tag/Makefile.in	Sat May 17 23:34:13 2003
@@ -105,7 +105,7 @@
 CPPFLAGS = @CPPFLAGS@
 LDFLAGS = @LDFLAGS@
 LIBS = @LIBS@
-libid3tag_la_LIBADD = 
+libid3tag_la_LIBADD = -lz
 @DEBUG_FALSE@libid3tag_la_OBJECTS =  version.lo ucs4.lo latin1.lo \
 @DEBUG_FALSE@utf16.lo utf8.lo parse.lo render.lo field.lo frametype.lo \
 @DEBUG_FALSE@compat.lo genre.lo frame.lo crc.lo util.lo tag.lo file.lo
