diff -uNbr agqt/ag/Makefile agqt-new/ag/Makefile
--- agqt/ag/Makefile	2002-02-20 04:10:33.000000000 -0500
+++ agqt-new/ag/Makefile	2002-07-28 10:59:38.000000000 -0500
@@ -1,7 +1,7 @@
 OBJ		=	main.o server.o cfgfile.o stringproc.o web.o \
 			myqueue.o printproc.o versionlist.o common.o
 G++		=	g++
-CXXFLAGS	=	-O6 -g
+CXXFLAGS	=	-O6 -g -Wno-deprecated
 
 ag: $(OBJ)
 	$(G++) -o ag $(OBJ)
diff -uNbr agqt/ag/main.cc agqt-new/ag/main.cc
--- agqt/ag/main.cc	2002-02-20 05:26:44.000000000 -0500
+++ agqt-new/ag/main.cc	2002-07-28 10:59:38.000000000 -0500
@@ -6,6 +6,10 @@
 #include "myqueue.h"
 #include "printproc.h"
 
+#include <sys/types.h>
+#include <sys/time.h>
+#include <sys/resource.h>
+
 int
 main(int argc, char *argv[])
 {
@@ -24,6 +28,11 @@
 		exit(-1);
 	}
 
+        struct rlimit rlim;
+        getrlimit(RLIMIT_STACK, &rlim);
+        rlim.rlim_cur=600000;
+        setrlimit(RLIMIT_STACK, &rlim);
+
 	int timeWanted = 0;
 	int versionNum = 1;
 	if ( (argc > 2) && (argv[2][0] == 'T') )
diff -uNbr agqt/ag/myqueue.h agqt-new/ag/myqueue.h
--- agqt/ag/myqueue.h	2002-01-04 13:22:09.000000000 -0500
+++ agqt-new/ag/myqueue.h	2002-07-28 10:59:38.000000000 -0500
@@ -1,7 +1,7 @@
 #ifndef MYQUEUE_H
 #define MYQUEUE_H
 
-#include <queue>
+#include <queue.h>
 
 struct fileInfo
 {
diff -uNbr agqt/agrename.pl agqt-new/agrename.pl
--- agqt/agrename.pl	1969-12-31 19:00:00.000000000 -0500
+++ agqt-new/agrename.pl	2002-07-28 10:59:38.000000000 -0500
@@ -0,0 +1,89 @@
+#!/usr/bin/perl
+#
+# agqt file renamer
+#
+# Written and copyright 2001 by Andy Grundman.
+# THIS SOFTWARE COMES WITH ABSOLUTELY NO WARRANTY!
+#
+# Usage:
+#   agrename.pl --path=/path/to/agfiles --artist="Artist Name" --album="Album Name" --format=[rename format (see below)]
+#
+# Rename formats:
+#   "album": All files will be moved into a directory called "Artist - Album" 
+#            and will be named "track number - track name.mp3"  This is the default.
+#   create your own format, this is left as an exercise for the reader!
+#
+# Note: If the AllMusic track listing contains a shortened version of the track name,
+# you will need to rename it after running this script.
+#
+
+use strict;
+use Getopt::Long;
+
+# path to 'am' binary
+my $am = 'am';	# if located in PATH
+
+my ($path, $artist, $album, $format);
+$format = "album";
+GetOptions( 'path=s', \$path, 'artist=s' => \$artist, 'album=s' => \$album, 'format' => \$format );
+unless ($path && $album && $artist) {
+	usage();
+}
+
+chdir($path) or die "Invalid path: $path";
+
+my $albumDir;
+if ($format eq "album") {
+	$albumDir = $artist . " - " . $album;
+	mkdir($albumDir);
+}
+# create new formats here
+
+# get correct track listing using 'am'
+my $trackNum = 0;	
+my $cmd = $am . " \"" . $artist . "\" \"" . $album . "\" show-tracks";
+open AM, "$cmd |";
+while (<AM>) {
+	if (/^Track/) {
+		chomp;
+		$trackNum++;
+		my $title = $_;
+		$title =~ s/^Track(.*): //;
+		
+		my $search = "ls -1 | grep -i \"" . $artist . " - " . $title . ".mp3\"";
+		my $filename = `$search`;
+		chomp($filename);
+		
+		if ($filename) {
+		
+			if ($trackNum < 10) {
+				$trackNum = "0" . $trackNum;
+			}
+			
+			my $dest;
+			if ($format eq "album") {
+				$dest = "$albumDir/$trackNum - $title.mp3";
+			}
+			# add more formats here
+			
+			$cmd = "mv \"" . $filename . "\" \"$dest\"";		
+			`$cmd`;
+			
+			print "Moving $filename to $albumDir/$trackNum - $title.mp3\n";
+		} else {
+			print "Cannot find file for track $trackNum - $title\n";
+		}
+		
+	}
+}
+close AM;
+
+sub usage {
+	print "Usage: agrename.pl --path=/path/to/agfiles --artist=\"Artist Name\" --album=\"Album Name\" [--format=rename format (see below)]\n";
+	print "Rename mp3 files downloaded by AudioGalaxy to their correct track order.\n";
+	print "Example: agrename.pl --path=/var/AGSatellite/files --artist=\"vic chesnutt\" --album=\"about to choke\" --format=album\n\n";
+	print "Format options:\n";
+	print "  album: All files will be moved into a directory called \"Artist - Album\"\n";
+	print "         and will be named \"track number - track name.mp3\".  This is the default.\n";
+	exit(0);
+}
diff -uNbr agqt/am/Makefile agqt-new/am/Makefile
--- agqt/am/Makefile	2002-02-19 22:24:19.000000000 -0500
+++ agqt-new/am/Makefile	2002-07-28 10:59:38.000000000 -0500
@@ -1,7 +1,7 @@
 OBJ		=	main.o server.o stringproc.o myqueue.o \
 			ag_interface.o web.o common.o
 G++		=	g++
-CXXFLAGS	=	-O6 -g
+CXXFLAGS	=	-O6 -g -Wno-deprecated
 
 am: $(OBJ)
 	$(G++) -o am $(OBJ)
diff -uNbr agqt/am/main.cc agqt-new/am/main.cc
--- agqt/am/main.cc	2002-02-20 04:39:20.000000000 -0500
+++ agqt-new/am/main.cc	2002-07-28 10:59:38.000000000 -0500
@@ -4,6 +4,10 @@
 #include "web.h"
 #include "common.h"
 
+#include <sys/types.h>
+#include <sys/time.h>
+#include <sys/resource.h>
+
 int
 main(int argc, char *argv[])
 {
@@ -18,7 +22,9 @@
 		fprintf(stderr, "\tor\n");
 		fprintf(stderr, "%s \"artist name\" \"album URL\" [logfile name]\n", argv[0]);
 		fprintf(stderr, "\tor\n");
-		fprintf(stderr, "%s \"artist name\" sh [logfile name]\n\n", argv[0]);
+		fprintf(stderr, "%s \"artist name\" sh [logfile name]\n", argv[0]);
+		fprintf(stderr, "\tor to display the tracks only:\n");
+		fprintf(stderr, "%s \"artist name\" \"album name or url\" show-tracks\n\n", argv[0]);
 		fprintf(stderr, "If album name is missing, all available albums by artist will be listed.\n");
 		fprintf(stderr, "If album name is \"sh\", the listed song highlights are retrieved.\n");
 		fprintf(stderr, "And yes, the quote marks *are* required.\n");
@@ -31,6 +37,11 @@
 		exit(-1);
 	}
 
+        struct rlimit rlim;
+        getrlimit(RLIMIT_STACK, &rlim);
+        rlim.rlim_cur=600000;
+        setrlimit(RLIMIT_STACK, &rlim);
+
 	if (argv[2] && strstr(argv[2], "www.allmusic.com/cg/amg.dll?p=amg&sql="))
 	{
 		// Album URL specified on command line.
diff -uNbr agqt/am/myqueue.h agqt-new/am/myqueue.h
--- agqt/am/myqueue.h	2002-01-04 13:22:09.000000000 -0500
+++ agqt-new/am/myqueue.h	2002-07-28 10:59:38.000000000 -0500
@@ -1,7 +1,7 @@
 #ifndef MYQUEUE_H
 #define MYQUEUE_H
 
-#include <queue>
+#include <queue.h>
 
 void 			q_push		(char *);
 void			q_pop		();
diff -uNbr agqt/am/web.cc agqt-new/am/web.cc
--- agqt/am/web.cc	2002-04-19 05:24:16.000000000 -0500
+++ agqt-new/am/web.cc	2002-07-28 10:59:38.000000000 -0500
@@ -172,6 +172,11 @@
 {
 	int inlist = FALSE;
 	int finished = FALSE;
+	int showTracks = FALSE;
+	
+	if (logfile && strstr(logfile, "show-tracks")) {
+		showTracks = TRUE;
+	}
 
 	class server *S = new class server("www.allmusic.com", 80);
 	S->send("GET %s HTTP/1.1\n", URL);
@@ -231,11 +236,16 @@
 				}
 
 				removeAccents(title);
-				printf("\nFound song: %s, calling ag...\n", title);
 				numSongs++;
+				if (showTracks == TRUE) {
+					printf("Track %02d: %s\n", numSongs, title);
+				} else {
+					printf("\nFound song: %s, calling ag...\n", title);
 				// Keep track of unsuccessful queue attempts.
 				if (!queueSong(artist, title, logfile))
 					q_push(strdup(title)); 
+				}
+
 				free(title);
 			}
 			
diff -uNbr agqt/fdb/Makefile agqt-new/fdb/Makefile
--- agqt/fdb/Makefile	2002-02-19 22:28:05.000000000 -0500
+++ agqt-new/fdb/Makefile	2002-07-28 10:59:38.000000000 -0500
@@ -1,7 +1,7 @@
 OBJ		=	main.o server.o stringproc.o myqueue.o \
 			ag_interface.o web.o common.o
 G++		=	g++
-CXXFLAGS	=	-O6 -g
+CXXFLAGS	=	-O6 -g -Wno-deprecated
 
 fdb: $(OBJ)
 	$(G++) -o fdb $(OBJ)
diff -uNbr agqt/fdb/main.cc agqt-new/fdb/main.cc
--- agqt/fdb/main.cc	2002-02-21 23:08:19.000000000 -0500
+++ agqt-new/fdb/main.cc	2002-07-28 10:59:38.000000000 -0500
@@ -4,6 +4,10 @@
 #include "web.h"
 #include "common.h"
 
+#include <sys/types.h>
+#include <sys/time.h>
+#include <sys/resource.h>
+
 int
 main(int argc, char *argv[])
 {
@@ -27,6 +31,11 @@
                 exit(-1);
         }
 
+        struct rlimit rlim;
+        getrlimit(RLIMIT_STACK, &rlim);
+        rlim.rlim_cur=600000;
+        setrlimit(RLIMIT_STACK, &rlim);
+
 	if (argv[2] == NULL)
 		// Don't get songs if we're just looking for album list
 		searchArtist(argv[1], argv[2], argv[3]);
