diff -Naur nautilus-1.0.6/Makefile.in nautilus-new/Makefile.in
--- nautilus-1.0.6/Makefile.in	2001-11-07 22:54:44.000000000 -0500
+++ nautilus-new/Makefile.in	2002-04-15 17:42:27.000000000 -0400
@@ -226,7 +226,7 @@
 
 @HAVE_APPLETS_TRUE@APPLETS_SUBDIRS = applets
 
-SUBDIRS =  	data 				icons 				cut-n-paste-code 		libnautilus 			libnautilus-adapter		libnautilus-private 		helper-utilities 		helper-scripts			src 				components			$(APPLETS_SUBDIRS)		test 				user-guide			help				po 				omf-install			$(NULL)
+SUBDIRS =  	data 				icons 				cut-n-paste-code 		libnautilus 			libnautilus-adapter		libnautilus-private 		helper-utilities 		helper-scripts			src 				components			$(APPLETS_SUBDIRS)		user-guide			help				po 				omf-install			$(NULL)
 
 
 EXTRA_DIST =  	TRADEMARK_NOTICE				COPYING.LIB					COPYING-DOCS					HACKING						MAINTAINERS					nautilus.spec.in				nautilus-clean.sh				nautilus.desktop				nautilus.desktop.in				nautilus-help.desktop				nautilus-help.desktop.in			nautilus-launch-icon.png			gnome-starthere-96.png				gnome-starthere.png				nautilusConf.sh.in				nautilus-config.in				ChangeLog-20000414				ChangeLog-20000625				ChangeLog-20000831				ChangeLog-20001018				xml-i18n-extract.in				xml-i18n-merge.in				xml-i18n-update.in				Makefile.shared					$(DESKTOP_IN_FILES)				$(DESKTOP_FILES)				$(NULL)
diff -Naur nautilus-1.0.6/components/Makefile.in nautilus-new/components/Makefile.in
--- nautilus-1.0.6/components/Makefile.in	2001-11-07 22:57:49.000000000 -0500
+++ nautilus-new/components/Makefile.in	2002-04-15 17:42:27.000000000 -0400
@@ -222,7 +222,7 @@
 
 @BUILD_MOZILLA_COMPONENT_TRUE@MOZILLA_COMPONENT_SUBDIRS = mozilla
 
-SUBDIRS =  	adapter 					hardware 					help 						history 					image-viewer 					loser 						music 						news						notes 						sample 						text 						throbber 					tree 						$(MOZILLA_COMPONENT_SUBDIRS) 			$(NULL)
+SUBDIRS =  	adapter 					hardware 					help 						history 					image-viewer 					loser 						music 						news						notes 						text 						throbber 					tree 						$(MOZILLA_COMPONENT_SUBDIRS) 			$(NULL)
 
 mkinstalldirs = $(SHELL) $(top_srcdir)/mkinstalldirs
 CONFIG_HEADER = ../config.h
diff -Naur nautilus-1.0.6/components/help/Makefile.in nautilus-new/components/help/Makefile.in
--- nautilus-1.0.6/components/help/Makefile.in	2001-11-07 22:58:32.000000000 -0500
+++ nautilus-new/components/help/Makefile.in	2002-04-16 10:36:36.000000000 -0400
@@ -238,7 +238,7 @@
 mapdatadir = $(datadir)/nautilus/components/hyperbola/maps
 mapdata_DATA = pages.map
 
-module_flags = -export_dynamic -avoid-version
+module_flags = -export_dynamic -avoid-version -module
 modulesdir = $(libdir)/vfs/modules
 
 libvfs_help_la_SOURCES = help-method.c help-method.h
diff -Naur nautilus-1.0.6/components/help/converters/gnome-db2html2/Makefile.in nautilus-new/components/help/converters/gnome-db2html2/Makefile.in
--- nautilus-1.0.6/components/help/converters/gnome-db2html2/Makefile.in	2001-11-07 22:58:37.000000000 -0500
+++ nautilus-new/components/help/converters/gnome-db2html2/Makefile.in	2002-04-15 17:42:28.000000000 -0400
@@ -221,7 +221,7 @@
 INCLUDES =  	-DGNOMELOCALEDIR=\""$(datadir)/locale"\"		$(GNOMEUI_CFLAGS)						$(XML_CFLAGS)
 
 
-LDADD =  	$(GNOME_LIBS) $(XML_LIBS)
+LDADD =  	$(GNOME_LIBS) $(XML_LIBS) $(INTLLIBS)
 
 
 bin_PROGRAMS = gnome-db2html2
diff -Naur nautilus-1.0.6/components/help/converters/gnome-info2html2/Makefile.in nautilus-new/components/help/converters/gnome-info2html2/Makefile.in
--- nautilus-1.0.6/components/help/converters/gnome-info2html2/Makefile.in	2001-11-07 22:58:36.000000000 -0500
+++ nautilus-new/components/help/converters/gnome-info2html2/Makefile.in	2002-04-15 17:42:28.000000000 -0400
@@ -221,7 +221,7 @@
 INCLUDES =  	$(GNOMEUI_CFLAGS)	
 
 
-LDADD =  	$(GNOME_LIBS) $(Z_LIBS) $(BZ_LIBS)
+LDADD =  	$(GNOME_LIBS) $(Z_LIBS) $(BZ_LIBS) $(INTLLIBS)
 
 
 bin_PROGRAMS = gnome-info2html2
diff -Naur nautilus-1.0.6/components/sample/Makefile.in nautilus-new/components/sample/Makefile.in
--- nautilus-1.0.6/components/sample/Makefile.in	2001-11-07 23:00:36.000000000 -0500
+++ nautilus-new/components/sample/Makefile.in	2002-04-15 17:42:28.000000000 -0400
@@ -224,7 +224,7 @@
 
 NULL = 
 
-INCLUDES =  	-DG_LOG_DOMAIN=\"Nautilus-Sample\"			-DDATADIR=\"$(datadir)\"				-DICON_DIR=\"$(datadir)/pixmaps/nautilus\"		-I$(top_srcdir)						-I$(top_builddir)					$(GNOMEUI_CFLAGS)                       		$(BONOBO_CFLAGS)					$(OAF_CFLAGS)
+INCLUDES =  	-DG_LOG_DOMAIN=\"Nautilus-Sample\"			-DDATADIR=\"$(datadir)\"				-DICON_DIR=\"$(datadir)/pixmaps/nautilus\"		-I$(top_srcdir)						-I$(top_builddir)					$(GNOMEUI_CFLAGS)                       		$(BONOBO_CFLAGS)					$(OAF_CFLAGS)					$(GDK_PIXBUF_CFLAGS)
 
 
 bin_PROGRAMS =  	nautilus-sample-content-view
diff -Naur nautilus-1.0.6/components/tree/Makefile.in nautilus-new/components/tree/Makefile.in
--- nautilus-1.0.6/components/tree/Makefile.in	2001-11-07 23:01:18.000000000 -0500
+++ nautilus-new/components/tree/Makefile.in	2002-04-15 21:49:34.000000000 -0400
@@ -247,7 +247,7 @@
 CPPFLAGS = @CPPFLAGS@
 LDFLAGS = @LDFLAGS@
 LIBS = @LIBS@
-libnautilus_tree_view_la_LDFLAGS = 
+libnautilus_tree_view_la_LDFLAGS = -module
 libnautilus_tree_view_la_LIBADD = 
 libnautilus_tree_view_la_OBJECTS =  nautilus-tree-change-queue.lo \
 nautilus-tree-expansion-state.lo nautilus-tree-model.lo \
@@ -544,6 +544,11 @@
 
 @XML_I18N_MERGE_OAF_RULE@
 
+Nautilus_View_tree.oaf.in:	Makefile
+	sed -e "s|\@LIBDIR\@|$(libdir)|" Nautilus_View_tree.oaf.in > \
+	Nautilus_View_tree.oaf.in.tmp; \
+	mv -f Nautilus_View_tree.oaf.in.tmp Nautilus_View_tree.oaf.in;
+
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
 .NOEXPORT:
diff -Naur nautilus-1.0.6/components/tree/Nautilus_View_tree.oaf.in nautilus-new/components/tree/Nautilus_View_tree.oaf.in
--- nautilus-1.0.6/components/tree/Nautilus_View_tree.oaf.in	2001-01-05 08:10:49.000000000 -0500
+++ nautilus-new/components/tree/Nautilus_View_tree.oaf.in	2002-04-15 18:57:51.000000000 -0400
@@ -1,6 +1,6 @@
 <oaf_info>
 
-<oaf_server iid="OAFIID:nautilus_tree_view:2d826a6e-1669-4a45-94b8-23d65d22802d" type="shlib" location="libnautilus-tree-view.so">
+<oaf_server iid="OAFIID:nautilus_tree_view:2d826a6e-1669-4a45-94b8-23d65d22802d" type="shlib" location="@LIBDIR@/libnautilus-tree-view.so">
 	<oaf_attribute name="repo_ids" type="stringv">
 		<item value="IDL:Bonobo/Unknown:1.0"/>
 		<item value="IDL:Bonobo/Control:1.0"/>
diff -Naur nautilus-1.0.6/configure nautilus-new/configure
--- nautilus-1.0.6/configure	2001-11-07 20:55:40.000000000 -0500
+++ nautilus-new/configure	2002-04-15 17:42:28.000000000 -0400
@@ -3478,7 +3478,7 @@
     # FIXME: Relying on posixy $() will cause problems for
     #        cross-compilation, but unfortunately the echo tests do not
     #        yet detect zsh echo's removal of \ escapes.
-    archive_cmds='$nonopt $(test "x$module" = xyes && echo -bundle || echo -dynamiclib) $allow_undefined_flag -o $lib $libobjs $deplibs$linker_flags -install_name $rpath/$soname $verstring'
+    archive_cmds='$nonopt $(test x$module = xyes && echo -bundle || echo -dynamiclib) $allow_undefined_flag -o $lib $libobjs $deplibs$linker_flags $(test .$module != .yes && echo -install_name $rpath/$soname $verstring)'
     # We need to add '_' to the symbols in $export_symbols first
     #archive_expsym_cmds="$archive_cmds"' && strip -s $export_symbols'
     hardcode_direct=yes
@@ -6966,7 +6966,7 @@
 
 _mozilla_all_nspr_flags="$_mozilla_nspr_libs $_mozilla_plc_libs $_mozilla_plds_libs"
 
-CC="g++"
+CC="c++"
 
 LDFLAGS="$_save_ldflags -L$_mozilla_lib_place $_mozilla_all_nspr_flags -lxpcom"
 echo $ac_n "checking for xpcom""... $ac_c" 1>&6
@@ -7251,7 +7251,7 @@
 
 	found_mozilla=true
 
-	MOZILLA_COMPONENT_RPATH_FLAGS=-Wl,-rpath=$_mozilla_lib_place
+	MOZILLA_COMPONENT_RPATH_FLAGS=
 	MOZILLA_COMPONENT_LDFLAGS=-L$_mozilla_lib_place
 	MOZILLA_COMPONENT_LIBS="$_mozilla_gtk_moz_embed_libs \
 				$_mozilla_gtk_super_win_libs \
diff -Naur nautilus-1.0.6/libnautilus-private/filesystem-attributes.xml nautilus-new/libnautilus-private/filesystem-attributes.xml
--- nautilus-1.0.6/libnautilus-private/filesystem-attributes.xml	2001-10-02 11:15:57.000000000 -0400
+++ nautilus-new/libnautilus-private/filesystem-attributes.xml	2002-04-15 17:42:28.000000000 -0400
@@ -9,6 +9,7 @@
     <filesystem name="ext2"     _default_volume_name="Ext2 Linux Volume" trash="yes"/>
     <filesystem name="ext3"     _default_volume_name="Ext3 Linux Volume" trash="yes"/>
     <filesystem name="fat"      _default_volume_name="MSDOS Volume" trash="yes"/>
+    <filesystem name="hfs"      _default_volume_name="Macintosh Volume" trash="yes"/>
     <filesystem name="iso9660"  _default_volume_name="CDROM Volume"/>
     <filesystem name="jfs"      _default_volume_name="JFS Volume" trash="yes"/>
     <filesystem name="hpfs"     _default_volume_name="Windows NT Volume"/>
diff -Naur nautilus-1.0.6/libnautilus-private/nautilus-customization-data.c nautilus-new/libnautilus-private/nautilus-customization-data.c
--- nautilus-1.0.6/libnautilus-private/nautilus-customization-data.c	2001-09-20 14:34:48.000000000 -0400
+++ nautilus-new/libnautilus-private/nautilus-customization-data.c	2002-04-15 17:42:28.000000000 -0400
@@ -32,6 +32,8 @@
 #include <libgnome/gnome-util.h>
 #include <libgnome/gnome-i18n.h>
 
+#include <eel/eel-xml-extensions.h>
+
 #include <libxml/parser.h>
 #include <libxml/xmlmemory.h>
 
@@ -51,7 +53,6 @@
 #include <eel/eel-gdk-extensions.h>
 #include <eel/eel-gtk-extensions.h>
 #include <eel/eel-scalable-font.h>
-#include <eel/eel-xml-extensions.h>
 #include <eel/eel-string.h>
 
 typedef enum {
diff -Naur nautilus-1.0.6/libnautilus-private/nautilus-desktop-file-loader.c nautilus-new/libnautilus-private/nautilus-desktop-file-loader.c
--- nautilus-1.0.6/libnautilus-private/nautilus-desktop-file-loader.c	2001-09-26 12:37:15.000000000 -0400
+++ nautilus-new/libnautilus-private/nautilus-desktop-file-loader.c	2002-04-15 17:42:28.000000000 -0400
@@ -39,7 +39,6 @@
 #include <ctype.h>
 #include <locale.h>
 #include <iconv.h>
-#include <langinfo.h>
 
 typedef struct NautilusDesktopFileSection NautilusDesktopFileSection;
 typedef struct NautilusDesktopFileAddition NautilusDesktopFileAddition;
@@ -1040,7 +1039,7 @@
         if (!initialized) {
                 /* whee, we are totally unportable (broken too) */
                 setlocale (LC_CTYPE, "");
-                locale = nl_langinfo (CODESET);
+                locale = "UTF-8";
 		initialized = TRUE;
         }
         G_UNLOCK (init_validate);
diff -Naur nautilus-1.0.6/libnautilus-private/nautilus-medusa-support.c nautilus-new/libnautilus-private/nautilus-medusa-support.c
--- nautilus-1.0.6/libnautilus-private/nautilus-medusa-support.c	2001-04-11 14:36:16.000000000 -0400
+++ nautilus-new/libnautilus-private/nautilus-medusa-support.c	2002-04-15 17:42:28.000000000 -0400
@@ -31,6 +31,7 @@
 
 #include <eel/eel-glib-extensions.h>
 #include <eel/eel-string.h>
+#include <stdint.h>
 #include <dirent.h>
 #include <stdio.h>
 #include <sys/types.h>
diff -Naur nautilus-1.0.6/libnautilus-private/nautilus-volume-monitor.c nautilus-new/libnautilus-private/nautilus-volume-monitor.c
--- nautilus-1.0.6/libnautilus-private/nautilus-volume-monitor.c	2001-10-09 20:16:39.000000000 -0400
+++ nautilus-new/libnautilus-private/nautilus-volume-monitor.c	2002-04-16 02:47:40.000000000 -0400
@@ -96,6 +96,10 @@
 #include <sys/ucred.h>
 #endif
 
+#ifdef __APPLE__
+#define HAVE_SYS_MOUNT_H
+#endif
+
 #ifdef HAVE_SYS_MOUNT_H
 #include <sys/mount.h>
 #endif
@@ -963,90 +967,28 @@
 static GList *
 get_mount_list (NautilusVolumeMonitor *monitor) 
 {
-        GList *volumes;
-        NautilusVolume *volume;
-	static time_t last_mtime = 0;
-        static FILE *fh = NULL;
-        static GList *saved_list = NULL;
-        const char *file_name;
-	const char *separator;
-	char line[PATH_MAX * 3];
-	char device_name[sizeof (line)];
-	EelStringList *list;
-	char *device_path, *mount_path, *file_system_type_name;
-	struct stat sb;
-
-	volumes = NULL;
-        
-	if (mnttab_exists) { 
-		file_name = "/etc/mnttab";
-		separator = "\t";
-	} else {
-		file_name = "/proc/mounts";
-		separator = " ";
-	}
-
-	/* /proc/mounts mtime never changes, so stat /etc/mtab.
-	 * Isn't this lame?
-	 */
-	if (stat ("/etc/mtab", &sb) < 0) {
-		g_warning ("Unable to stat %s: %s", file_name,
-			   g_strerror (errno));
+	GList *volumes = NULL;
+	struct statfs *fsinfos = NULL;
+	int fsnum, fsidx;
+	
+	fsnum = getmntinfo(&fsinfos, MNT_WAIT);
+	if (fsnum == -1) {
+		g_warning ("Unable to stat filesystems.");
 		return NULL;
 	}
 	
-	if (sb.st_mtime == last_mtime) {
-		return copy_mount_list (saved_list);
-	}
-
-	last_mtime = sb.st_mtime;
-	
-	if (fh == NULL) {
-                fh = fopen (file_name, "r");
-                if (fh == NULL) {
-                        g_warning ("Unable to open %s: %s", file_name, strerror (errno));
-                        return NULL;
-                }
-        } else {		
-                rewind (fh);
-        }
-
-	while (fgets (line, sizeof(line), fh)) {
-                if (sscanf (line, "%s", device_name) != 1) {
-                        continue;
-		}
-
-                list = eel_string_list_new_from_tokens (line, separator, FALSE);
-                if (list == NULL) {
-                        continue;
-		}
-                
-                /* The string list needs to have at least 3 items per line.
-                 * We need to find at least device path, mount path and file system type.
-                 */
-                if (eel_string_list_get_length (list) >= 3) {
-                        device_path = eel_string_list_nth (list, 0);
-                        mount_path = eel_string_list_nth (list, 1);
-                        file_system_type_name = eel_string_list_nth (list, 2);
-                        volume = create_volume (device_path, mount_path);
-			if (eel_string_list_get_length (list) >= 4 &&
-			    option_list_has_option (eel_string_list_nth (list, 3), MNTOPT_RO))
-				volume->is_read_only = TRUE;
-                        volumes = finish_creating_volume_and_prepend
-				(monitor, volume, file_system_type_name, volumes);
- 			
-                        g_free (device_path);
-                        g_free (mount_path);
-                        g_free (file_system_type_name);
-                }
-
-		eel_string_list_free (list);
+	for (fsidx = 0; fsidx < fsnum; ++fsidx) {
+		struct statfs afs = fsinfos[fsidx];
+		NautilusVolume *volume = NULL;
+		
+		volume = create_volume (afs.f_mntfromname, afs.f_mntonname);
+		if (afs.f_flags & MNT_RDONLY)
+			volume->is_read_only = TRUE;
+		volumes = finish_creating_volume_and_prepend
+			(monitor, volume, afs.f_fstypename, volumes);
   	}
-
-	free_mount_list (saved_list);
-        saved_list = volumes;
-	
-        return copy_mount_list (volumes);
+  	
+  	return volumes;
 }
 
 #endif /* !SOLARIS_MNT */
diff -Naur nautilus-1.0.6/ltmain.sh nautilus-new/ltmain.sh
--- nautilus-1.0.6/ltmain.sh	2001-11-07 20:55:25.000000000 -0500
+++ nautilus-new/ltmain.sh	2002-04-15 17:42:28.000000000 -0400
@@ -2892,7 +2892,12 @@
 	if test -n "$export_symbols" && test -n "$archive_expsym_cmds"; then
 	  eval cmds=\"$archive_expsym_cmds\"
 	else
-	  eval cmds=\"$archive_cmds\"
+	  save_deplibs="$deplibs"
+	  for conv in $convenience; do
+	    deplibs="${deplibs%$conv*} ${deplibs#*$conv}"
+	  done
+ 	  eval cmds=\"$archive_cmds\"
+	  deplibs="$save_deplibs"
 	fi
 	IFS="${IFS= 	}"; save_ifs="$IFS"; IFS='~'
 	for cmd in $cmds; do
diff -Naur nautilus-1.0.6/omf-install/Makefile.in nautilus-new/omf-install/Makefile.in
--- nautilus-1.0.6/omf-install/Makefile.in	2001-11-07 23:02:57.000000000 -0500
+++ nautilus-new/omf-install/Makefile.in	2002-04-17 01:36:41.000000000 -0400
@@ -325,7 +325,7 @@
 	-for file in $(srcdir)/*.omf; do 			\
 		$(INSTALL_DATA) $(srcdir)/$$file $(DESTDIR)$(omf_dest_dir); \
 	done
-	-scrollkeeper-update -p $(scrollkeeper_localstate_dir) 
+#	-scrollkeeper-update -p $(scrollkeeper_localstate_dir) 
 
 uninstall-local:
 	-for file in $(srcdir)/*.omf; do 			\
@@ -333,7 +333,7 @@
 		rm -f $(omf_dest_dir)/$$basefile; 		\
         done
 	-rmdir $(omf_dest_dir)
-	-scrollkeeper-update -p $(scrollkeeper_localstate_dir) 
+#	-scrollkeeper-update -p $(scrollkeeper_localstate_dir) 
 
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
