diff -uNbr upclient-4.2.1.23/src/stats-bsd.c upclient-4.2.1.23-new/src/stats-bsd.c
--- upclient-4.2.1.23/src/stats-bsd.c	2001-06-16 14:12:53.000000000 -0500
+++ upclient-4.2.1.23-new/src/stats-bsd.c	2002-07-28 20:31:56.000000000 -0500
@@ -21,6 +21,10 @@
 #include <sys/dkstat.h>
 #include <sys/sysctl.h>
 #include <sys/utsname.h>
+#include <rpcsvc/rstat.h>
+#ifndef CP_IDLE
+#define CP_IDLE 4
+#endif
 
 /**
  * @desc	Get statistics
diff -uNbr upclient-4.2.1.23/src/stats-ult.c upclient-4.2.1.23-new/src/stats-ult.c
--- upclient-4.2.1.23/src/stats-ult.c	2000-12-26 13:58:32.000000000 -0500
+++ upclient-4.2.1.23-new/src/stats-ult.c	2002-07-28 20:32:36.000000000 -0500
@@ -19,6 +19,10 @@
 #include <sys/utsname.h>
 #include <sys/cpudata.h>
 #include <sys/fixpoint.h>
+#include <sys/dkstat.h>
+#ifndef CP_IDLE
+#define CP_IDLE 4
+#endif
 
 /**
  * @desc	Get statistics
