diff -ruN upclient-4.2.1.23.orig/src/upclient.h upclient-4.2.1.23/src/upclient.h
--- upclient-4.2.1.23.orig/src/upclient.h	Thu May 31 17:19:44 2001
+++ upclient-4.2.1.23/src/upclient.h	Wed Jul 31 13:20:26 2002
@@ -49,6 +49,11 @@
 #define OSLEVEL_SIZE	64 /* Protocol is not clear on this one */
 #define CPU_SIZE	64 /* Protocol is not clear on this one either */
 
+/* Needed for darwin 6+ and OSX 10.2+ */
+#ifdef __APPLE_CC__
+# define CPUSTATES 5
+# define CP_IDLE 4
+#endif /* __APPLE_CC__ */
 
 /**
  * Sanity check
