diff -uNbr xc/config/cf/darwin.cf xc-new/config/cf/darwin.cf
--- xc/config/cf/darwin.cf	Tue Jan 15 16:22:31 2002
+++ xc-new/config/cf/darwin.cf	Wed Jul 24 19:46:39 2002
@@ -62,7 +62,7 @@
 #define MathLibrary             /**/
 
 /* we don't have a termcap library */
-#define TermcapLibrary          /**/
+#define TermcapLibrary -lncurses
 
 /* we don't have a dbm library */
 #define DBMLibrary              /**/
@@ -125,7 +125,7 @@
 #define BuildServer             YES
 
 /* our custom X server */
-#define XDarwinServer           YES
+#define XDarwinServer           NO
 
 /* Thanks to the IOKit, the X server does not have to be SetUID. */
 #ifndef InstallXserverSetUID
diff -uNbr xc/include/Xos_r.h xc-new/include/Xos_r.h
--- xc/include/Xos_r.h	Fri Dec 14 14:53:26 2001
+++ xc-new/include/Xos_r.h	Wed Jul 24 19:45:02 2002
@@ -254,7 +254,7 @@
  * fields.
  */
    
-#if defined(__NetBSD__) || defined(__FreeBSD__) || defined(__OpenBSD__)
+#if defined(__NetBSD__) || defined(__FreeBSD__) || defined(__OpenBSD__) || defined(__APPLE__)
 static __inline__ void _Xpw_copyPasswd(_Xgetpwparams p)
 {
    memcpy(&(p).pws, (p).pwp, sizeof(struct passwd));
