diff -uNbr xc/config/cf/darwin.cf xc-new/config/cf/darwin.cf
--- xc/config/cf/darwin.cf	2002-01-15 16:22:31.000000000 -0500
+++ xc-new/config/cf/darwin.cf	2002-07-30 21:13:32.000000000 -0500
@@ -27,12 +27,12 @@
 #define HasUsableFileMmap       YES
 #define HasZlib                 YES
 #ifndef HasShm
-# define HasShm                 YES
+#define HasShm                 YES
 #endif
 #define HasNdbm                 YES
 
 #ifndef HasGroff
-# define HasGroff               YES
+#define HasGroff               YES
 #endif
 
 /* Libtool on Darwin always uses minor numbers */
@@ -50,6 +50,16 @@
 # define XFree86Devel           NO
 #endif
 
+#ifdef PpcDarwinArchitecture
+# define DarwinMachineDefines   -D__powerpc__
+# define OptimizedCDebugFlags   -O2
+#endif /* PpcDarwinArchitecture */
+
+#ifdef i386DarwinArchitecture
+# define DarwinMachineDefines   -D__i386__
+# define OptimizedCDebugFlags   -O2 -fno-strength-reduce
+#endif /* i386DarwinArchitecture */
+
 /*
  * This enables Quartz compatibility.
  * Turn off to build on raw Darwin systems.
@@ -59,13 +69,13 @@
 #endif
 
 /* we don't need -lm */
-#define MathLibrary             /**/
+#define MathLibrary             -lm
 
 /* we don't have a termcap library */
-#define TermcapLibrary          /**/
+#define TermcapLibrary		-lncurses
 
 /* we don't have a dbm library */
-#define DBMLibrary              /**/
+#define DBMLibrary              -lgdbm
 
 #define MkdirHierCmd            mkdir -p
 
@@ -83,9 +93,9 @@
 # define DefaultCCOptions       -Wall -Wpointer-arith -Wstrict-prototypes \
                                 -Wmissing-prototypes -Wmissing-declarations \
                                 -Wredundant-decls -Wnested-externs \
-                                -no-cpp-precomp
+                                -no-cpp-precomp -I@prefix@/include -L@prefix@/lib
 #else
-# define DefaultCCOptions       -Wall -Wpointer-arith -no-cpp-precomp
+# define DefaultCCOptions       -Wall -Wpointer-arith -no-cpp-precomp -I@prefix@/include -L@prefix@/lib
 #endif
 #endif
 
@@ -96,7 +106,7 @@
 
 #ifdef PpcDarwinArchitecture
 # define DarwinMachineDefines   -D__powerpc__
-# define OptimizedCDebugFlags   -O2
+# define OptimizedCDebugFlags   -Os
 # define ByteOrder              X_BIG_ENDIAN
 #endif /* PpcDarwinArchitecture */
 
@@ -125,7 +135,7 @@
 #define BuildServer             YES
 
 /* our custom X server */
-#define XDarwinServer           YES
+#define XDarwinServer           YES
 
 /* Thanks to the IOKit, the X server does not have to be SetUID. */
 #ifndef InstallXserverSetUID
diff -uNbr xc/config/cf/darwinLib.rules xc-new/config/cf/darwinLib.rules
--- xc/config/cf/darwinLib.rules	2002-01-15 19:39:59.000000000 -0500
+++ xc-new/config/cf/darwinLib.rules	2002-07-30 21:13:32.000000000 -0500
@@ -1,4 +1,4 @@
-XCOMM $XFree86: xc/config/cf/darwinLib.rules,v 1.4 2002/01/16 00:39:59 keithp Exp $
+XCOMM $XFree86: xc/config/cf/darwinLib.rules,v 1.5 2002/07/17 01:06:04 torrey Exp $
 /*
  * Darwin/Mac OS X shared library rules
  */
@@ -57,14 +57,14 @@
  * the library gone for long periods.
  */
 #ifndef SharedLibraryTarget
-#ifdef UseInstalled
-#define LinkBuildSonameLibrary(lib) true
-#else
-#define LinkBuildSonameLibrary(lib) (RemoveFile($(BUILDLIBDIR)/lib); \
+# ifdef UseInstalled
+#  define LinkBuildSonameLibrary(lib) true
+# else
+#  define LinkBuildSonameLibrary(lib) (RemoveFile($(BUILDLIBDIR)/lib); \
 	cd $(BUILDLIBDIR); $(LN) $(BUILDINCTOP)/$(CURRENT_DIR)/lib .)
-#endif
+# endif
 
-#define SharedLibraryTarget(libname,rev,solist,down,up)			@@\
+# define SharedLibraryTarget(libname,rev,solist,down,up)		@@\
 AllTarget(Concat(lib,libname.rev.dylib))				@@\
 									@@\
 Concat(lib,libname.rev.dylib):  solist $(EXTRALIBRARYDEPS)		@@\
@@ -93,18 +93,18 @@
  * SharedDepLibraryTarget - generate rules to create a shared library.
  */
 #ifndef SharedDepLibraryTarget
-#ifdef UseInstalled
-#ifndef LinkBuildSonameLibrary
-#define LinkBuildSonameLibrary(lib) true
-#endif
-#else
-#ifndef LinkBuildSonameLibrary
-#define LinkBuildSonameLibrary(lib) (RemoveFile($(BUILDLIBDIR)/lib); \
+# ifdef UseInstalled
+#  ifndef LinkBuildSonameLibrary
+#   define LinkBuildSonameLibrary(lib) true
+#  endif
+# else
+#  ifndef LinkBuildSonameLibrary
+#   define LinkBuildSonameLibrary(lib) (RemoveFile($(BUILDLIBDIR)/lib); \
 	cd $(BUILDLIBDIR); $(LN) $(BUILDINCTOP)/$(CURRENT_DIR)/lib .)
-#endif
-#endif
+#  endif
+# endif
 
-#define SharedDepLibraryTarget(libname,rev,deplist,solist,down,up)	@@\
+# define SharedDepLibraryTarget(libname,rev,deplist,solist,down,up)	@@\
 AllTarget(Concat(lib,libname.rev.dylib))				@@\
 									@@\
 Concat(lib,libname.rev.dylib):  deplist $(EXTRALIBRARYDEPS)		@@\
@@ -129,6 +129,46 @@
 
 #endif /* SharedDepLibraryTarget */
 
+/*
+ * SharedDepCplusplusLibraryTarget - generate rules to create a shared library.
+ */
+#ifndef SharedDepCplusplusLibraryTarget
+# ifdef UseInstalled
+#  ifndef LinkBuildSonameLibrary
+#   define LinkBuildSonameLibrary(lib) true
+#  endif
+# else
+#  ifndef LinkBuildSonameLibrary
+#   define LinkBuildSonameLibrary(lib) (RemoveFile($(BUILDLIBDIR)/lib); \
+	cd $(BUILDLIBDIR); $(LN) $(BUILDINCTOP)/$(CURRENT_DIR)/lib .)
+#  endif
+# endif
+
+# define SharedDepCplusplusLibraryTarget(libname,rev,deplist,solist,down,up)	@@\
+AllTarget(Concat(lib,libname.rev.dylib))				@@\
+									@@\
+Concat(lib,libname.rev.dylib):  deplist $(EXTRALIBRARYDEPS)		@@\
+	$(RM) $@~							@@\
+	@MAJREV=`expr rev : '\([^.]*\)'`; \				@@\
+	  INSTALLNAME=Concat(lib,libname.$$MAJREV.dylib); \		@@\
+	  set -x; (cd down; $(CXX) -o up/$@~ $(SHLIBLDFLAGS) -install_name $(USRLIBDIR)/$$INSTALLNAME -current_version rev -compatibility_version rev solist $(REQUIREDLIBS)); \ @@\
+	  $(RM) $$INSTALLNAME; $(LN) $@ $$INSTALLNAME; \		@@\
+	  LinkBuildSonameLibrary($$INSTALLNAME)				@@\
+	$(RM) $@							@@\
+	$(MV) $@~ $@							@@\
+	@if $(SOSYMLINK); then (set -x; \				@@\
+	  $(RM) Concat(lib,libname.dylib); \				@@\
+	  $(LN) $@ Concat(lib,libname.dylib)); fi			@@\
+	LinkBuildLibrary($@)						@@\
+	LinkBuildLibraryMaybe(Concat(lib,libname.dylib),$(SOSYMLINK))	@@\
+									@@\
+clean::									@@\
+	@MAJREV=`expr rev : '\([^.]*\)'`; \				@@\
+	set -x; $(RM) Concat(lib,libname.$$MAJREV.dylib)		@@\
+	$(RM) Concat(lib,libname.rev.dylib) Concat(lib,libname.dylib)
+
+#endif /* SharedDepCplusplusLibraryTarget */
+
 #ifndef SharedDepModuleTarget
 #define SharedDepModuleTarget(name,deps,solist)				@@\
 AllTarget(name)								@@\
diff -uNbr xc/include/Xos_r.h xc-new/include/Xos_r.h
--- xc/include/Xos_r.h	2001-12-14 14:53:26.000000000 -0500
+++ xc-new/include/Xos_r.h	2002-07-30 21:03:48.000000000 -0500
@@ -254,7 +254,7 @@
  * fields.
  */
    
-#if defined(__NetBSD__) || defined(__FreeBSD__) || defined(__OpenBSD__)
+#if defined(__NetBSD__) || defined(__FreeBSD__) || defined(__OpenBSD__) || defined(__APPLE__)
 static __inline__ void _Xpw_copyPasswd(_Xgetpwparams p)
 {
    memcpy(&(p).pws, (p).pwp, sizeof(struct passwd));
diff -uNbr xc/lib/GLU/libnurbs/internals/Imakefile xc-new/lib/GLU/libnurbs/internals/Imakefile
--- xc/lib/GLU/libnurbs/internals/Imakefile	2001-06-11 02:23:11.000000000 -0500
+++ xc-new/lib/GLU/libnurbs/internals/Imakefile	2002-07-30 21:21:49.000000000 -0500
@@ -143,8 +143,12 @@
 	-I$(TOP)/include \
 	-I$(TOP)/include/GL
 
-#if SystemV4 || defined(DarwinArchitecture)
+#if SystemV4
 OSDEFINES = -DNEEDCEILF
+#elif defined(DarwinArchitecture)
+# if OSMajorVersion <= 5
+OSDEFINES = -DNEEDCEILF
+# endif
 #else
 OSDEFINES = -D_EXTENSIONS_
 #endif
diff -uNbr xc/programs/Xserver/hw/darwin/bundle/XDarwin.pbproj/project.pbxproj xc-new/programs/Xserver/hw/darwin/bundle/XDarwin.pbproj/project.pbxproj
--- xc/programs/Xserver/hw/darwin/bundle/XDarwin.pbproj/project.pbxproj	2002-01-14 02:12:22.000000000 -0500
+++ xc-new/programs/Xserver/hw/darwin/bundle/XDarwin.pbproj/project.pbxproj	2002-07-30 21:03:57.000000000 -0500
@@ -324,7 +324,7 @@
 				0A79E1A4004499A1CE6F79C2,
 			);
 			buildSettings = {
-				INSTALL_PATH = "";
+				INSTALL_PATH = /;
 				OPTIMIZATION_CFLAGS = "";
 				OTHER_CFLAGS = "";
 				OTHER_LDFLAGS = "";
@@ -338,7 +338,7 @@
 			);
 			isa = PBXApplicationTarget;
 			name = XDarwin;
-			productInstallPath = "";
+			productInstallPath = /;
 			productName = XDarwin;
 			productReference = 0A79E19E004499A1CE6F79C2;
 			productSettingsXML = "<?xml version=\"1.0\" encoding=\"UTF-8\"?>
diff -uNbr xc/programs/xterm/Tekproc.c xc-new/programs/xterm/Tekproc.c
--- xc/programs/xterm/Tekproc.c	2002-01-05 17:05:02.000000000 -0500
+++ xc-new/programs/xterm/Tekproc.c	2002-07-30 21:03:48.000000000 -0500
@@ -105,7 +105,7 @@
 #include <data.h>
 #include <error.h>
 #include <menu.h>
-
+#include "menu.h"
 #ifdef MINIX
 #include <sys/nbio.h>
 
diff -uNbr xc/programs/xterm/charproc.c xc-new/programs/xterm/charproc.c
--- xc/programs/xterm/charproc.c	2002-01-07 16:02:44.000000000 -0500
+++ xc-new/programs/xterm/charproc.c	2002-07-30 21:03:48.000000000 -0500
@@ -128,6 +128,8 @@
 #include <xcharmouse.h>
 #include <charclass.h>
 
+#include "menu.h"
+
 #if OPT_ZICONBEEP || OPT_TOOLBAR
 #define HANDLE_STRUCT_NOTIFY 1
 #endif
diff -uNbr xc/programs/xterm/fontutils.c xc-new/programs/xterm/fontutils.c
--- xc/programs/xterm/fontutils.c	2001-06-18 14:09:26.000000000 -0500
+++ xc-new/programs/xterm/fontutils.c	2002-07-30 21:03:48.000000000 -0500
@@ -48,7 +48,7 @@
 #include <xterm.h>
 
 #include <stdio.h>
-
+#include "menu.h"
 /* from X11/Xlibint.h - not all vendors install this file */
 #define CI_NONEXISTCHAR(cs) (((cs)->width == 0) && \
 			     (((cs)->rbearing|(cs)->lbearing| \
diff -uNbr xc/programs/xterm/menu.c xc-new/programs/xterm/menu.c
--- xc/programs/xterm/menu.c	2002-01-05 17:05:03.000000000 -0500
+++ xc-new/programs/xterm/menu.c	2002-07-30 21:03:48.000000000 -0500
@@ -51,6 +51,7 @@
 #include <xterm.h>
 #include <data.h>
 #include <menu.h>
+#include "menu.h"
 #include <fontutils.h>
 
 #include <X11/StringDefs.h>
diff -uNbr xc/programs/xterm/misc.c xc-new/programs/xterm/misc.c
--- xc/programs/xterm/misc.c	2001-10-23 20:21:24.000000000 -0500
+++ xc-new/programs/xterm/misc.c	2002-07-30 21:03:48.000000000 -0500
@@ -80,6 +80,7 @@
 #include <data.h>
 #include <error.h>
 #include <menu.h>
+#include "menu.h"
 #include <fontutils.h>
 #include <xcharmouse.h>
 #include <xstrings.h>
diff -uNbr xc/programs/xterm/scrollbar.c xc-new/programs/xterm/scrollbar.c
--- xc/programs/xterm/scrollbar.c	2001-09-08 20:07:26.000000000 -0500
+++ xc-new/programs/xterm/scrollbar.c	2002-07-30 21:03:48.000000000 -0500
@@ -67,6 +67,7 @@
 #include <data.h>
 #include <error.h>
 #include <menu.h>
+#include "menu.h"
 #include <xcharmouse.h>
 
 /* Event handlers */
diff -uNbr xc/programs/xterm/util.c xc-new/programs/xterm/util.c
--- xc/programs/xterm/util.c	2002-01-05 17:05:03.000000000 -0500
+++ xc-new/programs/xterm/util.c	2002-07-30 21:03:48.000000000 -0500
@@ -63,6 +63,7 @@
 #include <data.h>
 #include <error.h>
 #include <menu.h>
+#include "menu.h"
 #include <fontutils.h>
 
 #if OPT_WIDE_CHARS
