diff -uNbr xc/config/cf/darwin.cf xc-new/config/cf/darwin.cf
--- xc/config/cf/darwin.cf	Sat Aug 10 16:46:41 2002
+++ xc-new/config/cf/darwin.cf	Sun Aug 11 13:28:06 2002
@@ -24,6 +24,9 @@
 #define HasSnprintf             YES
 #define HasPutenv               YES
 #define HasBSD44Sockets         YES
+#define HasCplusplus		YES
+#define HasPosixThreads		YES
+#define HasThreadSafeAPI	YES
 #define HasUsableFileMmap       YES
 #define HasZlib                 YES
 #ifndef HasShm
@@ -35,6 +38,22 @@
 # define HasGroff               YES
 #endif
 
+/*
+ * Define OS capabilities based on version
+ */
+
+#if OSMajorVersion >= 6
+# ifndef HasNCurses
+#  define HasNCurses		YES
+# endif
+# define HasTermcap		YES
+# define TermcapLibrary		-L@FINKPREFIX@/lib -lncurses
+#else
+# define HasTermcap		NO
+# define TermcapLibrary		/**/
+# define DBMLibrary		/**/
+#endif
+
 /* Libtool on Darwin always uses minor numbers */
 #define LibtoolMinorVersions	YES
 
@@ -82,14 +101,15 @@
 #endif
 
 /* we don't need -lm */
-#define MathLibrary             /**/
+#define MathLibrary             -lm
 
-#define TermcapLibrary          -lcurses
+#define MkdirHierCmd            mkdir -p
 
-/* we don't have a dbm library */
-#define DBMLibrary              /**/
+#define BuildGLXLibrary		YES
+#define BuildGLULibrary		YES
+#define ThreadedX		YES
 
-#define MkdirHierCmd            mkdir -p
+#define FontconfigDir		ProjectRoot/etc
 
 /* our cpp isn't in /lib/cpp and it doesn't like -undef */
 #define CppCmd                  /usr/bin/cpp
@@ -115,10 +135,12 @@
 #ifndef LibraryCCOptions
 # define LibraryCCOptions       DefaultCCOptions -fno-common
 #endif
+#define SystemMTDefines		-D_REENTRANT
+#define MTSafeAPIDefines	-DXUSE_MTSAFE_API -DXNO_MTSAFE_PWDAPI
 
 #ifdef PpcDarwinArchitecture
 # define DarwinMachineDefines   -D__powerpc__
-# define OptimizedCDebugFlags   -O2
+# define OptimizedCDebugFlags   -Os
 # define ByteOrder              X_BIG_ENDIAN
 #endif /* PpcDarwinArchitecture */
 
@@ -138,7 +160,7 @@
  * Darwin's setlocale() doesn't work properly, so use X11's instead.
  */
 #define StandardDefines         DarwinMachineDefines -D__DARWIN__ \
-                                -DNO_ALLOCA -DX_LOCALE -DCSRG_BASED
+                                -DNO_ALLOCA -DX_LOCALE -DCSRG_BASED -I@FINKPREFIX@/include
 
 /*
  * Set this to NO to just build the client libs, which should work without
diff -uNbr xc/include/Xos_r.h xc-new/include/Xos_r.h
--- xc/include/Xos_r.h	Sat Aug 10 16:48:46 2002
+++ xc-new/include/Xos_r.h	Sat Aug 10 17:18:25 2002
@@ -256,7 +256,7 @@
  * fields.
  */
    
-#if defined(__NetBSD__) || defined(__FreeBSD__) || defined(__OpenBSD__)
+#if defined(__NetBSD__) || defined(__FreeBSD__) || defined(__OpenBSD__) || defined(__APPLE__)
 static __inline__ void _Xpw_copyPasswd(_Xgetpwparams p)
 {
    memcpy(&(p).pws, (p).pwp, sizeof(struct passwd));
diff -uNbr xc/programs/xterm/Tekproc.c xc-new/programs/xterm/Tekproc.c
--- xc/programs/xterm/Tekproc.c	Sat Aug 10 16:50:28 2002
+++ xc-new/programs/xterm/Tekproc.c	Sat Aug 10 17:20:21 2002
@@ -109,7 +109,7 @@
 #include <Tekparse.h>
 #include <data.h>
 #include <error.h>
-#include <menu.h>
+#include "menu.h"
 
 #ifdef MINIX
 #include <sys/nbio.h>
diff -uNbr xc/programs/xterm/button.c xc-new/programs/xterm/button.c
--- xc/programs/xterm/button.c	Sat Aug 10 16:50:28 2002
+++ xc-new/programs/xterm/button.c	Sat Aug 10 17:22:36 2002
@@ -71,7 +71,7 @@
 
 #include <data.h>
 #include <error.h>
-#include <menu.h>
+#include "menu.h"
 #include <xcharmouse.h>
 #include <charclass.h>
 
diff -uNbr xc/programs/xterm/charproc.c xc-new/programs/xterm/charproc.c
--- xc/programs/xterm/charproc.c	Sat Aug 10 16:50:29 2002
+++ xc-new/programs/xterm/charproc.c	Sat Aug 10 17:21:13 2002
@@ -129,7 +129,7 @@
 #include <VTparse.h>
 #include <data.h>
 #include <error.h>
-#include <menu.h>
+#include "menu.h"
 #include <main.h>
 #include <fontutils.h>
 #include <xcharmouse.h>
diff -uNbr xc/programs/xterm/fontutils.c xc-new/programs/xterm/fontutils.c
--- xc/programs/xterm/fontutils.c	Sat Aug 10 16:50:30 2002
+++ xc-new/programs/xterm/fontutils.c	Sat Aug 10 17:21:05 2002
@@ -43,7 +43,7 @@
 #include <fontutils.h>
 
 #include <data.h>
-#include <menu.h>
+#include "menu.h"
 #include <xstrings.h>
 #include <xterm.h>
 
diff -uNbr xc/programs/xterm/main.c xc-new/programs/xterm/main.c
--- xc/programs/xterm/main.c	Sat Aug 10 16:50:30 2002
+++ xc-new/programs/xterm/main.c	Sat Aug 10 17:20:55 2002
@@ -118,7 +118,7 @@
 
 #include <data.h>
 #include <error.h>
-#include <menu.h>
+#include "menu.h"
 #include <main.h>
 #include <xstrings.h>
 #include <xterm_io.h>
diff -uNbr xc/programs/xterm/menu.c xc-new/programs/xterm/menu.c
--- xc/programs/xterm/menu.c	Sat Aug 10 16:50:30 2002
+++ xc-new/programs/xterm/menu.c	Sat Aug 10 17:20:49 2002
@@ -49,7 +49,7 @@
 
 #include <xterm.h>
 #include <data.h>
-#include <menu.h>
+#include "menu.h"
 #include <fontutils.h>
 
 #include <X11/StringDefs.h>
diff -uNbr xc/programs/xterm/misc.c xc-new/programs/xterm/misc.c
--- xc/programs/xterm/misc.c	Sat Aug 10 16:50:31 2002
+++ xc-new/programs/xterm/misc.c	Sat Aug 10 17:20:41 2002
@@ -79,7 +79,7 @@
 
 #include <data.h>
 #include <error.h>
-#include <menu.h>
+#include "menu.h"
 #include <fontutils.h>
 #include <xcharmouse.h>
 #include <xstrings.h>
diff -uNbr xc/programs/xterm/os2main.c xc-new/programs/xterm/os2main.c
--- xc/programs/xterm/os2main.c	Sat Aug 10 16:50:31 2002
+++ xc-new/programs/xterm/os2main.c	Sat Aug 10 17:20:35 2002
@@ -89,7 +89,7 @@
 
 #include <data.h>
 #include <error.h>
-#include <menu.h>
+#include "menu.h"
 #include <main.h>
 #include <xstrings.h>
 #include <xterm_io.h>
diff -uNbr xc/programs/xterm/scrollbar.c xc-new/programs/xterm/scrollbar.c
--- xc/programs/xterm/scrollbar.c	Sat Aug 10 16:50:31 2002
+++ xc-new/programs/xterm/scrollbar.c	Sat Aug 10 17:20:29 2002
@@ -72,7 +72,7 @@
 
 #include <data.h>
 #include <error.h>
-#include <menu.h>
+#include "menu.h"
 #include <xcharmouse.h>
 
 /* Event handlers */
diff -uNbr xc/programs/xterm/util.c xc-new/programs/xterm/util.c
--- xc/programs/xterm/util.c	Sat Aug 10 16:50:31 2002
+++ xc-new/programs/xterm/util.c	Sat Aug 10 17:20:13 2002
@@ -62,7 +62,7 @@
 
 #include <data.h>
 #include <error.h>
-#include <menu.h>
+#include "menu.h"
 #include <fontutils.h>
 
 #if OPT_WIDE_CHARS
