diff -Nur xc/config/cf/darwin.cf xc-patched/config/cf/darwin.cf
--- xc/config/cf/darwin.cf	Tue Jan 15 16:22:31 2002
+++ xc-patched/config/cf/darwin.cf	Tue Jul 30 14:29:57 2002
@@ -27,12 +27,12 @@
 #define HasUsableFileMmap       YES
 #define HasZlib                 YES
 #ifndef HasShm
-# define HasShm                 YES
+#define HasShm                 YES
 #endif
 #define HasNdbm                 YES
 
 #ifndef HasGroff
-# define HasGroff               YES
+#define HasGroff               YES
 #endif
 
 /* Libtool on Darwin always uses minor numbers */
@@ -59,13 +59,13 @@
 #endif
 
 /* we don't need -lm */
-#define MathLibrary             /**/
+#define MathLibrary             -lm
 
 /* we don't have a termcap library */
-#define TermcapLibrary          /**/
+#define TermcapLibrary		-lncurses
 
 /* we don't have a dbm library */
-#define DBMLibrary              /**/
+#define DBMLibrary              -lgdbm
 
 #define MkdirHierCmd            mkdir -p
 
@@ -83,9 +83,9 @@
 # define DefaultCCOptions       -Wall -Wpointer-arith -Wstrict-prototypes \
                                 -Wmissing-prototypes -Wmissing-declarations \
                                 -Wredundant-decls -Wnested-externs \
-                                -no-cpp-precomp
+                                -no-cpp-precomp -I@prefix@/include -L@prefix@/lib -lstdc++
 #else
-# define DefaultCCOptions       -Wall -Wpointer-arith -no-cpp-precomp
+# define DefaultCCOptions       -Wall -Wpointer-arith -no-cpp-precomp -I@prefix@/include -L@prefix@/lib -lstdc++
 #endif
 #endif
 
@@ -96,7 +96,7 @@
 
 #ifdef PpcDarwinArchitecture
 # define DarwinMachineDefines   -D__powerpc__
-# define OptimizedCDebugFlags   -O2
+# define OptimizedCDebugFlags   -Os -mdynamic-no-pic
 # define ByteOrder              X_BIG_ENDIAN
 #endif /* PpcDarwinArchitecture */
 
@@ -125,7 +125,7 @@
 #define BuildServer             YES
 
 /* our custom X server */
-#define XDarwinServer           YES
+#define XDarwinServer           NO
 
 /* Thanks to the IOKit, the X server does not have to be SetUID. */
 #ifndef InstallXserverSetUID
diff -Nur xc/include/Xos_r.h xc-patched/include/Xos_r.h
--- xc/include/Xos_r.h	Fri Dec 14 14:53:26 2001
+++ xc-patched/include/Xos_r.h	Tue Jul 30 14:28:12 2002
@@ -254,7 +254,7 @@
  * fields.
  */
    
-#if defined(__NetBSD__) || defined(__FreeBSD__) || defined(__OpenBSD__)
+#if defined(__NetBSD__) || defined(__FreeBSD__) || defined(__OpenBSD__) || defined(__APPLE__)
 static __inline__ void _Xpw_copyPasswd(_Xgetpwparams p)
 {
    memcpy(&(p).pws, (p).pwp, sizeof(struct passwd));
diff -Nur xc/programs/xterm/Tekproc.c xc-patched/programs/xterm/Tekproc.c
--- xc/programs/xterm/Tekproc.c	Sat Jan  5 17:05:02 2002
+++ xc-patched/programs/xterm/Tekproc.c	Tue Jul 30 13:44:58 2002
@@ -105,7 +105,7 @@
 #include <data.h>
 #include <error.h>
 #include <menu.h>
-
+#include "menu.h"
 #ifdef MINIX
 #include <sys/nbio.h>
 
diff -Nur xc/programs/xterm/charproc.c xc-patched/programs/xterm/charproc.c
--- xc/programs/xterm/charproc.c	Mon Jan  7 16:02:44 2002
+++ xc-patched/programs/xterm/charproc.c	Tue Jul 30 13:39:37 2002
@@ -128,6 +128,8 @@
 #include <xcharmouse.h>
 #include <charclass.h>
 
+#include "menu.h"
+
 #if OPT_ZICONBEEP || OPT_TOOLBAR
 #define HANDLE_STRUCT_NOTIFY 1
 #endif
diff -Nur xc/programs/xterm/fontutils.c xc-patched/programs/xterm/fontutils.c
--- xc/programs/xterm/fontutils.c	Mon Jun 18 15:09:26 2001
+++ xc-patched/programs/xterm/fontutils.c	Tue Jul 30 13:42:25 2002
@@ -48,7 +48,7 @@
 #include <xterm.h>
 
 #include <stdio.h>
-
+#include "menu.h"
 /* from X11/Xlibint.h - not all vendors install this file */
 #define CI_NONEXISTCHAR(cs) (((cs)->width == 0) && \
 			     (((cs)->rbearing|(cs)->lbearing| \
diff -Nur xc/programs/xterm/menu.c xc-patched/programs/xterm/menu.c
--- xc/programs/xterm/menu.c	Sat Jan  5 17:05:03 2002
+++ xc-patched/programs/xterm/menu.c	Tue Jul 30 13:43:03 2002
@@ -51,6 +51,7 @@
 #include <xterm.h>
 #include <data.h>
 #include <menu.h>
+#include "menu.h"
 #include <fontutils.h>
 
 #include <X11/StringDefs.h>
diff -Nur xc/programs/xterm/misc.c xc-patched/programs/xterm/misc.c
--- xc/programs/xterm/misc.c	Tue Oct 23 21:21:24 2001
+++ xc-patched/programs/xterm/misc.c	Tue Jul 30 13:43:38 2002
@@ -80,6 +80,7 @@
 #include <data.h>
 #include <error.h>
 #include <menu.h>
+#include "menu.h"
 #include <fontutils.h>
 #include <xcharmouse.h>
 #include <xstrings.h>
diff -Nur xc/programs/xterm/scrollbar.c xc-patched/programs/xterm/scrollbar.c
--- xc/programs/xterm/scrollbar.c	Sat Sep  8 21:07:26 2001
+++ xc-patched/programs/xterm/scrollbar.c	Tue Jul 30 14:03:13 2002
@@ -67,6 +67,7 @@
 #include <data.h>
 #include <error.h>
 #include <menu.h>
+#include "menu.h"
 #include <xcharmouse.h>
 
 /* Event handlers */
diff -Nur xc/programs/xterm/util.c xc-patched/programs/xterm/util.c
--- xc/programs/xterm/util.c	Sat Jan  5 17:05:03 2002
+++ xc-patched/programs/xterm/util.c	Tue Jul 30 14:04:00 2002
@@ -63,6 +63,7 @@
 #include <data.h>
 #include <error.h>
 #include <menu.h>
+#include "menu.h"
 #include <fontutils.h>
 
 #if OPT_WIDE_CHARS
