diff -uNbr xIrc-2.3.8/Makefile.in xIrc-2.3.8-new/Makefile.in
--- xIrc-2.3.8/Makefile.in	2001-07-04 09:47:20.000000000 -0500
+++ xIrc-2.3.8-new/Makefile.in	2002-07-28 20:55:05.000000000 -0500
@@ -86,7 +86,7 @@
 xIrc_SOURCES = xIrc.cpp xIrcBanDialog.cpp xIrcChanNickBox.cpp xIrcChannelQuery.cpp xIrcCommands.cpp xIrcConnect1.cpp xIrcConnect2.cpp xIrcConnect3.cpp xIrcConnectDialog.cpp xIrcDccChatFrame.cpp xIrcDccFile.cpp xIrcDccQuery.cpp xIrcDccSocket.cpp xIrcIgnoreQuery.cpp xIrcInviteBox.cpp xIrcMircServerParse.cpp xIrcMsgDispatch.cpp xIrcMsgDispatchEntry.cpp xIrcMsgFrame1.cpp xIrcMsgFrame2.cpp xIrcMsgQuery.cpp xIrcNickActionQuery.cpp xIrcNickCombo.cpp xIrcNickList.cpp xIrcNickListEntry.cpp xIrcNickQuery.cpp xIrcNotifyQuery.cpp xIrcNotifyTable.cpp xIrcPeopleEdit.cpp xIrcPeopleEntry.cpp xIrcPeopleList.cpp xIrcPeopleTable.cpp xIrcServerEdit.cpp xIrcServerEntry.cpp xIrcServerList.cpp xIrcServerQuery.cpp xIrcServerTable.cpp xIrcSocket.cpp
 
 
-xIrc_LDADD = moc_xIrcBanDialog.o moc_xIrcChanNickBox.o moc_xIrcChannelQuery.o moc_xIrcConnect.o moc_xIrcConnectDialog.o moc_xIrcDccChatFrame.o moc_xIrcDccFile.o moc_xIrcDccQuery.o moc_xIrcDccSocket.o moc_xIrcIgnoreQuery.o moc_xIrcInviteBox.o moc_xIrcMsgDispatchEntry.o moc_xIrcMsgFrame.o moc_xIrcMsgQuery.o moc_xIrcNickActionQuery.o moc_xIrcNickCombo.o moc_xIrcNickQuery.o moc_xIrcNotifyQuery.o moc_xIrcNotifyTable.o moc_xIrcPeopleEdit.o moc_xIrcPeopleTable.o moc_xIrcServerEdit.o moc_xIrcServerQuery.o moc_xIrcServerTable.o moc_xIrcSocket.o -L./libX++ -lX++
+xIrc_LDADD = moc_xIrcBanDialog.o moc_xIrcChanNickBox.o moc_xIrcChannelQuery.o moc_xIrcConnect.o moc_xIrcConnectDialog.o moc_xIrcDccChatFrame.o moc_xIrcDccFile.o moc_xIrcDccQuery.o moc_xIrcDccSocket.o moc_xIrcIgnoreQuery.o moc_xIrcInviteBox.o moc_xIrcMsgDispatchEntry.o moc_xIrcMsgFrame.o moc_xIrcMsgQuery.o moc_xIrcNickActionQuery.o moc_xIrcNickCombo.o moc_xIrcNickQuery.o moc_xIrcNotifyQuery.o moc_xIrcNotifyTable.o moc_xIrcPeopleEdit.o moc_xIrcPeopleTable.o moc_xIrcServerEdit.o moc_xIrcServerQuery.o moc_xIrcServerTable.o moc_xIrcSocket.o -L./libX++ -lX++ -lX11
 
 
 datadir = $(libdir)/xIrc
diff -uNbr xIrc-2.3.8/xIrcConnect3.cpp xIrc-2.3.8-new/xIrcConnect3.cpp
--- xIrc-2.3.8/xIrcConnect3.cpp	2001-02-25 17:09:16.000000000 -0500
+++ xIrc-2.3.8-new/xIrcConnect3.cpp	2002-07-28 20:55:31.000000000 -0500
@@ -61,6 +61,7 @@
    switch (txtSel.iData)
    {
       case xIrcNickActionQuery::DccChat:
+       {
          struct sockaddr_in addr = pSocket->socketName();
          sprintf(buf, "%d", addr.sin_addr.s_addr);
          if ((pDccChatFrame = new xIrcDccChatFrame(wdtPrv, NULL, txtSel.text)) != NULL)
@@ -84,6 +85,7 @@
             }
          }
          return;
+       }
                   
       case xIrcNickActionQuery::PrivChat:
          if (findMsgFrame(txtSel.text) == NULL)
