--- /dev/null	Tue Jun 22 13:02:40 2004
+++ ./javac.in	Tue Jun 22 13:05:58 2004
@@ -0,0 +1,25 @@
+#!/bin/bash
+
+if [ ${LC_MESSAGES} ] ; then
+    language="${LC_MESSAGES}"
+elif [ ${LC_ALL} ] ; then
+    language="${LC_ALL}"
+elif [ ${LANG} ] ; then
+    language="${LANG}"
+else
+    language="ja_JP.UTF-8"
+fi
+
+case $language in
+    *utf*8*|*UTF*8*)   nkf_arg="-w" ; enc="UTF-8" ;;
+    *euc*|*EUC*|*ujis*|*UJIS*)   nkf_arg="-e" ; enc="EUCJIS" ;;
+    *sjis*|*SJIS*|*PCK*) nkf_arg="-s" ; enc="SJIS" ;;
+    *) nkf_arg="-w" ; enc="UTF-8" ;;
+esac
+
+# convert only stderr into $language 
+pipename=${TMPDIR:-/tmp}/javac-ja.$$
+mkfifo "$pipename" || exit 1
+trap 'rm -f "$pipename"' 0
+nkf ${nkf_arg} -u "$pipename" &
+/usr/bin/javac -encoding $enc "$@" 2>"$pipename"
