diff -ru audacity-src-1.1.3/configure audacity-src-1.1.3_patched/configure
--- audacity-src-1.1.3/configure	Thu Mar 20 05:38:07 2003
+++ audacity-src-1.1.3_patched/configure	Sun Apr 20 16:47:04 2003
@@ -3757,13 +3757,12 @@
 	LIBS="-framework AudioUnit -framework AudioToolbox $LIBS"
 	LIBS="-framework CoreAudio $LIBS -lz"
 
-	AFTERBUILD="cp ../audacity ../Audacity.app && `$WX_CONFIG --rezflags` ../Audacity.app ../mac/audacity.r -c auDy && /Developer/Tools/SetFile -a avBstClinmed ../Audacity.app"
+	AFTERBUILD=""
 
 	if [ "$use_ladspa" = "default" ] ; then
 		use_ladspa="no";
 	fi
 
-	OPTOBJS="$OPTOBJS \$(OBJDIR)/effects/VST/LoadVSTMac.o"
 	OPTOBJS="$OPTOBJS \$(OBJDIR)/effects/VST/VSTEffect.o"
 
 ;;
