diff -ru synaptic-0.16/common/indexcopy.cc synaptic-0.16_patched/common/indexcopy.cc
--- synaptic-0.16/common/indexcopy.cc	Wed Aug 15 21:44:09 2001
+++ synaptic-0.16_patched/common/indexcopy.cc	Sun Jun 15 13:02:10 2003
@@ -13,8 +13,6 @@
 
 #include "config.h"
 
-#include "indexcopy.h"
-
 #include <apt-pkg/error.h>
 #include <apt-pkg/progress.h>
 #include <apt-pkg/strutl.h>
@@ -27,6 +25,7 @@
 #include <sys/stat.h>
 #include <stdio.h>
 									/*}}}*/
+#include "indexcopy.h"
 
 // IndexCopy::CopyPackages - Copy the package files from the CD		/*{{{*/
 // ---------------------------------------------------------------------
diff -ru synaptic-0.16/common/indexcopy.h synaptic-0.16_patched/common/indexcopy.h
--- synaptic-0.16/common/indexcopy.h	Tue Jul 24 00:01:08 2001
+++ synaptic-0.16_patched/common/indexcopy.h	Sun Jun 15 13:00:59 2003
@@ -10,7 +10,7 @@
 #ifndef INDEXCOPY_H
 #define INDEXCOPY_H
 
-#include <vector>
+#include <vector.h>
 #include <string>
 
 class pkgTagSection;
diff -ru synaptic-0.16/common/rcdscanner.cc synaptic-0.16_patched/common/rcdscanner.cc
--- synaptic-0.16/common/rcdscanner.cc	Mon Jul 23 23:58:20 2001
+++ synaptic-0.16_patched/common/rcdscanner.cc	Sun Jun 15 12:50:05 2003
@@ -5,8 +5,8 @@
 #include <stdio.h>
 #include <dirent.h>
 #include <unistd.h>
-#include <iostream>
-#include <fstream>
+#include <iostream.h>
+#include <fstream.h>
 #include <algorithm>
 
 #include <apt-pkg/error.h>
@@ -136,7 +136,7 @@
     string File = _config->FindFile("Dir::Etc::sourcelist");
 
     // Open the stream for reading
-    ifstream F(File.c_str(),ios::in | ios::nocreate);
+    ifstream F(File.c_str(),ios::in | ((std::ios_base::openmode)0) );
     if (!F != 0)
 	return _error->Errno("ifstream::ifstream","Opening %s",File.c_str());
     
diff -ru synaptic-0.16/common/rconfiguration.cc synaptic-0.16_patched/common/rconfiguration.cc
--- synaptic-0.16/common/rconfiguration.cc	Fri Aug 10 20:56:02 2001
+++ synaptic-0.16_patched/common/rconfiguration.cc	Sun Jun 15 12:13:24 2003
@@ -180,7 +180,7 @@
 		    unsigned Depth)
 {   
    // Open the stream for reading
-   ifstream F(FName.c_str(),ios::in | ios::nocreate);
+    ifstream F(FName.c_str(),ios::in | ((std::ios_base::openmode)0) );
    if (!F != 0)
       return _error->Errno("ifstream::ifstream","Opening configuration file %s",FName.c_str());
    
diff -ru synaptic-0.16/common/rpackage.cc synaptic-0.16_patched/common/rpackage.cc
--- synaptic-0.16/common/rpackage.cc	Fri Aug 10 20:56:02 2001
+++ synaptic-0.16_patched/common/rpackage.cc	Sun Jun 15 12:30:03 2003
@@ -172,8 +172,8 @@
     if (descr.size() > sizeof(descrBuffer))
 	return "Description Too Long";
 
-    p = descr.begin();
-    end = descr.end();
+    p = &descr[0];
+    end = &descr[descr.size()];
     
     while (p != end) {
 	switch (state) {
diff -ru synaptic-0.16/common/rpackage.h synaptic-0.16_patched/common/rpackage.h
--- synaptic-0.16/common/rpackage.h	Wed Jun 27 19:26:50 2001
+++ synaptic-0.16_patched/common/rpackage.h	Sun Jun 15 12:26:08 2003
@@ -29,7 +29,7 @@
 #ifndef _RPACKAGE_H_
 #define _RPACKAGE_H_
 
-#include <vector>
+#include <vector.h>
 
 #include <apt-pkg/pkgcache.h>
 
diff -ru synaptic-0.16/common/rpackagefilter.h synaptic-0.16_patched/common/rpackagefilter.h
--- synaptic-0.16/common/rpackagefilter.h	Wed Jan 24 20:14:24 2001
+++ synaptic-0.16_patched/common/rpackagefilter.h	Sun Jun 15 12:30:37 2003
@@ -24,7 +24,7 @@
 #ifndef _RPACKAGEFILTER_H_
 # define _RPACKAGEFILTER_H_
 
-# include <vector>
+# include <vector.h>
 # include <string>
 # include <fstream.h>
 
diff -ru synaptic-0.16/common/rpackagelister.cc synaptic-0.16_patched/common/rpackagelister.cc
--- synaptic-0.16/common/rpackagelister.cc	Fri Aug 10 20:56:02 2001
+++ synaptic-0.16_patched/common/rpackagelister.cc	Sun Jun 15 12:48:17 2003
@@ -798,7 +798,7 @@
     
     bool AuthFailed = false;
     Failed = false;
-    for (pkgAcquire::Item **I = Fetcher.ItemsBegin(); I != Fetcher.ItemsEnd(); I++)
+    for (pkgAcquire::ItemIterator I = Fetcher.ItemsBegin(); I != Fetcher.ItemsEnd(); I++)
     {
 	if ((*I)->Status == pkgAcquire::Item::StatDone)
 	    continue;
@@ -880,7 +880,7 @@
 
 	// Print out errors
 	bool Failed = false;
-	for (pkgAcquire::Item **I = fetcher.ItemsBegin();
+	for (pkgAcquire::ItemIterator I = fetcher.ItemsBegin();
 	     I != fetcher.ItemsEnd(); 
 	     I++) {
 	    if ((*I)->Status == pkgAcquire::Item::StatDone &&
diff -ru synaptic-0.16/common/rpackagelister.h synaptic-0.16_patched/common/rpackagelister.h
--- synaptic-0.16/common/rpackagelister.h	Fri Aug 10 20:56:02 2001
+++ synaptic-0.16_patched/common/rpackagelister.h	Sun Jun 15 12:31:53 2003
@@ -26,8 +26,8 @@
 #define _RPACKAGELISTER_H_
 
 
-#include <vector>
-#include <map>
+#include <vector.h>
+#include <map.h>
 
 #include <regex.h>
 
diff -ru synaptic-0.16/wings/rwaboutpanel.cc synaptic-0.16_patched/wings/rwaboutpanel.cc
--- synaptic-0.16/wings/rwaboutpanel.cc	Tue Jun 26 19:49:51 2001
+++ synaptic-0.16_patched/wings/rwaboutpanel.cc	Sun Jun 15 13:26:10 2003
@@ -5,13 +5,11 @@
 
 #include "i18n.h"
 
-#include "rwaboutpanel.h"
-
-
 #include "cnc.xpm"
 #include "deb.xpm"
 #include "logo.xpm"
 
+#include "rwaboutpanel.h"
 
 static void closeWindow(WMWidget *self, void *data)
 {
diff -ru synaptic-0.16/wings/rwwindow.cc synaptic-0.16_patched/wings/rwwindow.cc
--- synaptic-0.16/wings/rwwindow.cc	Tue Feb 20 22:00:37 2001
+++ synaptic-0.16_patched/wings/rwwindow.cc	Sun Jun 15 13:29:51 2003
@@ -5,7 +5,7 @@
 
 #include "rwwindow.h"
 
-
+using namespace std;
 
 void RWWindow::windowCloseAction(WMWidget *win, void *data)
 {
diff -ru synaptic-0.16/wings/rwwindow.h synaptic-0.16_patched/wings/rwwindow.h
--- synaptic-0.16/wings/rwwindow.h	Thu Jan 25 00:30:48 2001
+++ synaptic-0.16_patched/wings/rwwindow.h	Sun Jun 15 13:27:44 2003
@@ -7,6 +7,8 @@
 
 #include <string>
 
+using namespace std;
+
 class RWWindow {   
 protected:
    WMWindow *_win;
diff -ru synaptic-0.16/wings/synaptic.cc synaptic-0.16_patched/wings/synaptic.cc
--- synaptic-0.16/wings/synaptic.cc	Fri Jul  6 21:26:31 2001
+++ synaptic-0.16_patched/wings/synaptic.cc	Sun Jun 15 13:05:36 2003
@@ -26,15 +26,14 @@
 
 #include "i18n.h"
 
-#include "rconfiguration.h"
-#include "rpackagelister.h"
-
 #include <apt-pkg/configuration.h>
 #include <apt-pkg/error.h>
 #ifdef HAVE_RPM
 # include <apt-pkg/rpmfactory.h>
 #endif
 
+#include "rconfiguration.h"
+#include "rpackagelister.h"
 
 #include <unistd.h>
 
