diff -Naur epiphany-1.2.2-old/configure epiphany-1.2.2/configure
--- epiphany-1.2.2-old/configure	Mon Mar 29 08:44:17 2004
+++ epiphany-1.2.2/configure	Fri Apr  9 22:56:17 2004
@@ -21275,6 +21275,7 @@
  1.5a) MOZILLA_SNAPSHOT=10 ;;
  1.5b) MOZILLA_SNAPSHOT=11 ;;
   1.5) MOZILLA_SNAPSHOT=11 ;;
+1.5.1) MOZILLA_SNAPSHOT=11 ;;
  1.6a) MOZILLA_SNAPSHOT=12 ;;
  1.6b) MOZILLA_SNAPSHOT=13 ;;
   1.6) MOZILLA_SNAPSHOT=13 ;;
diff -Naur epiphany-1.2.2-old/help/C/Makefile.in epiphany-1.2.2/help/C/Makefile.in
--- epiphany-1.2.2-old/help/C/Makefile.in	Mon Mar 29 08:44:07 2004
+++ epiphany-1.2.2/help/C/Makefile.in	Fri Apr  9 22:56:17 2004
@@ -456,7 +456,7 @@
 	for file in $(omffile); do \
 		$(INSTALL_DATA) $$file.out $(DESTDIR)$(omf_dest_dir)/$$file; \
 	done
-	-scrollkeeper-update -p $(scrollkeeper_localstate_dir) -o $(DESTDIR)$(omf_dest_dir)
+	#-scrollkeeper-update -p $(scrollkeeper_localstate_dir) -o $(DESTDIR)$(omf_dest_dir)
 
 uninstall-local-omf:
 	-for file in $(srcdir)/*.omf; do \
@@ -464,7 +464,7 @@
 		rm -f $(omf_dest_dir)/$$basefile; \
 	done
 	-rmdir $(omf_dest_dir)
-	-scrollkeeper-update -p $(scrollkeeper_localstate_dir)
+	#-scrollkeeper-update -p $(scrollkeeper_localstate_dir)
 
 clean-local-omf:
 	-for file in $(omffile); do \
diff -Naur epiphany-1.2.2-old/src/ephy-completion-model.h epiphany-1.2.2/src/ephy-completion-model.h
--- epiphany-1.2.2-old/src/ephy-completion-model.h	Wed Nov 19 14:19:57 2003
+++ epiphany-1.2.2/src/ephy-completion-model.h	Fri Apr  9 22:55:48 2004
@@ -34,7 +34,7 @@
 
 typedef struct EphyCompletionModelPrivate EphyCompletionModelPrivate;
 
-enum
+extern enum
 {
 	EPHY_COMPLETION_TEXT_COL,
 	EPHY_COMPLETION_ACTION_COL,
