--- anjuta-2.4.2.orig/libanjuta/anjuta-launcher.c	2008-04-06 20:13:21.000000000 +0200
+++ anjuta-2.4.2.new/libanjuta/anjuta-launcher.c	2008-10-17 15:08:35.000000000 +0200
@@ -38,7 +38,7 @@
 
 #if !defined(__sun) && !defined(__NetBSD__)
 #  ifndef FREEBSD
-#    include <pty.h>
+#    include <util.h>
 #  else
 #    include <libutil.h>
 #  endif
@@ -67,6 +67,8 @@
 #ifndef __MAX_BAUD
 #ifdef __CYGWIN__
 #define __MAX_BAUD B256000
+#elif defined __APPLE__
+#define __MAX_BAUD B230400
 #else
 #define __MAX_BAUD B460800
 #endif
--- anjuta-2.4.2.orig/plugins/terminal/terminal.c	2008-04-06 20:13:09.000000000 +0200
+++ anjuta-2.4.2.new/plugins/terminal/terminal.c	2008-10-17 17:27:49.000000000 +0200
@@ -63,7 +63,12 @@
 #include <gtk/gtk.h>
 #include <libanjuta/anjuta-plugin.h>
 
+#ifdef __APPLE__
+#include <crt_externs.h>
+#define environ (*_NSGetEnviron())
+#else
 extern char **environ;
+#endif
 
 extern GType terminal_plugin_get_type (GTypeModule *module);
 #define ANJUTA_PLUGIN_TERMINAL_TYPE         (terminal_plugin_get_type (NULL))
--- anjuta-2.4.2.orig/plugins/file-manager/Makefile.in	2008-05-26 23:40:36.000000000 +0200
+++ anjuta-2.4.2.new/plugins/file-manager/Makefile.in	2008-10-22 12:13:26.000000000 +0200
@@ -441,6 +441,7 @@
 	file-view.c \
 	file-model.c \
 	file-model.h
+libfile_manager_la_LDFLAGS = $(ANJUTA_PLUGIN_LDFLAGS)
 
 
 # Plugin dependencies
--- anjuta-2.4.2.orig/plugins/language-manager/Makefile.in	2008-05-26 23:40:39.000000000 +0200
+++ anjuta-2.4.2.new/plugins/language-manager/Makefile.in	2008-10-22 12:15:57.000000000 +0200
@@ -429,6 +429,7 @@
 
 # Plugin sources
 liblanguage_manager_la_SOURCES = plugin.c plugin.h
+liblanguage_manager_la_LDFLAGS = $(ANJUTA_PLUGIN_LDFLAGS)
 
 # Plugin dependencies
 liblanguage_manager_la_LIBADD = \
