--- orig/lib/FrameMaker/MifTree/MifTreeTags	2006-05-02 05:43:09.000000000 -0500
+++ new/lib/FrameMaker/MifTree/MifTreeTags	2007-03-19 20:07:47.000000000 -0500
@@ -1025,6 +1025,7 @@
                                ElementEnd
                                FLocked
                                FNote
+			       FNoteRef
                                FPairKern
                                FSize
                                Font
@@ -1842,6 +1843,7 @@
   FLocked                                        => 'boolean',
   FNoAdvance                                     => 'boolean',
   FNote                                          => 'ID',
+  FNoteRef                                       => 'integer',
   FNoteStartNum                                  => 'integer',
   FNumericUnderline                              => 'boolean',
   FOutline                                       => 'boolean',
@@ -2300,6 +2302,7 @@
   ElseIf
   Exclusion
   FNote
+  FNoteRef
   FPlatformName
   FmtChangeList
   Font
