diff -Naur --exclude='*~' E/Makefile E.fink/Makefile
--- E/Makefile	2009-08-06 05:27:01.000000000 +0100
+++ E.fink/Makefile	2009-12-03 11:43:15.000000000 +0000
@@ -142,10 +142,10 @@
 
 install: E
 	-sh -c 'mkdir -p $(EXECPATH)'
-	-sh -c 'cp PROVER/eprover $(EXECPATH)'
-	-sh -c 'cp PROVER/epclextract $(EXECPATH)'
-	-sh -c 'cp PROVER/eproof $(EXECPATH)'
-	-sh -c 'cp  PROVER/eground $(EXECPATH)'	
+	-sh -c 'cp PROVER/eprover $(DESTDIR)$(EXECPATH)'
+	-sh -c 'cp PROVER/epclextract $(DESTDIR)$(EXECPATH)'
+	-sh -c 'cp PROVER/eproof $(DESTDIR)$(EXECPATH)'
+	-sh -c 'cp  PROVER/eground $(DESTDIR)$(EXECPATH)'	
 
 # Also remake documentation
 
