diff -Naur --exclude='*~' ikiwiki/Makefile.PL ikiwiki.fink/Makefile.PL
--- ikiwiki/Makefile.PL	2010-01-10 05:02:38.000000000 +0100
+++ ikiwiki.fink/Makefile.PL	2010-02-22 02:59:45.000000000 +0100
@@ -122,10 +122,10 @@
 	
 	# These might fail if a regular user is installing into a home
 	# directory.
-	-install -d $(DESTDIR)/etc/ikiwiki
-	-install -m 0644 wikilist $(DESTDIR)/etc/ikiwiki
-	-install -m 0644 auto.setup $(DESTDIR)/etc/ikiwiki
-	-install -m 0644 auto-blog.setup $(DESTDIR)/etc/ikiwiki
+	-install -d $(DESTDIR)$(PREFIX)/etc/ikiwiki
+	-install -m 0644 wikilist $(DESTDIR)$(PREFIX)/etc/ikiwiki
+	-install -m 0644 auto.setup $(DESTDIR)$(PREFIX)/etc/ikiwiki
+	-install -m 0644 auto-blog.setup $(DESTDIR)$(PREFIX)/etc/ikiwiki
 }
 }
 
