diff -ru cups-1.1.14/Makedefs.in cups-1.1.14-patched/Makedefs.in
--- cups-1.1.14/Makedefs.in	2002-01-02 18:58:33.000000000 +0100
+++ cups-1.1.14-patched/Makedefs.in	2002-05-31 17:05:43.000000000 +0200
@@ -82,12 +82,12 @@
 #
 
 ARFLAGS	=	@ARFLAGS@
-CFLAGS	=	@CFLAGS@ -I.. $(OPTIONS)
-CXXFLAGS =	@CXXFLAGS@ -I.. $(OPTIONS)
+CFLAGS	=	$(RC_CFLAGS) @CFLAGS@ -I.. $(OPTIONS)
+CXXFLAGS =	$(RC_CFLAGS) @CXXFLAGS@ -I.. $(OPTIONS)
 DSOFLAGS =	@DSOFLAGS@
 DSOLIBS	=	@DSOLIBS@
 IMGLIBS	=	@IMGLIBS@ -lm
-LDFLAGS	=	@LDFLAGS@ $(OPTIM)
+LDFLAGS	=	$(RC_CFLAGS) @LDFLAGS@ $(OPTIM)
 LINKCUPS =	@LINKCUPS@
 LINKCUPSIMAGE =	@LINKCUPSIMAGE@
 LIBS	=	$(LINKCUPS) $(NETLIBS) @LIBS@
@@ -126,6 +126,8 @@
 sysconfdir	=	@sysconfdir@
 top_srcdir	=	@top_srcdir@
 
+BUILDROOT	=	$(DSTROOT)
+
 AMANDIR		=	$(BUILDROOT)@AMANDIR@
 BINDIR		=	$(BUILDROOT)@bindir@
 DATADIR		=	$(BUILDROOT)@CUPS_DATADIR@
diff -ru cups-1.1.14/Makefile cups-1.1.14-patched/Makefile
--- cups-1.1.14/Makefile	2002-02-12 20:05:36.000000000 +0100
+++ cups-1.1.14-patched/Makefile	2002-05-31 14:43:22.000000000 +0200
@@ -94,7 +94,7 @@
 	if test "x$(INITDIR)" = "x" -a "x$(INITDDIR)" != "x"; then \
 		$(INSTALL_DIR) $(BUILDROOT)$(INITDDIR); \
 		if test "$(INITDDIR)" = "/System/Library/StartupItems/PrintingServices"; then \
-			$(INSTALL_SCRIPT) cups.sh $(BUILDROOT)$(INITDDIR)/PrintingServices.sh; \
+			$(INSTALL_SCRIPT) cups.sh $(BUILDROOT)$(INITDDIR)/PrintingServices; \
 			$(INSTALL_DATA) cups.plist $(BUILDROOT)$(INITDDIR)/StartupParameters.plist; \
 			$(INSTALL_DIR) $(BUILDROOT)$(INITDDIR)/Resources/English.lproj; \
 			$(INSTALL_DATA) cups.strings $(BUILDROOT)$(INITDDIR)/Resources/English.lproj/Localizable.strings; \
diff -ru cups-1.1.14/configure cups-1.1.14-patched/configure
--- cups-1.1.14/configure	2002-02-13 21:23:11.000000000 +0100
+++ cups-1.1.14-patched/configure	2002-05-31 18:31:14.000000000 +0200
@@ -2898,7 +2898,7 @@
 	if test "$prefix" = "/"; then
 		exec_prefix="/usr"
 	else
-		exec_prefix="$prefix"
+		exec_prefix="\${prefix}"
 	fi
 fi
 
@@ -2910,7 +2910,7 @@
 	if test "$prefix" = "/"; then
 		datadir="/usr/share"
 	else
-		datadir="$prefix/share"
+		datadir="\${prefix}/share"
 	fi
 fi
 
@@ -2922,7 +2922,7 @@
 	if test "$prefix" = "/"; then
 		localstatedir="/var"
 	else
-		localstatedir="$prefix/var"
+		localstatedir="\${prefix}/var"
 	fi
 fi
 
@@ -2930,7 +2930,7 @@
 	if test "$prefix" = "/"; then
 		sysconfdir="/etc"
 	else
-		sysconfdir="$prefix/etc"
+		sysconfdir="\${prefix}/etc"
 	fi
 fi
 
@@ -3230,11 +3230,11 @@
 			DSO="\$(CC)"
 			DSOFLAGS="$DSOFLAGS -Wl,-soname,\$@ -shared \$(OPTIM)"
 			;;
-		Darwin* | FreeBSD*)
+		Darwin*)
 			LIBCUPS="libcups.2.dylib"
 			LIBCUPSIMAGE="libcupsimage.2.dylib"
-			DSO="ld"
-			DSOFLAGS="$DSOFLAGS -dylib /usr/lib/dylib1.o -lc"
+			DSO="\$(CC)"
+			DSOFLAGS="$DSOFLAGS \$(RC_CFLAGS) -dynamiclib"
 			;;
 		AIX*)
 			LIBCUPS="libcups_s.a"
@@ -3285,7 +3285,7 @@
 	# rather than to the executables.  This makes things smaller if you
 	# are using any static libraries, and it also allows us to distribute
 	# a single DSO rather than a bunch...
-	DSOLIBS="\$(LIBPNG) \$(LIBTIFF) \$(LIBJPEG) \$(LIBZ)"
+	DSOLIBS="-L\$(prefix)/lib \$(LIBPNG) \$(LIBTIFF) \$(LIBJPEG) \$(LIBZ)"
 	IMGLIBS=""
 
 	# The *BSD, HP-UX, and Solaris run-time linkers need help when
diff -ru cups-1.1.14/cups/Makefile cups-1.1.14-patched/cups/Makefile
--- cups-1.1.14/cups/Makefile	2002-01-23 18:25:34.000000000 +0100
+++ cups-1.1.14-patched/cups/Makefile	2002-05-31 17:06:00.000000000 +0200
@@ -75,7 +75,7 @@
 	$(INSTALL_DIR) $(LIBDIR)
 	$(INSTALL_LIB) $(LIBCUPS) $(LIBDIR)
 	if test $(LIBCUPS) = "libcups.so.2" -o $(LIBCUPS) = "libcups.sl.2"; then \
-		$(INSTALL_LIB) libcups.a $(LIBDIR); \
+		strip -x $(LIBDIR)/$(LIBCUPS); \
 		$(RM) $(LIBDIR)/`basename $(LIBCUPS) .2`; \
 		$(LN) $(LIBCUPS) $(LIBDIR)/`basename $(LIBCUPS) .2`; \
 	fi
@@ -107,9 +107,9 @@
 libcups.2.dylib:	$(LIBOBJS) ../Makedefs
 	echo Linking $@...
 	$(DSO) $(DSOFLAGS) -o $@ \
-		-dylib_install_name $(LIBDIR)/libcups.dylib \
-		-dylib_current_version 2.0.4 \
-		-dylib_compatibility_version 2.0.0 \
+		-install_name $(libdir)/$@ \
+		-current_version 2.0.4 \
+		-compatibility_version 2.0.0 \
 		$(LIBOBJS) $(SSLLIBS)
 	$(RM) libcups.dylib
 	$(LN) $@ libcups.dylib
diff -ru cups-1.1.14/filter/Makefile cups-1.1.14-patched/filter/Makefile
--- cups-1.1.14/filter/Makefile	2002-01-23 23:30:42.000000000 +0100
+++ cups-1.1.14-patched/filter/Makefile	2002-05-31 14:58:09.000000000 +0200
@@ -116,9 +116,9 @@
 libcupsimage.2.dylib:	$(IMAGEOBJS) ../Makedefs
 	echo Linking $@...
 	$(DSO) $(DSOFLAGS) -o $@  \
-		-dylib_install_name $(LIBDIR)/libcupsimage.dylib \
-		-dylib_current_version 2.0.0 \
-		-dylib_compatibility_version 2.0.0 \
+		-install_name $(libdir)/$@ \
+		-current_version 2.0.0 \
+		-compatibility_version 2.0.0 \
 		$(IMAGEOBJS) $(DSOLIBS) $(LINKCUPS) -lm -lcc_dynamic
 	$(RM) libcupsimage.dylib
 	$(LN) $@ libcupsimage.dylib
diff -ru cups-1.1.14/systemv/Makefile cups-1.1.14-patched/systemv/Makefile
--- cups-1.1.14/systemv/Makefile	2002-01-02 18:59:18.000000000 +0100
+++ cups-1.1.14-patched/systemv/Makefile	2002-05-31 15:02:39.000000000 +0200
@@ -51,6 +51,7 @@
 
 install:
 	$(INSTALL_DIR) $(BINDIR)
+	$(INSTALL_DIR) $(SBINDIR)
 	$(INSTALL_BIN) accept $(SBINDIR)
 	$(RM) $(SBINDIR)/reject
 	$(LN) accept $(SBINDIR)/reject
