diff -uNr kdebase-3.3.92/admin/ltmain.sh kdebase-3.3.92-new/admin/ltmain.sh
--- kdebase-3.3.92/admin/ltmain.sh	Mon Jan  3 09:50:24 2005
+++ kdebase-3.3.92-new/admin/ltmain.sh	Sat Feb  5 15:43:30 2005
@@ -4453,6 +4453,7 @@
 	# We should set the shlibpath_var
 	rpath=
 	for dir in $temp_rpath; do
+     if test "$dir" != "@FINKPREFIX@/lib"; then
 	  case $dir in
 	  [\\/]* | [A-Za-z]:[\\/]*)
 	    # Absolute path.
@@ -4463,6 +4464,7 @@
 	    rpath="$rpath\$thisdir/$dir:"
 	    ;;
 	  esac
+     fi
 	done
 	temp_rpath="$rpath"
       fi
diff -uNr kdebase-3.3.92/configure kdebase-3.3.92-new/configure
--- kdebase-3.3.92/configure	Fri Feb  4 07:48:02 2005
+++ kdebase-3.3.92-new/configure	Sat Feb  5 15:43:31 2005
@@ -8255,6 +8255,19 @@
     lt_cv_sys_max_cmd_len=8192;
     ;;
 
+  netbsd* | freebsd* | openbsd* | darwin* )
+    # This has been around since 386BSD, at least.  Likely further.
+    if test -x /sbin/sysctl; then
+      lt_cv_sys_max_cmd_len=`/sbin/sysctl -n kern.argmax`
+    elif test -x /usr/sbin/sysctl; then
+      lt_cv_sys_max_cmd_len=`/usr/sbin/sysctl -n kern.argmax`
+    else
+      lt_cv_sys_max_cmd_len=65536 # usable default for *BSD
+    fi
+    # And add a safety zone
+    lt_cv_sys_max_cmd_len=`expr $lt_cv_sys_max_cmd_len \/ 4`
+    ;;
+
  *)
     # If test is not a shell built-in, we'll probably end up computing a
     # maximum length that is only half of the actual maximum length, but
diff -uNr kdebase-3.3.92/kdeprint/kprinter/printwrapper.cpp kdebase-3.3.92-new/kdeprint/kprinter/printwrapper.cpp
--- kdebase-3.3.92/kdeprint/kprinter/printwrapper.cpp	Fri Nov 26 07:07:25 2004
+++ kdebase-3.3.92-new/kdeprint/kprinter/printwrapper.cpp	Sat Feb  5 15:43:31 2005
@@ -137,10 +137,6 @@
 {
 	KCmdLineArgs	*args = KCmdLineArgs::parsedArgs();
 
-#if defined(HAVE_SIGACTION) && !defined(HAVE_SIGSET)
-	struct sigaction action;
-#endif /* HAVE_SIGACTION && !HAVE_SIGSET*/
-
 	// read variables from command line
 	QString	printer = args->getOption("d");
 	QString	title = args->getOption("t");
@@ -324,6 +320,10 @@
 			errormsg(i18n("Nothing to print."));
 
 		// print from stdin
+
+#if defined(HAVE_SIGACTION) && !defined(HAVE_SIGSET)
+	struct sigaction action;
+#endif /* HAVE_SIGACTION && !HAVE_SIGSET*/
 
 #  if defined(HAVE_SIGSET)
 		sigset(SIGHUP, signal_handler);
diff -uNr kdebase-3.3.92/kioslave/info/kde-info2html.conf kdebase-3.3.92-new/kioslave/info/kde-info2html.conf
--- kdebase-3.3.92/kioslave/info/kde-info2html.conf	Tue Feb 24 06:30:11 2004
+++ kdebase-3.3.92-new/kioslave/info/kde-info2html.conf	Sat Feb  5 15:43:31 2005
@@ -25,6 +25,7 @@
 
 #-- location of info files.
 our @INFODIR = (
+	    "@FINKPREFIX@/share/info",
 	    "/usr/share/info",
 	    "/usr/info",
 	    "/usr/lib/info",
diff -uNr kdebase-3.3.92/kioslave/man/kio_man.cpp kdebase-3.3.92-new/kioslave/man/kio_man.cpp
--- kdebase-3.3.92/kioslave/man/kio_man.cpp	Mon Nov 15 21:10:34 2004
+++ kdebase-3.3.92-new/kioslave/man/kio_man.cpp	Sat Feb  5 15:43:31 2005
@@ -50,7 +50,7 @@
 
 MANProtocol *MANProtocol::_self = 0;
 
-#define SGML2ROFF_DIRS "/usr/lib/sgml"
+#define SGML2ROFF_DIRS "@FINKPREFIX@/share/sgml"
 
 /*
  * Drop trailing ".section[.gz]" from name
@@ -862,6 +862,7 @@
                         "/usr/sunpc/man",
                         "/usr/ncd/man",
                         "/usr/newsprint/man",
+                        "@FINKPREFIX@/share/man",
                         NULL };
 
 
diff -uNr kdebase-3.3.92/startkde kdebase-3.3.92-new/startkde
--- kdebase-3.3.92/startkde	Thu Feb  3 18:18:46 2005
+++ kdebase-3.3.92-new/startkde	Sat Feb  5 15:43:31 2005
@@ -3,6 +3,14 @@
 #  DEFAULT KDE STARTUP SCRIPT ( KDE-3.3.92 )
 #
 
+source @FINKPREFIX@/bin/init.sh
+
+for KEY in Up Down Left Right; do
+	FOO=$(xmodmap -pk | grep "($KEY)" | grep -i slash | cut -f 1); xmodmap -e"keycode $FOO = $KEY"
+done
+
+source "@FINKPREFIX@/bin/init.sh"
+
 # When the X server dies we get a HUP signal from xinit. We must ignore it
 # because we still need to do some cleanup.
 trap 'echo GOT SIGHUP' HUP
@@ -179,21 +187,24 @@
 
 echo 'startkde: Starting up...'  1>&2
 
-# run KPersonalizer before the session, if this is the first login
-if kreadconfig --file kpersonalizerrc --group General --key FirstLogin --default true --type bool; then
-    # start only dcopserver, don't start whole kdeinit (takes too long)
-    echo 'startkde: Running kpersonalizer...'  1>&2
-    dcopserver
-    kwin --lock &
-    kpersonalizer --before-session
-    # handle kpersonalizer restarts (language change)
-    while test $? -eq 1; do
-        kpersonalizer --r --before-session
-    done
-    dcopserver_shutdown
-    # shutdown will also make kwin quit, give it time to do so
-    sleep 1
-fi
+# on osx we skip this, the defaults are sane and most people
+# do bad things like "mac os menus" which makes it worse
+
+## run KPersonalizer before the session, if this is the first login
+#if kreadconfig --file kpersonalizerrc --group General --key FirstLogin --default true --type bool; then
+#    # start only dcopserver, don't start whole kdeinit (takes too long)
+#    echo 'startkde: Running kpersonalizer...'  1>&2
+#    dcopserver
+#    kwin --lock &
+#    kpersonalizer --before-session
+#    # handle kpersonalizer restarts (language change)
+#    while test $? -eq 1; do
+#        kpersonalizer --r --before-session
+#    done
+#    dcopserver_shutdown
+#    # shutdown will also make kwin quit, give it time to do so
+#    sleep 1
+#fi
 
 # the splashscreen and progress indicator
 splash=`kreadconfig --file ksplashrc --group KSplash --key Theme`
@@ -227,7 +238,29 @@
 # We only check for 255 which means that the ksmserver process could not be 
 # started, any problems thereafter, e.g. ksmserver failing to initialize, 
 # will remain undetected.
-test -n "$KDEWM" && KDEWM="--windowmanager $KDEWM"
+
+USE_PROXY=0
+if test -n "$KDEWM"; then
+  if test -z `expr "$KDEWM" : '\(.*quartz-wm.*\)'`; then
+    USE_PROXY=1
+  fi
+  KDEWM="--windowmanager $KDEWM"
+else
+  KDEWM="--windowmanager kwin"
+  USE_PROXY=1
+fi
+
+if test "$USE_PROXY" = "1"; then
+  if test -x /usr/X11R6/bin/quartz-wm; then
+    /usr/X11R6/bin/quartz-wm --only-proxy >/dev/null 2>&1 &
+  else
+    AUTOCUTSEL=`which autocutsel 2>/dev/null`
+    if test -n "$AUTOCUTSEL" && test -x "$AUTOCUTSEL"; then
+      $AUTOCUTSEL >/dev/null 2>&1 &
+    fi
+  fi
+fi
+
 kwrapper ksmserver $KDEWM 
 if test $? -eq 255; then
   # Startup error
