diff -Naur eel-2.8.0-old/eel/eel-glib-extensions.c eel-2.8.0/eel/eel-glib-extensions.c
--- eel-2.8.0-old/eel/eel-glib-extensions.c	Tue Mar  2 09:26:20 2004
+++ eel-2.8.0/eel/eel-glib-extensions.c	Wed Oct  6 23:22:49 2004
@@ -38,6 +38,9 @@
 #include <sys/utsname.h>
 #include <time.h>
 
+#include <crt_externs.h>
+#define environ (*_NSGetEnviron())
+
 /* Legal conversion specifiers, as specified in the C standard. */
 #define C_STANDARD_STRFTIME_CHARACTERS "aAbBcdHIjmMpSUwWxXyYZ"
 #define C_STANDARD_NUMERIC_STRFTIME_CHARACTERS "dHIjmMSUwWyY"
diff -Naur eel-2.8.0-old/ltmain.sh eel-2.8.0/ltmain.sh
--- eel-2.8.0-old/ltmain.sh	Mon Aug  9 13:07:34 2004
+++ eel-2.8.0/ltmain.sh	Wed Oct  6 23:22:49 2004
@@ -5266,10 +5266,10 @@
 
 # Directory that this library needs to be installed in:
 libdir='$install_libdir'"
-	  if test "$installed" = no && test "$need_relink" = yes; then
-	    $echo >> $output "\
-relink_command=\"$relink_command\""
-	  fi
+#	  if test "$installed" = no && test "$need_relink" = yes; then
+#	    $echo >> $output "\
+#relink_command=\"$relink_command\""
+#	  fi
 	done
       fi
 
diff -Naur eel-2.8.0-old/po/Makefile.in.in eel-2.8.0/po/Makefile.in.in
--- eel-2.8.0-old/po/Makefile.in.in	Mon Sep 13 04:15:52 2004
+++ eel-2.8.0/po/Makefile.in.in	Wed Oct  6 23:22:49 2004
@@ -27,7 +27,7 @@
 exec_prefix = @exec_prefix@
 datadir = @datadir@
 libdir = @libdir@
-localedir = $(libdir)/locale
+localedir = $(datadir)/locale
 gnulocaledir = $(datadir)/locale
 gettextsrcdir = $(datadir)/glib-2.0/gettext/po
 subdir = po
