diff -uNbr gq-0.6.0/configure gq-0.6.0-new/configure
--- gq-0.6.0/configure	2002-07-05 01:51:55.000000000 -0500
+++ gq-0.6.0-new/configure	2002-07-28 18:33:16.000000000 -0500
@@ -2024,8 +2024,13 @@
 # Check whether --with-kerberos-prefix or --without-kerberos-prefix was given.
 if test "${with_kerberos_prefix+set}" = set; then
   withval="$with_kerberos_prefix"
-  
+   if test -f /usr/lib/libdes524.dylib; then
+     LIBS="-L$with_kerberos_prefix/lib -lkrb5 -ldes524 $LIBS"
+   else
+     if test if /usr/lib/libdes425.dylib; then
 	     LIBS="-L$with_kerberos_prefix/lib -lkrb5 -ldes425 $LIBS"
+     fi
+   fi
 	     cat >> confdefs.h <<\EOF
 #define HAVE_KERBEROS 1
 EOF
diff -uNbr gq-0.6.0/src/Makefile.in gq-0.6.0-new/src/Makefile.in
--- gq-0.6.0/src/Makefile.in	2002-07-05 01:52:07.000000000 -0500
+++ gq-0.6.0-new/src/Makefile.in	2002-07-28 18:32:19.000000000 -0500
@@ -114,7 +114,7 @@
 DEFS = @DEFS@ -I. -I$(srcdir) -I..
 CPPFLAGS = @CPPFLAGS@
 LDFLAGS = @LDFLAGS@
-LIBS = @LIBS@
+LIBS = @LIBS@ -liconv
 gq_OBJECTS =  mainwin.o browse.o schema.o schemabrowse.o template.o gq.o \
 search.o tinput.o configfile.o prefs.o util.o errorchain.o ldif.o \
 formfill.o input.o encode.o filter.o debug.o syntax.o dt_entry.o \
