
--- configure.in.in	4 Jan 2003 04:27:29 -0000	1.1.1.5
+++ configure.in.in	4 Jan 2003 04:43:31 -0000	1.7
@@ -191,6 +191,18 @@
 AC_SUBST(LIB_KDEUI, '$(top_builddir)/kdeui/libkdeui.la')
 AC_SUBST(LIB_KIO, '$(top_builddir)/kio/libkio.la')
 AC_SUBST(LIB_KPARTS, '$(top_builddir)/kparts/libkparts.la')
+AC_SUBST(LIB_KFORMULA, '$(top_builddir)/kformula/libkformula.la')
+AC_SUBST(LIB_KSYCOCA, '$(top_builddir)/kio/libkio.la')
+AC_SUBST(LIB_KFILE, '$(top_builddir)/kio/libkio.la')
+AC_SUBST(LIB_KAB, '$(top_builddir)/kab/libkab.la')
+AC_SUBST(LIB_KHTML, '$(top_builddir)/khtml/libkhtml.la')
+AC_SUBST(LIB_DCOP, '$(top_builddir)/dcop/libDCOP.la')
+AC_SUBST(LIB_KDESU, '$(top_builddir)/kdesu/libkdesu.la')
+AC_SUBST(LIB_KDEFX, '$(top_builddir)/kdefx/libkdefx.la')
+AC_SUBST(LIB_KSPELL, '$(top_builddir)/kspell/libkspell.la')
+AC_SUBST(LIB_KDEPRINT, '$(top_builddir)/kdeprint/libkdeprint.la')
+AC_SUBST(LIB_KDED, '$(top_builddir)/kded/libkded.la')
+AC_SUBST(LIB_KDEFAKES, '$(top_builddir)/kdecore/libkdefakes.la')
 
 AC_SUBST(EXTRA_SUBDIRS)
 
--- kde.pot	14 Jan 2003 19:45:33 -0000	1.1.1.13
+++ kde.pot	21 Jan 2003 04:54:33 -0000	1.1.1.14
@@ -6,7 +6,7 @@
 msgid ""
 msgstr ""
 "Project-Id-Version: PACKAGE VERSION\n"
-"POT-Creation-Date: 2003-01-09 04:32+0100\n"
+"POT-Creation-Date: 2003-01-18 01:50+0100\n"
 "PO-Revision-Date: YEAR-MO-DA HO:MI+ZONE\n"
 "Last-Translator: FULL NAME <EMAIL@ADDRESS>\n"
 "Language-Team: LANGUAGE <LL@li.org>\n"
--- arts/kde/mcop-dcop/Makefile.am	10 Jul 2002 04:41:04 -0000	1.1.1.2
+++ arts/kde/mcop-dcop/Makefile.am	26 Nov 2002 21:00:48 -0000	1.2
@@ -1,16 +1,23 @@
 
 INCLUDES= -I$(includedir)/arts -I$(top_srcdir)/arts/kde $(all_includes)
 
+lib_LTLIBRARIES = libkmcop_main.la kmcop.la
 bin_PROGRAMS = kmcop
-lib_LTLIBRARIES = kmcop.la
 
-kmcop_la_SOURCES = mcopdcopobject.cpp kmcop.cpp kmcop.skel
-kmcop_la_LIBADD	= ../libartskde.la
+libkmcop_main_la_SOURCES = mcopdcopobject.cpp kmcop.cpp kmcop.skel
+libkmcop_main_la_LIBADD  = ../libartskde.la
+libkmcop_main_la_LDFLAGS = $(all_libraries) -version-info 1:0
+
+kmcop_la_SOURCES = kmcop_la_main.cpp
+kmcop_la_LIBADD	= ../libartskde.la libkmcop_main.la
 kmcop_la_LDFLAGS = $(all_libraries) -module -avoid-version
 kmcop_la_METASOURCES = AUTO
 
-kmcop_SOURCES = dummy.cpp
-kmcop_LDADD = kmcop.la
+kmcop_la_main.cpp: kmcop_main.cpp
+	cat kmcop_main.cpp > kmcop_la_main.cpp
+
+kmcop_SOURCES = kmcop_main.cpp
+kmcop_LDADD = libkmcop_main.la
 kmcop_LDFLAGS = $(all_libraries) $(KDE_RPATH)
 
 noinst_HEADERS = kmcop.h
@@ -20,6 +27,3 @@
 
 messages:
 	$(XGETTEXT) kmcop.cpp -o $(podir)/kmcop.pot
-
-dummy.cpp:
-	echo > dummy.cpp
--- arts/kde/mcop-dcop/kmcop.cpp	25 Jun 2002 02:46:53 -0000	1.1.1.1
+++ arts/kde/mcop-dcop/kmcop.cpp	26 Nov 2002 21:00:48 -0000	1.2
@@ -46,7 +46,9 @@
 	QPtrList<MCOPDCOPObject> list;
 };
 
-int main(int argc, char **argv)
+extern "C" int kdemain(int, char**);
+
+int kdemain(int argc, char **argv)
 {
     KAboutData aboutdata("kmcop", I18N_NOOP("KMCOP"),
                          "0.1", I18N_NOOP("KDE MCOP-DCOP Bridge"),
RCS file: arts/kde/mcop-dcop/kmcop_main.cpp
diff -N arts/kde/mcop-dcop/kmcop_main.cpp
--- /dev/null	1 Jan 1970 00:00:00 -0000
+++ arts/kde/mcop-dcop/kmcop_main.cpp	26 Nov 2002 21:00:48 -0000	1.1
@@ -0,0 +1,27 @@
+/*
+ *
+ * Copyright (c) 2001 Nick Hudson <skrll@netbsd.org>
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+ * AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN
+ * AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+extern "C" int kdemain(int, char* []);
+
+int main( int argc, char* argv[] )
+{
+        return kdemain(argc, argv);
+}
--- arts/knotify/Makefile.am	10 Jul 2002 04:41:09 -0000	1.1.1.2
+++ arts/knotify/Makefile.am	26 Nov 2002 21:00:48 -0000	1.10
@@ -3,18 +3,29 @@
 
 ####### Files
 
+lib_LTLIBRARIES = libknotify_main.la knotify.la
 bin_PROGRAMS	= knotify
-lib_LTLIBRARIES = knotify.la
 
-knotify_la_SOURCES	= knotify.cpp knotify.skel
-knotify_la_LIBADD	= -lsoundserver_idl -lqtmcop $(LIB_KDEUI)
+libknotify_main_la_SOURCES     = knotify.cpp knotify.skel
+libknotify_main_la_LIBADD      = -lsoundserver_idl -lqtmcop $(LIB_KDEUI) $(LIB_KDECORE) -lmcop -lkmedia2_idl -lartsflow_idl
+libknotify_main_la_LDFLAGS     = $(all_libraries) -avoid-version
+libknotify_main_la_METASOURCES = AUTO
+
+knotify_la_SOURCES     = knotify_la_main.cpp
+knotify_la_LIBADD      = libknotify_main.la
 knotify_la_LDFLAGS = $(all_libraries) -module -avoid-version
 knotify_la_METASOURCES = AUTO
 
-knotify_SOURCES = dummy.cpp
-knotify_LDADD = knotify.la
+knotify_la_main.cpp: stub_main.cpp
+	cat stub_main.cpp > knotify_la_main.cpp
+
+knotify_SOURCES = knotify_main.cpp
+knotify_LDADD = libknotify_main.la
 knotify_LDFLAGS  = $(all_libraries) $(KDE_RPATH)
 
+knotify_main.cpp: stub_main.cpp
+	cat stub_main.cpp > knotify_main.cpp
+
 check_PROGRAMS		= knotifytest
 knotifytest_SOURCES	= knotifytest.cpp
 knotifytest_LDADD	= $(LIB_KDECORE)
@@ -28,5 +39,3 @@
 messages:
 	$(XGETTEXT) knotify.cpp -o $(podir)/knotify.pot
 
-dummy.cpp:
-	echo > dummy.cpp
--- arts/knotify/knotify.cpp	20 Nov 2002 00:53:48 -0000	1.1.1.5
+++ arts/knotify/knotify.cpp	26 Nov 2002 21:00:48 -0000	1.2
@@ -81,7 +81,9 @@
 };
 
 
-int main(int argc, char **argv)
+extern "C" int kdemain(int, char**);
+
+int kdemain(int argc, char **argv)
 {
     KAboutData aboutdata("knotify", I18N_NOOP("KNotify"),
                          "3.0", I18N_NOOP("KDE Notification Server"),
RCS file: arts/knotify/stub_main.cpp
diff -N arts/knotify/stub_main.cpp
--- /dev/null	1 Jan 1970 00:00:00 -0000
+++ arts/knotify/stub_main.cpp	26 Nov 2002 21:00:48 -0000	1.1
@@ -0,0 +1,27 @@
+/*
+ *
+ * Copyright (c) 2001 Nick Hudson <skrll@netbsd.org>
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+ * AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN
+ * AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+extern "C" int kdemain(int, char* []);
+
+int main( int argc, char* argv[] )
+{
+	return kdemain(argc, argv);
+}
--- dcop/Makefile.am	1 Oct 2002 03:04:42 -0000	1.1.1.4
+++ dcop/Makefile.am	28 Nov 2002 00:06:58 -0000	1.6
@@ -26,10 +26,10 @@
 
 INCLUDES=  $(all_includes)
 
+lib_LTLIBRARIES  = libDCOP.la libdcopserver_main.la dcopserver.la
 bin_PROGRAMS = dcopserver dcopserver_shutdown
 check_PROGRAMS = testdcop testdcopc
 
-lib_LTLIBRARIES  = libDCOP.la dcopserver.la
 libDCOP_la_LIBADD = KDE-ICE/libkICE.la $(LIB_QT)
 libDCOP_la_LDFLAGS = $(KDE_RPATH) -version-info 5:0:1 -no-undefined $(all_libraries)
 
@@ -45,13 +45,20 @@
 # dcopstub needs to go first for --enable-final to work
 libDCOP_la_SOURCES = dcopstub.cpp dcopref.cpp dcopobject.cpp dcopclient.cpp
 
-dcopserver_la_LDFLAGS = -module -avoid-version $(all_libraries)
-dcopserver_la_SOURCES = dcopserver.cpp dcopsignals.cpp
-dcopserver_la_LIBADD   = libDCOP.la
-
-dcopserver_LDFLAGS = $(KDE_RPATH) $(all_libraries)
-dcopserver_SOURCES = dummy.cpp
-dcopserver_LDADD   = dcopserver.la
+libdcopserver_main_la_LDFLAGS = -avoid-version $(all_libraries)
+libdcopserver_main_la_SOURCES = dcopserver.cpp dcopsignals.cpp
+libdcopserver_main_la_LIBADD  = libDCOP.la
+
+dcopserver_la_LDFLAGS = -avoid-version $(all_libraries) -module
+dcopserver_la_SOURCES = dcopserver_la_main.cpp
+dcopserver_la_LIBADD  = libdcopserver_main.la
+
+dcopserver_la_main.cpp: dcopserver_main.cpp
+	cat dcopserver_main.cpp > dcopserver_la_main.cpp
+
+dcopserver_LDFLAGS = $(KDE_RPATH) $(all_libraries) -module
+dcopserver_SOURCES = dcopserver_main.cpp
+dcopserver_LDADD   = libdcopserver_main.la
 
 dcopserver_shutdown_SOURCES = dcopserver_shutdown.c
 
@@ -64,9 +71,6 @@
 testdcopc_LDADD =  KDE-ICE/libkICE.la $(LIBSM)
 
 EXTRA_DIST = HOWTO ICE
-
-dummy.cpp:
-	echo > dummy.cpp
 
 DOXYGEN_REFERENCES = kdecore
 DOXYGEN_EXCLUDE = KDE-ICE/*
--- dcop/dcopserver.cpp	4 Jan 2003 04:27:39 -0000	1.1.1.4
+++ dcop/dcopserver.cpp	4 Jan 2003 04:43:31 -0000	1.3
@@ -1544,7 +1544,9 @@
 "Copyright (C) 1999-2001, The KDE Developers <http://www.kde.org>\n"
 ;
 
-int main( int argc, char* argv[] )
+extern "C" int kdemain(int, char**);
+
+int kdemain( int argc, char* argv[] )
 {
     bool serverid = false;
     bool nofork = false;
RCS file: dcop/dcopserver_main.cpp
diff -N dcop/dcopserver_main.cpp
--- /dev/null	1 Jan 1970 00:00:00 -0000
+++ dcop/dcopserver_main.cpp	26 Nov 2002 21:00:48 -0000	1.1
@@ -0,0 +1,27 @@
+/*
+ *
+ * Copyright (c) 2001 Nick Hudson <skrll@netbsd.org>
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+ * AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN
+ * AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+extern "C" int kdemain(int, char* []);
+
+int main( int argc, char* argv[] )
+{
+	return kdemain(argc, argv);
+}
Binary files /tmp/cvsdElOgB and /tmp/cvs8pPc4v differ
--- interfaces/kmediaplayer/Makefile.am	10 Jul 2002 04:41:58 -0000	1.1.1.1
+++ interfaces/kmediaplayer/Makefile.am	24 Nov 2002 07:11:20 -0000	1.2
@@ -3,7 +3,7 @@
 lib_LTLIBRARIES = libkmediaplayer.la
 
 libkmediaplayer_la_SOURCES = player.cpp playerdcopobject.skel view.cpp
-libkmediaplayer_la_LDFLAGS = $(all_libraries) $(KDE_RPATH) -no-undefined
+libkmediaplayer_la_LDFLAGS = $(all_libraries) $(KDE_RPATH) -no-undefined -avoid-version
 libkmediaplayer_la_LIBADD = $(LIB_KPARTS)
 libkmediaplayer_la_METASOURCES = AUTO
 
--- interfaces/kscript/Makefile.am	10 Jul 2002 04:41:59 -0000	1.1.1.2
+++ interfaces/kscript/Makefile.am	24 Nov 2002 07:11:20 -0000	1.2
@@ -9,7 +9,7 @@
 #libkscriptloader_la_LIBADD = $(LIB_KPARTS)
 
 libkscript_la_SOURCES = scriptmanager.cpp
-libkscript_la_LDFLAGS = $(all_libraries) $(KDE_RPATH) -no-undefined
+libkscript_la_LDFLAGS = $(all_libraries) $(KDE_RPATH) -no-undefined -avoid-version
 libkscript_la_LIBADD = $(LIB_KIO)
 
 kscriptinclude_HEADERS = scriptinterface.h scriptclientinterface.h scriptmanager.h
--- interfaces/kscript/sample/Makefile.am	25 Jun 2002 02:46:54 -0000	1.1.1.1
+++ interfaces/kscript/sample/Makefile.am	24 Nov 2002 07:11:20 -0000	1.2
@@ -3,7 +3,7 @@
 lib_LTLIBRARIES = libshellscript.la
 
 libshellscript_la_SOURCES = shellscript.cpp
-libshellscript_la_LDFLAGS = $(all_libraries) $(KDE_RPATH) -no-undefined
+libshellscript_la_LDFLAGS = $(all_libraries) $(KDE_RPATH) -no-undefined -avoid-version
 libshellscript_la_LIBADD = ../libkscript.la
 
 
--- interfaces/ktexteditor/Makefile.am	14 Aug 2002 14:00:12 -0000	1.1.1.3
+++ interfaces/ktexteditor/Makefile.am	24 Nov 2002 07:11:21 -0000	1.2
@@ -21,7 +21,7 @@
     viewstatusmsgdcopinterface.skel selectioninterfaceext.cpp selectionextdcopinterface.skel
     
     
-libktexteditor_la_LDFLAGS = $(all_libraries) $(KDE_RPATH)
+libktexteditor_la_LDFLAGS = $(all_libraries) $(KDE_RPATH) -avoid-version
 libktexteditor_la_LIBADD = $(LIB_KPARTS) $(top_builddir)/dcop/libDCOP.la
 
 ktexteditorinclude_HEADERS = document.h view.h editor.h plugin.h editinterface.h undointerface.h \
--- kabc/Makefile.am	20 Nov 2002 00:54:01 -0000	1.1.1.6
+++ kabc/Makefile.am	23 Nov 2002 03:50:34 -0000	1.7
@@ -4,8 +4,8 @@
   -I$(top_srcdir)/kabc -I$(top_srcdir)/kab $(all_includes)
 
 lib_LTLIBRARIES   = libkabc.la
-libkabc_la_LDFLAGS = $(all_libraries) -no-undefined -version-info 2:0:1 -module
-libkabc_la_LIBADD  = vcard/libvcard.la ../kdeui/libkdeui.la ../kio/libkio.la
+libkabc_la_LDFLAGS = $(all_libraries) -no-undefined -version-info 2:0:1
+libkabc_la_LIBADD  = $(top_builddir)/kabc/vcard/libvcard.la $(top_builddir)/kdeui/libkdeui.la $(top_builddir)/kio/libkio.la
 
 libkabc_la_SOURCES = \
   address.cpp addressbook.cpp addressee.cpp addresseedialog.cpp agent.cpp \
--- kabc/plugins/ldap/configure.in.in	14 Jan 2003 19:45:52 -0000	1.1.1.3
+++ kabc/plugins/ldap/configure.in.in	9 Feb 2003 17:00:14 -0000	1.8
@@ -19,7 +19,7 @@
   fi
   if test $with_ldap = FOUND ; then
     with_ldap=NOTFOUND
-    for ext in la so sl a ; do
+    for ext in la so sl a dylib; do
       AC_FIND_FILE(libldap.$ext, $kde_libraries /usr/lib /usr/local/lib,
         ldap_libdir)
       if test -r $ldap_libdir/libldap.$ext ; then
@@ -38,6 +38,12 @@
 case "$with_ldap" in
 no) AC_MSG_RESULT(no) ;;
 NOTFOUND) AC_MSG_RESULT(searched but not found) ;;
+framework)
+  LDAP_LIBS="-framework LDAP"
+  AC_DEFINE_UNQUOTED(HAVE_LIBLDAP, 1, [Define if you have LDAP libraries])
+  LDAP_SUBDIR="ldap"
+  AC_MSG_RESULT(Apple framework)
+  ;;
 *)
   if test "x$with_ldap" = "xFOUND" ; then
     msg="incs=$ldap_incdir libs=$ldap_libdir"
--- kabc/vcard/Makefile.am	10 Jul 2002 04:42:34 -0000	1.1.1.2
+++ kabc/vcard/Makefile.am	24 Nov 2002 07:11:21 -0000	1.5
@@ -7,6 +7,7 @@
 
 libvcard_la_SOURCES = vCard-all.cpp
 libvcard_la_LIBADD = $(LIB_KDECORE)
+libvcard_la_LDFLAGS = -avoid-version
 
 check_PROGRAMS = testwrite testread
 
--- kate/interfaces/Makefile.am	10 Jul 2002 04:42:56 -0000	1.1.1.2
+++ kate/interfaces/Makefile.am	24 Nov 2002 07:11:21 -0000	1.2
@@ -4,7 +4,7 @@
 
 libkatepartinterfaces_la_SOURCES = interfaces.cpp
 libkatepartinterfaces_la_LIBADD   = $(top_builddir)/interfaces/ktexteditor/libktexteditor.la
-libkatepartinterfaces_la_LDFLAGS = $(all_libraries) $(KDE_RPATH) -no-undefined
+libkatepartinterfaces_la_LDFLAGS = $(all_libraries) $(KDE_RPATH) -no-undefined -avoid-version
 
 kateinclude_HEADERS = document.h view.h
 kateincludedir = $(includedir)/kate
--- kate/part/kateview.cpp	20 Nov 2002 00:54:10 -0000	1.1.1.7
+++ kate/part/kateview.cpp	21 Jan 2003 04:56:05 -0000	1.1.1.8
@@ -19,7 +19,7 @@
    Boston, MA 02111-1307, USA.
 */
 
-// $Id$
+// $Id$
 
 #include "kateview.h"
 #include "kateview.moc"
@@ -414,39 +414,42 @@
     i18n("Select to Matching Bracket"),      SHIFT +  CTRL + Key_6,
     this, SLOT(shiftToMatchingBracket()),
     ac, "select_matching_bracket" );
-
-  new KAction(
-    i18n("Transpose Characters"),           CTRL          + Key_T,
-    this, SLOT(transpose()),
-    ac, "transpose_char" );
-// new KAction(
-//    i18n("Transpose Words"),                CTRL + SHIFT + Key_T,
-//    this, SLOT(transposeWord()),
-//    ac, "transpose_word" );
-// new KAction(
-//    i18n("Transpose Line"),                 CTRL + SHIFT + Key_T, ??? What key combo?
-//    this, SLOT(transposeLine()),
-//    ac, "transpose_line" );
-
-//  new KAction(
-//    i18n("Delete Word"),                    CTRL + Key_K, ??? What key combo?
-//    this, SLOT(killWord()),
-//    ac, "delete_word" );
-  new KAction(
-    i18n("Delete Line"),                    CTRL + Key_K,
-    this, SLOT(killLine()),
-    ac, "delete_line" );
-
-  new KAction(
-    i18n("Delete Word Left"),               CTRL + Key_Backspace,
-    this, SLOT(deleteWordLeft()),
-    ac, "delete_word_left" );
-
-  new KAction(
-    i18n("Delete Word Right"),              CTRL + Key_Delete,
-    this, SLOT(deleteWordRight()),
-    ac, "delete_word_right" );
-
+  // anders: shortcuts doing any changes should not be created in browserextension
+  if ( !m_doc->m_bReadOnly )
+  {
+    new KAction(
+      i18n("Transpose Characters"),           CTRL          + Key_T,
+      this, SLOT(transpose()),
+      ac, "transpose_char" );
+//   new KAction(
+//      i18n("Transpose Words"),                CTRL + SHIFT + Key_T,
+//      this, SLOT(transposeWord()),
+//      ac, "transpose_word" );
+//   new KAction(
+//      i18n("Transpose Line"),                 CTRL + SHIFT + Key_T, ??? What key combo?
+//      this, SLOT(transposeLine()),
+//      ac, "transpose_line" );
+
+//   new KAction(
+//     i18n("Delete Word"),                    CTRL + Key_K, ??? What key combo?
+//     this, SLOT(killWord()),
+//     ac, "delete_word" );
+    new KAction(
+      i18n("Delete Line"),                    CTRL + Key_K,
+      this, SLOT(killLine()),
+      ac, "delete_line" );
+
+    new KAction(
+      i18n("Delete Word Left"),               CTRL + Key_Backspace,
+      this, SLOT(deleteWordLeft()),
+      ac, "delete_word_left" );
+
+    new KAction(
+      i18n("Delete Word Right"),              CTRL + Key_Delete,
+      this, SLOT(deleteWordRight()),
+      ac, "delete_word_right" );
+  }
+  
   connect( this, SIGNAL(gotFocus(Kate::View*)),
            this, SLOT(slotGotFocus()) );
   connect( this, SIGNAL(lostFocus(Kate::View*)),
--- kconf_update/Makefile.am	25 Jun 2002 02:46:55 -0000	1.1.1.1
+++ kconf_update/Makefile.am	26 Nov 2002 21:00:48 -0000	1.4
@@ -18,22 +18,26 @@
 
 INCLUDES= $(all_includes)
 
-lib_LTLIBRARIES = kconf_update.la
-
+lib_LTLIBRARIES = libkconf_update_main.la kconf_update.la
 bin_PROGRAMS = kconf_update
-kconf_update_SOURCES = dummy.cpp
-kconf_update_LDADD = kconf_update.la
-kconf_update_LDFLAGS = $(KDE_RPATH) $(all_libraries)
 
-kconf_update_la_SOURCES = kconf_update.cpp
-kconf_update_la_LIBADD = ../kdecore/libkdecore.la
-kconf_update_la_LDFLAGS = $(all_libraries) -module -avoid-version
+libkconf_update_main_la_SOURCES = kconf_update.cpp
+libkconf_update_main_la_LIBADD = ../kdecore/libkdecore.la
+libkconf_update_main_la_LDFLAGS = $(all_libraries) -avoid-version
+
+kconf_update_la_SOURCES = kconf_update_la_main.cpp
+kconf_update_la_LIBADD = libkconf_update_main.la
+kconf_update_la_LDFLAGS = $(all_libraries) -avoid-version -module
+
+kconf_update_la_main.cpp: kconf_update_main.cpp
+	cat kconf_update_main.cpp > kconf_update_la_main.cpp
+
+kconf_update_SOURCES = kconf_update_main.cpp
+kconf_update_LDADD = libkconf_update_main.la
+kconf_update_LDFLAGS = $(KDE_RPATH) $(all_libraries)
 
 METASOURCES = AUTO
 
 # Internal
 # noinst_HEADERS = kconf_update.h
-
-dummy.cpp: 
-	echo > dummy.cpp
 
--- kconf_update/kconf_update.cpp	12 Nov 2002 03:56:45 -0000	1.1.1.3
+++ kconf_update/kconf_update.cpp	26 Nov 2002 21:00:48 -0000	1.2
@@ -779,7 +779,9 @@
 }
 
 
-int main(int argc, char **argv)
+extern "C" int kdemain(int, char**);
+
+int kdemain(int argc, char **argv)
 {
    KAboutData aboutData("kconf_update", I18N_NOOP("KConf Update"),
                         "1.0.2",
RCS file: kconf_update/kconf_update_main.cpp
diff -N kconf_update/kconf_update_main.cpp
--- /dev/null	1 Jan 1970 00:00:00 -0000
+++ kconf_update/kconf_update_main.cpp	26 Nov 2002 21:00:48 -0000	1.1
@@ -0,0 +1,27 @@
+/*
+ *
+ * Copyright (c) 2001 Nick Hudson <skrll@netbsd.org>
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+ * AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN
+ * AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+extern "C" int kdemain(int, char* []);
+
+int main( int argc, char* argv[] )
+{
+        return kdemain(argc, argv);
+}
--- kdecore/Makefile.am	20 Nov 2002 00:54:24 -0000	1.1.1.6
+++ kdecore/Makefile.am	21 Nov 2002 21:28:35 -0000	1.4
@@ -1,6 +1,6 @@
 #    This file is part of the KDE libraries
 #
-#    $Id$
+#    $Id$
 #
 #    Copyright (C) 1996-1997 Matthias Kalle Dalheimer (kalle@kde.org)
 #			  (C) 1997 Stephan Kulow (coolo@kde.org)
--- kdecore/fakes.c	4 Jan 2003 04:28:21 -0000	1.1.1.5
+++ kdecore/fakes.c	4 Jan 2003 04:43:31 -0000	1.3
@@ -19,6 +19,7 @@
    Boston, MA 02111-1307, USA.
 */
 
+#include <sys/types.h>
 #include <config.h>
 
 #ifndef HAVE_SETENV
@@ -137,7 +138,7 @@
 #include <errno.h>
 #include <stdio.h>
 
-int getdomainname(char *name, size_t len)
+int getdomainname(char *name, unsigned int len)
 {
         struct utsname uts;
         struct hostent *hent;
--- kdecore/kapplication.cpp	17 Jan 2003 14:29:00 -0000	1.1.1.12
+++ kdecore/kapplication.cpp	21 Jan 2003 04:58:46 -0000	1.1.1.13
@@ -18,7 +18,7 @@
     Boston, MA 02111-1307, USA.
         */
 
-// $Id$
+// $Id$
 
 #include "config.h"
 
@@ -276,14 +276,14 @@
 
      bool destMatch(const KURL &url, const KURL &base)
      {
-        if (destProtWildCard)
+        if (destProtEqual)
         {
-           if (!destProt.isEmpty() && !url.protocol().startsWith(destProt))
+           if (url.protocol() != base.protocol())
               return false;
         }
-        else if (destProtEqual)
+        else if (destProtWildCard)
         {
-           if (url.protocol() != base.protocol())
+           if (!destProt.isEmpty() && !url.protocol().startsWith(destProt))
               return false;
         }
         else
@@ -2383,9 +2383,11 @@
   d->urlActionRestrictions.append( new KApplicationPrivate::URLActionRule
   ("redirect", QString::null, QString::null, QString::null, "file", QString::null, QString::null, true));
   d->urlActionRestrictions.append( new KApplicationPrivate::URLActionRule
-  ("redirect", "http*", QString::null, QString::null, "file", QString::null, QString::null, false));
+  ("redirect", "http", QString::null, QString::null, "file", QString::null, QString::null, false));
+  d->urlActionRestrictions.append( new KApplicationPrivate::URLActionRule
+  ("redirect", "ftp", QString::null, QString::null, "file", QString::null, QString::null, false));
   d->urlActionRestrictions.append( new KApplicationPrivate::URLActionRule
-  ("redirect", "webdav*", QString::null, QString::null, "file", QString::null, QString::null, false));
+  ("redirect", "webdav", QString::null, QString::null, "file", QString::null, QString::null, false));
 
   d->urlActionRestrictions.append( new KApplicationPrivate::URLActionRule
   ("redirect", QString::null, QString::null, QString::null, "about", QString::null, QString::null, true));
--- kdecore/kglobalsettings.cpp	29 Oct 2002 02:36:56 -0000	1.1.1.3
+++ kdecore/kglobalsettings.cpp	29 Oct 2002 02:58:40 -0000	1.4
@@ -455,7 +455,7 @@
         *s_desktopPath += "/";
 
     // Trash Path
-    *s_trashPath = *s_desktopPath + i18n("Trash") + "/";
+    *s_trashPath = *s_desktopPath + "." + i18n("Trash") + "/";
     *s_trashPath = config->readEntry( "Trash" , *s_trashPath);
     if ( (*s_trashPath)[0] != '/' )
       s_trashPath->prepend( QDir::homeDirPath() + "/" );
--- kdecore/kiconloader.cpp	14 Jan 2003 19:46:12 -0000	1.1.1.10
+++ kdecore/kiconloader.cpp	14 Jan 2003 20:04:30 -0000	1.6
@@ -457,7 +457,7 @@
     return icon;
 }
 
-inline QString KIconLoader::unknownIconPath( int size ) const
+QString KIconLoader::unknownIconPath( int size ) const
 {
     static const QString &str_unknown = KGlobal::staticQString("unknown");
 
--- kdecore/klibloader.cpp	10 Jul 2002 04:43:40 -0000	1.1.1.2
+++ kdecore/klibloader.cpp	14 Nov 2002 21:41:20 -0000	1.5
@@ -366,13 +366,8 @@
     else
     {
       libfile = instance->dirs()->findResource( "module", libname );
-      if ( libfile.isEmpty() )
-      {
+      if ( libfile.isEmpty() ) {
         libfile = instance->dirs()->findResource( "lib", libname );
-#ifndef NDEBUG
-        if ( !libfile.isEmpty() && libname.left(3) == "lib" ) // don't warn for kdeinit modules
-          kdDebug(150) << "library " << libname << " not found under 'module' but under 'lib'" << endl;
-#endif
       }
       if ( libfile.isEmpty() )
       {
--- kded/Makefile.am	10 Jul 2002 04:44:10 -0000	1.1.1.2
+++ kded/Makefile.am	26 Nov 2002 21:00:49 -0000	1.4
@@ -19,32 +19,46 @@
 
 INCLUDES= -I$(srcdir)/.. -I$(top_srcdir) $(all_includes)
 
-lib_LTLIBRARIES = kded.la kbuildsycoca.la
+lib_LTLIBRARIES = libkded_main.la kded.la libkbuildsycoca_main.la kbuildsycoca.la
 
-kded_la_LDFLAGS = $(all_libraries) -module -avoid-version
-kded_la_LIBADD = $(LIB_KIO)
-kded_la_SOURCES = kded.cpp kbuildservicetypefactory.cpp \
+libkded_main_la_LDFLAGS = $(all_libraries) -avoid-version
+libkded_main_la_LIBADD = $(LIB_KIO)
+libkded_main_la_SOURCES = kded.cpp kbuildservicetypefactory.cpp \
 			kbuildservicefactory.cpp \
 			kbuildservicegroupfactory.cpp kdedmodule.cpp
 
-kbuildsycoca_la_LDFLAGS = $(all_libraries) -module -avoid-version
-kbuildsycoca_la_LIBADD = $(LIB_KIO)
-kbuildsycoca_la_SOURCES = kbuildsycoca.cpp kbuildservicetypefactory.cpp \
+kded_la_LDFLAGS = $(all_libraries) -module -avoid-version
+kded_la_LIBADD  = libkded_main.la
+kded_la_SOURCES = kded_la_main.cpp
+
+kded_la_main.cpp: kded_main.cpp
+	cat kded_main.cpp > kded_la_main.cpp
+
+libkbuildsycoca_main_la_LDFLAGS = $(all_libraries) -avoid-version
+libkbuildsycoca_main_la_LIBADD = $(LIB_KIO)
+libkbuildsycoca_main_la_SOURCES = kbuildsycoca.cpp kbuildservicetypefactory.cpp \
 				kbuildservicefactory.cpp \
 				kbuildservicegroupfactory.cpp \
 				kbuildimageiofactory.cpp \
 				kbuildprotocolinfofactory.cpp \
 				kctimefactory.cpp
 
+kbuildsycoca_la_LDFLAGS = $(all_libraries) -module -avoid-version
+kbuildsycoca_la_LIBADD  = libkbuildsycoca_main.la
+kbuildsycoca_la_SOURCES = kbuildsycoca_la_main.cpp
+
+kbuildsycoca_la_main.cpp: kbuildsycoca_main.cpp
+	cat kbuildsycoca_main.cpp > kbuildsycoca_la_main.cpp
+
 bin_PROGRAMS = kded kbuildsycoca kdontchangethehostname
 
 kded_LDFLAGS = $(all_libraries) $(KDE_RPATH)
-kded_LDADD = kded.la
-kded_SOURCES = dummy.cpp
+kded_LDADD   = libkded_main.la
+kded_SOURCES = kded_main.cpp
 
 kbuildsycoca_LDFLAGS = $(all_libraries) $(KDE_RPATH)
-kbuildsycoca_LDADD = kbuildsycoca.la
-kbuildsycoca_SOURCES = dummy.cpp
+kbuildsycoca_LDADD = libkbuildsycoca_main.la
+kbuildsycoca_SOURCES = kbuildsycoca_main.cpp
 
 kdontchangethehostname_LDFLAGS = $(all_libraries) $(KDE_RPATH)
 kdontchangethehostname_LDADD = $(LIB_KDECORE) 
@@ -64,6 +78,3 @@
 
 update_DATA = kded.upd
 updatedir = $(kde_datadir)/kconf_update
-
-dummy.cpp:
-	echo > dummy.cpp
--- kded/kbuildsycoca.cpp	1 Oct 2002 03:05:26 -0000	1.1.1.4
+++ kded/kbuildsycoca.cpp	26 Nov 2002 21:00:49 -0000	1.2
@@ -502,7 +502,9 @@
 static const char *appName = "kbuildsycoca";
 static const char *appVersion = "1.0";
 
-int main(int argc, char **argv)
+extern "C" int kdemain(int, char**);
+
+int kdemain(int argc, char **argv)
 {
    KLocale::setMainCatalogue("kdelibs");
    KAboutData d(appName, I18N_NOOP("KBuildSycoca"), appVersion,
RCS file: kded/kbuildsycoca_main.cpp
diff -N kded/kbuildsycoca_main.cpp
--- /dev/null	1 Jan 1970 00:00:00 -0000
+++ kded/kbuildsycoca_main.cpp	26 Nov 2002 21:00:49 -0000	1.1
@@ -0,0 +1,27 @@
+/*
+ *
+ * Copyright (c) 2001 Nick Hudson <skrll@netbsd.org>
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+ * AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN
+ * AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+extern "C" int kdemain(int, char* []);
+
+int main( int argc, char* argv[] )
+{
+        return kdemain(argc, argv);
+}
--- kded/kded.cpp	14 Aug 2002 14:00:31 -0000	1.1.1.3
+++ kded/kded.cpp	26 Nov 2002 21:00:49 -0000	1.2
@@ -653,7 +653,9 @@
   bool startup;
 };
 
-int main(int argc, char *argv[])
+extern "C" int kdemain(int, char**);
+
+int kdemain(int argc, char **argv)
 {
      KAboutData aboutData( "kded", I18N_NOOP("KDE Daemon"),
         "$Id$",
RCS file: kded/kded_main.cpp
diff -N kded/kded_main.cpp
--- /dev/null	1 Jan 1970 00:00:00 -0000
+++ kded/kded_main.cpp	26 Nov 2002 21:00:49 -0000	1.1
@@ -0,0 +1,27 @@
+/*
+ *
+ * Copyright (c) 2001 Nick Hudson <skrll@netbsd.org>
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+ * AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN
+ * AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+extern "C" int kdemain(int, char* []);
+
+int main( int argc, char* argv[] )
+{
+        return kdemain(argc, argv);
+}
--- kdeprint/Makefile.am	1 Oct 2002 03:05:31 -0000	1.1.1.4
+++ kdeprint/Makefile.am	26 Nov 2002 21:00:49 -0000	1.9
@@ -10,9 +10,8 @@
 kde_module_LTLIBRARIES = kded_kdeprintd.la
 
 kded_kdeprintd_la_SOURCES = kdeprintd.skel kdeprintd.cpp kprintprocess.cpp
-kded_kdeprintd_la_LIBADD = $(LIB_KIO)
+kded_kdeprintd_la_LIBADD = $(LIB_KIO) $(top_builddir)/kded/libkded_main.la
 kded_kdeprintd_la_LDFLAGS = $(all_libraries) -module -avoid-version
-#kded_kdeprintd_la_METASOURCES = AUTO
 
 lib_LTLIBRARIES = libkdeprint.la
 
--- kdeprint/cups/cupsdconf2/Makefile.am	1 Oct 2002 03:05:37 -0000	1.1.1.3
+++ kdeprint/cups/cupsdconf2/Makefile.am	26 Nov 2002 21:00:49 -0000	1.4
@@ -3,22 +3,32 @@
 INCLUDES= -I$(top_srcdir) -I$(top_srcdir)/kio -I$(top_srcdir)/kfile $(all_includes)
 
 bin_PROGRAMS = cupsdconf
-cupsdconf_SOURCES = main.cpp
 
-cupsdconf_LDADD   =  libcupsdconf.la
+cupsdconf_SOURCES = cupsdconf_main.cpp
+cupsdconf_LDADD   = libcupsdconf_main.la
 cupsdconf_LDFLAGS = $(all_libraries) $(KDE_RPATH)
 
+cupsdconf_main.cpp: stub_main.cpp
+	cat stub_main.cpp > cupsdconf_main.cpp
+
 # library creation
-lib_LTLIBRARIES = libcupsdconf.la
-libcupsdconf_la_SOURCES = cupsdconf.cpp cupsddialog.cpp cupsdpage.cpp cupsdcomment.cpp cupsdsplash.cpp \
+lib_LTLIBRARIES = libcupsdconf_main.la cupsdconf.la
+libcupsdconf_main_la_SOURCES = cupsdconf.cpp cupsddialog.cpp cupsdpage.cpp cupsdcomment.cpp cupsdsplash.cpp \
 			  cupsdserverpage.cpp cupsdlogpage.cpp cupsdjobspage.cpp cupsdfilterpage.cpp \
 			  qdirlineedit.cpp cupsddirpage.cpp portdialog.cpp cupsdnetworkpage.cpp \
 			  editlist.cpp cupsdbrowsingpage.cpp browsedialog.cpp cupsdsecuritypage.cpp \
 			  locationdialog.cpp addressdialog.cpp cups-util.c \
-			  qdirmultilineedit.cpp sizewidget.cpp
-libcupsdconf_la_METASOURCES = AUTO
-libcupsdconf_la_LDFLAGS = $(all_libraries) -module -avoid-version -no-undefined
-libcupsdconf_la_LIBADD = $(LIB_KIO) $(LIB_CUPS)
+			  qdirmultilineedit.cpp sizewidget.cpp main.cpp
+libcupsdconf_main_la_METASOURCES = AUTO
+libcupsdconf_main_la_LDFLAGS = $(all_libraries) -avoid-version
+libcupsdconf_main_la_LIBADD = $(LIB_KIO) $(LIB_CUPS)
+
+cupsdconf_la_SOURCES = cupsdconf_la_main.cpp
+cupsdconf_la_LDFLAGS = $(all_libraries) -avoid-version -module $(KDE_RPATH)
+cupsdconf_la_LIBADD  = libcupsdconf_main.la
+
+cupsdconf_la_main.cpp: stub_main.cpp
+	cat stub_main.cpp > cupsdconf_la_main.cpp
 
 noinst_HEADERS = cupsdconf.h cupsdpage.h cupsddialog.h cupsdcomment.h cupsdsplash.h \
 		 cupsdserverpage.h cupsdlogpage.h cupsdjobspage.h cupsdfilterpage.h \
--- kdeprint/cups/cupsdconf2/main.cpp	25 Jun 2002 02:46:57 -0000	1.1.1.1
+++ kdeprint/cups/cupsdconf2/main.cpp	26 Nov 2002 21:00:49 -0000	1.2
@@ -30,7 +30,9 @@
 	{ 0, 0, 0}
 };
 
-int main(int argc, char *argv[])
+extern "C" int kdemain(int, char**);
+
+int kdemain(int argc, char **argv)
 {
 	KCmdLineArgs::init(argc,argv,"cupsdconf",I18N_NOOP("A CUPS configuration tool"),"0.0.1");
 	KCmdLineArgs::addCmdLineOptions(options);
RCS file: kdeprint/cups/cupsdconf2/stub_main.cpp
diff -N kdeprint/cups/cupsdconf2/stub_main.cpp
--- /dev/null	1 Jan 1970 00:00:00 -0000
+++ kdeprint/cups/cupsdconf2/stub_main.cpp	26 Nov 2002 21:00:49 -0000	1.1
@@ -0,0 +1,28 @@
+/*
+ *
+ * Copyright (c) 2001 Nick Hudson <skrll@netbsd.org>
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+ * AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN
+ * AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+extern "C" int kdemain(int, char* []);
+
+int main( int argc, char* argv[] )
+{
+        return kdemain(argc, argv);
+}
+
--- kdeprint/management/Makefile.am	14 Aug 2002 14:04:14 -0000	1.1.1.3
+++ kdeprint/management/Makefile.am	26 Nov 2002 21:00:49 -0000	1.4
@@ -3,13 +3,16 @@
 lib_LTLIBRARIES = libkdeprint_management.la kaddprinterwizard.la
 bin_PROGRAMS = kaddprinterwizard
 
-kaddprinterwizard_SOURCES = dummy.cpp
-kaddprinterwizard_LDADD = kaddprinterwizard.la
+kaddprinterwizard_SOURCES = kaddprinterwizard_main.cpp
+kaddprinterwizard_LDADD = libkdeprint_management.la
 kaddprinterwizard_LDFLAGS = $(all_libraries) $(KDE_RPATH)
 
+kaddprinterwizard_main.cpp: kaddprinterwizard.cpp
+	cat kaddprinterwizard.cpp > kaddprinterwizard_main.cpp
+
 kaddprinterwizard_la_SOURCES = kaddprinterwizard.cpp
 kaddprinterwizard_la_LIBADD = libkdeprint_management.la
-kaddprinterwizard_la_LDFLAGS = -module -avoid-version $(all_libraries) $(KDE_RPATH)
+kaddprinterwizard_la_LDFLAGS = -avoid-version $(all_libraries) $(KDE_RPATH) -module
 
 libkdeprint_management_la_SOURCES = \
 	kmdbcreator.cpp kmdriverdb.cpp kmdriverdbwidget.cpp \
@@ -45,7 +48,4 @@
 
 dataicon_DATA = tileup.png tiledown.png side.png
 dataicondir = $(kde_datadir)/kdeprint
-
-dummy.cpp:
-	echo > dummy.cpp
 
--- kdeui/kcombobox.cpp	14 Jan 2003 19:46:46 -0000	1.1.1.8
+++ kdeui/kcombobox.cpp	21 Jan 2003 04:59:35 -0000	1.1.1.9
@@ -95,7 +95,7 @@
     installEventFilter( this );
     if ( lineEdit() )
     {
-        connect( lineEdit(), SIGNAL( returnPressed() ), 
+        connect( lineEdit(), SIGNAL( returnPressed() ),
                  this, SIGNAL( returnPressed() ));
     }
 }
@@ -279,7 +279,13 @@
 
     if ( d->klineEdit )
     {
-        connect( d->klineEdit, SIGNAL( returnPressed( const QString& )), 
+        // someone calling KComboBox::setEditable( false ) destroys our
+        // lineedit without us noticing. And KCompletionBase::delegate would
+        // be a dangling pointer then, so prevent that. Note: only do this
+        // when it is a KLineEdit!
+        connect( edit, SIGNAL( destroyed() ), SLOT( lineEditDeleted() ));
+        
+        connect( d->klineEdit, SIGNAL( returnPressed( const QString& )),
                  SIGNAL( returnPressed( const QString& ) ));
 
         connect( d->klineEdit, SIGNAL( completion( const QString& )),
@@ -328,6 +334,18 @@
             sel = count() - 1;
     }
     setCurrentItem(sel);
+}
+
+void KComboBox::lineEditDeleted()
+{
+    // yes, we need those ugly casts due to the multiple inheritance
+    // sender() is guaranteed to be a KLineEdit (see the connect() to the
+    // destroyed() signal
+    const KCompletionBase *base = static_cast<const KCompletionBase*>( static_cast<const KLineEdit*>( sender() ));
+
+    // is it our delegate, that is destroyed?
+    if ( base == delegate() )
+        setDelegate( 0L );
 }
 
 // *********************************************************************
--- kdeui/kcombobox.h	4 Jan 2003 04:29:37 -0000	1.1.1.6
+++ kdeui/kcombobox.h	21 Jan 2003 04:59:27 -0000	1.1.1.7
@@ -472,6 +472,9 @@
 
     virtual void wheelEvent( QWheelEvent *ev );
 
+private slots:
+    void lineEditDeleted();
+
 private:
     // Constants that represent the ID's of the popup menu.
     // TODO: See if we can replace this mess with KActionMenu
--- kdeui/kedittoolbar.cpp	10 Dec 2002 16:19:22 -0000	1.1.1.6
+++ kdeui/kedittoolbar.cpp	10 Dec 2002 16:37:43 -0000	1.7
@@ -142,13 +142,13 @@
    */
   ToolbarList findToolbars(QDomElement elem)
   {
-    static const QString &tagToolbar = KGlobal::staticQString( "ToolBar" );
-    static const QString &attrNoEdit = KGlobal::staticQString( "noEdit" );
+    // static const QString &tagToolbar = KGlobal::staticQString( "ToolBar" );
+    // static const QString &attrNoEdit = KGlobal::staticQString( "noEdit" );
     ToolbarList list;
 
     for( ; !elem.isNull(); elem = elem.nextSibling().toElement() )
     {
-      if (elem.tagName() == tagToolbar && elem.attribute( attrNoEdit ) != "true" )
+      if (elem.tagName() == KGlobal::staticQString( "ToolBar" ) && elem.attribute( KGlobal::staticQString( "noEdit" ) ) != "true" )
         list.append(elem);
 
       QDomElement child = elem.firstChild().toElement();
--- kdeui/klineeditdlg.cpp	29 Oct 2002 02:37:33 -0000	1.1.1.3
+++ kdeui/klineeditdlg.cpp	21 Jan 2003 04:59:36 -0000	1.1.1.4
@@ -54,7 +54,10 @@
   connect( this, SIGNAL(user1Clicked()), this, SLOT(slotClear()) );
   edit->setText( _value );
   if ( _value.isEmpty() )
+  {
       enableButtonOK( false );
+      enableButton(KDialogBase::User1, false);
+  }
   edit->setSelection(0, edit->text().length());
   edit->setFocus();
 }
--- kdeui/kxmlguifactory.cpp	1 Oct 2002 03:06:00 -0000	1.1.1.3
+++ kdeui/kxmlguifactory.cpp	1 Oct 2002 03:59:10 -0000	1.5
@@ -49,14 +49,10 @@
 public:
     KXMLGUIFactoryPrivate()
     {
-        static const QString &defaultMergingName = KGlobal::staticQString( "<default>" );
-        static const QString &actionList = KGlobal::staticQString( "actionlist" );
-        static const QString &name = KGlobal::staticQString( "name" );
-
         m_rootNode = new ContainerNode( 0L, QString::null, 0L );
-        m_defaultMergingName = defaultMergingName;
-        tagActionList = actionList;
-        attrName = name;
+        m_defaultMergingName = KGlobal::staticQString( "<default>" );
+        tagActionList = KGlobal::staticQString( "actionlist" );
+        attrName = KGlobal::staticQString( "name" );
     }
     ~KXMLGUIFactoryPrivate()
     {
--- khtml/Makefile.am	4 Jan 2003 04:30:33 -0000	1.1.1.5
+++ khtml/Makefile.am	4 Jan 2003 04:43:31 -0000	1.4
@@ -26,8 +26,8 @@
 
 SUBDIRS = misc dom css xml html rendering pics $(JAVA_SUBDIR) . ecma kmultipart
 
-lib_LTLIBRARIES = libkhtml.la
-kde_module_LTLIBRARIES = khtmlimagepart.la
+lib_LTLIBRARIES = libkhtml.la khtml.la
+kde_module_LTLIBRARIES = libkhtmlimagepart_main.la khtmlimagepart.la
 
 libkhtml_la_SOURCES = khtmlview.cpp khtml_part.cpp khtml_run.cpp khtml_factory.cpp khtml_settings.cc \
 		      khtml_events.cpp khtml_ext.cpp khtml_pagecache.cpp khtml_iface.skel khtml_iface.cc \
@@ -40,7 +40,7 @@
 noinst_HEADERS = design.h testkhtml.h testrender.h test_regression.h khtml_ext.h \
 		 khtml_pagecache.h khtmlimage.h khtmlpart_p.h khtml_printsettings.h
 
-libkhtml_la_LDFLAGS = -version-info 5:0:1 -module -no-undefined $(all_libraries)
+libkhtml_la_LDFLAGS = -version-info 5:0:1 -no-undefined $(all_libraries)
 libkhtml_la_LIBADD = ./xml/libkhtmlxml.la ./html/libkhtmlhtml.la \
 	./rendering/libkhtmlrender.la ./css/libkhtmlcss.la \
 	./misc/libkhtmlmisc.la ./dom/libkhtmldom.la $(LIBKJAVA) \
@@ -49,9 +49,23 @@
 	$(top_builddir)/kdeprint/libkdeprint.la  \
 	$(top_builddir)/kutils/libkutils.la 
 
-khtmlimagepart_la_SOURCES = khtmlimage.cpp
+khtml_la_LDFLAGS = -avoid-version -no-undefined $(all_libraries) -module
+khtml_la_LIBADD  = libkhtml.la
+khtml_la_SOURCES = dummy.cpp
+
+dummy.cpp:
+	echo > dummy.cpp
+
+libkhtmlimagepart_main_la_SOURCES = khtmlimage.cpp
+libkhtmlimagepart_main_la_LDFLAGS = $(KDE_RPATH) $(all_libraries) -avoid-version
+libkhtmlimagepart_main_la_LIBADD  = $(LIB_QT) $(LIB_KDEUI) $(LIB_KDECORE) $(LIB_KPARTS) libkhtml.la
+
+khtmlimagepart_la_SOURCES = khtmlimagepart_dummy.cpp
 khtmlimagepart_la_LDFLAGS = -module $(KDE_PLUGIN) $(all_libraries)
-khtmlimagepart_la_LIBADD = libkhtml.la
+khtmlimagepart_la_LIBADD  = $(LIB_QT) $(LIB_KDEUI) $(LIB_KDECORE) $(LIB_KPARTS) libkhtml.la
+
+khtmlimagepart_dummy.cpp:
+	echo > khtmlimagepart_dummy.cpp
 
 INCLUDES = -I$(top_srcdir)/kjs -I$(top_srcdir)/kimgio \
 	-I$(srcdir)/java -I$(top_srcdir)/dcop -I$(srcdir)/misc \
--- khtml/khtml_part.cpp	14 Jan 2003 19:47:13 -0000	1.1.1.12
+++ khtml/khtml_part.cpp	21 Jan 2003 04:59:49 -0000	1.1.1.13
@@ -7,6 +7,8 @@
  *                     2000 Simon Hausmann <hausmann@kde.org>
  *                     2000 Stefan Schimanski <1Stein@gmx.de>
  *                     2001 George Staikos <staikos@kde.org>
+ *                     2001-2003 Dirk Mueller <mueller@kde.org>
+ *                     2002 Apple Computer, Inc.
  *
  * This library is free software; you can redistribute it and/or
  * modify it under the terms of the GNU Library General Public
@@ -803,7 +805,7 @@
 
   // TODO: handle child target correctly! currently the script are always executed fur the parent
   if ( url.find( QString::fromLatin1( "javascript:" ), 0, false ) == 0 ) {
-      executeScript( url.right( url.length() - 11) );
+      executeScript( KURL::decode_string( url.right( url.length() - 11) ) );
       return;
   }
 
@@ -1374,7 +1376,7 @@
 {
     if ( !d->m_decoder ) {
         d->m_decoder = new khtml::Decoder();
-        if(d->m_encoding != QString::null)
+        if(!d->m_encoding.isNull())
             d->m_decoder->setEncoding(d->m_encoding.latin1(), d->m_haveEncoding);
         else
             d->m_decoder->setEncoding(settings()->encoding().latin1(), d->m_haveEncoding);
@@ -1456,6 +1458,7 @@
 
 void KHTMLPart::slotFinishedParsing()
 {
+  qDebug( "slotFinishedParsing: %s", kdBacktrace( ).latin1( ) ) ;  
   d->m_doc->setParsing(false);
   checkEmitLoadEvent();
   disconnect(d->m_doc,SIGNAL(finishedParsing()),this,SLOT(slotFinishedParsing()));
@@ -1715,7 +1718,8 @@
 void KHTMLPart::scheduleRedirection( int delay, const QString &url, bool doLockHistory )
 {
   //kdDebug(6050) << "KHTMLPart::scheduleRedirection delay=" << delay << " url=" << url << endl;
-    if( d->m_redirectURL.isEmpty() || delay < d->m_delayRedirect )
+    if ( delay < 24*60*60 &&
+       ( d->m_redirectURL.isEmpty() || delay < d->m_delayRedirect ) )
     {
        d->m_delayRedirect = delay;
        d->m_redirectURL = url;
@@ -2446,7 +2450,7 @@
   }
 
   if (url.find( QString::fromLatin1( "javascript:" ),0, false ) != -1 ) {
-    QString jscode = url.mid( url.find( "javascript:", 0, false ) );
+    QString jscode = KURL::decode_string( url.mid( url.find( "javascript:", 0, false ) ) );
     jscode = KStringHandler::rsqueeze( jscode, 80 ); // truncate if too long
     setStatusBarText( QStyleSheet::escape( jscode ), BarHoverText );
     return;
@@ -2602,7 +2606,7 @@
 
   if ( url.find( QString::fromLatin1( "javascript:" ), 0, false ) == 0 )
   {
-    executeScript( url.right( url.length() - 11 ) );
+    executeScript( KURL::decode_string( url.right( url.length() - 11 ) ) );
     return;
   }
 
@@ -2969,7 +2973,7 @@
   // Support for <frame src="javascript:string">
   if ( url.find( QString::fromLatin1( "javascript:" ), 0, false ) == 0 )
   {
-      QVariant res = executeScript( DOM::Node(frame->element()), url.right( url.length() - 11) );
+      QVariant res = executeScript( DOM::Node(frame->element()), KURL::decode_string( url.right( url.length() - 11) ) );
       KURL myurl;
       myurl.setProtocol("javascript");
       if ( res.type() == QVariant::String )
@@ -4193,7 +4197,7 @@
   d->m_settings = new KHTMLSettings(*KHTMLFactory::defaultHTMLSettings());
 
   QApplication::setOverrideCursor( waitCursor );
-  if(d->m_doc) d->m_doc->recalcStyle( NodeImpl::Force );
+  if(d->m_doc) d->m_doc->updateStyleSelector();
   QApplication::restoreOverrideCursor();
 }
 
--- khtml/ecma/kjs_html.cpp	14 Jan 2003 19:47:23 -0000	1.1.1.11
+++ khtml/ecma/kjs_html.cpp	21 Jan 2003 05:00:05 -0000	1.1.1.12
@@ -700,6 +700,7 @@
   name		KJS::HTMLElement::FormName	DontDelete
   acceptCharset	KJS::HTMLElement::FormAcceptCharset	DontDelete
   action	KJS::HTMLElement::FormAction	DontDelete
+  encoding	KJS::HTMLElement::FormEncType	DontDelete
   enctype	KJS::HTMLElement::FormEncType	DontDelete
   method	KJS::HTMLElement::FormMethod	DontDelete
   target	KJS::HTMLElement::FormTarget	DontDelete
--- khtml/ecma/kjs_html.lut.h	14 Jan 2003 19:47:23 -0000	1.1.1.6
+++ khtml/ecma/kjs_html.lut.h	21 Jan 2003 05:00:02 -0000	1.1.1.7
@@ -221,7 +221,7 @@
 namespace KJS {
 
 const struct HashEntry HTMLFormElementTableEntries[] = {
-   { "action", KJS::HTMLElement::FormAction, DontDelete, 0, &HTMLFormElementTableEntries[12] },
+   { "action", KJS::HTMLElement::FormAction, DontDelete, 0, &HTMLFormElementTableEntries[13] },
    { "enctype", KJS::HTMLElement::FormEncType, DontDelete, 0, 0 },
    { 0, 0, 0, 0, 0 },
    { "elements", KJS::HTMLElement::FormElements, DontDelete|ReadOnly, 0, &HTMLFormElementTableEntries[11] },
@@ -232,11 +232,12 @@
    { "reset", KJS::HTMLElement::FormReset, DontDelete|Function, 0, 0 },
    { "target", KJS::HTMLElement::FormTarget, DontDelete, 0, 0 },
    { "name", KJS::HTMLElement::FormName, DontDelete, 0, 0 },
+   { "encoding", KJS::HTMLElement::FormEncType, DontDelete, 0, &HTMLFormElementTableEntries[12] },
    { "method", KJS::HTMLElement::FormMethod, DontDelete, 0, 0 },
    { "submit", KJS::HTMLElement::FormSubmit, DontDelete|Function, 0, 0 }
 };
 
-const struct HashTable HTMLFormElementTable = { 2, 13, HTMLFormElementTableEntries, 11 };
+const struct HashTable HTMLFormElementTable = { 2, 14, HTMLFormElementTableEntries, 11 };
 
 }; // namespace
 
--- khtml/ecma/kjs_window.cpp	17 Jan 2003 14:30:26 -0000	1.1.1.15
+++ khtml/ecma/kjs_window.cpp	21 Jan 2003 05:00:01 -0000	1.1.1.16
@@ -405,7 +405,47 @@
   }
 
   const HashEntry* entry = Lookup::findEntry(&WindowTable, p);
-  if (entry)
+
+  // properties that work on all windows
+  if (entry) {
+    switch(entry->value) {
+    case Closed:
+      return Boolean( false );
+    case _Location:
+      // No isSafeScript test here, we must be able to _set_ location.href (#49819)
+      return Value(location());
+    case Frames:
+      return Value(frames(exec));
+    case Opener:
+      if (!m_part->opener())
+        return Null();    // ### a null Window might be better, but == null
+      else                // doesn't work yet
+        return retrieve(m_part->opener());
+    case Parent:
+      return retrieve(m_part->parentPart() ? m_part->parentPart() : (KHTMLPart*)m_part);
+    case _Window:
+    case Self:
+      return retrieve(m_part);
+    case Top: {
+      KHTMLPart *p = m_part;
+      while (p->parentPart())
+        p = p->parentPart();
+      return retrieve(p);
+    }
+    case Alert:
+    case Confirm:
+    case Prompt:
+    case Open:
+    case Focus:
+    case Blur:
+      return lookupOrCreateFunction<WindowFunc>(exec,p,this,entry->value,entry->params,entry->attr);
+    default:
+      break;
+    }
+  }
+
+  // properties that only work on safe windows
+  if (isSafeScript(exec) && entry)
   {
     //kdDebug(6070) << "token: " << entry->value << endl;
     switch( entry->value ) {
@@ -416,8 +456,6 @@
     case Status:
       return String(UString(m_part->jsStatusBarText()));
     case Document:
-      // no isSafeScript() test. Getting hold of the document object itself is allowed
-      // (to test it for null, for instance).
       if (m_part->document().isNull()) {
         kdDebug(6070) << "Document.write: adding <HTML><BODY> to create document" << endl;
         m_part->begin();
@@ -437,8 +475,6 @@
       return getCSSRuleConstructor(exec);
     case EventCtor:
       return getEventConstructor(exec);
-    case Frames:
-      return Value(frames(exec));
     case _History:
       return Value(history ? history :
                    (const_cast<Window*>(this)->history = new History(exec,m_part)));
@@ -462,9 +498,6 @@
       return Number(m_part->view()->visibleWidth());
     case Length:
       return Number(m_part->frames().count());
-    case _Location:
-      // No isSafeScript test here, we must be able to _set_ location.href (#49819)
-      return Value(location());
     case Name:
       return String(m_part->name());
     case _Navigator:
@@ -484,11 +517,6 @@
 #endif
     case OffscreenBuffering:
       return Boolean(true);
-    case Opener:
-      if (!m_part->opener())
-        return Null();    // ### a null Window might be better, but == null
-      else                // doesn't work yet
-        return retrieve(m_part->opener());
     case OuterHeight:
     case OuterWidth:
     {
@@ -502,8 +530,6 @@
       return Number(m_part->view()->contentsX());
     case PageYOffset:
       return Number(m_part->view()->contentsY());
-    case Parent:
-      return retrieve(m_part->parentPart() ? m_part->parentPart() : (KHTMLPart*)m_part);
     case Personalbar:
       return Undefined(); // ###
     case ScreenLeft:
@@ -532,15 +558,6 @@
     }
     case Scrollbars:
       return Undefined(); // ###
-    case Self:
-    case _Window:
-      return retrieve(m_part);
-    case Top: {
-      KHTMLPart *p = m_part;
-      while (p->parentPart())
-        p = p->parentPart();
-      return retrieve(p);
-    }
     case _Screen:
       return Value(screen ? screen :
                    (const_cast<Window*>(this)->screen = new Screen(exec)));
@@ -548,12 +565,6 @@
       return Value(new ImageConstructorImp(exec, m_part->document()));
     case Option:
       return Value(new OptionConstructorImp(exec, m_part->document()));
-    case Alert:
-    case Confirm:
-    case Prompt:
-    case Open:
-    case Focus:
-    case Blur:
     case Close:
     case Scroll: // compatibility
     case ScrollBy:
@@ -572,125 +583,53 @@
     case SetInterval:
     case ClearInterval:
     case Print:
-      if (isSafeScript(exec))
-        return lookupOrCreateFunction<WindowFunc>(exec,p,this,entry->value,entry->params,entry->attr);
-      else
-        return Undefined();
+      return lookupOrCreateFunction<WindowFunc>(exec,p,this,entry->value,entry->params,entry->attr);
     case Onabort:
-      if (isSafeScript(exec))
-        return getListener(exec,DOM::EventImpl::ABORT_EVENT);
-      else
-        return Undefined();
+      return getListener(exec,DOM::EventImpl::ABORT_EVENT);
     case Onblur:
-      if (isSafeScript(exec))
-        return getListener(exec,DOM::EventImpl::BLUR_EVENT);
-      else
-        return Undefined();
+      return getListener(exec,DOM::EventImpl::BLUR_EVENT);
     case Onchange:
-      if (isSafeScript(exec))
-        return getListener(exec,DOM::EventImpl::CHANGE_EVENT);
-      else
-        return Undefined();
+      return getListener(exec,DOM::EventImpl::CHANGE_EVENT);
     case Onclick:
-      if (isSafeScript(exec))
-        return getListener(exec,DOM::EventImpl::KHTML_ECMA_CLICK_EVENT);
-      else
-        return Undefined();
+      return getListener(exec,DOM::EventImpl::KHTML_ECMA_CLICK_EVENT);
     case Ondblclick:
-      if (isSafeScript(exec))
-        return getListener(exec,DOM::EventImpl::KHTML_ECMA_DBLCLICK_EVENT);
-      else
-        return Undefined();
+      return getListener(exec,DOM::EventImpl::KHTML_ECMA_DBLCLICK_EVENT);
     case Ondragdrop:
-      if (isSafeScript(exec))
-        return getListener(exec,DOM::EventImpl::KHTML_DRAGDROP_EVENT);
-      else
-        return Undefined();
+      return getListener(exec,DOM::EventImpl::KHTML_DRAGDROP_EVENT);
     case Onerror:
-      if (isSafeScript(exec))
-        return getListener(exec,DOM::EventImpl::KHTML_ERROR_EVENT);
-      else
-        return Undefined();
+      return getListener(exec,DOM::EventImpl::KHTML_ERROR_EVENT);
     case Onfocus:
-      if (isSafeScript(exec))
-        return getListener(exec,DOM::EventImpl::FOCUS_EVENT);
-      else
-        return Undefined();
+      return getListener(exec,DOM::EventImpl::FOCUS_EVENT);
     case Onkeydown:
-      if (isSafeScript(exec))
-        return getListener(exec,DOM::EventImpl::KHTML_KEYDOWN_EVENT);
-      else
-        return Undefined();
+      return getListener(exec,DOM::EventImpl::KHTML_KEYDOWN_EVENT);
     case Onkeypress:
-      if (isSafeScript(exec))
-        return getListener(exec,DOM::EventImpl::KHTML_KEYPRESS_EVENT);
-      else
-        return Undefined();
+      return getListener(exec,DOM::EventImpl::KHTML_KEYPRESS_EVENT);
     case Onkeyup:
-      if (isSafeScript(exec))
-        return getListener(exec,DOM::EventImpl::KHTML_KEYUP_EVENT);
-      else
-        return Undefined();
+      return getListener(exec,DOM::EventImpl::KHTML_KEYUP_EVENT);
     case Onload:
-      if (isSafeScript(exec))
-        return getListener(exec,DOM::EventImpl::LOAD_EVENT);
-      else
-        return Undefined();
+      return getListener(exec,DOM::EventImpl::LOAD_EVENT);
     case Onmousedown:
-      if (isSafeScript(exec))
-        return getListener(exec,DOM::EventImpl::MOUSEDOWN_EVENT);
-      else
-        return Undefined();
+      return getListener(exec,DOM::EventImpl::MOUSEDOWN_EVENT);
     case Onmousemove:
-      if (isSafeScript(exec))
-        return getListener(exec,DOM::EventImpl::MOUSEMOVE_EVENT);
-      else
-        return Undefined();
+      return getListener(exec,DOM::EventImpl::MOUSEMOVE_EVENT);
     case Onmouseout:
-      if (isSafeScript(exec))
-        return getListener(exec,DOM::EventImpl::MOUSEOUT_EVENT);
-      else
-        return Undefined();
+      return getListener(exec,DOM::EventImpl::MOUSEOUT_EVENT);
     case Onmouseover:
-      if (isSafeScript(exec))
-        return getListener(exec,DOM::EventImpl::MOUSEOVER_EVENT);
-      else
-        return Undefined();
+      return getListener(exec,DOM::EventImpl::MOUSEOVER_EVENT);
     case Onmouseup:
-      if (isSafeScript(exec))
-        return getListener(exec,DOM::EventImpl::MOUSEUP_EVENT);
-      else
-        return Undefined();
+      return getListener(exec,DOM::EventImpl::MOUSEUP_EVENT);
     case Onmove:
-      if (isSafeScript(exec))
-        return getListener(exec,DOM::EventImpl::KHTML_MOVE_EVENT);
-      else
-        return Undefined();
+      return getListener(exec,DOM::EventImpl::KHTML_MOVE_EVENT);
     case Onreset:
-      if (isSafeScript(exec))
-        return getListener(exec,DOM::EventImpl::RESET_EVENT);
-      else
-        return Undefined();
+      return getListener(exec,DOM::EventImpl::RESET_EVENT);
     case Onresize:
-      if (isSafeScript(exec))
-        return getListener(exec,DOM::EventImpl::RESIZE_EVENT);
-      else
-        return Undefined();
+      return getListener(exec,DOM::EventImpl::RESIZE_EVENT);
     case Onselect:
-      if (isSafeScript(exec))
-        return getListener(exec,DOM::EventImpl::SELECT_EVENT);
-      else
-        return Undefined();
+      return getListener(exec,DOM::EventImpl::SELECT_EVENT);
     case Onsubmit:
-      if (isSafeScript(exec))
-        return getListener(exec,DOM::EventImpl::SUBMIT_EVENT);
-      else
-        return Undefined();
+      return getListener(exec,DOM::EventImpl::SUBMIT_EVENT);
     case Onunload:
-      if (isSafeScript(exec))
-        return getListener(exec,DOM::EventImpl::UNLOAD_EVENT);
-      else
-        return Undefined();
+      return getListener(exec,DOM::EventImpl::UNLOAD_EVENT);
     }
   }
   KHTMLPart *kp = m_part->findFrame( p.qstring() );
@@ -1215,6 +1154,9 @@
       KHTMLPart* p = Window::retrieveActive(exec)->m_part;
       if ( p )
         url = p->htmlDocument().completeURL(str).string();
+      if ( !p ||
+           !static_cast<DOM::DocumentImpl*>(p->htmlDocument().handle())->isURLAllowed(url.url()) )
+          return Undefined();
     }
 
     KParts::URLArgs uargs;
--- khtml/html/html_formimpl.cpp	14 Jan 2003 19:47:26 -0000	1.1.1.12
+++ khtml/html/html_formimpl.cpp	21 Jan 2003 05:06:21 -0000	1.15
@@ -20,7 +20,7 @@
  * the Free Software Foundation, Inc., 59 Temple Place - Suite 330,
  * Boston, MA 02111-1307, USA.
  *
- * $Id$
+ * $Id$
  */
 
 #undef FORMS_DEBUG
@@ -842,9 +842,8 @@
 
     RenderStyle* _style = getDocument()->styleSelector()->styleForElement(this);
     _style->ref();
-    if (parentNode()->renderer() && _style->display() != NONE)
+    if (parentNode()->renderer() && _style->display() != NONE) {
         m_render = new RenderFieldset(this);
-    if (m_render) {
         m_render->setStyle(_style);
     }
     HTMLGenericFormElementImpl::attach();
@@ -1396,6 +1395,12 @@
     }
 }
 
+void HTMLLabelElementImpl::attach()
+{
+    // skip the generic handler
+    HTMLElementImpl::attach();
+}
+
 #if 0
 ElementImpl *HTMLLabelElementImpl::formElement()
 {
@@ -1431,11 +1436,10 @@
     addCSSProperty(CSS_PROP_PADDING_RIGHT, "1px");
     RenderStyle* _style = getDocument()->styleSelector()->styleForElement(this);
     _style->ref();
-    if (parentNode()->renderer() && _style->display() != NONE)
+    if (parentNode()->renderer() && _style->display() != NONE) {
         m_render = new RenderLegend(this);
-    if (m_render)
         m_render->setStyle(_style);
-
+    }
     HTMLGenericFormElementImpl::attach();
     _style->deref();
 }
--- khtml/html/html_formimpl.h	14 Jan 2003 19:47:26 -0000	1.1.1.9
+++ khtml/html/html_formimpl.h	21 Jan 2003 05:06:21 -0000	1.11
@@ -20,7 +20,7 @@
  * the Free Software Foundation, Inc., 59 Temple Place - Suite 330,
  * Boston, MA 02111-1307, USA.
  *
- * $Id$
+ * $Id$
  */
 #ifndef HTML_FORMIMPL_H
 #define HTML_FORMIMPL_H
@@ -314,6 +314,7 @@
 
     virtual Id id() const;
     virtual void parseAttribute(AttributeImpl *attr);
+    virtual void attach();
 
  private:
     DOMString m_formElementID;
RCS file: khtml/java/tests/badapplets/.cvsignore
diff -N khtml/java/tests/badapplets/.cvsignore
--- /dev/null	1 Jan 1970 00:00:00 -0000
+++ khtml/java/tests/badapplets/.cvsignore	21 Jan 2003 05:10:44 -0000	1.1.1.1
@@ -0,0 +1,2 @@
+*.class 
+Makefile.rules.in
--- khtml/misc/loader.cpp	14 Jan 2003 19:47:31 -0000	1.1.1.9
+++ khtml/misc/loader.cpp	21 Jan 2003 05:00:15 -0000	1.1.1.10
@@ -1030,7 +1030,7 @@
 
     emit requestStarted( req->m_docLoader, req->object );
 
-    servePendingRequests();
+    QTimer::singleShot( 0, this, SLOT( servePendingRequests() ) );
 }
 
 void Loader::servePendingRequests()
@@ -1112,7 +1112,7 @@
 #endif
 
   delete r;
-  servePendingRequests();
+  QTimer::singleShot( 0, this, SLOT( servePendingRequests() ) );
 }
 
 void Loader::slotData( KIO::Job*job, const QByteArray &data )
--- khtml/misc/loader.h	1 Oct 2002 03:06:36 -0000	1.1.1.3
+++ khtml/misc/loader.h	21 Jan 2003 05:00:15 -0000	1.1.1.4
@@ -397,8 +397,6 @@
     protected slots:
 	void slotFinished( KIO::Job * );
 	void slotData( KIO::Job *, const QByteArray & );
-
-    private:
 	void servePendingRequests();
 
 	QPtrList<Request> m_requestsPending;
--- khtml/rendering/render_form.cpp	14 Jan 2003 19:47:32 -0000	1.1.1.10
+++ khtml/rendering/render_form.cpp	21 Jan 2003 05:00:19 -0000	1.1.1.11
@@ -20,7 +20,7 @@
  * the Free Software Foundation, Inc., 59 Temple Place - Suite 330,
  * Boston, MA 02111-1307, USA.
  *
- * $Id$
+ * $Id$
  */
 
 #include <kdebug.h>
@@ -54,7 +54,6 @@
     setInline(true);   // our object is Inline
 
     m_state = 0;
-    m_button = 0;
     m_isDoubleClick = false;
 }
 
@@ -96,7 +95,7 @@
 void RenderFormElement::slotClicked()
 {
     ref();
-    QMouseEvent e2( QEvent::MouseButtonRelease, m_mousePos, m_button, m_state);
+    QMouseEvent e2( QEvent::MouseButtonRelease, m_mousePos, 1, m_state);
 
     element()->dispatchMouseEvent(&e2, EventImpl::CLICK_EVENT, m_isDoubleClick + 1);
     m_isDoubleClick = false;
@@ -106,7 +105,7 @@
 void RenderFormElement::slotPressed()
 {
     ref();
-    QMouseEvent e2( QEvent::MouseButtonPress, m_mousePos, m_button, m_state);
+    QMouseEvent e2( QEvent::MouseButtonPress, m_mousePos, 1, m_state);
     element()->dispatchMouseEvent(&e2, EventImpl::MOUSEDOWN_EVENT, 1);
     deref();
 }
@@ -114,7 +113,7 @@
 void RenderFormElement::slotReleased()
 {
     ref();
-    QMouseEvent e2( QEvent::MouseButtonRelease, m_mousePos, m_button, m_state);
+    QMouseEvent e2( QEvent::MouseButtonRelease, m_mousePos, 1, m_state);
     element()->dispatchMouseEvent(&e2, EventImpl::MOUSEUP_EVENT, 1);
     deref();
 }
@@ -253,7 +252,7 @@
 
     bool empty = raw.isEmpty();
     if ( empty )
-        raw = QString::fromLatin1(" ");
+        raw = QString::fromLatin1("X");
     QFontMetrics fm = pb->fontMetrics();
     QSize ts = fm.size( ShowPrefix, raw);
     QSize s(pb->style().sizeFromContents( QStyle::CT_PushButton, pb, ts )
@@ -665,6 +664,7 @@
 RenderFileButton::RenderFileButton(HTMLInputElementImpl *element)
     : RenderFormElement(element)
 {
+    // this sucks, it creates a grey background
     QHBox *w = new QHBox(view()->viewport());
 
     m_edit = new LineEditWidget(w);
@@ -690,18 +690,17 @@
     KHTMLAssert( !minMaxKnown() );
 
     const QFontMetrics &fm = style()->fontMetrics();
-    QSize s;
     int size = element()->size();
 
     int h = fm.lineSpacing();
     int w = fm.width( 'x' ) * (size > 0 ? size : 17); // "some"
-    w += 6 + fm.width( m_button->text() ) + 2*fm.width( ' ' );
-    s = QSize(w + 2 + 2*m_edit->frameWidth(),
-              QMAX(h, 14) + 2 + 2*m_edit->frameWidth())
+    QSize s = m_edit->style().sizeFromContents(QStyle::CT_LineEdit, m_edit,
+          QSize(w + 2 + 2*m_edit->frameWidth(), kMax(h, 14) + 2 + 2*m_edit->frameWidth()))
         .expandedTo(QApplication::globalStrut());
+    QSize bs = m_button->sizeHint();
 
-    setIntrinsicWidth( s.width() );
-    setIntrinsicHeight( s.height() );
+    setIntrinsicWidth( s.width() + bs.width() );
+    setIntrinsicHeight( kMax(s.height(), bs.height()) );
 
     RenderFormElement::calcMinMaxWidth();
 }
--- khtml/rendering/render_form.h	14 Jan 2003 19:47:32 -0000	1.1.1.7
+++ khtml/rendering/render_form.h	21 Jan 2003 05:00:20 -0000	1.1.1.8
@@ -20,7 +20,7 @@
  * the Free Software Foundation, Inc., 59 Temple Place - Suite 330,
  * Boston, MA 02111-1307, USA.
  *
- * $Id$
+ * $Id$
  */
 #ifndef RENDER_FORM_H
 #define RENDER_FORM_H
@@ -92,7 +92,6 @@
 
     QPoint m_mousePos;
     int m_state;
-    int m_button;
     bool m_isDoubleClick;
 };
 
--- khtml/xml/dom_docimpl.cpp	14 Jan 2003 19:47:36 -0000	1.1.1.8
+++ khtml/xml/dom_docimpl.cpp	21 Jan 2003 04:59:59 -0000	1.1.1.9
@@ -65,6 +65,7 @@
 #include "html/html_objectimpl.h"
 
 #include <kio/job.h>
+#include <kapplication.h>
 
 using namespace DOM;
 using namespace khtml;
@@ -1936,6 +1937,37 @@
     QPtrListIterator<NodeIteratorImpl> it(m_nodeIterators);
     for (; it.current(); ++it)
         it.current()->notifyBeforeNodeRemoval(n);
+}
+
+bool DocumentImpl::isURLAllowed(const QString& url) const
+{
+    KHTMLView *w = view();
+
+    KURL newURL(completeURL(url));
+    newURL.setRef(QString::null);
+
+    // Prohibit non-file URLs if we are asked to.
+    if (!w || w->part()->onlyLocalReferences() && newURL.protocol() != "file")
+        return false;
+
+    // do we allow this suburl ?
+    if ( !kapp || !kapp->authorizeURLAction("redirect", w->part()->url(), newURL) )
+        return false;
+
+    // We allow one level of self-reference because some sites depend on that.
+    // But we don't allow more than one.
+    bool foundSelfReference = false;
+    for (KHTMLPart *part = w->part(); part; part = part->parentPart()) {
+        KURL partURL = part->url();
+        partURL.setRef(QString::null);
+        if (partURL == newURL) {
+            if (foundSelfReference)
+                return false;
+            foundSelfReference = true;
+        }
+    }
+
+    return true;
 }
 
 AbstractViewImpl *DocumentImpl::defaultView() const
--- khtml/xml/dom_docimpl.h	14 Jan 2003 19:47:36 -0000	1.1.1.7
+++ khtml/xml/dom_docimpl.h	21 Jan 2003 04:59:58 -0000	1.1.1.8
@@ -20,7 +20,7 @@
  * the Free Software Foundation, Inc., 59 Temple Place - Suite 330,
  * Boston, MA 02111-1307, USA.
  *
- * $Id$
+ * $Id$
  */
 #ifndef _DOM_DocumentImpl_h_
 #define _DOM_DocumentImpl_h_
@@ -227,7 +227,7 @@
     QString baseTarget() const { return m_baseTarget; }
     void setBaseTarget(const QString& baseTarget) { m_baseTarget = baseTarget; }
 
-    QString completeURL(const QString& url) { return KURL(baseURL(),url,m_decoderMibEnum).url(); };
+    QString completeURL(const QString& url) const { return KURL(baseURL(),url,m_decoderMibEnum).url(); };
 
     // from cachedObjectClient
     virtual void setStyleSheet(const DOM::DOMString &url, const DOM::DOMString &sheetStr);
@@ -370,6 +370,7 @@
      * @param content The header value (value of the meta tag's "content" attribute)
      */
     void processHttpEquiv(const DOMString &equiv, const DOMString &content);
+    bool isURLAllowed(const QString& url) const;
 
 signals:
     void finishedParsing();
--- khtml/xml/dom_nodeimpl.cpp	14 Jan 2003 19:47:37 -0000	1.1.1.8
+++ khtml/xml/dom_nodeimpl.cpp	21 Jan 2003 04:59:57 -0000	1.1.1.9
@@ -457,6 +457,12 @@
 
 bool NodeImpl::dispatchEvent(EventImpl *evt, int &exceptioncode, bool tempEvent)
 {
+    // Not mentioned in spec: throw NOT_FOUND_ERR if evt is null
+    if (!evt) {
+        exceptioncode = DOMException::NOT_FOUND_ERR;
+        return false;
+    }
+
     evt->setTarget(this);
 
     // Since event handling code could cause this object to be deleted, grab a reference to the view now
@@ -522,7 +528,8 @@
                  !evt->defaultPrevented() && !evt->defaultHandled(); --it)
             it.current()->defaultEventHandler(evt);
 
-        if (evt->id() == EventImpl::CLICK_EVENT && !evt->defaultPrevented())
+        if (evt->id() == EventImpl::CLICK_EVENT && !evt->defaultPrevented() &&
+			static_cast<MouseEventImpl*>(evt)->button() == 0)
             dispatchUIEvent(EventImpl::DOMACTIVATE_EVENT, static_cast<UIEventImpl*>(evt)->detail());
     }
 
@@ -623,9 +630,9 @@
         default:
             break;
     }
-    bool ctrlKey = (_mouse->state() & Qt::ControlButton);
-    bool altKey = (_mouse->state() & Qt::AltButton);
-    bool shiftKey = (_mouse->state() & Qt::ShiftButton);
+    bool ctrlKey = (_mouse->state() & Qt::ControlButton) == Qt::ControlButton;
+    bool altKey = (_mouse->state() & Qt::AltButton) == Qt::AltButton;
+    bool shiftKey = (_mouse->state() & Qt::ShiftButton) == Qt::ShiftButton;
     bool metaKey = false; // ### qt support?
 
     EventImpl *evt = new MouseEventImpl(evtId,true,cancelable,getDocument()->defaultView(),
--- kimgio/Makefile.am	4 Jan 2003 04:31:10 -0000	1.1.1.5
+++ kimgio/Makefile.am	4 Jan 2003 04:43:31 -0000	1.8
@@ -7,7 +7,7 @@
 kde_module_LTLIBRARIES= kimg_eps.la kimg_xview.la kimg_krl.la \
 	         $(KIMGIO_TIFF_MODULES) kimg_ico.la
 
-KIMGIO_PLUGIN = -avoid-version -export-symbols-regex 'kimgio_.*_(read|write)'
+KIMGIO_PLUGIN = -avoid-version -export-symbols-regex 'kimgio_.*_(read|write)' $(all_libraries) $(LIB_KDECORE) $(LIB_QT)
 
 kimg_tiff_la_SOURCES = tiffr.cpp
 kimg_tiff_la_LDFLAGS = -module $(KIMGIO_PLUGIN) -no-undefined $(all_libraries)
--- kinit/Makefile.am	10 Jul 2002 04:49:13 -0000	1.1.1.2
+++ kinit/Makefile.am	26 Nov 2002 21:00:49 -0000	1.4
@@ -20,7 +20,7 @@
 SUBDIRS = . tests
 
 bin_PROGRAMS = kdeinit kdeinit_wrapper kshell kdeinit_shutdown lnusertemp kwrapper klauncher kioslave
-lib_LTLIBRARIES = klauncher.la
+lib_LTLIBRARIES = klauncher.la libklauncher_main.la
 
 kdeinit_SOURCES = kinit.cpp setproctitle.cpp
 
@@ -49,13 +49,20 @@
 kwrapper_SOURCES = kwrapper.c
 kwrapper_LDADD = $(LIBSOCKET)
 
-klauncher_la_LDFLAGS = $(all_libraries) -module -avoid-version
-klauncher_la_LIBADD = $(LIB_KIO)
-klauncher_la_SOURCES = klauncher.cpp klauncher_main.cpp autostart.cpp
+libklauncher_main_la_LDFLAGS = $(all_libraries) -avoid-version
+libklauncher_main_la_LIBADD = $(LIB_KIO)
+libklauncher_main_la_SOURCES = klauncher.cpp autostart.cpp
+
+klauncher_la_LDFLAGS = $(all_libraries) -avoid-version -module
+klauncher_la_LIBADD = $(LIB_KIO) libklauncher_main.la
+klauncher_la_SOURCES = klauncher_la_main.cpp
+
+klauncher_la_main.cpp: klauncher_main.cpp
+	cat klauncher_main.cpp > klauncher_la_main.cpp
 
 klauncher_LDFLAGS =  $(all_libraries) $(KDE_RPATH)
-klauncher_LDADD = klauncher.la
-klauncher_SOURCES = dummy.cpp
+klauncher_LDADD = libklauncher_main.la
+klauncher_SOURCES = klauncher_main.cpp
 
 METASOURCES = AUTO
 
@@ -64,6 +71,3 @@
 kwrapper.o: wrapper.c
 
 shell.o: wrapper.c
-
-dummy.cpp:
-	echo > dummy.cpp
--- kinit/kinit.cpp	14 Jan 2003 19:47:38 -0000	1.1.1.5
+++ kinit/kinit.cpp	14 Jan 2003 20:04:30 -0000	1.6
@@ -1430,6 +1430,7 @@
    kdeinit_setproctitle("Starting up...");
    kdeinit_library_path();
    unsetenv("LD_BIND_NOW");
+   unsetenv("DYLD_BIND_AT_LAUNCH");
    KApplication::loadedByKdeinit = true;
 
    d.maxname = strlen(argv[0]);
--- kinit/setproctitle.cpp	4 Jan 2003 04:31:11 -0000	1.1.1.3
+++ kinit/setproctitle.cpp	4 Jan 2003 04:43:31 -0000	1.4
@@ -72,6 +72,12 @@
 # define SPT_TYPE	SPT_REUSEARGV
 #endif
 
+#if defined(HAVE_NSGETENVIRON) && defined(HAVE_CRT_EXTERNS_H)
+#include <sys/time.h>
+#include <crt_externs.h>
+#define environ (*_NSGetEnviron())
+#endif
+
 #if SPT_TYPE != SPT_NONE && SPT_TYPE != SPT_BUILTIN
 
 # if SPT_TYPE == SPT_PSTAT
--- kio/Makefile.am	1 Oct 2002 03:06:46 -0000	1.1.1.4
+++ kio/Makefile.am	23 Nov 2002 03:50:36 -0000	1.7
@@ -20,7 +20,7 @@
 
 INCLUDES=  -I$(srcdir)/.. $(all_includes) $(SSL_INCLUDES)
 
-SUBDIRS = kssl kio bookmarks kfile . kpasswdserver misc pics tests
+SUBDIRS = kssl kio bookmarks kfile . kded kpasswdserver misc pics tests
 
 lib_LTLIBRARIES = libkio.la
 
RCS file: kio/kded/Makefile
diff -N kio/kded/Makefile
--- /dev/null	1 Jan 1970 00:00:00 -0000
+++ kio/kded/Makefile	23 Nov 2002 03:50:36 -0000	1.2
@@ -0,0 +1,8 @@
+# That's right!  OSX's linker is so anal, I had to totally
+# mess with the link order to get things to build.  Scary, scary.
+
+all:
+	pushd ../../kded; $(MAKE); popd
+
+clean:
+install:
--- kio/kfile/kurlrequesterdlg.cpp	5 Nov 2002 00:05:06 -0000	1.1.1.3
+++ kio/kfile/kurlrequesterdlg.cpp	21 Jan 2003 04:55:19 -0000	1.1.1.4
@@ -72,7 +72,9 @@
     urlRequester_->setFocus();
     connect( urlRequester_->lineEdit(), SIGNAL(textChanged(const QString&)),
              SLOT(slotTextChanged(const QString&)) );
-    enableButtonOK( !urlName.isEmpty() );
+    bool state = !urlName.isEmpty();
+    enableButtonOK( state );
+    enableButton( KDialogBase::User1, state );
     /*
     KFile::Mode mode = static_cast<KFile::Mode>( KFile::File |
             KFile::ExistingOnly );
@@ -83,7 +85,9 @@
 
 void KURLRequesterDlg::slotTextChanged(const QString & text)
 {
-    enableButtonOK( !text.isEmpty() );
+    bool state = !text.isEmpty();
+    enableButtonOK( state );
+    enableButton( KDialogBase::User1, state );
 }
 
 void KURLRequesterDlg::slotClear()
--- kio/kio/global.cpp	17 Jan 2003 14:30:48 -0000	1.1.1.6
+++ kio/kio/global.cpp	21 Jan 2003 04:55:01 -0000	1.1.1.7
@@ -1,7 +1,7 @@
 /* This file is part of the KDE libraries
    Copyright (C) 2000 David Faure <faure@kde.org>
 
-   $Id$
+   $Id$
 
    This library is free software; you can redistribute it and/or
    modify it under the terms of the GNU Library General Public
@@ -35,7 +35,6 @@
 #include "kio/job.h"
 
 #include <kdebug.h>
-#include <kapplication.h>
 #include <klocale.h>
 #include <kglobal.h>
 #include <kprotocolmanager.h>
@@ -1250,6 +1249,7 @@
 #define GETMNTENT(file, var) ((var = getmntent(file)) != 0)
 #define MOUNTPOINT(var) var->mnt_dir
 #define MOUNTTYPE(var) var->mnt_type
+#define HASMNTOPT(var, opt) hasmntopt(var, opt)
 #define FSNAME(var) var->mnt_fsname
 #elif defined(_AIX)
 /* we don't need this stuff */
@@ -1261,6 +1261,7 @@
 #define GETMNTENT(file, var) (getmntent(file, &var) == 0)
 #define MOUNTPOINT(var) var.mnt_mountp
 #define MOUNTTYPE(var) var.mnt_fstype
+#define HASMNTOPT(var, opt) hasmntopt(&var, opt)
 #define FSNAME(var) var.mnt_special
 #endif
 
@@ -1684,14 +1685,34 @@
         {
             mountPoint = QFile::decodeName( MOUNTPOINT(me) );
             check_mount_point(MOUNTTYPE(me), FSNAME(me), isautofs, isslow);
-
+kdDebug() << "***** point: " << mountPoint << ", " << FSNAME(me) << ", " << MOUNTPOINT(me) << endl;
             // we don't check if ismanual is Right, if /a/b is manually
             // mounted /a/b/c can't be automounted. At least IMO.
             if (ismanual == Unseen)
             {
-                struct fstab *ft = getfsfile(MOUNTPOINT(me));
-                if (!ft || strstr(ft->fs_mntops, "noauto"))
+                STRUCT_SETMNTENT fstab;
+                // TODO: #define FSTAB (FSTAB_FILE?), important for Solaris
+                if ((fstab = SETMNTENT("/etc/fstab", "r")) == 0)
+                    continue;
+
+                bool found = false;
+                STRUCT_MNTENT fe;
+                while (GETMNTENT(fstab, fe))
+                {
+                    if (!strcmp(FSNAME(me), FSNAME(fe)))
+                    {
+kdDebug() << "***** found: " << MOUNTPOINT(fe) << ", " << FSNAME(fe) <<endl;
+                        found = true;
+                        if (HASMNTOPT(fe, "noauto") || 
+                            !strcmp(MOUNTTYPE(fe), "supermount"))
+                            ismanual = Right;
+                        break;
+                    }
+                }
+                if (!found || !strcmp(MOUNTTYPE(me), "supermount"))
                   ismanual = Right;
+
+                ENDMNTENT(fstab);
             }
         }
     }
--- kio/kio/kurlcompletion.cpp	1 Oct 2002 03:06:59 -0000	1.1.1.3
+++ kio/kio/kurlcompletion.cpp	1 Oct 2002 03:59:13 -0000	1.4
@@ -740,6 +740,11 @@
 // Environment variables
 //
 
+#if defined(HAVE_NSGETENVIRON) && defined(HAVE_CRT_EXTERNS_H)   
+#include <crt_externs.h>                                 
+#define environ (*_NSGetEnviron()) 
+#endif
+
 extern char **environ; // Array of environment variables
 
 bool KURLCompletion::envCompletion(const MyURL &url, QString *match)
--- kio/kio/kurlcompletion.h	10 Jul 2002 04:49:47 -0000	1.1.1.2
+++ kio/kio/kurlcompletion.h	10 Jul 2002 05:12:46 -0000	1.3
@@ -23,6 +23,10 @@
 #ifndef KURLCOMPLETION_H
 #define KURLCOMPLETION_H
 
+#ifdef HAVE_CONFIG_H
+#include <config.h>
+#endif
+
 #include <kcompletion.h>
 #include <kio/jobclasses.h>
 #include <qstring.h>
--- kio/kpasswdserver/Makefile.am	1 Oct 2002 03:07:02 -0000	1.1.1.2
+++ kio/kpasswdserver/Makefile.am	26 Nov 2002 21:00:49 -0000	1.7
@@ -10,4 +10,4 @@
 kded_kpasswdserver_la_SOURCES = kpasswdserver.cpp kpasswdserver.skel 
 kded_kpasswdserver_la_METASOURCES = AUTO
 kded_kpasswdserver_la_LDFLAGS = $(all_libraries) -module -avoid-version
-kded_kpasswdserver_la_LIBADD = $(LIB_KIO)
+kded_kpasswdserver_la_LIBADD = $(LIB_KIO) $(top_builddir)/kded/libkded_main.la
--- kio/kssl/kopenssl.cc	4 Jan 2003 04:31:42 -0000	1.1.1.5
+++ kio/kssl/kopenssl.cc	4 Jan 2003 04:43:31 -0000	1.6
@@ -251,6 +251,9 @@
             #ifdef _AIX
             << "/opt/freeware/lib/"
 	    #endif
+	    #ifdef __APPLE__
+	    << "@PREFIX@/lib/"
+	    #endif
 	    << "/usr/lib/"
 	    << "/usr/ssl/lib/"
             << "/usr/local/lib/"
@@ -269,6 +272,7 @@
 	     #elif defined(__APPLE__)
 	     << "libssl.dylib"
 	     << "libssl.0.9.dylib"
+	     << "libssl.0.9.6.dylib"
              #else
              << "libssl.so"
 	     << "libssl.so.0"
@@ -286,6 +290,7 @@
 	     #elif defined(__APPLE__)
 	     << "libcrypto.dylib"
 	     << "libcrypto.0.9.dylib"
+	     << "libcrypto.0.9.6.dylib"
 	     #else
              << "libcrypto.so"
 	     << "libcrypto.so.0"
--- kio/misc/Makefile.am	4 Jan 2003 04:31:49 -0000	1.1.1.4
+++ kio/misc/Makefile.am	4 Jan 2003 04:43:32 -0000	1.5
@@ -22,25 +22,29 @@
 
 SUBDIRS = . ksendbugmail kpac kdesasl kssld kfile
 
-lib_LTLIBRARIES = kio_uiserver.la
+lib_LTLIBRARIES = kio_uiserver.la libkio_uiserver_main.la
 bin_PROGRAMS = kio_uiserver kmailservice ktelnetservice
 
-kio_uiserver_SOURCES = dummy.cpp
-kio_uiserver_LDADD = kio_uiserver.la
-kio_uiserver_LDFLAGS = $(KDE_RPATH) $(all_libraries)
+libkio_uiserver_main_la_SOURCES = uiserver.cpp uiserver.skel observer.stub
+libkio_uiserver_main_la_LIBADD = $(LIB_KIO)
+libkio_uiserver_main_la_LDFLAGS = $(all_libraries) -avoid-version
+
+kio_uiserver_la_SOURCES = kio_uiserver_la_main.cpp
+kio_uiserver_la_LIBADD = libkio_uiserver_main.la
+kio_uiserver_la_LDFLAGS = $(all_libraries) -avoid-version -module
+
+kio_uiserver_la_main.cpp: kio_uiserver_main.cpp
+	cat kio_uiserver_main.cpp > kio_uiserver_la_main.cpp
 
-kio_uiserver_la_SOURCES = uiserver.cpp uiserver.skel observer.stub
-kio_uiserver_la_LIBADD = $(LIB_KIO)
-kio_uiserver_la_LDFLAGS = $(all_libraries) -module -avoid-version
+kio_uiserver_SOURCES = kio_uiserver_main.cpp
+kio_uiserver_LDADD = libkio_uiserver_main.la
+kio_uiserver_LDFLAGS = $(KDE_RPATH) $(all_libraries)
 
 observer_DIR = $(top_srcdir)/kio/kio
 
 METASOURCES = AUTO
 
 kde_services_DATA = kio_uiserver.desktop
-
-dummy.cpp: uiserver.kidl
-	echo > dummy.cpp
 
 kmailservice_SOURCES = kmailservice.cpp
 kmailservice_LDADD = $(LIB_KDECORE)
RCS file: kio/misc/kio_uiserver_main.cpp
diff -N kio/misc/kio_uiserver_main.cpp
--- /dev/null	1 Jan 1970 00:00:00 -0000
+++ kio/misc/kio_uiserver_main.cpp	26 Nov 2002 21:00:50 -0000	1.1
@@ -0,0 +1,27 @@
+/*
+ *
+ * Copyright (c) 2001 Nick Hudson <skrll@netbsd.org>
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+ * AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN
+ * AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+extern "C" int kdemain(int, char* []);
+
+int main( int argc, char* argv[] )
+{
+        return kdemain(argc, argv);
+}
--- kio/misc/uiserver.cpp	1 Dec 2002 16:29:16 -0000	1.1.1.4
+++ kio/misc/uiserver.cpp	1 Dec 2002 16:52:51 -0000	1.3
@@ -1013,7 +1013,9 @@
 
 //------------------------------------------------------------
 
-int main(int argc, char **argv)
+extern "C" int kdemain(int, char**);
+
+int kdemain(int argc, char **argv)
 {
     KLocale::setMainCatalogue("kdelibs");
     //  GS 5/2001 - I changed the name to "KDE" to make it look better
--- kio/misc/kpac/dhcp.h	10 Jul 2002 04:50:03 -0000	1.1.1.2
+++ kio/misc/kpac/dhcp.h	10 Jul 2002 05:12:46 -0000	1.3
@@ -29,6 +29,10 @@
 
 #define DHCP_OPT_LEN 312
 
+#ifndef uint8_t
+#include <stdint.h>
+#endif
+
 struct dhcp_msg
 {
 #define DHCP_BOOTREQUEST 1
--- kio/misc/kpac/kpac_dhcp_helper.c	10 Jul 2002 04:50:04 -0000	1.1.1.2
+++ kio/misc/kpac/kpac_dhcp_helper.c	29 Oct 2002 14:19:18 -0000	1.2
@@ -19,6 +19,7 @@
 /* $Id$ */
 
 #include <config.h>
+#include <sys/types.h>
 #include <grp.h>
 
 #include <unistd.h>
--- kio/misc/kssld/Makefile.am	4 Jan 2003 04:31:50 -0000	1.1.1.3
+++ kio/misc/kssld/Makefile.am	4 Jan 2003 04:43:32 -0000	1.8
@@ -21,7 +21,7 @@
 kde_module_LTLIBRARIES = kded_kssld.la
 
 kded_kssld_la_LDFLAGS = $(all_libraries) -module -avoid-version
-kded_kssld_la_LIBADD = $(LIB_KIO)
+kded_kssld_la_LIBADD = $(LIB_KIO) $(top_builddir)/kded/libkded_main.la
 kded_kssld_la_SOURCES = kssld.cpp kssld.skel
 
 METASOURCES = AUTO
--- kioslave/http/Makefile.am	29 Oct 2002 02:38:44 -0000	1.1.1.4
+++ kioslave/http/Makefile.am	26 Nov 2002 21:00:50 -0000	1.4
@@ -9,7 +9,7 @@
 ####### Files
 
 bin_PROGRAMS = kio_http_cache_cleaner
-lib_LTLIBRARIES = kio_http_cache_cleaner.la
+lib_LTLIBRARIES = kio_http_cache_cleaner.la libkio_http_cache_cleaner_main.la
 kde_module_LTLIBRARIES = kio_http.la
 
 kio_http_la_SOURCES = http.cc httpfilter.cc
@@ -17,19 +17,26 @@
 kio_http_la_LIBADD = $(LIB_KIO) $(LIBZ)
 kio_http_la_LDFLAGS = $(all_libraries) -module $(KDE_PLUGIN)
 
-kio_http_cache_cleaner_SOURCES = dummy.cpp
-kio_http_cache_cleaner_LDADD   = kio_http_cache_cleaner.la
+kio_http_cache_cleaner_SOURCES = http_cache_cleaner_main.cpp
+kio_http_cache_cleaner_LDADD = libkio_http_cache_cleaner_main.la
 kio_http_cache_cleaner_LDFLAGS = $(KDE_RPATH)
 
-kio_http_cache_cleaner_la_SOURCES = http_cache_cleaner.cpp
-kio_http_cache_cleaner_la_LIBADD  = $(LIB_KIO)
-kio_http_cache_cleaner_la_LDFLAGS = -module -avoid-version
+http_cache_cleaner_main.cpp: stub_main.cpp
+	cat stub_main.cpp > http_cache_cleaner_main.cpp
+
+kio_http_cache_cleaner_la_SOURCES = http_cache_cleaner_la_main.cpp
+kio_http_cache_cleaner_la_LIBADD  = libkio_http_cache_cleaner_main.la
+kio_http_cache_cleaner_la_LDFLAGS = -avoid-version -module
+
+http_cache_cleaner_la_main.cpp: stub_main.cpp
+	cat stub_main.cpp > http_cache_cleaner_la_main.cpp
+
+libkio_http_cache_cleaner_main_la_SOURCES = http_cache_cleaner.cpp
+libkio_http_cache_cleaner_main_la_LIBADD  = $(LIB_KIO)
+libkio_http_cache_cleaner_main_la_LDFLAGS = -avoid-version
 
 noinst_HEADERS = http.h
 
 kdelnkdir = $(kde_servicesdir)
 kdelnk_DATA = http_cache_cleaner.desktop http.protocol https.protocol \
               webdav.protocol webdavs.protocol
-
-dummy.cpp:
-	echo > dummy.cpp
--- kioslave/http/http.cc	17 Jan 2003 14:31:04 -0000	1.1.1.12
+++ kioslave/http/http.cc	17 Jan 2003 14:42:13 -0000	1.11
@@ -34,9 +34,9 @@
 #include <netinet/in.h>  // Required for AIX
 #include <netinet/tcp.h>
 
+#include <unistd.h>
 /*
 #include <netdb.h>
-#include <unistd.h>
 #include <sys/time.h>
 #include <sys/wait.h>
 */
--- kioslave/http/http_cache_cleaner.cpp	25 Jun 2002 02:47:04 -0000	1.1.1.1
+++ kioslave/http/http_cache_cleaner.cpp	26 Nov 2002 21:00:50 -0000	1.2
@@ -187,7 +187,9 @@
    }
 }
 
-int main(int argc, char **argv)
+extern "C" int kdemain(int, char**);
+
+int kdemain(int argc, char **argv)
 {
    KLocale::setMainCatalogue("kdelibs");
    KCmdLineArgs::init( argc, argv, appName, description, version, true);
RCS file: kioslave/http/stub_main.cpp
diff -N kioslave/http/stub_main.cpp
--- /dev/null	1 Jan 1970 00:00:00 -0000
+++ kioslave/http/stub_main.cpp	26 Nov 2002 21:00:50 -0000	1.1
@@ -0,0 +1,28 @@
+/*
+ *
+ * Copyright (c) 2001 Nick Hudson <skrll@netbsd.org>
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+ * AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN
+ * AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+extern "C" int kdemain(int, char* []);
+
+int main( int argc, char* argv[] )
+{
+        return kdemain(argc, argv);
+}
+
--- kioslave/http/kcookiejar/Makefile.am	1 Oct 2002 03:07:20 -0000	1.1.1.3
+++ kioslave/http/kcookiejar/Makefile.am	26 Nov 2002 21:00:50 -0000	1.7
@@ -9,25 +9,25 @@
 lib_LTLIBRARIES = kcookiejar.la
 kde_module_LTLIBRARIES = kded_kcookiejar.la
 
-kcookiejar_la_SOURCES = main.cpp
 METASOURCES = AUTO
+kcookiejar_la_SOURCES = main.cpp
 kcookiejar_la_LDFLAGS = $(all_libraries) -module -avoid-version
 kcookiejar_la_LIBADD = $(LIB_KDECORE)
 
 kded_kcookiejar_la_SOURCES = kcookiejar.cpp kcookieserver.cpp \
 			kcookieserver.skel kcookiewin.cpp 
 kded_kcookiejar_la_LDFLAGS = $(all_libraries) -module -avoid-version
-kded_kcookiejar_la_LIBADD = $(LIB_KIO)
+kded_kcookiejar_la_LIBADD = $(LIB_KIO) $(top_builddir)/kded/libkded_main.la
 
-kcookiejar_SOURCES = dummy.cpp
-kcookiejar_LDADD = kcookiejar.la
+kcookiejar_SOURCES = kcookiejar_main.cpp
+kcookiejar_LDADD = $(LIB_KDECORE)
 kcookiejar_LDFLAGS = $(all_libraries) $(KDE_RPATH)
 
+kcookiejar_main.cpp: main.cpp
+	cat main.cpp > kcookiejar_main.cpp
+
 kded_DATA = kcookiejar.desktop
 kdeddir = $(kde_servicesdir)/kded
 
 update_DATA = kcookiescfg.upd
 updatedir = $(kde_datadir)/kconf_update
-
-dummy.cpp:
-	echo > dummy.cpp
--- libkscreensaver/Makefile.am	14 Aug 2002 14:04:21 -0000	1.1.1.3
+++ libkscreensaver/Makefile.am	23 Nov 2002 03:50:37 -0000	1.6
@@ -1,7 +1,7 @@
 lib_LTLIBRARIES = libkscreensaver.la
 libkscreensaver_la_SOURCES = main.cpp kscreensaver.cpp
 libkscreensaver_la_LIBADD  = $(LIB_KIO)
-libkscreensaver_la_LDFLAGS = $(all_libraries) $(KDE_RPATH) -version-info 5:0:1
+libkscreensaver_la_LDFLAGS = $(all_libraries) $(KDE_RPATH) -version-info 5:0:1 -undefined suppress -flat_namespace
 
 include_HEADERS = kscreensaver.h kscreensaver_vroot.h
 
--- libltdl/ltdl.c	10 Jul 2002 04:51:21 -0000	1.1.1.2
+++ libltdl/ltdl.c	26 Nov 2002 21:00:50 -0000	1.7
@@ -1544,11 +1544,14 @@
   /* try to open the old library first; if it was dlpreopened,
      we want the preopened version of it, even if a dlopenable
      module is available */
+/*	HACKALERT: Why? Mac OS/Darwin can't open static libs, it is
+	a waste to even try. Disabled for kde.	*/
+#if 0
   if (old_name && tryall_dlopen(handle, old_name) == 0)
     {
       return 0;
     }
-
+#endif
   /* try to open the dynamic library */
   if (dlname)
     {
--- libltdl/ltdl.m4	10 Jul 2002 04:51:21 -0000	1.1.1.2
+++ libltdl/ltdl.m4	22 Jan 2003 02:54:22 -0000	1.4
@@ -103,6 +103,9 @@
 	osf*)
 	  libltdl_cv_sys_dlopen_deplibs=yes
 	  ;;
+	darwin*)
+	  libltdl_cv_sys_dlopen_deplibs=yes
+	  ;;
 	esac
 ])
 if test "$libltdl_cv_sys_dlopen_deplibs" != yes; then
@@ -442,6 +445,8 @@
 if test x"$ac_cv_sys_symbol_underscore" = xyes; then
   if test x"$ac_cv_func_dlopen" = xyes ||
      test x"$ac_cv_lib_dl_dlopen" = xyes ; then
+        ac_save_LIBS="$LIBS"
+        LIBS="$LIBS $LIBADD_DL"
 	AC_CACHE_CHECK([whether we have to add an underscore for dlsym],
 		libltdl_cv_need_uscore, [dnl
 		AC_TRY_RUN([
@@ -495,6 +500,7 @@
 fi
 
 if test x"$libltdl_cv_need_uscore" = xyes; then
+  LIBS="$ac_save_LIBS"
   AC_DEFINE(NEED_USCORE, 1,
     [Define if dlsym() requires a leading underscode in symbol names. ])
 fi
RCS file: pics/crystalsvg/cr16-mime-Mime.png
diff -N pics/crystalsvg/cr16-mime-Mime.png
RCS file: pics/crystalsvg/cr16-mime-mime.png
diff -N pics/crystalsvg/cr16-mime-mime.png
Binary files /tmp/cvsbUICiQ and /dev/null differ
