diff -uNbr mozilla/.mozconfig mozilla-patched/.mozconfig
--- mozilla/.mozconfig	Wed Dec 31 19:00:00 1969
+++ mozilla-patched/.mozconfig	Sat Dec 14 01:34:10 2002
@@ -0,0 +1,15 @@
+ac_add_options --with-pthreads
+ac_add_options --with-system-jpeg=/sw
+ac_add_options --with-system-zlib=/sw
+ac_add_options --with-system-png=/sw
+ac_add_options --disable-tests
+ac_add_options --disable-debug
+ac_add_options --disable-short-wchar
+ac_add_options --enable-optimize
+ac_add_options --enable-strip
+ac_add_options --enable-crypto
+ac_add_options --enable-mathml
+ac_add_options --enable-extensions=all
+ac_add_options --with-default-mozilla-five-home=/sw/lib/mozilla
+mk_add_options MOZ_MAKE_FLAGS=-j4
+
diff -uNbr mozilla/build/unix/mozilla-config.in mozilla-patched/build/unix/mozilla-config.in
--- mozilla/build/unix/mozilla-config.in	Wed Apr 24 22:51:15 2002
+++ mozilla-patched/build/unix/mozilla-config.in	Fri Dec 13 21:25:03 2002
@@ -55,7 +55,7 @@
       echo_exec_prefix=yes
       ;;
     --version)
-      echo -1
+      echo "%MOZILLA_VERSION%"
       exit 0
       ;;
     --cflags)
diff -uNbr mozilla/config/autoconf.mk.in mozilla-patched/config/autoconf.mk.in
--- mozilla/config/autoconf.mk.in	Tue Oct 15 19:34:42 2002
+++ mozilla-patched/config/autoconf.mk.in	Fri Dec 13 21:25:03 2002
@@ -30,14 +30,14 @@
 prefix		= @prefix@
 exec_prefix	= @exec_prefix@
 bindir		= @bindir@
-includedir	= @includedir@/mozilla-$(MOZILLA_VERSION)
+includedir	= @includedir@/mozilla
 libdir		= @libdir@
 datadir		= @datadir@
 mandir		= @mandir@
-idldir		= @datadir@/idl/mozilla-$(MOZILLA_VERSION)
+idldir		= @datadir@/idl/mozilla
 
-mozappdir	= $(libdir)/mozilla-$(MOZILLA_VERSION)
-mredir		= $(libdir)/mre/mre-$(MOZILLA_VERSION)
+mozappdir	= $(libdir)/mozilla
+mredir		= $(libdir)/mre
 mrelibdir	= $(mredir)/lib
 
 DIST		= $(DEPTH)/dist
diff -uNbr mozilla/config/rules.mk mozilla-patched/config/rules.mk
--- mozilla/config/rules.mk	Fri Oct 25 01:59:03 2002
+++ mozilla-patched/config/rules.mk	Fri Dec 13 21:25:03 2002
@@ -447,7 +447,7 @@
 ifdef IS_COMPONENT
 EXTRA_DSO_LDOPTS	+= -bundle
 else
-EXTRA_DSO_LDOPTS	+= -dynamiclib -install_name @executable_path/\$@ -compatibility_version 1 -current_version 1
+EXTRA_DSO_LDOPTS	+= -dynamiclib -install_name /sw/lib/mozilla/\$@ -compatibility_version 1 -current_version 1
 endif
 endif
 
diff -uNbr mozilla/configure mozilla-patched/configure
--- mozilla/configure	Sat Nov 30 14:59:50 2002
+++ mozilla-patched/configure	Sat Dec 14 01:34:10 2002
@@ -4663,34 +4663,34 @@
 
 *-darwin*) 
     _PEDANTIC=
-    CFLAGS="$CFLAGS -fpascal-strings -traditional-cpp -fno-common"
-    CXXFLAGS="$CXXFLAGS -fpascal-strings -traditional-cpp -fno-common"
+    CFLAGS="$CFLAGS -no-cpp-precomp -fno-common"
+    CXXFLAGS="$CXXFLAGS -no-cpp-precomp -fno-common"
     DLL_SUFFIX=".dylib"
     DSO_LDOPTS=''
     STRIP="$STRIP -x -S"
     case "${target_cpu}" in
     powerpc*)
-        cat >> confdefs.h <<\EOF
-#define XP_MACOSX 1
-EOF
-
-        cat >> confdefs.h <<\EOF
-#define TARGET_CARBON 1
-EOF
-
-        cat >> confdefs.h <<\EOF
-#define TARGET_API_MAC_CARBON 1
-EOF
-
-        no_x=yes
+#        cat >> confdefs.h <<\EOF
+##define XP_MACOSX 1
+#EOF
+#
+#        cat >> confdefs.h <<\EOF
+##define TARGET_CARBON 1
+#EOF
+#
+#        cat >> confdefs.h <<\EOF
+##define TARGET_API_MAC_CARBON 1
+#EOF
+#
+#        no_x=yes
         MACOSX=1
-        _PLATFORM_DEFAULT_TOOLKIT='mac'
+#        _PLATFORM_DEFAULT_TOOLKIT='mac'
         DSO_LDOPTS="$DSO_LDOPTS -arch ppc"
-        TK_LIBS='-framework Carbon /System/Library/Frameworks/Carbon.framework/Carbon'
-        TK_CFLAGS="-I/Developer/Headers/FlatCarbon -F/System/Library/Frameworks"
-        CFLAGS="$CFLAGS $TK_CFLAGS"
-        CXXFLAGS="$CXXFLAGS $TK_CFLAGS"
-        MOZ_ENABLE_POSTSCRIPT=
+#        TK_LIBS='-framework Carbon /System/Library/Frameworks/Carbon.framework/Carbon'
+#        TK_CFLAGS="-I/Developer/Headers/FlatCarbon -F/System/Library/Frameworks"
+#        CFLAGS="$CFLAGS $TK_CFLAGS"
+#        CXXFLAGS="$CXXFLAGS $TK_CFLAGS"
+#        MOZ_ENABLE_POSTSCRIPT=
         ;;
     esac
     ;;
@@ -11028,7 +11028,7 @@
 #include "confdefs.h"
 
 #include <ft2build.h>
-#include FT_FREETYPE_H
+#include <freetype/freetype.h>
 #include <stdio.h>
 #include <stdlib.h>
 
@@ -14425,9 +14425,9 @@
 #define XP_UNIX 1
 EOF
 
-  cat >> confdefs.h <<\EOF
-#define UNIX_EMBED 1
-EOF
+#  cat >> confdefs.h <<\EOF
+##define UNIX_EMBED 1
+#EOF
 
   cat >> confdefs.h <<\EOF
 #define UNIX_LDAP 1
diff -uNbr mozilla/directory/c-sdk/config/config.mk mozilla-patched/directory/c-sdk/config/config.mk
--- mozilla/directory/c-sdk/config/config.mk	Tue Mar 26 16:51:57 2002
+++ mozilla-patched/directory/c-sdk/config/config.mk	Fri Dec 13 21:25:03 2002
@@ -60,10 +60,10 @@
 
 NFSPWD		= $(MOD_DEPTH)/config/nfspwd
 
-CFLAGS		= $(CC_ONLY_FLAGS) $(OPTIMIZER) $(OS_CFLAGS)\
-		  $(XP_DEFINE) $(DEFINES) $(INCLUDES) $(XCFLAGS)
-CCCFLAGS	= $(CCC_ONLY_FLAGS) $(OPTIMIZER) $(OS_CFLAGS)\
-		  $(XP_DEFINE) $(DEFINES) $(INCLUDES) $(XCFLAGS)
+CFLAGS		= $(CC_ONLY_FLAGS) $(OPTIMIZER) $(XP_DEFINE)\
+		  $(DEFINES) $(INCLUDES) $(XCFLAGS) $(OS_CFLAGS)
+CCCFLAGS	= $(CCC_ONLY_FLAGS) $(OPTIMIZER) $(XP_DEFINE)\
+		  $(DEFINES) $(INCLUDES) $(XCFLAGS) $(OS_CFLAGS)
 # For purify
 NOMD_CFLAGS	= $(CC_ONLY_FLAGS) $(OPTIMIZER) $(NOMD_OS_CFLAGS)\
 		  $(XP_DEFINE) $(DEFINES) $(INCLUDES) $(XCFLAGS)
diff -uNbr mozilla/directory/c-sdk/configure mozilla-patched/directory/c-sdk/configure
--- mozilla/directory/c-sdk/configure	Thu Aug 22 18:44:21 2002
+++ mozilla-patched/directory/c-sdk/configure	Fri Dec 13 21:25:03 2002
@@ -3483,7 +3483,7 @@
 
         CPU_ARCH=ppc
     fi
-    DSO_LDOPTS='-dynamiclib -compatibility_version 1 -current_version 1 -all_load -install_name @executable_path/$@'
+    DSO_LDOPTS='-dynamiclib -compatibility_version 1 -current_version 1 -all_load -install_name /sw/lib/mozilla/$@'
     # Use the standard preprocessor (cpp)
     CFLAGS="$CFLAGS -no-cpp-precomp"
     MKSHLIB='$(CC) -arch $(CPU_ARCH) $(DSO_LDOPTS) -o $@'
diff -uNbr mozilla/fink/chrome.d/default mozilla-patched/fink/chrome.d/default
--- mozilla/fink/chrome.d/default	Wed Dec 31 19:00:00 1969
+++ mozilla-patched/fink/chrome.d/default	Fri Dec 13 21:25:03 2002
@@ -0,0 +1,2 @@
+skin,install,select,modern/1.0
+locale,install,select,en-US
diff -uNbr mozilla/fink/mozilla mozilla-patched/fink/mozilla
--- mozilla/fink/mozilla	Wed Dec 31 19:00:00 1969
+++ mozilla-patched/fink/mozilla	Fri Dec 13 21:25:03 2002
@@ -0,0 +1,99 @@
+#!/bin/sh
+#
+# The contents of this file are subject to the Netscape Public
+# License Version 1.1 (the "License"); you may not use this file
+# except in compliance with the License. You may obtain a copy of
+# the License at http://www.mozilla.org/NPL/
+#
+# Software distributed under the License is distributed on an "AS
+# IS" basis, WITHOUT WARRANTY OF ANY KIND, either express or
+# implied. See the License for the specific language governing
+# rights and limitations under the License.
+#
+# The Original Code is mozilla.org code.
+#
+# The Initial Developer of the Original Code is Netscape
+# Communications Corporation.  Portions created by Netscape are
+# Copyright (C) 1998 Netscape Communications Corporation. All
+# Rights Reserved.
+#
+# Contributor(s): 
+#
+
+## 
+## Usage:
+##
+## $ mozilla
+##
+## This script is meant to run a mozilla program from the mozilla
+## rpm installation.
+##
+## The script will setup all the environment voodoo needed to make
+## mozilla work.
+
+##
+## Variables
+##
+MOZ_PROGRAM="/sw/lib/mozilla/mozilla-bin"
+
+##
+## Set MOZILLA_FIVE_HOME
+##
+MOZILLA_FIVE_HOME="/sw/lib/mozilla"
+export MOZILLA_FIVE_HOME
+
+##
+## Set DYLD_LIBRARY_PATH
+##
+DYLD_LIBRARY_PATH=/sw/lib/mozilla:$DYLD_LIBRARY_PATH
+export DYLD_LIBRARY_PATH
+
+# If there is no command line argument at all then try to open a new
+# window in an already running instance.
+
+if [ -z "$1" ]; then
+  $MOZ_PROGRAM -remote "openurl(about:blank,new-window)" 2>/dev/null 1>/dev/null
+  RETURN_VAL=$?
+  if [ "$RETURN_VAL" -eq "2" ]; then
+    $MOZ_PROGRAM ${1+"$@"}
+    RETURN_VAL=$?
+  fi
+  exit $RETURN_VAL
+fi
+
+# If there's a command line argument but it doesn't begin with a -
+# it's probably a url.  Try to send it to a running instance.
+
+USE_EXIST=0
+opt="$1"
+case "$opt" in
+  -*) ;;
+  *) USE_EXIST=1 ;;
+esac
+
+if [ "${USE_EXIST}" -eq "1" ]; then
+  # check to make sure that the command contains at least a :/ in it.
+  echo $opt | grep -e ':/' 2>/dev/null > /dev/null
+  RETURN_VAL=$?
+  if [ "$RETURN_VAL" -eq "1" ]; then
+    # does it begin with a / ?
+    echo $opt | grep -e '^/' 2>/dev/null > /dev/null
+    RETURN_VAL=$?
+    if [ "$RETURN_VAL" -eq "0" ]; then
+      opt="file:$opt"
+    elif [ -e `pwd`/$opt ]; then
+      opt="`pwd`/$opt"
+    else
+      opt="http://$opt"
+    fi
+  fi
+  $MOZ_PROGRAM -remote "openurl($opt,new-window)" 2>/dev/null >/dev/null
+  RETURN_VAL=$?
+  if [ "$RETURN_VAL" -eq "2" ]; then
+    $MOZ_PROGRAM ${1+"$@"}
+    RETURN_VAL=$?
+  fi
+  exit $RETURN_VAL
+fi
+
+$MOZ_PROGRAM ${1+"$@"}
diff -uNbr mozilla/fink/mozilla-address.desktop mozilla-patched/fink/mozilla-address.desktop
--- mozilla/fink/mozilla-address.desktop	Wed Dec 31 19:00:00 1969
+++ mozilla-patched/fink/mozilla-address.desktop	Fri Dec 13 21:25:03 2002
@@ -0,0 +1,8 @@
+[Desktop Entry]
+Name=Mozilla Addressbook
+Comment=Addressbook from Mozilla Mailnews
+Exec=mozilla -addressbook
+Icon=gnome-calendar.png
+Terminal=false
+MultipleArgs=false
+Type=Application
diff -uNbr mozilla/fink/mozilla-chatzilla.desktop mozilla-patched/fink/mozilla-chatzilla.desktop
--- mozilla/fink/mozilla-chatzilla.desktop	Wed Dec 31 19:00:00 1969
+++ mozilla-patched/fink/mozilla-chatzilla.desktop	Fri Dec 13 21:25:03 2002
@@ -0,0 +1,8 @@
+[Desktop Entry]
+Name=Chatzilla
+Comment=IRC client
+Exec=mozilla -chat
+Icon=gnome-irc.png
+Terminal=false
+MultipleArgs=false
+Type=Application
diff -uNbr mozilla/fink/mozilla-editor.desktop mozilla-patched/fink/mozilla-editor.desktop
--- mozilla/fink/mozilla-editor.desktop	Wed Dec 31 19:00:00 1969
+++ mozilla-patched/fink/mozilla-editor.desktop	Fri Dec 13 21:25:03 2002
@@ -0,0 +1,8 @@
+[Desktop Entry]
+Name=Mozilla Editor
+Comment=HTML Composer
+Exec=mozilla -edit
+Icon=gnome-word.png
+Terminal=false
+MultipleArgs=false
+Type=Application
diff -uNbr mozilla/fink/mozilla-mail-composer.desktop mozilla-patched/fink/mozilla-mail-composer.desktop
--- mozilla/fink/mozilla-mail-composer.desktop	Wed Dec 31 19:00:00 1969
+++ mozilla-patched/fink/mozilla-mail-composer.desktop	Fri Dec 13 21:25:03 2002
@@ -0,0 +1,8 @@
+[Desktop Entry]
+Name=Mozilla Mail Composer
+Comment=Mozilla Mail Composer
+Exec=mozilla -compose
+Icon=mozilla-mail-composer.gif
+Terminal=false
+MultipleArgs=false
+Type=Application
diff -uNbr mozilla/fink/mozilla-mail.desktop mozilla-patched/fink/mozilla-mail.desktop
--- mozilla/fink/mozilla-mail.desktop	Wed Dec 31 19:00:00 1969
+++ mozilla-patched/fink/mozilla-mail.desktop	Fri Dec 13 21:25:03 2002
@@ -0,0 +1,8 @@
+[Desktop Entry]
+Name=Mozilla Mail
+Comment=Mozilla Mail
+Exec=mozilla -mail
+Icon=mozilla-mail.gif
+Terminal=false
+MultipleArgs=false
+Type=Application
diff -uNbr mozilla/fink/mozilla-news.desktop mozilla-patched/fink/mozilla-news.desktop
--- mozilla/fink/mozilla-news.desktop	Wed Dec 31 19:00:00 1969
+++ mozilla-patched/fink/mozilla-news.desktop	Fri Dec 13 21:25:03 2002
@@ -0,0 +1,8 @@
+[Desktop Entry]
+Name=Mozilla News
+Comment=Mozilla Mail
+Exec=mozilla -news
+Icon=gnome-news.png
+Terminal=false
+MultipleArgs=false
+Type=Application
diff -uNbr mozilla/fink/mozilla.desktop mozilla-patched/fink/mozilla.desktop
--- mozilla/fink/mozilla.desktop	Wed Dec 31 19:00:00 1969
+++ mozilla-patched/fink/mozilla.desktop	Fri Dec 13 21:25:03 2002
@@ -0,0 +1,8 @@
+[Desktop Entry]
+Name=Mozilla
+Comment=Mozilla Web Browser
+Exec=mozilla 
+Icon=mozilla.png
+Terminal=false
+MultipleArgs=false
+Type=Application
diff -uNbr mozilla/fink/prefs.js mozilla-patched/fink/prefs.js
--- mozilla/fink/prefs.js	Wed Dec 31 19:00:00 1969
+++ mozilla-patched/fink/prefs.js	Fri Dec 13 21:25:03 2002
@@ -0,0 +1,17 @@
+// XIM style
+pref("xim.input_style", "over-the-spot");
+
+// TryeType
+pref("font.FreeType2.enable", true);
+pref("font.freetype2.shared-library", "/sw/lib/mozilla/libfreetype.6.so");
+pref("font.FreeType2.autohinted", false);
+pref("font.FreeType2.unhinted", false);
+pref("font.antialias.min",        10);
+pref("font.directory.truetype.1", "/usr/X11R6/lib/X11/fonts/TTF");
+pref("font.directory.truetype.2", "/sw/share/fonts/truetype");
+
+// AA with Bitmap scaling.
+pref("font.scale.aa_bitmap.enable", true);
+pref("font.scale.aa_bitmap.always", false);
+pref("font.scale.aa_bitmap.min", 6);
+
diff -uNbr mozilla/fink/update-mozilla-chrome mozilla-patched/fink/update-mozilla-chrome
--- mozilla/fink/update-mozilla-chrome	Wed Dec 31 19:00:00 1969
+++ mozilla-patched/fink/update-mozilla-chrome	Fri Dec 13 21:25:03 2002
@@ -0,0 +1,22 @@
+#!/bin/sh -e
+
+ulimit -s unlimited || true
+
+MOZILLA_FIVE_HOME=/sw/lib/mozilla
+CHROME_DIR=/sw/var/lib/mozilla/chrome.d
+
+TMPFILE=`tempfile`
+cat ${CHROME_DIR}/* >>${TMPFILE}
+mv ${TMPFILE} ${MOZILLA_FIVE_HOME}/chrome/installed-chrome.txt
+chmod 644 ${MOZILLA_FIVE_HOME}/chrome/installed-chrome.txt
+
+rm -rf ${MOZILLA_FIVE_HOME}/chrome/overlayinfo
+rm -f  ${MOZILLA_FIVE_HOME}/chrome/*.rdf
+rm -f  ${MOZILLA_FIVE_HOME}/component.reg
+rm -f  ${MOZILLA_FIVE_HOME}/components/*.dat
+
+sudo -H ${MOZILLA_FIVE_HOME}/regxpcom || true
+sudo -H ${MOZILLA_FIVE_HOME}/regchrome || true
+
+exit 0
+
diff -uNbr mozilla/gfx/src/gtk/Makefile.in mozilla-patched/gfx/src/gtk/Makefile.in
--- mozilla/gfx/src/gtk/Makefile.in	Mon Oct 14 23:58:05 2002
+++ mozilla-patched/gfx/src/gtk/Makefile.in	Fri Dec 13 21:25:03 2002
@@ -193,6 +193,8 @@
 ifdef MOZ_ENABLE_FREETYPE2
 DEFINES         += -DMOZ_ENABLE_FREETYPE2
 INCLUDES        += $(FT2_CFLAGS)
+CC		+= $(FT2_CFLAGS)
+CXX		+= $(FT2_CFLAGS)
 # due to runtime linking with PR_LoadLibrary
 # EXTRA_DSO_LDOPTS does not need $(FT2_LIBS)
 endif
diff -uNbr mozilla/modules/libpref/src/nsPrefService.cpp mozilla-patched/modules/libpref/src/nsPrefService.cpp
--- mozilla/modules/libpref/src/nsPrefService.cpp	Sat Sep  7 13:11:33 2002
+++ mozilla-patched/modules/libpref/src/nsPrefService.cpp	Fri Dec 13 21:25:03 2002
@@ -530,6 +530,7 @@
 #elif defined(XP_BEOS)
       "beos.js"
 #endif
+      , "fink.js"
   };
 
   rv = NS_GetSpecialDirectory(NS_APP_PREF_DEFAULTS_50_DIR, getter_AddRefs(defaultPrefDir));
diff -uNbr mozilla/nsprpub/configure mozilla-patched/nsprpub/configure
--- mozilla/nsprpub/configure	Thu Sep 26 08:46:21 2002
+++ mozilla-patched/nsprpub/configure	Fri Dec 13 21:25:03 2002
@@ -3278,7 +3278,7 @@
 
         CPU_ARCH=ppc
     fi
-    DSO_LDOPTS='-dynamiclib -compatibility_version 1 -current_version 1 -all_load -install_name @executable_path/$@'
+    DSO_LDOPTS='-dynamiclib -compatibility_version 1 -current_version 1 -all_load -install_name /sw/lib/mozilla/$@'
     # Use the standard preprocessor (cpp)
     CFLAGS="$CFLAGS -no-cpp-precomp"
     MKSHLIB='$(CC) -arch $(CPU_ARCH) $(DSO_LDOPTS) -o $@'
diff -uNbr mozilla/other-licenses/Xft/fontconfig/src/Makefile.in mozilla-patched/other-licenses/Xft/fontconfig/src/Makefile.in
--- mozilla/other-licenses/Xft/fontconfig/src/Makefile.in	Thu Apr 25 15:03:35 2002
+++ mozilla-patched/other-licenses/Xft/fontconfig/src/Makefile.in	Fri Dec 13 21:25:04 2002
@@ -46,3 +46,5 @@
 ifdef MOZ_XFT_SYSTEM_FREETYPE2
 CFLAGS		+= $(FT2_CFLAGS)
 endif
+
+CFLAGS := -I$(MOZ_BUILD_ROOT)/dist/include/expat-1.95 $(CFLAGS)
diff -uNbr mozilla/other-licenses/expat/xmlwf/Makefile.in mozilla-patched/other-licenses/expat/xmlwf/Makefile.in
--- mozilla/other-licenses/expat/xmlwf/Makefile.in	Mon Feb 25 01:53:11 2002
+++ mozilla-patched/other-licenses/expat/xmlwf/Makefile.in	Fri Dec 13 21:25:04 2002
@@ -23,7 +23,7 @@
 LIBDIR= ../lib/.libs
 INCDIR= ../lib
 
-LDFLAGS= @LDFLAGS@ -static
+LDFLAGS= @LDFLAGS@
 CFLAGS= @CFLAGS@ -I$(INCDIR)
 CC = @CC@
 
diff -uNbr mozilla/other-licenses/freetype/builds/unix/configure mozilla-patched/other-licenses/freetype/builds/unix/configure
--- mozilla/other-licenses/freetype/builds/unix/configure	Thu Apr 25 15:00:54 2002
+++ mozilla-patched/other-licenses/freetype/builds/unix/configure	Fri Dec 13 21:25:04 2002
@@ -5715,7 +5715,7 @@
     # FIXME: Relying on posixy $() will cause problems for
     #        cross-compilation, but unfortunately the echo tests do not
     #        yet detect zsh echo's removal of \ escapes.
-    archive_cmds='$nonopt $(test "x$module" = xyes && echo -bundle || echo -dynamiclib) $allow_undefined_flag -o $lib $libobjs $deplibs$linker_flags -install_name $rpath/$soname $verstring'
+    archive_cmds='$nonopt $(test .$module = .yes && echo -bundle || echo -dynamiclib) $allow_undefined_flag -o $lib $libobjs $deplibs$linker_flags $(test .$module != .yes && echo -install_name $rpath/$soname $verstring)'
     # We need to add '_' to the symbols in $export_symbols first
     #archive_expsym_cmds="$archive_cmds"' && strip -s $export_symbols'
     hardcode_direct=yes
diff -uNbr mozilla/other-licenses/freetype/builds/unix/unix-cc.in mozilla-patched/other-licenses/freetype/builds/unix/unix-cc.in
--- mozilla/other-licenses/freetype/builds/unix/unix-cc.in	Mon Feb 25 01:57:25 2002
+++ mozilla-patched/other-licenses/freetype/builds/unix/unix-cc.in	Fri Dec 13 21:25:04 2002
@@ -76,6 +76,6 @@
 # Library linking
 #
 LINK_LIBRARY = $(LIBTOOL) --mode=link $(CCraw) -o $@ $(OBJECTS_LIST) \
-                          -rpath $(libdir) -version-info $(version_info)
+                          -rpath $(libdir) -version-info $(version_info) -module
 
 # EOF
diff -uNbr mozilla/other-licenses/freetype/include/freetype/config/ftoption.h mozilla-patched/other-licenses/freetype/include/freetype/config/ftoption.h
--- mozilla/other-licenses/freetype/include/freetype/config/ftoption.h	Thu Apr 25 14:57:53 2002
+++ mozilla-patched/other-licenses/freetype/include/freetype/config/ftoption.h	Fri Dec 13 21:25:04 2002
@@ -311,7 +311,7 @@
   /* embedded bitmaps in all formats using the SFNT module (namely         */
   /* TrueType & OpenType).                                                 */
   /*                                                                       */
-#define TT_CONFIG_OPTION_EMBEDDED_BITMAPS
+#undef TT_CONFIG_OPTION_EMBEDDED_BITMAPS
 
 
   /*************************************************************************/
@@ -375,7 +375,7 @@
   /* By undefining this, you will only compile the code necessary to load  */
   /* TrueType glyphs without hinting.                                      */
   /*                                                                       */
-#undef TT_CONFIG_OPTION_BYTECODE_INTERPRETER
+#define TT_CONFIG_OPTION_BYTECODE_INTERPRETER
 
 
   /*************************************************************************/
diff -uNbr mozilla/security/coreconf/Darwin.mk mozilla-patched/security/coreconf/Darwin.mk
--- mozilla/security/coreconf/Darwin.mk	Tue Jun 11 20:21:19 2002
+++ mozilla-patched/security/coreconf/Darwin.mk	Fri Dec 13 21:25:04 2002
@@ -57,7 +57,7 @@
 # definitions so that the linker can catch multiply-defined symbols.
 # Also, common symbols are not allowed with Darwin dynamic libraries.
 
-OS_CFLAGS	= $(DSO_CFLAGS) $(OS_REL_CFLAGS) -Wmost -fpascal-strings -traditional-cpp -fno-common -pipe -DDARWIN -DHAVE_STRERROR -DHAVE_BSD_FLOCK
+OS_CFLAGS	= $(DSO_CFLAGS) $(OS_REL_CFLAGS) -Wmost -no-cpp-precomp -fno-common -pipe -DDARWIN -DHAVE_STRERROR -DHAVE_BSD_FLOCK
 
 ifdef BUILD_OPT
 OPTIMIZER	= -O2
@@ -66,7 +66,7 @@
 ARCH		= darwin
 
 # May override this with -bundle to create a loadable module.
-DSO_LDOPTS	= -dynamiclib -compatibility_version 1 -current_version 1 -install_name @executable_path/$(notdir $@)
+DSO_LDOPTS	= -dynamiclib -compatibility_version 1 -current_version 1 -install_name /sw/lib/mozilla/$(notdir $@)
 
 MKSHLIB		= $(CC) -arch $(CPU_ARCH) $(DSO_LDOPTS)
 DLL_SUFFIX	= dylib
diff -uNbr mozilla/security/nss/cmd/platlibs.mk mozilla-patched/security/nss/cmd/platlibs.mk
--- mozilla/security/nss/cmd/platlibs.mk	Wed Sep 18 19:25:27 2002
+++ mozilla-patched/security/nss/cmd/platlibs.mk	Fri Dec 13 21:25:04 2002
@@ -177,7 +177,7 @@
 endif
 
 ifeq ($(OS_ARCH), Darwin)
-EXTRA_SHARED_LIBS += -dylib_file @executable_path/libsoftokn3.dylib:$(DIST)/lib/libsoftokn3.dylib
+EXTRA_SHARED_LIBS += -dylib_file /sw/lib/mozilla/libsoftokn3.dylib:$(DIST)/lib/libsoftokn3.dylib
 endif
 
 # Use the Solaris ld, which knows where to find libsoftokn3.so.
diff -uNbr mozilla/security/nss/lib/smime/config.mk mozilla-patched/security/nss/lib/smime/config.mk
--- mozilla/security/nss/lib/smime/config.mk	Fri Feb 15 18:38:43 2002
+++ mozilla-patched/security/nss/lib/smime/config.mk	Fri Dec 13 21:25:04 2002
@@ -60,7 +60,7 @@
 	$(NULL)
 
 ifeq ($(OS_ARCH), Darwin)
-EXTRA_SHARED_LIBS += -dylib_file @executable_path/libsoftokn3.dylib:$(DIST)/lib/libsoftokn3.dylib
+EXTRA_SHARED_LIBS += -dylib_file /sw/lib/mozilla/libsoftokn3.dylib:$(DIST)/lib/libsoftokn3.dylib
 endif
 
 endif
diff -uNbr mozilla/security/nss/lib/ssl/config.mk mozilla-patched/security/nss/lib/ssl/config.mk
--- mozilla/security/nss/lib/ssl/config.mk	Thu Feb 21 23:23:22 2002
+++ mozilla-patched/security/nss/lib/ssl/config.mk	Fri Dec 13 21:25:04 2002
@@ -64,7 +64,7 @@
 endif
 
 ifeq ($(OS_ARCH), Darwin)
-EXTRA_SHARED_LIBS += -dylib_file @executable_path/libsoftokn3.dylib:$(DIST)/lib/libsoftokn3.dylib
+EXTRA_SHARED_LIBS += -dylib_file /sw/lib/mozilla/libsoftokn3.dylib:$(DIST)/lib/libsoftokn3.dylib
 endif
 
 endif
diff -uNbr mozilla/xft.pc mozilla-patched/xft.pc
--- mozilla/xft.pc	Wed Dec 31 19:00:00 1969
+++ mozilla-patched/xft.pc	Fri Dec 13 22:31:16 2002
@@ -0,0 +1,14 @@
+prefix=/usr/X11R6
+exec_prefix=/usr/X11R6/bin
+libdir=/usr/X11R6/lib
+includedir=/usr/X11R6/include
+freetypelibs=-L/usr/X11R6/lib -lfreetype
+freetypecflags=-I/usr/X11R6/include/freetype2
+xrenderlibs=-L/usr/X11R6/lib -lXrender
+xrendercflags=-I/usr/X11R6/include
+
+Name: Xft
+Description: X FreeType library
+Version: 1.1.0
+Libs: -L${libdir} -lXft ${freetypelibs} ${xrenderlibs}
+Cflags: -I${includedir} ${freetypecflags} ${xrendercflags}
diff -uNbr mozilla/xpfe/bootstrap/Makefile.in mozilla-patched/xpfe/bootstrap/Makefile.in
--- mozilla/xpfe/bootstrap/Makefile.in	Fri Nov  8 10:33:48 2002
+++ mozilla-patched/xpfe/bootstrap/Makefile.in	Fri Dec 13 21:25:04 2002
@@ -250,7 +250,7 @@
 endif
 
 ifeq ($(OS_ARCH),Darwin)
-EXTRA_DSO_LDOPTS := $(subst -dynamiclib -install_name @executable_path/\$@ -compatibility_version 1 -current_version 1,-execute,$(EXTRA_DSO_LDOPTS))
+EXTRA_DSO_LDOPTS := $(subst -dynamiclib -install_name /sw/lib/mozilla/\$@ -compatibility_version 1 -current_version 1,-execute,$(EXTRA_DSO_LDOPTS))
 
 ifdef BUILD_STATIC_LIBS
 LDFLAGS	+= -framework QuickTime $(TK_LIBS)
