diff -uNbr postgresql-7.3.1/configure postgresql-7.3.1-new/configure
--- postgresql-7.3.1/configure	Tue Dec 17 21:07:17 2002
+++ postgresql-7.3.1-new/configure	Mon Dec 30 23:55:09 2002
@@ -8330,8 +8330,8 @@
 
 if test "$with_pam" = yes ; then
   if test "${ac_cv_header_security_pam_appl_h+set}" = set; then
-  echo "$as_me:$LINENO: checking for security/pam_appl.h" >&5
-echo $ECHO_N "checking for security/pam_appl.h... $ECHO_C" >&6
+  echo "$as_me:$LINENO: checking for pam/pam_appl.h" >&5
+echo $ECHO_N "checking for pam/pam_appl.h... $ECHO_C" >&6
 if test "${ac_cv_header_security_pam_appl_h+set}" = set; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 fi
@@ -8339,13 +8339,13 @@
 echo "${ECHO_T}$ac_cv_header_security_pam_appl_h" >&6
 else
   # Is the header compilable?
-echo "$as_me:$LINENO: checking security/pam_appl.h usability" >&5
-echo $ECHO_N "checking security/pam_appl.h usability... $ECHO_C" >&6
+echo "$as_me:$LINENO: checking pam/pam_appl.h usability" >&5
+echo $ECHO_N "checking pam/pam_appl.h usability... $ECHO_C" >&6
 cat >conftest.$ac_ext <<_ACEOF
 #line $LINENO "configure"
 #include "confdefs.h"
 $ac_includes_default
-#include <security/pam_appl.h>
+#include <pam/pam_appl.h>
 _ACEOF
 rm -f conftest.$ac_objext
 if { (eval echo "$as_me:$LINENO: \"$ac_compile\"") >&5
@@ -8370,12 +8370,12 @@
 echo "${ECHO_T}$ac_header_compiler" >&6
 
 # Is the header present?
-echo "$as_me:$LINENO: checking security/pam_appl.h presence" >&5
-echo $ECHO_N "checking security/pam_appl.h presence... $ECHO_C" >&6
+echo "$as_me:$LINENO: checking pam/pam_appl.h presence" >&5
+echo $ECHO_N "checking pam/pam_appl.h presence... $ECHO_C" >&6
 cat >conftest.$ac_ext <<_ACEOF
 #line $LINENO "configure"
 #include "confdefs.h"
-#include <security/pam_appl.h>
+#include <pam/pam_appl.h>
 _ACEOF
 if { (eval echo "$as_me:$LINENO: \"$ac_cpp conftest.$ac_ext\"") >&5
   (eval $ac_cpp conftest.$ac_ext) 2>conftest.er1
@@ -8407,20 +8407,20 @@
 # So?  What about this header?
 case $ac_header_compiler:$ac_header_preproc in
   yes:no )
-    { echo "$as_me:$LINENO: WARNING: security/pam_appl.h: accepted by the compiler, rejected by the preprocessor!" >&5
-echo "$as_me: WARNING: security/pam_appl.h: accepted by the compiler, rejected by the preprocessor!" >&2;}
-    { echo "$as_me:$LINENO: WARNING: security/pam_appl.h: proceeding with the preprocessor's result" >&5
-echo "$as_me: WARNING: security/pam_appl.h: proceeding with the preprocessor's result" >&2;};;
+    { echo "$as_me:$LINENO: WARNING: pam/pam_appl.h: accepted by the compiler, rejected by the preprocessor!" >&5
+echo "$as_me: WARNING: pam/pam_appl.h: accepted by the compiler, rejected by the preprocessor!" >&2;}
+    { echo "$as_me:$LINENO: WARNING: pam/pam_appl.h: proceeding with the preprocessor's result" >&5
+echo "$as_me: WARNING: pam/pam_appl.h: proceeding with the preprocessor's result" >&2;};;
   no:yes )
-    { echo "$as_me:$LINENO: WARNING: security/pam_appl.h: present but cannot be compiled" >&5
-echo "$as_me: WARNING: security/pam_appl.h: present but cannot be compiled" >&2;}
-    { echo "$as_me:$LINENO: WARNING: security/pam_appl.h: check for missing prerequisite headers?" >&5
-echo "$as_me: WARNING: security/pam_appl.h: check for missing prerequisite headers?" >&2;}
-    { echo "$as_me:$LINENO: WARNING: security/pam_appl.h: proceeding with the preprocessor's result" >&5
-echo "$as_me: WARNING: security/pam_appl.h: proceeding with the preprocessor's result" >&2;};;
+    { echo "$as_me:$LINENO: WARNING: pam/pam_appl.h: present but cannot be compiled" >&5
+echo "$as_me: WARNING: pam/pam_appl.h: present but cannot be compiled" >&2;}
+    { echo "$as_me:$LINENO: WARNING: pam/pam_appl.h: check for missing prerequisite headers?" >&5
+echo "$as_me: WARNING: pam/pam_appl.h: check for missing prerequisite headers?" >&2;}
+    { echo "$as_me:$LINENO: WARNING: pam/pam_appl.h: proceeding with the preprocessor's result" >&5
+echo "$as_me: WARNING: pam/pam_appl.h: proceeding with the preprocessor's result" >&2;};;
 esac
-echo "$as_me:$LINENO: checking for security/pam_appl.h" >&5
-echo $ECHO_N "checking for security/pam_appl.h... $ECHO_C" >&6
+echo "$as_me:$LINENO: checking for pam/pam_appl.h" >&5
+echo $ECHO_N "checking for pam/pam_appl.h... $ECHO_C" >&6
 if test "${ac_cv_header_security_pam_appl_h+set}" = set; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
@@ -8433,8 +8433,8 @@
 if test $ac_cv_header_security_pam_appl_h = yes; then
   :
 else
-  { { echo "$as_me:$LINENO: error: header file <security/pam_appl.h> is required for PAM" >&5
-echo "$as_me: error: header file <security/pam_appl.h> is required for PAM" >&2;}
+  { { echo "$as_me:$LINENO: error: header file <pam/pam_appl.h> is required for PAM" >&5
+echo "$as_me: error: header file <pam/pam_appl.h> is required for PAM" >&2;}
    { (exit 1); exit 1; }; }
 fi
 
diff -uNbr postgresql-7.3.1/src/Makefile.shlib postgresql-7.3.1-new/src/Makefile.shlib
--- postgresql-7.3.1/src/Makefile.shlib	Wed Oct  9 12:21:54 2002
+++ postgresql-7.3.1-new/src/Makefile.shlib	Mon Dec 30 23:27:58 2002
@@ -81,8 +81,16 @@
 endif
 
 ifeq ($(PORTNAME), darwin)
-  shlib			:= lib$(NAME)$(DLSUFFIX).$(SO_MAJOR_VERSION).$(SO_MINOR_VERSION)
-  LINK.shared		= $(COMPILER) $(DARWIN_NAMESPACE_SPEC) -bundle -undefined suppress
+  shlib			:= lib$(NAME).$(SO_MAJOR_VERSION).$(SO_MINOR_VERSION)$(DLSUFFIX)
+  ifneq ($(SO_MAJOR_VERSION), 0)
+    version_link	:= -compatibility_version $(SO_MAJOR_VERSION) -current_version $(SO_MAJOR_VERSION).$(SO_MINOR_VERSION)
+  endif
+  ifeq ($(DLSUFFIX), .so)
+    LINK.shared		= $(COMPILER) $(DARWIN_NAMESPACE_SPEC) -bundle -multiply_defined suppress
+  else
+    LINK.shared		= $(COMPILER) $(DARWIN_NAMESPACE_SPEC) -install_name $(libdir)/lib$(NAME).$(SO_MAJOR_VERSION)$(DLSUFFIX) $(version_link) -dynamiclib -multiply_defined suppress
+  endif
+  SHLIB_LINK		:= -L$(libpq_builddir) $(SHLIB_LINK)
 endif
 
 ifeq ($(PORTNAME), openbsd)
@@ -252,9 +260,9 @@
 $(shlib): $(OBJS)
 	$(LINK.shared) $(OBJS) $(SHLIB_LINK) -o $@
 # If we're using major and minor versions, then make a symlink to major-version-only.
-ifneq ($(shlib), lib$(NAME)$(DLSUFFIX).$(SO_MAJOR_VERSION))
-	rm -f lib$(NAME)$(DLSUFFIX).$(SO_MAJOR_VERSION)
-	$(LN_S) $(shlib) lib$(NAME)$(DLSUFFIX).$(SO_MAJOR_VERSION)
+ifneq ($(shlib), lib$(NAME).$(SO_MAJOR_VERSION)$(DLSUFFIX))
+	rm -f lib$(NAME).$(SO_MAJOR_VERSION)$(DLSUFFIX)
+	$(LN_S) $(shlib) lib$(NAME).$(SO_MAJOR_VERSION)$(DLSUFFIX)
 endif
 # Make sure we have a link to a name without any version numbers
 ifneq ($(shlib), lib$(NAME)$(DLSUFFIX))
@@ -310,10 +318,10 @@
 install-lib-shared: $(shlib)
 	$(INSTALL_SHLIB) $< $(DESTDIR)$(libdir)/$(shlib)
 ifneq ($(PORTNAME), win)
-ifneq ($(shlib), lib$(NAME)$(DLSUFFIX).$(SO_MAJOR_VERSION))
+ifneq ($(shlib), lib$(NAME).$(SO_MAJOR_VERSION)$(DLSUFFIX))
 	cd $(DESTDIR)$(libdir) && \
-	rm -f lib$(NAME)$(DLSUFFIX).$(SO_MAJOR_VERSION) && \
-	$(LN_S) $(shlib) lib$(NAME)$(DLSUFFIX).$(SO_MAJOR_VERSION)
+	rm -f lib$(NAME).$(SO_MAJOR_VERSION)$(DLSUFFIX) && \
+	$(LN_S) $(shlib) lib$(NAME).$(SO_MAJOR_VERSION)$(DLSUFFIX)
 endif
 ifneq ($(shlib), lib$(NAME)$(DLSUFFIX))
 	cd $(DESTDIR)$(libdir) && \
diff -uNbr postgresql-7.3.1/src/backend/libpq/auth.c postgresql-7.3.1-new/src/backend/libpq/auth.c
--- postgresql-7.3.1/src/backend/libpq/auth.c	Wed Sep  4 19:31:34 2002
+++ postgresql-7.3.1-new/src/backend/libpq/auth.c	Mon Dec 30 23:55:30 2002
@@ -42,7 +42,7 @@
 char	   *pg_krb_server_keyfile;
 
 #ifdef USE_PAM
-#include <security/pam_appl.h>
+#include <pam/pam_appl.h>
 
 #define PGSQL_PAM_SERVICE "postgresql"	/* Service name passed to PAM */
 
diff -uNbr postgresql-7.3.1/src/backend/utils/mb/conversion_procs/proc.mk postgresql-7.3.1-new/src/backend/utils/mb/conversion_procs/proc.mk
--- postgresql-7.3.1/src/backend/utils/mb/conversion_procs/proc.mk	Thu Sep  5 14:28:46 2002
+++ postgresql-7.3.1-new/src/backend/utils/mb/conversion_procs/proc.mk	Mon Dec 30 23:27:58 2002
@@ -1,7 +1,8 @@
 SRCS		+= $(NAME).c
 OBJS		+= $(NAME).o
 
-SHLIB_LINK 	:= $(BE_DLLLIBS)
+override DLSUFFIX = .so
+SHLIB_LINK 	:= $(BE_DLLLIBS) -undefined suppress -flat_namespace
 
 SO_MAJOR_VERSION := 0
 SO_MINOR_VERSION := 0
diff -uNbr postgresql-7.3.1/src/backend/utils/misc/guc.c postgresql-7.3.1-new/src/backend/utils/misc/guc.c
--- postgresql-7.3.1/src/backend/utils/misc/guc.c	Mon Dec  2 00:21:02 2002
+++ postgresql-7.3.1-new/src/backend/utils/misc/guc.c	Mon Dec 30 23:27:58 2002
@@ -834,12 +834,12 @@
 
 	{
 		{"unix_socket_group", PGC_POSTMASTER}, &Unix_socket_group,
-		"", NULL, NULL
+		"pgsql", NULL, NULL
 	},
 
 	{
 		{"unix_socket_directory", PGC_POSTMASTER}, &UnixSocketDir,
-		"", NULL, NULL
+		"/tmp", NULL, NULL
 	},
 
 	{
diff -uNbr postgresql-7.3.1/src/backend/utils/misc/postgresql.conf.sample postgresql-7.3.1-new/src/backend/utils/misc/postgresql.conf.sample
--- postgresql-7.3.1/src/backend/utils/misc/postgresql.conf.sample	Mon Sep  2 01:42:54 2002
+++ postgresql-7.3.1-new/src/backend/utils/misc/postgresql.conf.sample	Mon Dec 30 23:27:58 2002
@@ -27,18 +27,18 @@
 #
 #	Connection Parameters
 #
-#tcpip_socket = false
-#ssl = false
+tcpip_socket = true
+ssl = false
 
-#max_connections = 32
+max_connections = 32
 #superuser_reserved_connections = 2
 
 #port = 5432 
 #hostname_lookup = false
 #show_source_port = false
 
-#unix_socket_directory = ''
-#unix_socket_group = ''
+unix_socket_directory = '/tmp'
+unix_socket_group = 'pgsql'
 #unix_socket_permissions = 0777	# octal
 
 #virtual_host = ''
diff -uNbr postgresql-7.3.1/src/bin/initdb/initdb.sh postgresql-7.3.1-new/src/bin/initdb/initdb.sh
--- postgresql-7.3.1/src/bin/initdb/initdb.sh	Mon Nov 25 16:42:11 2002
+++ postgresql-7.3.1-new/src/bin/initdb/initdb.sh	Mon Dec 30 23:27:58 2002
@@ -1091,11 +1091,8 @@
 echo
 echo "Success. You can now start the database server using:"
 echo ""
-echo "    $PGPATH/postmaster -D $PGDATA"
-echo "or"
-# (Advertise -l option here, otherwise we have a background
-#  process writing to the terminal.)
-echo "    $PGPATH/pg_ctl -D $PGDATA -l logfile start"
-echo
+echo "    sudo -u root sudo -u pgsql $PGPATH/pg_ctl \\"
+echo "        -D $PGDATA -l /sw/var/log/postgresql/logfile start"
+echo ""
 
 exit 0
diff -uNbr postgresql-7.3.1/src/bin/pg_ctl/pg_ctl.sh postgresql-7.3.1-new/src/bin/pg_ctl/pg_ctl.sh
--- postgresql-7.3.1/src/bin/pg_ctl/pg_ctl.sh	Fri Oct 18 18:05:35 2002
+++ postgresql-7.3.1-new/src/bin/pg_ctl/pg_ctl.sh	Mon Dec 30 23:27:58 2002
@@ -14,6 +14,10 @@
 
 CMDNAME=`basename $0`
 
+ulimit -d 24000 >/dev/null 2>&1
+ulimit -n 512   >/dev/null 2>&1
+ulimit -s 4096  >/dev/null 2>&1
+
 help="\
 $CMDNAME is a utility to start, stop, restart, reload configuration files,
 or report the status of a PostgreSQL server.
diff -uNbr postgresql-7.3.1/src/interfaces/ecpg/lib/Makefile postgresql-7.3.1-new/src/interfaces/ecpg/lib/Makefile
--- postgresql-7.3.1/src/interfaces/ecpg/lib/Makefile	Tue Dec 10 23:08:04 2002
+++ postgresql-7.3.1-new/src/interfaces/ecpg/lib/Makefile	Mon Dec 30 23:27:58 2002
@@ -17,7 +17,7 @@
 SO_MINOR_VERSION= 4.1
 
 override CPPFLAGS := -I$(top_srcdir)/src/interfaces/ecpg/include -I$(libpq_srcdir) $(CPPFLAGS)
-
+override LDFLAGS  := -L$(libpq_builddir) $(LDFLAGS)
 
 OBJS= execute.o typename.o descriptor.o data.o error.o prepare.o memory.o \
 	connect.o misc.o
diff -uNbr postgresql-7.3.1/src/interfaces/libpgtcl/Makefile postgresql-7.3.1-new/src/interfaces/libpgtcl/Makefile
--- postgresql-7.3.1/src/interfaces/libpgtcl/Makefile	Tue Dec 10 23:08:05 2002
+++ postgresql-7.3.1-new/src/interfaces/libpgtcl/Makefile	Mon Dec 30 23:27:58 2002
@@ -17,10 +17,11 @@
 SO_MINOR_VERSION= 3
 
 override CPPFLAGS := -I$(libpq_srcdir) $(CPPFLAGS)
+override LDFLAGS  := -L$(libpq_builddir) $(LDFLAGS)
 
 OBJS= pgtcl.o pgtclCmds.o pgtclId.o
 
-SHLIB_LINK = $(libpq)
+SHLIB_LINK = $(libpq) -ltcl
 
 # If crypt is a separate library, rather than part of libc, it may need
 # to be referenced separately to keep (broken) linkers happy.  (This is
diff -uNbr postgresql-7.3.1/src/interfaces/python/GNUmakefile postgresql-7.3.1-new/src/interfaces/python/GNUmakefile
--- postgresql-7.3.1/src/interfaces/python/GNUmakefile	Thu Dec 13 13:39:04 2001
+++ postgresql-7.3.1-new/src/interfaces/python/GNUmakefile	Mon Dec 30 23:27:58 2002
@@ -17,7 +17,9 @@
 
 include $(top_srcdir)/src/Makefile.shlib
 
+override DLSUFFIX := .so
 override CPPFLAGS := -I$(libpq_srcdir) $(CPPFLAGS) $(python_includespec)
+override SHLIB_LINK := -undefined suppress -flat_namespace -L$(libpq_builddir) $(SHLIB_LINK)
 
 all: all-lib
 
diff -uNbr postgresql-7.3.1/src/makefiles/Makefile.darwin postgresql-7.3.1-new/src/makefiles/Makefile.darwin
--- postgresql-7.3.1/src/makefiles/Makefile.darwin	Fri Oct  5 17:15:38 2001
+++ postgresql-7.3.1-new/src/makefiles/Makefile.darwin	Mon Dec 30 23:27:58 2002
@@ -1,13 +1,16 @@
 AROPT = cr
 AWK= awk
  
-DLSUFFIX = .so
+DLSUFFIX = .dylib
 CFLAGS_SL =
-ifeq (,$(filter $(host_os), darwin1.0 darwin1.1 darwin1.2))
-DARWIN_NAMESPACE_SPEC = -flat_namespace
-endif
+CFLAGS += -fno-common
+#ifeq (,$(filter $(host_os), darwin1.0 darwin1.1 darwin1.2))
+#DARWIN_NAMESPACE_SPEC = -flat_namespace
+#endif
  
 %.so: %.o
-	$(CC) $(DARWIN_NAMESPACE_SPEC) -bundle -undefined suppress -o $@ $<
+	$(CC) $(DARWIN_NAMESPACE_SPEC) -bundle -multiply_defined suppress -o $@ $<
+%.dylib: %.o
+	$(CC) $(DARWIN_NAMESPACE_SPEC) -dynamiclib -multiply_defined suppress -o $@ $<
 
 sqlmansect = 7
diff -uNbr postgresql-7.3.1/src/pl/plperl/GNUmakefile postgresql-7.3.1-new/src/pl/plperl/GNUmakefile
--- postgresql-7.3.1/src/pl/plperl/GNUmakefile	Tue May 28 12:57:53 2002
+++ postgresql-7.3.1-new/src/pl/plperl/GNUmakefile	Mon Dec 30 23:27:58 2002
@@ -19,14 +19,14 @@
 endif
 
 override CPPFLAGS := -I$(srcdir) -I$(perl_archlibexp)/CORE $(CPPFLAGS)
-
+override DLSUFFIX = .so
 
 NAME = plperl
 SO_MAJOR_VERSION = 0
 SO_MINOR_VERSION = 0
 
 OBJS = plperl.o eloglvl.o SPI.o
-SHLIB_LINK = $(perl_embed_ldflags) $(BE_DLLLIBS)
+SHLIB_LINK = $(perl_embed_ldflags) $(BE_DLLLIBS) -undefined suppress -flat_namespace
 
 include $(top_srcdir)/src/Makefile.shlib
 
diff -uNbr postgresql-7.3.1/src/pl/plpgsql/src/Makefile postgresql-7.3.1-new/src/pl/plpgsql/src/Makefile
--- postgresql-7.3.1/src/pl/plpgsql/src/Makefile	Fri Nov  1 17:52:34 2002
+++ postgresql-7.3.1-new/src/pl/plpgsql/src/Makefile	Mon Dec 30 23:27:58 2002
@@ -19,11 +19,12 @@
 SO_MAJOR_VERSION= 1
 SO_MINOR_VERSION= 0
 
+override DLSUFFIX = .so
 override CPPFLAGS := -I$(srcdir) $(CPPFLAGS)
-SHLIB_LINK = $(BE_DLLLIBS)
+SHLIB_LINK = $(BE_DLLLIBS) -undefined suppress -flat_namespace
 rpath :=
 
-OBJS = pl_gram.o pl_handler.o pl_comp.o pl_exec.o pl_funcs.o
+OBJS = pl_gram.o pl_handler.o pl_comp.o pl_exec.o pl_funcs.o $(top_builddir)/src/backend/executor/SUBSYS.o
 
 ifneq ($(PORTNAME), qnx4)
 all: all-lib
diff -uNbr postgresql-7.3.1/src/pl/plpython/Makefile postgresql-7.3.1-new/src/pl/plpython/Makefile
--- postgresql-7.3.1/src/pl/plpython/Makefile	Thu Sep  5 14:28:46 2002
+++ postgresql-7.3.1-new/src/pl/plpython/Makefile	Mon Dec 30 23:27:58 2002
@@ -8,14 +8,13 @@
 # On some platforms we can only build PL/Python if libpython is a
 # shared library.  Since there is no official way to determine this,
 # we see if there is a file that is named like a shared library.
-ifneq (,$(wildcard $(python_configdir)/libpython*$(DLSUFFIX)*))
+ifneq (,$(wildcard $(python_configdir)/libpython*.dylib*))
 shared_libpython = yes
 endif
 
 # If we don't have a shared library and the platform doesn't allow it
 # to work without, we have to skip it.
 ifneq (,$(findstring yes, $(shared_libpython)$(allow_nonpic_in_shlib)))
-
 override CPPFLAGS := -I$(srcdir) $(python_includespec) $(CPPFLAGS)
 rpath :=
 
@@ -24,7 +23,8 @@
 SO_MINOR_VERSION = 0
 OBJS = plpython.o
 
-SHLIB_LINK = $(BE_DLLLIBS) $(python_libspec)
+SHLIB_LINK = $(BE_DLLLIBS) $(python_libspec) -undefined suppress -flat_namespace
+override DLSUFFIX = .so
 
 include $(top_srcdir)/src/Makefile.shlib
 
diff -uNbr postgresql-7.3.1/src/pl/tcl/Makefile postgresql-7.3.1-new/src/pl/tcl/Makefile
--- postgresql-7.3.1/src/pl/tcl/Makefile	Fri May 24 14:10:17 2002
+++ postgresql-7.3.1-new/src/pl/tcl/Makefile	Mon Dec 30 23:30:56 2002
@@ -25,6 +25,7 @@
 endif
 endif
 
+override DLSUFFIX = .so
 
 # The following attempts to figure out what libraries need to be
 # linked with pltcl.  The information comes from the tclConfig.sh
@@ -32,7 +33,7 @@
 
 ifneq ($(TCL_SHLIB_LD_LIBS),)
 # link command for a shared lib must mention shared libs it uses
-SHLIB_LINK = $(TCL_LIB_SPEC) $(TCL_LIBS) -lc
+SHLIB_LINK = $(TCL_LIB_SPEC) $(TCL_LIBS) -lc -undefined suppress -flat_namespace
 else
 ifeq ($(PORTNAME), hpux)
 # link command for a shared lib must mention shared libs it uses,
