diff -Naur unzip-5.42/fileio.c unzip-new/fileio.c
--- unzip-5.42/fileio.c	Sat Jan 13 13:40:12 2001
+++ unzip-new/fileio.c	Thu May 16 22:29:22 2002
@@ -53,7 +53,9 @@
 
 #define __FILEIO_C      /* identifies this source module */
 #define UNZIP_INTERNAL
+#include <fcntl.h>
 #include "unzip.h"
+#include "unzpriv.h"
 #ifdef WINDLL
 #  include "windll/windll.h"
 #  include <setjmp.h>
diff -Naur unzip-5.42/globals.h unzip-new/globals.h
--- unzip-5.42/globals.h	Sat Jan 13 13:39:58 2001
+++ unzip-new/globals.h	Thu May 16 22:17:38 2002
@@ -135,6 +135,8 @@
 #ifndef __globals_h
 #define __globals_h
 
+#include <sys/types.h>
+#include <sys/stat.h>
 #ifdef USE_ZLIB
 #  include "zlib.h"
 #endif
diff -Naur unzip-5.42/unix/Makefile unzip-new/unix/Makefile
--- unzip-5.42/unix/Makefile	Thu Apr 27 17:23:26 2000
+++ unzip-new/unix/Makefile	Thu May 16 22:41:07 2002
@@ -47,9 +47,9 @@
 AS = as
 LOC = $(LOCAL_UNZIP)
 AF = $(LOC)
-CF = -O -I. $(LOC)
+CF = -O -I. $(LOC) -DUNIX
 LF = -o unzip
-LF2 = -s
+LF2 = 
 
 # UnZipSFX flags
 SL = -o unzipsfx
diff -Naur unzip-5.42/unzpriv.h unzip-new/unzpriv.h
--- unzip-5.42/unzpriv.h	Wed Jan 10 14:53:56 2001
+++ unzip-new/unzpriv.h	Thu May 16 22:28:28 2002
@@ -21,6 +21,7 @@
 
 #ifndef __unzpriv_h   /* prevent multiple inclusions */
 #define __unzpriv_h
+#include <sys/stat.h>
 
 /* First thing: Signal all following code that we compile UnZip utilities! */
 #ifndef UNZIP
