diff -uNbr boson-all-0.8/boson/Makefile.am boson-all-0.8-patched/boson/Makefile.am
--- boson-all-0.8/boson/Makefile.am	Mon Mar 24 10:09:47 2003
+++ boson-all-0.8-patched/boson/Makefile.am	Sun Apr 27 11:54:42 2003
@@ -98,7 +98,7 @@
 		$(top_builddir)/boson/bosonwidgets/libbosonwidgets.la \
 		-lkdegames -lqtmcop -lsoundserver_idl -lartskde \
 		$(GLLIB) -l3ds \
-		$(LIB_KDEUI)
+		$(LIB_KDEUI) -lXmu
 boson_SOURCES = \
 		$(BOSON_SRC)\
 		main.cpp
@@ -114,7 +114,7 @@
 		$(top_builddir)/boson/bosonwidgets/libbosonwidgets.la \
 		-lkdegames -lqtmcop -lsoundserver_idl -lartskde \
 		$(GLLIB) -l3ds \
-		-lkio $(LIB_KDEUI)
+		-lkio $(LIB_KDEUI) -lXmu
 bounit_SOURCES = \
 		$(BOSON_SRC) \
 		bounitmain.cpp bosonsearchpathswidget.ui \
@@ -130,7 +130,7 @@
 bocursor_LDFLAGS = $(KDE_RPATH) $(all_libraries)
 bocursor_LDADD = \
 		$(top_builddir)/boson/bodebug/libbodebug.la \
-		-lkio $(LIB_KDEUI) $(GLLIB)
+		-lkio $(LIB_KDEUI) $(GLLIB) -lXmu
 bocursor_SOURCES = \
 		$(COMMON_SRC) \
 		bocursormain.cpp bosoncursoreditor.cpp bosoncursor.cpp \
@@ -148,7 +148,7 @@
 		$(top_builddir)/boson/bosonwidgets/libbosonwidgets.la \
 		-lkdegames -lqtmcop -lsoundserver_idl -lartskde \
 		$(GLLIB) -l3ds \
-		$(LIB_KDEUI)
+		$(LIB_KDEUI) -lXmu
 borender_SOURCES = \
 		$(BOSON_SRC) \
 		kgamemodeldebug.cpp \
@@ -159,7 +159,7 @@
 		$(top_builddir)/boson/bodebug/libbodebug.la \
 		$(top_builddir)/boson/info/libbosoninfo.la \
 		$(GLLIB) \
-		-lkio $(LIB_KDEUI)
+		-lkio $(LIB_KDEUI) -lXmu
 boinfo_SOURCES = \
 		bosonglwidget.cpp \
 		boinfomain.cpp
