diff -uNr gemrb-0.2.6/configure gemrb-0.2.6-patched/configure
--- gemrb-0.2.6/configure	2005-12-06 01:22:46.000000000 -0500
+++ gemrb-0.2.6-patched/configure	2006-03-01 00:12:53.000000000 -0500
@@ -24791,7 +24791,7 @@
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
   ac_check_lib_save_LIBS=$LIBS
-LIBS="-lopenal $LIBPTHREAD $LIBS"
+LIBS="-Wl,-framework,OpenAL $LIBPTHREAD $LIBS"
 cat >conftest.$ac_ext <<_ACEOF
 /* confdefs.h.  */
 _ACEOF
@@ -24934,7 +24934,7 @@
    { (exit 1); exit 1; }; }
 		fi
 	else
-		OPENAL_LIBS="-lopenal"
+		OPENAL_LIBS="-Wl,-framework,OpenAL"
 	fi
 
 
@@ -25113,10 +25113,7 @@
 echo $ECHO_N "checking for python libraries... $ECHO_C" >&6
 py_prefix=`$PYTHON -c "import sys; print sys.prefix"`
 py_exec_prefix=`$PYTHON -c "import sys; print sys.exec_prefix"`
-PYTHON_LIBS="-L${py_prefix}/lib -lpython${PYTHON_VERSION}"
-if test "$py_prefix" != "$py_exec_prefix"; then
-  PYTHON_LIBS="$PYTHON_LIBS -L${py_exec_prefix}/lib -lpython${PYTHON_VERSION}"
-fi
+PYTHON_LIBS=`$PYTHON -c "from distutils import sysconfig; print '/'.join(sysconfig.get_config_vars('LIBPL','LIBRARY'))+' '+sysconfig.get_config_var('LIBS')+' '+sysconfig.get_config_var('LINKFORSHARED')"`
 
 save_CPPFLAGS="$CPPFLAGS"
 CPPFLAGS="$CPPFLAGS $PYTHON_INCLUDES"
