diff -uNbr spacehulk-1.3.1/Makefile.am.in spacehulk-1.3.1-patched/Makefile.am.in
--- spacehulk-1.3.1/Makefile.am.in	Wed Dec 31 19:00:00 1969
+++ spacehulk-1.3.1-patched/Makefile.am.in	Wed Mar 26 14:36:15 2003
@@ -0,0 +1,13 @@
+AUTOMAKE_OPTIONS = foreign 1.5
+
+$(top_srcdir)/configure.in: $(top_srcdir)/configure.files $(shell test -f $(top_srcdir)/configure.files && sed -e "s%^%\$(top_srcdir)/%" $(top_srcdir)/configure.files) $(top_srcdir)/subdirs
+	cd $(top_srcdir) && $(MAKE) -f admin/Makefile.common configure.in
+
+$(top_srcdir)/subdirs:
+	cd $(top_srcdir) && $(MAKE) -f admin/Makefile.common subdirs
+
+$(top_srcdir)/acinclude.m4: $(top_srcdir)/admin/acinclude.m4.in $(top_srcdir)/admin/libtool.m4.in
+	@cd $(top_srcdir) && cat admin/acinclude.m4.in admin/libtool.m4.in > acinclude.m4
+
+$(top_srcdir)/configure.files :
+	cd $(top_srcdir) && $(MAKE) -f admin/Makefile.common configure.files
diff -uNbr spacehulk-1.3.1/configure.in.in spacehulk-1.3.1-patched/configure.in.in
--- spacehulk-1.3.1/configure.in.in	Wed Dec 31 19:00:00 1969
+++ spacehulk-1.3.1-patched/configure.in.in	Wed Mar 26 15:10:22 2003
@@ -0,0 +1,39 @@
+AC_INIT(acinclude.m4) dnl a source file from your sub dir
+
+dnl This is so we can use kde-common
+AC_CONFIG_AUX_DIR(admin)
+
+AC_CANONICAL_SYSTEM 
+
+dnl Automake doc recommends to do this only here. (Janos)
+AM_INIT_AUTOMAKE(spacehulk,1.3.1)
+
+dnl KDE_SET_PREFIX
+
+AC_PREFIX_DEFAULT(/usr/local)
+if test "x$prefix" = "xNONE"; then
+dnl  prefix=$ac_default_prefix
+dnl  bindir=$ac_default_prefix/games
+dnl  datadir=$ac_default_prefix/share/games
+dnl  ac_configure_args="$ac_configure_args --prefix $prefix --bindir $bindir --datadir $datadir"
+  ac_configure_args="$ac_configure_args --prefix $prefix"
+fi
+
+dnl generate the config header
+AM_CONFIG_HEADER(config.h) dnl at the distribution this done
+
+dnl Checks for programs.
+AC_CHECK_COMPILERS
+
+AC_ENABLE_SHARED(yes)
+AC_ENABLE_STATIC(no)
+KDE_PROG_LIBTOOL
+
+all_libraries="$all_libraries $USER_LDFLAGS"
+all_includes="$all_includes $USER_INCLUDES"
+AC_SUBST(all_includes)
+AC_SUBST(all_libraries)
+
+dnl Check for QT
+KDE_USE_QT
+AC_PATH_QT
diff -uNbr spacehulk-1.3.1/src/Makefile.am spacehulk-1.3.1-patched/src/Makefile.am
--- spacehulk-1.3.1/src/Makefile.am	Wed Mar 19 08:05:06 2003
+++ spacehulk-1.3.1-patched/src/Makefile.am	Wed Mar 26 15:10:01 2003
@@ -64,8 +64,8 @@
 INCLUDES= $(all_includes) -Wall -DQT_THREAD_SUPPORT -DDATA_DIR=\"$(pkgdatadir)\"
 
 #qspacehulk_LDADD   = $(LIB_QPE) $(LIB_QT) $(LIBSOCKET) -lqt-mt
-qspacehulk_LDADD   = -lqt-mt
-xspacehulk_LDADD   = -lqt-mt
+qspacehulk_LDADD   = -lqt-mt $(LIBZ)
+xspacehulk_LDADD   = -lqt-mt $(LIBZ)
 #qspacehulk_METASOURCES = AUTO
 
 # the library search path. 
