diff -ruN -x '*~' xboard-4.2.7/Makefile.in xboard-4.2.7.patched/Makefile.in
--- xboard-4.2.7/Makefile.in	2003-11-28 16:12:57.000000000 -0500
+++ xboard-4.2.7.patched/Makefile.in	2005-08-02 16:01:14.000000000 -0400
@@ -70,8 +70,6 @@
 	$(INSTALL_DATA) $(srcdir)/xboard.man $(man6dir)/xboard$(man6ext)
 	$(INSTALL_DATA) $(srcdir)/zic2xpm.man $(man6dir)/zic2xpm$(man6ext)
 	$(INSTALL_DATA) xboard.info $(infodir)/xboard.info
-	-PATH=$$PATH:/sbin:/usr/sbin install-info \
-		$(infodir)/xboard.info $(infodir)/dir
 
 installdirs: mkinstalldirs
 	$(srcdir)/mkinstalldirs $(bindir) $(mandir) $(man6dir) $(infodir)
