diff -Nurd unison-2.27.57.ori/Makefile.OCaml unison-2.27.57/Makefile.OCaml
--- unison-2.27.57.ori/Makefile.OCaml	2007-05-07 12:42:29.000000000 -0300
+++ unison-2.27.57/Makefile.OCaml	2010-01-22 02:58:51.000000000 -0200
@@ -149,7 +149,7 @@
 # builds in build/Default/, and earlier versions use build/
 macexecutable: $(NAME)-blob.o
 	sed -e's/@@VERSION@@/$(VERSION)/' $(UIMACDIR)/Info.plist.template > $(UIMACDIR)/Info.plist
-	(cd $(UIMACDIR); xcodebuild OCAMLLIBDIR="$(OCAMLLIBDIR)")
+	(cd $(UIMACDIR); xcodebuild ARCHS=@ARCH@ OCAMLLIBDIR="$(OCAMLLIBDIR)")
 	if [ -e $(UIMACDIR)/build/Default ]; then \
 	  gcc $(UIMACDIR)/cltool.c -o $(UIMACDIR)/build/Default/Unison.app/Contents/MacOS/cltool -framework Carbon; \
 	else \
diff -Nurd unison-2.27.57.ori/uimacnew/English.lproj/MainMenu.nib/classes.nib unison-2.27.57/uimacnew/English.lproj/MainMenu.nib/classes.nib
--- unison-2.27.57.ori/uimacnew/English.lproj/MainMenu.nib/classes.nib	2007-10-03 23:53:48.000000000 -0300
+++ unison-2.27.57/uimacnew/English.lproj/MainMenu.nib/classes.nib	2010-01-22 02:58:03.000000000 -0200
@@ -1,127 +1,259 @@
-{
-    IBClasses = (
-        {
-            ACTIONS = {
-                copyLR = id; 
-                copyRL = id; 
-                forceNewer = id; 
-                forceOlder = id; 
-                ignoreExt = id; 
-                ignoreName = id; 
-                ignorePath = id; 
-                leaveAlone = id; 
-                merge = id; 
-                revert = id; 
-                selectConflicts = id; 
-                showDiff = id; 
-            }; 
-            CLASS = FirstResponder; 
-            LANGUAGE = ObjC; 
-            SUPERCLASS = NSObject; 
-        }, 
-        {
-            CLASS = MessageProgressIndicator; 
-            LANGUAGE = ObjC; 
-            SUPERCLASS = NSProgressIndicator; 
-        }, 
-        {
-            ACTIONS = {
-                cancelProfileButton = id; 
-                cltoolNoButton = id; 
-                cltoolYesButton = id; 
-                createButton = id; 
-                endPasswordWindow = id; 
-                installCommandLineTool = id; 
-                onlineHelp = id; 
-                openButton = id; 
-                raiseAboutWindow = id; 
-                raiseCltoolWindow = id; 
-                raiseWindow = id; 
-                rescan = id; 
-                restartButton = id; 
-                saveProfileButton = id; 
-                syncButton = id; 
-                tableModeChanged = id; 
-            }; 
-            CLASS = MyController; 
-            LANGUAGE = ObjC; 
-            OUTLETS = {
-                ConnectingView = NSView; 
-                aboutWindow = NSWindow; 
-                chooseProfileView = NSView; 
-                cltoolPref = NSButton; 
-                cltoolWindow = NSWindow; 
-                detailsTextView = NSTextView; 
-                diffView = NSTextView; 
-                diffWindow = NSWindow; 
-                mainWindow = NSWindow; 
-                notificationController = NotificationController; 
-                passwordCancelButton = NSButton; 
-                passwordPrompt = NSTextField; 
-                passwordText = NSTextField; 
-                passwordWindow = NSWindow; 
-                preferencesController = PreferencesController; 
-                preferencesView = NSView; 
-                profileController = ProfileController; 
-                progressBar = NSProgressIndicator; 
-                statusText = NSTextField; 
-                tableModeSelector = NSSegmentedControl; 
-                tableView = ReconTableView; 
-                updatesText = NSTextField; 
-                updatesView = NSView; 
-                versionText = NSTextField; 
-            }; 
-            SUPERCLASS = NSObject; 
-        }, 
-        {CLASS = NSSegmentedControl; LANGUAGE = ObjC; SUPERCLASS = NSControl; }, 
-        {CLASS = NotificationController; LANGUAGE = ObjC; SUPERCLASS = NSObject; }, 
-        {
-            ACTIONS = {anyEnter = id; localClick = id; remoteClick = id; }; 
-            CLASS = PreferencesController; 
-            LANGUAGE = ObjC; 
-            OUTLETS = {
-                firstRootText = NSTextField; 
-                localButtonCell = NSButtonCell; 
-                profileNameText = NSTextField; 
-                remoteButtonCell = NSButtonCell; 
-                secondRootHost = NSTextField; 
-                secondRootText = NSTextField; 
-                secondRootUser = NSTextField; 
-            }; 
-            SUPERCLASS = NSObject; 
-        }, 
-        {
-            CLASS = ProfileController; 
-            LANGUAGE = ObjC; 
-            OUTLETS = {tableView = NSTableView; }; 
-            SUPERCLASS = NSObject; 
-        }, 
-        {
-            CLASS = ProfileTableView; 
-            LANGUAGE = ObjC; 
-            OUTLETS = {myController = MyController; }; 
-            SUPERCLASS = NSTableView; 
-        }, 
-        {CLASS = ReconItem; LANGUAGE = ObjC; SUPERCLASS = NSObject; }, 
-        {
-            ACTIONS = {
-                copyLR = id; 
-                copyRL = id; 
-                forceNewer = id; 
-                forceOlder = id; 
-                ignoreExt = id; 
-                ignoreName = id; 
-                ignorePath = id; 
-                leaveAlone = id; 
-                merge = id; 
-                revert = id; 
-                selectConflicts = id; 
-                showDiff = id; 
-            }; 
-            CLASS = ReconTableView; 
-            LANGUAGE = ObjC; 
-            SUPERCLASS = NSOutlineView; 
-        }
-    ); 
-    IBVersion = 1; 
-}
\ No newline at end of file
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+<plist version="1.0">
+<dict>
+	<key>IBClasses</key>
+	<array>
+		<dict>
+			<key>CLASS</key>
+			<string>NSSegmentedControl</string>
+			<key>LANGUAGE</key>
+			<string>ObjC</string>
+			<key>SUPERCLASS</key>
+			<string>NSControl</string>
+		</dict>
+		<dict>
+			<key>CLASS</key>
+			<string>ProfileTableView</string>
+			<key>LANGUAGE</key>
+			<string>ObjC</string>
+			<key>OUTLETS</key>
+			<dict>
+				<key>myController</key>
+				<string>MyController</string>
+			</dict>
+			<key>SUPERCLASS</key>
+			<string>NSTableView</string>
+		</dict>
+		<dict>
+			<key>CLASS</key>
+			<string>ProfileController</string>
+			<key>LANGUAGE</key>
+			<string>ObjC</string>
+			<key>OUTLETS</key>
+			<dict>
+				<key>tableView</key>
+				<string>NSTableView</string>
+			</dict>
+			<key>SUPERCLASS</key>
+			<string>NSObject</string>
+		</dict>
+		<dict>
+			<key>CLASS</key>
+			<string>NotificationController</string>
+			<key>LANGUAGE</key>
+			<string>ObjC</string>
+			<key>SUPERCLASS</key>
+			<string>NSObject</string>
+		</dict>
+		<dict>
+			<key>ACTIONS</key>
+			<dict>
+				<key>anyEnter</key>
+				<string>id</string>
+				<key>localClick</key>
+				<string>id</string>
+				<key>remoteClick</key>
+				<string>id</string>
+			</dict>
+			<key>CLASS</key>
+			<string>PreferencesController</string>
+			<key>LANGUAGE</key>
+			<string>ObjC</string>
+			<key>OUTLETS</key>
+			<dict>
+				<key>firstRootText</key>
+				<string>NSTextField</string>
+				<key>localButtonCell</key>
+				<string>NSButtonCell</string>
+				<key>profileNameText</key>
+				<string>NSTextField</string>
+				<key>remoteButtonCell</key>
+				<string>NSButtonCell</string>
+				<key>secondRootHost</key>
+				<string>NSTextField</string>
+				<key>secondRootText</key>
+				<string>NSTextField</string>
+				<key>secondRootUser</key>
+				<string>NSTextField</string>
+			</dict>
+			<key>SUPERCLASS</key>
+			<string>NSObject</string>
+		</dict>
+		<dict>
+			<key>ACTIONS</key>
+			<dict>
+				<key>copyLR</key>
+				<string>id</string>
+				<key>copyRL</key>
+				<string>id</string>
+				<key>forceNewer</key>
+				<string>id</string>
+				<key>forceOlder</key>
+				<string>id</string>
+				<key>ignoreExt</key>
+				<string>id</string>
+				<key>ignoreName</key>
+				<string>id</string>
+				<key>ignorePath</key>
+				<string>id</string>
+				<key>leaveAlone</key>
+				<string>id</string>
+				<key>merge</key>
+				<string>id</string>
+				<key>revert</key>
+				<string>id</string>
+				<key>selectConflicts</key>
+				<string>id</string>
+				<key>showDiff</key>
+				<string>id</string>
+			</dict>
+			<key>CLASS</key>
+			<string>FirstResponder</string>
+			<key>LANGUAGE</key>
+			<string>ObjC</string>
+			<key>SUPERCLASS</key>
+			<string>NSObject</string>
+		</dict>
+		<dict>
+			<key>CLASS</key>
+			<string>MessageProgressIndicator</string>
+			<key>LANGUAGE</key>
+			<string>ObjC</string>
+			<key>SUPERCLASS</key>
+			<string>NSProgressIndicator</string>
+		</dict>
+		<dict>
+			<key>ACTIONS</key>
+			<dict>
+				<key>cancelProfileButton</key>
+				<string>id</string>
+				<key>cltoolNoButton</key>
+				<string>id</string>
+				<key>cltoolYesButton</key>
+				<string>id</string>
+				<key>createButton</key>
+				<string>id</string>
+				<key>endPasswordWindow</key>
+				<string>id</string>
+				<key>installCommandLineTool</key>
+				<string>id</string>
+				<key>onlineHelp</key>
+				<string>id</string>
+				<key>openButton</key>
+				<string>id</string>
+				<key>raiseAboutWindow</key>
+				<string>id</string>
+				<key>raiseCltoolWindow</key>
+				<string>id</string>
+				<key>raiseWindow</key>
+				<string>id</string>
+				<key>rescan</key>
+				<string>id</string>
+				<key>restartButton</key>
+				<string>id</string>
+				<key>saveProfileButton</key>
+				<string>id</string>
+				<key>syncButton</key>
+				<string>id</string>
+				<key>tableModeChanged</key>
+				<string>id</string>
+			</dict>
+			<key>CLASS</key>
+			<string>MyController</string>
+			<key>LANGUAGE</key>
+			<string>ObjC</string>
+			<key>OUTLETS</key>
+			<dict>
+				<key>ConnectingView</key>
+				<string>NSView</string>
+				<key>aboutWindow</key>
+				<string>NSWindow</string>
+				<key>chooseProfileView</key>
+				<string>NSView</string>
+				<key>cltoolPref</key>
+				<string>NSButton</string>
+				<key>cltoolWindow</key>
+				<string>NSWindow</string>
+				<key>detailsTextView</key>
+				<string>NSTextView</string>
+				<key>diffView</key>
+				<string>NSTextView</string>
+				<key>diffWindow</key>
+				<string>NSWindow</string>
+				<key>mainWindow</key>
+				<string>NSWindow</string>
+				<key>notificationController</key>
+				<string>NotificationController</string>
+				<key>passwordCancelButton</key>
+				<string>NSButton</string>
+				<key>passwordPrompt</key>
+				<string>NSTextField</string>
+				<key>passwordText</key>
+				<string>NSTextField</string>
+				<key>passwordWindow</key>
+				<string>NSWindow</string>
+				<key>preferencesController</key>
+				<string>PreferencesController</string>
+				<key>preferencesView</key>
+				<string>NSView</string>
+				<key>profileController</key>
+				<string>ProfileController</string>
+				<key>progressBar</key>
+				<string>NSProgressIndicator</string>
+				<key>statusText</key>
+				<string>NSTextField</string>
+				<key>tableModeSelector</key>
+				<string>NSSegmentedControl</string>
+				<key>tableView</key>
+				<string>ReconTableView</string>
+				<key>updatesText</key>
+				<string>NSTextField</string>
+				<key>updatesView</key>
+				<string>NSView</string>
+				<key>versionText</key>
+				<string>NSTextField</string>
+			</dict>
+			<key>SUPERCLASS</key>
+			<string>NSObject</string>
+		</dict>
+		<dict>
+			<key>ACTIONS</key>
+			<dict>
+				<key>copyLR</key>
+				<string>id</string>
+				<key>copyRL</key>
+				<string>id</string>
+				<key>forceNewer</key>
+				<string>id</string>
+				<key>forceOlder</key>
+				<string>id</string>
+				<key>ignoreExt</key>
+				<string>id</string>
+				<key>ignoreName</key>
+				<string>id</string>
+				<key>ignorePath</key>
+				<string>id</string>
+				<key>leaveAlone</key>
+				<string>id</string>
+				<key>merge</key>
+				<string>id</string>
+				<key>revert</key>
+				<string>id</string>
+				<key>selectConflicts</key>
+				<string>id</string>
+				<key>showDiff</key>
+				<string>id</string>
+			</dict>
+			<key>CLASS</key>
+			<string>ReconTableView</string>
+			<key>LANGUAGE</key>
+			<string>ObjC</string>
+			<key>SUPERCLASS</key>
+			<string>NSOutlineView</string>
+		</dict>
+	</array>
+	<key>IBVersion</key>
+	<string>1</string>
+</dict>
+</plist>
diff -Nurd unison-2.27.57.ori/uimacnew/English.lproj/MainMenu.nib/info.nib unison-2.27.57/uimacnew/English.lproj/MainMenu.nib/info.nib
--- unison-2.27.57.ori/uimacnew/English.lproj/MainMenu.nib/info.nib	2007-10-03 23:53:48.000000000 -0300
+++ unison-2.27.57/uimacnew/English.lproj/MainMenu.nib/info.nib	2010-01-22 02:58:03.000000000 -0200
@@ -1,41 +1,18 @@
 <?xml version="1.0" encoding="UTF-8"?>
-<!DOCTYPE plist PUBLIC "-//Apple Computer//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
 <plist version="1.0">
 <dict>
-	<key>IBDocumentLocation</key>
-	<string>48 53 538 389 0 0 1920 1178 </string>
-	<key>IBEditorPositions</key>
-	<dict>
-		<key>197</key>
-		<string>745 623 383 323 0 0 1920 1178 </string>
-		<key>198</key>
-		<string>584 479 697 588 0 0 1920 1178 </string>
-		<key>29</key>
-		<string>256 460 294 44 0 0 1920 1178 </string>
-		<key>307</key>
-		<string>659 639 554 292 0 0 1920 1178 </string>
-		<key>423</key>
-		<string>491 475 891 619 0 0 1920 1178 </string>
-	</dict>
 	<key>IBFramework Version</key>
-	<string>446.1</string>
-	<key>IBLockedObjects</key>
-	<array>
-		<integer>234</integer>
-		<integer>402</integer>
-		<integer>21</integer>
-	</array>
+	<string>644</string>
+	<key>IBOldestOS</key>
+	<integer>5</integer>
 	<key>IBOpenObjects</key>
 	<array>
-		<integer>402</integer>
-		<integer>29</integer>
-		<integer>197</integer>
-		<integer>21</integer>
-		<integer>307</integer>
-		<integer>423</integer>
-		<integer>198</integer>
+		<integer>57</integer>
 	</array>
 	<key>IBSystem Version</key>
-	<string>8P2137</string>
+	<string>9D34</string>
+	<key>targetFramework</key>
+	<string>IBCocoaFramework</string>
 </dict>
 </plist>
Binary files unison-2.27.57.ori/uimacnew/English.lproj/MainMenu.nib/keyedobjects.nib and unison-2.27.57/uimacnew/English.lproj/MainMenu.nib/keyedobjects.nib differ
Binary files unison-2.27.57.ori/uimacnew/English.lproj/MainMenu.nib/objects.nib and unison-2.27.57/uimacnew/English.lproj/MainMenu.nib/objects.nib differ
diff -Nurd unison-2.27.57.ori/uimacnew/MyController.m unison-2.27.57/uimacnew/MyController.m
--- unison-2.27.57.ori/uimacnew/MyController.m	2007-10-03 23:53:48.000000000 -0300
+++ unison-2.27.57/uimacnew/MyController.m	2010-01-22 02:58:03.000000000 -0200
@@ -109,13 +109,6 @@
 
     [mainWindow display];
     [mainWindow makeKeyAndOrderFront:nil];
-
-    /* unless user has clicked Don't ask me again, ask about cltool */
-    if ( ([[NSUserDefaults standardUserDefaults]
-            integerForKey:@"CheckCltool"]==doAsk) &&
-        (![[NSFileManager defaultManager]
-            fileExistsAtPath:@"/usr/bin/unison"]) )
-            [self raiseCltoolWindow:nil];
 }
 
 - (void)chooseProfiles
