diff -Nurd unison-2.32.52.ori/Makefile.OCaml unison-2.32.52/Makefile.OCaml
--- unison-2.32.52.ori/Makefile.OCaml	2009-06-18 06:29:40.000000000 -0300
+++ unison-2.32.52/Makefile.OCaml	2010-01-22 04:13:15.000000000 -0200
@@ -162,7 +162,7 @@
 # builds in build/Default/, and earlier versions use build/
 macexecutable: $(NAME)-blob.o
 #	sed -e's/@@VERSION@@/$(VERSION)/' $(UIMACDIR)/Info.plist.template > $(UIMACDIR)/Info.plist
-	(cd $(UIMACDIR); xcodebuild OCAMLLIBDIR="$(OCAMLLIBDIR)" SYMROOT=build)
+	(cd $(UIMACDIR); xcodebuild -sdk macosx`sw_vers -productVersion | cut -d. -f1,2` ARCHS=@ARCH@ OCAMLLIBDIR="$(OCAMLLIBDIR)" SYMROOT=build)
 	if [ -e $(UIMACDIR)/build/Default ]; then \
 	  gcc -mmacosx-version-min=$(MINOSXVERSION) $(UIMACDIR)/cltool.c -o $(UIMACDIR)/build/Default/Unison.app/Contents/MacOS/cltool -framework Carbon; \
 	else \
