diff -uNr kdebase-3.2.91/kioslave/info/kde-info2html.conf kdebase-3.2.91-new/kioslave/info/kde-info2html.conf
--- kdebase-3.2.91/kioslave/info/kde-info2html.conf	2004-05-22 16:26:52.000000000 -0400
+++ kdebase-3.2.91-new/kioslave/info/kde-info2html.conf	2004-07-02 08:57:17.000000000 -0400
@@ -25,6 +25,7 @@
 
 #-- location of info files.
 our @INFODIR = (
+	    "@PREFIX@/share/info",
 	    "/usr/share/info",
 	    "/usr/info",
 	    "/usr/lib/info",
diff -uNr kdebase-3.2.91/kioslave/info/kde-info2html.conf.orig kdebase-3.2.91-new/kioslave/info/kde-info2html.conf.orig
--- kdebase-3.2.91/kioslave/info/kde-info2html.conf.orig	1969-12-31 19:00:00.000000000 -0500
+++ kdebase-3.2.91-new/kioslave/info/kde-info2html.conf.orig	2004-05-22 16:26:52.000000000 -0400
@@ -0,0 +1,43 @@
+# -*- perl -*-
+package info2html::config;
+#-----------------------------------------------------------------
+#                           info2html.conf
+#-----------------------------------------------------------------
+# PURPOSE
+#   configuration settings for the 'info2html' script.
+#
+# AUTHOR
+#   Karl Guggisberg  <guggis@iam.unibe.ch>
+#
+# HISTORY
+#   15.10.93 V 1.0b   
+#   16.10.93 V 1.0c  multple info files possible
+#   28.6.94  V 1.0d  some minor changes
+#   8.4.95   V 1.1   some changements
+#----------------------------------------------------------------
+
+use strict;
+#use vars qw(@ISA @EXPORT);
+#
+#@ISA = qw(Exporter);
+#@EXPORT = qw(@INFODIR $DOC_URL);
+
+
+#-- location of info files.
+our @INFODIR = (
+	    "/usr/share/info",
+	    "/usr/info",
+	    "/usr/lib/info",
+#	    "/usr/lib/teTeX/info",
+	    "/usr/local/info",
+	    "/usr/local/lib/info",
+	    "/usr/X11R6/info",
+	    "/usr/X11R6/lib/info",
+	    "/usr/X11R6/lib/xemacs/info"
+	    );
+
+
+#-- URL for documentation of info2html
+our $DOC_URL = 'http://info2html.sourceforge.net/';
+
+1;
diff -uNr kdebase-3.2.91/kioslave/man/kio_man.cpp kdebase-3.2.91-new/kioslave/man/kio_man.cpp
--- kdebase-3.2.91/kioslave/man/kio_man.cpp	2004-06-22 13:32:52.000000000 -0400
+++ kdebase-3.2.91-new/kioslave/man/kio_man.cpp	2004-07-02 08:57:17.000000000 -0400
@@ -50,7 +50,7 @@
 
 MANProtocol *MANProtocol::_self = 0;
 
-#define SGML2ROFF_DIRS "/usr/lib/sgml"
+#define SGML2ROFF_DIRS "@PREFIX@/share/sgml"
 
 /*
  * Drop trailing ".section[.gz]" from name
diff -uNr kdebase-3.2.91/kwin/client.cpp kdebase-3.2.91-new/kwin/client.cpp
--- kdebase-3.2.91/kwin/client.cpp	2004-06-22 13:32:58.000000000 -0400
+++ kdebase-3.2.91-new/kwin/client.cpp	2004-07-02 09:03:46.000000000 -0400
@@ -1388,7 +1388,7 @@
  */
 QCString Client::staticWindowRole(WId w)
     {
-    return getStringProperty(w, qt_window_role);
+    return getStringProperty(w, qt_window_role).lower();
     }
 
 /*!
@@ -1608,7 +1608,7 @@
  */
 void Client::setCursor( Position m )
     {
-    if ( !isResizable() || isShade() || noBorder())
+    if( !isResizable() || isShade())
         {
         m = PositionCenter;
         }
diff -uNr kdebase-3.2.91/kwin/events.cpp kdebase-3.2.91-new/kwin/events.cpp
--- kdebase-3.2.91/kwin/events.cpp	2004-06-22 13:32:58.000000000 -0400
+++ kdebase-3.2.91-new/kwin/events.cpp	2004-07-02 09:03:46.000000000 -0400
@@ -853,7 +853,7 @@
             else if (e->atom == atoms->wm_client_leader )
                 getWmClientLeader();
             else if( e->atom == qt_window_role )
-                window_role = getStringProperty( window(), qt_window_role );
+                window_role = staticWindowRole( window());
             break;
         }
     }
diff -uNr kdebase-3.2.91/kwin/geometry.cpp kdebase-3.2.91-new/kwin/geometry.cpp
--- kdebase-3.2.91/kwin/geometry.cpp	2004-06-26 09:24:02.000000000 -0400
+++ kdebase-3.2.91-new/kwin/geometry.cpp	2004-07-02 09:03:46.000000000 -0400
@@ -1540,8 +1540,16 @@
     if( !isMovable() || !isResizable() || isToolbar()) // SELI isToolbar() ?
         return false;
     QSize max = maxSize();
+#if 0
     if( max.width() < 32767 || max.height() < 32767 ) // sizes are 16bit with X
         return false;
+#else
+    // apparently there are enough apps which specify some arbitrary value
+    // for their maximum size just for the fun of it
+    QSize areasize = workspace()->clientArea( MaximizeArea, this ).size();
+    if( max.width() < areasize.width() || max.height() < areasize.height())
+        return false;
+#endif
     return true;
     }
 
diff -uNr kdebase-3.2.91/kwin/kcmkwin/kwinoptions/windows.cpp kdebase-3.2.91-new/kwin/kcmkwin/kwinoptions/windows.cpp
--- kdebase-3.2.91/kwin/kcmkwin/kwinoptions/windows.cpp	2004-06-22 13:32:58.000000000 -0400
+++ kdebase-3.2.91-new/kwin/kcmkwin/kwinoptions/windows.cpp	2004-07-02 09:03:46.000000000 -0400
@@ -54,7 +54,7 @@
 #define KWIN_MINIMIZE_ANIM         "AnimateMinimize"
 #define KWIN_MINIMIZE_ANIM_SPEED   "AnimateMinimizeSpeed"
 #define KWIN_RESIZE_OPAQUE         "ResizeMode"
-#define KWIN_GEOMETRY		   "GeometryTip"
+#define KWIN_GEOMETRY              "GeometryTip"
 #define KWIN_AUTORAISE_INTERVAL    "AutoRaiseInterval"
 #define KWIN_AUTORAISE             "AutoRaise"
 #define KWIN_DELAYFOCUS_INTERVAL   "DelayFocusInterval"
@@ -206,42 +206,38 @@
 
     kbdBox = new QButtonGroup(i18n("Navigation"), this);
     kbdBox->setColumnLayout( 0, Qt::Horizontal );
-    QGridLayout *kLay = new QGridLayout(kbdBox->layout(), 4, 4,
-                                        KDialog::spacingHint());
-    QLabel *altTabLabel = new QLabel( i18n("Walk through windows mode:"), kbdBox);
-    kLay->addWidget(altTabLabel, 1, 0);
-    kdeMode = new QRadioButton(i18n("&KDE"), kbdBox);
-    kLay->addWidget(kdeMode, 1, 1);
-    cdeMode = new QRadioButton(i18n("CD&E"), kbdBox);
-    kLay->addWidget(cdeMode, 1, 2);
+    QVBoxLayout *kLay = new QVBoxLayout(kbdBox->layout(), KDialog::spacingHint());
+
+    altTabPopup = new QCheckBox( i18n("Show window list while switching windows"), kbdBox );
+    kLay->addWidget( altTabPopup );
 
     wtstr = i18n("Hold down the Alt key and press the Tab key repeatedly to walk"
                  " through the windows on the current desktop (the Alt+Tab"
-                 " combination can be reconfigured). The two different modes mean:<ul>"
-                 "<li><b>KDE</b>: a nice widget is shown, displaying the icons of all windows to"
-                 " walk through and the title of the currently selected one;"
-                 "<li><b>CDE</b>: the focus is passed to a new window each time Tab is pressed."
-                 " No fancy widget.</li></ul>");
-    QWhatsThis::add( altTabLabel, wtstr );
-    QWhatsThis::add( kdeMode, wtstr );
-    QWhatsThis::add( cdeMode, wtstr );
+                 " combination can be reconfigured).\n\n"
+                 "If this checkbox is checked"
+                 " a popup widget is shown, displaying the icons of all windows to"
+                 " walk through and the title of the currently selected one.\n\n"
+                 "Otherwise, the focus is passed to a new window each time Tab"
+                 " is pressed, with no popup widget.  In addition, the previously"
+                 " activated window will be sent to the back in this mode.");
+    QWhatsThis::add( altTabPopup, wtstr );
 
     traverseAll = new QCheckBox( i18n( "&Traverse windows on all desktops" ), kbdBox );
-    kLay->addMultiCellWidget( traverseAll, 2, 2, 0, 2 );
+    kLay->addWidget( traverseAll );
 
     wtstr = i18n( "Leave this option disabled if you want to limit walking through"
                   " windows to the current desktop." );
     QWhatsThis::add( traverseAll, wtstr );
 
     rollOverDesktops = new QCheckBox( i18n("Desktop navi&gation wraps around"), kbdBox );
-    kLay->addMultiCellWidget(rollOverDesktops, 3, 3, 0, 2);
+    kLay->addWidget(rollOverDesktops);
 
     wtstr = i18n( "Enable this option if you want keyboard or active desktop border navigation beyond"
                   " the edge of a desktop to take you to the opposite edge of the new desktop." );
     QWhatsThis::add( rollOverDesktops, wtstr );
 
     showPopupinfo = new QCheckBox( i18n("Popup desktop name on desktop &switch"), kbdBox );
-    kLay->addMultiCellWidget(showPopupinfo, 4, 4, 0, 2);
+    kLay->addWidget(showPopupinfo);
 
     wtstr = i18n( "Enable this option if you wish to see the current desktop"
                   " name popup whenever the current desktop is changed." );
@@ -256,8 +252,7 @@
     connect(fcsBox, SIGNAL(clicked(int)), SLOT(changed()));
     connect(autoRaise, SIGNAL(valueChanged(int)), SLOT(changed()));
     connect(delayFocus, SIGNAL(valueChanged(int)), SLOT(changed()));
-    connect(kdeMode, SIGNAL(clicked()), SLOT(changed()));
-    connect(cdeMode, SIGNAL(clicked()), SLOT(changed()));
+    connect(altTabPopup, SIGNAL(clicked()), SLOT(changed()));
     connect(traverseAll, SIGNAL(clicked()), SLOT(changed()));
     connect(rollOverDesktops, SIGNAL(clicked()), SLOT(changed()));
     connect(showPopupinfo, SIGNAL(clicked()), SLOT(changed()));
@@ -357,8 +352,7 @@
 }
 
 void KFocusConfig::setAltTabMode(bool a) {
-    kdeMode->setChecked(a);
-    cdeMode->setChecked(!a);
+    altTabPopup->setChecked(a);
 }
 
 void KFocusConfig::setTraverseAll(bool a) {
@@ -458,7 +452,7 @@
     else
         config->writeEntry(KWIN_CLICKRAISE, "off");
 
-    if (kdeMode->isChecked())
+    if (altTabPopup->isChecked())
         config->writeEntry(KWIN_ALTTABMODE, "KDE");
     else
         config->writeEntry(KWIN_ALTTABMODE, "CDE");
@@ -774,9 +768,9 @@
     geometryTipOn = new QCheckBox(i18n("Display window &geometry when moving or resizing"), windowsBox);
     bLay->addWidget(geometryTipOn);
     QWhatsThis::add(geometryTipOn, i18n("Enable this option if you want a window's geometry to be displayed"
-			    		" while it is being moved or resized. The window position relative"
-					" to the top-left corner of the screen is displayed together with"
-					" its size."));
+                                        " while it is being moved or resized. The window position relative"
+                                        " to the top-left corner of the screen is displayed together with"
+                                        " its size."));
 
     QGridLayout *rLay = new QGridLayout(2,3);
     bLay->addLayout(rLay);
diff -uNr kdebase-3.2.91/kwin/kcmkwin/kwinoptions/windows.h kdebase-3.2.91-new/kwin/kcmkwin/kwinoptions/windows.h
--- kdebase-3.2.91/kwin/kcmkwin/kwinoptions/windows.h	2004-06-22 13:32:58.000000000 -0400
+++ kdebase-3.2.91-new/kwin/kcmkwin/kwinoptions/windows.h	2004-07-02 09:03:46.000000000 -0400
@@ -111,8 +111,7 @@
   KIntNumInput *delayFocus;
 
   QButtonGroup *kbdBox;
-  QRadioButton *kdeMode;
-  QRadioButton *cdeMode;
+  QCheckBox    *altTabPopup;
   QCheckBox    *traverseAll;
   QCheckBox    *rollOverDesktops;
   QCheckBox    *showPopupinfo;
diff -uNr kdebase-3.2.91/kwin/kcmkwin/kwinrules/Makefile.am kdebase-3.2.91-new/kwin/kcmkwin/kwinrules/Makefile.am
--- kdebase-3.2.91/kwin/kcmkwin/kwinrules/Makefile.am	2004-06-26 09:24:22.000000000 -0400
+++ kdebase-3.2.91-new/kwin/kcmkwin/kwinrules/Makefile.am	2004-07-02 09:03:46.000000000 -0400
@@ -2,6 +2,8 @@
 AM_CPPFLAGS = $(all_includes) -DKCMRULES
 INCLUDES = -I$(top_srcdir)/kwin/lib
 
+KDE_OPTIONS = nofinal
+
 kde_module_LTLIBRARIES = kcm_kwinrules.la 
 
 kcm_kwinrules_la_SOURCES = main.cpp ruleswidget.cpp ruleslist.cpp kwinsrc.cpp detectwidget.cpp \
@@ -9,7 +11,7 @@
 kcm_kwinrules_la_LDFLAGS = -module -avoid-version $(all_libraries) -no-undefined
 kcm_kwinrules_la_LIBADD = $(LIB_KDEUI)
 
-messages:
-	$(XGETTEXT) *.cpp -o $(podir)/kcmkwm.pot
+messages: rc.cpp
+	$(XGETTEXT) *.cpp -o $(podir)/kcmkwinrules.pot
 
 xdg_apps_DATA = kwinrules.desktop
diff -uNr kdebase-3.2.91/kwin/kcmkwin/kwinrules/detectwidget.cpp kdebase-3.2.91-new/kwin/kcmkwin/kwinrules/detectwidget.cpp
--- kdebase-3.2.91/kwin/kcmkwin/kwinrules/detectwidget.cpp	2004-06-26 09:24:22.000000000 -0400
+++ kdebase-3.2.91-new/kwin/kcmkwin/kwinrules/detectwidget.cpp	2004-07-02 09:03:46.000000000 -0400
@@ -55,7 +55,7 @@
         readWindow( window );
     }
 
-static QCString getStringProperty(WId w, Atom prop, char separator)
+static QCString getStringProperty(WId w, Atom prop, char separator=0)
     {
     Atom type;
     int format, status;
@@ -66,9 +66,9 @@
     status = XGetWindowProperty( qt_xdisplay(), w, prop, 0, 10000,
                                  FALSE, XA_STRING, &type, &format,
                                  &nitems, &extra, &data );
-    if ( status == Success) 
+    if ( status == Success)
         {
-        if (data && separator) 
+        if (data && separator)
             {
             for (int i=0; i<(int)nitems; i++)
                 if (!data[i] && i+1<(int)nitems)
@@ -128,7 +128,7 @@
         I18N_NOOP( "Override Type" ),
         I18N_NOOP( "Standalone Menubar" ),
         I18N_NOOP( "Utility Window" ),
-        I18N_NOOP( "Splashscreen" )
+        I18N_NOOP( "Splash Screen" )
         };
     widget->class_label->setText( wmclass_class + " (" + wmclass_name + ' ' + wmclass_class + ")" );
     widget->role_label->setText( role );
diff -uNr kdebase-3.2.91/kwin/kcmkwin/kwinrules/detectwidgetbase.ui kdebase-3.2.91-new/kwin/kcmkwin/kwinrules/detectwidgetbase.ui
--- kdebase-3.2.91/kwin/kcmkwin/kwinrules/detectwidgetbase.ui	2004-06-26 09:24:22.000000000 -0400
+++ kdebase-3.2.91-new/kwin/kcmkwin/kwinrules/detectwidgetbase.ui	2004-07-02 09:03:46.000000000 -0400
@@ -21,7 +21,7 @@
                 <cstring>textLabel9</cstring>
             </property>
             <property name="text">
-                <string>Information about the selected window:</string>
+                <string>Information About Selected Window</string>
             </property>
         </widget>
         <widget class="Line">
diff -uNr kdebase-3.2.91/kwin/kcmkwin/kwinrules/main.cpp kdebase-3.2.91-new/kwin/kcmkwin/kwinrules/main.cpp
--- kdebase-3.2.91/kwin/kcmkwin/kwinrules/main.cpp	2004-06-21 13:12:23.000000000 -0400
+++ kdebase-3.2.91-new/kwin/kcmkwin/kwinrules/main.cpp	2004-07-02 09:03:46.000000000 -0400
@@ -31,7 +31,7 @@
 KCModule *create_kwinrules( QWidget *parent, const char *name )
     {
     //CT there's need for decision: kwm or kwin?
-    KGlobal::locale()->insertCatalogue( "kcmkwm" );
+    KGlobal::locale()->insertCatalogue( "kcmkwinrules" );
     return new KWinInternal::KCMRules( parent, name );
     }
 
diff -uNr kdebase-3.2.91/kwin/kcmkwin/kwinrules/ruleswidget.cpp kdebase-3.2.91-new/kwin/kcmkwin/kwinrules/ruleswidget.cpp
--- kdebase-3.2.91/kwin/kcmkwin/kwinrules/ruleswidget.cpp	2004-06-26 09:24:22.000000000 -0400
+++ kdebase-3.2.91-new/kwin/kcmkwin/kwinrules/ruleswidget.cpp	2004-07-02 09:03:47.000000000 -0400
@@ -29,6 +29,7 @@
 #include <qregexp.h>
 #include <qwhatsthis.h>
 #include <assert.h>
+#include <kmessagebox.h>
 
 #include "../../rules.h"
 
@@ -187,6 +188,8 @@
 
 static QString sizeToStr( const QSize& s )
     {
+    if( !s.isValid())
+        return QString::null;
     return QString::number( s.width()) + "," + QString::number( s.height());
     }
 
@@ -413,7 +416,7 @@
     rules->types = 0;
     bool all_types = true;
     for( int i = 0;
-         i <= 9;
+         i < types->count();
          ++i )
         if( !types->isSelected( i ))
             all_types = false;
@@ -511,6 +514,26 @@
     detect_dlg = NULL;
     }
 
+bool RulesWidget::finalCheck()
+    {
+    bool all_types = true;
+    for( int i = 0;
+         i < types->count();
+         ++i )
+        if( !types->isSelected( i ))
+            all_types = false;
+    if( wmclass_match->currentItem() == Rules::UnimportantMatch && all_types )
+        {
+        if( KMessageBox::warningContinueCancel( topLevelWidget(),
+            i18n( "You have specified the window class as unimportant.\n"
+                  "This means the settings will possibly apply to windows from all applications. "
+                  "If you really want to create a generic setting, it is recommended you at least "
+                  "limit the window types to avoid special window types." )) != KMessageBox::Continue )
+            return false;
+        }
+    return true;
+    }
+
 RulesDialog::RulesDialog( QWidget* parent, const char* name )
 : KDialogBase( parent, name, true, "", Ok | Cancel )
     {
@@ -528,8 +551,10 @@
 
 void RulesDialog::accept()
     {
-    KDialogBase::accept();
+    if( !widget->finalCheck())
+        return;
     rules = widget->rules();
+    KDialogBase::accept();
     }
 
 } // namespace
diff -uNr kdebase-3.2.91/kwin/kcmkwin/kwinrules/ruleswidget.h kdebase-3.2.91-new/kwin/kcmkwin/kwinrules/ruleswidget.h
--- kdebase-3.2.91/kwin/kcmkwin/kwinrules/ruleswidget.h	2004-06-26 09:24:22.000000000 -0400
+++ kdebase-3.2.91-new/kwin/kcmkwin/kwinrules/ruleswidget.h	2004-07-02 09:03:47.000000000 -0400
@@ -38,6 +38,7 @@
         RulesWidget( QWidget* parent = NULL, const char* name = NULL );
         void setRules( Rules* r );
         Rules* rules() const;
+        bool finalCheck();
     signals:
         void changed( bool state );
     protected slots:
diff -uNr kdebase-3.2.91/kwin/kcmkwin/kwinrules/ruleswidgetbase.ui kdebase-3.2.91-new/kwin/kcmkwin/kwinrules/ruleswidgetbase.ui
--- kdebase-3.2.91/kwin/kcmkwin/kwinrules/ruleswidgetbase.ui	2004-06-26 09:24:22.000000000 -0400
+++ kdebase-3.2.91-new/kwin/kcmkwin/kwinrules/ruleswidgetbase.ui	2004-07-02 09:03:47.000000000 -0400
@@ -99,12 +99,12 @@
                                 </item>
                                 <item>
                                     <property name="text">
-                                        <string>Substring match</string>
+                                        <string>Substring Match</string>
                                     </property>
                                 </item>
                                 <item>
                                     <property name="text">
-                                        <string>Regular expression</string>
+                                        <string>Regular Expression</string>
                                     </property>
                                 </item>
                                 <property name="name">
@@ -195,12 +195,12 @@
                                 </item>
                                 <item>
                                     <property name="text">
-                                        <string>Substring match</string>
+                                        <string>Substring Match</string>
                                     </property>
                                 </item>
                                 <item>
                                     <property name="text">
-                                        <string>Regular expression</string>
+                                        <string>Regular Expression</string>
                                     </property>
                                 </item>
                                 <property name="name">
@@ -400,7 +400,7 @@
                         </item>
                         <item>
                             <property name="text">
-                                <string>Splashscreen</string>
+                                <string>Splash Screen</string>
                             </property>
                         </item>
                         <item>
@@ -459,12 +459,12 @@
                                 </item>
                                 <item>
                                     <property name="text">
-                                        <string>Substring match</string>
+                                        <string>Substring Match</string>
                                     </property>
                                 </item>
                                 <item>
                                     <property name="text">
-                                        <string>Regular expression</string>
+                                        <string>Regular Expression</string>
                                     </property>
                                 </item>
                                 <property name="name">
@@ -555,12 +555,12 @@
                                 </item>
                                 <item>
                                     <property name="text">
-                                        <string>Substring match</string>
+                                        <string>Substring Match</string>
                                     </property>
                                 </item>
                                 <item>
                                     <property name="text">
-                                        <string>Regular expression</string>
+                                        <string>Regular Expression</string>
                                     </property>
                                 </item>
                                 <property name="name">
@@ -651,12 +651,12 @@
                                 </item>
                                 <item>
                                     <property name="text">
-                                        <string>Substring match</string>
+                                        <string>Substring Match</string>
                                     </property>
                                 </item>
                                 <item>
                                     <property name="text">
-                                        <string>Regular expression</string>
+                                        <string>Regular Expression</string>
                                     </property>
                                 </item>
                                 <property name="name">
@@ -1725,7 +1725,7 @@
                         </item>
                         <item>
                             <property name="text">
-                                <string>Splashscreen</string>
+                                <string>Splash Screen</string>
                             </property>
                         </item>
                         <item>
diff -uNr kdebase-3.2.91/kwin/manage.cpp kdebase-3.2.91-new/kwin/manage.cpp
--- kdebase-3.2.91/kwin/manage.cpp	2004-06-22 13:32:58.000000000 -0400
+++ kdebase-3.2.91-new/kwin/manage.cpp	2004-07-02 09:03:46.000000000 -0400
@@ -25,7 +25,6 @@
 #include "rules.h"
 
 extern Time qt_x_time;
-extern Atom qt_window_role;
 
 namespace KWinInternal
 {
@@ -106,7 +105,7 @@
         }
     ignore_focus_stealing = options->checkIgnoreFocusStealing( this ); // TODO change to rules
 
-    window_role = getStringProperty( w, qt_window_role );
+    window_role = staticWindowRole( w );
     // first only read the caption text, so that setupWindowRules() can use it for matching,
     // and only then really set the caption using setCaption(), which checks for duplicates etc.
     // and also relies on rules already existing
diff -uNr kdebase-3.2.91/kwin/rules.cpp kdebase-3.2.91-new/kwin/rules.cpp
--- kdebase-3.2.91/kwin/rules.cpp	2004-06-26 09:24:03.000000000 -0400
+++ kdebase-3.2.91-new/kwin/rules.cpp	2004-07-02 09:03:46.000000000 -0400
@@ -127,10 +127,10 @@
         sizerule = UnusedSetRule;
     READ_FORCE_RULE( minsize, Size, );
     if( !minsize.isValid())
-        minsizerule = UnusedForceRule;
+        minsize = QSize( 1, 1 );
     READ_FORCE_RULE( maxsize, Size, );
     if( maxsize.isEmpty())
-        maxsizerule = UnusedForceRule;
+        maxsize = QSize( 32767, 32767 );
     READ_FORCE_RULE( ignoreposition, Bool, );
     READ_SET_RULE( desktop, Num, );
     type = readType( cfg, "type" );
diff -uNr kdebase-3.2.91/kwin/useractions.cpp kdebase-3.2.91-new/kwin/useractions.cpp
--- kdebase-3.2.91/kwin/useractions.cpp	2004-06-22 13:32:58.000000000 -0400
+++ kdebase-3.2.91-new/kwin/useractions.cpp	2004-07-02 09:03:46.000000000 -0400
@@ -54,10 +54,13 @@
         advanced_popup->setCheckable( TRUE );
         advanced_popup->setFont(KGlobalSettings::menuFont());
         connect( advanced_popup, SIGNAL( activated(int) ), this, SLOT( clientPopupActivated(int) ) );
-        advanced_popup->insertItem( SmallIconSet( "up" ), i18n("Keep &Above Others"), Options::KeepAboveOp );
-        advanced_popup->insertItem( SmallIconSet( "down" ), i18n("Keep &Below Others"), Options::KeepBelowOp );
-        advanced_popup->insertItem( SmallIconSet( "window_fullscreen" ), i18n("&Fullscreen"), Options::FullScreenOp );
-        advanced_popup->insertItem( i18n("&No Border"), Options::NoBorderOp );
+        advanced_popup->insertItem( SmallIconSet( "up" ),
+            i18n("Keep &Above Others")+'\t'+keys->shortcut("Window Above Other Windows").seq(0).toString(), Options::KeepAboveOp );
+        advanced_popup->insertItem( SmallIconSet( "down" ),
+            i18n("Keep &Below Others")+'\t'+keys->shortcut("Window Below Other Windows").seq(0).toString(), Options::KeepBelowOp );
+        advanced_popup->insertItem( SmallIconSet( "window_fullscreen" ),
+            i18n("&Fullscreen")+'\t'+keys->shortcut("Window Fullscreen").seq(0).toString(), Options::FullScreenOp );
+        advanced_popup->insertItem( i18n("&No Border")+'\t'+keys->shortcut("Window No Border").seq(0).toString(), Options::NoBorderOp );
         advanced_popup->insertItem( SmallIconSet( "filesave" ), i18n("&Special Window Settings..."), Options::WindowRulesOp );
 
         popup->insertItem(i18n("Ad&vanced"), advanced_popup );
@@ -194,6 +197,10 @@
     cutWalkThroughWindowsReverse = keys->shortcut("Walk Through Windows (Reverse)");
 
     keys->updateConnections();
+    
+    delete popup;
+    popup = NULL; // so that it's recreated next time
+    desk_popup = NULL;
     }
 
 
diff -uNr kdebase-3.2.91/kwin/utils.h kdebase-3.2.91-new/kwin/utils.h
--- kdebase-3.2.91/kwin/utils.h	2004-06-26 09:24:03.000000000 -0400
+++ kdebase-3.2.91-new/kwin/utils.h	2004-07-02 09:03:46.000000000 -0400
@@ -161,7 +161,7 @@
     };
 
 
-static QCString getStringProperty(WId w, Atom prop, char separator=0);
+QCString getStringProperty(WId w, Atom prop, char separator=0);
 void updateXTime();
 void grabXServer();
 void ungrabXServer();
diff -uNr kdebase-3.2.91/libkonq/Makefile.am kdebase-3.2.91-new/libkonq/Makefile.am
--- kdebase-3.2.91/libkonq/Makefile.am	2004-05-22 16:27:06.000000000 -0400
+++ kdebase-3.2.91-new/libkonq/Makefile.am	2004-07-02 08:57:17.000000000 -0400
@@ -21,7 +21,7 @@
 
 lib_LTLIBRARIES = libkonq.la
 libkonq_la_LDFLAGS = $(all_libraries) -version-info 6:0:2 -no-undefined
-libkonq_la_LIBADD = $(LIB_KPARTS)
+libkonq_la_LIBADD = $(LIB_KPARTS) $(LIBZ)
 
 libkonq_la_SOURCES = konq_popupmenu.cc knewmenu.cc \
    konq_xmlguiclient.cc\
diff -uNr kdebase-3.2.91/startkde kdebase-3.2.91-new/startkde
--- kdebase-3.2.91/startkde	2004-05-22 16:26:34.000000000 -0400
+++ kdebase-3.2.91-new/startkde	2004-07-02 08:57:17.000000000 -0400
@@ -27,6 +27,8 @@
   esac
 fi
 
+source "@PREFIX@/bin/init.sh"
+
 # Boot sequence:
 #
 # kdeinit is used to fork off processes which improves memory usage
@@ -87,6 +89,11 @@
 # Where X is the original font database that was set up before this script
 # runs.
 
+/usr/X11R6/bin/xmodmap -e "keycode 74 = Right"
+/usr/X11R6/bin/xmodmap -e "keycode 78 = Left"
+/usr/X11R6/bin/xmodmap -e "keycode 80 = Down"
+/usr/X11R6/bin/xmodmap -e "keycode 85 = Up"
+
 usr_odir=$HOME/.fonts/kde-override
 usr_fdir=$HOME/.fonts
 
@@ -191,7 +198,7 @@
     done
     dcopserver_shutdown
     # shutdown will also make kwin quit, give it time to do so
-    sleep 1
+    sleep 5
 fi
 
 # the splashscreen and progress indicator
@@ -215,7 +222,29 @@
 # We only check for 255 which means that the ksmserver process could not be 
 # started, any problems thereafter, e.g. ksmserver failing to initialize, 
 # will remain undetected.
-test -n "$KDEWM" && KDEWM="--windowmanager $KDEWM"
+
+USE_PROXY=0
+if test -n "$KDEWM"; then
+	if test -z `expr "$KDEWM" : '\(.*quartz-wm.*\)'`; then
+		USE_PROXY=1
+	fi
+	KDEWM="--windowmanager $KDEWM"
+else
+	KDEWM="--windowmanager kwin"
+	USE_PROXY=1
+fi
+
+if test "$USE_PROXY" = "1"; then
+	if test -x /usr/X11R6/bin/quartz-wm; then
+		/usr/X11R6/bin/quartz-wm --only-proxy >/dev/null 2>&1 &
+	else
+		AUTOCUTSEL=`which autocutsel 2>/dev/null`
+		if test -n "$AUTOCUTSEL" && test -x "$AUTOCUTSEL"; then
+			$AUTOCUTSEL >/dev/null 2>&1 &
+		fi
+	fi
+fi
+
 kwrapper ksmserver $KDEWM 
 if test $? -eq 255; then
   # Startup error
diff -uNr kdebase-3.2.91/startkde.orig kdebase-3.2.91-new/startkde.orig
--- kdebase-3.2.91/startkde.orig	1969-12-31 19:00:00.000000000 -0500
+++ kdebase-3.2.91-new/startkde.orig	2004-05-22 16:26:34.000000000 -0400
@@ -0,0 +1,242 @@
+#!/bin/sh
+#
+#  DEFAULT KDE STARTUP SCRIPT ( KDE-3.3 )
+#
+
+# When the X server dies we get a HUP signal from xinit. We must ignore it
+# because we still need to do some cleanup.
+trap 'echo GOT SIGHUP' HUP
+
+# Check if a KDE session already is running
+dcop kdesktop KDesktopIface isRunning >/dev/null 2>/dev/null
+if [ $? -eq 0 ]; then
+	echo "ERROR: KDE seems to be already running on this display."
+	xmessage -geometry 500x100 "ERROR: KDE seems to be already running on this display." > /dev/null 2>/dev/null
+	exit 1
+fi
+
+# we have to unset this for Darwin since it will screw up KDE's dynamic-loading
+unset DYLD_FORCE_FLAT_NAMESPACE
+
+# in case we have been started with full pathname spec without being in PATH
+bindir=`echo "$0" | sed -n 's,^\(/.*\)/[^/][^/]*$,\1,p'`
+if [ -n "$bindir" ]; then
+  case $PATH in
+    $bindir|$bindir:*|*:$bindir|*:$bindir:*) ;;
+    *) PATH=$bindir:$PATH; export PATH;;
+  esac
+fi
+
+# Boot sequence:
+#
+# kdeinit is used to fork off processes which improves memory usage
+# and startup time.
+#
+# * kdeinit starts the dcopserver and klauncher first.
+# * Then kded is started. kded is responsible for keeping the sycoca
+#   database up to date. When an up to date database is present it goes
+#   into the background and the startup continues.
+# * Then kdeinit starts kcminit. kcminit performs initialisation of
+#   certain devices according to the user's settings
+#
+# * Then ksmserver is started which in turn starts
+#   1) the window manager (kwin)
+#   2) everything in $KDEDIR/share/autostart (kdesktop, kicker, etc.)
+#   3) the rest of the session.
+
+# The user's personal KDE directory is usually ~/.kde, but this setting
+# may be overridden by setting KDEHOME.
+
+kdehome=$HOME/.kde
+test -n "$KDEHOME" && kdehome=`echo "$KDEHOME"|sed "s,^~/,$HOME/,"`
+
+# Source scripts found in <localprefix>/env/*.sh and <prefixes>/env/*.sh
+# (where <localprefix> is $KDEHOME or ~/.kde, and <prefixes> is where KDE is installed)
+#
+# This is where you can define environment variables that will be available to
+# all KDE programs, so this is where you can run agents using e.g. eval `ssh-agent`
+# or eval `gpg-agent --daemon`.
+# Note: if you do that, you should also put "ssh-agent -k" as a shutdown script
+#
+# (see end of this file).
+# For anything else (that doesn't set env vars, or that needs a window manager),
+# better use the Autostart folder.
+
+exepath=`kde-config --path exe`
+
+for prefix in `echo "$exepath" | sed -e 's^/bin/^/env/^g;s^:^ ^g'`; do
+  for file in "$prefix"*.sh; do
+    test -r "$file" && . "$file"
+  done
+done
+
+# Activate the kde font directories.
+#
+# There are 4 directories that may be used for supplying fonts for KDE.
+#
+# There are two system directories. These belong to the administrator.
+# There are two user directories, where the user may add her own fonts.
+#
+# The 'override' versions are for fonts that should come first in the list,
+# i.e. if you have a font in your 'override' directory, it will be used in
+# preference to any other.
+#
+# The preference order looks like this:
+# user override, system override, X, user, system
+#
+# Where X is the original font database that was set up before this script
+# runs.
+
+usr_odir=$HOME/.fonts/kde-override
+usr_fdir=$HOME/.fonts
+
+# Add any user-installed font directories to the X font path
+kde_fontpaths=$usr_fdir/fontpaths
+do_usr_fdir=1
+do_usr_odir=1
+if test -r "$kde_fontpaths" ; then
+    savifs=$IFS
+    IFS="
+"
+    for fpath in `grep -v '^[ 	]*#' < "$kde_fontpaths"` ; do
+        rfpath=`echo $fpath | sed "s:^~:$HOME:g"`
+        if test -s "$rfpath"/fonts.dir; then
+            xset fp+ "$rfpath"
+            if test "$rfpath" = "$usr_fdir"; then
+                do_usr_fdir=0
+            fi
+            if test "$rfpath" = "$usr_odir"; then
+                do_usr_odir=0
+            fi
+        fi
+    done
+    IFS=$savifs
+fi
+
+if test -n "$KDEDIRS"; then
+  kdedirs_first=`echo "$KDEDIRS"|sed -e 's/:.*//'`
+  sys_odir=$kdedirs_first/share/fonts/override
+  sys_fdir=$kdedirs_first/share/fonts
+else
+  sys_odir=$KDEDIR/share/fonts/override
+  sys_fdir=$KDEDIR/share/fonts
+fi
+
+# We run mkfontdir on the user's font dirs (if we have permission) to pick
+# up any new fonts they may have installed. If mkfontdir fails, we still
+# add the user's dirs to the font path, as they might simply have been made
+# read-only by the administrator, for whatever reason.
+
+# Only do usr_fdir and usr_odir if they are *not* listed in fontpaths
+test -d "$sys_odir" && xset +fp "$sys_odir"
+test $do_usr_odir -eq 1 && test -d "$usr_odir" && (mkfontdir "$usr_odir" ; xset +fp "$usr_odir")
+test $do_usr_fdir -eq 1 && test -d "$usr_fdir" && (mkfontdir "$usr_fdir" ; xset fp+ "$usr_fdir")
+test -d "$sys_fdir" && xset fp+ "$sys_fdir"
+
+# Ask X11 to rebuild its font list.
+xset fp rehash
+
+# if the user has overwritten fonts, the cursor font may be different now
+# so don't move this up
+
+# Set a left cursor instead of the standard X11 "X" cursor, since I've heard
+# from some users that they're confused and don't know what to do. This is
+# especially necessary on slow machines, where starting KDE takes one or two
+# minutes until anything appears on the screen.
+#
+# Set the background to plain grey.
+# The standard X background is nasty, causing moire effects and exploding
+# people's heads. We use colours from the standard KDE palette for those with
+# palettised displays.
+
+test "$XDM_MANAGED" || bkg="-solid #C0C0C0"
+xsetroot -cursor_name left_ptr $bkg
+
+# Get Ghostscript to look into user's KDE fonts dir for additional Fontmap
+if test -n "$GS_LIB" ; then
+    GS_LIB=$usr_fdir:$GS_LIB
+    export GS_LIB
+else
+    GS_LIB=$usr_fdir
+    export GS_LIB
+fi
+
+# Link "tmp" resource to directory in /tmp
+# Creates a directory /tmp/kde-$USER and links $KDEHOME/tmp-$HOSTNAME to it.
+lnusertemp tmp >/dev/null
+
+# Link "socket" resource to directory in /tmp
+# Creates a directory /tmp/ksocket-$USER and links $KDEHOME/socket-$HOSTNAME to it.
+lnusertemp socket >/dev/null
+
+# Link "cache" resource to directory in /var/tmp
+# Creates a directory /var/tmp/kdecache-$USER and links $KDEHOME/cache-$HOSTNAME to it.
+lnusertemp cache >/dev/null
+
+# In case of dcop sockets left by a previous session, cleanup
+dcopserver_shutdown
+
+echo 'startkde: Starting up...'  1>&2
+
+# run KPersonalizer before the session, if this is the first login
+if kreadconfig --file kpersonalizerrc --group General --key FirstLogin --default true --type bool; then
+    # start only dcopserver, don't start whole kdeinit (takes too long)
+    echo 'startkde: Running kpersonalizer...'  1>&2
+    dcopserver
+    kwin --lock &
+    kpersonalizer --before-session
+    # handle kpersonalizer restarts (language change)
+    while test $? -eq 1; do
+        kpersonalizer --r --before-session
+    done
+    dcopserver_shutdown
+    # shutdown will also make kwin quit, give it time to do so
+    sleep 1
+fi
+
+# the splashscreen and progress indicator
+ksplash --nodcop
+
+# We set LD_BIND_NOW to increase the efficiency of kdeinit.
+# kdeinit unsets this variable before loading applications.
+LD_BIND_NOW=true kdeinit +kcminit +knotify
+if test $? -ne 0; then
+  # Startup error
+  echo 'startkde: Could not start kdeinit. Check your installation.'  1>&2
+  xmessage -geometry 500x100 "Could not start kdeinit. Check your installation."
+fi
+
+# finally, give the session control to the session manager
+# if the KDEWM environment variable has been set, then it will be used as KDE's
+# window manager instead of kwin.
+# if KDEWM is not set, ksmserver will ensure kwin is started.
+# kwrapper is used to reduce startup time and memory usage
+# kwrapper does not return usefull error codes such as the exit code of ksmserver.
+# We only check for 255 which means that the ksmserver process could not be 
+# started, any problems thereafter, e.g. ksmserver failing to initialize, 
+# will remain undetected.
+test -n "$KDEWM" && KDEWM="--windowmanager $KDEWM"
+kwrapper ksmserver $KDEWM 
+if test $? -eq 255; then
+  # Startup error
+  echo 'startkde: Could not start ksmserver. Check your installation.'  1>&2
+  xmessage -geometry 500x100 "Could not start ksmserver. Check your installation."
+fi
+
+echo 'startkde: Shutting down...'  1>&2
+
+# Clean up
+kdeinit_shutdown
+dcopserver_shutdown
+artsshell -q terminate
+
+echo 'startkde: Running shutdown scripts...'  1>&2
+
+# Run scripts found in $KDEDIRS/shutdown
+for prefix in `echo "$exepath" | sed -e 's^/bin/^/shutdown/^g;s^:^ ^g'`; do
+  for file in `ls "$prefix" 2> /dev/null | egrep -v '(~|\.bak)$'`; do
+    test -x "$prefix$file" && "$prefix$file"
+  done
+done
+
+echo 'startkde: Done.'  1>&2
