diff -Naur evolution-1.5.92.1-old/po/Makefile.in.in evolution-1.5.92.1/po/Makefile.in.in
--- evolution-1.5.92.1-old/po/Makefile.in.in	Thu Aug 12 14:04:24 2004
+++ evolution-1.5.92.1/po/Makefile.in.in	Thu Aug 12 14:10:54 2004
@@ -28,7 +28,7 @@
 exec_prefix = @exec_prefix@
 datadir = @datadir@
 libdir = @libdir@
-localedir = $(libdir)/locale
+localedir = $(datadir)/locale
 gnulocaledir = $(datadir)/locale
 gettextsrcdir = $(datadir)/glib-2.0/gettext/po
 subdir = po
