diff -ur scribus-0.9.8_ori/scribus/libabout/Makefile.in scribus-0.9.8_corr/scribus/libabout/Makefile.in
--- scribus-0.9.8_ori/scribus/libabout/Makefile.in	Wed Mar 26 19:06:08 2003
+++ scribus-0.9.8_corr/scribus/libabout/Makefile.in	Fri Mar 28 08:45:56 2003
@@ -186,11 +186,11 @@
 target_vendor = @target_vendor@
 x_includes = @x_includes@
 x_libraries = @x_libraries@
-lib_LTLIBRARIES = libabout.la
+lib_LTLIBRARIES = libabout.so
 
 INCLUDES = $(all_includes)
 
-libabout_la_LDFLAGS = -version-info 0:0:0
+libabout_la_LDFLAGS = -bundle -flat_namespace -undefined suppress
 
 #>- libabout_la_METASOURCES = AUTO
 
@@ -199,7 +199,7 @@
 
 EXTRA_DIST = about.cpp about.h 
 
-AM_LDFLAGS = -s  
+AM_LDFLAGS = 
 subdir = scribus/libabout
 mkinstalldirs = $(SHELL) $(top_srcdir)/admin/mkinstalldirs
 CONFIG_HEADER = $(top_builddir)/config.h
@@ -281,7 +281,7 @@
 	echo "rm -f \"$${dir}/so_locations\""; \
 	rm -f "$${dir}/so_locations"; \
 	done
-libabout.la: $(libabout_la_OBJECTS) $(libabout_la_DEPENDENCIES) 
+libabout.so: $(libabout_la_OBJECTS) $(libabout_la_DEPENDENCIES) 
 	$(CXXLINK) -rpath $(libdir) $(libabout_la_LDFLAGS) $(libabout_la_OBJECTS) $(libabout_la_LIBADD) $(LIBS)
 
 mostlyclean-compile:
diff -ur scribus-0.9.8_ori/scribus/libpdf/Makefile.in scribus-0.9.8_corr/scribus/libpdf/Makefile.in
--- scribus-0.9.8_ori/scribus/libpdf/Makefile.in	Wed Mar 26 19:06:08 2003
+++ scribus-0.9.8_corr/scribus/libpdf/Makefile.in	Fri Mar 28 08:47:11 2003
@@ -187,11 +187,11 @@
 x_includes = @x_includes@
 x_libraries = @x_libraries@
 AM_CPPFLAGS = -I$(top_builddir)/scribus
-lib_LTLIBRARIES = libpdf.la
+lib_LTLIBRARIES = libpdf.so
 
 INCLUDES = $(all_includes)
 
-libpdf_la_LDFLAGS = -version-info 0:0:0
+libpdf_la_LDFLAGS = -bundle -flat_namespace -undefined suppress
 
 #>- libpdf_la_METASOURCES = AUTO
 
@@ -200,7 +200,7 @@
 
 EXTRA_DIST = pdflib.cpp pdflib.h 
 
-AM_LDFLAGS = -s 
+AM_LDFLAGS = 
 subdir = scribus/libpdf
 mkinstalldirs = $(SHELL) $(top_srcdir)/admin/mkinstalldirs
 CONFIG_HEADER = $(top_builddir)/config.h
@@ -282,7 +282,7 @@
 	echo "rm -f \"$${dir}/so_locations\""; \
 	rm -f "$${dir}/so_locations"; \
 	done
-libpdf.la: $(libpdf_la_OBJECTS) $(libpdf_la_DEPENDENCIES) 
+libpdf.so: $(libpdf_la_OBJECTS) $(libpdf_la_DEPENDENCIES) 
 	$(CXXLINK) -rpath $(libdir) $(libpdf_la_LDFLAGS) $(libpdf_la_OBJECTS) $(libpdf_la_LIBADD) $(LIBS)
 
 mostlyclean-compile:
diff -ur scribus-0.9.8_ori/scribus/libpostscript/Makefile.in scribus-0.9.8_corr/scribus/libpostscript/Makefile.in
--- scribus-0.9.8_ori/scribus/libpostscript/Makefile.in	Wed Mar 26 19:06:08 2003
+++ scribus-0.9.8_corr/scribus/libpostscript/Makefile.in	Fri Mar 28 08:47:47 2003
@@ -187,11 +187,11 @@
 x_includes = @x_includes@
 x_libraries = @x_libraries@
 AM_CPPFLAGS = -I$(top_builddir)/scribus
-lib_LTLIBRARIES = libpostscript.la
+lib_LTLIBRARIES = libpostscript.so
 
 INCLUDES = $(all_includes)
 
-libpostscript_la_LDFLAGS = -version-info 0:0:0
+libpostscript_la_LDFLAGS = -bundle -flat_namespace -undefined suppress
 
 #>- libpostscript_la_METASOURCES = AUTO
 
@@ -200,7 +200,7 @@
 
 EXTRA_DIST = pslib.cpp pslib.h 
 
-AM_LDFLAGS = -s 
+AM_LDFLAGS =  
 subdir = scribus/libpostscript
 mkinstalldirs = $(SHELL) $(top_srcdir)/admin/mkinstalldirs
 CONFIG_HEADER = $(top_builddir)/config.h
@@ -282,7 +282,7 @@
 	echo "rm -f \"$${dir}/so_locations\""; \
 	rm -f "$${dir}/so_locations"; \
 	done
-libpostscript.la: $(libpostscript_la_OBJECTS) $(libpostscript_la_DEPENDENCIES) 
+libpostscript.so: $(libpostscript_la_OBJECTS) $(libpostscript_la_DEPENDENCIES) 
 	$(CXXLINK) -rpath $(libdir) $(libpostscript_la_LDFLAGS) $(libpostscript_la_OBJECTS) $(libpostscript_la_LIBADD) $(LIBS)
 
 mostlyclean-compile:
diff -ur scribus-0.9.8_ori/scribus/libprefs/Makefile.in scribus-0.9.8_corr/scribus/libprefs/Makefile.in
--- scribus-0.9.8_ori/scribus/libprefs/Makefile.in	Wed Mar 26 19:06:08 2003
+++ scribus-0.9.8_corr/scribus/libprefs/Makefile.in	Fri Mar 28 08:48:39 2003
@@ -187,11 +187,11 @@
 x_includes = @x_includes@
 x_libraries = @x_libraries@
 AM_CPPFLAGS = -I$(top_builddir)/scribus
-lib_LTLIBRARIES = libpreferences.la
+lib_LTLIBRARIES = libpreferences.so
 
 INCLUDES = $(all_includes)
 
-libpreferences_la_LDFLAGS = -version-info 0:0:0
+libpreferences_la_LDFLAGS = -bundle -flat_namespace -undefined suppress
 
 #>- libpreferences_la_METASOURCES = AUTO
 
@@ -199,7 +199,7 @@
 KDE_OPTIONS = qtonly
 
 EXTRA_DIST = prefs.cpp prefs.h 
-AM_LDFLAGS = -s 
+AM_LDFLAGS =  
 subdir = scribus/libprefs
 mkinstalldirs = $(SHELL) $(top_srcdir)/admin/mkinstalldirs
 CONFIG_HEADER = $(top_builddir)/config.h
@@ -281,7 +281,7 @@
 	echo "rm -f \"$${dir}/so_locations\""; \
 	rm -f "$${dir}/so_locations"; \
 	done
-libpreferences.la: $(libpreferences_la_OBJECTS) $(libpreferences_la_DEPENDENCIES) 
+libpreferences.so: $(libpreferences_la_OBJECTS) $(libpreferences_la_DEPENDENCIES) 
 	$(CXXLINK) -rpath $(libdir) $(libpreferences_la_LDFLAGS) $(libpreferences_la_OBJECTS) $(libpreferences_la_LIBADD) $(LIBS)
 
 mostlyclean-compile:
diff -ur scribus-0.9.8_ori/scribus/plugins/libchar/Makefile.in scribus-0.9.8_corr/scribus/plugins/libchar/Makefile.in
--- scribus-0.9.8_ori/scribus/plugins/libchar/Makefile.in	Wed Mar 26 19:06:08 2003
+++ scribus-0.9.8_corr/scribus/plugins/libchar/Makefile.in	Fri Mar 28 08:49:10 2003
@@ -187,11 +187,11 @@
 x_includes = @x_includes@
 x_libraries = @x_libraries@
 AM_CPPFLAGS = -I$(top_builddir)/scribus
-lib_LTLIBRARIES = libcharselect.la
+lib_LTLIBRARIES = libcharselect.so
 
 INCLUDES = $(all_includes)
 
-libcharselect_la_LDFLAGS = -version-info 0:0:0
+libcharselect_la_LDFLAGS = -bundle -flat_namespace -undefined suppress
 
 #>- libcharselect_la_METASOURCES = AUTO
 
@@ -200,7 +200,7 @@
 
 EXTRA_DIST = charselect.cpp charselect.h 
 
-AM_LDFLAGS = -s 
+AM_LDFLAGS =  
 subdir = scribus/plugins/libchar
 mkinstalldirs = $(SHELL) $(top_srcdir)/admin/mkinstalldirs
 CONFIG_HEADER = $(top_builddir)/config.h
@@ -282,7 +282,7 @@
 	echo "rm -f \"$${dir}/so_locations\""; \
 	rm -f "$${dir}/so_locations"; \
 	done
-libcharselect.la: $(libcharselect_la_OBJECTS) $(libcharselect_la_DEPENDENCIES) 
+libcharselect.so: $(libcharselect_la_OBJECTS) $(libcharselect_la_DEPENDENCIES) 
 	$(CXXLINK) -rpath $(libdir) $(libcharselect_la_LDFLAGS) $(libcharselect_la_OBJECTS) $(libcharselect_la_LIBADD) $(LIBS)
 
 mostlyclean-compile:
