--- TeXmacs-1.0.1.7-src/configure~	Sun Mar  2 20:28:24 2003
+++ TeXmacs-1.0.1.7-src/configure	Mon Mar  3 23:55:08 2003
@@ -1076,7 +1076,7 @@
 
 echo $ac_n "checking version of guile""... $ac_c" 1>&6
 echo "configure:1079: checking version of guile" >&5
-GUILE_VERSION=`guile --version | grep 'Guile \([0-9]\)' | sed 's/Guile \([0-9]*\)/\1/' -`
+GUILE_VERSION=`guile --version | grep 'Guile \([0-9]\)' | sed 's/Guile \([0-9]*\)/\1/'`
 echo "$ac_t""`echo $GUILE_VERSION`" 1>&6
 
 #--------------------------------------------------------------------
