--- TeXmacs-1.0.1.6-src/src/Guile/glue.hh~	Sun Jan 26 13:28:54 2003
+++ TeXmacs-1.0.1.6-src/src/Guile/glue.hh	Wed Feb 26 20:14:52 2003
@@ -27,13 +27,4 @@
 #define scm_is_widget(t) \
   (SCM_NIMP (t) && (((long) SCM_CAR (t)) == widget_tag))
 
-class make_scheme_widget_rep: public make_widget_rep {
-  // use "the_server" for scheme evaluation
-  scheme_tree p;
-public:
-  make_scheme_widget_rep (scheme_tree p2): p(p2) {}
-  ostream& print (ostream& out) { return out << p; }
-  widget get_widget (display dis);
-};
-
 #endif // defined GLUE_HH
