--- TeXmacs-1.0.1.13-src/misc/scripts/texmacs.in        Sun Jan 26 13:28:54 2003
+++ TeXmacs-1.0.1.13-src.corr/misc/scripts/texmacs.in   Sat May 17 13:46:22 2003
@@ -1,5 +1,7 @@
 #!/bin/sh
 
+ulimit -s 8192
+
 if [ -z "$TEXMACS_PATH" ] ; then
   TEXMACS_PATH=@tmdata@
   export TEXMACS_PATH
@@ -11,6 +13,9 @@
   TEXMACS_BIN_PATH="$TEXMACS_PATH"
   export TEXMACS_BIN_PATH
 fi
+
+export GUILE_LOAD_PATH="$TEXMACS_PATH/progs:GUILEPATH"
+PATH="GUILEPATH/scripts/binoverride:$PATH"
 
 PATH="$TEXMACS_BIN_PATH/bin:$PATH"
 @CONFIG_LIB_PATH@="$TEXMACS_BIN_PATH/lib${@CONFIG_LIB_PATH@+":$@CONFIG_LIB_PATH
@"}"
