--- TeXmacs-1.0.1.6-src/src/Guile/glue.hh~	Sun Jan 26 13:28:54 2003
+++ TeXmacs-1.0.1.6-src/src/Guile/glue.hh	Wed Feb 26 20:14:52 2003
@@ -27,13 +27,4 @@
 #define scm_is_widget(t) \
   (SCM_NIMP (t) && (((long) SCM_CAR (t)) == widget_tag))
 
-class make_scheme_widget_rep: public make_widget_rep {
-  // use "the_server" for scheme evaluation
-  scheme_tree p;
-public:
-  make_scheme_widget_rep (scheme_tree p2): p(p2) {}
-  ostream& print (ostream& out) { return out << p; }
-  widget get_widget (display dis);
-};
-
 #endif // defined GLUE_HH
--- TeXmacs-1.0.1.6-src/TeXmacs-1.0.1.6/progs/extensions/gui/menu-define.scm~	Mon Feb 24 17:44:12 2003
+++ TeXmacs-1.0.1.6-src/TeXmacs-1.0.1.6/progs/extensions/gui/menu-define.scm	Thu Feb 27 11:13:11 2003
@@ -117,8 +117,8 @@
 	    (else (menu-format-error p)))))
 
 (define-table menu-pre-table
-  (group (:string) ,id)
-  (symbol (:string :*) ,id)
+  (group (:string?) ,id)
+  (symbol (:string? :*) ,id)
   (link (:1) ,id)
   (horizontal (:*)
     ,(lambda (p) `(horizontal ,@(menu-pre-list (cdr p)))))
