diff -Naur cvs2svn-1.0.0-rc4-orig/svntest/main.py cvs2svn-1.0.0-rc4/svntest/main.py
--- cvs2svn-1.0.0-rc4-orig/svntest/main.py	2004-06-23 06:43:38.000000000 +0200
+++ cvs2svn-1.0.0-rc4/svntest/main.py	2004-08-06 11:57:04.000000000 +0200
@@ -516,7 +516,10 @@
 
   if testnum is None:
     for n in range(1, len(test_list)):
-      if run_one_test(n, test_list):
+      test_exit_code = run_one_test(n, test_list)
+      if test_exit_code == 2:
+        test_exit_code = 0  # ignore skipped test cases
+      if test_exit_code:
         exit_code = 1
   else:
     exit_code = run_one_test(testnum, test_list)
