diff -Naur auctex-11.55.orig/fink/auctex auctex-11.55/fink/auctex
--- auctex-11.55.orig/fink/auctex	Thu Jan  1 09:00:00 1970
+++ auctex-11.55/fink/auctex	Fri Mar 18 05:54:10 2005
@@ -0,0 +1,43 @@
+Document: auctex
+Title: AUCTeX: An integrated environment for writing TeX/LaTeX documents
+Author: Kresten Krab Thorup, Per Abrahamsen, David Kastrup
+Abstract: AUCTeX is a comprehensive customizable integrated environment for
+ writing input files for TeX/LaTeX using GNU Emacs.
+ .
+ AUCTeX lets you run TeX/LaTeX and other LaTeX-related tools, such as a output
+ filters or post processor from inside Emacs.  Especially `running LaTeX' is
+ interesting, as AUCTeX lets you browse through the errors TeX reported, while
+ it moves the cursor directly to the reported error, and displays some
+ documentation for that particular error.  This will even work when the
+ document is spread over several files.
+ .
+ AUCTeX automatically indents your `LaTeX-source', not only as you write it --
+ you can also let it indent and format an entire document.  It has a special
+ outline feature, which can greatly help you `getting an overview' of a
+ document.
+ .
+ Apart from these special features, AUCTeX provides a large range of handy
+ Emacs macros, which in several different ways can help you write your LaTeX
+ documents fast and painlessly.
+ .
+  All features of AUCTeX are documented using the GNU Emacs online
+ documentation system.  That is, documentation for any command is just a key
+ click away!
+ .
+ AUCTeX is written entirely in Emacs-Lisp, and hence you can easily add new
+ features for your own needs.  It was not made as part of any particular
+ employment or project (apart from the AUCTeX project itself).  AUCTeX is
+ distributed under the `GNU Emacs General Public License' and may therefore
+ almost freely be copied and redistributed.
+Section: Apps/Editors
+
+Format: texinfo
+Files: @PREFIX@/share/doc/auctex/src/*.texi
+
+Format: info
+Index: @PREFIX@/share/info/auctex.info
+Files: @PREFIX@/share/info/auctex.info*
+
+Format: HTML
+Index: @PREFIX@/share/doc/auctex/HTML/index.html
+Files: @PREFIX@/share/doc/auctex/HTML/*.html
diff -Naur auctex-11.55.orig/fink/auctex-cron auctex-11.55/fink/auctex-cron
--- auctex-11.55.orig/fink/auctex-cron	Thu Jan  1 09:00:00 1970
+++ auctex-11.55/fink/auctex-cron	Fri Mar 18 05:42:52 2005
@@ -0,0 +1,27 @@
+#!/bin/bash -e
+#
+# Update AUCTeX auto-loads.
+# Installed by the Fink auctex package.
+#
+# Copyright (C) 1997, 98, 99, 2000, 01, 02, 03, 04 by Davide Giovanni
+# Maria Salvetti.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License as published by the
+# Free Software Foundation; either version 2 of the License, or (at your
+# option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to: The Free Software Foundation, Inc.,
+# 59 Temple Place, Suite 330, Boston, MA 02111-1307, USA.
+
+set -o posix
+
+test -x @PREFIX@/sbin/update-auctex-elisp && update-auctex-elisp &>/dev/null
+
+exit 0
diff -Naur auctex-11.55.orig/fink/auctex-install auctex-11.55/fink/auctex-install
--- auctex-11.55.orig/fink/auctex-install	Thu Jan  1 09:00:00 1970
+++ auctex-11.55/fink/auctex-install	Thu Mar 24 16:28:11 2005
@@ -0,0 +1,108 @@
+#!/bin/bash -e
+#
+# elisp install file for Fink auctex package.
+#
+# Copyright (C) 1997, 98, 99, 2000, 01, 02, 03, 04 by Davide Giovanni
+# Maria Salvetti.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License as published by the
+# Free Software Foundation; either version 2 of the License, or (at your
+# option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to: The Free Software Foundation, Inc.,
+# 59 Temple Place, Suite 330, Boston, MA 02111-1307, USA.
+
+set -o posix
+
+source @PREFIX@/share/debconf/confmodule
+db_version 2.0
+db_get auctex/logfile || true; _db_logfile=${RET}
+db_get auctex/default || true; _db_default=${RET}
+db_get auctex/doauto || true; _db_doauto=${RET}
+db_get auctex/doautofg || true; _db_doautofg=${RET}
+
+FLAVOR=${1}
+
+INSTALL="install -o root -g admin"
+INSTDIR="${INSTALL} -m 755 -d"
+
+do_install () {
+    echo >&2 -n "install/auctex: Setting up for ${FLAVOR}... "
+    cd @PREFIX@/share/emacs/site-lisp/auctex/; umask 0022;
+    ${INSTDIR} @PREFIX@/share/${FLAVOR}/site-lisp/auctex/
+    ./configure --disable-build-dir-test \
+	--prefix=@PREFIX@/ \
+	--with-emacs=${FLAVOR} \
+	--with-lispdir=@PREFIX@/share/${FLAVOR}/site-lisp/ \
+	--with-auto-dir=@PREFIX@/var/lib/auctex/${FLAVOR} \
+	>> @PREFIX@/share/${FLAVOR}/site-lisp/auctex//CompilationLog 2>&1
+    make INSTALL="${INSTALL}" MKINSTALLDIRS="${INSTDIR}" \
+	install-lisp install-contrib \
+	>> @PREFIX@/share/${FLAVOR}/site-lisp/auctex//CompilationLog 2>&1
+    make distclean \
+	>> @PREFIX@/share/${FLAVOR}/site-lisp/auctex//CompilationLog 2>&1
+    find @PREFIX@/share/${FLAVOR}/site-lisp/auctex/ -type f -name \*.el \
+	-print0 | xargs --null rm -f dummy \
+	>> @PREFIX@/share/${FLAVOR}/site-lisp/auctex//CompilationLog 2>&1
+    gzip -9fq @PREFIX@/share/${FLAVOR}/site-lisp/auctex//CompilationLog
+    echo -e >@PREFIX@/etc/${FLAVOR}/site-start.d//50auctex.el \
+	";;; This file is automatically generated.\n"
+    echo -e >>@PREFIX@/etc/${FLAVOR}/site-start.d//50auctex.el \
+	"(if (and nil (fboundp 'fink-pkg-add-load-path-item))\n" \
+	"   (progn (fink-pkg-add-load-path-item \"@PREFIX@/share/${FLAVOR}/site-lisp/auctex/\")\n" \
+	"          (fink-pkg-add-load-path-item \"@PREFIX@/share/emacs/site-lisp/auctex/\"))\n" \
+	" (progn (add-to-list 'load-path \"@PREFIX@/share/${FLAVOR}/site-lisp/auctex/\" 'append)\n" \
+	"        (add-to-list 'load-path \"@PREFIX@/share/emacs/site-lisp/auctex/\" 'append)))"
+    if [ "${_db_default}" = "true" ]; then
+	echo -e >>@PREFIX@/etc/${FLAVOR}/site-start.d//50auctex.el \
+	"\n(require 'tex-site)"
+    fi
+    echo >&2 "done."
+    return 0
+}
+
+do_auto () {
+    case "${_db_doauto}" in
+	(Foreground)
+	case "${_db_doautofg}" in
+	    (Console)
+	    rm -f ${_db_logfile}
+	    @PREFIX@/sbin//update-auctex-elisp ${FLAVOR} ;;
+	    (File)
+	    echo >&2 -n "update-auctex-elisp: "
+	    echo >&2 "Further output will appear in: ${_db_logfile}."
+	    echo >&2 -n "auctex: "
+	    echo >&2 -n "Waiting for update-auctex-elisp to terminate... "
+	    @PREFIX@/sbin//update-auctex-elisp ${FLAVOR} >> ${_db_logfile} 2>&1
+	    echo >&2 "done." ;;
+	    (*) echo >&2 \
+		"${0##*/}: Debconf passed unknown value \`${_db_doautofg}'." ;;
+	esac ;;
+	(Background)
+	# Closing fd 3 is debconf important.
+						@PREFIX@/sbin//update-auctex-elisp ${FLAVOR} >> ${_db_logfile} 2>&1 3>&- &
+	echo >&2 -n "update-auctex-elisp[${!}]: "
+	echo >&2 "Further output will appear in: ${_db_logfile}." ;;
+	(None) rm -f ${_db_logfile} ;;
+	(*) echo >&2 \
+	    "${0##*/}: Debconf passed unknown value \`${_db_doauto}'." ;;
+    esac
+    return 0
+}
+
+case "${FLAVOR}" in
+    (emacs) : ;;
+    (emacs21|emacs21-carbon)
+        do_install ${FLAVOR}
+        do_auto ${FLAVOR} ;;
+    (*) echo >&2 "install/auctex: Ignoring emacsen flavor ${FLAVOR}." ;;
+esac
+
+exit 0
diff -Naur auctex-11.55.orig/fink/auctex-remove auctex-11.55/fink/auctex-remove
--- auctex-11.55.orig/fink/auctex-remove	Thu Jan  1 09:00:00 1970
+++ auctex-11.55/fink/auctex-remove	Fri Mar 18 06:44:30 2005
@@ -0,0 +1,57 @@
+#!/bin/bash -e
+#
+# elisp remove file for Fink auctex package.
+#
+# Copyright (C) 1997, 98, 99, 2000, 01, 02, 03, 04 by Davide Giovanni
+# Maria Salvetti.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License as published by the
+# Free Software Foundation; either version 2 of the License, or (at your
+# option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to: The Free Software Foundation, Inc.,
+# 59 Temple Place, Suite 330, Boston, MA 02111-1307, USA.
+
+set -o posix
+
+FLAVOR=${1}
+
+undo_install () {
+    echo >&2 -n "remove/auctex: Removing for ${FLAVOR}... "
+    rm -f @PREFIX@/etc/${FLAVOR}/site-start.d//50auctex.el
+    rm -f @PREFIX@/share/${FLAVOR}/site-lisp/auctex//../tex-site.el
+    rm -f @PREFIX@/share/${FLAVOR}/site-lisp/auctex//CompilationLog{,.gz}
+    rm -f @PREFIX@/share/${FLAVOR}/site-lisp/auctex//*.elc
+    rm -f @PREFIX@/share/${FLAVOR}/site-lisp/auctex//style//*.elc
+    rm -f @PREFIX@/share/${FLAVOR}/site-lisp/auctex//style//.nosearch
+    test -d @PREFIX@/share/${FLAVOR}/site-lisp/auctex//style/ && \
+	rmdir --ignore-fail-on-non-empty \
+	@PREFIX@/share/${FLAVOR}/site-lisp/auctex//style/
+    test -d @PREFIX@/share/${FLAVOR}/site-lisp/auctex/ && \
+	rmdir --ignore-fail-on-non-empty \
+	@PREFIX@/share/${FLAVOR}/site-lisp/auctex/
+    echo >&2 "done."
+    return 0
+}
+
+undo_auto () {
+    rm -rf @PREFIX@/var/lib/auctex//${FLAVOR}/
+    return 0
+}
+
+case "${FLAVOR}" in
+    (emacs) : ;;
+    (emacs21|emacs21-carbon)
+        undo_auto
+        undo_install ;;
+    (*) echo >&2 "remove/auctex: Ignoring emacsen flavor ${FLAVOR}." ;;
+esac
+
+exit 0
diff -Naur auctex-11.55.orig/fink/config auctex-11.55/fink/config
--- auctex-11.55.orig/fink/config	Thu Jan  1 09:00:00 1970
+++ auctex-11.55/fink/config	Fri Mar 18 05:09:39 2005
@@ -0,0 +1,53 @@
+#!/bin/bash -e
+#
+# config configuration file for Fink auctex package.
+#
+# Copyright (C) 1997, 98, 99, 2000, 01, 02, 03, 04 by Davide Giovanni
+# Maria Salvetti.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License as published by the
+# Free Software Foundation; either version 2 of the License, or (at your
+# option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to: The Free Software Foundation, Inc.,
+# 59 Temple Place, Suite 330, Boston, MA 02111-1307, USA.
+
+set -o posix
+
+source @PREFIX@/share/debconf/confmodule
+db_version 2.0
+
+case "${1}" in
+    (configure|reconfigure)
+    if egrep "\brequire[[:space:]]+[^)]*'tex-site\b" \
+	@PREFIX@/etc/emacs/site-start.el 2>/dev/null \
+	| egrep -v ";.*\brequire\b" &>/dev/null; then
+	db_set auctex/default false || true
+	db_text low auctex/alreadydefault || true
+    else
+	db_input medium auctex/default || true
+    fi
+    db_go || true
+    LOGFILE=$(mktemp /tmp/update-auctex-elisp.XXXXXXX)
+    db_register auctex/doauto auctex/logfile || true
+    db_set auctex/logfile ${LOGFILE} || true
+    db_subst auctex/doauto LOGFILE ${LOGFILE} || true
+    db_subst auctex/doautofg LOGFILE ${LOGFILE} || true
+    db_input low auctex/doauto || true
+    db_go || true
+    db_get auctex/doauto || true
+    if [ "${RET}" = "Foreground" ]; then
+			db_input low auctex/doautofg || true
+	db_go || true
+    fi ;;
+    (*) echo >&2 "${0##*/}: Called with unknown argument \`${1}'." ;;
+esac
+
+exit 0
diff -Naur auctex-11.55.orig/fink/postinst auctex-11.55/fink/postinst
--- auctex-11.55.orig/fink/postinst	Thu Jan  1 09:00:00 1970
+++ auctex-11.55/fink/postinst	Fri Mar 18 05:20:31 2005
@@ -0,0 +1,52 @@
+#!/bin/bash -e
+#
+# postinst configuration file for Fink auctex package.
+#
+# Copyright (C) 1997, 98, 99, 2000, 01, 02, 03, 04 by Davide Giovanni
+# Maria Salvetti.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License as published by the
+# Free Software Foundation; either version 2 of the License, or (at your
+# option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to: The Free Software Foundation, Inc.,
+# 59 Temple Place, Suite 330, Boston, MA 02111-1307, USA.
+
+set -o posix
+
+source @PREFIX@/share/debconf/confmodule
+db_version 2.0
+
+#---B-E-G-I-N---O-L-D---S-T-U-F-F---keep-it,-it-doesn't-hurt-------------------
+for i in @PREFIX@/lib/emacs/common/auctex/auto/ @PREFIX@/var/lib/emacs/auctex/; do
+    if test -d ${i}
+	then rm -rf ${i}
+    fi
+done
+for i in @PREFIX@/lib/emacs/common/auctex @PREFIX@/lib/emacs/common; do
+    if test -d ${i}
+	then find ${i%/*} -name ${i##*/} -type d -empty \
+	    | xargs --no-run-if-empty rmdir
+    fi
+done
+#---E-N-D---O-L-D---S-T-U-F-F--------------------------------------------------
+
+case "${1}" in
+    (abort-remove|abort-upgrade|configure)
+                    @PREFIX@/lib/emacsen-common/emacs-package-install auctex
+        test -x @PREFIX@/sbin/install-docs && install-docs --install @PREFIX@/share/doc//auctex
+        install-info --section "Document Preparation" "Document Preparation" --menuentry="AUCTeX" --description="An integrated environment for TeX and LaTeX" --quiet @PREFIX@/share/info//auctex.info
+    ;;
+    (abort-deconfigure)
+        : ;;
+    (*) echo >&2 "${0##*/}: Called with unknown argument \`${1}'." ;;
+esac
+
+exit 0
diff -Naur auctex-11.55.orig/fink/postrm auctex-11.55/fink/postrm
--- auctex-11.55.orig/fink/postrm	Thu Jan  1 09:00:00 1970
+++ auctex-11.55/fink/postrm	Fri Mar 18 05:24:17 2005
@@ -0,0 +1,34 @@
+#!/bin/bash -e
+#
+# postrm configuration file for Fink auctex package.
+#
+# Copyright (C) 1997, 98, 99, 2000, 01, 02, 03, 04 by Davide Giovanni
+# Maria Salvetti.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License as published by the
+# Free Software Foundation; either version 2 of the License, or (at your
+# option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to: The Free Software Foundation, Inc.,
+# 59 Temple Place, Suite 330, Boston, MA 02111-1307, USA.
+
+set -o posix
+
+case "${1}" in
+    (abort-install|abort-upgrade|failed-upgrade|remove|upgrade)
+                        : ;;
+    (disappear|purge)
+                if test -e @PREFIX@/share/debconf/confmodule;
+	then source @PREFIX@/share/debconf/confmodule; db_version 2.0; db_purge
+    fi ;;
+    (*) echo >&2 "${0##*/}: Called with unknown argument \`${1}'." ;;
+esac
+
+exit 0
diff -Naur auctex-11.55.orig/fink/preinst auctex-11.55/fink/preinst
--- auctex-11.55.orig/fink/preinst	Thu Jan  1 09:00:00 1970
+++ auctex-11.55/fink/preinst	Fri Mar 18 05:25:24 2005
@@ -0,0 +1,30 @@
+#!/bin/bash -e
+#
+# preinst configuration file for Fink auctex package.
+#
+# Copyright (C) 1997, 98, 99, 2000, 01, 02, 03, 04 by Davide Giovanni
+# Maria Salvetti.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License as published by the
+# Free Software Foundation; either version 2 of the License, or (at your
+# option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to: The Free Software Foundation, Inc.,
+# 59 Temple Place, Suite 330, Boston, MA 02111-1307, USA.
+
+set -o posix
+
+case "${1}" in
+    (abort-upgrade|install|upgrade)
+                : ;;
+    (*) echo >&2 "${0##*/}: Called with unknown argument \`${1}'." ;;
+esac
+
+exit 0
diff -Naur auctex-11.55.orig/fink/prerm auctex-11.55/fink/prerm
--- auctex-11.55.orig/fink/prerm	Thu Jan  1 09:00:00 1970
+++ auctex-11.55/fink/prerm	Fri Mar 18 05:27:11 2005
@@ -0,0 +1,36 @@
+#!/bin/bash -e
+#
+# postinst configuration file for Fink auctex package.
+#
+# Copyright (C) 1997, 98, 99, 2000, 01, 02, 03, 04 by Davide Giovanni
+# Maria Salvetti.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License as published by the
+# Free Software Foundation; either version 2 of the License, or (at your
+# option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to: The Free Software Foundation, Inc.,
+# 59 Temple Place, Suite 330, Boston, MA 02111-1307, USA.
+
+set -o posix
+
+case "${1}" in
+    (deconfigure)
+            : ;;
+    (remove|upgrade)
+                install-info --quiet --remove @PREFIX@/share/info//auctex.info
+        test -x @PREFIX@/sbin/install-docs && install-docs --remove auctex
+        @PREFIX@/lib/emacsen-common/emacs-package-remove auctex ;;
+    (failed-upgrade)
+        : ;;
+    (*) echo >&2 "${0##*/}: Called with unknown argument \`${1}'." ;;
+esac
+
+exit 0
diff -Naur auctex-11.55.orig/fink/templates auctex-11.55/fink/templates
--- auctex-11.55.orig/fink/templates	Thu Jan  1 09:00:00 1970
+++ auctex-11.55/fink/templates	Fri Mar 18 05:55:51 2005
@@ -0,0 +1,244 @@
+Template: auctex/default
+Type: boolean
+Default: true
+Description: Should AUCTeX be auto-loaded by default at your site?
+ Emacs has its own standard (La)TeX major modes, that will be globally (i.e.,
+ for all users on this site) overridden by AUCTeX if you answer yes to this
+ question.
+ .
+ This is generally a good thing, since AUCTeX is so much better; however you
+ may not want it to happen, and instead let single users at your site decide
+ by themselves if they should load this package.
+ .
+ If you answer no, people who desire to use it will have to put the string
+ "(require 'tex-site)" in their "~/.emacs" to load it.
+Description-it: Abilitare AUCTeX automaticamente su questa macchina?
+ Emacs normalmente ha i propri modi maggiori originali per (La)TeX; se si
+ risponde affermativamente a questa domanda, questi verranno globalmente
+ sostituiti da quelli di AUCTeX (sostituiti, cio.béË, per tutti gli utenti di
+ questa macchina).
+ .
+ In genere questa éË una buona cosa, perchéÈ i modi di AUCTeX sono molto
+ migliori degli originali; puéÚ darsi, tuttavia, che si preferisca lasciare a
+ ciascun singolo utente la decisione di far uso o meno di questo pacchetto.
+ .
+ Se la risposta saré‡ negativa, gli utenti che ne vorranno far uso dovranno
+ mettere manualmente la stringa "(require 'tex-site)" nel loro "~/.emacs" per
+ abilitare AUCTeX.
+Description-de: Soll AUCTeX systemweit automatisch geladen werden?
+ Emacs hat standardmé‰éﬂige (La)TeX-Modi, die global (also fé¸r alle Nutzer auf
+ Ihrem System) é¸berschrieben werden, wenn Sie diese Frage bejahen.
+ .
+ Das ist generell ein gutes Ding, da AUCTeX 'ne Ecke besser ist -- wie auch
+ immer, Sie kéˆnnten das nicht méˆgen und jedem Nutzer die Wahl lassen wollen,
+ ob sie das Paket einbinden.
+ .
+ Wenn Sie die Frage verneinen, mé¸ssen Nutzer, die AUCTeX nutzen wollen,
+ "(require 'tex-site)" in ihr "~/.emacs" einfé¸gen.
+Description-es: éøDebe cargarse AUCTeX por defecto en su sistema?
+ Emacs tiene sus modos (La)TeX esté·ndar que seré·n reemplazados globalmente
+ (para todos los usuarios del sistema) por AUCTeX si responde SéÌ a esta
+ pregunta.
+ .
+ Generalmente es una buena idea ya que AUCTeX es mucho mejor.  Sin embargo
+ puede que no le guste la idea y prefiera que los usuarios de su sistema
+ decidan por séÌ mismos si quieren cargar este paquete.
+ .
+ Si responde No la gente que quiera usarlo deberé· incluir la cadena
+ "(require 'tex-site)" en su "~/.emacs" para cargarlo.
+Description-pt_BR: AUCTeX deve ser auto-carregado por padr.Aé„o em seu sistema?
+ Emacs tem seus préÛprios modos (La)TeX padréıes, que seré„o cancelados
+ globalmente (isto éÈ, para todos os usué·rios deste sistema) pelo AUCTeX se
+ vocéÍ responder sim a esta questé„o.
+ .
+ Isto geralmente éÈ algo bom, jé· que o AUCTeX éÈ muito melhor; entretanto vocéÍ
+ pode né„o querer que isso aconteéÁa, e ao invéÈs disso deixar que os usué·rios de
+ seu sistema escolham por si préÛprios se devem carregar este pacote.
+ .
+ Se vocéÍ responder né„o, as pessoas que desejarem usé·-lo teré„o que colocar as
+ palavras "(require 'tex-site)" em seus "~/.emacs" para carregé·-lo.
+
+Template: auctex/alreadydefault
+Type: note
+Description: AUCTeX seems to be already auto-loaded at your site.  Good.
+ By inspection of the file "@PREFIX@/etc/emacs/site-start.el" AUCTeX seems to be
+ already auto-loaded by default at your site.
+ .
+ If this is not true, or if it's not what you want, please search that file
+ looking for "(require 'tex-site)" resembling forms and fix them; then run
+ "dpkg-reconfigure auctex" if needed (recommended).
+Description-it: AUCTeX sembra gi.bé‡ automaticamente abilitato su questa macchina.  Bene.
+ Controllando il file "@PREFIX@/etc/emacs/site-start.el", pare che AUCTeX sia gié‡ automaticamente
+ abilitato su questa macchina.
+ .
+ Se questo non éË vero, o se non éË quel che si desidera, si vadano a cercare in
+ quel file Lisp form del tipo "(require 'tex-site)" e li si correggano;
+ poi si faccia girare "dpkg-reconfigure auctex" per configurare di
+ nuovo AUCTeX.
+Description-de: AUCTeX scheint schon standardmé‰éﬂig eingebunden zu sein. Fein.
+ Nach Kontrolle der Datei "@PREFIX@/etc/emacs/site-start.el" scheint AUCTeX bereits
+ standardmé‰éﬂig auf Ihrem System eingebunden zu werden.
+ .
+ Wenn das nicht stimmt oder wenn das nicht das ist, was Sie wollen, dann
+ entfernen Sie bitte die "(require 'tex-site)"-Aufrufe in der genannten
+ Datei. Danach wird der Aufruf von "dpkg-reconfigure auctex" empfohlen.
+Description-es: Parece que AUCTeX ya se esté· cargando automé·ticamente. Bien.
+ Tras inspeccionar el fichero "@PREFIX@/etc/emacs/site-start.el" parece que AUCTeX
+ esta ya configurado para cargarse por defecto en su sistema.
+ .
+ Si no es verdad o si no lo quiere, por favor, busque en el fichero la frase
+ "(require 'tex-site)" y modiféÌquela.
+ DespuéÈs ejecute "dpkg-reconfigure auctex" si es necesario (recomendado).
+Description-pt_BR: AUCTeX parece j.Aé· estar se carregando automaticamente. Bom.
+ Inspecionando o arquivo "@PREFIX@/etc/emacs/site-start.el", AUCTeX parece jé· estar
+ se carregando automaticamente por padré„o em seu sistema.
+ .
+ Se isso né„o for verdadeiro, ou se né„o éÈ o que vocéÍ quer, por favor, procure
+ nesse arquivo por algo como "(require 'tex-site)" e corrija-o; enté„o execute
+ "dpkg-reconfigure auctex" se necessé·rio (recomendado).
+
+Template: auctex/doauto
+Type: select
+Choices: Background, Foreground, None
+Choices-it: Sottofondo, Primo piano, Nessuno
+Choices-de: Hintergund, Vordergrund, nicht
+Choices-es: Segundo plano, Primer plano, Omitir
+Choices-pt_BR: Background, Foreground, Nenhum
+Default: Background
+Description: How should (La)TeX macros be parsed?
+ To greatly improve AUCTeX performances, every TeX macro package and LaTeX
+ style file currently installed at your site will be parsed.
+ .
+ This may take a lot of time, so it should be probably done in background;
+ however you may also choose to have it done in foreground, or to skip the
+ parsing at all instead.  If you choose Background, you will find a detailed
+ log of the process in the file: ${LOGFILE}.
+ .
+ A weekly cron job will also take care of updating the informations now
+ collected, so usually you won't need to worry about them whenever you install
+ new (La)TeX packages or remove old ones.
+ .
+ Anyway, you can update them by hand running update-auctex-elisp.
+Description-it: In che modo si deve effettuare la scansione delle macro per (La)TeX?
+ Per migliorare grandemente le proprie prestazioni, AUCTeX scandir.bé‡ tutti i
+ pacchetti di macro per TeX ed i file di stile per LaTeX attualmente
+ installati su questa macchina.
+ .
+ Questa operazione potrebbe richiedere molto tempo ed éË quindi probabilmente
+ preferibile eseguirla in sottofondo; tuttavia, si puéÚ anche scegliere di
+ eseguirla in primo piano o di non eseguirla affatto.  Se si sceglie
+ Sottofondo verré‡ generata una cronistoria dettagliata dell'operazione nel
+ file: ${LOGFILE}.
+ .
+ Le informazioni raccolte adesso saranno aggiornate settimanalmente per mezzo
+ del demone cron, di modo che normalmente non saré‡ necessario preoccuparsene
+ di persona al momento dell'installazione o della rimozione di pacchetti
+ aggiuntivi per (La)TeX.
+ .
+ In ogni caso, le informazioni si possono sempre aggiornare a mano lanciando
+ update-auctex-elisp.
+Description-de: Wie sollen (La)TeX-Makros behandelt werden?
+ Zur Beschleunigung von AUCTeX kann jedes installierte TeX-Makro und jeder
+ LaTeX-Style im Voraus bearbeitet werden.
+ .
+ Das kann eine Weile dauern, daher wird die Bearbeitung im Hintergrund
+ empfohlen. Die Bearbeitung kann allerdings auch im Vordergrund erfolgen --
+ die weitere Installation wird dann allerdings bis zum Abschluéﬂ der
+ Bearbeitung unterbrochen. Die Voraus-Bearbeitung kann auch ganz
+ unterbleiben. Wenn Sie die Bearbeitung im Hintergrund wé‰hlen, finden Sie
+ eine detaillierte Protokollausgabe in ${LOGFILE}.
+ .
+ Weiterhin wird ein wéˆchentlicher cron-Job auf die Fortschreibung der nun
+ erhobenen Informationen achten, so daéﬂ Sie sich fé¸r gewéˆhnlich nicht weiter
+ darum ké¸mmern mé¸ssen, wenn Sie LaTeX-Pakete installieren oder entfernen.
+ .
+ Die Aktualisierung kann zudem von Hand durch Aufruf von
+ "update-auctex-elisp" erfolgen.
+Description-es: éøComo deben analizarse las macros (La)TeX?
+ Para mejorar el rendimiento de AUCTeX se analizaré· el contenido de cada
+ paquete de macros TeX y ficheros de estilo LaTeX instalado en su sistema.
+ .
+ Esto puede tardar mucho tiempo, por lo que se deberéÌa hacer en segundo
+ plano.  Sin embargo puede elegir hacerlo en primer plano u omitir este
+ paso.  Si elige hacerlo en segundo plano encontraré· un detallado registro
+ del proceso en el fichero: ${LOGFILE}.
+ .
+ AséÌ mismo, una tarea semanal actualizaré· la informaciéÛn recolectada en esta
+ sesiéÛn por lo que no se tiene que preocupar cuando instale nuevos paquetes
+ de (La)TeX o borre los antiguos.
+ .
+ De cualquier forma puede actualizarlos ejecutando update-auctex-elisp en
+ cualquier momento.
+Description-pt_BR: Como macros (La)TeX devem ser interpretados?
+ Para melhorar enormemente a performance de AUCTeX, todo pacote de macro TeX
+ e arquivo de estilo LaTeX instalados atualmente em seu sistema ser.Aé„o
+ interpretados.
+ .
+ Isso pode consumir um bom tempo, enté„o pode ser feito provavelmente em
+ background; entretanto, vocéÍ pode optar por ter isso feito em foreground, ou
+ ao invéÈs disso pular o processo de interpretaéÁé„o. Se vocéÍ escolher
+ Background, encontraré· um log detalhado do processo no arquivo: ${LOGFILE}.
+ .
+ Um trabalho semanal do cron tambéÈm tomaré· o cuidado de atualizar as
+ informaéÁéıes obtidas agora, enté„o vocéÍ normalmente né„o precisaré· preocupar-se
+ com elas mesmo quando instalar novos pacotes (La)TeX ou remover antigos.
+ .
+ De qualquer forma, vocéÍ pode atualizé·-las manualmente executando
+ update-auctex-elisp.
+
+Template: auctex/doautofg
+Type: select
+Choices: Console, File
+Choices-it: Video, File
+Choices-de: Konsole, Datei
+Choices-es: Consola, Fichero
+Choices-pt_BR: Console, Arquivo
+Default: File
+Description: Where should parsing output go?
+ You chose to parse TeX macro packages and LaTeX style files in foreground.
+ This is ok, but normally it will also generate quite a lot of output.
+ .
+ By default -- i.e., answering File to this question -- further output from
+ the parsing process will go to a file named: ${LOGFILE}.
+ .
+ If you answer Console, you will see all of the (boring) output right on your
+ terminal.
+Description-it: Dove vanno inviati i messaggi della scansione?
+ I pacchetti di macro per TeX ed i file di stile per LaTeX saranno scanditi
+ in primo piano come .béË stato richiesto.  Questo va bene, ma di solito produce
+ molti messaggi informativi.
+ .
+ Rispondendo File a questa domanda i messaggi originati dalla scansione
+ verranno scritti in un file di nome ${LOGFILE}.
+ .
+ Rispondendo Video, i messaggi (noiosi) verranno visualizzati sul proprio
+ terminale.
+Description-de: Wo soll die Protokoll-Ausgabe hingehen?
+ Sie wé‰hlten die Bearbeitung der Tex-Makros und LaTeX-Styles im Vordergrund.
+ Das ist schon in Ordnung, aber fé¸r gewéˆhnlich wird dabei eine Menge
+ langweiliger Kram ausgegeben.
+ .
+ Standardmé‰éﬂig (also bei der Wahl von "Datei") wird die weitere Ausgabe des
+ Bearbeitungsprozesses in die Datei ${LOGFILE} umgeleitet.
+ .
+ Wenn Sie mit "Konsole" antworten, wird die gesamte Ausgabe direkt auf Ihrem
+ Terminal erscheinen.
+Description-es: éøA donde se debe enviar la informaciéÛn al analizar?
+ Ha elegido analizar los paquetes de macros TeX y ficheros de estilo LaTeX
+ en primer plano.
+ .
+ Por defecto, respondiendo Fichero a esta pregunta, los mensajes del
+ proceso de analizado se guardaré·n en un fichero llamado: ${LOGFILE}.
+ .
+ Si responde Consola, veré· todos los (aburridos) mensajes en su terminal.
+Description-pt_BR: Onde deve ir a sa.AéÌda da interpretaéÁé„o?
+ VocéÍ escolheu interpretar pacotes de macros TeX e arquivos de estilo LaTeX em
+ foreground.  Isto esté· bem, mas normalmente isso tambéÈm geraré· bastante
+ saéÌda.
+ .
+ Por padré„o -- isto éÈ, respondendo Arquivo a esta questé„o -- a saéÌda adicional
+ do processo de interpretaéÁé„o iré· para um arquivo nomeado: ${LOGFILE}.
+ .
+ Se vocéÍ responder Console, veré· toda a saéÌda (cansativa) direto em seu
+ terminal.
+
diff -Naur auctex-11.55.orig/fink/tex-site.el auctex-11.55/fink/tex-site.el
--- auctex-11.55.orig/fink/tex-site.el	Thu Jan  1 09:00:00 1970
+++ auctex-11.55/fink/tex-site.el	Fri Mar 18 06:57:26 2005
@@ -0,0 +1,106 @@
+;;; tex-site.el - Site specific variables.  Don't edit.
+
+;; Copyright (C) 1991, 2000, 2001 Kresten Krab Thorup
+;; Copyright (C) 1993, 1994, 1997, 1999,
+;;               2003, 2004 Free Software Foundation, Inc.
+
+;; Author: Per Abrahamsen <abraham@dina.kvl.dk>
+;; Maintainer: auc-tex@sunsite.dk
+;; Keywords: tex
+
+;; This file is part of AUCTeX.
+
+;; AUCTeX is free software; you can redistribute it and/or modify it
+;; under the terms of the GNU General Public License as published by
+;; the Free Software Foundation; either version 2, or (at your option)
+;; any later version.
+
+;; AUCTeX is distributed in the hope that it will be useful, but
+;; WITHOUT ANY WARRANTY; without even the implied warranty of
+;; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+;; General Public License for more details.
+
+;; You should have received a copy of the GNU General Public License
+;; along with AUCTeX; see the file COPYING.  If not, write to the Free
+;; Software Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
+;; 02111-1307, USA.
+
+;;; Commentary:
+
+;; This file used to contain variables customized for the local site.
+;; This way of customizing AUCTeX should no longer be used.  All
+;; relevant variables can be customized by using
+;; M-x customize-variable RET
+;; on them.  Copying stuff into this file, as was recommended
+;; previously, would interfere with the operation of customize.
+;;
+;; Basically, this file contains a bunch of autoloads and
+;; automatically generated information.  AUCTeX is enabled by placing
+;; (require 'tex-site)
+;; into a user's .emacs file, or the administrator can insert it in
+;; the site-start.el file for a site-wide default.
+;;
+;; The right place for this file is in the `site-lisp' directory.
+
+;;; Code:
+
+(if (< emacs-major-version 21)
+  (error "AUCTeX requires Emacs 21 or later"))
+
+(defvar no-doc
+  "This function is part of AUCTeX, but has not yet been loaded.
+Full documentation will be available after autoloading the function."
+  "Documentation for autoload functions.")
+
+(defvar TeX-lisp-directory "@PREFIX@/share/emacs/site-lisp/auctex" 
+   "*The directory where most of the AUCTeX lisp files are located.
+tex-site.el (wherein this variable is defined) resides in its parent
+directory. For the location of lisp files associated with styles, see
+the variables TeX-style-* (hand-generated lisp) and TeX-auto-*
+(automatically generated lisp).")
+
+(defvar TeX-image-directory (expand-file-name
+			     "images"
+			     TeX-lisp-directory)
+  "*The directory where the AUCTeX images are located.")
+
+(defcustom TeX-auto-global "@PREFIX@/var/auctex/emacs"
+  "*Directory containing automatically generated information.
+Must end with a directory separator.
+
+For storing automatic extracted information about the TeX macros
+shared by all users of a site."
+  :group 'TeX-file
+  :type 'directory)
+
+;;; Autoloads:
+
+(add-to-list 'auto-mode-alist '("\\.dtx\\'" . doctex-mode))
+(add-to-list 'auto-mode-alist '("\\.drv\\'" . latex-mode))
+
+(add-to-list 'load-path TeX-lisp-directory)
+
+;; This hook will store bibitems when you save a BibTeX buffer.
+(add-hook 'bibtex-mode-hook 'BibTeX-auto-store)
+(autoload 'BibTeX-auto-store "latex" no-doc t)
+
+(autoload 'tex-mode "tex" no-doc t)
+(autoload 'plain-tex-mode "tex" no-doc t)
+(autoload 'ams-tex-mode "tex" no-doc t)
+(autoload 'context-mode "context" no-doc t)
+(autoload 'doctex-mode "latex" no-doc t)
+(autoload 'TeX-auto-generate "tex" no-doc t)
+(autoload 'TeX-auto-generate-global "tex" no-doc t)
+(autoload 'TeX-insert-quote "tex" no-doc t)
+(autoload 'TeX-submit-bug-report "tex" no-doc t)
+(autoload 'japanese-plain-tex-mode "tex-jp" no-doc t)
+(autoload 'japanese-latex-mode "tex-jp" no-doc t)
+(autoload 'texinfo-mode "tex-info" no-doc t)
+(autoload 'latex-mode "latex" no-doc t)
+
+(add-to-list 'load-path TeX-image-directory)
+(autoload 'LaTeX-install-toolbar "tex-bar")
+
+(provide 'tex-site)
+
+;;; tex-site.el ends here
diff -Naur auctex-11.55.orig/fink/update-auctex-elisp auctex-11.55/fink/update-auctex-elisp
--- auctex-11.55.orig/fink/update-auctex-elisp	Thu Jan  1 09:00:00 1970
+++ auctex-11.55/fink/update-auctex-elisp	Fri Mar 18 05:36:22 2005
@@ -0,0 +1,35 @@
+#!/bin/bash -e
+#
+# Update AUCTeX auto-loads.
+# Installed by the Fink auctex package.
+#
+# Copyright (C) 1997, 98, 99, 2000, 01, 02, 03, 04 by Davide Giovanni
+# Maria Salvetti.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License as published by the
+# Free Software Foundation; either version 2 of the License, or (at your
+# option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to: The Free Software Foundation, Inc.,
+# 59 Temple Place, Suite 330, Boston, MA 02111-1307, USA.
+
+set -o posix
+
+FLAVORS=${*:-"emacs21 emacs21-carbon"}
+
+for FLAVOR in ${FLAVORS}; do
+    if [ -x @PREFIX@/bin/${FLAVOR} ]; then
+	${FLAVOR} --no-site-file --no-init-file --batch \
+		  --load=@PREFIX@/share/${FLAVOR}/site-lisp/auctex//../tex-site.el \
+		  --funcall=TeX-auto-generate-global
+    fi
+done
+
+rm -f $(find @PREFIX@/var/lib/auctex/ -type f -name \*.el)
diff -Naur auctex-11.55.orig/fink/update-auctex-elisp.8 auctex-11.55/fink/update-auctex-elisp.8
--- auctex-11.55.orig/fink/update-auctex-elisp.8	Thu Jan  1 09:00:00 1970
+++ auctex-11.55/fink/update-auctex-elisp.8	Fri Mar 18 06:01:33 2005
@@ -0,0 +1,56 @@
+.\" Well, you know... this is an -*- nroff -*- source file.
+.\"
+.\" Man page for update-auctex-elisp.
+.\" Copyright 1998, 99, 2000, 01, 04 Davide Giovanni Maria Salvetti.
+.\"
+.\" This manual page is free software; you can redistribute it and/or
+ \" modify it under the terms of the GNU General Public License as
+ \" published by the Free Software Foundation; either version 2 of the
+ \" License, or (at your option) any later version.
+.\"
+.\" This manual page is distributed in the hope that it will be useful,
+ \" but WITHOUT ANY WARRANTY; without even the implied warranty of
+ \" MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ \" General Public License for more details.
+.\"
+.\" You should have received a copy of the GNU General Public License
+ \" along with this manual page; if not, write to: The Free Software
+ \" Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307,
+ \" USA.
+.\"
+.TH Update\-AUCTeX\-elisp 8 "17 February 1998" "Fink Project"
+.SH NAME
+update\-auctex\-elisp \- update AUCTeX automatically generated TeX
+macros and LaTeX styles informations.
+.SH SYNOPSIS
+.B update\-auctex\-elisp
+.RI [ FLAVOR .\|.\|. ]
+.SH DESCRIPTION
+The
+.B update\-auctex\-elisp
+script updates the automatically generated TeX and LaTeX informations
+used by AUCTeX and stored in
+.B @PREFIX@/var/lib/auctex/.
+.PP
+It should be called after each major (La)TeX style files installation or upgrade.
+.PP
+If no arguments are specified, update\-auctex\-elisp caters to every
+supported Emacs flavor installed on the system; else it updates only the
+specified flavors, if they are installed.
+.PP
+On Fink system
+.B update\-auctex\-elisp
+is called weekly by a
+.BR anacron (8)
+driven mechanism, to ensure up-to-date informations.
+.SH BUGS
+None known.
+.SH FILES
+.B @PREFIX@/var/lib/auctex/:
+the hierarchy containing the byte-compiled files.
+.SH SEE ALSO
+.BR emacs (1),
+.BR tex (1),
+.BR latex (1).
+.SH AUTHOR
+Davide G. M. Salvetti <salve@debian.org>.
diff -Naur auctex-11.55.orig/fink/update-auctex-install auctex-11.55/fink/update-auctex-install
--- auctex-11.55.orig/fink/update-auctex-install	Thu Jan  1 09:00:00 1970
+++ auctex-11.55/fink/update-auctex-install	Fri Mar 18 05:51:00 2005
@@ -0,0 +1,38 @@
+#!/bin/bash -e
+#
+# Update AUCTeX installation.
+# Installed by the Fink auctex package.
+#
+# Copyright (C) 1997, 98, 99, 2000, 01, 02, 03, 04 by Davide Giovanni
+# Maria Salvetti.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License as published by the
+# Free Software Foundation; either version 2 of the License, or (at your
+# option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to: The Free Software Foundation, Inc.,
+# 59 Temple Place, Suite 330, Boston, MA 02111-1307, USA.
+
+set -o posix
+
+source @PREFIX@/share/debconf/confmodule
+db_version 2.0
+
+FLAVORS=${*:-"emacs21 emacs21-carbon"}
+
+for FLAVOR in ${FLAVORS}; do
+    if [ -x @PREFIX@/bin/${FLAVOR} ]; then
+	LOGFILE=$(mktemp /tmp/update-auctex-elisp.XXXXXXX)
+	db_set auctex/logfile ${LOGFILE} || true
+	@PREFIX@/lib/emacsen-common/packages//remove/auctex ${FLAVOR}
+	@PREFIX@/lib/emacsen-common/packages//install/auctex ${FLAVOR}
+    fi
+done
+
diff -Naur auctex-11.55.orig/fink/update-auctex-install.8 auctex-11.55/fink/update-auctex-install.8
--- auctex-11.55.orig/fink/update-auctex-install.8	Thu Jan  1 09:00:00 1970
+++ auctex-11.55/fink/update-auctex-install.8	Fri Mar 18 06:02:02 2005
@@ -0,0 +1,58 @@
+.\" Well, you know... this is an -*- nroff -*- source file.
+.\"
+.\" Man page for update-auctex-install.
+.\" Copyright 2004 Davide Giovanni Maria Salvetti.
+.\"
+.\" This manual page is free software; you can redistribute it and/or
+ \" modify it under the terms of the GNU General Public License as
+ \" published by the Free Software Foundation; either version 2 of the
+ \" License, or (at your option) any later version.
+.\"
+.\" This manual page is distributed in the hope that it will be useful,
+ \" but WITHOUT ANY WARRANTY; without even the implied warranty of
+ \" MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ \" General Public License for more details.
+.\"
+.\" You should have received a copy of the GNU General Public License
+ \" along with this manual page; if not, write to: The Free Software
+ \" Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307,
+ \" USA.
+.\"
+.TH Update\-AUCTeX\-install 8 "19 May 2004" "Fink Project"
+.SH NAME
+update\-auctex\-install \- update AUCTeX installation.
+.SH SYNOPSIS
+.B update\-auctex\-install
+.RI [ FLAVOR .\|.\|. ]
+.SH DESCRIPTION
+The
+.B update\-auctex\-install
+script updates AUCTeX installation.
+.PP
+It should be called after each major (La)TeX installation or upgrade.
+.PP
+If no arguments are specified, update\-auctex\-install caters to every
+supported Emacs flavor installed on the system; else it updates only the
+specified flavors, if they are installed.
+.PP
+On Fink system
+.B update\-auctex\-install
+is not called weekly by a
+.BR anacron (8)
+driven mechanism, because it modifies files installed in the
+.B @PREFIX@/
+hierarchy, and that would not be policy compliant.  You may call it by hand,
+or have the maintainer of relevant packages add a call for it in their
+.B postinst
+package scripts.
+.SH BUGS
+None known.
+.SH FILES
+.B @PREFIX@/:
+the hierarchy containing AUCTeX installed files.
+.SH SEE ALSO
+.BR emacs (1),
+.BR tex (1),
+.BR latex (1).
+.SH AUTHOR
+Davide G. M. Salvetti <salve@debian.org>.
