diff -Nru emacs-21.3.50-20041229.work/01carbon-emacs.el emacs/01carbon-emacs.el
--- emacs-21.3.50-20041229.work/01carbon-emacs.el	Thu Jan  1 09:00:00 1970
+++ emacs/01carbon-emacs.el	Fri Mar 18 15:30:42 2005
@@ -0,0 +1,18 @@
+(if (eq window-system 'mac)
+  (progn
+    ;; append "PREFIX/sbin" "PREFIX/bin" to exec-path
+    (if (not (member "@PREFIX@/sbin" exec-path))
+    	(setq exec-path (append '("@PREFIX@/sbin") exec-path))
+    )
+    (if (not (member "@PREFIX@/bin" exec-path))
+	(setq exec-path (append '("@PREFIX@/bin") exec-path))
+      )
+
+    ;; inline image support
+    (auto-image-file-mode)
+    ;; inline patch
+    (setq mac-im-keystroke "\x18\x1b ")
+    (global-set-key mac-im-keystroke 'mac-im-update)
+    (define-key esc-map " " 'just-one-space)
+  )
+)
diff -Nru emacs-21.3.50-20041229.work/Emacs.app emacs/Emacs.app
--- emacs-21.3.50-20041229.work/Emacs.app	Thu Jan  1 09:00:00 1970
+++ emacs/Emacs.app	Fri Mar 18 15:25:00 2005
@@ -0,0 +1,2 @@
+#!/bin/sh
+exec @PREFIX@/Applications/Emacs.app/Contents/MacOS/Emacs "$@"
diff -Nru emacs-21.3.50-20041229.work/Makefile.in emacs/Makefile.in
--- emacs-21.3.50-20041229.work/Makefile.in	Sat Dec 11 09:03:11 2004
+++ emacs/Makefile.in	Fri Mar 18 15:24:45 2005
@@ -418,7 +418,7 @@
 	      (cd $${dir}; tar -chf - . ) \
 		| (cd $${dest}; umask 022; \
                    tar -xvf - && cat > /dev/null) || exit 1; \
-	      find $${dest} -exec chown $${LOGNAME:-$$USERNAME} {} ';' ;\
+	      find $${dest} -exec chown root:admin {} ';' ;\
 	      for subdir in `find $${dest} -type d ! -name RCS ! -name CVS -print` ; do \
 		chmod a+rx $${subdir} ; \
 		rm -rf $${subdir}/RCS ; \
@@ -458,7 +458,7 @@
 	   echo "Copying etc/DOC-* to ${docdir} ..." ; \
 	   (cd ./etc; tar -chf - DOC*) \
 	     |(cd ${docdir}; umask 022; tar -xvf - && cat > /dev/null) || exit 1; \
-	   (cd $(docdir); chown $${LOGNAME:-$$USERNAME} DOC*; chmod a+r DOC*; \
+	   (cd $(docdir); chown root:admin DOC*; chmod a+r DOC*; \
 	    if test "`echo DOC-*`" != "DOC-*"; then rm DOC; fi); \
 	else true; fi
 	-unset CDPATH; \
@@ -470,7 +470,7 @@
 	   echo "Copying lisp/*.el and lisp/*.elc to ${lispdir} ..." ; \
 	   (cd lisp; tar -chf - *.el *.elc) \
 	     |(cd ${lispdir}; umask 022; tar -xvf - && cat > /dev/null) || exit 1; \
-	   (cd ${lispdir}; find . -exec chown $${LOGNAME:-$$USERNAME} {} ';') ; \
+	   (cd ${lispdir}; find . -exec chown root:admin {} ';') ; \
 	else true; fi
 	-unset CDPATH; \
 	thisdir=`/bin/pwd`; \
diff -Nru emacs-21.3.50-20041229.work/configure emacs/configure
--- emacs-21.3.50-20041229.work/configure	Sat Dec 18 04:35:43 2004
+++ emacs/configure	Fri Mar 18 15:24:46 2005
@@ -21633,6 +21633,7 @@
 echo "$as_me: error: can't find current emacs version in \`${srcdir}/lisp/version.el'." >&2;}
    { (exit 1); exit 1; }; }
 fi
+version=${version}-carbon
 
 ### Specify what sort of things we'll be editing into Makefile and config.h.
 ### Use configuration here uncanonicalized to avoid exceeding size limits.
diff -Nru emacs-21.3.50-20041229.work/finktmp/carbon-font.el emacs/finktmp/carbon-font.el
--- emacs-21.3.50-20041229.work/finktmp/carbon-font.el	Fri Mar 18 15:22:42 2005
+++ emacs/finktmp/carbon-font.el	Fri Mar 18 15:24:17 2005
@@ -173,10 +173,20 @@
   "Variable if courier new font is installed or not.")
 
 (defvar carbon-font-monaco_cy-exist (x-list-fonts "*-monaco cy-*")
-  "Variable if courier new font is installed or not.")
+  "Variable if monaco cy font is installed or not.")
 
 (defvar carbon-font-monaco_ce-exist (x-list-fonts "*-monaco ce-*")
-  "Variable if courier new font is installed or not.")
+  "Variable if monaco ce european font is installed or not.")
+
+(defun carbon-font-spec-monaco_ce-or-some (size)
+  (format "latin-iso8859-2:-apple-%s-medium-r-normal--%d-*-75-75-m-*-mac-centraleurroman"
+	  (if carbon-font-monaco_ce-exist "monaco ce" "*")
+	  size))
+
+(defun carbon-font-spec-monaco_cy-or-some (size)
+  (format "cyrillic-iso8859-5:-apple-%s-medium-r-normal--%d-*-75-75-m-*-mac-cyrillic"
+	  (if carbon-font-monaco_cy-exist "monaco cy" "*")
+	  size))
 
 (if carbon-font-ayuthaya-exist
     (create-fontset-from-fontset-spec
@@ -189,8 +199,8 @@
       ",korean-ksc5601:-apple-applegothic-medium-r-normal--28-288-75-75-m-288-ksc5601.1989-0" 
       ",ascii:-apple-ayuthaya-medium-r-normal--22-*-75-75-m-*-mac-*"
       ",latin-iso8859-1:-apple-monaco-medium-r-normal--24-*-75-75-m-*-mac-roman"
-      ",latin-iso8859-2:-apple-monaco ce-medium-r-normal--24-*-75-75-m-*-mac-centraleurroman"
-      ",cyrillic-iso8859-5:-apple-monaco cy-medium-r-normal--24-*-75-75-m-*-mac-cyrillic"
+      "," (carbon-font-spec-monaco_ce-or-some 24)
+      "," (carbon-font-spec-monaco_cy-or-some 24)
       )))
 
 (if carbon-font-ayuthaya-exist
@@ -204,8 +214,8 @@
       ",korean-ksc5601:-apple-applegothic-medium-r-normal--24-240-75-75-m-240-ksc5601.1989-0" 
       ",ascii:-apple-ayuthaya-medium-r-normal--19-*-75-75-m-*-mac-*"
       ",latin-iso8859-1:-apple-monaco-medium-r-normal--20-*-75-75-m-*-mac-roman"
-      ",latin-iso8859-2:-apple-monaco ce-medium-r-normal--20-*-75-75-m-*-mac-centraleurroman"
-      ",cyrillic-iso8859-5:-apple-monaco cy-medium-r-normal--20-*-75-75-m-*-mac-cyrillic"
+      "," (carbon-font-spec-monaco_ce-or-some 20)
+      "," (carbon-font-spec-monaco_cy-or-some 20)
       )))
 
 (if carbon-font-ayuthaya-exist
@@ -219,12 +229,12 @@
       ",korean-ksc5601:-apple-applegothic-medium-r-normal--22-216-75-75-m-216-ksc5601.1989-0" 
       ",ascii:-apple-ayuthaya-medium-r-normal--18-*-75-75-m-*-mac-*"
       ",latin-iso8859-1:-apple-monaco-medium-r-normal--18-*-75-75-m-*-mac-roman"
-      ",latin-iso8859-2:-apple-monaco ce-medium-r-normal--18-*-75-75-m-*-mac-centraleurroman"
-      ",cyrillic-iso8859-5:-apple-monaco cy-medium-r-normal--18-*-75-75-m-*-mac-cyrillic"
+      "," (carbon-font-spec-monaco_ce-or-some 18)
+      "," (carbon-font-spec-monaco_cy-or-some 18)
       )))
 
 (if carbon-font-ayuthaya-exist
-    (create-fontset-from-fontset-spec
+   (create-fontset-from-fontset-spec
      (concat 
       "-*-fixed-medium-r-normal-*-16-*-*-*-*-*-fontset-hiraginomaru16" 
       ",japanese-jisx0208:-apple-ヒラギノ丸ゴ pro w4-medium-r-normal--20-200-75-75-m-200-jisx0208.1983-sjis" 
@@ -234,8 +244,8 @@
       ",korean-ksc5601:-apple-applegothic-medium-r-normal--20-200-75-75-m-200-ksc5601.1989-0" 
       ",ascii:-apple-ayuthaya-medium-r-normal--16-*-75-75-m-*-mac-*"
       ",latin-iso8859-1:-apple-monaco-medium-r-normal--16-*-75-75-m-*-mac-roman"
-      ",latin-iso8859-2:-apple-monaco ce-medium-r-normal--16-*-75-75-m-*-mac-centraleurroman"
-      ",cyrillic-iso8859-5:-apple-monaco cy-medium-r-normal--16-*-75-75-m-*-mac-cyrillic"
+      "," (carbon-font-spec-monaco_ce-or-some 16)
+      "," (carbon-font-spec-monaco_cy-or-some 16)
       )))
 
 (create-fontset-from-fontset-spec
@@ -248,12 +258,8 @@
   ",korean-ksc5601:-apple-applegothic-medium-r-normal--16-160-75-75-m-160-ksc5601.1989-0" 
   ",ascii:-apple-monaco-medium-r-normal--14-*-75-75-m-*-mac-roman"
   ",latin-iso8859-1:-apple-monaco-medium-r-normal--14-*-75-75-m-*-mac-roman"
-  (if carbon-font-monaco_ce-exist
-      ",latin-iso8859-2:-apple-monaco ce-medium-r-normal--14-*-75-75-m-*-mac-centraleurroman"
-    ",latin-iso8859-2:-apple-*-medium-r-normal--14-*-75-75-m-*-mac-centraleurroman")
-  (if carbon-font-monaco_cy-exist
-      ",cyrillic-iso8859-5:-apple-monaco cy-medium-r-normal--14-*-75-75-m-*-mac-cyrillic"
-    ",cyrillic-iso8859-5:-apple-*-medium-r-normal--14-*-75-75-m-*-mac-cyrillic")
+  "," (carbon-font-spec-monaco_ce-or-some 14)
+  "," (carbon-font-spec-monaco_cy-or-some 14)
   ))
 
 (create-fontset-from-fontset-spec
@@ -266,12 +272,8 @@
   ",korean-ksc5601:-apple-applegothic-medium-r-normal--14-140-75-75-m-140-ksc5601.1989-0" 
   ",ascii:-apple-monaco-medium-r-normal--12-*-75-75-m-*-mac-roman"
   ",latin-iso8859-1:-apple-monaco-medium-r-normal--12-*-75-75-m-*-mac-roman"
-  (if carbon-font-monaco_ce-exist
-      ",latin-iso8859-2:-apple-monac ce-medium-r-normal--12-*-75-75-m-*-mac-centraleurroman"
-    ",latin-iso8859-2:-apple-*-medium-r-normal--12-*-75-75-m-*-mac-centraleurroman")
-  (if carbon-font-monaco_cy-exist
-      ",cyrillic-iso8859-5:-apple-monaco cy-medium-r-normal--12-*-75-75-m-*-mac-cyrillic"
-    ",cyrillic-iso8859-5:-apple-*-medium-r-normal--12-*-75-75-m-*-mac-cyrillic")
+  "," (carbon-font-spec-monaco_ce-or-some 12)
+  "," (carbon-font-spec-monaco_cy-or-some 12)
   ))
 
 (create-fontset-from-fontset-spec
@@ -284,12 +286,8 @@
   ",korean-ksc5601:-apple-applegothic-medium-r-normal--12-120-75-75-m-120-ksc5601.1989-0" 
   ",ascii:-apple-monaco-medium-r-normal--10-*-75-75-m-*-mac-roman"
   ",latin-iso8859-1:-apple-monaco-medium-r-normal--10-*-75-75-m-*-mac-roman"
-  (if carbon-font-monaco_ce-exist
-      ",latin-iso8859-2:-apple-monac ce-medium-r-normal--10-*-75-75-m-*-mac-centraleurroman"
-    ",latin-iso8859-2:-apple-*-medium-r-normal--10-*-75-75-m-*-mac-centraleurroman")
-  (if carbon-font-monaco_cy-exist
-      ",cyrillic-iso8859-5:-apple-monaco cy-medium-r-normal--10-*-75-75-m-*-mac-cyrillic"
-    ",cyrillic-iso8859-5:-apple-*-medium-r-normal--10-*-75-75-m-*-mac-cyrillic")
+  "," (carbon-font-spec-monaco_ce-or-some 10)
+  "," (carbon-font-spec-monaco_cy-or-some 10)
   ))
 
 
@@ -303,8 +301,8 @@
   ",korean-ksc5601:-apple-applegothic-medium-r-normal--12-120-75-75-m-120-ksc5601.1989-0" 
   ",ascii:-apple-monaco-medium-r-normal--9-*-75-75-m-*-mac-*"
   ",latin-iso8859-1:-apple-monaco-medium-r-normal--9-*-75-75-m-*-mac-roman"
-  ",latin-iso8859-2:-apple-monaco ce-medium-r-normal--9-*-75-75-m-*-mac-centraleurroman"
-  ",cyrillic-iso8859-5:-apple-monaco cy-medium-r-normal--9-*-75-75-m-*-mac-cyrillic"
+  "," (carbon-font-spec-monaco_ce-or-some 9)
+  "," (carbon-font-spec-monaco_cy-or-some 9)
   ))
 
 (if carbon-font-ayuthaya-exist
@@ -318,6 +316,8 @@
       ",korean-ksc5601:-apple-applegothic-medium-r-normal--10-100-75-75-m-100-ksc5601.1989-0" 
       ",ascii:-apple-ayuthaya-medium-r-normal--8-*-75-75-m-*-mac-*"
       ",latin-iso8859-1:-apple-monaco-medium-r-normal--8-*-75-75-m-*-mac-roman"
+      ;; "," (carbon-font-spec-monaco_ce-or-some 8) ;; You might want to uncomment this line
+      ;; "," (carbon-font-spec-monaco_cy-or-some 8) ;; You might want to uncomment this line
       )))
 
 (if carbon-font-ayuthaya-exist
@@ -331,6 +331,8 @@
       ",korean-ksc5601:-apple-applegothic-medium-r-normal--8-80-75-75-m-80-ksc5601.1989-0" 
       ",ascii:-apple-ayuthaya-medium-r-normal--7-*-75-75-m-*-mac-*"
       ",latin-iso8859-1:-apple-monaco-medium-r-normal--7-*-75-75-m-*-mac-roman"
+      ;; "," (carbon-font-spec-monaco_ce-or-some 7) ;; You might want to uncomment this line
+      ;; "," (carbon-font-spec-monaco_cy-or-some 7) ;; You might want to uncomment this line
       )))
 
 ;;
@@ -348,8 +350,8 @@
       ",korean-ksc5601:-apple-applegothic-medium-r-normal--28-288-75-75-m-288-ksc5601.1989-0" 
       ",ascii:-apple-ayuthaya-*-*-normal--22-*-75-75-m-*-mac-*"
       ",latin-iso8859-1:-apple-monaco-*-*-normal--24-*-75-75-m-*-mac-roman"
-      ",latin-iso8859-2:-apple-monaco ce-medium-r-normal--24-*-75-75-m-*-mac-centraleurroman"
-      ",cyrillic-iso8859-5:-apple-monaco cy-medium-r-normal--24-*-75-75-m-*-mac-cyrillic"
+      "," (carbon-font-spec-monaco_ce-or-some 24)
+      "," (carbon-font-spec-monaco_cy-or-some 24)
       )))
 
 (if carbon-font-ayuthaya-exist
@@ -363,8 +365,8 @@
       ",korean-ksc5601:-apple-applegothic-medium-r-normal--24-240-75-75-m-240-ksc5601.1989-0" 
       ",ascii:-apple-ayuthaya-*-*-normal--19-*-75-75-m-*-mac-*"
       ",latin-iso8859-1:-apple-monaco-*-*-normal--20-*-75-75-m-*-mac-roman"
-      ",latin-iso8859-2:-apple-monaco ce-medium-r-normal--20-*-75-75-m-*-mac-centraleurroman"
-      ",cyrillic-iso8859-5:-apple-monaco cy-medium-r-normal--20-*-75-75-m-*-mac-cyrillic"
+      "," (carbon-font-spec-monaco_ce-or-some 20)
+      "," (carbon-font-spec-monaco_cy-or-some 20)
       )))
 
 (if carbon-font-ayuthaya-exist
@@ -378,8 +380,8 @@
       ",korean-ksc5601:-apple-applegothic-medium-r-normal--22-216-75-75-m-216-ksc5601.1989-0" 
       ",ascii:-apple-ayuthaya-*-*-normal--18-*-75-75-m-*-mac-*"
       ",latin-iso8859-1:-apple-monaco-*-*-normal--18-*-75-75-m-*-mac-roman"
-      ",latin-iso8859-2:-apple-monaco ce-medium-r-normal--18-*-75-75-m-*-mac-centraleurroman"
-      ",cyrillic-iso8859-5:-apple-monaco cy-medium-r-normal--18-*-75-75-m-*-mac-cyrillic"
+      "," (carbon-font-spec-monaco_ce-or-some 18)
+      "," (carbon-font-spec-monaco_cy-or-some 18)
       )))
 
 (if carbon-font-ayuthaya-exist
@@ -393,8 +395,8 @@
       ",korean-ksc5601:-apple-applegothic-medium-r-normal--20-200-75-75-m-200-ksc5601.1989-0" 
       ",ascii:-apple-ayuthaya-*-*-normal--16-*-75-75-m-*-mac-*"
       ",latin-iso8859-1:-apple-monaco-*-*-normal--16-*-75-75-m-*-mac-roman"
-      ",latin-iso8859-2:-apple-monaco ce-medium-r-normal--16-*-75-75-m-*-mac-centraleurroman"
-      ",cyrillic-iso8859-5:-apple-monaco cy-medium-r-normal--16-*-75-75-m-*-mac-cyrillic"
+      "," (carbon-font-spec-monaco_ce-or-some 16)
+      "," (carbon-font-spec-monaco_cy-or-some 16)
       )))
 
 (create-fontset-from-fontset-spec
@@ -407,12 +409,8 @@
   ",korean-ksc5601:-apple-applegothic-medium-r-normal--16-160-75-75-m-160-ksc5601.1989-0" 
   ",ascii:-apple-monaco-*-*-normal--14-*-75-75-m-*-mac-roman"
   ",latin-iso8859-1:-apple-monaco-*-*-normal--14-*-75-75-m-*-mac-roman"
-  (if carbon-font-monaco_ce-exist
-      ",latin-iso8859-2:-apple-monac ce-medium-r-normal--14-*-75-75-m-*-mac-centraleurroman"
-    ",latin-iso8859-2:-apple-*-medium-r-normal--14-*-75-75-m-*-mac-centraleurroman")
-  (if carbon-font-monaco_cy-exist
-      ",cyrillic-iso8859-5:-apple-monaco cy-medium-r-normal--14-*-75-75-m-*-mac-cyrillic"
-    ",cyrillic-iso8859-5:-apple-*-medium-r-normal--14-*-75-75-m-*-mac-cyrillic")
+  "," (carbon-font-spec-monaco_ce-or-some 14)
+  "," (carbon-font-spec-monaco_cy-or-some 14)
   ))
 
 (create-fontset-from-fontset-spec
@@ -425,12 +423,8 @@
   ",korean-ksc5601:-apple-applegothic-medium-r-normal--14-140-75-75-m-140-ksc5601.1989-0" 
   ",ascii:-apple-monaco-*-*-normal--12-*-75-75-m-*-mac-roman"
   ",latin-iso8859-1:-apple-monaco-*-*-normal--12-*-75-75-m-*-mac-roman"
-  (if carbon-font-monaco_ce-exist
-      ",latin-iso8859-2:-apple-monac ce-medium-r-normal--12-*-75-75-m-*-mac-centraleurroman"
-    ",latin-iso8859-2:-apple-*-medium-r-normal--12-*-75-75-m-*-mac-centraleurroman")
-  (if carbon-font-monaco_cy-exist
-      ",cyrillic-iso8859-5:-apple-monaco cy-medium-r-normal--12-*-75-75-m-*-mac-cyrillic"
-    ",cyrillic-iso8859-5:-apple-*-medium-r-normal--12-*-75-75-m-*-mac-cyrillic")
+  "," (carbon-font-spec-monaco_ce-or-some 12)
+  "," (carbon-font-spec-monaco_cy-or-some 12)
   ))
 
 (create-fontset-from-fontset-spec
@@ -443,12 +437,8 @@
   ",korean-ksc5601:-apple-applegothic-medium-r-normal--12-120-75-75-m-120-ksc5601.1989-0" 
   ",ascii:-apple-monaco-*-*-normal--10-*-75-75-m-*-mac-roman"
   ",latin-iso8859-1:-apple-monaco-*-*-normal--10-*-75-75-m-*-mac-roman"
-  (if carbon-font-monaco_ce-exist
-      ",latin-iso8859-2:-apple-monac ce-medium-r-normal--10-*-75-75-m-*-mac-centraleurroman"
-    ",latin-iso8859-2:-apple-*-medium-r-normal--10-*-75-75-m-*-mac-centraleurroman")
-  (if carbon-font-monaco_cy-exist
-      ",cyrillic-iso8859-5:-apple-monaco cy-medium-r-normal--10-*-75-75-m-*-mac-cyrillic"
-    ",cyrillic-iso8859-5:-apple-*-medium-r-normal--10-*-75-75-m-*-mac-cyrillic")
+  "," (carbon-font-spec-monaco_ce-or-some 10)
+  "," (carbon-font-spec-monaco_cy-or-some 10)
   ))
 
 (create-fontset-from-fontset-spec
@@ -461,8 +451,8 @@
   ",korean-ksc5601:-apple-applegothic-medium-r-normal--12-120-75-75-m-120-ksc5601.1989-0" 
   ",ascii:-apple-monaco-*-*-normal--9-*-75-75-m-*-mac-*"
   ",latin-iso8859-1:-apple-monaco-*-*-normal--9-*-75-75-m-*-mac-roman"
-  ",latin-iso8859-2:-apple-monaco ce-medium-r-normal--9-*-75-75-m-*-mac-centraleurroman"
-  ",cyrillic-iso8859-5:-apple-monaco cy-medium-r-normal--9-*-75-75-m-*-mac-cyrillic"
+  "," (carbon-font-spec-monaco_ce-or-some 9)
+  "," (carbon-font-spec-monaco_cy-or-some 9)
   ))
 
 (if carbon-font-ayuthaya-exist
@@ -476,8 +466,8 @@
       ",korean-ksc5601:-apple-applegothic-medium-r-normal--10-100-75-75-m-100-ksc5601.1989-0" 
       ",ascii:-apple-ayuthaya-medium-r-normal--8-*-75-75-m-*-mac-*"
       ",latin-iso8859-1:-apple-monaco-medium-r-normal--8-*-75-75-m-*-mac-roman"
-      ",latin-iso8859-2:-apple-monaco ce-medium-r-normal--8-*-75-75-m-*-mac-centraleurroman"
-      ",cyrillic-iso8859-5:-apple-monaco cy-medium-r-normal--8-*-75-75-m-*-mac-cyrillic"
+      "," (carbon-font-spec-monaco_ce-or-some 8)
+      "," (carbon-font-spec-monaco_cy-or-some 8)
       )))
 
 (if carbon-font-ayuthaya-exist
@@ -491,8 +481,8 @@
       ",korean-ksc5601:-apple-applegothic-medium-r-normal--8-80-75-75-m-80-ksc5601.1989-0" 
       ",ascii:-apple-ayuthaya-medium-r-normal--7-*-75-75-m-*-mac-*"
       ",latin-iso8859-1:-apple-monaco-medium-r-normal--7-*-75-75-m-*-mac-roman"
-      ",latin-iso8859-2:-apple-monaco ce-medium-r-normal--7-*-75-75-m-*-mac-centraleurroman"
-      ",cyrillic-iso8859-5:-apple-monaco cy-medium-r-normal--7-*-75-75-m-*-mac-cyrillic"
+      "," (carbon-font-spec-monaco_ce-or-some 7)
+      "," (carbon-font-spec-monaco_cy-or-some 7)
       )))
 
 ;;
@@ -568,7 +558,8 @@
   ",ascii:-apple-courier-*-*-normal--14-*-75-75-m-*-mac-roman"
   ",latin-iso8859-1:-apple-courier-*-*-normal--14-*-75-75-m-*-mac-roman"
   ",latin-iso8859-2:-apple-courier ce-medium-r-normal--14-*-75-75-m-*-mac-centraleurroman"
-  ",cyrillic-iso8850-5:-apple-*-medium-r-normal--*-*-75-75-m-*-mac-cyrillic"
+  ",cyrillic-iso8850-5:-apple-*-medium-r-normal--*-*-75-75-m-*-mac-cyrillic" ;; wrong?
+  ;; ",cyrillic-iso8850-5:-apple-*-medium-r-normal--14-*-75-75-m-*-mac-cyrillic"
   ))
 
 (create-fontset-from-fontset-spec
@@ -658,8 +649,8 @@
       ",korean-ksc5601:-apple-applegothic-medium-r-normal--28-288-75-75-m-288-ksc5601.1989-0" 
       ",ascii:-apple-ayuthaya-medium-r-normal--24-*-75-75-m-*-mac-*"
       ",latin-iso8859-1:-apple-monaco-medium-r-normal--24-*-75-75-m-*-mac-roman"
-      ",latin-iso8859-2:-apple-monaco ce-medium-r-normal--24-*-75-75-m-*-mac-centraleurroman"
-      ",cyrillic-iso8859-5:-apple-monaco cy-medium-r-normal--24-*-75-75-m-*-mac-cyrillic"
+      "," (carbon-font-spec-monaco_ce-or-some 24)
+      "," (carbon-font-spec-monaco_cy-or-some 24)
       )))
 
 (if carbon-font-ayuthaya-exist
@@ -673,8 +664,8 @@
       ",korean-ksc5601:-apple-applegothic-medium-r-normal--24-240-75-75-m-240-ksc5601.1989-0" 
       ",ascii:-apple-ayuthaya-medium-r-normal--20-*-75-75-m-*-mac-*"
       ",latin-iso8859-1:-apple-monaco-medium-r-normal--20-*-75-75-m-*-mac-roman"
-      ",latin-iso8859-2:-apple-monaco ce-medium-r-normal--20-*-75-75-m-*-mac-centraleurroman"
-      ",cyrillic-iso8859-5:-apple-monaco cy-medium-r-normal--20-*-75-75-m-*-mac-cyrillic"
+      "," (carbon-font-spec-monaco_ce-or-some 20)
+      "," (carbon-font-spec-monaco_cy-or-some 20)
       )))
 
 (if carbon-font-ayuthaya-exist
@@ -688,8 +679,8 @@
       ",korean-ksc5601:-apple-applegothic-medium-r-normal--22-216-75-75-m-216-ksc5601.1989-0" 
       ",ascii:-apple-ayuthaya-medium-r-normal--18-*-75-75-m-*-mac-*"
       ",latin-iso8859-1:-apple-monaco-medium-r-normal--18-*-75-75-m-*-mac-roman"
-      ",latin-iso8859-2:-apple-monaco ce-medium-r-normal--18-*-75-75-m-*-mac-centraleurroman"
-      ",cyrillic-iso8859-5:-apple-monaco cy-medium-r-normal--18-*-75-75-m-*-mac-cyrillic"
+      "," (carbon-font-spec-monaco_ce-or-some 18)
+      "," (carbon-font-spec-monaco_cy-or-some 18)
       )))
 
 (if carbon-font-ayuthaya-exist
@@ -703,8 +694,8 @@
       ",korean-ksc5601:-apple-applegothic-medium-r-normal--20-200-75-75-m-200-ksc5601.1989-0" 
       ",ascii:-apple-ayuthaya-medium-r-normal--16-*-75-75-m-*-mac-*"
       ",latin-iso8859-1:-apple-monaco-medium-r-normal--16-*-75-75-m-*-mac-roman"
-      ",latin-iso8859-2:-apple-monaco ce-medium-r-normal--16-*-75-75-m-*-mac-centraleurroman"
-      ",cyrillic-iso8859-5:-apple-monaco cy-medium-r-normal--16-*-75-75-m-*-mac-cyrillic"
+      "," (carbon-font-spec-monaco_ce-or-some 16)
+      "," (carbon-font-spec-monaco_cy-or-some 16)
       )))
 
 (if carbon-font-ayuthaya-exist
@@ -718,8 +709,8 @@
       ",korean-ksc5601:-apple-applegothic-medium-r-normal--18-180-75-75-m-180-ksc5601.1989-0" 
       ",ascii:-apple-ayuthaya-medium-r-normal--15-*-75-75-m-*-mac-*"
       ",latin-iso8859-1:-apple-monaco-medium-r-normal--15-*-75-75-m-*-mac-roman"
-      ",latin-iso8859-2:-apple-monaco ce-medium-r-normal--15-*-75-75-m-*-mac-centraleurroman"
-      ",cyrillic-iso8859-5:-apple-monaco cy-medium-r-normal--15-*-75-75-m-*-mac-cyrillic"
+      "," (carbon-font-spec-monaco_ce-or-some 15)
+      "," (carbon-font-spec-monaco_cy-or-some 15)
       )))
 
 (create-fontset-from-fontset-spec
@@ -732,12 +723,8 @@
   ",korean-ksc5601:-apple-applegothic-medium-r-normal--16-160-75-75-m-160-ksc5601.1989-0" 
   ",ascii:-apple-monaco-medium-r-normal--14-*-75-75-m-*-mac-roman"
   ",latin-iso8859-1:-apple-monaco-medium-r-normal--14-*-75-75-m-*-mac-roman"
-  (if carbon-font-monaco_ce-exist
-      ",latin-iso8859-2:-apple-monac ce-medium-r-normal--14-*-75-75-m-*-mac-centraleurroman"
-    ",latin-iso8859-2:-apple-*-medium-r-normal--14-*-75-75-m-*-mac-centraleurroman")
-  (if carbon-font-monaco_cy-exist
-      ",cyrillic-iso8859-5:-apple-monaco cy-medium-r-normal--14-*-75-75-m-*-mac-cyrillic"
-    ",cyrillic-iso8859-5:-apple-*-medium-r-normal--14-*-75-75-m-*-mac-cyrillic")
+  "," (carbon-font-spec-monaco_ce-or-some 14)
+  "," (carbon-font-spec-monaco_cy-or-some 14)
   ))
 
 (create-fontset-from-fontset-spec
@@ -750,12 +737,8 @@
   ",korean-ksc5601:-apple-applegothic-medium-r-normal--14-140-75-75-m-140-ksc5601.1989-0" 
   ",ascii:-apple-monaco-medium-r-normal--12-*-75-75-m-*-mac-roman"
   ",latin-iso8859-1:-apple-monaco-medium-r-normal--12-*-75-75-m-*-mac-roman"
-  (if carbon-font-monaco_ce-exist
-      ",latin-iso8859-2:-apple-monac ce-medium-r-normal--12-*-75-75-m-*-mac-centraleurroman"
-    ",latin-iso8859-2:-apple-*-medium-r-normal--12-*-75-75-m-*-mac-centraleurroman")
-  (if carbon-font-monaco_cy-exist
-      ",cyrillic-iso8859-5:-apple-monaco cy-medium-r-normal--12-*-75-75-m-*-mac-cyrillic"
-    ",cyrillic-iso8859-5:-apple-*-medium-r-normal--12-*-75-75-m-*-mac-cyrillic")
+  "," (carbon-font-spec-monaco_ce-or-some 12)
+  "," (carbon-font-spec-monaco_cy-or-some 12)
   ))
 
 (create-fontset-from-fontset-spec
@@ -768,12 +751,8 @@
   ",korean-ksc5601:-apple-applegothic-medium-r-normal--12-120-75-75-m-120-ksc5601.1989-0" 
   ",ascii:-apple-monaco-medium-r-normal--10-*-75-75-m-*-mac-roman"
   ",latin-iso8859-1:-apple-monaco-medium-r-normal--10-*-75-75-m-*-mac-roman"
-  (if carbon-font-monaco_ce-exist
-      ",latin-iso8859-2:-apple-monac ce-medium-r-normal--10-*-75-75-m-*-mac-centraleurroman"
-    ",latin-iso8859-2:-apple-*-medium-r-normal--10-*-75-75-m-*-mac-centraleurroman")
-  (if carbon-font-monaco_cy-exist
-      ",cyrillic-iso8859-5:-apple-monaco cy-medium-r-normal--10-*-75-75-m-*-mac-cyrillic"
-    ",cyrillic-iso8859-5:-apple-*-medium-r-normal--10-*-75-75-m-*-mac-cyrillic")
+  "," (carbon-font-spec-monaco_ce-or-some 10)
+  "," (carbon-font-spec-monaco_cy-or-some 10)
   ))
 
 (create-fontset-from-fontset-spec
@@ -786,8 +765,8 @@
   ",korean-ksc5601:-apple-applegothic-medium-r-normal--12-120-75-75-m-120-ksc5601.1989-0" 
   ",ascii:-apple-monaco-medium-r-normal--9-*-75-75-m-*-mac-*"
   ",latin-iso8859-1:-apple-monaco-medium-r-normal--9-*-75-75-m-*-mac-roman"
-  ",latin-iso8859-2:-apple-monaco ce-medium-r-normal--9-*-75-75-m-*-mac-centraleurroman"
-  ",cyrillic-iso8859-5:-apple-monaco cy-medium-r-normal--9-*-75-75-m-*-mac-cyrillic"
+  "," (carbon-font-spec-monaco_ce-or-some 9)
+  "," (carbon-font-spec-monaco_cy-or-some 9)
   ))
 
 (if carbon-font-ayuthaya-exist
@@ -801,8 +780,8 @@
       ",korean-ksc5601:-apple-applegothic-medium-r-normal--10-100-75-75-m-100-ksc5601.1989-0" 
       ",ascii:-apple-ayuthaya-medium-r-normal--8-*-75-75-m-*-mac-*"
       ",latin-iso8859-1:-apple-monaco-medium-r-normal--8-*-75-75-m-*-mac-roman"
-      ",latin-iso8859-2:-apple-monaco ce-medium-r-normal--8-*-75-75-m-*-mac-centraleurroman"
-      ",cyrillic-iso8859-5:-apple-monaco cy-medium-r-normal--8-*-75-75-m-*-mac-cyrillic"
+      "," (carbon-font-spec-monaco_ce-or-some 8)
+      "," (carbon-font-spec-monaco_cy-or-some 8)
       )))
 
 (if carbon-font-ayuthaya-exist
@@ -816,8 +795,8 @@
       ",korean-ksc5601:-apple-applegothic-medium-r-normal--8-80-75-75-m-80-ksc5601.1989-0" 
       ",ascii:-apple-ayuthaya-medium-r-normal--7-*-75-75-m-*-mac-*"
       ",latin-iso8859-1:-apple-monaco-medium-r-normal--7-*-75-75-m-*-mac-roman"
-      ",latin-iso8859-2:-apple-monaco ce-medium-r-normal--7-*-75-75-m-*-mac-centraleurroman"
-      ",cyrillic-iso8859-5:-apple-monaco cy-medium-r-normal--7-*-75-75-m-*-mac-cyrillic"
+      "," (carbon-font-spec-monaco_ce-or-some 7)
+      "," (carbon-font-spec-monaco_cy-or-some 7)
       )))
 
 ;;
diff -Nru emacs-21.3.50-20041229.work/fix-info emacs/fix-info
--- emacs-21.3.50-20041229.work/fix-info	Thu Jan  1 09:00:00 1970
+++ emacs/fix-info	Fri Mar 18 15:24:46 2005
@@ -0,0 +1,11 @@
+#!/bin/sh
+
+THE_INFO_PREFIX=$1
+
+for file in ada-mode autotype calc ccmode cl dired-x ebrowse ediff efaq \
+            eintr elisp emacs emacs-mime emacs-xtra eshell eudc flymake forms \
+            gnus idlwave info message mh-e pcl-cvs pgg reftex sc ses sieve \
+            smtpmail speedbar tramp vip viper widget woman
+do
+    ./mangle-info ${THE_INFO_PREFIX}/${file}
+done
diff -Nru emacs-21.3.50-20041229.work/lisp/startup.el emacs/lisp/startup.el
--- emacs-21.3.50-20041229.work/lisp/startup.el	Wed Dec 29 00:30:39 2004
+++ emacs/lisp/startup.el	Fri Mar 18 15:24:46 2005
@@ -286,6 +286,10 @@
 (defvar pure-space-overflow nil
   "Non-nil if building Emacs overflowed pure space.")
 
+(defconst fink-emacs-flavor 'emacs21-carbon
+  "A symbol representing the particular fink flavor of emacs running.
+Something like 'emacs20, 'xemacs20, etc.")
+
 (defun normal-top-level-add-subdirs-to-load-path ()
   "Add all subdirectories of current directory to `load-path'.
 More precisely, this uses only the subdirectories whose names
@@ -794,8 +798,19 @@
     ;; Run the site-start library if it exists.  The point of this file is
     ;; that it is run before .emacs.  There is no point in doing this after
     ;; .emacs; that is useless.
-    (if site-run-file
-	(load site-run-file t t))
+
+    ;; (if site-run-file                                                       
+    ;;     (load site-run-file t t))                                           
+                                                                               
+    ;; Fink version of site-start.                                           
+    (if site-run-file                                                          
+        (progn                                                                 
+          ;; Load all the fink package snippets.                             
+          ;; It's in here because we want -q to kill it too.                   
+          (if (load "fink-startup" t t nil)                                  
+              (fink-startup fink-emacs-flavor))                            
+          ;; Now the normal site file...                                       
+          (load site-run-file t t nil)))                                       
 
     ;; Sites should not disable this.  Only individuals should disable
     ;; the startup message.
diff -Nru emacs-21.3.50-20041229.work/mangle-info emacs/mangle-info
--- emacs-21.3.50-20041229.work/mangle-info	Thu Jan  1 09:00:00 1970
+++ emacs/mangle-info	Fri Mar 18 15:24:46 2005
@@ -0,0 +1,17 @@
+#!/usr/bin/perl -w -i
+
+use English;
+use strict;
+
+$RS = undef;
+
+my $prefix = "emacs-21-carbon";
+
+my $x = <>;
+if(!($x =~ m/^(\s*START-INFO-DIR-ENTRY\s+\*\s*[^:]+:\s*)\(([^\)]+)\)/mo)) {
+  die "Couldn't find START-INFO-DIR-ENTRY.";
+}
+   
+print ${PREMATCH};
+print "$1($prefix/$2)";
+print ${POSTMATCH};
diff -Nru emacs-21.3.50-20041229.work/src/macfns.c emacs/src/macfns.c
--- emacs-21.3.50-20041229.work/src/macfns.c	Tue Dec 28 02:27:30 2004
+++ emacs/src/macfns.c	Fri Mar 18 15:24:46 2005
@@ -1778,6 +1778,8 @@
       do_pending_window_change (0);
     }
   adjust_glyphs (f);
+
+  hide_mac_menubar(f);
 }
 
 
diff -Nru emacs-21.3.50-20041229.work/src/macterm.c emacs/src/macterm.c
--- emacs-21.3.50-20041229.work/src/macterm.c	Fri Mar 18 15:21:00 2005
+++ emacs/src/macterm.c	Fri Mar 18 15:24:46 2005
@@ -298,6 +298,11 @@
 extern void menubar_selection_callback (FRAME_PTR, int);
 extern void set_frame_menubar (FRAME_PTR, int, int);
 
+/* Make Transparency */
+int mac_transparency_alpha;
+
+void hide_mac_menubar P_ ((struct frame *));
+
 /* X display function emulation */
 
 void
@@ -4865,6 +4870,17 @@
     }
 }
 
+void
+hide_mac_menubar(f)
+     struct frame *f;
+{
+  if( !FRAME_EXTERNAL_MENU_BAR(f) && IsMenuBarVisible())
+    HideMenuBar();
+  if( FRAME_EXTERNAL_MENU_BAR(f) && !IsMenuBarVisible())
+    ShowMenuBar();
+}
+
+
 
 /* Icons.  */
 
@@ -5222,6 +5238,8 @@
   f->win_gravity = NorthWestGravity;
   x_wm_set_size_hint (f, (long) 0, 0);
 
+  SetWindowAlpha (f->output_data.mac->mWP, ((float)mac_transparency_alpha/100));
+
   SizeWindow (FRAME_MAC_WINDOW (f), pixelwidth, pixelheight, 0);
 
   /* Now, strictly speaking, we can't be sure that this is accurate,
@@ -9733,6 +9751,11 @@
 command, this enables the Mac keyboard to be used to enter non-ASCII
 characters directly.  */);
   mac_keyboard_text_encoding = kTextEncodingMacRoman;
+
+  DEFVAR_INT ("mac-transparency-alpha", &mac_transparency_alpha,
+    doc: /* Transparency ratio, can be set integer between 0 to 100.
+0 means fully transparent, 100 means non transparent */);
+  mac_transparency_alpha = 100;
 }
 
 /* arch-tag: f2259165-4454-4c04-a029-a133c8af7b5b
diff -Nru emacs-21.3.50-20041229.work/src/macterm.h emacs/src/macterm.h
--- emacs-21.3.50-20041229.work/src/macterm.h	Tue Dec 28 02:27:30 2004
+++ emacs/src/macterm.h	Fri Mar 18 15:24:46 2005
@@ -613,5 +613,7 @@
 extern CFStringRef cfstring_create_with_utf8_cstring P_ ((const char *));
 #endif
 
+extern void hide_mac_menubar P_ ((struct frame *));
+
 /* arch-tag: 6b4ca125-5bef-476d-8ee8-31ed808b7e79
    (do not change this comment) */
diff -Nru emacs-21.3.50-20041229.work/src/w32term.c emacs/src/w32term.c
--- emacs-21.3.50-20041229.work/src/w32term.c	Sun Dec 12 06:12:45 2004
+++ emacs/src/w32term.c	Fri Mar 18 15:24:46 2005
@@ -264,6 +264,11 @@
 
 static Lisp_Object Qvendor_specific_keysyms;
 
+/* Make Transparency */
+int w32_transparency_alpha;
+
+
+
 
 /***********************************************************************
 			      Debugging
@@ -5500,6 +5505,13 @@
     AdjustWindowRect(&rect, f->output_data.w32->dwStyle,
 		     FRAME_EXTERNAL_MENU_BAR (f));
 
+    SetWindowLong(FRAME_W32_WINDOW (f),GWL_EXSTYLE,WS_EX_LAYERED);
+    SetLayeredWindowAttributes(FRAME_W32_WINDOW (f),
+                               RGB(255, 255, 255),
+                               (int)((float)w32_transparency_alpha/(float)100*(float)255),
+                               LWA_ALPHA);
+    
+
     my_set_window_pos (FRAME_W32_WINDOW (f),
 		       NULL,
 		       0, 0,
@@ -6492,6 +6504,11 @@
 
 When this variable is set, other variables affecting the appearance of
 the cursor have no effect.  */);
+
+  DEFVAR_INT ("w32-transparency-alpha", &w32_transparency_alpha,
+             doc: /* Transparency ratio, can be set integer between 0 to 100.
+0 means fully transparent, 100 means non transparent */);
+  w32_transparency_alpha = 100;
 
   /* Initialize w32_use_visible_system_caret based on whether a screen
      reader is in use.  */
