diff -Nurd gstreamer-0.8.12-orig/common/m4/as-docbook.m4 gstreamer-0.8.12/common/m4/as-docbook.m4
--- gstreamer-0.8.12-orig/common/m4/as-docbook.m4	2004-02-22 16:18:08.000000000 +0100
+++ gstreamer-0.8.12/common/m4/as-docbook.m4	2006-02-28 00:10:31.000000000 +0100
@@ -14,7 +14,7 @@
   TYPE_UC=XML
   DOCBOOK_VERSION=4.1.2
 
-  if test ! -f /etc/xml/catalog; then
+  if test ! -f @PREFIX@/etc/xml/catalog; then
     for i in /usr/share/sgml/docbook/stylesheet/xsl/nwalsh /usr/share/sgml/docbook/xsl-stylesheets/;
     do
       if test -d "$i"; then
@@ -22,7 +22,7 @@
       fi
     done
   else
-    XML_CATALOG=/etc/xml/catalog
+    XML_CATALOG=@PREFIX@/etc/xml/catalog
     CAT_ENTRY_START='<!--'
     CAT_ENTRY_END='-->'
   fi
