diff -Nurd inkscape-0.44.1pre2-orig/share/extensions/Makefile.am inkscape-0.44.1pre2/share/extensions/Makefile.am
--- inkscape-0.44.1pre2-orig/share/extensions/Makefile.am	2006-09-06 07:43:07.000000000 +0200
+++ inkscape-0.44.1pre2/share/extensions/Makefile.am	2006-09-07 08:50:52.000000000 +0200
@@ -71,7 +71,6 @@
 	dia.inx \
 	epsi_output.inx \
 	pdf_output.inx \
-	pdf_output_via_gs_on_win32.inx \
 	txt2svg.inx \
 	dots.inx \
 	ffmet.inx \
diff -Nurd inkscape-0.44.1pre2-orig/share/extensions/Makefile.in inkscape-0.44.1pre2/share/extensions/Makefile.in
--- inkscape-0.44.1pre2-orig/share/extensions/Makefile.in	2006-09-06 08:26:40.000000000 +0200
+++ inkscape-0.44.1pre2/share/extensions/Makefile.in	2006-09-07 08:52:27.000000000 +0200
@@ -294,7 +294,6 @@
 	dia.inx \
 	epsi_output.inx \
 	pdf_output.inx \
-	pdf_output_via_gs_on_win32.inx \
 	txt2svg.inx \
 	dots.inx \
 	ffmet.inx \
