diff -Nurd mlview-0.9.0-orig/configure mlview-0.9.0/configure
--- mlview-0.9.0-orig/configure	2005-08-07 10:16:32.000000000 +0200
+++ mlview-0.9.0/configure	2006-04-30 17:12:06.000000000 +0200
@@ -2272,7 +2272,7 @@
 \* 10000 + $MLVIEW_MINOR_VERSION \
 \* 100 + $MLVIEW_MICRO_VERSION`
 
-MLVIEW_EXE=$prefix/bin/mlv
+MLVIEW_EXE=$prefix/bin/mlview
 
 
 
diff -Nurd mlview-0.9.0-orig/mlview.desktop.in mlview-0.9.0/mlview.desktop.in
--- mlview-0.9.0-orig/mlview.desktop.in	2005-03-12 18:58:45.000000000 +0100
+++ mlview-0.9.0/mlview.desktop.in	2006-04-30 15:53:34.000000000 +0200
@@ -7,7 +7,7 @@
 Categories=Application;Development;
 Comment=A generic simple XML editor for GNOME
 Icon=mlview-app-icon.xpm
-TryExec=mlv
+TryExec=mlview
 Exec=@MLVIEW_EXE@ %F
 Terminal=false
 
diff -Nurd mlview-0.9.0-orig/src/mlview-exec-command-dialog.cc mlview-0.9.0/src/mlview-exec-command-dialog.cc
--- mlview-0.9.0-orig/src/mlview-exec-command-dialog.cc	2005-08-07 09:47:54.000000000 +0200
+++ mlview-0.9.0/src/mlview-exec-command-dialog.cc	2006-04-30 15:53:02.000000000 +0200
@@ -24,6 +24,13 @@
  *See COPYRIGHT file for copyright information.
  */
 
+#ifdef __APPLE__
+#include <crt_externs.h>
+#define environ (*_NSGetEnviron())
+#else
+extern char **environ;
+#endif
+
 #include <unistd.h>
 #include <stdlib.h>
 #include <gnome.h>
diff -Nurd mlview-0.9.0-orig/tests/firstplugin.xml mlview-0.9.0/tests/firstplugin.xml
--- mlview-0.9.0-orig/tests/firstplugin.xml	2005-08-07 10:30:32.000000000 +0200
+++ mlview-0.9.0/tests/firstplugin.xml	2006-04-30 16:02:03.000000000 +0200
@@ -1,5 +1,5 @@
 <plugin>
-  <file>/usr/local/lib/mlview/plugins/libfirstplugin.so</file>
+  <file>@PREFIX@/lib/mlview/plugins/libfirstplugin.dylib</file>
   <name>FirstPlugin</name>
   <description>MlView's first plugin !</description>
   <load>first_plugin_load</load>
