diff -Nurd -x'*~' yelp-2.12.2.orig/configure yelp-2.12.2/configure
--- yelp-2.12.2.orig/configure	2005-11-27 15:57:40.000000000 -0500
+++ yelp-2.12.2/configure	2005-12-07 16:50:06.000000000 -0500
@@ -24159,18 +24159,10 @@
 _ACEOF
 
 
-case $host in
-  *-linux-gnu | *-irix6*)
 cat >>confdefs.h <<\_ACEOF
 #define HAVE_MAN_WHEREIS 1
-_ACEOF
-;;
-  *-solaris*)
-cat >>confdefs.h <<\_ACEOF
 #define HAVE_MAN_DEBUG 1
 _ACEOF
-;;
-esac
 
 echo "$as_me:$LINENO: checking for poptGetArgs in -lpopt" >&5
 echo $ECHO_N "checking for poptGetArgs in -lpopt... $ECHO_C" >&6
diff -Nurd -x'*~' yelp-2.12.2.orig/po/Makefile.in.in yelp-2.12.2/po/Makefile.in.in
--- yelp-2.12.2.orig/po/Makefile.in.in	2005-11-27 15:57:23.000000000 -0500
+++ yelp-2.12.2/po/Makefile.in.in	2005-12-07 16:50:06.000000000 -0500
@@ -28,7 +28,7 @@
 exec_prefix = @exec_prefix@
 datadir = @datadir@
 libdir = @libdir@
-localedir = $(libdir)/locale
+localedir = $(datadir)/locale
 gnulocaledir = $(datadir)/locale
 gettextsrcdir = $(datadir)/glib-2.0/gettext/po
 subdir = po
