diff -uNbr libpng-1.2.5/scripts/makefile.darwin libpng-1.2.5-new/scripts/makefile.darwin
--- libpng-1.2.5/scripts/makefile.darwin	Thu Oct  3 07:32:34 2002
+++ libpng-1.2.5-new/scripts/makefile.darwin	Sat Nov  9 02:00:32 2002
@@ -84,17 +84,15 @@
 	ln -sf $(LIBNAME).$(PNGVER).dylib $(LIBNAME).$(PNGMAJ).dylib
 
 $(LIBNAME).$(PNGVER).dylib: $(OBJSDLL)
-	$(CC) -dynamiclib \
-	 -install_name $(DL)/$(LIBNAME).$(PNGMAJ).dylib \
-	 -flat_namespace -undefined suppress \
+	gcc-prebind $(CC) -dynamiclib \
+	 -install_name $(LIBPATH)/$(LIBNAME).$(PNGMAJ).dylib \
 	 -o $(LIBNAME).$(PNGVER).dylib \
 	 $(OBJSDLL)
 
 libpng.3.$(PNGMIN).dylib: $(OBJSDLL)
-	$(CC) -dynamiclib \
-	 -install_name $(DL)/libpng.3.dylib \
+	gcc-prebind $(CC) -dynamiclib \
+	 -install_name $(LIBPATH)/libpng.3.dylib \
 	 -current_version 3 -compatibility_version 3 \
-	 -flat_namespace -undefined suppress \
 	 -o libpng.3.$(PNGMIN).dylib \
 	 $(OBJSDLL)
 
