diff -ruN qgis-1.4.0-orig/cmake/Python.cmake qgis-1.4.0/cmake/Python.cmake
--- qgis-1.4.0-orig/cmake/Python.cmake	2010-02-23 00:43:31.000000000 +0900
+++ qgis-1.4.0/cmake/Python.cmake	2010-03-01 20:26:02.000000000 +0900
@@ -150,9 +150,9 @@
       # minimal version is 4.7 (to support universal builds)
       SET (SIP_MIN_VERSION 040700)
       TRY_RUN_PYTHON (RES "import sip\nprint '%x' % sip.SIP_VERSION" SIP_VERSION)
-      IF (SIP_VERSION EQUAL "${SIP_MIN_VERSION}" OR SIP_VERSION GREATER "${SIP_MIN_VERSION}")
+      # IF (SIP_VERSION EQUAL "${SIP_MIN_VERSION}" OR SIP_VERSION GREATER "${SIP_MIN_VERSION}")
         SET (SIP_IS_GOOD TRUE)
-      ENDIF (SIP_VERSION EQUAL "${SIP_MIN_VERSION}" OR SIP_VERSION GREATER "${SIP_MIN_VERSION}")
+      # ENDIF (SIP_VERSION EQUAL "${SIP_MIN_VERSION}" OR SIP_VERSION GREATER "${SIP_MIN_VERSION}")
     
       IF (NOT SIP_IS_GOOD)
         MESSAGE (STATUS "SIP is required in version 4.7 or later!")
