Index: perlmod/Fink/ChangeLog
===================================================================
RCS file: /cvsroot/fink/fink/perlmod/Fink/ChangeLog,v
retrieving revision 1.710
diff -u -r1.710 ChangeLog
--- perlmod/Fink/ChangeLog	30 Sep 2004 04:17:28 -0000	1.710
+++ perlmod/Fink/ChangeLog	4 Oct 2004 21:12:55 -0000
@@ -1,3 +1,9 @@
+2004-09-29  Benjamin Reed  <rangerrick@users.sourceforge.net>
+
+	* PkgVersion.pm, Validation.pm: Implemented AppBundles:
+	which does essentially what JarFiles does, copies an application
+	bundle to a known directory (%p/Applications)
+
 2004-09-28  Daniel Macks  <dmacks@netspace.org>
 
 	* Engine.pm: Implemented 'fink show-deps' (cmd_show_deps() and
Index: perlmod/Fink/PkgVersion.pm
===================================================================
RCS file: /cvsroot/fink/fink/perlmod/Fink/PkgVersion.pm,v
retrieving revision 1.303
diff -u -r1.303 PkgVersion.pm
--- perlmod/Fink/PkgVersion.pm	23 Sep 2004 10:01:59 -0000	1.303
+++ perlmod/Fink/PkgVersion.pm	4 Oct 2004 21:12:55 -0000
@@ -39,7 +39,7 @@
 use Fink::Bootstrap qw(&get_bsbase);
 use Fink::Command qw(mkdir_p rm_f rm_rf symlink_f du_sk);
 
-use File::Basename qw(&dirname);
+use File::Basename qw(&dirname &basename);
 
 use POSIX qw(uname);
 
@@ -1973,6 +1973,16 @@
 		}
 	}
 
+	# generate commands to install App bundles
+	if ($self->has_param("AppBundles")) {
+		$install_script .= "\n/usr/bin/install -d -m 755 %i/Applications";
+      for my $bundle (split(/\s+/, $self->param("AppBundles"))) {
+        $bundle =~ s/\'/\\\'/gsi;
+        my $shortname = basename($bundle);
+        $install_script .= "\ncp -pR '$bundle' '%i/Applications/'"
+      }
+	}
+
 	# generate commands to install jar files
 	if ($self->has_param("JarFiles")) {
 		my (@jarfiles, $jarfile, $jarfilelist);
Index: perlmod/Fink/Validation.pm
===================================================================
RCS file: /cvsroot/fink/fink/perlmod/Fink/Validation.pm,v
retrieving revision 1.143
diff -u -r1.143 Validation.pm
--- perlmod/Fink/Validation.pm	21 Sep 2004 22:20:13 -0000	1.143
+++ perlmod/Fink/Validation.pm	4 Oct 2004 21:12:55 -0000
@@ -170,6 +170,7 @@
 #  install phase:
 		 'updatepod',
 		 'installscript',
+		 'appbundles',
 		 'jarfiles',
 		 'docfiles',
 		 'shlibs',
