Index: inject.pl
===================================================================
RCS file: /cvsroot/fink/fink/inject.pl,v
retrieving revision 1.10
diff -u -r1.10 inject.pl
--- inject.pl	22 Dec 2002 22:09:01 -0000	1.10
+++ inject.pl	11 Jan 2003 03:53:26 -0000
@@ -22,7 +22,7 @@
 #
 
 $| = 1;
-use v5.6.0;  # perl 5.6.0 or newer required
+use 5.006;  # perl 5.6.0 or newer required
 use strict;
 
 use FindBin;
Index: perlmod/Fink/PkgVersion.pm
===================================================================
RCS file: /cvsroot/fink/fink/perlmod/Fink/PkgVersion.pm,v
retrieving revision 1.102
diff -u -r1.102 PkgVersion.pm
--- perlmod/Fink/PkgVersion.pm	6 Jan 2003 03:31:45 -0000	1.102
+++ perlmod/Fink/PkgVersion.pm	11 Jan 2003 03:53:27 -0000
@@ -1213,10 +1213,17 @@
   if ($self->has_param("CompileScript")) {
     $compile_script = $self->param("CompileScript");
   } elsif ($self->param("_type") eq "perl") {
-    $compile_script =
-      "perl Makefile.PL PREFIX=\%p INSTALLPRIVLIB=\%p/lib/perl5 INSTALLARCHLIB=\%p/lib/perl5/darwin INSTALLSITELIB=\%p/lib/perl5 INSTALLSITEARCH=\%p/lib/perl5/darwin INSTALLMAN1DIR=\%p/share/man/man1 INSTALLMAN3DIR=\%p/share/man/man3\n".
-      "make\n".
-      "make test";
+    if (-d $basepath . '/lib/perl' and -x $basepath . '/bin/perl') {
+      $compile_script =
+        "${basepath}/bin/perl Makefile.PL INSTALLDIRS=vendor\n".
+        "make\n".
+        "make test";
+    } else {
+      $compile_script =
+        "/usr/bin/perl Makefile.PL PREFIX=\%p INSTALLPRIVLIB=\%p/lib/perl5 INSTALLARCHLIB=\%p/lib/perl5/darwin INSTALLSITELIB=\%p/lib/perl5 INSTALLSITEARCH=\%p/lib/perl5/darwin INSTALLMAN1DIR=\%p/share/man/man1 INSTALLMAN3DIR=\%p/share/man/man3\n".
+        "make\n".
+        "make test";
+    }
   } else {
     $compile_script = 
       "./configure \%c\n".
@@ -1276,8 +1283,23 @@
       # Now run the custom install script
       $self->run_script($self->param("InstallScript"), "installing");
     } elsif ($self->param("_type") eq "perl") {
-      $install_script .= 
-	"make install PREFIX=\%i INSTALLPRIVLIB=\%i/lib/perl5 INSTALLARCHLIB=\%i/lib/perl5/darwin INSTALLSITELIB=\%i/lib/perl5 INSTALLSITEARCH=\%i/lib/perl5/darwin INSTALLMAN1DIR=\%i/share/man/man1 INSTALLMAN3DIR=\%i/share/man/man3\n";
+      if (-d $basepath . '/lib/perl' and $basepath . '/bin/perl') {
+	my $vendorpath;
+	my $install_args;
+        for my $var ('vendorprefix', 'installvendorarch', 'installvendorlib', 'installvendorbin', 'installscript', 'installman1dir', 'installman3dir') {
+          chomp($vendorpath = `${basepath}/bin/perl -V:${var}`);
+	  if ($vendorpath =~ s/^${var}=\'${basepath}(.*)\';$/\%i$1/) {
+	    $install_args .= ' ' . uc($var) . '=' . $vendorpath;
+	  } else {
+	    warn "couldn't get perl path for '${var}'";
+	  }
+        }
+        $install_script .= 
+	  "make install INSTALLDIRS=vendor ${install_args}\n";
+      } else {
+        $install_script .= 
+	  "make install PREFIX=\%i INSTALLPRIVLIB=\%i/lib/perl5 INSTALLARCHLIB=\%i/lib/perl5/darwin INSTALLSITELIB=\%i/lib/perl5 INSTALLSITEARCH=\%i/lib/perl5/darwin INSTALLMAN1DIR=\%i/share/man/man1 INSTALLMAN3DIR=\%i/share/man/man3\n";
+      }
     } elsif (not $do_splitoff) {
       $install_script .= "make install prefix=\%i\n";
     } 
