Index: perlmod/Fink/PkgVersion.pm
===================================================================
RCS file: /cvsroot/fink/fink/perlmod/Fink/PkgVersion.pm,v
retrieving revision 1.176
diff -u -r1.176 PkgVersion.pm
--- perlmod/Fink/PkgVersion.pm	20 Nov 2003 18:22:03 -0000	1.176
+++ perlmod/Fink/PkgVersion.pm	21 Nov 2003 15:33:02 -0000
@@ -1010,33 +1010,6 @@
 		return;
 	}
 
-	my ($gcc);
-	my %gcchash = ('2.95.2' => '2', '2.95' => '2', '3.1' => '3', '3.3' => '3.3');
-
-	if ($self->has_param("GCC")) {
-		$gcc = $self->param("GCC");
-		chomp(my $gcc_select = `gcc_select`);
-		if (not $gcc_select =~ s/^.*gcc version (\S+)\s+.*$/$1/gs) {
-			$gcc_select = 'an unknown version';
-		}
-		if (not exists $gcchash{$gcc}) {
-			$gcchash{$gcc} = $gcc;
-		}
-		if ($gcc_select !~ /^$gcc/) {
-			die <<END;
-
-This package must be compiled with GCC $gcc, but you currently have $gcc_select selected.
-To correct this problem, run the command:
-
-	sudo gcc_select $gcchash{$gcc}
-
-You may need to install a more recent version of the Developer Tools to be able
-to do so.
-
-END
-		}
-	}
-
 	$bdir = $self->get_fullname();
 
 	$verbosity = "";
@@ -2005,10 +1978,26 @@
 
 sub set_env {
 	my $self = shift;
-	my ($varname, $s, $expand);
+	my ($varname, $s, $expand, $postfix);
+
+	my %gccpostfix = (
+		'2.95.2' => '2',
+		'2.95'   => '2',
+		'3.1'    => '3',
+		'3.3'    => '-3.3',
+	);
+	
+	if ($config->param("Distribution") ge "10.2-gcc3.3") {
+		$postfix = $gccpostfix{$self->param_default('GCC', '3.3')};
+	} else {
+		$postfix = $gccpostfix{$self->param_default('GCC', '3.1')};
+	}
+
 	my %defaults = (
 		"CPPFLAGS"                 => "-I\%p/include",
 		"LDFLAGS"                  => "-L\%p/lib",
+		"CC"                       => "gcc$postfix",
+		"CXX"                      => "g++$postfix",
 		"LD_PREBIND"               => 1,
 		"LD_PREBIND_ALLOW_OVERLAP" => 1,
 		"LD_SEG_ADDR_TABLE"        => "$basepath/var/lib/fink/prebound/seg_addr_table",
@@ -2044,6 +2033,7 @@
 		my @vars = `sh -c ". $basepath/bin/init.sh ; /usr/bin/env"`;
 		chomp @vars;
 		%ENV = map { split /=/,$_,2 } @vars;
+		$defaults{'MAKEFLAGS'} = $ENV{'MAKEFLAGS'} if (exists $ENV{'MAKEFLAGS'});
 	}
 
 	# set variables according to the info file
