? perlmod/Fink/.PkgVersion.pm.swp
Index: perlmod/Fink/ChangeLog
===================================================================
RCS file: /cvsroot/fink/fink/perlmod/Fink/ChangeLog,v
retrieving revision 1.532
diff -u -u -r1.532 ChangeLog
--- perlmod/Fink/ChangeLog	30 Jan 2004 22:11:47 -0000	1.532
+++ perlmod/Fink/ChangeLog	1 Feb 2004 04:00:14 -0000
@@ -1,3 +1,8 @@
+2004-01-31  BEnjamin Reed  <rangerrick@users.sourceforge.net>
+
+	* PkgVersion.pm, Valication.pm: Add support for BundleFiles
+	and all the related stuff to symlink stuff to /sw/Applications
+
 2004-01-30  Daniel Macks  <dmacks@netspace.org>
 
 	* PkgVersion.pm: Added second parameter to add_splitoff() for
Index: perlmod/Fink/PkgVersion.pm
===================================================================
RCS file: /cvsroot/fink/fink/perlmod/Fink/PkgVersion.pm,v
retrieving revision 1.219
diff -u -u -r1.219 PkgVersion.pm
--- perlmod/Fink/PkgVersion.pm	30 Jan 2004 22:11:47 -0000	1.219
+++ perlmod/Fink/PkgVersion.pm	1 Feb 2004 04:00:15 -0000
@@ -1690,8 +1690,30 @@
 		}
 	}
 
+	# generate commands to install app bundles
+	if ($self->has_param("BundleFiles")) {
+		my ($bundle, $escaped_bundle);
+		for $bundle (split(/\s+/, $self->param("BundleFiles"))) {
+			$escaped_bundle = $bundle;
+			$escaped_bundle =~ s/'/\\\'/g;
+			if (/\.framework$/) {
+				$install_script .= "\n/usr/bin/install -d -m 755 '\%i/Library/Frameworks'";
+				$install_script .= "\n/bin/mv -f '$escaped_bundle' '\%i/Library/Frameworks/'";
+			} else {
+				$install_script .= "\n/usr/bin/install -d -m 755 '\%i/.Applications'";
+				$install_script .= "\n/bin/mv -f '$escaped_bundle' '\%i/.Applications'";
+				$install_script .= "\n/usr/bin/install -d -m 755 '\%i/var/lib/fink/bundles'";
+				$install_script .= "\necho '$escaped_bundle' >> '\%i/var/lib/fink/bundles/\%n.list'";
+			}
+		}
+	}
+
+	# clean out old GNU info directories, just in case
 	$install_script .= "\n/bin/rm -f %i/info/dir %i/info/dir.old %i/share/info/dir %i/share/info/dir.old";
 
+	# handle resource forks nicely -- these get recombined in the postinstall
+	$install_script .= "\n[ -x /Developer/Tools/SplitForks ] \&\& /Developer/Tools/SplitForks %d";
+
 	### install
 
 	$self->run_script($install_script, "installing");
@@ -1932,11 +1954,22 @@
 	### create scripts as neccessary
 
 	foreach $scriptname (qw(preinst postinst prerm postrm)) {
+
+		$scriptbody = "";
+
+		# fix the resource bundles of all BundlFiles
+		if ($scriptname eq "postinst" and $self->has_param("BundleFiles")) {
+			$scriptbody .= "\n [ -x /System/Library/CoreServices/FixupResourceForks ] \&\& /System/Library/CoreServices/FixupResourceForks ";
+			my $bundle;
+			for $bundle ($self->has_param("BundleFiles")) {
+				$bundle =~ s/.*\///;
+				$scriptbody .= "\"\%p/.Applications/$bundle\" ";
+			}
+		}
+
 		# get script piece from package description
 		if ($self->has_param($scriptname."Script")) {
-			$scriptbody = $self->param($scriptname."Script");
-		} else {
-			$scriptbody = "";
+			$scriptbody .= $self->param($scriptname."Script");
 		}
 
 		# add UpdatePOD Code
@@ -1982,6 +2015,24 @@
 			}
 		}
 
+		# fix the Applications directory
+		if ($self->has_param("BundleFiles")) {
+			if ($scriptname eq "postinst") {
+				$scriptbody .=
+					"\ncat '\%p/var/lib/fink/bundles/\%n.list' | while read FILENAME; do" .
+					"\n  if [ ! -e \"\%p/Applications/\$FILENAME\" ]; then" .
+					"\n    /usr/bin/install -d -m 755 \"\%p/Applications\"" .
+					"\n    ln -sf \"\%p/.Applications/\$FILENAME\" \"%p/Applications/\"" .
+					"\n  fi" .
+					"\ndone";
+			} elsif ($scriptname eq "postrm") {
+				$scriptbody .=
+					"\ncat '\%p/var/lib/fink/bundles/\%n.list' | while read FILENAME; do" .
+					"\n  rm -rf \"\%p/Applications/\$FILENAME\"" .
+					"\ndone";
+			}
+		}
+
 		# add auto-generated parts
 		if ($self->has_param("InfoDocs")) {
 			if ($scriptname eq "postinst") {
@@ -2200,8 +2251,8 @@
 	my $self = shift;
 	my ($varname, $s, $expand);
 	my %defaults = (
-		"CPPFLAGS"                 => "-I\%p/include",
-		"LDFLAGS"                  => "-L\%p/lib",
+		"CPPFLAGS"                 => "-I\%p/include -F\%p/Library/Frameworks",
+		"LDFLAGS"                  => "-L\%p/lib -F\%p/Library/Frameworks",
 		"LD_PREBIND"               => 1,
 		"LD_PREBIND_ALLOW_OVERLAP" => 1,
 		"LD_SEG_ADDR_TABLE"        => "$basepath/var/lib/fink/prebound/seg_addr_table",
Index: perlmod/Fink/Validation.pm
===================================================================
RCS file: /cvsroot/fink/fink/perlmod/Fink/Validation.pm,v
retrieving revision 1.98
diff -u -u -r1.98 Validation.pm
--- perlmod/Fink/Validation.pm	28 Jan 2004 11:18:08 -0000	1.98
+++ perlmod/Fink/Validation.pm	1 Feb 2004 04:00:17 -0000
@@ -182,6 +182,7 @@
 #  install phase:
 		 'updatepod',
 		 'installscript',
+		 'bundlefiles',
 		 'jarfiles',
 		 'docfiles',
 		 'shlibs',
@@ -229,6 +230,7 @@
 #  install phase:
 		 'updatepod',
 		 'installscript',
+		 'bundlefiles',
 		 'jarfiles',
 		 'docfiles',
 		 'shlibs',
