Index: perlmod/Fink/Engine.pm
===================================================================
RCS file: /cvsroot/fink/fink/perlmod/Fink/Engine.pm,v
retrieving revision 1.140
diff -u -r1.140 Engine.pm
--- perlmod/Fink/Engine.pm	13 Dec 2003 19:04:15 -0000	1.140
+++ perlmod/Fink/Engine.pm	16 Dec 2003 01:26:32 -0000
@@ -1047,6 +1047,23 @@
 		$ep = [ Fink::Package->package_by_name($ep)->get_all_versions() ];
 	}
 
+	# check for recommends/suggests
+	foreach $pkgname (@queue) {
+		$item = $deps{$pkgname};
+		for ('suggests', 'recommends') {
+			if ($item->[PKGVER]->has_param($_)) {
+				foreach $pkgspec (split(/\s*,\s*/, $item->[PKGVER]->param($_))) {
+					$package = Fink::PkgVersion->match_package($pkgspec);
+					if (defined $package) {
+						print $_, ": ", $package->get_name(), "\n";
+					} else {
+						warn $pkgname, " suggests '", $pkgspec, "', but a match could not be found!\n";
+					}
+				}
+			}
+		}
+	}
+
 	# recursively expand dependencies
 	while ($#queue >= 0) {
 		$pkgname = shift @queue;
