Index: perlmod/Fink/PkgVersion.pm
===================================================================
RCS file: /cvsroot/fink/fink/perlmod/Fink/PkgVersion.pm,v
retrieving revision 1.253
diff -u -u -r1.253 PkgVersion.pm
--- perlmod/Fink/PkgVersion.pm	11 Apr 2004 20:01:26 -0000	1.253
+++ perlmod/Fink/PkgVersion.pm	16 Apr 2004 13:08:58 -0000
@@ -2382,6 +2382,7 @@
 		my @vars = `sh -c ". $basepath/bin/init.sh ; /usr/bin/env"`;
 		chomp @vars;
 		%ENV = map { split /=/,$_,2 } @vars;
+		$defaults{'MAKEFLAGS'} = $ENV{'MAKEFLAGS'} if (exists $ENV{'MAKEFLAGS'});
 	}
 
 	# set variables according to the info file
