Index: perlmod/Fink/PkgVersion.pm
===================================================================
RCS file: /cvsroot/fink/fink/perlmod/Fink/PkgVersion.pm,v
retrieving revision 1.278
diff -u -u -r1.278 PkgVersion.pm
--- perlmod/Fink/PkgVersion.pm	1 Jul 2004 14:39:56 -0000	1.278
+++ perlmod/Fink/PkgVersion.pm	1 Jul 2004 14:44:32 -0000
@@ -2554,6 +2554,7 @@
 		my @vars = `sh -c ". $basepath/bin/init.sh ; /usr/bin/env"`;
 		chomp @vars;
 		%ENV = map { split /=/,$_,2 } @vars;
+		$defaults{'MAKEFLAGS'} = $ENV{'MAKEFLAGS'} if (exists $ENV{'MAKEFLAGS'});
 	}
 
 	# set variables according to the info file
