Index: perlmod/Fink/PkgVersion.pm
===================================================================
RCS file: /cvsroot/fink/fink/perlmod/Fink/PkgVersion.pm,v
retrieving revision 1.271
diff -u -r1.271 PkgVersion.pm
--- perlmod/Fink/PkgVersion.pm	22 Jun 2004 14:41:00 -0000	1.271
+++ perlmod/Fink/PkgVersion.pm	23 Jun 2004 02:59:52 -0000
@@ -2520,6 +2520,7 @@
 		my @vars = `sh -c ". $basepath/bin/init.sh ; /usr/bin/env"`;
 		chomp @vars;
 		%ENV = map { split /=/,$_,2 } @vars;
+		$defaults{'MAKEFLAGS'} = $ENV{'MAKEFLAGS'} if (exists $ENV{'MAKEFLAGS'});
 	}
 
 	# set variables according to the info file
