Index: perlmod/Fink/PkgVersion.pm
===================================================================
RCS file: /cvsroot/fink/fink/perlmod/Fink/PkgVersion.pm,v
retrieving revision 1.219
diff -u -u -r1.219 PkgVersion.pm
--- perlmod/Fink/PkgVersion.pm	30 Jan 2004 22:11:47 -0000	1.219
+++ perlmod/Fink/PkgVersion.pm	1 Feb 2004 01:52:55 -0000
@@ -2240,6 +2240,7 @@
 		my @vars = `sh -c ". $basepath/bin/init.sh ; /usr/bin/env"`;
 		chomp @vars;
 		%ENV = map { split /=/,$_,2 } @vars;
+		$defaults{'MAKEFLAGS'} = $ENV{'MAKEFLAGS'} if (exists $ENV{'MAKEFLAGS'});
 	}
 
 	# set variables according to the info file
