Index: perlmod/Fink/PkgVersion.pm
===================================================================
RCS file: /cvsroot/fink/fink/perlmod/Fink/PkgVersion.pm,v
retrieving revision 1.218
diff -u -r1.218 PkgVersion.pm
--- perlmod/Fink/PkgVersion.pm	23 Jan 2004 09:35:05 -0000	1.218
+++ perlmod/Fink/PkgVersion.pm	31 Jan 2004 02:48:25 -0000
@@ -2239,6 +2239,7 @@
 		my @vars = `sh -c ". $basepath/bin/init.sh ; /usr/bin/env"`;
 		chomp @vars;
 		%ENV = map { split /=/,$_,2 } @vars;
+		$defaults{'MAKEFLAGS'} = $ENV{'MAKEFLAGS'} if (exists $ENV{'MAKEFLAGS'});
 	}
 
 	# set variables according to the info file
