diff -ruN Macaulay2-0.9.2/Macaulay2/README.MacOSX Macaulay2-0.9.2-patched/Macaulay2/README.MacOSX
--- Macaulay2-0.9.2/Macaulay2/README.MacOSX	Wed Dec 31 19:00:00 1969
+++ Macaulay2-0.9.2-patched/Macaulay2/README.MacOSX	Wed May  7 08:58:13 2003
@@ -0,0 +1,42 @@
+
+
+           Macaulay2 on Mac OS X
+
+
+1. Installation
+
+The easiest way to install Macaulay2 under Mac OS X is to use the Fink
+system <http://fink.sourceforge.net>.  Follow the instructions at the
+Fink website to install Fink itself, and then install Fink's "macaulay2"
+package.  You can choose the binary installation method, or the source
+installation method (which will download the source files and compile
+them on your machine).  You may also wish to install emacs or xemacs
+using Fink, for the best interface to Macaulay2.
+
+Fink can also be used to install other software of interest to
+mathematicians.  In most cases, you simply install the relevant Fink
+package.  In two special cases -- the X11 windowing system, and TeX --
+it is possible to install the software using another method, and then
+tell Fink about it (by installing Fink's packages "system-xfree86" and
+"system-tetex", respectively).
+
+
+2. Running Macaulay2
+
+The best way to run Macaulay2 is under emacs or xemacs.  Before running
+Macaulay2 for the first time, edit the .emacs file in your home directory
+to include the line
+
+ (load "@PREFIX@/lib/Macaulay2-0.9.2/emacs/M2-init.el")
+
+For further information about running Macaulay2 under emacs, please consult
+the file
+
+  @PREFIX@/lib/Macaulay2-0.9.2/emacs/emacs-hlp.txt
+
+
+  The stacksize under Mac OS X is not always large enough to handle 
+computations in Macaulay2.  (The symptom is a segmentation fault when you try
+a large calculation.)  The cure is to run the command "limit stacksize 5m"
+prior to the command "M2".  (You can experiment with different values of
+the stacksize to find a suitable one for your computation.)
