diff -ruN Singular-3-0-0/factory/NTLconvert.h Singular-3-0-0-patched/factory/NTLconvert.h
--- Singular-3-0-0/factory/NTLconvert.h	2004-01-19 12:26:20.000000000 +0100
+++ Singular-3-0-0-patched/factory/NTLconvert.h	2006-07-08 18:37:47.000000000 +0200
@@ -30,7 +30,7 @@
 #include <NTL/GF2EXFactoring.h>
 
 #ifdef NTL_CLIENT               // in <NTL/tools.h>: using of name space NTL
-NTL_CLIENT
+NTL_USE_NNS
 #endif
 
 ZZ_pX convertFacCF2NTLZZpX(CanonicalForm f);
diff -ruN Singular-3-0-0/factory/imm.h Singular-3-0-0-patched/factory/imm.h
--- Singular-3-0-0/factory/imm.h	2001-06-27 15:20:37.000000000 +0200
+++ Singular-3-0-0-patched/factory/imm.h	2006-07-08 18:37:08.000000000 +0200
@@ -41,7 +41,7 @@
 
 inline int imm2int ( const InternalCF * const imm )
 {
-    return (int)imm >> 2;
+    return (int)(long)imm >> 2;
 }
 
 inline InternalCF * int2imm ( int i )
@@ -85,7 +85,7 @@
 inline int is_imm ( const InternalCF * const ptr )
 {
     // returns 0 if ptr is not immediate
-    return ( (int)ptr & 3 );
+    return ( (int)(long)ptr & 3 );
 }
 
 //{{{ inline int imm_isone, imm_isone_p, imm_isone_gf ( const InternalCF * const ptr )
