diff -ruN Singular-3-0-0/libfac/charset/alg_factor.cc Singular-3-0-0-ptched/libfac/charset/alg_factor.cc
--- Singular-3-0-0/libfac/charset/alg_factor.cc	2005-07-08 11:18:15.000000000 +0200
+++ Singular-3-0-0-ptched/libfac/charset/alg_factor.cc	2006-07-08 18:46:28.000000000 +0200
@@ -31,6 +31,8 @@
 #include "timing.h"
 TIMING_DEFINE_PRINT(newfactoras_time);
 
+Variable rootOf(const CanonicalForm &, char name);
+
 static Varlist
 Var_is_in_AS(const Varlist & uord, const CFList & Astar);
 
diff -ruN Singular-3-0-0/libfac/charset/csutil.cc Singular-3-0-0-ptched/libfac/charset/csutil.cc
--- Singular-3-0-0/libfac/charset/csutil.cc	2003-05-28 13:52:52.000000000 +0200
+++ Singular-3-0-0-ptched/libfac/charset/csutil.cc	2006-07-08 18:47:12.000000000 +0200
@@ -311,7 +311,7 @@
   int count=0;
 
   for ( CFIterator j=init; j.hasTerms(); j++ )
-    if (j.coeff() != 1 ) count += 1;
+    if (j.coeff().isOne() ) count += 1;
   //  if ( init != 1 ){
   //  cout << "nopower: f is " << init << endl;
   //  cout << "nopower: count is " << count << endl;}
diff -ruN Singular-3-0-0/libfac/factor/Factor.cc Singular-3-0-0-ptched/libfac/factor/Factor.cc
--- Singular-3-0-0/libfac/factor/Factor.cc	2004-12-10 11:15:06.000000000 +0100
+++ Singular-3-0-0-ptched/libfac/factor/Factor.cc	2006-07-08 18:48:01.000000000 +0200
@@ -47,6 +47,8 @@
 
 void out_cf(char *s1,const CanonicalForm &f,char *s2);
 
+Variable rootOf(const CanonicalForm &, char name);
+
 /*
 * a wrapper for factorize over algebraic extensions:
 * does a sanity check and, if nec., a conversion
