diff -ruN squid-2.6.STABLE16/src/cf.data.pre squid-2.6.STABLE16-patched/src/cf.data.pre
--- squid-2.6.STABLE16/src/cf.data.pre	2007-09-05 14:50:15.000000000 -0700
+++ squid-2.6.STABLE16-patched/src/cf.data.pre	2007-09-12 06:37:09.000000000 -0700
@@ -539,6 +539,8 @@
 	  # NOTE: proxy_auth requires a EXTERNAL authentication program
 	  # to check username/password combinations (see
 	  # auth_param directive).
+# Allow proxy connections from localhost                                      
+http_access allow localhost                                                    
 	  #
 	  # NOTE: proxy_auth can't be used in a transparent proxy as
 	  # the browser needs to be configured for using a proxy in order
@@ -3412,7 +3414,7 @@
 NAME: visible_hostname
 TYPE: string
 LOC: Config.visibleHostname
-DEFAULT: none
+DEFAULT: localhost
 DOC_START
 	If you want to present a special hostname in error messages, etc,
 	define this.  Otherwise, the return value of gethostname()
