diff -ruN tetex-src-2.99.3.20041109-beta/Makefile.in tetex-src-2.99.3.20041109-beta-patched/Makefile.in
--- tetex-src-2.99.3.20041109-beta/Makefile.in	Sat Nov  6 17:09:05 2004
+++ tetex-src-2.99.3.20041109-beta-patched/Makefile.in	Mon Nov  8 22:21:49 2004
@@ -63,25 +63,6 @@
 	test -x texk/web2c/omegafonts/omfonts && \
 	  cd texk/web2c/omegafonts; $(MAKE) install-links
 	$(srcdir)/selfautofix "$(bindir)" "$(texmf)/web2c"
-	- if test -f $(scriptdir)/mktexlsr; then \
-	  TEXMFMAIN=$(texmf) PATH=$(scriptdir):$(bindir):$(PATH) \
-	    $(scriptdir)/mktexlsr; \
-	else :; fi
-	- if test -f $(scriptdir)/fmtutil; then \
-	  TEXMFMAIN=$(texmf) PATH=$(scriptdir):$(bindir):$(PATH) \
-	    $(scriptdir)/fmtutil --all; \
-	else :; fi
-	- if test -f $(scriptdir)/texlinks; then \
-	  TEXMFMAIN=$(texmf) PATH=$(scriptdir):$(bindir):$(PATH) \
-	    $(scriptdir)/texlinks; \
-	else :; fi
-	- if test -f $(scriptdir)/updmap \
-	  && TEXMFMAIN=$(texmf) PATH=$(scriptdir):$(bindir):$(PATH) \
-	       $(scriptdir)/kpsewhich --format='web2c files' updmap.cfg >/dev/null; \
-	then \
-	  TEXMFMAIN=$(texmf) PATH=$(scriptdir):$(bindir):$(PATH) \
-	    $(scriptdir)/updmap; \
-	else :; fi
 
 install-exec:
 	for dir in $(INSTDIRS); do \
diff -ruN tetex-src-2.99.3.20041109-beta/README.fink tetex-src-2.99.3.20041109-beta-patched/README.fink
--- tetex-src-2.99.3.20041109-beta/README.fink	Wed Dec 31 19:00:00 1969
+++ tetex-src-2.99.3.20041109-beta-patched/README.fink	Mon Nov  8 21:38:50 2004
@@ -0,0 +1,86 @@
+             README for Fink package: tetex-beta-20040208-1
+
+The Fink package for teTeX 2.0 follows Thomas Esser's distribution quite
+closely.  This file provides some advice for users, and documents the
+differences between the Fink package and the upstream version.
+
+MISSING ITEMS
+
+The following things are removed from the teTeX package because they are
+supplied elsewhere in the Fink system: texinfo (supplied in Fink's texinfo
+package), and readlink (supplied in Fink's debianutils package).
+
+TETEX PACKAGES IN FINK
+
+As of teTeX 2.0, a Fink user will find the following teTeX packages
+available to be installed on the system: 
+  tetex-texmf, tetex-base, tetex-dev, tetex-shlibs, tetex-nox, tetex.
+
+The "tetex" package replaces the former "bundle-tetex" package, and
+installs everything that is needed.
+
+The "tetex-nox" package is an alternative to this which does not depend on
+the X11 libraries.  Users of this package should be sure to also install
+the "texinfo" and "debianutils" packages, which are not installed by
+default.
+
+The "tetex-texmf", "tetex-shlibs", and "tetex-base" packages are various
+components of the teTeX system, and will be installed when needed.  The
+"tetex-dev" package is used when certain other TeX related packages are
+being compiled, but is not needed when programs are run.
+
+CONFIGURING TETEX
+
+Users can adjust the configuration of their copy of teTeX by running the
+program "texconfig".  The configuration files are stored in the directory
+@PREFIX@/var/lib/texmf .
+
+INSTALLING MORE TEX STYLE FILES, ETC.
+
+Users can install any private TeX style files, or TeX macro packages which
+they haven't installed via Fink or don't wish to share with other users of
+their computer, in the directory
+  ~/Library/texmf
+(i.e., a folder named texmf within the Library folder in the user's Home).
+Each user would need to create the corresponding directory if they wished
+to put things into it.
+
+If the user's ~/Library/texmf directory contains many files, it can be
+indexed with "mktexlsr ~/Library/texmf", which will speed up searching.
+Once this has been done, it must be done again any time something new is
+added to the directory (or else the file ~/Library/texmf/ls-R can be
+removed instead, to turn off the indexing of this directory).
+
+USING THE "ORIGINAL" XDVI
+
+Although teTeX comes with a version of xdvi installed (often called xdvik),
+some users may prefer to use the "original" xdvi program maintained by Paul
+Vojta.  This can be accomplished by installing Fink's xdvi package along
+side of tetex-base.  When this has been done, Vojta's package is called by
+the command "xdvi", whereas the teTeX-supplied package is called by the
+command "xdvik".
+
+USING AN UPDATED TEXI2HTML
+
+The texi2html program supplied with teTeX is not the most up-to-date one.
+If you want a more up-to-date one, install Fink's texi2html package.
+
+TECHNICAL NOTE: SHARED LIBRARIES
+
+Fink's tetex packages are built using shared libraries, by patching teTeX
+to make this possible.  We will be working with the teTeX maintainers to
+get these patches moved "upstream" for some future teTeX release.
+
+TECHNICAL NOTE: PATCHES
+
+In addition to the patch for shared libraries (a patch to klibtool), the
+Makefile is patched so as to prevent execution of the final configuration
+step (which in the Fink system is handled in a PostInstall script), and
+texmf.in is patched to implement the choices of directories and provide
+support for Paul Vojta's xdvi.  Finally, this README.fink file is supplied
+via a patch. 
+
+
+                                            Dave Morrison
+                                            <dmrrsn@users.sourceforge.net>
+                                            21 February 2003
diff -ruN tetex-src-2.99.3.20041109-beta/texk/kpathsea/texmf.in tetex-src-2.99.3.20041109-beta-patched/texk/kpathsea/texmf.in
--- tetex-src-2.99.3.20041109-beta/texk/kpathsea/texmf.in	Sun Oct 17 16:14:15 2004
+++ tetex-src-2.99.3.20041109-beta-patched/texk/kpathsea/texmf.in	Mon Nov  8 21:38:50 2004
@@ -67,10 +67,10 @@
 
 % If defined, teTeX's texconfig stores variable runtime data here
 % (instead of the TEXMFMAIN tree).
-%   VARTEXMF = @texmf@-var
+VARTEXMF = @PREFIX@/var/lib/texmf
 
 % User texmf trees can be catered for like this...
-HOMETEXMF = $HOME/texmf
+HOMETEXMF = $HOME/Library/texmf
 
 % Now, list all the texmf trees. If you have multiple trees you can
 % use shell brace notation, like this:
@@ -79,7 +79,7 @@
 %   - list $VARTEXMF in the TEXMF definition;
 %   - make sure that $VARTEXMF precedes $TEXMFMAIN in the TEXMF definition.
 % TEXMF = $TEXMFMAIN
-TEXMF = {$HOMETEXMF,!!$TEXMFLOCAL,!!$TEXMFMAIN,!!$TEXMFDIST}
+TEXMF = {$HOMETEXMF,!!$TEXMFLOCAL,!!$VARTEXMF,!!$TEXMFMAIN,!!$TEXMFDIST}
 
 % The system trees.  These are the trees that are shared by all the users.
 SYSTEXMF = $TEXMFLOCAL;$TEXMFMAIN;$TEXMFDIST
@@ -87,14 +87,14 @@
 % Where generated fonts may be written.  This tree is used when the sources
 % were found in a system tree and either that tree wasn't writable, or the
 % varfonts feature was enabled in MT_FEATURES in mktex.cnf.
-VARTEXFONTS = @vartexfonts@
+VARTEXFONTS = $VARTEXMF/fonts
 
 % Where to look for ls-R files.  There need not be an ls-R in the
 % directories in this path, but if there is one, Kpathsea will use it.
 %
 % Remove $VARTEXFONTS from TEXMFDBS if the VARTEXFONTS directory is below
 % one of the TEXMF directories (avoids overlapping ls-R files).
-TEXMFDBS = $TEXMF;$VARTEXFONTS
+TEXMFDBS = $TEXMF
 
 % It may be convenient to define TEXMF like this:
 %   TEXMF = {$HOMETEXMF,!!$TEXMFLOCAL,!!$TEXMFMAIN,$HOME}
@@ -115,6 +115,11 @@
 %   VARTEXFONTS = $VARTEXMF/fonts
 % and do not mention it in TEXMFDBS (but _do_ mention VARTEXMF).
 
+% Additional lines for use by xdvi:
+PKFONTS.XDvi    = .:$TEXMF/%s:$VARTEXFONTS/pk/{%m,modeless}//
+VFFONTS.XDvi    = .:$TEXMF/%s
+PSHEADERS.XDvi  = .:$TEXMF/%q{dvips,fonts/type1}//
+TEXPICTS.XDvi   = .:$TEXMF/%q{dvips,tex}//
 
 %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
 % Usually you will not need to edit any of the other variables in part 1. %
