Index: perlmod/Fink/Engine.pm
===================================================================
RCS file: /cvsroot/fink/fink/perlmod/Fink/Engine.pm,v
retrieving revision 1.67
diff -u -b -r1.67 Engine.pm
--- perlmod/Fink/Engine.pm	25 Sep 2002 19:01:42 -0000	1.67
+++ perlmod/Fink/Engine.pm	25 Sep 2002 19:53:15 -0000
@@ -1078,8 +1078,10 @@
 	foreach my $splitoff (@{$parent->{_splitoffs}}) {
 	  $already_rebuilt{$splitoff->get_name()} = 1;
 	  if ($splitoff->is_installed() or exists $deps{$splitoff->get_name()}) {
-	    push(@batch_install, $splitoff) unless ($already_activated{$splitoff->get_name()} or $splitoff->is_installed());
+	    unless ($already_activated{$splitoff->get_name()} or $splitoff->is_installed()) {
+	      push(@batch_install, $splitoff);
 	    $already_activated{$splitoff->get_name()} = 1;
+	    }
 	  }
 	}
       }
