Index: perlmod/Fink/Engine.pm
===================================================================
RCS file: /cvsroot/fink/fink/perlmod/Fink/Engine.pm,v
retrieving revision 1.66
diff -u -b -r1.66 Engine.pm
--- perlmod/Fink/Engine.pm	19 Sep 2002 20:03:02 -0000	1.66
+++ perlmod/Fink/Engine.pm	24 Sep 2002 19:01:30 -0000
@@ -1078,7 +1078,7 @@
 	foreach my $splitoff (@{$parent->{_splitoffs}}) {
 	  $already_rebuilt{$splitoff->get_name()} = 1;
 	  if ($splitoff->is_installed() or exists $deps{$splitoff->get_name()}) {
-	    push(@batch_install, $splitoff) unless ($already_activated{$splitoff->get_name()});
+	    push(@batch_install, $splitoff) unless ($already_activated{$splitoff->get_name()} or $splitoff->is_installed());
 	    $already_activated{$splitoff->get_name()} = 1;
 	  }
 	}
Index: perlmod/Fink/PkgVersion.pm
===================================================================
RCS file: /cvsroot/fink/fink/perlmod/Fink/PkgVersion.pm,v
retrieving revision 1.81
diff -u -b -r1.81 PkgVersion.pm
--- perlmod/Fink/PkgVersion.pm	19 Sep 2002 20:03:03 -0000	1.81
+++ perlmod/Fink/PkgVersion.pm	24 Sep 2002 19:01:30 -0000
@@ -700,13 +700,14 @@
       }
 
       $package = Fink::Package->package_by_name($depname);
-      push(@{$package->{_versionspecs}}, $versionspec) unless ($versionspec =~ /^\s*$/);
 
       if (not defined $package) {
 	print "WARNING: While resolving dependency \"$depspec\" for package \"".$self->get_fullname()."\", package \"$depname\" was not found.\n";
 	next;
       }
 
+      push(@{$package->{_versionspecs}}, $versionspec) unless ($versionspec =~ /^\s*$/);
+
       if ($versionspec) {
 	push @$altlist, $package->get_matching_versions($versionspec);
       } else {
@@ -1636,18 +1637,19 @@
 	  die "can't find package ".$package->get_debname()."\n";
 	}
 
-    push(@installable, $deb);
+    push(@installable, $package);
   }
 
   if (@installable == 0) {
     die "no installable .deb files found!\n";
   }
 
-  if (&execute("dpkg -i @installable")) {
+  my @deb_installable = map { $_->find_debfile() } @installable;
+  if (&execute("dpkg -i @deb_installable")) {
     if (@installable == 1) {
       die "can't install package ".$installable[0]->get_fullname()."\n";
     } else {
-      die "can't batch-install packages: @installable\n";
+      die "can't batch-install packages: @deb_installable\n";
     }
   }
 
