Index: perlmod/Fink/PkgVersion.pm
===================================================================
RCS file: /cvsroot/fink/fink/perlmod/Fink/PkgVersion.pm,v
retrieving revision 1.98
diff -u -u -r1.98 PkgVersion.pm
--- perlmod/Fink/PkgVersion.pm	11 Dec 2002 22:45:43 -0000	1.98
+++ perlmod/Fink/PkgVersion.pm	12 Dec 2002 14:55:00 -0000
@@ -1724,7 +1724,8 @@
   my ($varname, $s, $expand);
   my %defaults = ( "CPPFLAGS" => "-isystem \%p/include",
 		   "LIBRARY_PATH" => "\%p/lib",
-		   "DYLD_LIBRARY_PATH" => "\%p/lib" );
+		   "DYLD_LIBRARY_PATH" => "\%p/lib",
+		   "MACOSX_DEPLOYMENT_TARGET" => '10.1' );
   my $bsbase = get_bsbase();
 
   # clean the environment
@@ -1753,7 +1754,8 @@
 		    "LD", "LDFLAGS", "LIBS",
 		    "MAKE", "MFLAGS",
 		    "LIBRARY_PATH",
-		    "DYLD_LIBRARY_PATH") {
+		    "DYLD_LIBRARY_PATH",
+		    "MACOSX_DEPLOYMENT_TARGET") {
     if ($self->has_param("Set$varname")) {
       $s = $self->param("Set$varname");
       if (exists $defaults{$varname} and
