Index: perlmod/Fink/Engine.pm
===================================================================
RCS file: /cvsroot/fink/fink/perlmod/Fink/Engine.pm,v
retrieving revision 1.65
diff -u -b -u -b -r1.65 Engine.pm
--- perlmod/Fink/Engine.pm	21 Aug 2002 15:59:15 -0000	1.65
+++ perlmod/Fink/Engine.pm	16 Sep 2002 20:47:13 -0000
@@ -923,16 +923,46 @@
 
       # the dice are rolled...
 
-      if (exists $deps{$dname}) {
-	die "Internal error: node for $dname already exists\n";
-      }
-
       $pnode = Fink::Package->package_by_name($dname);
       @vlist = ();
       foreach $dp (@$dep) {
 	if ($dp->get_name() eq $dname) {
 	  push @vlist, $dp->get_fullversion();
 	}
+      }
+
+      if (exists $deps{$dname}) {
+        # node exists, we need to generate the version list
+        # based on multiple packages
+        @vlist = ();
+
+        # first, get the current list of acceptable packages
+        # this will run get_matching_versions over every version spec
+        # for this package
+        my $package = Fink::Package->package_by_name($dname);
+        my @existing_matches;
+        for my $spec (@{$package->{_versionspecs}}) {
+          push(@existing_matches, $package->get_matching_versions($spec, @existing_matches));
+          if (@existing_matches == 0) {
+            print "unable to resolve version conflict on multiple dependencies\n";
+            for my $spec (@{$package->{_versionspecs}}) {
+              print "  $dname $spec\n";
+            }
+            exit 1;
+          }
+        }
+
+        for (@existing_matches) {
+          push(@vlist, $_->get_fullversion());
+        }
+
+        unless (@vlist > 0) {
+          print "unable to resolve version conflict on multiple dependencies\n";
+          for my $spec (@{$package->{_versionspecs}}) {
+            print "  $dname $spec\n";
+          }
+          exit 1;
+        }
       }
 
       # add node to graph
Index: perlmod/Fink/Package.pm
===================================================================
RCS file: /cvsroot/fink/fink/perlmod/Fink/Package.pm,v
retrieving revision 1.22
diff -u -b -u -b -r1.22 Package.pm
--- perlmod/Fink/Package.pm	4 Aug 2002 19:42:24 -0000	1.22
+++ perlmod/Fink/Package.pm	16 Sep 2002 20:47:13 -0000
@@ -148,6 +148,7 @@
 sub get_matching_versions {
   my $self = shift;
   my $spec = shift;
+  my @include_list = @_;
   my (@list, $version, $vo, $relation, $reqversion);
 
   if ($spec =~ /^\s*(<<|<=|=|>=|>>)\s*([0-9a-zA-Z.\+-]+)\s*$/) {
@@ -181,7 +182,20 @@
     }
   }
 
+  if (@include_list > 0) {
+    my @match_list;
+    # if we've been given a list to choose from, return the
+    # intersection of the two
+    for my $vo (@list) {
+      my $version = $vo->get_version();
+      if (grep(/^${version}$/, @include_list)) {
+        push(@match_list, $vo);
+      }
+    }
+    return @match_list;
+  } else {
   return @list;
+  }
 }
 
 sub get_all_providers {
Index: perlmod/Fink/PkgVersion.pm
===================================================================
RCS file: /cvsroot/fink/fink/perlmod/Fink/PkgVersion.pm,v
retrieving revision 1.80
diff -u -b -u -b -r1.80 PkgVersion.pm
--- perlmod/Fink/PkgVersion.pm	21 Aug 2002 16:10:07 -0000	1.80
+++ perlmod/Fink/PkgVersion.pm	16 Sep 2002 20:47:13 -0000
@@ -700,6 +700,8 @@
       }
 
       $package = Fink::Package->package_by_name($depname);
+      push(@{$package->{_versionspecs}}, $versionspec) unless ($versionspec =~ /^\s*$/);
+
       if (not defined $package) {
 	print "WARNING: While resolving dependency \"$depspec\" for package \"".$self->get_fullname()."\", package \"$depname\" was not found.\n";
 	next;
