diff -Naur debian.old/defoma debian.new/defoma
--- debian.old/defoma	Wed Apr 17 23:50:53 2002
+++ debian.new/defoma	Mon Nov 11 02:40:35 2002
@@ -15,7 +15,7 @@
 
 my ($Id, $IdCmap, $IdSub);
 
-my $configfile = "$DEFOMA_TEST_DIR/etc/defoma/config/x-ttcidfont-conf.conf";
+my $configfile = "$DEFOMA_TEST_DIR@PREFIX@/etc/defoma/config/x-ttcidfont-conf.conf";
 my $PkgDir = "$ROOTDIR/x-ttcidfont-conf.d";
 my $FontRootDir = "$PkgDir/dirs";
 my $Method;
diff -Naur debian.old/x-ttcidfont-conf.conf debian.new/x-ttcidfont-conf.conf
--- debian.old/x-ttcidfont-conf.conf	Thu Jan  1 09:00:00 1970
+++ debian.new/x-ttcidfont-conf.conf	Mon Nov 11 02:50:16 2002
@@ -0,0 +1,9 @@
+# IF YOU EDIT THIS FILE MANUALLY, RUN:
+#   defoma-app update x-ttcidfont-conf truetype
+# TO REFLECT THE CHANGE.
+# The following variable decides which method is used for
+# handling truetype fonts.
+X_TRUETYPE_METHOD=freetype
+# The following variable decides if X calculate font
+# metrics in the very lazy way when the method is xtt.
+XTT_VL=n
