diff -uNr mono-debugger-0.11/arch/Makefile.in mono-debugger-0.11-new/arch/Makefile.in
--- mono-debugger-0.11/arch/Makefile.in	2005-12-15 19:37:02.000000000 -0500
+++ mono-debugger-0.11-new/arch/Makefile.in	2005-12-19 16:15:19.000000000 -0500
@@ -51,8 +51,7 @@
 CONFIG_HEADER = $(top_builddir)/config.h
 CONFIG_CLEAN_FILES =
 LTLIBRARIES = $(noinst_LTLIBRARIES)
-libmonodebuggerbfdglue_la_DEPENDENCIES = opcodes/libopcodes.la \
-	bfd/libbfd.la
+libmonodebuggerbfdglue_la_DEPENDENCIES = 
 am_libmonodebuggerbfdglue_la_OBJECTS = bfdglue.lo
 libmonodebuggerbfdglue_la_OBJECTS =  \
 	$(am_libmonodebuggerbfdglue_la_OBJECTS)
@@ -221,16 +220,16 @@
 target_cpu = @target_cpu@
 target_os = @target_os@
 target_vendor = @target_vendor@
-SUBDIRS = libiberty bfd opcodes
+SUBDIRS = 
 noinst_LTLIBRARIES = libmonodebuggerbfdglue.la
 libmonodebuggerbfdglue_la_SOURCES = \
 	bfdglue.c		\
 	bfdglue.h
 
 testbfd_LDADD = libmonodebuggerbfdglue.la
-libmonodebuggerbfdglue_la_LIBADD = opcodes/libopcodes.la bfd/libbfd.la @BASE_DEPENDENCIES_LIBS@
-libmonodebuggerbfdglue_la_LDFLAGS = -no-undefined -module -export-dynamic -shared
-INCLUDES = -I$(srcdir)/bfd -I$(srcdir)/bfd/include -I$(srcdir)/opcodes @BASE_DEPENDENCIES_CFLAGS@
+libmonodebuggerbfdglue_la_LIBADD = -L$(prefix)/lib/libopcodes/lib -L$(prefix)/lib/libbfd/lib -L$(prefix)/lib/libiberty -lopcodes -lbfd -liberty @BASE_DEPENDENCIES_LIBS@
+libmonodebuggerbfdglue_la_LDFLAGS = -no-undefined -module -export-dynamic
+INCLUDES = -I$(prefix)/lib/libbfd/include -I$(prefix)/lib/libopcodes/include -I$(preix)/lib/libiberty/include @BASE_DEPENDENCIES_CFLAGS@
 CLEANFILES = lib*.a lib*.dll
 EXTRA_DIST = \
 	Architecture.cs Architecture_I386.cs Architecture_X86_64.cs BfdContainer.cs Bfd.cs \
