diff -uNbr monopd-0.6.0/Makefile.in monopd-0.6.0-patched/Makefile.in
--- monopd-0.6.0/Makefile.in	Thu Oct 24 01:47:15 2002
+++ monopd-0.6.0-patched/Makefile.in	Sun Oct 27 19:54:55 2002
@@ -72,7 +72,7 @@
 
 SUBDIRS = games conf doc
 
-CPPFLAGS = -DMONOPD_DATADIR=\"$(pkgdatadir)\" -DMONOPD_CONFIGDIR=\"$(sysconfdir)\"
+CPPFLAGS += -DMONOPD_DATADIR=\"$(pkgdatadir)\" -DMONOPD_CONFIGDIR=\"$(sysconfdir)\"
 
 bin_PROGRAMS = monopd
 
@@ -107,7 +107,7 @@
 monopd_LDFLAGS =
 
 DEFS = @DEFS@
-DEFAULT_INCLUDES =  -I. -I$(srcdir) -I.
+DEFAULT_INCLUDES =  -I. -I$(srcdir) -I. -I$(prefix)/include
 LDFLAGS = @LDFLAGS@
 LIBS = @LIBS@
 depcomp = $(SHELL) $(top_srcdir)/depcomp
diff -uNbr monopd-0.6.0/server.cc monopd-0.6.0-patched/server.cc
--- monopd-0.6.0/server.cc	Wed Oct 23 19:19:46 2002
+++ monopd-0.6.0-patched/server.cc	Sun Oct 27 21:08:46 2002
@@ -15,13 +15,14 @@
 // the Free Software Foundation, Inc., 59 Temple Place - Suite 330,
 // Boston, MA 02111-1307, USA.
 
+#include <sys/types.h>
 #include <dirent.h>
 #include <netdb.h>
 #include <stdio.h>
 #include <sys/time.h>
 #include <sys/socket.h>
-#include <sys/types.h>
 #include <unistd.h>
+#include <netinet/in.h>
 
 #include <string>
 
diff -uNbr monopd-0.6.0/server.h monopd-0.6.0-patched/server.h
--- monopd-0.6.0/server.h	Mon Oct 21 00:35:02 2002
+++ monopd-0.6.0-patched/server.h	Sun Oct 27 21:03:14 2002
@@ -25,6 +25,8 @@
 class Player;
 class Socket;
 
+#include <vector>
+
 class MonopdServer
 {
 public:
