diff -Naur libgnomeprint-2.7.1.old/configure libgnomeprint-2.7.1.new/configure
--- libgnomeprint-2.7.1.old/configure	2004-07-31 10:42:21.000000000 +1000
+++ libgnomeprint-2.7.1.new/configure	2004-08-29 22:19:24.000000000 +1000
@@ -23476,7 +23476,7 @@
 GNOME_PRINT_DATA_DIR=$GNOME_PRINT_DATA_DIR
 
 
-LIBGNOMEPRINT_CFLAGS="$GP_CFLAGS $FREETYPE_CFLAGS"
+LIBGNOMEPRINT_CFLAGS="$FREETYPE_CFLAGS $GP_CFLAGS"
 LIBGNOMEPRINT_LIBS="$GP_LIBS $FREETYPE_LIBS $Z_LIBS"
 
 
diff -Naur libgnomeprint-2.7.1.old/libgnomeprint/gnome-fontmap.c libgnomeprint-2.7.1.new/libgnomeprint/gnome-fontmap.c
--- libgnomeprint-2.7.1.old/libgnomeprint/gnome-fontmap.c	2004-06-22 01:13:56.000000000 +1000
+++ libgnomeprint-2.7.1.new/libgnomeprint/gnome-fontmap.c	2004-08-29 22:18:36.000000000 +1000
@@ -219,6 +219,8 @@
 		font_type = GP_FONT_ENTRY_TYPE1;
 	else if (strcasecmp (c, ".ttf") == 0)
 		font_type = GP_FONT_ENTRY_TRUETYPE;
+	else if (strcasecmp (c, ".otf") == 0)
+		font_type = GP_FONT_ENTRY_TRUETYPE;
 	else if (g_ascii_strcasecmp (c, ".ttc") == 0)
 		font_type = GP_FONT_ENTRY_TRUETYPE;
 	else if (strcasecmp (c, ".font") == 0) /* See #102400 */
 		font_type = GP_FONT_ENTRY_TRUETYPE;
 	else {
