diff -Naur pango-1.5.2.old/Makefile.in pango-1.5.2.new/Makefile.in
--- pango-1.5.2.old/Makefile.in	2004-08-03 07:12:01.000000000 +1000
+++ pango-1.5.2.new/Makefile.in	2004-08-29 23:18:46.000000000 +1000
@@ -234,7 +234,7 @@
 
 AUTOMAKE_OPTIONS = 1.7
 
-SUBDIRS = pango modules examples docs tools tests
+SUBDIRS = pango modules examples docs tools
 
 EXTRA_DIST = \
 	pango.pc.in	\
diff -Naur pango-1.5.2.old/ltmain.sh pango-1.5.2.new/ltmain.sh
--- pango-1.5.2.old/ltmain.sh	2004-04-12 12:01:57.000000000 +1000
+++ pango-1.5.2.new/ltmain.sh	2004-08-29 23:19:26.000000000 +1000
@@ -5266,10 +5266,10 @@
 
 # Directory that this library needs to be installed in:
 libdir='$install_libdir'"
-	  if test "$installed" = no && test "$need_relink" = yes; then
-	    $echo >> $output "\
-relink_command=\"$relink_command\""
-	  fi
+#	  if test "$installed" = no && test "$need_relink" = yes; then
+#	    $echo >> $output "\
+#relink_command=\"$relink_command\""
+#	  fi
 	done
       fi
 
diff -Naur pango-1.5.2.old/pango/Makefile.in pango-1.5.2.new/pango/Makefile.in
--- pango-1.5.2.old/pango/Makefile.in	2004-08-03 07:12:05.000000000 +1000
+++ pango-1.5.2.new/pango/Makefile.in	2004-08-29 23:20:20.000000000 +1000
@@ -383,7 +383,7 @@
 
 
 libpangoxft_1_0_la_LDFLAGS = -version-info $(LT_VERSION_INFO) $(no_undefined)
-libpangoxft_1_0_la_LIBADD = libpangoft2-$(PANGO_API_VERSION).la libpango-$(PANGO_API_VERSION).la $(XFT_LIBS) $(GLIB_LIBS)
+libpangoxft_1_0_la_LIBADD = libpangoft2-$(PANGO_API_VERSION).la libpango-$(PANGO_API_VERSION).la $(XFT_LIBS) $(GLIB_LIBS) $(X_LIBS)
 libpangoxft_1_0_la_DEPENDENCIES = libpangoft2-$(PANGO_API_VERSION).la libpango-$(PANGO_API_VERSION).la
 
 libpangoxft_1_0_la_SOURCES = \
diff -Naur pango-1.5.2.old/update-pango-modules pango-1.5.2.new/update-pango-modules
--- pango-1.5.2.old/update-pango-modules	1970-01-01 10:00:00.000000000 +1000
+++ pango-1.5.2.new/update-pango-modules	2004-08-29 23:21:04.000000000 +1000
@@ -0,0 +1,20 @@
+#!/bin/sh -e
+#
+# update-pango-modules
+#
+#   Package which install pango module(s) should call update-pango-modules in
+#   it's PostInstScript/PostRmScript to update pango.module file.
+#
+
+CONFDIR=@PREFIX@/etc/pango
+CONFFILE=$CONFDIR/pango.modules
+
+mkdir -p $CONFDIR
+
+TMPFILE=`tempfile`
+pango-querymodules >$TMPFILE
+
+mv -f $TMPFILE $CONFFILE
+chmod 0644 $CONFFILE
+
+exit 0
\ No newline at end of file
