--- mono-1.1.13/configure	2006-01-06 14:26:30.000000000 -0500
+++ mono-1.1.13-new/configure	2006-01-11 07:53:18.000000000 -0500
@@ -35519,7 +35519,7 @@
 	INTL="libintl.dylib"
 	SQLITE="libsqlite.0.dylib"
 	SQLITE3="libsqlite3.0.dylib"
-	X11="libX11.dylib"
+	X11="libX11.6.dylib"
 	;;
      *-*-*netbsd*)
 	LIBC="libc.so.12"
--- mono-1.1.13/mono/metadata/Makefile.in	2006-01-06 14:26:24.000000000 -0500
+++ mono-1.1.13-new/mono/metadata/Makefile.in	2006-01-11 07:53:19.000000000 -0500
@@ -68,7 +68,7 @@
 am_libmonoruntime_static_la_OBJECTS = $(am__objects_1)
 libmonoruntime_static_la_OBJECTS =  \
 	$(am_libmonoruntime_static_la_OBJECTS)
-libmonoruntime_la_LIBADD =
+libmonoruntime_la_LIBADD = $(ICU_LIBS)
 am_libmonoruntime_la_OBJECTS = reflection.lo object.lo icall.lo \
 	decimal.lo boehm-gc.lo null-gc.lo gc.lo marshal.lo monitor.lo \
 	threads.lo threadpool.lo file-io.lo socket-io.lo exception.lo \
--- mono-1.1.13.6/data/config.in	2006-03-24 13:00:15.000000000 -0500
+++ mono-1.1.13.6-new/data/config.in	2006-03-28 22:47:43.000000000 -0500
@@ -12,5 +12,5 @@
 	<dllmap dll="sqlite" target="@SQLITE@"/>
 	<dllmap dll="sqlite3" target="@SQLITE3@"/>
 	<dllmap dll="libX11" target="@X11@"/>
-	<dllmap dll="libcairo-2.dll" target="libcairo.so.2"/>
+	<dllmap dll="libcairo-2.dll" target="libcairo@libsuffix@"/>
 </configuration>
