diff -Naur control-center-2.4.0-old/configure control-center-2.4.0/configure
--- control-center-2.4.0-old/configure	Mon Sep  8 10:57:09 2003
+++ control-center-2.4.0/configure	Thu Oct 30 10:41:32 2003
@@ -21608,7 +21608,7 @@
 
 fi
 
-x_libs="$X_PRE_LIBS $X_LIBS -lX11 $X_EXTRA_LIBS"
+x_libs="$X_PRE_LIBS $X_LIBS -lX11 -lXext $X_EXTRA_LIBS"
 
 savecppflags=$CPPFLAGS
 CPPFLAGS="$CPPFLAGS $X_CFLAGS"
diff -Naur control-center-2.4.0-old/typing-break/Makefile.in control-center-2.4.0/typing-break/Makefile.in
--- control-center-2.4.0-old/typing-break/Makefile.in	Mon Sep  8 10:58:41 2003
+++ control-center-2.4.0/typing-break/Makefile.in	Thu Oct 30 10:42:12 2003
@@ -204,7 +204,7 @@
 	egg-spawn.h
 
 
-gnome_typing_monitor_LDADD = @GNOME_LIBS@ -L/usr/X11R6/lib -lXss
+gnome_typing_monitor_LDADD = @GNOME_LIBS@ -L/usr/X11R6/lib -lXss -lXext
 gnome_typing_monitor_LDFLAGS = -export-dynamic
 
 imagedir = $(GNOMECC_PIXMAPS_DIR)
