diff -Naur gnome-applets-2.4.0-old/Makefile.in gnome-applets-2.4.0/Makefile.in
--- gnome-applets-2.4.0-old/Makefile.in	Mon Sep  8 20:01:45 2003
+++ gnome-applets-2.4.0/Makefile.in	Mon Oct 27 11:14:40 2003
@@ -163,13 +163,13 @@
 
 @APPLET_ACCESSX_TRUE@accessx_status_SUBDIR = accessx-status
 
-always_built_SUBDIRS =  	drivemount		charpick		geyes			mailcheck		mini-commander		gkb-new			wireless		stickynotes
+always_built_SUBDIRS =  	drivemount		charpick		geyes			mailcheck		mini-commander		stickynotes
 
 
 SUBDIRS =  	po				screen-exec			$(always_built_SUBDIRS)		$(mixer_SUBDIRS)		$(modemlights_SUBDIRS)		$(cdplayer_SUBDIRS)		$(gtop_SUBDIRS)			$(vfs_SUBDIRS)			$(battstat_SUBDIR)		$(accessx_status_SUBDIR)
 
 
-DIST_SUBDIRS =  	po			screen-exec		mailcheck		mixer			modemlights		cdplayer		drivemount		charpick		geyes			mini-commander		gkb-new			battstat		multiload		gtik			gweather 		mini-commander		man			wireless		accessx-status  	stickynotes
+DIST_SUBDIRS =  	po			screen-exec		mailcheck		mixer			modemlights		cdplayer		drivemount		charpick		geyes			mini-commander		battstat		multiload		gtik			gweather 		mini-commander		man			accessx-status  	stickynotes
 
 
 EXTRA_DIST =  	COPYING-DOCS			gnome-applets.spec.in		gnome-applets.spec		applet-docs.make		intltool-update.in		intltool-extract.in		intltool-merge.in		xmldocs.make			omf.make
diff -Naur gnome-applets-2.4.0-old/charpick/charpick.c gnome-applets-2.4.0/charpick/charpick.c
--- gnome-applets-2.4.0-old/charpick/charpick.c	Wed Sep  3 19:51:46 2003
+++ gnome-applets-2.4.0/charpick/charpick.c	Mon Oct 27 11:09:46 2003
@@ -3,9 +3,11 @@
  */
 
 #include <config.h>
+/*
 #ifndef sun
   #include <tgmath.h>
 #endif
+*/
 #include <panel-applet.h>
 #include <egg-screen-help.h>
 #include "charpick.h"
diff -Naur gnome-applets-2.4.0-old/configure gnome-applets-2.4.0/configure
--- gnome-applets-2.4.0-old/configure	Mon Sep  8 20:00:28 2003
+++ gnome-applets-2.4.0/configure	Mon Oct 27 11:09:46 2003
@@ -11932,6 +11932,9 @@
 	echo "warning: ${host} is not supported by battstat_applet, not building" >&2
 	build_battstat_applet=no
 	;;
+  *darwin*)
+	build_battstat_applet=no
+	;;
 esac
 
 
diff -Naur gnome-applets-2.4.0-old/gkb-new/preset.c gnome-applets-2.4.0/gkb-new/preset.c
--- gnome-applets-2.4.0-old/gkb-new/preset.c	Sun Jun 29 12:25:11 2003
+++ gnome-applets-2.4.0/gkb-new/preset.c	Mon Oct 27 11:09:46 2003
@@ -29,6 +29,7 @@
  * USA
  */
 
+#include <sys/types.h>
 #include <dirent.h>
 #include "gkb.h"
 
diff -Naur gnome-applets-2.4.0-old/multiload/Makefile.in gnome-applets-2.4.0/multiload/Makefile.in
--- gnome-applets-2.4.0-old/multiload/Makefile.in	Mon Sep  8 20:07:07 2003
+++ gnome-applets-2.4.0/multiload/Makefile.in	Mon Oct 27 11:09:46 2003
@@ -148,7 +148,7 @@
 multiload_applet_2_SOURCES =  			global.h				linux-proc.h				load-graph.h				cpuload.c				linux-proc.c				load-graph.c				loadavg.c				main.c				memload.c				netload.c				properties.c				swapload.c
 
 
-multiload_applet_2_LDADD =  	$(top_builddir)/screen-exec/libscreen-exec.la					$(GNOME_APPLETS_LIBS) 					$(GTOP_APPLETS_LIBS)
+multiload_applet_2_LDADD =  	$(top_builddir)/screen-exec/libscreen-exec.la					$(GNOME_APPLETS_LIBS) 					$(GTOP_APPLETS_LIBS)	$(X_LIBS)
 
 
 pixmaps = gnome-cpu.png
