diff -uNbr openldap-2.1.3/build/ltmain.sh openldap-2.1.3-new/build/ltmain.sh
--- openldap-2.1.3/build/ltmain.sh	Mon Feb 11 00:45:59 2002
+++ openldap-2.1.3-new/build/ltmain.sh	Thu Aug  8 23:53:05 2002
@@ -1744,7 +1744,7 @@
 
 	  if test "$installed" = no; then
 	    notinst_deplibs="$notinst_deplibs $lib"
-	    need_relink=yes
+	    need_relink=no
 	  fi
 
 	  if test -n "$old_archive_from_expsyms_cmds"; then
diff -uNbr openldap-2.1.3/configure openldap-2.1.3-new/configure
--- openldap-2.1.3/configure	Tue Jul  9 21:27:33 2002
+++ openldap-2.1.3-new/configure	Fri Aug  9 10:28:21 2002
@@ -797,7 +797,7 @@
 
 ac_ext=c
 # CFLAGS is not in ac_cpp because -g, -O, etc. are not valid cpp options.
-ac_cpp='$CPP $CPPFLAGS'
+ac_cpp='$CPP $CPPFLAGS $CFLAGS'
 ac_compile='${CC-cc} -c $CFLAGS $CPPFLAGS conftest.$ac_ext 1>&5'
 ac_link='${CC-cc} -o conftest${ac_exeext} $CFLAGS $CPPFLAGS $LDFLAGS conftest.$ac_ext $LIBS 1>&5'
 cross_compiling=$ac_cv_prog_cc_cross
@@ -10421,7 +10421,7 @@
   echo $ac_n "(cached) $ac_c" 1>&6
 else
   ac_save_LIBS="$LIBS"
-LIBS="-lkrb5 -l$krb5crypto -lcom_err $LIBS"
+LIBS="-prebind -framework Kerberos $LIBS"
 cat > conftest.$ac_ext <<EOF
 #line 10427 "configure"
 #include "confdefs.h"
@@ -10446,7 +10446,7 @@
 if eval "test \"`echo '$ac_cv_lib_'$ac_lib_var`\" = yes"; then
   echo "$ac_t""yes" 1>&6
   have_krb5=yes
-				KRB5_LIBS="-lkrb5 -l$krb5crypto -lcom_err"
+				KRB5_LIBS="-prebind -framework Kerberos"
 else
   echo "$ac_t""no" 1>&6
 have_krb5=no
