Index: src/Makefile.global.in
===================================================================
RCS file: /projects/cvsroot/pgsql/src/Makefile.global.in,v
retrieving revision 1.141
diff -u -b -u -b -r1.141 Makefile.global.in
--- src/Makefile.global.in	2001/12/20 21:23:05	1.141
+++ src/Makefile.global.in	2002/07/06 14:37:13
@@ -94,6 +94,7 @@
 override pkgincludedir := $(pkgincludedir)/postgresql
 endif
 endif
+includedir = $(pkgincludedir)
 includedir_server = $(pkgincludedir)/server
 includedir_internal = $(pkgincludedir)/internal
 
Index: src/Makefile.shlib
===================================================================
RCS file: /projects/cvsroot/pgsql/src/Makefile.shlib,v
retrieving revision 1.56
diff -u -b -u -b -r1.56 Makefile.shlib
--- src/Makefile.shlib	2001/11/11 19:20:53	1.56
+++ src/Makefile.shlib	2002/07/06 14:37:13
@@ -112,8 +112,9 @@
 endif
 
 ifeq ($(PORTNAME), darwin)
-  shlib			:= lib$(NAME)$(DLSUFFIX).$(SO_MAJOR_VERSION).$(SO_MINOR_VERSION)
+  shlib			:= lib$(NAME).$(SO_MAJOR_VERSION).$(SO_MINOR_VERSION)$(DLSUFFIX)
   LINK.shared		= $(COMPILER) $(DARWIN_NAMESPACE_SPEC) -bundle -undefined suppress
+  SHLIB_LINK		:= -L$(libpq_builddir) $(SHLIB_LINK)
 endif
 
 ifeq ($(PORTNAME), openbsd)
@@ -305,9 +306,9 @@
 $(shlib): $(OBJS)
 	$(LINK.shared) $(OBJS) $(SHLIB_LINK) -o $@
 # If we're using major and minor versions, then make a symlink to major-version-only.
-ifneq ($(shlib), lib$(NAME)$(DLSUFFIX).$(SO_MAJOR_VERSION))
-	rm -f lib$(NAME)$(DLSUFFIX).$(SO_MAJOR_VERSION)
-	$(LN_S) $(shlib) lib$(NAME)$(DLSUFFIX).$(SO_MAJOR_VERSION)
+ifneq ($(shlib), lib$(NAME).$(SO_MAJOR_VERSION)$(DLSUFFIX))
+	rm -f lib$(NAME).$(SO_MAJOR_VERSION)$(DLSUFFIX)
+	$(LN_S) $(shlib) lib$(NAME).$(SO_MAJOR_VERSION)$(DLSUFFIX)
 endif
 # Make sure we have a link to a name without any version numbers
 ifneq ($(shlib), lib$(NAME)$(DLSUFFIX))
@@ -363,10 +364,10 @@
 install-lib-shared: $(shlib)
 	$(INSTALL_SHLIB) $< $(DESTDIR)$(libdir)/$(shlib)
 ifneq ($(PORTNAME), win)
-ifneq ($(shlib), lib$(NAME)$(DLSUFFIX).$(SO_MAJOR_VERSION))
+ifneq ($(shlib), lib$(NAME).$(SO_MAJOR_VERSION)$(DLSUFFIX))
 	cd $(DESTDIR)$(libdir) && \
-	rm -f lib$(NAME)$(DLSUFFIX).$(SO_MAJOR_VERSION) && \
-	$(LN_S) $(shlib) lib$(NAME)$(DLSUFFIX).$(SO_MAJOR_VERSION)
+	rm -f lib$(NAME).$(SO_MAJOR_VERSION)$(DLSUFFIX) && \
+	$(LN_S) $(shlib) lib$(NAME).$(SO_MAJOR_VERSION)$(DLSUFFIX)
 endif
 ifneq ($(shlib), lib$(NAME)$(DLSUFFIX))
 	cd $(DESTDIR)$(libdir) && \
Index: src/backend/utils/misc/guc.c
===================================================================
RCS file: /projects/cvsroot/pgsql/src/backend/utils/misc/guc.c,v
retrieving revision 1.58
diff -u -b -u -b -r1.58 guc.c
--- src/backend/utils/misc/guc.c	2001/10/30 05:38:56	1.58
+++ src/backend/utils/misc/guc.c	2002/07/06 14:37:13
@@ -576,12 +576,12 @@
 
 	{
 		"unix_socket_group", PGC_POSTMASTER, &Unix_socket_group,
-		"", NULL, NULL
+		"pgsql", NULL, NULL
 	},
 
 	{
 		"unix_socket_directory", PGC_POSTMASTER, &UnixSocketDir,
-		"", NULL, NULL
+		"/tmp", NULL, NULL
 	},
 
 	{
Index: src/backend/utils/misc/postgresql.conf.sample
===================================================================
RCS file: /projects/cvsroot/pgsql/src/backend/utils/misc/postgresql.conf.sample,v
retrieving revision 1.28
diff -u -b -u -b -r1.28 postgresql.conf.sample
--- src/backend/utils/misc/postgresql.conf.sample	2002/01/04 05:50:25	1.28
+++ src/backend/utils/misc/postgresql.conf.sample	2002/07/06 14:37:14
@@ -23,17 +23,17 @@
 #
 #	Connection Parameters
 #
-#tcpip_socket = false
-#ssl = false
+tcpip_socket = true
+ssl = false
 
-#max_connections = 32
+max_connections = 32
 
 #port = 5432 
 #hostname_lookup = false
 #show_source_port = false
 
-#unix_socket_directory = ''
-#unix_socket_group = ''
+unix_socket_directory = '/tmp'
+unix_socket_group = 'pgsql'
 #unix_socket_permissions = 0777
 
 #virtual_host = ''
Index: src/bin/pg_ctl/pg_ctl.sh
===================================================================
RCS file: /projects/cvsroot/pgsql/src/bin/pg_ctl/pg_ctl.sh,v
retrieving revision 1.25
diff -u -b -u -b -r1.25 pg_ctl.sh
--- src/bin/pg_ctl/pg_ctl.sh	2001/09/29 03:09:32	1.25
+++ src/bin/pg_ctl/pg_ctl.sh	2002/07/06 14:37:14
@@ -14,6 +14,10 @@
 
 CMDNAME=`basename $0`
 
+ulimit -d 24000
+ulimit -n 512
+ulimit -s 4096
+
 help="\
 $CMDNAME is a utility to start, stop, restart, reload configuration files,
 or report the status of a PostgreSQL server.
Index: src/interfaces/ecpg/lib/Makefile
===================================================================
RCS file: /projects/cvsroot/pgsql/src/interfaces/ecpg/lib/Makefile,v
retrieving revision 1.15
diff -u -b -u -b -r1.15 Makefile
--- src/interfaces/ecpg/lib/Makefile	2002/01/08 20:41:26	1.15
+++ src/interfaces/ecpg/lib/Makefile	2002/07/06 14:37:14
@@ -17,7 +17,7 @@
 SO_MINOR_VERSION= 3.0
 
 override CPPFLAGS := -I$(top_srcdir)/src/interfaces/ecpg/include -I$(libpq_srcdir) $(CPPFLAGS)
-
+override LDFLAGS := -L$(libpq_builddir) $(LDFLAGS)
 
 OBJS= execute.o typename.o descriptor.o data.o error.o prepare.o memory.o \
 	connect.o misc.o
Index: src/interfaces/libpgeasy/Makefile
===================================================================
RCS file: /projects/cvsroot/pgsql/src/interfaces/libpgeasy/Makefile,v
retrieving revision 1.7
diff -u -b -u -b -r1.7 Makefile
--- src/interfaces/libpgeasy/Makefile	2001/05/11 01:46:33	1.7
+++ src/interfaces/libpgeasy/Makefile	2002/07/06 14:37:14
@@ -17,6 +17,7 @@
 SO_MINOR_VERSION= 2
 
 override CPPFLAGS := -I$(libpq_srcdir) $(CPPFLAGS)
+override LDFLAGS := -L$(libpq_builddir) $(LDFLAGS)
 
 OBJS= libpgeasy.o halt.o
 
Index: src/interfaces/libpgtcl/Makefile
===================================================================
RCS file: /projects/cvsroot/pgsql/src/interfaces/libpgtcl/Makefile,v
retrieving revision 1.26
diff -u -b -u -b -r1.26 Makefile
--- src/interfaces/libpgtcl/Makefile	2001/05/11 01:46:33	1.26
+++ src/interfaces/libpgtcl/Makefile	2002/07/06 14:37:14
@@ -17,6 +17,7 @@
 SO_MINOR_VERSION= 2
 
 override CPPFLAGS := -I$(libpq_srcdir) $(CPPFLAGS)
+override LDFLAGS := -L$(libpq_builddir) $(LDFLAGS)
 
 OBJS= pgtcl.o pgtclCmds.o pgtclId.o
 
Index: src/interfaces/perl5/GNUmakefile
===================================================================
RCS file: /projects/cvsroot/pgsql/src/interfaces/perl5/GNUmakefile,v
retrieving revision 1.6
diff -u -b -u -b -r1.6 GNUmakefile
--- src/interfaces/perl5/GNUmakefile	2001/08/26 22:28:04	1.6
+++ src/interfaces/perl5/GNUmakefile	2002/07/06 14:37:14
@@ -12,14 +12,9 @@
 
 .NOTPARALLEL:
 
-# This would allow a non-root install of the Perl module, but it's not
-# quite implemented yet.
-ifeq ($(mysterious_feature),yes)
-perl_installsitearch	= $(pkglibdir)
-perl_installsitelib	= $(pkglibdir)
-perl_installman3dir	= $(mandir)/man3
-endif
-
+perl_installsitearch	= $(prefix)/lib/perl5/darwin
+perl_installsitelib	= $(prefix)/lib/perl5
+perl_installman3dir	= $(prefix)/share/man/man3
 
 all: Makefile libpq-all
 	$(MAKE) -f $< all VPATH=$(VPATH)
Index: src/interfaces/python/GNUmakefile
===================================================================
RCS file: /projects/cvsroot/pgsql/src/interfaces/python/GNUmakefile,v
retrieving revision 1.13
diff -u -b -u -b -r1.13 GNUmakefile
--- src/interfaces/python/GNUmakefile	2001/12/13 18:39:04	1.13
+++ src/interfaces/python/GNUmakefile	2002/07/06 14:37:14
@@ -18,6 +18,7 @@
 include $(top_srcdir)/src/Makefile.shlib
 
 override CPPFLAGS := -I$(libpq_srcdir) $(CPPFLAGS) $(python_includespec)
+override SHLIB_LINK := -L$(libpq_builddir) $(SHLIB_LINK)
 
 all: all-lib
 
Index: src/pl/plpython/Makefile
===================================================================
RCS file: /projects/cvsroot/pgsql/src/pl/plpython/Makefile,v
retrieving revision 1.10
diff -u -b -u -b -r1.10 Makefile
--- src/pl/plpython/Makefile	2001/09/16 16:11:11	1.10
+++ src/pl/plpython/Makefile	2002/07/06 14:37:14
@@ -8,7 +8,7 @@
 # On some platforms we can only build PL/Python if libpython is a
 # shared library.  Since there is no official way to determine this,
 # we see if there is a file that is named like a shared library.
-ifneq (,$(wildcard $(python_configdir)/libpython*$(DLSUFFIX)*))
+ifneq (,$(wildcard $(python_configdir)/libpython*.dylib*))
 shared_libpython = yes
 endif
 
