Index: perlmod/Fink/PkgVersion.pm
===================================================================
RCS file: /cvsroot/fink/fink/perlmod/Fink/PkgVersion.pm,v
retrieving revision 1.81
diff -u -b -r1.81 PkgVersion.pm
--- perlmod/Fink/PkgVersion.pm	19 Sep 2002 20:03:03 -0000	1.81
+++ perlmod/Fink/PkgVersion.pm	25 Sep 2002 14:55:53 -0000
@@ -1719,6 +1719,10 @@
 	delete $ENV{$varname};
       }
     }
+  }
+  unless ($self->has_param("NoOptFlags") or not defined $config->param("optflags")) {
+    $ENV{'CFLAGS'}   .= " " . $config->param("optflags");
+    $ENV{'CXXFLAGS'} .= " " . $config->param("optflags");
   }
 }
 
