diff -urN httpd-2.0.47.orig/Makefile.in httpd-2.0.47/Makefile.in
--- httpd-2.0.47.orig/Makefile.in	2003-05-17 19:11:37.000000000 +0200
+++ httpd-2.0.47/Makefile.in	2003-07-12 17:22:15.000000000 +0200
@@ -58,7 +58,7 @@
 	    					echo "<IfDefine SSL>"; \
 	    				fi; \
 	    				if test $$j != "^EOL^"; then \
-	    					echo "LoadModule $${j}_module $(rel_libexecdir)/mod_$${j}.so"; \
+	    					echo "LoadModule $${j}_module mod_$${j}.so"; \
 	    				fi; \
 	    				if test "x$$j" = "xssl"; then \
 	    					echo "</IfDefine>"; \
diff -urN httpd-2.0.47.orig/docs/conf/httpd-std.conf.in httpd-2.0.47/docs/conf/httpd-std.conf.in
--- httpd-2.0.47.orig/docs/conf/httpd-std.conf.in	2003-07-06 16:43:30.000000000 +0200
+++ httpd-2.0.47/docs/conf/httpd-std.conf.in	2003-07-12 17:23:50.000000000 +0200
@@ -264,8 +264,12 @@
 #  when the value of (unsigned)Group is above 60000; 
 #  don't use Group #-1 on these systems!
 #
-User nobody
-Group #-1
+#User nobody
+#Group #-1
+
+User www
+Group www
+
 </IfModule>
 </IfModule>
 
@@ -982,25 +986,32 @@
 # with the URL of http://servername/server-status
 # Change the ".example.com" to match your domain to enable.
 #
-#<Location /server-status>
-#    SetHandler server-status
-#    Order deny,allow
-#    Deny from all
-#    Allow from .example.com
-#</Location>
+<IfDefine STATUS>
+   <IfModule mod_status.c>
+      <Location /server-status>
+         SetHandler server-status
+         Order deny,allow
+         Deny from all
+         Allow from localhost
+      </Location>
+   </IfModule>
+</IfDefine>
 
 #
 # Allow remote server configuration reports, with the URL of
 #  http://servername/server-info (requires that mod_info.c be loaded).
 # Change the ".example.com" to match your domain to enable.
 #
-#<Location /server-info>
-#    SetHandler server-info
-#    Order deny,allow
-#    Deny from all
-#    Allow from .example.com
-#</Location>
-
+<IfDefine INFO>
+   <IfModule mod_info.c>
+      <Location /server-info>
+         SetHandler server-info
+         Order deny,allow
+         Deny from all
+         Allow from localhost
+      </Location>
+   </IfModule>
+</IfDefine>
 
 #
 # Bring in additional module-specific configurations
